<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light_v2-e830ce163aae.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark_v2-bc3c552285ea.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed_v2-191796a86a54.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast_v2-e0a0b6e42616.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind_v2-4fd40c9324e8.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind_v2-2f299c61eceb.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast_v2-05d723322773.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia_v2-5679d6101565.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia_v2-0ff66d76cc55.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","repository_suggester_elastic_search","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-6a292b1161ed.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         kcrid001/KatieCriderAdvancedGenomicsLog: log of work in the 2021 Spring Advanced Genomics class
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="90A8:5F4D:35FAED:4C4CC3:6605CBC3" data-turbo-transient="true" name="request-id">
         <meta content="22f598a296c639e8b750dc2cf1a20323836185b0d8b152a2e56d5461ad76529a" data-turbo-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6bnVsbCwicmVxdWVzdF9pZCI6IjkwQTg6NUY0RDozNUZBRUQ6NEM0Q0MzOjY2MDVDQkMzIiwidmlzaXRvcl9pZCI6IjU1NjM2ODU5OTEwODQ4MjkyODYiLCJyZWdpb25fZWRnZSI6ImlhZCIsInJlZ2lvbl9yZW5kZXIiOiJpYWQifQ==" data-turbo-transient="true" name="visitor-payload">
           <meta content="fbaba9f341bbf57ca036f57960ac649b506046e57bca0eb8531de28bea7c1fa7" data-turbo-transient="true" name="visitor-hmac">
            <meta content="repository:332006677" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="126681664" name="octolytics-actor-id">
               <meta content="ninenia" name="octolytics-actor-login">
                <meta content="57d6abdd600494be8ac2f82b25b957e4c0d029a4d648c10497790171de0492de" name="octolytics-actor-hash">
                 <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
                  <meta content="ninenia" name="user-login"/>
                  <link href="/sessions/sudo_modal" rel="sudo-modal"/>
                  <meta content="width=device-width" name="viewport"/>
                  <meta content="log of work in the 2021 Spring Advanced Genomics class - kcrid001/KatieCriderAdvancedGenomicsLog" name="description"/>
                  <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
                  <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
                  <meta content="1401488693436528" property="fb:app_id"/>
                  <meta content="app-id=1477376905, app-argument=https://github.com/kcrid001/KatieCriderAdvancedGenomicsLog" name="apple-itunes-app">
                   <meta content="https://opengraph.githubassets.com/5d9a0db4092193c413aa650e4e7faf07f916d5740f4fdad460e3d7a8a3800bc0/kcrid001/KatieCriderAdvancedGenomicsLog" name="twitter:image:src">
                    <meta content="@github" name="twitter:site">
                     <meta content="summary_large_image" name="twitter:card">
                      <meta content="kcrid001/KatieCriderAdvancedGenomicsLog: log of work in the 2021 Spring Advanced Genomics class" name="twitter:title">
                       <meta content="log of work in the 2021 Spring Advanced Genomics class - kcrid001/KatieCriderAdvancedGenomicsLog" name="twitter:description">
                        <meta content="https://opengraph.githubassets.com/5d9a0db4092193c413aa650e4e7faf07f916d5740f4fdad460e3d7a8a3800bc0/kcrid001/KatieCriderAdvancedGenomicsLog" property="og:image"/>
                        <meta content="log of work in the 2021 Spring Advanced Genomics class - kcrid001/KatieCriderAdvancedGenomicsLog" property="og:image:alt"/>
                        <meta content="1200" property="og:image:width"/>
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="kcrid001/KatieCriderAdvancedGenomicsLog: log of work in the 2021 Spring Advanced Genomics class" property="og:title"/>
                        <meta content="https://github.com/kcrid001/KatieCriderAdvancedGenomicsLog" property="og:url"/>
                        <meta content="log of work in the 2021 Spring Advanced Genomics class - kcrid001/KatieCriderAdvancedGenomicsLog" property="og:description"/>
                        <link data-refresh-url="/_alive" data-session-id="fae53153a7a7c8d08a43b9810520b9ce4325ce8b4e99a06c5fc3f52c85cc22fe" href="wss://alive.github.com/_sockets/u/126681664/ws?session=eyJ2IjoiVjMiLCJ1IjoxMjY2ODE2NjQsInMiOjEzMDQyODIxMTYsImMiOjMwNTgzMzY4NDQsInQiOjE3MTE2NTU4Nzl9--dbb845f4c8a50354bbd89a49820e18cc6133e027857e63ee70b40ed6e698f8a1" rel="shared-web-socket"/>
                        <link href="/assets-cdn/worker/socket-worker-9cc1149b224c.js" rel="shared-web-socket-src"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="all" name="keyboard-shortcuts-preference"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="de8e1521fc37a178caaed17097d0545118b4eb0f01833d1d1eeeb65ac20b6ca7" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="1dd47e41bae3fb619dfa564356c6dee453350e97ff939ad02473ed6906f90b45" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/kcrid001/KatieCriderAdvancedGenomicsLog git https://github.com/kcrid001/KatieCriderAdvancedGenomicsLog.git" name="go-import"/>
                        <meta content="77846945" name="octolytics-dimension-user_id">
                         <meta content="kcrid001" name="octolytics-dimension-user_login">
                          <meta content="332006677" name="octolytics-dimension-repository_id">
                           <meta content="kcrid001/KatieCriderAdvancedGenomicsLog" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="332006677" name="octolytics-dimension-repository_network_root_id">
                               <meta content="kcrid001/KatieCriderAdvancedGenomicsLog" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/kcrid001/KatieCriderAdvancedGenomicsLog" rel="canonical"/>
                                <meta content="logged-in env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <meta content="/windows-tile.png" name="msapplication-TileImage"/>
                                 <meta content="#ffffff" name="msapplication-TileColor"/>
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-in env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-in env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="p-3 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_allex_crc32_lib_crc32_esm_js-node_modules_github_mini-throttle_dist_deco-b38cad-748e74df23ce.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_clipboard-copy-element_dist_index_esm_js-node_modules_delegated-e-b37f7d-2f24d321a3fb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_command-palette_items_help-item_ts-app_assets_modules_github_comman-48ad9d-a9ed17f2ebff.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/command-palette-f620e569db0b.js" type="application/javascript">
    </script>
    <header class="AppHeader">
     <div class="AppHeader-globalBar pb-2 js-global-bar">
      <div class="AppHeader-globalBar-start">
       <deferred-side-panel data-url="/_side-panels/global">
        <include-fragment data-target="deferred-side-panel.fragment">
         <button aria-label="Open global navigation menu" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-bg-transparent p-0 color-fg-muted" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-38bf0078-02a3-4e6f-96b7-1a4c23c49c9d" data-view-component="true" id="dialog-show-dialog-38bf0078-02a3-4e6f-96b7-1a4c23c49c9d" type="button">
          <svg aria-hidden="true" class="octicon octicon-three-bars Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1 2.75A.75.75 0 0 1 1.75 2h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 2.75Zm0 5A.75.75 0 0 1 1.75 7h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 7.75ZM1.75 12h12.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Z">
           </path>
          </svg>
         </button>
         <dialog-helper>
          <dialog aria-describedby="dialog-38bf0078-02a3-4e6f-96b7-1a4c23c49c9d-description" aria-labelledby="dialog-38bf0078-02a3-4e6f-96b7-1a4c23c49c9d-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-left SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-38bf0078-02a3-4e6f-96b7-1a4c23c49c9d">
           <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title sr-only" id="dialog-38bf0078-02a3-4e6f-96b7-1a4c23c49c9d-title">
               Global navigation
              </h1>
              <div class="d-flex" data-view-component="true">
               <div class="AppHeader-logo position-relative" data-view-component="true">
                <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
                 <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
                 </path>
                </svg>
               </div>
              </div>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-38bf0078-02a3-4e6f-96b7-1a4c23c49c9d" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="dialog-38bf0078-02a3-4e6f-96b7-1a4c23c49c9d-title">
            <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
             <div class="d-flex flex-column mb-3" data-view-component="true">
              <nav aria-label="Site navigation" class="ActionList" data-view-component="true">
               <nav-list>
                <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"HOME","label":null}' data-hotkey="g d" data-view-component="true" href="/dashboard" id="item-380131d0-73cf-43c6-9b67-bbd6abd46e3e">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-home" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M6.906.664a1.749 1.749 0 0 1 2.187 0l5.25 4.2c.415.332.657.835.657 1.367v7.019A1.75 1.75 0 0 1 13.25 15h-3.5a.75.75 0 0 1-.75-.75V9H7v5.25a.75.75 0 0 1-.75.75h-3.5A1.75 1.75 0 0 1 1 13.25V6.23c0-.531.242-1.034.657-1.366l5.25-4.2Zm1.25 1.171a.25.25 0 0 0-.312 0l-5.25 4.2a.25.25 0 0 0-.094.196v7.019c0 .138.112.25.25.25H5.5V8.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 .75.75v5.25h2.75a.25.25 0 0 0 .25-.25V6.23a.25.25 0 0 0-.094-.195Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Home
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"ISSUES","label":null}' data-hotkey="g i" data-view-component="true" href="/issues" id="item-350a7620-85ed-4599-a79d-a27d673807ba">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS","label":null}' data-hotkey="g p" data-view-component="true" href="/pulls" id="item-ea4242e8-b87c-47aa-8220-a8eea036096e">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="projects" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROJECTS","label":null}' data-view-component="true" href="/projects" id="item-7a7803a3-ab74-440a-b81c-28d44e5a3c9b">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"DISCUSSIONS","label":null}' data-view-component="true" href="/discussions" id="item-3cf576e3-053e-4b5d-809e-7230451d2cb2">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Discussions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"CODESPACES","label":null}' data-view-component="true" href="https://github.com/codespaces" id="item-3baa9603-f1dc-45f8-bf78-894d1dcedd01">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Codespaces
                   </span>
                  </a>
                 </li>
                 <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"EXPLORE","label":null}' data-view-component="true" href="/explore" id="item-9e936df5-9127-45f0-a475-eefd6af832b0">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-telescope" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.184 1.143v-.001l1.422 2.464a1.75 1.75 0 0 1-.757 2.451L3.104 11.713a1.75 1.75 0 0 1-2.275-.702l-.447-.775a1.75 1.75 0 0 1 .53-2.32L11.682.573a1.748 1.748 0 0 1 2.502.57Zm-4.709 9.32h-.001l2.644 3.863a.75.75 0 1 1-1.238.848l-1.881-2.75v2.826a.75.75 0 0 1-1.5 0v-2.826l-1.881 2.75a.75.75 0 1 1-1.238-.848l2.049-2.992a.746.746 0 0 1 .293-.253l1.809-.87a.749.749 0 0 1 .944.252ZM9.436 3.92h-.001l-4.97 3.39.942 1.63 5.42-2.61Zm3.091-2.108h.001l-1.85 1.26 1.505 2.605 2.016-.97a.247.247 0 0 0 .13-.151.247.247 0 0 0-.022-.199l-1.422-2.464a.253.253 0 0 0-.161-.119.254.254 0 0 0-.197.038ZM1.756 9.157a.25.25 0 0 0-.075.33l.447.775a.25.25 0 0 0 .325.1l1.598-.769-.83-1.436-1.465 1Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Explore
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"MARKETPLACE","label":null}' data-view-component="true" href="/marketplace" id="item-7211300f-6508-4bff-b5ca-67cf59fd193d">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Marketplace
                   </span>
                  </a>
                 </li>
                </ul>
               </nav-list>
              </nav>
              <div class="my-3 d-flex flex-justify-center height-full" data-view-component="true">
               <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                </circle>
                <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                </path>
               </svg>
              </div>
             </div>
             <div class="flex-1" data-view-component="true">
             </div>
             <div class="px-2" data-view-component="true">
              <p class="color-fg-subtle text-small text-light">
               © 2024 GitHub, Inc.
              </p>
              <div class="d-flex flex-wrap text-small text-light" data-view-component="true">
               <a class="Link mr-2" data-view-component="true" href="https://github.com/about" target="_blank">
                About
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.blog" target="_blank">
                Blog
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service" target="_blank">
                Terms
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement" target="_blank">
                Privacy
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.com/security" target="_blank">
                Security
               </a>
               <a class="Link mr-3" data-view-component="true" href="https://www.githubstatus.com/" target="_blank">
                Status
               </a>
              </div>
             </div>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </include-fragment>
       </deferred-side-panel>
       <a aria-label="Homepage " class="AppHeader-logo ml-2" data-analytics-event='{"category":"Header","action":"go to dashboard","label":"icon:logo"}' data-hotkey="g d" data-turbo="false" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github v-align-middle color-fg-default" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="AppHeader-context">
        <div class="AppHeader-context-compact">
         <button aria-expanded="false" aria-haspopup="dialog" aria-label="Page context: kcrid001 / KatieCriderAdvancedGenomicsLog" class="AppHeader-context-compact-trigger Truncate Button--secondary Button--medium Button box-shadow-none" data-show-dialog-id="context-region-dialog" data-view-component="true" id="dialog-show-context-region-dialog" type="button">
          <span class="Button-content">
           <span class="Button-label">
            <span class="AppHeader-context-compact-lead">
             <span class="AppHeader-context-compact-parentItem">
              kcrid001
             </span>
             <span class="no-wrap">
              /
             </span>
            </span>
            <strong class="AppHeader-context-compact-mainItem d-flex flex-items-center Truncate">
             <span class="Truncate-text">
              KatieCriderAdvancedGenomicsLog
             </span>
            </strong>
           </span>
          </span>
         </button>
         <dialog-helper>
          <dialog aria-describedby="context-region-dialog-description" aria-labelledby="context-region-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-view-component="true" id="context-region-dialog">
           <div class="Overlay-header" data-view-component="true">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title" id="context-region-dialog-title">
               Navigate back to
              </h1>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="context-region-dialog" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="context-region-dialog-title">
            <div class="Overlay-body" data-view-component="true">
             <ul class="list-style-none" role="list">
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"kcrid001","screen_size":"compact"}' data-view-component="true" href="/kcrid001">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-person mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                  </path>
                 </svg>
                 kcrid001
                </span>
               </a>
              </li>
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"KatieCriderAdvancedGenomicsLog","screen_size":"compact"}' data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-repo mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                  </path>
                 </svg>
                 KatieCriderAdvancedGenomicsLog
                </span>
               </a>
              </li>
             </ul>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </div>
        <div class="AppHeader-context-full">
         <nav aria-label="Page context" role="navigation">
          <ul class="list-style-none" role="list">
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"kcrid001","screen_size":"full"}' data-hovercard-type="user" data-hovercard-url="/users/kcrid001/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" data-view-component="true" href="/kcrid001">
             <span class="AppHeader-context-item-label">
              kcrid001
             </span>
            </a>
            <span class="AppHeader-context-item-separator">
             /
            </span>
           </li>
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"KatieCriderAdvancedGenomicsLog","screen_size":"full"}' data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog">
             <span class="AppHeader-context-item-label">
              KatieCriderAdvancedGenomicsLog
             </span>
            </a>
           </li>
          </ul>
         </nav>
        </div>
       </div>
      </div>
      <div class="AppHeader-globalBar-end">
       <div class="AppHeader-search">
        <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;input type="hidden" value="qev0ATZz7NzkCfe-3uubDTLI_ZRE5V0T5KiWZyakkOD5yPiJpP4zE1OTr5B7fHlWBkQ9f3DO6II60-jZi2AtCA" data-csrf="true" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="kcrid001" data-current-repository="kcrid001/KatieCriderAdvancedGenomicsLog" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="1OoTWazuBtUkV78zZIlEfniTgo3Vwa96ykDaxoP_JCucIARziqhPoHP7zeqCjLRvKjfAJPOmXxWmopttZME_Zw" data-header-redesign-enabled="true" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="true" data-max-custom-scopes="10" data-scope="repo:kcrid001/KatieCriderAdvancedGenomicsLog">
         <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center height-auto color-bg-transparent border-0 color-fg-subtle mx-0" data-action="click:qbsearch-input#searchInputContainerClicked">
          <button aria-expanded="false" aria-haspopup="dialog" aria-label="Search or jump to…" class="AppHeader-button AppHeader-search-whenNarrow" data-action="click:qbsearch-input#handleExpand" type="button">
           <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
            </path>
           </svg>
          </button>
          <div class="AppHeader-search-whenRegular">
           <div class="AppHeader-search-wrap AppHeader-search-wrap--hasTrailing">
            <div class="AppHeader-search-control">
             <label aria-label="Search or jump to…" class="AppHeader-search-visual--leading" for="AppHeader-searchInput">
              <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
               </path>
              </svg>
             </label>
             <button aria-describedby="search-error-message-flash" class="AppHeader-searchButton form-control input-contrast text-left color-fg-subtle no-wrap" data-action="click:qbsearch-input#handleExpand" data-analytics-event='{"category":"Global navigation","action":"SEARCH","label":null}' data-hotkey="s,/" data-target="qbsearch-input.inputButton" type="button">
              <div class="overflow-hidden">
               <span data-target="qbsearch-input.inputButtonText" id="qb-input-query">
                Type
                <kbd class="AppHeader-search-kbd">
                 /
                </kbd>
                to search
               </span>
              </div>
             </button>
            </div>
            <button class="AppHeader-search-action--trailing js-activate-command-palette" data-analytics-event='{"category":"SiteHeaderComponent","action":"command_palette","label":"open command palette"}' id="AppHeader-commandPalette-button" type="button">
             <svg aria-hidden="true" class="octicon octicon-command-palette" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m6.354 8.04-4.773 4.773a.75.75 0 1 0 1.061 1.06L7.945 8.57a.75.75 0 0 0 0-1.06L2.642 2.206a.75.75 0 0 0-1.06 1.061L6.353 8.04ZM8.75 11.5a.75.75 0 0 0 0 1.5h5.5a.75.75 0 0 0 0-1.5h-5.5Z">
              </path>
             </svg>
            </button>
            <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="AppHeader-commandPalette-button" id="tooltip-ceb3deb6-b60a-417f-a2af-7406019966c2" popover="manual">
             Command palette
            </tool-tip>
           </div>
          </div>
          <input class="js-site-search-type-field" name="type" type="hidden"/>
          <div class="Overlay--hidden" data-modal-dialog-overlay="">
           <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-medium Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
            <h1 class="sr-only" id="search-suggestions-dialog-header">
             Search code, repositories, users, issues, pull requests...
            </h1>
            <div class="Overlay-body Overlay-body--paddingNone">
             <div data-view-component="true">
              <div class="search-suggestions position-absolute width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                 <div class="FormControl FormControl--fullWidth">
                  <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                   Search
                  </label>
                  <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                   <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                    <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </span>
                   <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                    <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                    </div>
                    <div class="QueryBuilder-InputWrapper">
                     <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                     </div>
                     <input aria-describedby="validation-86d35f63-2ec9-4800-bc80-9952ebf4fde7" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                     </input>
                    </div>
                   </div>
                   <span class="sr-only" id="query-builder-test-clear">
                    Clear
                   </span>
                   <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                    <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <template id="search-icon">
                   <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-icon">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="file-code-icon">
                   <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                    </path>
                   </svg>
                  </template>
                  <template id="history-icon">
                   <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                    </path>
                   </svg>
                  </template>
                  <template id="repo-icon">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="bookmark-icon">
                   <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="plus-circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                    </path>
                   </svg>
                  </template>
                  <template id="trash-icon">
                   <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="team-icon">
                   <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="project-icon">
                   <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                    </path>
                   </svg>
                  </template>
                  <template id="pencil-icon">
                   <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                    </path>
                   </svg>
                  </template>
                  <template id="copilot-icon">
                   <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                    </path>
                    <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="workflow-icon">
                   <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="book-icon">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-review-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="codespaces-icon">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-discussion-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="organization-icon">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="rocket-icon">
                   <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="shield-check-icon">
                   <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="heart-icon">
                   <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="server-icon">
                   <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="globe-icon">
                   <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                    </path>
                   </svg>
                  </template>
                  <template id="issue-opened-icon">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-mobile-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="package-icon">
                   <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                    </path>
                   </svg>
                  </template>
                  <template id="credit-card-icon">
                   <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                    </path>
                    <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                    </path>
                   </svg>
                  </template>
                  <template id="play-icon">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </template>
                  <template id="gift-icon">
                   <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-square-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-desktop-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                    </path>
                   </svg>
                  </template>
                  <div class="position-relative">
                   <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                   </ul>
                  </div>
                  <div class="FormControl-inlineValidation" hidden="hidden" id="validation-86d35f63-2ec9-4800-bc80-9952ebf4fde7">
                   <span class="FormControl-inlineValidation--visual">
                    <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                     <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                     </path>
                    </svg>
                   </span>
                   <span>
                   </span>
                  </div>
                 </div>
                 <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                 </div>
                </query-builder>
               </form>
               <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                 Search syntax tips
                </a>
                <div class="d-flex flex-1">
                </div>
                <button class="Button--link Button--medium Button color-fg-accent text-normal ml-2" data-action="click:qbsearch-input#showFeedbackDialog" data-view-component="true" type="button">
                 <span class="Button-content">
                  <span class="Button-label">
                   Give feedback
                  </span>
                 </span>
                </button>
               </div>
              </div>
             </div>
            </div>
           </modal-dialog>
          </div>
         </div>
         <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
         </div>
         <div class="color-fg-default">
          <dialog-helper>
           <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
            <div class="Overlay-header" data-view-component="true">
             <div class="Overlay-headerContentWrap">
              <div class="Overlay-titleWrap">
               <h1 class="Overlay-title" id="feedback-dialog-title">
                Provide feedback
               </h1>
              </div>
              <div class="Overlay-actionWrap">
               <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
              </div>
             </div>
            </div>
            <scrollable-region data-labelled-by="feedback-dialog-title">
             <div class="Overlay-body" data-view-component="true">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
               <input name="authenticity_token" type="hidden" value="MEneFhg8np3b41SYZQqpviVdFvXn2hqz1OmAloSQKtmi-8cu0OqtB2vBWpxbE6hP-FbbNr36_ObaaZUscsEvYg"/>
               <p>
                We read every piece of feedback, and take your input very seriously.
               </p>
               <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
               <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
               <label for="include_email" style="font-weight: normal">
                Include my email address so I can be contacted
               </label>
              </form>
             </div>
            </scrollable-region>
            <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
             <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
              Cancel
             </button>
             <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
              Submit feedback
             </button>
            </div>
           </dialog>
          </dialog-helper>
          <custom-scopes data-target="qbsearch-input.customScopesManager">
           <dialog-helper>
            <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
             <div class="Overlay-header Overlay-header--divided" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                 Saved searches
                </h1>
                <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                 Use saved searches to filter your results more quickly
                </h2>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="custom-scopes-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <div data-target="custom-scopes.customScopesModalDialogFlash">
               </div>
               <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                 <input name="authenticity_token" type="hidden" value="slxakY9eCFvBUX_wIqxiAPt_nP-bDSBebgI16n80xuf-rjgr199Wgoo4wl1oea89Mi_46M0IYoXNWuRSYHOfsg">
                  <div data-target="custom-scopes.customScopesModalDialogFlash">
                  </div>
                  <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                  <div class="form-group">
                   <label for="custom_scope_name">
                    Name
                   </label>
                   <auto-check required="" src="/search/custom_scopes/check_name">
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                    <input data-csrf="true" type="hidden" value="S1h2ldoF2kXCcheeG8XzsHeh004ZIWucojAw7pX4NP1TBTszjYdzRkFmXh1bl63awWbWyTf4iwTPkJI8rZfnNQ">
                    </input>
                   </auto-check>
                  </div>
                  <div class="form-group">
                   <label for="custom_scope_query">
                    Query
                   </label>
                   <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                  </div>
                  <p class="text-small color-fg-muted">
                   To see all available qualifiers, see our
                   <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                    documentation
                   </a>
                   .
                  </p>
                 </input>
                </form>
               </div>
               <div data-target="custom-scopes.manageCustomScopesForm">
                <div data-target="custom-scopes.list">
                </div>
               </div>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
              <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
               Create saved search
              </button>
             </div>
            </dialog>
           </dialog-helper>
          </custom-scopes>
         </div>
        </qbsearch-input>
        <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="v9DnmgpKtJagLkqTU0oSfhrDnHU60ymHXFQHyn5dpPl4hYi4E01y9Ef10G6a8WltyntEv79177dTN1lv5snF8w">
        </input>
       </div>
       <div class="AppHeader-actions position-relative">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="global-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="global-create-menu-button" popovertarget="global-create-menu-overlay" type="button">
           <span class="Button-content">
            <span class="Button-visual Button-leadingVisual">
             <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
              </path>
             </svg>
            </span>
            <span class="Button-label">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </span>
           </span>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="global-create-menu-button" id="tooltip-ba9307a4-8c4a-4bbf-b62a-02ddf62f2a84" popover="manual">
           Create new...
          </tool-tip>
          <anchored-position align="end" anchor="global-create-menu-button" anchor-offset="normal" data-view-component="true" id="global-create-menu-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="global-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="global-create-menu-list" role="menu">
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-8602d6af-b8d2-4a99-8af8-e05dea6870d1" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New repository
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-4543178a-e4fe-47a6-bd9f-ff75cac35d91" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Import repository
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-8e350287-b566-4651-bb04-96228bea826e" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New codespace
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-76059064-e652-421c-8296-8c83f63f7231" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New gist
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-1d464ecd-ffa7-4d58-9070-a9f1ef8c2139" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New organization
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
        <a aria-labelledby="tooltip-982d2dbb-bb2c-4e3d-8e1d-566cd7d6afed" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"ISSUES_HEADER","label":null}' data-view-component="true" href="/issues" id="icon-button-217ffc06-7fe9-4846-9e90-a2984902effc">
         <svg aria-hidden="true" class="octicon octicon-issue-opened Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-217ffc06-7fe9-4846-9e90-a2984902effc" id="tooltip-982d2dbb-bb2c-4e3d-8e1d-566cd7d6afed" popover="manual">
         Issues
        </tool-tip>
        <a aria-labelledby="tooltip-fe0cff60-8521-4e52-9507-6d0cc9633537" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS_HEADER","label":null}' data-view-component="true" href="/pulls" id="icon-button-cf19213a-c6dd-4720-bdd7-e01fab169050">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-cf19213a-c6dd-4720-bdd7-e01fab169050" id="tooltip-fe0cff60-8521-4e52-9507-6d0cc9633537" popover="manual">
         Pull requests
        </tool-tip>
       </div>
       <notification-indicator class="js-socket-channel" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY1NTg3OX0=--ce97df1ca3950af7725eebbb2b1ce1107a7c388af79fdffb6a440e55990fd8ab" data-fetch-indicator-enabled="true" data-fetch-indicator-src="/notifications/indicator" data-header-redesign-enabled="true" data-indicator-mode="none" data-tooltip-global="You have unread notifications" data-tooltip-none="You have no unread notifications" data-tooltip-unavailable="Notifications are unavailable at the moment." data-view-component="true">
        <a aria-label="Notifications" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"NOTIFICATIONS_HEADER","label":null}' data-hotkey="g n" data-target="notification-indicator.link" data-view-component="true" href="/notifications" id="AppHeader-notifications-button">
         <svg aria-hidden="true" class="octicon octicon-inbox Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.8 2.06A1.75 1.75 0 0 1 4.41 1h7.18c.7 0 1.333.417 1.61 1.06l2.74 6.395c.04.093.06.194.06.295v4.5A1.75 1.75 0 0 1 14.25 15H1.75A1.75 1.75 0 0 1 0 13.25v-4.5c0-.101.02-.202.06-.295Zm1.61.44a.25.25 0 0 0-.23.152L1.887 8H4.75a.75.75 0 0 1 .6.3L6.625 10h2.75l1.275-1.7a.75.75 0 0 1 .6-.3h2.863L11.82 2.652a.25.25 0 0 0-.23-.152Zm10.09 7h-2.875l-1.275 1.7a.75.75 0 0 1-.6.3h-3.5a.75.75 0 0 1-.6-.3L4.375 9.5H1.5v3.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-target="notification-indicator.tooltip" data-type="description" data-view-component="true" for="AppHeader-notifications-button" id="tooltip-32a0bbc2-748f-4bcf-9c32-f1bfd369d8b1" popover="manual">
         Notifications
        </tool-tip>
       </notification-indicator>
       <div class="AppHeader-user">
        <deferred-side-panel data-url="/_side-panels/user?react_global_nav=false&amp;repository_id=332006677">
         <include-fragment data-target="deferred-side-panel.fragment">
          <user-drawer-side-panel>
           <button aria-label="Open user account menu" class="AppHeader-logo Button--invisible Button--medium Button Button--invisible-noVisuals color-bg-transparent p-0" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-f59f32d8-db06-4105-9260-207a34fbf59b" data-view-component="true" id="dialog-show-dialog-f59f32d8-db06-4105-9260-207a34fbf59b" type="button">
            <span class="Button-content">
             <span class="Button-label">
              <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
             </span>
            </span>
           </button>
           <dialog-helper>
            <dialog aria-describedby="dialog-f59f32d8-db06-4105-9260-207a34fbf59b-description" aria-labelledby="dialog-f59f32d8-db06-4105-9260-207a34fbf59b-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-right SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-f59f32d8-db06-4105-9260-207a34fbf59b">
             <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title sr-only" id="dialog-f59f32d8-db06-4105-9260-207a34fbf59b-title">
                 Account menu
                </h1>
                <div class="d-flex" data-view-component="true">
                 <div class="AppHeader-logo position-relative" data-view-component="true">
                  <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
                 </div>
                 <div class="overflow-hidden d-flex width-full" data-view-component="true">
                  <div class="lh-condensed overflow-hidden d-flex flex-column flex-justify-center ml-2 f5 mr-auto width-full" data-view-component="true">
                   <span class="Truncate text-bold" data-view-component="true">
                    <span class="Truncate-text" data-view-component="true">
                     ninenia
                    </span>
                   </span>
                  </div>
                  <action-menu class="d-sm-none d-md-none d-lg-none" data-select-variant="none" data-view-component="true">
                   <focus-group direction="vertical" mnemonics="" retain="">
                    <button aria-controls="user-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="user-create-menu-button" popovertarget="user-create-menu-overlay" type="button">
                     <span class="Button-content">
                      <span class="Button-visual Button-leadingVisual">
                       <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                        </path>
                       </svg>
                      </span>
                      <span class="Button-label">
                       <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                        </path>
                       </svg>
                      </span>
                     </span>
                    </button>
                    <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="user-create-menu-button" id="tooltip-6b357dff-4103-4537-9a34-02e1106cb72a" popover="manual">
                     Create new...
                    </tool-tip>
                    <anchored-position align="end" anchor="user-create-menu-button" anchor-offset="normal" data-view-component="true" id="user-create-menu-overlay" popover="auto" side="outside-bottom">
                     <div class="Overlay Overlay--size-auto" data-view-component="true">
                      <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
                       <action-list>
                        <div data-view-component="true">
                         <ul aria-labelledby="user-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="user-create-menu-list" role="menu">
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-0fa75b72-097c-4273-aa6a-ae2e93e453e6" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New repository
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-d25a2521-8c63-406a-bce9-e0b44bbbd086" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             Import repository
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-4695f229-48a2-4f19-8d9e-ba51473d77c8" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                              </path>
                              <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New codespace
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-72c05346-82b1-4c87-9700-4bba62887140" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New gist
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-98ea5bc3-e570-4c1b-b456-d2f112c2185c" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New organization
                            </span>
                           </a>
                          </li>
                         </ul>
                        </div>
                       </action-list>
                      </div>
                     </div>
                    </anchored-position>
                   </focus-group>
                  </action-menu>
                 </div>
                </div>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-f59f32d8-db06-4105-9260-207a34fbf59b" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="dialog-f59f32d8-db06-4105-9260-207a34fbf59b-title">
              <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
               <div class="d-flex flex-column mb-3" data-view-component="true">
                <nav aria-label="User navigation" class="ActionList" data-view-component="true">
                 <nav-list>
                  <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-74c3ee27-0d78-47b1-8868-8f380ee221b8" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROFILE","label":null}' data-view-component="true" href="https://github.com/ninenia" id="item-e81eb0a8-0705-4b26-93ee-fe8d3121b80a">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-person" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your profile
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-93a1b8bd-267b-46d9-9bb6-597985843e67" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_REPOSITORIES","label":null}' data-view-component="true" href="/ninenia?tab=repositories" id="item-ae6c0a82-40b5-4f86-9246-292a68406563">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your repositories
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_PROJECTS","label":null}' data-view-component="true" href="/ninenia?tab=projects" id="item-e70336be-1077-4c98-88c5-fde17683ddb3">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your projects
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-0a06302d-52c1-4241-94ea-7292e5e6aebc" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-47e212d0-1817-49af-b713-4d554c2d5d77" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_STARS","label":null}' data-view-component="true" href="/ninenia?tab=stars" id="item-40a94f6d-a547-4b03-8909-c367e53194be">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-star" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your stars
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SPONSORS","label":null}' data-view-component="true" href="/sponsors/accounts" id="item-9e043b51-5268-44dc-ba20-3e93ebf1a105">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your sponsors
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_GISTS","label":null}' data-view-component="true" href="https://gist.github.com/mine" id="item-4a39813e-4e41-459c-96ca-79bf842db3fa">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your gists
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-5a9e3607-b2ba-4ae4-8e4a-86877a4fa426" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-ccb55382-97ed-4b4c-87ab-ccbc22608fdf" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SETTINGS","label":null}' data-view-component="true" href="/settings/profile" id="item-d128a8ee-8692-4378-bc15-cb510faaf8da">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-gear" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Settings
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SUPPORT","label":null}' data-view-component="true" href="https://support.github.com" id="item-a34520c8-e87e-4dff-ad6e-fe40e43350b7">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Support
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"COMMUNITY","label":null}' data-view-component="true" href="https://community.github.com" id="item-68ae46c4-694e-4d0a-84f5-be1c8597644b">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Community
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent" data-analytics-event='{"category":"Global navigation","action":"LOGOUT","label":null}' data-view-component="true" href="/logout" id="item-c163675b-09b3-4110-8f85-fc02610f0386">
                     <span class="ActionListItem-label" data-view-component="true">
                      Sign out
                     </span>
                    </a>
                   </li>
                  </ul>
                 </nav-list>
                </nav>
               </div>
              </div>
             </scrollable-region>
            </dialog>
           </dialog-helper>
          </user-drawer-side-panel>
         </include-fragment>
        </deferred-side-panel>
       </div>
       <div class="position-absolute mt-2">
        <site-header-logged-in-user-menu>
        </site-header-logged-in-user-menu>
       </div>
      </div>
     </div>
     <div class="AppHeader-localBar">
      <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav" data-pjax="#js-repo-pjax-container" data-view-component="true">
       <ul class="UnderlineNav-body list-style-none" data-view-component="true">
        <li class="d-inline-flex" data-view-component="true">
         <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /kcrid001/KatieCriderAdvancedGenomicsLog" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog" id="code-tab">
          <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
           </path>
          </svg>
          <span data-content="Code">
           Code
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /kcrid001/KatieCriderAdvancedGenomicsLog/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/issues" id="issues-tab">
          <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
           </path>
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
           </path>
          </svg>
          <span data-content="Issues">
           Issues
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /kcrid001/KatieCriderAdvancedGenomicsLog/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/pulls" id="pull-requests-tab">
          <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
           </path>
          </svg>
          <span data-content="Pull requests">
           Pull requests
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /kcrid001/KatieCriderAdvancedGenomicsLog/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/actions" id="actions-tab">
          <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
           </path>
          </svg>
          <span data-content="Actions">
           Actions
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /kcrid001/KatieCriderAdvancedGenomicsLog/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/projects" id="projects-tab">
          <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
           </path>
          </svg>
          <span data-content="Projects">
           Projects
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /kcrid001/KatieCriderAdvancedGenomicsLog/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/security" id="security-tab">
          <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
           </path>
          </svg>
          <span data-content="Security">
           Security
          </span>
          <include-fragment accept="text/fragment+html" src="/kcrid001/KatieCriderAdvancedGenomicsLog/security/overall-count">
          </include-fragment>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /kcrid001/KatieCriderAdvancedGenomicsLog/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/pulse" id="insights-tab">
          <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
           </path>
          </svg>
          <span data-content="Insights">
           Insights
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
       </ul>
       <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="action-menu-258fcf3b-bf8d-4c5f-99c3-0638e5913741-list" aria-haspopup="true" aria-labelledby="tooltip-2d2d338b-777a-4e3e-b602-2ff2b87c376f" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-258fcf3b-bf8d-4c5f-99c3-0638e5913741-button" popovertarget="action-menu-258fcf3b-bf8d-4c5f-99c3-0638e5913741-overlay" type="button">
           <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
           </svg>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-258fcf3b-bf8d-4c5f-99c3-0638e5913741-button" id="tooltip-2d2d338b-777a-4e3e-b602-2ff2b87c376f" popover="manual">
           Additional navigation options
          </tool-tip>
          <anchored-position align="start" anchor="action-menu-258fcf3b-bf8d-4c5f-99c3-0638e5913741-button" anchor-offset="normal" data-view-component="true" id="action-menu-258fcf3b-bf8d-4c5f-99c3-0638e5913741-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="action-menu-258fcf3b-bf8d-4c5f-99c3-0638e5913741-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-258fcf3b-bf8d-4c5f-99c3-0638e5913741-list" role="menu">
                <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog" id="item-2fa2a7a5-4c66-4da9-8072-1ee7264c9153" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Code
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/issues" id="item-a4ba227e-b693-4ffe-b3a4-6e702a3a04b3" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Issues
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/pulls" id="item-bcf35e68-f60a-4100-be42-177f60eca7a5" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Pull requests
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/actions" id="item-2e41bdd9-82e8-4891-9a61-8ccddc49c092" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Actions
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/projects" id="item-892b291b-2b28-4095-8278-40164a8be8d2" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Projects
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/security" id="item-61ee16a5-bfc8-40ed-80d8-4648a7e822eb" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Security
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/pulse" id="item-e4a1c5ac-2d8b-4b09-bd06-499d7e2809de" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Insights
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
       </div>
      </nav>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-71e147cd-0d05-4c1b-9646-93211741e0d3" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-10554cfc-63ee-47c4-84e0-ea4225a42c0c" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-10554cfc-63ee-47c4-84e0-ea4225a42c0c" id="tooltip-71e147cd-0d05-4c1b-9646-93211741e0d3" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <notification-shelf-watcher class="js-socket-channel" data-base-url="https://github.com/notifications/beta/shelf" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY1NTg3OX0=--ce97df1ca3950af7725eebbb2b1ce1107a7c388af79fdffb6a440e55990fd8ab" data-view-component="true">
   </notification-shelf-watcher>
   <div data-initial="" data-target="notification-shelf-watcher.placeholder" hidden="">
   </div>
   <details class="details-reset details-overlay details-overlay-dark js-command-palette-dialog" data-turbo-replace="" id="command-palette-pjax-container">
    <summary aria-label="Command palette trigger" tabindex="-1">
    </summary>
    <details-dialog aria-label="Command palette" class="command-palette-details-dialog d-flex flex-column flex-justify-center height-fit">
     <command-palette activation-hotkey="Mod+k,Mod+Alt+k" class="command-palette color-bg-default rounded-3 border color-shadow-small" command-mode-hotkey="Mod+Shift+K" data-action="
        command-palette-input-ready:command-palette#inputReady
        command-palette-page-stack-updated:command-palette#updateInputScope
        itemsUpdated:command-palette#itemsUpdated
        keydown:command-palette#onKeydown
        loadingStateChanged:command-palette#loadingStateChanged
        selectedItemChanged:command-palette#selectedItemChanged
        pageFetchError:command-palette#pageFetchError
      " return-to="/kcrid001/KatieCriderAdvancedGenomicsLog" user-id="126681664">
      <command-palette-mode data-char="#" data-placeholder="Search issues and pull requests" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="#" data-placeholder="Search issues, pull requests, discussions, and projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="!" data-placeholder="Search projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a user, organization, or repository" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a repository" data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode data-char="/" data-placeholder="Search files" data-scope-types='["repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="?">
      </command-palette-mode>
      <command-palette-mode data-char="&gt;" data-placeholder="Run a command">
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode class="js-command-palette-default-mode" data-char="" data-placeholder="Search or jump to...">
      </command-palette-mode>
      <command-palette-input data-action="
          command-palette-input:command-palette#onInput
          command-palette-select:command-palette#onSelect
          command-palette-descope:command-palette#onDescope
          command-palette-cleared:command-palette#onInputClear
        " placeholder="Search or jump to...">
       <div class="js-search-icon d-flex flex-items-center mr-2" style="height: 26px">
        <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
         </path>
        </svg>
       </div>
       <div class="js-spinner d-flex flex-items-center mr-2 color-fg-muted" hidden="">
        <svg aria-label="Loading" class="anim-rotate" fill="none" height="16" viewbox="0 0 16 16" width="16">
         <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
         </circle>
         <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
         </path>
        </svg>
       </div>
       <command-palette-scope>
        <div class="color-fg-subtle" data-target="command-palette-scope.placeholder" hidden="">
         /
         <span class="text-semibold color-fg-default">
          ...
         </span>
         /
        </div>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDQ6VXNlcjc3ODQ2OTQ1" data-targets="command-palette-scope.tokens" data-text="kcrid001" data-type="owner" data-value="kcrid001" style="white-space:nowrap;line-height:20px;">
         kcrid001
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDEwOlJlcG9zaXRvcnkzMzIwMDY2Nzc=" data-targets="command-palette-scope.tokens" data-text="KatieCriderAdvancedGenomicsLog" data-type="repository" data-value="KatieCriderAdvancedGenomicsLog" style="white-space:nowrap;line-height:20px;">
         KatieCriderAdvanc...
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
       </command-palette-scope>
       <div class="command-palette-input-group flex-1 form-control border-0 box-shadow-none" style="z-index: 0">
        <div class="command-palette-typeahead position-absolute d-flex flex-items-center Truncate">
         <span class="typeahead-segment input-mirror" data-target="command-palette-input.mirror">
         </span>
         <span class="Truncate-text" data-target="command-palette-input.typeaheadText">
         </span>
         <span class="typeahead-segment" data-target="command-palette-input.typeaheadPlaceholder">
         </span>
        </div>
        <input aria-label="Hidden input for typeahead" class="js-overlay-input typeahead-input d-none" disabled="" tabindex="-1"/>
        <input aria-autocomplete="list" aria-controls="command-palette-page-stack" aria-expanded="false" aria-haspopup="listbox" aria-label="Command palette input" autocapitalize="off" autocomplete="off" autocorrect="off" class="js-input typeahead-input form-control border-0 box-shadow-none input-block width-full no-focus-indicator" data-action="
              input:command-palette-input#onInput
              keydown:command-palette-input#onKeydown
            " role="combobox" spellcheck="false" type="text"/>
       </div>
       <div class="position-relative d-inline-block" data-view-component="true">
        <button aria-keyshortcuts="Control+Backspace" class="btn-octicon command-palette-input-clear-button" data-action="click:command-palette-input#onClear keypress:command-palette-input#onClear" data-target="command-palette-input.clearButton" data-view-component="true" hidden="hidden" id="command-palette-clear-button" type="button">
         <svg aria-hidden="true" class="octicon octicon-x-circle-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
          </path>
         </svg>
        </button>
        <tool-tip class="sr-only position-absolute" data-direction="w" data-type="label" data-view-component="true" for="command-palette-clear-button" id="tooltip-60629cf6-e928-4a19-84b8-1aa3b85507d6" popover="manual">
         Clear Command Palette
        </tool-tip>
       </div>
      </command-palette-input>
      <command-palette-page-stack data-action="command-palette-page-octicons-cached:command-palette-page-stack#cacheOcticons" data-default-scope-id="MDEwOlJlcG9zaXRvcnkzMzIwMDY2Nzc=" data-default-scope-type="Repository">
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search discussions
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           !
          </kbd>
          to search projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search teams
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='[""]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search people and organizations
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           &gt;
          </kbd>
          to activate command mode
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Go to your accessibility settings to change your keyboard shortcuts
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type author:@me to search your content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:pr to filter to pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:issue to filter to issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:project to filter to projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:open to filter to open content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="mx-3 my-2 flash flash-error d-flex flex-items-center" data-on-error="" data-scope-types="*">
        <div>
         <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
          </path>
         </svg>
        </div>
        <div class="px-2">
         We’ve encountered an error and some results aren't available at this time. Type a new search or try again later.
        </div>
       </command-palette-tip>
       <command-palette-tip class="h4 color-fg-default pl-3 pb-2 pt-3" data-match-mode="[^?]|^$" data-on-empty="" data-scope-types="*">
        No results matched your search
       </command-palette-tip>
       <div hidden="">
        <div data-octicon-id="arrow-right-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="arrow-right-color-fg-default" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-default" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="codespaces-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
          </path>
          <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="copy-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-copy color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
          </path>
          <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="dash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-dash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 7.75A.75.75 0 0 1 2.75 7h10a.75.75 0 0 1 0 1.5h-10A.75.75 0 0 1 2 7.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="file-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-file color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="gear-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-gear color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="lock-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-lock color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M4 4a4 4 0 0 1 8 0v2h.25c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 12.25 15h-8.5A1.75 1.75 0 0 1 2 13.25v-5.5C2 6.784 2.784 6 3.75 6H4Zm8.25 3.5h-8.5a.25.25 0 0 0-.25.25v5.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25ZM10.5 6V4a2.5 2.5 0 1 0-5 0v2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="moon-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-moon color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M9.598 1.591a.749.749 0 0 1 .785-.175 7.001 7.001 0 1 1-8.967 8.967.75.75 0 0 1 .961-.96 5.5 5.5 0 0 0 7.046-7.046.75.75 0 0 1 .175-.786Zm1.616 1.945a7 7 0 0 1-7.678 7.678 5.499 5.499 0 1 0 7.678-7.678Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="person-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-person color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="pencil-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-pencil color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="issue-opened-open" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-issue-opened open" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="git-pull-request-draft-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request-draft color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M3.25 1A2.25 2.25 0 0 1 4 5.372v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.251 2.251 0 0 1 3.25 1Zm9.5 14a2.25 2.25 0 1 1 0-4.5 2.25 2.25 0 0 1 0 4.5ZM2.5 3.25a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0ZM3.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm9.5 0a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM14 7.5a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm0-4.25a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="search-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sun-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sun color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8Zm0-1.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Zm5.657-8.157a.75.75 0 0 1 0 1.061l-1.061 1.06a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.06-1.06a.75.75 0 0 1 1.06 0Zm-9.193 9.193a.75.75 0 0 1 0 1.06l-1.06 1.061a.75.75 0 1 1-1.061-1.06l1.06-1.061a.75.75 0 0 1 1.061 0ZM8 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V.75A.75.75 0 0 1 8 0ZM3 8a.75.75 0 0 1-.75.75H.75a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 3 8Zm13 0a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 16 8Zm-8 5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 8 13Zm3.536-1.464a.75.75 0 0 1 1.06 0l1.061 1.06a.75.75 0 0 1-1.06 1.061l-1.061-1.06a.75.75 0 0 1 0-1.061ZM2.343 2.343a.75.75 0 0 1 1.061 0l1.06 1.061a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-1.06-1.06a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sync-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sync color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.705 8.005a.75.75 0 0 1 .834.656 5.5 5.5 0 0 0 9.592 2.97l-1.204-1.204a.25.25 0 0 1 .177-.427h3.646a.25.25 0 0 1 .25.25v3.646a.25.25 0 0 1-.427.177l-1.38-1.38A7.002 7.002 0 0 1 1.05 8.84a.75.75 0 0 1 .656-.834ZM8 2.5a5.487 5.487 0 0 0-4.131 1.869l1.204 1.204A.25.25 0 0 1 4.896 6H1.25A.25.25 0 0 1 1 5.75V2.104a.25.25 0 0 1 .427-.177l1.38 1.38A7.002 7.002 0 0 1 14.95 7.16a.75.75 0 0 1-1.49.178A5.5 5.5 0 0 0 8 2.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="trash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-trash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="key-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-key color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="comment-discussion-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-slash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell-slash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="paintbrush-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-paintbrush color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.134 1.535c.7-.509 1.416-.942 2.076-1.155.649-.21 1.463-.267 2.069.34.603.601.568 1.411.368 2.07-.202.668-.624 1.39-1.125 2.096-1.011 1.424-2.496 2.987-3.775 4.249-1.098 1.084-2.132 1.839-3.04 2.3a3.744 3.744 0 0 1-1.055 3.217c-.431.431-1.065.691-1.657.861-.614.177-1.294.287-1.914.357A21.151 21.151 0 0 1 .797 16H.743l.007-.75H.749L.742 16a.75.75 0 0 1-.743-.742l.743-.008-.742.007v-.054a21.25 21.25 0 0 1 .13-2.284c.067-.647.187-1.287.358-1.914.17-.591.43-1.226.86-1.657a3.746 3.746 0 0 1 3.227-1.054c.466-.893 1.225-1.907 2.314-2.982 1.271-1.255 2.833-2.75 4.245-3.777ZM1.62 13.089c-.051.464-.086.929-.104 1.395.466-.018.932-.053 1.396-.104a10.511 10.511 0 0 0 1.668-.309c.526-.151.856-.325 1.011-.48a2.25 2.25 0 1 0-3.182-3.182c-.155.155-.329.485-.48 1.01a10.515 10.515 0 0 0-.309 1.67Zm10.396-10.34c-1.224.89-2.605 2.189-3.822 3.384l1.718 1.718c1.21-1.205 2.51-2.597 3.387-3.833.47-.662.78-1.227.912-1.662.134-.444.032-.551.009-.575h-.001V1.78c-.014-.014-.113-.113-.548.027-.432.14-.995.462-1.655.942Zm-4.832 7.266-.001.001a9.859 9.859 0 0 0 1.63-1.142L7.155 7.216a9.7 9.7 0 0 0-1.161 1.607c.482.302.889.71 1.19 1.192Z">
          </path>
         </svg>
        </div>
        <command-palette-item-group data-default-priority="0" data-group-hint="" data-group-id="top" data-group-limits="{}" data-group-title="Top result">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="1" data-group-hint="Type &gt; to filter" data-group-id="commands" data-group-limits='{"static_items_page":50,"issue":50,"pull_request":50,"discussion":50}' data-group-title="Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="2" data-group-hint="Type &gt; to filter" data-group-id="global_commands" data-group-limits='{"issue":0,"pull_request":0,"discussion":0}' data-group-title="Global Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="3" data-group-hint="" data-group-id="this_page" data-group-limits="{}" data-group-title="This Page">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="4" data-group-hint="" data-group-id="files" data-group-limits="{}" data-group-title="Files">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="5" data-group-hint="" data-group-id="default" data-group-limits='{"static_items_page":50}' data-group-title="Default">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="6" data-group-hint="" data-group-id="pages" data-group-limits='{"repository":10}' data-group-title="Pages">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="7" data-group-hint="" data-group-id="access_policies" data-group-limits="{}" data-group-title="Access Policies">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="8" data-group-hint="" data-group-id="organizations" data-group-limits="{}" data-group-title="Organizations">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="9" data-group-hint="" data-group-id="repositories" data-group-limits="{}" data-group-title="Repositories">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="10" data-group-hint="Type # to filter" data-group-id="references" data-group-limits="{}" data-group-title="Issues, pull requests, and discussions">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="11" data-group-hint="" data-group-id="teams" data-group-limits="{}" data-group-title="Teams">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="12" data-group-hint="" data-group-id="users" data-group-limits="{}" data-group-title="Users">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="13" data-group-hint="" data-group-id="memex_projects" data-group-limits="{}" data-group-title="Projects">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="14" data-group-hint="" data-group-id="projects" data-group-limits="{}" data-group-title="Projects (classic)">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="15" data-group-hint="" data-group-id="footer" data-group-limits="{}" data-group-title="Footer">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="16" data-group-hint="" data-group-id="modes_help" data-group-limits="{}" data-group-title="Modes">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="17" data-group-hint="" data-group-id="filters_help" data-group-limits="{}" data-group-title="Use filters in issues, pull requests, discussions, and projects">
        </command-palette-item-group>
        <command-palette-page data-page-title="kcrid001" data-scope-id="MDQ6VXNlcjc3ODQ2OTQ1" data-scope-type="owner" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
        <command-palette-page data-page-title="KatieCriderAdvancedGenomicsLog" data-scope-id="MDEwOlJlcG9zaXRvcnkzMzIwMDY2Nzc=" data-scope-type="repository" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
       </div>
       <command-palette-page data-is-root="">
       </command-palette-page>
       <command-palette-page data-page-title="kcrid001" data-scope-id="MDQ6VXNlcjc3ODQ2OTQ1" data-scope-type="owner">
       </command-palette-page>
       <command-palette-page data-page-title="KatieCriderAdvancedGenomicsLog" data-scope-id="MDEwOlJlcG9zaXRvcnkzMzIwMDY2Nzc=" data-scope-type="repository">
       </command-palette-page>
      </command-palette-page-stack>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="search-links">
      </server-defined-provider>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="help">
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues
         </strong>
         and
         <strong>
          pull requests
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues, pull requests, discussions,
         </strong>
         and
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="@" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          organizations, repositories,
         </strong>
         and
         <strong>
          users
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          @
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="!" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          !
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="/" data-scope-types='["repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          files
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          /
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="&gt;">
        <span data-target="command-palette-help.titleElement">
         Activate
         <strong>
          command mode
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          &gt;
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:pr">
        <span data-target="command-palette-help.titleElement">
         Filter to pull requests
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:pr
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:issue">
        <span data-target="command-palette-help.titleElement">
         Filter to issues
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:issue
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:discussion" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:discussion
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:project" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to projects
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:project
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:open">
        <span data-target="command-palette-help.titleElement">
         Filter to open issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:open
         </kbd>
        </span>
       </command-palette-help>
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/commands" data-supported-modes="[]" data-supports-commands="" data-targets="command-palette.serverDefinedProviderElements" data-type="commands">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_page_navigation" data-supported-modes='[""]' data-supported-scope-types='["","owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to" data-supported-modes='["@","@"]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to_members_only" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_members_only_prefetched" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/files" data-supported-modes='["/"]' data-supported-scope-types='["repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="files">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/discussions" data-supported-modes='["#"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/projects" data-supported-modes='["#","!"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/recent_issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/teams" data-supported-modes='["@",""]' data-supported-scope-types='["owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/name_with_owner_repository" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
     </command-palette>
    </details-dialog>
   </details>
   <div class="position-fixed bottom-0 left-0 ml-5 mb-5 js-command-palette-toasts" style="z-index: 1000">
    <div class="Toast Toast--loading" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="Toast--spinner" height="18" viewbox="0 0 32 32" width="18">
       <path d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4" fill="#959da5">
       </path>
       <path d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z" fill="#ffffff">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--error" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-stop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M4.47.22A.749.749 0 0 1 5 0h6c.199 0 .389.079.53.22l4.25 4.25c.141.14.22.331.22.53v6a.749.749 0 0 1-.22.53l-4.25 4.25A.749.749 0 0 1 11 16H5a.749.749 0 0 1-.53-.22L.22 11.53A.749.749 0 0 1 0 11V5c0-.199.079-.389.22-.53Zm.84 1.28L1.5 5.31v5.38l3.81 3.81h5.38l3.81-3.81V5.31L10.69 1.5ZM8 4a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 4Zm0 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--warning" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--success" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-info" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.5 7.75A.75.75 0 0 1 7.25 7h1a.75.75 0 0 1 .75.75v2.75h.25a.75.75 0 0 1 0 1.5h-2a.75.75 0 0 1 0-1.5h.25v-2h-.25a.75.75 0 0 1-.75-.75ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
   </div>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <template class="js-user-list-create-dialog-template" data-label="Create list">
       <div class="Box-header">
        <h2 class="Box-title">
         Create list
        </h2>
       </div>
       <!-- '"` -->
       <!-- </textarea></xmp> -->
       <form accept-charset="UTF-8" action="/stars/ninenia/lists" class="Box-body d-flex flex-column p-3 js-user-list-form" data-turbo="false" method="post">
        <input autocomplete="off" name="authenticity_token" type="hidden" value="18emLwrPguWmhThq4yeJNyYd0m0o9bLXQ6aNX80Z-MMHqwwLnxVUQkW4kg1w_xCRgx4ZGKzSm-vdDKjqecjudA">
         <p class="color-fg-subtle mb-3">
          Create a list to organize your starred repositories.
         </p>
         <input name="repository_id" type="hidden" value="{{ repositoryId }}"/>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <auto-check required="" src="/stars/ninenia/list-check?attr=name">
           <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
            <input aria-label="List name" autofocus="" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="32" maxlength="32" name="user_list[name]" placeholder="⭐️ Name this list" required="" type="text" value="{{ placeholderName }}"/>
           </text-expander>
           <input data-csrf="true" type="hidden" value="1Xq91gxv0STKGaThGvkLmDhCAalAKKIHfI5zV8Szsf-1UuEnw0GDQBxSO6X5-8-uOahf2OzaHD3QpAmeJtsFUQ">
           </input>
          </auto-check>
          <p class="note error position-relative js-user-list-error" hidden="">
           Name .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           32 remaining
          </p>
         </div>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
           <textarea aria-label="List description" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="160" maxlength="160" name="user_list[description]" placeholder="Write a description" style="height: 74px; min-height: 74px"></textarea>
          </text-expander>
          <p class="note error position-relative js-user-list-error" hidden="">
           Description .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           160 remaining
          </p>
         </div>
         <div class="js-user-list-base flash flash-error mx-0 mt-0 mb-2" data-generic-message="Unable to save your list at this time." data-view-component="true" hidden="hidden">
          .
         </div>
         <button class="Button--primary Button--medium Button Button--fullWidth mt-2" data-disable-invalid="true" data-submitting-message="Creating..." data-view-component="true" disabled="disabled" type="submit">
          <span class="Button-content">
           <span class="Button-label">
            Create
           </span>
          </span>
         </button>
         <p class="note mt-2 mb-0">
          <strong>
           Tip:
          </strong>
          type
          <code>
           :
          </code>
          to add emoji to the name or description.
         </p>
        </input>
       </form>
       <div class="Box-footer Box-row--gray text-small color-fg-muted d-flex flex-items-baseline py-2" data-view-component="true">
        <span aria-label="Feature Release Label: Beta" class="Label Label--success Label--inline px-2 mr-2" data-view-component="true" title="Feature Release Label: Beta">
         Beta
        </span>
        <span class="mr-1">
         Lists are currently in beta.
        </span>
        <a href="https://github.com//github/feedback/discussions/categories/lists">
         Share feedback and report bugs.
        </a>
       </div>
      </template>
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header">
       <div class="d-flex flex-wrap flex-justify-end mb-3 container-xl px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal" id="repo-title-component">
          <img alt="Owner avatar" class="avatar mr-2 d-none d-md-block avatar-user" height="24" src="https://avatars.githubusercontent.com/u/77846945?s=48&amp;v=4" width="24"/>
          <strong class="mr-2 flex-self-stretch d-none d-md-block" itemprop="name">
           <a class="color-fg-default" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/kcrid001/KatieCriderAdvancedGenomicsLog">
            KatieCriderAdvancedGenomicsLog
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1 d-none d-md-block">
           Public
          </span>
         </div>
         <div class="d-none d-md-block">
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <include-fragment loading="lazy" src="/notifications/332006677/watch_subscription?aria_id_prefix=repository-details&amp;button_block=false&amp;show_count=true">
            <div data-hide-on-error="">
             <button aria-label="Watching a repository" class="tooltipped tooltipped-n Button--secondary Button--small Button" data-view-component="true" disabled="disabled" type="button">
              <span class="Button-content">
               <span class="Button-visual Button-leadingVisual">
                <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                 </path>
                </svg>
               </span>
               <span class="Button-label">
                Watch
               </span>
              </span>
             </button>
            </div>
            <p data-show-on-error="" hidden="">
             <span class="d-flex pt-1">
              <svg aria-hidden="true" class="octicon octicon-alert mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
               </path>
              </svg>
              Couldn't load subscription status.
              <button class="Button--link Button--medium Button" data-retry-button="" data-view-component="true" type="button">
               <span class="Button-content">
                <span class="Button-label">
                 Retry
                </span>
               </span>
              </button>
             </span>
            </p>
           </include-fragment>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a class="btn-sm btn BtnGroup-item" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":332006677,"originating_url":"https://github.com/kcrid001/KatieCriderAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="d040935d8e241e262a4e966d7d4cfb3b7a0bbe4fe789bdb6f1754f13607c147e" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/fork" icon="repo-forked" id="fork-button">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             Fork
             <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
              0
             </span>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="fork-button" id="tooltip-24c66640-4b49-46d5-9340-0dbe7f44b101" popover="manual">
              Fork your own copy of kcrid001/KatieCriderAdvancedGenomicsLog
             </tool-tip>
            </a>
            <details class="details-reset details-overlay BtnGroup-parent d-inline-block position-relative" data-view-component="true" group_item="true" id="my-forks-menu-332006677">
             <summary aria-label="See your forks of this repository" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
              <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
               </path>
              </svg>
             </summary>
             <details-menu class="SelectMenu right-0" role="menu" src="/kcrid001/KatieCriderAdvancedGenomicsLog/my_forks_menu_content?can_fork=true">
              <div class="SelectMenu-modal">
               <button aria-label="Close menu" class="SelectMenu-closeButton position-absolute right-0 m-2" data-toggle-for="my-forks-menu-332006677" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
               <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-5dc520">
                <div class="SelectMenu-list">
                 <include-fragment aria-label="Loading" class="SelectMenu-loading">
                  <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                   <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                   </circle>
                   <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                   </path>
                  </svg>
                 </include-fragment>
                </div>
               </div>
              </div>
             </details-menu>
            </details>
           </div>
          </li>
          <li>
           <template class="js-unstar-confirmation-dialog-template">
            <div class="Box-header">
             <h2 class="Box-title">
              Unstar this repository?
             </h2>
            </div>
            <div class="Box-body">
             <p class="mb-3">
              This will remove {{ repoNameWithOwner }} from the {{ listsWithCount }} that it's been added to.
             </p>
             <div class="form-actions">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="{{ confirmUrl }}" class="js-social-confirmation-form" data-turbo="false" method="post">
               <input name="authenticity_token" type="hidden" value="{{ confirmCsrfToken }}"/>
               <input name="confirm" type="hidden" value="true"/>
               <button class="btn-danger btn width-full" data-close-dialog="true" data-view-component="true" type="submit">
                Unstar
               </button>
              </form>
             </div>
            </div>
           </template>
           <div class="js-toggler-container js-social-container starring-container d-flex" data-view-component="true">
            <div class="starred BtnGroup flex-1 ml-0" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/kcrid001/KatieCriderAdvancedGenomicsLog/unstar" class="js-social-form BtnGroup-parent flex-auto js-deferred-toggler-target" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="8wiZEDqbx_w3YkqUn0-2MqluLFd75mWr-ebSTZJe6rfQUa3sECOhGeQZNEClsbAcI8orqOBuhF0p62CRv_VdkQ">
               <input class="js-confirm-csrf-token" data-csrf="true" type="hidden" value="WAYrLoalhcWw1iW9vO10QNg-ttZF8l_4qXYq0-UtO4t7Xx_SrB3jIGOtW2mGE3JuUpqxKd56vg55e5gPyIaMrQ">
                <input name="context" type="hidden" value="repository"/>
                <button aria-label="Unstar this repository (0)" class="rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":332006677,"originating_url":"https://github.com/kcrid001/KatieCriderAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="f1ddccb5568e0e613110739b5e9fd4511c64dc3dfe27ee599debb007286de500" data-view-component="true" type="submit">
                 <svg aria-hidden="true" class="octicon octicon-star-fill starred-button-icon d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                  </path>
                 </svg>
                 <span class="d-inline" data-view-component="true">
                  Starred
                 </span>
                 <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-unstar" title="0">
                  0
                 </span>
                </button>
               </input>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-332006677-starred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/kcrid001/KatieCriderAdvancedGenomicsLog/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-332006677-starred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-539bdb">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
            <div class="unstarred BtnGroup ml-0 flex-1" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/kcrid001/KatieCriderAdvancedGenomicsLog/star" class="js-social-form BtnGroup-parent flex-auto" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="SrSdqkXYv8AcHXzuHgJxaUjTn7MG6yG8NgAesm4iURBnUmBg0XcApMqZL9-QhOzbl0yGc-6CHKInPUmK1i4gLQ">
               <input name="context" type="hidden" value="repository"/>
               <button aria-label="Star this repository (0)" class="js-toggler-target rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":332006677,"originating_url":"https://github.com/kcrid001/KatieCriderAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="679ed782b4ada6a14544547ab8659ba5b89e25a3474dae3ad271575e841d324d" data-view-component="true" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
                <span class="d-inline" data-view-component="true">
                 Star
                </span>
                <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
                 0
                </span>
               </button>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-332006677-unstarred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/kcrid001/KatieCriderAdvancedGenomicsLog/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-332006677-unstarred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-71824c">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div class="container-xl">
        <div data-turbo-replace="" id="responsive-meta-container">
         <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
          <div class="d-flex gap-2 mt-n3 mb-3 flex-wrap">
           <div class="d-flex flex-row gap-2">
            <notifications-list-subscription-form data-action="notifications-dialog-label-toggled:notifications-list-subscription-form#handleDialogLabelToggle">
             <details class="details-reset details-overlay details-overlay-dark" data-action="toggle:notifications-list-subscription-form#detailsToggled" data-target="notifications-list-subscription-form.details">
              <summary aria-label="Notifications" class="Button--iconOnly Button--secondary Button--medium Button p-0" data-ga-click="Repository, click Watch settings, action:files#disambiguate" data-hydro-click='{"event_type":"repository.click","payload":{"target":"WATCH_BUTTON","repository_id":332006677,"originating_url":"https://github.com/kcrid001/KatieCriderAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="2603207ce5f9e1a1902df2fe7026f8d6bfe08b9df959fe190c344e0110b52b88" data-view-component="true" id="notifications-button">
               <span class="Button-content">
                <span class="Button-label">
                 <span data-menu-button="">
                  <span data-target="notifications-list-subscription-form.unwatchButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.stopIgnoringButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-bell-slash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.watchButtonCopy">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                 </span>
                </span>
               </span>
              </summary>
              <details-dialog aria-label="Notification settings" class="anim-fade-in fast Box Box--overlay">
               <div data-target="notifications-list-subscription-form.menu">
                <div class="Box-header">
                 <button aria-label="Close dialog" class="Box-btn-octicon btn-octicon float-right" data-action="click:notifications-list-subscription-form#closeMenu" data-close-dialog="" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                 <h3 class="Box-title">
                  Notifications
                 </h3>
                </div>
                <div class="SelectMenu-list mb-0" role="menu">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitForm" data-target="notifications-list-subscription-form.form" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="6lXxtZXQa-nTr1FPM-l6kfMCKbUVQjack01AN7EAqfzylf3Pi28e6D-8ZkXvCfzmrCe6oMAYnUixtrLD5rEXZQ">
                   <input name="repository_id" type="hidden" value="332006677"/>
                   <button aria-checked="true" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="included">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Participating and @mentions
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Only receive notifications from this repository when participating or @mentioned.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="subscribed">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      All Activity
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Notified of all notifications on this repository.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="ignore">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Ignore
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Never be notified.
                     </div>
                    </div>
                   </button>
                  </input>
                 </form>
                 <button aria-checked="false" aria-haspopup="true" class="SelectMenu-item flex-items-start pr-3" data-action="click:notifications-list-subscription-form#openCustomDialog" data-target="notifications-list-subscription-form.customButton" role="menuitemradio" type="button">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                    </path>
                   </svg>
                  </span>
                  <div>
                   <div class="d-flex flex-items-start flex-justify-between">
                    <div class="f5 text-bold">
                     Custom
                    </div>
                    <div class="f5 pr-1">
                     <svg aria-hidden="true" class="octicon octicon-arrow-right" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </div>
                   <div class="text-small color-fg-muted text-normal pb-1">
                    Select events you want to be notified of in addition to participating and @mentions.
                   </div>
                  </div>
                 </button>
                 <div class="px-3 py-2 d-flex color-bg-subtle flex-items-center">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile SelectMenu-icon SelectMenu-icon--device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </span>
                  <span classname="text-small color-fg-muted text-normal pb-1">
                   Get push notifications on
                   <a href="https://apps.apple.com/app/apple-store/id1477376905?ct=watch-dropdown&amp;mt=8&amp;pt=524675" rel="noopener noreferrer" target="_blank">
                    iOS
                   </a>
                   or
                   <a href="https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dwatch-dropdown%26utm_medium%3Dweb%26utm_source%3Dgithub" rel="noopener noreferrer" target="_blank">
                    Android
                   </a>
                   .
                  </span>
                 </div>
                </div>
               </div>
               <div data-target="notifications-list-subscription-form.customDialog" hidden="">
                <header class="SelectMenu-header pb-0 border-bottom-0 px-2 px-sm-3">
                 <h1 class="f3 SelectMenu-title d-inline-flex">
                  <button aria-label="Return to menu" class="color-bg-default border-0 px-2 py-0 m-0 Link--secondary f5" data-action="click:notifications-list-subscription-form#closeCustomDialog" type="button">
                   <svg aria-hidden="true" class="octicon octicon-arrow-left" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z">
                    </path>
                   </svg>
                  </button>
                  Custom
                 </h1>
                </header>
                <div class="overflow-visible">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitCustomForm" data-target="notifications-list-subscription-form.customform" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="33nPvJd34ihA1nj1kWynENOinQD6GoHW46UXr4Kb13LHucPGiciXKazFT_9NjCFnjIcOFS9AKgLBXuVb1Spp6w">
                   <input name="repository_id" type="hidden" value="332006677"/>
                   <fieldset>
                    <legend>
                     <div class="text-small color-fg-muted pt-0 pr-3 pb-3 pl-6 pl-sm-5 border-bottom mb-3">
                      Select events you want to be notified of in addition to participating and @mentions.
                     </div>
                    </legend>
                    <div data-target="notifications-list-subscription-form.labelInputs">
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Issue"/>
                      Issues
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="PullRequest"/>
                      Pull requests
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Release"/>
                      Releases
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input aria-describedby="Discussion-disabled" aria-disabled="true" data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Discussion"/>
                      Discussions
                     </label>
                     <div class="color-fg-muted" id="Discussion-disabled">
                      &gt;
                          Discussions are not enabled for this repository
                     </div>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="SecurityAlert"/>
                      Security alerts
                     </label>
                    </div>
                   </fieldset>
                   <div class="pt-2 pb-3 px-3 d-flex flex-justify-start flex-row-reverse">
                    <button class="Button--primary Button--medium Button ml-2" data-target="notifications-list-subscription-form.customSubmit" data-view-component="true" disabled="disabled" name="do" type="submit" value="custom">
                     <span class="Button-content">
                      <span class="Button-label">
                       Apply
                      </span>
                     </span>
                    </button>
                    <button class="Button--secondary Button--medium Button" data-action="click:notifications-list-subscription-form#resetForm" data-close-dialog="" data-view-component="true" type="button">
                     <span class="Button-content">
                      <span class="Button-label">
                       Cancel
                      </span>
                     </span>
                    </button>
                   </div>
                  </input>
                 </form>
                </div>
               </div>
              </details-dialog>
             </details>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="notifications-button" id="tooltip-fd1cbf97-eee4-4211-9dc0-ef2e211a8846" popover="manual">
              Notification settings
             </tool-tip>
            </notifications-list-subscription-form>
            <div class="BtnGroup d-flex" data-view-component="true">
             <a aria-labelledby="tooltip-592818f8-df39-4233-9f57-abd78b5622f5" class="Button Button--iconOnly Button--secondary Button--medium" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":332006677,"originating_url":"https://github.com/kcrid001/KatieCriderAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="d040935d8e241e262a4e966d7d4cfb3b7a0bbe4fe789bdb6f1754f13607c147e" data-view-component="true" group_item="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/fork" id="fork-icon-button" variant="small">
              <svg aria-hidden="true" class="octicon octicon-repo-forked Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
               </path>
              </svg>
             </a>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="fork-icon-button" id="tooltip-592818f8-df39-4233-9f57-abd78b5622f5" popover="manual">
              Fork your own copy of kcrid001/KatieCriderAdvancedGenomicsLog
             </tool-tip>
            </div>
            <div class="js-toggler-container starring-container" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/kcrid001/KatieCriderAdvancedGenomicsLog/unstar" class="starred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="bMmejQF8uEr8nXSrNp6ZSDfmwRXWaEYwLGH8s-wRG5lPkKpxK8Tery_mCn8MYJ9mvULG6k3gp8b8bE5vwbqsvw">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-885afcf4-0098-4c7d-90aa-edd1aa9cbfd8" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target starred-button-icon" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":332006677,"originating_url":"https://github.com/kcrid001/KatieCriderAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="f1ddccb5568e0e613110739b5e9fd4511c64dc3dfe27ee599debb007286de500" data-view-component="true" id="icon-button-85b7afc2-91f2-40dd-8e71-16116843a401" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-85b7afc2-91f2-40dd-8e71-16116843a401" id="tooltip-885afcf4-0098-4c7d-90aa-edd1aa9cbfd8" popover="manual">
                Unstar this repository
               </tool-tip>
              </input>
             </form>
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/kcrid001/KatieCriderAdvancedGenomicsLog/star" class="unstarred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="6laY19wiEmzuzNuugqpnDuqNGY616rPSBZSrdCsbeQfHsGUdSI2tCDhIiJ8MLPq8NRIATl2DjswUqfxMkxcIOg">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-8d65ab6a-1595-45ba-979a-0242e1c335e1" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":332006677,"originating_url":"https://github.com/kcrid001/KatieCriderAdvancedGenomicsLog","user_id":126681664}}' data-hydro-click-hmac="679ed782b4ada6a14544547ab8659ba5b89e25a3474dae3ad271575e841d324d" data-view-component="true" id="icon-button-66c4b9d6-8646-4966-9496-3073591fe4f7" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-66c4b9d6-8646-4966-9496-3073591fe4f7" id="tooltip-8d65ab6a-1595-45ba-979a-0242e1c335e1" popover="manual">
                Star this repository
               </tool-tip>
              </input>
             </form>
            </div>
           </div>
           <div class="d-flex flex-row gap-2">
           </div>
          </div>
          <p class="f4 mb-3 color-fg-muted">
           log of work in the 2021 Spring Advanced Genomics class
          </p>
          <div class="mb-3">
           <ul aria-label="Repository details" class="d-flex flex-wrap mb-2 gap-2">
            <a class="Link--secondary no-underline d-block mr-2" href="/kcrid001/KatieCriderAdvancedGenomicsLog/stargazers" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             stars
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/kcrid001/KatieCriderAdvancedGenomicsLog/forks" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             forks
            </a>
            <div class="color-fg-muted mr-2" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-eye mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              1
             </span>
             watching
            </div>
            <a class="Link--secondary no-underline d-block mr-2" href="/kcrid001/KatieCriderAdvancedGenomicsLog/branches" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
              </path>
             </svg>
             <include-fragment src="/kcrid001/KatieCriderAdvancedGenomicsLog/branch-count">
              <span>
               Branches
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/kcrid001/KatieCriderAdvancedGenomicsLog/tags" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
              </path>
             </svg>
             <include-fragment src="/kcrid001/KatieCriderAdvancedGenomicsLog/tag-count">
              <span>
               Tags
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/kcrid001/KatieCriderAdvancedGenomicsLog/activity" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
              </path>
             </svg>
             <span>
              Activity
             </span>
            </a>
           </ul>
           <div class="mb-2 d-flex color-fg-muted">
            <div class="d-flex flex-items-center" style="height: 21px">
             <svg aria-hidden="true" class="octicon octicon-globe flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
              </path>
             </svg>
            </div>
            <span class="flex-auto min-width-0 width-fit">
             Public repository
            </span>
           </div>
          </div>
         </div>
        </div>
        <div class="border-bottom mx-xl-5">
        </div>
       </div>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <a class="d-none js-github-dev-shortcut" data-hotkey=".,Mod+Alt+." href="https://github.dev/">
         Open in github.dev
        </a>
        <a class="d-none js-github-dev-new-tab-shortcut" data-hotkey="Shift+.,Shift+&gt;,&gt;" href="https://github.dev/" rel="noopener noreferrer" target="_blank">
         Open in a new github.dev tab
        </a>
        <a class="d-none" data-hotkey=",,Mod+Alt+," href="/codespaces/new/kcrid001/KatieCriderAdvancedGenomicsLog?resume=1" target="_blank">
         Open in codespace
        </a>
        <h1 class="sr-only">
         kcrid001/KatieCriderAdvancedGenomicsLog
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/kcrid001/KatieCriderAdvancedGenomicsLog/spoofed_commit_check/b7e15c42127e797254e5a58a5f306d2fec0582d8">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":true,"path":"/","repo":{"id":332006677,"defaultBranch":"main","name":"KatieCriderAdvancedGenomicsLog","ownerLogin":"kcrid001","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2021-01-23T00:49:31.000+08:00","ownerAvatar":"https://avatars.githubusercontent.com/u/77846945?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":{"id":126681664,"login":"ninenia","userEmail":"2945518981@qq.com"},"refInfo":{"name":"main","listCacheKey":"v0:1611334199.0","canEdit":true,"refType":"branch","currentOid":"b7e15c42127e797254e5a58a5f306d2fec0582d8"},"tree":{"items":[{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":1,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/kcrid001/KatieCriderAdvancedGenomicsLog/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":true,"httpUrl":"https://github.com/kcrid001/KatieCriderAdvancedGenomicsLog.git","showCloneWarning":false,"sshUrl":"git@github.com:kcrid001/KatieCriderAdvancedGenomicsLog.git","sshCertificatesRequired":false,"sshCertificatesAvailable":false,"ghCliUrl":"gh repo clone kcrid001/KatieCriderAdvancedGenomicsLog","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"x-github-client://openRepo/https://github.com/kcrid001/KatieCriderAdvancedGenomicsLog","showVisualStudioCloneButton":true,"visualStudioCloneUrl":"git-client://clone?repo=https%3A%2F%2Fgithub.com%2Fkcrid001%2FKatieCriderAdvancedGenomicsLog","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/kcrid001/KatieCriderAdvancedGenomicsLog/archive/refs/heads/main.zip"}},"repoPolicyInfo":{"allowed":true,"canBill":true,"changesWouldBeSafe":true,"disabledByBusiness":false,"disabledByOrganization":false,"hasIpAllowLists":false},"currentUserIsEnterpriseManaged":false,"enterpriseManagedBusinessName":null,"codespacesEnabled":true,"hasAccessToCodespaces":true},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"16","overviewFiles":[{"displayName":"README.md","repoName":"KatieCriderAdvancedGenomicsLog","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eKatie Crider's Advanced Genomics Log\u003c/h1\u003e\u003ca id=\"user-content-katie-criders-advanced-genomics-log\" class=\"anchor\" aria-label=\"Permalink: Katie Crider's Advanced Genomics Log\" href=\"#katie-criders-advanced-genomics-log\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSpring 2021\u003c/h2\u003e\u003ca id=\"user-content-spring-2021\" class=\"anchor\" aria-label=\"Permalink: Spring 2021\" href=\"#spring-2021\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 02 Exercises: 1/22/21\u003c/h2\u003e\u003ca id=\"user-content-day-02-exercises-12221\" class=\"anchor\" aria-label=\"Permalink: Day 02 Exercises: 1/22/21\" href=\"#day-02-exercises-12221\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#exercises day02:\n\n#1- Logon to the cluster @turing.hpc.odu.edu\n(base) Katies-MacBook-Air-4:~ katiecrider$ ssh kcrid001@turing.hpc.odu.edu             \n[kcrid001@turing1 ~]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider\n\n#2- Make a directory in your course workspace called data\n[kcrid001@turing1 katiecrider]$ mkdir data\n\n#3- Make a directory called exercises in your data directory\n[kcrid001@turing1 katiecrider]$ cd data\n[kcrid001@turing1 data]$ mkdir exercises\n[kcrid001@turing1 data]$ cd exercises\n\n#4- execute a pwd command and start a log of your commands with a header for today's date in your README.md github logfile in your workspace\n[kcrid001@turing1 exercises]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/exercises\n\n#5- cp the Exercise2.fasta.gz and Exercise2.fastq.tar.gz files into your exercises directory from the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02 directory\n[kcrid001@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise2.fast* ./\n[kcrid001@turing1 exercises]$ ls\nExercise2.fasta.gz  Exercise2.fastq.tar.gz\n\n#6- unzip and untar the files\n[kcrid001@turing1 exercises]$ gunzip Exercise2.fasta.gz\n[kcrid001@turing1 exercises]$ ls\nExercise2.fasta  Exercise2.fastq.tar.gz\n[kcrid001@turing1 exercises]$ tar -zxvf Exercise2.fastq.tar.gz\nExercise2.fastq\n[kcrid001@turing1 exercises]$ ls\nExercise2.fasta  Exercise2.fastq  Exercise2.fastq.tar.gz\n[kcrid001@turing1 exercises]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/exercises\n\n#7- add these commands to your notebook file\nDone!\n\n#8- calculate how many sequences are in each file and add these results to your notebook file\n[kcrid001@turing1 exercises]$ head -1 Exercise2.fasta\n\u0026gt;scaffold10078|size20675\n[kcrid001@turing1 exercises]$ grep -c \\\u0026gt; Exercise2.fasta\n138\n[kcrid001@turing1 exercises]$ head -4 Exercise2.fastq\n@HS3:541:HAYTUADXX:1:1101:1297:1938 1:N:0:AGTTCC\nNCGGTGACAGGATTTTCAGGGCAGGAGGGACATTCGGGAAGGGGGAGGGAGATAAAAGAGGCAA\n+\n#1=DBDDFHHHHHJJJIJJJJJJJJIHJJIIIIJJJJJGIIJJJJAHFDBD?CDDDDDDDDDDD\n[kcrid001@turing1 exercises]$ wc -l Exercise2.fastq\n245216 Exercise2.fastq\n[kcrid001@turing1 exercises]$ echo 245216/4 | bc\n61304\n[kcrid001@turing1 exercises]$ grep -c \u0026quot;@HS3:541\u0026quot; Exercise2.fastq\n61304\n\n#9 - cp the avg_cov_len_fasta_advbioinf.py from the /cm/shared/courses/dbarshis/21AdvGenomics/scripts directory into your class scripts directory\n[kcrid001@turing1 exercises]$ cd ../../\n[kcrid001@turing1 katiecrider]$ cd scripts\n[kcrid001@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics//scripts/avg_cov_len_fasta_advbioinf.py ./ \n\n#10- start an interactive compute session and re-navigate to your exercises directory\nI had trouble doing this with Turing so I switched over to Wahab\n\n[kcrid001@turing1 katiecrider]$ exit\nlogout\nConnection to turing.hpc.odu.edu closed.\n(base) Katies-MacBook-Air-4:~ katiecrider$ ssh kcrid001@wahab.hpc.odu.edu\nwahab-01:~\u0026gt; cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/\nwahab-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider\u0026gt; salloc\nsalloc: Pending job allocation 133738\nsalloc: job 133738 queued and waiting for resources\nsalloc: job 133738 has been allocated resources\nsalloc: Granted job allocation 133738\nThis session will be terminated in 7 days. If your application requires\na longer excution time, please use command \u0026quot;salloc -t N-0\u0026quot; where N is the\nnumber of days that you need.\n\n#11- run the avg_cov_len_fasta_DJB.py script on your Exercise2.fasta file by typing the path to the script followed by the Exercise2.fasta file name\ne2-w6420b-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider\u0026gt; cd data/exercises/\ne2-w6420b-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/exercises\u0026gt; ../../scripts/avg_cov_len_fasta_advbioinf.py Exercise2.fasta\nThe total number of sequences is 138\nThe average sequence length is 126640\nThe total number of bases is 17476447\nThe minimum sequence length is 1122\nThe maximum sequence length is 562680\nThe N50 is 187037\nMedian Length = 92612\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 138\ncontigs \u0026gt;= 1000bp = 138\ncontigs \u0026gt;= 2000bp = 135\n\n#12- did you add all these entries to your notebook file, including the results of the script?\nYes\n\n#13- push your notebook file to your github page\n(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ git add README.md\n(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ git commit -m 'updating notebook with day02 exercises'\n(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ git push -u origin main\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eexercises day02:\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1- Logon to the cluster @turing.hpc.odu.edu\u003c/span\u003e\n(base) Katies-MacBook-Air-4:\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e katiecrider$ ssh kcrid001@turing.hpc.odu.edu             \n[kcrid001@turing1 \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2- Make a directory in your course workspace called data\u003c/span\u003e\n[kcrid001@turing1 katiecrider]$ mkdir data\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3- Make a directory called exercises in your data directory\u003c/span\u003e\n[kcrid001@turing1 katiecrider]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e data\n[kcrid001@turing1 data]$ mkdir exercises\n[kcrid001@turing1 data]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e exercises\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4- execute a pwd command and start a log of your commands with a header for today's date in your README.md github logfile in your workspace\u003c/span\u003e\n[kcrid001@turing1 exercises]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/exercises\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e5- cp the Exercise2.fasta.gz and Exercise2.fastq.tar.gz files into your exercises directory from the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02 directory\u003c/span\u003e\n[kcrid001@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise2.fast\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e ./\n[kcrid001@turing1 exercises]$ ls\nExercise2.fasta.gz  Exercise2.fastq.tar.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e6- unzip and untar the files\u003c/span\u003e\n[kcrid001@turing1 exercises]$ gunzip Exercise2.fasta.gz\n[kcrid001@turing1 exercises]$ ls\nExercise2.fasta  Exercise2.fastq.tar.gz\n[kcrid001@turing1 exercises]$ tar -zxvf Exercise2.fastq.tar.gz\nExercise2.fastq\n[kcrid001@turing1 exercises]$ ls\nExercise2.fasta  Exercise2.fastq  Exercise2.fastq.tar.gz\n[kcrid001@turing1 exercises]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/exercises\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e7- add these commands to your notebook file\u003c/span\u003e\nDone\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e8- calculate how many sequences are in each file and add these results to your notebook file\u003c/span\u003e\n[kcrid001@turing1 exercises]$ head -1 Exercise2.fasta\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003escaffold10078\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003esize20675\n[kcrid001@turing1 exercises]$ grep -c \u003cspan class=\"pl-cce\"\u003e\\\u0026gt;\u003c/span\u003e Exercise2.fasta\n138\n[kcrid001@turing1 exercises]$ head -4 Exercise2.fastq\n@HS3:541:HAYTUADXX:1:1101:1297:1938 1:N:0:AGTTCC\nNCGGTGACAGGATTTTCAGGGCAGGAGGGACATTCGGGAAGGGGGAGGGAGATAAAAGAGGCAA\n+\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1=DBDDFHHHHHJJJIJJJJJJJJIHJJIIIIJJJJJGIIJJJJAHFDBD?CDDDDDDDDDDD\u003c/span\u003e\n[kcrid001@turing1 exercises]$ wc -l Exercise2.fastq\n245216 Exercise2.fastq\n[kcrid001@turing1 exercises]$ \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e 245216/4 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e bc\n61304\n[kcrid001@turing1 exercises]$ grep -c \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e@HS3:541\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e Exercise2.fastq\n61304\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e9 - cp the avg_cov_len_fasta_advbioinf.py from the /cm/shared/courses/dbarshis/21AdvGenomics/scripts directory into your class scripts directory\u003c/span\u003e\n[kcrid001@turing1 exercises]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ../../\n[kcrid001@turing1 katiecrider]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e scripts\n[kcrid001@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics//scripts/avg_cov_len_fasta_advbioinf.py ./ \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e10- start an interactive compute session and re-navigate to your exercises directory\u003c/span\u003e\nI had trouble doing this with Turing so I switched over to Wahab\n\n[kcrid001@turing1 katiecrider]$ \u003cspan class=\"pl-c1\"\u003eexit\u003c/span\u003e\n\u003cspan class=\"pl-c1\"\u003elogout\u003c/span\u003e\nConnection to turing.hpc.odu.edu closed.\n(base) Katies-MacBook-Air-4:\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e katiecrider$ ssh kcrid001@wahab.hpc.odu.edu\nwahab-01:\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/\nwahab-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e salloc\nsalloc: Pending job allocation 133738\nsalloc: job 133738 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 133738 has been allocated resources\nsalloc: Granted job allocation 133738\nThis session will be terminated \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e 7 days. If your application requires\na longer excution time, please use \u003cspan class=\"pl-c1\"\u003ecommand\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esalloc -t N-0\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e where N is the\nnumber of days that you need.\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e11- run the avg_cov_len_fasta_DJB.py script on your Exercise2.fasta file by typing the path to the script followed by the Exercise2.fasta file name\u003c/span\u003e\ne2-w6420b-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e data/exercises/\ne2-w6420b-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/exercises\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e ../../scripts/avg_cov_len_fasta_advbioinf.py Exercise2.fasta\nThe total number of sequences is 138\nThe average sequence length is 126640\nThe total number of bases is 17476447\nThe minimum sequence length is 1122\nThe maximum sequence length is 562680\nThe N50 is 187037\nMedian Length = 92612\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 150bp = 0\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 500bp = 138\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 1000bp = 138\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 2000bp = 135\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e12- did you add all these entries to your notebook file, including the results of the script?\u003c/span\u003e\nYes\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e13- push your notebook file to your github page\u003c/span\u003e\n(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ git add README.md\n(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ git commit -m \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eupdating notebook with day02 exercises\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\n(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ git push -u origin main\n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 03 Exercises/Day 2 HW: 1/27/21\u003c/h2\u003e\u003ca id=\"user-content-day-03-exercisesday-2-hw-12721\" class=\"anchor\" aria-label=\"Permalink: Day 03 Exercises/Day 2 HW: 1/27/21\" href=\"#day-03-exercisesday-2-hw-12721\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#exercises day03 /Day02HW:\n\n#1 Write an sbatch script to cp the files /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/ into your own data directory\n\n(base) Katies-MacBook-Air-4:21sp_advgenomics katiecrider$ nano KCridCopyLane04.sh\n(base) Katies-MacBook-Air-4:21sp_advgenomics katiecrider$ scp KCridCopyLane04.sh kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/KCridCopyLane04.sh\nkcrid001@turing.hpc.odu.edu's password: \nKCridCopyLane04.sh                                              100%  325    10.6KB/s   00:00    \n\n[kcrid001@coreV2-25-072 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@coreV2-25-072 data]$ ls\nexercises  KCridCopyLane04.sh\n\n#2 Add the content of your sbatch script to your logfile\n\n[kcrid001@coreV2-25-072 data]$ cat KCridCopyLane04.sh \n#!/bin/bash -l\n\n#SBATCH -o KCridCopyLane04.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KCridCopyLane04\n\n\ncp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB04-* /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n\n\n#3 submit the slurm script (sbatch scripname.sh) and verify that it's working (by squeue -u yourusername multiple times and checking the destination directory to make sure the files are being created)\n\n[kcrid001@coreV2-25-072 data]$ sbatch KCridCopyLane04.sh \nSubmitted batch job 9270446\n[kcrid001@coreV2-25-072 data]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9270424      main       sh kcrid001  R      46:39      1 coreV2-25-072 \n           9270446      main KCridCop kcrid001  R      10:45      1 coreV2-25-072 \n#4 Make sure this is all documented on your github notebook\n  #Done\n\n#5 Write a sbatch script to gunzip all the fastq.gz files\n\n[kcrid001@coreV2-25-072 data]$ nano KCridGunzipLane04.sh\n[kcrid001@coreV2-25-072 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@coreV2-25-072 data]$ cat KCridGunzipLane04.sh\n#!/bin/bash -l\n\n#SBATCH -o KCridGunzipLane04.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KCridGunzipLane04\n\ngunzip *.fastq.gz\n\n[kcrid001@coreV2-25-072 data]$ sbatch KCridGunzipLane04.sh \nSubmitted batch job 9270468\n[kcrid001@coreV2-25-072 data]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9270424      main       sh kcrid001  R      57:21      1 coreV2-25-072 \n           9270468      main KCridGun kcrid001  R       0:34      1 coreV2-25-072 \n[kcrid001@coreV2-25-072 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@coreV2-25-072 data]$ ls\nexercises                          HADB04-J_S74_L005_R1_001.fastq.gz\nHADB04-A_S65_L005_R1_001.fastq     HADB04-K_S75_L005_R1_001.fastq.gz\nHADB04-A_S65_L005_R1_001.fastq.gz  HADB04-L_S76_L005_R1_001.fastq.gz\nHADB04-B_S66_L005_R1_001.fastq.gz  HADB04-M_S77_L005_R1_001.fastq.gz\nHADB04-C_S67_L005_R1_001.fastq.gz  HADB04-N_S78_L005_R1_001.fastq.gz\nHADB04-D_S68_L005_R1_001.fastq.gz  HADB04-O_S79_L005_R1_001.fastq.gz\nHADB04-E_S69_L005_R1_001.fastq.gz  HADB04-P_S80_L005_R1_001.fastq.gz\nHADB04-F_S70_L005_R1_001.fastq.gz  KCridCopyLane04.sh\nHADB04-G_S71_L005_R1_001.fastq.gz  KCridCopyLane04.txt\nHADB04-H_S72_L005_R1_001.fastq.gz  KCridGunzipLane04.sh\nHADB04-I_S73_L005_R1_001.fastq.gz  KCridGunzipLane04.txt\n[kcrid001@coreV2-25-072 data]$ ls\nexercises                       HADB04-G_S71_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq\nHADB04-A_S65_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq\nHADB04-B_S66_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq\nHADB04-C_S67_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  KCridCopyLane04.sh\nHADB04-D_S68_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  KCridCopyLane04.txt\nHADB04-E_S69_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq  KCridGunzipLane04.sh\nHADB04-F_S70_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq  KCridGunzipLane04.txt\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eexercises day03 /Day02HW:\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1 Write an sbatch script to cp the files /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/ into your own data directory\u003c/span\u003e\n\n(base) Katies-MacBook-Air-4:21sp_advgenomics katiecrider$ nano KCridCopyLane04.sh\n(base) Katies-MacBook-Air-4:21sp_advgenomics katiecrider$ scp KCridCopyLane04.sh kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/KCridCopyLane04.sh\nkcrid001@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password: \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eKCridCopyLane04.sh                                              100%  325    10.6KB/s   00:00    \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV2-25-072 data]$ pwd\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV2-25-072 data]$ ls\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eexercises  KCridCopyLane04.sh\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#2 Add the content of your sbatch script to your logfile\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV2-25-072 data]$ cat KCridCopyLane04.sh \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#!/bin/bash -l\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#SBATCH -o KCridCopyLane04.txt\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#SBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#SBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#SBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#SBATCH --job-name=KCridCopyLane04\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ecp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB04-* /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#3 submit the slurm script (sbatch scripname.sh) and verify that it\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003es working (by squeue -u yourusername multiple \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e and checking the destination directory to make sure the files are being created)\n\n[kcrid001@coreV2-25-072 data]$ sbatch KCridCopyLane04.sh \nSubmitted batch job 9270446\n[kcrid001@coreV2-25-072 data]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9270424      main       sh kcrid001  R      46:39      1 coreV2-25-072 \n           9270446      main KCridCop kcrid001  R      10:45      1 coreV2-25-072 \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4 Make sure this is all documented on your github notebook\u003c/span\u003e\n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eDone\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e5 Write a sbatch script to gunzip all the fastq.gz files\u003c/span\u003e\n\n[kcrid001@coreV2-25-072 data]$ nano KCridGunzipLane04.sh\n[kcrid001@coreV2-25-072 data]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@coreV2-25-072 data]$ cat KCridGunzipLane04.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KCridGunzipLane04.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KCridGunzipLane04\u003c/span\u003e\n\ngunzip \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq.gz\n\n[kcrid001@coreV2-25-072 data]$ sbatch KCridGunzipLane04.sh \nSubmitted batch job 9270468\n[kcrid001@coreV2-25-072 data]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9270424      main       sh kcrid001  R      57:21      1 coreV2-25-072 \n           9270468      main KCridGun kcrid001  R       0:34      1 coreV2-25-072 \n[kcrid001@coreV2-25-072 data]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@coreV2-25-072 data]$ ls\nexercises                          HADB04-J_S74_L005_R1_001.fastq.gz\nHADB04-A_S65_L005_R1_001.fastq     HADB04-K_S75_L005_R1_001.fastq.gz\nHADB04-A_S65_L005_R1_001.fastq.gz  HADB04-L_S76_L005_R1_001.fastq.gz\nHADB04-B_S66_L005_R1_001.fastq.gz  HADB04-M_S77_L005_R1_001.fastq.gz\nHADB04-C_S67_L005_R1_001.fastq.gz  HADB04-N_S78_L005_R1_001.fastq.gz\nHADB04-D_S68_L005_R1_001.fastq.gz  HADB04-O_S79_L005_R1_001.fastq.gz\nHADB04-E_S69_L005_R1_001.fastq.gz  HADB04-P_S80_L005_R1_001.fastq.gz\nHADB04-F_S70_L005_R1_001.fastq.gz  KCridCopyLane04.sh\nHADB04-G_S71_L005_R1_001.fastq.gz  KCridCopyLane04.txt\nHADB04-H_S72_L005_R1_001.fastq.gz  KCridGunzipLane04.sh\nHADB04-I_S73_L005_R1_001.fastq.gz  KCridGunzipLane04.txt\n[kcrid001@coreV2-25-072 data]$ ls\nexercises                       HADB04-G_S71_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq\nHADB04-A_S65_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq\nHADB04-B_S66_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq\nHADB04-C_S67_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  KCridCopyLane04.sh\nHADB04-D_S68_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  KCridCopyLane04.txt\nHADB04-E_S69_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq  KCridGunzipLane04.sh\nHADB04-F_S70_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq  KCridGunzipLane04.txt\n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 03 HW: 1/28/21\u003c/h2\u003e\u003ca id=\"user-content-day-03-hw-12821\" class=\"anchor\" aria-label=\"Permalink: Day 03 HW: 1/28/21\" href=\"#day-03-hw-12821\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#HW day03:\n\n#1. cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 directory (and files) to your sandbox\n\n[kcrid001@turing1 katiecrider]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider\n[kcrid001@turing1 katiecrider]$ cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/ ./\n[kcrid001@turing1 katiecrider]$ ls\ndata  day03  groundrules.txt  kcrid_exercise1.txt  scripts\n\n\n#2. mkdir a fastq directory in your data directory and mv all the .fastq files into this directory\n\n[kcrid001@turing1 katiecrider]$ cd data\n[kcrid001@turing1 data]$ ls\nexercises                       HADB04-G_S71_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq\nHADB04-A_S65_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq\nHADB04-B_S66_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq\nHADB04-C_S67_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  KCridCopyLane04.sh\nHADB04-D_S68_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  KCridCopyLane04.txt\nHADB04-E_S69_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq  KCridGunzipLane04.sh\nHADB04-F_S70_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq  KCridGunzipLane04.txt\n[kcrid001@turing1 data]$ mkdir fastq\n[kcrid001@turing1 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@turing1 data]$ mv *.fastq ./fastq/\n[kcrid001@turing1 data]$ ls\nexercises  fastq  KCridCopyLane04.sh  KCridCopyLane04.txt  KCridGunzipLane04.sh  KCridGunzipLane04.txt\n[kcrid001@turing1 data]$ cd fastq\n[kcrid001@turing1 fastq]$ ls\nHADB04-A_S65_L005_R1_001.fastq  HADB04-G_S71_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq\nHADB04-B_S66_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq\nHADB04-C_S67_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq\nHADB04-D_S68_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq\nHADB04-E_S69_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq\nHADB04-F_S70_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq\n\n\n#3. cp the renamingtable_complete.txt from the day03 directory into your fastq directory and the /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your sandbox scripts folder and less the new script and check out the usage statement\n\n[kcrid001@turing1 fastq]$ cd ../\n[kcrid001@turing1 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@turing1 data]$ cp ../day03/renamingtable_complete.txt ./fastq/\n[kcrid001@turing1 data]$ cd fastq\n[kcrid001@turing1 fastq]$ ls\nHADB04-A_S65_L005_R1_001.fastq  HADB04-G_S71_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq\nHADB04-B_S66_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq\nHADB04-C_S67_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq\nHADB04-D_S68_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq\nHADB04-E_S69_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  renamingtable_complete.txt\nHADB04-F_S70_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fasq\n\n\n[kcrid001@turing1 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/scripts\n[kcrid001@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py ./\n[kcrid001@turing1 scripts]$ ls\navg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py\n\n[kcrid001@turing1 scripts]$ head renamer_advbioinf.py \n#!/usr/bin/env python\n####usage renamer.py renamingtable\n#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column\nimport sys\nimport os\n\nfin=open(sys.argv[1],'r')\nlinecount=0\nfor line in fin:\n\tlinecount+=1\n\n\n#4. run the renamer_advbioinf.py script in your fastq folder using the renamingtable_complete.txt as practice and verify the output to the screen by hand\n\n[kcrid001@turing1 scripts]$ cd ../data/fastq/\n[kcrid001@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq\n[kcrid001@turing1 fastq]$ ../../scripts/renamer_advbioinf.py renamingtable_complete.txt\nmv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq\nmv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq\nmv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq\nmv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq\nmv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq\nmv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq\nmv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq\nmv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq\nmv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq\nmv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq\nmv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq\nmv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq\nmv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq\nmv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq\nmv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq\nmv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq\nmv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq\nmv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq\nmv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq\nmv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq\nmv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq\nmv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq\nmv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq\nmv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq\nmv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq\nmv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq\nmv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq\nmv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq\nmv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq\nmv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq\nmv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq\nmv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq\nmv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq\nmv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq\nmv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq\nmv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq\nmv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq\nmv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq\nmv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq\nmv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq\nmv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq\nmv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq\nmv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq\nmv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq\nmv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq\nmv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq\nmv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq\nmv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq\nmv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq\nmv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq\nmv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq\nmv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq\nmv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq\nmv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq\nmv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq\nmv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq\nmv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq\nmv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq\nmv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq\nmv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq\nmv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq\nmv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq\nmv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq\nmv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq\nmv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq\nmv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq\nmv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq\nmv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq\nmv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq\nmv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq\nmv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq\nmv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq\nmv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq\nmv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq\nmv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq\nmv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq\nmv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq\nmv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq\nmv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq\nmv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq\nmv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq\nmv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq\nmv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq\nmv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq\nmv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq\nmv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq\nmv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq\nmv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq\nmv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq\nmv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq\nmv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq\nmv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq\nmv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq\nmv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq\nmv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq\nmv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq\n\n\n\n\n#5. Uncomment the last line of the renaming script in your scripts folder that starts with os.popen and comment out the next to last line that starts with print\n\n[kcrid001@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq\n[kcrid001@turing1 fastq]$ nano ../../scripts/renamer_advbioinf.py\n[kcrid001@turing1 fastq]$ cat ../../scripts/renamer_advbioinf.py \n#!/usr/bin/env python\n####usage renamer.py renamingtable\n#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column\nimport sys\nimport os\n\nfin=open(sys.argv[1],'r')\nlinecount=0\nfor line in fin:\n\tlinecount+=1\n\tif linecount\u0026gt;=2:\n\t\tcols=line.rstrip().split('\\t')\n#\t\tprint 'mv %s %s' %(cols[0], cols[1])\n\t\tos.popen('mv %s %s' %(cols[0], cols[1]))\n\n\n#6. write a sbatch script and submit it to rename all the .fastq files according to the renaming table using your renamer_advbioinf.py script\n\n[kcrid001@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq\n[kcrid001@turing1 fastq]$ nano KCridRenamer.sh\n[kcrid001@turing1 fastq]$ ls\nHADB04-A_S65_L005_R1_001.fastq  HADB04-G_S71_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq\nHADB04-B_S66_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq\nHADB04-C_S67_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq\nHADB04-D_S68_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq\nHADB04-E_S69_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  KCridRenamer.sh\nHADB04-F_S70_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq  renamingtable_complete.txt\n[kcrid001@turing1 fastq]$ cat KCridRenamer.sh \n#!/bin/bash -l\n\n#SBATCH -o KCridRenamer.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=FastqRename\n\n../../scripts/renamer_advbioinf.py renamingtable_complete.txt\n\n[kcrid001@turing1 fastq]$ sbatch ./KCridRenamer.sh\nSubmitted batch job 9270863\n\n#7. Make sure this is all documented on your github page\n* Yep\n\n#8. The naming convention for the files is as follows:\n  * SOURCEPOPULATION_SYMBIOTICSTATE_GENOTYPE_TEMPERATURE.fastq\n  * There are 2 sources: Virginia and Rhode Island\n  * There are 2 symbiotic states: Brown and White\n  \n#9. Next, you're going to start the process of adapter clipping and quality trimming all the renamed .fastq files in batches, by lane\n\n#10. cp the script /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py into your scripts directory\n\n[kcrid001@turing1 fastq]$ cd ../\n[kcrid001@turing1 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@turing1 data]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py ../scripts/\n\n\n#11. Less/head the new script and check out the usage statement\n[kcrid001@turing1 data]$ head -14 ../scripts/Trimclipfilterstatsbatch_advbioinf.py\n#!/usr/bin/env python\n# Written by Dan Barshis\n\nimport sys, os\n\n########### Usage #############\n# Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles\n# This should be run from within the folder with all of your original .fastqstrim\n# Will quality trim multiple SINGLE-END fastq files\n# Things to customize for your particular platform:\n# qualoffset = 33 Quality score offset\n# -t option in qualitytrim (the lower threshold quality score for trimming)\n# -l option in qualitytrim and adapterclip (the length threshold for throwing out short reads)\n\n\n#12. cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt into the working directory with your fastq files\n\n[kcrid001@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq\n[kcrid001@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt ./\n[kcrid001@turing1 fastq]$ ls\nadapterlist_advbioinf.txt   RI_B_04_14.fastq  RI_W_04_14.fastq  VA_B_04_14.fastq  VA_W_04_14.fastq\nKCridRenamer.sh             RI_B_04_18.fastq  RI_W_04_18.fastq  VA_B_04_18.fastq  VA_W_04_18.fastq\nKCridRenamer.txt            RI_B_04_22.fastq  RI_W_04_22.fastq  VA_B_04_22.fastq  VA_W_04_22.fastq\nrenamingtable_complete.txt  RI_B_05_14.fastq  RI_W_05_14.fastq  VA_B_05_14.fastq  VA_W_05_14.fastq\n\n#13. Make a sbatch script for the Trimclipfilter... script and run it on your fastq files\n[kcrid001@turing1 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq\n[kcrid001@turing1 fastq]$ nano TestTrimClip.sh\n[kcrid001@turing1 fastq]$ cat TestTrimClip.sh\n#!/bin/bash -l\n\n#SBATCH -o KCridtestTrimclip.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KCridTrimTest\n\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq\n\n[kcrid001@turing1 fastq]$ salloc\nsalloc: Pending job allocation 9271050\nsalloc: job 9271050 queued and waiting for resources\nsalloc: job 9271050 has been allocated resources\nsalloc: Granted job allocation 9271050\n[kcrid001@coreV1-22-005 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq\n[kcrid001@coreV1-22-005 fastq]$ sbatch TestTrimClip.sh\nSubmitted batch job 9271051\n[kcrid001@coreV1-22-005 fastq]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9271051      main KCridTri kcrid001  R      10:22      1 coreV2-22-007 \n           9271050      main       sh kcrid001  R      11:22      1 coreV1-22-005 \n\n#I was confused on if I had to do the FullTrimClip next, because they look like the same thing, but I think the TestTrimClip is working (see list below)\n\n[kcrid001@coreV1-22-005 fastq]$ ls\nadapterlist_advbioinf.txt  renamingtable_complete.txt  RI_W_04_14.fastq  trimclipstats.txt  VA_W_04_18.fastq\nFullTrimClip.sh            RI_B_04_14_clipped.fastq    RI_W_04_18.fastq  VA_B_04_14.fastq   VA_W_04_22.fastq\nKCridFullTrimclip.txt      RI_B_04_14.fastq            RI_W_04_22.fastq  VA_B_04_18.fastq   VA_W_05_14.fastq\nKCridRenamer.sh            RI_B_04_18.fastq            RI_W_05_14.fastq  VA_B_04_22.fastq\nKCridRenamer.txt           RI_B_04_22.fastq            Testing           VA_B_05_14.fastq\nKCridtestTrimclip.txt      RI_B_05_14.fastq            TestTrimClip.sh   VA_W_04_14.fastq\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eHW day03:\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1. cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 directory (and files) to your sandbox\u003c/span\u003e\n\n[kcrid001@turing1 katiecrider]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider\n[kcrid001@turing1 katiecrider]$ cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/ ./\n[kcrid001@turing1 katiecrider]$ ls\ndata  day03  groundrules.txt  kcrid_exercise1.txt  scripts\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2. mkdir a fastq directory in your data directory and mv all the .fastq files into this directory\u003c/span\u003e\n\n[kcrid001@turing1 katiecrider]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e data\n[kcrid001@turing1 data]$ ls\nexercises                       HADB04-G_S71_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq\nHADB04-A_S65_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq\nHADB04-B_S66_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq\nHADB04-C_S67_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  KCridCopyLane04.sh\nHADB04-D_S68_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  KCridCopyLane04.txt\nHADB04-E_S69_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq  KCridGunzipLane04.sh\nHADB04-F_S70_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq  KCridGunzipLane04.txt\n[kcrid001@turing1 data]$ mkdir fastq\n[kcrid001@turing1 data]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@turing1 data]$ mv \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq ./fastq/\n[kcrid001@turing1 data]$ ls\nexercises  fastq  KCridCopyLane04.sh  KCridCopyLane04.txt  KCridGunzipLane04.sh  KCridGunzipLane04.txt\n[kcrid001@turing1 data]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e fastq\n[kcrid001@turing1 fastq]$ ls\nHADB04-A_S65_L005_R1_001.fastq  HADB04-G_S71_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq\nHADB04-B_S66_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq\nHADB04-C_S67_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq\nHADB04-D_S68_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq\nHADB04-E_S69_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq\nHADB04-F_S70_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3. cp the renamingtable_complete.txt from the day03 directory into your fastq directory and the /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your sandbox scripts folder and less the new script and check out the usage statement\u003c/span\u003e\n\n[kcrid001@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ../\n[kcrid001@turing1 data]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@turing1 data]$ cp ../day03/renamingtable_complete.txt ./fastq/\n[kcrid001@turing1 data]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e fastq\n[kcrid001@turing1 fastq]$ ls\nHADB04-A_S65_L005_R1_001.fastq  HADB04-G_S71_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq\nHADB04-B_S66_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq\nHADB04-C_S67_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq\nHADB04-D_S68_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq\nHADB04-E_S69_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  renamingtable_complete.txt\nHADB04-F_S70_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fasq\n\n\n[kcrid001@turing1 scripts]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/scripts\n[kcrid001@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py ./\n[kcrid001@turing1 scripts]$ ls\navg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py\n\n[kcrid001@turing1 scripts]$ head renamer_advbioinf.py \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/usr/bin/env python\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e###usage renamer.py renamingtable\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column\u003c/span\u003e\nimport sys\nimport os\n\nfin=open(sys.argv[1],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003er\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\nlinecount=0\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eline\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e fin:\n\tlinecount+=1\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4. run the renamer_advbioinf.py script in your fastq folder using the renamingtable_complete.txt as practice and verify the output to the screen by hand\u003c/span\u003e\n\n[kcrid001@turing1 scripts]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ../data/fastq/\n[kcrid001@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq\n[kcrid001@turing1 fastq]$ ../../scripts/renamer_advbioinf.py renamingtable_complete.txt\nmv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq\nmv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq\nmv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq\nmv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq\nmv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq\nmv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq\nmv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq\nmv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq\nmv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq\nmv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq\nmv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq\nmv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq\nmv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq\nmv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq\nmv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq\nmv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq\nmv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq\nmv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq\nmv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq\nmv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq\nmv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq\nmv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq\nmv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq\nmv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq\nmv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq\nmv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq\nmv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq\nmv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq\nmv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq\nmv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq\nmv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq\nmv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq\nmv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq\nmv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq\nmv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq\nmv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq\nmv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq\nmv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq\nmv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq\nmv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq\nmv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq\nmv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq\nmv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq\nmv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq\nmv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq\nmv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq\nmv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq\nmv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq\nmv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq\nmv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq\nmv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq\nmv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq\nmv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq\nmv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq\nmv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq\nmv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq\nmv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq\nmv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq\nmv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq\nmv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq\nmv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq\nmv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq\nmv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq\nmv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq\nmv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq\nmv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq\nmv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq\nmv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq\nmv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq\nmv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq\nmv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq\nmv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq\nmv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq\nmv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq\nmv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq\nmv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq\nmv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq\nmv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq\nmv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq\nmv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq\nmv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq\nmv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq\nmv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq\nmv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq\nmv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq\nmv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq\nmv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq\nmv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq\nmv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq\nmv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq\nmv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq\nmv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq\nmv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq\nmv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq\nmv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq\nmv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq\n\n\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e5. Uncomment the last line of the renaming script in your scripts folder that starts with os.popen and comment out the next to last line that starts with print\u003c/span\u003e\n\n[kcrid001@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq\n[kcrid001@turing1 fastq]$ nano ../../scripts/renamer_advbioinf.py\n[kcrid001@turing1 fastq]$ cat ../../scripts/renamer_advbioinf.py \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/usr/bin/env python\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e###usage renamer.py renamingtable\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column\u003c/span\u003e\nimport sys\nimport os\n\nfin=open(sys.argv[1],\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003er\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\nlinecount=0\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eline\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e fin:\n\tlinecount+=1\n\t\u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e linecount\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e=2:\n\t\tcols=line.rstrip().split(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\\t\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e\t\tprint 'mv %s %s' %(cols[0], cols[1])\u003c/span\u003e\n\t\tos.popen(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emv %s %s\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e %(cols[0], cols[1]))\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e6. write a sbatch script and submit it to rename all the .fastq files according to the renaming table using your renamer_advbioinf.py script\u003c/span\u003e\n\n[kcrid001@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq\n[kcrid001@turing1 fastq]$ nano KCridRenamer.sh\n[kcrid001@turing1 fastq]$ ls\nHADB04-A_S65_L005_R1_001.fastq  HADB04-G_S71_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq\nHADB04-B_S66_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq\nHADB04-C_S67_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq\nHADB04-D_S68_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq\nHADB04-E_S69_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  KCridRenamer.sh\nHADB04-F_S70_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq  renamingtable_complete.txt\n[kcrid001@turing1 fastq]$ cat KCridRenamer.sh \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KCridRenamer.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=FastqRename\u003c/span\u003e\n\n../../scripts/renamer_advbioinf.py renamingtable_complete.txt\n\n[kcrid001@turing1 fastq]$ sbatch ./KCridRenamer.sh\nSubmitted batch job 9270863\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e7. Make sure this is all documented on your github page\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e Yep\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e8. The naming convention for the files is as follows:\u003c/span\u003e\n  \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e SOURCEPOPULATION_SYMBIOTICSTATE_GENOTYPE_TEMPERATURE.fastq\n  \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e There are 2 sources: Virginia and Rhode Island\n  \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e There are 2 symbiotic states: Brown and White\n  \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e9. Next, you're going to start the process of adapter clipping and quality trimming all the renamed .fastq files in batches, by lane\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e10. cp the script /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py into your scripts directory\u003c/span\u003e\n\n[kcrid001@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ../\n[kcrid001@turing1 data]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@turing1 data]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py ../scripts/\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e11. Less/head the new script and check out the usage statement\u003c/span\u003e\n[kcrid001@turing1 data]$ head -14 ../scripts/Trimclipfilterstatsbatch_advbioinf.py\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/usr/bin/env python\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Written by Dan Barshis\u003c/span\u003e\n\nimport sys, os\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e########## Usage #############\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e This should be run from within the folder with all of your original .fastqstrim\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Will quality trim multiple SINGLE-END fastq files\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Things to customize for your particular platform:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e qualoffset = 33 Quality score offset\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e -t option in qualitytrim (the lower threshold quality score for trimming)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e -l option in qualitytrim and adapterclip (the length threshold for throwing out short reads)\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e12. cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt into the working directory with your fastq files\u003c/span\u003e\n\n[kcrid001@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq\n[kcrid001@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt ./\n[kcrid001@turing1 fastq]$ ls\nadapterlist_advbioinf.txt   RI_B_04_14.fastq  RI_W_04_14.fastq  VA_B_04_14.fastq  VA_W_04_14.fastq\nKCridRenamer.sh             RI_B_04_18.fastq  RI_W_04_18.fastq  VA_B_04_18.fastq  VA_W_04_18.fastq\nKCridRenamer.txt            RI_B_04_22.fastq  RI_W_04_22.fastq  VA_B_04_22.fastq  VA_W_04_22.fastq\nrenamingtable_complete.txt  RI_B_05_14.fastq  RI_W_05_14.fastq  VA_B_05_14.fastq  VA_W_05_14.fastq\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e13. Make a sbatch script for the Trimclipfilter... script and run it on your fastq files\u003c/span\u003e\n[kcrid001@turing1 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq\n[kcrid001@turing1 fastq]$ nano TestTrimClip.sh\n[kcrid001@turing1 fastq]$ cat TestTrimClip.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KCridtestTrimclip.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KCridTrimTest\u003c/span\u003e\n\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq\n\n[kcrid001@turing1 fastq]$ salloc\nsalloc: Pending job allocation 9271050\nsalloc: job 9271050 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9271050 has been allocated resources\nsalloc: Granted job allocation 9271050\n[kcrid001@coreV1-22-005 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq\n[kcrid001@coreV1-22-005 fastq]$ sbatch TestTrimClip.sh\nSubmitted batch job 9271051\n[kcrid001@coreV1-22-005 fastq]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9271051      main KCridTri kcrid001  R      10:22      1 coreV2-22-007 \n           9271050      main       sh kcrid001  R      11:22      1 coreV1-22-005 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eI was confused on if I had to do the FullTrimClip next, because they look like the same thing, but I think the TestTrimClip is working (see list below)\u003c/span\u003e\n\n[kcrid001@coreV1-22-005 fastq]$ ls\nadapterlist_advbioinf.txt  renamingtable_complete.txt  RI_W_04_14.fastq  trimclipstats.txt  VA_W_04_18.fastq\nFullTrimClip.sh            RI_B_04_14_clipped.fastq    RI_W_04_18.fastq  VA_B_04_14.fastq   VA_W_04_22.fastq\nKCridFullTrimclip.txt      RI_B_04_14.fastq            RI_W_04_22.fastq  VA_B_04_18.fastq   VA_W_05_14.fastq\nKCridRenamer.sh            RI_B_04_18.fastq            RI_W_05_14.fastq  VA_B_04_22.fastq\nKCridRenamer.txt           RI_B_04_22.fastq            Testing           VA_B_05_14.fastq\nKCridtestTrimclip.txt      RI_B_05_14.fastq            TestTrimClip.sh   VA_W_04_14.fastq\n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay 04 HW: 1/29/21\u003c/h2\u003e\u003ca id=\"user-content-day-04-hw-12921\" class=\"anchor\" aria-label=\"Permalink: Day 04 HW: 1/29/21\" href=\"#day-04-hw-12921\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#HW Day 04:\n\n#1. Add your trimclipstats.txt output to the full class datafile /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt using the following steps\n#1a. run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h to examine usage\n[kcrid001@coreV1-22-005 fastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq\n[kcrid001@coreV1-22-005 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h\nWritten by Peter Schafran pscha005@odu.edu 5-Oct-2015\n\nThis script takes a stats output file from fastx_clipper and converts it into a table.\n\nUsage: Schafran_trimstatstable.py [-c, -v, -h] inputfile.txt outputfile.txt\n\nOptions (-c and -v must be listed separately to run together):\n-h\tDisplay this help message\n-c\tUse comma delimiter instead of tabs\n-v\tVerbose mode (print steps to stdout)\n\n#1b. Run the script on your data with the outputfilename YOURNAME_trimclipstatsout.txt\n[kcrid001@coreV2-22-007 fastq]$ cd filteringstats\n[kcrid001@coreV2-22-007 filteringstats]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/filteringstats\n[kcrid001@coreV2-22-007 filteringstats]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt KCrid_trimclipstatsout.txt\n\n#1c. Add YOURNAME_trimclipstatsout.txt to the class file by running tail -n +2 YOURNAME_trimclipstatsout.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt\n\n[kcrid001@coreV2-25-047 filteringstats]$ tail -n +2 KCrid_trimclipstatsout.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt\n\n\n#2. Now we're going to map our reads back to our assembly using the bowtie2 alignment algorithm (starting to follow this pipeline https://github.com/bethsheets/SNPcalling_tutorial)\n#3. write a sbatch script to do the following commands in sequence on your _clippedtrimmedfilterd.fastq datafiles from your lane of data\n[kcrid001@coreV2-25-047 filteringstats]$ cd ../\n[kcrid001@coreV2-25-047 fastq]$ cd QCFastqs/\n[kcrid001@coreV2-25-047 QCFastqs]$ ls\nRI_B_04_14_clippedtrimmed.fastq  RI_W_04_22_clippedtrimmed.fastq  VA_W_04_14_clippedtrimmed.fastq\nRI_B_04_18_clippedtrimmed.fastq  RI_W_05_14_clippedtrimmed.fastq  VA_W_04_18_clippedtrimmed.fastq\nRI_B_04_22_clippedtrimmed.fastq  VA_B_04_14_clippedtrimmed.fastq  VA_W_04_22_clippedtrimmed.fastq\nRI_B_05_14_clippedtrimmed.fastq  VA_B_04_18_clippedtrimmed.fastq  VA_W_05_14_clippedtrimmed.fastq\nRI_W_04_14_clippedtrimmed.fastq  VA_B_04_22_clippedtrimmed.fastq\nRI_W_04_18_clippedtrimmed.fastq  VA_B_05_14_clippedtrimmed.fastq\n[kcrid001@coreV2-25-047 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs\n[kcrid001@coreV2-25-047 QCFastqs]$ nano bowtiealn.sh\n[kcrid001@coreV2-25-047 QCFastqs]$ cat bowtiealn.sh \n#!/bin/bash -l\n\n#SBATCH -o KCridbowtie2.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=newbowtie\n\nmodule load bowtie2/2.4\nfor i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \\\n--rg SM:${i%_clippedtrimmed.fastq} \\\n--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U $i \\\n\u0026gt; ${i%_clippedtrimmedfilterd.fastq}.sam --no-unal -k 5; done\n[kcrid001@coreV2-25-047 QCFastqs]$ sbatch bowtiealn.sh \nSubmitted batch job 9271852\n[kcrid001@coreV2-25-047 QCFastqs]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9271852      main newbowti kcrid001  R       0:01      1 coreV2-25-005 \n           9271848      main       sh kcrid001  R      18:01      1 coreV2-25-047 \n\n#4-Submit and add everything to your logfile\n*yep!\n\n## Day05 HW 2/3/21\n```sh\n##Day05HW\n\n1-Team up with a partner for this one and work only on a combined set of data for your two lanes of sequences\n\n[kcrid001@coreV3-23-050 fastq]$ mkdir KCKCfastq\n[kcrid001@coreV3-23-050 fastq]$ ls\nfilteringstats         KCridRenamer.txt            renamingtable_complete.txt\nFullTrimClip.sh        KCridtestTrimclip.txt       Testing\nKCKCfastq              KCrid_trimclipstatsout.txt  TestTrimClip.sh\nKCridFullTrimclip.txt  originalfastqs\nKCridRenamer.sh        QCFastqs\n[kcrid001@coreV3-23-050 fastq]$ cd KCKCfastq/\n[kcrid001@coreV3-23-050 KCKCfastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq\n\n[kcrid001@coreV3-23-050 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs\n[kcrid001@coreV3-23-050 QCFastqs]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs/*.fastq /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq\n# we both copied our clippedtrimmed.fastq files to the new KCKCfastq directory (Katie C \u0026amp; Kristina)\n[kcrid001@coreV3-23-050 KCKCfastq]$ ls\nKCKCTrinitydenovo.sh              VA_B_01_14_clippedtrimmed.fastq\nRI_B_01_14_clippedtrimmed.fastq   VA_B_01_18_clippedtrimmed.fastq\nRI_B_01_18_clippedtrimmed.fastq   VA_B_01_22_clippedtrimmed.fastq\nRI_B_01_22_clippedtrimmed.fastq   VA_B_04_14_clippedtrimmed.fastq\nRI_B_04_14_clippedtrimmed.fastq   VA_B_04_18_clippedtrimmed.fastq\nRI_B_04_18_clippedtrimmed.fastq   VA_B_04_22_clippedtrimmed.fastq\nRI_B_04_22_clippedtrimmed.fastq   VA_B_05_14_clippedtrimmed.fastq\nRI_B_05_14_clippedtrimmed.fastq   VA_B_09_SNP_clippedtrimmed.fastq\nRI_B_08_SNP_clippedtrimmed.fastq  VA_W_01_14_clippedtrimmed.fastq\nRI_W_01_14_clippedtrimmed.fastq   VA_W_01_18_clippedtrimmed.fastq\nRI_W_01_18_clippedtrimmed.fastq   VA_W_01_22_clippedtrimmed.fastq\nRI_W_01_22_clippedtrimmed.fastq   VA_W_04_14_clippedtrimmed.fastq\nRI_W_04_14_clippedtrimmed.fastq   VA_W_04_18_clippedtrimmed.fastq\nRI_W_04_18_clippedtrimmed.fastq   VA_W_04_22_clippedtrimmed.fastq\nRI_W_04_22_clippedtrimmed.fastq   VA_W_05_14_clippedtrimmed.fastq\nRI_W_05_14_clippedtrimmed.fastq   VA_W_08_SNP_clippedtrimmed.fastq\nRI_W_08_SNP_clippedtrimmed.fastq\n# we have 32 files, 16 for each of us. All files have copied\n\n2-Run the Trinity denovo assembler on your clippedtrimmed.fastq files for your two lanes together\n3-Modify the below sbatch script (note the differences in the header compared to the previous one)\n\n[kcrid001@coreV3-23-050 KCKCfastq]$ nano KCKCTrinitydenovo.sh\n[kcrid001@coreV3-23-050 KCKCfastq]$ cat KCKCTrinitydenovo.sh \n#!/bin/bash -l\n\n#SBATCH -o KCKCTrinitydenovo.txt\n#SBATCH -n 32\n#SBATCH -p himem\n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KCKCTrinitydenovo\n\nenable_lmod\nmodule load container_env trinity\n\ncrun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_04_14_clippedtrimmed.fastq,VA_B_04_14_clippedtrimmed.fastq,RI_B_04_18_clippedtrimmed.fastq,VA_B_04_18_clippedtrimmed.fastq,RI_B_04_22_clippedtrimmed.fastq,VA_B_04_22_clippedtrimmed.fastq,RI_B_05_14_clippedtrimmed.fastq,VA_B_05_14_clippedtrimmed.fastq,RI_W_04_14_clippedtrimmed.fastq,VA_W_04_14_clippedtrimmed.fastq,RI_W_04_18_clippedtrimmed.fastq,VA_W_04_18_clippedtrimmed.fastq,RI_W_04_22_clippedtrimmed.fastq,VA_W_04_22_clippedtrimmed.fastq,RI_W_05_14_clippedtrimmed.fastq,VA_W_05_14_clippedtrimmed.fastq,RI_B_01_14_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq --CPU 32\n\n4-Check https://trinityrnaseq.github.io/ for usage info\n\n5-Submit your trinity script\n[kcrid001@coreV3-23-050 KCKCfastq]$ sbatch KCKCTrinitydenovo.sh\nSubmitted batch job 9272362\n[kcrid001@coreV3-23-050 KCKCfastq]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eHW Day 04:\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1. Add your trimclipstats.txt output to the full class datafile /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt using the following steps\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1a. run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h to examine usage\u003c/span\u003e\n[kcrid001@coreV1-22-005 fastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq\n[kcrid001@coreV1-22-005 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h\nWritten by Peter Schafran pscha005@odu.edu 5-Oct-2015\n\nThis script takes a stats output file from fastx_clipper and converts it into a table.\n\nUsage: Schafran_trimstatstable.py [-c, -v, \u003cspan class=\"pl-k\"\u003e-h\u003c/span\u003e] inputfile.txt outputfile.txt\n\nOptions (-c and -v must be listed separately to run together):\n-h\tDisplay this \u003cspan class=\"pl-c1\"\u003ehelp\u003c/span\u003e message\n-c\tUse comma delimiter instead of tabs\n-v\tVerbose mode (print steps to stdout)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1b. Run the script on your data with the outputfilename YOURNAME_trimclipstatsout.txt\u003c/span\u003e\n[kcrid001@coreV2-22-007 fastq]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e filteringstats\n[kcrid001@coreV2-22-007 filteringstats]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/filteringstats\n[kcrid001@coreV2-22-007 filteringstats]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt KCrid_trimclipstatsout.txt\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1c. Add YOURNAME_trimclipstatsout.txt to the class file by running tail -n +2 YOURNAME_trimclipstatsout.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt\u003c/span\u003e\n\n[kcrid001@coreV2-25-047 filteringstats]$ tail -n +2 KCrid_trimclipstatsout.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2. Now we're going to map our reads back to our assembly using the bowtie2 alignment algorithm (starting to follow this pipeline https://github.com/bethsheets/SNPcalling_tutorial)\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3. write a sbatch script to do the following commands in sequence on your _clippedtrimmedfilterd.fastq datafiles from your lane of data\u003c/span\u003e\n[kcrid001@coreV2-25-047 filteringstats]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e ../\n[kcrid001@coreV2-25-047 fastq]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e QCFastqs/\n[kcrid001@coreV2-25-047 QCFastqs]$ ls\nRI_B_04_14_clippedtrimmed.fastq  RI_W_04_22_clippedtrimmed.fastq  VA_W_04_14_clippedtrimmed.fastq\nRI_B_04_18_clippedtrimmed.fastq  RI_W_05_14_clippedtrimmed.fastq  VA_W_04_18_clippedtrimmed.fastq\nRI_B_04_22_clippedtrimmed.fastq  VA_B_04_14_clippedtrimmed.fastq  VA_W_04_22_clippedtrimmed.fastq\nRI_B_05_14_clippedtrimmed.fastq  VA_B_04_18_clippedtrimmed.fastq  VA_W_05_14_clippedtrimmed.fastq\nRI_W_04_14_clippedtrimmed.fastq  VA_B_04_22_clippedtrimmed.fastq\nRI_W_04_18_clippedtrimmed.fastq  VA_B_05_14_clippedtrimmed.fastq\n[kcrid001@coreV2-25-047 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs\n[kcrid001@coreV2-25-047 QCFastqs]$ nano bowtiealn.sh\n[kcrid001@coreV2-25-047 QCFastqs]$ cat bowtiealn.sh \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KCridbowtie2.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=newbowtie\u003c/span\u003e\n\nmodule load bowtie2/2.4\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_clippedtrimmed.fastq\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e bowtie2 --rg-id \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_clippedtrimmed.fastq}\u003c/span\u003e \\\n--rg SM:\u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_clippedtrimmed.fastq}\u003c/span\u003e \\\n--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U \u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e \\\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_clippedtrimmedfilterd.fastq}\u003c/span\u003e.sam --no-unal -k 5\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n[kcrid001@coreV2-25-047 QCFastqs]$ sbatch bowtiealn.sh \nSubmitted batch job 9271852\n[kcrid001@coreV2-25-047 QCFastqs]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9271852      main newbowti kcrid001  R       0:01      1 coreV2-25-005 \n           9271848      main       sh kcrid001  R      18:01      1 coreV2-25-047 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4-Submit and add everything to your logfile\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eyep\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Day05 HW 2/3/21\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003esh\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Day05HW\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e1-Team up with a partner \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e this one and work only on a combined \u003cspan class=\"pl-c1\"\u003eset\u003c/span\u003e of data \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e your two lanes of sequences\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-050 fastq]$ mkdir KCKCfastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-050 fastq]$ ls\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003efilteringstats         KCridRenamer.txt            renamingtable_complete.txt\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eFullTrimClip.sh        KCridtestTrimclip.txt       Testing\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eKCKCfastq              KCrid_trimclipstatsout.txt  TestTrimClip.sh\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eKCridFullTrimclip.txt  originalfastqs\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eKCridRenamer.sh        QCFastqs\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-050 fastq]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e KCKCfastq/\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-050 KCKCfastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-050 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-050 QCFastqs]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs/\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.fastq /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e we both copied our clippedtrimmed.fastq files to the new KCKCfastq directory (Katie C \u0026amp; Kristina)\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-050 KCKCfastq]$ ls\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eKCKCTrinitydenovo.sh              VA_B_01_14_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_01_14_clippedtrimmed.fastq   VA_B_01_18_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_01_18_clippedtrimmed.fastq   VA_B_01_22_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_01_22_clippedtrimmed.fastq   VA_B_04_14_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_04_14_clippedtrimmed.fastq   VA_B_04_18_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_04_18_clippedtrimmed.fastq   VA_B_04_22_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_04_22_clippedtrimmed.fastq   VA_B_05_14_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_05_14_clippedtrimmed.fastq   VA_B_09_SNP_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_08_SNP_clippedtrimmed.fastq  VA_W_01_14_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_01_14_clippedtrimmed.fastq   VA_W_01_18_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_01_18_clippedtrimmed.fastq   VA_W_01_22_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_01_22_clippedtrimmed.fastq   VA_W_04_14_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_04_14_clippedtrimmed.fastq   VA_W_04_18_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_04_18_clippedtrimmed.fastq   VA_W_04_22_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_04_22_clippedtrimmed.fastq   VA_W_05_14_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_05_14_clippedtrimmed.fastq   VA_W_08_SNP_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_08_SNP_clippedtrimmed.fastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e we have 32 files, 16 for each of us. All files have copied\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e2-Run the Trinity denovo assembler on your clippedtrimmed.fastq files \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e your two lanes together\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e3-Modify the below sbatch script (note the differences \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e the header compared to the previous one)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-050 KCKCfastq]$ nano KCKCTrinitydenovo.sh\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-050 KCKCfastq]$ cat KCKCTrinitydenovo.sh \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KCKCTrinitydenovo.txt\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 32\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -p himem\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KCKCTrinitydenovo\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eenable_lmod\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emodule load container_env trinity\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ecrun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_04_14_clippedtrimmed.fastq,VA_B_04_14_clippedtrimmed.fastq,RI_B_04_18_clippedtrimmed.fastq,VA_B_04_18_clippedtrimmed.fastq,RI_B_04_22_clippedtrimmed.fastq,VA_B_04_22_clippedtrimmed.fastq,RI_B_05_14_clippedtrimmed.fastq,VA_B_05_14_clippedtrimmed.fastq,RI_W_04_14_clippedtrimmed.fastq,VA_W_04_14_clippedtrimmed.fastq,RI_W_04_18_clippedtrimmed.fastq,VA_W_04_18_clippedtrimmed.fastq,RI_W_04_22_clippedtrimmed.fastq,VA_W_04_22_clippedtrimmed.fastq,RI_W_05_14_clippedtrimmed.fastq,VA_W_05_14_clippedtrimmed.fastq,RI_B_01_14_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq --CPU 32\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e4-Check https://trinityrnaseq.github.io/ \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e usage info\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e5-Submit your trinity script\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-050 KCKCfastq]$ sbatch KCKCTrinitydenovo.sh\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eSubmitted batch job 9272362\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-050 KCKCfastq]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay06 HW\u003c/h2\u003e\u003ca id=\"user-content-day06-hw\" class=\"anchor\" aria-label=\"Permalink: Day06 HW\" href=\"#day06-hw\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#day06HW\n####Make sure to add all this to your logfile as you go, including the output from the various scripts/greps/etc.\n###Assembly quality assessment (following the first two steps in the recommended trinity assessment protocol from https://github.com/trinityrnaseq/trinityrnaseq/wiki/Transcriptome-Assembly-Quality-Assessment)\n\n\n#1- start an interactive session via salloc and run the /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl script on your Trinity.fasta output from your assembly\n\n[kcrid001@turing1 trinity_out_dir]$ salloc\nsalloc: Pending job allocation 9272857\nsalloc: job 9272857 queued and waiting for resources\nsalloc: job 9272857 has been allocated resources\nsalloc: Granted job allocation 9272857\n[kcrid001@coreV2-22-028 trinity_out_dir]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir\n\n[kcrid001@coreV2-22-028 trinity_out_dir]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta\n\n\n################################\n## Counts of transcripts, etc.\n################################\nTotal trinity 'genes':\t39219\nTotal trinity transcripts:\t41303\nPercent GC: 44.54\n\n########################################\nStats based on ALL transcript contigs:\n########################################\n\n\tContig N10: 1323\n\tContig N20: 783\n\tContig N30: 562\n\tContig N40: 440\n\tContig N50: 364\n\n\tMedian contig length: 277\n\tAverage contig: 371.08\n\tTotal assembled bases: 15326666\n\n\n#####################################################\n## Stats based on ONLY LONGEST ISOFORM per 'GENE':\n#####################################################\n\n\tContig N10: 1094\n\tContig N20: 682\n\tContig N30: 511\n\tContig N40: 410\n\tContig N50: 347\n\n\tMedian contig length: 274\n\tAverage contig: 355.69\n\tTotal assembled bases: 13949619\n\n\n#2- compare this with the output from avg_cov_len_fasta_advbioinf.py on our class reference assembly (/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta) and add both to your logfile\n\n[kcrid001@coreV2-22-028 trinity_out_dir]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py Trinity.fasta\nThe total number of sequences is 41303\nThe average sequence length is 371\nThe total number of bases is 15326666\nThe minimum sequence length is 184\nThe maximum sequence length is 10771\nThe N50 is 364\nMedian Length = 627\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 5936\ncontigs \u0026gt;= 1000bp = 1364\ncontigs \u0026gt;= 2000bp = 265\n\n\n#3- less or head your bowtie2 job output file to look at your alignment statistics and calculate the following from the information:\n\ta-the mean percent \u0026quot;overall alignment rate\u0026quot;\n\tb-the mean percent reads \u0026quot;aligned exactly 1 time\u0026quot;\n\tc-the mean number of reads \u0026quot;aligned exactly 1 time\u0026quot;\n\td-the mean percent reads \u0026quot;aligned \u0026gt;1 times\u0026quot;\n\t\n\thint use grep and paste into excel\n\n[kcrid001@coreV2-22-028 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs\n[kcrid001@coreV2-22-028 QCFastqs]$ less KCridbowtie2.txt \n    366410 (2.11%) aligned exactly 1 time\n    196159 (1.13%) aligned \u0026gt;1 times\n3.25% overall alignment rate\n28980402 reads; of these:\n  28980402 (100.00%) were unpaired; of these:\n    28428026 (98.09%) aligned 0 times\n    302622 (1.04%) aligned exactly 1 time\n    249754 (0.86%) aligned \u0026gt;1 times\n1.91% overall alignment rate\n22997803 reads; of these:\n  22997803 (100.00%) were unpaired; of these:\n    22542136 (98.02%) aligned 0 times\n    281735 (1.23%) aligned exactly 1 time\n    173932 (0.76%) aligned \u0026gt;1 times\n1.98% overall alignment rate\n18509735 reads; of these:\n  18509735 (100.00%) were unpaired; of these:\n    18027510 (97.39%) aligned 0 times\n    262310 (1.42%) aligned exactly 1 time\n    219915 (1.19%) aligned \u0026gt;1 times\n2.61% overall alignment rate\n15268908 reads; of these:\n  15268908 (100.00%) were unpaired; of these:\n    14979814 (98.11%) aligned 0 times\n    178865 (1.17%) aligned exactly 1 time\n    110229 (0.72%) aligned \u0026gt;1 times\n1.89% overall alignment rate\n\n\nkcrid001@coreV2-22-028 QCFastqs]$ grep \u0026quot;overall alignment rate\u0026quot; KCridbowtie2.txt | cut -d \u0026quot; \u0026quot; -f 1 | cut -d \u0026quot;%\u0026quot; -f 1\n2.48\n1.70\n1.45\n2.32\n6.81\n2.25\n2.12\n1.85\n2.00\n2.70\n2.36\n3.25\n1.91\n1.98\n2.61\n1.89\n[kcrid001@coreV2-22-028 QCFastqs]$ grep \u0026quot;aligned exactly 1 time\u0026quot; KCridbowtie2.txt | cut -d \u0026quot;(\u0026quot; -f 2 | cut -d \u0026quot;%\u0026quot; -f 1\n1.41\n1.12\n0.85\n1.66\n1.18\n1.26\n1.06\n1.10\n1.10\n1.51\n1.53\n2.11\n1.04\n1.23\n1.42\n1.17\n[kcrid001@coreV2-22-028 QCFastqs]$ grep \u0026quot;aligned exactly 1 time\u0026quot; KCridbowtie2.txt | cut -d \u0026quot; \u0026quot; -f 5\n370715\n569956\n82036\n677289\n38461\n303192\n11115\n409225\n189559\n281489\n300187\n366410\n302622\n281735\n262310\n178865\n[kcrid001@coreV2-22-028 QCFastqs]$ grep \u0026quot;aligned \u0026gt;1 times\u0026quot; KCridbowtie2.txt | cut -d \u0026quot;(\u0026quot; -f 2 | cut -d \u0026quot;%\u0026quot; -f 1\n1.07\n0.58\n0.60\n0.67\n5.63\n1.00\n1.06\n0.74\n0.90\n1.19\n0.83\n1.13\n0.86\n0.76\n1.19\n0.72\n\n# copy these values to excel, calculate averages as needed\n# answers\n\t# a - 2.48% - the mean percent \u0026quot;overall alignment rate\u0026quot;\n\t# b - 1.30% - the mean percent reads \u0026quot;aligned exactly 1 time\u0026quot;\n\t# c - 289072.88 - the mean number of reads \u0026quot;aligned exactly 1 time\u0026quot;\n\t# d - 1.18% - the mean percent reads \u0026quot;aligned \u0026gt;1 times\u0026quot;\n\n\n#4- add your statistics as single rows to the shared table /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt as tab-delimited text in the following order:\nLaneX_yourinitials\tb-the mean percent \u0026quot;overall alignment rate\u0026quot;\tc-the mean percent reads \u0026quot;aligned exactly 1 time\u0026quot;\td-the mean number of reads \u0026quot;aligned exactly 1 time\u0026quot;\te-the mean percent reads \u0026quot;aligned \u0026gt;1 times\u0026quot;\n\n[kcrid001@coreV2-22-028 QCFastqs]$ nano alignstats.txt\n[kcrid001@coreV2-22-028 QCFastqs]$ cat alignstats.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\n[kcrid001@coreV2-22-028 QCFastqs]$ cat /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\nLaneX\toverallalignmentrate\tpercsinglyaligned\tnumsinglyaligned\tpercmultiplyaligned\nLaneX_djb\t2.495714286\t1.251428571\t247656.7143\t1.247142857\nLaneX_djb\t2.495714286\t1.251428571\t247656.7143\t1.247142857\nLane1AP\t5.378125\t1.3725\t300361.8125\t4.005\nlane5_sg\t2.03%\t1.11%\t235455.75\t0.92%\nLaneX_jcw\t2.495714286\t1.251428571\t247656.7143\t1.247142857\nLane4_KCrid\t2.48\t1.30\t289072.88\t1.18\n\n#5- Data cleanup and archiving:\n*a-mv your Trinity.fasta file from your trinity_out_dir to a folder called testassembly in your data directory\n\t\n[kcrid001@turing1 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@turing1 data]$ mkdir testassembly\n[kcrid001@turing1 data]$ ls\nexercises  KCridCopyLane04.sh   KCridGunzipLane04.sh   testassembly\nfastq      KCridCopyLane04.txt  KCridGunzipLane04.txt\n[kcrid001@turing1 trinity_out_dir]$ salloc\nsalloc: Pending job allocation 9275008\nsalloc: job 9275008 queued and waiting for resources\nsalloc: job 9275008 has been allocated resources\nsalloc: Granted job allocation 9275008\n[kcrid001@coreV1-22-016 trinity_out_dir]$ mv /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir/Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly\n\n*b-set up a sbatch script to:\n\t\trm -r YOURtrinity_out_dir\n\t\trm -r YOURoriginalfastqs\n\t\trm -r YOURfilteringstats # as long as you've already appended your filteringstats output to the class table as part of homework_day04.txt\n\n[kcrid001@coreV1-22-016 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@coreV1-22-016 data]$ nano Cleanup.txt\n[kcrid001@coreV1-22-016 data]$ cat Cleanup.txt \n#!/bin/bash -l\n\n#SBATCH -o Cleanup.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=Cleanup\n\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/originalfastqs\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/filteringstats\n\n[kcrid001@coreV1-22-016 data]$ sbatch Cleanup.sh \nSubmitted batch job 9275062\n[kcrid001@coreV1-22-016 data]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9275062      main  Cleanup kcrid001  R       0:07      1 coreV1-22-016 \n           9275008      main       sh kcrid001  R      22:37      1 coreV1-22-016 \n\n#6- submit a blast against sprot from your testassembly folder using the following command\n\n[kcrid001@coreV1-22-016 testassembly]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly\n[kcrid001@coreV1-22-016 testassembly]$ ls\nTrinity.fasta\n[kcrid001@coreV1-22-016 testassembly]$ nano Blast.sh\n[kcrid001@coreV1-22-016 testassembly]$ cat Blast.sh \n#!/bin/bash -l\n\n#SBATCH -o Blast.txt\n#SBATCH -n 6         \n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=Blast\n\nenable_lmod\nmodule load container_env blast\nblastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \\\n        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6\n[kcrid001@coreV1-22-016 testassembly]$ sbatch Blast.sh \nSubmitted batch job 9275071\n[kcrid001@coreV1-22-016 testassembly]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9275071      main    Blast kcrid001  R       1:53      1 coreV3-23-027 \n           9275062      main  Cleanup kcrid001  R       6:23      1 coreV1-22-016 \n           9275008      main       sh kcrid001  R      28:53      1 coreV1-22-016 \n\n###Exploring our SAM files, check out http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#sam-output for bowtie2 specific output and http://bio-bwa.sourceforge.net/bwa.shtml#4 for general SAM output\n\n#7- head one of your .sam files to look at the header\n\n[kcrid001@coreV2-22-028 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs\n[kcrid001@coreV2-22-028 QCFastqs]$ head RI_B_04_14_clippedtrimmed.fastq.sam\n@HD\tVN:1.0\tSO:unsorted\n@SQ\tSN:TR78|c0_g1_i1_coral\tLN:1109\n@SQ\tSN:TR78|c0_g2_i1_coral\tLN:1109\n@SQ\tSN:TR79|c0_g1_i1_coral\tLN:610\n@SQ\tSN:TR79|c0_g2_i1_coral\tLN:1549\n@SQ\tSN:TR87|c0_g1_i1_coral\tLN:732\n@SQ\tSN:TR93|c0_g1_i1_coral\tLN:550\n@SQ\tSN:TR101|c0_g1_i1_coral\tLN:673\n@SQ\tSN:TR104|c0_g1_i1_coral\tLN:607\n@SQ\tSN:TR105|c0_g1_i1_coral\tLN:587\n\n\n#8- grep -v '@' your.sam | head to look at the sequence read lines, why does this work to exclude the header lines?\n\t#because the -v flag means to omit lines with the indicated symbol\n\n[kcrid001@coreV2-22-028 QCFastqs]$ grep -v '@' RI_B_04_14_clippedtrimmed.fastq.sam | head\nK00188:59:HMTFHBBXX:5:1101:6055:1648\t16\tTR13320|c0_g2_i1_coral\t2\t253M4I44M\t*\t0\t0\tTTGAATCATTCTTGTCTACTCCAAGGGGTCTACTTTCCTCAAGGTTTCCNJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJJJJJFFA#A\tAS:i:-30\tXN:i:0\tXM:i:3\tXO:i:1\tXG:i:4\tNM:i:7\tMD:Z:0A1C42A1\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:9597:1666\t0\tTR35685|c1_g2_i1_coral\t535\t2520M\t*\t0\t0\tGACAACCCCGTCTGTGGCGC\tAAFFFJJJJJJJJJJJJJJJ\tAS:i:-6\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:6A13\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:26281:1666\t16\tTR59703|c0_g1_i3_sym\t244\t1\t51M\t*\t0\t0\tCTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:26281:1666\t272\tTR59703|c0_g1_i1_sym\t277\t2551M\t*\t0\t0\tCTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:26281:1666\t272\tTR59703|c0_g1_i3_coral\t244\t2551M\t*\t0\t0\tCTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:26281:1666\t272\tTR59703|c0_g1_i1_coral\t277\t2551M\t*\t0\t0\tCTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:9557:1701\t0\tTR30871|c0_g1_i1_coral\t16\t253M3I45M\t*\t0\t0\tGTGATAATGATGTTGGTGATAATGATGATGATGGTGGTGGTTGTGGTGTTAAFFFJJJJJFFJJJJJFJFJJAFJJJJJJJJJJFJJAJJJJJFJJFJFFJ\tAS:i:-26\tXN:i:0\tXM:i:2\tXO:i:1\tXG:i:3\tNM:i:5\tMD:Z:6G5A35\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:28270:1701\t16\tTR5604|c0_g1_i1_coral\t515\t2551M\t*\t0\t0\tCTAAGGTTTGCTTTAAAGCACTTAGTTCAGAATCCTTTGTCTGTACAATTJJJJJJFJJJJJJJJJJJJJJJJJJJJF-JJJFJFJJJJJJJJJJJFFFAA\tAS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:28371:1701\t16\tTR59703|c0_g1_i3_sym\t244\t1\t51M\t*\t0\t0\tCTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTJJJ7JJJJJJFJJJJJJJJJJJJJJJFAJJJJJJJJAJJJJJAJJJFFFAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:28371:1701\t272\tTR59703|c0_g1_i1_sym\t277\t2551M\t*\t0\t0\tCTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTJJJ7JJJJJJFJJJJJJJJJJJJJJJFAJJJJJJJJAJJJJJAJJJFFFAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_04_14\n\n\n#9- in an interactive session run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py on 2-3 of your .sam files using * to select 2-3 at the same time.\n\n[kcrid001@coreV1-22-016 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs\n[kcrid001@coreV1-22-016 QCFastqs]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py RI_B_04_14_clippedtrimmed.fastq.sam RI_B_04_18_clippedtrimmed.fastq.sam RI_B_04_22_clippedtrimmed.fastq.sam\nRI_B_04_14_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nRI_B_04_18_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\nRI_B_04_22_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\n\tread reverse strand\n\tnot primary alignment\n256 :\n\tnot primary alignment\n16 :\n\tread reverse strand\n\n\n#10- we need to run the read sorting step required for SNP calling, so if you have time, set up and run the following script on your .sam files to finish before Wednesday:\n\n#!/bin/bash -l\n\n#SBATCH -o KCridBamandSort.txt\n#SBATCH -n 1         \n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KCridBamandSort\n\nenable_lmod\nmodule load samtools/1\nfor i in *.sam; do `samtools view -bS $i \u0026gt; ${i%.sam}_UNSORTED.bam`; done\n\nfor i in *UNSORTED.bam; do samtools sort $i \u0026gt; ${i%_UNSORTED.bam}.bam\nsamtools index ${i%_UNSORTED.bam}.bam\ndone\n\n[kcrid001@coreV1-22-016 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs\n[kcrid001@coreV1-22-016 QCFastqs]$ cat bamandsort.sh\n#!/bin/bash -l\n\n#SBATCH -o KCridBamandSort.txt\n#SBATCH -n 1         \n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KCridBamandSort\n\nenable_lmod\nmodule load samtools/1\nfor i in *.sam; do `samtools view -bS $i \u0026gt; ${i%.sam}_UNSORTED.bam`; done\n\nfor i in *UNSORTED.bam; do samtools sort $i \u0026gt; ${i%_UNSORTED.bam}.bam\nsamtools index ${i%_UNSORTED.bam}.bam\ndone\n\n[kcrid001@coreV1-22-016 QCFastqs]$ sbatch bamandsort.sh\nSubmitted batch job 9275211\n[kcrid001@coreV1-22-016 QCFastqs]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9275071      main    Blast kcrid001  R      16:38      1 coreV3-23-027 \n           9275062      main  Cleanup kcrid001  R      21:08      1 coreV1-22-016 \n           9275008      main       sh kcrid001  R      43:38      1 coreV1-22-016 \n           9275211      main KCridBam kcrid001  R       0:08      1 coreV2-22-007 \n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eday06HW\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e###Make sure to add all this to your logfile as you go, including the output from the various scripts/greps/etc.\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e##Assembly quality assessment (following the first two steps in the recommended trinity assessment protocol from https://github.com/trinityrnaseq/trinityrnaseq/wiki/Transcriptome-Assembly-Quality-Assessment)\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1- start an interactive session via salloc and run the /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl script on your Trinity.fasta output from your assembly\u003c/span\u003e\n\n[kcrid001@turing1 trinity_out_dir]$ salloc\nsalloc: Pending job allocation 9272857\nsalloc: job 9272857 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9272857 has been allocated resources\nsalloc: Granted job allocation 9272857\n[kcrid001@coreV2-22-028 trinity_out_dir]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir\n\n[kcrid001@coreV2-22-028 trinity_out_dir]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e###############################\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Counts of transcripts, etc.\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e###############################\u003c/span\u003e\nTotal trinity \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egenes\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e:\t39219\nTotal trinity transcripts:\t41303\nPercent GC: 44.54\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#######################################\u003c/span\u003e\nStats based on ALL transcript contigs:\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#######################################\u003c/span\u003e\n\n\tContig N10: 1323\n\tContig N20: 783\n\tContig N30: 562\n\tContig N40: 440\n\tContig N50: 364\n\n\tMedian contig length: 277\n\tAverage contig: 371.08\n\tTotal assembled bases: 15326666\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e####################################################\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# Stats based on ONLY LONGEST ISOFORM per 'GENE':\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e####################################################\u003c/span\u003e\n\n\tContig N10: 1094\n\tContig N20: 682\n\tContig N30: 511\n\tContig N40: 410\n\tContig N50: 347\n\n\tMedian contig length: 274\n\tAverage contig: 355.69\n\tTotal assembled bases: 13949619\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2- compare this with the output from avg_cov_len_fasta_advbioinf.py on our class reference assembly (/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta) and add both to your logfile\u003c/span\u003e\n\n[kcrid001@coreV2-22-028 trinity_out_dir]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py Trinity.fasta\nThe total number of sequences is 41303\nThe average sequence length is 371\nThe total number of bases is 15326666\nThe minimum sequence length is 184\nThe maximum sequence length is 10771\nThe N50 is 364\nMedian Length = 627\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 150bp = 0\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 500bp = 5936\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 1000bp = 1364\ncontigs \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e= 2000bp = 265\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3- less or head your bowtie2 job output file to look at your alignment statistics and calculate the following from the information:\u003c/span\u003e\n\ta-the mean percent \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoverall alignment rate\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\tb-the mean percent reads \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\tc-the mean number of reads \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\td-the mean percent reads \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned \u0026gt;1 times\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\t\n\thint use grep and paste into excel\n\n[kcrid001@coreV2-22-028 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs\n[kcrid001@coreV2-22-028 QCFastqs]$ less KCridbowtie2.txt \n    366410 (2.11%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    196159 (1.13%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n3.25% overall alignment rate\n28980402 reads\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e of these:\n  28980402 (100.00%) were unpaired\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e of these:\n    28428026 (98.09%) aligned 0 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    302622 (1.04%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    249754 (0.86%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n1.91% overall alignment rate\n22997803 reads\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e of these:\n  22997803 (100.00%) were unpaired\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e of these:\n    22542136 (98.02%) aligned 0 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    281735 (1.23%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    173932 (0.76%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n1.98% overall alignment rate\n18509735 reads\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e of these:\n  18509735 (100.00%) were unpaired\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e of these:\n    18027510 (97.39%) aligned 0 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    262310 (1.42%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    219915 (1.19%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n2.61% overall alignment rate\n15268908 reads\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e of these:\n  15268908 (100.00%) were unpaired\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e of these:\n    14979814 (98.11%) aligned 0 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n    178865 (1.17%) aligned exactly 1 \u003cspan class=\"pl-k\"\u003etime\u003c/span\u003e\n    110229 (0.72%) aligned \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e1 \u003cspan class=\"pl-c1\"\u003etimes\u003c/span\u003e\n1.89% overall alignment rate\n\n\nkcrid001@coreV2-22-028 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoverall alignment rate\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e KCridbowtie2.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 1 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e%\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 1\n2.48\n1.70\n1.45\n2.32\n6.81\n2.25\n2.12\n1.85\n2.00\n2.70\n2.36\n3.25\n1.91\n1.98\n2.61\n1.89\n[kcrid001@coreV2-22-028 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e KCridbowtie2.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e(\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 2 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e%\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 1\n1.41\n1.12\n0.85\n1.66\n1.18\n1.26\n1.06\n1.10\n1.10\n1.51\n1.53\n2.11\n1.04\n1.23\n1.42\n1.17\n[kcrid001@coreV2-22-028 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e KCridbowtie2.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 5\n370715\n569956\n82036\n677289\n38461\n303192\n11115\n409225\n189559\n281489\n300187\n366410\n302622\n281735\n262310\n178865\n[kcrid001@coreV2-22-028 QCFastqs]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned \u0026gt;1 times\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e KCridbowtie2.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e(\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 2 \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e cut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e%\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 1\n1.07\n0.58\n0.60\n0.67\n5.63\n1.00\n1.06\n0.74\n0.90\n1.19\n0.83\n1.13\n0.86\n0.76\n1.19\n0.72\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e copy these values to excel, calculate averages as needed\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e answers\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e a - 2.48% - the mean percent \"overall alignment rate\"\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e b - 1.30% - the mean percent reads \"aligned exactly 1 time\"\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e c - 289072.88 - the mean number of reads \"aligned exactly 1 time\"\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e d - 1.18% - the mean percent reads \"aligned \u0026gt;1 times\"\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4- add your statistics as single rows to the shared table /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt as tab-delimited text in the following order:\u003c/span\u003e\nLaneX_yourinitials\tb-the mean percent \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eoverall alignment rate\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\tc-the mean percent reads \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\td-the mean number of reads \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned exactly 1 time\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\te-the mean percent reads \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ealigned \u0026gt;1 times\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n[kcrid001@coreV2-22-028 QCFastqs]$ nano alignstats.txt\n[kcrid001@coreV2-22-028 QCFastqs]$ cat alignstats.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u0026gt;\u003c/span\u003e /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\n[kcrid001@coreV2-22-028 QCFastqs]$ cat /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\nLaneX\toverallalignmentrate\tpercsinglyaligned\tnumsinglyaligned\tpercmultiplyaligned\nLaneX_djb\t2.495714286\t1.251428571\t247656.7143\t1.247142857\nLaneX_djb\t2.495714286\t1.251428571\t247656.7143\t1.247142857\nLane1AP\t5.378125\t1.3725\t300361.8125\t4.005\nlane5_sg\t2.03%\t1.11%\t235455.75\t0.92%\nLaneX_jcw\t2.495714286\t1.251428571\t247656.7143\t1.247142857\nLane4_KCrid\t2.48\t1.30\t289072.88\t1.18\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e5- Data cleanup and archiving:\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003ea-mv your Trinity.fasta file from your trinity_out_dir to a folder called testassembly \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e your data directory\n\t\n[kcrid001@turing1 data]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@turing1 data]$ mkdir testassembly\n[kcrid001@turing1 data]$ ls\nexercises  KCridCopyLane04.sh   KCridGunzipLane04.sh   testassembly\nfastq      KCridCopyLane04.txt  KCridGunzipLane04.txt\n[kcrid001@turing1 trinity_out_dir]$ salloc\nsalloc: Pending job allocation 9275008\nsalloc: job 9275008 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9275008 has been allocated resources\nsalloc: Granted job allocation 9275008\n[kcrid001@coreV1-22-016 trinity_out_dir]$ mv /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir/Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly\n\n\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eb-set up a sbatch script to:\n\t\trm -r YOURtrinity_out_dir\n\t\trm -r YOURoriginalfastqs\n\t\trm -r YOURfilteringstats \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e as long as you've already appended your filteringstats output to the class table as part of homework_day04.txt\u003c/span\u003e\n\n[kcrid001@coreV1-22-016 data]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data\n[kcrid001@coreV1-22-016 data]$ nano Cleanup.txt\n[kcrid001@coreV1-22-016 data]$ cat Cleanup.txt \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o Cleanup.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=Cleanup\u003c/span\u003e\n\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/originalfastqs\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/filteringstats\n\n[kcrid001@coreV1-22-016 data]$ sbatch Cleanup.sh \nSubmitted batch job 9275062\n[kcrid001@coreV1-22-016 data]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9275062      main  Cleanup kcrid001  R       0:07      1 coreV1-22-016 \n           9275008      main       sh kcrid001  R      22:37      1 coreV1-22-016 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e6- submit a blast against sprot from your testassembly folder using the following command\u003c/span\u003e\n\n[kcrid001@coreV1-22-016 testassembly]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly\n[kcrid001@coreV1-22-016 testassembly]$ ls\nTrinity.fasta\n[kcrid001@coreV1-22-016 testassembly]$ nano Blast.sh\n[kcrid001@coreV1-22-016 testassembly]$ cat Blast.sh \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o Blast.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 6         \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=Blast\u003c/span\u003e\n\nenable_lmod\nmodule load container_env blast\nblastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \\\n        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6\n[kcrid001@coreV1-22-016 testassembly]$ sbatch Blast.sh \nSubmitted batch job 9275071\n[kcrid001@coreV1-22-016 testassembly]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9275071      main    Blast kcrid001  R       1:53      1 coreV3-23-027 \n           9275062      main  Cleanup kcrid001  R       6:23      1 coreV1-22-016 \n           9275008      main       sh kcrid001  R      28:53      1 coreV1-22-016 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e##Exploring our SAM files, check out http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#sam-output for bowtie2 specific output and http://bio-bwa.sourceforge.net/bwa.shtml#4 for general SAM output\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e7- head one of your .sam files to look at the header\u003c/span\u003e\n\n[kcrid001@coreV2-22-028 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs\n[kcrid001@coreV2-22-028 QCFastqs]$ head RI_B_04_14_clippedtrimmed.fastq.sam\n@HD\tVN:1.0\tSO:unsorted\n@SQ\tSN:TR78\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral\tLN:1109\n@SQ\tSN:TR78\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral\tLN:1109\n@SQ\tSN:TR79\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral\tLN:610\n@SQ\tSN:TR79\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral\tLN:1549\n@SQ\tSN:TR87\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral\tLN:732\n@SQ\tSN:TR93\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral\tLN:550\n@SQ\tSN:TR101\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral\tLN:673\n@SQ\tSN:TR104\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral\tLN:607\n@SQ\tSN:TR105\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral\tLN:587\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e8- grep -v '@' your.sam | head to look at the sequence read lines, why does this work to exclude the header lines?\u003c/span\u003e\n\t\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ebecause the -v flag means to omit lines with the indicated symbol\u003c/span\u003e\n\n[kcrid001@coreV2-22-028 QCFastqs]$ grep -v \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e@\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e RI_B_04_14_clippedtrimmed.fastq.sam \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e head\nK00188:59:HMTFHBBXX:5:1101:6055:1648\t16\tTR13320\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral\t2\t253M4I44M\t\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\t0\t0\tTTGAATCATTCTTGTCTACTCCAAGGGGTCTACTTTCCTCAAGGTTTCCNJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJJJJJFFA#A\tAS:i:-30\tXN:i:0\tXM:i:3\tXO:i:1\tXG:i:4\tNM:i:7\tMD:Z:0A1C42A1\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:9597:1666\t0\tTR35685\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g2_i1_coral\t535\t2520M\t\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\t0\t0\tGACAACCCCGTCTGTGGCGC\tAAFFFJJJJJJJJJJJJJJJ\tAS:i:-6\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:6A13\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:26281:1666\t16\tTR59703\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i3_sym\t244\t1\t51M\t\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\t0\t0\tCTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:26281:1666\t272\tTR59703\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_sym\t277\t2551M\t\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\t0\t0\tCTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:26281:1666\t272\tTR59703\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i3_coral\t244\t2551M\t\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\t0\t0\tCTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:26281:1666\t272\tTR59703\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral\t277\t2551M\t\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\t0\t0\tCTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:9557:1701\t0\tTR30871\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral\t16\t253M3I45M\t\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\t0\t0\tGTGATAATGATGTTGGTGATAATGATGATGATGGTGGTGGTTGTGGTGTTAAFFFJJJJJFFJJJJJFJFJJAFJJJJJJJJJJFJJAJJJJJFJJFJFFJ\tAS:i:-26\tXN:i:0\tXM:i:2\tXO:i:1\tXG:i:3\tNM:i:5\tMD:Z:6G5A35\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:28270:1701\t16\tTR5604\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral\t515\t2551M\t\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\t0\t0\tCTAAGGTTTGCTTTAAAGCACTTAGTTCAGAATCCTTTGTCTGTACAATTJJJJJJFJJJJJJJJJJJJJJJJJJJJF-JJJFJFJJJJJJJJJJJFFFAA\tAS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:28371:1701\t16\tTR59703\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i3_sym\t244\t1\t51M\t\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\t0\t0\tCTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTJJJ7JJJJJJFJJJJJJJJJJJJJJJFAJJJJJJJJAJJJJJAJJJFFFAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_04_14\nK00188:59:HMTFHBBXX:5:1101:28371:1701\t272\tTR59703\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_sym\t277\t2551M\t\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\t0\t0\tCTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTJJJ7JJJJJJFJJJJJJJJJJJJJJJFAJJJJJJJJAJJJJJAJJJFFFAA\tAS:i:0\tXS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:51\tYT:Z:UU\tRG:Z:RI_B_04_14\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e9- in an interactive session run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py on 2-3 of your .sam files using * to select 2-3 at the same time.\u003c/span\u003e\n\n[kcrid001@coreV1-22-016 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs\n[kcrid001@coreV1-22-016 QCFastqs]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py RI_B_04_14_clippedtrimmed.fastq.sam RI_B_04_18_clippedtrimmed.fastq.sam RI_B_04_22_clippedtrimmed.fastq.sam\nRI_B_04_14_clippedtrimmed.fastq.sam\n[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e272\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e256\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e]\n0 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n272 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\n\tnot primary alignment\n256 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\tnot primary alignment\n16 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\nRI_B_04_18_clippedtrimmed.fastq.sam\n[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e272\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e256\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e]\n0 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n272 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\n\tnot primary alignment\n256 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\tnot primary alignment\n16 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\nRI_B_04_22_clippedtrimmed.fastq.sam\n[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e0\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e272\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e256\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e16\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e]\n0 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n272 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\n\tnot primary alignment\n256 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\tnot primary alignment\n16 \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e\n\t\u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e reverse strand\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e10- we need to run the read sorting step required for SNP calling, so if you have time, set up and run the following script on your .sam files to finish before Wednesday:\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KCridBamandSort.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1         \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KCridBamandSort\u003c/span\u003e\n\nenable_lmod\nmodule load samtools/1\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.sam\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003esamtools view -bS \u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.sam}\u003c/span\u003e_UNSORTED.bam\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eUNSORTED.bam\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e samtools sort \u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_UNSORTED.bam}\u003c/span\u003e.bam\nsamtools index \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_UNSORTED.bam}\u003c/span\u003e.bam\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n[kcrid001@coreV1-22-016 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs\n[kcrid001@coreV1-22-016 QCFastqs]$ cat bamandsort.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KCridBamandSort.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1         \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KCridBamandSort\u003c/span\u003e\n\nenable_lmod\nmodule load samtools/1\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.sam\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003esamtools view -bS \u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.sam}\u003c/span\u003e_UNSORTED.bam\u003cspan class=\"pl-pds\"\u003e`\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ei\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eUNSORTED.bam\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edo\u003c/span\u003e samtools sort \u003cspan class=\"pl-smi\"\u003e$i\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_UNSORTED.bam}\u003c/span\u003e.bam\nsamtools index \u003cspan class=\"pl-smi\"\u003e${i\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e_UNSORTED.bam}\u003c/span\u003e.bam\n\u003cspan class=\"pl-k\"\u003edone\u003c/span\u003e\n\n[kcrid001@coreV1-22-016 QCFastqs]$ sbatch bamandsort.sh\nSubmitted batch job 9275211\n[kcrid001@coreV1-22-016 QCFastqs]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9275071      main    Blast kcrid001  R      16:38      1 coreV3-23-027 \n           9275062      main  Cleanup kcrid001  R      21:08      1 coreV1-22-016 \n           9275008      main       sh kcrid001  R      43:38      1 coreV1-22-016 \n           9275211      main KCridBam kcrid001  R       0:08      1 coreV2-22-007 \n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay07 HW\u003c/h2\u003e\u003ca id=\"user-content-day07-hw\" class=\"anchor\" aria-label=\"Permalink: Day07 HW\" href=\"#day07-hw\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#day07HW\n\n#1- Run the following command on your sprot output file to process into the contig length/match format that trinity examines\n\n[kcrid001@turing1 testassembly]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly\n[kcrid001@turing1 testassembly]$ ls\nBlast.sh  Blast.txt  blastx.outfmt6  Trinity.fasta\n[kcrid001@turing1 testassembly]$ nano blastparsing.sh\n[kcrid001@turing1 testassembly]$ cat blastparsing.sh \n#!/bin/bash -l\n\n#SBATCH -o KCridBlastParse.TXT\n#SBATCH -n 1\n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KCridBlastParse\n\n/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta\n[kcrid001@turing1 testassembly]$ salloc\nsalloc: Pending job allocation 9276475\nsalloc: job 9276475 queued and waiting for resources\nsalloc: job 9276475 has been allocated resources\nsalloc: Granted job allocation 9276475\n[kcrid001@coreV3-23-040 testassembly]$ sbatch blastparsing.sh \nSubmitted batch job 9276476\n[kcrid001@coreV3-23-040 testassembly]$ ls\nblastparsing.sh  blastx.outfmt6            blastx.outfmt6.w_pct_hit_length\nBlast.sh         blastx.outfmt6.hist       KCridBlastParse.TXT\nBlast.txt        blastx.outfmt6.hist.list  Trinity.fasta\n[kcrid001@coreV3-23-040 testassembly]$ cat blastx.outfmt6.hist\n#hit_pct_cov_bin\tcount_in_bin\t\u0026gt;bin_below\n100\t212\t212\n90\t98\t310\n80\t104\t414\n70\t137\t551\n60\t182\t733\n50\t256\t989\n40\t413\t1402\n30\t719\t2121\n20\t1440\t3561\n10\t984\t4545\n\n#2- Rm UNSORTED.bam from your QCFastqs directory (or wherever your .bams and .sams are)\n\n[kcrid001@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs\n[kcrid001@coreV3-23-040 QCFastqs]$ nano KCridrmunsortbam.sh\n[kcrid001@coreV3-23-040 QCFastqs]$ cat KCridrmunsortbam.sh\n\n#!/bin/bash -l\n\n#SBATCH -o KCridrmunsortbam.txt\n#SBATCH -n 1         \n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KCridrmunsortbam\n\nrm *UNSORTED.bam\n\n[kcrid001@coreV3-23-040 QCFastqs]$ sbatch KCridrmunsortbam.sh \nSubmitted batch job 9276495\n\n\n#3- Run the following to start genotyping your SNPs for filtering next class\n\n[kcrid001@coreV3-23-040 QCFastqs]$ nano KCridfreebayessubref.sh\n[kcrid001@coreV3-23-040 QCFastqs]$ cat KCridfreebayessubref.sh\n#!/bin/bash -l\n\n#SBATCH -o KCridfreebayessubref.txt\n#SBATCH -n 1         \n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KCridfreebayessubref\n\nenable_lmod\nmodule load dDocent\nfreebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.bam \u0026gt; KCridEmergedfastqs.vcf\n[kcrid001@coreV3-23-040 QCFastqs]$ sbatch KCridfreebayessubref.sh \nSubmitted batch job 9276573\n[kcrid001@coreV3-23-040 QCFastqs]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9276475      main       sh kcrid001  R    1:42:22      1 coreV3-23-040 \n           9276573      main KCridfre kcrid001  R       1:22      1 coreV3-23-046 \"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eday07HW\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1- Run the following command on your sprot output file to process into the contig length/match format that trinity examines\u003c/span\u003e\n\n[kcrid001@turing1 testassembly]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly\n[kcrid001@turing1 testassembly]$ ls\nBlast.sh  Blast.txt  blastx.outfmt6  Trinity.fasta\n[kcrid001@turing1 testassembly]$ nano blastparsing.sh\n[kcrid001@turing1 testassembly]$ cat blastparsing.sh \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KCridBlastParse.TXT\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KCridBlastParse\u003c/span\u003e\n\n/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta\n[kcrid001@turing1 testassembly]$ salloc\nsalloc: Pending job allocation 9276475\nsalloc: job 9276475 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9276475 has been allocated resources\nsalloc: Granted job allocation 9276475\n[kcrid001@coreV3-23-040 testassembly]$ sbatch blastparsing.sh \nSubmitted batch job 9276476\n[kcrid001@coreV3-23-040 testassembly]$ ls\nblastparsing.sh  blastx.outfmt6            blastx.outfmt6.w_pct_hit_length\nBlast.sh         blastx.outfmt6.hist       KCridBlastParse.TXT\nBlast.txt        blastx.outfmt6.hist.list  Trinity.fasta\n[kcrid001@coreV3-23-040 testassembly]$ cat blastx.outfmt6.hist\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ehit_pct_cov_bin\tcount_in_bin\t\u0026gt;bin_below\u003c/span\u003e\n100\t212\t212\n90\t98\t310\n80\t104\t414\n70\t137\t551\n60\t182\t733\n50\t256\t989\n40\t413\t1402\n30\t719\t2121\n20\t1440\t3561\n10\t984\t4545\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2- Rm UNSORTED.bam from your QCFastqs directory (or wherever your .bams and .sams are)\u003c/span\u003e\n\n[kcrid001@turing1 QCFastqs]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs\n[kcrid001@coreV3-23-040 QCFastqs]$ nano KCridrmunsortbam.sh\n[kcrid001@coreV3-23-040 QCFastqs]$ cat KCridrmunsortbam.sh\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KCridrmunsortbam.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1         \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KCridrmunsortbam\u003c/span\u003e\n\nrm \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003eUNSORTED.bam\n\n[kcrid001@coreV3-23-040 QCFastqs]$ sbatch KCridrmunsortbam.sh \nSubmitted batch job 9276495\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3- Run the following to start genotyping your SNPs for filtering next class\u003c/span\u003e\n\n[kcrid001@coreV3-23-040 QCFastqs]$ nano KCridfreebayessubref.sh\n[kcrid001@coreV3-23-040 QCFastqs]$ cat KCridfreebayessubref.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KCridfreebayessubref.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1         \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KCridfreebayessubref\u003c/span\u003e\n\nenable_lmod\nmodule load dDocent\nfreebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.bam \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e KCridEmergedfastqs.vcf\n[kcrid001@coreV3-23-040 QCFastqs]$ sbatch KCridfreebayessubref.sh \nSubmitted batch job 9276573\n[kcrid001@coreV3-23-040 QCFastqs]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9276475      main       sh kcrid001  R    1:42:22      1 coreV3-23-040 \n           9276573      main KCridfre kcrid001  R       1:22      1 coreV3-23-046 \u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay08 HW\u003c/h2\u003e\u003ca id=\"user-content-day08-hw\" class=\"anchor\" aria-label=\"Permalink: Day08 HW\" href=\"#day08-hw\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#day08HW\n\n#1- Clean up your data directory by:\n\t-Making a SAMS folder and mv all your .sam files into that directory\n\t-Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory\n\t-rm your _unfiltered.vcf files if you have any\n\t-rm your .fastq files\n\t-Make a VCF folder in your data directory and mv your YOURNAMEmergedfastqs.vcf into this directory (if your freebayes job didn't complete then skip this step)\n\n#2- Start an interactive session via salloc\n\n[kcrid001@turing1 QCFastqs]$ cd ../../\n[kcrid001@turing1 data]$ mkdir SAMS\n[kcrid001@turing1 QCFastqs]$ mv *.sam ../../SAMS/\n[kcrid001@turing1 QCFastqs]$ salloc\nsalloc: Pending job allocation 9278244\nsalloc: job 9278244 queued and waiting for resources\nsalloc: job 9278244 has been allocated resources\nsalloc: Granted job allocation 9278244\n[kcrid001@coreV3-23-040 QCFastqs]$ mkdir ../../BAMS\n[kcrid001@coreV3-23-040 QCFastqs]$ mv *.bam *.bam.bai ../../BAMS/\n\n\n#3- cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf to your VCF folder\n\n[kcrid001@coreV3-23-040 QCFastqs]$ mkdir ../../VCF\n[kcrid001@coreV3-23-040 QCFastqs]$ cd ../../VCF\n[kcrid001@coreV3-23-040 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF\n[kcrid001@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf ./\n\n\n#4- Determine the number of individuals and variant sites in the class vcf file (and yours if it worked) using:\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n[kcrid001@coreV3-23-040 VCF]$ enable_lmod\n[kcrid001@coreV3-23-040 VCF]$ module load dDocent\n[kcrid001@coreV3-23-040 VCF]$ vcftools\n\nVCFtools (0.1.14)\n© Adam Auton and Anthony Marcketta 2009\n\nProcess Variant Call Format files\n\nFor a list of options, please go to:\n\thttps://vcftools.github.io/man_latest.html\n\nAlternatively, a man page is available, type:\n\tman vcftools\n\nQuestions, comments, and suggestions should be emailed to:\n\tvcftools-help@lists.sourceforge.net\n\n[kcrid001@coreV3-23-040 VCF]$  vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 1214003 out of a possible 1214003 Sites\nRun Time = 14.00 seconds\n\n\n[kcrid001@coreV3-23-040 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF\n[kcrid001@coreV3-23-040 VCF]$ cp ../fastq/QCFastqs/KCridEmergedfastqs.vcf ./\n[kcrid001@coreV3-23-040 VCF]$ vcftools --vcf KCridEmergedfastqs.vcf \n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf KCridEmergedfastqs.vcf\n\n\\After filtering, kept 16 out of 16 Individuals\nAfter filtering, kept 21896 out of a possible 21896 Sites\nRun Time = 0.00 seconds\n\n\n#5- cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt into your directory with your .vcf files\n\n[kcrid001@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt ./\n[kcrid001@coreV3-23-040 VCF]$ ls\nGoodCoralGenelistForVCFSubsetter.txt  mergedfastq_HEAAstrangiaAssembly.vcf\nKCridEmergedfastqs.vcf                out.log\n\n\n#6- Run our host vcf extractor on your merged vcf file using the following syntax:\n\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\n\n\n[kcrid001@coreV3-23-040 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF\n[kcrid001@coreV3-23-040 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\nRead in ContigList\n^CTraceback (most recent call last):\n  File \u0026quot;/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py\u0026quot;, line 38, in \u0026lt;module\u0026gt;\n    if Contig in ContigsToKeep.keys(): #is our contig one we want to keep\nKeyboardInterrupt\n[kcrid001@turing1 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\n[kcrid001@turing1 VCF]$ ls\nGoodCoralGenelistForVCFSubsetter.txt\nKCridEmergedfastqs.vcf\nmergedfastq_HEAAstrangiaAssembly_subset18sp.vcf\nmergedfastq_HEAAstrangiaAssembly_subset.vcf\nmergedfastq_HEAAstrangiaAssembly.vcf\nout.log\n\n\n#7- Compare the number of variant sites in your three files (YOURNAMEmergedfastqs.vcf,  mergedfastq_HEAAstrangiaAssembly.vcf, and  mergedfastq_HEAAstrangiaAssembly_subset.vcf) using:\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf\n\n\n[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 383467 out of a possible 383467 Sites\nRun Time = 6.00 seconds\n[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly.vcf\n\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 1214003 out of a possible 1214003 Sites\nRun Time = 15.00 seconds\n[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf KCridEmergedfastqs.vcf\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf KCridEmergedfastqs.vcf\n\nAfter filtering, kept 16 out of 16 Individuals\nAfter filtering, kept 21896 out of a possible 21896 Sites\nRun Time = 0.00 seconds\n\n\n\n#8- Work through the VCF filtering tutorial until the following step:\n#http://www.ddocent.com/filtering/\n#Now that we have a list of individuals to remove, we can feed that directly into VCFtools for filtering.\n\n#vcftools --vcf raw.g5mac3dp3.recode.vcf --remove lowDP.indv --recode --recode-INFO-all --out raw.g5mac3dplm\n\n\n[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\t--recode-INFO-all\n\t--minQ 30\n\t--max-missing 0.5\n\t--out raw.g5mac3\n\t--recode\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 133004 out of a possible 383467 Sites\nRun Time = 25.00 seconds\n\n[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf raw.g5mac3.recode.vcf\n\t--recode-INFO-all\n\t--minDP 3\n\t--out raw.g5mac3dp3\n\t--recode\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 133004 out of a possible 133004 Sites\nRun Time = 26.00 seconds\n\n[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf raw.g5mac3dp3.recode.vcf\n\t--missing-indv\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting Individual Missingness\nAfter filtering, kept 133004 out of a possible 133004 Sites\nRun Time = 2.00 seconds\n\n[kcrid001@coreV3-23-002 VCF]$ cat out.imiss\nINDV\tN_DATA\tN_GENOTYPES_FILTERED\tN_MISS\tF_MISS\nRI_W_06_merged\t133004\t0\t86665\t0.651597\nRI_W_07_merged\t133004\t0\t82904\t0.62332\nVA_B_03_merged\t133004\t0\t72661\t0.546307\nRI_W_02_merged\t133004\t0\t94767\t0.712512\nRI_W_04_merged\t133004\t0\t92258\t0.693648\nVA_W_09_SNP_clipped\t133004\t0\t24563\t0.184679\nRI_B_08_SNP_clipped\t133004\t0\t116118\t0.873041\nVA_W_08_SNP_clipped\t133004\t0\t109256\t0.821449\nVA_B_08_SNP_clipped\t133004\t0\t124485\t0.935949\nVA_W_02_merged\t133004\t0\t92331\t0.694197\nVA_B_07_merged\t133004\t0\t80522\t0.60541\nRI_B_05_merged\t133004\t0\t55934\t0.420544\nVA_W_06_merged\t133004\t0\t84347\t0.634169\nVA_W_04_merged\t133004\t0\t64460\t0.484647\nVA_W_01_merged\t133004\t0\t86924\t0.653544\nVA_B_10_SNP_clipped\t133004\t0\t107967\t0.811758\nVA_B_06_merged\t133004\t0\t73841\t0.555179\nVA_W_05_merged\t133004\t0\t85184\t0.640462\nRI_B_09_SNP_clipped\t133004\t0\t111872\t0.841118\nVA_W_10_SNP_clipped\t133004\t0\t108814\t0.818126\nRI_W_08_SNP_clipped\t133004\t0\t101731\t0.764872\nRI_B_06_merged\t133004\t0\t103948\t0.78154\nRI_W_10_SNP_clipped\t133004\t0\t120375\t0.905048\nRI_B_04_merged\t133004\t0\t59639\t0.4484\nVA_W_03_merged\t133004\t0\t82018\t0.616658\nRI_B_07_merged\t133004\t0\t91804\t0.690235\nRI_W_05_merged\t133004\t0\t73522\t0.55278\nRI_W_09_SNP_clipped\t133004\t0\t116492\t0.875853\nVA_B_01_merged\t133004\t0\t59653\t0.448505\nVA_B_09_SNP_clipped\t133004\t0\t103404\t0.77745\nRI_B_10_SNP_clipped\t133004\t0\t108283\t0.814133\nRI_W_01_merged\t133004\t0\t95024\t0.714445\nRI_B_01_merged\t133004\t0\t89764\t0.674897\nVA_B_04_merged\t133004\t0\t74178\t0.557713\nRI_B_02_merged\t133004\t0\t114857\t0.86356\nRI_W_03_merged\t133004\t0\t49393\t0.371365\nVA_B_02_merged\t133004\t0\t102051\t0.767278\nVA_W_07_merged\t133004\t0\t83818\t0.630192\nVA_B_05_merged\t133004\t0\t69075\t0.519345\nRI_B_03_merged\t133004\t0\t104979\t0.789292\n\n\n[kcrid001@coreV3-23-002 VCF]$ bash\nkcrid001@coreV3-23-002:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF$ mawk '!/IN/' out.imiss | cut -f5 \u0026gt; totalmissing\nkcrid001@coreV3-23-002:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF$ \nkcrid001@coreV3-23-002:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF$ gnuplot \u0026lt;\u0026lt; \\EOF \n\u0026gt; set terminal dumb size 120, 30\n\u0026gt; set autoscale \n\u0026gt; unset label\n\u0026gt; set title \u0026quot;Histogram of % missing data per individual\u0026quot;\n\u0026gt; set ylabel \u0026quot;Number of Occurrences\u0026quot;\n\u0026gt; set xlabel \u0026quot;% of missing data\u0026quot;\n\u0026gt; #set yr [0:100000]\n\u0026gt; binwidth=0.01\n\u0026gt; bin(x,width)=width*floor(x/width) + binwidth/2.0\n\u0026gt; plot 'totalmissing' using (bin($1,binwidth)):(1.0) smooth freq with boxes\n\u0026gt; pause -1\n\u0026gt; EOF\n\n                                         Histogram of % missing data per individual\n  Number of Occurrences\n      3 ++----------+-----------+-----------+-----------+------****--+---------***-----------***---------+----------++\n        +           +           +           +           +      *'totalmissing' using (bin($1,binwidth)):(1.0) ****** +\n        |                                                      *  *            * *           * *                     |\n        |                                                      *  *            * *           * *                     |\n        |                                                      *  *            * *           * *                     |\n        |                                                      *  *            * *           * *                     |\n    2.5 ++                                                     *  *            * *           * *                    ++\n        |                                                      *  *            * *           * *                     |\n        |                                                      *  *            * *           * *                     |\n        |                                                      *  *            * *           * *                     |\n        |                                                      *  *            * *           * *                     |\n        |                                                      *  *            * *           * *                     |\n      2 ++                                       ****          *  *     ** **  * ************* *     ****           ++\n        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\n        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\n        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\n        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\n        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\n    1.5 ++                                       *  *          *  *     ** **  * *    *  **  * *     *  *           ++\n        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\n        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\n        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\n        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\n        +           +           +           +    *  *   +      *  *  +  ** **  * *    *  **  * *     *  *+           +\n      1 ********************************************************************************************************----++\n       0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1\n                                                      % of missing data\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eday08HW\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1- Clean up your data directory by:\u003c/span\u003e\n\t-Making a SAMS folder and mv all your .sam files into that directory\n\t-Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory\n\t-rm your _unfiltered.vcf files \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e you have any\n\t-rm your .fastq files\n\t-Make a VCF folder \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e your data directory and mv your YOURNAMEmergedfastqs.vcf into this directory (if your freebayes job didn\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003et complete then skip this step)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#2- Start an interactive session via salloc\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@turing1 QCFastqs]$ cd ../../\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@turing1 data]$ mkdir SAMS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@turing1 QCFastqs]$ mv *.sam ../../SAMS/\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@turing1 QCFastqs]$ salloc\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esalloc: Pending job allocation 9278244\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esalloc: job 9278244 queued and waiting for resources\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esalloc: job 9278244 has been allocated resources\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003esalloc: Granted job allocation 9278244\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 QCFastqs]$ mkdir ../../BAMS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 QCFastqs]$ mv *.bam *.bam.bai ../../BAMS/\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#3- cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf to your VCF folder\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 QCFastqs]$ mkdir ../../VCF\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 QCFastqs]$ cd ../../VCF\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 VCF]$ pwd\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf ./\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#4- Determine the number of individuals and variant sites in the class vcf file (and yours if it worked) using:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 VCF]$ enable_lmod\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 VCF]$ module load dDocent\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 VCF]$ vcftools\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools (0.1.14)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e© Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eProcess Variant Call Format files\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eFor a list of options, please go to:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\thttps://vcftools.github.io/man_latest.html\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAlternatively, a man page is available, type:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tman vcftools\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eQuestions, comments, and suggestions should be emailed to:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\tvcftools-help@lists.sourceforge.net\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 VCF]$  vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools - 0.1.14\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(C) Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eParameters as interpreted:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 40 out of 40 Individuals\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 1214003 out of a possible 1214003 Sites\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRun Time = 14.00 seconds\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 VCF]$ pwd\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 VCF]$ cp ../fastq/QCFastqs/KCridEmergedfastqs.vcf ./\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 VCF]$ vcftools --vcf KCridEmergedfastqs.vcf \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools - 0.1.14\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(C) Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eParameters as interpreted:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--vcf KCridEmergedfastqs.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\\After filtering, kept 16 out of 16 Individuals\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 21896 out of a possible 21896 Sites\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRun Time = 0.00 seconds\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#5- cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt into your directory with your .vcf files\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt ./\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 VCF]$ ls\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eGoodCoralGenelistForVCFSubsetter.txt  mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eKCridEmergedfastqs.vcf                out.log\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#6- Run our host vcf extractor on your merged vcf file using the following syntax:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 VCF]$ pwd\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-040 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRead in ContigList\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e^CTraceback (most recent call last):\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e  File \"/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py\", line 38, in \u0026lt;module\u0026gt;\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e    if Contig in ContigsToKeep.keys(): #is our contig one we want to keep\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eKeyboardInterrupt\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@turing1 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@turing1 VCF]$ ls\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eGoodCoralGenelistForVCFSubsetter.txt\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eKCridEmergedfastqs.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emergedfastq_HEAAstrangiaAssembly_subset18sp.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emergedfastq_HEAAstrangiaAssembly_subset.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003emergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eout.log\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#7- Compare the number of variant sites in your three files (YOURNAMEmergedfastqs.vcf,  mergedfastq_HEAAstrangiaAssembly.vcf, and  mergedfastq_HEAAstrangiaAssembly_subset.vcf) using:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools - 0.1.14\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(C) Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eParameters as interpreted:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 40 out of 40 Individuals\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 383467 out of a possible 383467 Sites\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRun Time = 6.00 seconds\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools - 0.1.14\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(C) Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eParameters as interpreted:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 40 out of 40 Individuals\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 1214003 out of a possible 1214003 Sites\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRun Time = 15.00 seconds\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf KCridEmergedfastqs.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools - 0.1.14\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(C) Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eParameters as interpreted:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--vcf KCridEmergedfastqs.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 16 out of 16 Individuals\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 21896 out of a possible 21896 Sites\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRun Time = 0.00 seconds\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#8- Work through the VCF filtering tutorial until the following step:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#http://www.ddocent.com/filtering/\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#Now that we have a list of individuals to remove, we can feed that directly into VCFtools for filtering.\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#vcftools --vcf raw.g5mac3dp3.recode.vcf --remove lowDP.indv --recode --recode-INFO-all --out raw.g5mac3dplm\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools - 0.1.14\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(C) Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eParameters as interpreted:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--recode-INFO-all\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--minQ 30\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--max-missing 0.5\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--out raw.g5mac3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--recode\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 40 out of 40 Individuals\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eOutputting VCF file...\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 133004 out of a possible 383467 Sites\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRun Time = 25.00 seconds\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools - 0.1.14\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(C) Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eParameters as interpreted:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--vcf raw.g5mac3.recode.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--recode-INFO-all\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--minDP 3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--out raw.g5mac3dp3\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--recode\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 40 out of 40 Individuals\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eOutputting VCF file...\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 133004 out of a possible 133004 Sites\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRun Time = 26.00 seconds\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVCFtools - 0.1.14\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(C) Adam Auton and Anthony Marcketta 2009\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eParameters as interpreted:\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--vcf raw.g5mac3dp3.recode.vcf\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\t--missing-indv\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 40 out of 40 Individuals\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eOutputting Individual Missingness\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eAfter filtering, kept 133004 out of a possible 133004 Sites\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRun Time = 2.00 seconds\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-002 VCF]$ cat out.imiss\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eINDV\tN_DATA\tN_GENOTYPES_FILTERED\tN_MISS\tF_MISS\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_06_merged\t133004\t0\t86665\t0.651597\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_07_merged\t133004\t0\t82904\t0.62332\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_03_merged\t133004\t0\t72661\t0.546307\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_02_merged\t133004\t0\t94767\t0.712512\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_04_merged\t133004\t0\t92258\t0.693648\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_09_SNP_clipped\t133004\t0\t24563\t0.184679\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_08_SNP_clipped\t133004\t0\t116118\t0.873041\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_08_SNP_clipped\t133004\t0\t109256\t0.821449\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_08_SNP_clipped\t133004\t0\t124485\t0.935949\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_02_merged\t133004\t0\t92331\t0.694197\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_07_merged\t133004\t0\t80522\t0.60541\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_05_merged\t133004\t0\t55934\t0.420544\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_06_merged\t133004\t0\t84347\t0.634169\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_04_merged\t133004\t0\t64460\t0.484647\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_01_merged\t133004\t0\t86924\t0.653544\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_10_SNP_clipped\t133004\t0\t107967\t0.811758\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_06_merged\t133004\t0\t73841\t0.555179\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_05_merged\t133004\t0\t85184\t0.640462\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_09_SNP_clipped\t133004\t0\t111872\t0.841118\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_10_SNP_clipped\t133004\t0\t108814\t0.818126\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_08_SNP_clipped\t133004\t0\t101731\t0.764872\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_06_merged\t133004\t0\t103948\t0.78154\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_10_SNP_clipped\t133004\t0\t120375\t0.905048\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_04_merged\t133004\t0\t59639\t0.4484\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_03_merged\t133004\t0\t82018\t0.616658\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_07_merged\t133004\t0\t91804\t0.690235\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_05_merged\t133004\t0\t73522\t0.55278\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_09_SNP_clipped\t133004\t0\t116492\t0.875853\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_01_merged\t133004\t0\t59653\t0.448505\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_09_SNP_clipped\t133004\t0\t103404\t0.77745\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_10_SNP_clipped\t133004\t0\t108283\t0.814133\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_01_merged\t133004\t0\t95024\t0.714445\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_01_merged\t133004\t0\t89764\t0.674897\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_04_merged\t133004\t0\t74178\t0.557713\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_02_merged\t133004\t0\t114857\t0.86356\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_W_03_merged\t133004\t0\t49393\t0.371365\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_02_merged\t133004\t0\t102051\t0.767278\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_W_07_merged\t133004\t0\t83818\t0.630192\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eVA_B_05_merged\t133004\t0\t69075\t0.519345\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_03_merged\t133004\t0\t104979\t0.789292\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e[kcrid001@coreV3-23-002 VCF]$ bash\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ekcrid001@coreV3-23-002:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF$ mawk \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e/IN/\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e out.imiss | cut -f5 \u0026gt; totalmissing\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ekcrid001@coreV3-23-002:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF$ \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ekcrid001@coreV3-23-002:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF$ gnuplot \u0026lt;\u0026lt; \\EOF \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set terminal dumb size 120, 30\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set autoscale \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; unset label\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set title \"Histogram of % missing data per individual\"\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set ylabel \"Number of Occurrences\"\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; set xlabel \"% of missing data\"\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; #set yr [0:100000]\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; binwidth=0.01\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; bin(x,width)=width*floor(x/width) + binwidth/2.0\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; plot \u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003etotalmissing\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e using (bin($1,binwidth)):(1.0) smooth freq with boxes\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; pause -1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u0026gt; EOF\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e                                         Histogram of % missing data per individual\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e  Number of Occurrences\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      3 ++----------+-----------+-----------+-----------+------****--+---------***-----------***---------+----------++\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        +           +           +           +           +      *\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003etotalmissing\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e using (bin($1,binwidth)):(1.0) ****** +\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                      *  *            * *           * *                     |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                      *  *            * *           * *                     |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                      *  *            * *           * *                     |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                      *  *            * *           * *                     |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e    2.5 ++                                                     *  *            * *           * *                    ++\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                      *  *            * *           * *                     |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                      *  *            * *           * *                     |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                      *  *            * *           * *                     |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                      *  *            * *           * *                     |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                                      *  *            * *           * *                     |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      2 ++                                       ****          *  *     ** **  * ************* *     ****           ++\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e    1.5 ++                                       *  *          *  *     ** **  * *    *  **  * *     *  *           ++\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e        +           +           +           +    *  *   +      *  *  +  ** **  * *    *  **  * *     *  *+           +\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e      1 ********************************************************************************************************----++\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e       0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e                                                      % of missing data\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay09 HW\u003c/h2\u003e\u003ca id=\"user-content-day09-hw\" class=\"anchor\" aria-label=\"Permalink: Day09 HW\" href=\"#day09-hw\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#day09HW - VCF to Genepop to PCAs\n\n\n#1-  Run one of the following sets of prescribed filters on your mergedfastq_HEAAstrangiaAssembly_subset.vcf, note these are fairly conservative filters\n\n##Half the class run the following\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\n##the other half run the filters we used from the paper\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\n\n\n[kcrid001@turing1 data]$ salloc\nsalloc: Pending job allocation 9280290\nsalloc: job 9280290 queued and waiting for resources\nsalloc: job 9280290 has been allocated resources\nsalloc: Granted job allocation 9280290\n[kcrid001@coreV3-23-004 data]$ cd VCF/\n[kcrid001@coreV3-23-004 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF\n[kcrid001@coreV3-23-004 VCF]$ ls\nGoodCoralGenelistForVCFSubsetter.txt             out.log\nKCridEmergedfastqs.vcf                           raw.g5mac3dp3.log\nmergedfastq_HEAAstrangiaAssembly_subset18sp.vcf  raw.g5mac3dp3.recode.vcf\nmergedfastq_HEAAstrangiaAssembly_subset.vcf      raw.g5mac3.log\nmergedfastq_HEAAstrangiaAssembly.vcf             raw.g5mac3.recode.vcf\nout.imiss                                        totalmissing\n[kcrid001@coreV3-23-004 VCF]$ enable_lmod\n[kcrid001@coreV3-23-004 VCF]$ module load dDocent\n[kcrid001@coreV3-23-004 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 383467 out of a possible 383467 Sites\nRun Time = 6.00 seconds\n[kcrid001@coreV3-23-004 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\t--recode-INFO-all\n\t--maf 0.015\n\t--max-alleles 2\n\t--minDP 3\n\t--minGQ 20\n\t--hwe 0.01\n\t--minQ 30\n\t--max-missing 0.5\n\t--out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\t--recode\n\t--remove-indels\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 16843 out of a possible 383467 Sites\nRun Time = 11.00 seconds\n## these numbers look wrong, so I used the other subset vcf file that was in my directory\n\n[kcrid001@coreV3-23-004 VCF]$ ls\n18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.log\n18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf\nGoodCoralGenelistForVCFSubsetter.txt\nKCridEmergedfastqs.vcf\nmergedfastq_HEAAstrangiaAssembly_subset18sp.vcf\nmergedfastq_HEAAstrangiaAssembly_subset.vcf\nmergedfastq_HEAAstrangiaAssembly.vcf\nout.imiss\nout.log\nraw.g5mac3dp3.log\nraw.g5mac3dp3.recode.vcf\nraw.g5mac3.log\nraw.g5mac3.recode.vcf\ntotalmissing\n[kcrid001@coreV3-23-004 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf\n\t--recode-INFO-all\n\t--maf 0.015\n\t--max-alleles 2\n\t--minDP 3\n\t--minGQ 20\n\t--hwe 0.01\n\t--minQ 30\n\t--max-missing 0.5\n\t--out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\t--recode\n\t--remove-indels\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 18718 out of a possible 432676 Sites\nRun Time = 14.00 seconds\n\n[kcrid001@coreV3-23-004 VCF]$ vcftools --vcf 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf --missing-indv\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf\n\t--missing-indv\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting Individual Missingness\nAfter filtering, kept 18718 out of a possible 18718 Sites\nRun Time = 1.00 seconds\n[kcrid001@coreV3-23-004 VCF]$ ls\n18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.log\n18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf\nGoodCoralGenelistForVCFSubsetter.txt\nKCridEmergedfastqs.vcf\nmergedfastq_HEAAstrangiaAssembly_subset18sp.vcf\nmergedfastq_HEAAstrangiaAssembly_subset.vcf\nmergedfastq_HEAAstrangiaAssembly.vcf\nout.imiss\nout.log\nraw.g5mac3dp3.log\nraw.g5mac3dp3.recode.vcf\nraw.g5mac3.log\nraw.g5mac3.recode.vcf\ntotalmissing\n[kcrid001@coreV3-23-004 VCF]$ cat out.imiss\nINDV\tN_DATA\tN_GENOTYPES_FILTERED\tN_MISS\tF_MISS\nRI_W_06_merged\t18718\t0\t5674\t0.303131\nRI_W_07_merged\t18718\t0\t4701\t0.251149\nVA_B_03_merged\t18718\t0\t3474\t0.185597\nRI_W_02_merged\t18718\t0\t7137\t0.381291\nRI_W_04_merged\t18718\t0\t7811\t0.417299\nVA_W_09_SNP_clipped\t18718\t0\t786\t0.0419917\nRI_B_08_SNP_clipped\t18718\t0\t14130\t0.754888\nVA_W_08_SNP_clipped\t18718\t0\t11806\t0.63073\nVA_B_08_SNP_clipped\t18718\t0\t16264\t0.868896\nVA_W_02_merged\t18718\t0\t7067\t0.377551\nVA_B_07_merged\t18718\t0\t4916\t0.262635\nRI_B_05_merged\t18718\t0\t2439\t0.130302\nVA_W_06_merged\t18718\t0\t4639\t0.247836\nVA_W_04_merged\t18718\t0\t3015\t0.161075\nVA_W_01_merged\t18718\t0\t5970\t0.318944\nVA_B_10_SNP_clipped\t18718\t0\t10773\t0.575542\nVA_B_06_merged\t18718\t0\t3608\t0.192756\nVA_W_05_merged\t18718\t0\t5904\t0.315418\nRI_B_09_SNP_clipped\t18718\t0\t12981\t0.693504\nVA_W_10_SNP_clipped\t18718\t0\t10890\t0.581793\nRI_W_08_SNP_clipped\t18718\t0\t9193\t0.491132\nRI_B_06_merged\t18718\t0\t8881\t0.474463\nRI_W_10_SNP_clipped\t18718\t0\t14714\t0.786088\nRI_B_04_merged\t18718\t0\t2421\t0.129341\nVA_W_03_merged\t18718\t0\t4740\t0.253232\nRI_B_07_merged\t18718\t0\t6586\t0.351854\nRI_W_05_merged\t18718\t0\t3762\t0.200983\nRI_W_09_SNP_clipped\t18718\t0\t13901\t0.742654\nVA_B_01_merged\t18718\t0\t3472\t0.18549\nVA_B_09_SNP_clipped\t18718\t0\t9209\t0.491986\nRI_B_10_SNP_clipped\t18718\t0\t11091\t0.592531\nRI_W_01_merged\t18718\t0\t7875\t0.420718\nRI_B_01_merged\t18718\t0\t6067\t0.324127\nVA_B_04_merged\t18718\t0\t4853\t0.259269\nRI_B_02_merged\t18718\t0\t13042\t0.696762\nRI_W_03_merged\t18718\t0\t1968\t0.105139\nVA_B_02_merged\t18718\t0\t10252\t0.547708\nVA_W_07_merged\t18718\t0\t4860\t0.259643\nVA_B_05_merged\t18718\t0\t3311\t0.176889\nRI_B_03_merged\t18718\t0\t9817\t0.524468\n\n#2- Make a population file containing two columns with no header, tab delimited text the first column should be the individual name and the second column the population to which that individual belongs\n\n[kcrid001@coreV3-23-004 VCF]$ grep '#CHROM' mergedfastq_HEAAstrangiaAssembly.vcf\n#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tRI_W_06_merged\tRI_W_07_merged\tVA_B_03_merged\tRI_W_02_merged\tRI_W_04_merged\tVA_W_09_SNP_clipped\tRI_B_08_SNP_clipped\tVA_W_08_SNP_clipped\tVA_B_08_SNP_clipped\tVA_W_02_merged\tVA_B_07_merged\tRI_B_05_merged\tVA_W_06_merged\tVA_W_04_merged\tVA_W_01_merged\tVA_B_10_SNP_clipped\tVA_B_06_mergedVA_W_05_merged\tRI_B_09_SNP_clipped\tVA_W_10_SNP_clipped\tRI_W_08_SNP_clipped\tRI_B_06_merged\tRI_W_10_SNP_clipped\tRI_B_04_merged\tVA_W_03_merged\tRI_B_07_merged\tRI_W_05_mergedRI_W_09_SNP_clipped\tVA_B_01_merged\tVA_B_09_SNP_clipped\tRI_B_10_SNP_clipped\tRI_W_01_merged\tRI_B_01_merged\tVA_B_04_merged\tRI_B_02_merged\tRI_W_03_merged\tVA_B_02_merged\tVA_W_07_merged\tVA_B_05_merged\tRI_B_03_merged\n[kcrid001@coreV3-23-004 VCF]$ grep '#CHROM' mergedfastq_HEAAstrangiaAssembly.vcf \u0026gt; samplenames.txt\n[kcrid001@coreV3-23-004 VCF]$ cat samplenames.txt \n#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tRI_W_06_merged\tRI_W_07_merged\tVA_B_03_merged\tRI_W_02_merged\tRI_W_04_merged\tVA_W_09_SNP_clipped\tRI_B_08_SNP_clipped\tVA_W_08_SNP_clipped\tVA_B_08_SNP_clipped\tVA_W_02_merged\tVA_B_07_merged\tRI_B_05_merged\tVA_W_06_merged\tVA_W_04_merged\tVA_W_01_merged\tVA_B_10_SNP_clipped\tVA_B_06_mergedVA_W_05_merged\tRI_B_09_SNP_clipped\tVA_W_10_SNP_clipped\tRI_W_08_SNP_clipped\tRI_B_06_merged\tRI_W_10_SNP_clipped\tRI_B_04_merged\tVA_W_03_merged\tRI_B_07_merged\tRI_W_05_mergedRI_W_09_SNP_clipped\tVA_B_01_merged\tVA_B_09_SNP_clipped\tRI_B_10_SNP_clipped\tRI_W_01_merged\tRI_B_01_merged\tVA_B_04_merged\tRI_B_02_merged\tRI_W_03_merged\tVA_B_02_merged\tVA_W_07_merged\tVA_B_05_merged\tRI_B_03_merged\n[kcrid001@coreV3-23-004 VCF]$ cut -f 1 out.imiss\nINDV\nRI_W_06_merged\nRI_W_07_merged\nVA_B_03_merged\nRI_W_02_merged\nRI_W_04_merged\nVA_W_09_SNP_clipped\nRI_B_08_SNP_clipped\nVA_W_08_SNP_clipped\nVA_B_08_SNP_clipped\nVA_W_02_merged\nVA_B_07_merged\nRI_B_05_merged\nVA_W_06_merged\nVA_W_04_merged\nVA_W_01_merged\nVA_B_10_SNP_clipped\nVA_B_06_merged\nVA_W_05_merged\nRI_B_09_SNP_clipped\nVA_W_10_SNP_clipped\nRI_W_08_SNP_clipped\nRI_B_06_merged\nRI_W_10_SNP_clipped\nRI_B_04_merged\nVA_W_03_merged\nRI_B_07_merged\nRI_W_05_merged\nRI_W_09_SNP_clipped\nVA_B_01_merged\nVA_B_09_SNP_clipped\nRI_B_10_SNP_clipped\nRI_W_01_merged\nRI_B_01_merged\nVA_B_04_merged\nRI_B_02_merged\nRI_W_03_merged\nVA_B_02_merged\nVA_W_07_merged\nVA_B_05_merged\nRI_B_03_merged\n\n# paste in excel and use the 'left' command to get the first 4 characters of each name. copy and paste to the cluster\n\n[kcrid001@coreV3-23-004 VCF]$ nano popfile.txt\n[kcrid001@coreV3-23-004 VCF]$ cat popfile.txt \nRI_W_06_merged\tRI_W\nRI_W_07_merged\tRI_W\nVA_B_03_merged\tVA_B\nRI_W_02_merged\tRI_W\nRI_W_04_merged\tRI_W\nVA_W_09_SNP_clipped\tVA_W\nRI_B_08_SNP_clipped\tRI_B\nVA_W_08_SNP_clipped\tVA_W\nVA_B_08_SNP_clipped\tVA_B\nVA_W_02_merged\tVA_W\nVA_B_07_merged\tVA_B\nRI_B_05_merged\tRI_B\nVA_W_06_merged\tVA_W\nVA_W_04_merged\tVA_W\nVA_W_01_merged\tVA_W\nVA_B_10_SNP_clipped\tVA_B\nVA_B_06_merged\tVA_B\nVA_W_05_merged\tVA_W\nRI_B_09_SNP_clipped\tRI_B\nVA_W_10_SNP_clipped\tVA_W\nRI_W_08_SNP_clipped\tRI_W\nRI_B_06_merged\tRI_B\nRI_W_10_SNP_clipped\tRI_W\nRI_B_04_merged\tRI_B\nVA_W_03_merged\tVA_W\nRI_B_07_merged\tRI_B\nRI_W_05_merged\tRI_W\nRI_W_09_SNP_clipped\tRI_W\nVA_B_01_merged\tVA_B\nVA_B_09_SNP_clipped\tVA_B\nRI_B_10_SNP_clipped\tRI_B\nRI_W_01_merged\tRI_W\nRI_B_01_merged\tRI_B\nVA_B_04_merged\tVA_B\nRI_B_02_merged\tRI_B\nRI_W_03_merged\tRI_W\nVA_B_02_merged\tVA_B\nVA_W_07_merged\tVA_W\nVA_B_05_merged\tVA_B\nRI_B_03_merged\tRI_B\n\n\n#3- Convert your filtered .vcf file to genepop format using the following command:\n\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py YOURFILTERED.vcf YOUR_PopFile.txt\n\n[kcrid001@coreV3-23-004 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF\n[kcrid001@coreV3-23-004 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf popfile.txt\nIndivs with genotypes in vcf file: RI_W_06_merged\tRI_W_07_merged\tVA_B_03_merged\tRI_W_02_merged\tRI_W_04_merged\tVA_W_09_SNP_clipped\tRI_B_08_SNP_clipped\tVA_W_08_SNP_clipped\tVA_B_08_SNP_clipped\tVA_W_02_merged\tVA_B_07_merged\tRI_B_05_merged\tVA_W_06_merged\tVA_W_04_merged\tVA_W_01_merged\tVA_B_10_SNP_clipped\tVA_B_06_merged\tVA_W_05_merged\tRI_B_09_SNP_clipped\tVA_W_10_SNP_clipped\tRI_W_08_SNP_clipped\tRI_B_06_merged\tRI_W_10_SNP_clipped\tRI_B_04_merged\tVA_W_03_merged\tRI_B_07_merged\tRI_W_05_merged\tRI_W_09_SNP_clipped\tVA_B_01_merged\tVA_B_09_SNP_clipped\tRI_B_10_SNP_clipped\tRI_W_01_merged\tRI_B_01_merged\tVA_B_04_merged\tRI_B_02_merged\tRI_W_03_merged\tVA_B_02_merged\tVA_W_07_merged\tVA_B_05_merged\tRI_B_03_merged\n44 18718 18718 18718 18718 40\n\n#4- SCP your YOURFILE_allfilters.recode_subset_genepop.gen file to your laptop\n(base) Katies-MacBook-Air-4:day09 katiecrider$ pwd\n/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day09\n(base) Katies-MacBook-Air-4:day09 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF/18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen ./\n\n#5- Switch to the adegenet_PCAs.R script and follow through the steps to produce some of the figures.\n\ninstall.packages(\u0026quot;deldir\u0026quot;)\nlibrary(\u0026quot;adegenet\u0026quot;)\nlibrary(\u0026quot;ape\u0026quot;)\nlibrary(\u0026quot;pegas\u0026quot;)\nlibrary(\u0026quot;seqinr\u0026quot;)\nlibrary(\u0026quot;ggplot2\u0026quot;)\nlibrary(\u0026quot;viridis\u0026quot;)\nlibrary(\u0026quot;scales\u0026quot;)\nlibrary(\u0026quot;factoextra\u0026quot;)\n\n#set working directory\nsetwd(\u0026quot;/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day09\u0026quot;)\n\ndatafile\u0026lt;-read.genepop('18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen', ncode=2)\n\n\nsum(is.na(datafile$tab))\ndatafile #shows info\nYOURdata\u0026lt;-scaleGen(datafile, NA.method='mean')\nX\u0026lt;-YOURdata\nY\u0026lt;-as.factor(substring(pop(datafile),1,4)) #change the second number here to change how many letters/numbers you keep in the name\npca1 \u0026lt;- dudi.pca(X,cent=T, scale=T, scannf=F, nf=3)\nsummary(pca1)\n\n\n#### PCAs ####\n# individual labels\ns.label(pca1$li)\n\n# population elipses\ns.class(pca1$li, pop(datafile))\n\n#color symbols, pop names\n#pdf(\u0026quot;YOURINITIALS_ColorPCA1v2.pdf\u0026quot;)\ncol \u0026lt;- c(\u0026quot;blue\u0026quot;,\u0026quot;red\u0026quot;, \u0026quot;green\u0026quot;, \u0026quot;black\u0026quot;)\ns.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)\nadd.scatter.eig(pca1$eig[1:3], 3,1,2, posi=\u0026quot;topright\u0026quot;)\ntitle(\u0026quot;PCA of DJB_data\\naxes 1-2\u0026quot;)\n#dev.off()\n\n#### Snapclust ####\na.clust\u0026lt;-snapclust(datafile, k = 4)\nclass(a.clust)\nnames(a.clust)\na.tab \u0026lt;- table(pop(datafile), a.clust$group)\ntable.value(a.tab, col.labels = 1:2)\n\ncompoplot(a.clust)\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eday09HW - VCF to Genepop to PCAs\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1-  Run one of the following sets of prescribed filters on your mergedfastq_HEAAstrangiaAssembly_subset.vcf, note these are fairly conservative filters\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Half the class run the following\u003c/span\u003e\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#the other half run the filters we used from the paper\u003c/span\u003e\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\n\n\n[kcrid001@turing1 data]$ salloc\nsalloc: Pending job allocation 9280290\nsalloc: job 9280290 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9280290 has been allocated resources\nsalloc: Granted job allocation 9280290\n[kcrid001@coreV3-23-004 data]$ \u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e VCF/\n[kcrid001@coreV3-23-004 VCF]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF\n[kcrid001@coreV3-23-004 VCF]$ ls\nGoodCoralGenelistForVCFSubsetter.txt             out.log\nKCridEmergedfastqs.vcf                           raw.g5mac3dp3.log\nmergedfastq_HEAAstrangiaAssembly_subset18sp.vcf  raw.g5mac3dp3.recode.vcf\nmergedfastq_HEAAstrangiaAssembly_subset.vcf      raw.g5mac3.log\nmergedfastq_HEAAstrangiaAssembly.vcf             raw.g5mac3.recode.vcf\nout.imiss                                        totalmissing\n[kcrid001@coreV3-23-004 VCF]$ enable_lmod\n[kcrid001@coreV3-23-004 VCF]$ module load dDocent\n[kcrid001@coreV3-23-004 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\nAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 383467 out of a possible 383467 Sites\nRun Time = 6.00 seconds\n[kcrid001@coreV3-23-004 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n\t--recode-INFO-all\n\t--maf 0.015\n\t--max-alleles 2\n\t--minDP 3\n\t--minGQ 20\n\t--hwe 0.01\n\t--minQ 30\n\t--max-missing 0.5\n\t--out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\t--recode\n\t--remove-indels\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 16843 out of a possible 383467 Sites\nRun Time = 11.00 seconds\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# these numbers look wrong, so I used the other subset vcf file that was in my directory\u003c/span\u003e\n\n[kcrid001@coreV3-23-004 VCF]$ ls\n18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.log\n18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf\nGoodCoralGenelistForVCFSubsetter.txt\nKCridEmergedfastqs.vcf\nmergedfastq_HEAAstrangiaAssembly_subset18sp.vcf\nmergedfastq_HEAAstrangiaAssembly_subset.vcf\nmergedfastq_HEAAstrangiaAssembly.vcf\nout.imiss\nout.log\nraw.g5mac3dp3.log\nraw.g5mac3dp3.recode.vcf\nraw.g5mac3.log\nraw.g5mac3.recode.vcf\ntotalmissing\n[kcrid001@coreV3-23-004 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\nVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf\n\t--recode-INFO-all\n\t--maf 0.015\n\t--max-alleles 2\n\t--minDP 3\n\t--minGQ 20\n\t--hwe 0.01\n\t--minQ 30\n\t--max-missing 0.5\n\t--out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters\n\t--recode\n\t--remove-indels\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 18718 out of a possible 432676 Sites\nRun Time = 14.00 seconds\n\n[kcrid001@coreV3-23-004 VCF]$ vcftools --vcf 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf --missing-indv\n\nVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\n\nParameters as interpreted:\n\t--vcf 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf\n\t--missing-indv\n\nAfter filtering, kept 40 out of 40 Individuals\nOutputting Individual Missingness\nAfter filtering, kept 18718 out of a possible 18718 Sites\nRun Time = 1.00 seconds\n[kcrid001@coreV3-23-004 VCF]$ ls\n18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.log\n18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf\nGoodCoralGenelistForVCFSubsetter.txt\nKCridEmergedfastqs.vcf\nmergedfastq_HEAAstrangiaAssembly_subset18sp.vcf\nmergedfastq_HEAAstrangiaAssembly_subset.vcf\nmergedfastq_HEAAstrangiaAssembly.vcf\nout.imiss\nout.log\nraw.g5mac3dp3.log\nraw.g5mac3dp3.recode.vcf\nraw.g5mac3.log\nraw.g5mac3.recode.vcf\ntotalmissing\n[kcrid001@coreV3-23-004 VCF]$ cat out.imiss\nINDV\tN_DATA\tN_GENOTYPES_FILTERED\tN_MISS\tF_MISS\nRI_W_06_merged\t18718\t0\t5674\t0.303131\nRI_W_07_merged\t18718\t0\t4701\t0.251149\nVA_B_03_merged\t18718\t0\t3474\t0.185597\nRI_W_02_merged\t18718\t0\t7137\t0.381291\nRI_W_04_merged\t18718\t0\t7811\t0.417299\nVA_W_09_SNP_clipped\t18718\t0\t786\t0.0419917\nRI_B_08_SNP_clipped\t18718\t0\t14130\t0.754888\nVA_W_08_SNP_clipped\t18718\t0\t11806\t0.63073\nVA_B_08_SNP_clipped\t18718\t0\t16264\t0.868896\nVA_W_02_merged\t18718\t0\t7067\t0.377551\nVA_B_07_merged\t18718\t0\t4916\t0.262635\nRI_B_05_merged\t18718\t0\t2439\t0.130302\nVA_W_06_merged\t18718\t0\t4639\t0.247836\nVA_W_04_merged\t18718\t0\t3015\t0.161075\nVA_W_01_merged\t18718\t0\t5970\t0.318944\nVA_B_10_SNP_clipped\t18718\t0\t10773\t0.575542\nVA_B_06_merged\t18718\t0\t3608\t0.192756\nVA_W_05_merged\t18718\t0\t5904\t0.315418\nRI_B_09_SNP_clipped\t18718\t0\t12981\t0.693504\nVA_W_10_SNP_clipped\t18718\t0\t10890\t0.581793\nRI_W_08_SNP_clipped\t18718\t0\t9193\t0.491132\nRI_B_06_merged\t18718\t0\t8881\t0.474463\nRI_W_10_SNP_clipped\t18718\t0\t14714\t0.786088\nRI_B_04_merged\t18718\t0\t2421\t0.129341\nVA_W_03_merged\t18718\t0\t4740\t0.253232\nRI_B_07_merged\t18718\t0\t6586\t0.351854\nRI_W_05_merged\t18718\t0\t3762\t0.200983\nRI_W_09_SNP_clipped\t18718\t0\t13901\t0.742654\nVA_B_01_merged\t18718\t0\t3472\t0.18549\nVA_B_09_SNP_clipped\t18718\t0\t9209\t0.491986\nRI_B_10_SNP_clipped\t18718\t0\t11091\t0.592531\nRI_W_01_merged\t18718\t0\t7875\t0.420718\nRI_B_01_merged\t18718\t0\t6067\t0.324127\nVA_B_04_merged\t18718\t0\t4853\t0.259269\nRI_B_02_merged\t18718\t0\t13042\t0.696762\nRI_W_03_merged\t18718\t0\t1968\t0.105139\nVA_B_02_merged\t18718\t0\t10252\t0.547708\nVA_W_07_merged\t18718\t0\t4860\t0.259643\nVA_B_05_merged\t18718\t0\t3311\t0.176889\nRI_B_03_merged\t18718\t0\t9817\t0.524468\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2- Make a population file containing two columns with no header, tab delimited text the first column should be the individual name and the second column the population to which that individual belongs\u003c/span\u003e\n\n[kcrid001@coreV3-23-004 VCF]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e#CHROM\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e mergedfastq_HEAAstrangiaAssembly.vcf\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eCHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tRI_W_06_merged\tRI_W_07_merged\tVA_B_03_merged\tRI_W_02_merged\tRI_W_04_merged\tVA_W_09_SNP_clipped\tRI_B_08_SNP_clipped\tVA_W_08_SNP_clipped\tVA_B_08_SNP_clipped\tVA_W_02_merged\tVA_B_07_merged\tRI_B_05_merged\tVA_W_06_merged\tVA_W_04_merged\tVA_W_01_merged\tVA_B_10_SNP_clipped\tVA_B_06_mergedVA_W_05_merged\tRI_B_09_SNP_clipped\tVA_W_10_SNP_clipped\tRI_W_08_SNP_clipped\tRI_B_06_merged\tRI_W_10_SNP_clipped\tRI_B_04_merged\tVA_W_03_merged\tRI_B_07_merged\tRI_W_05_mergedRI_W_09_SNP_clipped\tVA_B_01_merged\tVA_B_09_SNP_clipped\tRI_B_10_SNP_clipped\tRI_W_01_merged\tRI_B_01_merged\tVA_B_04_merged\tRI_B_02_merged\tRI_W_03_merged\tVA_B_02_merged\tVA_W_07_merged\tVA_B_05_merged\tRI_B_03_merged\u003c/span\u003e\n[kcrid001@coreV3-23-004 VCF]$ grep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e#CHROM\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e mergedfastq_HEAAstrangiaAssembly.vcf \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e samplenames.txt\n[kcrid001@coreV3-23-004 VCF]$ cat samplenames.txt \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eCHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tRI_W_06_merged\tRI_W_07_merged\tVA_B_03_merged\tRI_W_02_merged\tRI_W_04_merged\tVA_W_09_SNP_clipped\tRI_B_08_SNP_clipped\tVA_W_08_SNP_clipped\tVA_B_08_SNP_clipped\tVA_W_02_merged\tVA_B_07_merged\tRI_B_05_merged\tVA_W_06_merged\tVA_W_04_merged\tVA_W_01_merged\tVA_B_10_SNP_clipped\tVA_B_06_mergedVA_W_05_merged\tRI_B_09_SNP_clipped\tVA_W_10_SNP_clipped\tRI_W_08_SNP_clipped\tRI_B_06_merged\tRI_W_10_SNP_clipped\tRI_B_04_merged\tVA_W_03_merged\tRI_B_07_merged\tRI_W_05_mergedRI_W_09_SNP_clipped\tVA_B_01_merged\tVA_B_09_SNP_clipped\tRI_B_10_SNP_clipped\tRI_W_01_merged\tRI_B_01_merged\tVA_B_04_merged\tRI_B_02_merged\tRI_W_03_merged\tVA_B_02_merged\tVA_W_07_merged\tVA_B_05_merged\tRI_B_03_merged\u003c/span\u003e\n[kcrid001@coreV3-23-004 VCF]$ cut -f 1 out.imiss\nINDV\nRI_W_06_merged\nRI_W_07_merged\nVA_B_03_merged\nRI_W_02_merged\nRI_W_04_merged\nVA_W_09_SNP_clipped\nRI_B_08_SNP_clipped\nVA_W_08_SNP_clipped\nVA_B_08_SNP_clipped\nVA_W_02_merged\nVA_B_07_merged\nRI_B_05_merged\nVA_W_06_merged\nVA_W_04_merged\nVA_W_01_merged\nVA_B_10_SNP_clipped\nVA_B_06_merged\nVA_W_05_merged\nRI_B_09_SNP_clipped\nVA_W_10_SNP_clipped\nRI_W_08_SNP_clipped\nRI_B_06_merged\nRI_W_10_SNP_clipped\nRI_B_04_merged\nVA_W_03_merged\nRI_B_07_merged\nRI_W_05_merged\nRI_W_09_SNP_clipped\nVA_B_01_merged\nVA_B_09_SNP_clipped\nRI_B_10_SNP_clipped\nRI_W_01_merged\nRI_B_01_merged\nVA_B_04_merged\nRI_B_02_merged\nRI_W_03_merged\nVA_B_02_merged\nVA_W_07_merged\nVA_B_05_merged\nRI_B_03_merged\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e paste in excel and use the 'left' command to get the first 4 characters of each name. copy and paste to the cluster\u003c/span\u003e\n\n[kcrid001@coreV3-23-004 VCF]$ nano popfile.txt\n[kcrid001@coreV3-23-004 VCF]$ cat popfile.txt \nRI_W_06_merged\tRI_W\nRI_W_07_merged\tRI_W\nVA_B_03_merged\tVA_B\nRI_W_02_merged\tRI_W\nRI_W_04_merged\tRI_W\nVA_W_09_SNP_clipped\tVA_W\nRI_B_08_SNP_clipped\tRI_B\nVA_W_08_SNP_clipped\tVA_W\nVA_B_08_SNP_clipped\tVA_B\nVA_W_02_merged\tVA_W\nVA_B_07_merged\tVA_B\nRI_B_05_merged\tRI_B\nVA_W_06_merged\tVA_W\nVA_W_04_merged\tVA_W\nVA_W_01_merged\tVA_W\nVA_B_10_SNP_clipped\tVA_B\nVA_B_06_merged\tVA_B\nVA_W_05_merged\tVA_W\nRI_B_09_SNP_clipped\tRI_B\nVA_W_10_SNP_clipped\tVA_W\nRI_W_08_SNP_clipped\tRI_W\nRI_B_06_merged\tRI_B\nRI_W_10_SNP_clipped\tRI_W\nRI_B_04_merged\tRI_B\nVA_W_03_merged\tVA_W\nRI_B_07_merged\tRI_B\nRI_W_05_merged\tRI_W\nRI_W_09_SNP_clipped\tRI_W\nVA_B_01_merged\tVA_B\nVA_B_09_SNP_clipped\tVA_B\nRI_B_10_SNP_clipped\tRI_B\nRI_W_01_merged\tRI_W\nRI_B_01_merged\tRI_B\nVA_B_04_merged\tVA_B\nRI_B_02_merged\tRI_B\nRI_W_03_merged\tRI_W\nVA_B_02_merged\tVA_B\nVA_W_07_merged\tVA_W\nVA_B_05_merged\tVA_B\nRI_B_03_merged\tRI_B\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3- Convert your filtered .vcf file to genepop format using the following command:\u003c/span\u003e\n\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py YOURFILTERED.vcf YOUR_PopFile.txt\n\n[kcrid001@coreV3-23-004 VCF]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF\n[kcrid001@coreV3-23-004 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf popfile.txt\nIndivs with genotypes \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e vcf file: RI_W_06_merged\tRI_W_07_merged\tVA_B_03_merged\tRI_W_02_merged\tRI_W_04_merged\tVA_W_09_SNP_clipped\tRI_B_08_SNP_clipped\tVA_W_08_SNP_clipped\tVA_B_08_SNP_clipped\tVA_W_02_merged\tVA_B_07_merged\tRI_B_05_merged\tVA_W_06_merged\tVA_W_04_merged\tVA_W_01_merged\tVA_B_10_SNP_clipped\tVA_B_06_merged\tVA_W_05_merged\tRI_B_09_SNP_clipped\tVA_W_10_SNP_clipped\tRI_W_08_SNP_clipped\tRI_B_06_merged\tRI_W_10_SNP_clipped\tRI_B_04_merged\tVA_W_03_merged\tRI_B_07_merged\tRI_W_05_merged\tRI_W_09_SNP_clipped\tVA_B_01_merged\tVA_B_09_SNP_clipped\tRI_B_10_SNP_clipped\tRI_W_01_merged\tRI_B_01_merged\tVA_B_04_merged\tRI_B_02_merged\tRI_W_03_merged\tVA_B_02_merged\tVA_W_07_merged\tVA_B_05_merged\tRI_B_03_merged\n44 18718 18718 18718 18718 40\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4- SCP your YOURFILE_allfilters.recode_subset_genepop.gen file to your laptop\u003c/span\u003e\n(base) Katies-MacBook-Air-4:day09 katiecrider$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day09\n(base) Katies-MacBook-Air-4:day09 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF/18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen ./\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e5- Switch to the adegenet_PCAs.R script and follow through the steps to produce some of the figures.\u003c/span\u003e\n\ninstall.packages(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edeldir\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eadegenet\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eape\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epegas\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eseqinr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eggplot2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eviridis\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003escales\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efactoextra\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eset working directory\u003c/span\u003e\nsetwd(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day09\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\ndatafile\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-read.genepop(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, ncode=2)\n\n\nsum(is.na(datafile\u003cspan class=\"pl-smi\"\u003e$tab\u003c/span\u003e))\ndatafile \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eshows info\u003c/span\u003e\nYOURdata\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-scaleGen(datafile, NA.method=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emean\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\nX\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-YOURdata\nY\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-as.factor(substring(pop(datafile),1,4)) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003echange the second number here to change how many letters/numbers you keep in the name\u003c/span\u003e\npca1 \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- dudi.pca(X,cent=T, scale=T, scannf=F, nf=3)\nsummary(pca1)\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e### PCAs ####\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e individual labels\u003c/span\u003e\ns.label(pca1\u003cspan class=\"pl-smi\"\u003e$li\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e population elipses\u003c/span\u003e\ns.class(pca1\u003cspan class=\"pl-smi\"\u003e$li\u003c/span\u003e, pop(datafile))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ecolor symbols, pop names\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003epdf(\"YOURINITIALS_ColorPCA1v2.pdf\")\u003c/span\u003e\ncol \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egreen\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ns.class(pca1\u003cspan class=\"pl-smi\"\u003e$li\u003c/span\u003e, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)\nadd.scatter.eig(pca1\u003cspan class=\"pl-smi\"\u003e$eig\u003c/span\u003e[1:3], 3,1,2, posi=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etopright\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ntitle(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePCA of DJB_data\\naxes 1-2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003edev.off()\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e### Snapclust ####\u003c/span\u003e\na.clust\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-snapclust(datafile, k = 4)\nclass(a.clust)\nnames(a.clust)\na.tab \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- table(pop(datafile), a.clust\u003cspan class=\"pl-smi\"\u003e$group\u003c/span\u003e)\ntable.value(a.tab, col.labels = 1:2)\n\ncompoplot(a.clust)\n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay10 HW\u003c/h2\u003e\u003ca id=\"user-content-day10-hw\" class=\"anchor\" aria-label=\"Permalink: Day10 HW\" href=\"#day10-hw\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#1- Work through the adegenet_PCAs.R script and follow through the steps to produce some of the figures.\n\nsetwd(\u0026quot;/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day09\u0026quot;)\ndatafile\u0026lt;-read.genepop('coral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen', ncode=2)\n#follow the same scripts as day09 #5, with the other genepop files\n\n###run step 2 as an sbatch .sh script because it will take a while to finish\n#2- cd into your SAMS folder containing your .sams and run the following as an sbatch script on your sam files to generate read mapping counts from each individual file:\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam\n\n[kcrid001@turing1 SAMS]$ salloc\nsalloc: Pending job allocation 9280850\nsalloc: job 9280850 queued and waiting for resources\nsalloc: job 9280850 has been allocated resources\nsalloc: Granted job allocation 9280850\n[kcrid001@coreV2-25-015 SAMS]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/SAMS\n[kcrid001@coreV2-25-015 SAMS]$ nano SAMSreadmap.sh\n[kcrid001@coreV2-25-015 SAMS]$ cat SAMSreadmap.sh \n#!/bin/bash -l\n\n#SBATCH -o SAMSreadmap.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=SAMSreadmap\n\n\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam\n[kcrid001@coreV2-25-015 SAMS]$ sbatch SAMSreadmap.sh \nSubmitted batch job 9280858\n[kcrid001@coreV2-25-015 SAMS]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9280858      main SAMSread kcrid001  R       0:07      1 coreV1-22-016 \n           9280850      main       sh kcrid001  R    1:08:38      1 coreV2-25-015 \n\n\n#3- Once your job from step 1 is finished, start an salloc session and run the following on your outputted _counts.txt files:\n\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt KCridFullCounts_summed.txt NoMatch *_counts.txt\n\n\n[kcrid001@coreV2-25-015 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt KCridFullCounts_summed.txt NoMatch *_counts.txt\nHits not matchedRI_B_04_14_clippedtrimmed.fastq_counts.txt=1698\tRI_B_04_18_clippedtrimmed.fastq_counts.txt=1698\tRI_B_04_22_clippedtrimmed.fastq_counts.txt=1698\tRI_B_05_14_clippedtrimmed.fastq_counts.txt=1698\tRI_W_04_14_clippedtrimmed.fastq_counts.txt=1698\tRI_W_04_18_clippedtrimmed.fastq_counts.txt=1698\tRI_W_04_22_clippedtrimmed.fastq_counts.txt=1698\tRI_W_05_14_clippedtrimmed.fastq_counts.txt=1698\tVA_B_04_14_clippedtrimmed.fastq_counts.txt=1698\tVA_B_04_18_clippedtrimmed.fastq_counts.txt=1698\tVA_B_04_22_clippedtrimmed.fastq_counts.txt=1698\tVA_B_05_14_clippedtrimmed.fastq_counts.txt=1698\tVA_W_04_14_clippedtrimmed.fastq_counts.txt=1698\tVA_W_04_18_clippedtrimmed.fastq_counts.txt=1698\tVA_W_04_22_clippedtrimmed.fastq_counts.txt=1698\tVA_W_05_14_clippedtrimmed.fastq_counts.txt=1698\n\n#4- scp YOURNAMEFullCounts_summed.txt to your laptop\n\n(base) Katies-MacBook-Air-4:day10 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/SAMS/KCridFullCounts_summed.txt ./\nkcrid001@turing.hpc.odu.edu's password: \nKCridFullCounts_summed.txt                              100%  885KB   5.4MB/s   00:00    \n(base) Katies-MacBook-Air-4:day10 katiecrider$ ls\nKCridFullCounts_summed.txt\ncoral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen\ncoral_66_cloneremoved_highoutliers.filtered1SNPper_genepop.gen\nhomework_day10.txt\n(base) Katies-MacBook-Air-4:day10 katiecrider$ pwd\n/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day10\n\n#5- edit the first line of YOURNAMEFullCounts_summed.txt to remove the _counts.txt_UniqueTotReads from each sample name to just retain the actual informative part of the sample name (e.g., RI_W_06_18)\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1- Work through the adegenet_PCAs.R script and follow through the steps to produce some of the figures.\u003c/span\u003e\n\nsetwd(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day09\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndatafile\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-read.genepop(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecoral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, ncode=2)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003efollow the same scripts as day09 #5, with the other genepop files\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e##run step 2 as an sbatch .sh script because it will take a while to finish\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2- cd into your SAMS folder containing your .sams and run the following as an sbatch script on your sam files to generate read mapping counts from each individual file:\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.sam\n\n[kcrid001@turing1 SAMS]$ salloc\nsalloc: Pending job allocation 9280850\nsalloc: job 9280850 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9280850 has been allocated resources\nsalloc: Granted job allocation 9280850\n[kcrid001@coreV2-25-015 SAMS]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/SAMS\n[kcrid001@coreV2-25-015 SAMS]$ nano SAMSreadmap.sh\n[kcrid001@coreV2-25-015 SAMS]$ cat SAMSreadmap.sh \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o SAMSreadmap.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=SAMSreadmap\u003c/span\u003e\n\n\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e.sam\n[kcrid001@coreV2-25-015 SAMS]$ sbatch SAMSreadmap.sh \nSubmitted batch job 9280858\n[kcrid001@coreV2-25-015 SAMS]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9280858      main SAMSread kcrid001  R       0:07      1 coreV1-22-016 \n           9280850      main       sh kcrid001  R    1:08:38      1 coreV2-25-015 \n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3- Once your job from step 1 is finished, start an salloc session and run the following on your outputted _counts.txt files:\u003c/span\u003e\n\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt KCridFullCounts_summed.txt NoMatch \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_counts.txt\n\n\n[kcrid001@coreV2-25-015 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt KCridFullCounts_summed.txt NoMatch \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e_counts.txt\nHits not matchedRI_B_04_14_clippedtrimmed.fastq_counts.txt=1698\tRI_B_04_18_clippedtrimmed.fastq_counts.txt=1698\tRI_B_04_22_clippedtrimmed.fastq_counts.txt=1698\tRI_B_05_14_clippedtrimmed.fastq_counts.txt=1698\tRI_W_04_14_clippedtrimmed.fastq_counts.txt=1698\tRI_W_04_18_clippedtrimmed.fastq_counts.txt=1698\tRI_W_04_22_clippedtrimmed.fastq_counts.txt=1698\tRI_W_05_14_clippedtrimmed.fastq_counts.txt=1698\tVA_B_04_14_clippedtrimmed.fastq_counts.txt=1698\tVA_B_04_18_clippedtrimmed.fastq_counts.txt=1698\tVA_B_04_22_clippedtrimmed.fastq_counts.txt=1698\tVA_B_05_14_clippedtrimmed.fastq_counts.txt=1698\tVA_W_04_14_clippedtrimmed.fastq_counts.txt=1698\tVA_W_04_18_clippedtrimmed.fastq_counts.txt=1698\tVA_W_04_22_clippedtrimmed.fastq_counts.txt=1698\tVA_W_05_14_clippedtrimmed.fastq_counts.txt=1698\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4- scp YOURNAMEFullCounts_summed.txt to your laptop\u003c/span\u003e\n\n(base) Katies-MacBook-Air-4:day10 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/SAMS/KCridFullCounts_summed.txt ./\nkcrid001@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password: \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eKCridFullCounts_summed.txt                              100%  885KB   5.4MB/s   00:00    \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(base) Katies-MacBook-Air-4:day10 katiecrider$ ls\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eKCridFullCounts_summed.txt\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ecoral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ecoral_66_cloneremoved_highoutliers.filtered1SNPper_genepop.gen\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ehomework_day10.txt\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(base) Katies-MacBook-Air-4:day10 katiecrider$ pwd\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day10\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#5- edit the first line of YOURNAMEFullCounts_summed.txt to remove the _counts.txt_UniqueTotReads from each sample name to just retain the actual informative part of the sample name (e.g., RI_W_06_18)\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay11 HW\u003c/h2\u003e\u003ca id=\"user-content-day11-hw\" class=\"anchor\" aria-label=\"Permalink: Day11 HW\" href=\"#day11-hw\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n#1- Download the DESeq2Script_advbioinf.R script and work through on the shared djbFullCounts_summed.txt file\n\n#/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day11/DESeq2Script_advbioinf.R\n\n#copy this and the summed.txt file to the Genomics folder\n\n#make sure to have downloaded these packages already\n\n\u0026gt;library(DESeq2)\n\u0026gt;library(gplots)\n\u0026gt;vignette('DESeq2')\n\n\u0026gt;setwd(\u0026quot;/Users/katiecrider/Desktop/Genomics\u0026quot;)  # The drag and drop from finder works in R, too.\n\u0026gt; conds\u0026lt;-data.frame(\u0026quot;Sample\u0026quot;=names(countsTable))\n\u0026gt; conds$Origin\u0026lt;-factor(substring(conds$Sample,1,2))\n\u0026gt; conds$SymState\u0026lt;-factor(substring(conds$Sample,4,4))\n\u0026gt; conds$Temp\u0026lt;-factor(substring(conds$Sample,9,10))\n\u0026gt; conds$Origin_SymState\u0026lt;-factor(paste(conds$Origin,conds$SymState, sep=\u0026quot;_\u0026quot;))\n\u0026gt; conds$Origin_SymState_Temp\u0026lt;-factor(paste(conds$Origin,conds$SymState,conds$Temp, sep=\u0026quot;_\u0026quot;))\n\u0026gt; conds\n       Sample Origin SymState Temp Origin_SymState Origin_SymState_Temp\n1  RI_B_01_14     RI        B   14            RI_B              RI_B_14\n2  RI_B_01_18     RI        B   18            RI_B              RI_B_18\n3  RI_B_01_22     RI        B   22            RI_B              RI_B_22\n4  RI_B_02_14     RI        B   14            RI_B              RI_B_14\n5  RI_B_02_18     RI        B   18            RI_B              RI_B_18\n6  RI_B_02_22     RI        B   22            RI_B              RI_B_22\n7  RI_B_03_14     RI        B   14            RI_B              RI_B_14\n8  RI_B_03_22     RI        B   22            RI_B              RI_B_22\n9  RI_B_04_14     RI        B   14            RI_B              RI_B_14\n10 RI_B_04_18     RI        B   18            RI_B              RI_B_18\n11 RI_B_04_22     RI        B   22            RI_B              RI_B_22\n12 RI_B_05_14     RI        B   14            RI_B              RI_B_14\n13 RI_B_05_18     RI        B   18            RI_B              RI_B_18\n14 RI_B_05_22     RI        B   22            RI_B              RI_B_22\n15 RI_B_06_14     RI        B   14            RI_B              RI_B_14\n16 RI_B_06_18     RI        B   18            RI_B              RI_B_18\n17 RI_B_06_22     RI        B   22            RI_B              RI_B_22\n18 RI_B_07_14     RI        B   14            RI_B              RI_B_14\n19 RI_B_07_18     RI        B   18            RI_B              RI_B_18\n20 RI_B_07_22     RI        B   22            RI_B              RI_B_22\n21 RI_W_01_14     RI        W   14            RI_W              RI_W_14\n22 RI_W_01_18     RI        W   18            RI_W              RI_W_18\n23 RI_W_01_22     RI        W   22            RI_W              RI_W_22\n24 RI_W_02_14     RI        W   14            RI_W              RI_W_14\n25 RI_W_02_18     RI        W   18            RI_W              RI_W_18\n26 RI_W_02_22     RI        W   22            RI_W              RI_W_22\n27 RI_W_03_14     RI        W   14            RI_W              RI_W_14\n28 RI_W_03_18     RI        W   18            RI_W              RI_W_18\n29 RI_W_03_22     RI        W   22            RI_W              RI_W_22\n30 RI_W_04_14     RI        W   14            RI_W              RI_W_14\n31 RI_W_04_18     RI        W   18            RI_W              RI_W_18\n32 RI_W_04_22     RI        W   22            RI_W              RI_W_22\n33 RI_W_05_14     RI        W   14            RI_W              RI_W_14\n34 RI_W_05_18     RI        W   18            RI_W              RI_W_18\n35 RI_W_05_22     RI        W   22            RI_W              RI_W_22\n36 RI_W_06_14     RI        W   14            RI_W              RI_W_14\n37 RI_W_06_18     RI        W   18            RI_W              RI_W_18\n38 RI_W_06_22     RI        W   22            RI_W              RI_W_22\n39 RI_W_07_14     RI        W   14            RI_W              RI_W_14\n40 RI_W_07_18     RI        W   18            RI_W              RI_W_18\n41 RI_W_07_22     RI        W   22            RI_W              RI_W_22\n42 VA_B_01_14     VA        B   14            VA_B              VA_B_14\n43 VA_B_01_18     VA        B   18            VA_B              VA_B_18\n44 VA_B_01_22     VA        B   22            VA_B              VA_B_22\n45 VA_B_02_14     VA        B   14            VA_B              VA_B_14\n46 VA_B_02_18     VA        B   18            VA_B              VA_B_18\n47 VA_B_02_22     VA        B   22            VA_B              VA_B_22\n48 VA_B_03_14     VA        B   14            VA_B              VA_B_14\n49 VA_B_03_18     VA        B   18            VA_B              VA_B_18\n50 VA_B_03_22     VA        B   22            VA_B              VA_B_22\n51 VA_B_04_14     VA        B   14            VA_B              VA_B_14\n52 VA_B_04_18     VA        B   18            VA_B              VA_B_18\n53 VA_B_04_22     VA        B   22            VA_B              VA_B_22\n54 VA_B_05_14     VA        B   14            VA_B              VA_B_14\n55 VA_B_05_18     VA        B   18            VA_B              VA_B_18\n56 VA_B_05_22     VA        B   22            VA_B              VA_B_22\n57 VA_B_06_14     VA        B   14            VA_B              VA_B_14\n58 VA_B_06_18     VA        B   18            VA_B              VA_B_18\n59 VA_B_06_22     VA        B   22            VA_B              VA_B_22\n60 VA_B_07_14     VA        B   14            VA_B              VA_B_14\n61 VA_B_07_18     VA        B   18            VA_B              VA_B_18\n62 VA_B_07_22     VA        B   22            VA_B              VA_B_22\n63 VA_W_01_14     VA        W   14            VA_W              VA_W_14\n64 VA_W_01_18     VA        W   18            VA_W              VA_W_18\n65 VA_W_02_14     VA        W   14            VA_W              VA_W_14\n66 VA_W_02_18     VA        W   18            VA_W              VA_W_18\n67 VA_W_03_14     VA        W   14            VA_W              VA_W_14\n68 VA_W_03_18     VA        W   18            VA_W              VA_W_18\n69 VA_W_03_22     VA        W   22            VA_W              VA_W_22\n70 VA_W_04_14     VA        W   14            VA_W              VA_W_14\n71 VA_W_04_18     VA        W   18            VA_W              VA_W_18\n72 VA_W_04_22     VA        W   22            VA_W              VA_W_22\n73 VA_W_05_14     VA        W   14            VA_W              VA_W_14\n74 VA_W_05_18     VA        W   18            VA_W              VA_W_18\n75 VA_W_05_22     VA        W   22            VA_W              VA_W_22\n76 VA_W_06_14     VA        W   14            VA_W              VA_W_14\n77 VA_W_06_18     VA        W   18            VA_W              VA_W_18\n78 VA_W_06_22     VA        W   22            VA_W              VA_W_22\n79 VA_W_07_14     VA        W   14            VA_W              VA_W_14\n80 VA_W_07_18     VA        W   18            VA_W              VA_W_18\n81 VA_W_07_22     VA        W   22            VA_W              VA_W_22\n\u0026gt; prop.null \u0026lt;- apply(countsTable, 2, function(x) 100*mean(x==0))\n\u0026gt; \n\u0026gt; barplot(prop.null, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u0026gt; \n\u0026gt; pdf(file=\u0026quot;OverallNullCounts.pdf\u0026quot;,14, 14)\n\u0026gt; barplot(prop.null, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u0026gt; dev.off()\nRStudioGD \n        2 \n\u0026gt; \n\u0026gt; # how many genes have mean count \u0026gt;=3 \n\u0026gt; means = apply(countsTable, 1, mean)\n\u0026gt; table(means\u0026gt;=3)\n\nTRUE \n7543 \n\u0026gt; \n\u0026gt; countsTable\u0026lt;-countsTable[means\u0026gt;=3,]\n\u0026gt; dim(countsTable)\n[1] 7543   81\n\u0026gt; \n\u0026gt; prop.nullv2 \u0026lt;- apply(countsTable, 2, function(x) 100*mean(x==0))\n\u0026gt; pdf(file=\u0026quot;Mean3NullCounts.pdf\u0026quot;,14, 14)\n\u0026gt; barplot(prop.nullv2, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u0026gt; dev.off()\nRStudioGD \n        2 \n\u0026gt; \n\u0026gt; #make count data set\n\u0026gt; dds \u0026lt;- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=~ Origin_SymState_Temp)\n\u0026gt; \n\u0026gt; dds \u0026lt;- DESeq(dds)\nestimating size factors\nestimating dispersions\ngene-wise dispersion estimates\nmean-dispersion relationship\nfinal dispersion estimates\nfitting model and testing\n-- replacing outliers and refitting for 8 genes\n-- DESeq argument 'minReplicatesForReplace' = 7 \n-- original counts are preserved in counts(dds)\nestimating dispersions\nfitting model and testing\n\u0026gt; \n\u0026gt; #transform counts to variance stablized counts\n\u0026gt; vstCounts\u0026lt;-varianceStabilizingTransformation(dds)\n\u0026gt; \n\u0026gt; #plot cluster dendrogram across samples\n\u0026gt; dists\u0026lt;-dist(t(assay(vstCounts)))\n\u0026gt; plot(hclust(dists))\n\u0026gt; \n\u0026gt; plotPCA(vstCounts, intgroup=\u0026quot;Origin\u0026quot;)\n\u0026gt; plotPCA(vstCounts, intgroup=\u0026quot;SymState\u0026quot;)\n\u0026gt; plotPCA(vstCounts, intgroup=\u0026quot;Temp\u0026quot;)\n\u0026gt; plotPCA(vstCounts, intgroup=\u0026quot;Origin_SymState\u0026quot;)\n\u0026gt; plotPCA(vstCounts, intgroup=\u0026quot;Origin_SymState_Temp\u0026quot;)\n\u0026gt; resultsNames(dds)\n [1] \u0026quot;Intercept\u0026quot;                              \n [2] \u0026quot;Origin_SymState_Temp_RI_B_18_vs_RI_B_14\u0026quot;\n [3] \u0026quot;Origin_SymState_Temp_RI_B_22_vs_RI_B_14\u0026quot;\n [4] \u0026quot;Origin_SymState_Temp_RI_W_14_vs_RI_B_14\u0026quot;\n [5] \u0026quot;Origin_SymState_Temp_RI_W_18_vs_RI_B_14\u0026quot;\n [6] \u0026quot;Origin_SymState_Temp_RI_W_22_vs_RI_B_14\u0026quot;\n [7] \u0026quot;Origin_SymState_Temp_VA_B_14_vs_RI_B_14\u0026quot;\n [8] \u0026quot;Origin_SymState_Temp_VA_B_18_vs_RI_B_14\u0026quot;\n [9] \u0026quot;Origin_SymState_Temp_VA_B_22_vs_RI_B_14\u0026quot;\n[10] \u0026quot;Origin_SymState_Temp_VA_W_14_vs_RI_B_14\u0026quot;\n[11] \u0026quot;Origin_SymState_Temp_VA_W_18_vs_RI_B_14\u0026quot;\n[12] \u0026quot;Origin_SymState_Temp_VA_W_22_vs_RI_B_14\u0026quot;\n\u0026gt; \n\u0026gt; dim(dds)\n[1] 7543   81\n\u0026gt; \n\u0026gt; ###Figure out which contrast you want to examine (i.e. which two groups do you want to compare)\n\u0026gt; res \u0026lt;- results(dds, contrast=c(\u0026quot;Origin_SymState_Temp\u0026quot;, \u0026quot;RI_B_18\u0026quot;, \u0026quot;VA_B_18\u0026quot;))\n\u0026gt; head(res)\nlog2 fold change (MLE): Origin_SymState_Temp RI_B_18 vs VA_B_18 \nWald test p-value: Origin_SymState_Temp RI_B_18 vs VA_B_18 \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE      stat\n                       \u0026lt;numeric\u0026gt;      \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\nTR10083|c0_g2_i1_coral   5.92149      -0.124352  0.881723 -0.141032\nTR10090|c0_g1_i1_coral   2.41467       0.403769  0.959279  0.420909\nTR10090|c0_g2_i1_coral   3.75861      -0.835423  0.865239 -0.965540\nTR10103|c0_g1_i1_coral   2.98376      -0.801464  0.848308 -0.944780\nTR10108|c0_g1_i1_coral   5.73440      -0.526701  0.837420 -0.628957\nTR10132|c0_g1_i1_coral   8.52630       1.071572  1.160128  0.923667\n                          pvalue      padj\n                       \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\nTR10083|c0_g2_i1_coral  0.887844  0.996375\nTR10090|c0_g1_i1_coral  0.673822  0.990719\nTR10090|c0_g2_i1_coral  0.334274  0.964876\nTR10103|c0_g1_i1_coral  0.344771  0.967913\nTR10108|c0_g1_i1_coral  0.529377  0.987729\nTR10132|c0_g1_i1_coral  0.355660  0.967913\n\u0026gt; summary(res)\n\nout of 7542 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 3, 0.04%\nLFC \u0026lt; 0 (down)     : 5, 0.066%\noutliers [1]       : 4, 0.053%\nlow counts [2]     : 1, 0.013%\n(mean count \u0026lt; 1)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\n\n\u0026gt; \n\u0026gt; res \u0026lt;- results(dds, contrast=c(\u0026quot;Origin_SymState_Temp\u0026quot;, \u0026quot;RI_W_18\u0026quot;, \u0026quot;VA_W_18\u0026quot;))\n\u0026gt; summary(res)\n\nout of 7542 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 22, 0.29%\nLFC \u0026lt; 0 (down)     : 38, 0.5%\noutliers [1]       : 4, 0.053%\nlow counts [2]     : 3801, 50%\n(mean count \u0026lt; 6)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\n\n\u0026gt; \n\u0026gt; res \u0026lt;- results(dds, contrast=c(\u0026quot;Origin_SymState_Temp\u0026quot;, \u0026quot;RI_B_14\u0026quot;, \u0026quot;VA_B_14\u0026quot;))\n\u0026gt; summary(res)\n\nout of 7542 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 8, 0.11%\nLFC \u0026lt; 0 (down)     : 4, 0.053%\noutliers [1]       : 4, 0.053%\nlow counts [2]     : 1, 0.013%\n(mean count \u0026lt; 1)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\n\n\u0026gt; \n\u0026gt; res \u0026lt;- results(dds, contrast=c(\u0026quot;Origin_SymState_Temp\u0026quot;, \u0026quot;RI_W_14\u0026quot;, \u0026quot;VA_W_14\u0026quot;))\n\u0026gt; summary(res)\n\nout of 7542 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 10, 0.13%\nLFC \u0026lt; 0 (down)     : 3, 0.04%\noutliers [1]       : 4, 0.053%\nlow counts [2]     : 5263, 70%\n(mean count \u0026lt; 8)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\n\n\u0026gt; \n\u0026gt; res \u0026lt;- results(dds, contrast=c(\u0026quot;Origin_SymState_Temp\u0026quot;, \u0026quot;RI_B_22\u0026quot;, \u0026quot;VA_B_22\u0026quot;))\n\u0026gt; summary(res)\n\nout of 7542 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 8, 0.11%\nLFC \u0026lt; 0 (down)     : 2, 0.027%\noutliers [1]       : 4, 0.053%\nlow counts [2]     : 1, 0.013%\n(mean count \u0026lt; 1)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\n\n\u0026gt; \n\u0026gt; res \u0026lt;- results(dds, contrast=c(\u0026quot;Origin_SymState_Temp\u0026quot;, \u0026quot;RI_W_22\u0026quot;, \u0026quot;VA_W_22\u0026quot;))\n\u0026gt; summary(res)\n\nout of 7542 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 11, 0.15%\nLFC \u0026lt; 0 (down)     : 5, 0.066%\noutliers [1]       : 4, 0.053%\nlow counts [2]     : 146, 1.9%\n(mean count \u0026lt; 2)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\n\n\u0026gt; \n\u0026gt; head(res[order(res$padj),])\nlog2 fold change (MLE): Origin_SymState_Temp RI_W_22 vs VA_W_22 \nWald test p-value: Origin_SymState_Temp RI_W_22 vs VA_W_22 \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE      stat\n                       \u0026lt;numeric\u0026gt;      \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\nTR42879|c0_g2_i1_coral  15.41058       17.05753  3.137478   5.43670\nTR13286|c1_g1_i1_coral  32.61026       -3.38044  0.648393  -5.21357\nTR47617|c0_g1_i1_coral   3.74656       19.72616  3.835159   5.14351\nTR8327|c0_g3_i1_coral    8.73277       19.73384  4.063853   4.85594\nTR29560|c0_g1_i1_coral  12.79556        2.74246  0.651668   4.20838\nTR58072|c1_g1_i1_coral  53.85583        1.38039  0.330572   4.17576\n                            pvalue        padj\n                         \u0026lt;numeric\u0026gt;   \u0026lt;numeric\u0026gt;\nTR42879|c0_g2_i1_coral 5.42760e-08 0.000401263\nTR13286|c1_g1_i1_coral 1.85241e-07 0.000664526\nTR47617|c0_g1_i1_coral 2.69658e-07 0.000664526\nTR8327|c0_g3_i1_coral  1.19814e-06 0.002214469\nTR29560|c0_g1_i1_coral 2.57208e-05 0.036593740\nTR58072|c1_g1_i1_coral 2.96987e-05 0.036593740\n\u0026gt; plotCounts(dds, \u0026quot;TR2367|c0_g1_i1_coral\u0026quot;, intgroup=\u0026quot;Origin_SymState_Temp\u0026quot;)\n\u0026gt; \n\u0026gt; #count the number of significantly differentially expressed genes\n\u0026gt; sum(res$padj \u0026lt; 0.3, na.rm =T)\n[1] 30\n\u0026gt; sum(res$padj \u0026lt; 0.2, na.rm =T)\n[1] 20\n\u0026gt; sum(res$padj \u0026lt; 0.1, na.rm =T)\n[1] 16\n\u0026gt; sum(res$pvalue \u0026lt; 0.05, na.rm =T)\n[1] 343\n\u0026gt; summary(res)\n\nout of 7542 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 11, 0.15%\nLFC \u0026lt; 0 (down)     : 5, 0.066%\noutliers [1]       : 4, 0.053%\nlow counts [2]     : 146, 1.9%\n(mean count \u0026lt; 2)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\n\u0026gt; scaledcounts = counts(dds, normalized=T)\n\u0026gt; head(scaledcounts)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14\nTR10083|c0_g2_i1_coral          0    0.84279   2.292953   0.000000\nTR10090|c0_g1_i1_coral          0    2.52837   1.146477   0.000000\nTR10090|c0_g2_i1_coral          0    3.37116   2.292953   3.553607\nTR10103|c0_g1_i1_coral          0    0.84279   4.585907   0.000000\nTR10108|c0_g1_i1_coral          0    1.68558   4.585907   0.000000\nTR10132|c0_g1_i1_coral          0    5.89953   1.146477  17.768037\n                       RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_22\nTR10083|c0_g2_i1_coral   3.273656   4.312157  12.353943   1.269319\nTR10090|c0_g1_i1_coral  13.094624   0.000000   0.000000   0.000000\nTR10090|c0_g2_i1_coral   0.000000   0.000000   0.000000   3.807956\nTR10103|c0_g1_i1_coral   0.000000   8.624315   1.235394   3.807956\nTR10108|c0_g1_i1_coral   6.547312   0.000000  13.589337  15.231824\nTR10132|c0_g1_i1_coral  49.104839   0.000000   6.176972   2.538637\n                       RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10083|c0_g2_i1_coral  12.576774  8.1319863   0.000000   4.298440\nTR10090|c0_g1_i1_coral   2.028512  4.0659932   6.219661   0.537305\nTR10090|c0_g2_i1_coral   4.868429  3.6593938   0.000000   0.537305\nTR10103|c0_g1_i1_coral   6.085536  3.2527945   0.000000   4.835745\nTR10108|c0_g1_i1_coral   2.839917  4.8791918  15.549153   5.641703\nTR10132|c0_g1_i1_coral   0.000000  0.8131986   3.109831   1.343263\n                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18\nTR10083|c0_g2_i1_coral          0  2.3823188  18.130668   3.272361\nTR10090|c0_g1_i1_coral          0  0.0000000   0.000000   1.963417\nTR10090|c0_g2_i1_coral          0  2.3823188   0.000000   4.581305\nTR10103|c0_g1_i1_coral          0  3.9705313   0.000000   3.272361\nTR10108|c0_g1_i1_coral          0  0.7941063   6.043556   9.817083\nTR10132|c0_g1_i1_coral          0  1.5882125   0.000000   1.308944\n                       RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22\nTR10083|c0_g2_i1_coral   19.38574  1.3676909  7.9128854   36.52879\nTR10090|c0_g1_i1_coral    0.00000  0.6838455  1.9782213    0.00000\nTR10090|c0_g2_i1_coral    0.00000  3.4192273  0.9891107    0.00000\nTR10103|c0_g1_i1_coral    0.00000  2.0515364  1.9782213    0.00000\nTR10108|c0_g1_i1_coral   42.00244  3.4192273  0.9891107    0.00000\nTR10132|c0_g1_i1_coral    0.00000  8.2061454 43.5208696   27.39659\n                       RI_W_01_14 RI_W_01_18 RI_W_01_22 RI_W_02_14\nTR10083|c0_g2_i1_coral   63.29565  10.257613   4.369327   0.000000\nTR10090|c0_g1_i1_coral    0.00000  11.625295   1.456442   0.000000\nTR10090|c0_g2_i1_coral    0.00000  11.625295   7.282212   4.027688\nTR10103|c0_g1_i1_coral   28.77075   7.522249   0.000000   0.000000\nTR10108|c0_g1_i1_coral    0.00000   2.735363   1.456442   8.055376\nTR10132|c0_g1_i1_coral    0.00000   1.367682   0.000000   0.000000\n                       RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18\nTR10083|c0_g2_i1_coral  4.8924878   3.593465   8.078028   5.344367\nTR10090|c0_g1_i1_coral  1.9569951   0.000000   4.039014   1.068873\nTR10090|c0_g2_i1_coral  4.8924878   4.791286   2.885010   3.206620\nTR10103|c0_g1_i1_coral  0.9784976   3.593465   2.885010   3.206620\nTR10108|c0_g1_i1_coral  6.8494829   2.395643   2.308008   4.631785\nTR10132|c0_g1_i1_coral  5.8709854  44.319396   1.154004   1.068873\n                       RI_W_03_22 RI_W_04_14 RI_W_04_18 RI_W_04_22\nTR10083|c0_g2_i1_coral  4.4092608   12.32321  2.9466826    17.8865\nTR10090|c0_g1_i1_coral  1.4697536    0.00000  3.4377964     0.0000\nTR10090|c0_g2_i1_coral  1.4697536    0.00000  7.3667065     0.0000\nTR10103|c0_g1_i1_coral  0.9798357    0.00000  0.9822275     0.0000\nTR10108|c0_g1_i1_coral  1.9596715   20.53868 12.7689580     0.0000\nTR10132|c0_g1_i1_coral  0.9798357   28.75416  7.8578203     0.0000\n                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14\nTR10083|c0_g2_i1_coral  3.0758634  0.9718221   5.123934  0.9786798\nTR10090|c0_g1_i1_coral  0.8788181  7.7745767   0.000000  0.0000000\nTR10090|c0_g2_i1_coral  3.5152724  9.7182208   1.280983  2.9360395\nTR10103|c0_g1_i1_coral  3.9546815  1.9436442   2.561967  9.7867983\nTR10108|c0_g1_i1_coral  5.2729086  1.9436442   6.404917 10.7654782\nTR10132|c0_g1_i1_coral  1.7576362  9.7182208  25.619669  4.8933992\n                       RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18\nTR10083|c0_g2_i1_coral    1.28024   5.416550   4.050051   5.034881\nTR10090|c0_g1_i1_coral    0.00000   4.333240   0.000000   0.000000\nTR10090|c0_g2_i1_coral    1.28024   7.583169   2.025025   1.006976\nTR10103|c0_g1_i1_coral    1.28024   2.166620   5.400067  12.083715\nTR10108|c0_g1_i1_coral    0.00000   5.416550   0.000000   1.006976\nTR10132|c0_g1_i1_coral    1.28024  41.165776   1.350017   2.013953\n                       RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10083|c0_g2_i1_coral   7.075440   2.275221   1.310363   2.753792\nTR10090|c0_g1_i1_coral   0.000000   6.825663   6.879404   1.652275\nTR10090|c0_g2_i1_coral   5.660352   9.100884   4.258679   5.507584\nTR10103|c0_g1_i1_coral   1.415088   3.412832   1.965544   2.753792\nTR10108|c0_g1_i1_coral   5.660352   3.412832   6.879404   2.753792\nTR10132|c0_g1_i1_coral  89.150546   1.137611   0.982772   0.000000\n                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14\nTR10083|c0_g2_i1_coral          0   13.08723  17.568884  15.220246\nTR10090|c0_g1_i1_coral          0    0.00000   0.000000   5.534635\nTR10090|c0_g2_i1_coral          0    0.00000   0.000000  10.377440\nTR10103|c0_g1_i1_coral          0    0.00000   1.952098   2.075488\nTR10108|c0_g1_i1_coral          0    0.00000   0.000000  13.836587\nTR10132|c0_g1_i1_coral          0    0.00000   7.808393   8.993782\n                       VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18\nTR10083|c0_g2_i1_coral  13.204264   7.514330   0.000000  0.8308866\nTR10090|c0_g1_i1_coral   1.980640   7.514330   0.000000  4.1544330\nTR10090|c0_g2_i1_coral   3.961279   3.415604   6.526643  2.9081031\nTR10103|c0_g1_i1_coral   3.961279   3.415604   4.079152  0.0000000\nTR10108|c0_g1_i1_coral  17.165543  20.493627   2.447491  4.5698763\nTR10132|c0_g1_i1_coral  28.389167   8.197451   0.000000  2.4926598\n                       VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10083|c0_g2_i1_coral   6.207028   1.933974  2.9919941   1.718262\nTR10090|c0_g1_i1_coral   3.342246   4.512607  0.9973314   6.873047\nTR10090|c0_g2_i1_coral   0.000000   1.289316  0.4986657   5.154785\nTR10103|c0_g1_i1_coral   0.000000   0.000000  2.9919941   0.000000\nTR10108|c0_g1_i1_coral   3.342246   8.380556  4.9866569   0.000000\nTR10132|c0_g1_i1_coral   0.000000   0.000000  4.4879912   5.154785\n                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14\nTR10083|c0_g2_i1_coral   1.213849   0.000000  5.4126287   3.585182\nTR10090|c0_g1_i1_coral   7.283093   0.000000  0.4510524   0.000000\nTR10090|c0_g2_i1_coral   0.000000   3.961279  1.3531572   1.434073\nTR10103|c0_g1_i1_coral   4.855395   6.338047  3.6084191   1.434073\nTR10108|c0_g1_i1_coral  10.317715   3.961279  3.1573667   2.151109\nTR10132|c0_g1_i1_coral  16.386958   2.376767 13.0805193   8.604437\n                       VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18\nTR10083|c0_g2_i1_coral  0.6877679   1.987011   6.431470   7.522889\nTR10090|c0_g1_i1_coral  3.4388395   1.987011   3.675126   1.074698\nTR10090|c0_g2_i1_coral 14.4431257   5.961033   0.000000   3.761445\nTR10103|c0_g1_i1_coral  7.5654468   3.974022   0.000000   3.761445\nTR10108|c0_g1_i1_coral  2.0633037   4.967527   1.837563   4.298794\nTR10132|c0_g1_i1_coral 15.8186615   1.987011   0.000000  10.209636\n                       VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_18\nTR10083|c0_g2_i1_coral   0.000000   4.374769  2.8394448    0.00000\nTR10090|c0_g1_i1_coral   6.346543   0.624967  4.9690284    0.00000\nTR10090|c0_g2_i1_coral   3.966589  12.499340  6.3887508   36.13826\nTR10103|c0_g1_i1_coral   4.759907   0.624967  6.3887508    0.00000\nTR10108|c0_g1_i1_coral   0.000000   8.749538  2.8394448    0.00000\nTR10132|c0_g1_i1_coral   2.379953  12.499340  0.7098612    0.00000\n                       VA_W_03_22 VA_W_04_14 VA_W_04_18 VA_W_04_22\nTR10083|c0_g2_i1_coral   6.328591  0.9650112   3.976631   6.009333\nTR10090|c0_g1_i1_coral   8.629897  2.4125279   5.302175   3.698051\nTR10090|c0_g2_i1_coral   2.876632  1.9300224   7.069566  10.169640\nTR10103|c0_g1_i1_coral   2.301306  3.3775391   3.534783   2.773538\nTR10108|c0_g1_i1_coral   3.451959  6.2725727  18.557612   7.858358\nTR10132|c0_g1_i1_coral  12.081856  2.4125279   0.000000   0.000000\n                       VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14\nTR10083|c0_g2_i1_coral  0.0000000   3.511099  11.297065   4.584168\nTR10090|c0_g1_i1_coral  0.0000000   1.003171  11.297065   0.000000\nTR10090|c0_g2_i1_coral  0.0000000   2.507928   6.778239   2.750501\nTR10103|c0_g1_i1_coral  0.7675164   3.009514   2.259413   7.334669\nTR10108|c0_g1_i1_coral  1.5350328   5.015856  35.020902   3.667335\nTR10132|c0_g1_i1_coral  2.3025492   8.025369   2.259413   8.251503\n                       VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18\nTR10083|c0_g2_i1_coral  2.3825360   0.000000  9.2975805   1.022439\nTR10090|c0_g1_i1_coral  0.7941787   1.585195  1.4303970   1.022439\nTR10090|c0_g2_i1_coral  0.7941787   0.000000  5.0063895   3.067318\nTR10103|c0_g1_i1_coral  2.3825360   3.962988  4.2911910   0.000000\nTR10108|c0_g1_i1_coral  6.3534292   4.755585  2.8607940   1.022439\nTR10132|c0_g1_i1_coral 27.0020742  23.777926  0.7151985   6.134635\n                       VA_W_07_22\nTR10083|c0_g2_i1_coral  0.8584103\nTR10090|c0_g1_i1_coral  6.0088720\nTR10090|c0_g2_i1_coral  2.5752309\nTR10103|c0_g1_i1_coral  1.7168206\nTR10108|c0_g1_i1_coral  1.7168206\nTR10132|c0_g1_i1_coral  7.7256926\n\u0026gt; genes4heatmap\u0026lt;-res[res$pvalue \u0026lt;0.05 \u0026amp; !is.na(res$pvalue),]\n\u0026gt; names(genes4heatmap)\n[1] \u0026quot;baseMean\u0026quot;       \u0026quot;log2FoldChange\u0026quot; \u0026quot;lfcSE\u0026quot;          \u0026quot;stat\u0026quot;          \n[5] \u0026quot;pvalue\u0026quot;         \u0026quot;padj\u0026quot;          \n\u0026gt; head(genes4heatmap)\nlog2 fold change (MLE): Origin_SymState_Temp RI_W_22 vs VA_W_22 \nWald test p-value: Origin_SymState_Temp RI_W_22 vs VA_W_22 \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE      stat\n                       \u0026lt;numeric\u0026gt;      \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\nTR10090|c0_g1_i1_coral   2.41467       -2.16024  1.066582  -2.02538\nTR10450|c0_g1_i1_coral   3.06295       -3.43891  1.358159  -2.53204\nTR10768|c0_g1_i1_coral   4.11391        1.64720  0.816317   2.01785\nTR10773|c0_g1_i1_coral   2.74300        3.89665  1.286283   3.02939\nTR11771|c0_g1_i1_coral   2.86558        2.26910  0.996534   2.27700\nTR11938|c0_g1_i2_coral   3.17991        1.92891  0.932250   2.06910\n                          pvalue      padj\n                       \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\nTR10090|c0_g1_i1_coral 0.0428281  0.996985\nTR10450|c0_g1_i1_coral 0.0113402  0.882510\nTR10768|c0_g1_i1_coral 0.0436070  0.996985\nTR10773|c0_g1_i1_coral 0.0024505  0.437812\nTR11771|c0_g1_i1_coral 0.0227864  0.989064\nTR11938|c0_g1_i2_coral 0.0385371  0.996985\n\u0026gt; dim(genes4heatmap)\n[1] 343   6\n\u0026gt; data4heatmap\u0026lt;-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]\n\u0026gt; dim(data4heatmap)\n[1] 343  81\n\u0026gt; head(data4heatmap)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14\nTR10090|c0_g1_i1_coral    0.00000    2.52837   1.146477   0.000000\nTR10450|c0_g1_i1_coral    0.00000    2.52837   0.000000   0.000000\nTR10768|c0_g1_i1_coral   10.84813    4.21395   4.585907   7.107215\nTR10773|c0_g1_i1_coral    0.00000    5.89953   2.292953   3.553607\nTR11771|c0_g1_i1_coral    0.00000    2.52837   3.439430   0.000000\nTR11938|c0_g1_i2_coral    0.00000    2.52837   0.000000   0.000000\n                       RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_22\nTR10090|c0_g1_i1_coral   13.09462   0.000000   0.000000   0.000000\nTR10450|c0_g1_i1_coral    0.00000   6.468236   0.000000   1.269319\nTR10768|c0_g1_i1_coral    0.00000   8.624315   1.235394   1.269319\nTR10773|c0_g1_i1_coral    0.00000   0.000000   1.235394   0.000000\nTR11771|c0_g1_i1_coral    0.00000   0.000000   1.235394   5.077275\nTR11938|c0_g1_i2_coral    0.00000   4.312157   2.470789   3.807956\n                       RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10090|c0_g1_i1_coral  2.0285120  4.0659932   6.219661  0.5373050\nTR10450|c0_g1_i1_coral  2.4342144  0.4065993  18.658984  0.2686525\nTR10768|c0_g1_i1_coral  0.8114048  4.8791918   0.000000  2.9551775\nTR10773|c0_g1_i1_coral  0.0000000  0.4065993   0.000000  0.0000000\nTR11771|c0_g1_i1_coral  1.6228096  4.0659932   0.000000  1.0746100\nTR11938|c0_g1_i2_coral  0.0000000  1.2197979   9.329492  4.2984400\n                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18\nTR10090|c0_g1_i1_coral   0.000000   0.000000    0.00000   1.963417\nTR10450|c0_g1_i1_coral  15.182145   6.352850    0.00000   1.308944\nTR10768|c0_g1_i1_coral   0.000000   1.588213    0.00000   4.581305\nTR10773|c0_g1_i1_coral   0.000000   0.000000    0.00000   0.000000\nTR11771|c0_g1_i1_coral   6.900975   1.588213   12.08711   4.581305\nTR11938|c0_g1_i2_coral   4.140585   3.176425    0.00000   1.963417\n                       RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22\nTR10090|c0_g1_i1_coral    0.00000  0.6838455  1.9782213    0.00000\nTR10450|c0_g1_i1_coral   42.00244  2.0515364  5.9346640    0.00000\nTR10768|c0_g1_i1_coral    0.00000  2.7353818  8.9019960   45.66098\nTR10773|c0_g1_i1_coral    0.00000  0.0000000  0.9891107    0.00000\nTR11771|c0_g1_i1_coral    0.00000  2.0515364  7.9128854    0.00000\nTR11938|c0_g1_i2_coral    0.00000  0.0000000  0.0000000    0.00000\n                       RI_W_01_14 RI_W_01_18 RI_W_01_22 RI_W_02_14\nTR10090|c0_g1_i1_coral    0.00000  11.625295   1.456442   0.000000\nTR10450|c0_g1_i1_coral    0.00000   0.000000   0.000000   4.027688\nTR10768|c0_g1_i1_coral   17.26245   7.522249   1.456442   0.000000\nTR10773|c0_g1_i1_coral    0.00000   7.522249  16.020867   0.000000\nTR11771|c0_g1_i1_coral    0.00000   2.051523   2.912885   0.000000\nTR11938|c0_g1_i2_coral    0.00000   4.103045   1.456442   2.013844\n                       RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18\nTR10090|c0_g1_i1_coral   1.956995   0.000000   4.039014   1.068873\nTR10450|c0_g1_i1_coral   0.000000   1.197822   1.154004   0.000000\nTR10768|c0_g1_i1_coral   4.892488   5.989108   4.616016   2.850329\nTR10773|c0_g1_i1_coral  14.677463   5.989108  21.926075  12.470189\nTR11771|c0_g1_i1_coral   5.870985  14.373858   4.616016   1.781456\nTR11938|c0_g1_i2_coral   2.935493  10.780394   4.039014   1.425164\n                       RI_W_03_22 RI_W_04_14 RI_W_04_18 RI_W_04_22\nTR10090|c0_g1_i1_coral  1.4697536    0.00000   3.437796     0.0000\nTR10450|c0_g1_i1_coral  0.0000000    0.00000   4.911138     0.0000\nTR10768|c0_g1_i1_coral  6.8588501    0.00000   2.946683    17.8865\nTR10773|c0_g1_i1_coral 26.9454824    0.00000   1.964455     0.0000\nTR11771|c0_g1_i1_coral  0.4899179    0.00000   1.964455     0.0000\nTR11938|c0_g1_i2_coral  3.4294250   12.32321   3.928910     0.0000\n                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14\nTR10090|c0_g1_i1_coral  0.8788181  7.7745767   0.000000   0.000000\nTR10450|c0_g1_i1_coral  1.3182272  0.9718221   0.000000   1.957360\nTR10768|c0_g1_i1_coral  5.7123177  0.9718221   3.842950   0.000000\nTR10773|c0_g1_i1_coral  5.2729086  1.9436442   0.000000   4.893399\nTR11771|c0_g1_i1_coral  4.3940905  1.9436442   5.123934   4.893399\nTR11938|c0_g1_i2_coral  2.6364543  2.9154662   5.123934   6.850759\n                       RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18\nTR10090|c0_g1_i1_coral   0.000000   4.333240  0.0000000   0.000000\nTR10450|c0_g1_i1_coral   1.280240   0.000000  0.6750084   2.013953\nTR10768|c0_g1_i1_coral   7.681439   5.416550  6.7500843   3.020929\nTR10773|c0_g1_i1_coral   2.560480   9.749789  1.3500169   2.013953\nTR11771|c0_g1_i1_coral   0.000000   3.249930  0.6750084   2.013953\nTR11938|c0_g1_i2_coral   0.000000   2.166620  1.3500169   8.055810\n                       RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10090|c0_g1_i1_coral   0.000000   6.825663   6.879404  1.6522753\nTR10450|c0_g1_i1_coral   0.000000   5.688053  15.724353  2.7537921\nTR10768|c0_g1_i1_coral   2.830176   1.137611   6.224223  3.8553090\nTR10773|c0_g1_i1_coral   0.000000   5.688053   3.603497  2.7537921\nTR11771|c0_g1_i1_coral  14.150880   3.412832   1.965544  0.5507584\nTR11938|c0_g1_i2_coral   2.830176   3.412832   3.931088  1.1015169\n                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14\nTR10090|c0_g1_i1_coral   0.000000   0.000000   0.000000   5.534635\nTR10450|c0_g1_i1_coral   0.000000   8.328238   0.000000   0.000000\nTR10768|c0_g1_i1_coral   1.417949   0.000000   3.904196   4.842806\nTR10773|c0_g1_i1_coral   1.417949   0.000000   5.856295   0.000000\nTR11771|c0_g1_i1_coral   0.000000   0.000000   0.000000   9.685611\nTR11938|c0_g1_i2_coral   4.253846   0.000000   0.000000   2.075488\n                       VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18\nTR10090|c0_g1_i1_coral   1.980640  7.5143298   0.000000   4.154433\nTR10450|c0_g1_i1_coral   1.320426  2.7324836   0.000000   0.000000\nTR10768|c0_g1_i1_coral   2.640853  2.0493627   1.631661   1.246330\nTR10773|c0_g1_i1_coral   1.320426  0.6831209   0.000000   6.647093\nTR11771|c0_g1_i1_coral   1.980640  0.0000000   0.000000   2.908103\nTR11938|c0_g1_i2_coral   3.301066  3.4156045   2.447491   4.985320\n                       VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10090|c0_g1_i1_coral   3.342246   4.512607  0.9973314   6.873047\nTR10450|c0_g1_i1_coral   1.909855   1.611645  1.4959971   0.000000\nTR10768|c0_g1_i1_coral   0.000000   3.867949  2.4933284   0.000000\nTR10773|c0_g1_i1_coral   7.639419   1.289316  0.9973314   0.000000\nTR11771|c0_g1_i1_coral   0.000000   2.578632  2.9919941   5.154785\nTR11938|c0_g1_i2_coral   7.161955   2.256303  3.9893255   1.718262\n                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14\nTR10090|c0_g1_i1_coral   7.283093   0.000000  0.4510524   0.000000\nTR10450|c0_g1_i1_coral   1.820773   0.000000  4.5105239   0.000000\nTR10768|c0_g1_i1_coral   6.676168   2.376767  4.5105239   2.151109\nTR10773|c0_g1_i1_coral   2.427698   0.000000  4.9615763   1.434073\nTR11771|c0_g1_i1_coral   3.641546   5.545791  1.3531572   4.302218\nTR11938|c0_g1_i2_coral   7.283093   0.000000  3.6084191   7.170364\n                       VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18\nTR10090|c0_g1_i1_coral   3.438839  1.9870109   3.675126   1.074698\nTR10450|c0_g1_i1_coral   1.375536  0.9935055   0.000000   5.373492\nTR10768|c0_g1_i1_coral   1.375536  2.9805164   1.837563   8.060239\nTR10773|c0_g1_i1_coral   2.751072  5.9610327   5.512689   1.074698\nTR11771|c0_g1_i1_coral   0.000000  4.9675273   5.512689   1.612048\nTR11938|c0_g1_i2_coral   8.253215 10.9285600   2.756344   3.224095\n                       VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_18\nTR10090|c0_g1_i1_coral   6.346543   0.624967   4.969028    0.00000\nTR10450|c0_g1_i1_coral   0.000000   1.874901   4.969028   31.88670\nTR10768|c0_g1_i1_coral   1.586636   6.249670   2.839445    0.00000\nTR10773|c0_g1_i1_coral   0.000000   3.749802   1.419722    0.00000\nTR11771|c0_g1_i1_coral   1.586636   3.124835   7.098612   21.25780\nTR11938|c0_g1_i2_coral   1.586636   6.874637   4.259167   23.38358\n                       VA_W_03_22 VA_W_04_14 VA_W_04_18 VA_W_04_22\nTR10090|c0_g1_i1_coral   8.629897  2.4125279   5.302175  3.6980510\nTR10450|c0_g1_i1_coral   1.150653  0.9650112   0.000000  1.8490255\nTR10768|c0_g1_i1_coral   1.725979  1.4475168   3.976631  2.3112819\nTR10773|c0_g1_i1_coral   2.301306  0.9650112   0.000000  0.4622564\nTR11771|c0_g1_i1_coral   1.725979  0.0000000   0.000000  0.4622564\nTR11938|c0_g1_i2_coral   1.725979  4.8250559   1.767392  0.4622564\n                       VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14\nTR10090|c0_g1_i1_coral          0  1.0031712  11.297065   0.000000\nTR10450|c0_g1_i1_coral          0  0.0000000  12.426772   1.833667\nTR10768|c0_g1_i1_coral          0  6.5206127   0.000000   2.750501\nTR10773|c0_g1_i1_coral          0  0.0000000   0.000000   0.000000\nTR11771|c0_g1_i1_coral          0  0.5015856   1.129707   1.833667\nTR11938|c0_g1_i2_coral          0  2.5079280   0.000000   1.833667\n                       VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18\nTR10090|c0_g1_i1_coral  0.7941787  1.5851950  1.4303970   1.022439\nTR10450|c0_g1_i1_coral  1.5883573  3.9629876  2.1455955   0.000000\nTR10768|c0_g1_i1_coral  5.5592506  0.0000000  2.8607940   8.179514\nTR10773|c0_g1_i1_coral  1.5883573  0.0000000  0.0000000   0.000000\nTR11771|c0_g1_i1_coral  2.3825360  0.7925975  0.7151985   4.089757\nTR11938|c0_g1_i2_coral  0.7941787  2.3777926  1.4303970   0.000000\n                       VA_W_07_22\nTR10090|c0_g1_i1_coral  6.0088720\nTR10450|c0_g1_i1_coral  3.4336412\nTR10768|c0_g1_i1_coral  3.4336412\nTR10773|c0_g1_i1_coral  0.0000000\nTR11771|c0_g1_i1_coral  2.5752309\nTR11938|c0_g1_i2_coral  0.8584103\n\u0026gt; temp = as.matrix(rowMeans(data4heatmap))\n\u0026gt; head(temp)\n                           [,1]\nTR10090|c0_g1_i1_coral 2.414669\nTR10450|c0_g1_i1_coral 3.062953\nTR10768|c0_g1_i1_coral 4.113910\nTR10773|c0_g1_i1_coral 2.742998\nTR11771|c0_g1_i1_coral 2.865582\nTR11938|c0_g1_i2_coral 3.179912\n\u0026gt; scaledmatrix\u0026lt;-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)\n\u0026gt; data4heatmapscaled = data4heatmap/scaledmatrix\n\u0026gt; head(data4heatmapscaled)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14\nTR10090|c0_g1_i1_coral   0.000000  1.0470877  0.4747967   0.000000\nTR10450|c0_g1_i1_coral   0.000000  0.8254680  0.0000000   0.000000\nTR10768|c0_g1_i1_coral   2.636938  1.0243174  1.1147318   1.727606\nTR10773|c0_g1_i1_coral   0.000000  2.1507597  0.8359295   1.295519\nTR11771|c0_g1_i1_coral   0.000000  0.8823235  1.2002554   0.000000\nTR11938|c0_g1_i2_coral   0.000000  0.7951069  0.0000000   0.000000\n                       RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_22\nTR10090|c0_g1_i1_coral   5.422948   0.000000  0.0000000  0.0000000\nTR10450|c0_g1_i1_coral   0.000000   2.111764  0.0000000  0.4144100\nTR10768|c0_g1_i1_coral   0.000000   2.096379  0.3002969  0.3085431\nTR10773|c0_g1_i1_coral   0.000000   0.000000  0.4503810  0.0000000\nTR11771|c0_g1_i1_coral   0.000000   0.000000  0.4311147  1.7718129\nTR11938|c0_g1_i2_coral   0.000000   1.356062  0.7769990  1.1975035\n                       RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10090|c0_g1_i1_coral  0.8400788  1.6838720   2.575782 0.22251707\nTR10450|c0_g1_i1_coral  0.7947279  0.1327475   6.091827 0.08771028\nTR10768|c0_g1_i1_coral  0.1972344  1.1860229   0.000000 0.71833786\nTR10773|c0_g1_i1_coral  0.0000000  0.1482317   0.000000 0.00000000\nTR11771|c0_g1_i1_coral  0.5663107  1.4189067   0.000000 0.37500589\nTR11938|c0_g1_i2_coral  0.0000000  0.3835949   2.933884 1.35174810\n                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18\nTR10090|c0_g1_i1_coral   0.000000  0.0000000   0.000000  0.8131205\nTR10450|c0_g1_i1_coral   4.956701  2.0740929   0.000000  0.4273471\nTR10768|c0_g1_i1_coral   0.000000  0.3860591   0.000000  1.1136133\nTR10773|c0_g1_i1_coral   0.000000  0.0000000   0.000000  0.0000000\nTR11771|c0_g1_i1_coral   2.408228  0.5542374   4.218031  1.5987348\nTR11938|c0_g1_i2_coral   1.302107  0.9989034   0.000000  0.6174437\n                       RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22\nTR10090|c0_g1_i1_coral    0.00000  0.2832047  0.8192517    0.00000\nTR10450|c0_g1_i1_coral   13.71305  0.6697903  1.9375626    0.00000\nTR10768|c0_g1_i1_coral    0.00000  0.6649104  2.1638771   11.09917\nTR10773|c0_g1_i1_coral    0.00000  0.0000000  0.3605947    0.00000\nTR11771|c0_g1_i1_coral    0.00000  0.7159232  2.7613540    0.00000\nTR11938|c0_g1_i2_coral    0.00000  0.0000000  0.0000000    0.00000\n                       RI_W_01_14 RI_W_01_18 RI_W_01_22 RI_W_02_14\nTR10090|c0_g1_i1_coral   0.000000  4.8144470  0.6031645  0.0000000\nTR10450|c0_g1_i1_coral   0.000000  0.0000000  0.0000000  1.3149687\nTR10768|c0_g1_i1_coral   4.196117  1.8284914  0.3540287  0.0000000\nTR10773|c0_g1_i1_coral   0.000000  2.7423457  5.8406405  0.0000000\nTR11771|c0_g1_i1_coral   0.000000  0.7159184  1.0165074  0.0000000\nTR11938|c0_g1_i2_coral   0.000000  1.2903015  0.4580134  0.6333018\n                       RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18\nTR10090|c0_g1_i1_coral  0.8104611  0.0000000  1.6726989  0.4426584\nTR10450|c0_g1_i1_coral  0.0000000  0.3910675  0.3767618  0.0000000\nTR10768|c0_g1_i1_coral  1.1892549  1.4558187  1.1220507  0.6928515\nTR10773|c0_g1_i1_coral  5.3508832  2.1834164  7.9934703  4.5461892\nTR11771|c0_g1_i1_coral  2.0487936  5.0160350  1.6108477  0.6216733\nTR11938|c0_g1_i2_coral  0.9231364  3.3901547  1.2701653  0.4481773\n                       RI_W_03_22 RI_W_04_14 RI_W_04_18 RI_W_04_22\nTR10090|c0_g1_i1_coral  0.6086771   0.000000  1.4237135    0.00000\nTR10450|c0_g1_i1_coral  0.0000000   0.000000  1.6033994    0.00000\nTR10768|c0_g1_i1_coral  1.6672338   0.000000  0.7162729    4.34781\nTR10773|c0_g1_i1_coral  9.8233684   0.000000  0.7161707    0.00000\nTR11771|c0_g1_i1_coral  0.1709663   0.000000  0.6855345    0.00000\nTR11938|c0_g1_i2_coral  1.0784654   3.875331  1.2355405    0.00000\n                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14\nTR10090|c0_g1_i1_coral  0.3639498  3.2197281  0.0000000  0.0000000\nTR10450|c0_g1_i1_coral  0.4303778  0.3172827  0.0000000  0.6390432\nTR10768|c0_g1_i1_coral  1.3885373  0.2362283  0.9341357  0.0000000\nTR10773|c0_g1_i1_coral  1.9223157  0.7085838  0.0000000  1.7839600\nTR11771|c0_g1_i1_coral  1.5334026  0.6782721  1.7880955  1.7076460\nTR11938|c0_g1_i2_coral  0.8290966  0.9168387  1.6113445  2.1543863\n                       RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18\nTR10090|c0_g1_i1_coral  0.0000000  1.7945483  0.0000000  0.0000000\nTR10450|c0_g1_i1_coral  0.4179756  0.0000000  0.2203783  0.6575198\nTR10768|c0_g1_i1_coral  1.8671868  1.3166426  1.6407952  0.7343205\nTR10773|c0_g1_i1_coral  0.9334602  3.5544277  0.4921683  0.7342157\nTR11771|c0_g1_i1_coral  0.0000000  1.1341257  0.2355572  0.7028076\nTR11938|c0_g1_i2_coral  0.0000000  0.6813458  0.4245453  2.5333437\n                       RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10090|c0_g1_i1_coral  0.0000000  2.8267494  2.8490054  0.6842658\nTR10450|c0_g1_i1_coral  0.0000000  1.8570484  5.1337224  0.8990643\nTR10768|c0_g1_i1_coral  0.6879528  0.2765278  1.5129700  0.9371398\nTR10773|c0_g1_i1_coral  0.0000000  2.0736626  1.3137075  1.0039351\nTR11771|c0_g1_i1_coral  4.9382226  1.1909734  0.6859145  0.1921978\nTR11938|c0_g1_i2_coral  0.8900171  1.0732472  1.2362254  0.3463985\n                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14\nTR10090|c0_g1_i1_coral  0.0000000   0.000000  0.0000000  2.2920887\nTR10450|c0_g1_i1_coral  0.0000000   2.719022  0.0000000  0.0000000\nTR10768|c0_g1_i1_coral  0.3446718   0.000000  0.9490232  1.1771782\nTR10773|c0_g1_i1_coral  0.5169339   0.000000  2.1349975  0.0000000\nTR11771|c0_g1_i1_coral  0.0000000   0.000000  0.0000000  3.3799808\nTR11938|c0_g1_i2_coral  1.3377245   0.000000  0.0000000  0.6526873\n                       VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18\nTR10090|c0_g1_i1_coral  0.8202531  3.1119506  0.0000000  1.7204981\nTR10450|c0_g1_i1_coral  0.4310958  0.8921074  0.0000000  0.0000000\nTR10768|c0_g1_i1_coral  0.6419325  0.4981544  0.3966204  0.3029551\nTR10773|c0_g1_i1_coral  0.4813807  0.2490417  0.0000000  2.4232946\nTR11771|c0_g1_i1_coral  0.6911824  0.0000000  0.0000000  1.0148387\nTR11938|c0_g1_i2_coral  1.0380998  1.0741192  0.7696726  1.5677539\n                       VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10090|c0_g1_i1_coral  1.3841426  1.8688306  0.4130303  2.8463728\nTR10450|c0_g1_i1_coral  0.6235337  0.5261736  0.4884165  0.0000000\nTR10768|c0_g1_i1_coral  0.0000000  0.9402122  0.6060726  0.0000000\nTR10773|c0_g1_i1_coral  2.7850616  0.4700390  0.3635917  0.0000000\nTR11771|c0_g1_i1_coral  0.0000000  0.8998635  1.0441141  1.7988618\nTR11938|c0_g1_i2_coral  2.2522495  0.7095490  1.2545396  0.5403489\n                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14\nTR10090|c0_g1_i1_coral  3.0161871  0.0000000  0.1867968  0.0000000\nTR10450|c0_g1_i1_coral  0.5944502  0.0000000  1.4726061  0.0000000\nTR10768|c0_g1_i1_coral  1.6228279  0.5777393  1.0964079  0.5228868\nTR10773|c0_g1_i1_coral  0.8850525  0.0000000  1.8088149  0.5228121\nTR11771|c0_g1_i1_coral  1.2707878  1.9353107  0.4722103  1.5013421\nTR11938|c0_g1_i2_coral  2.2903441  0.0000000  1.1347544  2.2548938\n                       VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18\nTR10090|c0_g1_i1_coral  1.4241454  0.8228917  1.5220000  0.4450708\nTR10450|c0_g1_i1_coral  0.4490880  0.3243619  0.0000000  1.7543500\nTR10768|c0_g1_i1_coral  0.3343621  0.7244972  0.4466706  1.9592646\nTR10773|c0_g1_i1_coral  1.0029432  2.1731814  2.0097310  0.3917970\nTR11771|c0_g1_i1_coral  0.0000000  1.7335145  1.9237590  0.5625552\nTR11938|c0_g1_i2_coral  2.5954223  3.4367492  0.8667989  1.0138946\n                       VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_18\nTR10090|c0_g1_i1_coral  2.6283285  0.2588210  2.0578510   0.000000\nTR10450|c0_g1_i1_coral  0.0000000  0.6121219  1.6222997  10.410443\nTR10768|c0_g1_i1_coral  0.3856758  1.5191556  0.6902058   0.000000\nTR10773|c0_g1_i1_coral  0.0000000  1.3670450  0.5175805   0.000000\nTR11771|c0_g1_i1_coral  0.5536871  1.0904714  2.4771976   7.418320\nTR11938|c0_g1_i2_coral  0.4989558  2.1618953  1.3393978   7.353531\n                       VA_W_03_22 VA_W_04_14 VA_W_04_18 VA_W_04_22\nTR10090|c0_g1_i1_coral  3.5739467  0.9991134  2.1958187  1.5314941\nTR10450|c0_g1_i1_coral  0.3756678  0.3150590  0.0000000  0.6036740\nTR10768|c0_g1_i1_coral  0.4195472  0.3518591  0.9666305  0.5618212\nTR10773|c0_g1_i1_coral  0.8389746  0.3518089  0.0000000  0.1685223\nTR11771|c0_g1_i1_coral  0.6023138  0.0000000  0.0000000  0.1613133\nTR11938|c0_g1_i2_coral  0.5427759  1.5173552  0.5557989  0.1453677\n                       VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14\nTR10090|c0_g1_i1_coral          0  0.4154488  4.6785155  0.0000000\nTR10450|c0_g1_i1_coral          0  0.0000000  4.0571206  0.5986599\nTR10768|c0_g1_i1_coral          0  1.5850158  0.0000000  0.6685856\nTR10773|c0_g1_i1_coral          0  0.0000000  0.0000000  0.0000000\nTR11771|c0_g1_i1_coral          0  0.1750380  0.3942329  0.6398936\nTR11938|c0_g1_i2_coral          0  0.7886784  0.0000000  0.5766409\n                       VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18\nTR10090|c0_g1_i1_coral  0.3288976  0.6564855  0.5923782  0.4234284\nTR10450|c0_g1_i1_coral  0.5185705  1.2938452  0.7004989  0.0000000\nTR10768|c0_g1_i1_coral  1.3513301  0.0000000  0.6953953  1.9882577\nTR10773|c0_g1_i1_coral  0.5790588  0.0000000  0.0000000  0.0000000\nTR11771|c0_g1_i1_coral  0.8314319  0.2765922  0.2495823  1.4271996\nTR11938|c0_g1_i2_coral  0.2497486  0.7477542  0.4498228  0.0000000\n                       VA_W_07_22\nTR10090|c0_g1_i1_coral  2.4884871\nTR10450|c0_g1_i1_coral  1.1210230\nTR10768|c0_g1_i1_coral  0.8346417\nTR10773|c0_g1_i1_coral  0.0000000\nTR11771|c0_g1_i1_coral  0.8986765\nTR11938|c0_g1_i2_coral  0.2699478\n\u0026gt; \n\u0026gt; dim(data4heatmapscaled)\n[1] 343  81\n\u0026gt; \n\u0026gt; pairs.breaks \u0026lt;- seq(0, 3.0, by=0.1)\n\u0026gt; length(pairs.breaks)\n[1] 31\n\u0026gt; mycol \u0026lt;- colorpanel(n=30, low=\u0026quot;black\u0026quot;, high=\u0026quot;yellow\u0026quot;)\n\u0026gt; pdf(file=\u0026quot;XXXXXX_byrow.pdf\u0026quot;,14,7)\n\u0026gt; heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(\u0026quot;row\u0026quot;), scale=\u0026quot;none\u0026quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u0026quot;none\u0026quot;, symkey = F, density.info = \u0026quot;density\u0026quot;, colsep=c(24), sepcolor=c(\u0026quot;white\u0026quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\n\u0026gt; dev.off()\nRStudioGD \n        2 \n\u0026gt; pdf(file=\u0026quot;XXXXXX_bycolumn.pdf\u0026quot;,14,7)\n\u0026gt; heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(\u0026quot;col\u0026quot;), scale=\u0026quot;none\u0026quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u0026quot;none\u0026quot;, symkey = F, density.info = \u0026quot;density\u0026quot;, colsep=c(24), sepcolor=c(\u0026quot;white\u0026quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\n\u0026gt; dev.off()\nRStudioGD \n        2\n\u0026gt; plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u0026quot;Origin\u0026quot;)\n\u0026gt; prop.nullv3 \u0026lt;- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100*mean(x==0))\n\u0026gt; pdf(file=\u0026quot;ResNullCounts.pdf\u0026quot;,14, 14)\n\u0026gt; barplot(prop.nullv3, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u0026gt; dev.off()\nRStudioGD \n        2 \n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1- Download the DESeq2Script_advbioinf.R script and work through on the shared djbFullCounts_summed.txt file\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day11/DESeq2Script_advbioinf.R\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ecopy this and the summed.txt file to the Genomics folder\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003emake sure to have downloaded these packages already\u003c/span\u003e\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003elibrary(DESeq2)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003elibrary(gplots)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003evignette(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eDESeq2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003esetwd(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/Users/katiecrider/Desktop/Genomics\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e The drag and drop from finder works in R, too.\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e conds\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-data.frame(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e=names(countsTable))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e conds\u003cspan class=\"pl-smi\"\u003e$Origin\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(substring(conds\u003cspan class=\"pl-smi\"\u003e$Sample\u003c/span\u003e,1,2))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e conds\u003cspan class=\"pl-smi\"\u003e$SymState\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(substring(conds\u003cspan class=\"pl-smi\"\u003e$Sample\u003c/span\u003e,4,4))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e conds\u003cspan class=\"pl-smi\"\u003e$Temp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(substring(conds\u003cspan class=\"pl-smi\"\u003e$Sample\u003c/span\u003e,9,10))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(paste(conds\u003cspan class=\"pl-smi\"\u003e$Origin\u003c/span\u003e,conds\u003cspan class=\"pl-smi\"\u003e$SymState\u003c/span\u003e, sep=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(paste(conds\u003cspan class=\"pl-smi\"\u003e$Origin\u003c/span\u003e,conds\u003cspan class=\"pl-smi\"\u003e$SymState\u003c/span\u003e,conds\u003cspan class=\"pl-smi\"\u003e$Temp\u003c/span\u003e, sep=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e conds\n       Sample Origin SymState Temp Origin_SymState Origin_SymState_Temp\n1  RI_B_01_14     RI        B   14            RI_B              RI_B_14\n2  RI_B_01_18     RI        B   18            RI_B              RI_B_18\n3  RI_B_01_22     RI        B   22            RI_B              RI_B_22\n4  RI_B_02_14     RI        B   14            RI_B              RI_B_14\n5  RI_B_02_18     RI        B   18            RI_B              RI_B_18\n6  RI_B_02_22     RI        B   22            RI_B              RI_B_22\n7  RI_B_03_14     RI        B   14            RI_B              RI_B_14\n8  RI_B_03_22     RI        B   22            RI_B              RI_B_22\n9  RI_B_04_14     RI        B   14            RI_B              RI_B_14\n10 RI_B_04_18     RI        B   18            RI_B              RI_B_18\n11 RI_B_04_22     RI        B   22            RI_B              RI_B_22\n12 RI_B_05_14     RI        B   14            RI_B              RI_B_14\n13 RI_B_05_18     RI        B   18            RI_B              RI_B_18\n14 RI_B_05_22     RI        B   22            RI_B              RI_B_22\n15 RI_B_06_14     RI        B   14            RI_B              RI_B_14\n16 RI_B_06_18     RI        B   18            RI_B              RI_B_18\n17 RI_B_06_22     RI        B   22            RI_B              RI_B_22\n18 RI_B_07_14     RI        B   14            RI_B              RI_B_14\n19 RI_B_07_18     RI        B   18            RI_B              RI_B_18\n20 RI_B_07_22     RI        B   22            RI_B              RI_B_22\n21 RI_W_01_14     RI        W   14            RI_W              RI_W_14\n22 RI_W_01_18     RI        W   18            RI_W              RI_W_18\n23 RI_W_01_22     RI        W   22            RI_W              RI_W_22\n24 RI_W_02_14     RI        W   14            RI_W              RI_W_14\n25 RI_W_02_18     RI        W   18            RI_W              RI_W_18\n26 RI_W_02_22     RI        W   22            RI_W              RI_W_22\n27 RI_W_03_14     RI        W   14            RI_W              RI_W_14\n28 RI_W_03_18     RI        W   18            RI_W              RI_W_18\n29 RI_W_03_22     RI        W   22            RI_W              RI_W_22\n30 RI_W_04_14     RI        W   14            RI_W              RI_W_14\n31 RI_W_04_18     RI        W   18            RI_W              RI_W_18\n32 RI_W_04_22     RI        W   22            RI_W              RI_W_22\n33 RI_W_05_14     RI        W   14            RI_W              RI_W_14\n34 RI_W_05_18     RI        W   18            RI_W              RI_W_18\n35 RI_W_05_22     RI        W   22            RI_W              RI_W_22\n36 RI_W_06_14     RI        W   14            RI_W              RI_W_14\n37 RI_W_06_18     RI        W   18            RI_W              RI_W_18\n38 RI_W_06_22     RI        W   22            RI_W              RI_W_22\n39 RI_W_07_14     RI        W   14            RI_W              RI_W_14\n40 RI_W_07_18     RI        W   18            RI_W              RI_W_18\n41 RI_W_07_22     RI        W   22            RI_W              RI_W_22\n42 VA_B_01_14     VA        B   14            VA_B              VA_B_14\n43 VA_B_01_18     VA        B   18            VA_B              VA_B_18\n44 VA_B_01_22     VA        B   22            VA_B              VA_B_22\n45 VA_B_02_14     VA        B   14            VA_B              VA_B_14\n46 VA_B_02_18     VA        B   18            VA_B              VA_B_18\n47 VA_B_02_22     VA        B   22            VA_B              VA_B_22\n48 VA_B_03_14     VA        B   14            VA_B              VA_B_14\n49 VA_B_03_18     VA        B   18            VA_B              VA_B_18\n50 VA_B_03_22     VA        B   22            VA_B              VA_B_22\n51 VA_B_04_14     VA        B   14            VA_B              VA_B_14\n52 VA_B_04_18     VA        B   18            VA_B              VA_B_18\n53 VA_B_04_22     VA        B   22            VA_B              VA_B_22\n54 VA_B_05_14     VA        B   14            VA_B              VA_B_14\n55 VA_B_05_18     VA        B   18            VA_B              VA_B_18\n56 VA_B_05_22     VA        B   22            VA_B              VA_B_22\n57 VA_B_06_14     VA        B   14            VA_B              VA_B_14\n58 VA_B_06_18     VA        B   18            VA_B              VA_B_18\n59 VA_B_06_22     VA        B   22            VA_B              VA_B_22\n60 VA_B_07_14     VA        B   14            VA_B              VA_B_14\n61 VA_B_07_18     VA        B   18            VA_B              VA_B_18\n62 VA_B_07_22     VA        B   22            VA_B              VA_B_22\n63 VA_W_01_14     VA        W   14            VA_W              VA_W_14\n64 VA_W_01_18     VA        W   18            VA_W              VA_W_18\n65 VA_W_02_14     VA        W   14            VA_W              VA_W_14\n66 VA_W_02_18     VA        W   18            VA_W              VA_W_18\n67 VA_W_03_14     VA        W   14            VA_W              VA_W_14\n68 VA_W_03_18     VA        W   18            VA_W              VA_W_18\n69 VA_W_03_22     VA        W   22            VA_W              VA_W_22\n70 VA_W_04_14     VA        W   14            VA_W              VA_W_14\n71 VA_W_04_18     VA        W   18            VA_W              VA_W_18\n72 VA_W_04_22     VA        W   22            VA_W              VA_W_22\n73 VA_W_05_14     VA        W   14            VA_W              VA_W_14\n74 VA_W_05_18     VA        W   18            VA_W              VA_W_18\n75 VA_W_05_22     VA        W   22            VA_W              VA_W_22\n76 VA_W_06_14     VA        W   14            VA_W              VA_W_14\n77 VA_W_06_18     VA        W   18            VA_W              VA_W_18\n78 VA_W_06_22     VA        W   22            VA_W              VA_W_22\n79 VA_W_07_14     VA        W   14            VA_W              VA_W_14\n80 VA_W_07_18     VA        W   18            VA_W              VA_W_18\n81 VA_W_07_22     VA        W   22            VA_W              VA_W_22\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e prop.null \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- apply(countsTable, 2, function(x) 100\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003emean(x==0))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e barplot(prop.null, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOverallNullCounts.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,14, 14)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e barplot(prop.null, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\nRStudioGD \n        2 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e how many genes have mean count \u0026gt;=3 \u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e means = apply(countsTable, 1, mean)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e table(means\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e=3)\n\nTRUE \n7543 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e countsTable\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-countsTable[means\u003cspan class=\"pl-k\"\u003e\u0026gt;=\u003c/span\u003e3,]\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dim(countsTable)\n[1] 7543   81\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e prop.nullv2 \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- apply(countsTable, 2, function(x) 100\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003emean(x==0))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMean3NullCounts.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,14, 14)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e barplot(prop.nullv2, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\nRStudioGD \n        2 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003emake count data set\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dds \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e Origin_SymState_Temp)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dds \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- DESeq(dds)\nestimating size factors\nestimating dispersions\ngene-wise dispersion estimates\nmean-dispersion relationship\nfinal dispersion estimates\nfitting model and testing\n-- replacing outliers and refitting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e 8 genes\n-- DESeq argument \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eminReplicatesForReplace\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e = 7 \n-- original counts are preserved \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e counts(dds)\nestimating dispersions\nfitting model and testing\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etransform counts to variance stablized counts\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e vstCounts\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-varianceStabilizingTransformation(dds)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eplot cluster dendrogram across samples\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dists\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-dist(t(assay(vstCounts)))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plot(hclust(dists))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plotPCA(vstCounts, intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plotPCA(vstCounts, intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSymState\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plotPCA(vstCounts, intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plotPCA(vstCounts, intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plotPCA(vstCounts, intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e resultsNames(dds)\n [1] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIntercept\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e                              \n [2] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp_RI_B_18_vs_RI_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n [3] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp_RI_B_22_vs_RI_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n [4] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp_RI_W_14_vs_RI_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n [5] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp_RI_W_18_vs_RI_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n [6] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp_RI_W_22_vs_RI_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n [7] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp_VA_B_14_vs_RI_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n [8] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp_VA_B_18_vs_RI_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n [9] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp_VA_B_22_vs_RI_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n[10] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp_VA_W_14_vs_RI_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n[11] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp_VA_W_18_vs_RI_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n[12] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp_VA_W_22_vs_RI_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dim(dds)\n[1] 7543   81\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e##Figure out which contrast you want to examine (i.e. which two groups do you want to compare)\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e res \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_B_18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_B_18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(res)\nlog2 fold change (MLE): Origin_SymState_Temp RI_B_18 vs VA_B_18 \nWald \u003cspan class=\"pl-c1\"\u003etest\u003c/span\u003e p-value: Origin_SymState_Temp RI_B_18 vs VA_B_18 \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE      stat\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e      \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   5.92149      -0.124352  0.881723 -0.141032\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.41467       0.403769  0.959279  0.420909\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.75861      -0.835423  0.865239 -0.965540\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.98376      -0.801464  0.848308 -0.944780\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.73440      -0.526701  0.837420 -0.628957\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   8.52630       1.071572  1.160128  0.923667\n                          pvalue      padj\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.887844  0.996375\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.673822  0.990719\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.334274  0.964876\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.344771  0.967913\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.529377  0.987729\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.355660  0.967913\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e summary(res)\n\nout of 7542 with nonzero total \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e count\nadjusted p-value \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1\nLFC \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e 0 (up)       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 3, 0.04%\nLFC \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0 (down)     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 5, 0.066%\noutliers [1]       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 4, 0.053%\nlow counts [2]     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 1, 0.013%\n(mean count \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 1)\n[1] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecooksCutoff\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n[2] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eindependentFiltering\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e res \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_W_18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_W_18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e summary(res)\n\nout of 7542 with nonzero total \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e count\nadjusted p-value \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1\nLFC \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e 0 (up)       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 22, 0.29%\nLFC \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0 (down)     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 38, 0.5%\noutliers [1]       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 4, 0.053%\nlow counts [2]     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 3801, 50%\n(mean count \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 6)\n[1] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecooksCutoff\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n[2] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eindependentFiltering\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e res \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_B_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e summary(res)\n\nout of 7542 with nonzero total \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e count\nadjusted p-value \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1\nLFC \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e 0 (up)       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 8, 0.11%\nLFC \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0 (down)     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 4, 0.053%\noutliers [1]       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 4, 0.053%\nlow counts [2]     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 1, 0.013%\n(mean count \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 1)\n[1] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecooksCutoff\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n[2] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eindependentFiltering\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e res \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_W_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_W_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e summary(res)\n\nout of 7542 with nonzero total \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e count\nadjusted p-value \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1\nLFC \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e 0 (up)       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 10, 0.13%\nLFC \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0 (down)     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 3, 0.04%\noutliers [1]       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 4, 0.053%\nlow counts [2]     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 5263, 70%\n(mean count \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 8)\n[1] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecooksCutoff\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n[2] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eindependentFiltering\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e res \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_B_22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_B_22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e summary(res)\n\nout of 7542 with nonzero total \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e count\nadjusted p-value \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1\nLFC \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e 0 (up)       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 8, 0.11%\nLFC \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0 (down)     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 2, 0.027%\noutliers [1]       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 4, 0.053%\nlow counts [2]     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 1, 0.013%\n(mean count \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 1)\n[1] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecooksCutoff\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n[2] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eindependentFiltering\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e res \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_W_22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_W_22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e summary(res)\n\nout of 7542 with nonzero total \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e count\nadjusted p-value \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1\nLFC \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e 0 (up)       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 11, 0.15%\nLFC \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0 (down)     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 5, 0.066%\noutliers [1]       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 4, 0.053%\nlow counts [2]     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 146, 1.9%\n(mean count \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 2)\n[1] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecooksCutoff\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n[2] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eindependentFiltering\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(res[order(res\u003cspan class=\"pl-smi\"\u003e$padj\u003c/span\u003e),])\nlog2 fold change (MLE): Origin_SymState_Temp RI_W_22 vs VA_W_22 \nWald \u003cspan class=\"pl-c1\"\u003etest\u003c/span\u003e p-value: Origin_SymState_Temp RI_W_22 vs VA_W_22 \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE      stat\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e      \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR42879\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  15.41058       17.05753  3.137478   5.43670\nTR13286\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  32.61026       -3.38044  0.648393  -5.21357\nTR47617\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.74656       19.72616  3.835159   5.14351\nTR8327\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g3_i1_coral    8.73277       19.73384  4.063853   4.85594\nTR29560\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  12.79556        2.74246  0.651668   4.20838\nTR58072\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  53.85583        1.38039  0.330572   4.17576\n                            pvalue        padj\n                         \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e   \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR42879\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral 5.42760e-08 0.000401263\nTR13286\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral 1.85241e-07 0.000664526\nTR47617\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 2.69658e-07 0.000664526\nTR8327\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g3_i1_coral  1.19814e-06 0.002214469\nTR29560\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 2.57208e-05 0.036593740\nTR58072\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral 2.96987e-05 0.036593740\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plotCounts(dds, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTR2367|c0_g1_i1_coral\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ecount the number of significantly differentially expressed genes\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e sum(res\u003cspan class=\"pl-smi\"\u003e$padj\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.3, na.rm =T)\n[1] 30\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e sum(res\u003cspan class=\"pl-smi\"\u003e$padj\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.2, na.rm =T)\n[1] 20\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e sum(res\u003cspan class=\"pl-smi\"\u003e$padj\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1, na.rm =T)\n[1] 16\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e sum(res\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.05, na.rm =T)\n[1] 343\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e summary(res)\n\nout of 7542 with nonzero total \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e count\nadjusted p-value \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1\nLFC \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e 0 (up)       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 11, 0.15%\nLFC \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0 (down)     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 5, 0.066%\noutliers [1]       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 4, 0.053%\nlow counts [2]     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 146, 1.9%\n(mean count \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 2)\n[1] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecooksCutoff\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n[2] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eindependentFiltering\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e scaledcounts = counts(dds, normalized=T)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(scaledcounts)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0    0.84279   2.292953   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0    2.52837   1.146477   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0    3.37116   2.292953   3.553607\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0    0.84279   4.585907   0.000000\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0    1.68558   4.585907   0.000000\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0    5.89953   1.146477  17.768037\n                       RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.273656   4.312157  12.353943   1.269319\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  13.094624   0.000000   0.000000   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000   0.000000   0.000000   3.807956\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   8.624315   1.235394   3.807956\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.547312   0.000000  13.589337  15.231824\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  49.104839   0.000000   6.176972   2.538637\n                       RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  12.576774  8.1319863   0.000000   4.298440\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.028512  4.0659932   6.219661   0.537305\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   4.868429  3.6593938   0.000000   0.537305\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.085536  3.2527945   0.000000   4.835745\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.839917  4.8791918  15.549153   5.641703\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.8131986   3.109831   1.343263\n                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0  2.3823188  18.130668   3.272361\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.0000000   0.000000   1.963417\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0  2.3823188   0.000000   4.581305\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  3.9705313   0.000000   3.272361\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.7941063   6.043556   9.817083\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  1.5882125   0.000000   1.308944\n                       RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   19.38574  1.3676909  7.9128854   36.52879\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000  0.6838455  1.9782213    0.00000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral    0.00000  3.4192273  0.9891107    0.00000\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000  2.0515364  1.9782213    0.00000\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   42.00244  3.4192273  0.9891107    0.00000\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000  8.2061454 43.5208696   27.39659\n                       RI_W_01_14 RI_W_01_18 RI_W_01_22 RI_W_02_14\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   63.29565  10.257613   4.369327   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000  11.625295   1.456442   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral    0.00000  11.625295   7.282212   4.027688\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   28.77075   7.522249   0.000000   0.000000\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   2.735363   1.456442   8.055376\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   1.367682   0.000000   0.000000\n                       RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  4.8924878   3.593465   8.078028   5.344367\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.9569951   0.000000   4.039014   1.068873\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  4.8924878   4.791286   2.885010   3.206620\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.9784976   3.593465   2.885010   3.206620\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  6.8494829   2.395643   2.308008   4.631785\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  5.8709854  44.319396   1.154004   1.068873\n                       RI_W_03_22 RI_W_04_14 RI_W_04_18 RI_W_04_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  4.4092608   12.32321  2.9466826    17.8865\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.4697536    0.00000  3.4377964     0.0000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  1.4697536    0.00000  7.3667065     0.0000\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.9798357    0.00000  0.9822275     0.0000\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.9596715   20.53868 12.7689580     0.0000\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.9798357   28.75416  7.8578203     0.0000\n                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  3.0758634  0.9718221   5.123934  0.9786798\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8788181  7.7745767   0.000000  0.0000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  3.5152724  9.7182208   1.280983  2.9360395\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  3.9546815  1.9436442   2.561967  9.7867983\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  5.2729086  1.9436442   6.404917 10.7654782\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.7576362  9.7182208  25.619669  4.8933992\n                       RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral    1.28024   5.416550   4.050051   5.034881\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   4.333240   0.000000   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral    1.28024   7.583169   2.025025   1.006976\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    1.28024   2.166620   5.400067  12.083715\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   5.416550   0.000000   1.006976\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    1.28024  41.165776   1.350017   2.013953\n                       RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   7.075440   2.275221   1.310363   2.753792\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   6.825663   6.879404   1.652275\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   5.660352   9.100884   4.258679   5.507584\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.415088   3.412832   1.965544   2.753792\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.660352   3.412832   6.879404   2.753792\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  89.150546   1.137611   0.982772   0.000000\n                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0   13.08723  17.568884  15.220246\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0    0.00000   0.000000   5.534635\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0    0.00000   0.000000  10.377440\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0    0.00000   1.952098   2.075488\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0    0.00000   0.000000  13.836587\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0    0.00000   7.808393   8.993782\n                       VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  13.204264   7.514330   0.000000  0.8308866\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.980640   7.514330   0.000000  4.1544330\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.961279   3.415604   6.526643  2.9081031\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.961279   3.415604   4.079152  0.0000000\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  17.165543  20.493627   2.447491  4.5698763\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  28.389167   8.197451   0.000000  2.4926598\n                       VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   6.207028   1.933974  2.9919941   1.718262\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.342246   4.512607  0.9973314   6.873047\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000   1.289316  0.4986657   5.154785\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000  2.9919941   0.000000\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.342246   8.380556  4.9866569   0.000000\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000  4.4879912   5.154785\n                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.213849   0.000000  5.4126287   3.585182\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   7.283093   0.000000  0.4510524   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000   3.961279  1.3531572   1.434073\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.855395   6.338047  3.6084191   1.434073\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  10.317715   3.961279  3.1573667   2.151109\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  16.386958   2.376767 13.0805193   8.604437\n                       VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.6877679   1.987011   6.431470   7.522889\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  3.4388395   1.987011   3.675126   1.074698\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral 14.4431257   5.961033   0.000000   3.761445\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  7.5654468   3.974022   0.000000   3.761445\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.0633037   4.967527   1.837563   4.298794\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 15.8186615   1.987011   0.000000  10.209636\n                       VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_18\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000   4.374769  2.8394448    0.00000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.346543   0.624967  4.9690284    0.00000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.966589  12.499340  6.3887508   36.13826\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.759907   0.624967  6.3887508    0.00000\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   8.749538  2.8394448    0.00000\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.379953  12.499340  0.7098612    0.00000\n                       VA_W_03_22 VA_W_04_14 VA_W_04_18 VA_W_04_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   6.328591  0.9650112   3.976631   6.009333\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   8.629897  2.4125279   5.302175   3.698051\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   2.876632  1.9300224   7.069566  10.169640\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.301306  3.3775391   3.534783   2.773538\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.451959  6.2725727  18.557612   7.858358\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  12.081856  2.4125279   0.000000   0.000000\n                       VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.0000000   3.511099  11.297065   4.584168\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000   1.003171  11.297065   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.0000000   2.507928   6.778239   2.750501\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.7675164   3.009514   2.259413   7.334669\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.5350328   5.015856  35.020902   3.667335\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.3025492   8.025369   2.259413   8.251503\n                       VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  2.3825360   0.000000  9.2975805   1.022439\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.7941787   1.585195  1.4303970   1.022439\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.7941787   0.000000  5.0063895   3.067318\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.3825360   3.962988  4.2911910   0.000000\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  6.3534292   4.755585  2.8607940   1.022439\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 27.0020742  23.777926  0.7151985   6.134635\n                       VA_W_07_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.8584103\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  6.0088720\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  2.5752309\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.7168206\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.7168206\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  7.7256926\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e genes4heatmap\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-res[res\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e0.05 \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!\u003c/span\u003eis.na(res\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e),]\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e names(genes4heatmap)\n[1] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebaseMean\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e       \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elog2FoldChange\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elfcSE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e          \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003estat\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e          \n[5] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epvalue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e         \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epadj\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e          \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(genes4heatmap)\nlog2 fold change (MLE): Origin_SymState_Temp RI_W_22 vs VA_W_22 \nWald \u003cspan class=\"pl-c1\"\u003etest\u003c/span\u003e p-value: Origin_SymState_Temp RI_W_22 vs VA_W_22 \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE      stat\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e      \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.41467       -2.16024  1.066582  -2.02538\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.06295       -3.43891  1.358159  -2.53204\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.11391        1.64720  0.816317   2.01785\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.74300        3.89665  1.286283   3.02939\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.86558        2.26910  0.996534   2.27700\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   3.17991        1.92891  0.932250   2.06910\n                          pvalue      padj\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 0.0428281  0.996985\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 0.0113402  0.882510\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 0.0436070  0.996985\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 0.0024505  0.437812\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 0.0227864  0.989064\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral 0.0385371  0.996985\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dim(genes4heatmap)\n[1] 343   6\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e data4heatmap\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dim(data4heatmap)\n[1] 343  81\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(data4heatmap)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000    2.52837   1.146477   0.000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000    2.52837   0.000000   0.000000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   10.84813    4.21395   4.585907   7.107215\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000    5.89953   2.292953   3.553607\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000    2.52837   3.439430   0.000000\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral    0.00000    2.52837   0.000000   0.000000\n                       RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   13.09462   0.000000   0.000000   0.000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   6.468236   0.000000   1.269319\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   8.624315   1.235394   1.269319\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   0.000000   1.235394   0.000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   0.000000   1.235394   5.077275\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral    0.00000   4.312157   2.470789   3.807956\n                       RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.0285120  4.0659932   6.219661  0.5373050\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.4342144  0.4065993  18.658984  0.2686525\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8114048  4.8791918   0.000000  2.9551775\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  0.4065993   0.000000  0.0000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.6228096  4.0659932   0.000000  1.0746100\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  0.0000000  1.2197979   9.329492  4.2984400\n                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000    0.00000   1.963417\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  15.182145   6.352850    0.00000   1.308944\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   1.588213    0.00000   4.581305\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000    0.00000   0.000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.900975   1.588213   12.08711   4.581305\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   4.140585   3.176425    0.00000   1.963417\n                       RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000  0.6838455  1.9782213    0.00000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   42.00244  2.0515364  5.9346640    0.00000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000  2.7353818  8.9019960   45.66098\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000  0.0000000  0.9891107    0.00000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000  2.0515364  7.9128854    0.00000\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral    0.00000  0.0000000  0.0000000    0.00000\n                       RI_W_01_14 RI_W_01_18 RI_W_01_22 RI_W_02_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000  11.625295   1.456442   0.000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   0.000000   0.000000   4.027688\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   17.26245   7.522249   1.456442   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   7.522249  16.020867   0.000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   2.051523   2.912885   0.000000\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral    0.00000   4.103045   1.456442   2.013844\n                       RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.956995   0.000000   4.039014   1.068873\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   1.197822   1.154004   0.000000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.892488   5.989108   4.616016   2.850329\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  14.677463   5.989108  21.926075  12.470189\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.870985  14.373858   4.616016   1.781456\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   2.935493  10.780394   4.039014   1.425164\n                       RI_W_03_22 RI_W_04_14 RI_W_04_18 RI_W_04_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.4697536    0.00000   3.437796     0.0000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000    0.00000   4.911138     0.0000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  6.8588501    0.00000   2.946683    17.8865\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 26.9454824    0.00000   1.964455     0.0000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.4899179    0.00000   1.964455     0.0000\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  3.4294250   12.32321   3.928910     0.0000\n                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8788181  7.7745767   0.000000   0.000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.3182272  0.9718221   0.000000   1.957360\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  5.7123177  0.9718221   3.842950   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  5.2729086  1.9436442   0.000000   4.893399\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  4.3940905  1.9436442   5.123934   4.893399\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  2.6364543  2.9154662   5.123934   6.850759\n                       RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   4.333240  0.0000000   0.000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.280240   0.000000  0.6750084   2.013953\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   7.681439   5.416550  6.7500843   3.020929\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.560480   9.749789  1.3500169   2.013953\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   3.249930  0.6750084   2.013953\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   0.000000   2.166620  1.3500169   8.055810\n                       RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   6.825663   6.879404  1.6522753\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   5.688053  15.724353  2.7537921\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.830176   1.137611   6.224223  3.8553090\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   5.688053   3.603497  2.7537921\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  14.150880   3.412832   1.965544  0.5507584\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   2.830176   3.412832   3.931088  1.1015169\n                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000   0.000000   5.534635\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   8.328238   0.000000   0.000000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.417949   0.000000   3.904196   4.842806\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.417949   0.000000   5.856295   0.000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000   0.000000   9.685611\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   4.253846   0.000000   0.000000   2.075488\n                       VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.980640  7.5143298   0.000000   4.154433\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.320426  2.7324836   0.000000   0.000000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.640853  2.0493627   1.631661   1.246330\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.320426  0.6831209   0.000000   6.647093\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.980640  0.0000000   0.000000   2.908103\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   3.301066  3.4156045   2.447491   4.985320\n                       VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.342246   4.512607  0.9973314   6.873047\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.909855   1.611645  1.4959971   0.000000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   3.867949  2.4933284   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   7.639419   1.289316  0.9973314   0.000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   2.578632  2.9919941   5.154785\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   7.161955   2.256303  3.9893255   1.718262\n                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   7.283093   0.000000  0.4510524   0.000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.820773   0.000000  4.5105239   0.000000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.676168   2.376767  4.5105239   2.151109\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.427698   0.000000  4.9615763   1.434073\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.641546   5.545791  1.3531572   4.302218\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   7.283093   0.000000  3.6084191   7.170364\n                       VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.438839  1.9870109   3.675126   1.074698\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.375536  0.9935055   0.000000   5.373492\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.375536  2.9805164   1.837563   8.060239\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.751072  5.9610327   5.512689   1.074698\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  4.9675273   5.512689   1.612048\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   8.253215 10.9285600   2.756344   3.224095\n                       VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.346543   0.624967   4.969028    0.00000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   1.874901   4.969028   31.88670\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.586636   6.249670   2.839445    0.00000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   3.749802   1.419722    0.00000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.586636   3.124835   7.098612   21.25780\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   1.586636   6.874637   4.259167   23.38358\n                       VA_W_03_22 VA_W_04_14 VA_W_04_18 VA_W_04_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   8.629897  2.4125279   5.302175  3.6980510\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.150653  0.9650112   0.000000  1.8490255\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.725979  1.4475168   3.976631  2.3112819\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.301306  0.9650112   0.000000  0.4622564\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.725979  0.0000000   0.000000  0.4622564\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   1.725979  4.8250559   1.767392  0.4622564\n                       VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  1.0031712  11.297065   0.000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.0000000  12.426772   1.833667\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  6.5206127   0.000000   2.750501\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.0000000   0.000000   0.000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.5015856   1.129707   1.833667\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral          0  2.5079280   0.000000   1.833667\n                       VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.7941787  1.5851950  1.4303970   1.022439\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.5883573  3.9629876  2.1455955   0.000000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  5.5592506  0.0000000  2.8607940   8.179514\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.5883573  0.0000000  0.0000000   0.000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.3825360  0.7925975  0.7151985   4.089757\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  0.7941787  2.3777926  1.4303970   0.000000\n                       VA_W_07_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  6.0088720\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  3.4336412\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  3.4336412\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.5752309\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  0.8584103\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e temp = as.matrix(rowMeans(data4heatmap))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(temp)\n                           [,1]\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 2.414669\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 3.062953\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 4.113910\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 2.742998\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 2.865582\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral 3.179912\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e scaledmatrix\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e data4heatmapscaled = data4heatmap/scaledmatrix\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(data4heatmapscaled)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  1.0470877  0.4747967   0.000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.8254680  0.0000000   0.000000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.636938  1.0243174  1.1147318   1.727606\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  2.1507597  0.8359295   1.295519\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.8823235  1.2002554   0.000000\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   0.000000  0.7951069  0.0000000   0.000000\n                       RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.422948   0.000000  0.0000000  0.0000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   2.111764  0.0000000  0.4144100\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   2.096379  0.3002969  0.3085431\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000  0.4503810  0.0000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000  0.4311147  1.7718129\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   0.000000   1.356062  0.7769990  1.1975035\n                       RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8400788  1.6838720   2.575782 0.22251707\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.7947279  0.1327475   6.091827 0.08771028\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.1972344  1.1860229   0.000000 0.71833786\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  0.1482317   0.000000 0.00000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.5663107  1.4189067   0.000000 0.37500589\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  0.0000000  0.3835949   2.933884 1.35174810\n                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.0000000   0.000000  0.8131205\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.956701  2.0740929   0.000000  0.4273471\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.3860591   0.000000  1.1136133\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.0000000   0.000000  0.0000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.408228  0.5542374   4.218031  1.5987348\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   1.302107  0.9989034   0.000000  0.6174437\n                       RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000  0.2832047  0.8192517    0.00000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   13.71305  0.6697903  1.9375626    0.00000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000  0.6649104  2.1638771   11.09917\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000  0.0000000  0.3605947    0.00000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000  0.7159232  2.7613540    0.00000\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral    0.00000  0.0000000  0.0000000    0.00000\n                       RI_W_01_14 RI_W_01_18 RI_W_01_22 RI_W_02_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  4.8144470  0.6031645  0.0000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.0000000  0.0000000  1.3149687\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.196117  1.8284914  0.3540287  0.0000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  2.7423457  5.8406405  0.0000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.7159184  1.0165074  0.0000000\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral   0.000000  1.2903015  0.4580134  0.6333018\n                       RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8104611  0.0000000  1.6726989  0.4426584\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  0.3910675  0.3767618  0.0000000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.1892549  1.4558187  1.1220507  0.6928515\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  5.3508832  2.1834164  7.9934703  4.5461892\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.0487936  5.0160350  1.6108477  0.6216733\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  0.9231364  3.3901547  1.2701653  0.4481773\n                       RI_W_03_22 RI_W_04_14 RI_W_04_18 RI_W_04_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.6086771   0.000000  1.4237135    0.00000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000   0.000000  1.6033994    0.00000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.6672338   0.000000  0.7162729    4.34781\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  9.8233684   0.000000  0.7161707    0.00000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.1709663   0.000000  0.6855345    0.00000\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  1.0784654   3.875331  1.2355405    0.00000\n                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.3639498  3.2197281  0.0000000  0.0000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.4303778  0.3172827  0.0000000  0.6390432\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.3885373  0.2362283  0.9341357  0.0000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.9223157  0.7085838  0.0000000  1.7839600\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.5334026  0.6782721  1.7880955  1.7076460\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  0.8290966  0.9168387  1.6113445  2.1543863\n                       RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  1.7945483  0.0000000  0.0000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.4179756  0.0000000  0.2203783  0.6575198\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.8671868  1.3166426  1.6407952  0.7343205\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.9334602  3.5544277  0.4921683  0.7342157\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  1.1341257  0.2355572  0.7028076\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  0.0000000  0.6813458  0.4245453  2.5333437\n                       RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  2.8267494  2.8490054  0.6842658\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  1.8570484  5.1337224  0.8990643\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.6879528  0.2765278  1.5129700  0.9371398\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  2.0736626  1.3137075  1.0039351\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  4.9382226  1.1909734  0.6859145  0.1921978\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  0.8900171  1.0732472  1.2362254  0.3463985\n                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000   0.000000  0.0000000  2.2920887\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000   2.719022  0.0000000  0.0000000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.3446718   0.000000  0.9490232  1.1771782\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.5169339   0.000000  2.1349975  0.0000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000   0.000000  0.0000000  3.3799808\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  1.3377245   0.000000  0.0000000  0.6526873\n                       VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8202531  3.1119506  0.0000000  1.7204981\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.4310958  0.8921074  0.0000000  0.0000000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.6419325  0.4981544  0.3966204  0.3029551\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.4813807  0.2490417  0.0000000  2.4232946\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.6911824  0.0000000  0.0000000  1.0148387\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  1.0380998  1.0741192  0.7696726  1.5677539\n                       VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.3841426  1.8688306  0.4130303  2.8463728\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.6235337  0.5261736  0.4884165  0.0000000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  0.9402122  0.6060726  0.0000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.7850616  0.4700390  0.3635917  0.0000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  0.8998635  1.0441141  1.7988618\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  2.2522495  0.7095490  1.2545396  0.5403489\n                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  3.0161871  0.0000000  0.1867968  0.0000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.5944502  0.0000000  1.4726061  0.0000000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.6228279  0.5777393  1.0964079  0.5228868\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8850525  0.0000000  1.8088149  0.5228121\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.2707878  1.9353107  0.4722103  1.5013421\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  2.2903441  0.0000000  1.1347544  2.2548938\n                       VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.4241454  0.8228917  1.5220000  0.4450708\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.4490880  0.3243619  0.0000000  1.7543500\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.3343621  0.7244972  0.4466706  1.9592646\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.0029432  2.1731814  2.0097310  0.3917970\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  1.7335145  1.9237590  0.5625552\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  2.5954223  3.4367492  0.8667989  1.0138946\n                       VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.6283285  0.2588210  2.0578510   0.000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  0.6121219  1.6222997  10.410443\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.3856758  1.5191556  0.6902058   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  1.3670450  0.5175805   0.000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.5536871  1.0904714  2.4771976   7.418320\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  0.4989558  2.1618953  1.3393978   7.353531\n                       VA_W_03_22 VA_W_04_14 VA_W_04_18 VA_W_04_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  3.5739467  0.9991134  2.1958187  1.5314941\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.3756678  0.3150590  0.0000000  0.6036740\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.4195472  0.3518591  0.9666305  0.5618212\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8389746  0.3518089  0.0000000  0.1685223\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.6023138  0.0000000  0.0000000  0.1613133\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  0.5427759  1.5173552  0.5557989  0.1453677\n                       VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.4154488  4.6785155  0.0000000\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.0000000  4.0571206  0.5986599\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  1.5850158  0.0000000  0.6685856\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.0000000  0.0000000  0.0000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.1750380  0.3942329  0.6398936\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral          0  0.7886784  0.0000000  0.5766409\n                       VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.3288976  0.6564855  0.5923782  0.4234284\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.5185705  1.2938452  0.7004989  0.0000000\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.3513301  0.0000000  0.6953953  1.9882577\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.5790588  0.0000000  0.0000000  0.0000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8314319  0.2765922  0.2495823  1.4271996\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  0.2497486  0.7477542  0.4498228  0.0000000\n                       VA_W_07_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.4884871\nTR10450\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.1210230\nTR10768\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8346417\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000\nTR11771\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8986765\nTR11938\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i2_coral  0.2699478\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dim(data4heatmapscaled)\n[1] 343  81\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pairs.breaks \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- seq(0, 3.0, by=0.1)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e length(pairs.breaks)\n[1] 31\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e mycol \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- colorpanel(n=30, low=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, high=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eyellow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eXXXXXX_byrow.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,14,7)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), scale=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, symkey = F, density.info = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edensity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, colsep=c(24), sepcolor=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\nRStudioGD \n        2 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eXXXXXX_bycolumn.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,14,7)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecol\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), scale=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, symkey = F, density.info = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edensity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, colsep=c(24), sepcolor=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\nRStudioGD \n        2\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e prop.nullv3 \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003emean(x==0))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eResNullCounts.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,14, 14)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e barplot(prop.nullv3, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\nRStudioGD \n        2 \n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay12 HW\u003c/h2\u003e\u003ca id=\"user-content-day12-hw\" class=\"anchor\" aria-label=\"Permalink: Day12 HW\" href=\"#day12-hw\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#1- mkdir a directory with your name in the \n\n#2- cp/mv your logfile to the /cm/shared/courses/dbarshis/18AdvBioinf/archive/YOURNAME directory\n\n#3- cp/mv your merged, unfiltered .vcf file YOURINITIALS_unfiltered.vcf into the /cm/shared/courses/dbarshis/18AdvBioinf/archive/YOURNAME directory\n\n#4- cp/mv any other files you might want to keep to the /cm/shared/courses/dbarshis/18AdvBioinf/archive/YOURNAME directory\n\n#5- create an sbatch script to rm -r your sandbox/YOURNAME directory\n\n#6- scp/winscp /cm/shared/courses/dbarshis/18AdvBioinf/classdata/Astrangia_poculata/expressioncounts/ALLLANESFullCounts_summed.txt to your laptop\n\n#7- scp/winscp /cm/shared/courses/dbarshis/18AdvBioinf/classdata/Astrangia_poculata/expressioncounts/ALLLANES_conditions.txt to your laptop\n\n#8- Download the DESeq2Script_advbioinf_day12.R script and work through to generate a PCA of your data.\n\n\n\u0026gt; library(DESeq2)\n\u0026gt; library(gplots)\n\u0026gt; library(goseq)\n\u0026gt; library(GO.db)\n\u0026gt; \n\u0026gt; setwd(\u0026quot;/Users/katiecrider/Desktop/Genomics\u0026quot;)  # The drag and drop from finder works in R, too.\n\u0026gt; \n\u0026gt; #useful functions\n\u0026gt; #head() - prints out the top 6 lines\n\u0026gt; #dim() - prints the dimensions of a variable\n\u0026gt; #nrow() - returns the number of rows in a vector or matrix\n\u0026gt; # ?[functionName] - opens documentation describing the function\n\u0026gt; \n\u0026gt; #read in your data to make counts table\n\u0026gt; countsTable \u0026lt;- read.delim('djbFullCounts_summed.txt', header=TRUE, stringsAsFactors=TRUE, row.names=1)\n\u0026gt; head(countsTable)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          0          1          6          0          0          0\nTR10083|c0_g2_i1_coral          0          1          2          0          1          2\nTR10090|c0_g1_i1_coral          0          3          1          0          4          0\nTR10090|c0_g2_i1_coral          0          4          2          1          0          0\nTR10103|c0_g1_i1_coral          0          1          4          0          0          4\n                       RI_B_03_14 RI_B_03_18 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          0          0          0          9          1          0\nTR10083|c0_g2_i1_coral         10          0          1         31         20          0\nTR10090|c0_g1_i1_coral          0          0          0          5         10          2\nTR10090|c0_g2_i1_coral          0          0          3         12          9          0\nTR10103|c0_g1_i1_coral          1          0          3         15          8          0\n                       RI_B_05_14 RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          2          0          0          0          2          0\nTR10083|c0_g2_i1_coral         16          0          3          3          5          6\nTR10090|c0_g1_i1_coral          2          0          0          0          3          0\nTR10090|c0_g2_i1_coral          2          0          3          0          7          0\nTR10103|c0_g1_i1_coral         18          0          5          0          5          0\n                       RI_B_07_14 RI_B_07_18 RI_B_07_22 RI_W_01_14 RI_W_01_18 RI_W_01_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          2          1          0          0          0          1\nTR10083|c0_g2_i1_coral          2          8          4         11         15          3\nTR10090|c0_g1_i1_coral          1          2          0          0         17          1\nTR10090|c0_g2_i1_coral          5          1          0          0         17          5\nTR10103|c0_g1_i1_coral          3          2          0          5         11          0\n                       RI_W_02_14 RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          1          1          2          0          5          0\nTR10083|c0_g2_i1_coral          0          5          3         14         15          9\nTR10090|c0_g1_i1_coral          0          2          0          7          3          3\nTR10090|c0_g2_i1_coral          2          5          4          5          9          3\nTR10103|c0_g1_i1_coral          0          1          3          5          9          2\n                       RI_W_04_14 RI_W_04_18 RI_W_04_22 RI_W_05_14 RI_W_05_18 RI_W_05_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          1          1          0          2          0          0\nTR10083|c0_g2_i1_coral          3          6          1          7          1          4\nTR10090|c0_g1_i1_coral          0          7          0          2          8          0\nTR10090|c0_g2_i1_coral          0         15          0          8         10          1\nTR10103|c0_g1_i1_coral          0          2          0          9          2          2\n                       RI_W_06_14 RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18 RI_W_07_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          2          6          0          4          0          0\nTR10083|c0_g2_i1_coral          1          1          5          6          5          5\nTR10090|c0_g1_i1_coral          0          0          4          0          0          0\nTR10090|c0_g2_i1_coral          3          1          7          3          1          4\nTR10103|c0_g1_i1_coral         10          1          2          8         12          1\n                       VA_B_01_14 VA_B_01_18 VA_B_01_22 VA_B_02_14 VA_B_02_18 VA_B_02_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          0          2          2          0          0          0\nTR10083|c0_g2_i1_coral          2          4          5          0         11          9\nTR10090|c0_g1_i1_coral          6         21          3          0          0          0\nTR10090|c0_g2_i1_coral          8         13         10          0          0          0\nTR10103|c0_g1_i1_coral          3          6          5          0          0          1\n                       VA_B_03_14 VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18 VA_B_04_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          1          0          6          0         10          4\nTR10083|c0_g2_i1_coral         22         20         11          0          2         13\nTR10090|c0_g1_i1_coral          8          3         11          0         10          7\nTR10090|c0_g2_i1_coral         15          6          5          8          7          0\nTR10103|c0_g1_i1_coral          3          6          5          5          0          0\n                       VA_B_05_14 VA_B_05_18 VA_B_05_22 VA_B_06_14 VA_B_06_18 VA_B_06_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          2          2          0         10          0          1\nTR10083|c0_g2_i1_coral          6          6          1          2          0         12\nTR10090|c0_g1_i1_coral         14          2          4         12          0          1\nTR10090|c0_g2_i1_coral          4          1          3          0          5          3\nTR10103|c0_g1_i1_coral          0          6          0          8          8          8\n                       VA_B_07_14 VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18 VA_W_01_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          0          4          1          1          3          0\nTR10083|c0_g2_i1_coral          5          1          2          7         14          0\nTR10090|c0_g1_i1_coral          0          5          2          4          2          0\nTR10090|c0_g2_i1_coral          2         21          6          0          7          0\nTR10103|c0_g1_i1_coral          2         11          4          0          7          0\n                       VA_W_02_14 VA_W_02_18 VA_W_02_22 VA_W_03_14 VA_W_03_18 VA_W_03_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          4          0          0          4          0          6\nTR10083|c0_g2_i1_coral          0          7          0          4          0         11\nTR10090|c0_g1_i1_coral          8          1          0          7          0         15\nTR10090|c0_g2_i1_coral          5         20          0          9         17          5\nTR10103|c0_g1_i1_coral          6          1          0          9          0          4\n                       VA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          2          0          2          0          3          0\nTR10083|c0_g2_i1_coral          2          9         13          0          7         10\nTR10090|c0_g1_i1_coral          5         12          8          0          2         10\nTR10090|c0_g2_i1_coral          4         16         22          0          5          6\nTR10103|c0_g1_i1_coral          7          8          6          1          6          2\n                       VA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          2          0          0          0          0          3\nTR10083|c0_g2_i1_coral          5          3          0         13          1          1\nTR10090|c0_g1_i1_coral          0          1          2          2          1          7\nTR10090|c0_g2_i1_coral          3          1          0          7          3          3\nTR10103|c0_g1_i1_coral          8          3          5          6          0          2\n\u0026gt; dim(countsTable)\n[1] 13381    84\n\u0026gt; \n\u0026gt; colSums(countsTable)\nRI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_18 \n     97406     207372     173843      47481      58036      84501     170330       5972 \nRI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14 RI_B_05_18 RI_B_05_22 RI_B_06_14 \n    199204     357124     544267      71600     630609     227313     225639      37012 \nRI_B_06_18 RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22 RI_W_01_14 RI_W_01_18 RI_W_01_22 \n    296071      72254     257730     218198      25369      39419     292612     174666 \nRI_W_02_14 RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18 \n     54184     192092     155178     281297     428532     359053      37579     300578 \nRI_W_04_22 RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22 RI_W_07_14 \n     10848     392818     214887     158683     186568     205135     198272     342795 \nRI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22 VA_B_02_14 VA_B_02_18 VA_B_02_22 \n    220630     176946     261824     548448     328198      79276     152742      85224 \nVA_B_03_14 VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 \n    226784     251696     271959     172636     264192     268290     323715     288121 \nVA_B_05_22 VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22 VA_W_01_14 \n     89943     232578     195521     362054     215616     234465     169742     164874 \nVA_W_01_18 VA_W_01_22 VA_W_02_14 VA_W_02_18 VA_W_02_22 VA_W_03_14 VA_W_03_18 VA_W_03_22 \n    321900        313     170250     241804        134     278000     138573     295382 \nVA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14 VA_W_06_18 \n    292041     280007     261344     178096     388532     157123     176579     188506 \nVA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22 \n    209003     228792     171481     233899 \n\u0026gt; colSums(countsTable)[order(colSums(countsTable))]\nVA_W_02_22 VA_W_01_22 RI_B_03_18 RI_W_04_22 RI_B_07_22 RI_B_06_14 RI_W_04_14 RI_W_01_14 \n       134        313       5972      10848      25369      37012      37579      39419 \nRI_B_02_14 RI_W_02_14 RI_B_02_18 RI_B_04_22 RI_B_06_22 VA_B_02_14 RI_B_02_22 VA_B_02_22 \n     47481      54184      58036      71600      72254      79276      84501      85224 \nVA_B_05_22 RI_B_01_14 VA_W_03_18 VA_B_02_18 RI_W_02_22 VA_W_05_22 RI_W_05_22 VA_W_01_14 \n     89943      97406     138573     152742     155178     157123     158683     164874 \nVA_B_07_22 VA_W_02_14 RI_B_03_14 VA_W_07_18 VA_B_04_14 RI_B_01_22 RI_W_01_22 VA_W_06_14 \n    169742     170250     170330     171481     172636     173843     174666     176579 \nRI_W_07_22 VA_W_05_14 RI_W_06_14 VA_W_06_18 RI_W_02_18 VA_B_06_18 RI_W_06_22 RI_B_03_22 \n    176946     178096     186568     188506     192092     195521     198272     199204 \nRI_W_06_18 RI_B_01_18 VA_W_06_22 RI_W_05_18 VA_B_07_14 RI_B_07_18 RI_W_07_18 RI_B_05_22 \n    205135     207372     209003     214887     215616     218198     220630     225639 \nVA_B_03_14 RI_B_05_18 VA_W_07_14 VA_B_06_14 VA_W_07_22 VA_B_07_18 VA_W_02_18 VA_B_03_18 \n    226784     227313     228792     232578     233899     234465     241804     251696 \nRI_B_07_14 VA_W_04_22 VA_B_01_14 VA_B_04_18 VA_B_04_22 VA_B_03_22 VA_W_03_14 VA_W_04_18 \n    257730     261344     261824     264192     268290     271959     278000     280007 \nRI_W_03_14 VA_B_05_18 VA_W_04_14 RI_W_01_18 VA_W_03_22 RI_B_06_18 RI_W_04_18 VA_W_01_18 \n    281297     288121     292041     292612     295382     296071     300578     321900 \nVA_B_05_14 VA_B_01_22 RI_W_07_14 RI_B_04_14 RI_W_03_22 VA_B_06_22 VA_W_05_18 RI_W_05_14 \n    323715     328198     342795     357124     359053     362054     388532     392818 \nRI_W_03_18 RI_B_04_18 VA_B_01_18 RI_B_05_14 \n    428532     544267     548448     630609 \n\u0026gt; \n\u0026gt; table(substring(names(countsTable),1,2))\n\nRI VA \n42 42 \n\u0026gt; table(substring(names(countsTable),4,4))\n\n B  W \n42 42 \n\u0026gt; table(substring(names(countsTable),9,10))\n\n14 18 22 \n28 28 28 \n\u0026gt; table(paste(substring(names(countsTable),1,2),substring(names(countsTable),4,4), sep=\u0026quot;_\u0026quot;))\n\nRI_B RI_W VA_B VA_W \n  21   21   21   21 \n\u0026gt; table(paste(substring(names(countsTable),1,2),substring(names(countsTable),4,4), substring(names(countsTable),9,10), sep=\u0026quot;_\u0026quot;))\n\nRI_B_14 RI_B_18 RI_B_22 RI_W_14 RI_W_18 RI_W_22 VA_B_14 VA_B_18 VA_B_22 VA_W_14 VA_W_18 VA_W_22 \n      7       7       7       7       7       7       7       7       7       7       7       7 \n\u0026gt; \n\u0026gt; LowCounts\u0026lt;-c(\u0026quot;VA_W_02_22\u0026quot;,\u0026quot;VA_W_01_22\u0026quot;,\u0026quot;RI_B_03_18\u0026quot;, \u0026quot;RI_W_01_14\u0026quot;, \u0026quot;RI_W_04_22\u0026quot;, \u0026quot;VA_W_03_18\u0026quot;)\n\u0026gt; names(countsTable)%in%LowCounts\n [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n[16] FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n[31] FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n[46] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n[61] FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE FALSE  TRUE FALSE FALSE FALSE FALSE\n[76] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n\u0026gt; !(names(countsTable)%in%LowCounts)\n [1]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[16]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[31]  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[46]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[61]  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE FALSE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE\n[76]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n\u0026gt; countsTable\u0026lt;-countsTable[,!(names(countsTable)%in%LowCounts)]\n\u0026gt; dim(countsTable)\n[1] 13381    78\n\u0026gt; \n\u0026gt; #make a table with conditions of each individual (e.g. \u0026quot;VA\u0026quot; and \u0026quot;RI\u0026quot;)  There should be the same number of conditions described as there are samples in your data file, and in the same order.\n\u0026gt; #NOTE: It is absolutely critical that the columns of the count matrix and the rows of the column data (information about samples) are in the same order. DESeq2 will not make guesses as to which column of the count matrix belongs to which row of the column data, these must be provided to DESeq2 already in consistent order.\n\u0026gt; #Here's an example\n\u0026gt; # Sample\t\tOrigin\tSymState\tTemp  Origin_SymState Origin_SymState_Temp\n\u0026gt; # RI_B_06_18\t\tRI\tB\t18  RI_B  RI_B_18\n\u0026gt; # VA_B_07_14\t\tVA\tB\t14  VA_B  VA_B_14\n\u0026gt; # VA_W_07_22\t\tVA\tW\t22  VA_W  VA_W_22\n\u0026gt; # safest way to do this is in R\n\u0026gt; \n\u0026gt; conds\u0026lt;-data.frame(\u0026quot;Sample\u0026quot;=names(countsTable))\n\u0026gt; conds$Origin\u0026lt;-factor(substring(conds$Sample,1,2))\n\u0026gt; conds$SymState\u0026lt;-factor(substring(conds$Sample,4,4))\n\u0026gt; conds$Temp\u0026lt;-factor(substring(conds$Sample,9,10))\n\u0026gt; conds$Origin_SymState\u0026lt;-factor(paste(conds$Origin,conds$SymState, sep=\u0026quot;_\u0026quot;))\n\u0026gt; conds$Origin_SymState_Temp\u0026lt;-factor(paste(conds$Origin,conds$SymState,conds$Temp, sep=\u0026quot;_\u0026quot;))\n\u0026gt; \n\u0026gt; ##Check null counts per sample\n\u0026gt; prop.null \u0026lt;- apply(countsTable, 2, function(x) 100*mean(x==0))\n\u0026gt; \n\u0026gt; barplot(prop.null, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u0026gt; \n\u0026gt; pdf(file=\u0026quot;OverallNullCountsv2.pdf\u0026quot;,14, 14)\n\u0026gt; barplot(prop.null, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u0026gt; dev.off()\nRStudioGD \n        2 \n\u0026gt; \n\u0026gt; # how many genes have mean count \u0026gt;=3 \n\u0026gt; means = apply(countsTable, 1, mean)\n\u0026gt; table(means\u0026gt;=3)\n\nFALSE  TRUE \n 5721  7660 \n\u0026gt; \n\u0026gt; countsTable\u0026lt;-countsTable[means\u0026gt;=3,]\n\u0026gt; dim(countsTable)\n[1] 7660   78\n\u0026gt; \n\u0026gt; prop.nullv2 \u0026lt;- apply(countsTable, 2, function(x) 100*mean(x==0))\n\u0026gt; \n\u0026gt; barplot(prop.nullv2, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u0026gt; \n\u0026gt; pdf(file=\u0026quot;Mean3NullCountsv2.pdf\u0026quot;,14, 14)\n\u0026gt; barplot(prop.nullv2, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u0026gt; dev.off()\nRStudioGD \n        2 \n\u0026gt; \n\u0026gt; #make count data sets\n\u0026gt; dds \u0026lt;- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=~ Origin + SymState + Temp + Origin:SymState)\n\u0026gt; \n\u0026gt; dds \u0026lt;- DESeq(dds)\nestimating size factors\nestimating dispersions\ngene-wise dispersion estimates\nmean-dispersion relationship\nfinal dispersion estimates\nfitting model and testing\n-- replacing outliers and refitting for 14 genes\n-- DESeq argument 'minReplicatesForReplace' = 7 \n-- original counts are preserved in counts(dds)\nestimating dispersions\nfitting model and testing\n\u0026gt; \n\u0026gt; dim(dds)\n[1] 7660   78\n\u0026gt; \n\u0026gt; #transform counts to variance stablized counts\n\u0026gt; vstCounts\u0026lt;-varianceStabilizingTransformation(dds)\n\u0026gt; \n\u0026gt; #plot cluster dendrogram across samples\n\u0026gt; dists\u0026lt;-dist(t(assay(vstCounts)))\n\u0026gt; plot(hclust(dists))\n\u0026gt; heatmap.2(as.matrix(dists), key=F, trace=\u0026quot;none\u0026quot;,\n+           col=colorpanel(100, \u0026quot;black\u0026quot;, \u0026quot;white\u0026quot;),\n+           margin=c(10, 10))\n\u0026gt; \n\u0026gt; ###Figure out which contrast you want to examine (i.e. which two groups do you want to compare)\n\u0026gt; resultsNames(dds)\n[1] \u0026quot;Intercept\u0026quot;          \u0026quot;Origin_VA_vs_RI\u0026quot;    \u0026quot;SymState_W_vs_B\u0026quot;    \u0026quot;Temp_18_vs_14\u0026quot;     \n[5] \u0026quot;Temp_22_vs_14\u0026quot;      \u0026quot;OriginVA.SymStateW\u0026quot;\n\u0026gt; \n\u0026gt; res \u0026lt;- results(dds)\n\u0026gt; head(res)\nlog2 fold change (MLE): OriginVA.SymStateW \nWald test p-value: OriginVA.SymStateW \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj\n                       \u0026lt;numeric\u0026gt;      \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\nTR10083|c0_g2_i1_coral   5.36648       0.107690  0.644646  0.167053  0.867328  0.999364\nTR10090|c0_g1_i1_coral   2.70436      -0.247209  0.831371 -0.297351  0.766198  0.995209\nTR10090|c0_g2_i1_coral   3.65756      -0.976997  0.654110 -1.493627  0.135273  0.958807\nTR10103|c0_g1_i1_coral   2.91279      -0.172220  0.641026 -0.268662  0.788189  0.995209\nTR10108|c0_g1_i1_coral   6.31144       0.366046  0.664401  0.550942  0.581673  0.991107\nTR10132|c0_g1_i1_coral   9.26737      -0.589357  0.975809 -0.603967  0.545865  0.990477\n\u0026gt; summary(res)\n\nout of 7659 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 2, 0.026%\nLFC \u0026lt; 0 (down)     : 9, 0.12%\noutliers [1]       : 10, 0.13%\nlow counts [2]     : 1, 0.013%\n(mean count \u0026lt; 1)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\n\n\u0026gt; \n\u0026gt; resSymState \u0026lt;- results(dds, contrast=c(\u0026quot;SymState\u0026quot;, \u0026quot;B\u0026quot;, \u0026quot;W\u0026quot;))\n\u0026gt; head(resSymState)\nlog2 fold change (MLE): SymState B vs W \nWald test p-value: SymState B vs W \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj\n                       \u0026lt;numeric\u0026gt;      \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\nTR10083|c0_g2_i1_coral   5.36648       0.304875  0.463588  0.657643 0.5107675  0.986343\nTR10090|c0_g1_i1_coral   2.70436      -0.340214  0.624845 -0.544477 0.5861132  0.995961\nTR10090|c0_g2_i1_coral   3.65756      -1.044272  0.488673 -2.136956 0.0326015  0.816270\nTR10103|c0_g1_i1_coral   2.91279      -0.316211  0.469123 -0.674047 0.5002815  0.986343\nTR10108|c0_g1_i1_coral   6.31144       0.257974  0.481341  0.535948 0.5919943  0.995961\nTR10132|c0_g1_i1_coral   9.26737      -0.604957  0.692218 -0.873939 0.3821514  0.986343\n\u0026gt; summary(resSymState)\n\nout of 7659 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 6, 0.078%\nLFC \u0026lt; 0 (down)     : 6, 0.078%\noutliers [1]       : 10, 0.13%\nlow counts [2]     : 1, 0.013%\n(mean count \u0026lt; 1)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\n\n\u0026gt; \n\u0026gt; resOrigin \u0026lt;- results(dds, contrast=c(\u0026quot;Origin\u0026quot;, \u0026quot;VA\u0026quot;, \u0026quot;RI\u0026quot;))\n\u0026gt; head(resOrigin)\nlog2 fold change (MLE): Origin VA vs RI \nWald test p-value: Origin VA vs RI \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE       stat    pvalue      padj\n                       \u0026lt;numeric\u0026gt;      \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;  \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\nTR10083|c0_g2_i1_coral   5.36648     -0.2712427  0.446225 -0.6078616 0.5432793  0.865661\nTR10090|c0_g1_i1_coral   2.70436      1.1125117  0.589306  1.8878328 0.0590484        NA\nTR10090|c0_g2_i1_coral   3.65756      0.9431346  0.474719  1.9867203 0.0469534  0.418043\nTR10103|c0_g1_i1_coral   2.91279      0.0386643  0.456720  0.0846565 0.9325345  0.984590\nTR10108|c0_g1_i1_coral   6.31144     -0.0179010  0.461958 -0.0387502 0.9690895  0.992150\nTR10132|c0_g1_i1_coral   9.26737     -0.2649903  0.677608 -0.3910675 0.6957473  0.921517\n\u0026gt; summary(resOrigin)\n\nout of 7659 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 73, 0.95%\nLFC \u0026lt; 0 (down)     : 104, 1.4%\noutliers [1]       : 10, 0.13%\nlow counts [2]     : 594, 7.8%\n(mean count \u0026lt; 3)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\n\n\u0026gt; \n\u0026gt; res14v18 \u0026lt;- results(dds, contrast=c(\u0026quot;Temp\u0026quot;, \u0026quot;14\u0026quot;, \u0026quot;18\u0026quot;))\n\u0026gt; head(res14v18)\nlog2 fold change (MLE): Temp 14 vs 18 \nWald test p-value: Temp 14 vs 18 \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE       stat    pvalue      padj\n                       \u0026lt;numeric\u0026gt;      \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;  \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\nTR10083|c0_g2_i1_coral   5.36648      0.0134106  0.389675  0.0344147 0.9725464  0.994763\nTR10090|c0_g1_i1_coral   2.70436     -0.7500403  0.493720 -1.5191625 0.1287216        NA\nTR10090|c0_g2_i1_coral   3.65756     -0.5278497  0.385227 -1.3702295 0.1706153        NA\nTR10103|c0_g1_i1_coral   2.91279      0.1448997  0.377624  0.3837147 0.7011899        NA\nTR10108|c0_g1_i1_coral   6.31144     -0.1171316  0.401865 -0.2914700 0.7706919  0.956296\nTR10132|c0_g1_i1_coral   9.26737     -1.1666726  0.592603 -1.9687238 0.0489848  0.440134\n\u0026gt; summary(res14v18)\n\nout of 7659 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 31, 0.4%\nLFC \u0026lt; 0 (down)     : 111, 1.4%\noutliers [1]       : 10, 0.13%\nlow counts [2]     : 2671, 35%\n(mean count \u0026lt; 4)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\n\n\u0026gt; \n\u0026gt; res14v22 \u0026lt;- results(dds, contrast=c(\u0026quot;Temp\u0026quot;, \u0026quot;14\u0026quot;, \u0026quot;22\u0026quot;))\n\u0026gt; head(res14v22)\nlog2 fold change (MLE): Temp 14 vs 22 \nWald test p-value: Temp 14 vs 22 \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj\n                       \u0026lt;numeric\u0026gt;      \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\nTR10083|c0_g2_i1_coral   5.36648      -0.381080  0.396906 -0.960127 0.3369915  0.796010\nTR10090|c0_g1_i1_coral   2.70436      -0.420391  0.517100 -0.812978 0.4162304        NA\nTR10090|c0_g2_i1_coral   3.65756      -0.149468  0.407436 -0.366851 0.7137303        NA\nTR10103|c0_g1_i1_coral   2.91279       0.327506  0.400038  0.818686 0.4129654        NA\nTR10108|c0_g1_i1_coral   6.31144      -0.504681  0.409220 -1.233275 0.2174733  0.699820\nTR10132|c0_g1_i1_coral   9.26737      -1.240058  0.602190 -2.059248 0.0394705  0.387444\n\u0026gt; summary(res14v22)\n\nout of 7659 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 64, 0.84%\nLFC \u0026lt; 0 (down)     : 97, 1.3%\noutliers [1]       : 10, 0.13%\nlow counts [2]     : 2226, 29%\n(mean count \u0026lt; 4)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\n\n\u0026gt; \n\u0026gt; res18v22 \u0026lt;- results(dds, contrast=c(\u0026quot;Temp\u0026quot;, \u0026quot;18\u0026quot;, \u0026quot;22\u0026quot;))\n\u0026gt; head(res18v22)\nlog2 fold change (MLE): Temp 18 vs 22 \nWald test p-value: Temp 18 vs 22 \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj\n                       \u0026lt;numeric\u0026gt;      \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\nTR10083|c0_g2_i1_coral   5.36648     -0.3944906  0.395010 -0.998685  0.317947  0.838585\nTR10090|c0_g1_i1_coral   2.70436      0.3296494  0.503430  0.654807  0.512592        NA\nTR10090|c0_g2_i1_coral   3.65756      0.3783816  0.396935  0.953259  0.340459  0.847759\nTR10103|c0_g1_i1_coral   2.91279      0.1826059  0.399214  0.457414  0.647374        NA\nTR10108|c0_g1_i1_coral   6.31144     -0.3875492  0.406736 -0.952828  0.340677  0.847759\nTR10132|c0_g1_i1_coral   9.26737     -0.0733858  0.596241 -0.123081  0.902043  0.991572\n\u0026gt; summary(res18v22)\n\nout of 7659 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 34, 0.44%\nLFC \u0026lt; 0 (down)     : 10, 0.13%\noutliers [1]       : 10, 0.13%\nlow counts [2]     : 1337, 17%\n(mean count \u0026lt; 3)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\n\n\u0026gt; \n\u0026gt; #count the number of significantly differentially expressed genes\n\u0026gt; sum(resOrigin$padj \u0026lt; 0.1, na.rm =T)\n[1] 177\n\u0026gt; sum(resOrigin$pvalue \u0026lt; 0.05, na.rm =T)\n[1] 866\n\u0026gt; \n\u0026gt; # Make a counts table that is scaled by the size factors\n\u0026gt; scaledcounts = counts(dds, normalize=T)\n\u0026gt; head(scaledcounts)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10083|c0_g2_i1_coral          0  0.8671827   2.447866   0.000000   3.455170   4.432653\nTR10090|c0_g1_i1_coral          0  2.6015481   1.223933   0.000000  13.820681   0.000000\nTR10090|c0_g2_i1_coral          0  3.4687308   2.447866   3.772786   0.000000   0.000000\nTR10103|c0_g1_i1_coral          0  0.8671827   4.895733   0.000000   0.000000   8.865306\nTR10108|c0_g1_i1_coral          0  1.7343654   4.895733   0.000000   6.910341   0.000000\nTR10132|c0_g1_i1_coral          0  6.0702789   1.223933  18.863931  51.827555   0.000000\n                       RI_B_03_14 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10083|c0_g2_i1_coral  12.871970   1.272998  12.424473  8.1878079   0.000000   4.563408\nTR10090|c0_g1_i1_coral   0.000000   0.000000   2.003947  4.0939039   6.765958   0.570426\nTR10090|c0_g2_i1_coral   0.000000   3.818995   4.809473  3.6845136   0.000000   0.570426\nTR10103|c0_g1_i1_coral   1.287197   3.818995   6.011842  3.2751232   0.000000   5.133834\nTR10108|c0_g1_i1_coral  14.159167  15.275980   2.805526  4.9126847  16.914895   5.989473\nTR10132|c0_g1_i1_coral   6.435985   2.545997   0.000000  0.8187808   3.382979   1.426065\n                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22 RI_B_07_14\nTR10083|c0_g2_i1_coral          0  2.4538305   19.55961   3.448566   18.78455  1.4936671\nTR10090|c0_g1_i1_coral          0  0.0000000    0.00000   2.069140    0.00000  0.7468335\nTR10090|c0_g2_i1_coral          0  2.4538305    0.00000   4.827992    0.00000  3.7341677\nTR10103|c0_g1_i1_coral          0  4.0897176    0.00000   3.448566    0.00000  2.2405006\nTR10108|c0_g1_i1_coral          0  0.8179435    6.51987  10.345698   40.69986  3.7341677\nTR10132|c0_g1_i1_coral          0  1.6358870    0.00000   1.379426    0.00000  8.9620025\n                       RI_B_07_18 RI_B_07_22 RI_W_01_18 RI_W_01_22 RI_W_02_14 RI_W_02_18\nTR10083|c0_g2_i1_coral   8.070182   36.85316  10.815190   4.646396   0.000000  4.9947759\nTR10090|c0_g1_i1_coral   2.017545    0.00000  12.257216   1.548799   0.000000  1.9979104\nTR10090|c0_g2_i1_coral   1.008773    0.00000  12.257216   7.743994   4.651361  4.9947759\nTR10103|c0_g1_i1_coral   2.017545    0.00000   7.931140   0.000000   0.000000  0.9989552\nTR10108|c0_g1_i1_coral   1.008773    0.00000   2.884051   1.548799   9.302721  6.9926863\nTR10132|c0_g1_i1_coral  44.386000   27.63987   1.442025   0.000000   0.000000  5.9937311\n                       RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18\nTR10083|c0_g2_i1_coral   3.615216   8.525621   5.407467   4.745758   12.82564  2.9768406\nTR10090|c0_g1_i1_coral   0.000000   4.262810   1.081493   1.581919    0.00000  3.4729807\nTR10090|c0_g2_i1_coral   4.820288   3.044865   3.244480   1.581919    0.00000  7.4421015\nTR10103|c0_g1_i1_coral   3.615216   3.044865   3.244480   1.054613    0.00000  0.9922802\nTR10108|c0_g1_i1_coral   2.410144   2.435892   4.686471   2.109226   21.37607 12.8996426\nTR10132|c0_g1_i1_coral  44.587667   1.217946   1.081493   1.054613   29.92650  7.9382416\n                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22\nTR10083|c0_g2_i1_coral  3.1630643   1.010704   5.526636   1.044347   1.336966   5.564881\nTR10090|c0_g1_i1_coral  0.9037326   8.085630   0.000000   0.000000   0.000000   4.451904\nTR10090|c0_g2_i1_coral  3.6149306  10.107037   1.381659   3.133041   1.336966   7.790833\nTR10103|c0_g1_i1_coral  4.0667969   2.021407   2.763318  10.443469   1.336966   2.225952\nTR10108|c0_g1_i1_coral  5.4223959   2.021407   6.908296  11.487816   0.000000   5.564881\nTR10132|c0_g1_i1_coral  1.8074653  10.107037  27.633182   5.221735   1.336966  42.293092\n                       RI_W_07_14 RI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10083|c0_g2_i1_coral   4.124636   5.593017   6.900578   2.551157   1.731769   2.845097\nTR10090|c0_g1_i1_coral   0.000000   0.000000   0.000000   7.653472   9.091789   1.707058\nTR10090|c0_g2_i1_coral   2.062318   1.118603   5.520462  10.204629   5.628251   5.690194\nTR10103|c0_g1_i1_coral   5.499514  13.423242   1.380116   3.826736   2.597654   2.845097\nTR10108|c0_g1_i1_coral   0.000000   1.118603   5.520462   3.826736   9.091789   2.845097\nTR10132|c0_g1_i1_coral   1.374879   2.237207  86.947282   1.275579   1.298827   0.000000\n                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14 VA_B_03_18 VA_B_03_22\nTR10083|c0_g2_i1_coral          0    14.2516  19.312637  16.029307  14.339493   7.833217\nTR10090|c0_g1_i1_coral          0     0.0000   0.000000   5.828839   2.150924   7.833217\nTR10090|c0_g2_i1_coral          0     0.0000   0.000000  10.929073   4.301848   3.560553\nTR10103|c0_g1_i1_coral          0     0.0000   2.145849   2.185815   4.301848   3.560553\nTR10108|c0_g1_i1_coral          0     0.0000   0.000000  14.572098  18.641340  21.363318\nTR10132|c0_g1_i1_coral          0     0.0000   8.583394   9.471863  30.829909   8.545327\n                       VA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10083|c0_g2_i1_coral   0.000000  0.9006809   6.731616   2.098865  3.3173156   1.813677\nTR10090|c0_g1_i1_coral   0.000000  4.5034046   3.624716   4.897352  1.1057719   7.254707\nTR10090|c0_g2_i1_coral   7.108517  3.1523832   0.000000   1.399244  0.5528859   5.441030\nTR10103|c0_g1_i1_coral   4.442823  0.0000000   0.000000   0.000000  3.3173156   0.000000\nTR10108|c0_g1_i1_coral   2.665694  4.9537450   3.624716   9.095083  5.5288593   0.000000\nTR10132|c0_g1_i1_coral   0.000000  2.7020428   0.000000   0.000000  4.9759734   5.441030\n                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22\nTR10083|c0_g2_i1_coral   1.383946   0.000000  5.9098781   3.728534  0.7405861   2.098499\nTR10090|c0_g1_i1_coral   8.303676   0.000000  0.4924898   0.000000  3.7029304   2.098499\nTR10090|c0_g2_i1_coral   0.000000   4.206352  1.4774695   1.491414 15.5523076   6.295498\nTR10103|c0_g1_i1_coral   5.535784   6.730163  3.9399187   1.491414  8.1464468   4.196999\nTR10108|c0_g1_i1_coral  11.763540   4.206352  3.4474289   2.237121  2.2217582   5.246248\nTR10132|c0_g1_i1_coral  18.683270   2.523811 14.2822054   8.948482 17.0334798   2.098499\n                       VA_W_01_14 VA_W_01_18 VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_22\nTR10083|c0_g2_i1_coral   6.699293   7.732656   0.000000  4.5789684  3.1096023   6.888125\nTR10090|c0_g1_i1_coral   3.828167   1.104665   6.978382  0.6541383  5.4418040   9.392897\nTR10090|c0_g2_i1_coral   0.000000   3.866328   4.361489 13.0827667  6.9966052   3.130966\nTR10103|c0_g1_i1_coral   0.000000   3.866328   5.233787  0.6541383  6.9966052   2.504773\nTR10108|c0_g1_i1_coral   1.914084   4.418660   0.000000  9.1579367  3.1096023   3.757159\nTR10132|c0_g1_i1_coral   0.000000  10.494319   2.616893 13.0827667  0.7774006  13.150056\n                       VA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22\nTR10083|c0_g2_i1_coral   1.026752   4.334366   6.410680  0.0000000   3.848517  12.379823\nTR10090|c0_g1_i1_coral   2.566880   5.779155   3.945034  0.0000000   1.099576  12.379823\nTR10090|c0_g2_i1_coral   2.053504   7.705540  10.848842  0.0000000   2.748941   7.427894\nTR10103|c0_g1_i1_coral   3.593631   3.852770   2.958775  0.8118702   3.298729   2.475965\nTR10108|c0_g1_i1_coral   6.673887  20.227043   8.383196  1.6237403   5.497882  38.377452\nTR10132|c0_g1_i1_coral   2.566880   0.000000   0.000000  2.4356105   8.796611   2.475965\n                       VA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22\nTR10083|c0_g2_i1_coral   5.042735  2.5895188   0.000000 10.2466895   1.127608  0.9473408\nTR10090|c0_g1_i1_coral   0.000000  0.8631729   1.764014  1.5764138   1.127608  6.6313859\nTR10090|c0_g2_i1_coral   3.025641  0.8631729   0.000000  5.5174482   3.382824  2.8420225\nTR10103|c0_g1_i1_coral   8.068376  2.5895188   4.410036  4.7292413   0.000000  1.8946817\nTR10108|c0_g1_i1_coral   4.034188  6.9053834   5.292043  3.1528275   1.127608  1.8946817\nTR10132|c0_g1_i1_coral   9.076923 29.3478796  26.460214  0.7882069   6.765648  8.5260676\n\n\u0026gt; #building heat map data\n\u0026gt; head(scaledcounts)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22\nTR10083|c0_g2_i1_coral          0  0.8671827   2.447866\nTR10090|c0_g1_i1_coral          0  2.6015481   1.223933\nTR10090|c0_g2_i1_coral          0  3.4687308   2.447866\nTR10103|c0_g1_i1_coral          0  0.8671827   4.895733\nTR10108|c0_g1_i1_coral          0  1.7343654   4.895733\nTR10132|c0_g1_i1_coral          0  6.0702789   1.223933\n                       RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10083|c0_g2_i1_coral   0.000000   3.455170   4.432653\nTR10090|c0_g1_i1_coral   0.000000  13.820681   0.000000\nTR10090|c0_g2_i1_coral   3.772786   0.000000   0.000000\nTR10103|c0_g1_i1_coral   0.000000   0.000000   8.865306\nTR10108|c0_g1_i1_coral   0.000000   6.910341   0.000000\nTR10132|c0_g1_i1_coral  18.863931  51.827555   0.000000\n                       RI_B_03_14 RI_B_03_22 RI_B_04_14\nTR10083|c0_g2_i1_coral  12.871970   1.272998  12.424473\nTR10090|c0_g1_i1_coral   0.000000   0.000000   2.003947\nTR10090|c0_g2_i1_coral   0.000000   3.818995   4.809473\nTR10103|c0_g1_i1_coral   1.287197   3.818995   6.011842\nTR10108|c0_g1_i1_coral  14.159167  15.275980   2.805526\nTR10132|c0_g1_i1_coral   6.435985   2.545997   0.000000\n                       RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10083|c0_g2_i1_coral  8.1878079   0.000000   4.563408\nTR10090|c0_g1_i1_coral  4.0939039   6.765958   0.570426\nTR10090|c0_g2_i1_coral  3.6845136   0.000000   0.570426\nTR10103|c0_g1_i1_coral  3.2751232   0.000000   5.133834\nTR10108|c0_g1_i1_coral  4.9126847  16.914895   5.989473\nTR10132|c0_g1_i1_coral  0.8187808   3.382979   1.426065\n                       RI_B_05_18 RI_B_05_22 RI_B_06_14\nTR10083|c0_g2_i1_coral          0  2.4538305   19.55961\nTR10090|c0_g1_i1_coral          0  0.0000000    0.00000\nTR10090|c0_g2_i1_coral          0  2.4538305    0.00000\nTR10103|c0_g1_i1_coral          0  4.0897176    0.00000\nTR10108|c0_g1_i1_coral          0  0.8179435    6.51987\nTR10132|c0_g1_i1_coral          0  1.6358870    0.00000\n                       RI_B_06_18 RI_B_06_22 RI_B_07_14\nTR10083|c0_g2_i1_coral   3.448566   18.78455  1.4936671\nTR10090|c0_g1_i1_coral   2.069140    0.00000  0.7468335\nTR10090|c0_g2_i1_coral   4.827992    0.00000  3.7341677\nTR10103|c0_g1_i1_coral   3.448566    0.00000  2.2405006\nTR10108|c0_g1_i1_coral  10.345698   40.69986  3.7341677\nTR10132|c0_g1_i1_coral   1.379426    0.00000  8.9620025\n                       RI_B_07_18 RI_B_07_22 RI_W_01_18\nTR10083|c0_g2_i1_coral   8.070182   36.85316  10.815190\nTR10090|c0_g1_i1_coral   2.017545    0.00000  12.257216\nTR10090|c0_g2_i1_coral   1.008773    0.00000  12.257216\nTR10103|c0_g1_i1_coral   2.017545    0.00000   7.931140\nTR10108|c0_g1_i1_coral   1.008773    0.00000   2.884051\nTR10132|c0_g1_i1_coral  44.386000   27.63987   1.442025\n                       RI_W_01_22 RI_W_02_14 RI_W_02_18\nTR10083|c0_g2_i1_coral   4.646396   0.000000  4.9947759\nTR10090|c0_g1_i1_coral   1.548799   0.000000  1.9979104\nTR10090|c0_g2_i1_coral   7.743994   4.651361  4.9947759\nTR10103|c0_g1_i1_coral   0.000000   0.000000  0.9989552\nTR10108|c0_g1_i1_coral   1.548799   9.302721  6.9926863\nTR10132|c0_g1_i1_coral   0.000000   0.000000  5.9937311\n                       RI_W_02_22 RI_W_03_14 RI_W_03_18\nTR10083|c0_g2_i1_coral   3.615216   8.525621   5.407467\nTR10090|c0_g1_i1_coral   0.000000   4.262810   1.081493\nTR10090|c0_g2_i1_coral   4.820288   3.044865   3.244480\nTR10103|c0_g1_i1_coral   3.615216   3.044865   3.244480\nTR10108|c0_g1_i1_coral   2.410144   2.435892   4.686471\nTR10132|c0_g1_i1_coral  44.587667   1.217946   1.081493\n                       RI_W_03_22 RI_W_04_14 RI_W_04_18\nTR10083|c0_g2_i1_coral   4.745758   12.82564  2.9768406\nTR10090|c0_g1_i1_coral   1.581919    0.00000  3.4729807\nTR10090|c0_g2_i1_coral   1.581919    0.00000  7.4421015\nTR10103|c0_g1_i1_coral   1.054613    0.00000  0.9922802\nTR10108|c0_g1_i1_coral   2.109226   21.37607 12.8996426\nTR10132|c0_g1_i1_coral   1.054613   29.92650  7.9382416\n                       RI_W_05_14 RI_W_05_18 RI_W_05_22\nTR10083|c0_g2_i1_coral  3.1630643   1.010704   5.526636\nTR10090|c0_g1_i1_coral  0.9037326   8.085630   0.000000\nTR10090|c0_g2_i1_coral  3.6149306  10.107037   1.381659\nTR10103|c0_g1_i1_coral  4.0667969   2.021407   2.763318\nTR10108|c0_g1_i1_coral  5.4223959   2.021407   6.908296\nTR10132|c0_g1_i1_coral  1.8074653  10.107037  27.633182\n                       RI_W_06_14 RI_W_06_18 RI_W_06_22\nTR10083|c0_g2_i1_coral   1.044347   1.336966   5.564881\nTR10090|c0_g1_i1_coral   0.000000   0.000000   4.451904\nTR10090|c0_g2_i1_coral   3.133041   1.336966   7.790833\nTR10103|c0_g1_i1_coral  10.443469   1.336966   2.225952\nTR10108|c0_g1_i1_coral  11.487816   0.000000   5.564881\nTR10132|c0_g1_i1_coral   5.221735   1.336966  42.293092\n                       RI_W_07_14 RI_W_07_18 RI_W_07_22\nTR10083|c0_g2_i1_coral   4.124636   5.593017   6.900578\nTR10090|c0_g1_i1_coral   0.000000   0.000000   0.000000\nTR10090|c0_g2_i1_coral   2.062318   1.118603   5.520462\nTR10103|c0_g1_i1_coral   5.499514  13.423242   1.380116\nTR10108|c0_g1_i1_coral   0.000000   1.118603   5.520462\nTR10132|c0_g1_i1_coral   1.374879   2.237207  86.947282\n                       VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10083|c0_g2_i1_coral   2.551157   1.731769   2.845097\nTR10090|c0_g1_i1_coral   7.653472   9.091789   1.707058\nTR10090|c0_g2_i1_coral  10.204629   5.628251   5.690194\nTR10103|c0_g1_i1_coral   3.826736   2.597654   2.845097\nTR10108|c0_g1_i1_coral   3.826736   9.091789   2.845097\nTR10132|c0_g1_i1_coral   1.275579   1.298827   0.000000\n                       VA_B_02_14 VA_B_02_18 VA_B_02_22\nTR10083|c0_g2_i1_coral          0    14.2516  19.312637\nTR10090|c0_g1_i1_coral          0     0.0000   0.000000\nTR10090|c0_g2_i1_coral          0     0.0000   0.000000\nTR10103|c0_g1_i1_coral          0     0.0000   2.145849\nTR10108|c0_g1_i1_coral          0     0.0000   0.000000\nTR10132|c0_g1_i1_coral          0     0.0000   8.583394\n                       VA_B_03_14 VA_B_03_18 VA_B_03_22\nTR10083|c0_g2_i1_coral  16.029307  14.339493   7.833217\nTR10090|c0_g1_i1_coral   5.828839   2.150924   7.833217\nTR10090|c0_g2_i1_coral  10.929073   4.301848   3.560553\nTR10103|c0_g1_i1_coral   2.185815   4.301848   3.560553\nTR10108|c0_g1_i1_coral  14.572098  18.641340  21.363318\nTR10132|c0_g1_i1_coral   9.471863  30.829909   8.545327\n                       VA_B_04_14 VA_B_04_18 VA_B_04_22\nTR10083|c0_g2_i1_coral   0.000000  0.9006809   6.731616\nTR10090|c0_g1_i1_coral   0.000000  4.5034046   3.624716\nTR10090|c0_g2_i1_coral   7.108517  3.1523832   0.000000\nTR10103|c0_g1_i1_coral   4.442823  0.0000000   0.000000\nTR10108|c0_g1_i1_coral   2.665694  4.9537450   3.624716\nTR10132|c0_g1_i1_coral   0.000000  2.7020428   0.000000\n                       VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10083|c0_g2_i1_coral   2.098865  3.3173156   1.813677\nTR10090|c0_g1_i1_coral   4.897352  1.1057719   7.254707\nTR10090|c0_g2_i1_coral   1.399244  0.5528859   5.441030\nTR10103|c0_g1_i1_coral   0.000000  3.3173156   0.000000\nTR10108|c0_g1_i1_coral   9.095083  5.5288593   0.000000\nTR10132|c0_g1_i1_coral   0.000000  4.9759734   5.441030\n                       VA_B_06_14 VA_B_06_18 VA_B_06_22\nTR10083|c0_g2_i1_coral   1.383946   0.000000  5.9098781\nTR10090|c0_g1_i1_coral   8.303676   0.000000  0.4924898\nTR10090|c0_g2_i1_coral   0.000000   4.206352  1.4774695\nTR10103|c0_g1_i1_coral   5.535784   6.730163  3.9399187\nTR10108|c0_g1_i1_coral  11.763540   4.206352  3.4474289\nTR10132|c0_g1_i1_coral  18.683270   2.523811 14.2822054\n                       VA_B_07_14 VA_B_07_18 VA_B_07_22\nTR10083|c0_g2_i1_coral   3.728534  0.7405861   2.098499\nTR10090|c0_g1_i1_coral   0.000000  3.7029304   2.098499\nTR10090|c0_g2_i1_coral   1.491414 15.5523076   6.295498\nTR10103|c0_g1_i1_coral   1.491414  8.1464468   4.196999\nTR10108|c0_g1_i1_coral   2.237121  2.2217582   5.246248\nTR10132|c0_g1_i1_coral   8.948482 17.0334798   2.098499\n                       VA_W_01_14 VA_W_01_18 VA_W_02_14\nTR10083|c0_g2_i1_coral   6.699293   7.732656   0.000000\nTR10090|c0_g1_i1_coral   3.828167   1.104665   6.978382\nTR10090|c0_g2_i1_coral   0.000000   3.866328   4.361489\nTR10103|c0_g1_i1_coral   0.000000   3.866328   5.233787\nTR10108|c0_g1_i1_coral   1.914084   4.418660   0.000000\nTR10132|c0_g1_i1_coral   0.000000  10.494319   2.616893\n                       VA_W_02_18 VA_W_03_14 VA_W_03_22\nTR10083|c0_g2_i1_coral  4.5789684  3.1096023   6.888125\nTR10090|c0_g1_i1_coral  0.6541383  5.4418040   9.392897\nTR10090|c0_g2_i1_coral 13.0827667  6.9966052   3.130966\nTR10103|c0_g1_i1_coral  0.6541383  6.9966052   2.504773\nTR10108|c0_g1_i1_coral  9.1579367  3.1096023   3.757159\nTR10132|c0_g1_i1_coral 13.0827667  0.7774006  13.150056\n                       VA_W_04_14 VA_W_04_18 VA_W_04_22\nTR10083|c0_g2_i1_coral   1.026752   4.334366   6.410680\nTR10090|c0_g1_i1_coral   2.566880   5.779155   3.945034\nTR10090|c0_g2_i1_coral   2.053504   7.705540  10.848842\nTR10103|c0_g1_i1_coral   3.593631   3.852770   2.958775\nTR10108|c0_g1_i1_coral   6.673887  20.227043   8.383196\nTR10132|c0_g1_i1_coral   2.566880   0.000000   0.000000\n                       VA_W_05_14 VA_W_05_18 VA_W_05_22\nTR10083|c0_g2_i1_coral  0.0000000   3.848517  12.379823\nTR10090|c0_g1_i1_coral  0.0000000   1.099576  12.379823\nTR10090|c0_g2_i1_coral  0.0000000   2.748941   7.427894\nTR10103|c0_g1_i1_coral  0.8118702   3.298729   2.475965\nTR10108|c0_g1_i1_coral  1.6237403   5.497882  38.377452\nTR10132|c0_g1_i1_coral  2.4356105   8.796611   2.475965\n                       VA_W_06_14 VA_W_06_18 VA_W_06_22\nTR10083|c0_g2_i1_coral   5.042735  2.5895188   0.000000\nTR10090|c0_g1_i1_coral   0.000000  0.8631729   1.764014\nTR10090|c0_g2_i1_coral   3.025641  0.8631729   0.000000\nTR10103|c0_g1_i1_coral   8.068376  2.5895188   4.410036\nTR10108|c0_g1_i1_coral   4.034188  6.9053834   5.292043\nTR10132|c0_g1_i1_coral   9.076923 29.3478796  26.460214\n                       VA_W_07_14 VA_W_07_18 VA_W_07_22\nTR10083|c0_g2_i1_coral 10.2466895   1.127608  0.9473408\nTR10090|c0_g1_i1_coral  1.5764138   1.127608  6.6313859\nTR10090|c0_g2_i1_coral  5.5174482   3.382824  2.8420225\nTR10103|c0_g1_i1_coral  4.7292413   0.000000  1.8946817\nTR10108|c0_g1_i1_coral  3.1528275   1.127608  1.8946817\nTR10132|c0_g1_i1_coral  0.7882069   6.765648  8.5260676\n\u0026gt; genes4heatmap\u0026lt;-resOrigin[resOrigin$pvalue \u0026lt;0.05 \u0026amp; !is.na(resOrigin$pvalue),]\n\u0026gt; names(genes4heatmap)\n[1] \u0026quot;baseMean\u0026quot;       \u0026quot;log2FoldChange\u0026quot; \u0026quot;lfcSE\u0026quot;         \n[4] \u0026quot;stat\u0026quot;           \u0026quot;pvalue\u0026quot;         \u0026quot;padj\u0026quot;          \n\u0026gt; head(genes4heatmap)\nlog2 fold change (MLE): Origin VA vs RI \nWald test p-value: Origin VA vs RI \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE\n                       \u0026lt;numeric\u0026gt;      \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\nTR10090|c0_g2_i1_coral   3.65756       0.943135  0.474719\nTR10487|c0_g1_i1_coral   6.57170       0.727123  0.363907\nTR1069|c0_g1_i1_coral    4.14205       4.490291  1.982381\nTR10748|c0_g1_i1_coral   2.93642       1.787984  0.559059\nTR10773|c0_g1_i1_coral   3.02516       1.758477  0.659498\nTR10773|c1_g1_i1_coral   2.32876       1.292223  0.632168\n                            stat     pvalue      padj\n                       \u0026lt;numeric\u0026gt;  \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\nTR10090|c0_g2_i1_coral   1.98672 0.04695340 0.4180430\nTR10487|c0_g1_i1_coral   1.99810 0.04570594 0.4125186\nTR1069|c0_g1_i1_coral    2.26510 0.02350656 0.3086533\nTR10748|c0_g1_i1_coral   3.19820 0.00138287 0.0720091\nTR10773|c0_g1_i1_coral   2.66639 0.00766715 0.1733954\nTR10773|c1_g1_i1_coral   2.04411 0.04094219        NA\n\u0026gt; dim(genes4heatmap)\n[1] 866   6\n\u0026gt; data4heatmap\u0026lt;-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]\n\u0026gt; dim(data4heatmap)\n[1] 866  78\n\u0026gt; head(data4heatmap)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22\nTR10090|c0_g2_i1_coral    0.00000   3.468731   2.447866\nTR10487|c0_g1_i1_coral   10.94714   5.203096   8.567532\nTR1069|c0_g1_i1_coral     0.00000   0.000000   0.000000\nTR10748|c0_g1_i1_coral    0.00000   0.000000   2.447866\nTR10773|c0_g1_i1_coral    0.00000   6.070279   2.447866\nTR10773|c1_g1_i1_coral    0.00000   2.601548   4.895733\n                       RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10090|c0_g2_i1_coral   3.772786    0.00000   0.000000\nTR10487|c0_g1_i1_coral   0.000000    0.00000   6.648979\nTR1069|c0_g1_i1_coral    7.545573    3.45517   6.648979\nTR10748|c0_g1_i1_coral   0.000000    0.00000   0.000000\nTR10773|c0_g1_i1_coral   3.772786    0.00000   0.000000\nTR10773|c1_g1_i1_coral   0.000000    0.00000   2.216326\n                       RI_B_03_14 RI_B_03_22 RI_B_04_14\nTR10090|c0_g2_i1_coral   0.000000   3.818995   4.809473\nTR10487|c0_g1_i1_coral   9.010379   5.091993   2.404737\nTR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000\nTR10748|c0_g1_i1_coral   2.574394   0.000000   4.007894\nTR10773|c0_g1_i1_coral   1.287197   0.000000   0.000000\nTR10773|c1_g1_i1_coral   5.148788   1.272998   0.000000\n                       RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10090|c0_g2_i1_coral  3.6845136          0   0.570426\nTR10487|c0_g1_i1_coral  4.5032943          0   9.126817\nTR1069|c0_g1_i1_coral   0.0000000          0   0.000000\nTR10748|c0_g1_i1_coral  2.8657328          0   1.711278\nTR10773|c0_g1_i1_coral  0.4093904          0   0.000000\nTR10773|c1_g1_i1_coral  0.0000000          0   0.285213\n                       RI_B_05_18 RI_B_05_22 RI_B_06_14\nTR10090|c0_g2_i1_coral   0.000000  2.4538305          0\nTR10487|c0_g1_i1_coral   5.033716  0.8179435          0\nTR1069|c0_g1_i1_coral    0.000000  0.0000000          0\nTR10748|c0_g1_i1_coral   0.000000  0.0000000          0\nTR10773|c0_g1_i1_coral   0.000000  0.0000000          0\nTR10773|c1_g1_i1_coral   0.000000  0.8179435          0\n                       RI_B_06_18 RI_B_06_22 RI_B_07_14\nTR10090|c0_g2_i1_coral   4.827992          0   3.734168\nTR10487|c0_g1_i1_coral   2.758853          0   5.227835\nTR1069|c0_g1_i1_coral    0.000000          0   0.000000\nTR10748|c0_g1_i1_coral   0.000000          0   2.987334\nTR10773|c0_g1_i1_coral   0.000000          0   0.000000\nTR10773|c1_g1_i1_coral   0.000000          0   0.000000\n                       RI_B_07_18 RI_B_07_22 RI_W_01_18\nTR10090|c0_g2_i1_coral   1.008773          0  12.257216\nTR10487|c0_g1_i1_coral   5.043864          0   8.652152\nTR1069|c0_g1_i1_coral    0.000000          0  10.094178\nTR10748|c0_g1_i1_coral   0.000000          0   0.000000\nTR10773|c0_g1_i1_coral   1.008773          0   7.931140\nTR10773|c1_g1_i1_coral   0.000000          0   5.047089\n                       RI_W_01_22 RI_W_02_14 RI_W_02_18\nTR10090|c0_g2_i1_coral   7.743994   4.651361  4.9947759\nTR10487|c0_g1_i1_coral   6.195195  16.279762  2.9968656\nTR1069|c0_g1_i1_coral    9.292792   0.000000  0.0000000\nTR10748|c0_g1_i1_coral   1.548799   4.651361  0.9989552\nTR10773|c0_g1_i1_coral  17.036786   0.000000 14.9843278\nTR10773|c1_g1_i1_coral  18.585585   0.000000 10.9885071\n                       RI_W_02_22 RI_W_03_14 RI_W_03_18\nTR10090|c0_g2_i1_coral   4.820288   3.044865   3.244480\nTR10487|c0_g1_i1_coral   1.205072   7.307675   7.209956\nTR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000\nTR10748|c0_g1_i1_coral   2.410144   2.435892   6.128462\nTR10773|c0_g1_i1_coral   6.025360  23.140971  12.617422\nTR10773|c1_g1_i1_coral   1.205072  10.352540   7.930951\n                       RI_W_03_22 RI_W_04_14 RI_W_04_18\nTR10090|c0_g2_i1_coral   1.581919    0.00000  7.4421015\nTR10487|c0_g1_i1_coral  14.237275   17.10086  5.4575411\nTR1069|c0_g1_i1_coral    0.000000    0.00000  0.0000000\nTR10748|c0_g1_i1_coral   4.218452    0.00000  0.4961401\nTR10773|c0_g1_i1_coral  29.001856    0.00000  1.9845604\nTR10773|c1_g1_i1_coral   7.382291    0.00000  1.4884203\n                       RI_W_05_14 RI_W_05_18 RI_W_05_22\nTR10090|c0_g2_i1_coral   3.614931  10.107037   1.381659\nTR10487|c0_g1_i1_coral   4.066797   4.042815   0.000000\nTR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000\nTR10748|c0_g1_i1_coral   2.711198   1.010704   0.000000\nTR10773|c0_g1_i1_coral   5.422396   2.021407   0.000000\nTR10773|c1_g1_i1_coral   3.614931   3.032111   0.000000\n                       RI_W_06_14 RI_W_06_18 RI_W_06_22\nTR10090|c0_g2_i1_coral   3.133041   1.336966   7.790833\nTR10487|c0_g1_i1_coral   4.177388  13.369662  10.016785\nTR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000\nTR10748|c0_g1_i1_coral   3.133041   2.673932   2.225952\nTR10773|c0_g1_i1_coral   5.221735   2.673932  10.016785\nTR10773|c1_g1_i1_coral   1.044347   2.673932   3.338928\n                       RI_W_07_14 RI_W_07_18 RI_W_07_22\nTR10090|c0_g2_i1_coral   2.062318   1.118603   5.520462\nTR10487|c0_g1_i1_coral   3.437196  13.423242   8.280694\nTR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000\nTR10748|c0_g1_i1_coral   2.749757   2.237207   0.000000\nTR10773|c0_g1_i1_coral   1.374879   2.237207   0.000000\nTR10773|c1_g1_i1_coral   2.062318   3.355810   2.760231\n                       VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10090|c0_g2_i1_coral  10.204629   5.628251   5.690194\nTR10487|c0_g1_i1_coral  10.204629   6.494135  10.242350\nTR1069|c0_g1_i1_coral   33.165045   9.524732  17.639603\nTR10748|c0_g1_i1_coral   1.275579   1.298827   5.121175\nTR10773|c0_g1_i1_coral   6.377893   4.762366   2.845097\nTR10773|c1_g1_i1_coral   8.929051   6.061193   6.259214\n                       VA_B_02_14 VA_B_02_18 VA_B_02_22\nTR10090|c0_g2_i1_coral   0.000000     0.0000   0.000000\nTR10487|c0_g1_i1_coral   0.000000     0.0000  17.166789\nTR1069|c0_g1_i1_coral   86.631891    44.0504  40.771123\nTR10748|c0_g1_i1_coral  30.575962     0.0000   2.145849\nTR10773|c0_g1_i1_coral   1.698665     0.0000   6.437546\nTR10773|c1_g1_i1_coral   0.000000     2.5912   0.000000\n                       VA_B_03_14 VA_B_03_18 VA_B_03_22\nTR10090|c0_g2_i1_coral  10.929073   4.301848  3.5605530\nTR10487|c0_g1_i1_coral  10.929073   7.886721  7.8332166\nTR1069|c0_g1_i1_coral    0.000000   0.000000  0.0000000\nTR10748|c0_g1_i1_coral   3.643024   7.886721  3.5605530\nTR10773|c0_g1_i1_coral   0.000000   1.433949  0.7121106\nTR10773|c1_g1_i1_coral   0.000000   3.584873  0.7121106\n                       VA_B_04_14 VA_B_04_18 VA_B_04_22\nTR10090|c0_g2_i1_coral  7.1085175   3.152383   0.000000\nTR10487|c0_g1_i1_coral 15.1055996   4.053064   8.285065\nTR1069|c0_g1_i1_coral   0.0000000   0.000000   0.000000\nTR10748|c0_g1_i1_coral  2.6656940   0.000000   8.802882\nTR10773|c0_g1_i1_coral  0.0000000   7.205447   8.285065\nTR10773|c1_g1_i1_coral  0.8885647   8.556469   2.589083\n                       VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10090|c0_g2_i1_coral   1.399244  0.5528859   5.441030\nTR10487|c0_g1_i1_coral   4.547541 17.1394640  21.764122\nTR1069|c0_g1_i1_coral    0.000000  0.0000000   0.000000\nTR10748|c0_g1_i1_coral   1.049433  1.6586578   3.627354\nTR10773|c0_g1_i1_coral   1.399244  1.1057719   0.000000\nTR10773|c1_g1_i1_coral   0.000000  4.4230875   0.000000\n                       VA_B_06_14 VA_B_06_18 VA_B_06_22\nTR10090|c0_g2_i1_coral   0.000000   4.206352   1.477470\nTR10487|c0_g1_i1_coral   0.000000   8.412704   2.462449\nTR1069|c0_g1_i1_coral   29.062864  16.825408   8.372327\nTR10748|c0_g1_i1_coral   6.919730   0.000000   2.954939\nTR10773|c0_g1_i1_coral   2.767892   0.000000   5.417388\nTR10773|c1_g1_i1_coral   0.000000   2.523811   2.462449\n                       VA_B_07_14 VA_B_07_18 VA_B_07_22\nTR10090|c0_g2_i1_coral  1.4914137  15.552308   6.295498\nTR10487|c0_g1_i1_coral  3.7285342   5.184103   8.393997\nTR1069|c0_g1_i1_coral   0.0000000   0.000000   0.000000\nTR10748|c0_g1_i1_coral  6.7113616   0.000000   4.196999\nTR10773|c0_g1_i1_coral  1.4914137   2.962344   6.295498\nTR10773|c1_g1_i1_coral  0.7457068   0.000000   4.196999\n                       VA_W_01_14 VA_W_01_18 VA_W_02_14\nTR10090|c0_g2_i1_coral   0.000000   3.866328  4.3614889\nTR10487|c0_g1_i1_coral   6.699293   4.970993  1.7445956\nTR1069|c0_g1_i1_coral    0.000000   0.000000  0.0000000\nTR10748|c0_g1_i1_coral   2.871126   2.209330  0.8722978\nTR10773|c0_g1_i1_coral   5.742251   1.104665  0.0000000\nTR10773|c1_g1_i1_coral   0.000000   3.866328  0.0000000\n                       VA_W_02_18 VA_W_03_14 VA_W_03_22\nTR10090|c0_g2_i1_coral  13.082767  6.9966052   3.130966\nTR10487|c0_g1_i1_coral  13.082767  3.1096023   9.392897\nTR1069|c0_g1_i1_coral    0.000000  0.0000000   0.000000\nTR10748|c0_g1_i1_coral   0.000000  0.7774006   3.757159\nTR10773|c0_g1_i1_coral   3.924830  1.5548011   2.504773\nTR10773|c1_g1_i1_coral   5.233107  0.7774006   2.504773\n                       VA_W_04_14 VA_W_04_18 VA_W_04_22\nTR10090|c0_g2_i1_coral   2.053504   7.705540 10.8488424\nTR10487|c0_g1_i1_coral   3.593631  13.003099  1.4793876\nTR1069|c0_g1_i1_coral    0.000000   0.000000  0.0000000\nTR10748|c0_g1_i1_coral   6.673887   9.150329  7.3969380\nTR10773|c0_g1_i1_coral   1.026752   0.000000  0.4931292\nTR10773|c1_g1_i1_coral   1.026752   0.000000  0.0000000\n                       VA_W_05_14 VA_W_05_18 VA_W_05_22\nTR10090|c0_g2_i1_coral    0.00000   2.748941   7.427894\nTR10487|c0_g1_i1_coral    0.00000   8.246822   8.665876\nTR1069|c0_g1_i1_coral     0.00000   0.000000   0.000000\nTR10748|c0_g1_i1_coral   14.61366   2.748941   9.903859\nTR10773|c0_g1_i1_coral    0.00000   0.000000   0.000000\nTR10773|c1_g1_i1_coral    0.00000   2.199153   1.237982\n                       VA_W_06_14 VA_W_06_18 VA_W_06_22\nTR10090|c0_g2_i1_coral   3.025641  0.8631729   0.000000\nTR10487|c0_g1_i1_coral  10.085470  1.7263459  11.466093\nTR1069|c0_g1_i1_coral    0.000000  0.0000000   0.000000\nTR10748|c0_g1_i1_coral   2.017094  6.0422105   4.410036\nTR10773|c0_g1_i1_coral   0.000000  1.7263459   0.000000\nTR10773|c1_g1_i1_coral   2.017094  0.8631729   5.292043\n                       VA_W_07_14 VA_W_07_18 VA_W_07_22\nTR10090|c0_g2_i1_coral   5.517448   3.382824   2.842023\nTR10487|c0_g1_i1_coral   7.093862   6.765648   3.789363\nTR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000\nTR10748|c0_g1_i1_coral   0.000000   3.382824   1.894682\nTR10773|c0_g1_i1_coral   0.000000   0.000000   0.000000\nTR10773|c1_g1_i1_coral   0.000000   0.000000   0.000000\n\u0026gt; \n\u0026gt; temp = as.matrix(rowMeans(data4heatmap))\n\u0026gt; head(temp)\n                           [,1]\nTR10090|c0_g2_i1_coral 3.657565\nTR10487|c0_g1_i1_coral 6.571698\nTR1069|c0_g1_i1_coral  4.142052\nTR10748|c0_g1_i1_coral 2.936423\nTR10773|c0_g1_i1_coral 3.025158\nTR10773|c1_g1_i1_coral 2.328759\n\u0026gt; scaledmatrix\u0026lt;-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)\n\u0026gt; data4heatmapscaled = data4heatmap/scaledmatrix\n\u0026gt; head(data4heatmapscaled)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22\nTR10090|c0_g2_i1_coral   0.000000  0.9483717  0.6692613\nTR10487|c0_g1_i1_coral   1.665801  0.7917431  1.3037015\nTR1069|c0_g1_i1_coral    0.000000  0.0000000  0.0000000\nTR10748|c0_g1_i1_coral   0.000000  0.0000000  0.8336218\nTR10773|c0_g1_i1_coral   0.000000  2.0065992  0.8091699\nTR10773|c1_g1_i1_coral   0.000000  1.1171391  2.1022922\n                       RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10090|c0_g2_i1_coral   1.031502  0.0000000  0.0000000\nTR10487|c0_g1_i1_coral   0.000000  0.0000000  1.0117598\nTR1069|c0_g1_i1_coral    1.821699  0.8341687  1.6052378\nTR10748|c0_g1_i1_coral   0.000000  0.0000000  0.0000000\nTR10773|c0_g1_i1_coral   1.247137  0.0000000  0.0000000\nTR10773|c1_g1_i1_coral   0.000000  0.0000000  0.9517198\n                       RI_B_03_14 RI_B_03_22 RI_B_04_14\nTR10090|c0_g2_i1_coral  0.0000000  1.0441360  1.3149387\nTR10487|c0_g1_i1_coral  1.3710885  0.7748368  0.3659232\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000\nTR10748|c0_g1_i1_coral  0.8767109  0.0000000  1.3648899\nTR10773|c0_g1_i1_coral  0.4254975  0.0000000  0.0000000\nTR10773|c1_g1_i1_coral  2.2109576  0.5466423  0.0000000\n                       RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10090|c0_g2_i1_coral  1.0073680          0  0.1559579\nTR10487|c0_g1_i1_coral  0.6852559          0  1.3888066\nTR1069|c0_g1_i1_coral   0.0000000          0  0.0000000\nTR10748|c0_g1_i1_coral  0.9759263          0  0.5827764\nTR10773|c0_g1_i1_coral  0.1353286          0  0.0000000\nTR10773|c1_g1_i1_coral  0.0000000          0  0.1224742\n                       RI_B_05_18 RI_B_05_22 RI_B_06_14\nTR10090|c0_g2_i1_coral  0.0000000  0.6708919          0\nTR10487|c0_g1_i1_coral  0.7659688  0.1244646          0\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000          0\nTR10748|c0_g1_i1_coral  0.0000000  0.0000000          0\nTR10773|c0_g1_i1_coral  0.0000000  0.0000000          0\nTR10773|c1_g1_i1_coral  0.0000000  0.3512357          0\n                       RI_B_06_18 RI_B_06_22 RI_B_07_14\nTR10090|c0_g2_i1_coral  1.3200019          0  1.0209437\nTR10487|c0_g1_i1_coral  0.4198082          0  0.7955075\nTR1069|c0_g1_i1_coral   0.0000000          0  0.0000000\nTR10748|c0_g1_i1_coral  0.0000000          0  1.0173377\nTR10773|c0_g1_i1_coral  0.0000000          0  0.0000000\nTR10773|c1_g1_i1_coral  0.0000000          0  0.0000000\n                       RI_B_07_18 RI_B_07_22 RI_W_01_18\nTR10090|c0_g2_i1_coral  0.2758045          0   3.351196\nTR10487|c0_g1_i1_coral  0.7675130          0   1.316578\nTR1069|c0_g1_i1_coral   0.0000000          0   2.436999\nTR10748|c0_g1_i1_coral  0.0000000          0   0.000000\nTR10773|c0_g1_i1_coral  0.3334612          0   2.621728\nTR10773|c1_g1_i1_coral  0.0000000          0   2.167287\n                       RI_W_01_22 RI_W_02_14 RI_W_02_18\nTR10090|c0_g2_i1_coral  2.1172541   1.271710  1.3656016\nTR10487|c0_g1_i1_coral  0.9427084   2.477254  0.4560261\nTR1069|c0_g1_i1_coral   2.2435236   0.000000  0.0000000\nTR10748|c0_g1_i1_coral  0.5274440   1.584023  0.3401946\nTR10773|c0_g1_i1_coral  5.6317019   0.000000  4.9532387\nTR10773|c1_g1_i1_coral  7.9808953   0.000000  4.7186100\n                       RI_W_02_22 RI_W_03_14 RI_W_03_18\nTR10090|c0_g2_i1_coral  1.3178956  0.8324842  0.8870602\nTR10487|c0_g1_i1_coral  0.1833730  1.1119920  1.0971223\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000\nTR10748|c0_g1_i1_coral  0.8207755  0.8295438  2.0870501\nTR10773|c0_g1_i1_coral  1.9917509  7.6495091  4.1708314\nTR10773|c1_g1_i1_coral  0.5174739  4.4455171  3.4056552\n                       RI_W_03_22 RI_W_04_14 RI_W_04_18\nTR10090|c0_g2_i1_coral  0.4325062   0.000000  2.0347150\nTR10487|c0_g1_i1_coral  2.1664531   2.602198  0.8304614\nTR1069|c0_g1_i1_coral   0.0000000   0.000000  0.0000000\nTR10748|c0_g1_i1_coral  1.4365953   0.000000  0.1689607\nTR10773|c0_g1_i1_coral  9.5868908   0.000000  0.6560188\nTR10773|c1_g1_i1_coral  3.1700530   0.000000  0.6391473\n                       RI_W_05_14 RI_W_05_18 RI_W_05_22\nTR10090|c0_g2_i1_coral  0.9883436  2.7633244  0.3777539\nTR10487|c0_g1_i1_coral  0.6188351  0.6151858  0.0000000\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000\nTR10748|c0_g1_i1_coral  0.9232995  0.3441955  0.0000000\nTR10773|c0_g1_i1_coral  1.7924342  0.6681991  0.0000000\nTR10773|c1_g1_i1_coral  1.5522989  1.3020286  0.0000000\n                       RI_W_06_14 RI_W_06_18 RI_W_06_22\nTR10090|c0_g2_i1_coral  0.8565921  0.3655346  2.1300602\nTR10487|c0_g1_i1_coral  0.6356634  2.0344306  1.5242310\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000\nTR10748|c0_g1_i1_coral  1.0669582  0.9106087  0.7580488\nTR10773|c0_g1_i1_coral  1.7261033  0.8838986  3.3111614\nTR10773|c1_g1_i1_coral  0.4484564  1.1482219  1.4337799\n                       RI_W_07_14 RI_W_07_18 RI_W_07_22\nTR10090|c0_g2_i1_coral  0.5638500  0.3058329   1.509327\nTR10487|c0_g1_i1_coral  0.5230302  2.0425836   1.260054\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000\nTR10748|c0_g1_i1_coral  0.9364308  0.7618816   0.000000\nTR10773|c0_g1_i1_coral  0.4544816  0.7395340   0.000000\nTR10773|c1_g1_i1_coral  0.8855865  1.4410293   1.185280\n                       VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10090|c0_g2_i1_coral  2.7900066  1.5387973   1.555733\nTR10487|c0_g1_i1_coral  1.5528148  0.9881975   1.558555\nTR1069|c0_g1_i1_coral   8.0069110  2.2995199   4.258662\nTR10748|c0_g1_i1_coral  0.4343988  0.4423160   1.744018\nTR10773|c0_g1_i1_coral  2.1082846  1.5742538   0.940479\nTR10773|c1_g1_i1_coral  3.8342522  2.6027562   2.687789\n                       VA_B_02_14 VA_B_02_18 VA_B_02_22\nTR10090|c0_g2_i1_coral  0.0000000   0.000000  0.0000000\nTR10487|c0_g1_i1_coral  0.0000000   0.000000  2.6122305\nTR1069|c0_g1_i1_coral  20.9152088  10.634922  9.8432176\nTR10748|c0_g1_i1_coral 10.4126549   0.000000  0.7307695\nTR10773|c0_g1_i1_coral  0.5615127   0.000000  2.1280034\nTR10773|c1_g1_i1_coral  0.0000000   1.112696  0.0000000\n                       VA_B_03_14 VA_B_03_18 VA_B_03_22\nTR10090|c0_g2_i1_coral   2.988074  1.1761509  0.9734764\nTR10487|c0_g1_i1_coral   1.663052  1.2001041  1.1919624\nTR1069|c0_g1_i1_coral    0.000000  0.0000000  0.0000000\nTR10748|c0_g1_i1_coral   1.240633  2.6858257  1.2125476\nTR10773|c0_g1_i1_coral   0.000000  0.4740081  0.2353962\nTR10773|c1_g1_i1_coral   0.000000  1.5393918  0.3057897\n                       VA_B_04_14 VA_B_04_18 VA_B_04_22\nTR10090|c0_g2_i1_coral  1.9435111  0.8618804   0.000000\nTR10487|c0_g1_i1_coral  2.2985841  0.6167454   1.260719\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000\nTR10748|c0_g1_i1_coral  0.9078031  0.0000000   2.997825\nTR10773|c0_g1_i1_coral  0.0000000  2.3818420   2.738722\nTR10773|c1_g1_i1_coral  0.3815614  3.6742606   1.111786\n                       VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10090|c0_g2_i1_coral  0.3825615  0.1511623   1.487610\nTR10487|c0_g1_i1_coral  0.6919888  2.6080725   3.311796\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000\nTR10748|c0_g1_i1_coral  0.3573847  0.5648565   1.235297\nTR10773|c0_g1_i1_coral  0.4625357  0.3655254   0.000000\nTR10773|c1_g1_i1_coral  0.0000000  1.8993321   0.000000\n                       VA_B_06_14 VA_B_06_18 VA_B_06_22\nTR10090|c0_g2_i1_coral  0.0000000   1.150042  0.4039490\nTR10487|c0_g1_i1_coral  0.0000000   1.280142  0.3747052\nTR1069|c0_g1_i1_coral   7.0165371   4.062094  2.0212992\nTR10748|c0_g1_i1_coral  2.3565165   0.000000  1.0063056\nTR10773|c0_g1_i1_coral  0.9149579   0.000000  1.7907788\nTR10773|c1_g1_i1_coral  0.0000000   1.083758  1.0574082\n                       VA_B_07_14 VA_B_07_18 VA_B_07_22\nTR10090|c0_g2_i1_coral  0.4077614  4.2520938   1.721227\nTR10487|c0_g1_i1_coral  0.5673624  0.7888529   1.277295\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000\nTR10748|c0_g1_i1_coral  2.2855567  0.0000000   1.429289\nTR10773|c0_g1_i1_coral  0.4930036  0.9792364   2.081048\nTR10773|c1_g1_i1_coral  0.3202164  0.0000000   1.802247\n                       VA_W_01_14 VA_W_01_18 VA_W_02_14\nTR10090|c0_g2_i1_coral  0.0000000  1.0570771  1.1924571\nTR10487|c0_g1_i1_coral  1.0194159  0.7564245  0.2654711\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000\nTR10748|c0_g1_i1_coral  0.9777629  0.7523882  0.2970613\nTR10773|c0_g1_i1_coral  1.8981660  0.3651595  0.0000000\nTR10773|c1_g1_i1_coral  0.0000000  1.6602522  0.0000000\n                       VA_W_02_18 VA_W_03_14 VA_W_03_22\nTR10090|c0_g2_i1_coral   3.576907  1.9129136  0.8560247\nTR10487|c0_g1_i1_coral   1.990774  0.4731810  1.4292954\nTR1069|c0_g1_i1_coral    0.000000  0.0000000  0.0000000\nTR10748|c0_g1_i1_coral   0.000000  0.2647441  1.2795019\nTR10773|c0_g1_i1_coral   1.297397  0.5139571  0.8279809\nTR10773|c1_g1_i1_coral   2.247165  0.3338261  1.0755824\n                       VA_W_04_14 VA_W_04_18 VA_W_04_22\nTR10090|c0_g2_i1_coral  0.5614402   2.106741  2.9661383\nTR10487|c0_g1_i1_coral  0.5468346   1.978651  0.2251150\nTR1069|c0_g1_i1_coral   0.0000000   0.000000  0.0000000\nTR10748|c0_g1_i1_coral  2.2727947   3.116148  2.5190299\nTR10773|c0_g1_i1_coral  0.3394044   0.000000  0.1630094\nTR10773|c1_g1_i1_coral  0.4409008   0.000000  0.0000000\n                       VA_W_05_14 VA_W_05_18 VA_W_05_22\nTR10090|c0_g2_i1_coral   0.000000  0.7515768   2.030831\nTR10487|c0_g1_i1_coral   0.000000  1.2548999   1.318666\nTR1069|c0_g1_i1_coral    0.000000  0.0000000   0.000000\nTR10748|c0_g1_i1_coral   4.976688  0.9361528   3.372763\nTR10773|c0_g1_i1_coral   0.000000  0.0000000   0.000000\nTR10773|c1_g1_i1_coral   0.000000  0.9443452   0.531606\n                       VA_W_06_14 VA_W_06_18 VA_W_06_22\nTR10090|c0_g2_i1_coral  0.8272283  0.2359966   0.000000\nTR10487|c0_g1_i1_coral  1.5346826  0.2626940   1.744769\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000\nTR10748|c0_g1_i1_coral  0.6869221  2.0576770   1.501839\nTR10773|c0_g1_i1_coral  0.0000000  0.5706631   0.000000\nTR10773|c1_g1_i1_coral  0.8661668  0.3706578   2.272473\n                       VA_W_07_14 VA_W_07_18 VA_W_07_22\nTR10090|c0_g2_i1_coral   1.508503  0.9248843  0.7770259\nTR10487|c0_g1_i1_coral   1.079457  1.0295131  0.5766186\nTR1069|c0_g1_i1_coral    0.000000  0.0000000  0.0000000\nTR10748|c0_g1_i1_coral   0.000000  1.1520220  0.6452345\nTR10773|c0_g1_i1_coral   0.000000  0.0000000  0.0000000\nTR10773|c1_g1_i1_coral   0.000000  0.0000000  0.0000000\n\u0026gt; \n\u0026gt; dim(data4heatmapscaled)\n[1] 866  78\n\u0026gt; \n\u0026gt; pairs.breaks \u0026lt;- seq(0, 3.0, by=0.1)\n\u0026gt; length(pairs.breaks)\n[1] 31\n\u0026gt; mycol \u0026lt;- colorpanel(n=30, low=\u0026quot;black\u0026quot;, high=\u0026quot;yellow\u0026quot;) \n\u0026gt; \n\u0026gt; pdf(file=\u0026quot;XXXXXX_byrow.pdf\u0026quot;,7,7)\n\u0026gt; heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(\u0026quot;row\u0026quot;), scale=\u0026quot;none\u0026quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u0026quot;none\u0026quot;, symkey = F, density.info = \u0026quot;density\u0026quot;, colsep=c(24), sepcolor=c(\u0026quot;white\u0026quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\n\u0026gt; dev.off()\nRStudioGD \n        2 \n\u0026gt; \n\u0026gt; pdf(file=\u0026quot;XXXXXX_bycolumn.pdf\u0026quot;,7,7)\n\u0026gt; heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(\u0026quot;col\u0026quot;), scale=\u0026quot;none\u0026quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u0026quot;none\u0026quot;, symkey = F, density.info = \u0026quot;density\u0026quot;, colsep=c(24), sepcolor=c(\u0026quot;white\u0026quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\n\u0026gt; dev.off()\nRStudioGD \n        2 \n\u0026gt; heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(\u0026quot;row\u0026quot;), scale=\u0026quot;none\u0026quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u0026quot;none\u0026quot;, symkey = F, density.info = \u0026quot;density\u0026quot;, colsep=c(24), sepcolor=c(\u0026quot;white\u0026quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\nError in plot.new() : figure margins too large\n\u0026gt; heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(\u0026quot;col\u0026quot;), scale=\u0026quot;none\u0026quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u0026quot;none\u0026quot;, symkey = F, density.info = \u0026quot;density\u0026quot;, colsep=c(24), sepcolor=c(\u0026quot;white\u0026quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\nError in .External.graphics(C_layout, num.rows, num.cols, mat, as.integer(num.figures),  : \n  invalid graphics state\n\u0026gt; plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u0026quot;Origin\u0026quot;)\n\u0026gt; plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u0026quot;Origin\u0026quot;)\n\u0026gt; plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u0026quot;Origin\u0026quot;)\n\u0026gt; plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u0026quot;Origin_SymState\u0026quot;)\n\u0026gt; plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u0026quot;Origin_SymState_Temp\u0026quot;)\n\u0026gt; pdf(file=\u0026quot;866_RIvsVA_PCAv2.pdf\u0026quot;,14, 14)\n\u0026gt; plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u0026quot;Origin\u0026quot;)\n\u0026gt; dev.off()\nRStudioGD \n        2 \n\u0026gt; prop.nullv3 \u0026lt;- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100*mean(x==0))\n\u0026gt; barplot(prop.nullv3, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u0026gt; pdf(file=\u0026quot;ResNullCountsv2.pdf\u0026quot;,14, 14)\n\u0026gt; barplot(prop.nullv3, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u0026gt; dev.off()\nRStudioGD \n        2 \n\u0026gt; barplot(prop.nullv3, main=\u0026quot;Percentage of null counts per sample\u0026quot;, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u0026gt; distsv2\u0026lt;-dist(t(assay(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),])))\n\u0026gt; plot(hclust(distsv2))\n\u0026gt; heatmap.2(as.matrix(distsv2), key=F, trace=\u0026quot;none\u0026quot;,\n+           col=colorpanel(100, \u0026quot;black\u0026quot;, \u0026quot;white\u0026quot;),\n+           margin=c(10, 10))\n\u0026gt; #### GOSeq analysis ####\n\u0026gt; res\u0026lt;-resOrigin\n\u0026gt; Annotations\u0026lt;-read.delim(\u0026quot;Assembly_GoodCoralSymbiont_suffixed_totalannotated.txt\u0026quot;)\n\u0026gt; head(res)\nlog2 fold change (MLE): Origin VA vs RI \nWald test p-value: Origin VA vs RI \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange\n                       \u0026lt;numeric\u0026gt;      \u0026lt;numeric\u0026gt;\nTR10083|c0_g2_i1_coral   5.36648     -0.2712427\nTR10090|c0_g1_i1_coral   2.70436      1.1125117\nTR10090|c0_g2_i1_coral   3.65756      0.9431346\nTR10103|c0_g1_i1_coral   2.91279      0.0386643\nTR10108|c0_g1_i1_coral   6.31144     -0.0179010\nTR10132|c0_g1_i1_coral   9.26737     -0.2649903\n                           lfcSE       stat    pvalue\n                       \u0026lt;numeric\u0026gt;  \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\nTR10083|c0_g2_i1_coral  0.446225 -0.6078616 0.5432793\nTR10090|c0_g1_i1_coral  0.589306  1.8878328 0.0590484\nTR10090|c0_g2_i1_coral  0.474719  1.9867203 0.0469534\nTR10103|c0_g1_i1_coral  0.456720  0.0846565 0.9325345\nTR10108|c0_g1_i1_coral  0.461958 -0.0387502 0.9690895\nTR10132|c0_g1_i1_coral  0.677608 -0.3910675 0.6957473\n                            padj\n                       \u0026lt;numeric\u0026gt;\nTR10083|c0_g2_i1_coral  0.865661\nTR10090|c0_g1_i1_coral        NA\nTR10090|c0_g2_i1_coral  0.418043\nTR10103|c0_g1_i1_coral  0.984590\nTR10108|c0_g1_i1_coral  0.992150\nTR10132|c0_g1_i1_coral  0.921517\n\u0026gt; sum(res$log2FoldChange!=0)\n[1] 7659\n\u0026gt; \n\u0026gt; genes\u0026lt;-as.integer(p.adjust(res$pvalue[res$log2FoldChange!=0 \u0026amp; !is.na(res$pvalue)], method=\u0026quot;BH\u0026quot;)\u0026lt;.05)\n\u0026gt; names(genes)\u0026lt;-row.names(res[res$log2FoldChange!=0 \u0026amp; !is.na(res$pvalue),])\n\u0026gt; table(genes)\ngenes\n   0    1 \n7533  116 \n\u0026gt; \n\u0026gt; genes\u0026lt;-as.integer(res$pvalue[res$log2FoldChange!=0 \u0026amp; !is.na(res$pvalue)]\u0026lt;.05)\n\u0026gt; names(genes)\u0026lt;-row.names(res[res$log2FoldChange!=0 \u0026amp; !is.na(res$pvalue),])\n\u0026gt; table(genes)\ngenes\n   0    1 \n6783  866 \n\u0026gt; \n\u0026gt; goterms\u0026lt;-strsplit(as.character(Annotations$GO), split=\u0026quot; // \u0026quot;)\n\u0026gt; names(goterms)\u0026lt;-Annotations$ContigName\n\u0026gt; \n\u0026gt; pwf\u0026lt;-nullp(genes, bias.data=Annotations[Annotations$ContigName%in%names(genes),\u0026quot;ContigLength\u0026quot;])\nWarning message:\nIn pcls(G) : initial point very close to some inequality constraints\n\u0026gt; GOGOGO\u0026lt;-goseq(pwf,gene2cat=goterms)\nUsing manually entered categories.\nCalculating the p-values...\n'select()' returned 1:1 mapping between keys\nand columns\n\u0026gt; GOGOGO$padj\u0026lt;-p.adjust(GOGOGO$over_represented_pvalue, method=\u0026quot;fdr\u0026quot;)\n\u0026gt; \n\u0026gt; sum(GOGOGO$padj\u0026lt;0.2, na.rm=T)\n[1] 3\n\u0026gt; sum(GOGOGO$padj\u0026lt;0.1, na.rm=T)\n[1] 0\n\u0026gt; sum(GOGOGO$padj\u0026lt;0.05, na.rm=T)\n[1] 0\n\u0026gt; GOGOGO[GOGOGO$padj\u0026lt;0.05,]\n[1] category                 over_represented_pvalue \n[3] under_represented_pvalue numDEInCat              \n[5] numInCat                 term                    \n[7] ontology                 padj                    \n\u0026lt;0 rows\u0026gt; (or 0-length row.names)\n\u0026gt; \n\u0026gt; GOTERM[[\u0026quot;SOMEGOTERM\u0026quot;]]\nNULL\n\u0026gt; \n\u0026gt; EnGOS\u0026lt;-GOGOGO$category[p.adjust(GOGOGO$over_represented_pvalue, method=\u0026quot;BH\u0026quot;)\u0026lt;.05]\n\u0026gt; \n\u0026gt; for(go in EnGOS){\n+   print(GOTERM[[go]])\n+   cat(\u0026quot;--------------------------------------\\n\u0026quot;)}\n\u0026gt; names(Annotations)\n [1] \u0026quot;ContigName\u0026quot;             \n [2] \u0026quot;ContigLength\u0026quot;           \n [3] \u0026quot;topnrMatch\u0026quot;             \n [4] \u0026quot;topnrEvalue\u0026quot;            \n [5] \u0026quot;topnrAlignmentLength\u0026quot;   \n [6] \u0026quot;topnrPercMatch\u0026quot;         \n [7] \u0026quot;nobadwordnrMatch\u0026quot;       \n [8] \u0026quot;nobadwordnrEvalue\u0026quot;      \n [9] \u0026quot;nobadwordnrAlignLength\u0026quot; \n[10] \u0026quot;nobadwordnrPercentMatch\u0026quot;\n[11] \u0026quot;Uniprotmatch\u0026quot;           \n[12] \u0026quot;X._identity\u0026quot;            \n[13] \u0026quot;evalue\u0026quot;                 \n[14] \u0026quot;ID\u0026quot;                     \n[15] \u0026quot;Description\u0026quot;            \n[16] \u0026quot;KEGG\u0026quot;                   \n[17] \u0026quot;KEGGKO\u0026quot;                 \n[18] \u0026quot;PFAM\u0026quot;                   \n[19] \u0026quot;GO\u0026quot;                     \n[20] \u0026quot;GOCC\u0026quot;                   \n[21] \u0026quot;GOBP\u0026quot;                   \n[22] \u0026quot;GOMF\u0026quot;                   \n[23] \u0026quot;Keywords\u0026quot;               \n\u0026gt; head(genes4heatmap)\nlog2 fold change (MLE): Origin VA vs RI \nWald test p-value: Origin VA vs RI \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange\n                       \u0026lt;numeric\u0026gt;      \u0026lt;numeric\u0026gt;\nTR10090|c0_g2_i1_coral   3.65756       0.943135\nTR10487|c0_g1_i1_coral   6.57170       0.727123\nTR1069|c0_g1_i1_coral    4.14205       4.490291\nTR10748|c0_g1_i1_coral   2.93642       1.787984\nTR10773|c0_g1_i1_coral   3.02516       1.758477\nTR10773|c1_g1_i1_coral   2.32876       1.292223\n                           lfcSE      stat     pvalue\n                       \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;  \u0026lt;numeric\u0026gt;\nTR10090|c0_g2_i1_coral  0.474719   1.98672 0.04695340\nTR10487|c0_g1_i1_coral  0.363907   1.99810 0.04570594\nTR1069|c0_g1_i1_coral   1.982381   2.26510 0.02350656\nTR10748|c0_g1_i1_coral  0.559059   3.19820 0.00138287\nTR10773|c0_g1_i1_coral  0.659498   2.66639 0.00766715\nTR10773|c1_g1_i1_coral  0.632168   2.04411 0.04094219\n                            padj\n                       \u0026lt;numeric\u0026gt;\nTR10090|c0_g2_i1_coral 0.4180430\nTR10487|c0_g1_i1_coral 0.4125186\nTR1069|c0_g1_i1_coral  0.3086533\nTR10748|c0_g1_i1_coral 0.0720091\nTR10773|c0_g1_i1_coral 0.1733954\nTR10773|c1_g1_i1_coral        NA\n\u0026gt; dim(genes4heatmap)\n[1] 866   6\n\u0026gt; \n\u0026gt; ResultsOutput\u0026lt;-cbind(\u0026quot;ContigName\u0026quot;=row.names(genes4heatmap),genes4heatmap, Annotations[Annotations$ContigName%in%row.names(genes4heatmap),])\n\u0026gt; dim(ResultsOutput)\n[1] 866  30\n\u0026gt; head(ResultsOutput)\nDataFrame with 6 rows and 30 columns\n              ContigName  baseMean log2FoldChange\n             \u0026lt;character\u0026gt; \u0026lt;numeric\u0026gt;      \u0026lt;numeric\u0026gt;\n1 TR10090|c0_g2_i1_coral   3.65756       0.943135\n2 TR10487|c0_g1_i1_coral   6.57170       0.727123\n3  TR1069|c0_g1_i1_coral   4.14205       4.490291\n4 TR10748|c0_g1_i1_coral   2.93642       1.787984\n5 TR10773|c0_g1_i1_coral   3.02516       1.758477\n6 TR10773|c1_g1_i1_coral   2.32876       1.292223\n      lfcSE      stat     pvalue      padj\n  \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;  \u0026lt;numeric\u0026gt; \u0026lt;numeric\u0026gt;\n1  0.474719   1.98672 0.04695340 0.4180430\n2  0.363907   1.99810 0.04570594 0.4125186\n3  1.982381   2.26510 0.02350656 0.3086533\n4  0.559059   3.19820 0.00138287 0.0720091\n5  0.659498   2.66639 0.00766715 0.1733954\n6  0.632168   2.04411 0.04094219        NA\n              ContigName ContigLength\n             \u0026lt;character\u0026gt;    \u0026lt;integer\u0026gt;\n1 TR10090|c0_g2_i1_coral          772\n2 TR10487|c0_g1_i1_coral         1007\n3  TR1069|c0_g1_i1_coral          815\n4 TR10748|c0_g1_i1_coral          516\n5 TR10773|c0_g1_i1_coral          580\n6 TR10773|c1_g1_i1_coral          586\n              topnrMatch  topnrEvalue\n             \u0026lt;character\u0026gt;  \u0026lt;character\u0026gt;\n1 gnl|BL_ORD_ID|738431..  7.27758e-78\n2 gnl|BL_ORD_ID|110398..  3.65321e-84\n3 gnl|BL_ORD_ID|110400.. 9.23041e-133\n4 gnl|BL_ORD_ID|110376..  2.29747e-78\n5 gnl|BL_ORD_ID|110383..  2.14754e-64\n6 gnl|BL_ORD_ID|110383..  1.33696e-65\n  topnrAlignmentLength topnrPercMatch\n           \u0026lt;character\u0026gt;    \u0026lt;character\u0026gt;\n1                  258           0.53\n2                  201           0.88\n3                  271           0.73\n4                  172           0.85\n5                  206           0.55\n6                  196           0.60\n        nobadwordnrMatch nobadwordnrEvalue\n             \u0026lt;character\u0026gt;       \u0026lt;character\u0026gt;\n1 gnl|BL_ORD_ID|726506..       3.99079e-52\n2 gnl|BL_ORD_ID|110398..       3.65321e-84\n3 gnl|BL_ORD_ID|110400..      9.23041e-133\n4 gnl|BL_ORD_ID|110376..       2.29747e-78\n5 gnl|BL_ORD_ID|110383..       2.14754e-64\n6 gnl|BL_ORD_ID|110383..       1.33696e-65\n  nobadwordnrAlignLength nobadwordnrPercentMatch\n             \u0026lt;character\u0026gt;             \u0026lt;character\u0026gt;\n1                    264                    0.45\n2                    201                    0.88\n3                    271                    0.73\n4                    172                    0.85\n5                    206                    0.55\n6                    196                    0.60\n  Uniprotmatch X._identity      evalue\n   \u0026lt;character\u0026gt; \u0026lt;character\u0026gt; \u0026lt;character\u0026gt;\n1   A0A1X7UWJ8      40.927    1.82e-51\n2       Q4R6F3      37.500    1.77e-06\n3       Q8N2E2      31.022    1.60e-30\n4   A0A2B4RJ64      70.690    9.57e-61\n5   A0A2B4RML7      46.635    3.49e-44\n6   A0A2B4RML7      47.847    2.78e-48\n                ID            Description\n       \u0026lt;character\u0026gt;            \u0026lt;character\u0026gt;\n1 A0A1X7UWJ8_AMPQE Uncharacterized prot..\n2      NGLY1_MACFA Peptide-N(4)-(N-acet..\n3       VWDE_HUMAN von Willebrand facto..\n4 A0A2B4RJ64_STYPI Uncharacterized prot..\n5 A0A2B4RML7_STYPI Uncharacterized prot..\n6 A0A2B4RML7_STYPI Uncharacterized prot..\n           KEGG      KEGGKO                   PFAM\n    \u0026lt;character\u0026gt; \u0026lt;character\u0026gt;            \u0026lt;character\u0026gt;\n1       No_KEGG   No_KEGGKO       PF05585; DUF1758\n2 mcf:101925902      K01456 PF04721; PAW,  PF094..\n3    hsa:221806   No_KEGGKO           PF00094; VWD\n4       No_KEGG   No_KEGGKO                   Pfam\n5       No_KEGG   No_KEGGKO                   Pfam\n6       No_KEGG   No_KEGGKO                   Pfam\n                      GO                   GOCC\n             \u0026lt;character\u0026gt;            \u0026lt;character\u0026gt;\n1             GO:0004190                No_GOBP\n2 GO:0005737 // GO:004.. glycoprotein catabol..\n3 GO:0009986 // GO:000.. anatomical structure..\n4             GO:0016021                No_GOBP\n5             No_GOcodes                No_GOBP\n6             No_GOcodes                No_GOBP\n                    GOBP                   GOMF\n             \u0026lt;character\u0026gt;            \u0026lt;character\u0026gt;\n1 aspartic-type endope..                No_GOCC\n2 metal ion binding (G.. cytoplasm (GO:000573..\n3 signaling receptor b.. cell surface (GO:000..\n4                No_GOMF integral component o..\n5                No_GOMF                No_GOCC\n6                No_GOMF                No_GOCC\n                Keywords\n             \u0026lt;character\u0026gt;\n1 Complete proteome {E..\n2 Acetylation;Cytoplas..\n3 Alternative splicing..\n4 Coiled coil {ECO:000..\n5 Complete proteome {E..\n6 Complete proteome {E..\n\u0026gt; write.table(ResultsOutput, file=\u0026quot;177OriginDEGswAnnotation.txt\u0026quot;, quote=F, row.names=F, sep=\u0026quot;\\t\u0026quot;)\n\u0026gt; \n\n\n\n#9- Optional, also plot a PCA using only the genes with a pvalue\u0026lt;0.05 for the \u0026quot;origin\u0026quot; and \u0026quot;symstate\u0026quot; variables\n\n#10- Even more optional, color code your symbols by temperature treatment as well as origin and symstate.\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1- mkdir a directory with your name in the \u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2- cp/mv your logfile to the /cm/shared/courses/dbarshis/18AdvBioinf/archive/YOURNAME directory\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3- cp/mv your merged, unfiltered .vcf file YOURINITIALS_unfiltered.vcf into the /cm/shared/courses/dbarshis/18AdvBioinf/archive/YOURNAME directory\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4- cp/mv any other files you might want to keep to the /cm/shared/courses/dbarshis/18AdvBioinf/archive/YOURNAME directory\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e5- create an sbatch script to rm -r your sandbox/YOURNAME directory\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e6- scp/winscp /cm/shared/courses/dbarshis/18AdvBioinf/classdata/Astrangia_poculata/expressioncounts/ALLLANESFullCounts_summed.txt to your laptop\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e7- scp/winscp /cm/shared/courses/dbarshis/18AdvBioinf/classdata/Astrangia_poculata/expressioncounts/ALLLANES_conditions.txt to your laptop\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e8- Download the DESeq2Script_advbioinf_day12.R script and work through to generate a PCA of your data.\u003c/span\u003e\n\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e library(DESeq2)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e library(gplots)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e library(goseq)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e library(GO.db)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e setwd(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e/Users/katiecrider/Desktop/Genomics\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e The drag and drop from finder works in R, too.\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003euseful functions\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ehead() - prints out the top 6 lines\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003edim() - prints the dimensions of a variable\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003enrow() - returns the number of rows in a vector or matrix\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e ?[functionName] - opens documentation describing the function\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eread in your data to make counts table\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e countsTable \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- read.delim(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003edjbFullCounts_summed.txt\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, header=TRUE, stringsAsFactors=TRUE, row.names=1)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(countsTable)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          0          0          0\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0          1          6          0          0          0\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0          1          2          0          1          2\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          3          1          0          4          0\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0          4          2          1          0          0\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          1          4          0          0          4\n                       RI_B_03_14 RI_B_03_18 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          0          0          0\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0          0          0          9          1          0\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral         10          0          1         31         20          0\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          5         10          2\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0          0          3         12          9          0\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          1          0          3         15          8          0\n                       RI_B_05_14 RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          0          0          0\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          2          0          0          0          2          0\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral         16          0          3          3          5          6\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          2          0          0          0          3          0\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          2          0          3          0          7          0\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral         18          0          5          0          5          0\n                       RI_B_07_14 RI_B_07_18 RI_B_07_22 RI_W_01_14 RI_W_01_18 RI_W_01_22\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          0          0          0\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          2          1          0          0          0          1\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          2          8          4         11         15          3\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          1          2          0          0         17          1\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          5          1          0          0         17          5\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          3          2          0          5         11          0\n                       RI_W_02_14 RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          0          0          0\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          1          1          2          0          5          0\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0          5          3         14         15          9\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          2          0          7          3          3\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          2          5          4          5          9          3\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          1          3          5          9          2\n                       RI_W_04_14 RI_W_04_18 RI_W_04_22 RI_W_05_14 RI_W_05_18 RI_W_05_22\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          0          0          0\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          1          1          0          2          0          0\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          3          6          1          7          1          4\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          7          0          2          8          0\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0         15          0          8         10          1\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          2          0          9          2          2\n                       RI_W_06_14 RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18 RI_W_07_22\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          0          0          0\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          2          6          0          4          0          0\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          1          1          5          6          5          5\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          4          0          0          0\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          3          1          7          3          1          4\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral         10          1          2          8         12          1\n                       VA_B_01_14 VA_B_01_18 VA_B_01_22 VA_B_02_14 VA_B_02_18 VA_B_02_22\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          0          0          0\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0          2          2          0          0          0\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          2          4          5          0         11          9\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          6         21          3          0          0          0\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          8         13         10          0          0          0\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          3          6          5          0          0          1\n                       VA_B_03_14 VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18 VA_B_04_22\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          0          0          0\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          1          0          6          0         10          4\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral         22         20         11          0          2         13\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          8          3         11          0         10          7\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral         15          6          5          8          7          0\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          3          6          5          5          0          0\n                       VA_B_05_14 VA_B_05_18 VA_B_05_22 VA_B_06_14 VA_B_06_18 VA_B_06_22\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          0          0          0\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          2          2          0         10          0          1\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          6          6          1          2          0         12\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral         14          2          4         12          0          1\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          4          1          3          0          5          3\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          6          0          8          8          8\n                       VA_B_07_14 VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18 VA_W_01_22\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          0          0          0\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0          4          1          1          3          0\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          5          1          2          7         14          0\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          5          2          4          2          0\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          2         21          6          0          7          0\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          2         11          4          0          7          0\n                       VA_W_02_14 VA_W_02_18 VA_W_02_22 VA_W_03_14 VA_W_03_18 VA_W_03_22\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          0          0          0\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          4          0          0          4          0          6\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0          7          0          4          0         11\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          8          1          0          7          0         15\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          5         20          0          9         17          5\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          6          1          0          9          0          4\n                       VA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          0          0          0\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          2          0          2          0          3          0\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          2          9         13          0          7         10\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          5         12          8          0          2         10\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          4         16         22          0          5          6\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          7          8          6          1          6          2\n                       VA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          0          0          0          0          0\nTR10054\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          2          0          0          0          0          3\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          5          3          0         13          1          1\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0          1          2          2          1          7\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          3          1          0          7          3          3\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          8          3          5          6          0          2\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dim(countsTable)\n[1] 13381    84\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e colSums(countsTable)\nRI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_18 \n     97406     207372     173843      47481      58036      84501     170330       5972 \nRI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14 RI_B_05_18 RI_B_05_22 RI_B_06_14 \n    199204     357124     544267      71600     630609     227313     225639      37012 \nRI_B_06_18 RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22 RI_W_01_14 RI_W_01_18 RI_W_01_22 \n    296071      72254     257730     218198      25369      39419     292612     174666 \nRI_W_02_14 RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18 \n     54184     192092     155178     281297     428532     359053      37579     300578 \nRI_W_04_22 RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22 RI_W_07_14 \n     10848     392818     214887     158683     186568     205135     198272     342795 \nRI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22 VA_B_02_14 VA_B_02_18 VA_B_02_22 \n    220630     176946     261824     548448     328198      79276     152742      85224 \nVA_B_03_14 VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 \n    226784     251696     271959     172636     264192     268290     323715     288121 \nVA_B_05_22 VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22 VA_W_01_14 \n     89943     232578     195521     362054     215616     234465     169742     164874 \nVA_W_01_18 VA_W_01_22 VA_W_02_14 VA_W_02_18 VA_W_02_22 VA_W_03_14 VA_W_03_18 VA_W_03_22 \n    321900        313     170250     241804        134     278000     138573     295382 \nVA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14 VA_W_06_18 \n    292041     280007     261344     178096     388532     157123     176579     188506 \nVA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22 \n    209003     228792     171481     233899 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e colSums(countsTable)[order(colSums(countsTable))]\nVA_W_02_22 VA_W_01_22 RI_B_03_18 RI_W_04_22 RI_B_07_22 RI_B_06_14 RI_W_04_14 RI_W_01_14 \n       134        313       5972      10848      25369      37012      37579      39419 \nRI_B_02_14 RI_W_02_14 RI_B_02_18 RI_B_04_22 RI_B_06_22 VA_B_02_14 RI_B_02_22 VA_B_02_22 \n     47481      54184      58036      71600      72254      79276      84501      85224 \nVA_B_05_22 RI_B_01_14 VA_W_03_18 VA_B_02_18 RI_W_02_22 VA_W_05_22 RI_W_05_22 VA_W_01_14 \n     89943      97406     138573     152742     155178     157123     158683     164874 \nVA_B_07_22 VA_W_02_14 RI_B_03_14 VA_W_07_18 VA_B_04_14 RI_B_01_22 RI_W_01_22 VA_W_06_14 \n    169742     170250     170330     171481     172636     173843     174666     176579 \nRI_W_07_22 VA_W_05_14 RI_W_06_14 VA_W_06_18 RI_W_02_18 VA_B_06_18 RI_W_06_22 RI_B_03_22 \n    176946     178096     186568     188506     192092     195521     198272     199204 \nRI_W_06_18 RI_B_01_18 VA_W_06_22 RI_W_05_18 VA_B_07_14 RI_B_07_18 RI_W_07_18 RI_B_05_22 \n    205135     207372     209003     214887     215616     218198     220630     225639 \nVA_B_03_14 RI_B_05_18 VA_W_07_14 VA_B_06_14 VA_W_07_22 VA_B_07_18 VA_W_02_18 VA_B_03_18 \n    226784     227313     228792     232578     233899     234465     241804     251696 \nRI_B_07_14 VA_W_04_22 VA_B_01_14 VA_B_04_18 VA_B_04_22 VA_B_03_22 VA_W_03_14 VA_W_04_18 \n    257730     261344     261824     264192     268290     271959     278000     280007 \nRI_W_03_14 VA_B_05_18 VA_W_04_14 RI_W_01_18 VA_W_03_22 RI_B_06_18 RI_W_04_18 VA_W_01_18 \n    281297     288121     292041     292612     295382     296071     300578     321900 \nVA_B_05_14 VA_B_01_22 RI_W_07_14 RI_B_04_14 RI_W_03_22 VA_B_06_22 VA_W_05_18 RI_W_05_14 \n    323715     328198     342795     357124     359053     362054     388532     392818 \nRI_W_03_18 RI_B_04_18 VA_B_01_18 RI_B_05_14 \n    428532     544267     548448     630609 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e table(substring(names(countsTable),1,2))\n\nRI VA \n42 42 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e table(substring(names(countsTable),4,4))\n\n B  W \n42 42 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e table(substring(names(countsTable),9,10))\n\n14 18 22 \n28 28 28 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e table(paste(substring(names(countsTable),1,2),substring(names(countsTable),4,4), sep=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\nRI_B RI_W VA_B VA_W \n  21   21   21   21 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e table(paste(substring(names(countsTable),1,2),substring(names(countsTable),4,4), substring(names(countsTable),9,10), sep=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\nRI_B_14 RI_B_18 RI_B_22 RI_W_14 RI_W_18 RI_W_22 VA_B_14 VA_B_18 VA_B_22 VA_W_14 VA_W_18 VA_W_22 \n      7       7       7       7       7       7       7       7       7       7       7       7 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e LowCounts\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_W_02_22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_W_01_22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_B_03_18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_W_01_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI_W_04_22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA_W_03_18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e names(countsTable)%in%LowCounts\n [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n[16] FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n[31] FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n[46] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n[61] FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE FALSE  TRUE FALSE FALSE FALSE FALSE\n[76] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e(names(countsTable)%in%LowCounts)\n [1]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[16]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[31]  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[46]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[61]  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE FALSE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE\n[76]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e countsTable\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-countsTable[,\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e(names(countsTable)%in%LowCounts)]\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dim(countsTable)\n[1] 13381    78\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003emake a table with conditions of each individual (e.g. \"VA\" and \"RI\")  There should be the same number of conditions described as there are samples in your data file, and in the same order.\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eNOTE: It is absolutely critical that the columns of the count matrix and the rows of the column data (information about samples) are in the same order. DESeq2 will not make guesses as to which column of the count matrix belongs to which row of the column data, these must be provided to DESeq2 already in consistent order.\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eHere's an example\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Sample\t\tOrigin\tSymState\tTemp  Origin_SymState Origin_SymState_Temp\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e RI_B_06_18\t\tRI\tB\t18  RI_B  RI_B_18\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e VA_B_07_14\t\tVA\tB\t14  VA_B  VA_B_14\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e VA_W_07_22\t\tVA\tW\t22  VA_W  VA_W_22\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e safest way to do this is in R\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e conds\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-data.frame(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e=names(countsTable))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e conds\u003cspan class=\"pl-smi\"\u003e$Origin\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(substring(conds\u003cspan class=\"pl-smi\"\u003e$Sample\u003c/span\u003e,1,2))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e conds\u003cspan class=\"pl-smi\"\u003e$SymState\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(substring(conds\u003cspan class=\"pl-smi\"\u003e$Sample\u003c/span\u003e,4,4))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e conds\u003cspan class=\"pl-smi\"\u003e$Temp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(substring(conds\u003cspan class=\"pl-smi\"\u003e$Sample\u003c/span\u003e,9,10))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(paste(conds\u003cspan class=\"pl-smi\"\u003e$Origin\u003c/span\u003e,conds\u003cspan class=\"pl-smi\"\u003e$SymState\u003c/span\u003e, sep=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-factor(paste(conds\u003cspan class=\"pl-smi\"\u003e$Origin\u003c/span\u003e,conds\u003cspan class=\"pl-smi\"\u003e$SymState\u003c/span\u003e,conds\u003cspan class=\"pl-smi\"\u003e$Temp\u003c/span\u003e, sep=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e#Check null counts per sample\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e prop.null \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- apply(countsTable, 2, function(x) 100\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003emean(x==0))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e barplot(prop.null, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOverallNullCountsv2.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,14, 14)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e barplot(prop.null, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\nRStudioGD \n        2 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e how many genes have mean count \u0026gt;=3 \u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e means = apply(countsTable, 1, mean)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e table(means\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e=3)\n\nFALSE  TRUE \n 5721  7660 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e countsTable\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-countsTable[means\u003cspan class=\"pl-k\"\u003e\u0026gt;=\u003c/span\u003e3,]\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dim(countsTable)\n[1] 7660   78\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e prop.nullv2 \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- apply(countsTable, 2, function(x) 100\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003emean(x==0))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e barplot(prop.nullv2, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMean3NullCountsv2.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,14, 14)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e barplot(prop.nullv2, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\nRStudioGD \n        2 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003emake count data sets\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dds \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e Origin + SymState + Temp + Origin:SymState)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dds \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- DESeq(dds)\nestimating size factors\nestimating dispersions\ngene-wise dispersion estimates\nmean-dispersion relationship\nfinal dispersion estimates\nfitting model and testing\n-- replacing outliers and refitting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e 14 genes\n-- DESeq argument \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eminReplicatesForReplace\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e = 7 \n-- original counts are preserved \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e counts(dds)\nestimating dispersions\nfitting model and testing\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dim(dds)\n[1] 7660   78\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etransform counts to variance stablized counts\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e vstCounts\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-varianceStabilizingTransformation(dds)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eplot cluster dendrogram across samples\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dists\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-dist(t(assay(vstCounts)))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plot(hclust(dists))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e heatmap.2(as.matrix(dists), key=F, trace=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n+           col=colorpanel(100, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n+           margin=c(10, 10))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e##Figure out which contrast you want to examine (i.e. which two groups do you want to compare)\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e resultsNames(dds)\n[1] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eIntercept\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e          \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_VA_vs_RI\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSymState_W_vs_B\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemp_18_vs_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e     \n[5] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemp_22_vs_14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e      \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOriginVA.SymStateW\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e res \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(res)\nlog2 fold change (MLE): OriginVA.SymStateW \nWald \u003cspan class=\"pl-c1\"\u003etest\u003c/span\u003e p-value: OriginVA.SymStateW \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e      \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   5.36648       0.107690  0.644646  0.167053  0.867328  0.999364\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.70436      -0.247209  0.831371 -0.297351  0.766198  0.995209\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.65756      -0.976997  0.654110 -1.493627  0.135273  0.958807\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.91279      -0.172220  0.641026 -0.268662  0.788189  0.995209\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.31144       0.366046  0.664401  0.550942  0.581673  0.991107\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   9.26737      -0.589357  0.975809 -0.603967  0.545865  0.990477\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e summary(res)\n\nout of 7659 with nonzero total \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e count\nadjusted p-value \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1\nLFC \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e 0 (up)       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 2, 0.026%\nLFC \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0 (down)     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 9, 0.12%\noutliers [1]       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 10, 0.13%\nlow counts [2]     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 1, 0.013%\n(mean count \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 1)\n[1] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecooksCutoff\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n[2] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eindependentFiltering\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e resSymState \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSymState\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eW\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(resSymState)\nlog2 fold change (MLE): SymState B vs W \nWald \u003cspan class=\"pl-c1\"\u003etest\u003c/span\u003e p-value: SymState B vs W \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e      \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   5.36648       0.304875  0.463588  0.657643 0.5107675  0.986343\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.70436      -0.340214  0.624845 -0.544477 0.5861132  0.995961\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.65756      -1.044272  0.488673 -2.136956 0.0326015  0.816270\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.91279      -0.316211  0.469123 -0.674047 0.5002815  0.986343\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.31144       0.257974  0.481341  0.535948 0.5919943  0.995961\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   9.26737      -0.604957  0.692218 -0.873939 0.3821514  0.986343\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e summary(resSymState)\n\nout of 7659 with nonzero total \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e count\nadjusted p-value \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1\nLFC \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e 0 (up)       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 6, 0.078%\nLFC \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0 (down)     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 6, 0.078%\noutliers [1]       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 10, 0.13%\nlow counts [2]     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 1, 0.013%\n(mean count \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 1)\n[1] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecooksCutoff\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n[2] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eindependentFiltering\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e resOrigin \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eVA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRI\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(resOrigin)\nlog2 fold change (MLE): Origin VA vs RI \nWald \u003cspan class=\"pl-c1\"\u003etest\u003c/span\u003e p-value: Origin VA vs RI \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE       stat    pvalue      padj\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e      \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   5.36648     -0.2712427  0.446225 -0.6078616 0.5432793  0.865661\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.70436      1.1125117  0.589306  1.8878328 0.0590484        NA\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.65756      0.9431346  0.474719  1.9867203 0.0469534  0.418043\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.91279      0.0386643  0.456720  0.0846565 0.9325345  0.984590\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.31144     -0.0179010  0.461958 -0.0387502 0.9690895  0.992150\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   9.26737     -0.2649903  0.677608 -0.3910675 0.6957473  0.921517\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e summary(resOrigin)\n\nout of 7659 with nonzero total \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e count\nadjusted p-value \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1\nLFC \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e 0 (up)       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 73, 0.95%\nLFC \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0 (down)     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 104, 1.4%\noutliers [1]       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 10, 0.13%\nlow counts [2]     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 594, 7.8%\n(mean count \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 3)\n[1] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecooksCutoff\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n[2] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eindependentFiltering\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e res14v18 \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(res14v18)\nlog2 fold change (MLE): Temp 14 vs 18 \nWald \u003cspan class=\"pl-c1\"\u003etest\u003c/span\u003e p-value: Temp 14 vs 18 \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE       stat    pvalue      padj\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e      \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   5.36648      0.0134106  0.389675  0.0344147 0.9725464  0.994763\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.70436     -0.7500403  0.493720 -1.5191625 0.1287216        NA\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.65756     -0.5278497  0.385227 -1.3702295 0.1706153        NA\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.91279      0.1448997  0.377624  0.3837147 0.7011899        NA\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.31144     -0.1171316  0.401865 -0.2914700 0.7706919  0.956296\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   9.26737     -1.1666726  0.592603 -1.9687238 0.0489848  0.440134\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e summary(res14v18)\n\nout of 7659 with nonzero total \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e count\nadjusted p-value \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1\nLFC \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e 0 (up)       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 31, 0.4%\nLFC \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0 (down)     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 111, 1.4%\noutliers [1]       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 10, 0.13%\nlow counts [2]     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 2671, 35%\n(mean count \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 4)\n[1] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecooksCutoff\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n[2] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eindependentFiltering\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e res14v22 \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e14\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(res14v22)\nlog2 fold change (MLE): Temp 14 vs 22 \nWald \u003cspan class=\"pl-c1\"\u003etest\u003c/span\u003e p-value: Temp 14 vs 22 \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e      \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   5.36648      -0.381080  0.396906 -0.960127 0.3369915  0.796010\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.70436      -0.420391  0.517100 -0.812978 0.4162304        NA\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.65756      -0.149468  0.407436 -0.366851 0.7137303        NA\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.91279       0.327506  0.400038  0.818686 0.4129654        NA\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.31144      -0.504681  0.409220 -1.233275 0.2174733  0.699820\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   9.26737      -1.240058  0.602190 -2.059248 0.0394705  0.387444\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e summary(res14v22)\n\nout of 7659 with nonzero total \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e count\nadjusted p-value \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1\nLFC \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e 0 (up)       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 64, 0.84%\nLFC \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0 (down)     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 97, 1.3%\noutliers [1]       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 10, 0.13%\nlow counts [2]     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 2226, 29%\n(mean count \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 4)\n[1] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecooksCutoff\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n[2] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eindependentFiltering\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e res18v22 \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- results(dds, contrast=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e18\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e22\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(res18v22)\nlog2 fold change (MLE): Temp 18 vs 22 \nWald \u003cspan class=\"pl-c1\"\u003etest\u003c/span\u003e p-value: Temp 18 vs 22 \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e      \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   5.36648     -0.3944906  0.395010 -0.998685  0.317947  0.838585\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.70436      0.3296494  0.503430  0.654807  0.512592        NA\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.65756      0.3783816  0.396935  0.953259  0.340459  0.847759\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.91279      0.1826059  0.399214  0.457414  0.647374        NA\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.31144     -0.3875492  0.406736 -0.952828  0.340677  0.847759\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   9.26737     -0.0733858  0.596241 -0.123081  0.902043  0.991572\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e summary(res18v22)\n\nout of 7659 with nonzero total \u003cspan class=\"pl-c1\"\u003eread\u003c/span\u003e count\nadjusted p-value \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1\nLFC \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e 0 (up)       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 34, 0.44%\nLFC \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0 (down)     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 10, 0.13%\noutliers [1]       \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 10, 0.13%\nlow counts [2]     \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e 1337, 17%\n(mean count \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 3)\n[1] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecooksCutoff\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n[2] see \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eindependentFiltering\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e argument of \u003cspan class=\"pl-k\"\u003e?\u003c/span\u003eresults\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ecount the number of significantly differentially expressed genes\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e sum(resOrigin\u003cspan class=\"pl-smi\"\u003e$padj\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.1, na.rm =T)\n[1] 177\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e sum(resOrigin\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e 0.05, na.rm =T)\n[1] 866\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Make a counts table that is scaled by the size factors\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e scaledcounts = counts(dds, normalize=T)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(scaledcounts)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0  0.8671827   2.447866   0.000000   3.455170   4.432653\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  2.6015481   1.223933   0.000000  13.820681   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0  3.4687308   2.447866   3.772786   0.000000   0.000000\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.8671827   4.895733   0.000000   0.000000   8.865306\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  1.7343654   4.895733   0.000000   6.910341   0.000000\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  6.0702789   1.223933  18.863931  51.827555   0.000000\n                       RI_B_03_14 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  12.871970   1.272998  12.424473  8.1878079   0.000000   4.563408\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000   2.003947  4.0939039   6.765958   0.570426\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000   3.818995   4.809473  3.6845136   0.000000   0.570426\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.287197   3.818995   6.011842  3.2751232   0.000000   5.133834\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  14.159167  15.275980   2.805526  4.9126847  16.914895   5.989473\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.435985   2.545997   0.000000  0.8187808   3.382979   1.426065\n                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22 RI_B_07_14\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0  2.4538305   19.55961   3.448566   18.78455  1.4936671\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.0000000    0.00000   2.069140    0.00000  0.7468335\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0  2.4538305    0.00000   4.827992    0.00000  3.7341677\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  4.0897176    0.00000   3.448566    0.00000  2.2405006\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.8179435    6.51987  10.345698   40.69986  3.7341677\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  1.6358870    0.00000   1.379426    0.00000  8.9620025\n                       RI_B_07_18 RI_B_07_22 RI_W_01_18 RI_W_01_22 RI_W_02_14 RI_W_02_18\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   8.070182   36.85316  10.815190   4.646396   0.000000  4.9947759\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.017545    0.00000  12.257216   1.548799   0.000000  1.9979104\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.008773    0.00000  12.257216   7.743994   4.651361  4.9947759\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.017545    0.00000   7.931140   0.000000   0.000000  0.9989552\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.008773    0.00000   2.884051   1.548799   9.302721  6.9926863\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  44.386000   27.63987   1.442025   0.000000   0.000000  5.9937311\n                       RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.615216   8.525621   5.407467   4.745758   12.82564  2.9768406\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   4.262810   1.081493   1.581919    0.00000  3.4729807\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   4.820288   3.044865   3.244480   1.581919    0.00000  7.4421015\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.615216   3.044865   3.244480   1.054613    0.00000  0.9922802\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.410144   2.435892   4.686471   2.109226   21.37607 12.8996426\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  44.587667   1.217946   1.081493   1.054613   29.92650  7.9382416\n                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  3.1630643   1.010704   5.526636   1.044347   1.336966   5.564881\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.9037326   8.085630   0.000000   0.000000   0.000000   4.451904\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  3.6149306  10.107037   1.381659   3.133041   1.336966   7.790833\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  4.0667969   2.021407   2.763318  10.443469   1.336966   2.225952\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  5.4223959   2.021407   6.908296  11.487816   0.000000   5.564881\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.8074653  10.107037  27.633182   5.221735   1.336966  42.293092\n                       RI_W_07_14 RI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   4.124636   5.593017   6.900578   2.551157   1.731769   2.845097\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000   0.000000   7.653472   9.091789   1.707058\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   2.062318   1.118603   5.520462  10.204629   5.628251   5.690194\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.499514  13.423242   1.380116   3.826736   2.597654   2.845097\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   1.118603   5.520462   3.826736   9.091789   2.845097\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.374879   2.237207  86.947282   1.275579   1.298827   0.000000\n                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14 VA_B_03_18 VA_B_03_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0    14.2516  19.312637  16.029307  14.339493   7.833217\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0     0.0000   0.000000   5.828839   2.150924   7.833217\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0     0.0000   0.000000  10.929073   4.301848   3.560553\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0     0.0000   2.145849   2.185815   4.301848   3.560553\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0     0.0000   0.000000  14.572098  18.641340  21.363318\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0     0.0000   8.583394   9.471863  30.829909   8.545327\n                       VA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000  0.9006809   6.731616   2.098865  3.3173156   1.813677\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  4.5034046   3.624716   4.897352  1.1057719   7.254707\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   7.108517  3.1523832   0.000000   1.399244  0.5528859   5.441030\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.442823  0.0000000   0.000000   0.000000  3.3173156   0.000000\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.665694  4.9537450   3.624716   9.095083  5.5288593   0.000000\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  2.7020428   0.000000   0.000000  4.9759734   5.441030\n                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.383946   0.000000  5.9098781   3.728534  0.7405861   2.098499\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   8.303676   0.000000  0.4924898   0.000000  3.7029304   2.098499\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000   4.206352  1.4774695   1.491414 15.5523076   6.295498\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.535784   6.730163  3.9399187   1.491414  8.1464468   4.196999\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  11.763540   4.206352  3.4474289   2.237121  2.2217582   5.246248\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  18.683270   2.523811 14.2822054   8.948482 17.0334798   2.098499\n                       VA_W_01_14 VA_W_01_18 VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   6.699293   7.732656   0.000000  4.5789684  3.1096023   6.888125\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.828167   1.104665   6.978382  0.6541383  5.4418040   9.392897\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000   3.866328   4.361489 13.0827667  6.9966052   3.130966\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   3.866328   5.233787  0.6541383  6.9966052   2.504773\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.914084   4.418660   0.000000  9.1579367  3.1096023   3.757159\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  10.494319   2.616893 13.0827667  0.7774006  13.150056\n                       VA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.026752   4.334366   6.410680  0.0000000   3.848517  12.379823\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.566880   5.779155   3.945034  0.0000000   1.099576  12.379823\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   2.053504   7.705540  10.848842  0.0000000   2.748941   7.427894\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.593631   3.852770   2.958775  0.8118702   3.298729   2.475965\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.673887  20.227043   8.383196  1.6237403   5.497882  38.377452\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.566880   0.000000   0.000000  2.4356105   8.796611   2.475965\n                       VA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   5.042735  2.5895188   0.000000 10.2466895   1.127608  0.9473408\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.8631729   1.764014  1.5764138   1.127608  6.6313859\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.025641  0.8631729   0.000000  5.5174482   3.382824  2.8420225\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   8.068376  2.5895188   4.410036  4.7292413   0.000000  1.8946817\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.034188  6.9053834   5.292043  3.1528275   1.127608  1.8946817\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   9.076923 29.3478796  26.460214  0.7882069   6.765648  8.5260676\n\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ebuilding heat map data\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(scaledcounts)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0  0.8671827   2.447866\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  2.6015481   1.223933\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0  3.4687308   2.447866\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.8671827   4.895733\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  1.7343654   4.895733\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  6.0702789   1.223933\n                       RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000   3.455170   4.432653\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  13.820681   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.772786   0.000000   0.000000\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000   8.865306\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   6.910341   0.000000\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  18.863931  51.827555   0.000000\n                       RI_B_03_14 RI_B_03_22 RI_B_04_14\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  12.871970   1.272998  12.424473\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000   2.003947\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000   3.818995   4.809473\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.287197   3.818995   6.011842\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  14.159167  15.275980   2.805526\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.435985   2.545997   0.000000\n                       RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  8.1878079   0.000000   4.563408\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  4.0939039   6.765958   0.570426\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  3.6845136   0.000000   0.570426\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  3.2751232   0.000000   5.133834\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  4.9126847  16.914895   5.989473\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8187808   3.382979   1.426065\n                       RI_B_05_18 RI_B_05_22 RI_B_06_14\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0  2.4538305   19.55961\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.0000000    0.00000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0  2.4538305    0.00000\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  4.0897176    0.00000\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  0.8179435    6.51987\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0  1.6358870    0.00000\n                       RI_B_06_18 RI_B_06_22 RI_B_07_14\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.448566   18.78455  1.4936671\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.069140    0.00000  0.7468335\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   4.827992    0.00000  3.7341677\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.448566    0.00000  2.2405006\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  10.345698   40.69986  3.7341677\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.379426    0.00000  8.9620025\n                       RI_B_07_18 RI_B_07_22 RI_W_01_18\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   8.070182   36.85316  10.815190\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.017545    0.00000  12.257216\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.008773    0.00000  12.257216\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.017545    0.00000   7.931140\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.008773    0.00000   2.884051\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  44.386000   27.63987   1.442025\n                       RI_W_01_22 RI_W_02_14 RI_W_02_18\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   4.646396   0.000000  4.9947759\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.548799   0.000000  1.9979104\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   7.743994   4.651361  4.9947759\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000  0.9989552\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.548799   9.302721  6.9926863\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000  5.9937311\n                       RI_W_02_22 RI_W_03_14 RI_W_03_18\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.615216   8.525621   5.407467\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   4.262810   1.081493\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   4.820288   3.044865   3.244480\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.615216   3.044865   3.244480\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.410144   2.435892   4.686471\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  44.587667   1.217946   1.081493\n                       RI_W_03_22 RI_W_04_14 RI_W_04_18\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   4.745758   12.82564  2.9768406\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.581919    0.00000  3.4729807\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.581919    0.00000  7.4421015\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.054613    0.00000  0.9922802\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.109226   21.37607 12.8996426\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.054613   29.92650  7.9382416\n                       RI_W_05_14 RI_W_05_18 RI_W_05_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  3.1630643   1.010704   5.526636\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.9037326   8.085630   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  3.6149306  10.107037   1.381659\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  4.0667969   2.021407   2.763318\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  5.4223959   2.021407   6.908296\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.8074653  10.107037  27.633182\n                       RI_W_06_14 RI_W_06_18 RI_W_06_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.044347   1.336966   5.564881\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000   4.451904\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.133041   1.336966   7.790833\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  10.443469   1.336966   2.225952\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  11.487816   0.000000   5.564881\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.221735   1.336966  42.293092\n                       RI_W_07_14 RI_W_07_18 RI_W_07_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   4.124636   5.593017   6.900578\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   2.062318   1.118603   5.520462\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.499514  13.423242   1.380116\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   1.118603   5.520462\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.374879   2.237207  86.947282\n                       VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   2.551157   1.731769   2.845097\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   7.653472   9.091789   1.707058\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  10.204629   5.628251   5.690194\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.826736   2.597654   2.845097\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.826736   9.091789   2.845097\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.275579   1.298827   0.000000\n                       VA_B_02_14 VA_B_02_18 VA_B_02_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0    14.2516  19.312637\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0     0.0000   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          0     0.0000   0.000000\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0     0.0000   2.145849\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0     0.0000   0.000000\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          0     0.0000   8.583394\n                       VA_B_03_14 VA_B_03_18 VA_B_03_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  16.029307  14.339493   7.833217\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.828839   2.150924   7.833217\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  10.929073   4.301848   3.560553\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.185815   4.301848   3.560553\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  14.572098  18.641340  21.363318\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   9.471863  30.829909   8.545327\n                       VA_B_04_14 VA_B_04_18 VA_B_04_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000  0.9006809   6.731616\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  4.5034046   3.624716\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   7.108517  3.1523832   0.000000\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.442823  0.0000000   0.000000\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.665694  4.9537450   3.624716\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  2.7020428   0.000000\n                       VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   2.098865  3.3173156   1.813677\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.897352  1.1057719   7.254707\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.399244  0.5528859   5.441030\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  3.3173156   0.000000\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   9.095083  5.5288593   0.000000\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  4.9759734   5.441030\n                       VA_B_06_14 VA_B_06_18 VA_B_06_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.383946   0.000000  5.9098781\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   8.303676   0.000000  0.4924898\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000   4.206352  1.4774695\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.535784   6.730163  3.9399187\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  11.763540   4.206352  3.4474289\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  18.683270   2.523811 14.2822054\n                       VA_B_07_14 VA_B_07_18 VA_B_07_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.728534  0.7405861   2.098499\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  3.7029304   2.098499\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.491414 15.5523076   6.295498\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.491414  8.1464468   4.196999\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.237121  2.2217582   5.246248\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   8.948482 17.0334798   2.098499\n                       VA_W_01_14 VA_W_01_18 VA_W_02_14\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   6.699293   7.732656   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.828167   1.104665   6.978382\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000   3.866328   4.361489\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   3.866328   5.233787\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.914084   4.418660   0.000000\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  10.494319   2.616893\n                       VA_W_02_18 VA_W_03_14 VA_W_03_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  4.5789684  3.1096023   6.888125\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.6541383  5.4418040   9.392897\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral 13.0827667  6.9966052   3.130966\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.6541383  6.9966052   2.504773\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  9.1579367  3.1096023   3.757159\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 13.0827667  0.7774006  13.150056\n                       VA_W_04_14 VA_W_04_18 VA_W_04_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.026752   4.334366   6.410680\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.566880   5.779155   3.945034\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   2.053504   7.705540  10.848842\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.593631   3.852770   2.958775\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.673887  20.227043   8.383196\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.566880   0.000000   0.000000\n                       VA_W_05_14 VA_W_05_18 VA_W_05_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.0000000   3.848517  12.379823\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000   1.099576  12.379823\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.0000000   2.748941   7.427894\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8118702   3.298729   2.475965\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.6237403   5.497882  38.377452\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.4356105   8.796611   2.475965\n                       VA_W_06_14 VA_W_06_18 VA_W_06_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   5.042735  2.5895188   0.000000\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.8631729   1.764014\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.025641  0.8631729   0.000000\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   8.068376  2.5895188   4.410036\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.034188  6.9053834   5.292043\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   9.076923 29.3478796  26.460214\n                       VA_W_07_14 VA_W_07_18 VA_W_07_22\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral 10.2466895   1.127608  0.9473408\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.5764138   1.127608  6.6313859\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  5.5174482   3.382824  2.8420225\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  4.7292413   0.000000  1.8946817\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  3.1528275   1.127608  1.8946817\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.7882069   6.765648  8.5260676\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e genes4heatmap\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-resOrigin[resOrigin\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e0.05 \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!\u003c/span\u003eis.na(resOrigin\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e),]\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e names(genes4heatmap)\n[1] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebaseMean\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e       \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elog2FoldChange\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elfcSE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e         \n[4] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003estat\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e           \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epvalue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e         \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epadj\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e          \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(genes4heatmap)\nlog2 fold change (MLE): Origin VA vs RI \nWald \u003cspan class=\"pl-c1\"\u003etest\u003c/span\u003e p-value: Origin VA vs RI \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange     lfcSE\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e      \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.65756       0.943135  0.474719\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.57170       0.727123  0.363907\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    4.14205       4.490291  1.982381\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.93642       1.787984  0.559059\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.02516       1.758477  0.659498\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   2.32876       1.292223  0.632168\n                            stat     pvalue      padj\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.98672 0.04695340 0.4180430\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.99810 0.04570594 0.4125186\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    2.26510 0.02350656 0.3086533\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.19820 0.00138287 0.0720091\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.66639 0.00766715 0.1733954\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   2.04411 0.04094219        NA\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dim(genes4heatmap)\n[1] 866   6\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e data4heatmap\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dim(data4heatmap)\n[1] 866  78\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(data4heatmap)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral    0.00000   3.468731   2.447866\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   10.94714   5.203096   8.567532\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral     0.00000   0.000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   0.000000   2.447866\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   6.070279   2.447866\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral    0.00000   2.601548   4.895733\n                       RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.772786    0.00000   0.000000\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000    0.00000   6.648979\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    7.545573    3.45517   6.648979\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000    0.00000   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.772786    0.00000   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000    0.00000   2.216326\n                       RI_B_03_14 RI_B_03_22 RI_B_04_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000   3.818995   4.809473\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   9.010379   5.091993   2.404737\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000   0.000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.574394   0.000000   4.007894\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.287197   0.000000   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   5.148788   1.272998   0.000000\n                       RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  3.6845136          0   0.570426\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  4.5032943          0   9.126817\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000          0   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.8657328          0   1.711278\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.4093904          0   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.0000000          0   0.285213\n                       RI_B_05_18 RI_B_05_22 RI_B_06_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000  2.4538305          0\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.033716  0.8179435          0\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000  0.0000000          0\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.0000000          0\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.0000000          0\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000  0.8179435          0\n                       RI_B_06_18 RI_B_06_22 RI_B_07_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   4.827992          0   3.734168\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.758853          0   5.227835\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000          0   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000          0   2.987334\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000          0   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000          0   0.000000\n                       RI_B_07_18 RI_B_07_22 RI_W_01_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.008773          0  12.257216\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.043864          0   8.652152\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000          0  10.094178\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000          0   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.008773          0   7.931140\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000          0   5.047089\n                       RI_W_01_22 RI_W_02_14 RI_W_02_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   7.743994   4.651361  4.9947759\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.195195  16.279762  2.9968656\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    9.292792   0.000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.548799   4.651361  0.9989552\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  17.036786   0.000000 14.9843278\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  18.585585   0.000000 10.9885071\n                       RI_W_02_22 RI_W_03_14 RI_W_03_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   4.820288   3.044865   3.244480\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.205072   7.307675   7.209956\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000   0.000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.410144   2.435892   6.128462\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.025360  23.140971  12.617422\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   1.205072  10.352540   7.930951\n                       RI_W_03_22 RI_W_04_14 RI_W_04_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.581919    0.00000  7.4421015\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  14.237275   17.10086  5.4575411\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000    0.00000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.218452    0.00000  0.4961401\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  29.001856    0.00000  1.9845604\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   7.382291    0.00000  1.4884203\n                       RI_W_05_14 RI_W_05_18 RI_W_05_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.614931  10.107037   1.381659\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.066797   4.042815   0.000000\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000   0.000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.711198   1.010704   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.422396   2.021407   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   3.614931   3.032111   0.000000\n                       RI_W_06_14 RI_W_06_18 RI_W_06_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.133041   1.336966   7.790833\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.177388  13.369662  10.016785\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000   0.000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.133041   2.673932   2.225952\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.221735   2.673932  10.016785\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   1.044347   2.673932   3.338928\n                       RI_W_07_14 RI_W_07_18 RI_W_07_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   2.062318   1.118603   5.520462\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.437196  13.423242   8.280694\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000   0.000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.749757   2.237207   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.374879   2.237207   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   2.062318   3.355810   2.760231\n                       VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  10.204629   5.628251   5.690194\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  10.204629   6.494135  10.242350\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   33.165045   9.524732  17.639603\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.275579   1.298827   5.121175\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.377893   4.762366   2.845097\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   8.929051   6.061193   6.259214\n                       VA_B_02_14 VA_B_02_18 VA_B_02_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000     0.0000   0.000000\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000     0.0000  17.166789\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   86.631891    44.0504  40.771123\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  30.575962     0.0000   2.145849\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.698665     0.0000   6.437546\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000     2.5912   0.000000\n                       VA_B_03_14 VA_B_03_18 VA_B_03_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  10.929073   4.301848  3.5605530\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  10.929073   7.886721  7.8332166\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000   0.000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.643024   7.886721  3.5605530\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   1.433949  0.7121106\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000   3.584873  0.7121106\n                       VA_B_04_14 VA_B_04_18 VA_B_04_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  7.1085175   3.152383   0.000000\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 15.1055996   4.053064   8.285065\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000   0.000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.6656940   0.000000   8.802882\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000   7.205447   8.285065\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.8885647   8.556469   2.589083\n                       VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.399244  0.5528859   5.441030\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.547541 17.1394640  21.764122\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000  0.0000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.049433  1.6586578   3.627354\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.399244  1.1057719   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000  4.4230875   0.000000\n                       VA_B_06_14 VA_B_06_18 VA_B_06_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000   4.206352   1.477470\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   8.412704   2.462449\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   29.062864  16.825408   8.372327\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.919730   0.000000   2.954939\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.767892   0.000000   5.417388\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000   2.523811   2.462449\n                       VA_B_07_14 VA_B_07_18 VA_B_07_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  1.4914137  15.552308   6.295498\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  3.7285342   5.184103   8.393997\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000   0.000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  6.7113616   0.000000   4.196999\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.4914137   2.962344   6.295498\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.7457068   0.000000   4.196999\n                       VA_W_01_14 VA_W_01_18 VA_W_02_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000   3.866328  4.3614889\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.699293   4.970993  1.7445956\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000   0.000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.871126   2.209330  0.8722978\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   5.742251   1.104665  0.0000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000   3.866328  0.0000000\n                       VA_W_02_18 VA_W_03_14 VA_W_03_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  13.082767  6.9966052   3.130966\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  13.082767  3.1096023   9.392897\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000  0.0000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.7774006   3.757159\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.924830  1.5548011   2.504773\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   5.233107  0.7774006   2.504773\n                       VA_W_04_14 VA_W_04_18 VA_W_04_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   2.053504   7.705540 10.8488424\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.593631  13.003099  1.4793876\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000   0.000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.673887   9.150329  7.3969380\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.026752   0.000000  0.4931292\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   1.026752   0.000000  0.0000000\n                       VA_W_05_14 VA_W_05_18 VA_W_05_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral    0.00000   2.748941   7.427894\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   8.246822   8.665876\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral     0.00000   0.000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   14.61366   2.748941   9.903859\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.00000   0.000000   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral    0.00000   2.199153   1.237982\n                       VA_W_06_14 VA_W_06_18 VA_W_06_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.025641  0.8631729   0.000000\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  10.085470  1.7263459  11.466093\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000  0.0000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.017094  6.0422105   4.410036\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  1.7263459   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   2.017094  0.8631729   5.292043\n                       VA_W_07_14 VA_W_07_18 VA_W_07_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   5.517448   3.382824   2.842023\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   7.093862   6.765648   3.789363\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000   0.000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   3.382824   1.894682\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000   0.000000   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000   0.000000   0.000000\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e temp = as.matrix(rowMeans(data4heatmap))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(temp)\n                           [,1]\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral 3.657565\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 6.571698\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  4.142052\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 2.936423\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 3.025158\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral 2.328759\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e scaledmatrix\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e data4heatmapscaled = data4heatmap/scaledmatrix\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(data4heatmapscaled)\n                       RI_B_01_14 RI_B_01_18 RI_B_01_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000  0.9483717  0.6692613\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.665801  0.7917431  1.3037015\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000  0.0000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.0000000  0.8336218\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  2.0065992  0.8091699\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000  1.1171391  2.1022922\n                       RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.031502  0.0000000  0.0000000\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.0000000  1.0117598\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    1.821699  0.8341687  1.6052378\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.0000000  0.0000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.247137  0.0000000  0.0000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000  0.0000000  0.9517198\n                       RI_B_03_14 RI_B_03_22 RI_B_04_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.0000000  1.0441360  1.3149387\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.3710885  0.7748368  0.3659232\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000  0.0000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8767109  0.0000000  1.3648899\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.4254975  0.0000000  0.0000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  2.2109576  0.5466423  0.0000000\n                       RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  1.0073680          0  0.1559579\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.6852559          0  1.3888066\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000          0  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.9759263          0  0.5827764\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.1353286          0  0.0000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.0000000          0  0.1224742\n                       RI_B_05_18 RI_B_05_22 RI_B_06_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.0000000  0.6708919          0\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.7659688  0.1244646          0\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000  0.0000000          0\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  0.0000000          0\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  0.0000000          0\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.0000000  0.3512357          0\n                       RI_B_06_18 RI_B_06_22 RI_B_07_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  1.3200019          0  1.0209437\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.4198082          0  0.7955075\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000          0  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000          0  1.0173377\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000          0  0.0000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.0000000          0  0.0000000\n                       RI_B_07_18 RI_B_07_22 RI_W_01_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.2758045          0   3.351196\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.7675130          0   1.316578\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000          0   2.436999\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000          0   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.3334612          0   2.621728\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.0000000          0   2.167287\n                       RI_W_01_22 RI_W_02_14 RI_W_02_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  2.1172541   1.271710  1.3656016\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.9427084   2.477254  0.4560261\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.2435236   0.000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.5274440   1.584023  0.3401946\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  5.6317019   0.000000  4.9532387\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  7.9808953   0.000000  4.7186100\n                       RI_W_02_22 RI_W_03_14 RI_W_03_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  1.3178956  0.8324842  0.8870602\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.1833730  1.1119920  1.0971223\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000  0.0000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.8207755  0.8295438  2.0870501\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.9917509  7.6495091  4.1708314\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.5174739  4.4455171  3.4056552\n                       RI_W_03_22 RI_W_04_14 RI_W_04_18\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.4325062   0.000000  2.0347150\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.1664531   2.602198  0.8304614\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000   0.000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.4365953   0.000000  0.1689607\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  9.5868908   0.000000  0.6560188\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  3.1700530   0.000000  0.6391473\n                       RI_W_05_14 RI_W_05_18 RI_W_05_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.9883436  2.7633244  0.3777539\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.6188351  0.6151858  0.0000000\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000  0.0000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.9232995  0.3441955  0.0000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.7924342  0.6681991  0.0000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  1.5522989  1.3020286  0.0000000\n                       RI_W_06_14 RI_W_06_18 RI_W_06_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.8565921  0.3655346  2.1300602\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.6356634  2.0344306  1.5242310\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000  0.0000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.0669582  0.9106087  0.7580488\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.7261033  0.8838986  3.3111614\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.4484564  1.1482219  1.4337799\n                       RI_W_07_14 RI_W_07_18 RI_W_07_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.5638500  0.3058329   1.509327\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.5230302  2.0425836   1.260054\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000  0.0000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.9364308  0.7618816   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.4544816  0.7395340   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.8855865  1.4410293   1.185280\n                       VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  2.7900066  1.5387973   1.555733\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.5528148  0.9881975   1.558555\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   8.0069110  2.2995199   4.258662\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.4343988  0.4423160   1.744018\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.1082846  1.5742538   0.940479\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  3.8342522  2.6027562   2.687789\n                       VA_B_02_14 VA_B_02_18 VA_B_02_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.0000000   0.000000  0.0000000\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000   0.000000  2.6122305\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  20.9152088  10.634922  9.8432176\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 10.4126549   0.000000  0.7307695\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.5615127   0.000000  2.1280034\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.0000000   1.112696  0.0000000\n                       VA_B_03_14 VA_B_03_18 VA_B_03_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   2.988074  1.1761509  0.9734764\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.663052  1.2001041  1.1919624\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000  0.0000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.240633  2.6858257  1.2125476\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.4740081  0.2353962\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000  1.5393918  0.3057897\n                       VA_B_04_14 VA_B_04_18 VA_B_04_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  1.9435111  0.8618804   0.000000\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.2985841  0.6167454   1.260719\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000  0.0000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.9078031  0.0000000   2.997825\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  2.3818420   2.738722\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.3815614  3.6742606   1.111786\n                       VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.3825615  0.1511623   1.487610\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.6919888  2.6080725   3.311796\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000  0.0000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.3573847  0.5648565   1.235297\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.4625357  0.3655254   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.0000000  1.8993321   0.000000\n                       VA_B_06_14 VA_B_06_18 VA_B_06_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.0000000   1.150042  0.4039490\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000   1.280142  0.3747052\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   7.0165371   4.062094  2.0212992\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.3565165   0.000000  1.0063056\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.9149579   0.000000  1.7907788\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.0000000   1.083758  1.0574082\n                       VA_B_07_14 VA_B_07_18 VA_B_07_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.4077614  4.2520938   1.721227\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.5673624  0.7888529   1.277295\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000  0.0000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.2855567  0.0000000   1.429289\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.4930036  0.9792364   2.081048\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.3202164  0.0000000   1.802247\n                       VA_W_01_14 VA_W_01_18 VA_W_02_14\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.0000000  1.0570771  1.1924571\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.0194159  0.7564245  0.2654711\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000  0.0000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.9777629  0.7523882  0.2970613\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.8981660  0.3651595  0.0000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.0000000  1.6602522  0.0000000\n                       VA_W_02_18 VA_W_03_14 VA_W_03_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.576907  1.9129136  0.8560247\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.990774  0.4731810  1.4292954\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000  0.0000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.2647441  1.2795019\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.297397  0.5139571  0.8279809\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   2.247165  0.3338261  1.0755824\n                       VA_W_04_14 VA_W_04_18 VA_W_04_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.5614402   2.106741  2.9661383\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.5468346   1.978651  0.2251150\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000   0.000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  2.2727947   3.116148  2.5190299\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.3394044   0.000000  0.1630094\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.4409008   0.000000  0.0000000\n                       VA_W_05_14 VA_W_05_18 VA_W_05_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   0.000000  0.7515768   2.030831\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  1.2548999   1.318666\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000  0.0000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.976688  0.9361528   3.372763\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.0000000   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000  0.9443452   0.531606\n                       VA_W_06_14 VA_W_06_18 VA_W_06_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.8272283  0.2359966   0.000000\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  1.5346826  0.2626940   1.744769\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.0000000  0.0000000   0.000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.6869221  2.0576770   1.501839\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.0000000  0.5706631   0.000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.8661668  0.3706578   2.272473\n                       VA_W_07_14 VA_W_07_18 VA_W_07_22\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   1.508503  0.9248843  0.7770259\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.079457  1.0295131  0.5766186\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    0.000000  0.0000000  0.0000000\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  1.1520220  0.6452345\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   0.000000  0.0000000  0.0000000\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   0.000000  0.0000000  0.0000000\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dim(data4heatmapscaled)\n[1] 866  78\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pairs.breaks \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- seq(0, 3.0, by=0.1)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e length(pairs.breaks)\n[1] 31\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e mycol \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- colorpanel(n=30, low=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, high=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eyellow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eXXXXXX_byrow.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,7,7)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), scale=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, symkey = F, density.info = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edensity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, colsep=c(24), sepcolor=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\nRStudioGD \n        2 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eXXXXXX_bycolumn.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,7,7)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecol\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), scale=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, symkey = F, density.info = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edensity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, colsep=c(24), sepcolor=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\nRStudioGD \n        2 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), scale=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, symkey = F, density.info = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edensity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, colsep=c(24), sepcolor=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\nError \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e \u003cspan class=\"pl-en\"\u003eplot.new\u003c/span\u003e() \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e figure margins too large\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecol\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), scale=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, keysize=1, breaks=pairs.breaks, col=mycol, trace = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, symkey = F, density.info = \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edensity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, colsep=c(24), sepcolor=c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\nError \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e .External.graphics(C_layout, num.rows, num.cols, mat, as.integer(num.figures),  \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e \n  invalid graphics state\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin_SymState_Temp\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e866_RIvsVA_PCAv2.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,14, 14)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOrigin\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\nRStudioGD \n        2 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e prop.nullv3 \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003emean(x==0))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e barplot(prop.nullv3, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pdf(file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eResNullCountsv2.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,14, 14)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e barplot(prop.nullv3, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-en\"\u003edev.off\u003c/span\u003e()\nRStudioGD \n        2 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e barplot(prop.nullv3, main=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePercentage of null counts per sample\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n+         horiz=TRUE, cex.names=0.5, las=1, \n+         col=conds\u003cspan class=\"pl-smi\"\u003e$Origin_SymState_Temp\u003c/span\u003e, ylab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eSamples\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, xlab=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e% of null counts\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e distsv\u003cspan class=\"pl-k\"\u003e2\u0026lt;\u003c/span\u003e-dist(t(assay(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),])))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e plot(hclust(distsv2))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e heatmap.2(as.matrix(distsv2), key=F, trace=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n+           col=colorpanel(100, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblack\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n+           margin=c(10, 10))\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e### GOSeq analysis ####\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e res\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-resOrigin\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e Annotations\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-read.delim(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eAssembly_GoodCoralSymbiont_suffixed_totalannotated.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(res)\nlog2 fold change (MLE): Origin VA vs RI \nWald \u003cspan class=\"pl-c1\"\u003etest\u003c/span\u003e p-value: Origin VA vs RI \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e      \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   5.36648     -0.2712427\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.70436      1.1125117\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.65756      0.9431346\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.91279      0.0386643\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.31144     -0.0179010\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   9.26737     -0.2649903\n                           lfcSE       stat    pvalue\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.446225 -0.6078616 0.5432793\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.589306  1.8878328 0.0590484\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.474719  1.9867203 0.0469534\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.456720  0.0846565 0.9325345\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.461958 -0.0387502 0.9690895\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.677608 -0.3910675 0.6957473\n                            padj\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10083\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.865661\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral        NA\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.418043\nTR10103\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.984590\nTR10108\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.992150\nTR10132\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.921517\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e sum(res\u003cspan class=\"pl-smi\"\u003e$log2FoldChange\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e=0)\n[1] 7659\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e genes\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-as.integer(p.adjust(res\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e[res\u003cspan class=\"pl-smi\"\u003e$log2FoldChange\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e0 \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!\u003c/span\u003eis.na(res\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e)], method=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e.05)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e names(genes)\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-row.names(res[res\u003cspan class=\"pl-smi\"\u003e$log2FoldChange\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e0 \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!\u003c/span\u003eis.na(res\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e),])\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e table(genes)\ngenes\n   0    1 \n7533  116 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e genes\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-as.integer(res\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e[res\u003cspan class=\"pl-smi\"\u003e$log2FoldChange\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e0 \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!\u003c/span\u003eis.na(res\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e)]\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e.05)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e names(genes)\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-row.names(res[res\u003cspan class=\"pl-smi\"\u003e$log2FoldChange\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e0 \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!\u003c/span\u003eis.na(res\u003cspan class=\"pl-smi\"\u003e$pvalue\u003c/span\u003e),])\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e table(genes)\ngenes\n   0    1 \n6783  866 \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e goterms\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-strsplit(as.character(Annotations\u003cspan class=\"pl-smi\"\u003e$GO\u003c/span\u003e), split=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e // \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e names(goterms)\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-Annotations\u003cspan class=\"pl-smi\"\u003e$ContigName\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pwf\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-nullp(genes, bias.data=Annotations[Annotations\u003cspan class=\"pl-smi\"\u003e$ContigName\u003c/span\u003e%in%names(genes),\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eContigLength\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e])\nWarning message:\nIn pcls(G) \u003cspan class=\"pl-c1\"\u003e:\u003c/span\u003e initial point very close to some inequality constraints\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e GOGOGO\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-goseq(pwf,gene2cat=goterms)\nUsing manually entered categories.\nCalculating the p-values...\n\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eselect()\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e returned 1:1 mapping between keys\nand columns\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e GOGOGO\u003cspan class=\"pl-smi\"\u003e$padj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-p.adjust(GOGOGO\u003cspan class=\"pl-smi\"\u003e$over_represented_pvalue\u003c/span\u003e, method=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003efdr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e sum(GOGOGO\u003cspan class=\"pl-smi\"\u003e$padj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e0.2, na.rm=T)\n[1] 3\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e sum(GOGOGO\u003cspan class=\"pl-smi\"\u003e$padj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e0.1, na.rm=T)\n[1] 0\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e sum(GOGOGO\u003cspan class=\"pl-smi\"\u003e$padj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e0.05, na.rm=T)\n[1] 0\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e GOGOGO[GOGOGO\u003cspan class=\"pl-smi\"\u003e$padj\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e0.05,]\n[1] category                 over_represented_pvalue \n[3] under_represented_pvalue numDEInCat              \n[5] numInCat                 term                    \n[7] ontology                 padj                    \n\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e0 rows\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e (or 0-length row.names)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e GOTERM[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSOMEGOTERM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\nNULL\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e EnGOS\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-GOGOGO\u003cspan class=\"pl-smi\"\u003e$category\u003c/span\u003e[p.adjust(GOGOGO\u003cspan class=\"pl-smi\"\u003e$over_represented_pvalue\u003c/span\u003e, method=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBH\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e.05]\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e for(go \u003cspan class=\"pl-k\"\u003ein\u003c/span\u003e EnGOS){\n+   print(GOTERM[[go]])\n+   cat(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e--------------------------------------\\n\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)}\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e names(Annotations)\n [1] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eContigName\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e             \n [2] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eContigLength\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e           \n [3] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etopnrMatch\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e             \n [4] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etopnrEvalue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e            \n [5] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etopnrAlignmentLength\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e   \n [6] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etopnrPercMatch\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e         \n [7] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enobadwordnrMatch\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e       \n [8] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enobadwordnrEvalue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e      \n [9] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enobadwordnrAlignLength\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \n[10] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enobadwordnrPercentMatch\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n[11] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eUniprotmatch\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e           \n[12] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eX._identity\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e            \n[13] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eevalue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e                 \n[14] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eID\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e                     \n[15] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDescription\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e            \n[16] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKEGG\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e                   \n[17] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKEGGKO\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e                 \n[18] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePFAM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e                   \n[19] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGO\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e                     \n[20] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGOCC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e                   \n[21] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGOBP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e                   \n[22] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGOMF\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e                   \n[23] \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eKeywords\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e               \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(genes4heatmap)\nlog2 fold change (MLE): Origin VA vs RI \nWald \u003cspan class=\"pl-c1\"\u003etest\u003c/span\u003e p-value: Origin VA vs RI \nDataFrame with 6 rows and 6 columns\n                        baseMean log2FoldChange\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e      \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.65756       0.943135\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.57170       0.727123\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral    4.14205       4.490291\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.93642       1.787984\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.02516       1.758477\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   2.32876       1.292223\n                           lfcSE      stat     pvalue\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral  0.474719   1.98672 0.04695340\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.363907   1.99810 0.04570594\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   1.982381   2.26510 0.02350656\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.559059   3.19820 0.00138287\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.659498   2.66639 0.00766715\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral  0.632168   2.04411 0.04094219\n                            padj\n                       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\nTR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral 0.4180430\nTR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 0.4125186\nTR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral  0.3086533\nTR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 0.0720091\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral 0.1733954\nTR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral        NA\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dim(genes4heatmap)\n[1] 866   6\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e ResultsOutput\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e-cbind(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eContigName\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e=row.names(genes4heatmap),genes4heatmap, Annotations[Annotations\u003cspan class=\"pl-smi\"\u003e$ContigName\u003c/span\u003e%in%row.names(genes4heatmap),])\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e dim(ResultsOutput)\n[1] 866  30\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e head(ResultsOutput)\nDataFrame with 6 rows and 30 columns\n              ContigName  baseMean log2FoldChange\n             \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e      \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\n1 TR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral   3.65756       0.943135\n2 TR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   6.57170       0.727123\n3  TR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   4.14205       4.490291\n4 TR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   2.93642       1.787984\n5 TR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral   3.02516       1.758477\n6 TR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral   2.32876       1.292223\n      lfcSE      stat     pvalue      padj\n  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003enumeric\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\n1  0.474719   1.98672 0.04695340 0.4180430\n2  0.363907   1.99810 0.04570594 0.4125186\n3  1.982381   2.26510 0.02350656 0.3086533\n4  0.559059   3.19820 0.00138287 0.0720091\n5  0.659498   2.66639 0.00766715 0.1733954\n6  0.632168   2.04411 0.04094219        NA\n              ContigName ContigLength\n             \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e    \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003einteger\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\n1 TR10090\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g2_i1_coral          772\n2 TR10487\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral         1007\n3  TR1069\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          815\n4 TR10748\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          516\n5 TR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec0_g1_i1_coral          580\n6 TR10773\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ec1_g1_i1_coral          586\n              topnrMatch  topnrEvalue\n             \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\n1 gnl\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003eBL_ORD_ID\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e738431..  7.27758e-78\n2 gnl\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003eBL_ORD_ID\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e110398..  3.65321e-84\n3 gnl\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003eBL_ORD_ID\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e110400.. 9.23041e-133\n4 gnl\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003eBL_ORD_ID\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e110376..  2.29747e-78\n5 gnl\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003eBL_ORD_ID\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e110383..  2.14754e-64\n6 gnl\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003eBL_ORD_ID\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e110383..  1.33696e-65\n  topnrAlignmentLength topnrPercMatch\n           \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e    \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\n1                  258           0.53\n2                  201           0.88\n3                  271           0.73\n4                  172           0.85\n5                  206           0.55\n6                  196           0.60\n        nobadwordnrMatch nobadwordnrEvalue\n             \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\n1 gnl\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003eBL_ORD_ID\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e726506..       3.99079e-52\n2 gnl\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003eBL_ORD_ID\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e110398..       3.65321e-84\n3 gnl\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003eBL_ORD_ID\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e110400..      9.23041e-133\n4 gnl\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003eBL_ORD_ID\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e110376..       2.29747e-78\n5 gnl\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003eBL_ORD_ID\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e110383..       2.14754e-64\n6 gnl\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003eBL_ORD_ID\u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e110383..       1.33696e-65\n  nobadwordnrAlignLength nobadwordnrPercentMatch\n             \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e             \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\n1                    264                    0.45\n2                    201                    0.88\n3                    271                    0.73\n4                    172                    0.85\n5                    206                    0.55\n6                    196                    0.60\n  Uniprotmatch X._identity      evalue\n   \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\n1   A0A1X7UWJ8      40.927    1.82e-51\n2       Q4R6F3      37.500    1.77e-06\n3       Q8N2E2      31.022    1.60e-30\n4   A0A2B4RJ64      70.690    9.57e-61\n5   A0A2B4RML7      46.635    3.49e-44\n6   A0A2B4RML7      47.847    2.78e-48\n                ID            Description\n       \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e            \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\n1 A0A1X7UWJ8_AMPQE Uncharacterized prot..\n2      NGLY1_MACFA Peptide-N(4)-(N-acet..\n3       VWDE_HUMAN von Willebrand facto..\n4 A0A2B4RJ64_STYPI Uncharacterized prot..\n5 A0A2B4RML7_STYPI Uncharacterized prot..\n6 A0A2B4RML7_STYPI Uncharacterized prot..\n           KEGG      KEGGKO                   PFAM\n    \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e            \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\n1       No_KEGG   No_KEGGKO       PF05585\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e DUF1758\n2 mcf:101925902      K01456 PF04721\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e PAW,  PF094..\n3    hsa:221806   No_KEGGKO           PF00094\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e VWD\n4       No_KEGG   No_KEGGKO                   Pfam\n5       No_KEGG   No_KEGGKO                   Pfam\n6       No_KEGG   No_KEGGKO                   Pfam\n                      GO                   GOCC\n             \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e            \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\n1             GO:0004190                No_GOBP\n2 GO:0005737 // GO:004.. glycoprotein catabol..\n3 GO:0009986 // GO:000.. anatomical structure..\n4             GO:0016021                No_GOBP\n5             No_GOcodes                No_GOBP\n6             No_GOcodes                No_GOBP\n                    GOBP                   GOMF\n             \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e            \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\n1 aspartic-type endope..                No_GOCC\n2 metal ion binding (G.. cytoplasm (GO:000573..\n3 signaling receptor b.. cell surface (GO:000..\n4                No_GOMF integral component o..\n5                No_GOMF                No_GOCC\n6                No_GOMF                No_GOCC\n                Keywords\n             \u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003echaracter\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\n1 Complete proteome {E..\n2 Acetylation\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003eCytoplas..\n3 Alternative splicing..\n4 Coiled coil {ECO:000..\n5 Complete proteome {E..\n6 Complete proteome {E..\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e write.table(ResultsOutput, file=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e177OriginDEGswAnnotation.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, quote=F, row.names=F, sep=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\\t\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \n\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e9- Optional, also plot a PCA using only the genes with a pvalue\u0026lt;0.05 for the \"origin\" and \"symstate\" variables\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e10- Even more optional, color code your symbols by temperature treatment as well as origin and symstate.\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay13HW\u003c/h2\u003e\u003ca id=\"user-content-day13hw\" class=\"anchor\" aria-label=\"Permalink: Day13HW\" href=\"#day13hw\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#1- create an sbatch script to rm -r your sandbox/YOURNAME directory\n\n[kcrid001@coreV3-23-046 sandboxes]$ cat SandboxRemovalKCrid.sh \n#!/bin/bash -l\n\n#SBATCH -o KCridsandboxremoval.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KCridsandboxremoval\n\nrm -r ./katiecrider\n[kcrid001@coreV3-23-046 sandboxes]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes\n\n\n#2- submit your sbatch script and add the entry to your logfile/readme\n\n#3- scp your github logfile to the /cm/shared/courses/dbarshis/21AdvGenomics/archive with the name LASTNAME_FIRSTNAME_readme.md\n\n\nscp README.md kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/archive/Crider_Katie_readme.md\n\n\n\n#4- scp the following files to your local computer\n\tscp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta ./\n\tscp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta.fai ./\n\tscp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam ./\n\tscp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam.bai ./\n#5- launch IGV on your local machine and load in 15079_Apoc_hostsym.fasta as the \u0026quot;genome\u0026quot; and RI_B_01.bam as the input file\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1- create an sbatch script to rm -r your sandbox/YOURNAME directory\u003c/span\u003e\n\n[kcrid001@coreV3-23-046 sandboxes]$ cat SandboxRemovalKCrid.sh \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KCridsandboxremoval.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KCridsandboxremoval\u003c/span\u003e\n\nrm -r ./katiecrider\n[kcrid001@coreV3-23-046 sandboxes]$ \u003cspan class=\"pl-c1\"\u003epwd\u003c/span\u003e\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2- submit your sbatch script and add the entry to your logfile/readme\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3- scp your github logfile to the /cm/shared/courses/dbarshis/21AdvGenomics/archive with the name LASTNAME_FIRSTNAME_readme.md\u003c/span\u003e\n\n\nscp README.md kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/archive/Crider_Katie_readme.md\n\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4- scp the following files to your local computer\u003c/span\u003e\n\tscp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta ./\n\tscp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta.fai ./\n\tscp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam ./\n\tscp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam.bai ./\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e5- launch IGV on your local machine and load in 15079_Apoc_hostsym.fasta as the \"genome\" and RI_B_01.bam as the input file\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDay13\u003c/h2\u003e\u003ca id=\"user-content-day13\" class=\"anchor\" aria-label=\"Permalink: Day13\" href=\"#day13\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"#1- create an sbatch script to rm -r your sandbox/YOURNAME directory\n#2- submit your sbatch script and add the entry to your logfile/readme\n\n[kcrid001@turing1 sandboxes]$ cat SandboxRemovalKCrid.sh\n#!/bin/bash -l\n\n#SBATCH -o KCridsandboxremoval.txt\n#SBATCH -n 1\n#SBATCH --mail-user=kcrid001@odu.edu\n#SBATCH --mail-type=END\n#SBATCH --job-name=KCridsandboxremoval\n\nrm -r ./katiecrider\n[kcrid001@turing1 sandboxes]$ salloc\nsalloc: Pending job allocation 9284556\nsalloc: job 9284556 queued and waiting for resources\nsalloc: job 9284556 has been allocated resources\nsalloc: Granted job allocation 9284556\n[kcrid001@coreV1-22-016 sandboxes]$ sbatch SandboxRemovalKCrid.sh\nSubmitted batch job 9284557\n[kcrid001@coreV1-22-016 sandboxes]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9284557      main KCridsan kcrid001  R       0:01      1 coreV1-22-016 \n           9284556      main       sh kcrid001  R       0:31      1 coreV1-22-016 \n\n#3- scp your github logfile to the /cm/shared/courses/dbarshis/21AdvGenomics/archive with the name LASTNAME_FIRSTNAME_readme.md\n\n(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ scp README.md kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/archive/Crider_Katie_readme.md\nkcrid001@turing.hpc.odu.edu's password: \nREADME.md                                         100%  207KB   1.1MB/s   00:00 \n\n#4- scp the following files to your local computer\n(base) Katies-MacBook-Air-4:Genomics katiecrider$ mkdir day13\n(base) Katies-MacBook-Air-4:Genomics katiecrider$ cd day13\n(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta ./\nkcrid001@turing.hpc.odu.edu's password: \n15079_Apoc_hostsym.fasta                          100%   13MB  16.6MB/s   00:00    \n(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta.fai ./\nkcrid001@turing.hpc.odu.edu's password: \n15079_Apoc_hostsym.fasta.fai                      100%  641KB   2.0MB/s   00:00    \n(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam ./\nkcrid001@turing.hpc.odu.edu's password: \nRI_B_01.bam                                       100%   44MB  17.5MB/s   00:02    \n(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam.bai ./\nkcrid001@turing.hpc.odu.edu's password: \nRI_B_01.bam.bai  \n\n\n#5- launch IGV on your local machine and load in 15079_Apoc_hostsym.fasta as the \u0026quot;genome\u0026quot; and RI_B_01.bam as the input file\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1- create an sbatch script to rm -r your sandbox/YOURNAME directory\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2- submit your sbatch script and add the entry to your logfile/readme\u003c/span\u003e\n\n[kcrid001@turing1 sandboxes]$ cat SandboxRemovalKCrid.sh\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#!\u003c/span\u003e/bin/bash -l\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -o KCridsandboxremoval.txt\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH -n 1\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-user=kcrid001@odu.edu\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --mail-type=END\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eSBATCH --job-name=KCridsandboxremoval\u003c/span\u003e\n\nrm -r ./katiecrider\n[kcrid001@turing1 sandboxes]$ salloc\nsalloc: Pending job allocation 9284556\nsalloc: job 9284556 queued and waiting \u003cspan class=\"pl-k\"\u003efor\u003c/span\u003e resources\nsalloc: job 9284556 has been allocated resources\nsalloc: Granted job allocation 9284556\n[kcrid001@coreV1-22-016 sandboxes]$ sbatch SandboxRemovalKCrid.sh\nSubmitted batch job 9284557\n[kcrid001@coreV1-22-016 sandboxes]$ squeue -u kcrid001\n             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) \n           9284557      main KCridsan kcrid001  R       0:01      1 coreV1-22-016 \n           9284556      main       sh kcrid001  R       0:31      1 coreV1-22-016 \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3- scp your github logfile to the /cm/shared/courses/dbarshis/21AdvGenomics/archive with the name LASTNAME_FIRSTNAME_readme.md\u003c/span\u003e\n\n(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ scp README.md kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/archive/Crider_Katie_readme.md\nkcrid001@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password: \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eREADME.md                                         100%  207KB   1.1MB/s   00:00 \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#4- scp the following files to your local computer\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(base) Katies-MacBook-Air-4:Genomics katiecrider$ mkdir day13\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(base) Katies-MacBook-Air-4:Genomics katiecrider$ cd day13\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta ./\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ekcrid001@turing.hpc.odu.edu\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003es password: \n15079_Apoc_hostsym.fasta                          100%   13MB  16.6MB/s   00:00    \n(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta.fai ./\nkcrid001@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password: \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e15079_Apoc_hostsym.fasta.fai                      100%  641KB   2.0MB/s   00:00    \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam ./\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003ekcrid001@turing.hpc.odu.edu\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003es password: \nRI_B_01.bam                                       100%   44MB  17.5MB/s   00:02    \n(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam.bai ./\nkcrid001@turing.hpc.odu.edu\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es password: \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003eRI_B_01.bam.bai  \u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e#5- launch IGV on your local machine and load in 15079_Apoc_hostsym.fasta as the \"genome\" and RI_B_01.bam as the input file\u003c/span\u003e\n\u003cspan class=\"pl-s\"\u003e\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"Katie Crider's Advanced Genomics Log","anchor":"katie-criders-advanced-genomics-log","htmlText":"Katie Crider's Advanced Genomics Log"},{"level":2,"text":"Spring 2021","anchor":"spring-2021","htmlText":"Spring 2021"},{"level":2,"text":"Day 02 Exercises: 1/22/21","anchor":"day-02-exercises-12221","htmlText":"Day 02 Exercises: 1/22/21"},{"level":2,"text":"Day 03 Exercises/Day 2 HW: 1/27/21","anchor":"day-03-exercisesday-2-hw-12721","htmlText":"Day 03 Exercises/Day 2 HW: 1/27/21"},{"level":2,"text":"Day 03 HW: 1/28/21","anchor":"day-03-hw-12821","htmlText":"Day 03 HW: 1/28/21"},{"level":2,"text":"Day 04 HW: 1/29/21","anchor":"day-04-hw-12921","htmlText":"Day 04 HW: 1/29/21"},{"level":2,"text":"Day06 HW","anchor":"day06-hw","htmlText":"Day06 HW"},{"level":2,"text":"Day07 HW","anchor":"day07-hw","htmlText":"Day07 HW"},{"level":2,"text":"Day08 HW","anchor":"day08-hw","htmlText":"Day08 HW"},{"level":2,"text":"Day09 HW","anchor":"day09-hw","htmlText":"Day09 HW"},{"level":2,"text":"Day10 HW","anchor":"day10-hw","htmlText":"Day10 HW"},{"level":2,"text":"Day11 HW","anchor":"day11-hw","htmlText":"Day11 HW"},{"level":2,"text":"Day12 HW","anchor":"day12-hw","htmlText":"Day12 HW"},{"level":2,"text":"Day13HW","anchor":"day13hw","htmlText":"Day13HW"},{"level":2,"text":"Day13","anchor":"day13","htmlText":"Day13"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fkcrid001%2FKatieCriderAdvancedGenomicsLog"}}],"overviewFilesProcessingTime":429.325021,"createFromTemplatePath":"/new?template_name=KatieCriderAdvancedGenomicsLog\u0026template_owner=kcrid001"}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":"https://github.dev/","enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":true,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":true}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cLeluw:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cLeluw:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cLeluw[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cLeluw:active{-webkit-transition:none;transition:none;}/*!sc*/
.cLeluw[data-inactive]{cursor:auto;}/*!sc*/
.cLeluw:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled [data-component=ButtonCounter],.cLeluw:disabled [data-component="leadingVisual"],.cLeluw:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cLeluw:focus{outline:solid 1px transparent;}}/*!sc*/
.cLeluw [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cLeluw[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cLeluw[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cLeluw[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cLeluw[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cLeluw[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cLeluw[data-block="block"]{width:100%;}/*!sc*/
.cLeluw[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cLeluw[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cLeluw [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cLeluw [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cLeluw [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cLeluw [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cLeluw:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cLeluw[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cLeluw[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cLeluw:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cLeluw:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cLeluw[data-size="small"][data-no-visuals]{margin-right:8px;height:28px;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,cLeluw,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/kcrid001/KatieCriderAdvancedGenomicsLog/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/kcrid001/KatieCriderAdvancedGenomicsLog/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <button hidden="">
                  </button>
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" aria-label="Add file" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R2hil5:" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-plus" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="react-directory-remove-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 pyyxt" id=":R2lil5:" tabindex="0" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Add file
                     </span>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </button>
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/kcrid001/KatieCriderAdvancedGenomicsLog/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           16 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/kcrid001/KatieCriderAdvancedGenomicsLog/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/kcrid001/KatieCriderAdvancedGenomicsLog/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/kcrid001/KatieCriderAdvancedGenomicsLog/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-label="Edit file" class="types__StyledButton-sc-ws60qy-0 cLeluw" data-component="IconButton" data-no-visuals="true" data-size="small" title="Edit file" type="button">
                    <svg aria-hidden="true" class="octicon octicon-pencil" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </button>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Katie Crider's Advanced Genomics Log
                     </h1>
                     <a aria-label="Permalink: Katie Crider's Advanced Genomics Log" class="anchor" href="#katie-criders-advanced-genomics-log" id="user-content-katie-criders-advanced-genomics-log">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Spring 2021
                     </h2>
                     <a aria-label="Permalink: Spring 2021" class="anchor" href="#spring-2021" id="user-content-spring-2021">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 02 Exercises: 1/22/21
                     </h2>
                     <a aria-label="Permalink: Day 02 Exercises: 1/22/21" class="anchor" href="#day-02-exercises-12221" id="user-content-day-02-exercises-12221">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#exercises day02:

#1- Logon to the cluster @turing.hpc.odu.edu
(base) Katies-MacBook-Air-4:~ katiecrider$ ssh kcrid001@turing.hpc.odu.edu             
[kcrid001@turing1 ~]$ cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider

#2- Make a directory in your course workspace called data
[kcrid001@turing1 katiecrider]$ mkdir data

#3- Make a directory called exercises in your data directory
[kcrid001@turing1 katiecrider]$ cd data
[kcrid001@turing1 data]$ mkdir exercises
[kcrid001@turing1 data]$ cd exercises

#4- execute a pwd command and start a log of your commands with a header for today's date in your README.md github logfile in your workspace
[kcrid001@turing1 exercises]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/exercises

#5- cp the Exercise2.fasta.gz and Exercise2.fastq.tar.gz files into your exercises directory from the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02 directory
[kcrid001@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise2.fast* ./
[kcrid001@turing1 exercises]$ ls
Exercise2.fasta.gz  Exercise2.fastq.tar.gz

#6- unzip and untar the files
[kcrid001@turing1 exercises]$ gunzip Exercise2.fasta.gz
[kcrid001@turing1 exercises]$ ls
Exercise2.fasta  Exercise2.fastq.tar.gz
[kcrid001@turing1 exercises]$ tar -zxvf Exercise2.fastq.tar.gz
Exercise2.fastq
[kcrid001@turing1 exercises]$ ls
Exercise2.fasta  Exercise2.fastq  Exercise2.fastq.tar.gz
[kcrid001@turing1 exercises]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/exercises

#7- add these commands to your notebook file
Done!

#8- calculate how many sequences are in each file and add these results to your notebook file
[kcrid001@turing1 exercises]$ head -1 Exercise2.fasta
&gt;scaffold10078|size20675
[kcrid001@turing1 exercises]$ grep -c \&gt; Exercise2.fasta
138
[kcrid001@turing1 exercises]$ head -4 Exercise2.fastq
@HS3:541:HAYTUADXX:1:1101:1297:1938 1:N:0:AGTTCC
NCGGTGACAGGATTTTCAGGGCAGGAGGGACATTCGGGAAGGGGGAGGGAGATAAAAGAGGCAA
+
#1=DBDDFHHHHHJJJIJJJJJJJJIHJJIIIIJJJJJGIIJJJJAHFDBD?CDDDDDDDDDDD
[kcrid001@turing1 exercises]$ wc -l Exercise2.fastq
245216 Exercise2.fastq
[kcrid001@turing1 exercises]$ echo 245216/4 | bc
61304
[kcrid001@turing1 exercises]$ grep -c &quot;@HS3:541&quot; Exercise2.fastq
61304

#9 - cp the avg_cov_len_fasta_advbioinf.py from the /cm/shared/courses/dbarshis/21AdvGenomics/scripts directory into your class scripts directory
[kcrid001@turing1 exercises]$ cd ../../
[kcrid001@turing1 katiecrider]$ cd scripts
[kcrid001@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics//scripts/avg_cov_len_fasta_advbioinf.py ./ 

#10- start an interactive compute session and re-navigate to your exercises directory
I had trouble doing this with Turing so I switched over to Wahab

[kcrid001@turing1 katiecrider]$ exit
logout
Connection to turing.hpc.odu.edu closed.
(base) Katies-MacBook-Air-4:~ katiecrider$ ssh kcrid001@wahab.hpc.odu.edu
wahab-01:~&gt; cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/
wahab-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider&gt; salloc
salloc: Pending job allocation 133738
salloc: job 133738 queued and waiting for resources
salloc: job 133738 has been allocated resources
salloc: Granted job allocation 133738
This session will be terminated in 7 days. If your application requires
a longer excution time, please use command &quot;salloc -t N-0&quot; where N is the
number of days that you need.

#11- run the avg_cov_len_fasta_DJB.py script on your Exercise2.fasta file by typing the path to the script followed by the Exercise2.fasta file name
e2-w6420b-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider&gt; cd data/exercises/
e2-w6420b-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/exercises&gt; ../../scripts/avg_cov_len_fasta_advbioinf.py Exercise2.fasta
The total number of sequences is 138
The average sequence length is 126640
The total number of bases is 17476447
The minimum sequence length is 1122
The maximum sequence length is 562680
The N50 is 187037
Median Length = 92612
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 138
contigs &gt;= 1000bp = 138
contigs &gt;= 2000bp = 135

#12- did you add all these entries to your notebook file, including the results of the script?
Yes

#13- push your notebook file to your github page
(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ git add README.md
(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ git commit -m 'updating notebook with day02 exercises'
(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ git push -u origin main
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>exercises day02:</span>

<span class="pl-c"><span class="pl-c">#</span>1- Logon to the cluster @turing.hpc.odu.edu</span>
(base) Katies-MacBook-Air-4:<span class="pl-k">~</span> katiecrider$ ssh kcrid001@turing.hpc.odu.edu             
[kcrid001@turing1 <span class="pl-k">~</span>]$ <span class="pl-c1">cd</span> /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider

<span class="pl-c"><span class="pl-c">#</span>2- Make a directory in your course workspace called data</span>
[kcrid001@turing1 katiecrider]$ mkdir data

<span class="pl-c"><span class="pl-c">#</span>3- Make a directory called exercises in your data directory</span>
[kcrid001@turing1 katiecrider]$ <span class="pl-c1">cd</span> data
[kcrid001@turing1 data]$ mkdir exercises
[kcrid001@turing1 data]$ <span class="pl-c1">cd</span> exercises

<span class="pl-c"><span class="pl-c">#</span>4- execute a pwd command and start a log of your commands with a header for today's date in your README.md github logfile in your workspace</span>
[kcrid001@turing1 exercises]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/exercises

<span class="pl-c"><span class="pl-c">#</span>5- cp the Exercise2.fasta.gz and Exercise2.fastq.tar.gz files into your exercises directory from the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02 directory</span>
[kcrid001@turing1 exercises]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise2.fast<span class="pl-k">*</span> ./
[kcrid001@turing1 exercises]$ ls
Exercise2.fasta.gz  Exercise2.fastq.tar.gz

<span class="pl-c"><span class="pl-c">#</span>6- unzip and untar the files</span>
[kcrid001@turing1 exercises]$ gunzip Exercise2.fasta.gz
[kcrid001@turing1 exercises]$ ls
Exercise2.fasta  Exercise2.fastq.tar.gz
[kcrid001@turing1 exercises]$ tar -zxvf Exercise2.fastq.tar.gz
Exercise2.fastq
[kcrid001@turing1 exercises]$ ls
Exercise2.fasta  Exercise2.fastq  Exercise2.fastq.tar.gz
[kcrid001@turing1 exercises]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/exercises

<span class="pl-c"><span class="pl-c">#</span>7- add these commands to your notebook file</span>
Done<span class="pl-k">!</span>

<span class="pl-c"><span class="pl-c">#</span>8- calculate how many sequences are in each file and add these results to your notebook file</span>
[kcrid001@turing1 exercises]$ head -1 Exercise2.fasta
<span class="pl-k">&gt;</span>scaffold10078<span class="pl-k">|</span>size20675
[kcrid001@turing1 exercises]$ grep -c <span class="pl-cce">\&gt;</span> Exercise2.fasta
138
[kcrid001@turing1 exercises]$ head -4 Exercise2.fastq
@HS3:541:HAYTUADXX:1:1101:1297:1938 1:N:0:AGTTCC
NCGGTGACAGGATTTTCAGGGCAGGAGGGACATTCGGGAAGGGGGAGGGAGATAAAAGAGGCAA
+
<span class="pl-c"><span class="pl-c">#</span>1=DBDDFHHHHHJJJIJJJJJJJJIHJJIIIIJJJJJGIIJJJJAHFDBD?CDDDDDDDDDDD</span>
[kcrid001@turing1 exercises]$ wc -l Exercise2.fastq
245216 Exercise2.fastq
[kcrid001@turing1 exercises]$ <span class="pl-c1">echo</span> 245216/4 <span class="pl-k">|</span> bc
61304
[kcrid001@turing1 exercises]$ grep -c <span class="pl-s"><span class="pl-pds">"</span>@HS3:541<span class="pl-pds">"</span></span> Exercise2.fastq
61304

<span class="pl-c"><span class="pl-c">#</span>9 - cp the avg_cov_len_fasta_advbioinf.py from the /cm/shared/courses/dbarshis/21AdvGenomics/scripts directory into your class scripts directory</span>
[kcrid001@turing1 exercises]$ <span class="pl-c1">cd</span> ../../
[kcrid001@turing1 katiecrider]$ <span class="pl-c1">cd</span> scripts
[kcrid001@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics//scripts/avg_cov_len_fasta_advbioinf.py ./ 

<span class="pl-c"><span class="pl-c">#</span>10- start an interactive compute session and re-navigate to your exercises directory</span>
I had trouble doing this with Turing so I switched over to Wahab

[kcrid001@turing1 katiecrider]$ <span class="pl-c1">exit</span>
<span class="pl-c1">logout</span>
Connection to turing.hpc.odu.edu closed.
(base) Katies-MacBook-Air-4:<span class="pl-k">~</span> katiecrider$ ssh kcrid001@wahab.hpc.odu.edu
wahab-01:<span class="pl-k">~</span><span class="pl-k">&gt;</span> <span class="pl-c1">cd</span> /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/
wahab-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider<span class="pl-k">&gt;</span> salloc
salloc: Pending job allocation 133738
salloc: job 133738 queued and waiting <span class="pl-k">for</span> resources
salloc: job 133738 has been allocated resources
salloc: Granted job allocation 133738
This session will be terminated <span class="pl-k">in</span> 7 days. If your application requires
a longer excution time, please use <span class="pl-c1">command</span> <span class="pl-s"><span class="pl-pds">"</span>salloc -t N-0<span class="pl-pds">"</span></span> where N is the
number of days that you need.

<span class="pl-c"><span class="pl-c">#</span>11- run the avg_cov_len_fasta_DJB.py script on your Exercise2.fasta file by typing the path to the script followed by the Exercise2.fasta file name</span>
e2-w6420b-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider<span class="pl-k">&gt;</span> <span class="pl-c1">cd</span> data/exercises/
e2-w6420b-01:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/exercises<span class="pl-k">&gt;</span> ../../scripts/avg_cov_len_fasta_advbioinf.py Exercise2.fasta
The total number of sequences is 138
The average sequence length is 126640
The total number of bases is 17476447
The minimum sequence length is 1122
The maximum sequence length is 562680
The N50 is 187037
Median Length = 92612
contigs <span class="pl-k">&lt;</span> 150bp = 0
contigs <span class="pl-k">&gt;</span>= 500bp = 138
contigs <span class="pl-k">&gt;</span>= 1000bp = 138
contigs <span class="pl-k">&gt;</span>= 2000bp = 135

<span class="pl-c"><span class="pl-c">#</span>12- did you add all these entries to your notebook file, including the results of the script?</span>
Yes

<span class="pl-c"><span class="pl-c">#</span>13- push your notebook file to your github page</span>
(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ git add README.md
(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ git commit -m <span class="pl-s"><span class="pl-pds">'</span>updating notebook with day02 exercises<span class="pl-pds">'</span></span>
(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ git push -u origin main
</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 03 Exercises/Day 2 HW: 1/27/21
                     </h2>
                     <a aria-label="Permalink: Day 03 Exercises/Day 2 HW: 1/27/21" class="anchor" href="#day-03-exercisesday-2-hw-12721" id="user-content-day-03-exercisesday-2-hw-12721">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#exercises day03 /Day02HW:

#1 Write an sbatch script to cp the files /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/ into your own data directory

(base) Katies-MacBook-Air-4:21sp_advgenomics katiecrider$ nano KCridCopyLane04.sh
(base) Katies-MacBook-Air-4:21sp_advgenomics katiecrider$ scp KCridCopyLane04.sh kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/KCridCopyLane04.sh
kcrid001@turing.hpc.odu.edu's password: 
KCridCopyLane04.sh                                              100%  325    10.6KB/s   00:00    

[kcrid001@coreV2-25-072 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@coreV2-25-072 data]$ ls
exercises  KCridCopyLane04.sh

#2 Add the content of your sbatch script to your logfile

[kcrid001@coreV2-25-072 data]$ cat KCridCopyLane04.sh 
#!/bin/bash -l

#SBATCH -o KCridCopyLane04.txt
#SBATCH -n 1
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KCridCopyLane04


cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB04-* /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data


#3 submit the slurm script (sbatch scripname.sh) and verify that it's working (by squeue -u yourusername multiple times and checking the destination directory to make sure the files are being created)

[kcrid001@coreV2-25-072 data]$ sbatch KCridCopyLane04.sh 
Submitted batch job 9270446
[kcrid001@coreV2-25-072 data]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9270424      main       sh kcrid001  R      46:39      1 coreV2-25-072 
           9270446      main KCridCop kcrid001  R      10:45      1 coreV2-25-072 
#4 Make sure this is all documented on your github notebook
  #Done

#5 Write a sbatch script to gunzip all the fastq.gz files

[kcrid001@coreV2-25-072 data]$ nano KCridGunzipLane04.sh
[kcrid001@coreV2-25-072 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@coreV2-25-072 data]$ cat KCridGunzipLane04.sh
#!/bin/bash -l

#SBATCH -o KCridGunzipLane04.txt
#SBATCH -n 1
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KCridGunzipLane04

gunzip *.fastq.gz

[kcrid001@coreV2-25-072 data]$ sbatch KCridGunzipLane04.sh 
Submitted batch job 9270468
[kcrid001@coreV2-25-072 data]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9270424      main       sh kcrid001  R      57:21      1 coreV2-25-072 
           9270468      main KCridGun kcrid001  R       0:34      1 coreV2-25-072 
[kcrid001@coreV2-25-072 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@coreV2-25-072 data]$ ls
exercises                          HADB04-J_S74_L005_R1_001.fastq.gz
HADB04-A_S65_L005_R1_001.fastq     HADB04-K_S75_L005_R1_001.fastq.gz
HADB04-A_S65_L005_R1_001.fastq.gz  HADB04-L_S76_L005_R1_001.fastq.gz
HADB04-B_S66_L005_R1_001.fastq.gz  HADB04-M_S77_L005_R1_001.fastq.gz
HADB04-C_S67_L005_R1_001.fastq.gz  HADB04-N_S78_L005_R1_001.fastq.gz
HADB04-D_S68_L005_R1_001.fastq.gz  HADB04-O_S79_L005_R1_001.fastq.gz
HADB04-E_S69_L005_R1_001.fastq.gz  HADB04-P_S80_L005_R1_001.fastq.gz
HADB04-F_S70_L005_R1_001.fastq.gz  KCridCopyLane04.sh
HADB04-G_S71_L005_R1_001.fastq.gz  KCridCopyLane04.txt
HADB04-H_S72_L005_R1_001.fastq.gz  KCridGunzipLane04.sh
HADB04-I_S73_L005_R1_001.fastq.gz  KCridGunzipLane04.txt
[kcrid001@coreV2-25-072 data]$ ls
exercises                       HADB04-G_S71_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq
HADB04-A_S65_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq
HADB04-B_S66_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq
HADB04-C_S67_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  KCridCopyLane04.sh
HADB04-D_S68_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  KCridCopyLane04.txt
HADB04-E_S69_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq  KCridGunzipLane04.sh
HADB04-F_S70_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq  KCridGunzipLane04.txt
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>exercises day03 /Day02HW:</span>

<span class="pl-c"><span class="pl-c">#</span>1 Write an sbatch script to cp the files /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/ into your own data directory</span>

(base) Katies-MacBook-Air-4:21sp_advgenomics katiecrider$ nano KCridCopyLane04.sh
(base) Katies-MacBook-Air-4:21sp_advgenomics katiecrider$ scp KCridCopyLane04.sh kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/KCridCopyLane04.sh
kcrid001@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password: </span>
<span class="pl-s">KCridCopyLane04.sh                                              100%  325    10.6KB/s   00:00    </span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV2-25-072 data]$ pwd</span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data</span>
<span class="pl-s">[kcrid001@coreV2-25-072 data]$ ls</span>
<span class="pl-s">exercises  KCridCopyLane04.sh</span>
<span class="pl-s"></span>
<span class="pl-s">#2 Add the content of your sbatch script to your logfile</span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV2-25-072 data]$ cat KCridCopyLane04.sh </span>
<span class="pl-s">#!/bin/bash -l</span>
<span class="pl-s"></span>
<span class="pl-s">#SBATCH -o KCridCopyLane04.txt</span>
<span class="pl-s">#SBATCH -n 1</span>
<span class="pl-s">#SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-s">#SBATCH --mail-type=END</span>
<span class="pl-s">#SBATCH --job-name=KCridCopyLane04</span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/HADB04-* /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data</span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">#3 submit the slurm script (sbatch scripname.sh) and verify that it<span class="pl-pds">'</span></span>s working (by squeue -u yourusername multiple <span class="pl-c1">times</span> and checking the destination directory to make sure the files are being created)

[kcrid001@coreV2-25-072 data]$ sbatch KCridCopyLane04.sh 
Submitted batch job 9270446
[kcrid001@coreV2-25-072 data]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9270424      main       sh kcrid001  R      46:39      1 coreV2-25-072 
           9270446      main KCridCop kcrid001  R      10:45      1 coreV2-25-072 
<span class="pl-c"><span class="pl-c">#</span>4 Make sure this is all documented on your github notebook</span>
  <span class="pl-c"><span class="pl-c">#</span>Done</span>

<span class="pl-c"><span class="pl-c">#</span>5 Write a sbatch script to gunzip all the fastq.gz files</span>

[kcrid001@coreV2-25-072 data]$ nano KCridGunzipLane04.sh
[kcrid001@coreV2-25-072 data]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@coreV2-25-072 data]$ cat KCridGunzipLane04.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KCridGunzipLane04.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KCridGunzipLane04</span>

gunzip <span class="pl-k">*</span>.fastq.gz

[kcrid001@coreV2-25-072 data]$ sbatch KCridGunzipLane04.sh 
Submitted batch job 9270468
[kcrid001@coreV2-25-072 data]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9270424      main       sh kcrid001  R      57:21      1 coreV2-25-072 
           9270468      main KCridGun kcrid001  R       0:34      1 coreV2-25-072 
[kcrid001@coreV2-25-072 data]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@coreV2-25-072 data]$ ls
exercises                          HADB04-J_S74_L005_R1_001.fastq.gz
HADB04-A_S65_L005_R1_001.fastq     HADB04-K_S75_L005_R1_001.fastq.gz
HADB04-A_S65_L005_R1_001.fastq.gz  HADB04-L_S76_L005_R1_001.fastq.gz
HADB04-B_S66_L005_R1_001.fastq.gz  HADB04-M_S77_L005_R1_001.fastq.gz
HADB04-C_S67_L005_R1_001.fastq.gz  HADB04-N_S78_L005_R1_001.fastq.gz
HADB04-D_S68_L005_R1_001.fastq.gz  HADB04-O_S79_L005_R1_001.fastq.gz
HADB04-E_S69_L005_R1_001.fastq.gz  HADB04-P_S80_L005_R1_001.fastq.gz
HADB04-F_S70_L005_R1_001.fastq.gz  KCridCopyLane04.sh
HADB04-G_S71_L005_R1_001.fastq.gz  KCridCopyLane04.txt
HADB04-H_S72_L005_R1_001.fastq.gz  KCridGunzipLane04.sh
HADB04-I_S73_L005_R1_001.fastq.gz  KCridGunzipLane04.txt
[kcrid001@coreV2-25-072 data]$ ls
exercises                       HADB04-G_S71_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq
HADB04-A_S65_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq
HADB04-B_S66_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq
HADB04-C_S67_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  KCridCopyLane04.sh
HADB04-D_S68_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  KCridCopyLane04.txt
HADB04-E_S69_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq  KCridGunzipLane04.sh
HADB04-F_S70_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq  KCridGunzipLane04.txt
</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 03 HW: 1/28/21
                     </h2>
                     <a aria-label="Permalink: Day 03 HW: 1/28/21" class="anchor" href="#day-03-hw-12821" id="user-content-day-03-hw-12821">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#HW day03:

#1. cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 directory (and files) to your sandbox

[kcrid001@turing1 katiecrider]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider
[kcrid001@turing1 katiecrider]$ cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/ ./
[kcrid001@turing1 katiecrider]$ ls
data  day03  groundrules.txt  kcrid_exercise1.txt  scripts


#2. mkdir a fastq directory in your data directory and mv all the .fastq files into this directory

[kcrid001@turing1 katiecrider]$ cd data
[kcrid001@turing1 data]$ ls
exercises                       HADB04-G_S71_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq
HADB04-A_S65_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq
HADB04-B_S66_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq
HADB04-C_S67_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  KCridCopyLane04.sh
HADB04-D_S68_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  KCridCopyLane04.txt
HADB04-E_S69_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq  KCridGunzipLane04.sh
HADB04-F_S70_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq  KCridGunzipLane04.txt
[kcrid001@turing1 data]$ mkdir fastq
[kcrid001@turing1 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@turing1 data]$ mv *.fastq ./fastq/
[kcrid001@turing1 data]$ ls
exercises  fastq  KCridCopyLane04.sh  KCridCopyLane04.txt  KCridGunzipLane04.sh  KCridGunzipLane04.txt
[kcrid001@turing1 data]$ cd fastq
[kcrid001@turing1 fastq]$ ls
HADB04-A_S65_L005_R1_001.fastq  HADB04-G_S71_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq
HADB04-B_S66_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq
HADB04-C_S67_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq
HADB04-D_S68_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq
HADB04-E_S69_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq
HADB04-F_S70_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq


#3. cp the renamingtable_complete.txt from the day03 directory into your fastq directory and the /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your sandbox scripts folder and less the new script and check out the usage statement

[kcrid001@turing1 fastq]$ cd ../
[kcrid001@turing1 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@turing1 data]$ cp ../day03/renamingtable_complete.txt ./fastq/
[kcrid001@turing1 data]$ cd fastq
[kcrid001@turing1 fastq]$ ls
HADB04-A_S65_L005_R1_001.fastq  HADB04-G_S71_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq
HADB04-B_S66_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq
HADB04-C_S67_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq
HADB04-D_S68_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq
HADB04-E_S69_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  renamingtable_complete.txt
HADB04-F_S70_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fasq


[kcrid001@turing1 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/scripts
[kcrid001@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py ./
[kcrid001@turing1 scripts]$ ls
avg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py

[kcrid001@turing1 scripts]$ head renamer_advbioinf.py 
#!/usr/bin/env python
####usage renamer.py renamingtable
#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column
import sys
import os

fin=open(sys.argv[1],'r')
linecount=0
for line in fin:
	linecount+=1


#4. run the renamer_advbioinf.py script in your fastq folder using the renamingtable_complete.txt as practice and verify the output to the screen by hand

[kcrid001@turing1 scripts]$ cd ../data/fastq/
[kcrid001@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq
[kcrid001@turing1 fastq]$ ../../scripts/renamer_advbioinf.py renamingtable_complete.txt
mv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq
mv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq
mv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq
mv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq
mv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq
mv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq
mv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq
mv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq
mv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq
mv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq
mv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq
mv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq
mv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq
mv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq
mv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq
mv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq
mv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq
mv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq
mv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq
mv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq
mv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq
mv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq
mv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq
mv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq
mv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq
mv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq
mv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq
mv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq
mv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq
mv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq
mv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq
mv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq
mv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq
mv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq
mv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq
mv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq
mv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq
mv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq
mv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq
mv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq
mv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq
mv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq
mv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq
mv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq
mv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq
mv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq
mv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq
mv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq
mv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq
mv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq
mv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq
mv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq
mv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq
mv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq
mv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq
mv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq
mv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq
mv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq
mv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq
mv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq
mv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq
mv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq
mv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq
mv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq
mv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq
mv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq
mv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq
mv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq
mv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq
mv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq
mv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq
mv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq
mv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq
mv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq
mv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq
mv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq
mv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq
mv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq
mv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq
mv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq
mv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq
mv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq
mv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq
mv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq
mv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq
mv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq
mv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq
mv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq
mv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq
mv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq
mv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq
mv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq
mv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq
mv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq
mv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq
mv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq




#5. Uncomment the last line of the renaming script in your scripts folder that starts with os.popen and comment out the next to last line that starts with print

[kcrid001@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq
[kcrid001@turing1 fastq]$ nano ../../scripts/renamer_advbioinf.py
[kcrid001@turing1 fastq]$ cat ../../scripts/renamer_advbioinf.py 
#!/usr/bin/env python
####usage renamer.py renamingtable
#### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column
import sys
import os

fin=open(sys.argv[1],'r')
linecount=0
for line in fin:
	linecount+=1
	if linecount&gt;=2:
		cols=line.rstrip().split('\t')
#		print 'mv %s %s' %(cols[0], cols[1])
		os.popen('mv %s %s' %(cols[0], cols[1]))


#6. write a sbatch script and submit it to rename all the .fastq files according to the renaming table using your renamer_advbioinf.py script

[kcrid001@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq
[kcrid001@turing1 fastq]$ nano KCridRenamer.sh
[kcrid001@turing1 fastq]$ ls
HADB04-A_S65_L005_R1_001.fastq  HADB04-G_S71_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq
HADB04-B_S66_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq
HADB04-C_S67_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq
HADB04-D_S68_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq
HADB04-E_S69_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  KCridRenamer.sh
HADB04-F_S70_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq  renamingtable_complete.txt
[kcrid001@turing1 fastq]$ cat KCridRenamer.sh 
#!/bin/bash -l

#SBATCH -o KCridRenamer.txt
#SBATCH -n 1
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=FastqRename

../../scripts/renamer_advbioinf.py renamingtable_complete.txt

[kcrid001@turing1 fastq]$ sbatch ./KCridRenamer.sh
Submitted batch job 9270863

#7. Make sure this is all documented on your github page
* Yep

#8. The naming convention for the files is as follows:
  * SOURCEPOPULATION_SYMBIOTICSTATE_GENOTYPE_TEMPERATURE.fastq
  * There are 2 sources: Virginia and Rhode Island
  * There are 2 symbiotic states: Brown and White
  
#9. Next, you're going to start the process of adapter clipping and quality trimming all the renamed .fastq files in batches, by lane

#10. cp the script /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py into your scripts directory

[kcrid001@turing1 fastq]$ cd ../
[kcrid001@turing1 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@turing1 data]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py ../scripts/


#11. Less/head the new script and check out the usage statement
[kcrid001@turing1 data]$ head -14 ../scripts/Trimclipfilterstatsbatch_advbioinf.py
#!/usr/bin/env python
# Written by Dan Barshis

import sys, os

########### Usage #############
# Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles
# This should be run from within the folder with all of your original .fastqstrim
# Will quality trim multiple SINGLE-END fastq files
# Things to customize for your particular platform:
# qualoffset = 33 Quality score offset
# -t option in qualitytrim (the lower threshold quality score for trimming)
# -l option in qualitytrim and adapterclip (the length threshold for throwing out short reads)


#12. cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt into the working directory with your fastq files

[kcrid001@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq
[kcrid001@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt ./
[kcrid001@turing1 fastq]$ ls
adapterlist_advbioinf.txt   RI_B_04_14.fastq  RI_W_04_14.fastq  VA_B_04_14.fastq  VA_W_04_14.fastq
KCridRenamer.sh             RI_B_04_18.fastq  RI_W_04_18.fastq  VA_B_04_18.fastq  VA_W_04_18.fastq
KCridRenamer.txt            RI_B_04_22.fastq  RI_W_04_22.fastq  VA_B_04_22.fastq  VA_W_04_22.fastq
renamingtable_complete.txt  RI_B_05_14.fastq  RI_W_05_14.fastq  VA_B_05_14.fastq  VA_W_05_14.fastq

#13. Make a sbatch script for the Trimclipfilter... script and run it on your fastq files
[kcrid001@turing1 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq
[kcrid001@turing1 fastq]$ nano TestTrimClip.sh
[kcrid001@turing1 fastq]$ cat TestTrimClip.sh
#!/bin/bash -l

#SBATCH -o KCridtestTrimclip.txt
#SBATCH -n 1
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KCridTrimTest

/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq

[kcrid001@turing1 fastq]$ salloc
salloc: Pending job allocation 9271050
salloc: job 9271050 queued and waiting for resources
salloc: job 9271050 has been allocated resources
salloc: Granted job allocation 9271050
[kcrid001@coreV1-22-005 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq
[kcrid001@coreV1-22-005 fastq]$ sbatch TestTrimClip.sh
Submitted batch job 9271051
[kcrid001@coreV1-22-005 fastq]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9271051      main KCridTri kcrid001  R      10:22      1 coreV2-22-007 
           9271050      main       sh kcrid001  R      11:22      1 coreV1-22-005 

#I was confused on if I had to do the FullTrimClip next, because they look like the same thing, but I think the TestTrimClip is working (see list below)

[kcrid001@coreV1-22-005 fastq]$ ls
adapterlist_advbioinf.txt  renamingtable_complete.txt  RI_W_04_14.fastq  trimclipstats.txt  VA_W_04_18.fastq
FullTrimClip.sh            RI_B_04_14_clipped.fastq    RI_W_04_18.fastq  VA_B_04_14.fastq   VA_W_04_22.fastq
KCridFullTrimclip.txt      RI_B_04_14.fastq            RI_W_04_22.fastq  VA_B_04_18.fastq   VA_W_05_14.fastq
KCridRenamer.sh            RI_B_04_18.fastq            RI_W_05_14.fastq  VA_B_04_22.fastq
KCridRenamer.txt           RI_B_04_22.fastq            Testing           VA_B_05_14.fastq
KCridtestTrimclip.txt      RI_B_05_14.fastq            TestTrimClip.sh   VA_W_04_14.fastq
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>HW day03:</span>

<span class="pl-c"><span class="pl-c">#</span>1. cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 directory (and files) to your sandbox</span>

[kcrid001@turing1 katiecrider]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider
[kcrid001@turing1 katiecrider]$ cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/ ./
[kcrid001@turing1 katiecrider]$ ls
data  day03  groundrules.txt  kcrid_exercise1.txt  scripts


<span class="pl-c"><span class="pl-c">#</span>2. mkdir a fastq directory in your data directory and mv all the .fastq files into this directory</span>

[kcrid001@turing1 katiecrider]$ <span class="pl-c1">cd</span> data
[kcrid001@turing1 data]$ ls
exercises                       HADB04-G_S71_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq
HADB04-A_S65_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq
HADB04-B_S66_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq
HADB04-C_S67_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  KCridCopyLane04.sh
HADB04-D_S68_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  KCridCopyLane04.txt
HADB04-E_S69_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq  KCridGunzipLane04.sh
HADB04-F_S70_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq  KCridGunzipLane04.txt
[kcrid001@turing1 data]$ mkdir fastq
[kcrid001@turing1 data]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@turing1 data]$ mv <span class="pl-k">*</span>.fastq ./fastq/
[kcrid001@turing1 data]$ ls
exercises  fastq  KCridCopyLane04.sh  KCridCopyLane04.txt  KCridGunzipLane04.sh  KCridGunzipLane04.txt
[kcrid001@turing1 data]$ <span class="pl-c1">cd</span> fastq
[kcrid001@turing1 fastq]$ ls
HADB04-A_S65_L005_R1_001.fastq  HADB04-G_S71_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq
HADB04-B_S66_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq
HADB04-C_S67_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq
HADB04-D_S68_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq
HADB04-E_S69_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq
HADB04-F_S70_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq


<span class="pl-c"><span class="pl-c">#</span>3. cp the renamingtable_complete.txt from the day03 directory into your fastq directory and the /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py script into your sandbox scripts folder and less the new script and check out the usage statement</span>

[kcrid001@turing1 fastq]$ <span class="pl-c1">cd</span> ../
[kcrid001@turing1 data]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@turing1 data]$ cp ../day03/renamingtable_complete.txt ./fastq/
[kcrid001@turing1 data]$ <span class="pl-c1">cd</span> fastq
[kcrid001@turing1 fastq]$ ls
HADB04-A_S65_L005_R1_001.fastq  HADB04-G_S71_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq
HADB04-B_S66_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq
HADB04-C_S67_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq
HADB04-D_S68_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq
HADB04-E_S69_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  renamingtable_complete.txt
HADB04-F_S70_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fasq


[kcrid001@turing1 scripts]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/scripts
[kcrid001@turing1 scripts]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py ./
[kcrid001@turing1 scripts]$ ls
avg_cov_len_fasta_advbioinf.py  renamer_advbioinf.py

[kcrid001@turing1 scripts]$ head renamer_advbioinf.py 
<span class="pl-c"><span class="pl-c">#!</span>/usr/bin/env python</span>
<span class="pl-c"><span class="pl-c">#</span>###usage renamer.py renamingtable</span>
<span class="pl-c"><span class="pl-c">#</span>### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column</span>
import sys
import os

fin=open(sys.argv[1],<span class="pl-s"><span class="pl-pds">'</span>r<span class="pl-pds">'</span></span>)
linecount=0
<span class="pl-k">for</span> <span class="pl-smi">line</span> <span class="pl-k">in</span> fin:
	linecount+=1


<span class="pl-c"><span class="pl-c">#</span>4. run the renamer_advbioinf.py script in your fastq folder using the renamingtable_complete.txt as practice and verify the output to the screen by hand</span>

[kcrid001@turing1 scripts]$ <span class="pl-c1">cd</span> ../data/fastq/
[kcrid001@turing1 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq
[kcrid001@turing1 fastq]$ ../../scripts/renamer_advbioinf.py renamingtable_complete.txt
mv HADB01-A_S17_L002_R1_001.fastq VA_W_01_14.fastq
mv HADB01-B_S18_L002_R1_001.fastq VA_B_01_14.fastq
mv HADB01-C_S19_L002_R1_001.fastq RI_W_01_14.fastq
mv HADB01-D_S20_L002_R1_001.fastq RI_B_01_14.fastq
mv HADB01-E_S21_L002_R1_001.fastq VA_W_01_22.fastq
mv HADB01-F_S22_L002_R1_001.fastq VA_B_01_22.fastq
mv HADB01-G_S23_L002_R1_001.fastq RI_W_01_22.fastq
mv HADB01-H_S24_L002_R1_001.fastq RI_B_01_22.fastq
mv HADB01-I_S25_L002_R1_001.fastq VA_W_01_18.fastq
mv HADB01-J_S26_L002_R1_001.fastq VA_B_01_18.fastq
mv HADB01-K_S27_L002_R1_001.fastq RI_W_01_18.fastq
mv HADB01-L_S28_L002_R1_001.fastq RI_B_01_18.fastq
mv HADB01-M_S29_L002_R1_001.fastq VA_W_08_SNP.fastq
mv HADB01-N_S30_L002_R1_001.fastq VA_B_09_SNP.fastq
mv HADB01-O_S31_L002_R1_001.fastq RI_W_08_SNP.fastq
mv HADB01-P_S32_L002_R1_001.fastq RI_B_08_SNP.fastq
mv HADB02-A_S33_L003_R1_001.fastq VA_W_02_14.fastq
mv HADB02-B_S34_L003_R1_001.fastq VA_B_02_14.fastq
mv HADB02-C_S35_L003_R1_001.fastq RI_W_02_14.fastq
mv HADB02-D_S36_L003_R1_001.fastq RI_B_02_14.fastq
mv HADB02-E_S37_L003_R1_001.fastq VA_W_02_22.fastq
mv HADB02-F_S38_L003_R1_001.fastq VA_B_02_22.fastq
mv HADB02-G_S39_L003_R1_001.fastq RI_W_02_22.fastq
mv HADB02-H_S40_L003_R1_001.fastq RI_B_02_22.fastq
mv HADB02-I_S41_L003_R1_001.fastq VA_W_02_18.fastq
mv HADB02-J_S42_L003_R1_001.fastq VA_B_02_18.fastq
mv HADB02-K_S43_L003_R1_001.fastq RI_W_02_18.fastq
mv HADB02-L_S44_L003_R1_001.fastq RI_B_02_18.fastq
mv HADB02-M_S45_L003_R1_001.fastq VA_W_09_SNP.fastq
mv HADB02-N_S46_L003_R1_001.fastq VA_B_08_SNP.fastq
mv HADB02-O_S47_L003_R1_001.fastq RI_W_09_SNP.fastq
mv HADB02-P_S48_L003_R1_001.fastq RI_B_09_SNP.fastq
mv HADB03-A_S49_L004_R1_001.fastq VA_W_03_14.fastq
mv HADB03-B_S50_L004_R1_001.fastq VA_B_03_14.fastq
mv HADB03-C_S51_L004_R1_001.fastq RI_W_03_14.fastq
mv HADB03-D_S52_L004_R1_001.fastq RI_B_03_14.fastq
mv HADB03-E_S53_L004_R1_001.fastq VA_W_03_22.fastq
mv HADB03-F_S54_L004_R1_001.fastq VA_B_03_22.fastq
mv HADB03-G_S55_L004_R1_001.fastq RI_W_03_22.fastq
mv HADB03-H_S56_L004_R1_001.fastq RI_B_03_22.fastq
mv HADB03-I_S57_L004_R1_001.fastq VA_W_03_18.fastq
mv HADB03-J_S58_L004_R1_001.fastq VA_B_03_18.fastq
mv HADB03-K_S59_L004_R1_001.fastq RI_W_03_18.fastq
mv HADB03-L_S60_L004_R1_001.fastq RI_B_03_18.fastq
mv HADB03-M_S61_L004_R1_001.fastq VA_W_10_SNP.fastq
mv HADB03-N_S62_L004_R1_001.fastq VA_B_10_SNP.fastq
mv HADB03-O_S63_L004_R1_001.fastq RI_W_10_SNP.fastq
mv HADB03-P_S64_L004_R1_001.fastq RI_B_10_SNP.fastq
mv HADB04-A_S65_L005_R1_001.fastq VA_W_04_14.fastq
mv HADB04-B_S66_L005_R1_001.fastq VA_B_04_14.fastq
mv HADB04-C_S67_L005_R1_001.fastq RI_W_04_14.fastq
mv HADB04-D_S68_L005_R1_001.fastq RI_B_04_14.fastq
mv HADB04-E_S69_L005_R1_001.fastq VA_W_04_22.fastq
mv HADB04-F_S70_L005_R1_001.fastq VA_B_04_22.fastq
mv HADB04-G_S71_L005_R1_001.fastq RI_W_04_22.fastq
mv HADB04-H_S72_L005_R1_001.fastq RI_B_04_22.fastq
mv HADB04-I_S73_L005_R1_001.fastq VA_W_04_18.fastq
mv HADB04-J_S74_L005_R1_001.fastq VA_B_04_18.fastq
mv HADB04-K_S75_L005_R1_001.fastq RI_W_04_18.fastq
mv HADB04-L_S76_L005_R1_001.fastq RI_B_04_18.fastq
mv HADB04-M_S77_L005_R1_001.fastq VA_W_05_14.fastq
mv HADB04-N_S78_L005_R1_001.fastq VA_B_05_14.fastq
mv HADB04-O_S79_L005_R1_001.fastq RI_W_05_14.fastq
mv HADB04-P_S80_L005_R1_001.fastq RI_B_05_14.fastq
mv HADB05-A_S81_L006_R1_001.fastq VA_W_05_22.fastq
mv HADB05-B_S82_L006_R1_001.fastq VA_B_05_22.fastq
mv HADB05-C_S83_L006_R1_001.fastq RI_W_05_22.fastq
mv HADB05-D_S84_L006_R1_001.fastq RI_B_05_22.fastq
mv HADB05-E_S85_L006_R1_001.fastq VA_W_05_18.fastq
mv HADB05-F_S86_L006_R1_001.fastq VA_B_05_18.fastq
mv HADB05-G_S87_L006_R1_001.fastq RI_W_05_18.fastq
mv HADB05-H_S88_L006_R1_001.fastq RI_B_05_18.fastq
mv HADB05-I_S89_L006_R1_001.fastq VA_W_06_14.fastq
mv HADB05-J_S90_L006_R1_001.fastq VA_B_06_14.fastq
mv HADB05-K_S91_L006_R1_001.fastq RI_W_06_14.fastq
mv HADB05-L_S92_L006_R1_001.fastq RI_B_06_14.fastq
mv HADB05-M_S93_L006_R1_001.fastq VA_W_06_22.fastq
mv HADB05-N_S94_L006_R1_001.fastq VA_B_06_22.fastq
mv HADB05-O_S95_L006_R1_001.fastq RI_W_06_22.fastq
mv HADB05-P_S96_L006_R1_001.fastq RI_B_06_22.fastq
mv HADB06-A_S97_L007_R1_001.fastq VA_W_06_18.fastq
mv HADB06-B_S98_L007_R1_001.fastq VA_B_06_18.fastq
mv HADB06-C_S99_L007_R1_001.fastq RI_W_06_18.fastq
mv HADB06-D_S100_L007_R1_001.fastq RI_B_06_18.fastq
mv HADB06-E_S101_L007_R1_001.fastq VA_W_07_14.fastq
mv HADB06-F_S102_L007_R1_001.fastq VA_B_07_14.fastq
mv HADB06-G_S103_L007_R1_001.fastq RI_W_07_14.fastq
mv HADB06-H_S104_L007_R1_001.fastq RI_B_07_14.fastq
mv HADB06-I_S105_L007_R1_001.fastq VA_W_07_22.fastq
mv HADB06-J_S106_L007_R1_001.fastq VA_B_07_22.fastq
mv HADB06-K_S107_L007_R1_001.fastq RI_W_07_22.fastq
mv HADB06-L_S108_L007_R1_001.fastq RI_B_07_22.fastq
mv HADB06-M_S109_L007_R1_001.fastq VA_W_07_18.fastq
mv HADB06-N_S110_L007_R1_001.fastq VA_B_07_18.fastq
mv HADB06-O_S111_L007_R1_001.fastq RI_W_07_18.fastq
mv HADB06-P_S112_L007_R1_001.fastq RI_B_07_18.fastq




<span class="pl-c"><span class="pl-c">#</span>5. Uncomment the last line of the renaming script in your scripts folder that starts with os.popen and comment out the next to last line that starts with print</span>

[kcrid001@turing1 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq
[kcrid001@turing1 fastq]$ nano ../../scripts/renamer_advbioinf.py
[kcrid001@turing1 fastq]$ cat ../../scripts/renamer_advbioinf.py 
<span class="pl-c"><span class="pl-c">#!</span>/usr/bin/env python</span>
<span class="pl-c"><span class="pl-c">#</span>###usage renamer.py renamingtable</span>
<span class="pl-c"><span class="pl-c">#</span>### this script take the entries in the first column of table and renames (mv's) them to files with the names in the second column</span>
import sys
import os

fin=open(sys.argv[1],<span class="pl-s"><span class="pl-pds">'</span>r<span class="pl-pds">'</span></span>)
linecount=0
<span class="pl-k">for</span> <span class="pl-smi">line</span> <span class="pl-k">in</span> fin:
	linecount+=1
	<span class="pl-k">if</span> linecount<span class="pl-k">&gt;</span>=2:
		cols=line.rstrip().split(<span class="pl-s"><span class="pl-pds">'</span>\t<span class="pl-pds">'</span></span>)
<span class="pl-c"><span class="pl-c">#</span>		print 'mv %s %s' %(cols[0], cols[1])</span>
		os.popen(<span class="pl-s"><span class="pl-pds">'</span>mv %s %s<span class="pl-pds">'</span></span> %(cols[0], cols[1]))


<span class="pl-c"><span class="pl-c">#</span>6. write a sbatch script and submit it to rename all the .fastq files according to the renaming table using your renamer_advbioinf.py script</span>

[kcrid001@turing1 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq
[kcrid001@turing1 fastq]$ nano KCridRenamer.sh
[kcrid001@turing1 fastq]$ ls
HADB04-A_S65_L005_R1_001.fastq  HADB04-G_S71_L005_R1_001.fastq  HADB04-M_S77_L005_R1_001.fastq
HADB04-B_S66_L005_R1_001.fastq  HADB04-H_S72_L005_R1_001.fastq  HADB04-N_S78_L005_R1_001.fastq
HADB04-C_S67_L005_R1_001.fastq  HADB04-I_S73_L005_R1_001.fastq  HADB04-O_S79_L005_R1_001.fastq
HADB04-D_S68_L005_R1_001.fastq  HADB04-J_S74_L005_R1_001.fastq  HADB04-P_S80_L005_R1_001.fastq
HADB04-E_S69_L005_R1_001.fastq  HADB04-K_S75_L005_R1_001.fastq  KCridRenamer.sh
HADB04-F_S70_L005_R1_001.fastq  HADB04-L_S76_L005_R1_001.fastq  renamingtable_complete.txt
[kcrid001@turing1 fastq]$ cat KCridRenamer.sh 
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KCridRenamer.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=FastqRename</span>

../../scripts/renamer_advbioinf.py renamingtable_complete.txt

[kcrid001@turing1 fastq]$ sbatch ./KCridRenamer.sh
Submitted batch job 9270863

<span class="pl-c"><span class="pl-c">#</span>7. Make sure this is all documented on your github page</span>
<span class="pl-k">*</span> Yep

<span class="pl-c"><span class="pl-c">#</span>8. The naming convention for the files is as follows:</span>
  <span class="pl-k">*</span> SOURCEPOPULATION_SYMBIOTICSTATE_GENOTYPE_TEMPERATURE.fastq
  <span class="pl-k">*</span> There are 2 sources: Virginia and Rhode Island
  <span class="pl-k">*</span> There are 2 symbiotic states: Brown and White
  
<span class="pl-c"><span class="pl-c">#</span>9. Next, you're going to start the process of adapter clipping and quality trimming all the renamed .fastq files in batches, by lane</span>

<span class="pl-c"><span class="pl-c">#</span>10. cp the script /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py into your scripts directory</span>

[kcrid001@turing1 fastq]$ <span class="pl-c1">cd</span> ../
[kcrid001@turing1 data]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@turing1 data]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py ../scripts/


<span class="pl-c"><span class="pl-c">#</span>11. Less/head the new script and check out the usage statement</span>
[kcrid001@turing1 data]$ head -14 ../scripts/Trimclipfilterstatsbatch_advbioinf.py
<span class="pl-c"><span class="pl-c">#!</span>/usr/bin/env python</span>
<span class="pl-c"><span class="pl-c">#</span> Written by Dan Barshis</span>

import sys, os

<span class="pl-c"><span class="pl-c">#</span>########## Usage #############</span>
<span class="pl-c"><span class="pl-c">#</span> Trimclipfilterstatsbatch.py barcodefile.txt anynumberoffastqfiles</span>
<span class="pl-c"><span class="pl-c">#</span> This should be run from within the folder with all of your original .fastqstrim</span>
<span class="pl-c"><span class="pl-c">#</span> Will quality trim multiple SINGLE-END fastq files</span>
<span class="pl-c"><span class="pl-c">#</span> Things to customize for your particular platform:</span>
<span class="pl-c"><span class="pl-c">#</span> qualoffset = 33 Quality score offset</span>
<span class="pl-c"><span class="pl-c">#</span> -t option in qualitytrim (the lower threshold quality score for trimming)</span>
<span class="pl-c"><span class="pl-c">#</span> -l option in qualitytrim and adapterclip (the length threshold for throwing out short reads)</span>


<span class="pl-c"><span class="pl-c">#</span>12. cp the /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt into the working directory with your fastq files</span>

[kcrid001@turing1 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq
[kcrid001@turing1 fastq]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03/adapterlist_advbioinf.txt ./
[kcrid001@turing1 fastq]$ ls
adapterlist_advbioinf.txt   RI_B_04_14.fastq  RI_W_04_14.fastq  VA_B_04_14.fastq  VA_W_04_14.fastq
KCridRenamer.sh             RI_B_04_18.fastq  RI_W_04_18.fastq  VA_B_04_18.fastq  VA_W_04_18.fastq
KCridRenamer.txt            RI_B_04_22.fastq  RI_W_04_22.fastq  VA_B_04_22.fastq  VA_W_04_22.fastq
renamingtable_complete.txt  RI_B_05_14.fastq  RI_W_05_14.fastq  VA_B_05_14.fastq  VA_W_05_14.fastq

<span class="pl-c"><span class="pl-c">#</span>13. Make a sbatch script for the Trimclipfilter... script and run it on your fastq files</span>
[kcrid001@turing1 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq
[kcrid001@turing1 fastq]$ nano TestTrimClip.sh
[kcrid001@turing1 fastq]$ cat TestTrimClip.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KCridtestTrimclip.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KCridTrimTest</span>

/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt <span class="pl-k">*</span>.fastq

[kcrid001@turing1 fastq]$ salloc
salloc: Pending job allocation 9271050
salloc: job 9271050 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9271050 has been allocated resources
salloc: Granted job allocation 9271050
[kcrid001@coreV1-22-005 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq
[kcrid001@coreV1-22-005 fastq]$ sbatch TestTrimClip.sh
Submitted batch job 9271051
[kcrid001@coreV1-22-005 fastq]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9271051      main KCridTri kcrid001  R      10:22      1 coreV2-22-007 
           9271050      main       sh kcrid001  R      11:22      1 coreV1-22-005 

<span class="pl-c"><span class="pl-c">#</span>I was confused on if I had to do the FullTrimClip next, because they look like the same thing, but I think the TestTrimClip is working (see list below)</span>

[kcrid001@coreV1-22-005 fastq]$ ls
adapterlist_advbioinf.txt  renamingtable_complete.txt  RI_W_04_14.fastq  trimclipstats.txt  VA_W_04_18.fastq
FullTrimClip.sh            RI_B_04_14_clipped.fastq    RI_W_04_18.fastq  VA_B_04_14.fastq   VA_W_04_22.fastq
KCridFullTrimclip.txt      RI_B_04_14.fastq            RI_W_04_22.fastq  VA_B_04_18.fastq   VA_W_05_14.fastq
KCridRenamer.sh            RI_B_04_18.fastq            RI_W_05_14.fastq  VA_B_04_22.fastq
KCridRenamer.txt           RI_B_04_22.fastq            Testing           VA_B_05_14.fastq
KCridtestTrimclip.txt      RI_B_05_14.fastq            TestTrimClip.sh   VA_W_04_14.fastq
</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day 04 HW: 1/29/21
                     </h2>
                     <a aria-label="Permalink: Day 04 HW: 1/29/21" class="anchor" href="#day-04-hw-12921" id="user-content-day-04-hw-12921">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#HW Day 04:

#1. Add your trimclipstats.txt output to the full class datafile /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt using the following steps
#1a. run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h to examine usage
[kcrid001@coreV1-22-005 fastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq
[kcrid001@coreV1-22-005 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h
Written by Peter Schafran pscha005@odu.edu 5-Oct-2015

This script takes a stats output file from fastx_clipper and converts it into a table.

Usage: Schafran_trimstatstable.py [-c, -v, -h] inputfile.txt outputfile.txt

Options (-c and -v must be listed separately to run together):
-h	Display this help message
-c	Use comma delimiter instead of tabs
-v	Verbose mode (print steps to stdout)

#1b. Run the script on your data with the outputfilename YOURNAME_trimclipstatsout.txt
[kcrid001@coreV2-22-007 fastq]$ cd filteringstats
[kcrid001@coreV2-22-007 filteringstats]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/filteringstats
[kcrid001@coreV2-22-007 filteringstats]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt KCrid_trimclipstatsout.txt

#1c. Add YOURNAME_trimclipstatsout.txt to the class file by running tail -n +2 YOURNAME_trimclipstatsout.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt

[kcrid001@coreV2-25-047 filteringstats]$ tail -n +2 KCrid_trimclipstatsout.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt


#2. Now we're going to map our reads back to our assembly using the bowtie2 alignment algorithm (starting to follow this pipeline https://github.com/bethsheets/SNPcalling_tutorial)
#3. write a sbatch script to do the following commands in sequence on your _clippedtrimmedfilterd.fastq datafiles from your lane of data
[kcrid001@coreV2-25-047 filteringstats]$ cd ../
[kcrid001@coreV2-25-047 fastq]$ cd QCFastqs/
[kcrid001@coreV2-25-047 QCFastqs]$ ls
RI_B_04_14_clippedtrimmed.fastq  RI_W_04_22_clippedtrimmed.fastq  VA_W_04_14_clippedtrimmed.fastq
RI_B_04_18_clippedtrimmed.fastq  RI_W_05_14_clippedtrimmed.fastq  VA_W_04_18_clippedtrimmed.fastq
RI_B_04_22_clippedtrimmed.fastq  VA_B_04_14_clippedtrimmed.fastq  VA_W_04_22_clippedtrimmed.fastq
RI_B_05_14_clippedtrimmed.fastq  VA_B_04_18_clippedtrimmed.fastq  VA_W_05_14_clippedtrimmed.fastq
RI_W_04_14_clippedtrimmed.fastq  VA_B_04_22_clippedtrimmed.fastq
RI_W_04_18_clippedtrimmed.fastq  VA_B_05_14_clippedtrimmed.fastq
[kcrid001@coreV2-25-047 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs
[kcrid001@coreV2-25-047 QCFastqs]$ nano bowtiealn.sh
[kcrid001@coreV2-25-047 QCFastqs]$ cat bowtiealn.sh 
#!/bin/bash -l

#SBATCH -o KCridbowtie2.txt
#SBATCH -n 1
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=newbowtie

module load bowtie2/2.4
for i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \
--rg SM:${i%_clippedtrimmed.fastq} \
--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U $i \
&gt; ${i%_clippedtrimmedfilterd.fastq}.sam --no-unal -k 5; done
[kcrid001@coreV2-25-047 QCFastqs]$ sbatch bowtiealn.sh 
Submitted batch job 9271852
[kcrid001@coreV2-25-047 QCFastqs]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9271852      main newbowti kcrid001  R       0:01      1 coreV2-25-005 
           9271848      main       sh kcrid001  R      18:01      1 coreV2-25-047 

#4-Submit and add everything to your logfile
*yep!

## Day05 HW 2/3/21
```sh
##Day05HW

1-Team up with a partner for this one and work only on a combined set of data for your two lanes of sequences

[kcrid001@coreV3-23-050 fastq]$ mkdir KCKCfastq
[kcrid001@coreV3-23-050 fastq]$ ls
filteringstats         KCridRenamer.txt            renamingtable_complete.txt
FullTrimClip.sh        KCridtestTrimclip.txt       Testing
KCKCfastq              KCrid_trimclipstatsout.txt  TestTrimClip.sh
KCridFullTrimclip.txt  originalfastqs
KCridRenamer.sh        QCFastqs
[kcrid001@coreV3-23-050 fastq]$ cd KCKCfastq/
[kcrid001@coreV3-23-050 KCKCfastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq

[kcrid001@coreV3-23-050 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs
[kcrid001@coreV3-23-050 QCFastqs]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs/*.fastq /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq
# we both copied our clippedtrimmed.fastq files to the new KCKCfastq directory (Katie C &amp; Kristina)
[kcrid001@coreV3-23-050 KCKCfastq]$ ls
KCKCTrinitydenovo.sh              VA_B_01_14_clippedtrimmed.fastq
RI_B_01_14_clippedtrimmed.fastq   VA_B_01_18_clippedtrimmed.fastq
RI_B_01_18_clippedtrimmed.fastq   VA_B_01_22_clippedtrimmed.fastq
RI_B_01_22_clippedtrimmed.fastq   VA_B_04_14_clippedtrimmed.fastq
RI_B_04_14_clippedtrimmed.fastq   VA_B_04_18_clippedtrimmed.fastq
RI_B_04_18_clippedtrimmed.fastq   VA_B_04_22_clippedtrimmed.fastq
RI_B_04_22_clippedtrimmed.fastq   VA_B_05_14_clippedtrimmed.fastq
RI_B_05_14_clippedtrimmed.fastq   VA_B_09_SNP_clippedtrimmed.fastq
RI_B_08_SNP_clippedtrimmed.fastq  VA_W_01_14_clippedtrimmed.fastq
RI_W_01_14_clippedtrimmed.fastq   VA_W_01_18_clippedtrimmed.fastq
RI_W_01_18_clippedtrimmed.fastq   VA_W_01_22_clippedtrimmed.fastq
RI_W_01_22_clippedtrimmed.fastq   VA_W_04_14_clippedtrimmed.fastq
RI_W_04_14_clippedtrimmed.fastq   VA_W_04_18_clippedtrimmed.fastq
RI_W_04_18_clippedtrimmed.fastq   VA_W_04_22_clippedtrimmed.fastq
RI_W_04_22_clippedtrimmed.fastq   VA_W_05_14_clippedtrimmed.fastq
RI_W_05_14_clippedtrimmed.fastq   VA_W_08_SNP_clippedtrimmed.fastq
RI_W_08_SNP_clippedtrimmed.fastq
# we have 32 files, 16 for each of us. All files have copied

2-Run the Trinity denovo assembler on your clippedtrimmed.fastq files for your two lanes together
3-Modify the below sbatch script (note the differences in the header compared to the previous one)

[kcrid001@coreV3-23-050 KCKCfastq]$ nano KCKCTrinitydenovo.sh
[kcrid001@coreV3-23-050 KCKCfastq]$ cat KCKCTrinitydenovo.sh 
#!/bin/bash -l

#SBATCH -o KCKCTrinitydenovo.txt
#SBATCH -n 32
#SBATCH -p himem
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KCKCTrinitydenovo

enable_lmod
module load container_env trinity

crun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_04_14_clippedtrimmed.fastq,VA_B_04_14_clippedtrimmed.fastq,RI_B_04_18_clippedtrimmed.fastq,VA_B_04_18_clippedtrimmed.fastq,RI_B_04_22_clippedtrimmed.fastq,VA_B_04_22_clippedtrimmed.fastq,RI_B_05_14_clippedtrimmed.fastq,VA_B_05_14_clippedtrimmed.fastq,RI_W_04_14_clippedtrimmed.fastq,VA_W_04_14_clippedtrimmed.fastq,RI_W_04_18_clippedtrimmed.fastq,VA_W_04_18_clippedtrimmed.fastq,RI_W_04_22_clippedtrimmed.fastq,VA_W_04_22_clippedtrimmed.fastq,RI_W_05_14_clippedtrimmed.fastq,VA_W_05_14_clippedtrimmed.fastq,RI_B_01_14_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq --CPU 32

4-Check https://trinityrnaseq.github.io/ for usage info

5-Submit your trinity script
[kcrid001@coreV3-23-050 KCKCfastq]$ sbatch KCKCTrinitydenovo.sh
Submitted batch job 9272362
[kcrid001@coreV3-23-050 KCKCfastq]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>HW Day 04:</span>

<span class="pl-c"><span class="pl-c">#</span>1. Add your trimclipstats.txt output to the full class datafile /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt using the following steps</span>
<span class="pl-c"><span class="pl-c">#</span>1a. run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h to examine usage</span>
[kcrid001@coreV1-22-005 fastq]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq
[kcrid001@coreV1-22-005 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h
Written by Peter Schafran pscha005@odu.edu 5-Oct-2015

This script takes a stats output file from fastx_clipper and converts it into a table.

Usage: Schafran_trimstatstable.py [-c, -v, <span class="pl-k">-h</span>] inputfile.txt outputfile.txt

Options (-c and -v must be listed separately to run together):
-h	Display this <span class="pl-c1">help</span> message
-c	Use comma delimiter instead of tabs
-v	Verbose mode (print steps to stdout)

<span class="pl-c"><span class="pl-c">#</span>1b. Run the script on your data with the outputfilename YOURNAME_trimclipstatsout.txt</span>
[kcrid001@coreV2-22-007 fastq]$ <span class="pl-c1">cd</span> filteringstats
[kcrid001@coreV2-22-007 filteringstats]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/filteringstats
[kcrid001@coreV2-22-007 filteringstats]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt KCrid_trimclipstatsout.txt

<span class="pl-c"><span class="pl-c">#</span>1c. Add YOURNAME_trimclipstatsout.txt to the class file by running tail -n +2 YOURNAME_trimclipstatsout.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt</span>

[kcrid001@coreV2-25-047 filteringstats]$ tail -n +2 KCrid_trimclipstatsout.txt <span class="pl-k">&gt;&gt;</span> /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt


<span class="pl-c"><span class="pl-c">#</span>2. Now we're going to map our reads back to our assembly using the bowtie2 alignment algorithm (starting to follow this pipeline https://github.com/bethsheets/SNPcalling_tutorial)</span>
<span class="pl-c"><span class="pl-c">#</span>3. write a sbatch script to do the following commands in sequence on your _clippedtrimmedfilterd.fastq datafiles from your lane of data</span>
[kcrid001@coreV2-25-047 filteringstats]$ <span class="pl-c1">cd</span> ../
[kcrid001@coreV2-25-047 fastq]$ <span class="pl-c1">cd</span> QCFastqs/
[kcrid001@coreV2-25-047 QCFastqs]$ ls
RI_B_04_14_clippedtrimmed.fastq  RI_W_04_22_clippedtrimmed.fastq  VA_W_04_14_clippedtrimmed.fastq
RI_B_04_18_clippedtrimmed.fastq  RI_W_05_14_clippedtrimmed.fastq  VA_W_04_18_clippedtrimmed.fastq
RI_B_04_22_clippedtrimmed.fastq  VA_B_04_14_clippedtrimmed.fastq  VA_W_04_22_clippedtrimmed.fastq
RI_B_05_14_clippedtrimmed.fastq  VA_B_04_18_clippedtrimmed.fastq  VA_W_05_14_clippedtrimmed.fastq
RI_W_04_14_clippedtrimmed.fastq  VA_B_04_22_clippedtrimmed.fastq
RI_W_04_18_clippedtrimmed.fastq  VA_B_05_14_clippedtrimmed.fastq
[kcrid001@coreV2-25-047 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs
[kcrid001@coreV2-25-047 QCFastqs]$ nano bowtiealn.sh
[kcrid001@coreV2-25-047 QCFastqs]$ cat bowtiealn.sh 
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KCridbowtie2.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=newbowtie</span>

module load bowtie2/2.4
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-k">*</span>_clippedtrimmed.fastq<span class="pl-k">;</span> <span class="pl-k">do</span> bowtie2 --rg-id <span class="pl-smi">${i<span class="pl-k">%</span>_clippedtrimmed.fastq}</span> \
--rg SM:<span class="pl-smi">${i<span class="pl-k">%</span>_clippedtrimmed.fastq}</span> \
--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U <span class="pl-smi">$i</span> \
<span class="pl-k">&gt;</span> <span class="pl-smi">${i<span class="pl-k">%</span>_clippedtrimmedfilterd.fastq}</span>.sam --no-unal -k 5<span class="pl-k">;</span> <span class="pl-k">done</span>
[kcrid001@coreV2-25-047 QCFastqs]$ sbatch bowtiealn.sh 
Submitted batch job 9271852
[kcrid001@coreV2-25-047 QCFastqs]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9271852      main newbowti kcrid001  R       0:01      1 coreV2-25-005 
           9271848      main       sh kcrid001  R      18:01      1 coreV2-25-047 

<span class="pl-c"><span class="pl-c">#</span>4-Submit and add everything to your logfile</span>
<span class="pl-k">*</span>yep<span class="pl-k">!</span>

<span class="pl-c"><span class="pl-c">#</span># Day05 HW 2/3/21</span>
<span class="pl-s"><span class="pl-pds">`</span><span class="pl-pds">`</span><span class="pl-pds">`</span>sh</span>
<span class="pl-s"><span class="pl-c"><span class="pl-c">#</span>#Day05HW</span></span>
<span class="pl-s"></span>
<span class="pl-s">1-Team up with a partner <span class="pl-k">for</span> this one and work only on a combined <span class="pl-c1">set</span> of data <span class="pl-k">for</span> your two lanes of sequences</span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV3-23-050 fastq]$ mkdir KCKCfastq</span>
<span class="pl-s">[kcrid001@coreV3-23-050 fastq]$ ls</span>
<span class="pl-s">filteringstats         KCridRenamer.txt            renamingtable_complete.txt</span>
<span class="pl-s">FullTrimClip.sh        KCridtestTrimclip.txt       Testing</span>
<span class="pl-s">KCKCfastq              KCrid_trimclipstatsout.txt  TestTrimClip.sh</span>
<span class="pl-s">KCridFullTrimclip.txt  originalfastqs</span>
<span class="pl-s">KCridRenamer.sh        QCFastqs</span>
<span class="pl-s">[kcrid001@coreV3-23-050 fastq]$ <span class="pl-c1">cd</span> KCKCfastq/</span>
<span class="pl-s">[kcrid001@coreV3-23-050 KCKCfastq]$ <span class="pl-c1">pwd</span></span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq</span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV3-23-050 QCFastqs]$ <span class="pl-c1">pwd</span></span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs</span>
<span class="pl-s">[kcrid001@coreV3-23-050 QCFastqs]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs/<span class="pl-k">*</span>.fastq /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq</span>
<span class="pl-s"><span class="pl-c"><span class="pl-c">#</span> we both copied our clippedtrimmed.fastq files to the new KCKCfastq directory (Katie C &amp; Kristina)</span></span>
<span class="pl-s">[kcrid001@coreV3-23-050 KCKCfastq]$ ls</span>
<span class="pl-s">KCKCTrinitydenovo.sh              VA_B_01_14_clippedtrimmed.fastq</span>
<span class="pl-s">RI_B_01_14_clippedtrimmed.fastq   VA_B_01_18_clippedtrimmed.fastq</span>
<span class="pl-s">RI_B_01_18_clippedtrimmed.fastq   VA_B_01_22_clippedtrimmed.fastq</span>
<span class="pl-s">RI_B_01_22_clippedtrimmed.fastq   VA_B_04_14_clippedtrimmed.fastq</span>
<span class="pl-s">RI_B_04_14_clippedtrimmed.fastq   VA_B_04_18_clippedtrimmed.fastq</span>
<span class="pl-s">RI_B_04_18_clippedtrimmed.fastq   VA_B_04_22_clippedtrimmed.fastq</span>
<span class="pl-s">RI_B_04_22_clippedtrimmed.fastq   VA_B_05_14_clippedtrimmed.fastq</span>
<span class="pl-s">RI_B_05_14_clippedtrimmed.fastq   VA_B_09_SNP_clippedtrimmed.fastq</span>
<span class="pl-s">RI_B_08_SNP_clippedtrimmed.fastq  VA_W_01_14_clippedtrimmed.fastq</span>
<span class="pl-s">RI_W_01_14_clippedtrimmed.fastq   VA_W_01_18_clippedtrimmed.fastq</span>
<span class="pl-s">RI_W_01_18_clippedtrimmed.fastq   VA_W_01_22_clippedtrimmed.fastq</span>
<span class="pl-s">RI_W_01_22_clippedtrimmed.fastq   VA_W_04_14_clippedtrimmed.fastq</span>
<span class="pl-s">RI_W_04_14_clippedtrimmed.fastq   VA_W_04_18_clippedtrimmed.fastq</span>
<span class="pl-s">RI_W_04_18_clippedtrimmed.fastq   VA_W_04_22_clippedtrimmed.fastq</span>
<span class="pl-s">RI_W_04_22_clippedtrimmed.fastq   VA_W_05_14_clippedtrimmed.fastq</span>
<span class="pl-s">RI_W_05_14_clippedtrimmed.fastq   VA_W_08_SNP_clippedtrimmed.fastq</span>
<span class="pl-s">RI_W_08_SNP_clippedtrimmed.fastq</span>
<span class="pl-s"><span class="pl-c"><span class="pl-c">#</span> we have 32 files, 16 for each of us. All files have copied</span></span>
<span class="pl-s"></span>
<span class="pl-s">2-Run the Trinity denovo assembler on your clippedtrimmed.fastq files <span class="pl-k">for</span> your two lanes together</span>
<span class="pl-s">3-Modify the below sbatch script (note the differences <span class="pl-k">in</span> the header compared to the previous one)</span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV3-23-050 KCKCfastq]$ nano KCKCTrinitydenovo.sh</span>
<span class="pl-s">[kcrid001@coreV3-23-050 KCKCfastq]$ cat KCKCTrinitydenovo.sh </span>
<span class="pl-s"><span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span></span>
<span class="pl-s"></span>
<span class="pl-s"><span class="pl-c"><span class="pl-c">#</span>SBATCH -o KCKCTrinitydenovo.txt</span></span>
<span class="pl-s"><span class="pl-c"><span class="pl-c">#</span>SBATCH -n 32</span></span>
<span class="pl-s"><span class="pl-c"><span class="pl-c">#</span>SBATCH -p himem</span></span>
<span class="pl-s"><span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span></span>
<span class="pl-s"><span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span></span>
<span class="pl-s"><span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KCKCTrinitydenovo</span></span>
<span class="pl-s"></span>
<span class="pl-s">enable_lmod</span>
<span class="pl-s">module load container_env trinity</span>
<span class="pl-s"></span>
<span class="pl-s">crun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_04_14_clippedtrimmed.fastq,VA_B_04_14_clippedtrimmed.fastq,RI_B_04_18_clippedtrimmed.fastq,VA_B_04_18_clippedtrimmed.fastq,RI_B_04_22_clippedtrimmed.fastq,VA_B_04_22_clippedtrimmed.fastq,RI_B_05_14_clippedtrimmed.fastq,VA_B_05_14_clippedtrimmed.fastq,RI_W_04_14_clippedtrimmed.fastq,VA_W_04_14_clippedtrimmed.fastq,RI_W_04_18_clippedtrimmed.fastq,VA_W_04_18_clippedtrimmed.fastq,RI_W_04_22_clippedtrimmed.fastq,VA_W_04_22_clippedtrimmed.fastq,RI_W_05_14_clippedtrimmed.fastq,VA_W_05_14_clippedtrimmed.fastq,RI_B_01_14_clippedtrimmed.fastq,RI_B_01_18_clippedtrimmed.fastq,RI_B_01_22_clippedtrimmed.fastq,RI_B_08_SNP_clippedtrimmed.fastq,RI_W_01_14_clippedtrimmed.fastq,RI_W_01_18_clippedtrimmed.fastq,RI_W_01_22_clippedtrimmed.fastq,RI_W_08_SNP_clippedtrimmed.fastq,VA_B_01_14_clippedtrimmed.fastq,VA_B_01_18_clippedtrimmed.fastq,VA_B_01_22_clippedtrimmed.fastq,VA_B_09_SNP_clippedtrimmed.fastq,VA_W_01_14_clippedtrimmed.fastq,VA_W_01_18_clippedtrimmed.fastq,VA_W_01_22_clippedtrimmed.fastq,VA_W_08_SNP_clippedtrimmed.fastq --CPU 32</span>
<span class="pl-s"></span>
<span class="pl-s">4-Check https://trinityrnaseq.github.io/ <span class="pl-k">for</span> usage info</span>
<span class="pl-s"></span>
<span class="pl-s">5-Submit your trinity script</span>
<span class="pl-s">[kcrid001@coreV3-23-050 KCKCfastq]$ sbatch KCKCTrinitydenovo.sh</span>
<span class="pl-s">Submitted batch job 9272362</span>
<span class="pl-s">[kcrid001@coreV3-23-050 KCKCfastq]$ <span class="pl-c1">pwd</span></span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq</span>
<span class="pl-s"></span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day06 HW
                     </h2>
                     <a aria-label="Permalink: Day06 HW" class="anchor" href="#day06-hw" id="user-content-day06-hw">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#day06HW
####Make sure to add all this to your logfile as you go, including the output from the various scripts/greps/etc.
###Assembly quality assessment (following the first two steps in the recommended trinity assessment protocol from https://github.com/trinityrnaseq/trinityrnaseq/wiki/Transcriptome-Assembly-Quality-Assessment)


#1- start an interactive session via salloc and run the /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl script on your Trinity.fasta output from your assembly

[kcrid001@turing1 trinity_out_dir]$ salloc
salloc: Pending job allocation 9272857
salloc: job 9272857 queued and waiting for resources
salloc: job 9272857 has been allocated resources
salloc: Granted job allocation 9272857
[kcrid001@coreV2-22-028 trinity_out_dir]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir

[kcrid001@coreV2-22-028 trinity_out_dir]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta


################################
## Counts of transcripts, etc.
################################
Total trinity 'genes':	39219
Total trinity transcripts:	41303
Percent GC: 44.54

########################################
Stats based on ALL transcript contigs:
########################################

	Contig N10: 1323
	Contig N20: 783
	Contig N30: 562
	Contig N40: 440
	Contig N50: 364

	Median contig length: 277
	Average contig: 371.08
	Total assembled bases: 15326666


#####################################################
## Stats based on ONLY LONGEST ISOFORM per 'GENE':
#####################################################

	Contig N10: 1094
	Contig N20: 682
	Contig N30: 511
	Contig N40: 410
	Contig N50: 347

	Median contig length: 274
	Average contig: 355.69
	Total assembled bases: 13949619


#2- compare this with the output from avg_cov_len_fasta_advbioinf.py on our class reference assembly (/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta) and add both to your logfile

[kcrid001@coreV2-22-028 trinity_out_dir]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py Trinity.fasta
The total number of sequences is 41303
The average sequence length is 371
The total number of bases is 15326666
The minimum sequence length is 184
The maximum sequence length is 10771
The N50 is 364
Median Length = 627
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 5936
contigs &gt;= 1000bp = 1364
contigs &gt;= 2000bp = 265


#3- less or head your bowtie2 job output file to look at your alignment statistics and calculate the following from the information:
	a-the mean percent &quot;overall alignment rate&quot;
	b-the mean percent reads &quot;aligned exactly 1 time&quot;
	c-the mean number of reads &quot;aligned exactly 1 time&quot;
	d-the mean percent reads &quot;aligned &gt;1 times&quot;
	
	hint use grep and paste into excel

[kcrid001@coreV2-22-028 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs
[kcrid001@coreV2-22-028 QCFastqs]$ less KCridbowtie2.txt 
    366410 (2.11%) aligned exactly 1 time
    196159 (1.13%) aligned &gt;1 times
3.25% overall alignment rate
28980402 reads; of these:
  28980402 (100.00%) were unpaired; of these:
    28428026 (98.09%) aligned 0 times
    302622 (1.04%) aligned exactly 1 time
    249754 (0.86%) aligned &gt;1 times
1.91% overall alignment rate
22997803 reads; of these:
  22997803 (100.00%) were unpaired; of these:
    22542136 (98.02%) aligned 0 times
    281735 (1.23%) aligned exactly 1 time
    173932 (0.76%) aligned &gt;1 times
1.98% overall alignment rate
18509735 reads; of these:
  18509735 (100.00%) were unpaired; of these:
    18027510 (97.39%) aligned 0 times
    262310 (1.42%) aligned exactly 1 time
    219915 (1.19%) aligned &gt;1 times
2.61% overall alignment rate
15268908 reads; of these:
  15268908 (100.00%) were unpaired; of these:
    14979814 (98.11%) aligned 0 times
    178865 (1.17%) aligned exactly 1 time
    110229 (0.72%) aligned &gt;1 times
1.89% overall alignment rate


kcrid001@coreV2-22-028 QCFastqs]$ grep &quot;overall alignment rate&quot; KCridbowtie2.txt | cut -d &quot; &quot; -f 1 | cut -d &quot;%&quot; -f 1
2.48
1.70
1.45
2.32
6.81
2.25
2.12
1.85
2.00
2.70
2.36
3.25
1.91
1.98
2.61
1.89
[kcrid001@coreV2-22-028 QCFastqs]$ grep &quot;aligned exactly 1 time&quot; KCridbowtie2.txt | cut -d &quot;(&quot; -f 2 | cut -d &quot;%&quot; -f 1
1.41
1.12
0.85
1.66
1.18
1.26
1.06
1.10
1.10
1.51
1.53
2.11
1.04
1.23
1.42
1.17
[kcrid001@coreV2-22-028 QCFastqs]$ grep &quot;aligned exactly 1 time&quot; KCridbowtie2.txt | cut -d &quot; &quot; -f 5
370715
569956
82036
677289
38461
303192
11115
409225
189559
281489
300187
366410
302622
281735
262310
178865
[kcrid001@coreV2-22-028 QCFastqs]$ grep &quot;aligned &gt;1 times&quot; KCridbowtie2.txt | cut -d &quot;(&quot; -f 2 | cut -d &quot;%&quot; -f 1
1.07
0.58
0.60
0.67
5.63
1.00
1.06
0.74
0.90
1.19
0.83
1.13
0.86
0.76
1.19
0.72

# copy these values to excel, calculate averages as needed
# answers
	# a - 2.48% - the mean percent &quot;overall alignment rate&quot;
	# b - 1.30% - the mean percent reads &quot;aligned exactly 1 time&quot;
	# c - 289072.88 - the mean number of reads &quot;aligned exactly 1 time&quot;
	# d - 1.18% - the mean percent reads &quot;aligned &gt;1 times&quot;


#4- add your statistics as single rows to the shared table /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt as tab-delimited text in the following order:
LaneX_yourinitials	b-the mean percent &quot;overall alignment rate&quot;	c-the mean percent reads &quot;aligned exactly 1 time&quot;	d-the mean number of reads &quot;aligned exactly 1 time&quot;	e-the mean percent reads &quot;aligned &gt;1 times&quot;

[kcrid001@coreV2-22-028 QCFastqs]$ nano alignstats.txt
[kcrid001@coreV2-22-028 QCFastqs]$ cat alignstats.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt
[kcrid001@coreV2-22-028 QCFastqs]$ cat /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt
LaneX	overallalignmentrate	percsinglyaligned	numsinglyaligned	percmultiplyaligned
LaneX_djb	2.495714286	1.251428571	247656.7143	1.247142857
LaneX_djb	2.495714286	1.251428571	247656.7143	1.247142857
Lane1AP	5.378125	1.3725	300361.8125	4.005
lane5_sg	2.03%	1.11%	235455.75	0.92%
LaneX_jcw	2.495714286	1.251428571	247656.7143	1.247142857
Lane4_KCrid	2.48	1.30	289072.88	1.18

#5- Data cleanup and archiving:
*a-mv your Trinity.fasta file from your trinity_out_dir to a folder called testassembly in your data directory
	
[kcrid001@turing1 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@turing1 data]$ mkdir testassembly
[kcrid001@turing1 data]$ ls
exercises  KCridCopyLane04.sh   KCridGunzipLane04.sh   testassembly
fastq      KCridCopyLane04.txt  KCridGunzipLane04.txt
[kcrid001@turing1 trinity_out_dir]$ salloc
salloc: Pending job allocation 9275008
salloc: job 9275008 queued and waiting for resources
salloc: job 9275008 has been allocated resources
salloc: Granted job allocation 9275008
[kcrid001@coreV1-22-016 trinity_out_dir]$ mv /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir/Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly

*b-set up a sbatch script to:
		rm -r YOURtrinity_out_dir
		rm -r YOURoriginalfastqs
		rm -r YOURfilteringstats # as long as you've already appended your filteringstats output to the class table as part of homework_day04.txt

[kcrid001@coreV1-22-016 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@coreV1-22-016 data]$ nano Cleanup.txt
[kcrid001@coreV1-22-016 data]$ cat Cleanup.txt 
#!/bin/bash -l

#SBATCH -o Cleanup.txt
#SBATCH -n 1
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=Cleanup

rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/originalfastqs
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/filteringstats

[kcrid001@coreV1-22-016 data]$ sbatch Cleanup.sh 
Submitted batch job 9275062
[kcrid001@coreV1-22-016 data]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9275062      main  Cleanup kcrid001  R       0:07      1 coreV1-22-016 
           9275008      main       sh kcrid001  R      22:37      1 coreV1-22-016 

#6- submit a blast against sprot from your testassembly folder using the following command

[kcrid001@coreV1-22-016 testassembly]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly
[kcrid001@coreV1-22-016 testassembly]$ ls
Trinity.fasta
[kcrid001@coreV1-22-016 testassembly]$ nano Blast.sh
[kcrid001@coreV1-22-016 testassembly]$ cat Blast.sh 
#!/bin/bash -l

#SBATCH -o Blast.txt
#SBATCH -n 6         
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=Blast

enable_lmod
module load container_env blast
blastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \
        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6
[kcrid001@coreV1-22-016 testassembly]$ sbatch Blast.sh 
Submitted batch job 9275071
[kcrid001@coreV1-22-016 testassembly]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9275071      main    Blast kcrid001  R       1:53      1 coreV3-23-027 
           9275062      main  Cleanup kcrid001  R       6:23      1 coreV1-22-016 
           9275008      main       sh kcrid001  R      28:53      1 coreV1-22-016 

###Exploring our SAM files, check out http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#sam-output for bowtie2 specific output and http://bio-bwa.sourceforge.net/bwa.shtml#4 for general SAM output

#7- head one of your .sam files to look at the header

[kcrid001@coreV2-22-028 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs
[kcrid001@coreV2-22-028 QCFastqs]$ head RI_B_04_14_clippedtrimmed.fastq.sam
@HD	VN:1.0	SO:unsorted
@SQ	SN:TR78|c0_g1_i1_coral	LN:1109
@SQ	SN:TR78|c0_g2_i1_coral	LN:1109
@SQ	SN:TR79|c0_g1_i1_coral	LN:610
@SQ	SN:TR79|c0_g2_i1_coral	LN:1549
@SQ	SN:TR87|c0_g1_i1_coral	LN:732
@SQ	SN:TR93|c0_g1_i1_coral	LN:550
@SQ	SN:TR101|c0_g1_i1_coral	LN:673
@SQ	SN:TR104|c0_g1_i1_coral	LN:607
@SQ	SN:TR105|c0_g1_i1_coral	LN:587


#8- grep -v '@' your.sam | head to look at the sequence read lines, why does this work to exclude the header lines?
	#because the -v flag means to omit lines with the indicated symbol

[kcrid001@coreV2-22-028 QCFastqs]$ grep -v '@' RI_B_04_14_clippedtrimmed.fastq.sam | head
K00188:59:HMTFHBBXX:5:1101:6055:1648	16	TR13320|c0_g2_i1_coral	2	253M4I44M	*	0	0	TTGAATCATTCTTGTCTACTCCAAGGGGTCTACTTTCCTCAAGGTTTCCNJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJJJJJFFA#A	AS:i:-30	XN:i:0	XM:i:3	XO:i:1	XG:i:4	NM:i:7	MD:Z:0A1C42A1	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:9597:1666	0	TR35685|c1_g2_i1_coral	535	2520M	*	0	0	GACAACCCCGTCTGTGGCGC	AAFFFJJJJJJJJJJJJJJJ	AS:i:-6	XN:i:0	XM:i:1	XO:i:0	XG:i:0	NM:i:1	MD:Z:6A13	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:26281:1666	16	TR59703|c0_g1_i3_sym	244	1	51M	*	0	0	CTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:26281:1666	272	TR59703|c0_g1_i1_sym	277	2551M	*	0	0	CTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:26281:1666	272	TR59703|c0_g1_i3_coral	244	2551M	*	0	0	CTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:26281:1666	272	TR59703|c0_g1_i1_coral	277	2551M	*	0	0	CTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:9557:1701	0	TR30871|c0_g1_i1_coral	16	253M3I45M	*	0	0	GTGATAATGATGTTGGTGATAATGATGATGATGGTGGTGGTTGTGGTGTTAAFFFJJJJJFFJJJJJFJFJJAFJJJJJJJJJJFJJAJJJJJFJJFJFFJ	AS:i:-26	XN:i:0	XM:i:2	XO:i:1	XG:i:3	NM:i:5	MD:Z:6G5A35	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:28270:1701	16	TR5604|c0_g1_i1_coral	515	2551M	*	0	0	CTAAGGTTTGCTTTAAAGCACTTAGTTCAGAATCCTTTGTCTGTACAATTJJJJJJFJJJJJJJJJJJJJJJJJJJJF-JJJFJFJJJJJJJJJJJFFFAA	AS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:28371:1701	16	TR59703|c0_g1_i3_sym	244	1	51M	*	0	0	CTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTJJJ7JJJJJJFJJJJJJJJJJJJJJJFAJJJJJJJJAJJJJJAJJJFFFAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:28371:1701	272	TR59703|c0_g1_i1_sym	277	2551M	*	0	0	CTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTJJJ7JJJJJJFJJJJJJJJJJJJJJJFAJJJJJJJJAJJJJJAJJJFFFAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_04_14


#9- in an interactive session run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py on 2-3 of your .sam files using * to select 2-3 at the same time.

[kcrid001@coreV1-22-016 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs
[kcrid001@coreV1-22-016 QCFastqs]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py RI_B_04_14_clippedtrimmed.fastq.sam RI_B_04_18_clippedtrimmed.fastq.sam RI_B_04_22_clippedtrimmed.fastq.sam
RI_B_04_14_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
RI_B_04_18_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand
RI_B_04_22_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
	read reverse strand
	not primary alignment
256 :
	not primary alignment
16 :
	read reverse strand


#10- we need to run the read sorting step required for SNP calling, so if you have time, set up and run the following script on your .sam files to finish before Wednesday:

#!/bin/bash -l

#SBATCH -o KCridBamandSort.txt
#SBATCH -n 1         
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KCridBamandSort

enable_lmod
module load samtools/1
for i in *.sam; do `samtools view -bS $i &gt; ${i%.sam}_UNSORTED.bam`; done

for i in *UNSORTED.bam; do samtools sort $i &gt; ${i%_UNSORTED.bam}.bam
samtools index ${i%_UNSORTED.bam}.bam
done

[kcrid001@coreV1-22-016 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs
[kcrid001@coreV1-22-016 QCFastqs]$ cat bamandsort.sh
#!/bin/bash -l

#SBATCH -o KCridBamandSort.txt
#SBATCH -n 1         
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KCridBamandSort

enable_lmod
module load samtools/1
for i in *.sam; do `samtools view -bS $i &gt; ${i%.sam}_UNSORTED.bam`; done

for i in *UNSORTED.bam; do samtools sort $i &gt; ${i%_UNSORTED.bam}.bam
samtools index ${i%_UNSORTED.bam}.bam
done

[kcrid001@coreV1-22-016 QCFastqs]$ sbatch bamandsort.sh
Submitted batch job 9275211
[kcrid001@coreV1-22-016 QCFastqs]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9275071      main    Blast kcrid001  R      16:38      1 coreV3-23-027 
           9275062      main  Cleanup kcrid001  R      21:08      1 coreV1-22-016 
           9275008      main       sh kcrid001  R      43:38      1 coreV1-22-016 
           9275211      main KCridBam kcrid001  R       0:08      1 coreV2-22-007 
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>day06HW</span>
<span class="pl-c"><span class="pl-c">#</span>###Make sure to add all this to your logfile as you go, including the output from the various scripts/greps/etc.</span>
<span class="pl-c"><span class="pl-c">#</span>##Assembly quality assessment (following the first two steps in the recommended trinity assessment protocol from https://github.com/trinityrnaseq/trinityrnaseq/wiki/Transcriptome-Assembly-Quality-Assessment)</span>


<span class="pl-c"><span class="pl-c">#</span>1- start an interactive session via salloc and run the /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl script on your Trinity.fasta output from your assembly</span>

[kcrid001@turing1 trinity_out_dir]$ salloc
salloc: Pending job allocation 9272857
salloc: job 9272857 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9272857 has been allocated resources
salloc: Granted job allocation 9272857
[kcrid001@coreV2-22-028 trinity_out_dir]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir

[kcrid001@coreV2-22-028 trinity_out_dir]$ /cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta


<span class="pl-c"><span class="pl-c">#</span>###############################</span>
<span class="pl-c"><span class="pl-c">#</span># Counts of transcripts, etc.</span>
<span class="pl-c"><span class="pl-c">#</span>###############################</span>
Total trinity <span class="pl-s"><span class="pl-pds">'</span>genes<span class="pl-pds">'</span></span>:	39219
Total trinity transcripts:	41303
Percent GC: 44.54

<span class="pl-c"><span class="pl-c">#</span>#######################################</span>
Stats based on ALL transcript contigs:
<span class="pl-c"><span class="pl-c">#</span>#######################################</span>

	Contig N10: 1323
	Contig N20: 783
	Contig N30: 562
	Contig N40: 440
	Contig N50: 364

	Median contig length: 277
	Average contig: 371.08
	Total assembled bases: 15326666


<span class="pl-c"><span class="pl-c">#</span>####################################################</span>
<span class="pl-c"><span class="pl-c">#</span># Stats based on ONLY LONGEST ISOFORM per 'GENE':</span>
<span class="pl-c"><span class="pl-c">#</span>####################################################</span>

	Contig N10: 1094
	Contig N20: 682
	Contig N30: 511
	Contig N40: 410
	Contig N50: 347

	Median contig length: 274
	Average contig: 355.69
	Total assembled bases: 13949619


<span class="pl-c"><span class="pl-c">#</span>2- compare this with the output from avg_cov_len_fasta_advbioinf.py on our class reference assembly (/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta) and add both to your logfile</span>

[kcrid001@coreV2-22-028 trinity_out_dir]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py Trinity.fasta
The total number of sequences is 41303
The average sequence length is 371
The total number of bases is 15326666
The minimum sequence length is 184
The maximum sequence length is 10771
The N50 is 364
Median Length = 627
contigs <span class="pl-k">&lt;</span> 150bp = 0
contigs <span class="pl-k">&gt;</span>= 500bp = 5936
contigs <span class="pl-k">&gt;</span>= 1000bp = 1364
contigs <span class="pl-k">&gt;</span>= 2000bp = 265


<span class="pl-c"><span class="pl-c">#</span>3- less or head your bowtie2 job output file to look at your alignment statistics and calculate the following from the information:</span>
	a-the mean percent <span class="pl-s"><span class="pl-pds">"</span>overall alignment rate<span class="pl-pds">"</span></span>
	b-the mean percent reads <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span>
	c-the mean number of reads <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span>
	d-the mean percent reads <span class="pl-s"><span class="pl-pds">"</span>aligned &gt;1 times<span class="pl-pds">"</span></span>
	
	hint use grep and paste into excel

[kcrid001@coreV2-22-028 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs
[kcrid001@coreV2-22-028 QCFastqs]$ less KCridbowtie2.txt 
    366410 (2.11%) aligned exactly 1 <span class="pl-k">time</span>
    196159 (1.13%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
3.25% overall alignment rate
28980402 reads<span class="pl-k">;</span> of these:
  28980402 (100.00%) were unpaired<span class="pl-k">;</span> of these:
    28428026 (98.09%) aligned 0 <span class="pl-c1">times</span>
    302622 (1.04%) aligned exactly 1 <span class="pl-k">time</span>
    249754 (0.86%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
1.91% overall alignment rate
22997803 reads<span class="pl-k">;</span> of these:
  22997803 (100.00%) were unpaired<span class="pl-k">;</span> of these:
    22542136 (98.02%) aligned 0 <span class="pl-c1">times</span>
    281735 (1.23%) aligned exactly 1 <span class="pl-k">time</span>
    173932 (0.76%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
1.98% overall alignment rate
18509735 reads<span class="pl-k">;</span> of these:
  18509735 (100.00%) were unpaired<span class="pl-k">;</span> of these:
    18027510 (97.39%) aligned 0 <span class="pl-c1">times</span>
    262310 (1.42%) aligned exactly 1 <span class="pl-k">time</span>
    219915 (1.19%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
2.61% overall alignment rate
15268908 reads<span class="pl-k">;</span> of these:
  15268908 (100.00%) were unpaired<span class="pl-k">;</span> of these:
    14979814 (98.11%) aligned 0 <span class="pl-c1">times</span>
    178865 (1.17%) aligned exactly 1 <span class="pl-k">time</span>
    110229 (0.72%) aligned <span class="pl-k">&gt;</span>1 <span class="pl-c1">times</span>
1.89% overall alignment rate


kcrid001@coreV2-22-028 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>overall alignment rate<span class="pl-pds">"</span></span> KCridbowtie2.txt <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span> -f 1 <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>%<span class="pl-pds">"</span></span> -f 1
2.48
1.70
1.45
2.32
6.81
2.25
2.12
1.85
2.00
2.70
2.36
3.25
1.91
1.98
2.61
1.89
[kcrid001@coreV2-22-028 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span> KCridbowtie2.txt <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>(<span class="pl-pds">"</span></span> -f 2 <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>%<span class="pl-pds">"</span></span> -f 1
1.41
1.12
0.85
1.66
1.18
1.26
1.06
1.10
1.10
1.51
1.53
2.11
1.04
1.23
1.42
1.17
[kcrid001@coreV2-22-028 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span> KCridbowtie2.txt <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span> -f 5
370715
569956
82036
677289
38461
303192
11115
409225
189559
281489
300187
366410
302622
281735
262310
178865
[kcrid001@coreV2-22-028 QCFastqs]$ grep <span class="pl-s"><span class="pl-pds">"</span>aligned &gt;1 times<span class="pl-pds">"</span></span> KCridbowtie2.txt <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>(<span class="pl-pds">"</span></span> -f 2 <span class="pl-k">|</span> cut -d <span class="pl-s"><span class="pl-pds">"</span>%<span class="pl-pds">"</span></span> -f 1
1.07
0.58
0.60
0.67
5.63
1.00
1.06
0.74
0.90
1.19
0.83
1.13
0.86
0.76
1.19
0.72

<span class="pl-c"><span class="pl-c">#</span> copy these values to excel, calculate averages as needed</span>
<span class="pl-c"><span class="pl-c">#</span> answers</span>
	<span class="pl-c"><span class="pl-c">#</span> a - 2.48% - the mean percent "overall alignment rate"</span>
	<span class="pl-c"><span class="pl-c">#</span> b - 1.30% - the mean percent reads "aligned exactly 1 time"</span>
	<span class="pl-c"><span class="pl-c">#</span> c - 289072.88 - the mean number of reads "aligned exactly 1 time"</span>
	<span class="pl-c"><span class="pl-c">#</span> d - 1.18% - the mean percent reads "aligned &gt;1 times"</span>


<span class="pl-c"><span class="pl-c">#</span>4- add your statistics as single rows to the shared table /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt as tab-delimited text in the following order:</span>
LaneX_yourinitials	b-the mean percent <span class="pl-s"><span class="pl-pds">"</span>overall alignment rate<span class="pl-pds">"</span></span>	c-the mean percent reads <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span>	d-the mean number of reads <span class="pl-s"><span class="pl-pds">"</span>aligned exactly 1 time<span class="pl-pds">"</span></span>	e-the mean percent reads <span class="pl-s"><span class="pl-pds">"</span>aligned &gt;1 times<span class="pl-pds">"</span></span>

[kcrid001@coreV2-22-028 QCFastqs]$ nano alignstats.txt
[kcrid001@coreV2-22-028 QCFastqs]$ cat alignstats.txt <span class="pl-k">&gt;&gt;</span> /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt
[kcrid001@coreV2-22-028 QCFastqs]$ cat /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt
LaneX	overallalignmentrate	percsinglyaligned	numsinglyaligned	percmultiplyaligned
LaneX_djb	2.495714286	1.251428571	247656.7143	1.247142857
LaneX_djb	2.495714286	1.251428571	247656.7143	1.247142857
Lane1AP	5.378125	1.3725	300361.8125	4.005
lane5_sg	2.03%	1.11%	235455.75	0.92%
LaneX_jcw	2.495714286	1.251428571	247656.7143	1.247142857
Lane4_KCrid	2.48	1.30	289072.88	1.18

<span class="pl-c"><span class="pl-c">#</span>5- Data cleanup and archiving:</span>
<span class="pl-k">*</span>a-mv your Trinity.fasta file from your trinity_out_dir to a folder called testassembly <span class="pl-k">in</span> your data directory
	
[kcrid001@turing1 data]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@turing1 data]$ mkdir testassembly
[kcrid001@turing1 data]$ ls
exercises  KCridCopyLane04.sh   KCridGunzipLane04.sh   testassembly
fastq      KCridCopyLane04.txt  KCridGunzipLane04.txt
[kcrid001@turing1 trinity_out_dir]$ salloc
salloc: Pending job allocation 9275008
salloc: job 9275008 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9275008 has been allocated resources
salloc: Granted job allocation 9275008
[kcrid001@coreV1-22-016 trinity_out_dir]$ mv /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir/Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly

<span class="pl-k">*</span>b-set up a sbatch script to:
		rm -r YOURtrinity_out_dir
		rm -r YOURoriginalfastqs
		rm -r YOURfilteringstats <span class="pl-c"><span class="pl-c">#</span> as long as you've already appended your filteringstats output to the class table as part of homework_day04.txt</span>

[kcrid001@coreV1-22-016 data]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data
[kcrid001@coreV1-22-016 data]$ nano Cleanup.txt
[kcrid001@coreV1-22-016 data]$ cat Cleanup.txt 
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o Cleanup.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=Cleanup</span>

rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/KCKCfastq/trinity_out_dir
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/originalfastqs
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/filteringstats

[kcrid001@coreV1-22-016 data]$ sbatch Cleanup.sh 
Submitted batch job 9275062
[kcrid001@coreV1-22-016 data]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9275062      main  Cleanup kcrid001  R       0:07      1 coreV1-22-016 
           9275008      main       sh kcrid001  R      22:37      1 coreV1-22-016 

<span class="pl-c"><span class="pl-c">#</span>6- submit a blast against sprot from your testassembly folder using the following command</span>

[kcrid001@coreV1-22-016 testassembly]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly
[kcrid001@coreV1-22-016 testassembly]$ ls
Trinity.fasta
[kcrid001@coreV1-22-016 testassembly]$ nano Blast.sh
[kcrid001@coreV1-22-016 testassembly]$ cat Blast.sh 
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o Blast.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 6         </span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=Blast</span>

enable_lmod
module load container_env blast
blastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 \
        -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6
[kcrid001@coreV1-22-016 testassembly]$ sbatch Blast.sh 
Submitted batch job 9275071
[kcrid001@coreV1-22-016 testassembly]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9275071      main    Blast kcrid001  R       1:53      1 coreV3-23-027 
           9275062      main  Cleanup kcrid001  R       6:23      1 coreV1-22-016 
           9275008      main       sh kcrid001  R      28:53      1 coreV1-22-016 

<span class="pl-c"><span class="pl-c">#</span>##Exploring our SAM files, check out http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#sam-output for bowtie2 specific output and http://bio-bwa.sourceforge.net/bwa.shtml#4 for general SAM output</span>

<span class="pl-c"><span class="pl-c">#</span>7- head one of your .sam files to look at the header</span>

[kcrid001@coreV2-22-028 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs
[kcrid001@coreV2-22-028 QCFastqs]$ head RI_B_04_14_clippedtrimmed.fastq.sam
@HD	VN:1.0	SO:unsorted
@SQ	SN:TR78<span class="pl-k">|</span>c0_g1_i1_coral	LN:1109
@SQ	SN:TR78<span class="pl-k">|</span>c0_g2_i1_coral	LN:1109
@SQ	SN:TR79<span class="pl-k">|</span>c0_g1_i1_coral	LN:610
@SQ	SN:TR79<span class="pl-k">|</span>c0_g2_i1_coral	LN:1549
@SQ	SN:TR87<span class="pl-k">|</span>c0_g1_i1_coral	LN:732
@SQ	SN:TR93<span class="pl-k">|</span>c0_g1_i1_coral	LN:550
@SQ	SN:TR101<span class="pl-k">|</span>c0_g1_i1_coral	LN:673
@SQ	SN:TR104<span class="pl-k">|</span>c0_g1_i1_coral	LN:607
@SQ	SN:TR105<span class="pl-k">|</span>c0_g1_i1_coral	LN:587


<span class="pl-c"><span class="pl-c">#</span>8- grep -v '@' your.sam | head to look at the sequence read lines, why does this work to exclude the header lines?</span>
	<span class="pl-c"><span class="pl-c">#</span>because the -v flag means to omit lines with the indicated symbol</span>

[kcrid001@coreV2-22-028 QCFastqs]$ grep -v <span class="pl-s"><span class="pl-pds">'</span>@<span class="pl-pds">'</span></span> RI_B_04_14_clippedtrimmed.fastq.sam <span class="pl-k">|</span> head
K00188:59:HMTFHBBXX:5:1101:6055:1648	16	TR13320<span class="pl-k">|</span>c0_g2_i1_coral	2	253M4I44M	<span class="pl-k">*</span>	0	0	TTGAATCATTCTTGTCTACTCCAAGGGGTCTACTTTCCTCAAGGTTTCCNJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJJJJJFFA#A	AS:i:-30	XN:i:0	XM:i:3	XO:i:1	XG:i:4	NM:i:7	MD:Z:0A1C42A1	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:9597:1666	0	TR35685<span class="pl-k">|</span>c1_g2_i1_coral	535	2520M	<span class="pl-k">*</span>	0	0	GACAACCCCGTCTGTGGCGC	AAFFFJJJJJJJJJJJJJJJ	AS:i:-6	XN:i:0	XM:i:1	XO:i:0	XG:i:0	NM:i:1	MD:Z:6A13	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:26281:1666	16	TR59703<span class="pl-k">|</span>c0_g1_i3_sym	244	1	51M	<span class="pl-k">*</span>	0	0	CTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:26281:1666	272	TR59703<span class="pl-k">|</span>c0_g1_i1_sym	277	2551M	<span class="pl-k">*</span>	0	0	CTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:26281:1666	272	TR59703<span class="pl-k">|</span>c0_g1_i3_coral	244	2551M	<span class="pl-k">*</span>	0	0	CTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:26281:1666	272	TR59703<span class="pl-k">|</span>c0_g1_i1_coral	277	2551M	<span class="pl-k">*</span>	0	0	CTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTFJJFJJFJJJJJJJJJJJFJJJJJJJJAJJJJJJJJFJJJJJJFJJFFFAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:9557:1701	0	TR30871<span class="pl-k">|</span>c0_g1_i1_coral	16	253M3I45M	<span class="pl-k">*</span>	0	0	GTGATAATGATGTTGGTGATAATGATGATGATGGTGGTGGTTGTGGTGTTAAFFFJJJJJFFJJJJJFJFJJAFJJJJJJJJJJFJJAJJJJJFJJFJFFJ	AS:i:-26	XN:i:0	XM:i:2	XO:i:1	XG:i:3	NM:i:5	MD:Z:6G5A35	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:28270:1701	16	TR5604<span class="pl-k">|</span>c0_g1_i1_coral	515	2551M	<span class="pl-k">*</span>	0	0	CTAAGGTTTGCTTTAAAGCACTTAGTTCAGAATCCTTTGTCTGTACAATTJJJJJJFJJJJJJJJJJJJJJJJJJJJF-JJJFJFJJJJJJJJJJJFFFAA	AS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:28371:1701	16	TR59703<span class="pl-k">|</span>c0_g1_i3_sym	244	1	51M	<span class="pl-k">*</span>	0	0	CTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTJJJ7JJJJJJFJJJJJJJJJJJJJJJFAJJJJJJJJAJJJJJAJJJFFFAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_04_14
K00188:59:HMTFHBBXX:5:1101:28371:1701	272	TR59703<span class="pl-k">|</span>c0_g1_i1_sym	277	2551M	<span class="pl-k">*</span>	0	0	CTGCTTCTAAACAGCTGGAGAAAGCTGAAATTCTGGAAAAGACAATTAGTJJJ7JJJJJJFJJJJJJJJJJJJJJJFAJJJJJJJJAJJJJJAJJJFFFAA	AS:i:0	XS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:51	YT:Z:UU	RG:Z:RI_B_04_14


<span class="pl-c"><span class="pl-c">#</span>9- in an interactive session run /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py on 2-3 of your .sam files using * to select 2-3 at the same time.</span>

[kcrid001@coreV1-22-016 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs
[kcrid001@coreV1-22-016 QCFastqs]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py RI_B_04_14_clippedtrimmed.fastq.sam RI_B_04_18_clippedtrimmed.fastq.sam RI_B_04_22_clippedtrimmed.fastq.sam
RI_B_04_14_clippedtrimmed.fastq.sam
[<span class="pl-s"><span class="pl-pds">'</span>0<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>272<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>256<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>16<span class="pl-pds">'</span></span>]
0 <span class="pl-c1">:</span>
272 <span class="pl-c1">:</span>
	<span class="pl-c1">read</span> reverse strand
	not primary alignment
256 <span class="pl-c1">:</span>
	not primary alignment
16 <span class="pl-c1">:</span>
	<span class="pl-c1">read</span> reverse strand
RI_B_04_18_clippedtrimmed.fastq.sam
[<span class="pl-s"><span class="pl-pds">'</span>0<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>272<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>256<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>16<span class="pl-pds">'</span></span>]
0 <span class="pl-c1">:</span>
272 <span class="pl-c1">:</span>
	<span class="pl-c1">read</span> reverse strand
	not primary alignment
256 <span class="pl-c1">:</span>
	not primary alignment
16 <span class="pl-c1">:</span>
	<span class="pl-c1">read</span> reverse strand
RI_B_04_22_clippedtrimmed.fastq.sam
[<span class="pl-s"><span class="pl-pds">'</span>0<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>272<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>256<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>16<span class="pl-pds">'</span></span>]
0 <span class="pl-c1">:</span>
272 <span class="pl-c1">:</span>
	<span class="pl-c1">read</span> reverse strand
	not primary alignment
256 <span class="pl-c1">:</span>
	not primary alignment
16 <span class="pl-c1">:</span>
	<span class="pl-c1">read</span> reverse strand


<span class="pl-c"><span class="pl-c">#</span>10- we need to run the read sorting step required for SNP calling, so if you have time, set up and run the following script on your .sam files to finish before Wednesday:</span>

<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KCridBamandSort.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1         </span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KCridBamandSort</span>

enable_lmod
module load samtools/1
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-k">*</span>.sam<span class="pl-k">;</span> <span class="pl-k">do</span> <span class="pl-s"><span class="pl-pds">`</span>samtools view -bS <span class="pl-smi">$i</span> <span class="pl-k">&gt;</span> <span class="pl-smi">${i<span class="pl-k">%</span>.sam}</span>_UNSORTED.bam<span class="pl-pds">`</span></span><span class="pl-k">;</span> <span class="pl-k">done</span>

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-k">*</span>UNSORTED.bam<span class="pl-k">;</span> <span class="pl-k">do</span> samtools sort <span class="pl-smi">$i</span> <span class="pl-k">&gt;</span> <span class="pl-smi">${i<span class="pl-k">%</span>_UNSORTED.bam}</span>.bam
samtools index <span class="pl-smi">${i<span class="pl-k">%</span>_UNSORTED.bam}</span>.bam
<span class="pl-k">done</span>

[kcrid001@coreV1-22-016 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs
[kcrid001@coreV1-22-016 QCFastqs]$ cat bamandsort.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KCridBamandSort.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1         </span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KCridBamandSort</span>

enable_lmod
module load samtools/1
<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-k">*</span>.sam<span class="pl-k">;</span> <span class="pl-k">do</span> <span class="pl-s"><span class="pl-pds">`</span>samtools view -bS <span class="pl-smi">$i</span> <span class="pl-k">&gt;</span> <span class="pl-smi">${i<span class="pl-k">%</span>.sam}</span>_UNSORTED.bam<span class="pl-pds">`</span></span><span class="pl-k">;</span> <span class="pl-k">done</span>

<span class="pl-k">for</span> <span class="pl-smi">i</span> <span class="pl-k">in</span> <span class="pl-k">*</span>UNSORTED.bam<span class="pl-k">;</span> <span class="pl-k">do</span> samtools sort <span class="pl-smi">$i</span> <span class="pl-k">&gt;</span> <span class="pl-smi">${i<span class="pl-k">%</span>_UNSORTED.bam}</span>.bam
samtools index <span class="pl-smi">${i<span class="pl-k">%</span>_UNSORTED.bam}</span>.bam
<span class="pl-k">done</span>

[kcrid001@coreV1-22-016 QCFastqs]$ sbatch bamandsort.sh
Submitted batch job 9275211
[kcrid001@coreV1-22-016 QCFastqs]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9275071      main    Blast kcrid001  R      16:38      1 coreV3-23-027 
           9275062      main  Cleanup kcrid001  R      21:08      1 coreV1-22-016 
           9275008      main       sh kcrid001  R      43:38      1 coreV1-22-016 
           9275211      main KCridBam kcrid001  R       0:08      1 coreV2-22-007 
</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day07 HW
                     </h2>
                     <a aria-label="Permalink: Day07 HW" class="anchor" href="#day07-hw" id="user-content-day07-hw">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#day07HW

#1- Run the following command on your sprot output file to process into the contig length/match format that trinity examines

[kcrid001@turing1 testassembly]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly
[kcrid001@turing1 testassembly]$ ls
Blast.sh  Blast.txt  blastx.outfmt6  Trinity.fasta
[kcrid001@turing1 testassembly]$ nano blastparsing.sh
[kcrid001@turing1 testassembly]$ cat blastparsing.sh 
#!/bin/bash -l

#SBATCH -o KCridBlastParse.TXT
#SBATCH -n 1
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KCridBlastParse

/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta
[kcrid001@turing1 testassembly]$ salloc
salloc: Pending job allocation 9276475
salloc: job 9276475 queued and waiting for resources
salloc: job 9276475 has been allocated resources
salloc: Granted job allocation 9276475
[kcrid001@coreV3-23-040 testassembly]$ sbatch blastparsing.sh 
Submitted batch job 9276476
[kcrid001@coreV3-23-040 testassembly]$ ls
blastparsing.sh  blastx.outfmt6            blastx.outfmt6.w_pct_hit_length
Blast.sh         blastx.outfmt6.hist       KCridBlastParse.TXT
Blast.txt        blastx.outfmt6.hist.list  Trinity.fasta
[kcrid001@coreV3-23-040 testassembly]$ cat blastx.outfmt6.hist
#hit_pct_cov_bin	count_in_bin	&gt;bin_below
100	212	212
90	98	310
80	104	414
70	137	551
60	182	733
50	256	989
40	413	1402
30	719	2121
20	1440	3561
10	984	4545

#2- Rm UNSORTED.bam from your QCFastqs directory (or wherever your .bams and .sams are)

[kcrid001@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs
[kcrid001@coreV3-23-040 QCFastqs]$ nano KCridrmunsortbam.sh
[kcrid001@coreV3-23-040 QCFastqs]$ cat KCridrmunsortbam.sh

#!/bin/bash -l

#SBATCH -o KCridrmunsortbam.txt
#SBATCH -n 1         
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KCridrmunsortbam

rm *UNSORTED.bam

[kcrid001@coreV3-23-040 QCFastqs]$ sbatch KCridrmunsortbam.sh 
Submitted batch job 9276495


#3- Run the following to start genotyping your SNPs for filtering next class

[kcrid001@coreV3-23-040 QCFastqs]$ nano KCridfreebayessubref.sh
[kcrid001@coreV3-23-040 QCFastqs]$ cat KCridfreebayessubref.sh
#!/bin/bash -l

#SBATCH -o KCridfreebayessubref.txt
#SBATCH -n 1         
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KCridfreebayessubref

enable_lmod
module load dDocent
freebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.bam &gt; KCridEmergedfastqs.vcf
[kcrid001@coreV3-23-040 QCFastqs]$ sbatch KCridfreebayessubref.sh 
Submitted batch job 9276573
[kcrid001@coreV3-23-040 QCFastqs]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9276475      main       sh kcrid001  R    1:42:22      1 coreV3-23-040 
           9276573      main KCridfre kcrid001  R       1:22      1 coreV3-23-046 " dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>day07HW</span>

<span class="pl-c"><span class="pl-c">#</span>1- Run the following command on your sprot output file to process into the contig length/match format that trinity examines</span>

[kcrid001@turing1 testassembly]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/testassembly
[kcrid001@turing1 testassembly]$ ls
Blast.sh  Blast.txt  blastx.outfmt6  Trinity.fasta
[kcrid001@turing1 testassembly]$ nano blastparsing.sh
[kcrid001@turing1 testassembly]$ cat blastparsing.sh 
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KCridBlastParse.TXT</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KCridBlastParse</span>

/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta
[kcrid001@turing1 testassembly]$ salloc
salloc: Pending job allocation 9276475
salloc: job 9276475 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9276475 has been allocated resources
salloc: Granted job allocation 9276475
[kcrid001@coreV3-23-040 testassembly]$ sbatch blastparsing.sh 
Submitted batch job 9276476
[kcrid001@coreV3-23-040 testassembly]$ ls
blastparsing.sh  blastx.outfmt6            blastx.outfmt6.w_pct_hit_length
Blast.sh         blastx.outfmt6.hist       KCridBlastParse.TXT
Blast.txt        blastx.outfmt6.hist.list  Trinity.fasta
[kcrid001@coreV3-23-040 testassembly]$ cat blastx.outfmt6.hist
<span class="pl-c"><span class="pl-c">#</span>hit_pct_cov_bin	count_in_bin	&gt;bin_below</span>
100	212	212
90	98	310
80	104	414
70	137	551
60	182	733
50	256	989
40	413	1402
30	719	2121
20	1440	3561
10	984	4545

<span class="pl-c"><span class="pl-c">#</span>2- Rm UNSORTED.bam from your QCFastqs directory (or wherever your .bams and .sams are)</span>

[kcrid001@turing1 QCFastqs]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/fastq/QCFastqs
[kcrid001@coreV3-23-040 QCFastqs]$ nano KCridrmunsortbam.sh
[kcrid001@coreV3-23-040 QCFastqs]$ cat KCridrmunsortbam.sh

<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KCridrmunsortbam.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1         </span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KCridrmunsortbam</span>

rm <span class="pl-k">*</span>UNSORTED.bam

[kcrid001@coreV3-23-040 QCFastqs]$ sbatch KCridrmunsortbam.sh 
Submitted batch job 9276495


<span class="pl-c"><span class="pl-c">#</span>3- Run the following to start genotyping your SNPs for filtering next class</span>

[kcrid001@coreV3-23-040 QCFastqs]$ nano KCridfreebayessubref.sh
[kcrid001@coreV3-23-040 QCFastqs]$ cat KCridfreebayessubref.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KCridfreebayessubref.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1         </span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KCridfreebayessubref</span>

enable_lmod
module load dDocent
freebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta <span class="pl-k">*</span>.bam <span class="pl-k">&gt;</span> KCridEmergedfastqs.vcf
[kcrid001@coreV3-23-040 QCFastqs]$ sbatch KCridfreebayessubref.sh 
Submitted batch job 9276573
[kcrid001@coreV3-23-040 QCFastqs]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9276475      main       sh kcrid001  R    1:42:22      1 coreV3-23-040 
           9276573      main KCridfre kcrid001  R       1:22      1 coreV3-23-046 </pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day08 HW
                     </h2>
                     <a aria-label="Permalink: Day08 HW" class="anchor" href="#day08-hw" id="user-content-day08-hw">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#day08HW

#1- Clean up your data directory by:
	-Making a SAMS folder and mv all your .sam files into that directory
	-Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory
	-rm your _unfiltered.vcf files if you have any
	-rm your .fastq files
	-Make a VCF folder in your data directory and mv your YOURNAMEmergedfastqs.vcf into this directory (if your freebayes job didn't complete then skip this step)

#2- Start an interactive session via salloc

[kcrid001@turing1 QCFastqs]$ cd ../../
[kcrid001@turing1 data]$ mkdir SAMS
[kcrid001@turing1 QCFastqs]$ mv *.sam ../../SAMS/
[kcrid001@turing1 QCFastqs]$ salloc
salloc: Pending job allocation 9278244
salloc: job 9278244 queued and waiting for resources
salloc: job 9278244 has been allocated resources
salloc: Granted job allocation 9278244
[kcrid001@coreV3-23-040 QCFastqs]$ mkdir ../../BAMS
[kcrid001@coreV3-23-040 QCFastqs]$ mv *.bam *.bam.bai ../../BAMS/


#3- cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf to your VCF folder

[kcrid001@coreV3-23-040 QCFastqs]$ mkdir ../../VCF
[kcrid001@coreV3-23-040 QCFastqs]$ cd ../../VCF
[kcrid001@coreV3-23-040 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF
[kcrid001@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf ./


#4- Determine the number of individuals and variant sites in the class vcf file (and yours if it worked) using:
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf
[kcrid001@coreV3-23-040 VCF]$ enable_lmod
[kcrid001@coreV3-23-040 VCF]$ module load dDocent
[kcrid001@coreV3-23-040 VCF]$ vcftools

VCFtools (0.1.14)
© Adam Auton and Anthony Marcketta 2009

Process Variant Call Format files

For a list of options, please go to:
	https://vcftools.github.io/man_latest.html

Alternatively, a man page is available, type:
	man vcftools

Questions, comments, and suggestions should be emailed to:
	vcftools-help@lists.sourceforge.net

[kcrid001@coreV3-23-040 VCF]$  vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly.vcf

After filtering, kept 40 out of 40 Individuals
After filtering, kept 1214003 out of a possible 1214003 Sites
Run Time = 14.00 seconds


[kcrid001@coreV3-23-040 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF
[kcrid001@coreV3-23-040 VCF]$ cp ../fastq/QCFastqs/KCridEmergedfastqs.vcf ./
[kcrid001@coreV3-23-040 VCF]$ vcftools --vcf KCridEmergedfastqs.vcf 

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf KCridEmergedfastqs.vcf

\After filtering, kept 16 out of 16 Individuals
After filtering, kept 21896 out of a possible 21896 Sites
Run Time = 0.00 seconds


#5- cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt into your directory with your .vcf files

[kcrid001@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt ./
[kcrid001@coreV3-23-040 VCF]$ ls
GoodCoralGenelistForVCFSubsetter.txt  mergedfastq_HEAAstrangiaAssembly.vcf
KCridEmergedfastqs.vcf                out.log


#6- Run our host vcf extractor on your merged vcf file using the following syntax:

/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf


[kcrid001@coreV3-23-040 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF
[kcrid001@coreV3-23-040 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf
Read in ContigList
^CTraceback (most recent call last):
  File &quot;/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py&quot;, line 38, in &lt;module&gt;
    if Contig in ContigsToKeep.keys(): #is our contig one we want to keep
KeyboardInterrupt
[kcrid001@turing1 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf
[kcrid001@turing1 VCF]$ ls
GoodCoralGenelistForVCFSubsetter.txt
KCridEmergedfastqs.vcf
mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf
mergedfastq_HEAAstrangiaAssembly_subset.vcf
mergedfastq_HEAAstrangiaAssembly.vcf
out.log


#7- Compare the number of variant sites in your three files (YOURNAMEmergedfastqs.vcf,  mergedfastq_HEAAstrangiaAssembly.vcf, and  mergedfastq_HEAAstrangiaAssembly_subset.vcf) using:
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf


[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf

After filtering, kept 40 out of 40 Individuals
After filtering, kept 383467 out of a possible 383467 Sites
Run Time = 6.00 seconds
[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly.vcf

After filtering, kept 40 out of 40 Individuals
After filtering, kept 1214003 out of a possible 1214003 Sites
Run Time = 15.00 seconds
[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf KCridEmergedfastqs.vcf

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf KCridEmergedfastqs.vcf

After filtering, kept 16 out of 16 Individuals
After filtering, kept 21896 out of a possible 21896 Sites
Run Time = 0.00 seconds



#8- Work through the VCF filtering tutorial until the following step:
#http://www.ddocent.com/filtering/
#Now that we have a list of individuals to remove, we can feed that directly into VCFtools for filtering.

#vcftools --vcf raw.g5mac3dp3.recode.vcf --remove lowDP.indv --recode --recode-INFO-all --out raw.g5mac3dplm


[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
	--recode-INFO-all
	--minQ 30
	--max-missing 0.5
	--out raw.g5mac3
	--recode

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 133004 out of a possible 383467 Sites
Run Time = 25.00 seconds

[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf raw.g5mac3.recode.vcf
	--recode-INFO-all
	--minDP 3
	--out raw.g5mac3dp3
	--recode

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 133004 out of a possible 133004 Sites
Run Time = 26.00 seconds

[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf raw.g5mac3dp3.recode.vcf
	--missing-indv

After filtering, kept 40 out of 40 Individuals
Outputting Individual Missingness
After filtering, kept 133004 out of a possible 133004 Sites
Run Time = 2.00 seconds

[kcrid001@coreV3-23-002 VCF]$ cat out.imiss
INDV	N_DATA	N_GENOTYPES_FILTERED	N_MISS	F_MISS
RI_W_06_merged	133004	0	86665	0.651597
RI_W_07_merged	133004	0	82904	0.62332
VA_B_03_merged	133004	0	72661	0.546307
RI_W_02_merged	133004	0	94767	0.712512
RI_W_04_merged	133004	0	92258	0.693648
VA_W_09_SNP_clipped	133004	0	24563	0.184679
RI_B_08_SNP_clipped	133004	0	116118	0.873041
VA_W_08_SNP_clipped	133004	0	109256	0.821449
VA_B_08_SNP_clipped	133004	0	124485	0.935949
VA_W_02_merged	133004	0	92331	0.694197
VA_B_07_merged	133004	0	80522	0.60541
RI_B_05_merged	133004	0	55934	0.420544
VA_W_06_merged	133004	0	84347	0.634169
VA_W_04_merged	133004	0	64460	0.484647
VA_W_01_merged	133004	0	86924	0.653544
VA_B_10_SNP_clipped	133004	0	107967	0.811758
VA_B_06_merged	133004	0	73841	0.555179
VA_W_05_merged	133004	0	85184	0.640462
RI_B_09_SNP_clipped	133004	0	111872	0.841118
VA_W_10_SNP_clipped	133004	0	108814	0.818126
RI_W_08_SNP_clipped	133004	0	101731	0.764872
RI_B_06_merged	133004	0	103948	0.78154
RI_W_10_SNP_clipped	133004	0	120375	0.905048
RI_B_04_merged	133004	0	59639	0.4484
VA_W_03_merged	133004	0	82018	0.616658
RI_B_07_merged	133004	0	91804	0.690235
RI_W_05_merged	133004	0	73522	0.55278
RI_W_09_SNP_clipped	133004	0	116492	0.875853
VA_B_01_merged	133004	0	59653	0.448505
VA_B_09_SNP_clipped	133004	0	103404	0.77745
RI_B_10_SNP_clipped	133004	0	108283	0.814133
RI_W_01_merged	133004	0	95024	0.714445
RI_B_01_merged	133004	0	89764	0.674897
VA_B_04_merged	133004	0	74178	0.557713
RI_B_02_merged	133004	0	114857	0.86356
RI_W_03_merged	133004	0	49393	0.371365
VA_B_02_merged	133004	0	102051	0.767278
VA_W_07_merged	133004	0	83818	0.630192
VA_B_05_merged	133004	0	69075	0.519345
RI_B_03_merged	133004	0	104979	0.789292


[kcrid001@coreV3-23-002 VCF]$ bash
kcrid001@coreV3-23-002:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF$ mawk '!/IN/' out.imiss | cut -f5 &gt; totalmissing
kcrid001@coreV3-23-002:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF$ 
kcrid001@coreV3-23-002:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF$ gnuplot &lt;&lt; \EOF 
&gt; set terminal dumb size 120, 30
&gt; set autoscale 
&gt; unset label
&gt; set title &quot;Histogram of % missing data per individual&quot;
&gt; set ylabel &quot;Number of Occurrences&quot;
&gt; set xlabel &quot;% of missing data&quot;
&gt; #set yr [0:100000]
&gt; binwidth=0.01
&gt; bin(x,width)=width*floor(x/width) + binwidth/2.0
&gt; plot 'totalmissing' using (bin($1,binwidth)):(1.0) smooth freq with boxes
&gt; pause -1
&gt; EOF

                                         Histogram of % missing data per individual
  Number of Occurrences
      3 ++----------+-----------+-----------+-----------+------****--+---------***-----------***---------+----------++
        +           +           +           +           +      *'totalmissing' using (bin($1,binwidth)):(1.0) ****** +
        |                                                      *  *            * *           * *                     |
        |                                                      *  *            * *           * *                     |
        |                                                      *  *            * *           * *                     |
        |                                                      *  *            * *           * *                     |
    2.5 ++                                                     *  *            * *           * *                    ++
        |                                                      *  *            * *           * *                     |
        |                                                      *  *            * *           * *                     |
        |                                                      *  *            * *           * *                     |
        |                                                      *  *            * *           * *                     |
        |                                                      *  *            * *           * *                     |
      2 ++                                       ****          *  *     ** **  * ************* *     ****           ++
        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |
        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |
        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |
        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |
        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |
    1.5 ++                                       *  *          *  *     ** **  * *    *  **  * *     *  *           ++
        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |
        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |
        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |
        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |
        +           +           +           +    *  *   +      *  *  +  ** **  * *    *  **  * *     *  *+           +
      1 ********************************************************************************************************----++
       0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1
                                                      % of missing data
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>day08HW</span>

<span class="pl-c"><span class="pl-c">#</span>1- Clean up your data directory by:</span>
	-Making a SAMS folder and mv all your .sam files into that directory
	-Make a BAMS folder and mv all your .bam files and .bam.bai files into that directory
	-rm your _unfiltered.vcf files <span class="pl-k">if</span> you have any
	-rm your .fastq files
	-Make a VCF folder <span class="pl-k">in</span> your data directory and mv your YOURNAMEmergedfastqs.vcf into this directory (if your freebayes job didn<span class="pl-s"><span class="pl-pds">'</span>t complete then skip this step)</span>
<span class="pl-s"></span>
<span class="pl-s">#2- Start an interactive session via salloc</span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@turing1 QCFastqs]$ cd ../../</span>
<span class="pl-s">[kcrid001@turing1 data]$ mkdir SAMS</span>
<span class="pl-s">[kcrid001@turing1 QCFastqs]$ mv *.sam ../../SAMS/</span>
<span class="pl-s">[kcrid001@turing1 QCFastqs]$ salloc</span>
<span class="pl-s">salloc: Pending job allocation 9278244</span>
<span class="pl-s">salloc: job 9278244 queued and waiting for resources</span>
<span class="pl-s">salloc: job 9278244 has been allocated resources</span>
<span class="pl-s">salloc: Granted job allocation 9278244</span>
<span class="pl-s">[kcrid001@coreV3-23-040 QCFastqs]$ mkdir ../../BAMS</span>
<span class="pl-s">[kcrid001@coreV3-23-040 QCFastqs]$ mv *.bam *.bam.bai ../../BAMS/</span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">#3- cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf to your VCF folder</span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV3-23-040 QCFastqs]$ mkdir ../../VCF</span>
<span class="pl-s">[kcrid001@coreV3-23-040 QCFastqs]$ cd ../../VCF</span>
<span class="pl-s">[kcrid001@coreV3-23-040 VCF]$ pwd</span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF</span>
<span class="pl-s">[kcrid001@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf ./</span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">#4- Determine the number of individuals and variant sites in the class vcf file (and yours if it worked) using:</span>
<span class="pl-s">/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s">[kcrid001@coreV3-23-040 VCF]$ enable_lmod</span>
<span class="pl-s">[kcrid001@coreV3-23-040 VCF]$ module load dDocent</span>
<span class="pl-s">[kcrid001@coreV3-23-040 VCF]$ vcftools</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools (0.1.14)</span>
<span class="pl-s">© Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Process Variant Call Format files</span>
<span class="pl-s"></span>
<span class="pl-s">For a list of options, please go to:</span>
<span class="pl-s">	https://vcftools.github.io/man_latest.html</span>
<span class="pl-s"></span>
<span class="pl-s">Alternatively, a man page is available, type:</span>
<span class="pl-s">	man vcftools</span>
<span class="pl-s"></span>
<span class="pl-s">Questions, comments, and suggestions should be emailed to:</span>
<span class="pl-s">	vcftools-help@lists.sourceforge.net</span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV3-23-040 VCF]$  vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools - 0.1.14</span>
<span class="pl-s">(C) Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Parameters as interpreted:</span>
<span class="pl-s">	--vcf mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">After filtering, kept 40 out of 40 Individuals</span>
<span class="pl-s">After filtering, kept 1214003 out of a possible 1214003 Sites</span>
<span class="pl-s">Run Time = 14.00 seconds</span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV3-23-040 VCF]$ pwd</span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF</span>
<span class="pl-s">[kcrid001@coreV3-23-040 VCF]$ cp ../fastq/QCFastqs/KCridEmergedfastqs.vcf ./</span>
<span class="pl-s">[kcrid001@coreV3-23-040 VCF]$ vcftools --vcf KCridEmergedfastqs.vcf </span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools - 0.1.14</span>
<span class="pl-s">(C) Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Parameters as interpreted:</span>
<span class="pl-s">	--vcf KCridEmergedfastqs.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">\After filtering, kept 16 out of 16 Individuals</span>
<span class="pl-s">After filtering, kept 21896 out of a possible 21896 Sites</span>
<span class="pl-s">Run Time = 0.00 seconds</span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">#5- cp the /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt into your directory with your .vcf files</span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV3-23-040 VCF]$ cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt ./</span>
<span class="pl-s">[kcrid001@coreV3-23-040 VCF]$ ls</span>
<span class="pl-s">GoodCoralGenelistForVCFSubsetter.txt  mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s">KCridEmergedfastqs.vcf                out.log</span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">#6- Run our host vcf extractor on your merged vcf file using the following syntax:</span>
<span class="pl-s"></span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV3-23-040 VCF]$ pwd</span>
<span class="pl-s">/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF</span>
<span class="pl-s">[kcrid001@coreV3-23-040 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s">Read in ContigList</span>
<span class="pl-s">^CTraceback (most recent call last):</span>
<span class="pl-s">  File "/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py", line 38, in &lt;module&gt;</span>
<span class="pl-s">    if Contig in ContigsToKeep.keys(): #is our contig one we want to keep</span>
<span class="pl-s">KeyboardInterrupt</span>
<span class="pl-s">[kcrid001@turing1 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s">[kcrid001@turing1 VCF]$ ls</span>
<span class="pl-s">GoodCoralGenelistForVCFSubsetter.txt</span>
<span class="pl-s">KCridEmergedfastqs.vcf</span>
<span class="pl-s">mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf</span>
<span class="pl-s">mergedfastq_HEAAstrangiaAssembly_subset.vcf</span>
<span class="pl-s">mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s">out.log</span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">#7- Compare the number of variant sites in your three files (YOURNAMEmergedfastqs.vcf,  mergedfastq_HEAAstrangiaAssembly.vcf, and  mergedfastq_HEAAstrangiaAssembly_subset.vcf) using:</span>
<span class="pl-s">/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf</span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools - 0.1.14</span>
<span class="pl-s">(C) Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Parameters as interpreted:</span>
<span class="pl-s">	--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">After filtering, kept 40 out of 40 Individuals</span>
<span class="pl-s">After filtering, kept 383467 out of a possible 383467 Sites</span>
<span class="pl-s">Run Time = 6.00 seconds</span>
<span class="pl-s">[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools - 0.1.14</span>
<span class="pl-s">(C) Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Parameters as interpreted:</span>
<span class="pl-s">	--vcf mergedfastq_HEAAstrangiaAssembly.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">After filtering, kept 40 out of 40 Individuals</span>
<span class="pl-s">After filtering, kept 1214003 out of a possible 1214003 Sites</span>
<span class="pl-s">Run Time = 15.00 seconds</span>
<span class="pl-s">[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf KCridEmergedfastqs.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools - 0.1.14</span>
<span class="pl-s">(C) Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Parameters as interpreted:</span>
<span class="pl-s">	--vcf KCridEmergedfastqs.vcf</span>
<span class="pl-s"></span>
<span class="pl-s">After filtering, kept 16 out of 16 Individuals</span>
<span class="pl-s">After filtering, kept 21896 out of a possible 21896 Sites</span>
<span class="pl-s">Run Time = 0.00 seconds</span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">#8- Work through the VCF filtering tutorial until the following step:</span>
<span class="pl-s">#http://www.ddocent.com/filtering/</span>
<span class="pl-s">#Now that we have a list of individuals to remove, we can feed that directly into VCFtools for filtering.</span>
<span class="pl-s"></span>
<span class="pl-s">#vcftools --vcf raw.g5mac3dp3.recode.vcf --remove lowDP.indv --recode --recode-INFO-all --out raw.g5mac3dplm</span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools - 0.1.14</span>
<span class="pl-s">(C) Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Parameters as interpreted:</span>
<span class="pl-s">	--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf</span>
<span class="pl-s">	--recode-INFO-all</span>
<span class="pl-s">	--minQ 30</span>
<span class="pl-s">	--max-missing 0.5</span>
<span class="pl-s">	--out raw.g5mac3</span>
<span class="pl-s">	--recode</span>
<span class="pl-s"></span>
<span class="pl-s">After filtering, kept 40 out of 40 Individuals</span>
<span class="pl-s">Outputting VCF file...</span>
<span class="pl-s">After filtering, kept 133004 out of a possible 383467 Sites</span>
<span class="pl-s">Run Time = 25.00 seconds</span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw.g5mac3dp3</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools - 0.1.14</span>
<span class="pl-s">(C) Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Parameters as interpreted:</span>
<span class="pl-s">	--vcf raw.g5mac3.recode.vcf</span>
<span class="pl-s">	--recode-INFO-all</span>
<span class="pl-s">	--minDP 3</span>
<span class="pl-s">	--out raw.g5mac3dp3</span>
<span class="pl-s">	--recode</span>
<span class="pl-s"></span>
<span class="pl-s">After filtering, kept 40 out of 40 Individuals</span>
<span class="pl-s">Outputting VCF file...</span>
<span class="pl-s">After filtering, kept 133004 out of a possible 133004 Sites</span>
<span class="pl-s">Run Time = 26.00 seconds</span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV3-23-002 VCF]$ vcftools --vcf raw.g5mac3dp3.recode.vcf --missing-indv</span>
<span class="pl-s"></span>
<span class="pl-s">VCFtools - 0.1.14</span>
<span class="pl-s">(C) Adam Auton and Anthony Marcketta 2009</span>
<span class="pl-s"></span>
<span class="pl-s">Parameters as interpreted:</span>
<span class="pl-s">	--vcf raw.g5mac3dp3.recode.vcf</span>
<span class="pl-s">	--missing-indv</span>
<span class="pl-s"></span>
<span class="pl-s">After filtering, kept 40 out of 40 Individuals</span>
<span class="pl-s">Outputting Individual Missingness</span>
<span class="pl-s">After filtering, kept 133004 out of a possible 133004 Sites</span>
<span class="pl-s">Run Time = 2.00 seconds</span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV3-23-002 VCF]$ cat out.imiss</span>
<span class="pl-s">INDV	N_DATA	N_GENOTYPES_FILTERED	N_MISS	F_MISS</span>
<span class="pl-s">RI_W_06_merged	133004	0	86665	0.651597</span>
<span class="pl-s">RI_W_07_merged	133004	0	82904	0.62332</span>
<span class="pl-s">VA_B_03_merged	133004	0	72661	0.546307</span>
<span class="pl-s">RI_W_02_merged	133004	0	94767	0.712512</span>
<span class="pl-s">RI_W_04_merged	133004	0	92258	0.693648</span>
<span class="pl-s">VA_W_09_SNP_clipped	133004	0	24563	0.184679</span>
<span class="pl-s">RI_B_08_SNP_clipped	133004	0	116118	0.873041</span>
<span class="pl-s">VA_W_08_SNP_clipped	133004	0	109256	0.821449</span>
<span class="pl-s">VA_B_08_SNP_clipped	133004	0	124485	0.935949</span>
<span class="pl-s">VA_W_02_merged	133004	0	92331	0.694197</span>
<span class="pl-s">VA_B_07_merged	133004	0	80522	0.60541</span>
<span class="pl-s">RI_B_05_merged	133004	0	55934	0.420544</span>
<span class="pl-s">VA_W_06_merged	133004	0	84347	0.634169</span>
<span class="pl-s">VA_W_04_merged	133004	0	64460	0.484647</span>
<span class="pl-s">VA_W_01_merged	133004	0	86924	0.653544</span>
<span class="pl-s">VA_B_10_SNP_clipped	133004	0	107967	0.811758</span>
<span class="pl-s">VA_B_06_merged	133004	0	73841	0.555179</span>
<span class="pl-s">VA_W_05_merged	133004	0	85184	0.640462</span>
<span class="pl-s">RI_B_09_SNP_clipped	133004	0	111872	0.841118</span>
<span class="pl-s">VA_W_10_SNP_clipped	133004	0	108814	0.818126</span>
<span class="pl-s">RI_W_08_SNP_clipped	133004	0	101731	0.764872</span>
<span class="pl-s">RI_B_06_merged	133004	0	103948	0.78154</span>
<span class="pl-s">RI_W_10_SNP_clipped	133004	0	120375	0.905048</span>
<span class="pl-s">RI_B_04_merged	133004	0	59639	0.4484</span>
<span class="pl-s">VA_W_03_merged	133004	0	82018	0.616658</span>
<span class="pl-s">RI_B_07_merged	133004	0	91804	0.690235</span>
<span class="pl-s">RI_W_05_merged	133004	0	73522	0.55278</span>
<span class="pl-s">RI_W_09_SNP_clipped	133004	0	116492	0.875853</span>
<span class="pl-s">VA_B_01_merged	133004	0	59653	0.448505</span>
<span class="pl-s">VA_B_09_SNP_clipped	133004	0	103404	0.77745</span>
<span class="pl-s">RI_B_10_SNP_clipped	133004	0	108283	0.814133</span>
<span class="pl-s">RI_W_01_merged	133004	0	95024	0.714445</span>
<span class="pl-s">RI_B_01_merged	133004	0	89764	0.674897</span>
<span class="pl-s">VA_B_04_merged	133004	0	74178	0.557713</span>
<span class="pl-s">RI_B_02_merged	133004	0	114857	0.86356</span>
<span class="pl-s">RI_W_03_merged	133004	0	49393	0.371365</span>
<span class="pl-s">VA_B_02_merged	133004	0	102051	0.767278</span>
<span class="pl-s">VA_W_07_merged	133004	0	83818	0.630192</span>
<span class="pl-s">VA_B_05_merged	133004	0	69075	0.519345</span>
<span class="pl-s">RI_B_03_merged	133004	0	104979	0.789292</span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">[kcrid001@coreV3-23-002 VCF]$ bash</span>
<span class="pl-s">kcrid001@coreV3-23-002:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF$ mawk <span class="pl-pds">'</span></span><span class="pl-k">!</span>/IN/<span class="pl-s"><span class="pl-pds">'</span> out.imiss | cut -f5 &gt; totalmissing</span>
<span class="pl-s">kcrid001@coreV3-23-002:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF$ </span>
<span class="pl-s">kcrid001@coreV3-23-002:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF$ gnuplot &lt;&lt; \EOF </span>
<span class="pl-s">&gt; set terminal dumb size 120, 30</span>
<span class="pl-s">&gt; set autoscale </span>
<span class="pl-s">&gt; unset label</span>
<span class="pl-s">&gt; set title "Histogram of % missing data per individual"</span>
<span class="pl-s">&gt; set ylabel "Number of Occurrences"</span>
<span class="pl-s">&gt; set xlabel "% of missing data"</span>
<span class="pl-s">&gt; #set yr [0:100000]</span>
<span class="pl-s">&gt; binwidth=0.01</span>
<span class="pl-s">&gt; bin(x,width)=width*floor(x/width) + binwidth/2.0</span>
<span class="pl-s">&gt; plot <span class="pl-pds">'</span></span>totalmissing<span class="pl-s"><span class="pl-pds">'</span> using (bin($1,binwidth)):(1.0) smooth freq with boxes</span>
<span class="pl-s">&gt; pause -1</span>
<span class="pl-s">&gt; EOF</span>
<span class="pl-s"></span>
<span class="pl-s">                                         Histogram of % missing data per individual</span>
<span class="pl-s">  Number of Occurrences</span>
<span class="pl-s">      3 ++----------+-----------+-----------+-----------+------****--+---------***-----------***---------+----------++</span>
<span class="pl-s">        +           +           +           +           +      *<span class="pl-pds">'</span></span>totalmissing<span class="pl-s"><span class="pl-pds">'</span> using (bin($1,binwidth)):(1.0) ****** +</span>
<span class="pl-s">        |                                                      *  *            * *           * *                     |</span>
<span class="pl-s">        |                                                      *  *            * *           * *                     |</span>
<span class="pl-s">        |                                                      *  *            * *           * *                     |</span>
<span class="pl-s">        |                                                      *  *            * *           * *                     |</span>
<span class="pl-s">    2.5 ++                                                     *  *            * *           * *                    ++</span>
<span class="pl-s">        |                                                      *  *            * *           * *                     |</span>
<span class="pl-s">        |                                                      *  *            * *           * *                     |</span>
<span class="pl-s">        |                                                      *  *            * *           * *                     |</span>
<span class="pl-s">        |                                                      *  *            * *           * *                     |</span>
<span class="pl-s">        |                                                      *  *            * *           * *                     |</span>
<span class="pl-s">      2 ++                                       ****          *  *     ** **  * ************* *     ****           ++</span>
<span class="pl-s">        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |</span>
<span class="pl-s">        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |</span>
<span class="pl-s">        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |</span>
<span class="pl-s">        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |</span>
<span class="pl-s">        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |</span>
<span class="pl-s">    1.5 ++                                       *  *          *  *     ** **  * *    *  **  * *     *  *           ++</span>
<span class="pl-s">        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |</span>
<span class="pl-s">        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |</span>
<span class="pl-s">        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |</span>
<span class="pl-s">        |                                        *  *          *  *     ** **  * *    *  **  * *     *  *            |</span>
<span class="pl-s">        +           +           +           +    *  *   +      *  *  +  ** **  * *    *  **  * *     *  *+           +</span>
<span class="pl-s">      1 ********************************************************************************************************----++</span>
<span class="pl-s">       0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1</span>
<span class="pl-s">                                                      % of missing data</span>
<span class="pl-s"></span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day09 HW
                     </h2>
                     <a aria-label="Permalink: Day09 HW" class="anchor" href="#day09-hw" id="user-content-day09-hw">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#day09HW - VCF to Genepop to PCAs


#1-  Run one of the following sets of prescribed filters on your mergedfastq_HEAAstrangiaAssembly_subset.vcf, note these are fairly conservative filters

##Half the class run the following
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters

##the other half run the filters we used from the paper
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters


[kcrid001@turing1 data]$ salloc
salloc: Pending job allocation 9280290
salloc: job 9280290 queued and waiting for resources
salloc: job 9280290 has been allocated resources
salloc: Granted job allocation 9280290
[kcrid001@coreV3-23-004 data]$ cd VCF/
[kcrid001@coreV3-23-004 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF
[kcrid001@coreV3-23-004 VCF]$ ls
GoodCoralGenelistForVCFSubsetter.txt             out.log
KCridEmergedfastqs.vcf                           raw.g5mac3dp3.log
mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf  raw.g5mac3dp3.recode.vcf
mergedfastq_HEAAstrangiaAssembly_subset.vcf      raw.g5mac3.log
mergedfastq_HEAAstrangiaAssembly.vcf             raw.g5mac3.recode.vcf
out.imiss                                        totalmissing
[kcrid001@coreV3-23-004 VCF]$ enable_lmod
[kcrid001@coreV3-23-004 VCF]$ module load dDocent
[kcrid001@coreV3-23-004 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf

After filtering, kept 40 out of 40 Individuals
After filtering, kept 383467 out of a possible 383467 Sites
Run Time = 6.00 seconds
[kcrid001@coreV3-23-004 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
	--recode-INFO-all
	--maf 0.015
	--max-alleles 2
	--minDP 3
	--minGQ 20
	--hwe 0.01
	--minQ 30
	--max-missing 0.5
	--out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters
	--recode
	--remove-indels

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 16843 out of a possible 383467 Sites
Run Time = 11.00 seconds
## these numbers look wrong, so I used the other subset vcf file that was in my directory

[kcrid001@coreV3-23-004 VCF]$ ls
18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.log
18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf
GoodCoralGenelistForVCFSubsetter.txt
KCridEmergedfastqs.vcf
mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf
mergedfastq_HEAAstrangiaAssembly_subset.vcf
mergedfastq_HEAAstrangiaAssembly.vcf
out.imiss
out.log
raw.g5mac3dp3.log
raw.g5mac3dp3.recode.vcf
raw.g5mac3.log
raw.g5mac3.recode.vcf
totalmissing
[kcrid001@coreV3-23-004 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf
	--recode-INFO-all
	--maf 0.015
	--max-alleles 2
	--minDP 3
	--minGQ 20
	--hwe 0.01
	--minQ 30
	--max-missing 0.5
	--out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters
	--recode
	--remove-indels

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 18718 out of a possible 432676 Sites
Run Time = 14.00 seconds

[kcrid001@coreV3-23-004 VCF]$ vcftools --vcf 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf --missing-indv

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf
	--missing-indv

After filtering, kept 40 out of 40 Individuals
Outputting Individual Missingness
After filtering, kept 18718 out of a possible 18718 Sites
Run Time = 1.00 seconds
[kcrid001@coreV3-23-004 VCF]$ ls
18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.log
18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf
GoodCoralGenelistForVCFSubsetter.txt
KCridEmergedfastqs.vcf
mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf
mergedfastq_HEAAstrangiaAssembly_subset.vcf
mergedfastq_HEAAstrangiaAssembly.vcf
out.imiss
out.log
raw.g5mac3dp3.log
raw.g5mac3dp3.recode.vcf
raw.g5mac3.log
raw.g5mac3.recode.vcf
totalmissing
[kcrid001@coreV3-23-004 VCF]$ cat out.imiss
INDV	N_DATA	N_GENOTYPES_FILTERED	N_MISS	F_MISS
RI_W_06_merged	18718	0	5674	0.303131
RI_W_07_merged	18718	0	4701	0.251149
VA_B_03_merged	18718	0	3474	0.185597
RI_W_02_merged	18718	0	7137	0.381291
RI_W_04_merged	18718	0	7811	0.417299
VA_W_09_SNP_clipped	18718	0	786	0.0419917
RI_B_08_SNP_clipped	18718	0	14130	0.754888
VA_W_08_SNP_clipped	18718	0	11806	0.63073
VA_B_08_SNP_clipped	18718	0	16264	0.868896
VA_W_02_merged	18718	0	7067	0.377551
VA_B_07_merged	18718	0	4916	0.262635
RI_B_05_merged	18718	0	2439	0.130302
VA_W_06_merged	18718	0	4639	0.247836
VA_W_04_merged	18718	0	3015	0.161075
VA_W_01_merged	18718	0	5970	0.318944
VA_B_10_SNP_clipped	18718	0	10773	0.575542
VA_B_06_merged	18718	0	3608	0.192756
VA_W_05_merged	18718	0	5904	0.315418
RI_B_09_SNP_clipped	18718	0	12981	0.693504
VA_W_10_SNP_clipped	18718	0	10890	0.581793
RI_W_08_SNP_clipped	18718	0	9193	0.491132
RI_B_06_merged	18718	0	8881	0.474463
RI_W_10_SNP_clipped	18718	0	14714	0.786088
RI_B_04_merged	18718	0	2421	0.129341
VA_W_03_merged	18718	0	4740	0.253232
RI_B_07_merged	18718	0	6586	0.351854
RI_W_05_merged	18718	0	3762	0.200983
RI_W_09_SNP_clipped	18718	0	13901	0.742654
VA_B_01_merged	18718	0	3472	0.18549
VA_B_09_SNP_clipped	18718	0	9209	0.491986
RI_B_10_SNP_clipped	18718	0	11091	0.592531
RI_W_01_merged	18718	0	7875	0.420718
RI_B_01_merged	18718	0	6067	0.324127
VA_B_04_merged	18718	0	4853	0.259269
RI_B_02_merged	18718	0	13042	0.696762
RI_W_03_merged	18718	0	1968	0.105139
VA_B_02_merged	18718	0	10252	0.547708
VA_W_07_merged	18718	0	4860	0.259643
VA_B_05_merged	18718	0	3311	0.176889
RI_B_03_merged	18718	0	9817	0.524468

#2- Make a population file containing two columns with no header, tab delimited text the first column should be the individual name and the second column the population to which that individual belongs

[kcrid001@coreV3-23-004 VCF]$ grep '#CHROM' mergedfastq_HEAAstrangiaAssembly.vcf
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	RI_W_06_merged	RI_W_07_merged	VA_B_03_merged	RI_W_02_merged	RI_W_04_merged	VA_W_09_SNP_clipped	RI_B_08_SNP_clipped	VA_W_08_SNP_clipped	VA_B_08_SNP_clipped	VA_W_02_merged	VA_B_07_merged	RI_B_05_merged	VA_W_06_merged	VA_W_04_merged	VA_W_01_merged	VA_B_10_SNP_clipped	VA_B_06_mergedVA_W_05_merged	RI_B_09_SNP_clipped	VA_W_10_SNP_clipped	RI_W_08_SNP_clipped	RI_B_06_merged	RI_W_10_SNP_clipped	RI_B_04_merged	VA_W_03_merged	RI_B_07_merged	RI_W_05_mergedRI_W_09_SNP_clipped	VA_B_01_merged	VA_B_09_SNP_clipped	RI_B_10_SNP_clipped	RI_W_01_merged	RI_B_01_merged	VA_B_04_merged	RI_B_02_merged	RI_W_03_merged	VA_B_02_merged	VA_W_07_merged	VA_B_05_merged	RI_B_03_merged
[kcrid001@coreV3-23-004 VCF]$ grep '#CHROM' mergedfastq_HEAAstrangiaAssembly.vcf &gt; samplenames.txt
[kcrid001@coreV3-23-004 VCF]$ cat samplenames.txt 
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	RI_W_06_merged	RI_W_07_merged	VA_B_03_merged	RI_W_02_merged	RI_W_04_merged	VA_W_09_SNP_clipped	RI_B_08_SNP_clipped	VA_W_08_SNP_clipped	VA_B_08_SNP_clipped	VA_W_02_merged	VA_B_07_merged	RI_B_05_merged	VA_W_06_merged	VA_W_04_merged	VA_W_01_merged	VA_B_10_SNP_clipped	VA_B_06_mergedVA_W_05_merged	RI_B_09_SNP_clipped	VA_W_10_SNP_clipped	RI_W_08_SNP_clipped	RI_B_06_merged	RI_W_10_SNP_clipped	RI_B_04_merged	VA_W_03_merged	RI_B_07_merged	RI_W_05_mergedRI_W_09_SNP_clipped	VA_B_01_merged	VA_B_09_SNP_clipped	RI_B_10_SNP_clipped	RI_W_01_merged	RI_B_01_merged	VA_B_04_merged	RI_B_02_merged	RI_W_03_merged	VA_B_02_merged	VA_W_07_merged	VA_B_05_merged	RI_B_03_merged
[kcrid001@coreV3-23-004 VCF]$ cut -f 1 out.imiss
INDV
RI_W_06_merged
RI_W_07_merged
VA_B_03_merged
RI_W_02_merged
RI_W_04_merged
VA_W_09_SNP_clipped
RI_B_08_SNP_clipped
VA_W_08_SNP_clipped
VA_B_08_SNP_clipped
VA_W_02_merged
VA_B_07_merged
RI_B_05_merged
VA_W_06_merged
VA_W_04_merged
VA_W_01_merged
VA_B_10_SNP_clipped
VA_B_06_merged
VA_W_05_merged
RI_B_09_SNP_clipped
VA_W_10_SNP_clipped
RI_W_08_SNP_clipped
RI_B_06_merged
RI_W_10_SNP_clipped
RI_B_04_merged
VA_W_03_merged
RI_B_07_merged
RI_W_05_merged
RI_W_09_SNP_clipped
VA_B_01_merged
VA_B_09_SNP_clipped
RI_B_10_SNP_clipped
RI_W_01_merged
RI_B_01_merged
VA_B_04_merged
RI_B_02_merged
RI_W_03_merged
VA_B_02_merged
VA_W_07_merged
VA_B_05_merged
RI_B_03_merged

# paste in excel and use the 'left' command to get the first 4 characters of each name. copy and paste to the cluster

[kcrid001@coreV3-23-004 VCF]$ nano popfile.txt
[kcrid001@coreV3-23-004 VCF]$ cat popfile.txt 
RI_W_06_merged	RI_W
RI_W_07_merged	RI_W
VA_B_03_merged	VA_B
RI_W_02_merged	RI_W
RI_W_04_merged	RI_W
VA_W_09_SNP_clipped	VA_W
RI_B_08_SNP_clipped	RI_B
VA_W_08_SNP_clipped	VA_W
VA_B_08_SNP_clipped	VA_B
VA_W_02_merged	VA_W
VA_B_07_merged	VA_B
RI_B_05_merged	RI_B
VA_W_06_merged	VA_W
VA_W_04_merged	VA_W
VA_W_01_merged	VA_W
VA_B_10_SNP_clipped	VA_B
VA_B_06_merged	VA_B
VA_W_05_merged	VA_W
RI_B_09_SNP_clipped	RI_B
VA_W_10_SNP_clipped	VA_W
RI_W_08_SNP_clipped	RI_W
RI_B_06_merged	RI_B
RI_W_10_SNP_clipped	RI_W
RI_B_04_merged	RI_B
VA_W_03_merged	VA_W
RI_B_07_merged	RI_B
RI_W_05_merged	RI_W
RI_W_09_SNP_clipped	RI_W
VA_B_01_merged	VA_B
VA_B_09_SNP_clipped	VA_B
RI_B_10_SNP_clipped	RI_B
RI_W_01_merged	RI_W
RI_B_01_merged	RI_B
VA_B_04_merged	VA_B
RI_B_02_merged	RI_B
RI_W_03_merged	RI_W
VA_B_02_merged	VA_B
VA_W_07_merged	VA_W
VA_B_05_merged	VA_B
RI_B_03_merged	RI_B


#3- Convert your filtered .vcf file to genepop format using the following command:

/cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py YOURFILTERED.vcf YOUR_PopFile.txt

[kcrid001@coreV3-23-004 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF
[kcrid001@coreV3-23-004 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf popfile.txt
Indivs with genotypes in vcf file: RI_W_06_merged	RI_W_07_merged	VA_B_03_merged	RI_W_02_merged	RI_W_04_merged	VA_W_09_SNP_clipped	RI_B_08_SNP_clipped	VA_W_08_SNP_clipped	VA_B_08_SNP_clipped	VA_W_02_merged	VA_B_07_merged	RI_B_05_merged	VA_W_06_merged	VA_W_04_merged	VA_W_01_merged	VA_B_10_SNP_clipped	VA_B_06_merged	VA_W_05_merged	RI_B_09_SNP_clipped	VA_W_10_SNP_clipped	RI_W_08_SNP_clipped	RI_B_06_merged	RI_W_10_SNP_clipped	RI_B_04_merged	VA_W_03_merged	RI_B_07_merged	RI_W_05_merged	RI_W_09_SNP_clipped	VA_B_01_merged	VA_B_09_SNP_clipped	RI_B_10_SNP_clipped	RI_W_01_merged	RI_B_01_merged	VA_B_04_merged	RI_B_02_merged	RI_W_03_merged	VA_B_02_merged	VA_W_07_merged	VA_B_05_merged	RI_B_03_merged
44 18718 18718 18718 18718 40

#4- SCP your YOURFILE_allfilters.recode_subset_genepop.gen file to your laptop
(base) Katies-MacBook-Air-4:day09 katiecrider$ pwd
/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day09
(base) Katies-MacBook-Air-4:day09 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF/18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen ./

#5- Switch to the adegenet_PCAs.R script and follow through the steps to produce some of the figures.

install.packages(&quot;deldir&quot;)
library(&quot;adegenet&quot;)
library(&quot;ape&quot;)
library(&quot;pegas&quot;)
library(&quot;seqinr&quot;)
library(&quot;ggplot2&quot;)
library(&quot;viridis&quot;)
library(&quot;scales&quot;)
library(&quot;factoextra&quot;)

#set working directory
setwd(&quot;/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day09&quot;)

datafile&lt;-read.genepop('18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen', ncode=2)


sum(is.na(datafile$tab))
datafile #shows info
YOURdata&lt;-scaleGen(datafile, NA.method='mean')
X&lt;-YOURdata
Y&lt;-as.factor(substring(pop(datafile),1,4)) #change the second number here to change how many letters/numbers you keep in the name
pca1 &lt;- dudi.pca(X,cent=T, scale=T, scannf=F, nf=3)
summary(pca1)


#### PCAs ####
# individual labels
s.label(pca1$li)

# population elipses
s.class(pca1$li, pop(datafile))

#color symbols, pop names
#pdf(&quot;YOURINITIALS_ColorPCA1v2.pdf&quot;)
col &lt;- c(&quot;blue&quot;,&quot;red&quot;, &quot;green&quot;, &quot;black&quot;)
s.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)
add.scatter.eig(pca1$eig[1:3], 3,1,2, posi=&quot;topright&quot;)
title(&quot;PCA of DJB_data\naxes 1-2&quot;)
#dev.off()

#### Snapclust ####
a.clust&lt;-snapclust(datafile, k = 4)
class(a.clust)
names(a.clust)
a.tab &lt;- table(pop(datafile), a.clust$group)
table.value(a.tab, col.labels = 1:2)

compoplot(a.clust)
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>day09HW - VCF to Genepop to PCAs</span>


<span class="pl-c"><span class="pl-c">#</span>1-  Run one of the following sets of prescribed filters on your mergedfastq_HEAAstrangiaAssembly_subset.vcf, note these are fairly conservative filters</span>

<span class="pl-c"><span class="pl-c">#</span>#Half the class run the following</span>
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters

<span class="pl-c"><span class="pl-c">#</span>#the other half run the filters we used from the paper</span>
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters


[kcrid001@turing1 data]$ salloc
salloc: Pending job allocation 9280290
salloc: job 9280290 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9280290 has been allocated resources
salloc: Granted job allocation 9280290
[kcrid001@coreV3-23-004 data]$ <span class="pl-c1">cd</span> VCF/
[kcrid001@coreV3-23-004 VCF]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF
[kcrid001@coreV3-23-004 VCF]$ ls
GoodCoralGenelistForVCFSubsetter.txt             out.log
KCridEmergedfastqs.vcf                           raw.g5mac3dp3.log
mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf  raw.g5mac3dp3.recode.vcf
mergedfastq_HEAAstrangiaAssembly_subset.vcf      raw.g5mac3.log
mergedfastq_HEAAstrangiaAssembly.vcf             raw.g5mac3.recode.vcf
out.imiss                                        totalmissing
[kcrid001@coreV3-23-004 VCF]$ enable_lmod
[kcrid001@coreV3-23-004 VCF]$ module load dDocent
[kcrid001@coreV3-23-004 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf

After filtering, kept 40 out of 40 Individuals
After filtering, kept 383467 out of a possible 383467 Sites
Run Time = 6.00 seconds
[kcrid001@coreV3-23-004 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
	--recode-INFO-all
	--maf 0.015
	--max-alleles 2
	--minDP 3
	--minGQ 20
	--hwe 0.01
	--minQ 30
	--max-missing 0.5
	--out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters
	--recode
	--remove-indels

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 16843 out of a possible 383467 Sites
Run Time = 11.00 seconds
<span class="pl-c"><span class="pl-c">#</span># these numbers look wrong, so I used the other subset vcf file that was in my directory</span>

[kcrid001@coreV3-23-004 VCF]$ ls
18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.log
18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf
GoodCoralGenelistForVCFSubsetter.txt
KCridEmergedfastqs.vcf
mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf
mergedfastq_HEAAstrangiaAssembly_subset.vcf
mergedfastq_HEAAstrangiaAssembly.vcf
out.imiss
out.log
raw.g5mac3dp3.log
raw.g5mac3dp3.recode.vcf
raw.g5mac3.log
raw.g5mac3.recode.vcf
totalmissing
[kcrid001@coreV3-23-004 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf --maf 0.015 --max-alleles 2 --max-missing 0.5 --minQ 30 --minGQ 20 --minDP 3 --remove-indels --hwe 0.01 --recode --recode-INFO-all --out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters

VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf
	--recode-INFO-all
	--maf 0.015
	--max-alleles 2
	--minDP 3
	--minGQ 20
	--hwe 0.01
	--minQ 30
	--max-missing 0.5
	--out 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters
	--recode
	--remove-indels

After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 18718 out of a possible 432676 Sites
Run Time = 14.00 seconds

[kcrid001@coreV3-23-004 VCF]$ vcftools --vcf 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf --missing-indv

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf
	--missing-indv

After filtering, kept 40 out of 40 Individuals
Outputting Individual Missingness
After filtering, kept 18718 out of a possible 18718 Sites
Run Time = 1.00 seconds
[kcrid001@coreV3-23-004 VCF]$ ls
18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.log
18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf
GoodCoralGenelistForVCFSubsetter.txt
KCridEmergedfastqs.vcf
mergedfastq_HEAAstrangiaAssembly_subset18sp.vcf
mergedfastq_HEAAstrangiaAssembly_subset.vcf
mergedfastq_HEAAstrangiaAssembly.vcf
out.imiss
out.log
raw.g5mac3dp3.log
raw.g5mac3dp3.recode.vcf
raw.g5mac3.log
raw.g5mac3.recode.vcf
totalmissing
[kcrid001@coreV3-23-004 VCF]$ cat out.imiss
INDV	N_DATA	N_GENOTYPES_FILTERED	N_MISS	F_MISS
RI_W_06_merged	18718	0	5674	0.303131
RI_W_07_merged	18718	0	4701	0.251149
VA_B_03_merged	18718	0	3474	0.185597
RI_W_02_merged	18718	0	7137	0.381291
RI_W_04_merged	18718	0	7811	0.417299
VA_W_09_SNP_clipped	18718	0	786	0.0419917
RI_B_08_SNP_clipped	18718	0	14130	0.754888
VA_W_08_SNP_clipped	18718	0	11806	0.63073
VA_B_08_SNP_clipped	18718	0	16264	0.868896
VA_W_02_merged	18718	0	7067	0.377551
VA_B_07_merged	18718	0	4916	0.262635
RI_B_05_merged	18718	0	2439	0.130302
VA_W_06_merged	18718	0	4639	0.247836
VA_W_04_merged	18718	0	3015	0.161075
VA_W_01_merged	18718	0	5970	0.318944
VA_B_10_SNP_clipped	18718	0	10773	0.575542
VA_B_06_merged	18718	0	3608	0.192756
VA_W_05_merged	18718	0	5904	0.315418
RI_B_09_SNP_clipped	18718	0	12981	0.693504
VA_W_10_SNP_clipped	18718	0	10890	0.581793
RI_W_08_SNP_clipped	18718	0	9193	0.491132
RI_B_06_merged	18718	0	8881	0.474463
RI_W_10_SNP_clipped	18718	0	14714	0.786088
RI_B_04_merged	18718	0	2421	0.129341
VA_W_03_merged	18718	0	4740	0.253232
RI_B_07_merged	18718	0	6586	0.351854
RI_W_05_merged	18718	0	3762	0.200983
RI_W_09_SNP_clipped	18718	0	13901	0.742654
VA_B_01_merged	18718	0	3472	0.18549
VA_B_09_SNP_clipped	18718	0	9209	0.491986
RI_B_10_SNP_clipped	18718	0	11091	0.592531
RI_W_01_merged	18718	0	7875	0.420718
RI_B_01_merged	18718	0	6067	0.324127
VA_B_04_merged	18718	0	4853	0.259269
RI_B_02_merged	18718	0	13042	0.696762
RI_W_03_merged	18718	0	1968	0.105139
VA_B_02_merged	18718	0	10252	0.547708
VA_W_07_merged	18718	0	4860	0.259643
VA_B_05_merged	18718	0	3311	0.176889
RI_B_03_merged	18718	0	9817	0.524468

<span class="pl-c"><span class="pl-c">#</span>2- Make a population file containing two columns with no header, tab delimited text the first column should be the individual name and the second column the population to which that individual belongs</span>

[kcrid001@coreV3-23-004 VCF]$ grep <span class="pl-s"><span class="pl-pds">'</span>#CHROM<span class="pl-pds">'</span></span> mergedfastq_HEAAstrangiaAssembly.vcf
<span class="pl-c"><span class="pl-c">#</span>CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	RI_W_06_merged	RI_W_07_merged	VA_B_03_merged	RI_W_02_merged	RI_W_04_merged	VA_W_09_SNP_clipped	RI_B_08_SNP_clipped	VA_W_08_SNP_clipped	VA_B_08_SNP_clipped	VA_W_02_merged	VA_B_07_merged	RI_B_05_merged	VA_W_06_merged	VA_W_04_merged	VA_W_01_merged	VA_B_10_SNP_clipped	VA_B_06_mergedVA_W_05_merged	RI_B_09_SNP_clipped	VA_W_10_SNP_clipped	RI_W_08_SNP_clipped	RI_B_06_merged	RI_W_10_SNP_clipped	RI_B_04_merged	VA_W_03_merged	RI_B_07_merged	RI_W_05_mergedRI_W_09_SNP_clipped	VA_B_01_merged	VA_B_09_SNP_clipped	RI_B_10_SNP_clipped	RI_W_01_merged	RI_B_01_merged	VA_B_04_merged	RI_B_02_merged	RI_W_03_merged	VA_B_02_merged	VA_W_07_merged	VA_B_05_merged	RI_B_03_merged</span>
[kcrid001@coreV3-23-004 VCF]$ grep <span class="pl-s"><span class="pl-pds">'</span>#CHROM<span class="pl-pds">'</span></span> mergedfastq_HEAAstrangiaAssembly.vcf <span class="pl-k">&gt;</span> samplenames.txt
[kcrid001@coreV3-23-004 VCF]$ cat samplenames.txt 
<span class="pl-c"><span class="pl-c">#</span>CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	RI_W_06_merged	RI_W_07_merged	VA_B_03_merged	RI_W_02_merged	RI_W_04_merged	VA_W_09_SNP_clipped	RI_B_08_SNP_clipped	VA_W_08_SNP_clipped	VA_B_08_SNP_clipped	VA_W_02_merged	VA_B_07_merged	RI_B_05_merged	VA_W_06_merged	VA_W_04_merged	VA_W_01_merged	VA_B_10_SNP_clipped	VA_B_06_mergedVA_W_05_merged	RI_B_09_SNP_clipped	VA_W_10_SNP_clipped	RI_W_08_SNP_clipped	RI_B_06_merged	RI_W_10_SNP_clipped	RI_B_04_merged	VA_W_03_merged	RI_B_07_merged	RI_W_05_mergedRI_W_09_SNP_clipped	VA_B_01_merged	VA_B_09_SNP_clipped	RI_B_10_SNP_clipped	RI_W_01_merged	RI_B_01_merged	VA_B_04_merged	RI_B_02_merged	RI_W_03_merged	VA_B_02_merged	VA_W_07_merged	VA_B_05_merged	RI_B_03_merged</span>
[kcrid001@coreV3-23-004 VCF]$ cut -f 1 out.imiss
INDV
RI_W_06_merged
RI_W_07_merged
VA_B_03_merged
RI_W_02_merged
RI_W_04_merged
VA_W_09_SNP_clipped
RI_B_08_SNP_clipped
VA_W_08_SNP_clipped
VA_B_08_SNP_clipped
VA_W_02_merged
VA_B_07_merged
RI_B_05_merged
VA_W_06_merged
VA_W_04_merged
VA_W_01_merged
VA_B_10_SNP_clipped
VA_B_06_merged
VA_W_05_merged
RI_B_09_SNP_clipped
VA_W_10_SNP_clipped
RI_W_08_SNP_clipped
RI_B_06_merged
RI_W_10_SNP_clipped
RI_B_04_merged
VA_W_03_merged
RI_B_07_merged
RI_W_05_merged
RI_W_09_SNP_clipped
VA_B_01_merged
VA_B_09_SNP_clipped
RI_B_10_SNP_clipped
RI_W_01_merged
RI_B_01_merged
VA_B_04_merged
RI_B_02_merged
RI_W_03_merged
VA_B_02_merged
VA_W_07_merged
VA_B_05_merged
RI_B_03_merged

<span class="pl-c"><span class="pl-c">#</span> paste in excel and use the 'left' command to get the first 4 characters of each name. copy and paste to the cluster</span>

[kcrid001@coreV3-23-004 VCF]$ nano popfile.txt
[kcrid001@coreV3-23-004 VCF]$ cat popfile.txt 
RI_W_06_merged	RI_W
RI_W_07_merged	RI_W
VA_B_03_merged	VA_B
RI_W_02_merged	RI_W
RI_W_04_merged	RI_W
VA_W_09_SNP_clipped	VA_W
RI_B_08_SNP_clipped	RI_B
VA_W_08_SNP_clipped	VA_W
VA_B_08_SNP_clipped	VA_B
VA_W_02_merged	VA_W
VA_B_07_merged	VA_B
RI_B_05_merged	RI_B
VA_W_06_merged	VA_W
VA_W_04_merged	VA_W
VA_W_01_merged	VA_W
VA_B_10_SNP_clipped	VA_B
VA_B_06_merged	VA_B
VA_W_05_merged	VA_W
RI_B_09_SNP_clipped	RI_B
VA_W_10_SNP_clipped	VA_W
RI_W_08_SNP_clipped	RI_W
RI_B_06_merged	RI_B
RI_W_10_SNP_clipped	RI_W
RI_B_04_merged	RI_B
VA_W_03_merged	VA_W
RI_B_07_merged	RI_B
RI_W_05_merged	RI_W
RI_W_09_SNP_clipped	RI_W
VA_B_01_merged	VA_B
VA_B_09_SNP_clipped	VA_B
RI_B_10_SNP_clipped	RI_B
RI_W_01_merged	RI_W
RI_B_01_merged	RI_B
VA_B_04_merged	VA_B
RI_B_02_merged	RI_B
RI_W_03_merged	RI_W
VA_B_02_merged	VA_B
VA_W_07_merged	VA_W
VA_B_05_merged	VA_B
RI_B_03_merged	RI_B


<span class="pl-c"><span class="pl-c">#</span>3- Convert your filtered .vcf file to genepop format using the following command:</span>

/cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py YOURFILTERED.vcf YOUR_PopFile.txt

[kcrid001@coreV3-23-004 VCF]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF
[kcrid001@coreV3-23-004 VCF]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode.vcf popfile.txt
Indivs with genotypes <span class="pl-k">in</span> vcf file: RI_W_06_merged	RI_W_07_merged	VA_B_03_merged	RI_W_02_merged	RI_W_04_merged	VA_W_09_SNP_clipped	RI_B_08_SNP_clipped	VA_W_08_SNP_clipped	VA_B_08_SNP_clipped	VA_W_02_merged	VA_B_07_merged	RI_B_05_merged	VA_W_06_merged	VA_W_04_merged	VA_W_01_merged	VA_B_10_SNP_clipped	VA_B_06_merged	VA_W_05_merged	RI_B_09_SNP_clipped	VA_W_10_SNP_clipped	RI_W_08_SNP_clipped	RI_B_06_merged	RI_W_10_SNP_clipped	RI_B_04_merged	VA_W_03_merged	RI_B_07_merged	RI_W_05_merged	RI_W_09_SNP_clipped	VA_B_01_merged	VA_B_09_SNP_clipped	RI_B_10_SNP_clipped	RI_W_01_merged	RI_B_01_merged	VA_B_04_merged	RI_B_02_merged	RI_W_03_merged	VA_B_02_merged	VA_W_07_merged	VA_B_05_merged	RI_B_03_merged
44 18718 18718 18718 18718 40

<span class="pl-c"><span class="pl-c">#</span>4- SCP your YOURFILE_allfilters.recode_subset_genepop.gen file to your laptop</span>
(base) Katies-MacBook-Air-4:day09 katiecrider$ <span class="pl-c1">pwd</span>
/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day09
(base) Katies-MacBook-Air-4:day09 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/VCF/18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen ./

<span class="pl-c"><span class="pl-c">#</span>5- Switch to the adegenet_PCAs.R script and follow through the steps to produce some of the figures.</span>

install.packages(<span class="pl-s"><span class="pl-pds">"</span>deldir<span class="pl-pds">"</span></span>)
library(<span class="pl-s"><span class="pl-pds">"</span>adegenet<span class="pl-pds">"</span></span>)
library(<span class="pl-s"><span class="pl-pds">"</span>ape<span class="pl-pds">"</span></span>)
library(<span class="pl-s"><span class="pl-pds">"</span>pegas<span class="pl-pds">"</span></span>)
library(<span class="pl-s"><span class="pl-pds">"</span>seqinr<span class="pl-pds">"</span></span>)
library(<span class="pl-s"><span class="pl-pds">"</span>ggplot2<span class="pl-pds">"</span></span>)
library(<span class="pl-s"><span class="pl-pds">"</span>viridis<span class="pl-pds">"</span></span>)
library(<span class="pl-s"><span class="pl-pds">"</span>scales<span class="pl-pds">"</span></span>)
library(<span class="pl-s"><span class="pl-pds">"</span>factoextra<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span>set working directory</span>
setwd(<span class="pl-s"><span class="pl-pds">"</span>/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day09<span class="pl-pds">"</span></span>)

datafile<span class="pl-k">&lt;</span>-read.genepop(<span class="pl-s"><span class="pl-pds">'</span>18718_mergedfastq_HEAAstrangiaAssembly_subset_ClassFilters.recode_genepop.gen<span class="pl-pds">'</span></span>, ncode=2)


sum(is.na(datafile<span class="pl-smi">$tab</span>))
datafile <span class="pl-c"><span class="pl-c">#</span>shows info</span>
YOURdata<span class="pl-k">&lt;</span>-scaleGen(datafile, NA.method=<span class="pl-s"><span class="pl-pds">'</span>mean<span class="pl-pds">'</span></span>)
X<span class="pl-k">&lt;</span>-YOURdata
Y<span class="pl-k">&lt;</span>-as.factor(substring(pop(datafile),1,4)) <span class="pl-c"><span class="pl-c">#</span>change the second number here to change how many letters/numbers you keep in the name</span>
pca1 <span class="pl-k">&lt;</span>- dudi.pca(X,cent=T, scale=T, scannf=F, nf=3)
summary(pca1)


<span class="pl-c"><span class="pl-c">#</span>### PCAs ####</span>
<span class="pl-c"><span class="pl-c">#</span> individual labels</span>
s.label(pca1<span class="pl-smi">$li</span>)

<span class="pl-c"><span class="pl-c">#</span> population elipses</span>
s.class(pca1<span class="pl-smi">$li</span>, pop(datafile))

<span class="pl-c"><span class="pl-c">#</span>color symbols, pop names</span>
<span class="pl-c"><span class="pl-c">#</span>pdf("YOURINITIALS_ColorPCA1v2.pdf")</span>
col <span class="pl-k">&lt;</span>- c(<span class="pl-s"><span class="pl-pds">"</span>blue<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>green<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>)
s.class(pca1<span class="pl-smi">$li</span>, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)
add.scatter.eig(pca1<span class="pl-smi">$eig</span>[1:3], 3,1,2, posi=<span class="pl-s"><span class="pl-pds">"</span>topright<span class="pl-pds">"</span></span>)
title(<span class="pl-s"><span class="pl-pds">"</span>PCA of DJB_data\naxes 1-2<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span>dev.off()</span>

<span class="pl-c"><span class="pl-c">#</span>### Snapclust ####</span>
a.clust<span class="pl-k">&lt;</span>-snapclust(datafile, k = 4)
class(a.clust)
names(a.clust)
a.tab <span class="pl-k">&lt;</span>- table(pop(datafile), a.clust<span class="pl-smi">$group</span>)
table.value(a.tab, col.labels = 1:2)

compoplot(a.clust)
</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day10 HW
                     </h2>
                     <a aria-label="Permalink: Day10 HW" class="anchor" href="#day10-hw" id="user-content-day10-hw">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#1- Work through the adegenet_PCAs.R script and follow through the steps to produce some of the figures.

setwd(&quot;/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day09&quot;)
datafile&lt;-read.genepop('coral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen', ncode=2)
#follow the same scripts as day09 #5, with the other genepop files

###run step 2 as an sbatch .sh script because it will take a while to finish
#2- cd into your SAMS folder containing your .sams and run the following as an sbatch script on your sam files to generate read mapping counts from each individual file:
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam

[kcrid001@turing1 SAMS]$ salloc
salloc: Pending job allocation 9280850
salloc: job 9280850 queued and waiting for resources
salloc: job 9280850 has been allocated resources
salloc: Granted job allocation 9280850
[kcrid001@coreV2-25-015 SAMS]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/SAMS
[kcrid001@coreV2-25-015 SAMS]$ nano SAMSreadmap.sh
[kcrid001@coreV2-25-015 SAMS]$ cat SAMSreadmap.sh 
#!/bin/bash -l

#SBATCH -o SAMSreadmap.txt
#SBATCH -n 1
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=SAMSreadmap


/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam
[kcrid001@coreV2-25-015 SAMS]$ sbatch SAMSreadmap.sh 
Submitted batch job 9280858
[kcrid001@coreV2-25-015 SAMS]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9280858      main SAMSread kcrid001  R       0:07      1 coreV1-22-016 
           9280850      main       sh kcrid001  R    1:08:38      1 coreV2-25-015 


#3- Once your job from step 1 is finished, start an salloc session and run the following on your outputted _counts.txt files:

/cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt KCridFullCounts_summed.txt NoMatch *_counts.txt


[kcrid001@coreV2-25-015 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt KCridFullCounts_summed.txt NoMatch *_counts.txt
Hits not matchedRI_B_04_14_clippedtrimmed.fastq_counts.txt=1698	RI_B_04_18_clippedtrimmed.fastq_counts.txt=1698	RI_B_04_22_clippedtrimmed.fastq_counts.txt=1698	RI_B_05_14_clippedtrimmed.fastq_counts.txt=1698	RI_W_04_14_clippedtrimmed.fastq_counts.txt=1698	RI_W_04_18_clippedtrimmed.fastq_counts.txt=1698	RI_W_04_22_clippedtrimmed.fastq_counts.txt=1698	RI_W_05_14_clippedtrimmed.fastq_counts.txt=1698	VA_B_04_14_clippedtrimmed.fastq_counts.txt=1698	VA_B_04_18_clippedtrimmed.fastq_counts.txt=1698	VA_B_04_22_clippedtrimmed.fastq_counts.txt=1698	VA_B_05_14_clippedtrimmed.fastq_counts.txt=1698	VA_W_04_14_clippedtrimmed.fastq_counts.txt=1698	VA_W_04_18_clippedtrimmed.fastq_counts.txt=1698	VA_W_04_22_clippedtrimmed.fastq_counts.txt=1698	VA_W_05_14_clippedtrimmed.fastq_counts.txt=1698

#4- scp YOURNAMEFullCounts_summed.txt to your laptop

(base) Katies-MacBook-Air-4:day10 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/SAMS/KCridFullCounts_summed.txt ./
kcrid001@turing.hpc.odu.edu's password: 
KCridFullCounts_summed.txt                              100%  885KB   5.4MB/s   00:00    
(base) Katies-MacBook-Air-4:day10 katiecrider$ ls
KCridFullCounts_summed.txt
coral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen
coral_66_cloneremoved_highoutliers.filtered1SNPper_genepop.gen
homework_day10.txt
(base) Katies-MacBook-Air-4:day10 katiecrider$ pwd
/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day10

#5- edit the first line of YOURNAMEFullCounts_summed.txt to remove the _counts.txt_UniqueTotReads from each sample name to just retain the actual informative part of the sample name (e.g., RI_W_06_18)
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>1- Work through the adegenet_PCAs.R script and follow through the steps to produce some of the figures.</span>

setwd(<span class="pl-s"><span class="pl-pds">"</span>/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day09<span class="pl-pds">"</span></span>)
datafile<span class="pl-k">&lt;</span>-read.genepop(<span class="pl-s"><span class="pl-pds">'</span>coral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen<span class="pl-pds">'</span></span>, ncode=2)
<span class="pl-c"><span class="pl-c">#</span>follow the same scripts as day09 #5, with the other genepop files</span>

<span class="pl-c"><span class="pl-c">#</span>##run step 2 as an sbatch .sh script because it will take a while to finish</span>
<span class="pl-c"><span class="pl-c">#</span>2- cd into your SAMS folder containing your .sams and run the following as an sbatch script on your sam files to generate read mapping counts from each individual file:</span>
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py <span class="pl-k">*</span>.sam

[kcrid001@turing1 SAMS]$ salloc
salloc: Pending job allocation 9280850
salloc: job 9280850 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9280850 has been allocated resources
salloc: Granted job allocation 9280850
[kcrid001@coreV2-25-015 SAMS]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/SAMS
[kcrid001@coreV2-25-015 SAMS]$ nano SAMSreadmap.sh
[kcrid001@coreV2-25-015 SAMS]$ cat SAMSreadmap.sh 
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o SAMSreadmap.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=SAMSreadmap</span>


/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py <span class="pl-k">*</span>.sam
[kcrid001@coreV2-25-015 SAMS]$ sbatch SAMSreadmap.sh 
Submitted batch job 9280858
[kcrid001@coreV2-25-015 SAMS]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9280858      main SAMSread kcrid001  R       0:07      1 coreV1-22-016 
           9280850      main       sh kcrid001  R    1:08:38      1 coreV2-25-015 


<span class="pl-c"><span class="pl-c">#</span>3- Once your job from step 1 is finished, start an salloc session and run the following on your outputted _counts.txt files:</span>

/cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt KCridFullCounts_summed.txt NoMatch <span class="pl-k">*</span>_counts.txt


[kcrid001@coreV2-25-015 SAMS]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt KCridFullCounts_summed.txt NoMatch <span class="pl-k">*</span>_counts.txt
Hits not matchedRI_B_04_14_clippedtrimmed.fastq_counts.txt=1698	RI_B_04_18_clippedtrimmed.fastq_counts.txt=1698	RI_B_04_22_clippedtrimmed.fastq_counts.txt=1698	RI_B_05_14_clippedtrimmed.fastq_counts.txt=1698	RI_W_04_14_clippedtrimmed.fastq_counts.txt=1698	RI_W_04_18_clippedtrimmed.fastq_counts.txt=1698	RI_W_04_22_clippedtrimmed.fastq_counts.txt=1698	RI_W_05_14_clippedtrimmed.fastq_counts.txt=1698	VA_B_04_14_clippedtrimmed.fastq_counts.txt=1698	VA_B_04_18_clippedtrimmed.fastq_counts.txt=1698	VA_B_04_22_clippedtrimmed.fastq_counts.txt=1698	VA_B_05_14_clippedtrimmed.fastq_counts.txt=1698	VA_W_04_14_clippedtrimmed.fastq_counts.txt=1698	VA_W_04_18_clippedtrimmed.fastq_counts.txt=1698	VA_W_04_22_clippedtrimmed.fastq_counts.txt=1698	VA_W_05_14_clippedtrimmed.fastq_counts.txt=1698

<span class="pl-c"><span class="pl-c">#</span>4- scp YOURNAMEFullCounts_summed.txt to your laptop</span>

(base) Katies-MacBook-Air-4:day10 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/katiecrider/data/SAMS/KCridFullCounts_summed.txt ./
kcrid001@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password: </span>
<span class="pl-s">KCridFullCounts_summed.txt                              100%  885KB   5.4MB/s   00:00    </span>
<span class="pl-s">(base) Katies-MacBook-Air-4:day10 katiecrider$ ls</span>
<span class="pl-s">KCridFullCounts_summed.txt</span>
<span class="pl-s">coral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen</span>
<span class="pl-s">coral_66_cloneremoved_highoutliers.filtered1SNPper_genepop.gen</span>
<span class="pl-s">homework_day10.txt</span>
<span class="pl-s">(base) Katies-MacBook-Air-4:day10 katiecrider$ pwd</span>
<span class="pl-s">/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day10</span>
<span class="pl-s"></span>
<span class="pl-s">#5- edit the first line of YOURNAMEFullCounts_summed.txt to remove the _counts.txt_UniqueTotReads from each sample name to just retain the actual informative part of the sample name (e.g., RI_W_06_18)</span>
<span class="pl-s"></span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day11 HW
                     </h2>
                     <a aria-label="Permalink: Day11 HW" class="anchor" href="#day11-hw" id="user-content-day11-hw">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
#1- Download the DESeq2Script_advbioinf.R script and work through on the shared djbFullCounts_summed.txt file

#/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day11/DESeq2Script_advbioinf.R

#copy this and the summed.txt file to the Genomics folder

#make sure to have downloaded these packages already

&gt;library(DESeq2)
&gt;library(gplots)
&gt;vignette('DESeq2')

&gt;setwd(&quot;/Users/katiecrider/Desktop/Genomics&quot;)  # The drag and drop from finder works in R, too.
&gt; conds&lt;-data.frame(&quot;Sample&quot;=names(countsTable))
&gt; conds$Origin&lt;-factor(substring(conds$Sample,1,2))
&gt; conds$SymState&lt;-factor(substring(conds$Sample,4,4))
&gt; conds$Temp&lt;-factor(substring(conds$Sample,9,10))
&gt; conds$Origin_SymState&lt;-factor(paste(conds$Origin,conds$SymState, sep=&quot;_&quot;))
&gt; conds$Origin_SymState_Temp&lt;-factor(paste(conds$Origin,conds$SymState,conds$Temp, sep=&quot;_&quot;))
&gt; conds
       Sample Origin SymState Temp Origin_SymState Origin_SymState_Temp
1  RI_B_01_14     RI        B   14            RI_B              RI_B_14
2  RI_B_01_18     RI        B   18            RI_B              RI_B_18
3  RI_B_01_22     RI        B   22            RI_B              RI_B_22
4  RI_B_02_14     RI        B   14            RI_B              RI_B_14
5  RI_B_02_18     RI        B   18            RI_B              RI_B_18
6  RI_B_02_22     RI        B   22            RI_B              RI_B_22
7  RI_B_03_14     RI        B   14            RI_B              RI_B_14
8  RI_B_03_22     RI        B   22            RI_B              RI_B_22
9  RI_B_04_14     RI        B   14            RI_B              RI_B_14
10 RI_B_04_18     RI        B   18            RI_B              RI_B_18
11 RI_B_04_22     RI        B   22            RI_B              RI_B_22
12 RI_B_05_14     RI        B   14            RI_B              RI_B_14
13 RI_B_05_18     RI        B   18            RI_B              RI_B_18
14 RI_B_05_22     RI        B   22            RI_B              RI_B_22
15 RI_B_06_14     RI        B   14            RI_B              RI_B_14
16 RI_B_06_18     RI        B   18            RI_B              RI_B_18
17 RI_B_06_22     RI        B   22            RI_B              RI_B_22
18 RI_B_07_14     RI        B   14            RI_B              RI_B_14
19 RI_B_07_18     RI        B   18            RI_B              RI_B_18
20 RI_B_07_22     RI        B   22            RI_B              RI_B_22
21 RI_W_01_14     RI        W   14            RI_W              RI_W_14
22 RI_W_01_18     RI        W   18            RI_W              RI_W_18
23 RI_W_01_22     RI        W   22            RI_W              RI_W_22
24 RI_W_02_14     RI        W   14            RI_W              RI_W_14
25 RI_W_02_18     RI        W   18            RI_W              RI_W_18
26 RI_W_02_22     RI        W   22            RI_W              RI_W_22
27 RI_W_03_14     RI        W   14            RI_W              RI_W_14
28 RI_W_03_18     RI        W   18            RI_W              RI_W_18
29 RI_W_03_22     RI        W   22            RI_W              RI_W_22
30 RI_W_04_14     RI        W   14            RI_W              RI_W_14
31 RI_W_04_18     RI        W   18            RI_W              RI_W_18
32 RI_W_04_22     RI        W   22            RI_W              RI_W_22
33 RI_W_05_14     RI        W   14            RI_W              RI_W_14
34 RI_W_05_18     RI        W   18            RI_W              RI_W_18
35 RI_W_05_22     RI        W   22            RI_W              RI_W_22
36 RI_W_06_14     RI        W   14            RI_W              RI_W_14
37 RI_W_06_18     RI        W   18            RI_W              RI_W_18
38 RI_W_06_22     RI        W   22            RI_W              RI_W_22
39 RI_W_07_14     RI        W   14            RI_W              RI_W_14
40 RI_W_07_18     RI        W   18            RI_W              RI_W_18
41 RI_W_07_22     RI        W   22            RI_W              RI_W_22
42 VA_B_01_14     VA        B   14            VA_B              VA_B_14
43 VA_B_01_18     VA        B   18            VA_B              VA_B_18
44 VA_B_01_22     VA        B   22            VA_B              VA_B_22
45 VA_B_02_14     VA        B   14            VA_B              VA_B_14
46 VA_B_02_18     VA        B   18            VA_B              VA_B_18
47 VA_B_02_22     VA        B   22            VA_B              VA_B_22
48 VA_B_03_14     VA        B   14            VA_B              VA_B_14
49 VA_B_03_18     VA        B   18            VA_B              VA_B_18
50 VA_B_03_22     VA        B   22            VA_B              VA_B_22
51 VA_B_04_14     VA        B   14            VA_B              VA_B_14
52 VA_B_04_18     VA        B   18            VA_B              VA_B_18
53 VA_B_04_22     VA        B   22            VA_B              VA_B_22
54 VA_B_05_14     VA        B   14            VA_B              VA_B_14
55 VA_B_05_18     VA        B   18            VA_B              VA_B_18
56 VA_B_05_22     VA        B   22            VA_B              VA_B_22
57 VA_B_06_14     VA        B   14            VA_B              VA_B_14
58 VA_B_06_18     VA        B   18            VA_B              VA_B_18
59 VA_B_06_22     VA        B   22            VA_B              VA_B_22
60 VA_B_07_14     VA        B   14            VA_B              VA_B_14
61 VA_B_07_18     VA        B   18            VA_B              VA_B_18
62 VA_B_07_22     VA        B   22            VA_B              VA_B_22
63 VA_W_01_14     VA        W   14            VA_W              VA_W_14
64 VA_W_01_18     VA        W   18            VA_W              VA_W_18
65 VA_W_02_14     VA        W   14            VA_W              VA_W_14
66 VA_W_02_18     VA        W   18            VA_W              VA_W_18
67 VA_W_03_14     VA        W   14            VA_W              VA_W_14
68 VA_W_03_18     VA        W   18            VA_W              VA_W_18
69 VA_W_03_22     VA        W   22            VA_W              VA_W_22
70 VA_W_04_14     VA        W   14            VA_W              VA_W_14
71 VA_W_04_18     VA        W   18            VA_W              VA_W_18
72 VA_W_04_22     VA        W   22            VA_W              VA_W_22
73 VA_W_05_14     VA        W   14            VA_W              VA_W_14
74 VA_W_05_18     VA        W   18            VA_W              VA_W_18
75 VA_W_05_22     VA        W   22            VA_W              VA_W_22
76 VA_W_06_14     VA        W   14            VA_W              VA_W_14
77 VA_W_06_18     VA        W   18            VA_W              VA_W_18
78 VA_W_06_22     VA        W   22            VA_W              VA_W_22
79 VA_W_07_14     VA        W   14            VA_W              VA_W_14
80 VA_W_07_18     VA        W   18            VA_W              VA_W_18
81 VA_W_07_22     VA        W   22            VA_W              VA_W_22
&gt; prop.null &lt;- apply(countsTable, 2, function(x) 100*mean(x==0))
&gt; 
&gt; barplot(prop.null, main=&quot;Percentage of null counts per sample&quot;, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
&gt; 
&gt; pdf(file=&quot;OverallNullCounts.pdf&quot;,14, 14)
&gt; barplot(prop.null, main=&quot;Percentage of null counts per sample&quot;, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
&gt; dev.off()
RStudioGD 
        2 
&gt; 
&gt; # how many genes have mean count &gt;=3 
&gt; means = apply(countsTable, 1, mean)
&gt; table(means&gt;=3)

TRUE 
7543 
&gt; 
&gt; countsTable&lt;-countsTable[means&gt;=3,]
&gt; dim(countsTable)
[1] 7543   81
&gt; 
&gt; prop.nullv2 &lt;- apply(countsTable, 2, function(x) 100*mean(x==0))
&gt; pdf(file=&quot;Mean3NullCounts.pdf&quot;,14, 14)
&gt; barplot(prop.nullv2, main=&quot;Percentage of null counts per sample&quot;, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
&gt; dev.off()
RStudioGD 
        2 
&gt; 
&gt; #make count data set
&gt; dds &lt;- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=~ Origin_SymState_Temp)
&gt; 
&gt; dds &lt;- DESeq(dds)
estimating size factors
estimating dispersions
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
fitting model and testing
-- replacing outliers and refitting for 8 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)
estimating dispersions
fitting model and testing
&gt; 
&gt; #transform counts to variance stablized counts
&gt; vstCounts&lt;-varianceStabilizingTransformation(dds)
&gt; 
&gt; #plot cluster dendrogram across samples
&gt; dists&lt;-dist(t(assay(vstCounts)))
&gt; plot(hclust(dists))
&gt; 
&gt; plotPCA(vstCounts, intgroup=&quot;Origin&quot;)
&gt; plotPCA(vstCounts, intgroup=&quot;SymState&quot;)
&gt; plotPCA(vstCounts, intgroup=&quot;Temp&quot;)
&gt; plotPCA(vstCounts, intgroup=&quot;Origin_SymState&quot;)
&gt; plotPCA(vstCounts, intgroup=&quot;Origin_SymState_Temp&quot;)
&gt; resultsNames(dds)
 [1] &quot;Intercept&quot;                              
 [2] &quot;Origin_SymState_Temp_RI_B_18_vs_RI_B_14&quot;
 [3] &quot;Origin_SymState_Temp_RI_B_22_vs_RI_B_14&quot;
 [4] &quot;Origin_SymState_Temp_RI_W_14_vs_RI_B_14&quot;
 [5] &quot;Origin_SymState_Temp_RI_W_18_vs_RI_B_14&quot;
 [6] &quot;Origin_SymState_Temp_RI_W_22_vs_RI_B_14&quot;
 [7] &quot;Origin_SymState_Temp_VA_B_14_vs_RI_B_14&quot;
 [8] &quot;Origin_SymState_Temp_VA_B_18_vs_RI_B_14&quot;
 [9] &quot;Origin_SymState_Temp_VA_B_22_vs_RI_B_14&quot;
[10] &quot;Origin_SymState_Temp_VA_W_14_vs_RI_B_14&quot;
[11] &quot;Origin_SymState_Temp_VA_W_18_vs_RI_B_14&quot;
[12] &quot;Origin_SymState_Temp_VA_W_22_vs_RI_B_14&quot;
&gt; 
&gt; dim(dds)
[1] 7543   81
&gt; 
&gt; ###Figure out which contrast you want to examine (i.e. which two groups do you want to compare)
&gt; res &lt;- results(dds, contrast=c(&quot;Origin_SymState_Temp&quot;, &quot;RI_B_18&quot;, &quot;VA_B_18&quot;))
&gt; head(res)
log2 fold change (MLE): Origin_SymState_Temp RI_B_18 vs VA_B_18 
Wald test p-value: Origin_SymState_Temp RI_B_18 vs VA_B_18 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE      stat
                       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
TR10083|c0_g2_i1_coral   5.92149      -0.124352  0.881723 -0.141032
TR10090|c0_g1_i1_coral   2.41467       0.403769  0.959279  0.420909
TR10090|c0_g2_i1_coral   3.75861      -0.835423  0.865239 -0.965540
TR10103|c0_g1_i1_coral   2.98376      -0.801464  0.848308 -0.944780
TR10108|c0_g1_i1_coral   5.73440      -0.526701  0.837420 -0.628957
TR10132|c0_g1_i1_coral   8.52630       1.071572  1.160128  0.923667
                          pvalue      padj
                       &lt;numeric&gt; &lt;numeric&gt;
TR10083|c0_g2_i1_coral  0.887844  0.996375
TR10090|c0_g1_i1_coral  0.673822  0.990719
TR10090|c0_g2_i1_coral  0.334274  0.964876
TR10103|c0_g1_i1_coral  0.344771  0.967913
TR10108|c0_g1_i1_coral  0.529377  0.987729
TR10132|c0_g1_i1_coral  0.355660  0.967913
&gt; summary(res)

out of 7542 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 3, 0.04%
LFC &lt; 0 (down)     : 5, 0.066%
outliers [1]       : 4, 0.053%
low counts [2]     : 1, 0.013%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

&gt; 
&gt; res &lt;- results(dds, contrast=c(&quot;Origin_SymState_Temp&quot;, &quot;RI_W_18&quot;, &quot;VA_W_18&quot;))
&gt; summary(res)

out of 7542 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 22, 0.29%
LFC &lt; 0 (down)     : 38, 0.5%
outliers [1]       : 4, 0.053%
low counts [2]     : 3801, 50%
(mean count &lt; 6)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

&gt; 
&gt; res &lt;- results(dds, contrast=c(&quot;Origin_SymState_Temp&quot;, &quot;RI_B_14&quot;, &quot;VA_B_14&quot;))
&gt; summary(res)

out of 7542 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 8, 0.11%
LFC &lt; 0 (down)     : 4, 0.053%
outliers [1]       : 4, 0.053%
low counts [2]     : 1, 0.013%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

&gt; 
&gt; res &lt;- results(dds, contrast=c(&quot;Origin_SymState_Temp&quot;, &quot;RI_W_14&quot;, &quot;VA_W_14&quot;))
&gt; summary(res)

out of 7542 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 10, 0.13%
LFC &lt; 0 (down)     : 3, 0.04%
outliers [1]       : 4, 0.053%
low counts [2]     : 5263, 70%
(mean count &lt; 8)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

&gt; 
&gt; res &lt;- results(dds, contrast=c(&quot;Origin_SymState_Temp&quot;, &quot;RI_B_22&quot;, &quot;VA_B_22&quot;))
&gt; summary(res)

out of 7542 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 8, 0.11%
LFC &lt; 0 (down)     : 2, 0.027%
outliers [1]       : 4, 0.053%
low counts [2]     : 1, 0.013%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

&gt; 
&gt; res &lt;- results(dds, contrast=c(&quot;Origin_SymState_Temp&quot;, &quot;RI_W_22&quot;, &quot;VA_W_22&quot;))
&gt; summary(res)

out of 7542 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 11, 0.15%
LFC &lt; 0 (down)     : 5, 0.066%
outliers [1]       : 4, 0.053%
low counts [2]     : 146, 1.9%
(mean count &lt; 2)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

&gt; 
&gt; head(res[order(res$padj),])
log2 fold change (MLE): Origin_SymState_Temp RI_W_22 vs VA_W_22 
Wald test p-value: Origin_SymState_Temp RI_W_22 vs VA_W_22 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE      stat
                       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
TR42879|c0_g2_i1_coral  15.41058       17.05753  3.137478   5.43670
TR13286|c1_g1_i1_coral  32.61026       -3.38044  0.648393  -5.21357
TR47617|c0_g1_i1_coral   3.74656       19.72616  3.835159   5.14351
TR8327|c0_g3_i1_coral    8.73277       19.73384  4.063853   4.85594
TR29560|c0_g1_i1_coral  12.79556        2.74246  0.651668   4.20838
TR58072|c1_g1_i1_coral  53.85583        1.38039  0.330572   4.17576
                            pvalue        padj
                         &lt;numeric&gt;   &lt;numeric&gt;
TR42879|c0_g2_i1_coral 5.42760e-08 0.000401263
TR13286|c1_g1_i1_coral 1.85241e-07 0.000664526
TR47617|c0_g1_i1_coral 2.69658e-07 0.000664526
TR8327|c0_g3_i1_coral  1.19814e-06 0.002214469
TR29560|c0_g1_i1_coral 2.57208e-05 0.036593740
TR58072|c1_g1_i1_coral 2.96987e-05 0.036593740
&gt; plotCounts(dds, &quot;TR2367|c0_g1_i1_coral&quot;, intgroup=&quot;Origin_SymState_Temp&quot;)
&gt; 
&gt; #count the number of significantly differentially expressed genes
&gt; sum(res$padj &lt; 0.3, na.rm =T)
[1] 30
&gt; sum(res$padj &lt; 0.2, na.rm =T)
[1] 20
&gt; sum(res$padj &lt; 0.1, na.rm =T)
[1] 16
&gt; sum(res$pvalue &lt; 0.05, na.rm =T)
[1] 343
&gt; summary(res)

out of 7542 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 11, 0.15%
LFC &lt; 0 (down)     : 5, 0.066%
outliers [1]       : 4, 0.053%
low counts [2]     : 146, 1.9%
(mean count &lt; 2)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results
&gt; scaledcounts = counts(dds, normalized=T)
&gt; head(scaledcounts)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14
TR10083|c0_g2_i1_coral          0    0.84279   2.292953   0.000000
TR10090|c0_g1_i1_coral          0    2.52837   1.146477   0.000000
TR10090|c0_g2_i1_coral          0    3.37116   2.292953   3.553607
TR10103|c0_g1_i1_coral          0    0.84279   4.585907   0.000000
TR10108|c0_g1_i1_coral          0    1.68558   4.585907   0.000000
TR10132|c0_g1_i1_coral          0    5.89953   1.146477  17.768037
                       RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_22
TR10083|c0_g2_i1_coral   3.273656   4.312157  12.353943   1.269319
TR10090|c0_g1_i1_coral  13.094624   0.000000   0.000000   0.000000
TR10090|c0_g2_i1_coral   0.000000   0.000000   0.000000   3.807956
TR10103|c0_g1_i1_coral   0.000000   8.624315   1.235394   3.807956
TR10108|c0_g1_i1_coral   6.547312   0.000000  13.589337  15.231824
TR10132|c0_g1_i1_coral  49.104839   0.000000   6.176972   2.538637
                       RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10083|c0_g2_i1_coral  12.576774  8.1319863   0.000000   4.298440
TR10090|c0_g1_i1_coral   2.028512  4.0659932   6.219661   0.537305
TR10090|c0_g2_i1_coral   4.868429  3.6593938   0.000000   0.537305
TR10103|c0_g1_i1_coral   6.085536  3.2527945   0.000000   4.835745
TR10108|c0_g1_i1_coral   2.839917  4.8791918  15.549153   5.641703
TR10132|c0_g1_i1_coral   0.000000  0.8131986   3.109831   1.343263
                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18
TR10083|c0_g2_i1_coral          0  2.3823188  18.130668   3.272361
TR10090|c0_g1_i1_coral          0  0.0000000   0.000000   1.963417
TR10090|c0_g2_i1_coral          0  2.3823188   0.000000   4.581305
TR10103|c0_g1_i1_coral          0  3.9705313   0.000000   3.272361
TR10108|c0_g1_i1_coral          0  0.7941063   6.043556   9.817083
TR10132|c0_g1_i1_coral          0  1.5882125   0.000000   1.308944
                       RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22
TR10083|c0_g2_i1_coral   19.38574  1.3676909  7.9128854   36.52879
TR10090|c0_g1_i1_coral    0.00000  0.6838455  1.9782213    0.00000
TR10090|c0_g2_i1_coral    0.00000  3.4192273  0.9891107    0.00000
TR10103|c0_g1_i1_coral    0.00000  2.0515364  1.9782213    0.00000
TR10108|c0_g1_i1_coral   42.00244  3.4192273  0.9891107    0.00000
TR10132|c0_g1_i1_coral    0.00000  8.2061454 43.5208696   27.39659
                       RI_W_01_14 RI_W_01_18 RI_W_01_22 RI_W_02_14
TR10083|c0_g2_i1_coral   63.29565  10.257613   4.369327   0.000000
TR10090|c0_g1_i1_coral    0.00000  11.625295   1.456442   0.000000
TR10090|c0_g2_i1_coral    0.00000  11.625295   7.282212   4.027688
TR10103|c0_g1_i1_coral   28.77075   7.522249   0.000000   0.000000
TR10108|c0_g1_i1_coral    0.00000   2.735363   1.456442   8.055376
TR10132|c0_g1_i1_coral    0.00000   1.367682   0.000000   0.000000
                       RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18
TR10083|c0_g2_i1_coral  4.8924878   3.593465   8.078028   5.344367
TR10090|c0_g1_i1_coral  1.9569951   0.000000   4.039014   1.068873
TR10090|c0_g2_i1_coral  4.8924878   4.791286   2.885010   3.206620
TR10103|c0_g1_i1_coral  0.9784976   3.593465   2.885010   3.206620
TR10108|c0_g1_i1_coral  6.8494829   2.395643   2.308008   4.631785
TR10132|c0_g1_i1_coral  5.8709854  44.319396   1.154004   1.068873
                       RI_W_03_22 RI_W_04_14 RI_W_04_18 RI_W_04_22
TR10083|c0_g2_i1_coral  4.4092608   12.32321  2.9466826    17.8865
TR10090|c0_g1_i1_coral  1.4697536    0.00000  3.4377964     0.0000
TR10090|c0_g2_i1_coral  1.4697536    0.00000  7.3667065     0.0000
TR10103|c0_g1_i1_coral  0.9798357    0.00000  0.9822275     0.0000
TR10108|c0_g1_i1_coral  1.9596715   20.53868 12.7689580     0.0000
TR10132|c0_g1_i1_coral  0.9798357   28.75416  7.8578203     0.0000
                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14
TR10083|c0_g2_i1_coral  3.0758634  0.9718221   5.123934  0.9786798
TR10090|c0_g1_i1_coral  0.8788181  7.7745767   0.000000  0.0000000
TR10090|c0_g2_i1_coral  3.5152724  9.7182208   1.280983  2.9360395
TR10103|c0_g1_i1_coral  3.9546815  1.9436442   2.561967  9.7867983
TR10108|c0_g1_i1_coral  5.2729086  1.9436442   6.404917 10.7654782
TR10132|c0_g1_i1_coral  1.7576362  9.7182208  25.619669  4.8933992
                       RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18
TR10083|c0_g2_i1_coral    1.28024   5.416550   4.050051   5.034881
TR10090|c0_g1_i1_coral    0.00000   4.333240   0.000000   0.000000
TR10090|c0_g2_i1_coral    1.28024   7.583169   2.025025   1.006976
TR10103|c0_g1_i1_coral    1.28024   2.166620   5.400067  12.083715
TR10108|c0_g1_i1_coral    0.00000   5.416550   0.000000   1.006976
TR10132|c0_g1_i1_coral    1.28024  41.165776   1.350017   2.013953
                       RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10083|c0_g2_i1_coral   7.075440   2.275221   1.310363   2.753792
TR10090|c0_g1_i1_coral   0.000000   6.825663   6.879404   1.652275
TR10090|c0_g2_i1_coral   5.660352   9.100884   4.258679   5.507584
TR10103|c0_g1_i1_coral   1.415088   3.412832   1.965544   2.753792
TR10108|c0_g1_i1_coral   5.660352   3.412832   6.879404   2.753792
TR10132|c0_g1_i1_coral  89.150546   1.137611   0.982772   0.000000
                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14
TR10083|c0_g2_i1_coral          0   13.08723  17.568884  15.220246
TR10090|c0_g1_i1_coral          0    0.00000   0.000000   5.534635
TR10090|c0_g2_i1_coral          0    0.00000   0.000000  10.377440
TR10103|c0_g1_i1_coral          0    0.00000   1.952098   2.075488
TR10108|c0_g1_i1_coral          0    0.00000   0.000000  13.836587
TR10132|c0_g1_i1_coral          0    0.00000   7.808393   8.993782
                       VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18
TR10083|c0_g2_i1_coral  13.204264   7.514330   0.000000  0.8308866
TR10090|c0_g1_i1_coral   1.980640   7.514330   0.000000  4.1544330
TR10090|c0_g2_i1_coral   3.961279   3.415604   6.526643  2.9081031
TR10103|c0_g1_i1_coral   3.961279   3.415604   4.079152  0.0000000
TR10108|c0_g1_i1_coral  17.165543  20.493627   2.447491  4.5698763
TR10132|c0_g1_i1_coral  28.389167   8.197451   0.000000  2.4926598
                       VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10083|c0_g2_i1_coral   6.207028   1.933974  2.9919941   1.718262
TR10090|c0_g1_i1_coral   3.342246   4.512607  0.9973314   6.873047
TR10090|c0_g2_i1_coral   0.000000   1.289316  0.4986657   5.154785
TR10103|c0_g1_i1_coral   0.000000   0.000000  2.9919941   0.000000
TR10108|c0_g1_i1_coral   3.342246   8.380556  4.9866569   0.000000
TR10132|c0_g1_i1_coral   0.000000   0.000000  4.4879912   5.154785
                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14
TR10083|c0_g2_i1_coral   1.213849   0.000000  5.4126287   3.585182
TR10090|c0_g1_i1_coral   7.283093   0.000000  0.4510524   0.000000
TR10090|c0_g2_i1_coral   0.000000   3.961279  1.3531572   1.434073
TR10103|c0_g1_i1_coral   4.855395   6.338047  3.6084191   1.434073
TR10108|c0_g1_i1_coral  10.317715   3.961279  3.1573667   2.151109
TR10132|c0_g1_i1_coral  16.386958   2.376767 13.0805193   8.604437
                       VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18
TR10083|c0_g2_i1_coral  0.6877679   1.987011   6.431470   7.522889
TR10090|c0_g1_i1_coral  3.4388395   1.987011   3.675126   1.074698
TR10090|c0_g2_i1_coral 14.4431257   5.961033   0.000000   3.761445
TR10103|c0_g1_i1_coral  7.5654468   3.974022   0.000000   3.761445
TR10108|c0_g1_i1_coral  2.0633037   4.967527   1.837563   4.298794
TR10132|c0_g1_i1_coral 15.8186615   1.987011   0.000000  10.209636
                       VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_18
TR10083|c0_g2_i1_coral   0.000000   4.374769  2.8394448    0.00000
TR10090|c0_g1_i1_coral   6.346543   0.624967  4.9690284    0.00000
TR10090|c0_g2_i1_coral   3.966589  12.499340  6.3887508   36.13826
TR10103|c0_g1_i1_coral   4.759907   0.624967  6.3887508    0.00000
TR10108|c0_g1_i1_coral   0.000000   8.749538  2.8394448    0.00000
TR10132|c0_g1_i1_coral   2.379953  12.499340  0.7098612    0.00000
                       VA_W_03_22 VA_W_04_14 VA_W_04_18 VA_W_04_22
TR10083|c0_g2_i1_coral   6.328591  0.9650112   3.976631   6.009333
TR10090|c0_g1_i1_coral   8.629897  2.4125279   5.302175   3.698051
TR10090|c0_g2_i1_coral   2.876632  1.9300224   7.069566  10.169640
TR10103|c0_g1_i1_coral   2.301306  3.3775391   3.534783   2.773538
TR10108|c0_g1_i1_coral   3.451959  6.2725727  18.557612   7.858358
TR10132|c0_g1_i1_coral  12.081856  2.4125279   0.000000   0.000000
                       VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14
TR10083|c0_g2_i1_coral  0.0000000   3.511099  11.297065   4.584168
TR10090|c0_g1_i1_coral  0.0000000   1.003171  11.297065   0.000000
TR10090|c0_g2_i1_coral  0.0000000   2.507928   6.778239   2.750501
TR10103|c0_g1_i1_coral  0.7675164   3.009514   2.259413   7.334669
TR10108|c0_g1_i1_coral  1.5350328   5.015856  35.020902   3.667335
TR10132|c0_g1_i1_coral  2.3025492   8.025369   2.259413   8.251503
                       VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18
TR10083|c0_g2_i1_coral  2.3825360   0.000000  9.2975805   1.022439
TR10090|c0_g1_i1_coral  0.7941787   1.585195  1.4303970   1.022439
TR10090|c0_g2_i1_coral  0.7941787   0.000000  5.0063895   3.067318
TR10103|c0_g1_i1_coral  2.3825360   3.962988  4.2911910   0.000000
TR10108|c0_g1_i1_coral  6.3534292   4.755585  2.8607940   1.022439
TR10132|c0_g1_i1_coral 27.0020742  23.777926  0.7151985   6.134635
                       VA_W_07_22
TR10083|c0_g2_i1_coral  0.8584103
TR10090|c0_g1_i1_coral  6.0088720
TR10090|c0_g2_i1_coral  2.5752309
TR10103|c0_g1_i1_coral  1.7168206
TR10108|c0_g1_i1_coral  1.7168206
TR10132|c0_g1_i1_coral  7.7256926
&gt; genes4heatmap&lt;-res[res$pvalue &lt;0.05 &amp; !is.na(res$pvalue),]
&gt; names(genes4heatmap)
[1] &quot;baseMean&quot;       &quot;log2FoldChange&quot; &quot;lfcSE&quot;          &quot;stat&quot;          
[5] &quot;pvalue&quot;         &quot;padj&quot;          
&gt; head(genes4heatmap)
log2 fold change (MLE): Origin_SymState_Temp RI_W_22 vs VA_W_22 
Wald test p-value: Origin_SymState_Temp RI_W_22 vs VA_W_22 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE      stat
                       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
TR10090|c0_g1_i1_coral   2.41467       -2.16024  1.066582  -2.02538
TR10450|c0_g1_i1_coral   3.06295       -3.43891  1.358159  -2.53204
TR10768|c0_g1_i1_coral   4.11391        1.64720  0.816317   2.01785
TR10773|c0_g1_i1_coral   2.74300        3.89665  1.286283   3.02939
TR11771|c0_g1_i1_coral   2.86558        2.26910  0.996534   2.27700
TR11938|c0_g1_i2_coral   3.17991        1.92891  0.932250   2.06910
                          pvalue      padj
                       &lt;numeric&gt; &lt;numeric&gt;
TR10090|c0_g1_i1_coral 0.0428281  0.996985
TR10450|c0_g1_i1_coral 0.0113402  0.882510
TR10768|c0_g1_i1_coral 0.0436070  0.996985
TR10773|c0_g1_i1_coral 0.0024505  0.437812
TR11771|c0_g1_i1_coral 0.0227864  0.989064
TR11938|c0_g1_i2_coral 0.0385371  0.996985
&gt; dim(genes4heatmap)
[1] 343   6
&gt; data4heatmap&lt;-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]
&gt; dim(data4heatmap)
[1] 343  81
&gt; head(data4heatmap)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14
TR10090|c0_g1_i1_coral    0.00000    2.52837   1.146477   0.000000
TR10450|c0_g1_i1_coral    0.00000    2.52837   0.000000   0.000000
TR10768|c0_g1_i1_coral   10.84813    4.21395   4.585907   7.107215
TR10773|c0_g1_i1_coral    0.00000    5.89953   2.292953   3.553607
TR11771|c0_g1_i1_coral    0.00000    2.52837   3.439430   0.000000
TR11938|c0_g1_i2_coral    0.00000    2.52837   0.000000   0.000000
                       RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_22
TR10090|c0_g1_i1_coral   13.09462   0.000000   0.000000   0.000000
TR10450|c0_g1_i1_coral    0.00000   6.468236   0.000000   1.269319
TR10768|c0_g1_i1_coral    0.00000   8.624315   1.235394   1.269319
TR10773|c0_g1_i1_coral    0.00000   0.000000   1.235394   0.000000
TR11771|c0_g1_i1_coral    0.00000   0.000000   1.235394   5.077275
TR11938|c0_g1_i2_coral    0.00000   4.312157   2.470789   3.807956
                       RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10090|c0_g1_i1_coral  2.0285120  4.0659932   6.219661  0.5373050
TR10450|c0_g1_i1_coral  2.4342144  0.4065993  18.658984  0.2686525
TR10768|c0_g1_i1_coral  0.8114048  4.8791918   0.000000  2.9551775
TR10773|c0_g1_i1_coral  0.0000000  0.4065993   0.000000  0.0000000
TR11771|c0_g1_i1_coral  1.6228096  4.0659932   0.000000  1.0746100
TR11938|c0_g1_i2_coral  0.0000000  1.2197979   9.329492  4.2984400
                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18
TR10090|c0_g1_i1_coral   0.000000   0.000000    0.00000   1.963417
TR10450|c0_g1_i1_coral  15.182145   6.352850    0.00000   1.308944
TR10768|c0_g1_i1_coral   0.000000   1.588213    0.00000   4.581305
TR10773|c0_g1_i1_coral   0.000000   0.000000    0.00000   0.000000
TR11771|c0_g1_i1_coral   6.900975   1.588213   12.08711   4.581305
TR11938|c0_g1_i2_coral   4.140585   3.176425    0.00000   1.963417
                       RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22
TR10090|c0_g1_i1_coral    0.00000  0.6838455  1.9782213    0.00000
TR10450|c0_g1_i1_coral   42.00244  2.0515364  5.9346640    0.00000
TR10768|c0_g1_i1_coral    0.00000  2.7353818  8.9019960   45.66098
TR10773|c0_g1_i1_coral    0.00000  0.0000000  0.9891107    0.00000
TR11771|c0_g1_i1_coral    0.00000  2.0515364  7.9128854    0.00000
TR11938|c0_g1_i2_coral    0.00000  0.0000000  0.0000000    0.00000
                       RI_W_01_14 RI_W_01_18 RI_W_01_22 RI_W_02_14
TR10090|c0_g1_i1_coral    0.00000  11.625295   1.456442   0.000000
TR10450|c0_g1_i1_coral    0.00000   0.000000   0.000000   4.027688
TR10768|c0_g1_i1_coral   17.26245   7.522249   1.456442   0.000000
TR10773|c0_g1_i1_coral    0.00000   7.522249  16.020867   0.000000
TR11771|c0_g1_i1_coral    0.00000   2.051523   2.912885   0.000000
TR11938|c0_g1_i2_coral    0.00000   4.103045   1.456442   2.013844
                       RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18
TR10090|c0_g1_i1_coral   1.956995   0.000000   4.039014   1.068873
TR10450|c0_g1_i1_coral   0.000000   1.197822   1.154004   0.000000
TR10768|c0_g1_i1_coral   4.892488   5.989108   4.616016   2.850329
TR10773|c0_g1_i1_coral  14.677463   5.989108  21.926075  12.470189
TR11771|c0_g1_i1_coral   5.870985  14.373858   4.616016   1.781456
TR11938|c0_g1_i2_coral   2.935493  10.780394   4.039014   1.425164
                       RI_W_03_22 RI_W_04_14 RI_W_04_18 RI_W_04_22
TR10090|c0_g1_i1_coral  1.4697536    0.00000   3.437796     0.0000
TR10450|c0_g1_i1_coral  0.0000000    0.00000   4.911138     0.0000
TR10768|c0_g1_i1_coral  6.8588501    0.00000   2.946683    17.8865
TR10773|c0_g1_i1_coral 26.9454824    0.00000   1.964455     0.0000
TR11771|c0_g1_i1_coral  0.4899179    0.00000   1.964455     0.0000
TR11938|c0_g1_i2_coral  3.4294250   12.32321   3.928910     0.0000
                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14
TR10090|c0_g1_i1_coral  0.8788181  7.7745767   0.000000   0.000000
TR10450|c0_g1_i1_coral  1.3182272  0.9718221   0.000000   1.957360
TR10768|c0_g1_i1_coral  5.7123177  0.9718221   3.842950   0.000000
TR10773|c0_g1_i1_coral  5.2729086  1.9436442   0.000000   4.893399
TR11771|c0_g1_i1_coral  4.3940905  1.9436442   5.123934   4.893399
TR11938|c0_g1_i2_coral  2.6364543  2.9154662   5.123934   6.850759
                       RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18
TR10090|c0_g1_i1_coral   0.000000   4.333240  0.0000000   0.000000
TR10450|c0_g1_i1_coral   1.280240   0.000000  0.6750084   2.013953
TR10768|c0_g1_i1_coral   7.681439   5.416550  6.7500843   3.020929
TR10773|c0_g1_i1_coral   2.560480   9.749789  1.3500169   2.013953
TR11771|c0_g1_i1_coral   0.000000   3.249930  0.6750084   2.013953
TR11938|c0_g1_i2_coral   0.000000   2.166620  1.3500169   8.055810
                       RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10090|c0_g1_i1_coral   0.000000   6.825663   6.879404  1.6522753
TR10450|c0_g1_i1_coral   0.000000   5.688053  15.724353  2.7537921
TR10768|c0_g1_i1_coral   2.830176   1.137611   6.224223  3.8553090
TR10773|c0_g1_i1_coral   0.000000   5.688053   3.603497  2.7537921
TR11771|c0_g1_i1_coral  14.150880   3.412832   1.965544  0.5507584
TR11938|c0_g1_i2_coral   2.830176   3.412832   3.931088  1.1015169
                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14
TR10090|c0_g1_i1_coral   0.000000   0.000000   0.000000   5.534635
TR10450|c0_g1_i1_coral   0.000000   8.328238   0.000000   0.000000
TR10768|c0_g1_i1_coral   1.417949   0.000000   3.904196   4.842806
TR10773|c0_g1_i1_coral   1.417949   0.000000   5.856295   0.000000
TR11771|c0_g1_i1_coral   0.000000   0.000000   0.000000   9.685611
TR11938|c0_g1_i2_coral   4.253846   0.000000   0.000000   2.075488
                       VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18
TR10090|c0_g1_i1_coral   1.980640  7.5143298   0.000000   4.154433
TR10450|c0_g1_i1_coral   1.320426  2.7324836   0.000000   0.000000
TR10768|c0_g1_i1_coral   2.640853  2.0493627   1.631661   1.246330
TR10773|c0_g1_i1_coral   1.320426  0.6831209   0.000000   6.647093
TR11771|c0_g1_i1_coral   1.980640  0.0000000   0.000000   2.908103
TR11938|c0_g1_i2_coral   3.301066  3.4156045   2.447491   4.985320
                       VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10090|c0_g1_i1_coral   3.342246   4.512607  0.9973314   6.873047
TR10450|c0_g1_i1_coral   1.909855   1.611645  1.4959971   0.000000
TR10768|c0_g1_i1_coral   0.000000   3.867949  2.4933284   0.000000
TR10773|c0_g1_i1_coral   7.639419   1.289316  0.9973314   0.000000
TR11771|c0_g1_i1_coral   0.000000   2.578632  2.9919941   5.154785
TR11938|c0_g1_i2_coral   7.161955   2.256303  3.9893255   1.718262
                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14
TR10090|c0_g1_i1_coral   7.283093   0.000000  0.4510524   0.000000
TR10450|c0_g1_i1_coral   1.820773   0.000000  4.5105239   0.000000
TR10768|c0_g1_i1_coral   6.676168   2.376767  4.5105239   2.151109
TR10773|c0_g1_i1_coral   2.427698   0.000000  4.9615763   1.434073
TR11771|c0_g1_i1_coral   3.641546   5.545791  1.3531572   4.302218
TR11938|c0_g1_i2_coral   7.283093   0.000000  3.6084191   7.170364
                       VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18
TR10090|c0_g1_i1_coral   3.438839  1.9870109   3.675126   1.074698
TR10450|c0_g1_i1_coral   1.375536  0.9935055   0.000000   5.373492
TR10768|c0_g1_i1_coral   1.375536  2.9805164   1.837563   8.060239
TR10773|c0_g1_i1_coral   2.751072  5.9610327   5.512689   1.074698
TR11771|c0_g1_i1_coral   0.000000  4.9675273   5.512689   1.612048
TR11938|c0_g1_i2_coral   8.253215 10.9285600   2.756344   3.224095
                       VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_18
TR10090|c0_g1_i1_coral   6.346543   0.624967   4.969028    0.00000
TR10450|c0_g1_i1_coral   0.000000   1.874901   4.969028   31.88670
TR10768|c0_g1_i1_coral   1.586636   6.249670   2.839445    0.00000
TR10773|c0_g1_i1_coral   0.000000   3.749802   1.419722    0.00000
TR11771|c0_g1_i1_coral   1.586636   3.124835   7.098612   21.25780
TR11938|c0_g1_i2_coral   1.586636   6.874637   4.259167   23.38358
                       VA_W_03_22 VA_W_04_14 VA_W_04_18 VA_W_04_22
TR10090|c0_g1_i1_coral   8.629897  2.4125279   5.302175  3.6980510
TR10450|c0_g1_i1_coral   1.150653  0.9650112   0.000000  1.8490255
TR10768|c0_g1_i1_coral   1.725979  1.4475168   3.976631  2.3112819
TR10773|c0_g1_i1_coral   2.301306  0.9650112   0.000000  0.4622564
TR11771|c0_g1_i1_coral   1.725979  0.0000000   0.000000  0.4622564
TR11938|c0_g1_i2_coral   1.725979  4.8250559   1.767392  0.4622564
                       VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14
TR10090|c0_g1_i1_coral          0  1.0031712  11.297065   0.000000
TR10450|c0_g1_i1_coral          0  0.0000000  12.426772   1.833667
TR10768|c0_g1_i1_coral          0  6.5206127   0.000000   2.750501
TR10773|c0_g1_i1_coral          0  0.0000000   0.000000   0.000000
TR11771|c0_g1_i1_coral          0  0.5015856   1.129707   1.833667
TR11938|c0_g1_i2_coral          0  2.5079280   0.000000   1.833667
                       VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18
TR10090|c0_g1_i1_coral  0.7941787  1.5851950  1.4303970   1.022439
TR10450|c0_g1_i1_coral  1.5883573  3.9629876  2.1455955   0.000000
TR10768|c0_g1_i1_coral  5.5592506  0.0000000  2.8607940   8.179514
TR10773|c0_g1_i1_coral  1.5883573  0.0000000  0.0000000   0.000000
TR11771|c0_g1_i1_coral  2.3825360  0.7925975  0.7151985   4.089757
TR11938|c0_g1_i2_coral  0.7941787  2.3777926  1.4303970   0.000000
                       VA_W_07_22
TR10090|c0_g1_i1_coral  6.0088720
TR10450|c0_g1_i1_coral  3.4336412
TR10768|c0_g1_i1_coral  3.4336412
TR10773|c0_g1_i1_coral  0.0000000
TR11771|c0_g1_i1_coral  2.5752309
TR11938|c0_g1_i2_coral  0.8584103
&gt; temp = as.matrix(rowMeans(data4heatmap))
&gt; head(temp)
                           [,1]
TR10090|c0_g1_i1_coral 2.414669
TR10450|c0_g1_i1_coral 3.062953
TR10768|c0_g1_i1_coral 4.113910
TR10773|c0_g1_i1_coral 2.742998
TR11771|c0_g1_i1_coral 2.865582
TR11938|c0_g1_i2_coral 3.179912
&gt; scaledmatrix&lt;-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)
&gt; data4heatmapscaled = data4heatmap/scaledmatrix
&gt; head(data4heatmapscaled)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14
TR10090|c0_g1_i1_coral   0.000000  1.0470877  0.4747967   0.000000
TR10450|c0_g1_i1_coral   0.000000  0.8254680  0.0000000   0.000000
TR10768|c0_g1_i1_coral   2.636938  1.0243174  1.1147318   1.727606
TR10773|c0_g1_i1_coral   0.000000  2.1507597  0.8359295   1.295519
TR11771|c0_g1_i1_coral   0.000000  0.8823235  1.2002554   0.000000
TR11938|c0_g1_i2_coral   0.000000  0.7951069  0.0000000   0.000000
                       RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_22
TR10090|c0_g1_i1_coral   5.422948   0.000000  0.0000000  0.0000000
TR10450|c0_g1_i1_coral   0.000000   2.111764  0.0000000  0.4144100
TR10768|c0_g1_i1_coral   0.000000   2.096379  0.3002969  0.3085431
TR10773|c0_g1_i1_coral   0.000000   0.000000  0.4503810  0.0000000
TR11771|c0_g1_i1_coral   0.000000   0.000000  0.4311147  1.7718129
TR11938|c0_g1_i2_coral   0.000000   1.356062  0.7769990  1.1975035
                       RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10090|c0_g1_i1_coral  0.8400788  1.6838720   2.575782 0.22251707
TR10450|c0_g1_i1_coral  0.7947279  0.1327475   6.091827 0.08771028
TR10768|c0_g1_i1_coral  0.1972344  1.1860229   0.000000 0.71833786
TR10773|c0_g1_i1_coral  0.0000000  0.1482317   0.000000 0.00000000
TR11771|c0_g1_i1_coral  0.5663107  1.4189067   0.000000 0.37500589
TR11938|c0_g1_i2_coral  0.0000000  0.3835949   2.933884 1.35174810
                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18
TR10090|c0_g1_i1_coral   0.000000  0.0000000   0.000000  0.8131205
TR10450|c0_g1_i1_coral   4.956701  2.0740929   0.000000  0.4273471
TR10768|c0_g1_i1_coral   0.000000  0.3860591   0.000000  1.1136133
TR10773|c0_g1_i1_coral   0.000000  0.0000000   0.000000  0.0000000
TR11771|c0_g1_i1_coral   2.408228  0.5542374   4.218031  1.5987348
TR11938|c0_g1_i2_coral   1.302107  0.9989034   0.000000  0.6174437
                       RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22
TR10090|c0_g1_i1_coral    0.00000  0.2832047  0.8192517    0.00000
TR10450|c0_g1_i1_coral   13.71305  0.6697903  1.9375626    0.00000
TR10768|c0_g1_i1_coral    0.00000  0.6649104  2.1638771   11.09917
TR10773|c0_g1_i1_coral    0.00000  0.0000000  0.3605947    0.00000
TR11771|c0_g1_i1_coral    0.00000  0.7159232  2.7613540    0.00000
TR11938|c0_g1_i2_coral    0.00000  0.0000000  0.0000000    0.00000
                       RI_W_01_14 RI_W_01_18 RI_W_01_22 RI_W_02_14
TR10090|c0_g1_i1_coral   0.000000  4.8144470  0.6031645  0.0000000
TR10450|c0_g1_i1_coral   0.000000  0.0000000  0.0000000  1.3149687
TR10768|c0_g1_i1_coral   4.196117  1.8284914  0.3540287  0.0000000
TR10773|c0_g1_i1_coral   0.000000  2.7423457  5.8406405  0.0000000
TR11771|c0_g1_i1_coral   0.000000  0.7159184  1.0165074  0.0000000
TR11938|c0_g1_i2_coral   0.000000  1.2903015  0.4580134  0.6333018
                       RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18
TR10090|c0_g1_i1_coral  0.8104611  0.0000000  1.6726989  0.4426584
TR10450|c0_g1_i1_coral  0.0000000  0.3910675  0.3767618  0.0000000
TR10768|c0_g1_i1_coral  1.1892549  1.4558187  1.1220507  0.6928515
TR10773|c0_g1_i1_coral  5.3508832  2.1834164  7.9934703  4.5461892
TR11771|c0_g1_i1_coral  2.0487936  5.0160350  1.6108477  0.6216733
TR11938|c0_g1_i2_coral  0.9231364  3.3901547  1.2701653  0.4481773
                       RI_W_03_22 RI_W_04_14 RI_W_04_18 RI_W_04_22
TR10090|c0_g1_i1_coral  0.6086771   0.000000  1.4237135    0.00000
TR10450|c0_g1_i1_coral  0.0000000   0.000000  1.6033994    0.00000
TR10768|c0_g1_i1_coral  1.6672338   0.000000  0.7162729    4.34781
TR10773|c0_g1_i1_coral  9.8233684   0.000000  0.7161707    0.00000
TR11771|c0_g1_i1_coral  0.1709663   0.000000  0.6855345    0.00000
TR11938|c0_g1_i2_coral  1.0784654   3.875331  1.2355405    0.00000
                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14
TR10090|c0_g1_i1_coral  0.3639498  3.2197281  0.0000000  0.0000000
TR10450|c0_g1_i1_coral  0.4303778  0.3172827  0.0000000  0.6390432
TR10768|c0_g1_i1_coral  1.3885373  0.2362283  0.9341357  0.0000000
TR10773|c0_g1_i1_coral  1.9223157  0.7085838  0.0000000  1.7839600
TR11771|c0_g1_i1_coral  1.5334026  0.6782721  1.7880955  1.7076460
TR11938|c0_g1_i2_coral  0.8290966  0.9168387  1.6113445  2.1543863
                       RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18
TR10090|c0_g1_i1_coral  0.0000000  1.7945483  0.0000000  0.0000000
TR10450|c0_g1_i1_coral  0.4179756  0.0000000  0.2203783  0.6575198
TR10768|c0_g1_i1_coral  1.8671868  1.3166426  1.6407952  0.7343205
TR10773|c0_g1_i1_coral  0.9334602  3.5544277  0.4921683  0.7342157
TR11771|c0_g1_i1_coral  0.0000000  1.1341257  0.2355572  0.7028076
TR11938|c0_g1_i2_coral  0.0000000  0.6813458  0.4245453  2.5333437
                       RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10090|c0_g1_i1_coral  0.0000000  2.8267494  2.8490054  0.6842658
TR10450|c0_g1_i1_coral  0.0000000  1.8570484  5.1337224  0.8990643
TR10768|c0_g1_i1_coral  0.6879528  0.2765278  1.5129700  0.9371398
TR10773|c0_g1_i1_coral  0.0000000  2.0736626  1.3137075  1.0039351
TR11771|c0_g1_i1_coral  4.9382226  1.1909734  0.6859145  0.1921978
TR11938|c0_g1_i2_coral  0.8900171  1.0732472  1.2362254  0.3463985
                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14
TR10090|c0_g1_i1_coral  0.0000000   0.000000  0.0000000  2.2920887
TR10450|c0_g1_i1_coral  0.0000000   2.719022  0.0000000  0.0000000
TR10768|c0_g1_i1_coral  0.3446718   0.000000  0.9490232  1.1771782
TR10773|c0_g1_i1_coral  0.5169339   0.000000  2.1349975  0.0000000
TR11771|c0_g1_i1_coral  0.0000000   0.000000  0.0000000  3.3799808
TR11938|c0_g1_i2_coral  1.3377245   0.000000  0.0000000  0.6526873
                       VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18
TR10090|c0_g1_i1_coral  0.8202531  3.1119506  0.0000000  1.7204981
TR10450|c0_g1_i1_coral  0.4310958  0.8921074  0.0000000  0.0000000
TR10768|c0_g1_i1_coral  0.6419325  0.4981544  0.3966204  0.3029551
TR10773|c0_g1_i1_coral  0.4813807  0.2490417  0.0000000  2.4232946
TR11771|c0_g1_i1_coral  0.6911824  0.0000000  0.0000000  1.0148387
TR11938|c0_g1_i2_coral  1.0380998  1.0741192  0.7696726  1.5677539
                       VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10090|c0_g1_i1_coral  1.3841426  1.8688306  0.4130303  2.8463728
TR10450|c0_g1_i1_coral  0.6235337  0.5261736  0.4884165  0.0000000
TR10768|c0_g1_i1_coral  0.0000000  0.9402122  0.6060726  0.0000000
TR10773|c0_g1_i1_coral  2.7850616  0.4700390  0.3635917  0.0000000
TR11771|c0_g1_i1_coral  0.0000000  0.8998635  1.0441141  1.7988618
TR11938|c0_g1_i2_coral  2.2522495  0.7095490  1.2545396  0.5403489
                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14
TR10090|c0_g1_i1_coral  3.0161871  0.0000000  0.1867968  0.0000000
TR10450|c0_g1_i1_coral  0.5944502  0.0000000  1.4726061  0.0000000
TR10768|c0_g1_i1_coral  1.6228279  0.5777393  1.0964079  0.5228868
TR10773|c0_g1_i1_coral  0.8850525  0.0000000  1.8088149  0.5228121
TR11771|c0_g1_i1_coral  1.2707878  1.9353107  0.4722103  1.5013421
TR11938|c0_g1_i2_coral  2.2903441  0.0000000  1.1347544  2.2548938
                       VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18
TR10090|c0_g1_i1_coral  1.4241454  0.8228917  1.5220000  0.4450708
TR10450|c0_g1_i1_coral  0.4490880  0.3243619  0.0000000  1.7543500
TR10768|c0_g1_i1_coral  0.3343621  0.7244972  0.4466706  1.9592646
TR10773|c0_g1_i1_coral  1.0029432  2.1731814  2.0097310  0.3917970
TR11771|c0_g1_i1_coral  0.0000000  1.7335145  1.9237590  0.5625552
TR11938|c0_g1_i2_coral  2.5954223  3.4367492  0.8667989  1.0138946
                       VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_18
TR10090|c0_g1_i1_coral  2.6283285  0.2588210  2.0578510   0.000000
TR10450|c0_g1_i1_coral  0.0000000  0.6121219  1.6222997  10.410443
TR10768|c0_g1_i1_coral  0.3856758  1.5191556  0.6902058   0.000000
TR10773|c0_g1_i1_coral  0.0000000  1.3670450  0.5175805   0.000000
TR11771|c0_g1_i1_coral  0.5536871  1.0904714  2.4771976   7.418320
TR11938|c0_g1_i2_coral  0.4989558  2.1618953  1.3393978   7.353531
                       VA_W_03_22 VA_W_04_14 VA_W_04_18 VA_W_04_22
TR10090|c0_g1_i1_coral  3.5739467  0.9991134  2.1958187  1.5314941
TR10450|c0_g1_i1_coral  0.3756678  0.3150590  0.0000000  0.6036740
TR10768|c0_g1_i1_coral  0.4195472  0.3518591  0.9666305  0.5618212
TR10773|c0_g1_i1_coral  0.8389746  0.3518089  0.0000000  0.1685223
TR11771|c0_g1_i1_coral  0.6023138  0.0000000  0.0000000  0.1613133
TR11938|c0_g1_i2_coral  0.5427759  1.5173552  0.5557989  0.1453677
                       VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14
TR10090|c0_g1_i1_coral          0  0.4154488  4.6785155  0.0000000
TR10450|c0_g1_i1_coral          0  0.0000000  4.0571206  0.5986599
TR10768|c0_g1_i1_coral          0  1.5850158  0.0000000  0.6685856
TR10773|c0_g1_i1_coral          0  0.0000000  0.0000000  0.0000000
TR11771|c0_g1_i1_coral          0  0.1750380  0.3942329  0.6398936
TR11938|c0_g1_i2_coral          0  0.7886784  0.0000000  0.5766409
                       VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18
TR10090|c0_g1_i1_coral  0.3288976  0.6564855  0.5923782  0.4234284
TR10450|c0_g1_i1_coral  0.5185705  1.2938452  0.7004989  0.0000000
TR10768|c0_g1_i1_coral  1.3513301  0.0000000  0.6953953  1.9882577
TR10773|c0_g1_i1_coral  0.5790588  0.0000000  0.0000000  0.0000000
TR11771|c0_g1_i1_coral  0.8314319  0.2765922  0.2495823  1.4271996
TR11938|c0_g1_i2_coral  0.2497486  0.7477542  0.4498228  0.0000000
                       VA_W_07_22
TR10090|c0_g1_i1_coral  2.4884871
TR10450|c0_g1_i1_coral  1.1210230
TR10768|c0_g1_i1_coral  0.8346417
TR10773|c0_g1_i1_coral  0.0000000
TR11771|c0_g1_i1_coral  0.8986765
TR11938|c0_g1_i2_coral  0.2699478
&gt; 
&gt; dim(data4heatmapscaled)
[1] 343  81
&gt; 
&gt; pairs.breaks &lt;- seq(0, 3.0, by=0.1)
&gt; length(pairs.breaks)
[1] 31
&gt; mycol &lt;- colorpanel(n=30, low=&quot;black&quot;, high=&quot;yellow&quot;)
&gt; pdf(file=&quot;XXXXXX_byrow.pdf&quot;,14,7)
&gt; heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(&quot;row&quot;), scale=&quot;none&quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = &quot;none&quot;, symkey = F, density.info = &quot;density&quot;, colsep=c(24), sepcolor=c(&quot;white&quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
&gt; dev.off()
RStudioGD 
        2 
&gt; pdf(file=&quot;XXXXXX_bycolumn.pdf&quot;,14,7)
&gt; heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(&quot;col&quot;), scale=&quot;none&quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = &quot;none&quot;, symkey = F, density.info = &quot;density&quot;, colsep=c(24), sepcolor=c(&quot;white&quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
&gt; dev.off()
RStudioGD 
        2
&gt; plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=&quot;Origin&quot;)
&gt; prop.nullv3 &lt;- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100*mean(x==0))
&gt; pdf(file=&quot;ResNullCounts.pdf&quot;,14, 14)
&gt; barplot(prop.nullv3, main=&quot;Percentage of null counts per sample&quot;, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
&gt; dev.off()
RStudioGD 
        2 
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>1- Download the DESeq2Script_advbioinf.R script and work through on the shared djbFullCounts_summed.txt file</span>

<span class="pl-c"><span class="pl-c">#</span>/Users/katiecrider/Desktop/Genomics/21sp_advgenomics/assignments_exercises/day11/DESeq2Script_advbioinf.R</span>

<span class="pl-c"><span class="pl-c">#</span>copy this and the summed.txt file to the Genomics folder</span>

<span class="pl-c"><span class="pl-c">#</span>make sure to have downloaded these packages already</span>

<span class="pl-k">&gt;</span>library(DESeq2)
<span class="pl-k">&gt;</span>library(gplots)
<span class="pl-k">&gt;</span>vignette(<span class="pl-s"><span class="pl-pds">'</span>DESeq2<span class="pl-pds">'</span></span>)

<span class="pl-k">&gt;</span>setwd(<span class="pl-s"><span class="pl-pds">"</span>/Users/katiecrider/Desktop/Genomics<span class="pl-pds">"</span></span>)  <span class="pl-c"><span class="pl-c">#</span> The drag and drop from finder works in R, too.</span>
<span class="pl-k">&gt;</span> conds<span class="pl-k">&lt;</span>-data.frame(<span class="pl-s"><span class="pl-pds">"</span>Sample<span class="pl-pds">"</span></span>=names(countsTable))
<span class="pl-k">&gt;</span> conds<span class="pl-smi">$Origin</span><span class="pl-k">&lt;</span>-factor(substring(conds<span class="pl-smi">$Sample</span>,1,2))
<span class="pl-k">&gt;</span> conds<span class="pl-smi">$SymState</span><span class="pl-k">&lt;</span>-factor(substring(conds<span class="pl-smi">$Sample</span>,4,4))
<span class="pl-k">&gt;</span> conds<span class="pl-smi">$Temp</span><span class="pl-k">&lt;</span>-factor(substring(conds<span class="pl-smi">$Sample</span>,9,10))
<span class="pl-k">&gt;</span> conds<span class="pl-smi">$Origin_SymState</span><span class="pl-k">&lt;</span>-factor(paste(conds<span class="pl-smi">$Origin</span>,conds<span class="pl-smi">$SymState</span>, sep=<span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> conds<span class="pl-smi">$Origin_SymState_Temp</span><span class="pl-k">&lt;</span>-factor(paste(conds<span class="pl-smi">$Origin</span>,conds<span class="pl-smi">$SymState</span>,conds<span class="pl-smi">$Temp</span>, sep=<span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> conds
       Sample Origin SymState Temp Origin_SymState Origin_SymState_Temp
1  RI_B_01_14     RI        B   14            RI_B              RI_B_14
2  RI_B_01_18     RI        B   18            RI_B              RI_B_18
3  RI_B_01_22     RI        B   22            RI_B              RI_B_22
4  RI_B_02_14     RI        B   14            RI_B              RI_B_14
5  RI_B_02_18     RI        B   18            RI_B              RI_B_18
6  RI_B_02_22     RI        B   22            RI_B              RI_B_22
7  RI_B_03_14     RI        B   14            RI_B              RI_B_14
8  RI_B_03_22     RI        B   22            RI_B              RI_B_22
9  RI_B_04_14     RI        B   14            RI_B              RI_B_14
10 RI_B_04_18     RI        B   18            RI_B              RI_B_18
11 RI_B_04_22     RI        B   22            RI_B              RI_B_22
12 RI_B_05_14     RI        B   14            RI_B              RI_B_14
13 RI_B_05_18     RI        B   18            RI_B              RI_B_18
14 RI_B_05_22     RI        B   22            RI_B              RI_B_22
15 RI_B_06_14     RI        B   14            RI_B              RI_B_14
16 RI_B_06_18     RI        B   18            RI_B              RI_B_18
17 RI_B_06_22     RI        B   22            RI_B              RI_B_22
18 RI_B_07_14     RI        B   14            RI_B              RI_B_14
19 RI_B_07_18     RI        B   18            RI_B              RI_B_18
20 RI_B_07_22     RI        B   22            RI_B              RI_B_22
21 RI_W_01_14     RI        W   14            RI_W              RI_W_14
22 RI_W_01_18     RI        W   18            RI_W              RI_W_18
23 RI_W_01_22     RI        W   22            RI_W              RI_W_22
24 RI_W_02_14     RI        W   14            RI_W              RI_W_14
25 RI_W_02_18     RI        W   18            RI_W              RI_W_18
26 RI_W_02_22     RI        W   22            RI_W              RI_W_22
27 RI_W_03_14     RI        W   14            RI_W              RI_W_14
28 RI_W_03_18     RI        W   18            RI_W              RI_W_18
29 RI_W_03_22     RI        W   22            RI_W              RI_W_22
30 RI_W_04_14     RI        W   14            RI_W              RI_W_14
31 RI_W_04_18     RI        W   18            RI_W              RI_W_18
32 RI_W_04_22     RI        W   22            RI_W              RI_W_22
33 RI_W_05_14     RI        W   14            RI_W              RI_W_14
34 RI_W_05_18     RI        W   18            RI_W              RI_W_18
35 RI_W_05_22     RI        W   22            RI_W              RI_W_22
36 RI_W_06_14     RI        W   14            RI_W              RI_W_14
37 RI_W_06_18     RI        W   18            RI_W              RI_W_18
38 RI_W_06_22     RI        W   22            RI_W              RI_W_22
39 RI_W_07_14     RI        W   14            RI_W              RI_W_14
40 RI_W_07_18     RI        W   18            RI_W              RI_W_18
41 RI_W_07_22     RI        W   22            RI_W              RI_W_22
42 VA_B_01_14     VA        B   14            VA_B              VA_B_14
43 VA_B_01_18     VA        B   18            VA_B              VA_B_18
44 VA_B_01_22     VA        B   22            VA_B              VA_B_22
45 VA_B_02_14     VA        B   14            VA_B              VA_B_14
46 VA_B_02_18     VA        B   18            VA_B              VA_B_18
47 VA_B_02_22     VA        B   22            VA_B              VA_B_22
48 VA_B_03_14     VA        B   14            VA_B              VA_B_14
49 VA_B_03_18     VA        B   18            VA_B              VA_B_18
50 VA_B_03_22     VA        B   22            VA_B              VA_B_22
51 VA_B_04_14     VA        B   14            VA_B              VA_B_14
52 VA_B_04_18     VA        B   18            VA_B              VA_B_18
53 VA_B_04_22     VA        B   22            VA_B              VA_B_22
54 VA_B_05_14     VA        B   14            VA_B              VA_B_14
55 VA_B_05_18     VA        B   18            VA_B              VA_B_18
56 VA_B_05_22     VA        B   22            VA_B              VA_B_22
57 VA_B_06_14     VA        B   14            VA_B              VA_B_14
58 VA_B_06_18     VA        B   18            VA_B              VA_B_18
59 VA_B_06_22     VA        B   22            VA_B              VA_B_22
60 VA_B_07_14     VA        B   14            VA_B              VA_B_14
61 VA_B_07_18     VA        B   18            VA_B              VA_B_18
62 VA_B_07_22     VA        B   22            VA_B              VA_B_22
63 VA_W_01_14     VA        W   14            VA_W              VA_W_14
64 VA_W_01_18     VA        W   18            VA_W              VA_W_18
65 VA_W_02_14     VA        W   14            VA_W              VA_W_14
66 VA_W_02_18     VA        W   18            VA_W              VA_W_18
67 VA_W_03_14     VA        W   14            VA_W              VA_W_14
68 VA_W_03_18     VA        W   18            VA_W              VA_W_18
69 VA_W_03_22     VA        W   22            VA_W              VA_W_22
70 VA_W_04_14     VA        W   14            VA_W              VA_W_14
71 VA_W_04_18     VA        W   18            VA_W              VA_W_18
72 VA_W_04_22     VA        W   22            VA_W              VA_W_22
73 VA_W_05_14     VA        W   14            VA_W              VA_W_14
74 VA_W_05_18     VA        W   18            VA_W              VA_W_18
75 VA_W_05_22     VA        W   22            VA_W              VA_W_22
76 VA_W_06_14     VA        W   14            VA_W              VA_W_14
77 VA_W_06_18     VA        W   18            VA_W              VA_W_18
78 VA_W_06_22     VA        W   22            VA_W              VA_W_22
79 VA_W_07_14     VA        W   14            VA_W              VA_W_14
80 VA_W_07_18     VA        W   18            VA_W              VA_W_18
81 VA_W_07_22     VA        W   22            VA_W              VA_W_22
<span class="pl-k">&gt;</span> prop.null <span class="pl-k">&lt;</span>- apply(countsTable, 2, function(x) 100<span class="pl-k">*</span>mean(x==0))
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> barplot(prop.null, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> pdf(file=<span class="pl-s"><span class="pl-pds">"</span>OverallNullCounts.pdf<span class="pl-pds">"</span></span>,14, 14)
<span class="pl-k">&gt;</span> barplot(prop.null, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)
<span class="pl-k">&gt;</span> <span class="pl-en">dev.off</span>()
RStudioGD 
        2 
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span> how many genes have mean count &gt;=3 </span>
<span class="pl-k">&gt;</span> means = apply(countsTable, 1, mean)
<span class="pl-k">&gt;</span> table(means<span class="pl-k">&gt;</span>=3)

TRUE 
7543 
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> countsTable<span class="pl-k">&lt;</span>-countsTable[means<span class="pl-k">&gt;=</span>3,]
<span class="pl-k">&gt;</span> dim(countsTable)
[1] 7543   81
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> prop.nullv2 <span class="pl-k">&lt;</span>- apply(countsTable, 2, function(x) 100<span class="pl-k">*</span>mean(x==0))
<span class="pl-k">&gt;</span> pdf(file=<span class="pl-s"><span class="pl-pds">"</span>Mean3NullCounts.pdf<span class="pl-pds">"</span></span>,14, 14)
<span class="pl-k">&gt;</span> barplot(prop.nullv2, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)
<span class="pl-k">&gt;</span> <span class="pl-en">dev.off</span>()
RStudioGD 
        2 
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>make count data set</span>
<span class="pl-k">&gt;</span> dds <span class="pl-k">&lt;</span>- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=<span class="pl-k">~</span> Origin_SymState_Temp)
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> dds <span class="pl-k">&lt;</span>- DESeq(dds)
estimating size factors
estimating dispersions
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
fitting model and testing
-- replacing outliers and refitting <span class="pl-k">for</span> 8 genes
-- DESeq argument <span class="pl-s"><span class="pl-pds">'</span>minReplicatesForReplace<span class="pl-pds">'</span></span> = 7 
-- original counts are preserved <span class="pl-k">in</span> counts(dds)
estimating dispersions
fitting model and testing
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>transform counts to variance stablized counts</span>
<span class="pl-k">&gt;</span> vstCounts<span class="pl-k">&lt;</span>-varianceStabilizingTransformation(dds)
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>plot cluster dendrogram across samples</span>
<span class="pl-k">&gt;</span> dists<span class="pl-k">&lt;</span>-dist(t(assay(vstCounts)))
<span class="pl-k">&gt;</span> plot(hclust(dists))
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> plotPCA(vstCounts, intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> plotPCA(vstCounts, intgroup=<span class="pl-s"><span class="pl-pds">"</span>SymState<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> plotPCA(vstCounts, intgroup=<span class="pl-s"><span class="pl-pds">"</span>Temp<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> plotPCA(vstCounts, intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> plotPCA(vstCounts, intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> resultsNames(dds)
 [1] <span class="pl-s"><span class="pl-pds">"</span>Intercept<span class="pl-pds">"</span></span>                              
 [2] <span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp_RI_B_18_vs_RI_B_14<span class="pl-pds">"</span></span>
 [3] <span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp_RI_B_22_vs_RI_B_14<span class="pl-pds">"</span></span>
 [4] <span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp_RI_W_14_vs_RI_B_14<span class="pl-pds">"</span></span>
 [5] <span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp_RI_W_18_vs_RI_B_14<span class="pl-pds">"</span></span>
 [6] <span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp_RI_W_22_vs_RI_B_14<span class="pl-pds">"</span></span>
 [7] <span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp_VA_B_14_vs_RI_B_14<span class="pl-pds">"</span></span>
 [8] <span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp_VA_B_18_vs_RI_B_14<span class="pl-pds">"</span></span>
 [9] <span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp_VA_B_22_vs_RI_B_14<span class="pl-pds">"</span></span>
[10] <span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp_VA_W_14_vs_RI_B_14<span class="pl-pds">"</span></span>
[11] <span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp_VA_W_18_vs_RI_B_14<span class="pl-pds">"</span></span>
[12] <span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp_VA_W_22_vs_RI_B_14<span class="pl-pds">"</span></span>
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> dim(dds)
[1] 7543   81
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>##Figure out which contrast you want to examine (i.e. which two groups do you want to compare)</span>
<span class="pl-k">&gt;</span> res <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI_B_18<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>VA_B_18<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> head(res)
log2 fold change (MLE): Origin_SymState_Temp RI_B_18 vs VA_B_18 
Wald <span class="pl-c1">test</span> p-value: Origin_SymState_Temp RI_B_18 vs VA_B_18 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE      stat
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>      <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   5.92149      -0.124352  0.881723 -0.141032
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.41467       0.403769  0.959279  0.420909
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.75861      -0.835423  0.865239 -0.965540
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   2.98376      -0.801464  0.848308 -0.944780
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   5.73440      -0.526701  0.837420 -0.628957
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   8.52630       1.071572  1.160128  0.923667
                          pvalue      padj
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  0.887844  0.996375
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.673822  0.990719
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.334274  0.964876
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  0.344771  0.967913
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  0.529377  0.987729
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  0.355660  0.967913
<span class="pl-k">&gt;</span> summary(res)

out of 7542 with nonzero total <span class="pl-c1">read</span> count
adjusted p-value <span class="pl-k">&lt;</span> 0.1
LFC <span class="pl-k">&gt;</span> 0 (up)       <span class="pl-c1">:</span> 3, 0.04%
LFC <span class="pl-k">&lt;</span> 0 (down)     <span class="pl-c1">:</span> 5, 0.066%
outliers [1]       <span class="pl-c1">:</span> 4, 0.053%
low counts [2]     <span class="pl-c1">:</span> 1, 0.013%
(mean count <span class="pl-k">&lt;</span> 1)
[1] see <span class="pl-s"><span class="pl-pds">'</span>cooksCutoff<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results
[2] see <span class="pl-s"><span class="pl-pds">'</span>independentFiltering<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results

<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> res <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI_W_18<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>VA_W_18<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> summary(res)

out of 7542 with nonzero total <span class="pl-c1">read</span> count
adjusted p-value <span class="pl-k">&lt;</span> 0.1
LFC <span class="pl-k">&gt;</span> 0 (up)       <span class="pl-c1">:</span> 22, 0.29%
LFC <span class="pl-k">&lt;</span> 0 (down)     <span class="pl-c1">:</span> 38, 0.5%
outliers [1]       <span class="pl-c1">:</span> 4, 0.053%
low counts [2]     <span class="pl-c1">:</span> 3801, 50%
(mean count <span class="pl-k">&lt;</span> 6)
[1] see <span class="pl-s"><span class="pl-pds">'</span>cooksCutoff<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results
[2] see <span class="pl-s"><span class="pl-pds">'</span>independentFiltering<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results

<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> res <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI_B_14<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>VA_B_14<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> summary(res)

out of 7542 with nonzero total <span class="pl-c1">read</span> count
adjusted p-value <span class="pl-k">&lt;</span> 0.1
LFC <span class="pl-k">&gt;</span> 0 (up)       <span class="pl-c1">:</span> 8, 0.11%
LFC <span class="pl-k">&lt;</span> 0 (down)     <span class="pl-c1">:</span> 4, 0.053%
outliers [1]       <span class="pl-c1">:</span> 4, 0.053%
low counts [2]     <span class="pl-c1">:</span> 1, 0.013%
(mean count <span class="pl-k">&lt;</span> 1)
[1] see <span class="pl-s"><span class="pl-pds">'</span>cooksCutoff<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results
[2] see <span class="pl-s"><span class="pl-pds">'</span>independentFiltering<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results

<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> res <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI_W_14<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>VA_W_14<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> summary(res)

out of 7542 with nonzero total <span class="pl-c1">read</span> count
adjusted p-value <span class="pl-k">&lt;</span> 0.1
LFC <span class="pl-k">&gt;</span> 0 (up)       <span class="pl-c1">:</span> 10, 0.13%
LFC <span class="pl-k">&lt;</span> 0 (down)     <span class="pl-c1">:</span> 3, 0.04%
outliers [1]       <span class="pl-c1">:</span> 4, 0.053%
low counts [2]     <span class="pl-c1">:</span> 5263, 70%
(mean count <span class="pl-k">&lt;</span> 8)
[1] see <span class="pl-s"><span class="pl-pds">'</span>cooksCutoff<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results
[2] see <span class="pl-s"><span class="pl-pds">'</span>independentFiltering<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results

<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> res <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI_B_22<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>VA_B_22<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> summary(res)

out of 7542 with nonzero total <span class="pl-c1">read</span> count
adjusted p-value <span class="pl-k">&lt;</span> 0.1
LFC <span class="pl-k">&gt;</span> 0 (up)       <span class="pl-c1">:</span> 8, 0.11%
LFC <span class="pl-k">&lt;</span> 0 (down)     <span class="pl-c1">:</span> 2, 0.027%
outliers [1]       <span class="pl-c1">:</span> 4, 0.053%
low counts [2]     <span class="pl-c1">:</span> 1, 0.013%
(mean count <span class="pl-k">&lt;</span> 1)
[1] see <span class="pl-s"><span class="pl-pds">'</span>cooksCutoff<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results
[2] see <span class="pl-s"><span class="pl-pds">'</span>independentFiltering<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results

<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> res <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI_W_22<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>VA_W_22<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> summary(res)

out of 7542 with nonzero total <span class="pl-c1">read</span> count
adjusted p-value <span class="pl-k">&lt;</span> 0.1
LFC <span class="pl-k">&gt;</span> 0 (up)       <span class="pl-c1">:</span> 11, 0.15%
LFC <span class="pl-k">&lt;</span> 0 (down)     <span class="pl-c1">:</span> 5, 0.066%
outliers [1]       <span class="pl-c1">:</span> 4, 0.053%
low counts [2]     <span class="pl-c1">:</span> 146, 1.9%
(mean count <span class="pl-k">&lt;</span> 2)
[1] see <span class="pl-s"><span class="pl-pds">'</span>cooksCutoff<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results
[2] see <span class="pl-s"><span class="pl-pds">'</span>independentFiltering<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results

<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> head(res[order(res<span class="pl-smi">$padj</span>),])
log2 fold change (MLE): Origin_SymState_Temp RI_W_22 vs VA_W_22 
Wald <span class="pl-c1">test</span> p-value: Origin_SymState_Temp RI_W_22 vs VA_W_22 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE      stat
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>      <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR42879<span class="pl-k">|</span>c0_g2_i1_coral  15.41058       17.05753  3.137478   5.43670
TR13286<span class="pl-k">|</span>c1_g1_i1_coral  32.61026       -3.38044  0.648393  -5.21357
TR47617<span class="pl-k">|</span>c0_g1_i1_coral   3.74656       19.72616  3.835159   5.14351
TR8327<span class="pl-k">|</span>c0_g3_i1_coral    8.73277       19.73384  4.063853   4.85594
TR29560<span class="pl-k">|</span>c0_g1_i1_coral  12.79556        2.74246  0.651668   4.20838
TR58072<span class="pl-k">|</span>c1_g1_i1_coral  53.85583        1.38039  0.330572   4.17576
                            pvalue        padj
                         <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>   <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR42879<span class="pl-k">|</span>c0_g2_i1_coral 5.42760e-08 0.000401263
TR13286<span class="pl-k">|</span>c1_g1_i1_coral 1.85241e-07 0.000664526
TR47617<span class="pl-k">|</span>c0_g1_i1_coral 2.69658e-07 0.000664526
TR8327<span class="pl-k">|</span>c0_g3_i1_coral  1.19814e-06 0.002214469
TR29560<span class="pl-k">|</span>c0_g1_i1_coral 2.57208e-05 0.036593740
TR58072<span class="pl-k">|</span>c1_g1_i1_coral 2.96987e-05 0.036593740
<span class="pl-k">&gt;</span> plotCounts(dds, <span class="pl-s"><span class="pl-pds">"</span>TR2367|c0_g1_i1_coral<span class="pl-pds">"</span></span>, intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>count the number of significantly differentially expressed genes</span>
<span class="pl-k">&gt;</span> sum(res<span class="pl-smi">$padj</span> <span class="pl-k">&lt;</span> 0.3, na.rm =T)
[1] 30
<span class="pl-k">&gt;</span> sum(res<span class="pl-smi">$padj</span> <span class="pl-k">&lt;</span> 0.2, na.rm =T)
[1] 20
<span class="pl-k">&gt;</span> sum(res<span class="pl-smi">$padj</span> <span class="pl-k">&lt;</span> 0.1, na.rm =T)
[1] 16
<span class="pl-k">&gt;</span> sum(res<span class="pl-smi">$pvalue</span> <span class="pl-k">&lt;</span> 0.05, na.rm =T)
[1] 343
<span class="pl-k">&gt;</span> summary(res)

out of 7542 with nonzero total <span class="pl-c1">read</span> count
adjusted p-value <span class="pl-k">&lt;</span> 0.1
LFC <span class="pl-k">&gt;</span> 0 (up)       <span class="pl-c1">:</span> 11, 0.15%
LFC <span class="pl-k">&lt;</span> 0 (down)     <span class="pl-c1">:</span> 5, 0.066%
outliers [1]       <span class="pl-c1">:</span> 4, 0.053%
low counts [2]     <span class="pl-c1">:</span> 146, 1.9%
(mean count <span class="pl-k">&lt;</span> 2)
[1] see <span class="pl-s"><span class="pl-pds">'</span>cooksCutoff<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results
[2] see <span class="pl-s"><span class="pl-pds">'</span>independentFiltering<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results
<span class="pl-k">&gt;</span> scaledcounts = counts(dds, normalized=T)
<span class="pl-k">&gt;</span> head(scaledcounts)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          0    0.84279   2.292953   0.000000
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0    2.52837   1.146477   0.000000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          0    3.37116   2.292953   3.553607
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          0    0.84279   4.585907   0.000000
TR10108<span class="pl-k">|</span>c0_g1_i1_coral          0    1.68558   4.585907   0.000000
TR10132<span class="pl-k">|</span>c0_g1_i1_coral          0    5.89953   1.146477  17.768037
                       RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   3.273656   4.312157  12.353943   1.269319
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  13.094624   0.000000   0.000000   0.000000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000   0.000000   0.000000   3.807956
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   8.624315   1.235394   3.807956
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   6.547312   0.000000  13.589337  15.231824
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  49.104839   0.000000   6.176972   2.538637
                       RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  12.576774  8.1319863   0.000000   4.298440
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.028512  4.0659932   6.219661   0.537305
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   4.868429  3.6593938   0.000000   0.537305
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   6.085536  3.2527945   0.000000   4.835745
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   2.839917  4.8791918  15.549153   5.641703
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.8131986   3.109831   1.343263
                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          0  2.3823188  18.130668   3.272361
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0  0.0000000   0.000000   1.963417
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          0  2.3823188   0.000000   4.581305
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          0  3.9705313   0.000000   3.272361
TR10108<span class="pl-k">|</span>c0_g1_i1_coral          0  0.7941063   6.043556   9.817083
TR10132<span class="pl-k">|</span>c0_g1_i1_coral          0  1.5882125   0.000000   1.308944
                       RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   19.38574  1.3676909  7.9128854   36.52879
TR10090<span class="pl-k">|</span>c0_g1_i1_coral    0.00000  0.6838455  1.9782213    0.00000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral    0.00000  3.4192273  0.9891107    0.00000
TR10103<span class="pl-k">|</span>c0_g1_i1_coral    0.00000  2.0515364  1.9782213    0.00000
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   42.00244  3.4192273  0.9891107    0.00000
TR10132<span class="pl-k">|</span>c0_g1_i1_coral    0.00000  8.2061454 43.5208696   27.39659
                       RI_W_01_14 RI_W_01_18 RI_W_01_22 RI_W_02_14
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   63.29565  10.257613   4.369327   0.000000
TR10090<span class="pl-k">|</span>c0_g1_i1_coral    0.00000  11.625295   1.456442   0.000000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral    0.00000  11.625295   7.282212   4.027688
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   28.77075   7.522249   0.000000   0.000000
TR10108<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   2.735363   1.456442   8.055376
TR10132<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   1.367682   0.000000   0.000000
                       RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  4.8924878   3.593465   8.078028   5.344367
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  1.9569951   0.000000   4.039014   1.068873
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  4.8924878   4.791286   2.885010   3.206620
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  0.9784976   3.593465   2.885010   3.206620
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  6.8494829   2.395643   2.308008   4.631785
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  5.8709854  44.319396   1.154004   1.068873
                       RI_W_03_22 RI_W_04_14 RI_W_04_18 RI_W_04_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  4.4092608   12.32321  2.9466826    17.8865
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  1.4697536    0.00000  3.4377964     0.0000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  1.4697536    0.00000  7.3667065     0.0000
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  0.9798357    0.00000  0.9822275     0.0000
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  1.9596715   20.53868 12.7689580     0.0000
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  0.9798357   28.75416  7.8578203     0.0000
                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  3.0758634  0.9718221   5.123934  0.9786798
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.8788181  7.7745767   0.000000  0.0000000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  3.5152724  9.7182208   1.280983  2.9360395
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  3.9546815  1.9436442   2.561967  9.7867983
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  5.2729086  1.9436442   6.404917 10.7654782
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  1.7576362  9.7182208  25.619669  4.8933992
                       RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18
TR10083<span class="pl-k">|</span>c0_g2_i1_coral    1.28024   5.416550   4.050051   5.034881
TR10090<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   4.333240   0.000000   0.000000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral    1.28024   7.583169   2.025025   1.006976
TR10103<span class="pl-k">|</span>c0_g1_i1_coral    1.28024   2.166620   5.400067  12.083715
TR10108<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   5.416550   0.000000   1.006976
TR10132<span class="pl-k">|</span>c0_g1_i1_coral    1.28024  41.165776   1.350017   2.013953
                       RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   7.075440   2.275221   1.310363   2.753792
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   6.825663   6.879404   1.652275
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   5.660352   9.100884   4.258679   5.507584
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   1.415088   3.412832   1.965544   2.753792
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   5.660352   3.412832   6.879404   2.753792
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  89.150546   1.137611   0.982772   0.000000
                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          0   13.08723  17.568884  15.220246
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0    0.00000   0.000000   5.534635
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          0    0.00000   0.000000  10.377440
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          0    0.00000   1.952098   2.075488
TR10108<span class="pl-k">|</span>c0_g1_i1_coral          0    0.00000   0.000000  13.836587
TR10132<span class="pl-k">|</span>c0_g1_i1_coral          0    0.00000   7.808393   8.993782
                       VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  13.204264   7.514330   0.000000  0.8308866
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   1.980640   7.514330   0.000000  4.1544330
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.961279   3.415604   6.526643  2.9081031
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   3.961279   3.415604   4.079152  0.0000000
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  17.165543  20.493627   2.447491  4.5698763
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  28.389167   8.197451   0.000000  2.4926598
                       VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   6.207028   1.933974  2.9919941   1.718262
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   3.342246   4.512607  0.9973314   6.873047
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000   1.289316  0.4986657   5.154785
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000  2.9919941   0.000000
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   3.342246   8.380556  4.9866569   0.000000
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000  4.4879912   5.154785
                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   1.213849   0.000000  5.4126287   3.585182
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   7.283093   0.000000  0.4510524   0.000000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000   3.961279  1.3531572   1.434073
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   4.855395   6.338047  3.6084191   1.434073
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  10.317715   3.961279  3.1573667   2.151109
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  16.386958   2.376767 13.0805193   8.604437
                       VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  0.6877679   1.987011   6.431470   7.522889
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  3.4388395   1.987011   3.675126   1.074698
TR10090<span class="pl-k">|</span>c0_g2_i1_coral 14.4431257   5.961033   0.000000   3.761445
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  7.5654468   3.974022   0.000000   3.761445
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  2.0633037   4.967527   1.837563   4.298794
TR10132<span class="pl-k">|</span>c0_g1_i1_coral 15.8186615   1.987011   0.000000  10.209636
                       VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_18
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   0.000000   4.374769  2.8394448    0.00000
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   6.346543   0.624967  4.9690284    0.00000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.966589  12.499340  6.3887508   36.13826
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   4.759907   0.624967  6.3887508    0.00000
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   8.749538  2.8394448    0.00000
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   2.379953  12.499340  0.7098612    0.00000
                       VA_W_03_22 VA_W_04_14 VA_W_04_18 VA_W_04_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   6.328591  0.9650112   3.976631   6.009333
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   8.629897  2.4125279   5.302175   3.698051
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   2.876632  1.9300224   7.069566  10.169640
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   2.301306  3.3775391   3.534783   2.773538
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   3.451959  6.2725727  18.557612   7.858358
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  12.081856  2.4125279   0.000000   0.000000
                       VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  0.0000000   3.511099  11.297065   4.584168
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000   1.003171  11.297065   0.000000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.0000000   2.507928   6.778239   2.750501
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  0.7675164   3.009514   2.259413   7.334669
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  1.5350328   5.015856  35.020902   3.667335
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  2.3025492   8.025369   2.259413   8.251503
                       VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  2.3825360   0.000000  9.2975805   1.022439
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.7941787   1.585195  1.4303970   1.022439
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.7941787   0.000000  5.0063895   3.067318
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  2.3825360   3.962988  4.2911910   0.000000
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  6.3534292   4.755585  2.8607940   1.022439
TR10132<span class="pl-k">|</span>c0_g1_i1_coral 27.0020742  23.777926  0.7151985   6.134635
                       VA_W_07_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  0.8584103
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  6.0088720
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  2.5752309
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  1.7168206
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  1.7168206
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  7.7256926
<span class="pl-k">&gt;</span> genes4heatmap<span class="pl-k">&lt;</span>-res[res<span class="pl-smi">$pvalue</span> <span class="pl-k">&lt;</span>0.05 <span class="pl-k">&amp;</span> <span class="pl-k">!</span>is.na(res<span class="pl-smi">$pvalue</span>),]
<span class="pl-k">&gt;</span> names(genes4heatmap)
[1] <span class="pl-s"><span class="pl-pds">"</span>baseMean<span class="pl-pds">"</span></span>       <span class="pl-s"><span class="pl-pds">"</span>log2FoldChange<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>lfcSE<span class="pl-pds">"</span></span>          <span class="pl-s"><span class="pl-pds">"</span>stat<span class="pl-pds">"</span></span>          
[5] <span class="pl-s"><span class="pl-pds">"</span>pvalue<span class="pl-pds">"</span></span>         <span class="pl-s"><span class="pl-pds">"</span>padj<span class="pl-pds">"</span></span>          
<span class="pl-k">&gt;</span> head(genes4heatmap)
log2 fold change (MLE): Origin_SymState_Temp RI_W_22 vs VA_W_22 
Wald <span class="pl-c1">test</span> p-value: Origin_SymState_Temp RI_W_22 vs VA_W_22 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE      stat
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>      <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.41467       -2.16024  1.066582  -2.02538
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   3.06295       -3.43891  1.358159  -2.53204
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   4.11391        1.64720  0.816317   2.01785
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   2.74300        3.89665  1.286283   3.02939
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   2.86558        2.26910  0.996534   2.27700
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   3.17991        1.92891  0.932250   2.06910
                          pvalue      padj
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10090<span class="pl-k">|</span>c0_g1_i1_coral 0.0428281  0.996985
TR10450<span class="pl-k">|</span>c0_g1_i1_coral 0.0113402  0.882510
TR10768<span class="pl-k">|</span>c0_g1_i1_coral 0.0436070  0.996985
TR10773<span class="pl-k">|</span>c0_g1_i1_coral 0.0024505  0.437812
TR11771<span class="pl-k">|</span>c0_g1_i1_coral 0.0227864  0.989064
TR11938<span class="pl-k">|</span>c0_g1_i2_coral 0.0385371  0.996985
<span class="pl-k">&gt;</span> dim(genes4heatmap)
[1] 343   6
<span class="pl-k">&gt;</span> data4heatmap<span class="pl-k">&lt;</span>-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]
<span class="pl-k">&gt;</span> dim(data4heatmap)
[1] 343  81
<span class="pl-k">&gt;</span> head(data4heatmap)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14
TR10090<span class="pl-k">|</span>c0_g1_i1_coral    0.00000    2.52837   1.146477   0.000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral    0.00000    2.52837   0.000000   0.000000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   10.84813    4.21395   4.585907   7.107215
TR10773<span class="pl-k">|</span>c0_g1_i1_coral    0.00000    5.89953   2.292953   3.553607
TR11771<span class="pl-k">|</span>c0_g1_i1_coral    0.00000    2.52837   3.439430   0.000000
TR11938<span class="pl-k">|</span>c0_g1_i2_coral    0.00000    2.52837   0.000000   0.000000
                       RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_22
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   13.09462   0.000000   0.000000   0.000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   6.468236   0.000000   1.269319
TR10768<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   8.624315   1.235394   1.269319
TR10773<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   0.000000   1.235394   0.000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   0.000000   1.235394   5.077275
TR11938<span class="pl-k">|</span>c0_g1_i2_coral    0.00000   4.312157   2.470789   3.807956
                       RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  2.0285120  4.0659932   6.219661  0.5373050
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  2.4342144  0.4065993  18.658984  0.2686525
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  0.8114048  4.8791918   0.000000  2.9551775
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  0.4065993   0.000000  0.0000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  1.6228096  4.0659932   0.000000  1.0746100
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  0.0000000  1.2197979   9.329492  4.2984400
                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000    0.00000   1.963417
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  15.182145   6.352850    0.00000   1.308944
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   1.588213    0.00000   4.581305
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000    0.00000   0.000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   6.900975   1.588213   12.08711   4.581305
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   4.140585   3.176425    0.00000   1.963417
                       RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22
TR10090<span class="pl-k">|</span>c0_g1_i1_coral    0.00000  0.6838455  1.9782213    0.00000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   42.00244  2.0515364  5.9346640    0.00000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral    0.00000  2.7353818  8.9019960   45.66098
TR10773<span class="pl-k">|</span>c0_g1_i1_coral    0.00000  0.0000000  0.9891107    0.00000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral    0.00000  2.0515364  7.9128854    0.00000
TR11938<span class="pl-k">|</span>c0_g1_i2_coral    0.00000  0.0000000  0.0000000    0.00000
                       RI_W_01_14 RI_W_01_18 RI_W_01_22 RI_W_02_14
TR10090<span class="pl-k">|</span>c0_g1_i1_coral    0.00000  11.625295   1.456442   0.000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   0.000000   0.000000   4.027688
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   17.26245   7.522249   1.456442   0.000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   7.522249  16.020867   0.000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   2.051523   2.912885   0.000000
TR11938<span class="pl-k">|</span>c0_g1_i2_coral    0.00000   4.103045   1.456442   2.013844
                       RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   1.956995   0.000000   4.039014   1.068873
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   1.197822   1.154004   0.000000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   4.892488   5.989108   4.616016   2.850329
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  14.677463   5.989108  21.926075  12.470189
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   5.870985  14.373858   4.616016   1.781456
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   2.935493  10.780394   4.039014   1.425164
                       RI_W_03_22 RI_W_04_14 RI_W_04_18 RI_W_04_22
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  1.4697536    0.00000   3.437796     0.0000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000    0.00000   4.911138     0.0000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  6.8588501    0.00000   2.946683    17.8865
TR10773<span class="pl-k">|</span>c0_g1_i1_coral 26.9454824    0.00000   1.964455     0.0000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  0.4899179    0.00000   1.964455     0.0000
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  3.4294250   12.32321   3.928910     0.0000
                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.8788181  7.7745767   0.000000   0.000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  1.3182272  0.9718221   0.000000   1.957360
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  5.7123177  0.9718221   3.842950   0.000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  5.2729086  1.9436442   0.000000   4.893399
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  4.3940905  1.9436442   5.123934   4.893399
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  2.6364543  2.9154662   5.123934   6.850759
                       RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   4.333240  0.0000000   0.000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   1.280240   0.000000  0.6750084   2.013953
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   7.681439   5.416550  6.7500843   3.020929
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   2.560480   9.749789  1.3500169   2.013953
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   3.249930  0.6750084   2.013953
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   0.000000   2.166620  1.3500169   8.055810
                       RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   6.825663   6.879404  1.6522753
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   5.688053  15.724353  2.7537921
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   2.830176   1.137611   6.224223  3.8553090
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   5.688053   3.603497  2.7537921
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  14.150880   3.412832   1.965544  0.5507584
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   2.830176   3.412832   3.931088  1.1015169
                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000   0.000000   5.534635
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   8.328238   0.000000   0.000000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   1.417949   0.000000   3.904196   4.842806
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   1.417949   0.000000   5.856295   0.000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000   0.000000   9.685611
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   4.253846   0.000000   0.000000   2.075488
                       VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   1.980640  7.5143298   0.000000   4.154433
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   1.320426  2.7324836   0.000000   0.000000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   2.640853  2.0493627   1.631661   1.246330
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   1.320426  0.6831209   0.000000   6.647093
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   1.980640  0.0000000   0.000000   2.908103
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   3.301066  3.4156045   2.447491   4.985320
                       VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   3.342246   4.512607  0.9973314   6.873047
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   1.909855   1.611645  1.4959971   0.000000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   3.867949  2.4933284   0.000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   7.639419   1.289316  0.9973314   0.000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   2.578632  2.9919941   5.154785
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   7.161955   2.256303  3.9893255   1.718262
                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   7.283093   0.000000  0.4510524   0.000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   1.820773   0.000000  4.5105239   0.000000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   6.676168   2.376767  4.5105239   2.151109
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   2.427698   0.000000  4.9615763   1.434073
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   3.641546   5.545791  1.3531572   4.302218
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   7.283093   0.000000  3.6084191   7.170364
                       VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   3.438839  1.9870109   3.675126   1.074698
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   1.375536  0.9935055   0.000000   5.373492
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   1.375536  2.9805164   1.837563   8.060239
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   2.751072  5.9610327   5.512689   1.074698
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  4.9675273   5.512689   1.612048
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   8.253215 10.9285600   2.756344   3.224095
                       VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_18
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   6.346543   0.624967   4.969028    0.00000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   1.874901   4.969028   31.88670
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   1.586636   6.249670   2.839445    0.00000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   3.749802   1.419722    0.00000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   1.586636   3.124835   7.098612   21.25780
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   1.586636   6.874637   4.259167   23.38358
                       VA_W_03_22 VA_W_04_14 VA_W_04_18 VA_W_04_22
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   8.629897  2.4125279   5.302175  3.6980510
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   1.150653  0.9650112   0.000000  1.8490255
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   1.725979  1.4475168   3.976631  2.3112819
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   2.301306  0.9650112   0.000000  0.4622564
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   1.725979  0.0000000   0.000000  0.4622564
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   1.725979  4.8250559   1.767392  0.4622564
                       VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0  1.0031712  11.297065   0.000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral          0  0.0000000  12.426772   1.833667
TR10768<span class="pl-k">|</span>c0_g1_i1_coral          0  6.5206127   0.000000   2.750501
TR10773<span class="pl-k">|</span>c0_g1_i1_coral          0  0.0000000   0.000000   0.000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral          0  0.5015856   1.129707   1.833667
TR11938<span class="pl-k">|</span>c0_g1_i2_coral          0  2.5079280   0.000000   1.833667
                       VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.7941787  1.5851950  1.4303970   1.022439
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  1.5883573  3.9629876  2.1455955   0.000000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  5.5592506  0.0000000  2.8607940   8.179514
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  1.5883573  0.0000000  0.0000000   0.000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  2.3825360  0.7925975  0.7151985   4.089757
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  0.7941787  2.3777926  1.4303970   0.000000
                       VA_W_07_22
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  6.0088720
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  3.4336412
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  3.4336412
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  2.5752309
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  0.8584103
<span class="pl-k">&gt;</span> temp = as.matrix(rowMeans(data4heatmap))
<span class="pl-k">&gt;</span> head(temp)
                           [,1]
TR10090<span class="pl-k">|</span>c0_g1_i1_coral 2.414669
TR10450<span class="pl-k">|</span>c0_g1_i1_coral 3.062953
TR10768<span class="pl-k">|</span>c0_g1_i1_coral 4.113910
TR10773<span class="pl-k">|</span>c0_g1_i1_coral 2.742998
TR11771<span class="pl-k">|</span>c0_g1_i1_coral 2.865582
TR11938<span class="pl-k">|</span>c0_g1_i2_coral 3.179912
<span class="pl-k">&gt;</span> scaledmatrix<span class="pl-k">&lt;</span>-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)
<span class="pl-k">&gt;</span> data4heatmapscaled = data4heatmap/scaledmatrix
<span class="pl-k">&gt;</span> head(data4heatmapscaled)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  1.0470877  0.4747967   0.000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.8254680  0.0000000   0.000000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   2.636938  1.0243174  1.1147318   1.727606
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  2.1507597  0.8359295   1.295519
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.8823235  1.2002554   0.000000
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   0.000000  0.7951069  0.0000000   0.000000
                       RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_22
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   5.422948   0.000000  0.0000000  0.0000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   2.111764  0.0000000  0.4144100
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   2.096379  0.3002969  0.3085431
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000  0.4503810  0.0000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000  0.4311147  1.7718129
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   0.000000   1.356062  0.7769990  1.1975035
                       RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.8400788  1.6838720   2.575782 0.22251707
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.7947279  0.1327475   6.091827 0.08771028
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  0.1972344  1.1860229   0.000000 0.71833786
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  0.1482317   0.000000 0.00000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  0.5663107  1.4189067   0.000000 0.37500589
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  0.0000000  0.3835949   2.933884 1.35174810
                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.0000000   0.000000  0.8131205
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   4.956701  2.0740929   0.000000  0.4273471
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.3860591   0.000000  1.1136133
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.0000000   0.000000  0.0000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   2.408228  0.5542374   4.218031  1.5987348
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   1.302107  0.9989034   0.000000  0.6174437
                       RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22
TR10090<span class="pl-k">|</span>c0_g1_i1_coral    0.00000  0.2832047  0.8192517    0.00000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   13.71305  0.6697903  1.9375626    0.00000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral    0.00000  0.6649104  2.1638771   11.09917
TR10773<span class="pl-k">|</span>c0_g1_i1_coral    0.00000  0.0000000  0.3605947    0.00000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral    0.00000  0.7159232  2.7613540    0.00000
TR11938<span class="pl-k">|</span>c0_g1_i2_coral    0.00000  0.0000000  0.0000000    0.00000
                       RI_W_01_14 RI_W_01_18 RI_W_01_22 RI_W_02_14
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  4.8144470  0.6031645  0.0000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.0000000  0.0000000  1.3149687
TR10768<span class="pl-k">|</span>c0_g1_i1_coral   4.196117  1.8284914  0.3540287  0.0000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  2.7423457  5.8406405  0.0000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.7159184  1.0165074  0.0000000
TR11938<span class="pl-k">|</span>c0_g1_i2_coral   0.000000  1.2903015  0.4580134  0.6333018
                       RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.8104611  0.0000000  1.6726989  0.4426584
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  0.3910675  0.3767618  0.0000000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  1.1892549  1.4558187  1.1220507  0.6928515
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  5.3508832  2.1834164  7.9934703  4.5461892
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  2.0487936  5.0160350  1.6108477  0.6216733
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  0.9231364  3.3901547  1.2701653  0.4481773
                       RI_W_03_22 RI_W_04_14 RI_W_04_18 RI_W_04_22
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.6086771   0.000000  1.4237135    0.00000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000   0.000000  1.6033994    0.00000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  1.6672338   0.000000  0.7162729    4.34781
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  9.8233684   0.000000  0.7161707    0.00000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  0.1709663   0.000000  0.6855345    0.00000
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  1.0784654   3.875331  1.2355405    0.00000
                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.3639498  3.2197281  0.0000000  0.0000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.4303778  0.3172827  0.0000000  0.6390432
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  1.3885373  0.2362283  0.9341357  0.0000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  1.9223157  0.7085838  0.0000000  1.7839600
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  1.5334026  0.6782721  1.7880955  1.7076460
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  0.8290966  0.9168387  1.6113445  2.1543863
                       RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  1.7945483  0.0000000  0.0000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.4179756  0.0000000  0.2203783  0.6575198
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  1.8671868  1.3166426  1.6407952  0.7343205
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.9334602  3.5544277  0.4921683  0.7342157
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  1.1341257  0.2355572  0.7028076
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  0.0000000  0.6813458  0.4245453  2.5333437
                       RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  2.8267494  2.8490054  0.6842658
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  1.8570484  5.1337224  0.8990643
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  0.6879528  0.2765278  1.5129700  0.9371398
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  2.0736626  1.3137075  1.0039351
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  4.9382226  1.1909734  0.6859145  0.1921978
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  0.8900171  1.0732472  1.2362254  0.3463985
                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000   0.000000  0.0000000  2.2920887
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000   2.719022  0.0000000  0.0000000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  0.3446718   0.000000  0.9490232  1.1771782
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.5169339   0.000000  2.1349975  0.0000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000   0.000000  0.0000000  3.3799808
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  1.3377245   0.000000  0.0000000  0.6526873
                       VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.8202531  3.1119506  0.0000000  1.7204981
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.4310958  0.8921074  0.0000000  0.0000000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  0.6419325  0.4981544  0.3966204  0.3029551
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.4813807  0.2490417  0.0000000  2.4232946
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  0.6911824  0.0000000  0.0000000  1.0148387
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  1.0380998  1.0741192  0.7696726  1.5677539
                       VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  1.3841426  1.8688306  0.4130303  2.8463728
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.6235337  0.5261736  0.4884165  0.0000000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  0.9402122  0.6060726  0.0000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  2.7850616  0.4700390  0.3635917  0.0000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  0.8998635  1.0441141  1.7988618
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  2.2522495  0.7095490  1.2545396  0.5403489
                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  3.0161871  0.0000000  0.1867968  0.0000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.5944502  0.0000000  1.4726061  0.0000000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  1.6228279  0.5777393  1.0964079  0.5228868
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.8850525  0.0000000  1.8088149  0.5228121
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  1.2707878  1.9353107  0.4722103  1.5013421
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  2.2903441  0.0000000  1.1347544  2.2548938
                       VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  1.4241454  0.8228917  1.5220000  0.4450708
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.4490880  0.3243619  0.0000000  1.7543500
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  0.3343621  0.7244972  0.4466706  1.9592646
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  1.0029432  2.1731814  2.0097310  0.3917970
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  1.7335145  1.9237590  0.5625552
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  2.5954223  3.4367492  0.8667989  1.0138946
                       VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_18
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  2.6283285  0.2588210  2.0578510   0.000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  0.6121219  1.6222997  10.410443
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  0.3856758  1.5191556  0.6902058   0.000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  1.3670450  0.5175805   0.000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  0.5536871  1.0904714  2.4771976   7.418320
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  0.4989558  2.1618953  1.3393978   7.353531
                       VA_W_03_22 VA_W_04_14 VA_W_04_18 VA_W_04_22
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  3.5739467  0.9991134  2.1958187  1.5314941
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.3756678  0.3150590  0.0000000  0.6036740
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  0.4195472  0.3518591  0.9666305  0.5618212
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.8389746  0.3518089  0.0000000  0.1685223
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  0.6023138  0.0000000  0.0000000  0.1613133
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  0.5427759  1.5173552  0.5557989  0.1453677
                       VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0  0.4154488  4.6785155  0.0000000
TR10450<span class="pl-k">|</span>c0_g1_i1_coral          0  0.0000000  4.0571206  0.5986599
TR10768<span class="pl-k">|</span>c0_g1_i1_coral          0  1.5850158  0.0000000  0.6685856
TR10773<span class="pl-k">|</span>c0_g1_i1_coral          0  0.0000000  0.0000000  0.0000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral          0  0.1750380  0.3942329  0.6398936
TR11938<span class="pl-k">|</span>c0_g1_i2_coral          0  0.7886784  0.0000000  0.5766409
                       VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.3288976  0.6564855  0.5923782  0.4234284
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  0.5185705  1.2938452  0.7004989  0.0000000
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  1.3513301  0.0000000  0.6953953  1.9882577
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.5790588  0.0000000  0.0000000  0.0000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  0.8314319  0.2765922  0.2495823  1.4271996
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  0.2497486  0.7477542  0.4498228  0.0000000
                       VA_W_07_22
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  2.4884871
TR10450<span class="pl-k">|</span>c0_g1_i1_coral  1.1210230
TR10768<span class="pl-k">|</span>c0_g1_i1_coral  0.8346417
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000
TR11771<span class="pl-k">|</span>c0_g1_i1_coral  0.8986765
TR11938<span class="pl-k">|</span>c0_g1_i2_coral  0.2699478
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> dim(data4heatmapscaled)
[1] 343  81
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> pairs.breaks <span class="pl-k">&lt;</span>- seq(0, 3.0, by=0.1)
<span class="pl-k">&gt;</span> length(pairs.breaks)
[1] 31
<span class="pl-k">&gt;</span> mycol <span class="pl-k">&lt;</span>- colorpanel(n=30, low=<span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, high=<span class="pl-s"><span class="pl-pds">"</span>yellow<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> pdf(file=<span class="pl-s"><span class="pl-pds">"</span>XXXXXX_byrow.pdf<span class="pl-pds">"</span></span>,14,7)
<span class="pl-k">&gt;</span> heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(<span class="pl-s"><span class="pl-pds">"</span>row<span class="pl-pds">"</span></span>), scale=<span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, keysize=1, breaks=pairs.breaks, col=mycol, trace = <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, symkey = F, density.info = <span class="pl-s"><span class="pl-pds">"</span>density<span class="pl-pds">"</span></span>, colsep=c(24), sepcolor=c(<span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
<span class="pl-k">&gt;</span> <span class="pl-en">dev.off</span>()
RStudioGD 
        2 
<span class="pl-k">&gt;</span> pdf(file=<span class="pl-s"><span class="pl-pds">"</span>XXXXXX_bycolumn.pdf<span class="pl-pds">"</span></span>,14,7)
<span class="pl-k">&gt;</span> heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(<span class="pl-s"><span class="pl-pds">"</span>col<span class="pl-pds">"</span></span>), scale=<span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, keysize=1, breaks=pairs.breaks, col=mycol, trace = <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, symkey = F, density.info = <span class="pl-s"><span class="pl-pds">"</span>density<span class="pl-pds">"</span></span>, colsep=c(24), sepcolor=c(<span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
<span class="pl-k">&gt;</span> <span class="pl-en">dev.off</span>()
RStudioGD 
        2
<span class="pl-k">&gt;</span> plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> prop.nullv3 <span class="pl-k">&lt;</span>- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100<span class="pl-k">*</span>mean(x==0))
<span class="pl-k">&gt;</span> pdf(file=<span class="pl-s"><span class="pl-pds">"</span>ResNullCounts.pdf<span class="pl-pds">"</span></span>,14, 14)
<span class="pl-k">&gt;</span> barplot(prop.nullv3, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)
<span class="pl-k">&gt;</span> <span class="pl-en">dev.off</span>()
RStudioGD 
        2 
</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day12 HW
                     </h2>
                     <a aria-label="Permalink: Day12 HW" class="anchor" href="#day12-hw" id="user-content-day12-hw">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#1- mkdir a directory with your name in the 

#2- cp/mv your logfile to the /cm/shared/courses/dbarshis/18AdvBioinf/archive/YOURNAME directory

#3- cp/mv your merged, unfiltered .vcf file YOURINITIALS_unfiltered.vcf into the /cm/shared/courses/dbarshis/18AdvBioinf/archive/YOURNAME directory

#4- cp/mv any other files you might want to keep to the /cm/shared/courses/dbarshis/18AdvBioinf/archive/YOURNAME directory

#5- create an sbatch script to rm -r your sandbox/YOURNAME directory

#6- scp/winscp /cm/shared/courses/dbarshis/18AdvBioinf/classdata/Astrangia_poculata/expressioncounts/ALLLANESFullCounts_summed.txt to your laptop

#7- scp/winscp /cm/shared/courses/dbarshis/18AdvBioinf/classdata/Astrangia_poculata/expressioncounts/ALLLANES_conditions.txt to your laptop

#8- Download the DESeq2Script_advbioinf_day12.R script and work through to generate a PCA of your data.


&gt; library(DESeq2)
&gt; library(gplots)
&gt; library(goseq)
&gt; library(GO.db)
&gt; 
&gt; setwd(&quot;/Users/katiecrider/Desktop/Genomics&quot;)  # The drag and drop from finder works in R, too.
&gt; 
&gt; #useful functions
&gt; #head() - prints out the top 6 lines
&gt; #dim() - prints the dimensions of a variable
&gt; #nrow() - returns the number of rows in a vector or matrix
&gt; # ?[functionName] - opens documentation describing the function
&gt; 
&gt; #read in your data to make counts table
&gt; countsTable &lt;- read.delim('djbFullCounts_summed.txt', header=TRUE, stringsAsFactors=TRUE, row.names=1)
&gt; head(countsTable)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          0          1          6          0          0          0
TR10083|c0_g2_i1_coral          0          1          2          0          1          2
TR10090|c0_g1_i1_coral          0          3          1          0          4          0
TR10090|c0_g2_i1_coral          0          4          2          1          0          0
TR10103|c0_g1_i1_coral          0          1          4          0          0          4
                       RI_B_03_14 RI_B_03_18 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          0          0          0          9          1          0
TR10083|c0_g2_i1_coral         10          0          1         31         20          0
TR10090|c0_g1_i1_coral          0          0          0          5         10          2
TR10090|c0_g2_i1_coral          0          0          3         12          9          0
TR10103|c0_g1_i1_coral          1          0          3         15          8          0
                       RI_B_05_14 RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          2          0          0          0          2          0
TR10083|c0_g2_i1_coral         16          0          3          3          5          6
TR10090|c0_g1_i1_coral          2          0          0          0          3          0
TR10090|c0_g2_i1_coral          2          0          3          0          7          0
TR10103|c0_g1_i1_coral         18          0          5          0          5          0
                       RI_B_07_14 RI_B_07_18 RI_B_07_22 RI_W_01_14 RI_W_01_18 RI_W_01_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          2          1          0          0          0          1
TR10083|c0_g2_i1_coral          2          8          4         11         15          3
TR10090|c0_g1_i1_coral          1          2          0          0         17          1
TR10090|c0_g2_i1_coral          5          1          0          0         17          5
TR10103|c0_g1_i1_coral          3          2          0          5         11          0
                       RI_W_02_14 RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          1          1          2          0          5          0
TR10083|c0_g2_i1_coral          0          5          3         14         15          9
TR10090|c0_g1_i1_coral          0          2          0          7          3          3
TR10090|c0_g2_i1_coral          2          5          4          5          9          3
TR10103|c0_g1_i1_coral          0          1          3          5          9          2
                       RI_W_04_14 RI_W_04_18 RI_W_04_22 RI_W_05_14 RI_W_05_18 RI_W_05_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          1          1          0          2          0          0
TR10083|c0_g2_i1_coral          3          6          1          7          1          4
TR10090|c0_g1_i1_coral          0          7          0          2          8          0
TR10090|c0_g2_i1_coral          0         15          0          8         10          1
TR10103|c0_g1_i1_coral          0          2          0          9          2          2
                       RI_W_06_14 RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18 RI_W_07_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          2          6          0          4          0          0
TR10083|c0_g2_i1_coral          1          1          5          6          5          5
TR10090|c0_g1_i1_coral          0          0          4          0          0          0
TR10090|c0_g2_i1_coral          3          1          7          3          1          4
TR10103|c0_g1_i1_coral         10          1          2          8         12          1
                       VA_B_01_14 VA_B_01_18 VA_B_01_22 VA_B_02_14 VA_B_02_18 VA_B_02_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          0          2          2          0          0          0
TR10083|c0_g2_i1_coral          2          4          5          0         11          9
TR10090|c0_g1_i1_coral          6         21          3          0          0          0
TR10090|c0_g2_i1_coral          8         13         10          0          0          0
TR10103|c0_g1_i1_coral          3          6          5          0          0          1
                       VA_B_03_14 VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18 VA_B_04_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          1          0          6          0         10          4
TR10083|c0_g2_i1_coral         22         20         11          0          2         13
TR10090|c0_g1_i1_coral          8          3         11          0         10          7
TR10090|c0_g2_i1_coral         15          6          5          8          7          0
TR10103|c0_g1_i1_coral          3          6          5          5          0          0
                       VA_B_05_14 VA_B_05_18 VA_B_05_22 VA_B_06_14 VA_B_06_18 VA_B_06_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          2          2          0         10          0          1
TR10083|c0_g2_i1_coral          6          6          1          2          0         12
TR10090|c0_g1_i1_coral         14          2          4         12          0          1
TR10090|c0_g2_i1_coral          4          1          3          0          5          3
TR10103|c0_g1_i1_coral          0          6          0          8          8          8
                       VA_B_07_14 VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18 VA_W_01_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          0          4          1          1          3          0
TR10083|c0_g2_i1_coral          5          1          2          7         14          0
TR10090|c0_g1_i1_coral          0          5          2          4          2          0
TR10090|c0_g2_i1_coral          2         21          6          0          7          0
TR10103|c0_g1_i1_coral          2         11          4          0          7          0
                       VA_W_02_14 VA_W_02_18 VA_W_02_22 VA_W_03_14 VA_W_03_18 VA_W_03_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          4          0          0          4          0          6
TR10083|c0_g2_i1_coral          0          7          0          4          0         11
TR10090|c0_g1_i1_coral          8          1          0          7          0         15
TR10090|c0_g2_i1_coral          5         20          0          9         17          5
TR10103|c0_g1_i1_coral          6          1          0          9          0          4
                       VA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          2          0          2          0          3          0
TR10083|c0_g2_i1_coral          2          9         13          0          7         10
TR10090|c0_g1_i1_coral          5         12          8          0          2         10
TR10090|c0_g2_i1_coral          4         16         22          0          5          6
TR10103|c0_g1_i1_coral          7          8          6          1          6          2
                       VA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          2          0          0          0          0          3
TR10083|c0_g2_i1_coral          5          3          0         13          1          1
TR10090|c0_g1_i1_coral          0          1          2          2          1          7
TR10090|c0_g2_i1_coral          3          1          0          7          3          3
TR10103|c0_g1_i1_coral          8          3          5          6          0          2
&gt; dim(countsTable)
[1] 13381    84
&gt; 
&gt; colSums(countsTable)
RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_18 
     97406     207372     173843      47481      58036      84501     170330       5972 
RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14 RI_B_05_18 RI_B_05_22 RI_B_06_14 
    199204     357124     544267      71600     630609     227313     225639      37012 
RI_B_06_18 RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22 RI_W_01_14 RI_W_01_18 RI_W_01_22 
    296071      72254     257730     218198      25369      39419     292612     174666 
RI_W_02_14 RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18 
     54184     192092     155178     281297     428532     359053      37579     300578 
RI_W_04_22 RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22 RI_W_07_14 
     10848     392818     214887     158683     186568     205135     198272     342795 
RI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22 VA_B_02_14 VA_B_02_18 VA_B_02_22 
    220630     176946     261824     548448     328198      79276     152742      85224 
VA_B_03_14 VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 
    226784     251696     271959     172636     264192     268290     323715     288121 
VA_B_05_22 VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22 VA_W_01_14 
     89943     232578     195521     362054     215616     234465     169742     164874 
VA_W_01_18 VA_W_01_22 VA_W_02_14 VA_W_02_18 VA_W_02_22 VA_W_03_14 VA_W_03_18 VA_W_03_22 
    321900        313     170250     241804        134     278000     138573     295382 
VA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14 VA_W_06_18 
    292041     280007     261344     178096     388532     157123     176579     188506 
VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22 
    209003     228792     171481     233899 
&gt; colSums(countsTable)[order(colSums(countsTable))]
VA_W_02_22 VA_W_01_22 RI_B_03_18 RI_W_04_22 RI_B_07_22 RI_B_06_14 RI_W_04_14 RI_W_01_14 
       134        313       5972      10848      25369      37012      37579      39419 
RI_B_02_14 RI_W_02_14 RI_B_02_18 RI_B_04_22 RI_B_06_22 VA_B_02_14 RI_B_02_22 VA_B_02_22 
     47481      54184      58036      71600      72254      79276      84501      85224 
VA_B_05_22 RI_B_01_14 VA_W_03_18 VA_B_02_18 RI_W_02_22 VA_W_05_22 RI_W_05_22 VA_W_01_14 
     89943      97406     138573     152742     155178     157123     158683     164874 
VA_B_07_22 VA_W_02_14 RI_B_03_14 VA_W_07_18 VA_B_04_14 RI_B_01_22 RI_W_01_22 VA_W_06_14 
    169742     170250     170330     171481     172636     173843     174666     176579 
RI_W_07_22 VA_W_05_14 RI_W_06_14 VA_W_06_18 RI_W_02_18 VA_B_06_18 RI_W_06_22 RI_B_03_22 
    176946     178096     186568     188506     192092     195521     198272     199204 
RI_W_06_18 RI_B_01_18 VA_W_06_22 RI_W_05_18 VA_B_07_14 RI_B_07_18 RI_W_07_18 RI_B_05_22 
    205135     207372     209003     214887     215616     218198     220630     225639 
VA_B_03_14 RI_B_05_18 VA_W_07_14 VA_B_06_14 VA_W_07_22 VA_B_07_18 VA_W_02_18 VA_B_03_18 
    226784     227313     228792     232578     233899     234465     241804     251696 
RI_B_07_14 VA_W_04_22 VA_B_01_14 VA_B_04_18 VA_B_04_22 VA_B_03_22 VA_W_03_14 VA_W_04_18 
    257730     261344     261824     264192     268290     271959     278000     280007 
RI_W_03_14 VA_B_05_18 VA_W_04_14 RI_W_01_18 VA_W_03_22 RI_B_06_18 RI_W_04_18 VA_W_01_18 
    281297     288121     292041     292612     295382     296071     300578     321900 
VA_B_05_14 VA_B_01_22 RI_W_07_14 RI_B_04_14 RI_W_03_22 VA_B_06_22 VA_W_05_18 RI_W_05_14 
    323715     328198     342795     357124     359053     362054     388532     392818 
RI_W_03_18 RI_B_04_18 VA_B_01_18 RI_B_05_14 
    428532     544267     548448     630609 
&gt; 
&gt; table(substring(names(countsTable),1,2))

RI VA 
42 42 
&gt; table(substring(names(countsTable),4,4))

 B  W 
42 42 
&gt; table(substring(names(countsTable),9,10))

14 18 22 
28 28 28 
&gt; table(paste(substring(names(countsTable),1,2),substring(names(countsTable),4,4), sep=&quot;_&quot;))

RI_B RI_W VA_B VA_W 
  21   21   21   21 
&gt; table(paste(substring(names(countsTable),1,2),substring(names(countsTable),4,4), substring(names(countsTable),9,10), sep=&quot;_&quot;))

RI_B_14 RI_B_18 RI_B_22 RI_W_14 RI_W_18 RI_W_22 VA_B_14 VA_B_18 VA_B_22 VA_W_14 VA_W_18 VA_W_22 
      7       7       7       7       7       7       7       7       7       7       7       7 
&gt; 
&gt; LowCounts&lt;-c(&quot;VA_W_02_22&quot;,&quot;VA_W_01_22&quot;,&quot;RI_B_03_18&quot;, &quot;RI_W_01_14&quot;, &quot;RI_W_04_22&quot;, &quot;VA_W_03_18&quot;)
&gt; names(countsTable)%in%LowCounts
 [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[16] FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[31] FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[46] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[61] FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE FALSE  TRUE FALSE FALSE FALSE FALSE
[76] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
&gt; !(names(countsTable)%in%LowCounts)
 [1]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[16]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[31]  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[46]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[61]  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE FALSE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE
[76]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
&gt; countsTable&lt;-countsTable[,!(names(countsTable)%in%LowCounts)]
&gt; dim(countsTable)
[1] 13381    78
&gt; 
&gt; #make a table with conditions of each individual (e.g. &quot;VA&quot; and &quot;RI&quot;)  There should be the same number of conditions described as there are samples in your data file, and in the same order.
&gt; #NOTE: It is absolutely critical that the columns of the count matrix and the rows of the column data (information about samples) are in the same order. DESeq2 will not make guesses as to which column of the count matrix belongs to which row of the column data, these must be provided to DESeq2 already in consistent order.
&gt; #Here's an example
&gt; # Sample		Origin	SymState	Temp  Origin_SymState Origin_SymState_Temp
&gt; # RI_B_06_18		RI	B	18  RI_B  RI_B_18
&gt; # VA_B_07_14		VA	B	14  VA_B  VA_B_14
&gt; # VA_W_07_22		VA	W	22  VA_W  VA_W_22
&gt; # safest way to do this is in R
&gt; 
&gt; conds&lt;-data.frame(&quot;Sample&quot;=names(countsTable))
&gt; conds$Origin&lt;-factor(substring(conds$Sample,1,2))
&gt; conds$SymState&lt;-factor(substring(conds$Sample,4,4))
&gt; conds$Temp&lt;-factor(substring(conds$Sample,9,10))
&gt; conds$Origin_SymState&lt;-factor(paste(conds$Origin,conds$SymState, sep=&quot;_&quot;))
&gt; conds$Origin_SymState_Temp&lt;-factor(paste(conds$Origin,conds$SymState,conds$Temp, sep=&quot;_&quot;))
&gt; 
&gt; ##Check null counts per sample
&gt; prop.null &lt;- apply(countsTable, 2, function(x) 100*mean(x==0))
&gt; 
&gt; barplot(prop.null, main=&quot;Percentage of null counts per sample&quot;, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
&gt; 
&gt; pdf(file=&quot;OverallNullCountsv2.pdf&quot;,14, 14)
&gt; barplot(prop.null, main=&quot;Percentage of null counts per sample&quot;, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
&gt; dev.off()
RStudioGD 
        2 
&gt; 
&gt; # how many genes have mean count &gt;=3 
&gt; means = apply(countsTable, 1, mean)
&gt; table(means&gt;=3)

FALSE  TRUE 
 5721  7660 
&gt; 
&gt; countsTable&lt;-countsTable[means&gt;=3,]
&gt; dim(countsTable)
[1] 7660   78
&gt; 
&gt; prop.nullv2 &lt;- apply(countsTable, 2, function(x) 100*mean(x==0))
&gt; 
&gt; barplot(prop.nullv2, main=&quot;Percentage of null counts per sample&quot;, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
&gt; 
&gt; pdf(file=&quot;Mean3NullCountsv2.pdf&quot;,14, 14)
&gt; barplot(prop.nullv2, main=&quot;Percentage of null counts per sample&quot;, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
&gt; dev.off()
RStudioGD 
        2 
&gt; 
&gt; #make count data sets
&gt; dds &lt;- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=~ Origin + SymState + Temp + Origin:SymState)
&gt; 
&gt; dds &lt;- DESeq(dds)
estimating size factors
estimating dispersions
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
fitting model and testing
-- replacing outliers and refitting for 14 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)
estimating dispersions
fitting model and testing
&gt; 
&gt; dim(dds)
[1] 7660   78
&gt; 
&gt; #transform counts to variance stablized counts
&gt; vstCounts&lt;-varianceStabilizingTransformation(dds)
&gt; 
&gt; #plot cluster dendrogram across samples
&gt; dists&lt;-dist(t(assay(vstCounts)))
&gt; plot(hclust(dists))
&gt; heatmap.2(as.matrix(dists), key=F, trace=&quot;none&quot;,
+           col=colorpanel(100, &quot;black&quot;, &quot;white&quot;),
+           margin=c(10, 10))
&gt; 
&gt; ###Figure out which contrast you want to examine (i.e. which two groups do you want to compare)
&gt; resultsNames(dds)
[1] &quot;Intercept&quot;          &quot;Origin_VA_vs_RI&quot;    &quot;SymState_W_vs_B&quot;    &quot;Temp_18_vs_14&quot;     
[5] &quot;Temp_22_vs_14&quot;      &quot;OriginVA.SymStateW&quot;
&gt; 
&gt; res &lt;- results(dds)
&gt; head(res)
log2 fold change (MLE): OriginVA.SymStateW 
Wald test p-value: OriginVA.SymStateW 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj
                       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
TR10083|c0_g2_i1_coral   5.36648       0.107690  0.644646  0.167053  0.867328  0.999364
TR10090|c0_g1_i1_coral   2.70436      -0.247209  0.831371 -0.297351  0.766198  0.995209
TR10090|c0_g2_i1_coral   3.65756      -0.976997  0.654110 -1.493627  0.135273  0.958807
TR10103|c0_g1_i1_coral   2.91279      -0.172220  0.641026 -0.268662  0.788189  0.995209
TR10108|c0_g1_i1_coral   6.31144       0.366046  0.664401  0.550942  0.581673  0.991107
TR10132|c0_g1_i1_coral   9.26737      -0.589357  0.975809 -0.603967  0.545865  0.990477
&gt; summary(res)

out of 7659 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 2, 0.026%
LFC &lt; 0 (down)     : 9, 0.12%
outliers [1]       : 10, 0.13%
low counts [2]     : 1, 0.013%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

&gt; 
&gt; resSymState &lt;- results(dds, contrast=c(&quot;SymState&quot;, &quot;B&quot;, &quot;W&quot;))
&gt; head(resSymState)
log2 fold change (MLE): SymState B vs W 
Wald test p-value: SymState B vs W 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj
                       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
TR10083|c0_g2_i1_coral   5.36648       0.304875  0.463588  0.657643 0.5107675  0.986343
TR10090|c0_g1_i1_coral   2.70436      -0.340214  0.624845 -0.544477 0.5861132  0.995961
TR10090|c0_g2_i1_coral   3.65756      -1.044272  0.488673 -2.136956 0.0326015  0.816270
TR10103|c0_g1_i1_coral   2.91279      -0.316211  0.469123 -0.674047 0.5002815  0.986343
TR10108|c0_g1_i1_coral   6.31144       0.257974  0.481341  0.535948 0.5919943  0.995961
TR10132|c0_g1_i1_coral   9.26737      -0.604957  0.692218 -0.873939 0.3821514  0.986343
&gt; summary(resSymState)

out of 7659 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 6, 0.078%
LFC &lt; 0 (down)     : 6, 0.078%
outliers [1]       : 10, 0.13%
low counts [2]     : 1, 0.013%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

&gt; 
&gt; resOrigin &lt;- results(dds, contrast=c(&quot;Origin&quot;, &quot;VA&quot;, &quot;RI&quot;))
&gt; head(resOrigin)
log2 fold change (MLE): Origin VA vs RI 
Wald test p-value: Origin VA vs RI 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE       stat    pvalue      padj
                       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
TR10083|c0_g2_i1_coral   5.36648     -0.2712427  0.446225 -0.6078616 0.5432793  0.865661
TR10090|c0_g1_i1_coral   2.70436      1.1125117  0.589306  1.8878328 0.0590484        NA
TR10090|c0_g2_i1_coral   3.65756      0.9431346  0.474719  1.9867203 0.0469534  0.418043
TR10103|c0_g1_i1_coral   2.91279      0.0386643  0.456720  0.0846565 0.9325345  0.984590
TR10108|c0_g1_i1_coral   6.31144     -0.0179010  0.461958 -0.0387502 0.9690895  0.992150
TR10132|c0_g1_i1_coral   9.26737     -0.2649903  0.677608 -0.3910675 0.6957473  0.921517
&gt; summary(resOrigin)

out of 7659 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 73, 0.95%
LFC &lt; 0 (down)     : 104, 1.4%
outliers [1]       : 10, 0.13%
low counts [2]     : 594, 7.8%
(mean count &lt; 3)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

&gt; 
&gt; res14v18 &lt;- results(dds, contrast=c(&quot;Temp&quot;, &quot;14&quot;, &quot;18&quot;))
&gt; head(res14v18)
log2 fold change (MLE): Temp 14 vs 18 
Wald test p-value: Temp 14 vs 18 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE       stat    pvalue      padj
                       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
TR10083|c0_g2_i1_coral   5.36648      0.0134106  0.389675  0.0344147 0.9725464  0.994763
TR10090|c0_g1_i1_coral   2.70436     -0.7500403  0.493720 -1.5191625 0.1287216        NA
TR10090|c0_g2_i1_coral   3.65756     -0.5278497  0.385227 -1.3702295 0.1706153        NA
TR10103|c0_g1_i1_coral   2.91279      0.1448997  0.377624  0.3837147 0.7011899        NA
TR10108|c0_g1_i1_coral   6.31144     -0.1171316  0.401865 -0.2914700 0.7706919  0.956296
TR10132|c0_g1_i1_coral   9.26737     -1.1666726  0.592603 -1.9687238 0.0489848  0.440134
&gt; summary(res14v18)

out of 7659 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 31, 0.4%
LFC &lt; 0 (down)     : 111, 1.4%
outliers [1]       : 10, 0.13%
low counts [2]     : 2671, 35%
(mean count &lt; 4)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

&gt; 
&gt; res14v22 &lt;- results(dds, contrast=c(&quot;Temp&quot;, &quot;14&quot;, &quot;22&quot;))
&gt; head(res14v22)
log2 fold change (MLE): Temp 14 vs 22 
Wald test p-value: Temp 14 vs 22 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj
                       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
TR10083|c0_g2_i1_coral   5.36648      -0.381080  0.396906 -0.960127 0.3369915  0.796010
TR10090|c0_g1_i1_coral   2.70436      -0.420391  0.517100 -0.812978 0.4162304        NA
TR10090|c0_g2_i1_coral   3.65756      -0.149468  0.407436 -0.366851 0.7137303        NA
TR10103|c0_g1_i1_coral   2.91279       0.327506  0.400038  0.818686 0.4129654        NA
TR10108|c0_g1_i1_coral   6.31144      -0.504681  0.409220 -1.233275 0.2174733  0.699820
TR10132|c0_g1_i1_coral   9.26737      -1.240058  0.602190 -2.059248 0.0394705  0.387444
&gt; summary(res14v22)

out of 7659 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 64, 0.84%
LFC &lt; 0 (down)     : 97, 1.3%
outliers [1]       : 10, 0.13%
low counts [2]     : 2226, 29%
(mean count &lt; 4)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

&gt; 
&gt; res18v22 &lt;- results(dds, contrast=c(&quot;Temp&quot;, &quot;18&quot;, &quot;22&quot;))
&gt; head(res18v22)
log2 fold change (MLE): Temp 18 vs 22 
Wald test p-value: Temp 18 vs 22 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj
                       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
TR10083|c0_g2_i1_coral   5.36648     -0.3944906  0.395010 -0.998685  0.317947  0.838585
TR10090|c0_g1_i1_coral   2.70436      0.3296494  0.503430  0.654807  0.512592        NA
TR10090|c0_g2_i1_coral   3.65756      0.3783816  0.396935  0.953259  0.340459  0.847759
TR10103|c0_g1_i1_coral   2.91279      0.1826059  0.399214  0.457414  0.647374        NA
TR10108|c0_g1_i1_coral   6.31144     -0.3875492  0.406736 -0.952828  0.340677  0.847759
TR10132|c0_g1_i1_coral   9.26737     -0.0733858  0.596241 -0.123081  0.902043  0.991572
&gt; summary(res18v22)

out of 7659 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 34, 0.44%
LFC &lt; 0 (down)     : 10, 0.13%
outliers [1]       : 10, 0.13%
low counts [2]     : 1337, 17%
(mean count &lt; 3)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results

&gt; 
&gt; #count the number of significantly differentially expressed genes
&gt; sum(resOrigin$padj &lt; 0.1, na.rm =T)
[1] 177
&gt; sum(resOrigin$pvalue &lt; 0.05, na.rm =T)
[1] 866
&gt; 
&gt; # Make a counts table that is scaled by the size factors
&gt; scaledcounts = counts(dds, normalize=T)
&gt; head(scaledcounts)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10083|c0_g2_i1_coral          0  0.8671827   2.447866   0.000000   3.455170   4.432653
TR10090|c0_g1_i1_coral          0  2.6015481   1.223933   0.000000  13.820681   0.000000
TR10090|c0_g2_i1_coral          0  3.4687308   2.447866   3.772786   0.000000   0.000000
TR10103|c0_g1_i1_coral          0  0.8671827   4.895733   0.000000   0.000000   8.865306
TR10108|c0_g1_i1_coral          0  1.7343654   4.895733   0.000000   6.910341   0.000000
TR10132|c0_g1_i1_coral          0  6.0702789   1.223933  18.863931  51.827555   0.000000
                       RI_B_03_14 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10083|c0_g2_i1_coral  12.871970   1.272998  12.424473  8.1878079   0.000000   4.563408
TR10090|c0_g1_i1_coral   0.000000   0.000000   2.003947  4.0939039   6.765958   0.570426
TR10090|c0_g2_i1_coral   0.000000   3.818995   4.809473  3.6845136   0.000000   0.570426
TR10103|c0_g1_i1_coral   1.287197   3.818995   6.011842  3.2751232   0.000000   5.133834
TR10108|c0_g1_i1_coral  14.159167  15.275980   2.805526  4.9126847  16.914895   5.989473
TR10132|c0_g1_i1_coral   6.435985   2.545997   0.000000  0.8187808   3.382979   1.426065
                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22 RI_B_07_14
TR10083|c0_g2_i1_coral          0  2.4538305   19.55961   3.448566   18.78455  1.4936671
TR10090|c0_g1_i1_coral          0  0.0000000    0.00000   2.069140    0.00000  0.7468335
TR10090|c0_g2_i1_coral          0  2.4538305    0.00000   4.827992    0.00000  3.7341677
TR10103|c0_g1_i1_coral          0  4.0897176    0.00000   3.448566    0.00000  2.2405006
TR10108|c0_g1_i1_coral          0  0.8179435    6.51987  10.345698   40.69986  3.7341677
TR10132|c0_g1_i1_coral          0  1.6358870    0.00000   1.379426    0.00000  8.9620025
                       RI_B_07_18 RI_B_07_22 RI_W_01_18 RI_W_01_22 RI_W_02_14 RI_W_02_18
TR10083|c0_g2_i1_coral   8.070182   36.85316  10.815190   4.646396   0.000000  4.9947759
TR10090|c0_g1_i1_coral   2.017545    0.00000  12.257216   1.548799   0.000000  1.9979104
TR10090|c0_g2_i1_coral   1.008773    0.00000  12.257216   7.743994   4.651361  4.9947759
TR10103|c0_g1_i1_coral   2.017545    0.00000   7.931140   0.000000   0.000000  0.9989552
TR10108|c0_g1_i1_coral   1.008773    0.00000   2.884051   1.548799   9.302721  6.9926863
TR10132|c0_g1_i1_coral  44.386000   27.63987   1.442025   0.000000   0.000000  5.9937311
                       RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18
TR10083|c0_g2_i1_coral   3.615216   8.525621   5.407467   4.745758   12.82564  2.9768406
TR10090|c0_g1_i1_coral   0.000000   4.262810   1.081493   1.581919    0.00000  3.4729807
TR10090|c0_g2_i1_coral   4.820288   3.044865   3.244480   1.581919    0.00000  7.4421015
TR10103|c0_g1_i1_coral   3.615216   3.044865   3.244480   1.054613    0.00000  0.9922802
TR10108|c0_g1_i1_coral   2.410144   2.435892   4.686471   2.109226   21.37607 12.8996426
TR10132|c0_g1_i1_coral  44.587667   1.217946   1.081493   1.054613   29.92650  7.9382416
                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22
TR10083|c0_g2_i1_coral  3.1630643   1.010704   5.526636   1.044347   1.336966   5.564881
TR10090|c0_g1_i1_coral  0.9037326   8.085630   0.000000   0.000000   0.000000   4.451904
TR10090|c0_g2_i1_coral  3.6149306  10.107037   1.381659   3.133041   1.336966   7.790833
TR10103|c0_g1_i1_coral  4.0667969   2.021407   2.763318  10.443469   1.336966   2.225952
TR10108|c0_g1_i1_coral  5.4223959   2.021407   6.908296  11.487816   0.000000   5.564881
TR10132|c0_g1_i1_coral  1.8074653  10.107037  27.633182   5.221735   1.336966  42.293092
                       RI_W_07_14 RI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10083|c0_g2_i1_coral   4.124636   5.593017   6.900578   2.551157   1.731769   2.845097
TR10090|c0_g1_i1_coral   0.000000   0.000000   0.000000   7.653472   9.091789   1.707058
TR10090|c0_g2_i1_coral   2.062318   1.118603   5.520462  10.204629   5.628251   5.690194
TR10103|c0_g1_i1_coral   5.499514  13.423242   1.380116   3.826736   2.597654   2.845097
TR10108|c0_g1_i1_coral   0.000000   1.118603   5.520462   3.826736   9.091789   2.845097
TR10132|c0_g1_i1_coral   1.374879   2.237207  86.947282   1.275579   1.298827   0.000000
                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14 VA_B_03_18 VA_B_03_22
TR10083|c0_g2_i1_coral          0    14.2516  19.312637  16.029307  14.339493   7.833217
TR10090|c0_g1_i1_coral          0     0.0000   0.000000   5.828839   2.150924   7.833217
TR10090|c0_g2_i1_coral          0     0.0000   0.000000  10.929073   4.301848   3.560553
TR10103|c0_g1_i1_coral          0     0.0000   2.145849   2.185815   4.301848   3.560553
TR10108|c0_g1_i1_coral          0     0.0000   0.000000  14.572098  18.641340  21.363318
TR10132|c0_g1_i1_coral          0     0.0000   8.583394   9.471863  30.829909   8.545327
                       VA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10083|c0_g2_i1_coral   0.000000  0.9006809   6.731616   2.098865  3.3173156   1.813677
TR10090|c0_g1_i1_coral   0.000000  4.5034046   3.624716   4.897352  1.1057719   7.254707
TR10090|c0_g2_i1_coral   7.108517  3.1523832   0.000000   1.399244  0.5528859   5.441030
TR10103|c0_g1_i1_coral   4.442823  0.0000000   0.000000   0.000000  3.3173156   0.000000
TR10108|c0_g1_i1_coral   2.665694  4.9537450   3.624716   9.095083  5.5288593   0.000000
TR10132|c0_g1_i1_coral   0.000000  2.7020428   0.000000   0.000000  4.9759734   5.441030
                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22
TR10083|c0_g2_i1_coral   1.383946   0.000000  5.9098781   3.728534  0.7405861   2.098499
TR10090|c0_g1_i1_coral   8.303676   0.000000  0.4924898   0.000000  3.7029304   2.098499
TR10090|c0_g2_i1_coral   0.000000   4.206352  1.4774695   1.491414 15.5523076   6.295498
TR10103|c0_g1_i1_coral   5.535784   6.730163  3.9399187   1.491414  8.1464468   4.196999
TR10108|c0_g1_i1_coral  11.763540   4.206352  3.4474289   2.237121  2.2217582   5.246248
TR10132|c0_g1_i1_coral  18.683270   2.523811 14.2822054   8.948482 17.0334798   2.098499
                       VA_W_01_14 VA_W_01_18 VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_22
TR10083|c0_g2_i1_coral   6.699293   7.732656   0.000000  4.5789684  3.1096023   6.888125
TR10090|c0_g1_i1_coral   3.828167   1.104665   6.978382  0.6541383  5.4418040   9.392897
TR10090|c0_g2_i1_coral   0.000000   3.866328   4.361489 13.0827667  6.9966052   3.130966
TR10103|c0_g1_i1_coral   0.000000   3.866328   5.233787  0.6541383  6.9966052   2.504773
TR10108|c0_g1_i1_coral   1.914084   4.418660   0.000000  9.1579367  3.1096023   3.757159
TR10132|c0_g1_i1_coral   0.000000  10.494319   2.616893 13.0827667  0.7774006  13.150056
                       VA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22
TR10083|c0_g2_i1_coral   1.026752   4.334366   6.410680  0.0000000   3.848517  12.379823
TR10090|c0_g1_i1_coral   2.566880   5.779155   3.945034  0.0000000   1.099576  12.379823
TR10090|c0_g2_i1_coral   2.053504   7.705540  10.848842  0.0000000   2.748941   7.427894
TR10103|c0_g1_i1_coral   3.593631   3.852770   2.958775  0.8118702   3.298729   2.475965
TR10108|c0_g1_i1_coral   6.673887  20.227043   8.383196  1.6237403   5.497882  38.377452
TR10132|c0_g1_i1_coral   2.566880   0.000000   0.000000  2.4356105   8.796611   2.475965
                       VA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22
TR10083|c0_g2_i1_coral   5.042735  2.5895188   0.000000 10.2466895   1.127608  0.9473408
TR10090|c0_g1_i1_coral   0.000000  0.8631729   1.764014  1.5764138   1.127608  6.6313859
TR10090|c0_g2_i1_coral   3.025641  0.8631729   0.000000  5.5174482   3.382824  2.8420225
TR10103|c0_g1_i1_coral   8.068376  2.5895188   4.410036  4.7292413   0.000000  1.8946817
TR10108|c0_g1_i1_coral   4.034188  6.9053834   5.292043  3.1528275   1.127608  1.8946817
TR10132|c0_g1_i1_coral   9.076923 29.3478796  26.460214  0.7882069   6.765648  8.5260676

&gt; #building heat map data
&gt; head(scaledcounts)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22
TR10083|c0_g2_i1_coral          0  0.8671827   2.447866
TR10090|c0_g1_i1_coral          0  2.6015481   1.223933
TR10090|c0_g2_i1_coral          0  3.4687308   2.447866
TR10103|c0_g1_i1_coral          0  0.8671827   4.895733
TR10108|c0_g1_i1_coral          0  1.7343654   4.895733
TR10132|c0_g1_i1_coral          0  6.0702789   1.223933
                       RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10083|c0_g2_i1_coral   0.000000   3.455170   4.432653
TR10090|c0_g1_i1_coral   0.000000  13.820681   0.000000
TR10090|c0_g2_i1_coral   3.772786   0.000000   0.000000
TR10103|c0_g1_i1_coral   0.000000   0.000000   8.865306
TR10108|c0_g1_i1_coral   0.000000   6.910341   0.000000
TR10132|c0_g1_i1_coral  18.863931  51.827555   0.000000
                       RI_B_03_14 RI_B_03_22 RI_B_04_14
TR10083|c0_g2_i1_coral  12.871970   1.272998  12.424473
TR10090|c0_g1_i1_coral   0.000000   0.000000   2.003947
TR10090|c0_g2_i1_coral   0.000000   3.818995   4.809473
TR10103|c0_g1_i1_coral   1.287197   3.818995   6.011842
TR10108|c0_g1_i1_coral  14.159167  15.275980   2.805526
TR10132|c0_g1_i1_coral   6.435985   2.545997   0.000000
                       RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10083|c0_g2_i1_coral  8.1878079   0.000000   4.563408
TR10090|c0_g1_i1_coral  4.0939039   6.765958   0.570426
TR10090|c0_g2_i1_coral  3.6845136   0.000000   0.570426
TR10103|c0_g1_i1_coral  3.2751232   0.000000   5.133834
TR10108|c0_g1_i1_coral  4.9126847  16.914895   5.989473
TR10132|c0_g1_i1_coral  0.8187808   3.382979   1.426065
                       RI_B_05_18 RI_B_05_22 RI_B_06_14
TR10083|c0_g2_i1_coral          0  2.4538305   19.55961
TR10090|c0_g1_i1_coral          0  0.0000000    0.00000
TR10090|c0_g2_i1_coral          0  2.4538305    0.00000
TR10103|c0_g1_i1_coral          0  4.0897176    0.00000
TR10108|c0_g1_i1_coral          0  0.8179435    6.51987
TR10132|c0_g1_i1_coral          0  1.6358870    0.00000
                       RI_B_06_18 RI_B_06_22 RI_B_07_14
TR10083|c0_g2_i1_coral   3.448566   18.78455  1.4936671
TR10090|c0_g1_i1_coral   2.069140    0.00000  0.7468335
TR10090|c0_g2_i1_coral   4.827992    0.00000  3.7341677
TR10103|c0_g1_i1_coral   3.448566    0.00000  2.2405006
TR10108|c0_g1_i1_coral  10.345698   40.69986  3.7341677
TR10132|c0_g1_i1_coral   1.379426    0.00000  8.9620025
                       RI_B_07_18 RI_B_07_22 RI_W_01_18
TR10083|c0_g2_i1_coral   8.070182   36.85316  10.815190
TR10090|c0_g1_i1_coral   2.017545    0.00000  12.257216
TR10090|c0_g2_i1_coral   1.008773    0.00000  12.257216
TR10103|c0_g1_i1_coral   2.017545    0.00000   7.931140
TR10108|c0_g1_i1_coral   1.008773    0.00000   2.884051
TR10132|c0_g1_i1_coral  44.386000   27.63987   1.442025
                       RI_W_01_22 RI_W_02_14 RI_W_02_18
TR10083|c0_g2_i1_coral   4.646396   0.000000  4.9947759
TR10090|c0_g1_i1_coral   1.548799   0.000000  1.9979104
TR10090|c0_g2_i1_coral   7.743994   4.651361  4.9947759
TR10103|c0_g1_i1_coral   0.000000   0.000000  0.9989552
TR10108|c0_g1_i1_coral   1.548799   9.302721  6.9926863
TR10132|c0_g1_i1_coral   0.000000   0.000000  5.9937311
                       RI_W_02_22 RI_W_03_14 RI_W_03_18
TR10083|c0_g2_i1_coral   3.615216   8.525621   5.407467
TR10090|c0_g1_i1_coral   0.000000   4.262810   1.081493
TR10090|c0_g2_i1_coral   4.820288   3.044865   3.244480
TR10103|c0_g1_i1_coral   3.615216   3.044865   3.244480
TR10108|c0_g1_i1_coral   2.410144   2.435892   4.686471
TR10132|c0_g1_i1_coral  44.587667   1.217946   1.081493
                       RI_W_03_22 RI_W_04_14 RI_W_04_18
TR10083|c0_g2_i1_coral   4.745758   12.82564  2.9768406
TR10090|c0_g1_i1_coral   1.581919    0.00000  3.4729807
TR10090|c0_g2_i1_coral   1.581919    0.00000  7.4421015
TR10103|c0_g1_i1_coral   1.054613    0.00000  0.9922802
TR10108|c0_g1_i1_coral   2.109226   21.37607 12.8996426
TR10132|c0_g1_i1_coral   1.054613   29.92650  7.9382416
                       RI_W_05_14 RI_W_05_18 RI_W_05_22
TR10083|c0_g2_i1_coral  3.1630643   1.010704   5.526636
TR10090|c0_g1_i1_coral  0.9037326   8.085630   0.000000
TR10090|c0_g2_i1_coral  3.6149306  10.107037   1.381659
TR10103|c0_g1_i1_coral  4.0667969   2.021407   2.763318
TR10108|c0_g1_i1_coral  5.4223959   2.021407   6.908296
TR10132|c0_g1_i1_coral  1.8074653  10.107037  27.633182
                       RI_W_06_14 RI_W_06_18 RI_W_06_22
TR10083|c0_g2_i1_coral   1.044347   1.336966   5.564881
TR10090|c0_g1_i1_coral   0.000000   0.000000   4.451904
TR10090|c0_g2_i1_coral   3.133041   1.336966   7.790833
TR10103|c0_g1_i1_coral  10.443469   1.336966   2.225952
TR10108|c0_g1_i1_coral  11.487816   0.000000   5.564881
TR10132|c0_g1_i1_coral   5.221735   1.336966  42.293092
                       RI_W_07_14 RI_W_07_18 RI_W_07_22
TR10083|c0_g2_i1_coral   4.124636   5.593017   6.900578
TR10090|c0_g1_i1_coral   0.000000   0.000000   0.000000
TR10090|c0_g2_i1_coral   2.062318   1.118603   5.520462
TR10103|c0_g1_i1_coral   5.499514  13.423242   1.380116
TR10108|c0_g1_i1_coral   0.000000   1.118603   5.520462
TR10132|c0_g1_i1_coral   1.374879   2.237207  86.947282
                       VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10083|c0_g2_i1_coral   2.551157   1.731769   2.845097
TR10090|c0_g1_i1_coral   7.653472   9.091789   1.707058
TR10090|c0_g2_i1_coral  10.204629   5.628251   5.690194
TR10103|c0_g1_i1_coral   3.826736   2.597654   2.845097
TR10108|c0_g1_i1_coral   3.826736   9.091789   2.845097
TR10132|c0_g1_i1_coral   1.275579   1.298827   0.000000
                       VA_B_02_14 VA_B_02_18 VA_B_02_22
TR10083|c0_g2_i1_coral          0    14.2516  19.312637
TR10090|c0_g1_i1_coral          0     0.0000   0.000000
TR10090|c0_g2_i1_coral          0     0.0000   0.000000
TR10103|c0_g1_i1_coral          0     0.0000   2.145849
TR10108|c0_g1_i1_coral          0     0.0000   0.000000
TR10132|c0_g1_i1_coral          0     0.0000   8.583394
                       VA_B_03_14 VA_B_03_18 VA_B_03_22
TR10083|c0_g2_i1_coral  16.029307  14.339493   7.833217
TR10090|c0_g1_i1_coral   5.828839   2.150924   7.833217
TR10090|c0_g2_i1_coral  10.929073   4.301848   3.560553
TR10103|c0_g1_i1_coral   2.185815   4.301848   3.560553
TR10108|c0_g1_i1_coral  14.572098  18.641340  21.363318
TR10132|c0_g1_i1_coral   9.471863  30.829909   8.545327
                       VA_B_04_14 VA_B_04_18 VA_B_04_22
TR10083|c0_g2_i1_coral   0.000000  0.9006809   6.731616
TR10090|c0_g1_i1_coral   0.000000  4.5034046   3.624716
TR10090|c0_g2_i1_coral   7.108517  3.1523832   0.000000
TR10103|c0_g1_i1_coral   4.442823  0.0000000   0.000000
TR10108|c0_g1_i1_coral   2.665694  4.9537450   3.624716
TR10132|c0_g1_i1_coral   0.000000  2.7020428   0.000000
                       VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10083|c0_g2_i1_coral   2.098865  3.3173156   1.813677
TR10090|c0_g1_i1_coral   4.897352  1.1057719   7.254707
TR10090|c0_g2_i1_coral   1.399244  0.5528859   5.441030
TR10103|c0_g1_i1_coral   0.000000  3.3173156   0.000000
TR10108|c0_g1_i1_coral   9.095083  5.5288593   0.000000
TR10132|c0_g1_i1_coral   0.000000  4.9759734   5.441030
                       VA_B_06_14 VA_B_06_18 VA_B_06_22
TR10083|c0_g2_i1_coral   1.383946   0.000000  5.9098781
TR10090|c0_g1_i1_coral   8.303676   0.000000  0.4924898
TR10090|c0_g2_i1_coral   0.000000   4.206352  1.4774695
TR10103|c0_g1_i1_coral   5.535784   6.730163  3.9399187
TR10108|c0_g1_i1_coral  11.763540   4.206352  3.4474289
TR10132|c0_g1_i1_coral  18.683270   2.523811 14.2822054
                       VA_B_07_14 VA_B_07_18 VA_B_07_22
TR10083|c0_g2_i1_coral   3.728534  0.7405861   2.098499
TR10090|c0_g1_i1_coral   0.000000  3.7029304   2.098499
TR10090|c0_g2_i1_coral   1.491414 15.5523076   6.295498
TR10103|c0_g1_i1_coral   1.491414  8.1464468   4.196999
TR10108|c0_g1_i1_coral   2.237121  2.2217582   5.246248
TR10132|c0_g1_i1_coral   8.948482 17.0334798   2.098499
                       VA_W_01_14 VA_W_01_18 VA_W_02_14
TR10083|c0_g2_i1_coral   6.699293   7.732656   0.000000
TR10090|c0_g1_i1_coral   3.828167   1.104665   6.978382
TR10090|c0_g2_i1_coral   0.000000   3.866328   4.361489
TR10103|c0_g1_i1_coral   0.000000   3.866328   5.233787
TR10108|c0_g1_i1_coral   1.914084   4.418660   0.000000
TR10132|c0_g1_i1_coral   0.000000  10.494319   2.616893
                       VA_W_02_18 VA_W_03_14 VA_W_03_22
TR10083|c0_g2_i1_coral  4.5789684  3.1096023   6.888125
TR10090|c0_g1_i1_coral  0.6541383  5.4418040   9.392897
TR10090|c0_g2_i1_coral 13.0827667  6.9966052   3.130966
TR10103|c0_g1_i1_coral  0.6541383  6.9966052   2.504773
TR10108|c0_g1_i1_coral  9.1579367  3.1096023   3.757159
TR10132|c0_g1_i1_coral 13.0827667  0.7774006  13.150056
                       VA_W_04_14 VA_W_04_18 VA_W_04_22
TR10083|c0_g2_i1_coral   1.026752   4.334366   6.410680
TR10090|c0_g1_i1_coral   2.566880   5.779155   3.945034
TR10090|c0_g2_i1_coral   2.053504   7.705540  10.848842
TR10103|c0_g1_i1_coral   3.593631   3.852770   2.958775
TR10108|c0_g1_i1_coral   6.673887  20.227043   8.383196
TR10132|c0_g1_i1_coral   2.566880   0.000000   0.000000
                       VA_W_05_14 VA_W_05_18 VA_W_05_22
TR10083|c0_g2_i1_coral  0.0000000   3.848517  12.379823
TR10090|c0_g1_i1_coral  0.0000000   1.099576  12.379823
TR10090|c0_g2_i1_coral  0.0000000   2.748941   7.427894
TR10103|c0_g1_i1_coral  0.8118702   3.298729   2.475965
TR10108|c0_g1_i1_coral  1.6237403   5.497882  38.377452
TR10132|c0_g1_i1_coral  2.4356105   8.796611   2.475965
                       VA_W_06_14 VA_W_06_18 VA_W_06_22
TR10083|c0_g2_i1_coral   5.042735  2.5895188   0.000000
TR10090|c0_g1_i1_coral   0.000000  0.8631729   1.764014
TR10090|c0_g2_i1_coral   3.025641  0.8631729   0.000000
TR10103|c0_g1_i1_coral   8.068376  2.5895188   4.410036
TR10108|c0_g1_i1_coral   4.034188  6.9053834   5.292043
TR10132|c0_g1_i1_coral   9.076923 29.3478796  26.460214
                       VA_W_07_14 VA_W_07_18 VA_W_07_22
TR10083|c0_g2_i1_coral 10.2466895   1.127608  0.9473408
TR10090|c0_g1_i1_coral  1.5764138   1.127608  6.6313859
TR10090|c0_g2_i1_coral  5.5174482   3.382824  2.8420225
TR10103|c0_g1_i1_coral  4.7292413   0.000000  1.8946817
TR10108|c0_g1_i1_coral  3.1528275   1.127608  1.8946817
TR10132|c0_g1_i1_coral  0.7882069   6.765648  8.5260676
&gt; genes4heatmap&lt;-resOrigin[resOrigin$pvalue &lt;0.05 &amp; !is.na(resOrigin$pvalue),]
&gt; names(genes4heatmap)
[1] &quot;baseMean&quot;       &quot;log2FoldChange&quot; &quot;lfcSE&quot;         
[4] &quot;stat&quot;           &quot;pvalue&quot;         &quot;padj&quot;          
&gt; head(genes4heatmap)
log2 fold change (MLE): Origin VA vs RI 
Wald test p-value: Origin VA vs RI 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE
                       &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;
TR10090|c0_g2_i1_coral   3.65756       0.943135  0.474719
TR10487|c0_g1_i1_coral   6.57170       0.727123  0.363907
TR1069|c0_g1_i1_coral    4.14205       4.490291  1.982381
TR10748|c0_g1_i1_coral   2.93642       1.787984  0.559059
TR10773|c0_g1_i1_coral   3.02516       1.758477  0.659498
TR10773|c1_g1_i1_coral   2.32876       1.292223  0.632168
                            stat     pvalue      padj
                       &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;
TR10090|c0_g2_i1_coral   1.98672 0.04695340 0.4180430
TR10487|c0_g1_i1_coral   1.99810 0.04570594 0.4125186
TR1069|c0_g1_i1_coral    2.26510 0.02350656 0.3086533
TR10748|c0_g1_i1_coral   3.19820 0.00138287 0.0720091
TR10773|c0_g1_i1_coral   2.66639 0.00766715 0.1733954
TR10773|c1_g1_i1_coral   2.04411 0.04094219        NA
&gt; dim(genes4heatmap)
[1] 866   6
&gt; data4heatmap&lt;-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]
&gt; dim(data4heatmap)
[1] 866  78
&gt; head(data4heatmap)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22
TR10090|c0_g2_i1_coral    0.00000   3.468731   2.447866
TR10487|c0_g1_i1_coral   10.94714   5.203096   8.567532
TR1069|c0_g1_i1_coral     0.00000   0.000000   0.000000
TR10748|c0_g1_i1_coral    0.00000   0.000000   2.447866
TR10773|c0_g1_i1_coral    0.00000   6.070279   2.447866
TR10773|c1_g1_i1_coral    0.00000   2.601548   4.895733
                       RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10090|c0_g2_i1_coral   3.772786    0.00000   0.000000
TR10487|c0_g1_i1_coral   0.000000    0.00000   6.648979
TR1069|c0_g1_i1_coral    7.545573    3.45517   6.648979
TR10748|c0_g1_i1_coral   0.000000    0.00000   0.000000
TR10773|c0_g1_i1_coral   3.772786    0.00000   0.000000
TR10773|c1_g1_i1_coral   0.000000    0.00000   2.216326
                       RI_B_03_14 RI_B_03_22 RI_B_04_14
TR10090|c0_g2_i1_coral   0.000000   3.818995   4.809473
TR10487|c0_g1_i1_coral   9.010379   5.091993   2.404737
TR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000
TR10748|c0_g1_i1_coral   2.574394   0.000000   4.007894
TR10773|c0_g1_i1_coral   1.287197   0.000000   0.000000
TR10773|c1_g1_i1_coral   5.148788   1.272998   0.000000
                       RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10090|c0_g2_i1_coral  3.6845136          0   0.570426
TR10487|c0_g1_i1_coral  4.5032943          0   9.126817
TR1069|c0_g1_i1_coral   0.0000000          0   0.000000
TR10748|c0_g1_i1_coral  2.8657328          0   1.711278
TR10773|c0_g1_i1_coral  0.4093904          0   0.000000
TR10773|c1_g1_i1_coral  0.0000000          0   0.285213
                       RI_B_05_18 RI_B_05_22 RI_B_06_14
TR10090|c0_g2_i1_coral   0.000000  2.4538305          0
TR10487|c0_g1_i1_coral   5.033716  0.8179435          0
TR1069|c0_g1_i1_coral    0.000000  0.0000000          0
TR10748|c0_g1_i1_coral   0.000000  0.0000000          0
TR10773|c0_g1_i1_coral   0.000000  0.0000000          0
TR10773|c1_g1_i1_coral   0.000000  0.8179435          0
                       RI_B_06_18 RI_B_06_22 RI_B_07_14
TR10090|c0_g2_i1_coral   4.827992          0   3.734168
TR10487|c0_g1_i1_coral   2.758853          0   5.227835
TR1069|c0_g1_i1_coral    0.000000          0   0.000000
TR10748|c0_g1_i1_coral   0.000000          0   2.987334
TR10773|c0_g1_i1_coral   0.000000          0   0.000000
TR10773|c1_g1_i1_coral   0.000000          0   0.000000
                       RI_B_07_18 RI_B_07_22 RI_W_01_18
TR10090|c0_g2_i1_coral   1.008773          0  12.257216
TR10487|c0_g1_i1_coral   5.043864          0   8.652152
TR1069|c0_g1_i1_coral    0.000000          0  10.094178
TR10748|c0_g1_i1_coral   0.000000          0   0.000000
TR10773|c0_g1_i1_coral   1.008773          0   7.931140
TR10773|c1_g1_i1_coral   0.000000          0   5.047089
                       RI_W_01_22 RI_W_02_14 RI_W_02_18
TR10090|c0_g2_i1_coral   7.743994   4.651361  4.9947759
TR10487|c0_g1_i1_coral   6.195195  16.279762  2.9968656
TR1069|c0_g1_i1_coral    9.292792   0.000000  0.0000000
TR10748|c0_g1_i1_coral   1.548799   4.651361  0.9989552
TR10773|c0_g1_i1_coral  17.036786   0.000000 14.9843278
TR10773|c1_g1_i1_coral  18.585585   0.000000 10.9885071
                       RI_W_02_22 RI_W_03_14 RI_W_03_18
TR10090|c0_g2_i1_coral   4.820288   3.044865   3.244480
TR10487|c0_g1_i1_coral   1.205072   7.307675   7.209956
TR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000
TR10748|c0_g1_i1_coral   2.410144   2.435892   6.128462
TR10773|c0_g1_i1_coral   6.025360  23.140971  12.617422
TR10773|c1_g1_i1_coral   1.205072  10.352540   7.930951
                       RI_W_03_22 RI_W_04_14 RI_W_04_18
TR10090|c0_g2_i1_coral   1.581919    0.00000  7.4421015
TR10487|c0_g1_i1_coral  14.237275   17.10086  5.4575411
TR1069|c0_g1_i1_coral    0.000000    0.00000  0.0000000
TR10748|c0_g1_i1_coral   4.218452    0.00000  0.4961401
TR10773|c0_g1_i1_coral  29.001856    0.00000  1.9845604
TR10773|c1_g1_i1_coral   7.382291    0.00000  1.4884203
                       RI_W_05_14 RI_W_05_18 RI_W_05_22
TR10090|c0_g2_i1_coral   3.614931  10.107037   1.381659
TR10487|c0_g1_i1_coral   4.066797   4.042815   0.000000
TR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000
TR10748|c0_g1_i1_coral   2.711198   1.010704   0.000000
TR10773|c0_g1_i1_coral   5.422396   2.021407   0.000000
TR10773|c1_g1_i1_coral   3.614931   3.032111   0.000000
                       RI_W_06_14 RI_W_06_18 RI_W_06_22
TR10090|c0_g2_i1_coral   3.133041   1.336966   7.790833
TR10487|c0_g1_i1_coral   4.177388  13.369662  10.016785
TR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000
TR10748|c0_g1_i1_coral   3.133041   2.673932   2.225952
TR10773|c0_g1_i1_coral   5.221735   2.673932  10.016785
TR10773|c1_g1_i1_coral   1.044347   2.673932   3.338928
                       RI_W_07_14 RI_W_07_18 RI_W_07_22
TR10090|c0_g2_i1_coral   2.062318   1.118603   5.520462
TR10487|c0_g1_i1_coral   3.437196  13.423242   8.280694
TR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000
TR10748|c0_g1_i1_coral   2.749757   2.237207   0.000000
TR10773|c0_g1_i1_coral   1.374879   2.237207   0.000000
TR10773|c1_g1_i1_coral   2.062318   3.355810   2.760231
                       VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10090|c0_g2_i1_coral  10.204629   5.628251   5.690194
TR10487|c0_g1_i1_coral  10.204629   6.494135  10.242350
TR1069|c0_g1_i1_coral   33.165045   9.524732  17.639603
TR10748|c0_g1_i1_coral   1.275579   1.298827   5.121175
TR10773|c0_g1_i1_coral   6.377893   4.762366   2.845097
TR10773|c1_g1_i1_coral   8.929051   6.061193   6.259214
                       VA_B_02_14 VA_B_02_18 VA_B_02_22
TR10090|c0_g2_i1_coral   0.000000     0.0000   0.000000
TR10487|c0_g1_i1_coral   0.000000     0.0000  17.166789
TR1069|c0_g1_i1_coral   86.631891    44.0504  40.771123
TR10748|c0_g1_i1_coral  30.575962     0.0000   2.145849
TR10773|c0_g1_i1_coral   1.698665     0.0000   6.437546
TR10773|c1_g1_i1_coral   0.000000     2.5912   0.000000
                       VA_B_03_14 VA_B_03_18 VA_B_03_22
TR10090|c0_g2_i1_coral  10.929073   4.301848  3.5605530
TR10487|c0_g1_i1_coral  10.929073   7.886721  7.8332166
TR1069|c0_g1_i1_coral    0.000000   0.000000  0.0000000
TR10748|c0_g1_i1_coral   3.643024   7.886721  3.5605530
TR10773|c0_g1_i1_coral   0.000000   1.433949  0.7121106
TR10773|c1_g1_i1_coral   0.000000   3.584873  0.7121106
                       VA_B_04_14 VA_B_04_18 VA_B_04_22
TR10090|c0_g2_i1_coral  7.1085175   3.152383   0.000000
TR10487|c0_g1_i1_coral 15.1055996   4.053064   8.285065
TR1069|c0_g1_i1_coral   0.0000000   0.000000   0.000000
TR10748|c0_g1_i1_coral  2.6656940   0.000000   8.802882
TR10773|c0_g1_i1_coral  0.0000000   7.205447   8.285065
TR10773|c1_g1_i1_coral  0.8885647   8.556469   2.589083
                       VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10090|c0_g2_i1_coral   1.399244  0.5528859   5.441030
TR10487|c0_g1_i1_coral   4.547541 17.1394640  21.764122
TR1069|c0_g1_i1_coral    0.000000  0.0000000   0.000000
TR10748|c0_g1_i1_coral   1.049433  1.6586578   3.627354
TR10773|c0_g1_i1_coral   1.399244  1.1057719   0.000000
TR10773|c1_g1_i1_coral   0.000000  4.4230875   0.000000
                       VA_B_06_14 VA_B_06_18 VA_B_06_22
TR10090|c0_g2_i1_coral   0.000000   4.206352   1.477470
TR10487|c0_g1_i1_coral   0.000000   8.412704   2.462449
TR1069|c0_g1_i1_coral   29.062864  16.825408   8.372327
TR10748|c0_g1_i1_coral   6.919730   0.000000   2.954939
TR10773|c0_g1_i1_coral   2.767892   0.000000   5.417388
TR10773|c1_g1_i1_coral   0.000000   2.523811   2.462449
                       VA_B_07_14 VA_B_07_18 VA_B_07_22
TR10090|c0_g2_i1_coral  1.4914137  15.552308   6.295498
TR10487|c0_g1_i1_coral  3.7285342   5.184103   8.393997
TR1069|c0_g1_i1_coral   0.0000000   0.000000   0.000000
TR10748|c0_g1_i1_coral  6.7113616   0.000000   4.196999
TR10773|c0_g1_i1_coral  1.4914137   2.962344   6.295498
TR10773|c1_g1_i1_coral  0.7457068   0.000000   4.196999
                       VA_W_01_14 VA_W_01_18 VA_W_02_14
TR10090|c0_g2_i1_coral   0.000000   3.866328  4.3614889
TR10487|c0_g1_i1_coral   6.699293   4.970993  1.7445956
TR1069|c0_g1_i1_coral    0.000000   0.000000  0.0000000
TR10748|c0_g1_i1_coral   2.871126   2.209330  0.8722978
TR10773|c0_g1_i1_coral   5.742251   1.104665  0.0000000
TR10773|c1_g1_i1_coral   0.000000   3.866328  0.0000000
                       VA_W_02_18 VA_W_03_14 VA_W_03_22
TR10090|c0_g2_i1_coral  13.082767  6.9966052   3.130966
TR10487|c0_g1_i1_coral  13.082767  3.1096023   9.392897
TR1069|c0_g1_i1_coral    0.000000  0.0000000   0.000000
TR10748|c0_g1_i1_coral   0.000000  0.7774006   3.757159
TR10773|c0_g1_i1_coral   3.924830  1.5548011   2.504773
TR10773|c1_g1_i1_coral   5.233107  0.7774006   2.504773
                       VA_W_04_14 VA_W_04_18 VA_W_04_22
TR10090|c0_g2_i1_coral   2.053504   7.705540 10.8488424
TR10487|c0_g1_i1_coral   3.593631  13.003099  1.4793876
TR1069|c0_g1_i1_coral    0.000000   0.000000  0.0000000
TR10748|c0_g1_i1_coral   6.673887   9.150329  7.3969380
TR10773|c0_g1_i1_coral   1.026752   0.000000  0.4931292
TR10773|c1_g1_i1_coral   1.026752   0.000000  0.0000000
                       VA_W_05_14 VA_W_05_18 VA_W_05_22
TR10090|c0_g2_i1_coral    0.00000   2.748941   7.427894
TR10487|c0_g1_i1_coral    0.00000   8.246822   8.665876
TR1069|c0_g1_i1_coral     0.00000   0.000000   0.000000
TR10748|c0_g1_i1_coral   14.61366   2.748941   9.903859
TR10773|c0_g1_i1_coral    0.00000   0.000000   0.000000
TR10773|c1_g1_i1_coral    0.00000   2.199153   1.237982
                       VA_W_06_14 VA_W_06_18 VA_W_06_22
TR10090|c0_g2_i1_coral   3.025641  0.8631729   0.000000
TR10487|c0_g1_i1_coral  10.085470  1.7263459  11.466093
TR1069|c0_g1_i1_coral    0.000000  0.0000000   0.000000
TR10748|c0_g1_i1_coral   2.017094  6.0422105   4.410036
TR10773|c0_g1_i1_coral   0.000000  1.7263459   0.000000
TR10773|c1_g1_i1_coral   2.017094  0.8631729   5.292043
                       VA_W_07_14 VA_W_07_18 VA_W_07_22
TR10090|c0_g2_i1_coral   5.517448   3.382824   2.842023
TR10487|c0_g1_i1_coral   7.093862   6.765648   3.789363
TR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000
TR10748|c0_g1_i1_coral   0.000000   3.382824   1.894682
TR10773|c0_g1_i1_coral   0.000000   0.000000   0.000000
TR10773|c1_g1_i1_coral   0.000000   0.000000   0.000000
&gt; 
&gt; temp = as.matrix(rowMeans(data4heatmap))
&gt; head(temp)
                           [,1]
TR10090|c0_g2_i1_coral 3.657565
TR10487|c0_g1_i1_coral 6.571698
TR1069|c0_g1_i1_coral  4.142052
TR10748|c0_g1_i1_coral 2.936423
TR10773|c0_g1_i1_coral 3.025158
TR10773|c1_g1_i1_coral 2.328759
&gt; scaledmatrix&lt;-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)
&gt; data4heatmapscaled = data4heatmap/scaledmatrix
&gt; head(data4heatmapscaled)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22
TR10090|c0_g2_i1_coral   0.000000  0.9483717  0.6692613
TR10487|c0_g1_i1_coral   1.665801  0.7917431  1.3037015
TR1069|c0_g1_i1_coral    0.000000  0.0000000  0.0000000
TR10748|c0_g1_i1_coral   0.000000  0.0000000  0.8336218
TR10773|c0_g1_i1_coral   0.000000  2.0065992  0.8091699
TR10773|c1_g1_i1_coral   0.000000  1.1171391  2.1022922
                       RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10090|c0_g2_i1_coral   1.031502  0.0000000  0.0000000
TR10487|c0_g1_i1_coral   0.000000  0.0000000  1.0117598
TR1069|c0_g1_i1_coral    1.821699  0.8341687  1.6052378
TR10748|c0_g1_i1_coral   0.000000  0.0000000  0.0000000
TR10773|c0_g1_i1_coral   1.247137  0.0000000  0.0000000
TR10773|c1_g1_i1_coral   0.000000  0.0000000  0.9517198
                       RI_B_03_14 RI_B_03_22 RI_B_04_14
TR10090|c0_g2_i1_coral  0.0000000  1.0441360  1.3149387
TR10487|c0_g1_i1_coral  1.3710885  0.7748368  0.3659232
TR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000
TR10748|c0_g1_i1_coral  0.8767109  0.0000000  1.3648899
TR10773|c0_g1_i1_coral  0.4254975  0.0000000  0.0000000
TR10773|c1_g1_i1_coral  2.2109576  0.5466423  0.0000000
                       RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10090|c0_g2_i1_coral  1.0073680          0  0.1559579
TR10487|c0_g1_i1_coral  0.6852559          0  1.3888066
TR1069|c0_g1_i1_coral   0.0000000          0  0.0000000
TR10748|c0_g1_i1_coral  0.9759263          0  0.5827764
TR10773|c0_g1_i1_coral  0.1353286          0  0.0000000
TR10773|c1_g1_i1_coral  0.0000000          0  0.1224742
                       RI_B_05_18 RI_B_05_22 RI_B_06_14
TR10090|c0_g2_i1_coral  0.0000000  0.6708919          0
TR10487|c0_g1_i1_coral  0.7659688  0.1244646          0
TR1069|c0_g1_i1_coral   0.0000000  0.0000000          0
TR10748|c0_g1_i1_coral  0.0000000  0.0000000          0
TR10773|c0_g1_i1_coral  0.0000000  0.0000000          0
TR10773|c1_g1_i1_coral  0.0000000  0.3512357          0
                       RI_B_06_18 RI_B_06_22 RI_B_07_14
TR10090|c0_g2_i1_coral  1.3200019          0  1.0209437
TR10487|c0_g1_i1_coral  0.4198082          0  0.7955075
TR1069|c0_g1_i1_coral   0.0000000          0  0.0000000
TR10748|c0_g1_i1_coral  0.0000000          0  1.0173377
TR10773|c0_g1_i1_coral  0.0000000          0  0.0000000
TR10773|c1_g1_i1_coral  0.0000000          0  0.0000000
                       RI_B_07_18 RI_B_07_22 RI_W_01_18
TR10090|c0_g2_i1_coral  0.2758045          0   3.351196
TR10487|c0_g1_i1_coral  0.7675130          0   1.316578
TR1069|c0_g1_i1_coral   0.0000000          0   2.436999
TR10748|c0_g1_i1_coral  0.0000000          0   0.000000
TR10773|c0_g1_i1_coral  0.3334612          0   2.621728
TR10773|c1_g1_i1_coral  0.0000000          0   2.167287
                       RI_W_01_22 RI_W_02_14 RI_W_02_18
TR10090|c0_g2_i1_coral  2.1172541   1.271710  1.3656016
TR10487|c0_g1_i1_coral  0.9427084   2.477254  0.4560261
TR1069|c0_g1_i1_coral   2.2435236   0.000000  0.0000000
TR10748|c0_g1_i1_coral  0.5274440   1.584023  0.3401946
TR10773|c0_g1_i1_coral  5.6317019   0.000000  4.9532387
TR10773|c1_g1_i1_coral  7.9808953   0.000000  4.7186100
                       RI_W_02_22 RI_W_03_14 RI_W_03_18
TR10090|c0_g2_i1_coral  1.3178956  0.8324842  0.8870602
TR10487|c0_g1_i1_coral  0.1833730  1.1119920  1.0971223
TR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000
TR10748|c0_g1_i1_coral  0.8207755  0.8295438  2.0870501
TR10773|c0_g1_i1_coral  1.9917509  7.6495091  4.1708314
TR10773|c1_g1_i1_coral  0.5174739  4.4455171  3.4056552
                       RI_W_03_22 RI_W_04_14 RI_W_04_18
TR10090|c0_g2_i1_coral  0.4325062   0.000000  2.0347150
TR10487|c0_g1_i1_coral  2.1664531   2.602198  0.8304614
TR1069|c0_g1_i1_coral   0.0000000   0.000000  0.0000000
TR10748|c0_g1_i1_coral  1.4365953   0.000000  0.1689607
TR10773|c0_g1_i1_coral  9.5868908   0.000000  0.6560188
TR10773|c1_g1_i1_coral  3.1700530   0.000000  0.6391473
                       RI_W_05_14 RI_W_05_18 RI_W_05_22
TR10090|c0_g2_i1_coral  0.9883436  2.7633244  0.3777539
TR10487|c0_g1_i1_coral  0.6188351  0.6151858  0.0000000
TR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000
TR10748|c0_g1_i1_coral  0.9232995  0.3441955  0.0000000
TR10773|c0_g1_i1_coral  1.7924342  0.6681991  0.0000000
TR10773|c1_g1_i1_coral  1.5522989  1.3020286  0.0000000
                       RI_W_06_14 RI_W_06_18 RI_W_06_22
TR10090|c0_g2_i1_coral  0.8565921  0.3655346  2.1300602
TR10487|c0_g1_i1_coral  0.6356634  2.0344306  1.5242310
TR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000
TR10748|c0_g1_i1_coral  1.0669582  0.9106087  0.7580488
TR10773|c0_g1_i1_coral  1.7261033  0.8838986  3.3111614
TR10773|c1_g1_i1_coral  0.4484564  1.1482219  1.4337799
                       RI_W_07_14 RI_W_07_18 RI_W_07_22
TR10090|c0_g2_i1_coral  0.5638500  0.3058329   1.509327
TR10487|c0_g1_i1_coral  0.5230302  2.0425836   1.260054
TR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000
TR10748|c0_g1_i1_coral  0.9364308  0.7618816   0.000000
TR10773|c0_g1_i1_coral  0.4544816  0.7395340   0.000000
TR10773|c1_g1_i1_coral  0.8855865  1.4410293   1.185280
                       VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10090|c0_g2_i1_coral  2.7900066  1.5387973   1.555733
TR10487|c0_g1_i1_coral  1.5528148  0.9881975   1.558555
TR1069|c0_g1_i1_coral   8.0069110  2.2995199   4.258662
TR10748|c0_g1_i1_coral  0.4343988  0.4423160   1.744018
TR10773|c0_g1_i1_coral  2.1082846  1.5742538   0.940479
TR10773|c1_g1_i1_coral  3.8342522  2.6027562   2.687789
                       VA_B_02_14 VA_B_02_18 VA_B_02_22
TR10090|c0_g2_i1_coral  0.0000000   0.000000  0.0000000
TR10487|c0_g1_i1_coral  0.0000000   0.000000  2.6122305
TR1069|c0_g1_i1_coral  20.9152088  10.634922  9.8432176
TR10748|c0_g1_i1_coral 10.4126549   0.000000  0.7307695
TR10773|c0_g1_i1_coral  0.5615127   0.000000  2.1280034
TR10773|c1_g1_i1_coral  0.0000000   1.112696  0.0000000
                       VA_B_03_14 VA_B_03_18 VA_B_03_22
TR10090|c0_g2_i1_coral   2.988074  1.1761509  0.9734764
TR10487|c0_g1_i1_coral   1.663052  1.2001041  1.1919624
TR1069|c0_g1_i1_coral    0.000000  0.0000000  0.0000000
TR10748|c0_g1_i1_coral   1.240633  2.6858257  1.2125476
TR10773|c0_g1_i1_coral   0.000000  0.4740081  0.2353962
TR10773|c1_g1_i1_coral   0.000000  1.5393918  0.3057897
                       VA_B_04_14 VA_B_04_18 VA_B_04_22
TR10090|c0_g2_i1_coral  1.9435111  0.8618804   0.000000
TR10487|c0_g1_i1_coral  2.2985841  0.6167454   1.260719
TR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000
TR10748|c0_g1_i1_coral  0.9078031  0.0000000   2.997825
TR10773|c0_g1_i1_coral  0.0000000  2.3818420   2.738722
TR10773|c1_g1_i1_coral  0.3815614  3.6742606   1.111786
                       VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10090|c0_g2_i1_coral  0.3825615  0.1511623   1.487610
TR10487|c0_g1_i1_coral  0.6919888  2.6080725   3.311796
TR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000
TR10748|c0_g1_i1_coral  0.3573847  0.5648565   1.235297
TR10773|c0_g1_i1_coral  0.4625357  0.3655254   0.000000
TR10773|c1_g1_i1_coral  0.0000000  1.8993321   0.000000
                       VA_B_06_14 VA_B_06_18 VA_B_06_22
TR10090|c0_g2_i1_coral  0.0000000   1.150042  0.4039490
TR10487|c0_g1_i1_coral  0.0000000   1.280142  0.3747052
TR1069|c0_g1_i1_coral   7.0165371   4.062094  2.0212992
TR10748|c0_g1_i1_coral  2.3565165   0.000000  1.0063056
TR10773|c0_g1_i1_coral  0.9149579   0.000000  1.7907788
TR10773|c1_g1_i1_coral  0.0000000   1.083758  1.0574082
                       VA_B_07_14 VA_B_07_18 VA_B_07_22
TR10090|c0_g2_i1_coral  0.4077614  4.2520938   1.721227
TR10487|c0_g1_i1_coral  0.5673624  0.7888529   1.277295
TR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000
TR10748|c0_g1_i1_coral  2.2855567  0.0000000   1.429289
TR10773|c0_g1_i1_coral  0.4930036  0.9792364   2.081048
TR10773|c1_g1_i1_coral  0.3202164  0.0000000   1.802247
                       VA_W_01_14 VA_W_01_18 VA_W_02_14
TR10090|c0_g2_i1_coral  0.0000000  1.0570771  1.1924571
TR10487|c0_g1_i1_coral  1.0194159  0.7564245  0.2654711
TR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000
TR10748|c0_g1_i1_coral  0.9777629  0.7523882  0.2970613
TR10773|c0_g1_i1_coral  1.8981660  0.3651595  0.0000000
TR10773|c1_g1_i1_coral  0.0000000  1.6602522  0.0000000
                       VA_W_02_18 VA_W_03_14 VA_W_03_22
TR10090|c0_g2_i1_coral   3.576907  1.9129136  0.8560247
TR10487|c0_g1_i1_coral   1.990774  0.4731810  1.4292954
TR1069|c0_g1_i1_coral    0.000000  0.0000000  0.0000000
TR10748|c0_g1_i1_coral   0.000000  0.2647441  1.2795019
TR10773|c0_g1_i1_coral   1.297397  0.5139571  0.8279809
TR10773|c1_g1_i1_coral   2.247165  0.3338261  1.0755824
                       VA_W_04_14 VA_W_04_18 VA_W_04_22
TR10090|c0_g2_i1_coral  0.5614402   2.106741  2.9661383
TR10487|c0_g1_i1_coral  0.5468346   1.978651  0.2251150
TR1069|c0_g1_i1_coral   0.0000000   0.000000  0.0000000
TR10748|c0_g1_i1_coral  2.2727947   3.116148  2.5190299
TR10773|c0_g1_i1_coral  0.3394044   0.000000  0.1630094
TR10773|c1_g1_i1_coral  0.4409008   0.000000  0.0000000
                       VA_W_05_14 VA_W_05_18 VA_W_05_22
TR10090|c0_g2_i1_coral   0.000000  0.7515768   2.030831
TR10487|c0_g1_i1_coral   0.000000  1.2548999   1.318666
TR1069|c0_g1_i1_coral    0.000000  0.0000000   0.000000
TR10748|c0_g1_i1_coral   4.976688  0.9361528   3.372763
TR10773|c0_g1_i1_coral   0.000000  0.0000000   0.000000
TR10773|c1_g1_i1_coral   0.000000  0.9443452   0.531606
                       VA_W_06_14 VA_W_06_18 VA_W_06_22
TR10090|c0_g2_i1_coral  0.8272283  0.2359966   0.000000
TR10487|c0_g1_i1_coral  1.5346826  0.2626940   1.744769
TR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000
TR10748|c0_g1_i1_coral  0.6869221  2.0576770   1.501839
TR10773|c0_g1_i1_coral  0.0000000  0.5706631   0.000000
TR10773|c1_g1_i1_coral  0.8661668  0.3706578   2.272473
                       VA_W_07_14 VA_W_07_18 VA_W_07_22
TR10090|c0_g2_i1_coral   1.508503  0.9248843  0.7770259
TR10487|c0_g1_i1_coral   1.079457  1.0295131  0.5766186
TR1069|c0_g1_i1_coral    0.000000  0.0000000  0.0000000
TR10748|c0_g1_i1_coral   0.000000  1.1520220  0.6452345
TR10773|c0_g1_i1_coral   0.000000  0.0000000  0.0000000
TR10773|c1_g1_i1_coral   0.000000  0.0000000  0.0000000
&gt; 
&gt; dim(data4heatmapscaled)
[1] 866  78
&gt; 
&gt; pairs.breaks &lt;- seq(0, 3.0, by=0.1)
&gt; length(pairs.breaks)
[1] 31
&gt; mycol &lt;- colorpanel(n=30, low=&quot;black&quot;, high=&quot;yellow&quot;) 
&gt; 
&gt; pdf(file=&quot;XXXXXX_byrow.pdf&quot;,7,7)
&gt; heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(&quot;row&quot;), scale=&quot;none&quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = &quot;none&quot;, symkey = F, density.info = &quot;density&quot;, colsep=c(24), sepcolor=c(&quot;white&quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
&gt; dev.off()
RStudioGD 
        2 
&gt; 
&gt; pdf(file=&quot;XXXXXX_bycolumn.pdf&quot;,7,7)
&gt; heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(&quot;col&quot;), scale=&quot;none&quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = &quot;none&quot;, symkey = F, density.info = &quot;density&quot;, colsep=c(24), sepcolor=c(&quot;white&quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
&gt; dev.off()
RStudioGD 
        2 
&gt; heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(&quot;row&quot;), scale=&quot;none&quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = &quot;none&quot;, symkey = F, density.info = &quot;density&quot;, colsep=c(24), sepcolor=c(&quot;white&quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
Error in plot.new() : figure margins too large
&gt; heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(&quot;col&quot;), scale=&quot;none&quot;, keysize=1, breaks=pairs.breaks, col=mycol, trace = &quot;none&quot;, symkey = F, density.info = &quot;density&quot;, colsep=c(24), sepcolor=c(&quot;white&quot;), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
Error in .External.graphics(C_layout, num.rows, num.cols, mat, as.integer(num.figures),  : 
  invalid graphics state
&gt; plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=&quot;Origin&quot;)
&gt; plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=&quot;Origin&quot;)
&gt; plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=&quot;Origin&quot;)
&gt; plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=&quot;Origin_SymState&quot;)
&gt; plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=&quot;Origin_SymState_Temp&quot;)
&gt; pdf(file=&quot;866_RIvsVA_PCAv2.pdf&quot;,14, 14)
&gt; plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=&quot;Origin&quot;)
&gt; dev.off()
RStudioGD 
        2 
&gt; prop.nullv3 &lt;- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100*mean(x==0))
&gt; barplot(prop.nullv3, main=&quot;Percentage of null counts per sample&quot;, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
&gt; pdf(file=&quot;ResNullCountsv2.pdf&quot;,14, 14)
&gt; barplot(prop.nullv3, main=&quot;Percentage of null counts per sample&quot;, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
&gt; dev.off()
RStudioGD 
        2 
&gt; barplot(prop.nullv3, main=&quot;Percentage of null counts per sample&quot;, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
&gt; distsv2&lt;-dist(t(assay(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),])))
&gt; plot(hclust(distsv2))
&gt; heatmap.2(as.matrix(distsv2), key=F, trace=&quot;none&quot;,
+           col=colorpanel(100, &quot;black&quot;, &quot;white&quot;),
+           margin=c(10, 10))
&gt; #### GOSeq analysis ####
&gt; res&lt;-resOrigin
&gt; Annotations&lt;-read.delim(&quot;Assembly_GoodCoralSymbiont_suffixed_totalannotated.txt&quot;)
&gt; head(res)
log2 fold change (MLE): Origin VA vs RI 
Wald test p-value: Origin VA vs RI 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange
                       &lt;numeric&gt;      &lt;numeric&gt;
TR10083|c0_g2_i1_coral   5.36648     -0.2712427
TR10090|c0_g1_i1_coral   2.70436      1.1125117
TR10090|c0_g2_i1_coral   3.65756      0.9431346
TR10103|c0_g1_i1_coral   2.91279      0.0386643
TR10108|c0_g1_i1_coral   6.31144     -0.0179010
TR10132|c0_g1_i1_coral   9.26737     -0.2649903
                           lfcSE       stat    pvalue
                       &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;
TR10083|c0_g2_i1_coral  0.446225 -0.6078616 0.5432793
TR10090|c0_g1_i1_coral  0.589306  1.8878328 0.0590484
TR10090|c0_g2_i1_coral  0.474719  1.9867203 0.0469534
TR10103|c0_g1_i1_coral  0.456720  0.0846565 0.9325345
TR10108|c0_g1_i1_coral  0.461958 -0.0387502 0.9690895
TR10132|c0_g1_i1_coral  0.677608 -0.3910675 0.6957473
                            padj
                       &lt;numeric&gt;
TR10083|c0_g2_i1_coral  0.865661
TR10090|c0_g1_i1_coral        NA
TR10090|c0_g2_i1_coral  0.418043
TR10103|c0_g1_i1_coral  0.984590
TR10108|c0_g1_i1_coral  0.992150
TR10132|c0_g1_i1_coral  0.921517
&gt; sum(res$log2FoldChange!=0)
[1] 7659
&gt; 
&gt; genes&lt;-as.integer(p.adjust(res$pvalue[res$log2FoldChange!=0 &amp; !is.na(res$pvalue)], method=&quot;BH&quot;)&lt;.05)
&gt; names(genes)&lt;-row.names(res[res$log2FoldChange!=0 &amp; !is.na(res$pvalue),])
&gt; table(genes)
genes
   0    1 
7533  116 
&gt; 
&gt; genes&lt;-as.integer(res$pvalue[res$log2FoldChange!=0 &amp; !is.na(res$pvalue)]&lt;.05)
&gt; names(genes)&lt;-row.names(res[res$log2FoldChange!=0 &amp; !is.na(res$pvalue),])
&gt; table(genes)
genes
   0    1 
6783  866 
&gt; 
&gt; goterms&lt;-strsplit(as.character(Annotations$GO), split=&quot; // &quot;)
&gt; names(goterms)&lt;-Annotations$ContigName
&gt; 
&gt; pwf&lt;-nullp(genes, bias.data=Annotations[Annotations$ContigName%in%names(genes),&quot;ContigLength&quot;])
Warning message:
In pcls(G) : initial point very close to some inequality constraints
&gt; GOGOGO&lt;-goseq(pwf,gene2cat=goterms)
Using manually entered categories.
Calculating the p-values...
'select()' returned 1:1 mapping between keys
and columns
&gt; GOGOGO$padj&lt;-p.adjust(GOGOGO$over_represented_pvalue, method=&quot;fdr&quot;)
&gt; 
&gt; sum(GOGOGO$padj&lt;0.2, na.rm=T)
[1] 3
&gt; sum(GOGOGO$padj&lt;0.1, na.rm=T)
[1] 0
&gt; sum(GOGOGO$padj&lt;0.05, na.rm=T)
[1] 0
&gt; GOGOGO[GOGOGO$padj&lt;0.05,]
[1] category                 over_represented_pvalue 
[3] under_represented_pvalue numDEInCat              
[5] numInCat                 term                    
[7] ontology                 padj                    
&lt;0 rows&gt; (or 0-length row.names)
&gt; 
&gt; GOTERM[[&quot;SOMEGOTERM&quot;]]
NULL
&gt; 
&gt; EnGOS&lt;-GOGOGO$category[p.adjust(GOGOGO$over_represented_pvalue, method=&quot;BH&quot;)&lt;.05]
&gt; 
&gt; for(go in EnGOS){
+   print(GOTERM[[go]])
+   cat(&quot;--------------------------------------\n&quot;)}
&gt; names(Annotations)
 [1] &quot;ContigName&quot;             
 [2] &quot;ContigLength&quot;           
 [3] &quot;topnrMatch&quot;             
 [4] &quot;topnrEvalue&quot;            
 [5] &quot;topnrAlignmentLength&quot;   
 [6] &quot;topnrPercMatch&quot;         
 [7] &quot;nobadwordnrMatch&quot;       
 [8] &quot;nobadwordnrEvalue&quot;      
 [9] &quot;nobadwordnrAlignLength&quot; 
[10] &quot;nobadwordnrPercentMatch&quot;
[11] &quot;Uniprotmatch&quot;           
[12] &quot;X._identity&quot;            
[13] &quot;evalue&quot;                 
[14] &quot;ID&quot;                     
[15] &quot;Description&quot;            
[16] &quot;KEGG&quot;                   
[17] &quot;KEGGKO&quot;                 
[18] &quot;PFAM&quot;                   
[19] &quot;GO&quot;                     
[20] &quot;GOCC&quot;                   
[21] &quot;GOBP&quot;                   
[22] &quot;GOMF&quot;                   
[23] &quot;Keywords&quot;               
&gt; head(genes4heatmap)
log2 fold change (MLE): Origin VA vs RI 
Wald test p-value: Origin VA vs RI 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange
                       &lt;numeric&gt;      &lt;numeric&gt;
TR10090|c0_g2_i1_coral   3.65756       0.943135
TR10487|c0_g1_i1_coral   6.57170       0.727123
TR1069|c0_g1_i1_coral    4.14205       4.490291
TR10748|c0_g1_i1_coral   2.93642       1.787984
TR10773|c0_g1_i1_coral   3.02516       1.758477
TR10773|c1_g1_i1_coral   2.32876       1.292223
                           lfcSE      stat     pvalue
                       &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;
TR10090|c0_g2_i1_coral  0.474719   1.98672 0.04695340
TR10487|c0_g1_i1_coral  0.363907   1.99810 0.04570594
TR1069|c0_g1_i1_coral   1.982381   2.26510 0.02350656
TR10748|c0_g1_i1_coral  0.559059   3.19820 0.00138287
TR10773|c0_g1_i1_coral  0.659498   2.66639 0.00766715
TR10773|c1_g1_i1_coral  0.632168   2.04411 0.04094219
                            padj
                       &lt;numeric&gt;
TR10090|c0_g2_i1_coral 0.4180430
TR10487|c0_g1_i1_coral 0.4125186
TR1069|c0_g1_i1_coral  0.3086533
TR10748|c0_g1_i1_coral 0.0720091
TR10773|c0_g1_i1_coral 0.1733954
TR10773|c1_g1_i1_coral        NA
&gt; dim(genes4heatmap)
[1] 866   6
&gt; 
&gt; ResultsOutput&lt;-cbind(&quot;ContigName&quot;=row.names(genes4heatmap),genes4heatmap, Annotations[Annotations$ContigName%in%row.names(genes4heatmap),])
&gt; dim(ResultsOutput)
[1] 866  30
&gt; head(ResultsOutput)
DataFrame with 6 rows and 30 columns
              ContigName  baseMean log2FoldChange
             &lt;character&gt; &lt;numeric&gt;      &lt;numeric&gt;
1 TR10090|c0_g2_i1_coral   3.65756       0.943135
2 TR10487|c0_g1_i1_coral   6.57170       0.727123
3  TR1069|c0_g1_i1_coral   4.14205       4.490291
4 TR10748|c0_g1_i1_coral   2.93642       1.787984
5 TR10773|c0_g1_i1_coral   3.02516       1.758477
6 TR10773|c1_g1_i1_coral   2.32876       1.292223
      lfcSE      stat     pvalue      padj
  &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;
1  0.474719   1.98672 0.04695340 0.4180430
2  0.363907   1.99810 0.04570594 0.4125186
3  1.982381   2.26510 0.02350656 0.3086533
4  0.559059   3.19820 0.00138287 0.0720091
5  0.659498   2.66639 0.00766715 0.1733954
6  0.632168   2.04411 0.04094219        NA
              ContigName ContigLength
             &lt;character&gt;    &lt;integer&gt;
1 TR10090|c0_g2_i1_coral          772
2 TR10487|c0_g1_i1_coral         1007
3  TR1069|c0_g1_i1_coral          815
4 TR10748|c0_g1_i1_coral          516
5 TR10773|c0_g1_i1_coral          580
6 TR10773|c1_g1_i1_coral          586
              topnrMatch  topnrEvalue
             &lt;character&gt;  &lt;character&gt;
1 gnl|BL_ORD_ID|738431..  7.27758e-78
2 gnl|BL_ORD_ID|110398..  3.65321e-84
3 gnl|BL_ORD_ID|110400.. 9.23041e-133
4 gnl|BL_ORD_ID|110376..  2.29747e-78
5 gnl|BL_ORD_ID|110383..  2.14754e-64
6 gnl|BL_ORD_ID|110383..  1.33696e-65
  topnrAlignmentLength topnrPercMatch
           &lt;character&gt;    &lt;character&gt;
1                  258           0.53
2                  201           0.88
3                  271           0.73
4                  172           0.85
5                  206           0.55
6                  196           0.60
        nobadwordnrMatch nobadwordnrEvalue
             &lt;character&gt;       &lt;character&gt;
1 gnl|BL_ORD_ID|726506..       3.99079e-52
2 gnl|BL_ORD_ID|110398..       3.65321e-84
3 gnl|BL_ORD_ID|110400..      9.23041e-133
4 gnl|BL_ORD_ID|110376..       2.29747e-78
5 gnl|BL_ORD_ID|110383..       2.14754e-64
6 gnl|BL_ORD_ID|110383..       1.33696e-65
  nobadwordnrAlignLength nobadwordnrPercentMatch
             &lt;character&gt;             &lt;character&gt;
1                    264                    0.45
2                    201                    0.88
3                    271                    0.73
4                    172                    0.85
5                    206                    0.55
6                    196                    0.60
  Uniprotmatch X._identity      evalue
   &lt;character&gt; &lt;character&gt; &lt;character&gt;
1   A0A1X7UWJ8      40.927    1.82e-51
2       Q4R6F3      37.500    1.77e-06
3       Q8N2E2      31.022    1.60e-30
4   A0A2B4RJ64      70.690    9.57e-61
5   A0A2B4RML7      46.635    3.49e-44
6   A0A2B4RML7      47.847    2.78e-48
                ID            Description
       &lt;character&gt;            &lt;character&gt;
1 A0A1X7UWJ8_AMPQE Uncharacterized prot..
2      NGLY1_MACFA Peptide-N(4)-(N-acet..
3       VWDE_HUMAN von Willebrand facto..
4 A0A2B4RJ64_STYPI Uncharacterized prot..
5 A0A2B4RML7_STYPI Uncharacterized prot..
6 A0A2B4RML7_STYPI Uncharacterized prot..
           KEGG      KEGGKO                   PFAM
    &lt;character&gt; &lt;character&gt;            &lt;character&gt;
1       No_KEGG   No_KEGGKO       PF05585; DUF1758
2 mcf:101925902      K01456 PF04721; PAW,  PF094..
3    hsa:221806   No_KEGGKO           PF00094; VWD
4       No_KEGG   No_KEGGKO                   Pfam
5       No_KEGG   No_KEGGKO                   Pfam
6       No_KEGG   No_KEGGKO                   Pfam
                      GO                   GOCC
             &lt;character&gt;            &lt;character&gt;
1             GO:0004190                No_GOBP
2 GO:0005737 // GO:004.. glycoprotein catabol..
3 GO:0009986 // GO:000.. anatomical structure..
4             GO:0016021                No_GOBP
5             No_GOcodes                No_GOBP
6             No_GOcodes                No_GOBP
                    GOBP                   GOMF
             &lt;character&gt;            &lt;character&gt;
1 aspartic-type endope..                No_GOCC
2 metal ion binding (G.. cytoplasm (GO:000573..
3 signaling receptor b.. cell surface (GO:000..
4                No_GOMF integral component o..
5                No_GOMF                No_GOCC
6                No_GOMF                No_GOCC
                Keywords
             &lt;character&gt;
1 Complete proteome {E..
2 Acetylation;Cytoplas..
3 Alternative splicing..
4 Coiled coil {ECO:000..
5 Complete proteome {E..
6 Complete proteome {E..
&gt; write.table(ResultsOutput, file=&quot;177OriginDEGswAnnotation.txt&quot;, quote=F, row.names=F, sep=&quot;\t&quot;)
&gt; 



#9- Optional, also plot a PCA using only the genes with a pvalue&lt;0.05 for the &quot;origin&quot; and &quot;symstate&quot; variables

#10- Even more optional, color code your symbols by temperature treatment as well as origin and symstate." dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>1- mkdir a directory with your name in the </span>

<span class="pl-c"><span class="pl-c">#</span>2- cp/mv your logfile to the /cm/shared/courses/dbarshis/18AdvBioinf/archive/YOURNAME directory</span>

<span class="pl-c"><span class="pl-c">#</span>3- cp/mv your merged, unfiltered .vcf file YOURINITIALS_unfiltered.vcf into the /cm/shared/courses/dbarshis/18AdvBioinf/archive/YOURNAME directory</span>

<span class="pl-c"><span class="pl-c">#</span>4- cp/mv any other files you might want to keep to the /cm/shared/courses/dbarshis/18AdvBioinf/archive/YOURNAME directory</span>

<span class="pl-c"><span class="pl-c">#</span>5- create an sbatch script to rm -r your sandbox/YOURNAME directory</span>

<span class="pl-c"><span class="pl-c">#</span>6- scp/winscp /cm/shared/courses/dbarshis/18AdvBioinf/classdata/Astrangia_poculata/expressioncounts/ALLLANESFullCounts_summed.txt to your laptop</span>

<span class="pl-c"><span class="pl-c">#</span>7- scp/winscp /cm/shared/courses/dbarshis/18AdvBioinf/classdata/Astrangia_poculata/expressioncounts/ALLLANES_conditions.txt to your laptop</span>

<span class="pl-c"><span class="pl-c">#</span>8- Download the DESeq2Script_advbioinf_day12.R script and work through to generate a PCA of your data.</span>


<span class="pl-k">&gt;</span> library(DESeq2)
<span class="pl-k">&gt;</span> library(gplots)
<span class="pl-k">&gt;</span> library(goseq)
<span class="pl-k">&gt;</span> library(GO.db)
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> setwd(<span class="pl-s"><span class="pl-pds">"</span>/Users/katiecrider/Desktop/Genomics<span class="pl-pds">"</span></span>)  <span class="pl-c"><span class="pl-c">#</span> The drag and drop from finder works in R, too.</span>
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>useful functions</span>
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>head() - prints out the top 6 lines</span>
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>dim() - prints the dimensions of a variable</span>
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>nrow() - returns the number of rows in a vector or matrix</span>
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span> ?[functionName] - opens documentation describing the function</span>
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>read in your data to make counts table</span>
<span class="pl-k">&gt;</span> countsTable <span class="pl-k">&lt;</span>- read.delim(<span class="pl-s"><span class="pl-pds">'</span>djbFullCounts_summed.txt<span class="pl-pds">'</span></span>, header=TRUE, stringsAsFactors=TRUE, row.names=1)
<span class="pl-k">&gt;</span> head(countsTable)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10054<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          0          0          0
TR10054<span class="pl-k">|</span>c0_g2_i1_coral          0          1          6          0          0          0
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          0          1          2          0          1          2
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0          3          1          0          4          0
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          0          4          2          1          0          0
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          0          1          4          0          0          4
                       RI_B_03_14 RI_B_03_18 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22
TR10054<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          0          0          0
TR10054<span class="pl-k">|</span>c0_g2_i1_coral          0          0          0          9          1          0
TR10083<span class="pl-k">|</span>c0_g2_i1_coral         10          0          1         31         20          0
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          5         10          2
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          0          0          3         12          9          0
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          1          0          3         15          8          0
                       RI_B_05_14 RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22
TR10054<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          0          0          0
TR10054<span class="pl-k">|</span>c0_g2_i1_coral          2          0          0          0          2          0
TR10083<span class="pl-k">|</span>c0_g2_i1_coral         16          0          3          3          5          6
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          2          0          0          0          3          0
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          2          0          3          0          7          0
TR10103<span class="pl-k">|</span>c0_g1_i1_coral         18          0          5          0          5          0
                       RI_B_07_14 RI_B_07_18 RI_B_07_22 RI_W_01_14 RI_W_01_18 RI_W_01_22
TR10054<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          0          0          0
TR10054<span class="pl-k">|</span>c0_g2_i1_coral          2          1          0          0          0          1
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          2          8          4         11         15          3
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          1          2          0          0         17          1
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          5          1          0          0         17          5
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          3          2          0          5         11          0
                       RI_W_02_14 RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22
TR10054<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          0          0          0
TR10054<span class="pl-k">|</span>c0_g2_i1_coral          1          1          2          0          5          0
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          0          5          3         14         15          9
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0          2          0          7          3          3
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          2          5          4          5          9          3
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          0          1          3          5          9          2
                       RI_W_04_14 RI_W_04_18 RI_W_04_22 RI_W_05_14 RI_W_05_18 RI_W_05_22
TR10054<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          0          0          0
TR10054<span class="pl-k">|</span>c0_g2_i1_coral          1          1          0          2          0          0
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          3          6          1          7          1          4
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0          7          0          2          8          0
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          0         15          0          8         10          1
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          0          2          0          9          2          2
                       RI_W_06_14 RI_W_06_18 RI_W_06_22 RI_W_07_14 RI_W_07_18 RI_W_07_22
TR10054<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          0          0          0
TR10054<span class="pl-k">|</span>c0_g2_i1_coral          2          6          0          4          0          0
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          1          1          5          6          5          5
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0          0          4          0          0          0
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          3          1          7          3          1          4
TR10103<span class="pl-k">|</span>c0_g1_i1_coral         10          1          2          8         12          1
                       VA_B_01_14 VA_B_01_18 VA_B_01_22 VA_B_02_14 VA_B_02_18 VA_B_02_22
TR10054<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          0          0          0
TR10054<span class="pl-k">|</span>c0_g2_i1_coral          0          2          2          0          0          0
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          2          4          5          0         11          9
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          6         21          3          0          0          0
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          8         13         10          0          0          0
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          3          6          5          0          0          1
                       VA_B_03_14 VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18 VA_B_04_22
TR10054<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          0          0          0
TR10054<span class="pl-k">|</span>c0_g2_i1_coral          1          0          6          0         10          4
TR10083<span class="pl-k">|</span>c0_g2_i1_coral         22         20         11          0          2         13
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          8          3         11          0         10          7
TR10090<span class="pl-k">|</span>c0_g2_i1_coral         15          6          5          8          7          0
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          3          6          5          5          0          0
                       VA_B_05_14 VA_B_05_18 VA_B_05_22 VA_B_06_14 VA_B_06_18 VA_B_06_22
TR10054<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          0          0          0
TR10054<span class="pl-k">|</span>c0_g2_i1_coral          2          2          0         10          0          1
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          6          6          1          2          0         12
TR10090<span class="pl-k">|</span>c0_g1_i1_coral         14          2          4         12          0          1
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          4          1          3          0          5          3
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          0          6          0          8          8          8
                       VA_B_07_14 VA_B_07_18 VA_B_07_22 VA_W_01_14 VA_W_01_18 VA_W_01_22
TR10054<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          0          0          0
TR10054<span class="pl-k">|</span>c0_g2_i1_coral          0          4          1          1          3          0
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          5          1          2          7         14          0
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0          5          2          4          2          0
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          2         21          6          0          7          0
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          2         11          4          0          7          0
                       VA_W_02_14 VA_W_02_18 VA_W_02_22 VA_W_03_14 VA_W_03_18 VA_W_03_22
TR10054<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          0          0          0
TR10054<span class="pl-k">|</span>c0_g2_i1_coral          4          0          0          4          0          6
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          0          7          0          4          0         11
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          8          1          0          7          0         15
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          5         20          0          9         17          5
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          6          1          0          9          0          4
                       VA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22
TR10054<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          0          0          0
TR10054<span class="pl-k">|</span>c0_g2_i1_coral          2          0          2          0          3          0
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          2          9         13          0          7         10
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          5         12          8          0          2         10
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          4         16         22          0          5          6
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          7          8          6          1          6          2
                       VA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22
TR10054<span class="pl-k">|</span>c0_g1_i1_coral          0          0          0          0          0          0
TR10054<span class="pl-k">|</span>c0_g2_i1_coral          2          0          0          0          0          3
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          5          3          0         13          1          1
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0          1          2          2          1          7
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          3          1          0          7          3          3
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          8          3          5          6          0          2
<span class="pl-k">&gt;</span> dim(countsTable)
[1] 13381    84
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> colSums(countsTable)
RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_18 
     97406     207372     173843      47481      58036      84501     170330       5972 
RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14 RI_B_05_18 RI_B_05_22 RI_B_06_14 
    199204     357124     544267      71600     630609     227313     225639      37012 
RI_B_06_18 RI_B_06_22 RI_B_07_14 RI_B_07_18 RI_B_07_22 RI_W_01_14 RI_W_01_18 RI_W_01_22 
    296071      72254     257730     218198      25369      39419     292612     174666 
RI_W_02_14 RI_W_02_18 RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18 
     54184     192092     155178     281297     428532     359053      37579     300578 
RI_W_04_22 RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22 RI_W_07_14 
     10848     392818     214887     158683     186568     205135     198272     342795 
RI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22 VA_B_02_14 VA_B_02_18 VA_B_02_22 
    220630     176946     261824     548448     328198      79276     152742      85224 
VA_B_03_14 VA_B_03_18 VA_B_03_22 VA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 
    226784     251696     271959     172636     264192     268290     323715     288121 
VA_B_05_22 VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22 VA_W_01_14 
     89943     232578     195521     362054     215616     234465     169742     164874 
VA_W_01_18 VA_W_01_22 VA_W_02_14 VA_W_02_18 VA_W_02_22 VA_W_03_14 VA_W_03_18 VA_W_03_22 
    321900        313     170250     241804        134     278000     138573     295382 
VA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22 VA_W_06_14 VA_W_06_18 
    292041     280007     261344     178096     388532     157123     176579     188506 
VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22 
    209003     228792     171481     233899 
<span class="pl-k">&gt;</span> colSums(countsTable)[order(colSums(countsTable))]
VA_W_02_22 VA_W_01_22 RI_B_03_18 RI_W_04_22 RI_B_07_22 RI_B_06_14 RI_W_04_14 RI_W_01_14 
       134        313       5972      10848      25369      37012      37579      39419 
RI_B_02_14 RI_W_02_14 RI_B_02_18 RI_B_04_22 RI_B_06_22 VA_B_02_14 RI_B_02_22 VA_B_02_22 
     47481      54184      58036      71600      72254      79276      84501      85224 
VA_B_05_22 RI_B_01_14 VA_W_03_18 VA_B_02_18 RI_W_02_22 VA_W_05_22 RI_W_05_22 VA_W_01_14 
     89943      97406     138573     152742     155178     157123     158683     164874 
VA_B_07_22 VA_W_02_14 RI_B_03_14 VA_W_07_18 VA_B_04_14 RI_B_01_22 RI_W_01_22 VA_W_06_14 
    169742     170250     170330     171481     172636     173843     174666     176579 
RI_W_07_22 VA_W_05_14 RI_W_06_14 VA_W_06_18 RI_W_02_18 VA_B_06_18 RI_W_06_22 RI_B_03_22 
    176946     178096     186568     188506     192092     195521     198272     199204 
RI_W_06_18 RI_B_01_18 VA_W_06_22 RI_W_05_18 VA_B_07_14 RI_B_07_18 RI_W_07_18 RI_B_05_22 
    205135     207372     209003     214887     215616     218198     220630     225639 
VA_B_03_14 RI_B_05_18 VA_W_07_14 VA_B_06_14 VA_W_07_22 VA_B_07_18 VA_W_02_18 VA_B_03_18 
    226784     227313     228792     232578     233899     234465     241804     251696 
RI_B_07_14 VA_W_04_22 VA_B_01_14 VA_B_04_18 VA_B_04_22 VA_B_03_22 VA_W_03_14 VA_W_04_18 
    257730     261344     261824     264192     268290     271959     278000     280007 
RI_W_03_14 VA_B_05_18 VA_W_04_14 RI_W_01_18 VA_W_03_22 RI_B_06_18 RI_W_04_18 VA_W_01_18 
    281297     288121     292041     292612     295382     296071     300578     321900 
VA_B_05_14 VA_B_01_22 RI_W_07_14 RI_B_04_14 RI_W_03_22 VA_B_06_22 VA_W_05_18 RI_W_05_14 
    323715     328198     342795     357124     359053     362054     388532     392818 
RI_W_03_18 RI_B_04_18 VA_B_01_18 RI_B_05_14 
    428532     544267     548448     630609 
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> table(substring(names(countsTable),1,2))

RI VA 
42 42 
<span class="pl-k">&gt;</span> table(substring(names(countsTable),4,4))

 B  W 
42 42 
<span class="pl-k">&gt;</span> table(substring(names(countsTable),9,10))

14 18 22 
28 28 28 
<span class="pl-k">&gt;</span> table(paste(substring(names(countsTable),1,2),substring(names(countsTable),4,4), sep=<span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>))

RI_B RI_W VA_B VA_W 
  21   21   21   21 
<span class="pl-k">&gt;</span> table(paste(substring(names(countsTable),1,2),substring(names(countsTable),4,4), substring(names(countsTable),9,10), sep=<span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>))

RI_B_14 RI_B_18 RI_B_22 RI_W_14 RI_W_18 RI_W_22 VA_B_14 VA_B_18 VA_B_22 VA_W_14 VA_W_18 VA_W_22 
      7       7       7       7       7       7       7       7       7       7       7       7 
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> LowCounts<span class="pl-k">&lt;</span>-c(<span class="pl-s"><span class="pl-pds">"</span>VA_W_02_22<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>VA_W_01_22<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>RI_B_03_18<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI_W_01_14<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI_W_04_22<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>VA_W_03_18<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> names(countsTable)%in%LowCounts
 [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[16] FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[31] FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[46] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[61] FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE FALSE  TRUE FALSE FALSE FALSE FALSE
[76] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
<span class="pl-k">&gt;</span> <span class="pl-k">!</span>(names(countsTable)%in%LowCounts)
 [1]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[16]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[31]  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[46]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[61]  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE FALSE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE
[76]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
<span class="pl-k">&gt;</span> countsTable<span class="pl-k">&lt;</span>-countsTable[,<span class="pl-k">!</span>(names(countsTable)%in%LowCounts)]
<span class="pl-k">&gt;</span> dim(countsTable)
[1] 13381    78
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>make a table with conditions of each individual (e.g. "VA" and "RI")  There should be the same number of conditions described as there are samples in your data file, and in the same order.</span>
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>NOTE: It is absolutely critical that the columns of the count matrix and the rows of the column data (information about samples) are in the same order. DESeq2 will not make guesses as to which column of the count matrix belongs to which row of the column data, these must be provided to DESeq2 already in consistent order.</span>
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>Here's an example</span>
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span> Sample		Origin	SymState	Temp  Origin_SymState Origin_SymState_Temp</span>
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span> RI_B_06_18		RI	B	18  RI_B  RI_B_18</span>
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span> VA_B_07_14		VA	B	14  VA_B  VA_B_14</span>
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span> VA_W_07_22		VA	W	22  VA_W  VA_W_22</span>
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span> safest way to do this is in R</span>
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> conds<span class="pl-k">&lt;</span>-data.frame(<span class="pl-s"><span class="pl-pds">"</span>Sample<span class="pl-pds">"</span></span>=names(countsTable))
<span class="pl-k">&gt;</span> conds<span class="pl-smi">$Origin</span><span class="pl-k">&lt;</span>-factor(substring(conds<span class="pl-smi">$Sample</span>,1,2))
<span class="pl-k">&gt;</span> conds<span class="pl-smi">$SymState</span><span class="pl-k">&lt;</span>-factor(substring(conds<span class="pl-smi">$Sample</span>,4,4))
<span class="pl-k">&gt;</span> conds<span class="pl-smi">$Temp</span><span class="pl-k">&lt;</span>-factor(substring(conds<span class="pl-smi">$Sample</span>,9,10))
<span class="pl-k">&gt;</span> conds<span class="pl-smi">$Origin_SymState</span><span class="pl-k">&lt;</span>-factor(paste(conds<span class="pl-smi">$Origin</span>,conds<span class="pl-smi">$SymState</span>, sep=<span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> conds<span class="pl-smi">$Origin_SymState_Temp</span><span class="pl-k">&lt;</span>-factor(paste(conds<span class="pl-smi">$Origin</span>,conds<span class="pl-smi">$SymState</span>,conds<span class="pl-smi">$Temp</span>, sep=<span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>#Check null counts per sample</span>
<span class="pl-k">&gt;</span> prop.null <span class="pl-k">&lt;</span>- apply(countsTable, 2, function(x) 100<span class="pl-k">*</span>mean(x==0))
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> barplot(prop.null, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> pdf(file=<span class="pl-s"><span class="pl-pds">"</span>OverallNullCountsv2.pdf<span class="pl-pds">"</span></span>,14, 14)
<span class="pl-k">&gt;</span> barplot(prop.null, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)
<span class="pl-k">&gt;</span> <span class="pl-en">dev.off</span>()
RStudioGD 
        2 
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span> how many genes have mean count &gt;=3 </span>
<span class="pl-k">&gt;</span> means = apply(countsTable, 1, mean)
<span class="pl-k">&gt;</span> table(means<span class="pl-k">&gt;</span>=3)

FALSE  TRUE 
 5721  7660 
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> countsTable<span class="pl-k">&lt;</span>-countsTable[means<span class="pl-k">&gt;=</span>3,]
<span class="pl-k">&gt;</span> dim(countsTable)
[1] 7660   78
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> prop.nullv2 <span class="pl-k">&lt;</span>- apply(countsTable, 2, function(x) 100<span class="pl-k">*</span>mean(x==0))
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> barplot(prop.nullv2, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> pdf(file=<span class="pl-s"><span class="pl-pds">"</span>Mean3NullCountsv2.pdf<span class="pl-pds">"</span></span>,14, 14)
<span class="pl-k">&gt;</span> barplot(prop.nullv2, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)
<span class="pl-k">&gt;</span> <span class="pl-en">dev.off</span>()
RStudioGD 
        2 
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>make count data sets</span>
<span class="pl-k">&gt;</span> dds <span class="pl-k">&lt;</span>- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=<span class="pl-k">~</span> Origin + SymState + Temp + Origin:SymState)
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> dds <span class="pl-k">&lt;</span>- DESeq(dds)
estimating size factors
estimating dispersions
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
fitting model and testing
-- replacing outliers and refitting <span class="pl-k">for</span> 14 genes
-- DESeq argument <span class="pl-s"><span class="pl-pds">'</span>minReplicatesForReplace<span class="pl-pds">'</span></span> = 7 
-- original counts are preserved <span class="pl-k">in</span> counts(dds)
estimating dispersions
fitting model and testing
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> dim(dds)
[1] 7660   78
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>transform counts to variance stablized counts</span>
<span class="pl-k">&gt;</span> vstCounts<span class="pl-k">&lt;</span>-varianceStabilizingTransformation(dds)
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>plot cluster dendrogram across samples</span>
<span class="pl-k">&gt;</span> dists<span class="pl-k">&lt;</span>-dist(t(assay(vstCounts)))
<span class="pl-k">&gt;</span> plot(hclust(dists))
<span class="pl-k">&gt;</span> heatmap.2(as.matrix(dists), key=F, trace=<span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>,
+           col=colorpanel(100, <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>),
+           margin=c(10, 10))
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>##Figure out which contrast you want to examine (i.e. which two groups do you want to compare)</span>
<span class="pl-k">&gt;</span> resultsNames(dds)
[1] <span class="pl-s"><span class="pl-pds">"</span>Intercept<span class="pl-pds">"</span></span>          <span class="pl-s"><span class="pl-pds">"</span>Origin_VA_vs_RI<span class="pl-pds">"</span></span>    <span class="pl-s"><span class="pl-pds">"</span>SymState_W_vs_B<span class="pl-pds">"</span></span>    <span class="pl-s"><span class="pl-pds">"</span>Temp_18_vs_14<span class="pl-pds">"</span></span>     
[5] <span class="pl-s"><span class="pl-pds">"</span>Temp_22_vs_14<span class="pl-pds">"</span></span>      <span class="pl-s"><span class="pl-pds">"</span>OriginVA.SymStateW<span class="pl-pds">"</span></span>
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> res <span class="pl-k">&lt;</span>- results(dds)
<span class="pl-k">&gt;</span> head(res)
log2 fold change (MLE): OriginVA.SymStateW 
Wald <span class="pl-c1">test</span> p-value: OriginVA.SymStateW 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>      <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   5.36648       0.107690  0.644646  0.167053  0.867328  0.999364
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.70436      -0.247209  0.831371 -0.297351  0.766198  0.995209
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.65756      -0.976997  0.654110 -1.493627  0.135273  0.958807
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   2.91279      -0.172220  0.641026 -0.268662  0.788189  0.995209
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   6.31144       0.366046  0.664401  0.550942  0.581673  0.991107
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   9.26737      -0.589357  0.975809 -0.603967  0.545865  0.990477
<span class="pl-k">&gt;</span> summary(res)

out of 7659 with nonzero total <span class="pl-c1">read</span> count
adjusted p-value <span class="pl-k">&lt;</span> 0.1
LFC <span class="pl-k">&gt;</span> 0 (up)       <span class="pl-c1">:</span> 2, 0.026%
LFC <span class="pl-k">&lt;</span> 0 (down)     <span class="pl-c1">:</span> 9, 0.12%
outliers [1]       <span class="pl-c1">:</span> 10, 0.13%
low counts [2]     <span class="pl-c1">:</span> 1, 0.013%
(mean count <span class="pl-k">&lt;</span> 1)
[1] see <span class="pl-s"><span class="pl-pds">'</span>cooksCutoff<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results
[2] see <span class="pl-s"><span class="pl-pds">'</span>independentFiltering<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results

<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> resSymState <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>SymState<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>B<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>W<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> head(resSymState)
log2 fold change (MLE): SymState B vs W 
Wald <span class="pl-c1">test</span> p-value: SymState B vs W 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>      <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   5.36648       0.304875  0.463588  0.657643 0.5107675  0.986343
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.70436      -0.340214  0.624845 -0.544477 0.5861132  0.995961
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.65756      -1.044272  0.488673 -2.136956 0.0326015  0.816270
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   2.91279      -0.316211  0.469123 -0.674047 0.5002815  0.986343
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   6.31144       0.257974  0.481341  0.535948 0.5919943  0.995961
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   9.26737      -0.604957  0.692218 -0.873939 0.3821514  0.986343
<span class="pl-k">&gt;</span> summary(resSymState)

out of 7659 with nonzero total <span class="pl-c1">read</span> count
adjusted p-value <span class="pl-k">&lt;</span> 0.1
LFC <span class="pl-k">&gt;</span> 0 (up)       <span class="pl-c1">:</span> 6, 0.078%
LFC <span class="pl-k">&lt;</span> 0 (down)     <span class="pl-c1">:</span> 6, 0.078%
outliers [1]       <span class="pl-c1">:</span> 10, 0.13%
low counts [2]     <span class="pl-c1">:</span> 1, 0.013%
(mean count <span class="pl-k">&lt;</span> 1)
[1] see <span class="pl-s"><span class="pl-pds">'</span>cooksCutoff<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results
[2] see <span class="pl-s"><span class="pl-pds">'</span>independentFiltering<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results

<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> resOrigin <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Origin<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>VA<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>RI<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> head(resOrigin)
log2 fold change (MLE): Origin VA vs RI 
Wald <span class="pl-c1">test</span> p-value: Origin VA vs RI 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE       stat    pvalue      padj
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>      <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>  <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   5.36648     -0.2712427  0.446225 -0.6078616 0.5432793  0.865661
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.70436      1.1125117  0.589306  1.8878328 0.0590484        NA
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.65756      0.9431346  0.474719  1.9867203 0.0469534  0.418043
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   2.91279      0.0386643  0.456720  0.0846565 0.9325345  0.984590
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   6.31144     -0.0179010  0.461958 -0.0387502 0.9690895  0.992150
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   9.26737     -0.2649903  0.677608 -0.3910675 0.6957473  0.921517
<span class="pl-k">&gt;</span> summary(resOrigin)

out of 7659 with nonzero total <span class="pl-c1">read</span> count
adjusted p-value <span class="pl-k">&lt;</span> 0.1
LFC <span class="pl-k">&gt;</span> 0 (up)       <span class="pl-c1">:</span> 73, 0.95%
LFC <span class="pl-k">&lt;</span> 0 (down)     <span class="pl-c1">:</span> 104, 1.4%
outliers [1]       <span class="pl-c1">:</span> 10, 0.13%
low counts [2]     <span class="pl-c1">:</span> 594, 7.8%
(mean count <span class="pl-k">&lt;</span> 3)
[1] see <span class="pl-s"><span class="pl-pds">'</span>cooksCutoff<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results
[2] see <span class="pl-s"><span class="pl-pds">'</span>independentFiltering<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results

<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> res14v18 <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>14<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>18<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> head(res14v18)
log2 fold change (MLE): Temp 14 vs 18 
Wald <span class="pl-c1">test</span> p-value: Temp 14 vs 18 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE       stat    pvalue      padj
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>      <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>  <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   5.36648      0.0134106  0.389675  0.0344147 0.9725464  0.994763
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.70436     -0.7500403  0.493720 -1.5191625 0.1287216        NA
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.65756     -0.5278497  0.385227 -1.3702295 0.1706153        NA
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   2.91279      0.1448997  0.377624  0.3837147 0.7011899        NA
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   6.31144     -0.1171316  0.401865 -0.2914700 0.7706919  0.956296
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   9.26737     -1.1666726  0.592603 -1.9687238 0.0489848  0.440134
<span class="pl-k">&gt;</span> summary(res14v18)

out of 7659 with nonzero total <span class="pl-c1">read</span> count
adjusted p-value <span class="pl-k">&lt;</span> 0.1
LFC <span class="pl-k">&gt;</span> 0 (up)       <span class="pl-c1">:</span> 31, 0.4%
LFC <span class="pl-k">&lt;</span> 0 (down)     <span class="pl-c1">:</span> 111, 1.4%
outliers [1]       <span class="pl-c1">:</span> 10, 0.13%
low counts [2]     <span class="pl-c1">:</span> 2671, 35%
(mean count <span class="pl-k">&lt;</span> 4)
[1] see <span class="pl-s"><span class="pl-pds">'</span>cooksCutoff<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results
[2] see <span class="pl-s"><span class="pl-pds">'</span>independentFiltering<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results

<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> res14v22 <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>14<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>22<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> head(res14v22)
log2 fold change (MLE): Temp 14 vs 22 
Wald <span class="pl-c1">test</span> p-value: Temp 14 vs 22 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>      <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   5.36648      -0.381080  0.396906 -0.960127 0.3369915  0.796010
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.70436      -0.420391  0.517100 -0.812978 0.4162304        NA
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.65756      -0.149468  0.407436 -0.366851 0.7137303        NA
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   2.91279       0.327506  0.400038  0.818686 0.4129654        NA
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   6.31144      -0.504681  0.409220 -1.233275 0.2174733  0.699820
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   9.26737      -1.240058  0.602190 -2.059248 0.0394705  0.387444
<span class="pl-k">&gt;</span> summary(res14v22)

out of 7659 with nonzero total <span class="pl-c1">read</span> count
adjusted p-value <span class="pl-k">&lt;</span> 0.1
LFC <span class="pl-k">&gt;</span> 0 (up)       <span class="pl-c1">:</span> 64, 0.84%
LFC <span class="pl-k">&lt;</span> 0 (down)     <span class="pl-c1">:</span> 97, 1.3%
outliers [1]       <span class="pl-c1">:</span> 10, 0.13%
low counts [2]     <span class="pl-c1">:</span> 2226, 29%
(mean count <span class="pl-k">&lt;</span> 4)
[1] see <span class="pl-s"><span class="pl-pds">'</span>cooksCutoff<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results
[2] see <span class="pl-s"><span class="pl-pds">'</span>independentFiltering<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results

<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> res18v22 <span class="pl-k">&lt;</span>- results(dds, contrast=c(<span class="pl-s"><span class="pl-pds">"</span>Temp<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>18<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>22<span class="pl-pds">"</span></span>))
<span class="pl-k">&gt;</span> head(res18v22)
log2 fold change (MLE): Temp 18 vs 22 
Wald <span class="pl-c1">test</span> p-value: Temp 18 vs 22 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE      stat    pvalue      padj
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>      <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   5.36648     -0.3944906  0.395010 -0.998685  0.317947  0.838585
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.70436      0.3296494  0.503430  0.654807  0.512592        NA
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.65756      0.3783816  0.396935  0.953259  0.340459  0.847759
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   2.91279      0.1826059  0.399214  0.457414  0.647374        NA
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   6.31144     -0.3875492  0.406736 -0.952828  0.340677  0.847759
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   9.26737     -0.0733858  0.596241 -0.123081  0.902043  0.991572
<span class="pl-k">&gt;</span> summary(res18v22)

out of 7659 with nonzero total <span class="pl-c1">read</span> count
adjusted p-value <span class="pl-k">&lt;</span> 0.1
LFC <span class="pl-k">&gt;</span> 0 (up)       <span class="pl-c1">:</span> 34, 0.44%
LFC <span class="pl-k">&lt;</span> 0 (down)     <span class="pl-c1">:</span> 10, 0.13%
outliers [1]       <span class="pl-c1">:</span> 10, 0.13%
low counts [2]     <span class="pl-c1">:</span> 1337, 17%
(mean count <span class="pl-k">&lt;</span> 3)
[1] see <span class="pl-s"><span class="pl-pds">'</span>cooksCutoff<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results
[2] see <span class="pl-s"><span class="pl-pds">'</span>independentFiltering<span class="pl-pds">'</span></span> argument of <span class="pl-k">?</span>results

<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>count the number of significantly differentially expressed genes</span>
<span class="pl-k">&gt;</span> sum(resOrigin<span class="pl-smi">$padj</span> <span class="pl-k">&lt;</span> 0.1, na.rm =T)
[1] 177
<span class="pl-k">&gt;</span> sum(resOrigin<span class="pl-smi">$pvalue</span> <span class="pl-k">&lt;</span> 0.05, na.rm =T)
[1] 866
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span> Make a counts table that is scaled by the size factors</span>
<span class="pl-k">&gt;</span> scaledcounts = counts(dds, normalize=T)
<span class="pl-k">&gt;</span> head(scaledcounts)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          0  0.8671827   2.447866   0.000000   3.455170   4.432653
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0  2.6015481   1.223933   0.000000  13.820681   0.000000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          0  3.4687308   2.447866   3.772786   0.000000   0.000000
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          0  0.8671827   4.895733   0.000000   0.000000   8.865306
TR10108<span class="pl-k">|</span>c0_g1_i1_coral          0  1.7343654   4.895733   0.000000   6.910341   0.000000
TR10132<span class="pl-k">|</span>c0_g1_i1_coral          0  6.0702789   1.223933  18.863931  51.827555   0.000000
                       RI_B_03_14 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  12.871970   1.272998  12.424473  8.1878079   0.000000   4.563408
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000   2.003947  4.0939039   6.765958   0.570426
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000   3.818995   4.809473  3.6845136   0.000000   0.570426
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   1.287197   3.818995   6.011842  3.2751232   0.000000   5.133834
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  14.159167  15.275980   2.805526  4.9126847  16.914895   5.989473
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   6.435985   2.545997   0.000000  0.8187808   3.382979   1.426065
                       RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22 RI_B_07_14
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          0  2.4538305   19.55961   3.448566   18.78455  1.4936671
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0  0.0000000    0.00000   2.069140    0.00000  0.7468335
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          0  2.4538305    0.00000   4.827992    0.00000  3.7341677
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          0  4.0897176    0.00000   3.448566    0.00000  2.2405006
TR10108<span class="pl-k">|</span>c0_g1_i1_coral          0  0.8179435    6.51987  10.345698   40.69986  3.7341677
TR10132<span class="pl-k">|</span>c0_g1_i1_coral          0  1.6358870    0.00000   1.379426    0.00000  8.9620025
                       RI_B_07_18 RI_B_07_22 RI_W_01_18 RI_W_01_22 RI_W_02_14 RI_W_02_18
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   8.070182   36.85316  10.815190   4.646396   0.000000  4.9947759
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.017545    0.00000  12.257216   1.548799   0.000000  1.9979104
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   1.008773    0.00000  12.257216   7.743994   4.651361  4.9947759
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   2.017545    0.00000   7.931140   0.000000   0.000000  0.9989552
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   1.008773    0.00000   2.884051   1.548799   9.302721  6.9926863
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  44.386000   27.63987   1.442025   0.000000   0.000000  5.9937311
                       RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   3.615216   8.525621   5.407467   4.745758   12.82564  2.9768406
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   4.262810   1.081493   1.581919    0.00000  3.4729807
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   4.820288   3.044865   3.244480   1.581919    0.00000  7.4421015
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   3.615216   3.044865   3.244480   1.054613    0.00000  0.9922802
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   2.410144   2.435892   4.686471   2.109226   21.37607 12.8996426
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  44.587667   1.217946   1.081493   1.054613   29.92650  7.9382416
                       RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  3.1630643   1.010704   5.526636   1.044347   1.336966   5.564881
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.9037326   8.085630   0.000000   0.000000   0.000000   4.451904
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  3.6149306  10.107037   1.381659   3.133041   1.336966   7.790833
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  4.0667969   2.021407   2.763318  10.443469   1.336966   2.225952
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  5.4223959   2.021407   6.908296  11.487816   0.000000   5.564881
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  1.8074653  10.107037  27.633182   5.221735   1.336966  42.293092
                       RI_W_07_14 RI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   4.124636   5.593017   6.900578   2.551157   1.731769   2.845097
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000   0.000000   7.653472   9.091789   1.707058
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   2.062318   1.118603   5.520462  10.204629   5.628251   5.690194
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   5.499514  13.423242   1.380116   3.826736   2.597654   2.845097
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   1.118603   5.520462   3.826736   9.091789   2.845097
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   1.374879   2.237207  86.947282   1.275579   1.298827   0.000000
                       VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14 VA_B_03_18 VA_B_03_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          0    14.2516  19.312637  16.029307  14.339493   7.833217
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0     0.0000   0.000000   5.828839   2.150924   7.833217
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          0     0.0000   0.000000  10.929073   4.301848   3.560553
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          0     0.0000   2.145849   2.185815   4.301848   3.560553
TR10108<span class="pl-k">|</span>c0_g1_i1_coral          0     0.0000   0.000000  14.572098  18.641340  21.363318
TR10132<span class="pl-k">|</span>c0_g1_i1_coral          0     0.0000   8.583394   9.471863  30.829909   8.545327
                       VA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   0.000000  0.9006809   6.731616   2.098865  3.3173156   1.813677
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  4.5034046   3.624716   4.897352  1.1057719   7.254707
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   7.108517  3.1523832   0.000000   1.399244  0.5528859   5.441030
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   4.442823  0.0000000   0.000000   0.000000  3.3173156   0.000000
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   2.665694  4.9537450   3.624716   9.095083  5.5288593   0.000000
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  2.7020428   0.000000   0.000000  4.9759734   5.441030
                       VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   1.383946   0.000000  5.9098781   3.728534  0.7405861   2.098499
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   8.303676   0.000000  0.4924898   0.000000  3.7029304   2.098499
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000   4.206352  1.4774695   1.491414 15.5523076   6.295498
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   5.535784   6.730163  3.9399187   1.491414  8.1464468   4.196999
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  11.763540   4.206352  3.4474289   2.237121  2.2217582   5.246248
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  18.683270   2.523811 14.2822054   8.948482 17.0334798   2.098499
                       VA_W_01_14 VA_W_01_18 VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   6.699293   7.732656   0.000000  4.5789684  3.1096023   6.888125
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   3.828167   1.104665   6.978382  0.6541383  5.4418040   9.392897
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000   3.866328   4.361489 13.0827667  6.9966052   3.130966
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   3.866328   5.233787  0.6541383  6.9966052   2.504773
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   1.914084   4.418660   0.000000  9.1579367  3.1096023   3.757159
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  10.494319   2.616893 13.0827667  0.7774006  13.150056
                       VA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   1.026752   4.334366   6.410680  0.0000000   3.848517  12.379823
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.566880   5.779155   3.945034  0.0000000   1.099576  12.379823
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   2.053504   7.705540  10.848842  0.0000000   2.748941   7.427894
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   3.593631   3.852770   2.958775  0.8118702   3.298729   2.475965
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   6.673887  20.227043   8.383196  1.6237403   5.497882  38.377452
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   2.566880   0.000000   0.000000  2.4356105   8.796611   2.475965
                       VA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   5.042735  2.5895188   0.000000 10.2466895   1.127608  0.9473408
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.8631729   1.764014  1.5764138   1.127608  6.6313859
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.025641  0.8631729   0.000000  5.5174482   3.382824  2.8420225
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   8.068376  2.5895188   4.410036  4.7292413   0.000000  1.8946817
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   4.034188  6.9053834   5.292043  3.1528275   1.127608  1.8946817
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   9.076923 29.3478796  26.460214  0.7882069   6.765648  8.5260676

<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>building heat map data</span>
<span class="pl-k">&gt;</span> head(scaledcounts)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          0  0.8671827   2.447866
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0  2.6015481   1.223933
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          0  3.4687308   2.447866
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          0  0.8671827   4.895733
TR10108<span class="pl-k">|</span>c0_g1_i1_coral          0  1.7343654   4.895733
TR10132<span class="pl-k">|</span>c0_g1_i1_coral          0  6.0702789   1.223933
                       RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   0.000000   3.455170   4.432653
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  13.820681   0.000000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.772786   0.000000   0.000000
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000   8.865306
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   6.910341   0.000000
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  18.863931  51.827555   0.000000
                       RI_B_03_14 RI_B_03_22 RI_B_04_14
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  12.871970   1.272998  12.424473
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000   2.003947
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000   3.818995   4.809473
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   1.287197   3.818995   6.011842
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  14.159167  15.275980   2.805526
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   6.435985   2.545997   0.000000
                       RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  8.1878079   0.000000   4.563408
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  4.0939039   6.765958   0.570426
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  3.6845136   0.000000   0.570426
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  3.2751232   0.000000   5.133834
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  4.9126847  16.914895   5.989473
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  0.8187808   3.382979   1.426065
                       RI_B_05_18 RI_B_05_22 RI_B_06_14
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          0  2.4538305   19.55961
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0  0.0000000    0.00000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          0  2.4538305    0.00000
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          0  4.0897176    0.00000
TR10108<span class="pl-k">|</span>c0_g1_i1_coral          0  0.8179435    6.51987
TR10132<span class="pl-k">|</span>c0_g1_i1_coral          0  1.6358870    0.00000
                       RI_B_06_18 RI_B_06_22 RI_B_07_14
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   3.448566   18.78455  1.4936671
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.069140    0.00000  0.7468335
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   4.827992    0.00000  3.7341677
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   3.448566    0.00000  2.2405006
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  10.345698   40.69986  3.7341677
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   1.379426    0.00000  8.9620025
                       RI_B_07_18 RI_B_07_22 RI_W_01_18
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   8.070182   36.85316  10.815190
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.017545    0.00000  12.257216
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   1.008773    0.00000  12.257216
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   2.017545    0.00000   7.931140
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   1.008773    0.00000   2.884051
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  44.386000   27.63987   1.442025
                       RI_W_01_22 RI_W_02_14 RI_W_02_18
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   4.646396   0.000000  4.9947759
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   1.548799   0.000000  1.9979104
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   7.743994   4.651361  4.9947759
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000  0.9989552
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   1.548799   9.302721  6.9926863
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000  5.9937311
                       RI_W_02_22 RI_W_03_14 RI_W_03_18
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   3.615216   8.525621   5.407467
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   4.262810   1.081493
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   4.820288   3.044865   3.244480
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   3.615216   3.044865   3.244480
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   2.410144   2.435892   4.686471
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  44.587667   1.217946   1.081493
                       RI_W_03_22 RI_W_04_14 RI_W_04_18
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   4.745758   12.82564  2.9768406
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   1.581919    0.00000  3.4729807
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   1.581919    0.00000  7.4421015
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   1.054613    0.00000  0.9922802
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   2.109226   21.37607 12.8996426
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   1.054613   29.92650  7.9382416
                       RI_W_05_14 RI_W_05_18 RI_W_05_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  3.1630643   1.010704   5.526636
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.9037326   8.085630   0.000000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  3.6149306  10.107037   1.381659
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  4.0667969   2.021407   2.763318
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  5.4223959   2.021407   6.908296
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  1.8074653  10.107037  27.633182
                       RI_W_06_14 RI_W_06_18 RI_W_06_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   1.044347   1.336966   5.564881
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000   4.451904
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.133041   1.336966   7.790833
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  10.443469   1.336966   2.225952
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  11.487816   0.000000   5.564881
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   5.221735   1.336966  42.293092
                       RI_W_07_14 RI_W_07_18 RI_W_07_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   4.124636   5.593017   6.900578
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000   0.000000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   2.062318   1.118603   5.520462
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   5.499514  13.423242   1.380116
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   1.118603   5.520462
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   1.374879   2.237207  86.947282
                       VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   2.551157   1.731769   2.845097
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   7.653472   9.091789   1.707058
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  10.204629   5.628251   5.690194
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   3.826736   2.597654   2.845097
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   3.826736   9.091789   2.845097
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   1.275579   1.298827   0.000000
                       VA_B_02_14 VA_B_02_18 VA_B_02_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral          0    14.2516  19.312637
TR10090<span class="pl-k">|</span>c0_g1_i1_coral          0     0.0000   0.000000
TR10090<span class="pl-k">|</span>c0_g2_i1_coral          0     0.0000   0.000000
TR10103<span class="pl-k">|</span>c0_g1_i1_coral          0     0.0000   2.145849
TR10108<span class="pl-k">|</span>c0_g1_i1_coral          0     0.0000   0.000000
TR10132<span class="pl-k">|</span>c0_g1_i1_coral          0     0.0000   8.583394
                       VA_B_03_14 VA_B_03_18 VA_B_03_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  16.029307  14.339493   7.833217
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   5.828839   2.150924   7.833217
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  10.929073   4.301848   3.560553
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   2.185815   4.301848   3.560553
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  14.572098  18.641340  21.363318
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   9.471863  30.829909   8.545327
                       VA_B_04_14 VA_B_04_18 VA_B_04_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   0.000000  0.9006809   6.731616
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  4.5034046   3.624716
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   7.108517  3.1523832   0.000000
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   4.442823  0.0000000   0.000000
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   2.665694  4.9537450   3.624716
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  2.7020428   0.000000
                       VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   2.098865  3.3173156   1.813677
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   4.897352  1.1057719   7.254707
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   1.399244  0.5528859   5.441030
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  3.3173156   0.000000
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   9.095083  5.5288593   0.000000
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  4.9759734   5.441030
                       VA_B_06_14 VA_B_06_18 VA_B_06_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   1.383946   0.000000  5.9098781
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   8.303676   0.000000  0.4924898
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000   4.206352  1.4774695
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   5.535784   6.730163  3.9399187
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  11.763540   4.206352  3.4474289
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  18.683270   2.523811 14.2822054
                       VA_B_07_14 VA_B_07_18 VA_B_07_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   3.728534  0.7405861   2.098499
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  3.7029304   2.098499
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   1.491414 15.5523076   6.295498
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   1.491414  8.1464468   4.196999
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   2.237121  2.2217582   5.246248
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   8.948482 17.0334798   2.098499
                       VA_W_01_14 VA_W_01_18 VA_W_02_14
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   6.699293   7.732656   0.000000
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   3.828167   1.104665   6.978382
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000   3.866328   4.361489
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   3.866328   5.233787
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   1.914084   4.418660   0.000000
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  10.494319   2.616893
                       VA_W_02_18 VA_W_03_14 VA_W_03_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  4.5789684  3.1096023   6.888125
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.6541383  5.4418040   9.392897
TR10090<span class="pl-k">|</span>c0_g2_i1_coral 13.0827667  6.9966052   3.130966
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  0.6541383  6.9966052   2.504773
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  9.1579367  3.1096023   3.757159
TR10132<span class="pl-k">|</span>c0_g1_i1_coral 13.0827667  0.7774006  13.150056
                       VA_W_04_14 VA_W_04_18 VA_W_04_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   1.026752   4.334366   6.410680
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.566880   5.779155   3.945034
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   2.053504   7.705540  10.848842
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   3.593631   3.852770   2.958775
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   6.673887  20.227043   8.383196
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   2.566880   0.000000   0.000000
                       VA_W_05_14 VA_W_05_18 VA_W_05_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  0.0000000   3.848517  12.379823
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000   1.099576  12.379823
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.0000000   2.748941   7.427894
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  0.8118702   3.298729   2.475965
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  1.6237403   5.497882  38.377452
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  2.4356105   8.796611   2.475965
                       VA_W_06_14 VA_W_06_18 VA_W_06_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   5.042735  2.5895188   0.000000
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.8631729   1.764014
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.025641  0.8631729   0.000000
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   8.068376  2.5895188   4.410036
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   4.034188  6.9053834   5.292043
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   9.076923 29.3478796  26.460214
                       VA_W_07_14 VA_W_07_18 VA_W_07_22
TR10083<span class="pl-k">|</span>c0_g2_i1_coral 10.2466895   1.127608  0.9473408
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  1.5764138   1.127608  6.6313859
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  5.5174482   3.382824  2.8420225
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  4.7292413   0.000000  1.8946817
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  3.1528275   1.127608  1.8946817
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  0.7882069   6.765648  8.5260676
<span class="pl-k">&gt;</span> genes4heatmap<span class="pl-k">&lt;</span>-resOrigin[resOrigin<span class="pl-smi">$pvalue</span> <span class="pl-k">&lt;</span>0.05 <span class="pl-k">&amp;</span> <span class="pl-k">!</span>is.na(resOrigin<span class="pl-smi">$pvalue</span>),]
<span class="pl-k">&gt;</span> names(genes4heatmap)
[1] <span class="pl-s"><span class="pl-pds">"</span>baseMean<span class="pl-pds">"</span></span>       <span class="pl-s"><span class="pl-pds">"</span>log2FoldChange<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span>lfcSE<span class="pl-pds">"</span></span>         
[4] <span class="pl-s"><span class="pl-pds">"</span>stat<span class="pl-pds">"</span></span>           <span class="pl-s"><span class="pl-pds">"</span>pvalue<span class="pl-pds">"</span></span>         <span class="pl-s"><span class="pl-pds">"</span>padj<span class="pl-pds">"</span></span>          
<span class="pl-k">&gt;</span> head(genes4heatmap)
log2 fold change (MLE): Origin VA vs RI 
Wald <span class="pl-c1">test</span> p-value: Origin VA vs RI 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange     lfcSE
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>      <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.65756       0.943135  0.474719
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   6.57170       0.727123  0.363907
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    4.14205       4.490291  1.982381
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   2.93642       1.787984  0.559059
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   3.02516       1.758477  0.659498
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   2.32876       1.292223  0.632168
                            stat     pvalue      padj
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>  <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   1.98672 0.04695340 0.4180430
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   1.99810 0.04570594 0.4125186
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    2.26510 0.02350656 0.3086533
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   3.19820 0.00138287 0.0720091
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   2.66639 0.00766715 0.1733954
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   2.04411 0.04094219        NA
<span class="pl-k">&gt;</span> dim(genes4heatmap)
[1] 866   6
<span class="pl-k">&gt;</span> data4heatmap<span class="pl-k">&lt;</span>-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]
<span class="pl-k">&gt;</span> dim(data4heatmap)
[1] 866  78
<span class="pl-k">&gt;</span> head(data4heatmap)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral    0.00000   3.468731   2.447866
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   10.94714   5.203096   8.567532
TR1069<span class="pl-k">|</span>c0_g1_i1_coral     0.00000   0.000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   0.000000   2.447866
TR10773<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   6.070279   2.447866
TR10773<span class="pl-k">|</span>c1_g1_i1_coral    0.00000   2.601548   4.895733
                       RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.772786    0.00000   0.000000
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   0.000000    0.00000   6.648979
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    7.545573    3.45517   6.648979
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   0.000000    0.00000   0.000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   3.772786    0.00000   0.000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000    0.00000   2.216326
                       RI_B_03_14 RI_B_03_22 RI_B_04_14
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000   3.818995   4.809473
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   9.010379   5.091993   2.404737
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000   0.000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   2.574394   0.000000   4.007894
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   1.287197   0.000000   0.000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   5.148788   1.272998   0.000000
                       RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  3.6845136          0   0.570426
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  4.5032943          0   9.126817
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000          0   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  2.8657328          0   1.711278
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.4093904          0   0.000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.0000000          0   0.285213
                       RI_B_05_18 RI_B_05_22 RI_B_06_14
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000  2.4538305          0
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   5.033716  0.8179435          0
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000  0.0000000          0
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.0000000          0
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.0000000          0
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000  0.8179435          0
                       RI_B_06_18 RI_B_06_22 RI_B_07_14
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   4.827992          0   3.734168
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   2.758853          0   5.227835
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000          0   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   0.000000          0   2.987334
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000          0   0.000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000          0   0.000000
                       RI_B_07_18 RI_B_07_22 RI_W_01_18
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   1.008773          0  12.257216
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   5.043864          0   8.652152
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000          0  10.094178
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   0.000000          0   0.000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   1.008773          0   7.931140
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000          0   5.047089
                       RI_W_01_22 RI_W_02_14 RI_W_02_18
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   7.743994   4.651361  4.9947759
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   6.195195  16.279762  2.9968656
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    9.292792   0.000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   1.548799   4.651361  0.9989552
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  17.036786   0.000000 14.9843278
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  18.585585   0.000000 10.9885071
                       RI_W_02_22 RI_W_03_14 RI_W_03_18
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   4.820288   3.044865   3.244480
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   1.205072   7.307675   7.209956
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000   0.000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   2.410144   2.435892   6.128462
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   6.025360  23.140971  12.617422
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   1.205072  10.352540   7.930951
                       RI_W_03_22 RI_W_04_14 RI_W_04_18
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   1.581919    0.00000  7.4421015
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  14.237275   17.10086  5.4575411
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000    0.00000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   4.218452    0.00000  0.4961401
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  29.001856    0.00000  1.9845604
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   7.382291    0.00000  1.4884203
                       RI_W_05_14 RI_W_05_18 RI_W_05_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.614931  10.107037   1.381659
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   4.066797   4.042815   0.000000
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000   0.000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   2.711198   1.010704   0.000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   5.422396   2.021407   0.000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   3.614931   3.032111   0.000000
                       RI_W_06_14 RI_W_06_18 RI_W_06_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.133041   1.336966   7.790833
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   4.177388  13.369662  10.016785
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000   0.000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   3.133041   2.673932   2.225952
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   5.221735   2.673932  10.016785
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   1.044347   2.673932   3.338928
                       RI_W_07_14 RI_W_07_18 RI_W_07_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   2.062318   1.118603   5.520462
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   3.437196  13.423242   8.280694
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000   0.000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   2.749757   2.237207   0.000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   1.374879   2.237207   0.000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   2.062318   3.355810   2.760231
                       VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  10.204629   5.628251   5.690194
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  10.204629   6.494135  10.242350
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   33.165045   9.524732  17.639603
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   1.275579   1.298827   5.121175
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   6.377893   4.762366   2.845097
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   8.929051   6.061193   6.259214
                       VA_B_02_14 VA_B_02_18 VA_B_02_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000     0.0000   0.000000
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   0.000000     0.0000  17.166789
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   86.631891    44.0504  40.771123
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  30.575962     0.0000   2.145849
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   1.698665     0.0000   6.437546
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000     2.5912   0.000000
                       VA_B_03_14 VA_B_03_18 VA_B_03_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  10.929073   4.301848  3.5605530
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  10.929073   7.886721  7.8332166
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000   0.000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   3.643024   7.886721  3.5605530
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   1.433949  0.7121106
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000   3.584873  0.7121106
                       VA_B_04_14 VA_B_04_18 VA_B_04_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  7.1085175   3.152383   0.000000
TR10487<span class="pl-k">|</span>c0_g1_i1_coral 15.1055996   4.053064   8.285065
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000   0.000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  2.6656940   0.000000   8.802882
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000   7.205447   8.285065
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.8885647   8.556469   2.589083
                       VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   1.399244  0.5528859   5.441030
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   4.547541 17.1394640  21.764122
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000  0.0000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   1.049433  1.6586578   3.627354
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   1.399244  1.1057719   0.000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000  4.4230875   0.000000
                       VA_B_06_14 VA_B_06_18 VA_B_06_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000   4.206352   1.477470
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   8.412704   2.462449
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   29.062864  16.825408   8.372327
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   6.919730   0.000000   2.954939
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   2.767892   0.000000   5.417388
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000   2.523811   2.462449
                       VA_B_07_14 VA_B_07_18 VA_B_07_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  1.4914137  15.552308   6.295498
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  3.7285342   5.184103   8.393997
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000   0.000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  6.7113616   0.000000   4.196999
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  1.4914137   2.962344   6.295498
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.7457068   0.000000   4.196999
                       VA_W_01_14 VA_W_01_18 VA_W_02_14
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000   3.866328  4.3614889
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   6.699293   4.970993  1.7445956
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000   0.000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   2.871126   2.209330  0.8722978
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   5.742251   1.104665  0.0000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000   3.866328  0.0000000
                       VA_W_02_18 VA_W_03_14 VA_W_03_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  13.082767  6.9966052   3.130966
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  13.082767  3.1096023   9.392897
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000  0.0000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.7774006   3.757159
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   3.924830  1.5548011   2.504773
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   5.233107  0.7774006   2.504773
                       VA_W_04_14 VA_W_04_18 VA_W_04_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   2.053504   7.705540 10.8488424
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   3.593631  13.003099  1.4793876
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000   0.000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   6.673887   9.150329  7.3969380
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   1.026752   0.000000  0.4931292
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   1.026752   0.000000  0.0000000
                       VA_W_05_14 VA_W_05_18 VA_W_05_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral    0.00000   2.748941   7.427894
TR10487<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   8.246822   8.665876
TR1069<span class="pl-k">|</span>c0_g1_i1_coral     0.00000   0.000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   14.61366   2.748941   9.903859
TR10773<span class="pl-k">|</span>c0_g1_i1_coral    0.00000   0.000000   0.000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral    0.00000   2.199153   1.237982
                       VA_W_06_14 VA_W_06_18 VA_W_06_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.025641  0.8631729   0.000000
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  10.085470  1.7263459  11.466093
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000  0.0000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   2.017094  6.0422105   4.410036
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  1.7263459   0.000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   2.017094  0.8631729   5.292043
                       VA_W_07_14 VA_W_07_18 VA_W_07_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   5.517448   3.382824   2.842023
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   7.093862   6.765648   3.789363
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000   0.000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   3.382824   1.894682
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000   0.000000   0.000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000   0.000000   0.000000
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> temp = as.matrix(rowMeans(data4heatmap))
<span class="pl-k">&gt;</span> head(temp)
                           [,1]
TR10090<span class="pl-k">|</span>c0_g2_i1_coral 3.657565
TR10487<span class="pl-k">|</span>c0_g1_i1_coral 6.571698
TR1069<span class="pl-k">|</span>c0_g1_i1_coral  4.142052
TR10748<span class="pl-k">|</span>c0_g1_i1_coral 2.936423
TR10773<span class="pl-k">|</span>c0_g1_i1_coral 3.025158
TR10773<span class="pl-k">|</span>c1_g1_i1_coral 2.328759
<span class="pl-k">&gt;</span> scaledmatrix<span class="pl-k">&lt;</span>-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)
<span class="pl-k">&gt;</span> data4heatmapscaled = data4heatmap/scaledmatrix
<span class="pl-k">&gt;</span> head(data4heatmapscaled)
                       RI_B_01_14 RI_B_01_18 RI_B_01_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000  0.9483717  0.6692613
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   1.665801  0.7917431  1.3037015
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000  0.0000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.0000000  0.8336218
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  2.0065992  0.8091699
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000  1.1171391  2.1022922
                       RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   1.031502  0.0000000  0.0000000
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.0000000  1.0117598
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    1.821699  0.8341687  1.6052378
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.0000000  0.0000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   1.247137  0.0000000  0.0000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000  0.0000000  0.9517198
                       RI_B_03_14 RI_B_03_22 RI_B_04_14
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.0000000  1.0441360  1.3149387
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  1.3710885  0.7748368  0.3659232
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000  0.0000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.8767109  0.0000000  1.3648899
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.4254975  0.0000000  0.0000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  2.2109576  0.5466423  0.0000000
                       RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  1.0073680          0  0.1559579
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.6852559          0  1.3888066
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000          0  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.9759263          0  0.5827764
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.1353286          0  0.0000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.0000000          0  0.1224742
                       RI_B_05_18 RI_B_05_22 RI_B_06_14
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.0000000  0.6708919          0
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.7659688  0.1244646          0
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000  0.0000000          0
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  0.0000000          0
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  0.0000000          0
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.0000000  0.3512357          0
                       RI_B_06_18 RI_B_06_22 RI_B_07_14
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  1.3200019          0  1.0209437
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.4198082          0  0.7955075
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000          0  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000          0  1.0173377
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000          0  0.0000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.0000000          0  0.0000000
                       RI_B_07_18 RI_B_07_22 RI_W_01_18
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.2758045          0   3.351196
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.7675130          0   1.316578
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000          0   2.436999
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000          0   0.000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.3334612          0   2.621728
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.0000000          0   2.167287
                       RI_W_01_22 RI_W_02_14 RI_W_02_18
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  2.1172541   1.271710  1.3656016
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.9427084   2.477254  0.4560261
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   2.2435236   0.000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.5274440   1.584023  0.3401946
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  5.6317019   0.000000  4.9532387
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  7.9808953   0.000000  4.7186100
                       RI_W_02_22 RI_W_03_14 RI_W_03_18
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  1.3178956  0.8324842  0.8870602
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.1833730  1.1119920  1.0971223
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000  0.0000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.8207755  0.8295438  2.0870501
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  1.9917509  7.6495091  4.1708314
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.5174739  4.4455171  3.4056552
                       RI_W_03_22 RI_W_04_14 RI_W_04_18
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.4325062   0.000000  2.0347150
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  2.1664531   2.602198  0.8304614
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000   0.000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  1.4365953   0.000000  0.1689607
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  9.5868908   0.000000  0.6560188
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  3.1700530   0.000000  0.6391473
                       RI_W_05_14 RI_W_05_18 RI_W_05_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.9883436  2.7633244  0.3777539
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.6188351  0.6151858  0.0000000
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000  0.0000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.9232995  0.3441955  0.0000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  1.7924342  0.6681991  0.0000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  1.5522989  1.3020286  0.0000000
                       RI_W_06_14 RI_W_06_18 RI_W_06_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.8565921  0.3655346  2.1300602
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.6356634  2.0344306  1.5242310
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000  0.0000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  1.0669582  0.9106087  0.7580488
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  1.7261033  0.8838986  3.3111614
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.4484564  1.1482219  1.4337799
                       RI_W_07_14 RI_W_07_18 RI_W_07_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.5638500  0.3058329   1.509327
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.5230302  2.0425836   1.260054
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000  0.0000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.9364308  0.7618816   0.000000
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.4544816  0.7395340   0.000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.8855865  1.4410293   1.185280
                       VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  2.7900066  1.5387973   1.555733
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  1.5528148  0.9881975   1.558555
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   8.0069110  2.2995199   4.258662
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.4343988  0.4423160   1.744018
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  2.1082846  1.5742538   0.940479
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  3.8342522  2.6027562   2.687789
                       VA_B_02_14 VA_B_02_18 VA_B_02_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.0000000   0.000000  0.0000000
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000   0.000000  2.6122305
TR1069<span class="pl-k">|</span>c0_g1_i1_coral  20.9152088  10.634922  9.8432176
TR10748<span class="pl-k">|</span>c0_g1_i1_coral 10.4126549   0.000000  0.7307695
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.5615127   0.000000  2.1280034
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.0000000   1.112696  0.0000000
                       VA_B_03_14 VA_B_03_18 VA_B_03_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   2.988074  1.1761509  0.9734764
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   1.663052  1.2001041  1.1919624
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000  0.0000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   1.240633  2.6858257  1.2125476
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.4740081  0.2353962
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000  1.5393918  0.3057897
                       VA_B_04_14 VA_B_04_18 VA_B_04_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  1.9435111  0.8618804   0.000000
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  2.2985841  0.6167454   1.260719
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000  0.0000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.9078031  0.0000000   2.997825
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  2.3818420   2.738722
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.3815614  3.6742606   1.111786
                       VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.3825615  0.1511623   1.487610
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.6919888  2.6080725   3.311796
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000  0.0000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.3573847  0.5648565   1.235297
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.4625357  0.3655254   0.000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.0000000  1.8993321   0.000000
                       VA_B_06_14 VA_B_06_18 VA_B_06_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.0000000   1.150042  0.4039490
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000   1.280142  0.3747052
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   7.0165371   4.062094  2.0212992
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  2.3565165   0.000000  1.0063056
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.9149579   0.000000  1.7907788
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.0000000   1.083758  1.0574082
                       VA_B_07_14 VA_B_07_18 VA_B_07_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.4077614  4.2520938   1.721227
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.5673624  0.7888529   1.277295
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000  0.0000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  2.2855567  0.0000000   1.429289
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.4930036  0.9792364   2.081048
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.3202164  0.0000000   1.802247
                       VA_W_01_14 VA_W_01_18 VA_W_02_14
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.0000000  1.0570771  1.1924571
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  1.0194159  0.7564245  0.2654711
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000  0.0000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.9777629  0.7523882  0.2970613
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  1.8981660  0.3651595  0.0000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.0000000  1.6602522  0.0000000
                       VA_W_02_18 VA_W_03_14 VA_W_03_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.576907  1.9129136  0.8560247
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   1.990774  0.4731810  1.4292954
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000  0.0000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.2647441  1.2795019
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   1.297397  0.5139571  0.8279809
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   2.247165  0.3338261  1.0755824
                       VA_W_04_14 VA_W_04_18 VA_W_04_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.5614402   2.106741  2.9661383
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.5468346   1.978651  0.2251150
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000   0.000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  2.2727947   3.116148  2.5190299
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.3394044   0.000000  0.1630094
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.4409008   0.000000  0.0000000
                       VA_W_05_14 VA_W_05_18 VA_W_05_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   0.000000  0.7515768   2.030831
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  1.2548999   1.318666
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000  0.0000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   4.976688  0.9361528   3.372763
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.0000000   0.000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000  0.9443452   0.531606
                       VA_W_06_14 VA_W_06_18 VA_W_06_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.8272283  0.2359966   0.000000
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  1.5346826  0.2626940   1.744769
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   0.0000000  0.0000000   0.000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.6869221  2.0576770   1.501839
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.0000000  0.5706631   0.000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.8661668  0.3706578   2.272473
                       VA_W_07_14 VA_W_07_18 VA_W_07_22
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   1.508503  0.9248843  0.7770259
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   1.079457  1.0295131  0.5766186
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    0.000000  0.0000000  0.0000000
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  1.1520220  0.6452345
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   0.000000  0.0000000  0.0000000
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   0.000000  0.0000000  0.0000000
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> dim(data4heatmapscaled)
[1] 866  78
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> pairs.breaks <span class="pl-k">&lt;</span>- seq(0, 3.0, by=0.1)
<span class="pl-k">&gt;</span> length(pairs.breaks)
[1] 31
<span class="pl-k">&gt;</span> mycol <span class="pl-k">&lt;</span>- colorpanel(n=30, low=<span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, high=<span class="pl-s"><span class="pl-pds">"</span>yellow<span class="pl-pds">"</span></span>) 
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> pdf(file=<span class="pl-s"><span class="pl-pds">"</span>XXXXXX_byrow.pdf<span class="pl-pds">"</span></span>,7,7)
<span class="pl-k">&gt;</span> heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(<span class="pl-s"><span class="pl-pds">"</span>row<span class="pl-pds">"</span></span>), scale=<span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, keysize=1, breaks=pairs.breaks, col=mycol, trace = <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, symkey = F, density.info = <span class="pl-s"><span class="pl-pds">"</span>density<span class="pl-pds">"</span></span>, colsep=c(24), sepcolor=c(<span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
<span class="pl-k">&gt;</span> <span class="pl-en">dev.off</span>()
RStudioGD 
        2 
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> pdf(file=<span class="pl-s"><span class="pl-pds">"</span>XXXXXX_bycolumn.pdf<span class="pl-pds">"</span></span>,7,7)
<span class="pl-k">&gt;</span> heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(<span class="pl-s"><span class="pl-pds">"</span>col<span class="pl-pds">"</span></span>), scale=<span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, keysize=1, breaks=pairs.breaks, col=mycol, trace = <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, symkey = F, density.info = <span class="pl-s"><span class="pl-pds">"</span>density<span class="pl-pds">"</span></span>, colsep=c(24), sepcolor=c(<span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
<span class="pl-k">&gt;</span> <span class="pl-en">dev.off</span>()
RStudioGD 
        2 
<span class="pl-k">&gt;</span> heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(<span class="pl-s"><span class="pl-pds">"</span>row<span class="pl-pds">"</span></span>), scale=<span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, keysize=1, breaks=pairs.breaks, col=mycol, trace = <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, symkey = F, density.info = <span class="pl-s"><span class="pl-pds">"</span>density<span class="pl-pds">"</span></span>, colsep=c(24), sepcolor=c(<span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
Error <span class="pl-k">in</span> <span class="pl-en">plot.new</span>() <span class="pl-c1">:</span> figure margins too large
<span class="pl-k">&gt;</span> heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(<span class="pl-s"><span class="pl-pds">"</span>col<span class="pl-pds">"</span></span>), scale=<span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, keysize=1, breaks=pairs.breaks, col=mycol, trace = <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, symkey = F, density.info = <span class="pl-s"><span class="pl-pds">"</span>density<span class="pl-pds">"</span></span>, colsep=c(24), sepcolor=c(<span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
Error <span class="pl-k">in</span> .External.graphics(C_layout, num.rows, num.cols, mat, as.integer(num.figures),  <span class="pl-c1">:</span> 
  invalid graphics state
<span class="pl-k">&gt;</span> plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin_SymState_Temp<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> pdf(file=<span class="pl-s"><span class="pl-pds">"</span>866_RIvsVA_PCAv2.pdf<span class="pl-pds">"</span></span>,14, 14)
<span class="pl-k">&gt;</span> plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=<span class="pl-s"><span class="pl-pds">"</span>Origin<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> <span class="pl-en">dev.off</span>()
RStudioGD 
        2 
<span class="pl-k">&gt;</span> prop.nullv3 <span class="pl-k">&lt;</span>- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100<span class="pl-k">*</span>mean(x==0))
<span class="pl-k">&gt;</span> barplot(prop.nullv3, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)
<span class="pl-k">&gt;</span> pdf(file=<span class="pl-s"><span class="pl-pds">"</span>ResNullCountsv2.pdf<span class="pl-pds">"</span></span>,14, 14)
<span class="pl-k">&gt;</span> barplot(prop.nullv3, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)
<span class="pl-k">&gt;</span> <span class="pl-en">dev.off</span>()
RStudioGD 
        2 
<span class="pl-k">&gt;</span> barplot(prop.nullv3, main=<span class="pl-s"><span class="pl-pds">"</span>Percentage of null counts per sample<span class="pl-pds">"</span></span>, 
+         horiz=TRUE, cex.names=0.5, las=1, 
+         col=conds<span class="pl-smi">$Origin_SymState_Temp</span>, ylab=<span class="pl-s"><span class="pl-pds">'</span>Samples<span class="pl-pds">'</span></span>, xlab=<span class="pl-s"><span class="pl-pds">'</span>% of null counts<span class="pl-pds">'</span></span>)
<span class="pl-k">&gt;</span> distsv<span class="pl-k">2&lt;</span>-dist(t(assay(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),])))
<span class="pl-k">&gt;</span> plot(hclust(distsv2))
<span class="pl-k">&gt;</span> heatmap.2(as.matrix(distsv2), key=F, trace=<span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>,
+           col=colorpanel(100, <span class="pl-s"><span class="pl-pds">"</span>black<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>),
+           margin=c(10, 10))
<span class="pl-k">&gt;</span> <span class="pl-c"><span class="pl-c">#</span>### GOSeq analysis ####</span>
<span class="pl-k">&gt;</span> res<span class="pl-k">&lt;</span>-resOrigin
<span class="pl-k">&gt;</span> Annotations<span class="pl-k">&lt;</span>-read.delim(<span class="pl-s"><span class="pl-pds">"</span>Assembly_GoodCoralSymbiont_suffixed_totalannotated.txt<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> head(res)
log2 fold change (MLE): Origin VA vs RI 
Wald <span class="pl-c1">test</span> p-value: Origin VA vs RI 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>      <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10083<span class="pl-k">|</span>c0_g2_i1_coral   5.36648     -0.2712427
TR10090<span class="pl-k">|</span>c0_g1_i1_coral   2.70436      1.1125117
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.65756      0.9431346
TR10103<span class="pl-k">|</span>c0_g1_i1_coral   2.91279      0.0386643
TR10108<span class="pl-k">|</span>c0_g1_i1_coral   6.31144     -0.0179010
TR10132<span class="pl-k">|</span>c0_g1_i1_coral   9.26737     -0.2649903
                           lfcSE       stat    pvalue
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>  <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  0.446225 -0.6078616 0.5432793
TR10090<span class="pl-k">|</span>c0_g1_i1_coral  0.589306  1.8878328 0.0590484
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.474719  1.9867203 0.0469534
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  0.456720  0.0846565 0.9325345
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  0.461958 -0.0387502 0.9690895
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  0.677608 -0.3910675 0.6957473
                            padj
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10083<span class="pl-k">|</span>c0_g2_i1_coral  0.865661
TR10090<span class="pl-k">|</span>c0_g1_i1_coral        NA
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.418043
TR10103<span class="pl-k">|</span>c0_g1_i1_coral  0.984590
TR10108<span class="pl-k">|</span>c0_g1_i1_coral  0.992150
TR10132<span class="pl-k">|</span>c0_g1_i1_coral  0.921517
<span class="pl-k">&gt;</span> sum(res<span class="pl-smi">$log2FoldChange</span><span class="pl-k">!</span>=0)
[1] 7659
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> genes<span class="pl-k">&lt;</span>-as.integer(p.adjust(res<span class="pl-smi">$pvalue</span>[res<span class="pl-smi">$log2FoldChange</span><span class="pl-k">!=</span>0 <span class="pl-k">&amp;</span> <span class="pl-k">!</span>is.na(res<span class="pl-smi">$pvalue</span>)], method=<span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>)<span class="pl-k">&lt;</span>.05)
<span class="pl-k">&gt;</span> names(genes)<span class="pl-k">&lt;</span>-row.names(res[res<span class="pl-smi">$log2FoldChange</span><span class="pl-k">!=</span>0 <span class="pl-k">&amp;</span> <span class="pl-k">!</span>is.na(res<span class="pl-smi">$pvalue</span>),])
<span class="pl-k">&gt;</span> table(genes)
genes
   0    1 
7533  116 
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> genes<span class="pl-k">&lt;</span>-as.integer(res<span class="pl-smi">$pvalue</span>[res<span class="pl-smi">$log2FoldChange</span><span class="pl-k">!=</span>0 <span class="pl-k">&amp;</span> <span class="pl-k">!</span>is.na(res<span class="pl-smi">$pvalue</span>)]<span class="pl-k">&lt;</span>.05)
<span class="pl-k">&gt;</span> names(genes)<span class="pl-k">&lt;</span>-row.names(res[res<span class="pl-smi">$log2FoldChange</span><span class="pl-k">!=</span>0 <span class="pl-k">&amp;</span> <span class="pl-k">!</span>is.na(res<span class="pl-smi">$pvalue</span>),])
<span class="pl-k">&gt;</span> table(genes)
genes
   0    1 
6783  866 
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> goterms<span class="pl-k">&lt;</span>-strsplit(as.character(Annotations<span class="pl-smi">$GO</span>), split=<span class="pl-s"><span class="pl-pds">"</span> // <span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> names(goterms)<span class="pl-k">&lt;</span>-Annotations<span class="pl-smi">$ContigName</span>
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> pwf<span class="pl-k">&lt;</span>-nullp(genes, bias.data=Annotations[Annotations<span class="pl-smi">$ContigName</span>%in%names(genes),<span class="pl-s"><span class="pl-pds">"</span>ContigLength<span class="pl-pds">"</span></span>])
Warning message:
In pcls(G) <span class="pl-c1">:</span> initial point very close to some inequality constraints
<span class="pl-k">&gt;</span> GOGOGO<span class="pl-k">&lt;</span>-goseq(pwf,gene2cat=goterms)
Using manually entered categories.
Calculating the p-values...
<span class="pl-s"><span class="pl-pds">'</span>select()<span class="pl-pds">'</span></span> returned 1:1 mapping between keys
and columns
<span class="pl-k">&gt;</span> GOGOGO<span class="pl-smi">$padj</span><span class="pl-k">&lt;</span>-p.adjust(GOGOGO<span class="pl-smi">$over_represented_pvalue</span>, method=<span class="pl-s"><span class="pl-pds">"</span>fdr<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> sum(GOGOGO<span class="pl-smi">$padj</span><span class="pl-k">&lt;</span>0.2, na.rm=T)
[1] 3
<span class="pl-k">&gt;</span> sum(GOGOGO<span class="pl-smi">$padj</span><span class="pl-k">&lt;</span>0.1, na.rm=T)
[1] 0
<span class="pl-k">&gt;</span> sum(GOGOGO<span class="pl-smi">$padj</span><span class="pl-k">&lt;</span>0.05, na.rm=T)
[1] 0
<span class="pl-k">&gt;</span> GOGOGO[GOGOGO<span class="pl-smi">$padj</span><span class="pl-k">&lt;</span>0.05,]
[1] category                 over_represented_pvalue 
[3] under_represented_pvalue numDEInCat              
[5] numInCat                 term                    
[7] ontology                 padj                    
<span class="pl-k">&lt;</span>0 rows<span class="pl-k">&gt;</span> (or 0-length row.names)
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> GOTERM[[<span class="pl-s"><span class="pl-pds">"</span>SOMEGOTERM<span class="pl-pds">"</span></span>]]
NULL
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> EnGOS<span class="pl-k">&lt;</span>-GOGOGO<span class="pl-smi">$category</span>[p.adjust(GOGOGO<span class="pl-smi">$over_represented_pvalue</span>, method=<span class="pl-s"><span class="pl-pds">"</span>BH<span class="pl-pds">"</span></span>)<span class="pl-k">&lt;</span>.05]
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> for(go <span class="pl-k">in</span> EnGOS){
+   print(GOTERM[[go]])
+   cat(<span class="pl-s"><span class="pl-pds">"</span>--------------------------------------\n<span class="pl-pds">"</span></span>)}
<span class="pl-k">&gt;</span> names(Annotations)
 [1] <span class="pl-s"><span class="pl-pds">"</span>ContigName<span class="pl-pds">"</span></span>             
 [2] <span class="pl-s"><span class="pl-pds">"</span>ContigLength<span class="pl-pds">"</span></span>           
 [3] <span class="pl-s"><span class="pl-pds">"</span>topnrMatch<span class="pl-pds">"</span></span>             
 [4] <span class="pl-s"><span class="pl-pds">"</span>topnrEvalue<span class="pl-pds">"</span></span>            
 [5] <span class="pl-s"><span class="pl-pds">"</span>topnrAlignmentLength<span class="pl-pds">"</span></span>   
 [6] <span class="pl-s"><span class="pl-pds">"</span>topnrPercMatch<span class="pl-pds">"</span></span>         
 [7] <span class="pl-s"><span class="pl-pds">"</span>nobadwordnrMatch<span class="pl-pds">"</span></span>       
 [8] <span class="pl-s"><span class="pl-pds">"</span>nobadwordnrEvalue<span class="pl-pds">"</span></span>      
 [9] <span class="pl-s"><span class="pl-pds">"</span>nobadwordnrAlignLength<span class="pl-pds">"</span></span> 
[10] <span class="pl-s"><span class="pl-pds">"</span>nobadwordnrPercentMatch<span class="pl-pds">"</span></span>
[11] <span class="pl-s"><span class="pl-pds">"</span>Uniprotmatch<span class="pl-pds">"</span></span>           
[12] <span class="pl-s"><span class="pl-pds">"</span>X._identity<span class="pl-pds">"</span></span>            
[13] <span class="pl-s"><span class="pl-pds">"</span>evalue<span class="pl-pds">"</span></span>                 
[14] <span class="pl-s"><span class="pl-pds">"</span>ID<span class="pl-pds">"</span></span>                     
[15] <span class="pl-s"><span class="pl-pds">"</span>Description<span class="pl-pds">"</span></span>            
[16] <span class="pl-s"><span class="pl-pds">"</span>KEGG<span class="pl-pds">"</span></span>                   
[17] <span class="pl-s"><span class="pl-pds">"</span>KEGGKO<span class="pl-pds">"</span></span>                 
[18] <span class="pl-s"><span class="pl-pds">"</span>PFAM<span class="pl-pds">"</span></span>                   
[19] <span class="pl-s"><span class="pl-pds">"</span>GO<span class="pl-pds">"</span></span>                     
[20] <span class="pl-s"><span class="pl-pds">"</span>GOCC<span class="pl-pds">"</span></span>                   
[21] <span class="pl-s"><span class="pl-pds">"</span>GOBP<span class="pl-pds">"</span></span>                   
[22] <span class="pl-s"><span class="pl-pds">"</span>GOMF<span class="pl-pds">"</span></span>                   
[23] <span class="pl-s"><span class="pl-pds">"</span>Keywords<span class="pl-pds">"</span></span>               
<span class="pl-k">&gt;</span> head(genes4heatmap)
log2 fold change (MLE): Origin VA vs RI 
Wald <span class="pl-c1">test</span> p-value: Origin VA vs RI 
DataFrame with 6 rows and 6 columns
                        baseMean log2FoldChange
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>      <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.65756       0.943135
TR10487<span class="pl-k">|</span>c0_g1_i1_coral   6.57170       0.727123
TR1069<span class="pl-k">|</span>c0_g1_i1_coral    4.14205       4.490291
TR10748<span class="pl-k">|</span>c0_g1_i1_coral   2.93642       1.787984
TR10773<span class="pl-k">|</span>c0_g1_i1_coral   3.02516       1.758477
TR10773<span class="pl-k">|</span>c1_g1_i1_coral   2.32876       1.292223
                           lfcSE      stat     pvalue
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>  <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10090<span class="pl-k">|</span>c0_g2_i1_coral  0.474719   1.98672 0.04695340
TR10487<span class="pl-k">|</span>c0_g1_i1_coral  0.363907   1.99810 0.04570594
TR1069<span class="pl-k">|</span>c0_g1_i1_coral   1.982381   2.26510 0.02350656
TR10748<span class="pl-k">|</span>c0_g1_i1_coral  0.559059   3.19820 0.00138287
TR10773<span class="pl-k">|</span>c0_g1_i1_coral  0.659498   2.66639 0.00766715
TR10773<span class="pl-k">|</span>c1_g1_i1_coral  0.632168   2.04411 0.04094219
                            padj
                       <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
TR10090<span class="pl-k">|</span>c0_g2_i1_coral 0.4180430
TR10487<span class="pl-k">|</span>c0_g1_i1_coral 0.4125186
TR1069<span class="pl-k">|</span>c0_g1_i1_coral  0.3086533
TR10748<span class="pl-k">|</span>c0_g1_i1_coral 0.0720091
TR10773<span class="pl-k">|</span>c0_g1_i1_coral 0.1733954
TR10773<span class="pl-k">|</span>c1_g1_i1_coral        NA
<span class="pl-k">&gt;</span> dim(genes4heatmap)
[1] 866   6
<span class="pl-k">&gt;</span> 
<span class="pl-k">&gt;</span> ResultsOutput<span class="pl-k">&lt;</span>-cbind(<span class="pl-s"><span class="pl-pds">"</span>ContigName<span class="pl-pds">"</span></span>=row.names(genes4heatmap),genes4heatmap, Annotations[Annotations<span class="pl-smi">$ContigName</span>%in%row.names(genes4heatmap),])
<span class="pl-k">&gt;</span> dim(ResultsOutput)
[1] 866  30
<span class="pl-k">&gt;</span> head(ResultsOutput)
DataFrame with 6 rows and 30 columns
              ContigName  baseMean log2FoldChange
             <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>      <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
1 TR10090<span class="pl-k">|</span>c0_g2_i1_coral   3.65756       0.943135
2 TR10487<span class="pl-k">|</span>c0_g1_i1_coral   6.57170       0.727123
3  TR1069<span class="pl-k">|</span>c0_g1_i1_coral   4.14205       4.490291
4 TR10748<span class="pl-k">|</span>c0_g1_i1_coral   2.93642       1.787984
5 TR10773<span class="pl-k">|</span>c0_g1_i1_coral   3.02516       1.758477
6 TR10773<span class="pl-k">|</span>c1_g1_i1_coral   2.32876       1.292223
      lfcSE      stat     pvalue      padj
  <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>  <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>numeric<span class="pl-k">&gt;</span>
1  0.474719   1.98672 0.04695340 0.4180430
2  0.363907   1.99810 0.04570594 0.4125186
3  1.982381   2.26510 0.02350656 0.3086533
4  0.559059   3.19820 0.00138287 0.0720091
5  0.659498   2.66639 0.00766715 0.1733954
6  0.632168   2.04411 0.04094219        NA
              ContigName ContigLength
             <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>    <span class="pl-k">&lt;</span>integer<span class="pl-k">&gt;</span>
1 TR10090<span class="pl-k">|</span>c0_g2_i1_coral          772
2 TR10487<span class="pl-k">|</span>c0_g1_i1_coral         1007
3  TR1069<span class="pl-k">|</span>c0_g1_i1_coral          815
4 TR10748<span class="pl-k">|</span>c0_g1_i1_coral          516
5 TR10773<span class="pl-k">|</span>c0_g1_i1_coral          580
6 TR10773<span class="pl-k">|</span>c1_g1_i1_coral          586
              topnrMatch  topnrEvalue
             <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>  <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>
1 gnl<span class="pl-k">|</span>BL_ORD_ID<span class="pl-k">|</span>738431..  7.27758e-78
2 gnl<span class="pl-k">|</span>BL_ORD_ID<span class="pl-k">|</span>110398..  3.65321e-84
3 gnl<span class="pl-k">|</span>BL_ORD_ID<span class="pl-k">|</span>110400.. 9.23041e-133
4 gnl<span class="pl-k">|</span>BL_ORD_ID<span class="pl-k">|</span>110376..  2.29747e-78
5 gnl<span class="pl-k">|</span>BL_ORD_ID<span class="pl-k">|</span>110383..  2.14754e-64
6 gnl<span class="pl-k">|</span>BL_ORD_ID<span class="pl-k">|</span>110383..  1.33696e-65
  topnrAlignmentLength topnrPercMatch
           <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>    <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>
1                  258           0.53
2                  201           0.88
3                  271           0.73
4                  172           0.85
5                  206           0.55
6                  196           0.60
        nobadwordnrMatch nobadwordnrEvalue
             <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>       <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>
1 gnl<span class="pl-k">|</span>BL_ORD_ID<span class="pl-k">|</span>726506..       3.99079e-52
2 gnl<span class="pl-k">|</span>BL_ORD_ID<span class="pl-k">|</span>110398..       3.65321e-84
3 gnl<span class="pl-k">|</span>BL_ORD_ID<span class="pl-k">|</span>110400..      9.23041e-133
4 gnl<span class="pl-k">|</span>BL_ORD_ID<span class="pl-k">|</span>110376..       2.29747e-78
5 gnl<span class="pl-k">|</span>BL_ORD_ID<span class="pl-k">|</span>110383..       2.14754e-64
6 gnl<span class="pl-k">|</span>BL_ORD_ID<span class="pl-k">|</span>110383..       1.33696e-65
  nobadwordnrAlignLength nobadwordnrPercentMatch
             <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>             <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>
1                    264                    0.45
2                    201                    0.88
3                    271                    0.73
4                    172                    0.85
5                    206                    0.55
6                    196                    0.60
  Uniprotmatch X._identity      evalue
   <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>
1   A0A1X7UWJ8      40.927    1.82e-51
2       Q4R6F3      37.500    1.77e-06
3       Q8N2E2      31.022    1.60e-30
4   A0A2B4RJ64      70.690    9.57e-61
5   A0A2B4RML7      46.635    3.49e-44
6   A0A2B4RML7      47.847    2.78e-48
                ID            Description
       <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>            <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>
1 A0A1X7UWJ8_AMPQE Uncharacterized prot..
2      NGLY1_MACFA Peptide-N(4)-(N-acet..
3       VWDE_HUMAN von Willebrand facto..
4 A0A2B4RJ64_STYPI Uncharacterized prot..
5 A0A2B4RML7_STYPI Uncharacterized prot..
6 A0A2B4RML7_STYPI Uncharacterized prot..
           KEGG      KEGGKO                   PFAM
    <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span> <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>            <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>
1       No_KEGG   No_KEGGKO       PF05585<span class="pl-k">;</span> DUF1758
2 mcf:101925902      K01456 PF04721<span class="pl-k">;</span> PAW,  PF094..
3    hsa:221806   No_KEGGKO           PF00094<span class="pl-k">;</span> VWD
4       No_KEGG   No_KEGGKO                   Pfam
5       No_KEGG   No_KEGGKO                   Pfam
6       No_KEGG   No_KEGGKO                   Pfam
                      GO                   GOCC
             <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>            <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>
1             GO:0004190                No_GOBP
2 GO:0005737 // GO:004.. glycoprotein catabol..
3 GO:0009986 // GO:000.. anatomical structure..
4             GO:0016021                No_GOBP
5             No_GOcodes                No_GOBP
6             No_GOcodes                No_GOBP
                    GOBP                   GOMF
             <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>            <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>
1 aspartic-type endope..                No_GOCC
2 metal ion binding (G.. cytoplasm (GO:000573..
3 signaling receptor b.. cell surface (GO:000..
4                No_GOMF integral component o..
5                No_GOMF                No_GOCC
6                No_GOMF                No_GOCC
                Keywords
             <span class="pl-k">&lt;</span>character<span class="pl-k">&gt;</span>
1 Complete proteome {E..
2 Acetylation<span class="pl-k">;</span>Cytoplas..
3 Alternative splicing..
4 Coiled coil {ECO:000..
5 Complete proteome {E..
6 Complete proteome {E..
<span class="pl-k">&gt;</span> write.table(ResultsOutput, file=<span class="pl-s"><span class="pl-pds">"</span>177OriginDEGswAnnotation.txt<span class="pl-pds">"</span></span>, quote=F, row.names=F, sep=<span class="pl-s"><span class="pl-pds">"</span>\t<span class="pl-pds">"</span></span>)
<span class="pl-k">&gt;</span> 



<span class="pl-c"><span class="pl-c">#</span>9- Optional, also plot a PCA using only the genes with a pvalue&lt;0.05 for the "origin" and "symstate" variables</span>

<span class="pl-c"><span class="pl-c">#</span>10- Even more optional, color code your symbols by temperature treatment as well as origin and symstate.</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day13HW
                     </h2>
                     <a aria-label="Permalink: Day13HW" class="anchor" href="#day13hw" id="user-content-day13hw">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='#1- create an sbatch script to rm -r your sandbox/YOURNAME directory

[kcrid001@coreV3-23-046 sandboxes]$ cat SandboxRemovalKCrid.sh 
#!/bin/bash -l

#SBATCH -o KCridsandboxremoval.txt
#SBATCH -n 1
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KCridsandboxremoval

rm -r ./katiecrider
[kcrid001@coreV3-23-046 sandboxes]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes


#2- submit your sbatch script and add the entry to your logfile/readme

#3- scp your github logfile to the /cm/shared/courses/dbarshis/21AdvGenomics/archive with the name LASTNAME_FIRSTNAME_readme.md


scp README.md kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/archive/Crider_Katie_readme.md



#4- scp the following files to your local computer
	scp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta ./
	scp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta.fai ./
	scp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam ./
	scp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam.bai ./
#5- launch IGV on your local machine and load in 15079_Apoc_hostsym.fasta as the "genome" and RI_B_01.bam as the input file' dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>1- create an sbatch script to rm -r your sandbox/YOURNAME directory</span>

[kcrid001@coreV3-23-046 sandboxes]$ cat SandboxRemovalKCrid.sh 
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KCridsandboxremoval.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KCridsandboxremoval</span>

rm -r ./katiecrider
[kcrid001@coreV3-23-046 sandboxes]$ <span class="pl-c1">pwd</span>
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes


<span class="pl-c"><span class="pl-c">#</span>2- submit your sbatch script and add the entry to your logfile/readme</span>

<span class="pl-c"><span class="pl-c">#</span>3- scp your github logfile to the /cm/shared/courses/dbarshis/21AdvGenomics/archive with the name LASTNAME_FIRSTNAME_readme.md</span>


scp README.md kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/archive/Crider_Katie_readme.md



<span class="pl-c"><span class="pl-c">#</span>4- scp the following files to your local computer</span>
	scp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta ./
	scp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta.fai ./
	scp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam ./
	scp YOURMIDASID@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam.bai ./
<span class="pl-c"><span class="pl-c">#</span>5- launch IGV on your local machine and load in 15079_Apoc_hostsym.fasta as the "genome" and RI_B_01.bam as the input file</span></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Day13
                     </h2>
                     <a aria-label="Permalink: Day13" class="anchor" href="#day13" id="user-content-day13">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="#1- create an sbatch script to rm -r your sandbox/YOURNAME directory
#2- submit your sbatch script and add the entry to your logfile/readme

[kcrid001@turing1 sandboxes]$ cat SandboxRemovalKCrid.sh
#!/bin/bash -l

#SBATCH -o KCridsandboxremoval.txt
#SBATCH -n 1
#SBATCH --mail-user=kcrid001@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=KCridsandboxremoval

rm -r ./katiecrider
[kcrid001@turing1 sandboxes]$ salloc
salloc: Pending job allocation 9284556
salloc: job 9284556 queued and waiting for resources
salloc: job 9284556 has been allocated resources
salloc: Granted job allocation 9284556
[kcrid001@coreV1-22-016 sandboxes]$ sbatch SandboxRemovalKCrid.sh
Submitted batch job 9284557
[kcrid001@coreV1-22-016 sandboxes]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9284557      main KCridsan kcrid001  R       0:01      1 coreV1-22-016 
           9284556      main       sh kcrid001  R       0:31      1 coreV1-22-016 

#3- scp your github logfile to the /cm/shared/courses/dbarshis/21AdvGenomics/archive with the name LASTNAME_FIRSTNAME_readme.md

(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ scp README.md kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/archive/Crider_Katie_readme.md
kcrid001@turing.hpc.odu.edu's password: 
README.md                                         100%  207KB   1.1MB/s   00:00 

#4- scp the following files to your local computer
(base) Katies-MacBook-Air-4:Genomics katiecrider$ mkdir day13
(base) Katies-MacBook-Air-4:Genomics katiecrider$ cd day13
(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta ./
kcrid001@turing.hpc.odu.edu's password: 
15079_Apoc_hostsym.fasta                          100%   13MB  16.6MB/s   00:00    
(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta.fai ./
kcrid001@turing.hpc.odu.edu's password: 
15079_Apoc_hostsym.fasta.fai                      100%  641KB   2.0MB/s   00:00    
(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam ./
kcrid001@turing.hpc.odu.edu's password: 
RI_B_01.bam                                       100%   44MB  17.5MB/s   00:02    
(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam.bai ./
kcrid001@turing.hpc.odu.edu's password: 
RI_B_01.bam.bai  


#5- launch IGV on your local machine and load in 15079_Apoc_hostsym.fasta as the &quot;genome&quot; and RI_B_01.bam as the input file
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>1- create an sbatch script to rm -r your sandbox/YOURNAME directory</span>
<span class="pl-c"><span class="pl-c">#</span>2- submit your sbatch script and add the entry to your logfile/readme</span>

[kcrid001@turing1 sandboxes]$ cat SandboxRemovalKCrid.sh
<span class="pl-c"><span class="pl-c">#!</span>/bin/bash -l</span>

<span class="pl-c"><span class="pl-c">#</span>SBATCH -o KCridsandboxremoval.txt</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH -n 1</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-user=kcrid001@odu.edu</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --mail-type=END</span>
<span class="pl-c"><span class="pl-c">#</span>SBATCH --job-name=KCridsandboxremoval</span>

rm -r ./katiecrider
[kcrid001@turing1 sandboxes]$ salloc
salloc: Pending job allocation 9284556
salloc: job 9284556 queued and waiting <span class="pl-k">for</span> resources
salloc: job 9284556 has been allocated resources
salloc: Granted job allocation 9284556
[kcrid001@coreV1-22-016 sandboxes]$ sbatch SandboxRemovalKCrid.sh
Submitted batch job 9284557
[kcrid001@coreV1-22-016 sandboxes]$ squeue -u kcrid001
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON) 
           9284557      main KCridsan kcrid001  R       0:01      1 coreV1-22-016 
           9284556      main       sh kcrid001  R       0:31      1 coreV1-22-016 

<span class="pl-c"><span class="pl-c">#</span>3- scp your github logfile to the /cm/shared/courses/dbarshis/21AdvGenomics/archive with the name LASTNAME_FIRSTNAME_readme.md</span>

(base) Katies-MacBook-Air-4:KatieCriderAdvancedGenomicsLog katiecrider$ scp README.md kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/archive/Crider_Katie_readme.md
kcrid001@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password: </span>
<span class="pl-s">README.md                                         100%  207KB   1.1MB/s   00:00 </span>
<span class="pl-s"></span>
<span class="pl-s">#4- scp the following files to your local computer</span>
<span class="pl-s">(base) Katies-MacBook-Air-4:Genomics katiecrider$ mkdir day13</span>
<span class="pl-s">(base) Katies-MacBook-Air-4:Genomics katiecrider$ cd day13</span>
<span class="pl-s">(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta ./</span>
<span class="pl-s">kcrid001@turing.hpc.odu.edu<span class="pl-pds">'</span></span>s password: 
15079_Apoc_hostsym.fasta                          100%   13MB  16.6MB/s   00:00    
(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta.fai ./
kcrid001@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password: </span>
<span class="pl-s">15079_Apoc_hostsym.fasta.fai                      100%  641KB   2.0MB/s   00:00    </span>
<span class="pl-s">(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam ./</span>
<span class="pl-s">kcrid001@turing.hpc.odu.edu<span class="pl-pds">'</span></span>s password: 
RI_B_01.bam                                       100%   44MB  17.5MB/s   00:02    
(base) Katies-MacBook-Air-4:day13 katiecrider$ scp kcrid001@turing.hpc.odu.edu:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/dan/data/BAMS/RI_B_01/RI_B_01.bam.bai ./
kcrid001@turing.hpc.odu.edu<span class="pl-s"><span class="pl-pds">'</span>s password: </span>
<span class="pl-s">RI_B_01.bam.bai  </span>
<span class="pl-s"></span>
<span class="pl-s"></span>
<span class="pl-s">#5- launch IGV on your local machine and load in 15079_Apoc_hostsym.fasta as the "genome" and RI_B_01.bam as the input file</span>
<span class="pl-s"></span></pre>
                    </div>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" id="react-codespace-csrf" type="hidden" value="kZ26Zx8GFbb64v_UhTMExmkqDvbBlqnTq8iXpc0YH5-GRSXIgw1t1vmuBa3TZ5kIH5AjFSE_0knaG7MOyLuBvA">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 log of work in the 2021 Spring Advanced Genomics class
                </p>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/kcrid001/KatieCriderAdvancedGenomicsLog/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fkcrid001%2FKatieCriderAdvancedGenomicsLog&amp;report=kcrid001+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/kcrid001/KatieCriderAdvancedGenomicsLog/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/kcrid001/packages?repo_name=KatieCriderAdvancedGenomicsLog">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/kcrid001/KatieCriderAdvancedGenomicsLog/used_by_list">
               </include-fragment>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <style>
    .user-mention[href$="/ninenia"] {
        color: var(--color-user-mention-fg);
        background-color: var(--color-user-mention-bg);
        border-radius: 2px;
        margin-left: -2px;
        margin-right: -2px;
        padding: 0 2px;
      }
   </style>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
