<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light_v2-e830ce163aae.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark_v2-bc3c552285ea.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed_v2-191796a86a54.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast_v2-e0a0b6e42616.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind_v2-4fd40c9324e8.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind_v2-2f299c61eceb.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast_v2-05d723322773.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia_v2-5679d6101565.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia_v2-0ff66d76cc55.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","repository_suggester_elastic_search","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-6a292b1161ed.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         lorenapr92/21SPStephGAdvanceGenomicslog
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="90A8:5F4D:3608A9:4C5F2F:6605CBC7" data-turbo-transient="true" name="request-id">
         <meta content="22f598a296c639e8b750dc2cf1a20323836185b0d8b152a2e56d5461ad76529a" data-turbo-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6bnVsbCwicmVxdWVzdF9pZCI6IjkwQTg6NUY0RDozNjA4QTk6NEM1RjJGOjY2MDVDQkM3IiwidmlzaXRvcl9pZCI6IjU1NjM2ODU5OTEwODQ4MjkyODYiLCJyZWdpb25fZWRnZSI6ImlhZCIsInJlZ2lvbl9yZW5kZXIiOiJpYWQifQ==" data-turbo-transient="true" name="visitor-payload">
           <meta content="fb3a21da2360d69daed4e8980de5d5624e1db52dcb1d6eddcd4179d6afe71d7a" data-turbo-transient="true" name="visitor-hmac">
            <meta content="repository:331923588" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="126681664" name="octolytics-actor-id">
               <meta content="ninenia" name="octolytics-actor-login">
                <meta content="57d6abdd600494be8ac2f82b25b957e4c0d029a4d648c10497790171de0492de" name="octolytics-actor-hash">
                 <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
                  <meta content="ninenia" name="user-login"/>
                  <link href="/sessions/sudo_modal" rel="sudo-modal"/>
                  <meta content="width=device-width" name="viewport"/>
                  <meta content="Contribute to lorenapr92/21SPStephGAdvanceGenomicslog development by creating an account on GitHub." name="description"/>
                  <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
                  <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
                  <meta content="1401488693436528" property="fb:app_id"/>
                  <meta content="app-id=1477376905, app-argument=https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog" name="apple-itunes-app">
                   <meta content="https://opengraph.githubassets.com/d49c3ccc166003bf38f0289cdcb09b6aa79d8130c91dbc279a87a4801f9c834d/lorenapr92/21SPStephGAdvanceGenomicslog" name="twitter:image:src">
                    <meta content="@github" name="twitter:site">
                     <meta content="summary_large_image" name="twitter:card">
                      <meta content="lorenapr92/21SPStephGAdvanceGenomicslog" name="twitter:title">
                       <meta content="Contribute to lorenapr92/21SPStephGAdvanceGenomicslog development by creating an account on GitHub." name="twitter:description">
                        <meta content="https://opengraph.githubassets.com/d49c3ccc166003bf38f0289cdcb09b6aa79d8130c91dbc279a87a4801f9c834d/lorenapr92/21SPStephGAdvanceGenomicslog" property="og:image"/>
                        <meta content="Contribute to lorenapr92/21SPStephGAdvanceGenomicslog development by creating an account on GitHub." property="og:image:alt"/>
                        <meta content="1200" property="og:image:width"/>
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="lorenapr92/21SPStephGAdvanceGenomicslog" property="og:title"/>
                        <meta content="https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog" property="og:url"/>
                        <meta content="Contribute to lorenapr92/21SPStephGAdvanceGenomicslog development by creating an account on GitHub." property="og:description"/>
                        <link data-refresh-url="/_alive" data-session-id="fae53153a7a7c8d08a43b9810520b9ce4325ce8b4e99a06c5fc3f52c85cc22fe" href="wss://alive.github.com/_sockets/u/126681664/ws?session=eyJ2IjoiVjMiLCJ1IjoxMjY2ODE2NjQsInMiOjEzMDQyODIxMTYsImMiOjI1OTYwODIyMjcsInQiOjE3MTE2NTU4ODR9--6a63f1ae10a078389a141d2547d0b6ee8d27f68a1a8c93f326c19997e1ccc8cd" rel="shared-web-socket"/>
                        <link href="/assets-cdn/worker/socket-worker-9cc1149b224c.js" rel="shared-web-socket-src"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="all" name="keyboard-shortcuts-preference"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="de8e1521fc37a178caaed17097d0545118b4eb0f01833d1d1eeeb65ac20b6ca7" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="1dd47e41bae3fb619dfa564356c6dee453350e97ff939ad02473ed6906f90b45" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/lorenapr92/21SPStephGAdvanceGenomicslog git https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog.git" name="go-import"/>
                        <meta content="37906567" name="octolytics-dimension-user_id">
                         <meta content="lorenapr92" name="octolytics-dimension-user_login">
                          <meta content="331923588" name="octolytics-dimension-repository_id">
                           <meta content="lorenapr92/21SPStephGAdvanceGenomicslog" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="331923588" name="octolytics-dimension-repository_network_root_id">
                               <meta content="lorenapr92/21SPStephGAdvanceGenomicslog" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog" rel="canonical"/>
                                <meta content="logged-in env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <meta content="/windows-tile.png" name="msapplication-TileImage"/>
                                 <meta content="#ffffff" name="msapplication-TileColor"/>
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-in env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-in env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="p-3 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_allex_crc32_lib_crc32_esm_js-node_modules_github_mini-throttle_dist_deco-b38cad-748e74df23ce.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_clipboard-copy-element_dist_index_esm_js-node_modules_delegated-e-b37f7d-2f24d321a3fb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_command-palette_items_help-item_ts-app_assets_modules_github_comman-48ad9d-a9ed17f2ebff.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/command-palette-f620e569db0b.js" type="application/javascript">
    </script>
    <header class="AppHeader">
     <div class="AppHeader-globalBar pb-2 js-global-bar">
      <div class="AppHeader-globalBar-start">
       <deferred-side-panel data-url="/_side-panels/global">
        <include-fragment data-target="deferred-side-panel.fragment">
         <button aria-label="Open global navigation menu" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-bg-transparent p-0 color-fg-muted" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-92769683-962f-4b83-bfc4-34c21b641218" data-view-component="true" id="dialog-show-dialog-92769683-962f-4b83-bfc4-34c21b641218" type="button">
          <svg aria-hidden="true" class="octicon octicon-three-bars Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1 2.75A.75.75 0 0 1 1.75 2h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 2.75Zm0 5A.75.75 0 0 1 1.75 7h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 7.75ZM1.75 12h12.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Z">
           </path>
          </svg>
         </button>
         <dialog-helper>
          <dialog aria-describedby="dialog-92769683-962f-4b83-bfc4-34c21b641218-description" aria-labelledby="dialog-92769683-962f-4b83-bfc4-34c21b641218-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-left SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-92769683-962f-4b83-bfc4-34c21b641218">
           <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title sr-only" id="dialog-92769683-962f-4b83-bfc4-34c21b641218-title">
               Global navigation
              </h1>
              <div class="d-flex" data-view-component="true">
               <div class="AppHeader-logo position-relative" data-view-component="true">
                <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
                 <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
                 </path>
                </svg>
               </div>
              </div>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-92769683-962f-4b83-bfc4-34c21b641218" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="dialog-92769683-962f-4b83-bfc4-34c21b641218-title">
            <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
             <div class="d-flex flex-column mb-3" data-view-component="true">
              <nav aria-label="Site navigation" class="ActionList" data-view-component="true">
               <nav-list>
                <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"HOME","label":null}' data-hotkey="g d" data-view-component="true" href="/dashboard" id="item-8d1042c4-db6a-45d2-a214-c13d6531b378">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-home" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M6.906.664a1.749 1.749 0 0 1 2.187 0l5.25 4.2c.415.332.657.835.657 1.367v7.019A1.75 1.75 0 0 1 13.25 15h-3.5a.75.75 0 0 1-.75-.75V9H7v5.25a.75.75 0 0 1-.75.75h-3.5A1.75 1.75 0 0 1 1 13.25V6.23c0-.531.242-1.034.657-1.366l5.25-4.2Zm1.25 1.171a.25.25 0 0 0-.312 0l-5.25 4.2a.25.25 0 0 0-.094.196v7.019c0 .138.112.25.25.25H5.5V8.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 .75.75v5.25h2.75a.25.25 0 0 0 .25-.25V6.23a.25.25 0 0 0-.094-.195Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Home
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"ISSUES","label":null}' data-hotkey="g i" data-view-component="true" href="/issues" id="item-cf414e2c-f2ed-4911-8b8f-7ac46f1d0807">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS","label":null}' data-hotkey="g p" data-view-component="true" href="/pulls" id="item-e90082b6-a1c8-4cb6-a11b-4409f9226f81">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="projects" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROJECTS","label":null}' data-view-component="true" href="/projects" id="item-5986d06b-5994-412f-a276-d8d518c37295">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"DISCUSSIONS","label":null}' data-view-component="true" href="/discussions" id="item-1c61ffc2-b3f9-4030-a6b2-8776b3aae0c2">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Discussions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"CODESPACES","label":null}' data-view-component="true" href="https://github.com/codespaces" id="item-49f3a5e8-d5b4-43b5-baa4-738c91d3c162">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Codespaces
                   </span>
                  </a>
                 </li>
                 <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"EXPLORE","label":null}' data-view-component="true" href="/explore" id="item-a6ce69a8-5ea7-4ec6-917a-d0a03ae5df07">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-telescope" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.184 1.143v-.001l1.422 2.464a1.75 1.75 0 0 1-.757 2.451L3.104 11.713a1.75 1.75 0 0 1-2.275-.702l-.447-.775a1.75 1.75 0 0 1 .53-2.32L11.682.573a1.748 1.748 0 0 1 2.502.57Zm-4.709 9.32h-.001l2.644 3.863a.75.75 0 1 1-1.238.848l-1.881-2.75v2.826a.75.75 0 0 1-1.5 0v-2.826l-1.881 2.75a.75.75 0 1 1-1.238-.848l2.049-2.992a.746.746 0 0 1 .293-.253l1.809-.87a.749.749 0 0 1 .944.252ZM9.436 3.92h-.001l-4.97 3.39.942 1.63 5.42-2.61Zm3.091-2.108h.001l-1.85 1.26 1.505 2.605 2.016-.97a.247.247 0 0 0 .13-.151.247.247 0 0 0-.022-.199l-1.422-2.464a.253.253 0 0 0-.161-.119.254.254 0 0 0-.197.038ZM1.756 9.157a.25.25 0 0 0-.075.33l.447.775a.25.25 0 0 0 .325.1l1.598-.769-.83-1.436-1.465 1Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Explore
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"MARKETPLACE","label":null}' data-view-component="true" href="/marketplace" id="item-9a2b23fe-cdc1-4966-9b33-b96f1e56bcab">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Marketplace
                   </span>
                  </a>
                 </li>
                </ul>
               </nav-list>
              </nav>
              <div class="my-3 d-flex flex-justify-center height-full" data-view-component="true">
               <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                </circle>
                <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                </path>
               </svg>
              </div>
             </div>
             <div class="flex-1" data-view-component="true">
             </div>
             <div class="px-2" data-view-component="true">
              <p class="color-fg-subtle text-small text-light">
               Â© 2024 GitHub, Inc.
              </p>
              <div class="d-flex flex-wrap text-small text-light" data-view-component="true">
               <a class="Link mr-2" data-view-component="true" href="https://github.com/about" target="_blank">
                About
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.blog" target="_blank">
                Blog
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service" target="_blank">
                Terms
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement" target="_blank">
                Privacy
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.com/security" target="_blank">
                Security
               </a>
               <a class="Link mr-3" data-view-component="true" href="https://www.githubstatus.com/" target="_blank">
                Status
               </a>
              </div>
             </div>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </include-fragment>
       </deferred-side-panel>
       <a aria-label="Homepage " class="AppHeader-logo ml-2" data-analytics-event='{"category":"Header","action":"go to dashboard","label":"icon:logo"}' data-hotkey="g d" data-turbo="false" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github v-align-middle color-fg-default" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="AppHeader-context">
        <div class="AppHeader-context-compact">
         <button aria-expanded="false" aria-haspopup="dialog" aria-label="Page context: lorenapr92 / 21SPStephGAdvanceGenomicslog" class="AppHeader-context-compact-trigger Truncate Button--secondary Button--medium Button box-shadow-none" data-show-dialog-id="context-region-dialog" data-view-component="true" id="dialog-show-context-region-dialog" type="button">
          <span class="Button-content">
           <span class="Button-label">
            <span class="AppHeader-context-compact-lead">
             <span class="AppHeader-context-compact-parentItem">
              lorenapr92
             </span>
             <span class="no-wrap">
              /
             </span>
            </span>
            <strong class="AppHeader-context-compact-mainItem d-flex flex-items-center Truncate">
             <span class="Truncate-text">
              21SPStephGAdvanceGenomicslog
             </span>
            </strong>
           </span>
          </span>
         </button>
         <dialog-helper>
          <dialog aria-describedby="context-region-dialog-description" aria-labelledby="context-region-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-view-component="true" id="context-region-dialog">
           <div class="Overlay-header" data-view-component="true">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title" id="context-region-dialog-title">
               Navigate back to
              </h1>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="context-region-dialog" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="context-region-dialog-title">
            <div class="Overlay-body" data-view-component="true">
             <ul class="list-style-none" role="list">
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"lorenapr92","screen_size":"compact"}' data-view-component="true" href="/lorenapr92">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-person mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                  </path>
                 </svg>
                 lorenapr92
                </span>
               </a>
              </li>
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"21SPStephGAdvanceGenomicslog","screen_size":"compact"}' data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-repo mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                  </path>
                 </svg>
                 21SPStephGAdvanceGenomicslog
                </span>
               </a>
              </li>
             </ul>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </div>
        <div class="AppHeader-context-full">
         <nav aria-label="Page context" role="navigation">
          <ul class="list-style-none" role="list">
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"lorenapr92","screen_size":"full"}' data-hovercard-type="user" data-hovercard-url="/users/lorenapr92/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" data-view-component="true" href="/lorenapr92">
             <span class="AppHeader-context-item-label">
              lorenapr92
             </span>
            </a>
            <span class="AppHeader-context-item-separator">
             /
            </span>
           </li>
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"21SPStephGAdvanceGenomicslog","screen_size":"full"}' data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog">
             <span class="AppHeader-context-item-label">
              21SPStephGAdvanceGenomicslog
             </span>
            </a>
           </li>
          </ul>
         </nav>
        </div>
       </div>
      </div>
      <div class="AppHeader-globalBar-end">
       <div class="AppHeader-search">
        <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;input type="hidden" value="oa-7xPlILJ41SEjosJSMZWqHYn6DEQXpI0elLFF_5TvxjLdMa8XzUYLSEMYVA24-Xguilbc6sHj9PNuS_LtY0w" data-csrf="true" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="lorenapr92" data-current-repository="lorenapr92/21SPStephGAdvanceGenomicslog" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="xk5bBGvjlA9bc64MM9OhBn613qQSPWAp5YkvhmfMf5iOhEwuTaXdegzf3NXV1lEXLBGcDTRakEaJa24tgPJk1A" data-header-redesign-enabled="true" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="true" data-max-custom-scopes="10" data-scope="repo:lorenapr92/21SPStephGAdvanceGenomicslog">
         <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center height-auto color-bg-transparent border-0 color-fg-subtle mx-0" data-action="click:qbsearch-input#searchInputContainerClicked">
          <button aria-expanded="false" aria-haspopup="dialog" aria-label="Search or jump toâ¦" class="AppHeader-button AppHeader-search-whenNarrow" data-action="click:qbsearch-input#handleExpand" type="button">
           <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
            </path>
           </svg>
          </button>
          <div class="AppHeader-search-whenRegular">
           <div class="AppHeader-search-wrap AppHeader-search-wrap--hasTrailing">
            <div class="AppHeader-search-control">
             <label aria-label="Search or jump toâ¦" class="AppHeader-search-visual--leading" for="AppHeader-searchInput">
              <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
               </path>
              </svg>
             </label>
             <button aria-describedby="search-error-message-flash" class="AppHeader-searchButton form-control input-contrast text-left color-fg-subtle no-wrap" data-action="click:qbsearch-input#handleExpand" data-analytics-event='{"category":"Global navigation","action":"SEARCH","label":null}' data-hotkey="s,/" data-target="qbsearch-input.inputButton" type="button">
              <div class="overflow-hidden">
               <span data-target="qbsearch-input.inputButtonText" id="qb-input-query">
                Type
                <kbd class="AppHeader-search-kbd">
                 /
                </kbd>
                to search
               </span>
              </div>
             </button>
            </div>
            <button class="AppHeader-search-action--trailing js-activate-command-palette" data-analytics-event='{"category":"SiteHeaderComponent","action":"command_palette","label":"open command palette"}' id="AppHeader-commandPalette-button" type="button">
             <svg aria-hidden="true" class="octicon octicon-command-palette" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m6.354 8.04-4.773 4.773a.75.75 0 1 0 1.061 1.06L7.945 8.57a.75.75 0 0 0 0-1.06L2.642 2.206a.75.75 0 0 0-1.06 1.061L6.353 8.04ZM8.75 11.5a.75.75 0 0 0 0 1.5h5.5a.75.75 0 0 0 0-1.5h-5.5Z">
              </path>
             </svg>
            </button>
            <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="AppHeader-commandPalette-button" id="tooltip-44b3ee87-f7a3-4c03-9b50-02f825c09d9e" popover="manual">
             Command palette
            </tool-tip>
           </div>
          </div>
          <input class="js-site-search-type-field" name="type" type="hidden"/>
          <div class="Overlay--hidden" data-modal-dialog-overlay="">
           <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-medium Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
            <h1 class="sr-only" id="search-suggestions-dialog-header">
             Search code, repositories, users, issues, pull requests...
            </h1>
            <div class="Overlay-body Overlay-body--paddingNone">
             <div data-view-component="true">
              <div class="search-suggestions position-absolute width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                 <div class="FormControl FormControl--fullWidth">
                  <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                   Search
                  </label>
                  <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                   <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                    <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </span>
                   <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                    <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                    </div>
                    <div class="QueryBuilder-InputWrapper">
                     <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                     </div>
                     <input aria-describedby="validation-f280e7e7-2c33-4bd6-92d5-e5ca6a2d731b" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                     </input>
                    </div>
                   </div>
                   <span class="sr-only" id="query-builder-test-clear">
                    Clear
                   </span>
                   <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                    <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <template id="search-icon">
                   <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-icon">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="file-code-icon">
                   <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                    </path>
                   </svg>
                  </template>
                  <template id="history-icon">
                   <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                    </path>
                   </svg>
                  </template>
                  <template id="repo-icon">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="bookmark-icon">
                   <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="plus-circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                    </path>
                   </svg>
                  </template>
                  <template id="trash-icon">
                   <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="team-icon">
                   <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="project-icon">
                   <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                    </path>
                   </svg>
                  </template>
                  <template id="pencil-icon">
                   <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                    </path>
                   </svg>
                  </template>
                  <template id="copilot-icon">
                   <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                    </path>
                    <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="workflow-icon">
                   <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="book-icon">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-review-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="codespaces-icon">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-discussion-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="organization-icon">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="rocket-icon">
                   <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="shield-check-icon">
                   <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="heart-icon">
                   <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="server-icon">
                   <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="globe-icon">
                   <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                    </path>
                   </svg>
                  </template>
                  <template id="issue-opened-icon">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-mobile-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="package-icon">
                   <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                    </path>
                   </svg>
                  </template>
                  <template id="credit-card-icon">
                   <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                    </path>
                    <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                    </path>
                   </svg>
                  </template>
                  <template id="play-icon">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </template>
                  <template id="gift-icon">
                   <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-square-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-desktop-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                    </path>
                   </svg>
                  </template>
                  <div class="position-relative">
                   <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                   </ul>
                  </div>
                  <div class="FormControl-inlineValidation" hidden="hidden" id="validation-f280e7e7-2c33-4bd6-92d5-e5ca6a2d731b">
                   <span class="FormControl-inlineValidation--visual">
                    <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                     <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                     </path>
                    </svg>
                   </span>
                   <span>
                   </span>
                  </div>
                 </div>
                 <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                 </div>
                </query-builder>
               </form>
               <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                 Search syntax tips
                </a>
                <div class="d-flex flex-1">
                </div>
                <button class="Button--link Button--medium Button color-fg-accent text-normal ml-2" data-action="click:qbsearch-input#showFeedbackDialog" data-view-component="true" type="button">
                 <span class="Button-content">
                  <span class="Button-label">
                   Give feedback
                  </span>
                 </span>
                </button>
               </div>
              </div>
             </div>
            </div>
           </modal-dialog>
          </div>
         </div>
         <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
         </div>
         <div class="color-fg-default">
          <dialog-helper>
           <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
            <div class="Overlay-header" data-view-component="true">
             <div class="Overlay-headerContentWrap">
              <div class="Overlay-titleWrap">
               <h1 class="Overlay-title" id="feedback-dialog-title">
                Provide feedback
               </h1>
              </div>
              <div class="Overlay-actionWrap">
               <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
              </div>
             </div>
            </div>
            <scrollable-region data-labelled-by="feedback-dialog-title">
             <div class="Overlay-body" data-view-component="true">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
               <input name="authenticity_token" type="hidden" value="tCEa2c81RQ5d8jhbrIWxa8QoaQr23kKyMH-osYQr-OImkwPhB-N2lO3QNl-SnLCaGSOkyaz-pOc-_70Lcnr9WQ"/>
               <p>
                We read every piece of feedback, and take your input very seriously.
               </p>
               <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
               <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
               <label for="include_email" style="font-weight: normal">
                Include my email address so I can be contacted
               </label>
              </form>
             </div>
            </scrollable-region>
            <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
             <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
              Cancel
             </button>
             <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
              Submit feedback
             </button>
            </div>
           </dialog>
          </dialog-helper>
          <custom-scopes data-target="qbsearch-input.customScopesManager">
           <dialog-helper>
            <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
             <div class="Overlay-header Overlay-header--divided" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                 Saved searches
                </h1>
                <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                 Use saved searches to filter your results more quickly
                </h2>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="custom-scopes-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <div data-target="custom-scopes.customScopesModalDialogFlash">
               </div>
               <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                 <input name="authenticity_token" type="hidden" value="fgKTtcmPacxq5uu_48HfWO7I1UVckA52-MI9SqSU-SEy8PEPkQ43FSGPVhKpFBJlJ5ixUgqVTK1bmuzyu9OgdA">
                  <div data-target="custom-scopes.customScopesModalDialogFlash">
                  </div>
                  <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                  <div class="form-group">
                   <label for="custom_scope_name">
                    Name
                   </label>
                   <auto-check required="" src="/search/custom_scopes/check_name">
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                    <input data-csrf="true" type="hidden" value="zEgGwVQfOjR9UhR2ycPUkmVLWQbBkudgGy9p32WszyXUFUtnA52TN_5GXfWJkYr404xcge9LB_h2j8sNXcMc7Q">
                    </input>
                   </auto-check>
                  </div>
                  <div class="form-group">
                   <label for="custom_scope_query">
                    Query
                   </label>
                   <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                  </div>
                  <p class="text-small color-fg-muted">
                   To see all available qualifiers, see our
                   <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                    documentation
                   </a>
                   .
                  </p>
                 </input>
                </form>
               </div>
               <div data-target="custom-scopes.manageCustomScopesForm">
                <div data-target="custom-scopes.list">
                </div>
               </div>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
              <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
               Create saved search
              </button>
             </div>
            </dialog>
           </dialog-helper>
          </custom-scopes>
         </div>
        </qbsearch-input>
        <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="mu85cb6V-43T5UNSjDpqAk0OdsoG18bsO-2rnbne6-JdulZTp5I97zQ-2a9FgRERnbauAINxANw0jvU4IUqK6A">
        </input>
       </div>
       <div class="AppHeader-actions position-relative">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="global-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="global-create-menu-button" popovertarget="global-create-menu-overlay" type="button">
           <span class="Button-content">
            <span class="Button-visual Button-leadingVisual">
             <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
              </path>
             </svg>
            </span>
            <span class="Button-label">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </span>
           </span>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="global-create-menu-button" id="tooltip-e3d5a5de-8f73-4231-bbf2-e13cc39ee6fd" popover="manual">
           Create new...
          </tool-tip>
          <anchored-position align="end" anchor="global-create-menu-button" anchor-offset="normal" data-view-component="true" id="global-create-menu-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="global-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="global-create-menu-list" role="menu">
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-ad393555-bfa5-42b2-9da9-4ff51da504e8" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New repository
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-2ed6a2a9-b126-4553-af3c-6897afaa473f" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Import repository
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-2d64cea6-1c79-41ef-a699-3cc1fd7bc7de" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New codespace
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-cf19d75c-a316-4084-977b-188d63fb7e15" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New gist
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-3fee10c9-bcb9-4cfb-84bf-c192c665b53c" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New organization
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
        <a aria-labelledby="tooltip-c5e6c5b8-22af-4455-9036-a65d490cafdb" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"ISSUES_HEADER","label":null}' data-view-component="true" href="/issues" id="icon-button-ea908a76-d84b-41b3-ac6b-f9368510be9f">
         <svg aria-hidden="true" class="octicon octicon-issue-opened Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-ea908a76-d84b-41b3-ac6b-f9368510be9f" id="tooltip-c5e6c5b8-22af-4455-9036-a65d490cafdb" popover="manual">
         Issues
        </tool-tip>
        <a aria-labelledby="tooltip-85fdee69-bb08-4990-a21e-47c36eff05fb" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS_HEADER","label":null}' data-view-component="true" href="/pulls" id="icon-button-bc015cd8-69d1-478a-82af-52cc4e9a3615">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-bc015cd8-69d1-478a-82af-52cc4e9a3615" id="tooltip-85fdee69-bb08-4990-a21e-47c36eff05fb" popover="manual">
         Pull requests
        </tool-tip>
       </div>
       <notification-indicator class="js-socket-channel" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY1NTg4NH0=--e22b30e7e471b7a5a4561c391a5ed0662f9c24f29c70daf0a902892f2158d1cb" data-fetch-indicator-enabled="true" data-fetch-indicator-src="/notifications/indicator" data-header-redesign-enabled="true" data-indicator-mode="none" data-tooltip-global="You have unread notifications" data-tooltip-none="You have no unread notifications" data-tooltip-unavailable="Notifications are unavailable at the moment." data-view-component="true">
        <a aria-label="Notifications" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"NOTIFICATIONS_HEADER","label":null}' data-hotkey="g n" data-target="notification-indicator.link" data-view-component="true" href="/notifications" id="AppHeader-notifications-button">
         <svg aria-hidden="true" class="octicon octicon-inbox Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.8 2.06A1.75 1.75 0 0 1 4.41 1h7.18c.7 0 1.333.417 1.61 1.06l2.74 6.395c.04.093.06.194.06.295v4.5A1.75 1.75 0 0 1 14.25 15H1.75A1.75 1.75 0 0 1 0 13.25v-4.5c0-.101.02-.202.06-.295Zm1.61.44a.25.25 0 0 0-.23.152L1.887 8H4.75a.75.75 0 0 1 .6.3L6.625 10h2.75l1.275-1.7a.75.75 0 0 1 .6-.3h2.863L11.82 2.652a.25.25 0 0 0-.23-.152Zm10.09 7h-2.875l-1.275 1.7a.75.75 0 0 1-.6.3h-3.5a.75.75 0 0 1-.6-.3L4.375 9.5H1.5v3.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-target="notification-indicator.tooltip" data-type="description" data-view-component="true" for="AppHeader-notifications-button" id="tooltip-203093a6-534d-4583-a6f9-32a0c1e6b82b" popover="manual">
         Notifications
        </tool-tip>
       </notification-indicator>
       <div class="AppHeader-user">
        <deferred-side-panel data-url="/_side-panels/user?react_global_nav=false&amp;repository_id=331923588">
         <include-fragment data-target="deferred-side-panel.fragment">
          <user-drawer-side-panel>
           <button aria-label="Open user account menu" class="AppHeader-logo Button--invisible Button--medium Button Button--invisible-noVisuals color-bg-transparent p-0" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-77c6c4ec-a682-4757-badb-c7967e419872" data-view-component="true" id="dialog-show-dialog-77c6c4ec-a682-4757-badb-c7967e419872" type="button">
            <span class="Button-content">
             <span class="Button-label">
              <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
             </span>
            </span>
           </button>
           <dialog-helper>
            <dialog aria-describedby="dialog-77c6c4ec-a682-4757-badb-c7967e419872-description" aria-labelledby="dialog-77c6c4ec-a682-4757-badb-c7967e419872-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-right SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-77c6c4ec-a682-4757-badb-c7967e419872">
             <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title sr-only" id="dialog-77c6c4ec-a682-4757-badb-c7967e419872-title">
                 Account menu
                </h1>
                <div class="d-flex" data-view-component="true">
                 <div class="AppHeader-logo position-relative" data-view-component="true">
                  <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
                 </div>
                 <div class="overflow-hidden d-flex width-full" data-view-component="true">
                  <div class="lh-condensed overflow-hidden d-flex flex-column flex-justify-center ml-2 f5 mr-auto width-full" data-view-component="true">
                   <span class="Truncate text-bold" data-view-component="true">
                    <span class="Truncate-text" data-view-component="true">
                     ninenia
                    </span>
                   </span>
                  </div>
                  <action-menu class="d-sm-none d-md-none d-lg-none" data-select-variant="none" data-view-component="true">
                   <focus-group direction="vertical" mnemonics="" retain="">
                    <button aria-controls="user-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="user-create-menu-button" popovertarget="user-create-menu-overlay" type="button">
                     <span class="Button-content">
                      <span class="Button-visual Button-leadingVisual">
                       <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                        </path>
                       </svg>
                      </span>
                      <span class="Button-label">
                       <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                        </path>
                       </svg>
                      </span>
                     </span>
                    </button>
                    <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="user-create-menu-button" id="tooltip-5a1c90f6-726d-406d-a4a6-c21dc003b482" popover="manual">
                     Create new...
                    </tool-tip>
                    <anchored-position align="end" anchor="user-create-menu-button" anchor-offset="normal" data-view-component="true" id="user-create-menu-overlay" popover="auto" side="outside-bottom">
                     <div class="Overlay Overlay--size-auto" data-view-component="true">
                      <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
                       <action-list>
                        <div data-view-component="true">
                         <ul aria-labelledby="user-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="user-create-menu-list" role="menu">
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-ba1798d0-5021-4760-bf11-9fc96021cf66" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New repository
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-e0ef00ff-d6a0-465b-a934-38f09977380e" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             Import repository
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-466129a3-d8c0-44e6-b7dd-7edff321118c" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                              </path>
                              <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New codespace
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-21e700a5-a6cd-43d0-addb-5a8d9081f2a2" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New gist
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-df8e6bd1-da84-4228-aaa1-6fedb31866e0" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New organization
                            </span>
                           </a>
                          </li>
                         </ul>
                        </div>
                       </action-list>
                      </div>
                     </div>
                    </anchored-position>
                   </focus-group>
                  </action-menu>
                 </div>
                </div>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-77c6c4ec-a682-4757-badb-c7967e419872" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="dialog-77c6c4ec-a682-4757-badb-c7967e419872-title">
              <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
               <div class="d-flex flex-column mb-3" data-view-component="true">
                <nav aria-label="User navigation" class="ActionList" data-view-component="true">
                 <nav-list>
                  <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-6c9ccd4f-ea79-40b1-a416-41473bdb9575" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROFILE","label":null}' data-view-component="true" href="https://github.com/ninenia" id="item-aa8e2043-cf9f-4a0b-bcb6-1ca64f183ad2">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-person" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your profile
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-c0b200c7-759d-4a92-8785-288d0ff5fa92" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_REPOSITORIES","label":null}' data-view-component="true" href="/ninenia?tab=repositories" id="item-559b43cc-d124-487d-8a6b-5ab9c20864fb">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your repositories
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_PROJECTS","label":null}' data-view-component="true" href="/ninenia?tab=projects" id="item-165b9f71-cb07-42f5-851a-d67437397b27">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your projects
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-e69b6bb3-5dca-4d20-b873-f3c98ea4e55e" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-8509a423-4067-4af7-a8ce-6a304699af5c" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_STARS","label":null}' data-view-component="true" href="/ninenia?tab=stars" id="item-3c8502b3-8790-41a7-89db-a570865d0203">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-star" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your stars
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SPONSORS","label":null}' data-view-component="true" href="/sponsors/accounts" id="item-fbbd347b-0a26-4ba1-95ef-17489e141d2b">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your sponsors
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_GISTS","label":null}' data-view-component="true" href="https://gist.github.com/mine" id="item-48ac99c5-9097-40b2-85e9-693aa8318cfb">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your gists
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-7ad0235a-94f2-4d6c-92c1-f62610bbcc1f" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-d12aee65-f190-4b7d-8bfc-f415812f3f92" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SETTINGS","label":null}' data-view-component="true" href="/settings/profile" id="item-0f603a64-d294-4e5e-a071-c9f20d16196a">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-gear" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Settings
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SUPPORT","label":null}' data-view-component="true" href="https://support.github.com" id="item-e4e8ca6d-9497-41b2-be0b-bc6035dc5a3a">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Support
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"COMMUNITY","label":null}' data-view-component="true" href="https://community.github.com" id="item-01753c0f-c575-4595-aae7-de303ecdf4ee">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Community
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent" data-analytics-event='{"category":"Global navigation","action":"LOGOUT","label":null}' data-view-component="true" href="/logout" id="item-2f76c412-a767-4d8c-8a46-b0fb54a764ea">
                     <span class="ActionListItem-label" data-view-component="true">
                      Sign out
                     </span>
                    </a>
                   </li>
                  </ul>
                 </nav-list>
                </nav>
               </div>
              </div>
             </scrollable-region>
            </dialog>
           </dialog-helper>
          </user-drawer-side-panel>
         </include-fragment>
        </deferred-side-panel>
       </div>
       <div class="position-absolute mt-2">
        <site-header-logged-in-user-menu>
        </site-header-logged-in-user-menu>
       </div>
      </div>
     </div>
     <div class="AppHeader-localBar">
      <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav" data-pjax="#js-repo-pjax-container" data-view-component="true">
       <ul class="UnderlineNav-body list-style-none" data-view-component="true">
        <li class="d-inline-flex" data-view-component="true">
         <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /lorenapr92/21SPStephGAdvanceGenomicslog" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog" id="code-tab">
          <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
           </path>
          </svg>
          <span data-content="Code">
           Code
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /lorenapr92/21SPStephGAdvanceGenomicslog/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/issues" id="issues-tab">
          <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
           </path>
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
           </path>
          </svg>
          <span data-content="Issues">
           Issues
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /lorenapr92/21SPStephGAdvanceGenomicslog/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/pulls" id="pull-requests-tab">
          <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
           </path>
          </svg>
          <span data-content="Pull requests">
           Pull requests
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /lorenapr92/21SPStephGAdvanceGenomicslog/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/actions" id="actions-tab">
          <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
           </path>
          </svg>
          <span data-content="Actions">
           Actions
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /lorenapr92/21SPStephGAdvanceGenomicslog/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/projects" id="projects-tab">
          <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
           </path>
          </svg>
          <span data-content="Projects">
           Projects
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /lorenapr92/21SPStephGAdvanceGenomicslog/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/security" id="security-tab">
          <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
           </path>
          </svg>
          <span data-content="Security">
           Security
          </span>
          <include-fragment accept="text/fragment+html" src="/lorenapr92/21SPStephGAdvanceGenomicslog/security/overall-count">
          </include-fragment>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /lorenapr92/21SPStephGAdvanceGenomicslog/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/pulse" id="insights-tab">
          <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
           </path>
          </svg>
          <span data-content="Insights">
           Insights
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
       </ul>
       <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="action-menu-b0f24470-ca13-4fb2-befb-59c025231555-list" aria-haspopup="true" aria-labelledby="tooltip-98d0902b-72d5-4814-9a76-bb482b511ae5" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-b0f24470-ca13-4fb2-befb-59c025231555-button" popovertarget="action-menu-b0f24470-ca13-4fb2-befb-59c025231555-overlay" type="button">
           <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
           </svg>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-b0f24470-ca13-4fb2-befb-59c025231555-button" id="tooltip-98d0902b-72d5-4814-9a76-bb482b511ae5" popover="manual">
           Additional navigation options
          </tool-tip>
          <anchored-position align="start" anchor="action-menu-b0f24470-ca13-4fb2-befb-59c025231555-button" anchor-offset="normal" data-view-component="true" id="action-menu-b0f24470-ca13-4fb2-befb-59c025231555-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="action-menu-b0f24470-ca13-4fb2-befb-59c025231555-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-b0f24470-ca13-4fb2-befb-59c025231555-list" role="menu">
                <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog" id="item-41ab9d84-0056-4348-bd80-7176b0f8a454" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Code
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/issues" id="item-5dd3c68c-1655-4b3c-b3ee-d6efc3e28765" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Issues
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/pulls" id="item-f4b8da04-1fdc-4a45-b877-16c3d2d284d7" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Pull requests
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/actions" id="item-5d0da160-7eaa-4d5d-b75b-d5754927dd2f" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Actions
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/projects" id="item-09d4ba44-e0d2-4e85-a2fc-dd0c25a17128" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Projects
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/security" id="item-e7849aad-d802-4244-8c5f-3731e4048ba1" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Security
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/pulse" id="item-54ec8a01-b24e-40af-af81-c0c505f29c97" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Insights
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
       </div>
      </nav>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-49f8108a-8a5c-4758-96c8-68d19799face" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-d352b164-d6b5-4e44-8034-b0834214ae18" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-d352b164-d6b5-4e44-8034-b0834214ae18" id="tooltip-49f8108a-8a5c-4758-96c8-68d19799face" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <notification-shelf-watcher class="js-socket-channel" data-base-url="https://github.com/notifications/beta/shelf" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY1NTg4NH0=--e22b30e7e471b7a5a4561c391a5ed0662f9c24f29c70daf0a902892f2158d1cb" data-view-component="true">
   </notification-shelf-watcher>
   <div data-initial="" data-target="notification-shelf-watcher.placeholder" hidden="">
   </div>
   <details class="details-reset details-overlay details-overlay-dark js-command-palette-dialog" data-turbo-replace="" id="command-palette-pjax-container">
    <summary aria-label="Command palette trigger" tabindex="-1">
    </summary>
    <details-dialog aria-label="Command palette" class="command-palette-details-dialog d-flex flex-column flex-justify-center height-fit">
     <command-palette activation-hotkey="Mod+k,Mod+Alt+k" class="command-palette color-bg-default rounded-3 border color-shadow-small" command-mode-hotkey="Mod+Shift+K" data-action="
        command-palette-input-ready:command-palette#inputReady
        command-palette-page-stack-updated:command-palette#updateInputScope
        itemsUpdated:command-palette#itemsUpdated
        keydown:command-palette#onKeydown
        loadingStateChanged:command-palette#loadingStateChanged
        selectedItemChanged:command-palette#selectedItemChanged
        pageFetchError:command-palette#pageFetchError
      " return-to="/lorenapr92/21SPStephGAdvanceGenomicslog" user-id="126681664">
      <command-palette-mode data-char="#" data-placeholder="Search issues and pull requests" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="#" data-placeholder="Search issues, pull requests, discussions, and projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="!" data-placeholder="Search projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a user, organization, or repository" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a repository" data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode data-char="/" data-placeholder="Search files" data-scope-types='["repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="?">
      </command-palette-mode>
      <command-palette-mode data-char="&gt;" data-placeholder="Run a command">
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode class="js-command-palette-default-mode" data-char="" data-placeholder="Search or jump to...">
      </command-palette-mode>
      <command-palette-input data-action="
          command-palette-input:command-palette#onInput
          command-palette-select:command-palette#onSelect
          command-palette-descope:command-palette#onDescope
          command-palette-cleared:command-palette#onInputClear
        " placeholder="Search or jump to...">
       <div class="js-search-icon d-flex flex-items-center mr-2" style="height: 26px">
        <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
         </path>
        </svg>
       </div>
       <div class="js-spinner d-flex flex-items-center mr-2 color-fg-muted" hidden="">
        <svg aria-label="Loading" class="anim-rotate" fill="none" height="16" viewbox="0 0 16 16" width="16">
         <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
         </circle>
         <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
         </path>
        </svg>
       </div>
       <command-palette-scope>
        <div class="color-fg-subtle" data-target="command-palette-scope.placeholder" hidden="">
         /
         <span class="text-semibold color-fg-default">
          ...
         </span>
         /
        </div>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDQ6VXNlcjM3OTA2NTY3" data-targets="command-palette-scope.tokens" data-text="lorenapr92" data-type="owner" data-value="lorenapr92" style="white-space:nowrap;line-height:20px;">
         lorenapr92
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDEwOlJlcG9zaXRvcnkzMzE5MjM1ODg=" data-targets="command-palette-scope.tokens" data-text="21SPStephGAdvanceGenomicslog" data-type="repository" data-value="21SPStephGAdvanceGenomicslog" style="white-space:nowrap;line-height:20px;">
         21SPStephGAdvance...
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
       </command-palette-scope>
       <div class="command-palette-input-group flex-1 form-control border-0 box-shadow-none" style="z-index: 0">
        <div class="command-palette-typeahead position-absolute d-flex flex-items-center Truncate">
         <span class="typeahead-segment input-mirror" data-target="command-palette-input.mirror">
         </span>
         <span class="Truncate-text" data-target="command-palette-input.typeaheadText">
         </span>
         <span class="typeahead-segment" data-target="command-palette-input.typeaheadPlaceholder">
         </span>
        </div>
        <input aria-label="Hidden input for typeahead" class="js-overlay-input typeahead-input d-none" disabled="" tabindex="-1"/>
        <input aria-autocomplete="list" aria-controls="command-palette-page-stack" aria-expanded="false" aria-haspopup="listbox" aria-label="Command palette input" autocapitalize="off" autocomplete="off" autocorrect="off" class="js-input typeahead-input form-control border-0 box-shadow-none input-block width-full no-focus-indicator" data-action="
              input:command-palette-input#onInput
              keydown:command-palette-input#onKeydown
            " role="combobox" spellcheck="false" type="text"/>
       </div>
       <div class="position-relative d-inline-block" data-view-component="true">
        <button aria-keyshortcuts="Control+Backspace" class="btn-octicon command-palette-input-clear-button" data-action="click:command-palette-input#onClear keypress:command-palette-input#onClear" data-target="command-palette-input.clearButton" data-view-component="true" hidden="hidden" id="command-palette-clear-button" type="button">
         <svg aria-hidden="true" class="octicon octicon-x-circle-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
          </path>
         </svg>
        </button>
        <tool-tip class="sr-only position-absolute" data-direction="w" data-type="label" data-view-component="true" for="command-palette-clear-button" id="tooltip-28cf650f-ac00-438d-8b77-65f8c9dff81e" popover="manual">
         Clear Command Palette
        </tool-tip>
       </div>
      </command-palette-input>
      <command-palette-page-stack data-action="command-palette-page-octicons-cached:command-palette-page-stack#cacheOcticons" data-default-scope-id="MDEwOlJlcG9zaXRvcnkzMzE5MjM1ODg=" data-default-scope-type="Repository">
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search discussions
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           !
          </kbd>
          to search projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search teams
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='[""]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search people and organizations
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           &gt;
          </kbd>
          to activate command mode
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Go to your accessibility settings to change your keyboard shortcuts
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type author:@me to search your content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:pr to filter to pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:issue to filter to issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:project to filter to projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:open to filter to open content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="mx-3 my-2 flash flash-error d-flex flex-items-center" data-on-error="" data-scope-types="*">
        <div>
         <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
          </path>
         </svg>
        </div>
        <div class="px-2">
         Weâve encountered an error and some results aren't available at this time. Type a new search or try again later.
        </div>
       </command-palette-tip>
       <command-palette-tip class="h4 color-fg-default pl-3 pb-2 pt-3" data-match-mode="[^?]|^$" data-on-empty="" data-scope-types="*">
        No results matched your search
       </command-palette-tip>
       <div hidden="">
        <div data-octicon-id="arrow-right-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="arrow-right-color-fg-default" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-default" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="codespaces-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
          </path>
          <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="copy-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-copy color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
          </path>
          <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="dash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-dash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 7.75A.75.75 0 0 1 2.75 7h10a.75.75 0 0 1 0 1.5h-10A.75.75 0 0 1 2 7.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="file-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-file color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="gear-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-gear color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="lock-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-lock color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M4 4a4 4 0 0 1 8 0v2h.25c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 12.25 15h-8.5A1.75 1.75 0 0 1 2 13.25v-5.5C2 6.784 2.784 6 3.75 6H4Zm8.25 3.5h-8.5a.25.25 0 0 0-.25.25v5.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25ZM10.5 6V4a2.5 2.5 0 1 0-5 0v2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="moon-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-moon color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M9.598 1.591a.749.749 0 0 1 .785-.175 7.001 7.001 0 1 1-8.967 8.967.75.75 0 0 1 .961-.96 5.5 5.5 0 0 0 7.046-7.046.75.75 0 0 1 .175-.786Zm1.616 1.945a7 7 0 0 1-7.678 7.678 5.499 5.499 0 1 0 7.678-7.678Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="person-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-person color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="pencil-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-pencil color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="issue-opened-open" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-issue-opened open" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="git-pull-request-draft-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request-draft color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M3.25 1A2.25 2.25 0 0 1 4 5.372v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.251 2.251 0 0 1 3.25 1Zm9.5 14a2.25 2.25 0 1 1 0-4.5 2.25 2.25 0 0 1 0 4.5ZM2.5 3.25a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0ZM3.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm9.5 0a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM14 7.5a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm0-4.25a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="search-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sun-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sun color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8Zm0-1.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Zm5.657-8.157a.75.75 0 0 1 0 1.061l-1.061 1.06a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.06-1.06a.75.75 0 0 1 1.06 0Zm-9.193 9.193a.75.75 0 0 1 0 1.06l-1.06 1.061a.75.75 0 1 1-1.061-1.06l1.06-1.061a.75.75 0 0 1 1.061 0ZM8 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V.75A.75.75 0 0 1 8 0ZM3 8a.75.75 0 0 1-.75.75H.75a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 3 8Zm13 0a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 16 8Zm-8 5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 8 13Zm3.536-1.464a.75.75 0 0 1 1.06 0l1.061 1.06a.75.75 0 0 1-1.06 1.061l-1.061-1.06a.75.75 0 0 1 0-1.061ZM2.343 2.343a.75.75 0 0 1 1.061 0l1.06 1.061a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-1.06-1.06a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sync-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sync color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.705 8.005a.75.75 0 0 1 .834.656 5.5 5.5 0 0 0 9.592 2.97l-1.204-1.204a.25.25 0 0 1 .177-.427h3.646a.25.25 0 0 1 .25.25v3.646a.25.25 0 0 1-.427.177l-1.38-1.38A7.002 7.002 0 0 1 1.05 8.84a.75.75 0 0 1 .656-.834ZM8 2.5a5.487 5.487 0 0 0-4.131 1.869l1.204 1.204A.25.25 0 0 1 4.896 6H1.25A.25.25 0 0 1 1 5.75V2.104a.25.25 0 0 1 .427-.177l1.38 1.38A7.002 7.002 0 0 1 14.95 7.16a.75.75 0 0 1-1.49.178A5.5 5.5 0 0 0 8 2.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="trash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-trash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="key-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-key color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="comment-discussion-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-slash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell-slash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="paintbrush-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-paintbrush color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.134 1.535c.7-.509 1.416-.942 2.076-1.155.649-.21 1.463-.267 2.069.34.603.601.568 1.411.368 2.07-.202.668-.624 1.39-1.125 2.096-1.011 1.424-2.496 2.987-3.775 4.249-1.098 1.084-2.132 1.839-3.04 2.3a3.744 3.744 0 0 1-1.055 3.217c-.431.431-1.065.691-1.657.861-.614.177-1.294.287-1.914.357A21.151 21.151 0 0 1 .797 16H.743l.007-.75H.749L.742 16a.75.75 0 0 1-.743-.742l.743-.008-.742.007v-.054a21.25 21.25 0 0 1 .13-2.284c.067-.647.187-1.287.358-1.914.17-.591.43-1.226.86-1.657a3.746 3.746 0 0 1 3.227-1.054c.466-.893 1.225-1.907 2.314-2.982 1.271-1.255 2.833-2.75 4.245-3.777ZM1.62 13.089c-.051.464-.086.929-.104 1.395.466-.018.932-.053 1.396-.104a10.511 10.511 0 0 0 1.668-.309c.526-.151.856-.325 1.011-.48a2.25 2.25 0 1 0-3.182-3.182c-.155.155-.329.485-.48 1.01a10.515 10.515 0 0 0-.309 1.67Zm10.396-10.34c-1.224.89-2.605 2.189-3.822 3.384l1.718 1.718c1.21-1.205 2.51-2.597 3.387-3.833.47-.662.78-1.227.912-1.662.134-.444.032-.551.009-.575h-.001V1.78c-.014-.014-.113-.113-.548.027-.432.14-.995.462-1.655.942Zm-4.832 7.266-.001.001a9.859 9.859 0 0 0 1.63-1.142L7.155 7.216a9.7 9.7 0 0 0-1.161 1.607c.482.302.889.71 1.19 1.192Z">
          </path>
         </svg>
        </div>
        <command-palette-item-group data-default-priority="0" data-group-hint="" data-group-id="top" data-group-limits="{}" data-group-title="Top result">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="1" data-group-hint="Type &gt; to filter" data-group-id="commands" data-group-limits='{"static_items_page":50,"issue":50,"pull_request":50,"discussion":50}' data-group-title="Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="2" data-group-hint="Type &gt; to filter" data-group-id="global_commands" data-group-limits='{"issue":0,"pull_request":0,"discussion":0}' data-group-title="Global Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="3" data-group-hint="" data-group-id="this_page" data-group-limits="{}" data-group-title="This Page">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="4" data-group-hint="" data-group-id="files" data-group-limits="{}" data-group-title="Files">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="5" data-group-hint="" data-group-id="default" data-group-limits='{"static_items_page":50}' data-group-title="Default">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="6" data-group-hint="" data-group-id="pages" data-group-limits='{"repository":10}' data-group-title="Pages">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="7" data-group-hint="" data-group-id="access_policies" data-group-limits="{}" data-group-title="Access Policies">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="8" data-group-hint="" data-group-id="organizations" data-group-limits="{}" data-group-title="Organizations">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="9" data-group-hint="" data-group-id="repositories" data-group-limits="{}" data-group-title="Repositories">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="10" data-group-hint="Type # to filter" data-group-id="references" data-group-limits="{}" data-group-title="Issues, pull requests, and discussions">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="11" data-group-hint="" data-group-id="teams" data-group-limits="{}" data-group-title="Teams">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="12" data-group-hint="" data-group-id="users" data-group-limits="{}" data-group-title="Users">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="13" data-group-hint="" data-group-id="memex_projects" data-group-limits="{}" data-group-title="Projects">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="14" data-group-hint="" data-group-id="projects" data-group-limits="{}" data-group-title="Projects (classic)">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="15" data-group-hint="" data-group-id="footer" data-group-limits="{}" data-group-title="Footer">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="16" data-group-hint="" data-group-id="modes_help" data-group-limits="{}" data-group-title="Modes">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="17" data-group-hint="" data-group-id="filters_help" data-group-limits="{}" data-group-title="Use filters in issues, pull requests, discussions, and projects">
        </command-palette-item-group>
        <command-palette-page data-page-title="lorenapr92" data-scope-id="MDQ6VXNlcjM3OTA2NTY3" data-scope-type="owner" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
        <command-palette-page data-page-title="21SPStephGAdvanceGenomicslog" data-scope-id="MDEwOlJlcG9zaXRvcnkzMzE5MjM1ODg=" data-scope-type="repository" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
       </div>
       <command-palette-page data-is-root="">
       </command-palette-page>
       <command-palette-page data-page-title="lorenapr92" data-scope-id="MDQ6VXNlcjM3OTA2NTY3" data-scope-type="owner">
       </command-palette-page>
       <command-palette-page data-page-title="21SPStephGAdvanceGenomicslog" data-scope-id="MDEwOlJlcG9zaXRvcnkzMzE5MjM1ODg=" data-scope-type="repository">
       </command-palette-page>
      </command-palette-page-stack>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="search-links">
      </server-defined-provider>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="help">
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues
         </strong>
         and
         <strong>
          pull requests
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues, pull requests, discussions,
         </strong>
         and
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="@" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          organizations, repositories,
         </strong>
         and
         <strong>
          users
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          @
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="!" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          !
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="/" data-scope-types='["repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          files
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          /
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="&gt;">
        <span data-target="command-palette-help.titleElement">
         Activate
         <strong>
          command mode
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          &gt;
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:pr">
        <span data-target="command-palette-help.titleElement">
         Filter to pull requests
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:pr
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:issue">
        <span data-target="command-palette-help.titleElement">
         Filter to issues
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:issue
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:discussion" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:discussion
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:project" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to projects
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:project
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:open">
        <span data-target="command-palette-help.titleElement">
         Filter to open issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:open
         </kbd>
        </span>
       </command-palette-help>
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/commands" data-supported-modes="[]" data-supports-commands="" data-targets="command-palette.serverDefinedProviderElements" data-type="commands">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_page_navigation" data-supported-modes='[""]' data-supported-scope-types='["","owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to" data-supported-modes='["@","@"]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to_members_only" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_members_only_prefetched" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/files" data-supported-modes='["/"]' data-supported-scope-types='["repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="files">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/discussions" data-supported-modes='["#"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/projects" data-supported-modes='["#","!"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/recent_issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/teams" data-supported-modes='["@",""]' data-supported-scope-types='["owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/name_with_owner_repository" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
     </command-palette>
    </details-dialog>
   </details>
   <div class="position-fixed bottom-0 left-0 ml-5 mb-5 js-command-palette-toasts" style="z-index: 1000">
    <div class="Toast Toast--loading" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="Toast--spinner" height="18" viewbox="0 0 32 32" width="18">
       <path d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4" fill="#959da5">
       </path>
       <path d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z" fill="#ffffff">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--error" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-stop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M4.47.22A.749.749 0 0 1 5 0h6c.199 0 .389.079.53.22l4.25 4.25c.141.14.22.331.22.53v6a.749.749 0 0 1-.22.53l-4.25 4.25A.749.749 0 0 1 11 16H5a.749.749 0 0 1-.53-.22L.22 11.53A.749.749 0 0 1 0 11V5c0-.199.079-.389.22-.53Zm.84 1.28L1.5 5.31v5.38l3.81 3.81h5.38l3.81-3.81V5.31L10.69 1.5ZM8 4a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 4Zm0 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--warning" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--success" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-info" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.5 7.75A.75.75 0 0 1 7.25 7h1a.75.75 0 0 1 .75.75v2.75h.25a.75.75 0 0 1 0 1.5h-2a.75.75 0 0 1 0-1.5h.25v-2h-.25a.75.75 0 0 1-.75-.75ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
   </div>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <template class="js-user-list-create-dialog-template" data-label="Create list">
       <div class="Box-header">
        <h2 class="Box-title">
         Create list
        </h2>
       </div>
       <!-- '"` -->
       <!-- </textarea></xmp> -->
       <form accept-charset="UTF-8" action="/stars/ninenia/lists" class="Box-body d-flex flex-column p-3 js-user-list-form" data-turbo="false" method="post">
        <input autocomplete="off" name="authenticity_token" type="hidden" value="ChyUBGoYP5J3B_lN4YCY1TeSPUL3RKF5JeGXCHNlmuHacD4g_8LpNZQ6UypyWAFzkpH2N3NjiEW7S7K9x7SMVg">
         <p class="color-fg-subtle mb-3">
          Create a list to organize your starred repositories.
         </p>
         <input name="repository_id" type="hidden" value="{{ repositoryId }}"/>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <auto-check required="" src="/stars/ninenia/list-check?attr=name">
           <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
            <input aria-label="List name" autofocus="" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="32" maxlength="32" name="user_list[name]" placeholder="â­ï¸ Name this list" required="" type="text" value="{{ placeholderName }}"/>
           </text-expander>
           <input data-csrf="true" type="hidden" value="Y4aGvK-u0LAv9ZXEJV1UDZstJyZPY6uqT-fdz1r0OKUDrtpNYICC1Pm-CoDGX5A7msd5V-ORFZDjzacGuJyMCw">
           </input>
          </auto-check>
          <p class="note error position-relative js-user-list-error" hidden="">
           Name .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           32 remaining
          </p>
         </div>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
           <textarea aria-label="List description" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="160" maxlength="160" name="user_list[description]" placeholder="Write a description" style="height: 74px; min-height: 74px"></textarea>
          </text-expander>
          <p class="note error position-relative js-user-list-error" hidden="">
           Description .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           160 remaining
          </p>
         </div>
         <div class="js-user-list-base flash flash-error mx-0 mt-0 mb-2" data-generic-message="Unable to save your list at this time." data-view-component="true" hidden="hidden">
          .
         </div>
         <button class="Button--primary Button--medium Button Button--fullWidth mt-2" data-disable-invalid="true" data-submitting-message="Creating..." data-view-component="true" disabled="disabled" type="submit">
          <span class="Button-content">
           <span class="Button-label">
            Create
           </span>
          </span>
         </button>
         <p class="note mt-2 mb-0">
          <strong>
           Tip:
          </strong>
          type
          <code>
           :
          </code>
          to add emoji to the name or description.
         </p>
        </input>
       </form>
       <div class="Box-footer Box-row--gray text-small color-fg-muted d-flex flex-items-baseline py-2" data-view-component="true">
        <span aria-label="Feature Release Label: Beta" class="Label Label--success Label--inline px-2 mr-2" data-view-component="true" title="Feature Release Label: Beta">
         Beta
        </span>
        <span class="mr-1">
         Lists are currently in beta.
        </span>
        <a href="https://github.com//github/feedback/discussions/categories/lists">
         Share feedback and report bugs.
        </a>
       </div>
      </template>
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header">
       <div class="d-flex flex-wrap flex-justify-end mb-3 container-xl px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal" id="repo-title-component">
          <img alt="Owner avatar" class="avatar mr-2 d-none d-md-block avatar-user" height="24" src="https://avatars.githubusercontent.com/u/37906567?s=48&amp;v=4" width="24"/>
          <strong class="mr-2 flex-self-stretch d-none d-md-block" itemprop="name">
           <a class="color-fg-default" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/lorenapr92/21SPStephGAdvanceGenomicslog">
            21SPStephGAdvanceGenomicslog
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1 d-none d-md-block">
           Public
          </span>
         </div>
         <div class="d-none d-md-block">
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <include-fragment loading="lazy" src="/notifications/331923588/watch_subscription?aria_id_prefix=repository-details&amp;button_block=false&amp;show_count=true">
            <div data-hide-on-error="">
             <button aria-label="Watching a repository" class="tooltipped tooltipped-n Button--secondary Button--small Button" data-view-component="true" disabled="disabled" type="button">
              <span class="Button-content">
               <span class="Button-visual Button-leadingVisual">
                <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                 </path>
                </svg>
               </span>
               <span class="Button-label">
                Watch
               </span>
              </span>
             </button>
            </div>
            <p data-show-on-error="" hidden="">
             <span class="d-flex pt-1">
              <svg aria-hidden="true" class="octicon octicon-alert mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
               </path>
              </svg>
              Couldn't load subscription status.
              <button class="Button--link Button--medium Button" data-retry-button="" data-view-component="true" type="button">
               <span class="Button-content">
                <span class="Button-label">
                 Retry
                </span>
               </span>
              </button>
             </span>
            </p>
           </include-fragment>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a class="btn-sm btn BtnGroup-item" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":331923588,"originating_url":"https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog","user_id":126681664}}' data-hydro-click-hmac="84e87093f7ea839c0f5a7ae5cfa41d6b9e97228317a7a7a39229119f69080907" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/fork" icon="repo-forked" id="fork-button">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             Fork
             <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
              0
             </span>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="fork-button" id="tooltip-5ae2b732-3d59-45e6-8ee2-849b6c2f60d7" popover="manual">
              Fork your own copy of lorenapr92/21SPStephGAdvanceGenomicslog
             </tool-tip>
            </a>
            <details class="details-reset details-overlay BtnGroup-parent d-inline-block position-relative" data-view-component="true" group_item="true" id="my-forks-menu-331923588">
             <summary aria-label="See your forks of this repository" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
              <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
               </path>
              </svg>
             </summary>
             <details-menu class="SelectMenu right-0" role="menu" src="/lorenapr92/21SPStephGAdvanceGenomicslog/my_forks_menu_content?can_fork=true">
              <div class="SelectMenu-modal">
               <button aria-label="Close menu" class="SelectMenu-closeButton position-absolute right-0 m-2" data-toggle-for="my-forks-menu-331923588" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
               <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-908f40">
                <div class="SelectMenu-list">
                 <include-fragment aria-label="Loading" class="SelectMenu-loading">
                  <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                   <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                   </circle>
                   <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                   </path>
                  </svg>
                 </include-fragment>
                </div>
               </div>
              </div>
             </details-menu>
            </details>
           </div>
          </li>
          <li>
           <template class="js-unstar-confirmation-dialog-template">
            <div class="Box-header">
             <h2 class="Box-title">
              Unstar this repository?
             </h2>
            </div>
            <div class="Box-body">
             <p class="mb-3">
              This will remove {{ repoNameWithOwner }} from the {{ listsWithCount }} that it's been added to.
             </p>
             <div class="form-actions">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="{{ confirmUrl }}" class="js-social-confirmation-form" data-turbo="false" method="post">
               <input name="authenticity_token" type="hidden" value="{{ confirmCsrfToken }}"/>
               <input name="confirm" type="hidden" value="true"/>
               <button class="btn-danger btn width-full" data-close-dialog="true" data-view-component="true" type="submit">
                Unstar
               </button>
              </form>
             </div>
            </div>
           </template>
           <div class="js-toggler-container js-social-container starring-container d-flex" data-view-component="true">
            <div class="starred BtnGroup flex-1 ml-0" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/lorenapr92/21SPStephGAdvanceGenomicslog/unstar" class="js-social-form BtnGroup-parent flex-auto js-deferred-toggler-target" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="DX2A-CJlJfL9vK_sU7wU-fm6RoHiGzsh-pRiVb0SZIIeyxKa-iT9ZQMC4AG-M2g8ELxIliTnphCmzZVLRI8lKg">
               <input class="js-confirm-csrf-token" data-csrf="true" type="hidden" value="NFCJM9jE7uiUf-LUsREwBoeRlfeoHsqcXaINo41Wcdon5htRAIU2f2rBrTlcnkzDbpeb4G7iV60B-_q9dMswcg">
                <input name="context" type="hidden" value="repository"/>
                <button aria-label="Unstar this repository (0)" class="rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":331923588,"originating_url":"https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog","user_id":126681664}}' data-hydro-click-hmac="f03b196c98be0522815a6353496874035d641100e1f93cc3e67578be94347b3f" data-view-component="true" type="submit">
                 <svg aria-hidden="true" class="octicon octicon-star-fill starred-button-icon d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                  </path>
                 </svg>
                 <span class="d-inline" data-view-component="true">
                  Starred
                 </span>
                 <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-unstar" title="0">
                  0
                 </span>
                </button>
               </input>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-331923588-starred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/lorenapr92/21SPStephGAdvanceGenomicslog/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-331923588-starred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-265941">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
            <div class="unstarred BtnGroup ml-0 flex-1" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/lorenapr92/21SPStephGAdvanceGenomicslog/star" class="js-social-form BtnGroup-parent flex-auto" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="825F7qXWQ7dw8YllKcFztY4vIgCMY3HDtqocl5nsKrDkhl_O5D0FpFDajdR7wWVYgQXXQbi4SFfXCxWrbEmUZw">
               <input name="context" type="hidden" value="repository"/>
               <button aria-label="Star this repository (0)" class="js-toggler-target rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":331923588,"originating_url":"https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog","user_id":126681664}}' data-hydro-click-hmac="9bc4d73d0efef14b1d1968e26ffbf647ae9f1c356b0217dfe5fed2e792a7d8ae" data-view-component="true" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
                <span class="d-inline" data-view-component="true">
                 Star
                </span>
                <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
                 0
                </span>
               </button>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-331923588-unstarred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/lorenapr92/21SPStephGAdvanceGenomicslog/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-331923588-unstarred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-5c9b2b">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div class="container-xl">
        <div data-turbo-replace="" id="responsive-meta-container">
         <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
          <div class="d-flex gap-2 mt-n3 mb-3 flex-wrap">
           <div class="d-flex flex-row gap-2">
            <notifications-list-subscription-form data-action="notifications-dialog-label-toggled:notifications-list-subscription-form#handleDialogLabelToggle">
             <details class="details-reset details-overlay details-overlay-dark" data-action="toggle:notifications-list-subscription-form#detailsToggled" data-target="notifications-list-subscription-form.details">
              <summary aria-label="Notifications" class="Button--iconOnly Button--secondary Button--medium Button p-0" data-ga-click="Repository, click Watch settings, action:files#disambiguate" data-hydro-click='{"event_type":"repository.click","payload":{"target":"WATCH_BUTTON","repository_id":331923588,"originating_url":"https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog","user_id":126681664}}' data-hydro-click-hmac="aadfb8746bd87298559f301084a22ce42e9c6e7e316718868381898c943bc3c0" data-view-component="true" id="notifications-button">
               <span class="Button-content">
                <span class="Button-label">
                 <span data-menu-button="">
                  <span data-target="notifications-list-subscription-form.unwatchButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.stopIgnoringButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-bell-slash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.watchButtonCopy">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                 </span>
                </span>
               </span>
              </summary>
              <details-dialog aria-label="Notification settings" class="anim-fade-in fast Box Box--overlay">
               <div data-target="notifications-list-subscription-form.menu">
                <div class="Box-header">
                 <button aria-label="Close dialog" class="Box-btn-octicon btn-octicon float-right" data-action="click:notifications-list-subscription-form#closeMenu" data-close-dialog="" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                 <h3 class="Box-title">
                  Notifications
                 </h3>
                </div>
                <div class="SelectMenu-list mb-0" role="menu">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitForm" data-target="notifications-list-subscription-form.form" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="w4_EmGflM5yGmbuV5peJI6fUkUHCdBe_lmqkO8VTbajbT8jieVpGnWqKjJ86dw9U-PECVBcuvGu0kVbPkuLTMQ">
                   <input name="repository_id" type="hidden" value="331923588"/>
                   <button aria-checked="true" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="included">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Participating and @mentions
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Only receive notifications from this repository when participating or @mentioned.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="subscribed">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      All Activity
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Notified of all notifications on this repository.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="ignore">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Ignore
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Never be notified.
                     </div>
                    </div>
                   </button>
                  </input>
                 </form>
                 <button aria-checked="false" aria-haspopup="true" class="SelectMenu-item flex-items-start pr-3" data-action="click:notifications-list-subscription-form#openCustomDialog" data-target="notifications-list-subscription-form.customButton" role="menuitemradio" type="button">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                    </path>
                   </svg>
                  </span>
                  <div>
                   <div class="d-flex flex-items-start flex-justify-between">
                    <div class="f5 text-bold">
                     Custom
                    </div>
                    <div class="f5 pr-1">
                     <svg aria-hidden="true" class="octicon octicon-arrow-right" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </div>
                   <div class="text-small color-fg-muted text-normal pb-1">
                    Select events you want to be notified of in addition to participating and @mentions.
                   </div>
                  </div>
                 </button>
                 <div class="px-3 py-2 d-flex color-bg-subtle flex-items-center">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile SelectMenu-icon SelectMenu-icon--device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </span>
                  <span classname="text-small color-fg-muted text-normal pb-1">
                   Get push notifications on
                   <a href="https://apps.apple.com/app/apple-store/id1477376905?ct=watch-dropdown&amp;mt=8&amp;pt=524675" rel="noopener noreferrer" target="_blank">
                    iOS
                   </a>
                   or
                   <a href="https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dwatch-dropdown%26utm_medium%3Dweb%26utm_source%3Dgithub" rel="noopener noreferrer" target="_blank">
                    Android
                   </a>
                   .
                  </span>
                 </div>
                </div>
               </div>
               <div data-target="notifications-list-subscription-form.customDialog" hidden="">
                <header class="SelectMenu-header pb-0 border-bottom-0 px-2 px-sm-3">
                 <h1 class="f3 SelectMenu-title d-inline-flex">
                  <button aria-label="Return to menu" class="color-bg-default border-0 px-2 py-0 m-0 Link--secondary f5" data-action="click:notifications-list-subscription-form#closeCustomDialog" type="button">
                   <svg aria-hidden="true" class="octicon octicon-arrow-left" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z">
                    </path>
                   </svg>
                  </button>
                  Custom
                 </h1>
                </header>
                <div class="overflow-visible">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitCustomForm" data-target="notifications-list-subscription-form.customform" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="8olIHa397hHfZUn0c68LC7HLUHAogCbgXFvoyOy7Cn3qSURns0KbEDN2fv6vT4187u7DZf3ajTR-oBo8uwq05A">
                   <input name="repository_id" type="hidden" value="331923588"/>
                   <fieldset>
                    <legend>
                     <div class="text-small color-fg-muted pt-0 pr-3 pb-3 pl-6 pl-sm-5 border-bottom mb-3">
                      Select events you want to be notified of in addition to participating and @mentions.
                     </div>
                    </legend>
                    <div data-target="notifications-list-subscription-form.labelInputs">
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Issue"/>
                      Issues
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="PullRequest"/>
                      Pull requests
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Release"/>
                      Releases
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input aria-describedby="Discussion-disabled" aria-disabled="true" data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Discussion"/>
                      Discussions
                     </label>
                     <div class="color-fg-muted" id="Discussion-disabled">
                      &gt;
                          Discussions are not enabled for this repository
                     </div>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="SecurityAlert"/>
                      Security alerts
                     </label>
                    </div>
                   </fieldset>
                   <div class="pt-2 pb-3 px-3 d-flex flex-justify-start flex-row-reverse">
                    <button class="Button--primary Button--medium Button ml-2" data-target="notifications-list-subscription-form.customSubmit" data-view-component="true" disabled="disabled" name="do" type="submit" value="custom">
                     <span class="Button-content">
                      <span class="Button-label">
                       Apply
                      </span>
                     </span>
                    </button>
                    <button class="Button--secondary Button--medium Button" data-action="click:notifications-list-subscription-form#resetForm" data-close-dialog="" data-view-component="true" type="button">
                     <span class="Button-content">
                      <span class="Button-label">
                       Cancel
                      </span>
                     </span>
                    </button>
                   </div>
                  </input>
                 </form>
                </div>
               </div>
              </details-dialog>
             </details>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="notifications-button" id="tooltip-89854f17-6f9c-4650-953d-990ff91ee2e1" popover="manual">
              Notification settings
             </tool-tip>
            </notifications-list-subscription-form>
            <div class="BtnGroup d-flex" data-view-component="true">
             <a aria-labelledby="tooltip-81a1499e-448e-43e7-ac01-478da38832d4" class="Button Button--iconOnly Button--secondary Button--medium" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":331923588,"originating_url":"https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog","user_id":126681664}}' data-hydro-click-hmac="84e87093f7ea839c0f5a7ae5cfa41d6b9e97228317a7a7a39229119f69080907" data-view-component="true" group_item="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/fork" id="fork-icon-button" variant="small">
              <svg aria-hidden="true" class="octicon octicon-repo-forked Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
               </path>
              </svg>
             </a>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="fork-icon-button" id="tooltip-81a1499e-448e-43e7-ac01-478da38832d4" popover="manual">
              Fork your own copy of lorenapr92/21SPStephGAdvanceGenomicslog
             </tool-tip>
            </div>
            <div class="js-toggler-container starring-container" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/lorenapr92/21SPStephGAdvanceGenomicslog/unstar" class="starred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="JIPjz8Jfr4fsqIZMUpZ-bnu-GwXGyerM6iRT_yG6-_o3NXGtGh53EBIWyaG_GQKrkrgVEgA1d_22faTh2Ce6Ug">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-93f164b7-f652-46ea-98ff-89f79d9bcc1f" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target starred-button-icon" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":331923588,"originating_url":"https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog","user_id":126681664}}' data-hydro-click-hmac="f03b196c98be0522815a6353496874035d641100e1f93cc3e67578be94347b3f" data-view-component="true" id="icon-button-ff16bc60-095a-4ec6-a124-eb8651c0de5d" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-ff16bc60-095a-4ec6-a124-eb8651c0de5d" id="tooltip-93f164b7-f652-46ea-98ff-89f79d9bcc1f" popover="manual">
                Unstar this repository
               </tool-tip>
              </input>
             </form>
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/lorenapr92/21SPStephGAdvanceGenomicslog/star" class="unstarred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="yhRp238gW2ZXDk3QdGxPMWUFUf_GnWaMeBUx6CRRaK7d_HP7PssddXclSWEmbFncai-kvvJGXxgZtDjU0fTWeQ">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-d061a123-874f-48b2-910b-0585964f6921" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":331923588,"originating_url":"https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog","user_id":126681664}}' data-hydro-click-hmac="9bc4d73d0efef14b1d1968e26ffbf647ae9f1c356b0217dfe5fed2e792a7d8ae" data-view-component="true" id="icon-button-8b3c5389-89bd-4310-926a-daf55064a8ec" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-8b3c5389-89bd-4310-926a-daf55064a8ec" id="tooltip-d061a123-874f-48b2-910b-0585964f6921" popover="manual">
                Star this repository
               </tool-tip>
              </input>
             </form>
            </div>
           </div>
           <div class="d-flex flex-row gap-2">
           </div>
          </div>
          <div class="mb-3">
           <ul aria-label="Repository details" class="d-flex flex-wrap mb-2 gap-2">
            <a class="Link--secondary no-underline d-block mr-2" href="/lorenapr92/21SPStephGAdvanceGenomicslog/stargazers" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             stars
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/lorenapr92/21SPStephGAdvanceGenomicslog/forks" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             forks
            </a>
            <div class="color-fg-muted mr-2" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-eye mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              1
             </span>
             watching
            </div>
            <a class="Link--secondary no-underline d-block mr-2" href="/lorenapr92/21SPStephGAdvanceGenomicslog/branches" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
              </path>
             </svg>
             <include-fragment src="/lorenapr92/21SPStephGAdvanceGenomicslog/branch-count">
              <span>
               Branches
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/lorenapr92/21SPStephGAdvanceGenomicslog/tags" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
              </path>
             </svg>
             <include-fragment src="/lorenapr92/21SPStephGAdvanceGenomicslog/tag-count">
              <span>
               Tags
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/lorenapr92/21SPStephGAdvanceGenomicslog/activity" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
              </path>
             </svg>
             <span>
              Activity
             </span>
            </a>
           </ul>
           <div class="mb-2 d-flex color-fg-muted">
            <div class="d-flex flex-items-center" style="height: 21px">
             <svg aria-hidden="true" class="octicon octicon-globe flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
              </path>
             </svg>
            </div>
            <span class="flex-auto min-width-0 width-fit">
             Public repository
            </span>
           </div>
          </div>
         </div>
        </div>
        <div class="border-bottom mx-xl-5">
        </div>
       </div>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <a class="d-none js-github-dev-shortcut" data-hotkey=".,Mod+Alt+." href="https://github.dev/">
         Open in github.dev
        </a>
        <a class="d-none js-github-dev-new-tab-shortcut" data-hotkey="Shift+.,Shift+&gt;,&gt;" href="https://github.dev/" rel="noopener noreferrer" target="_blank">
         Open in a new github.dev tab
        </a>
        <a class="d-none" data-hotkey=",,Mod+Alt+," href="/codespaces/new/lorenapr92/21SPStephGAdvanceGenomicslog?resume=1" target="_blank">
         Open in codespace
        </a>
        <h1 class="sr-only">
         lorenapr92/21SPStephGAdvanceGenomicslog
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/lorenapr92/21SPStephGAdvanceGenomicslog/spoofed_commit_check/2f6633bc3d010b35b01eacc5e4eee5ece91c6408">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":true,"path":"/","repo":{"id":331923588,"defaultBranch":"main","name":"21SPStephGAdvanceGenomicslog","ownerLogin":"lorenapr92","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2021-01-22T19:18:15.000+08:00","ownerAvatar":"https://avatars.githubusercontent.com/u/37906567?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":{"id":126681664,"login":"ninenia","userEmail":"2945518981@qq.com"},"refInfo":{"name":"main","listCacheKey":"v0:1611314316.0","canEdit":true,"refType":"branch","currentOid":"2f6633bc3d010b35b01eacc5e4eee5ece91c6408"},"tree":{"items":[{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":1,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/lorenapr92/21SPStephGAdvanceGenomicslog/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":true,"httpUrl":"https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog.git","showCloneWarning":false,"sshUrl":"git@github.com:lorenapr92/21SPStephGAdvanceGenomicslog.git","sshCertificatesRequired":false,"sshCertificatesAvailable":false,"ghCliUrl":"gh repo clone lorenapr92/21SPStephGAdvanceGenomicslog","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"x-github-client://openRepo/https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog","showVisualStudioCloneButton":true,"visualStudioCloneUrl":"git-client://clone?repo=https%3A%2F%2Fgithub.com%2Florenapr92%2F21SPStephGAdvanceGenomicslog","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/lorenapr92/21SPStephGAdvanceGenomicslog/archive/refs/heads/main.zip"}},"repoPolicyInfo":{"allowed":true,"canBill":true,"changesWouldBeSafe":true,"disabledByBusiness":false,"disabledByOrganization":false,"hasIpAllowLists":false},"currentUserIsEnterpriseManaged":false,"enterpriseManagedBusinessName":null,"codespacesEnabled":true,"hasAccessToCodespaces":true},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"17","overviewFiles":[{"displayName":"README.md","repoName":"21SPStephGAdvanceGenomicslog","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cp dir=\"auto\"\u003e#Stephanie Grizzard's README\n#21SPStephGAdvanceGenomicsLog repository\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#Exercise2\ncd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG\nmkdir data\nmkdir exercises\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/exercises\ncp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise2.fast* ./\nls -alh\n13M\ngunzip -c Exercise2.fasta.gz \u0026gt; Exercise2.fasta\nls -alh\n37M\ntar -zxvf Exercise2.fastq.tar.gz\nls -alh\n62M\ndrwxrwxrwx 2 spere004 users  142 Jan 22 06:38 .\ndrwxrwxrwx 5 spere004 users  141 Jan 22 06:34 ..\n-rwxrwxrwx 1 spere004 users  17M Jan 22 06:37 Exercise2.fasta\n-rwxr-xr-x 1 spere004 users 4.3M Jan 22 06:35 Exercise2.fasta.gz\n-rw-r--r-- 1 spere004 users  18M Sep  2  2015 Exercise2.fastq\n-rwxr-xr-x 1 spere004 users 4.3M Jan 22 06:35 Exercise2.fastq.tar.gz\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e[spere004@turing1 exercises]$ head -1 Exercise2.fasta\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003escaffold10078|size20675\n[spere004@turing1 exercises]$ grep -c '\u0026gt;' Exercise2.fasta\n138\n[spere004@turing1 exercises]$ head -1 Exercise2.fastq\n@HS3:541:HAYTUADXX:1:1101:1297:1938 1:N:0:AGTTCC\n[spere004@turing1 exercises]$ grep -c '@HS3' Exercise2.fastq\n61304\n[spere004@turing1 exercises]$ wc -l Exercise2.fastq\n245216 Exercise2.fastq\ncp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/scripts/\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003e[spere004@turing1 scripts]$ salloc\nsalloc: Pending job allocation 9268786\nsalloc: job 9268786 queued and waiting for resources\nsalloc: job 9268786 has been allocated resources\nsalloc: Granted job allocation 9268786\nThis session will be terminated in 7 days. If your application requires\na longer excution time, please use command \"salloc -t N-0\" where N is the\nnumber of days that you need.\ncd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/exercises\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e[spere004@coreV3-23-024 exercises]$ /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/scripts/avg_cov_len_fasta_advbioinf.py Exercise2.fasta\nThe total number of sequences is 138\nThe average sequence length is 126640\nThe total number of bases is 17476447\nThe minimum sequence length is 1122\nThe maximum sequence length is 562680\nThe N50 is 187037\nMedian Length = 92612\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 138\ncontigs \u0026gt;= 1000bp = 138\ncontigs \u0026gt;= 2000bp = 135\u003c/p\u003e\n\u003cp dir=\"auto\"\u003estephanieperez@MacBook-Pro-2 ~/D/D/21sp_advgenomics\u0026gt; cd 21SPStephGAdvancedGenomicsLog/\nstephanieperez@MacBook-Pro-2 ~/D/D/2/21SPStephGAdvancedGenomicsLog\u0026gt; ls\nREADME.md\nstephanieperez@MacBook-Pro-2 ~/D/D/2/21SPStephGAdvancedGenomicsLog\u0026gt;  git add README.md\nstephanieperez@MacBook-Pro-2 ~/D/D/2/21SPStephGAdvancedGenomicsLog\u0026gt;\ngit commit -m 'updating with day02 exercises'\n[main 552e36f] updating with day02 exercises\n1 file changed, 81 insertions(+)\nstephanieperez@MacBook-Pro-2 ~/D/D/2/21SPStephGAdvancedGenomicsLog\u0026gt; git push -u origin main\nEnumerating objects: 5, done.\nCounting objects: 100% (5/5), done.\nDelta compression using up to 8 threads\nCompressing objects: 100% (2/2), done.\nWriting objects: 100% (3/3), 1.37 KiB | 1.38 MiB/s, done.\nTotal 3 (delta 0), reused 0 (delta 0), pack-reused 0\nTo \u003ca href=\"https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog.git\"\u003ehttps://github.com/lorenapr92/21SPStephGAdvanceGenomicslog.git\u003c/a\u003e\ne726f87..552e36f  main -\u0026gt; main\nBranch 'main' set up to track remote branch 'main' from 'origin'.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#Homework2\ncd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/\nemacs emacs StephFQCp.sh\ninside shell script\n#!/bin/bash -l\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#SBATCH -o stephfastqcope.txt\u003cbr\u003e\n#SBATCH -n 1\u003cbr\u003e\n#SBATCH --mail-user=\u003ca href=\"mailto:spere004@odu.edu\"\u003espere004@odu.edu\u003c/a\u003e\u003cbr\u003e\n#SBATCH --mail-type=END\u003cbr\u003e\n#SBATCH --job-name=StephFQCp\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/*.fastq.gz /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esbatch StephFQCp.sh\n#Submitted batch job 9268702\nsqueue -u spere004\nJOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n9268702      main StephFQC spere004  R       0:19      1 coreV3-23-044\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eemacs StephGZfq.sh\ninside shell script\n#!/bin/bash -l\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#SBATCH -o stephgzfastqcopy.txt\u003cbr\u003e\n#SBATCH -n 1\u003cbr\u003e\n#SBATCH --mail-user=\u003ca href=\"mailto:spere004@odu.edu\"\u003espere004@odu.edu\u003c/a\u003e\u003cbr\u003e\n#SBATCH --mail-type=END\u003cbr\u003e\n#SBATCH --job-name=StephGZfq\u003c/p\u003e\n\u003cp dir=\"auto\"\u003egunzip /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/*.fastq.gz\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e[spere004@turing1 data]$ sbatch StephGZfq.sh\nSubmitted batch job 9268787\n[spere004@turing1 data]$ squeue -u spere004\nJOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n9268786      main       sh spere004  R       7:52      1 coreV3-23-024\n9268787      main StephGZf spere004  R       0:25      1 coreV1-22-005\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e[spere004@turing1 data]$ ls\nHADB01-A_S17_L002_R1_001.fastq     HADB03-C_S51_L004_R1_001.fastq.gz  HADB05-F_S86_L006_R1_001.fastq.gz\nHADB01-B_S18_L002_R1_001.fastq     HADB03-D_S52_L004_R1_001.fastq.gz  HADB05-G_S87_L006_R1_001.fastq.gz\nHADB01-C_S19_L002_R1_001.fastq     HADB03-E_S53_L004_R1_001.fastq.gz  HADB05-H_S88_L006_R1_001.fastq.gz\nHADB01-D_S20_L002_R1_001.fastq     HADB03-F_S54_L004_R1_001.fastq.gz  HADB05-I_S89_L006_R1_001.fastq.gz\nHADB01-E_S21_L002_R1_001.fastq     HADB03-G_S55_L004_R1_001.fastq.gz  HADB05-J_S90_L006_R1_001.fastq.gz\nHADB01-F_S22_L002_R1_001.fastq\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eDAY03\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ecp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 ./\npwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003emkdir fastq\nmv *.fastq fastq/\nls\ncd fastq/\nls\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003ecp renamingtable_complete.txt /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/scripts\ncp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py ./\nless renamer_advbioinf.py\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003emodule load python\npython renamer_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/renamingtable_complete.txt /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/*.fastq\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003enano renamer_advbioinf.py\n#!/usr/bin/env python\n####usage renamer.py renamingtable\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ethis script take the entries in the first column of table and renames (mv's) them to files with the nam$\u003c/h4\u003e\u003ca id=\"user-content-this-script-take-the-entries-in-the-first-column-of-table-and-renames-mvs-them-to-files-with-the-nam\" class=\"anchor\" aria-label=\"Permalink: this script take the entries in the first column of table and renames (mv's) them to files with the nam$\" href=\"#this-script-take-the-entries-in-the-first-column-of-table-and-renames-mvs-them-to-files-with-the-nam\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eimport sys\nimport os\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efin=open(sys.argv[1],'r')\nlinecount=0\nfor line in fin:\nlinecount+=1\nif linecount\u0026gt;=2:\ncols=line.rstrip().split('\\t')\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eprint 'mv %s %s' %(cols[0], cols[1])\u003c/h1\u003e\u003ca id=\"user-content-print-mv-s-s-cols0-cols1\" class=\"anchor\" aria-label=\"Permalink: print 'mv %s %s' %(cols[0], cols[1])\" href=\"#print-mv-s-s-cols0-cols1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"            os.popen('mv %s %s' %(cols[0], cols[1]))\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e            os.popen('mv %s %s' %(cols[0], cols[1]))\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003enano Steph_Renamer.sh\n#!/bin/bash -l\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e#SBATCH -o steph_renamercopy.txt\n#SBATCH -n 1\n#SBATCH --mail-user=\u003ca href=\"mailto:spere004@odu.edu\"\u003espere004@odu.edu\u003c/a\u003e\n#SBATCH --mail-type=END\n#SBATCH --job-name=Steph_Renamer\u003c/p\u003e\n\u003cp dir=\"auto\"\u003epython /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/scripts/renamer_advbioinf.py renamingtable_complete.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esbatch Steph_Renamer.sh\nSubmitted batch job 9271073\u003c/p\u003e\n\u003col start=\"7\" dir=\"auto\"\u003e\n\u003cli\u003eDONE\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e[spere004@turing1 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/scripts\n10. cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py ./\n11. less Trimclipfilterstatsbatch_advbioinf.py\n12. pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03\n[spere004@turing1 day03]$ cp adapterlist_advbioinf.txt /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/\n13. [spere004@turing1 fastq]$ nano Steph_Trimclipfilter.sh\n#!/bin/bash -l\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#SBATCH -o steph_trimmerclip.txt\n#SBATCH -n 4\n#SBATCH --mail-user=\u003ca href=\"mailto:spere004@odu.edu\"\u003espere004@odu.edu\u003c/a\u003e\n#SBATCH --mail-type=END\n#SBATCH --job-name=Steph_Trimmerclipfilter\u003c/p\u003e\n\u003cp dir=\"auto\"\u003epython /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e[spere004@turing1 fastq]$ sbatch Steph_Trimclipfilter.sh\nSubmitted batch job 9271091\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHomework04\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e1a.  [spere004@turing1 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h\nWritten by Peter Schafran \u003ca href=\"mailto:pscha005@odu.edu\"\u003epscha005@odu.edu\u003c/a\u003e 5-Oct-2015\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis script takes a stats output file from fastx_clipper and converts it into a table.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eUsage: Schafran_trimstatstable.py [-c, -v, -h] inputfile.txt outputfile.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eOptions (-c and -v must be listed separately to run together):\n-h\tDisplay this help message\n-c\tUse comma delimiter instead of tabs\n-v\tVerbose mode (print steps to stdout)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e1b.   pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/filteringstats\n/cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt Steph_trimclipstatsout.txt\nhead -10 Steph_trimclipstatsout.txt\ntail -n +2 Steph_trimclipstatsout.txt | head\n1c.  tail -n +2 Steph_trimclipstatsout.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e[spere004@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs\n3. nano Steph_clippedtrimmedbowtie.sh\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#!/bin/bash -l\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#SBATCH -o steph_clippedtrimmedbowtie.txt\n#SBATCH -n 1\n#SBATCH --mail-user=\u003ca href=\"mailto:spere004@odu.edu\"\u003espere004@odu.edu\u003c/a\u003e\n#SBATCH --mail-type=END\n#SBATCH --job-name=Steph_clippedtrimmedbowtie\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eenable_lmod\nmodule load bowtie2/2.4\nfor i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq} \u003cbr\u003e\n--rg SM:${i%_clippedtrimmed.fastq} \u003cbr\u003e\n--very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U $i \\\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e${i%_clippedtrimmedfilterd.fastq}.sam --no-unal -k 5; done\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003esbatch Steph_clippedtrimmedbowtie.sh\nSubmitted batch job 9271724\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eHomework05\n#Working with John. He copied his fastq files into my sandbox. Created a list with his fast names for shell script\npwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs\nls *.fastq | head -32\nnano Steph_trinity.sh\n#! /bin/bash -l\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#SBATCH -o steph_trinity.txt\n#SBATCH -n 32\n#SBATCH -p himem\n#SBATCH --mail-user=\u003ca href=\"mailto:spere004@odu.edu\"\u003espere004@odu.edu\u003c/a\u003e\n#SBATCH --mail-type=END\n#SBATCH --job-name=Steph_trinity\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eenable_lmod\nmodule load container_env trinity\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecrun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_05_18_clippedtrimmed.fastq,RI_B_05_22_clippedtrimmed.fastq,RI_B_06_14_clippedtrimmed.fastq,RI_B_06_22_clippedtrimmed.fastq,RI_W_05_18_clippedtrimmed.fastq,RI_W_05_22_clippedtrimmed.fastq,RI_W_06_14_clippedtrimmed.fastq,RI_W_06_22_clippedtrimmed.fastq,VA_B_05_18_clippedtrimmed.fastq,VA_B_05_22_clippedtrimmed.fastq,VA_B_06_14_clippedtrimmed.fastq,VA_B_06_22_clippedtrimmed.fastq,VA_W_05_18_clippedtrimmed.fastq,VA_W_05_22_clippedtrimmed.fastq,VA_W_06_14_clippedtrimmed.fastq,VA_W_06_22_clippedtrimmed.fastq,RI_B_02_14_clippedtrimmed.fastq,RI_B_02_18_clippedtrimmed.fastq,RI_B_02_22_clippedtrimmed.fastq,RI_B_09_SNP_clippedtrimmed.fastq,RI_W_02_14_clippedtrimmed.fastq,RI_W_02_18_clippedtrimmed.fastq,RI_W_02_22_clippedtrimmed.fastq,RI_W_09_SNP_clippedtrimmed.fastq,VA_B_02_14_clippedtrimmed.fastq,VA_B_02_18_clippedtrimmed.fastq,VA_B_02_22_clippedtrimmed.fastq,VA_B_08_SNP_clippedtrimmed.fastq,VA_W_02_14_clippedtrimmed.fastq,VA_W_02_18_clippedtrimmed.fastq,VA_W_02_22_clippedtrimmed.fastq,VA_W_09_SNP_clippedtrimmed.fastq --CPU 32\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esbatch Steph_trinity.sh\nSubmitted batch job 9272350\nFAILED\nsbatch Steph_trinity.sh\nSubmitted batch job 9272374\n#w/o --normalized reads\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#Homework06\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e[spere004@turing1 QCFastqs]$ salloc\nsalloc: Pending job allocation 9272855\n[spere004@coreV2-22-028 QCFastqs]$ cd djbtestassembly/\n[spere004@coreV2-22-028 djbtestassembly]$ ls\nTrinity.fasta\n/cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta\n################################\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCounts of transcripts, etc.\u003c/h2\u003e\u003ca id=\"user-content-counts-of-transcripts-etc\" class=\"anchor\" aria-label=\"Permalink: Counts of transcripts, etc.\" href=\"#counts-of-transcripts-etc\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e################################\nTotal trinity 'genes':\t20980\nTotal trinity transcripts:\t21992\nPercent GC: 46.21\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e########################################\nStats based on ALL transcript contigs:\n########################################\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"Contig N10: 1178\nContig N20: 694\nContig N30: 514\nContig N40: 414\nContig N50: 347\n\nMedian contig length: 273\nAverage contig: 356.95\nTotal assembled bases: 7850006\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eContig N10: 1178\nContig N20: 694\nContig N30: 514\nContig N40: 414\nContig N50: 347\n\nMedian contig length: 273\nAverage contig: 356.95\nTotal assembled bases: 7850006\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e#####################################################\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eStats based on ONLY LONGEST ISOFORM per 'GENE':\u003c/h2\u003e\u003ca id=\"user-content-stats-based-on-only-longest-isoform-per-gene\" class=\"anchor\" aria-label=\"Permalink: Stats based on ONLY LONGEST ISOFORM per 'GENE':\" href=\"#stats-based-on-only-longest-isoform-per-gene\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e#####################################################\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"Contig N10: 1027\nContig N20: 643\nContig N30: 486\nContig N40: 397\nContig N50: 336\n\nMedian contig length: 271\nAverage contig: 347.72\nTotal assembled bases: 7295061\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eContig N10: 1027\nContig N20: 643\nContig N30: 486\nContig N40: 397\nContig N50: 336\n\nMedian contig length: 271\nAverage contig: 347.72\nTotal assembled bases: 7295061\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e[spere004@coreV2-22-028 scripts]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/scripts/python avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta\nThe total number of sequences is 15079\nThe average sequence length is 876\nThe total number of bases is 13210470\nThe minimum sequence length is 500\nThe maximum sequence length is 10795\nThe N50 is 881\nMedian Length = 578\ncontigs \u0026lt; 150bp = 0\ncontigs \u0026gt;= 500bp = 15079\ncontigs \u0026gt;= 1000bp = 3660\ncontigs \u0026gt;= 2000bp = 536\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e[spere004@coreV2-22-028 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs\na) grep \"overall alignment rate\" steph_clippedtrimmedbowtie.txt\nb) grep \"aligned exactly 1 time\" steph_clippedtrimmedbowtie.txt | cut -d \" \" -f 5\nc) grep \"aligned exactly 1 time\" steph_clippedtrimmedbowtie.txt | cut -d \"%\" -f 1 | cut -d \"(\" -f 2\nd) grep \"aligned \u0026gt;1 time\" steph_clippedtrimmedbowtie.txt | cut -d \"%\" -f 1 | cut -d \"(\" -f 2\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003enano alignedstats_sg.txt\nlane5_sg\t2.03\t1.11\t235455.75\t0.92\ncat alignedstats_sg.txt \u0026gt;\u0026gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e[spere004@coreV2-22-028 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data\nmkdir testassembly\na) [spere004@coreV2-22-028 djbtestassembly]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs/djbtestassembly\nmv Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/testassembly/\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eb) /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq\nnano cleanup.sh\n#!/bin/bash -l\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#SBATCH -o cleanup.txt\n#SBATCH -n 1\n#SBATCH --mail-user=\u003ca href=\"mailto:spere004@odu.edu\"\u003espere004@odu.edu\u003c/a\u003e\n#SBATCH --mail-type=END\n#SBATCH --job-name=cleanup\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/originalfastqs\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/filteringstats\nrm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs/trinity_out_dir\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esbatch cleanup.sh\u003c/p\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003epwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data\nnano steph_blast.sh\n#!/bin/bash -l\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e#SBATCH -o blastout.txt\n#SBATCH -n 6\n#SBATCH --mail-user=\u003ca href=\"mailto:spere004@odu.edu\"\u003espere004@odu.edu\u003c/a\u003e\n#SBATCH --mail-type=END\n#SBATCH --job-name=steph_blast\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eenable_lmod\nmodule load container_env blast\nblastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esbatch steph_blast.sh\nSubmitted batch job 9273231\u003c/p\u003e\n\u003col start=\"7\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e[spere004@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs\nhead -20 RI_B_05_18_clippedtrimmed.fastq.sam\n@HD\tVN:1.0\tSO:unsorted\n@SQ\tSN:TR78|c0_g1_i1_coral\tLN:1109\n@SQ\tSN:TR78|c0_g2_i1_coral\tLN:1109\n@SQ\tSN:TR79|c0_g1_i1_coral\tLN:610\n@SQ\tSN:TR79|c0_g2_i1_coral\tLN:1549\n@SQ\tSN:TR87|c0_g1_i1_coral\tLN:732\n@SQ\tSN:TR93|c0_g1_i1_coral\tLN:550\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003egrep -v '@' RI_B_05_18_clippedtrimmed.fastq.sam | head\nK00188:59:HMTFHBBXX:6:1101:2788:1648\t0\tTR47986|c1_g3_i2_sym\t9\t255\t51M\t*\t0\t0\tCNGGCAGTATGTGCAAAATTTATCTCCTGTTCACCTTTAAGTAACTAACCA\tA#\u0026lt;AFF-FJAFAJJJJJJFJJJFJ-FJJJFFJJFJAJJJJJJJJJFJJJJJ\tAS:i:-1\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:1G49\tYT:Z:UU\tRG:Z:RI_B_05_18\nK00188:59:HMTFHBBXX:6:1101:16823:1666\t16\tTR47986|c1_g3_i2_sym\t684\t255\t51M\t*\t0\t0\tTAAT\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003esalloc\nsalloc: Pending job allocation 9273245\nsalloc: job 9273245 queued and waiting for resources\nsalloc: job 9273245 has been allocated resources\nsalloc: Granted job allocation 9273245\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py RI_B_05_*.sam\nRI_B_05_18_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\nread reverse strand\nnot primary alignment\n256 :\nnot primary alignment\n16 :\nread reverse strand\nRI_B_05_22_clippedtrimmed.fastq.sam\n['0', '272', '256', '16']\n0 :\n272 :\nread reverse strand\nnot primary alignment\n256 :\nnot primary alignment\n16 :\nread reverse strand\u003c/p\u003e\n\u003col start=\"10\" dir=\"auto\"\u003e\n\u003cli\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs\nnano steph_SNPcall.sh\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e#!/bin/bash -l\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#SBATCH -o steph_SNPcall.txt\n#SBATCH -n 1\n#SBATCH --mail-user=\u003ca href=\"mailto:spere004@odu.edu\"\u003espere004@odu.edu\u003c/a\u003e\n#SBATCH --mail-type=END\n#SBATCH --job-name=steph_SNPcall\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eenable_lmod\nmodule load samtools/1\nfor i in *.sam; do \u003ccode\u003esamtools view -bS $i \u0026gt; ${i%.sam}_UNSORTED.bam\u003c/code\u003e; done\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor i in *UNSORTED.bam; do samtools sort \u003cmath-renderer class=\"js-inline-math\" style=\"display: inline\" data-static-url=\"https://github.githubassets.com/static\" data-run-id=\"65e40d34dd7cab1ed98d9b508702929b\"\u003e$i \u0026amp;gt; $\u003c/math-renderer\u003e{i%_UNSORTED.bam}.bam\nsamtools index ${i%_UNSORTED.bam}.bam\ndone\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esbatch steph_SNPcall.sh\nSubmitted batch job 9273232\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#Homework07\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e[spere004@turing1 data]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/testassembly/\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003enano blast parsing.sh\n#!/bin/bash -l\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e#SBATCH -o blastparseout.txt\n#SBATCH -n 1\n#SBATCH --mail-user=\u003ca href=\"mailto:spere004@odu.edu\"\u003espere004@odu.edu\u003c/a\u003e\n#SBATCH --mail-type=END\n#SBATCH --job-name=blastparsing\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e/cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esbatch blastparsing.sh\nSubmitted batch job 9276480\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecat blastparseout.txt\n#hit_pct_cov_bin        count_in_bin    \u0026gt;bin_below\n100     143     143\n90      52      195\n80      77      272\n70      87      359\n60      110     469\n50      133     602\n40      219     821\n30      395     1216\n20      726     1942\n10      674     2616\u003c/p\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003e[spere004@turing1 QCFastqs]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs\nls *_UNSORTED.bam\nRI_B_05_18_clippedtrimmed.fastq_UNSORTED.bam  VA_B_05_18_clippedtrimmed.fastq_UNSORTED.bam\nRI_B_05_22_clippedtrimmed.fastq_UNSORTED.bam  VA_B_05_22_clippedtrimmed.fastq_UNSORTED.bam\nRI_B_06_14_clippedtrimmed.fastq_UNSORTED.bam  VA_B_06_14_clippedtrimmed.fastq_UNSORTED.bam\nRI_B_06_22_clippedtrimmed.fastq_UNSORTED.bam  VA_B_06_22_clippedtrimmed.fastq_UNSORTED.bam\nRI_W_05_18_clippedtrimmed.fastq_UNSORTED.bam  VA_W_05_18_clippedtrimmed.fastq_UNSORTED.bam\nRI_W_05_22_clippedtrimmed.fastq_UNSORTED.bam  VA_W_05_22_clippedtrimmed.fastq_UNSORTED.bam\nRI_W_06_14_clippedtrimmed.fastq_UNSORTED.bam  VA_W_06_14_clippedtrimmed.fastq_UNSORTED.bam\nRI_W_06_22_clippedtrimmed.fastq_UNSORTED.bam  VA_W_06_22_clippedtrimmed.fastq_UNSORTED.bam\nnano rmunsorted.sh\n#!/bin/bash -l\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e#SBATCH -o rmunsortout.txt\n#SBATCH -n 1\n#SBATCH --mail-user=\u003ca href=\"mailto:spere004@odu.edu\"\u003espere004@odu.edu\u003c/a\u003e\n#SBATCH --mail-type=END\n#SBATCH --job-name=rmunsorted\u003c/p\u003e\n\u003cp dir=\"auto\"\u003erm *_UNSORTED.bam\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esbatch rmunsorted.sh\nSubmitted batch job 9276496\nls *_UNSORTED.bam\nls: No match. ##all unsorted bam files are gone.\nrm rmunsorted.txt ##unnecessary txt file to keep\u003c/p\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs\nnano steph_bayes.sh\n#!/bin/bash -l\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e#SBATCH -o bayesout.txt\n#SBATCH -n 1\u003cbr\u003e\n#SBATCH --mail-user=\u003ca href=\"mailto:spere004@odu.edu\"\u003espere004@odu.edu\u003c/a\u003e\n#SBATCH --mail-type=END\n#SBATCH --job-name=steph_bayes\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eenable_lmod\nmodule load dDocent\nfreebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.bam \u0026gt; SGmergedfastqs.vcf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esbatch steph_bayes.sh\nSubmitted batch job 9276570\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e##Homework08\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data\nmkdir BAMs\nmkdir SAMs\nmkdir VCF\nsalloc\nsalloc: Pending job allocation 9278250\nsalloc: job 9278250 queued and waiting for resources\nsalloc: job 9278250 has been allocated resources\nsalloc: Granted job allocation 9278250\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs\nmv *.sam /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/SAMs/\nmv *.bam *.bam.bai /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/BAMs/\nNO unfiltered vcfs\nrm *.fastq\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eenable_lmod\nmodule load dDocent\nvcftools\u003c/p\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e[spere004@turing1 VCF]$ pwd\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/VCF\ncp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf ./\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e[spere004@turing1 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 1214003 out of a possible 1214003 Sites\nRun Time = 10.00 seconds\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e[spere004@turing1 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf SGmergedfastqs.vcf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eParameters as interpreted:\n--vcf SGmergedfastqs.vcf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAfter filtering, kept 16 out of 16 Individuals\nAfter filtering, kept 130505 out of a possible 130505 Sites\nRun Time = 1.00 seconds\u003c/p\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003ecp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt ./\n[spere004@turing1 VCF]$ ls\nGoodCoralGenelistForVCFSubsetter.txt  mergedfastq_HEAAstrangiaAssembly.vcf  out.log  SGmergedfastqs.vcf\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003egrep -c \"##contig\" SGmergedfastqs.vcf\n15079\u003c/p\u003e\n\u003col start=\"6\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e[spere004@coreV3-23-031 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 432676 out of a possible 432676 Sites\nRun Time = 6.00 seconds\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e--vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly.vcf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAfter filtering, kept 40 out of 40 Individuals\nAfter filtering, kept 1214003 out of a possible 1214003 Sites\nRun Time = 5.00 seconds\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e[spere004@coreV3-23-031 VCF]$ vcftools --vcf SGmergedfastqs.vcf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eParameters as interpreted:\n--vcf SGmergedfastqs.vcf\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAfter filtering, kept 16 out of 16 Individuals\nAfter filtering, kept 130505 out of a possible 130505 Sites\nRun Time = 1.00 seconds\u003c/p\u003e\n\u003col start=\"8\" dir=\"auto\"\u003e\n\u003cli\u003e[spere004@coreV3-23-031 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n--recode-INFO-all\n--mac 3\n--minQ 30\n--max-missing 0.5\n--out raw.g5mac3\n--recode\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 99853 out of a possible 432676 Sites\nRun Time = 31.00 seconds\u003c/p\u003e\n\u003cp dir=\"auto\"\u003evcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw5.g5mac3dp3\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eParameters as interpreted:\n--vcf raw.g5mac3.recode.vcf\n--recode-INFO-all\n--minDP 3\n--out raw5.g5mac3dp3\n--recode\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 99853 out of a possible 99853 Sites\nRun Time = 25.00 seconds\u003c/p\u003e\n\u003cp dir=\"auto\"\u003evcftools --vcf raw5.g5mac3dp3.recode.vcf --missing-indv\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eParameters as interpreted:\n--vcf raw5.g5mac3dp3.recode.vcf\n--missing-indv\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAfter filtering, kept 40 out of 40 Individuals\nOutputting Individual Missingness\nAfter filtering, kept 99853 out of a possible 99853 Sites\nRun Time = 2.00 seconds\n[spere004@coreV3-23-031 VCF]$ cat out.imiss\nINDV\tN_DATA\tN_GENOTYPES_FILTERED\tN_MISS\tF_MISS\nRI_W_06_merged\t99853\t0\t67075\t0.671737\nRI_W_07_merged\t99853\t0\t63917\t0.640111\nVA_B_03_merged\t99853\t0\t56586\t0.566693\nRI_W_02_merged\t99853\t0\t73105\t0.732126\nRI_W_04_merged\t99853\t0\t71159\t0.712638\nVA_W_09_SNP_clipped\t99853\t0\t19459\t0.194876\nRI_B_08_SNP_clipped\t99853\t0\t88110\t0.882397\nVA_W_08_SNP_clipped\t99853\t0\t83344\t0.834667\nVA_B_08_SNP_clipped\t99853\t0\t94221\t0.943597\nVA_W_02_merged\t99853\t0\t71050\t0.711546\nVA_B_07_merged\t99853\t0\t62687\t0.627793\nRI_B_05_merged\t99853\t0\t43945\t0.440097\nVA_W_06_merged\t99853\t0\t65164\t0.652599\nVA_W_04_merged\t99853\t0\t50911\t0.509859\nVA_W_01_merged\t99853\t0\t67150\t0.672489\nVA_B_10_SNP_clipped\t99853\t0\t82831\t0.829529\nVA_B_06_merged\t99853\t0\t57724\t0.57809\nVA_W_05_merged\t99853\t0\t66297\t0.663946\nRI_B_09_SNP_clipped\t99853\t0\t85107\t0.852323\nVA_W_10_SNP_clipped\t99853\t0\t83234\t0.833565\nRI_W_08_SNP_clipped\t99853\t0\t77802\t0.779165\nRI_B_06_merged\t99853\t0\t79766\t0.798834\nRI_W_10_SNP_clipped\t99853\t0\t91323\t0.914574\nRI_B_04_merged\t99853\t0\t46717\t0.467858\nVA_W_03_merged\t99853\t0\t64016\t0.641102\nRI_B_07_merged\t99853\t0\t70798\t0.709022\nRI_W_05_merged\t99853\t0\t57117\t0.572011\nRI_W_09_SNP_clipped\t99853\t0\t88492\t0.886223\nVA_B_01_merged\t99853\t0\t47217\t0.472865\nVA_B_09_SNP_clipped\t99853\t0\t79187\t0.793036\nRI_B_10_SNP_clipped\t99853\t0\t82779\t0.829009\nRI_W_01_merged\t99853\t0\t73169\t0.732767\nRI_B_01_merged\t99853\t0\t69619\t0.697215\nVA_B_04_merged\t99853\t0\t58190\t0.582757\nRI_B_02_merged\t99853\t0\t87439\t0.875677\nRI_W_03_merged\t99853\t0\t38760\t0.388171\nVA_B_02_merged\t99853\t0\t78381\t0.784964\nVA_W_07_merged\t99853\t0\t64592\t0.646871\nVA_B_05_merged\t99853\t0\t54401\t0.544811\nRI_B_03_merged\t99853\t0\t80369\t0.804873\n[spere004@coreV3-23-031 VCF]$ bash\n[spere004@coreV3-23-031 VCF]$ mawk '!/IN/' out.imiss | cut -f5 \u0026gt; totalmissing\n[spere004@coreV3-23-031 VCF]$ gnuplot \u0026lt;\u0026lt; \\EOF\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eset terminal dumb size 120, 30\nset autoscale\nunset label\nset title \"Histogram of % missing data per individual\"\nset ylabel \"Number of Occurrences\"\nset xlabel \"% of missing data\"\n#set yr [0:100000]\nbinwidth=0.01\nbin(x,width)=width*floor(x/width) + binwidth/2.0\nplot 'totalmissing' using (bin($1,binwidth)):(1.0) smooth freq with boxes\npause -1\nEOF\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"                                     Histogram of % missing data per individual\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e                                     Histogram of % missing data per individual\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNumber of Occurrences\n3 ++----------+-----------+-----------+-----------+------------+---**\u003cem\u003e-----+-----------+-----------+----------++\n+           +           +           +           +       'totalmissing' using (bin($1,binwidth)):(1.0) ****** +\n|                                                                * *                                         |\n|                                                                * *                                         |\n|                                                                * *                                         |\n|                                                                * *                                         |\n2.5 ++                                                               * *                                        ++\n|                                                                * *                                         |\n|                                                                * *                                         |\n|                                                                * *                                         |\n|                                                                * *                                         |\n|                                                                * *                                         |\n2 ++                                                       **      * * ***  ******    ** ****   ****          ++\n|                                                        **      * * * *  * *  *    ** ** *   *  *           |\n|                                                        **      * * * *  * *  *    ** ** *   *  *           |\n|                                                        **      * * * *  * *  *    ** ** *   *  *           |\n|                                                        **      * * * *  * *  *    ** ** *   *  *           |\n|                                                        **      * * * *  * *  *    ** ** *   *  *           |\n1.5 ++                                                       **      * * * *  * *  *    ** ** *   *  *          ++\n|                                                        **      * * * *  * *  *    ** ** *   *  *           |\n|                                                        **      * * * *  * *  *    ** ** *   *  *           |\n|                                                        **      * * * *  * *  *    ** ** *   *  *           |\n|                                                        **      * * * *  * *  *    ** ** *   *  *           |\n+           +           +           +           +        **  +   * * * * +\u003c/em\u003e *  *    ** ** *   *  *           +\n1 *********************************************************************************************************---++\n0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1\n% of missing data\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emawk '$5 \u0026gt; 0.5' out.imiss | cut -f1 \u0026gt; lowDP.indv\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e--vcf raw5.g5mac3dp3.recode.vcf --remove lowDP.indv --recode --recode-INFO-all --out raw.g5mac3dplm\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eVCFtools - 0.1.14\n(C) Adam Auton and Anthony Marcketta 2009\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eParameters as interpreted:\n--vcf raw5.g5mac3dp3.recode.vcf\n--exclude lowDP.indv\n--recode-INFO-all\n--out raw.g5mac3dplm\n--recode\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eExcluding individuals in 'exclude' list\nAfter filtering, kept 5 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 99853 out of a possible 99853 Sites\nRun Time = 10.00 seconds\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#Homework09\n/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/VCF\nsalloc\nenable_lmod\nmodule load dDocent\n#second half run\n/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eVCFtools - v0.1.12b\n(C) Adam Auton and Anthony Marcketta 2009\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eParameters as interpreted:\n--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf\n--recode-INFO-all\n--mac 3\n--maf 0.015\n--max-alleles 2\n--minDP 10\n--minQ 30\n--max-missing 0.5\n--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters\n--recode\n--remove-indels\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAfter filtering, kept 40 out of 40 Individuals\nOutputting VCF file...\nAfter filtering, kept 1578 out of a possible 432676 Sites\nRun Time = 6.00 seconds\u003c/p\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003egrep '#CHROM' mergedfastq_HEAAstrangiaAssembly_subset.vcf \u0026gt; sampleIDs.txt\nsed 's/\\s+/\\n/g' sampleIDs.txt \u0026gt; sampleIDs1.txt\nscp \u003ca href=\"mailto:spere004@turing.hpc.odu.edu\"\u003espere004@turing.hpc.odu.edu\u003c/a\u003e:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/VCF/sampleIDs1.txt .\n#Added the second column via excel and uploaded the file to the cluster again\nscp sampleIDs1.txt \u003ca href=\"mailto:spere004@turing.hpc.odu.edu\"\u003espere004@turing.hpc.odu.edu\u003c/a\u003e:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/VCF/sampleIDs2.txt\ncp sampleIDs2.txt genepop.txt\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf genepop.txt\nIndivs with genotypes in vcf file: RI_W_06_merged\tRI_W_07_merged\tVA_B_03_merged\tRI_W_02_merged\tRI_W_04_merged\tVA_W_09_SNP_clipped\tRI_B_08_SNP_clipped\tVA_W_08_SNP_clipped\tVA_B_08_SNP_clipped\tVA_W_02_merged\tVA_B_07_merged\tRI_B_05_merged\tVA_W_06_merged\tVA_W_04_merged\tVA_W_01_merged\tVA_B_10_SNP_clipped\tVA_B_06_merged\tVA_W_05_merged\tRI_B_09_SNP_clipped\tVA_W_10_SNP_clipped\tRI_W_08_SNP_clipped\tRI_B_06_merged\tRI_W_10_SNP_clipped\tRI_B_04_merged\tVA_W_03_merged\tRI_B_07_merged\tRI_W_05_merged\tRI_W_09_SNP_clipped\tVA_B_01_merged\tVA_B_09_SNP_clipped\tRI_B_10_SNP_clipped\tRI_W_01_merged\tRI_B_01_merged\tVA_B_04_merged\tRI_B_02_merged\tRI_W_03_merged\tVA_B_02_merged\tVA_W_07_merged\tVA_B_05_merged\tRI_B_03_merged\n44 1578 1578 1578 1578 40\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003escp \u003ca href=\"mailto:spere004@turing.hpc.odu.edu\"\u003espere004@turing.hpc.odu.edu\u003c/a\u003e:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/VCF/1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen .\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003e#Working on RStudio. Code below is r commands from day09 R script\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003esum(is.na(datafile$tab))\n[1] 35966\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003edatafile #shows info\n/// GENIND OBJECT /////////\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003e// 40 individuals; 1,578 loci; 3,156 alleles; size: 1.4 Mb\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e// Basic content\n@tab:  40 x 3156 matrix of allele counts\n@loc.n.all: number of alleles per locus (range: 2-2)\n@loc.fac: locus factor for the 3156 columns of @tab\n@all.names: list of allele names for each locus\n@ploidy: ploidy of each individual  (range: 2-2)\n@type:  codom\n@call: read.genepop(file = \"1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen\",\nncode = 2)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e// Optional content\n@pop: population of each individual (group size range: 40-40)\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eYOURdata\u0026lt;-scaleGen(datafile, NA.method='mean')\nWarning message:\nIn .local(x, ...) : Some scaling values are null.\nCorresponding alleles are removed.\nX\u0026lt;-YOURdata\nY\u0026lt;-as.factor(substring(pop(datafile),1,4))\npca1 \u0026lt;- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)\ns.label(pca1$li)\ns.class(pca1$li, pop(datafile))\ncol \u0026lt;- c(\"blue\",\"red\", \"green\", \"black\")\ns.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)\nadd.scatter.eig(pca1$eig[1:3], 3,1,2, posi=\"topright\")\ntitle(\"PCA of DJB_data\\naxes 1-2\")\na.clust\u0026lt;-snapclust(datafile, k = 2)\nclass(a.clust)\n[1] \"snapclust\" \"list\"\u003cbr\u003e\nnames(a.clust)\n[1] \"group\"     \"ll\"        \"proba\"     \"converged\" \"n.iter\"    \"n.param\"\u003cbr\u003e\na.tab \u0026lt;- table(pop(datafile), a.clust$group)\ntable.value(a.tab, col.labels = 1:2)\ncompoplot(a.clust)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003e#plots were created and save on computer\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#Homework10\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e#Following code is R scrip ran on 2 files in day10 folder. Figures are saved on local computer.\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003elibrary(\"ape\")\nlibrary(\"pegas\")\nlibrary(\"seqinr\")\nlibrary(\"ggplot2\")\nlibrary(\"adegenet\")\nsetwd(\"/Users/stephanieperez/Documents/Data_analysis_class/21sp_advgenomics\")\ndatafile\u0026lt;-read.genepop(\"coral_66_cloneremoved_highoutliers.filtered1SNPper_genepop.gen\", ncode=2)\ndatafile2\u0026lt;-read.genepop(\"coral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen\", ncode=2)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esum(is.na(datafile$tab))\ndatafile #shows info\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esum(is.na(datafile2$tab))\ndatafile2\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eYOURdata\u0026lt;-scaleGen(datafile, NA.method='mean')\nX\u0026lt;-YOURdata\nY\u0026lt;-as.factor(substring(pop(datafile),1,4))\npca1 \u0026lt;- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eYOUR2data\u0026lt;-scaleGen(datafile2, NA.method='mean')\nX2\u0026lt;-YOUR2data\nY2\u0026lt;-as.factor(substring(pop(datafile2),1,4))\npca12 \u0026lt;- dudi.pca(X2,cent=F, scale=F, scannf=F, nf=3)\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePCAs\u003c/h4\u003e\u003ca id=\"user-content-pcas\" class=\"anchor\" aria-label=\"Permalink: PCAs\" href=\"#pcas\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eindividual labels\u003c/h1\u003e\u003ca id=\"user-content-individual-labels\" class=\"anchor\" aria-label=\"Permalink: individual labels\" href=\"#individual-labels\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003es.label(pca1$li)\ns.label(pca12$li)\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003epopulation elipses\u003c/h1\u003e\u003ca id=\"user-content-population-elipses\" class=\"anchor\" aria-label=\"Permalink: population elipses\" href=\"#population-elipses\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003es.class(pca1$li, pop(datafile))\ns.class(pca12$li, pop(datafile2))\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#color symbols, pop names\n#pdf(\"YOURINITIALS_ColorPCA1v2.pdf\")\ncol \u0026lt;- c(\"blue\",\"red\", \"green\", \"black\")\ns.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)\nadd.scatter.eig(pca1$eig[1:3], 3,1,2, posi=\"topright\")\ntitle(\"PCA of DJB_data\\naxes 1-2\")\ndev.off()\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecol \u0026lt;- c(\"blue\",\"red\", \"green\", \"black\")\ns.class(pca12$li, Y2,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)\nadd.scatter.eig(pca12$eig[1:3], 3,1,2, posi=\"topright\")\ntitle(\"PCA of DJB_data\\naxes 1-2\")\ndev.off()\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSnapclust\u003c/h4\u003e\u003ca id=\"user-content-snapclust\" class=\"anchor\" aria-label=\"Permalink: Snapclust\" href=\"#snapclust\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ea.clust\u0026lt;-snapclust(datafile, k = 2)\nclass(a.clust)\nnames(a.clust)\na.tab \u0026lt;- table(pop(datafile), a.clust$group)\ntable.value(a.tab, col.labels = 1:2)\ncompoplot(a.clust)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ea.clust1\u0026lt;-snapclust(datafile2, k = 2)\nclass(a.clust1)\nnames(a.clust1)\na.tab \u0026lt;- table(pop(datafile2), a.clust$group)\ntable.value(a.tab, col.labels = 1:2)\ncompoplot(a.clust1)\u003c/p\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003ecd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/SAMs\nnano countexpression.sh\n#!/bin/bash -l\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e#SBATCH -o countexpression.txt\n#SBATCH -n 1\u003cbr\u003e\n#SBATCH --mail-user=\u003ca href=\"mailto:spere004@odu.edu\"\u003espere004@odu.edu\u003c/a\u003e\n#SBATCH --mail-type=END\n#SBATCH --job-name=countexpression\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esbatch countexpression.sh\u003c/p\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003esalloc\nsalloc: Pending job allocation 9280866\nsalloc: job 9280866 queued and waiting for resources\nsalloc: job 9280866 has been allocated resources\nsalloc: Granted job allocation 9280866\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003emv match_counts.txt countsoutput.txt\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e/cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt StephFullCounts_summed.txt NoMatch *_counts.txt\nHits not matchedmatch_counts.txt=13381\tRI_B_05_18_clippedtrimmed.fastq_counts.txt=1698\tRI_B_05_22_clippedtrimmed.fastq_counts.txt=1698\tRI_B_06_14_clippedtrimmed.fastq_counts.txt=1698\tRI_B_06_22_clippedtrimmed.fastq_counts.txt=1698\tRI_W_05_18_clippedtrimmed.fastq_counts.txt=1698\tRI_W_05_22_clippedtrimmed.fastq_counts.txt=1698\tRI_W_06_14_clippedtrimmed.fastq_counts.txt=1698\tRI_W_06_22_clippedtrimmed.fastq_counts.txt=1698\tVA_B_05_18_clippedtrimmed.fastq_counts.txt=1698\tVA_B_05_22_clippedtrimmed.fastq_counts.txt=1698\tVA_B_06_14_clippedtrimmed.fastq_counts.txt=1698\tVA_B_06_22_clippedtrimmed.fastq_counts.txt=1698\tVA_W_05_18_clippedtrimmed.fastq_counts.txt=1698\tVA_W_05_22_clippedtrimmed.fastq_counts.txt=1698\tVA_W_06_14_clippedtrimmed.fastq_counts.txt=1698\tVA_W_06_22_clippedtrimmed.fastq_counts.txt=1698\u003c/p\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003escp \u003ca href=\"mailto:spere004@turing.hpc.odu.edu\"\u003espere004@turing.hpc.odu.edu\u003c/a\u003e:/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/SAMs/StephFullCounts_summed.txt .\n\u003ca href=\"mailto:spere004@turing.hpc.odu.edu\"\u003espere004@turing.hpc.odu.edu\u003c/a\u003e's password:\nStephFullCounts_summed.txt                                                                100%  860KB   5.2MB/s   00:00\u003c/p\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003esed 's/_clippedtrimmed.fastq_counts.txt_UniqueTotReads//g' StephFullCounts_summed.txt \u0026gt; NewStephfullcount_summed.txt\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e##HOMEWORKDAY11\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003esetwd(\"/Users/stephanieperez/Documents/Data_analysis_class/21sp_advgenomics\")\ngetwd()\n[1] \"/Users/stephanieperez/Documents/Data_analysis_class/21sp_advgenomics\"\ncountsTable \u0026lt;- read.delim('djbFullCounts_summed.txt', header=TRUE, stringsAsFactors=TRUE, row.names=1)\nhead(countsTable)\nRI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10054|c0_g1_i1_coral          0          0          0          0          0          0\nTR10054|c0_g2_i1_coral          0          1          6          0          0          0\nTR10083|c0_g2_i1_coral          0          1          2          0          1          2\nTR10090|c0_g1_i1_coral          0          3          1          0          4          0\nTR10090|c0_g2_i1_coral          0          4          2          1          0          0\nTR10103|c0_g1_i1_coral          0          1          4          0          0          4\nRI_B_03_14 RI_B_03_18 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22\ndim(countsTable)\n[1] 13381    84\ncolSums(countsTable)\nRI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_18\n97406     207372     173843      47481      58036      84501     170330       5972\nRI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14 RI_B_05_18 RI_B_05_22 RI_B_06_14\n199204     357124     544267      71600     630609     227313     225639      37012\ncolSums(countsTable)[order(colSums(countsTable))]\nVA_W_02_22 VA_W_01_22 RI_B_03_18 RI_W_04_22 RI_B_07_22 RI_B_06_14 RI_W_04_14 RI_W_01_14\n134        313       5972      10848      25369      37012      37579      39419\nRI_B_02_14 RI_W_02_14 RI_B_02_18 RI_B_04_22 RI_B_06_22 VA_B_02_14 RI_B_02_22 VA_B_02_22\n47481      54184      58036      71600      72254      79276      84501      85224\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eLowCounts\u0026lt;-c(\"VA_W_02_22\",\"VA_W_01_22\",\"RI_B_03_18\")\nnames(countsTable)%in%LowCounts\n[1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE\n[15] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n[29] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n[43] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n[57] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE FALSE\n[71] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e!(names(countsTable)%in%LowCounts)\n[1]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[15]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[29]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[43]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[57]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE FALSE  TRUE\n[71]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\ncountsTable\u0026lt;-countsTable[,!(names(countsTable)%in%LowCounts)]\ndim(countsTable)\n[1] 13381    81\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003econds\u0026lt;-data.frame(\"Sample\"=names(countsTable))\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003econds$Origin\u0026lt;-factor(substring(conds$Sample,1,2))\nconds$SymState\u0026lt;-factor(substring(conds$Sample,4,4))\nconds$Temp\u0026lt;-factor(substring(conds$Sample,9,10))\nconds$Origin_SymState\u0026lt;-factor(paste(conds$Origin,conds$SymState, sep=\"\u003cem\u003e\"))\nconds$Origin_SymState_Temp\u0026lt;-factor(paste(conds$Origin,conds$SymState,conds$Temp, sep=\"\u003c/em\u003e\"))\u003c/p\u003e\n\u003cp dir=\"auto\"\u003econds\nSample Origin SymState Temp Origin_SymState Origin_SymState_Temp\n1  RI_B_01_14     RI        B   14            RI_B              RI_B_14\n2  RI_B_01_18     RI        B   18            RI_B              RI_B_18\n3  RI_B_01_22     RI        B   22            RI_B              RI_B_22\n4  RI_B_02_14     RI        B   14            RI_B              RI_B_14\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eprop.null \u0026lt;- apply(countsTable, 2, function(x) 100*mean(x==0))\nbarplot(prop.null, main=\"Percentage of null counts per sample\",\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    horiz=TRUE, cex.names=0.5, las=1, \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    horiz=TRUE, cex.names=0.5, las=1, \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003epdf(file=\"OverallNullCounts.pdf\",14, 14)\nbarplot(prop.null, main=\"Percentage of null counts per sample\",\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    horiz=TRUE, cex.names=0.5, las=1, \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    horiz=TRUE, cex.names=0.5, las=1, \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003edev.off()\nRStudioGD\n2\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003emeans = apply(countsTable, 1, mean)\ntable(means\u0026gt;=3)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eFALSE  TRUE\n5838  7543\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003ecountsTable\u0026lt;-countsTable[means\u0026gt;=3,]\ndim(countsTable)\n[1] 7543   81\nprop.nullv2 \u0026lt;- apply(countsTable, 2, function(x) 100*mean(x==0))\npdf(file=\"Mean3NullCounts.pdf\",14, 14)\nbarplot(prop.nullv2, main=\"Percentage of null counts per sample\",\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    horiz=TRUE, cex.names=0.5, las=1, \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    horiz=TRUE, cex.names=0.5, las=1, \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003edev.off()\nRStudioGD\n2\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003edds \u0026lt;- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=~ Origin_SymState_Temp)\ndds \u0026lt;- DESeq(dds)\nestimating size factors\nestimating dispersions\ngene-wise dispersion estimates\nmean-dispersion relationship\nfinal dispersion estimates\nfitting model and testing\n-- replacing outliers and refitting for 8 genes\n-- DESeq argument 'minReplicatesForReplace' = 7\n-- original counts are preserved in counts(dds)\nestimating dispersions\nfitting model and testing\nvstCounts\u0026lt;-varianceStabilizingTransformation(dds)\ndists\u0026lt;-dist(t(assay(vstCounts)))\nplot(hclust(dists))\nplotPCA(vstCounts, intgroup=\"Origin\")\nplotPCA(vstCounts, intgroup=\"SymState\")\nplotPCA(vstCounts, intgroup=\"Temp\")\nplotPCA(vstCounts, intgroup=\"Origin_SymState\")\nplotPCA(vstCounts, intgroup=\"Origin_SymState_Temp\")\nresultsNames(dds)\n[1] \"Intercept\"                               \"Origin_SymState_Temp_RI_B_18_vs_RI_B_14\"\n[3] \"Origin_SymState_Temp_RI_B_22_vs_RI_B_14\" \"Origin_SymState_Temp_RI_W_14_vs_RI_B_14\"\n[5] \"Origin_SymState_Temp_RI_W_18_vs_RI_B_14\" \"Origin_SymState_Temp_RI_W_22_vs_RI_B_14\"\n[7] \"Origin_SymState_Temp_VA_B_14_vs_RI_B_14\" \"Origin_SymState_Temp_VA_B_18_vs_RI_B_14\"\n[9] \"Origin_SymState_Temp_VA_B_22_vs_RI_B_14\" \"Origin_SymState_Temp_VA_W_14_vs_RI_B_14\"\n[11] \"Origin_SymState_Temp_VA_W_18_vs_RI_B_14\" \"Origin_SymState_Temp_VA_W_22_vs_RI_B_14\"\ndim(dds)\n[1] 7543   81\nres \u0026lt;- results(dds, contrast=c(\"Origin_SymState_Temp\", \"RI_B_18\", \"VA_B_18\"))\nhead(res)\nlog2 fold change (MLE): Origin_SymState_Temp RI_B_18 vs VA_B_18\nWald test p-value: Origin_SymState_Temp RI_B_18 vs VA_B_18\nDataFrame with 6 rows and 6 columns\nbaseMean     log2FoldChange             lfcSE\n                   \nTR10083|c0_g2_i1_coral 5.92149226796624 -0.124351582570554 0.881723031266384\nTR10090|c0_g1_i1_coral   2.414668728522  0.403769276033225 0.959279168358701\nTR10090|c0_g2_i1_coral 3.75861468695046  -0.83542301411573  0.86523878852701\nTR10103|c0_g1_i1_coral 2.98376089398939 -0.801464007457239 0.848307534238223\nTR10108|c0_g1_i1_coral 5.73440284734055 -0.526701449017958 0.837420233249731\nTR10132|c0_g1_i1_coral 8.52629945810969   1.07157187389725  1.16012842796493\nstat            pvalue              padj\n                  \nTR10083|c0_g2_i1_coral -0.141032476368405 0.887844286314782 0.996374898818957\nTR10090|c0_g1_i1_coral  0.420909042280218 0.673821502118031 0.990719409569674\nTR10090|c0_g2_i1_coral -0.965540409414564 0.334274206188308 0.964875734525325\nTR10103|c0_g1_i1_coral -0.944780017988347 0.344771195342852 0.967913215005047\nTR10108|c0_g1_i1_coral -0.628957156879308 0.529377105037453 0.987728806338776\nTR10132|c0_g1_i1_coral  0.923666594203699 0.355659929898546 0.967913215005047\nsummary(res)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eout of 7542 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 3, 0.04%\nLFC \u0026lt; 0 (down)     : 5, 0.066%\noutliers [1]       : 4, 0.053%\nlow counts [2]     : 1, 0.013%\n(mean count \u0026lt; 1)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eres \u0026lt;- results(dds, contrast=c(\"Origin_SymState_Temp\", \"RI_W_18\", \"VA_W_18\"))\nsummary(res)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eout of 7542 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 22, 0.29%\nLFC \u0026lt; 0 (down)     : 38, 0.5%\noutliers [1]       : 4, 0.053%\nlow counts [2]     : 3801, 50%\n(mean count \u0026lt; 6)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eres \u0026lt;- results(dds, contrast=c(\"Origin_SymState_Temp\", \"RI_B_14\", \"VA_B_14\"))\nsummary(res)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eout of 7542 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 8, 0.11%\nLFC \u0026lt; 0 (down)     : 4, 0.053%\noutliers [1]       : 4, 0.053%\nlow counts [2]     : 1, 0.013%\n(mean count \u0026lt; 1)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eres \u0026lt;- results(dds, contrast=c(\"Origin_SymState_Temp\", \"RI_W_14\", \"VA_W_14\"))\nsummary(res)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eout of 7542 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 10, 0.13%\nLFC \u0026lt; 0 (down)     : 3, 0.04%\noutliers [1]       : 4, 0.053%\nlow counts [2]     : 5263, 70%\n(mean count \u0026lt; 8)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eres \u0026lt;- results(dds, contrast=c(\"Origin_SymState_Temp\", \"RI_B_22\", \"VA_B_22\"))\nsummary(res)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eout of 7542 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 8, 0.11%\nLFC \u0026lt; 0 (down)     : 2, 0.027%\noutliers [1]       : 4, 0.053%\nlow counts [2]     : 1, 0.013%\n(mean count \u0026lt; 1)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eres \u0026lt;- results(dds, contrast=c(\"Origin_SymState_Temp\", \"RI_W_22\", \"VA_W_22\"))\nsummary(res)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eout of 7542 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 11, 0.15%\nLFC \u0026lt; 0 (down)     : 5, 0.066%\noutliers [1]       : 4, 0.053%\nlow counts [2]     : 146, 1.9%\n(mean count \u0026lt; 2)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003ehead(res[order(res$padj),])\nlog2 fold change (MLE): Origin_SymState_Temp RI_W_22 vs VA_W_22\nWald test p-value: Origin_SymState_Temp RI_W_22 vs VA_W_22\nDataFrame with 6 rows and 6 columns\nbaseMean    log2FoldChange             lfcSE\n                  \nTR42879|c0_g2_i1_coral  15.410578527665  17.0575331642567  3.13747844493701\nTR13286|c1_g1_i1_coral 32.6102567902141 -3.38044295663573 0.648393226272248\nTR47617|c0_g1_i1_coral  3.7465555529336  19.7261549909787  3.83515887756262\nTR8327|c0_g3_i1_coral  8.73276616692518  19.7338447748926  4.06385316874438\nTR29560|c0_g1_i1_coral 12.7955647888331  2.74246489048391 0.651667601875444\nTR58072|c1_g1_i1_coral 53.8558310969574   1.3803927876847 0.330572470141824\nstat               pvalue                 padj\n                        \nTR42879|c0_g2_i1_coral  5.43670130763213 5.42760110313753e-08 0.000401262549554957\nTR13286|c1_g1_i1_coral -5.21356920409335 1.85241378350407e-07 0.000664534402813938\nTR47617|c0_g1_i1_coral  5.14350399050882 2.69660923636117e-07 0.000664534402813938\nTR8327|c0_g3_i1_coral   4.85594433545684  1.1981435873852e-06  0.00221446888538469\nTR29560|c0_g1_i1_coral  4.20837998174426 2.57208081930306e-05   0.0365937404208763\nTR58072|c1_g1_i1_coral  4.17576450662234 2.96986937001566e-05   0.0365937404208763\nplotCounts(dds, \"TR2367|c0_g1_i1_coral\", intgroup=\"Origin_SymState_Temp\")\nplotCounts(dds, \"TR2367|c0_g1_i1_coral\", intgroup=\"Origin_SymState_Temp\")\nsum(res$padj \u0026lt; 0.3, na.rm =T)\n[1] 30\nsum(res$padj \u0026lt; 0.2, na.rm =T)\n[1] 20\nsum(res$padj \u0026lt; 0.1, na.rm =T)\n[1] 16\nsum(res$pvalue \u0026lt; 0.05, na.rm =T)\n[1] 343\nscaledcounts = counts(dds, normalized=T)\nhead(scaledcounts)\nRI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10083|c0_g2_i1_coral          0    0.84279   2.292953   0.000000   3.273656   4.312157\nTR10090|c0_g1_i1_coral          0    2.52837   1.146477   0.000000  13.094624   0.000000\nTR10090|c0_g2_i1_coral          0    3.37116   2.292953   3.553607   0.000000   0.000000\nTR10103|c0_g1_i1_coral          0    0.84279   4.585907   0.000000   0.000000   8.624315...\ngenes4heatmap\u0026lt;-res[res$pvalue \u0026lt;0.05 \u0026amp; !is.na(res$pvalue),]\nnames(genes4heatmap)\n[1] \"baseMean\"       \"log2FoldChange\" \"lfcSE\"          \"stat\"           \"pvalue\"\u003cbr\u003e\n[6] \"padj\"\u003cbr\u003e\nhead(genes4heatmap)\nlog2 fold change (MLE): Origin_SymState_Temp RI_W_22 vs VA_W_22\nWald test p-value: Origin_SymState_Temp RI_W_22 vs VA_W_22\nDataFrame with 6 rows and 6 columns\nbaseMean    log2FoldChange             lfcSE\n                  \nTR10090|c0_g1_i1_coral   2.414668728522  -2.1602364606262  1.06658202772732\nTR10450|c0_g1_i1_coral 3.06295343830589 -3.43890734797067  1.35815909767837\nTR10768|c0_g1_i1_coral 4.11391026506041  1.64720448435771   0.8163169025316\nTR10773|c0_g1_i1_coral 2.74299826965667  3.89664910540304  1.28628276466641\nTR11771|c0_g1_i1_coral 2.86558169571988  2.26910497576441 0.996533942762988\nTR11938|c0_g1_i2_coral 3.17991201457045  1.92891473796097  0.93224992931647\nstat              pvalue              padj\n                    \nTR10090|c0_g1_i1_coral -2.02538239391605   0.042828113882863 0.996984539439689\nTR10450|c0_g1_i1_coral -2.53203571941545   0.011340243841624 0.882509712853964\nTR10768|c0_g1_i1_coral  2.01784929265745  0.0436069594078913 0.996984539439689\nTR10773|c0_g1_i1_coral  3.02938763733929 0.00245050039091153 0.437812432964705\nTR11771|c0_g1_i1_coral  2.27699717831296  0.0227863941200996 0.989064009775494\nTR11938|c0_g1_i2_coral  2.06909614825636  0.0385370651149041 0.996984539439689\ndim(genes4heatmap)\n[1] 343   6\ndata4heatmap\u0026lt;-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]\ndim(data4heatmap)\n[1] 343  81\nhead(data4heatmap)\nRI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10090|c0_g1_i1_coral    0.00000    2.52837   1.146477   0.000000   13.09462   0.000000\nTR10450|c0_g1_i1_coral    0.00000    2.52837   0.000000   0.000000    0.00000   6.468236\nTR10768|c0_g1_i1_coral   10.84813    4.21395   4.585907   7.107215    0.00000   8.624315\nTR10773|c0_g1_i1_coral    0.00000    5.89953   2.292953   3.553607    0.00000   0.000000\nTR11771|c0_g1_i1_coral    0.00000    2.52837   3.439430   0.000000    0.00000   0.000000\nTR11938|c0_g1_i2_coral    0.00000    2.52837   0.000000   0.000000    0.00000   4.312157...\ntemp = as.matrix(rowMeans(data4heatmap))\nhead(temp)\n[,1]\nTR10090|c0_g1_i1_coral 2.414669\nTR10450|c0_g1_i1_coral 3.062953\nTR10768|c0_g1_i1_coral 4.113910\nTR10773|c0_g1_i1_coral 2.742998\nTR11771|c0_g1_i1_coral 2.865582\nTR11938|c0_g1_i2_coral 3.179912\nscaledmatrix\u0026lt;-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)\ndata4heatmapscaled = data4heatmap/scaledmatrix\nhead(data4heatmapscaled)\nRI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10090|c0_g1_i1_coral   0.000000  1.0470877  0.4747967   0.000000   5.422948   0.000000\nTR10450|c0_g1_i1_coral   0.000000  0.8254680  0.0000000   0.000000   0.000000   2.111764\nTR10768|c0_g1_i1_coral   2.636938  1.0243174  1.1147318   1.727606   0.000000   2.096379\nTR10773|c0_g1_i1_coral   0.000000  2.1507597  0.8359295   1.295519   0.000000   0.000000\nTR11771|c0_g1_i1_coral   0.000000  0.8823235  1.2002554   0.000000   0.000000   0.000000\nTR11938|c0_g1_i2_coral   0.000000  0.7951069  0.0000000   0.000000   0.000000   1.356062...\ndim(data4heatmapscaled)\n[1] 343  81\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003epairs.breaks \u0026lt;- seq(0, 3.0, by=0.1)\nlength(pairs.breaks)\n[1] 31\nmycol \u0026lt;- colorpanel(n=30, low=\"black\", high=\"yellow\")\npdf(file=\"SGdata_byrow.pdf\",14,7)\nheatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(\"row\"), scale=\"none\", keysize=1, breaks=pairs.breaks, col=mycol, trace = \"none\", symkey = F, density.info = \"density\", colsep=c(24), sepcolor=c(\"white\"), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\ndev.off()\nRStudioGD\n2\npdf(file=\"SGdata_bycolumn.pdf\",14,7)\nheatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(\"col\"), scale=\"none\", keysize=1, breaks=pairs.breaks, col=mycol, trace = \"none\", symkey = F, density.info = \"density\", colsep=c(24), sepcolor=c(\"white\"), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\ndev.off()\nRStudioGD\n2\nplotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\"Origin\")\nprop.nullv3 \u0026lt;- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100*mean(x==0))\npdf(file=\"ResNullCounts.pdf\",14, 14)\nbarplot(prop.nullv3, main=\"Percentage of null counts per sample\",\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    horiz=TRUE, cex.names=0.5, las=1, \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    horiz=TRUE, cex.names=0.5, las=1, \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003edev.off()\nRStudioGD\n2\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003e#Homeworkday12\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e_B_03_14 RI_B_05_18 VA_W_07_14 VA_B_06_14 VA_W_07_22 VA_B_07_18 VA_W_02_18 VA_B_03_18\n226784     227313     228792     232578     233899     234465     241804     251696\nRI_B_07_14 VA_W_04_22 VA_B_01_14 VA_B_04_18 VA_B_04_22 VA_B_03_22 VA_W_03_14 VA_W_04_18\n257730     261344     261824     264192     268290     271959     278000     280007\nRI_W_03_14 VA_B_05_18 VA_W_04_14 RI_W_01_18 VA_W_03_22 RI_B_06_18 RI_W_04_18 VA_W_01_18\n281297     288121     292041     292612     295382     296071     300578     321900\nVA_B_05_14 VA_B_01_22 RI_W_07_14 RI_B_04_14 RI_W_03_22 VA_B_06_22 VA_W_05_18 RI_W_05_14\n323715     328198     342795     357124     359053     362054     388532     392818\nRI_W_03_18 RI_B_04_18 VA_B_01_18 RI_B_05_14\n428532     544267     548448     630609\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003etable(substring(names(countsTable),1,2))\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eRI VA\n42 42\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003etable(substring(names(countsTable),4,4))\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eB  W\n42 42\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003etable(substring(names(countsTable),9,10))\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003e14 18 22\n28 28 28\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003etable(paste(substring(names(countsTable),1,2),substring(names(countsTable),4,4), sep=\"_\"))\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eRI_B RI_W VA_B VA_W\n21   21   21   21\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003etable(paste(substring(names(countsTable),1,2),substring(names(countsTable),4,4), substring(names(countsTable),9,10), sep=\"_\"))\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eRI_B_14 RI_B_18 RI_B_22 RI_W_14 RI_W_18 RI_W_22 VA_B_14 VA_B_18 VA_B_22 VA_W_14 VA_W_18\n7       7       7       7       7       7       7       7       7       7       7\nVA_W_22\n7\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eLowCounts\u0026lt;-c(\"VA_W_02_22\",\"VA_W_01_22\",\"RI_B_03_18\", \"RI_W_01_14\", \"RI_W_04_22\", \"VA_W_03_18\")\nnames(countsTable)%in%LowCounts\n[1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE\n[15] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE\n[29] FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n[43] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n[57] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE FALSE\n[71]  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n!(names(countsTable)%in%LowCounts)\n[1]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[15]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[29]  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[43]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n[57]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE FALSE  TRUE\n[71] FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\ncountsTable\u0026lt;-countsTable[,!(names(countsTable)%in%LowCounts)]\ndim(countsTable)\n[1] 13381    78\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#make a table with conditions of each individual (e.g. \"VA\" and \"RI\")  There should be the same number of conditions described as there are samples in your data file, and in the same order.\n#NOTE: It is absolutely critical that the columns of the count matrix and the rows of the column data (information about samples) are in the same order. DESeq2 will not make guesses as to which column of the count matrix belongs to which row of the column data, these must be provided to DESeq2 already in consistent order.\n#Here's an example\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSample\t\tOrigin\tSymState\tTemp  Origin_SymState Origin_SymState_Temp\u003c/h1\u003e\u003ca id=\"user-content-sampleoriginsymstatetemp--origin_symstate-origin_symstate_temp\" class=\"anchor\" aria-label=\"Permalink: Sample\t\tOrigin\tSymState\tTemp  Origin_SymState Origin_SymState_Temp\" href=\"#sampleoriginsymstatetemp--origin_symstate-origin_symstate_temp\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRI_B_06_18\t\tRI\tB\t18  RI_B  RI_B_18\u003c/h1\u003e\u003ca id=\"user-content-ri_b_06_18rib18--ri_b--ri_b_18\" class=\"anchor\" aria-label=\"Permalink: RI_B_06_18\t\tRI\tB\t18  RI_B  RI_B_18\" href=\"#ri_b_06_18rib18--ri_b--ri_b_18\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eVA_B_07_14\t\tVA\tB\t14  VA_B  VA_B_14\u003c/h1\u003e\u003ca id=\"user-content-va_b_07_14vab14--va_b--va_b_14\" class=\"anchor\" aria-label=\"Permalink: VA_B_07_14\t\tVA\tB\t14  VA_B  VA_B_14\" href=\"#va_b_07_14vab14--va_b--va_b_14\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eVA_W_07_22\t\tVA\tW\t22  VA_W  VA_W_22\u003c/h1\u003e\u003ca id=\"user-content-va_w_07_22vaw22--va_w--va_w_22\" class=\"anchor\" aria-label=\"Permalink: VA_W_07_22\t\tVA\tW\t22  VA_W  VA_W_22\" href=\"#va_w_07_22vaw22--va_w--va_w_22\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003esafest way to do this is in R\u003c/h1\u003e\u003ca id=\"user-content-safest-way-to-do-this-is-in-r\" class=\"anchor\" aria-label=\"Permalink: safest way to do this is in R\" href=\"#safest-way-to-do-this-is-in-r\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003econds\u0026lt;-data.frame(\"Sample\"=names(countsTable))\nconds$Origin\u0026lt;-factor(substring(conds$Sample,1,2))\nconds$SymState\u0026lt;-factor(substring(conds$Sample,4,4))\nconds$Temp\u0026lt;-factor(substring(conds$Sample,9,10))\nconds$Origin_SymState\u0026lt;-factor(paste(conds$Origin,conds$SymState, sep=\"\u003cem\u003e\"))\nconds$Origin_SymState_Temp\u0026lt;-factor(paste(conds$Origin,conds$SymState,conds$Temp, sep=\"\u003c/em\u003e\"))\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e##Check null counts per sample\nprop.null \u0026lt;- apply(countsTable, 2, function(x) 100*mean(x==0))\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ebarplot(prop.null, main=\"Percentage of null counts per sample\",\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    horiz=TRUE, cex.names=0.5, las=1, \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    horiz=TRUE, cex.names=0.5, las=1, \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003epdf(file=\"OverallNullCountsv2.pdf\",14, 14)\nbarplot(prop.null, main=\"Percentage of null counts per sample\",\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    horiz=TRUE, cex.names=0.5, las=1, \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    horiz=TRUE, cex.names=0.5, las=1, \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003edev.off()\nRStudioGD\n2\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ehow many genes have mean count \u0026gt;=3\u003c/h1\u003e\u003ca id=\"user-content-how-many-genes-have-mean-count-3\" class=\"anchor\" aria-label=\"Permalink: how many genes have mean count \u0026gt;=3\" href=\"#how-many-genes-have-mean-count-3\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003emeans = apply(countsTable, 1, mean)\ntable(means\u0026gt;=3)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eFALSE  TRUE\n5721  7660\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003ecountsTable\u0026lt;-countsTable[means\u0026gt;=3,]\ndim(countsTable)\n[1] 7660   78\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eprop.nullv2 \u0026lt;- apply(countsTable, 2, function(x) 100*mean(x==0))\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ebarplot(prop.nullv2, main=\"Percentage of null counts per sample\",\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    horiz=TRUE, cex.names=0.5, las=1, \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    horiz=TRUE, cex.names=0.5, las=1, \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003epdf(file=\"Mean3NullCountsv2.pdf\",14, 14)\nbarplot(prop.nullv2, main=\"Percentage of null counts per sample\",\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    horiz=TRUE, cex.names=0.5, las=1, \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    horiz=TRUE, cex.names=0.5, las=1, \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003edev.off()\nRStudioGD\n2\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#make count data sets\ndds \u0026lt;- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=~ Origin + SymState + Temp + Origin:SymState)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edds \u0026lt;- DESeq(dds)\nestimating size factors\nestimating dispersions\ngene-wise dispersion estimates\nmean-dispersion relationship\nfinal dispersion estimates\nfitting model and testing\n-- replacing outliers and refitting for 14 genes\n-- DESeq argument 'minReplicatesForReplace' = 7\n-- original counts are preserved in counts(dds)\nestimating dispersions\nfitting model and testing\u003c/p\u003e\n\u003cp dir=\"auto\"\u003edim(dds)\n[1] 7660   78\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#transform counts to variance stablized counts\nvstCounts\u0026lt;-varianceStabilizingTransformation(dds)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e#plot cluster dendrogram across samples\ndists\u0026lt;-dist(t(assay(vstCounts)))\nplot(hclust(dists))\nheatmap.2(as.matrix(dists), key=F, trace=\"none\",\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"      col=colorpanel(100, \u0026quot;black\u0026quot;, \u0026quot;white\u0026quot;),\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e      col=colorpanel(100, \"black\", \"white\"),\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"      margin=c(10, 10))\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e      margin=c(10, 10))\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003e###Figure out which contrast you want to examine (i.e. which two groups do you want to compare)\nresultsNames(dds)\n[1] \"Intercept\"          \"Origin_VA_vs_RI\"    \"SymState_W_vs_B\"    \"Temp_18_vs_14\"\u003cbr\u003e\n[5] \"Temp_22_vs_14\"      \"OriginVA.SymStateW\"\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eres \u0026lt;- results(dds)\nhead(res)\nlog2 fold change (MLE): OriginVA.SymStateW\nWald test p-value: OriginVA.SymStateW\nDataFrame with 6 rows and 6 columns\nbaseMean     log2FoldChange             lfcSE\n                   \nTR10083|c0_g2_i1_coral   5.366479327573  0.107690151743452  0.64464619766983\nTR10090|c0_g1_i1_coral 2.70436251126018 -0.247209127175208 0.831370968745152\nTR10090|c0_g2_i1_coral 3.65756460141883 -0.976996511249959 0.654110304614643\nTR10103|c0_g1_i1_coral  2.9127886212765 -0.172219518994785 0.641025633753981\nTR10108|c0_g1_i1_coral 6.31144026709945  0.366046450639525 0.664400660631174\nTR10132|c0_g1_i1_coral  9.2673703948612 -0.589356834713084 0.975809166670876\nstat            pvalue              padj\n                  \nTR10083|c0_g2_i1_coral   0.16705310933767 0.867328260788591 0.999364127033592\nTR10090|c0_g1_i1_coral -0.297351166289025 0.766198422057453  0.99520853313718\nTR10090|c0_g2_i1_coral  -1.49362654029054  0.13527325854353 0.958806816980756\nTR10103|c0_g1_i1_coral -0.268662452679515 0.788189448150222  0.99520853313718\nTR10108|c0_g1_i1_coral  0.550942333940163 0.581673205867577 0.991106673557066\nTR10132|c0_g1_i1_coral -0.603967307177248 0.545865378089277 0.990476522025752\nsummary(res)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eout of 7659 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 2, 0.026%\nLFC \u0026lt; 0 (down)     : 9, 0.12%\noutliers [1]       : 10, 0.13%\nlow counts [2]     : 1, 0.013%\n(mean count \u0026lt; 1)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eresSymState \u0026lt;- results(dds, contrast=c(\"SymState\", \"B\", \"W\"))\nhead(resSymState)\nlog2 fold change (MLE): SymState B vs W\nWald test p-value: SymState B vs W\nDataFrame with 6 rows and 6 columns\nbaseMean     log2FoldChange             lfcSE\n                   \nTR10083|c0_g2_i1_coral   5.366479327573  0.304875236273871 0.463587635582715\nTR10090|c0_g1_i1_coral 2.70436251126018 -0.340213618854429 0.624844624315529\nTR10090|c0_g2_i1_coral 3.65756460141883   -1.0442719434015 0.488672538582417\nTR10103|c0_g1_i1_coral  2.9127886212765 -0.316210568092073 0.469122505416545\nTR10108|c0_g1_i1_coral 6.31144026709945   0.25797411425412 0.481341478797278\nTR10132|c0_g1_i1_coral  9.2673703948612 -0.604956612553455 0.692218264872669\nstat             pvalue              padj\n                   \nTR10083|c0_g2_i1_coral  0.657643157136089  0.510767456004599 0.986343106634585\nTR10090|c0_g1_i1_coral -0.544477147782311  0.586113169806799 0.995961170585915\nTR10090|c0_g2_i1_coral  -2.13695647074996 0.0326015322469065 0.816269739453607\nTR10103|c0_g1_i1_coral -0.674046894875149  0.500281500123105 0.986343106634585\nTR10108|c0_g1_i1_coral  0.535948231385995   0.59199432935739 0.995961170585915\nTR10132|c0_g1_i1_coral -0.873939107435622  0.382151415850855 0.986343106634585\nsummary(resSymState)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eout of 7659 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 6, 0.078%\nLFC \u0026lt; 0 (down)     : 6, 0.078%\noutliers [1]       : 10, 0.13%\nlow counts [2]     : 1, 0.013%\n(mean count \u0026lt; 1)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eresOrigin \u0026lt;- results(dds, contrast=c(\"Origin\", \"VA\", \"RI\"))\nhead(resOrigin)\nlog2 fold change (MLE): Origin VA vs RI\nWald test p-value: Origin VA vs RI\nDataFrame with 6 rows and 6 columns\nbaseMean      log2FoldChange             lfcSE\n                    \nTR10083|c0_g2_i1_coral   5.366479327573  -0.271242746289843 0.446224512750353\nTR10090|c0_g1_i1_coral 2.70436251126018    1.11251170496561 0.589306266446119\nTR10090|c0_g2_i1_coral 3.65756460141883   0.943134641716211 0.474719393211504\nTR10103|c0_g1_i1_coral  2.9127886212765  0.0386642676540012 0.456719612483097\nTR10108|c0_g1_i1_coral 6.31144026709945 -0.0179009694277555 0.461957599914831\nTR10132|c0_g1_i1_coral  9.2673703948612  -0.264990306479184 0.677607561801266\nstat             pvalue              padj\n                   \nTR10083|c0_g2_i1_coral  -0.607861600022842  0.543279269700006 0.865660813784253\nTR10090|c0_g1_i1_coral    1.88783281005095 0.0590483989759312                NA\nTR10090|c0_g2_i1_coral    1.98672027139201 0.0469534041233123 0.418043000342497\nTR10103|c0_g1_i1_coral  0.0846564644854882  0.932534507970102 0.984590265229257\nTR10108|c0_g1_i1_coral -0.0387502433795997  0.969089515041603 0.992149683420423\nTR10132|c0_g1_i1_coral  -0.391067516682912  0.695747330435347 0.921516804624192\nsummary(resOrigin)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eout of 7659 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 73, 0.95%\nLFC \u0026lt; 0 (down)     : 104, 1.4%\noutliers [1]       : 10, 0.13%\nlow counts [2]     : 594, 7.8%\n(mean count \u0026lt; 3)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eres14v18 \u0026lt;- results(dds, contrast=c(\"Temp\", \"14\", \"18\"))\nhead(res14v18)\nlog2 fold change (MLE): Temp 14 vs 18\nWald test p-value: Temp 14 vs 18\nDataFrame with 6 rows and 6 columns\nbaseMean     log2FoldChange             lfcSE\n                   \nTR10083|c0_g2_i1_coral   5.366479327573 0.0134105654377919 0.389675116366814\nTR10090|c0_g1_i1_coral 2.70436251126018 -0.750040304511369 0.493719609600886\nTR10090|c0_g2_i1_coral 3.65756460141883 -0.527849679316855 0.385227211070661\nTR10103|c0_g1_i1_coral  2.9127886212765  0.144899722334601 0.377623543666657\nTR10108|c0_g1_i1_coral 6.31144026709945 -0.117131615020316 0.401865058532759\nTR10132|c0_g1_i1_coral  9.2673703948612  -1.16667257299433 0.592603491023266\nstat             pvalue              padj\n                   \nTR10083|c0_g2_i1_coral 0.0344147339014795  0.972546434477265 0.994762640063651\nTR10090|c0_g1_i1_coral  -1.51916247587915   0.12872160398251                NA\nTR10090|c0_g2_i1_coral  -1.37022947535249   0.17061528001709                NA\nTR10103|c0_g1_i1_coral  0.383714746510905  0.701189881652785                NA\nTR10108|c0_g1_i1_coral -0.291470015950062  0.770691872645245 0.956296438232868\nTR10132|c0_g1_i1_coral  -1.96872375992892 0.0489848213296003 0.440134464115494\nsummary(res14v18)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eout of 7659 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 31, 0.4%\nLFC \u0026lt; 0 (down)     : 111, 1.4%\noutliers [1]       : 10, 0.13%\nlow counts [2]     : 2671, 35%\n(mean count \u0026lt; 4)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eres14v22 \u0026lt;- results(dds, contrast=c(\"Temp\", \"14\", \"22\"))\nhead(res14v22)\nlog2 fold change (MLE): Temp 14 vs 22\nWald test p-value: Temp 14 vs 22\nDataFrame with 6 rows and 6 columns\nbaseMean     log2FoldChange             lfcSE\n                   \nTR10083|c0_g2_i1_coral   5.366479327573 -0.381080048190283 0.396906053114453\nTR10090|c0_g1_i1_coral 2.70436251126018 -0.420390942831918 0.517099767325245\nTR10090|c0_g2_i1_coral 3.65756460141883 -0.149468056890158 0.407435526480227\nTR10103|c0_g1_i1_coral  2.9127886212765  0.327505589102544 0.400037906787554\nTR10108|c0_g1_i1_coral 6.31144026709945 -0.504680765262664 0.409220095704197\nTR10132|c0_g1_i1_coral  9.2673703948612   -1.2400583742243 0.602190057747409\nstat             pvalue              padj\n                   \nTR10083|c0_g2_i1_coral  -0.96012657201878  0.336991516539358  0.79600966973284\nTR10090|c0_g1_i1_coral -0.812978402613553  0.416230440926914                NA\nTR10090|c0_g2_i1_coral -0.366850819763779  0.713730297691035                NA\nTR10103|c0_g1_i1_coral  0.818686388328871  0.412965364593821                NA\nTR10108|c0_g1_i1_coral  -1.23327463768414  0.217473316972006 0.699820201183028\nTR10132|c0_g1_i1_coral  -2.05924750545192 0.0394705344365933 0.387444149379081\nsummary(res14v22)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eout of 7659 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 64, 0.84%\nLFC \u0026lt; 0 (down)     : 97, 1.3%\noutliers [1]       : 10, 0.13%\nlow counts [2]     : 2226, 29%\n(mean count \u0026lt; 4)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eres18v22 \u0026lt;- results(dds, contrast=c(\"Temp\", \"18\", \"22\"))\nhead(res18v22)\nlog2 fold change (MLE): Temp 18 vs 22\nWald test p-value: Temp 18 vs 22\nDataFrame with 6 rows and 6 columns\nbaseMean      log2FoldChange             lfcSE\n                    \nTR10083|c0_g2_i1_coral   5.366479327573  -0.394490613628075 0.395010013602601\nTR10090|c0_g1_i1_coral 2.70436251126018   0.329649361679451 0.503429809209276\nTR10090|c0_g2_i1_coral 3.65756460141883   0.378381622426696 0.396934638805837\nTR10103|c0_g1_i1_coral  2.9127886212765   0.182605866767943 0.399213811730964\nTR10108|c0_g1_i1_coral 6.31144026709945  -0.387549150242348 0.406735578634103\nTR10132|c0_g1_i1_coral  9.2673703948612 -0.0733858012299701 0.596241474936225\nstat            pvalue              padj\n                  \nTR10083|c0_g2_i1_coral -0.998685096689603 0.317947262436469 0.838585264651832\nTR10090|c0_g1_i1_coral  0.654806997220173 0.512592024969438                NA\nTR10090|c0_g2_i1_coral  0.953259265971454 0.340458724704261 0.847758501629442\nTR10103|c0_g1_i1_coral  0.457413700132709  0.64737371896663                NA\nTR10108|c0_g1_i1_coral -0.952828251572713 0.340677097558174 0.847758501629442\nTR10132|c0_g1_i1_coral -0.123080671699029 0.902043216645959 0.991572144643207\nsummary(res18v22)\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cp dir=\"auto\"\u003eout of 7659 with nonzero total read count\nadjusted p-value \u0026lt; 0.1\nLFC \u0026gt; 0 (up)       : 34, 0.44%\nLFC \u0026lt; 0 (down)     : 10, 0.13%\noutliers [1]       : 10, 0.13%\nlow counts [2]     : 1337, 17%\n(mean count \u0026lt; 3)\n[1] see 'cooksCutoff' argument of ?results\n[2] see 'independentFiltering' argument of ?results\u003c/p\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003esum(resOrigin$padj \u0026lt; 0.1, na.rm =T)\n[1] 177\nsum(resOrigin$pvalue \u0026lt; 0.05, na.rm =T)\n[1] 866\nscaledcounts = counts(dds, normalize=T)\nhead(scaledcounts)\nRI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10083|c0_g2_i1_coral          0  0.8671827   2.447866   0.000000   3.455170   4.432653\nTR10090|c0_g1_i1_coral          0  2.6015481   1.223933   0.000000  13.820681   0.000000\nTR10090|c0_g2_i1_coral          0  3.4687308   2.447866   3.772786   0.000000   0.000000\nTR10103|c0_g1_i1_coral          0  0.8671827   4.895733   0.000000   0.000000   8.865306\nTR10108|c0_g1_i1_coral          0  1.7343654   4.895733   0.000000   6.910341   0.000000\nTR10132|c0_g1_i1_coral          0  6.0702789   1.223933  18.863931  51.827555   0.000000\nRI_B_03_14 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10083|c0_g2_i1_coral  12.871970   1.272998  12.424473  8.1878079   0.000000   4.563408\nTR10090|c0_g1_i1_coral   0.000000   0.000000   2.003947  4.0939039   6.765958   0.570426\nTR10090|c0_g2_i1_coral   0.000000   3.818995   4.809473  3.6845136   0.000000   0.570426\nTR10103|c0_g1_i1_coral   1.287197   3.818995   6.011842  3.2751232   0.000000   5.133834\nTR10108|c0_g1_i1_coral  14.159167  15.275980   2.805526  4.9126847  16.914895   5.989473\nTR10132|c0_g1_i1_coral   6.435985   2.545997   0.000000  0.8187808   3.382979   1.426065\nRI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22 RI_B_07_14\nTR10083|c0_g2_i1_coral          0  2.4538305   19.55961   3.448566   18.78455  1.4936671\nTR10090|c0_g1_i1_coral          0  0.0000000    0.00000   2.069140    0.00000  0.7468335\nTR10090|c0_g2_i1_coral          0  2.4538305    0.00000   4.827992    0.00000  3.7341677\nTR10103|c0_g1_i1_coral          0  4.0897176    0.00000   3.448566    0.00000  2.2405006\nTR10108|c0_g1_i1_coral          0  0.8179435    6.51987  10.345698   40.69986  3.7341677\nTR10132|c0_g1_i1_coral          0  1.6358870    0.00000   1.379426    0.00000  8.9620025\nRI_B_07_18 RI_B_07_22 RI_W_01_18 RI_W_01_22 RI_W_02_14 RI_W_02_18\nTR10083|c0_g2_i1_coral   8.070182   36.85316  10.815190   4.646396   0.000000  4.9947759\nTR10090|c0_g1_i1_coral   2.017545    0.00000  12.257216   1.548799   0.000000  1.9979104\nTR10090|c0_g2_i1_coral   1.008773    0.00000  12.257216   7.743994   4.651361  4.9947759\nTR10103|c0_g1_i1_coral   2.017545    0.00000   7.931140   0.000000   0.000000  0.9989552\nTR10108|c0_g1_i1_coral   1.008773    0.00000   2.884051   1.548799   9.302721  6.9926863\nTR10132|c0_g1_i1_coral  44.386000   27.63987   1.442025   0.000000   0.000000  5.9937311\nRI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18\nTR10083|c0_g2_i1_coral   3.615216   8.525621   5.407467   4.745758   12.82564  2.9768406\nTR10090|c0_g1_i1_coral   0.000000   4.262810   1.081493   1.581919    0.00000  3.4729807\nTR10090|c0_g2_i1_coral   4.820288   3.044865   3.244480   1.581919    0.00000  7.4421015\nTR10103|c0_g1_i1_coral   3.615216   3.044865   3.244480   1.054613    0.00000  0.9922802\nTR10108|c0_g1_i1_coral   2.410144   2.435892   4.686471   2.109226   21.37607 12.8996426\nTR10132|c0_g1_i1_coral  44.587667   1.217946   1.081493   1.054613   29.92650  7.9382416\nRI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22\nTR10083|c0_g2_i1_coral  3.1630643   1.010704   5.526636   1.044347   1.336966   5.564881\nTR10090|c0_g1_i1_coral  0.9037326   8.085630   0.000000   0.000000   0.000000   4.451904\nTR10090|c0_g2_i1_coral  3.6149306  10.107037   1.381659   3.133041   1.336966   7.790833\nTR10103|c0_g1_i1_coral  4.0667969   2.021407   2.763318  10.443469   1.336966   2.225952\nTR10108|c0_g1_i1_coral  5.4223959   2.021407   6.908296  11.487816   0.000000   5.564881\nTR10132|c0_g1_i1_coral  1.8074653  10.107037  27.633182   5.221735   1.336966  42.293092\nRI_W_07_14 RI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10083|c0_g2_i1_coral   4.124636   5.593017   6.900578   2.551157   1.731769   2.845097\nTR10090|c0_g1_i1_coral   0.000000   0.000000   0.000000   7.653472   9.091789   1.707058\nTR10090|c0_g2_i1_coral   2.062318   1.118603   5.520462  10.204629   5.628251   5.690194\nTR10103|c0_g1_i1_coral   5.499514  13.423242   1.380116   3.826736   2.597654   2.845097\nTR10108|c0_g1_i1_coral   0.000000   1.118603   5.520462   3.826736   9.091789   2.845097\nTR10132|c0_g1_i1_coral   1.374879   2.237207  86.947282   1.275579   1.298827   0.000000\nVA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14 VA_B_03_18 VA_B_03_22\nTR10083|c0_g2_i1_coral          0    14.2516  19.312637  16.029307  14.339493   7.833217\nTR10090|c0_g1_i1_coral          0     0.0000   0.000000   5.828839   2.150924   7.833217\nTR10090|c0_g2_i1_coral          0     0.0000   0.000000  10.929073   4.301848   3.560553\nTR10103|c0_g1_i1_coral          0     0.0000   2.145849   2.185815   4.301848   3.560553\nTR10108|c0_g1_i1_coral          0     0.0000   0.000000  14.572098  18.641340  21.363318\nTR10132|c0_g1_i1_coral          0     0.0000   8.583394   9.471863  30.829909   8.545327\nVA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10083|c0_g2_i1_coral   0.000000  0.9006809   6.731616   2.098865  3.3173156   1.813677\nTR10090|c0_g1_i1_coral   0.000000  4.5034046   3.624716   4.897352  1.1057719   7.254707\nTR10090|c0_g2_i1_coral   7.108517  3.1523832   0.000000   1.399244  0.5528859   5.441030\nTR10103|c0_g1_i1_coral   4.442823  0.0000000   0.000000   0.000000  3.3173156   0.000000\nTR10108|c0_g1_i1_coral   2.665694  4.9537450   3.624716   9.095083  5.5288593   0.000000\nTR10132|c0_g1_i1_coral   0.000000  2.7020428   0.000000   0.000000  4.9759734   5.441030\nVA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22\nTR10083|c0_g2_i1_coral   1.383946   0.000000  5.9098781   3.728534  0.7405861   2.098499\nTR10090|c0_g1_i1_coral   8.303676   0.000000  0.4924898   0.000000  3.7029304   2.098499\nTR10090|c0_g2_i1_coral   0.000000   4.206352  1.4774695   1.491414 15.5523076   6.295498\nTR10103|c0_g1_i1_coral   5.535784   6.730163  3.9399187   1.491414  8.1464468   4.196999\nTR10108|c0_g1_i1_coral  11.763540   4.206352  3.4474289   2.237121  2.2217582   5.246248\nTR10132|c0_g1_i1_coral  18.683270   2.523811 14.2822054   8.948482 17.0334798   2.098499\nVA_W_01_14 VA_W_01_18 VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_22\nTR10083|c0_g2_i1_coral   6.699293   7.732656   0.000000  4.5789684  3.1096023   6.888125\nTR10090|c0_g1_i1_coral   3.828167   1.104665   6.978382  0.6541383  5.4418040   9.392897\nTR10090|c0_g2_i1_coral   0.000000   3.866328   4.361489 13.0827667  6.9966052   3.130966\nTR10103|c0_g1_i1_coral   0.000000   3.866328   5.233787  0.6541383  6.9966052   2.504773\nTR10108|c0_g1_i1_coral   1.914084   4.418660   0.000000  9.1579367  3.1096023   3.757159\nTR10132|c0_g1_i1_coral   0.000000  10.494319   2.616893 13.0827667  0.7774006  13.150056\nVA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22\nTR10083|c0_g2_i1_coral   1.026752   4.334366   6.410680  0.0000000   3.848517  12.379823\nTR10090|c0_g1_i1_coral   2.566880   5.779155   3.945034  0.0000000   1.099576  12.379823\nTR10090|c0_g2_i1_coral   2.053504   7.705540  10.848842  0.0000000   2.748941   7.427894\nTR10103|c0_g1_i1_coral   3.593631   3.852770   2.958775  0.8118702   3.298729   2.475965\nTR10108|c0_g1_i1_coral   6.673887  20.227043   8.383196  1.6237403   5.497882  38.377452\nTR10132|c0_g1_i1_coral   2.566880   0.000000   0.000000  2.4356105   8.796611   2.475965\nVA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22\nTR10083|c0_g2_i1_coral   5.042735  2.5895188   0.000000 10.2466895   1.127608  0.9473408\nTR10090|c0_g1_i1_coral   0.000000  0.8631729   1.764014  1.5764138   1.127608  6.6313859\nTR10090|c0_g2_i1_coral   3.025641  0.8631729   0.000000  5.5174482   3.382824  2.8420225\nTR10103|c0_g1_i1_coral   8.068376  2.5895188   4.410036  4.7292413   0.000000  1.8946817\nTR10108|c0_g1_i1_coral   4.034188  6.9053834   5.292043  3.1528275   1.127608  1.8946817\nTR10132|c0_g1_i1_coral   9.076923 29.3478796  26.460214  0.7882069   6.765648  8.5260676\nhead(scaledcounts)\nRI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10083|c0_g2_i1_coral          0  0.8671827   2.447866   0.000000   3.455170   4.432653\nTR10090|c0_g1_i1_coral          0  2.6015481   1.223933   0.000000  13.820681   0.000000\nTR10090|c0_g2_i1_coral          0  3.4687308   2.447866   3.772786   0.000000   0.000000\nTR10103|c0_g1_i1_coral          0  0.8671827   4.895733   0.000000   0.000000   8.865306\nTR10108|c0_g1_i1_coral          0  1.7343654   4.895733   0.000000   6.910341   0.000000\nTR10132|c0_g1_i1_coral          0  6.0702789   1.223933  18.863931  51.827555   0.000000\nRI_B_03_14 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10083|c0_g2_i1_coral  12.871970   1.272998  12.424473  8.1878079   0.000000   4.563408\nTR10090|c0_g1_i1_coral   0.000000   0.000000   2.003947  4.0939039   6.765958   0.570426\nTR10090|c0_g2_i1_coral   0.000000   3.818995   4.809473  3.6845136   0.000000   0.570426\nTR10103|c0_g1_i1_coral   1.287197   3.818995   6.011842  3.2751232   0.000000   5.133834\nTR10108|c0_g1_i1_coral  14.159167  15.275980   2.805526  4.9126847  16.914895   5.989473\nTR10132|c0_g1_i1_coral   6.435985   2.545997   0.000000  0.8187808   3.382979   1.426065\nRI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22 RI_B_07_14\nTR10083|c0_g2_i1_coral          0  2.4538305   19.55961   3.448566   18.78455  1.4936671\nTR10090|c0_g1_i1_coral          0  0.0000000    0.00000   2.069140    0.00000  0.7468335\nTR10090|c0_g2_i1_coral          0  2.4538305    0.00000   4.827992    0.00000  3.7341677\nTR10103|c0_g1_i1_coral          0  4.0897176    0.00000   3.448566    0.00000  2.2405006\nTR10108|c0_g1_i1_coral          0  0.8179435    6.51987  10.345698   40.69986  3.7341677\nTR10132|c0_g1_i1_coral          0  1.6358870    0.00000   1.379426    0.00000  8.9620025\nRI_B_07_18 RI_B_07_22 RI_W_01_18 RI_W_01_22 RI_W_02_14 RI_W_02_18\nTR10083|c0_g2_i1_coral   8.070182   36.85316  10.815190   4.646396   0.000000  4.9947759\nTR10090|c0_g1_i1_coral   2.017545    0.00000  12.257216   1.548799   0.000000  1.9979104\nTR10090|c0_g2_i1_coral   1.008773    0.00000  12.257216   7.743994   4.651361  4.9947759\nTR10103|c0_g1_i1_coral   2.017545    0.00000   7.931140   0.000000   0.000000  0.9989552\nTR10108|c0_g1_i1_coral   1.008773    0.00000   2.884051   1.548799   9.302721  6.9926863\nTR10132|c0_g1_i1_coral  44.386000   27.63987   1.442025   0.000000   0.000000  5.9937311\nRI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18\nTR10083|c0_g2_i1_coral   3.615216   8.525621   5.407467   4.745758   12.82564  2.9768406\nTR10090|c0_g1_i1_coral   0.000000   4.262810   1.081493   1.581919    0.00000  3.4729807\nTR10090|c0_g2_i1_coral   4.820288   3.044865   3.244480   1.581919    0.00000  7.4421015\nTR10103|c0_g1_i1_coral   3.615216   3.044865   3.244480   1.054613    0.00000  0.9922802\nTR10108|c0_g1_i1_coral   2.410144   2.435892   4.686471   2.109226   21.37607 12.8996426\nTR10132|c0_g1_i1_coral  44.587667   1.217946   1.081493   1.054613   29.92650  7.9382416\nRI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22\nTR10083|c0_g2_i1_coral  3.1630643   1.010704   5.526636   1.044347   1.336966   5.564881\nTR10090|c0_g1_i1_coral  0.9037326   8.085630   0.000000   0.000000   0.000000   4.451904\nTR10090|c0_g2_i1_coral  3.6149306  10.107037   1.381659   3.133041   1.336966   7.790833\nTR10103|c0_g1_i1_coral  4.0667969   2.021407   2.763318  10.443469   1.336966   2.225952\nTR10108|c0_g1_i1_coral  5.4223959   2.021407   6.908296  11.487816   0.000000   5.564881\nTR10132|c0_g1_i1_coral  1.8074653  10.107037  27.633182   5.221735   1.336966  42.293092\nRI_W_07_14 RI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10083|c0_g2_i1_coral   4.124636   5.593017   6.900578   2.551157   1.731769   2.845097\nTR10090|c0_g1_i1_coral   0.000000   0.000000   0.000000   7.653472   9.091789   1.707058\nTR10090|c0_g2_i1_coral   2.062318   1.118603   5.520462  10.204629   5.628251   5.690194\nTR10103|c0_g1_i1_coral   5.499514  13.423242   1.380116   3.826736   2.597654   2.845097\nTR10108|c0_g1_i1_coral   0.000000   1.118603   5.520462   3.826736   9.091789   2.845097\nTR10132|c0_g1_i1_coral   1.374879   2.237207  86.947282   1.275579   1.298827   0.000000\nVA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14 VA_B_03_18 VA_B_03_22\nTR10083|c0_g2_i1_coral          0    14.2516  19.312637  16.029307  14.339493   7.833217\nTR10090|c0_g1_i1_coral          0     0.0000   0.000000   5.828839   2.150924   7.833217\nTR10090|c0_g2_i1_coral          0     0.0000   0.000000  10.929073   4.301848   3.560553\nTR10103|c0_g1_i1_coral          0     0.0000   2.145849   2.185815   4.301848   3.560553\nTR10108|c0_g1_i1_coral          0     0.0000   0.000000  14.572098  18.641340  21.363318\nTR10132|c0_g1_i1_coral          0     0.0000   8.583394   9.471863  30.829909   8.545327\nVA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10083|c0_g2_i1_coral   0.000000  0.9006809   6.731616   2.098865  3.3173156   1.813677\nTR10090|c0_g1_i1_coral   0.000000  4.5034046   3.624716   4.897352  1.1057719   7.254707\nTR10090|c0_g2_i1_coral   7.108517  3.1523832   0.000000   1.399244  0.5528859   5.441030\nTR10103|c0_g1_i1_coral   4.442823  0.0000000   0.000000   0.000000  3.3173156   0.000000\nTR10108|c0_g1_i1_coral   2.665694  4.9537450   3.624716   9.095083  5.5288593   0.000000\nTR10132|c0_g1_i1_coral   0.000000  2.7020428   0.000000   0.000000  4.9759734   5.441030\nVA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22\nTR10083|c0_g2_i1_coral   1.383946   0.000000  5.9098781   3.728534  0.7405861   2.098499\nTR10090|c0_g1_i1_coral   8.303676   0.000000  0.4924898   0.000000  3.7029304   2.098499\nTR10090|c0_g2_i1_coral   0.000000   4.206352  1.4774695   1.491414 15.5523076   6.295498\nTR10103|c0_g1_i1_coral   5.535784   6.730163  3.9399187   1.491414  8.1464468   4.196999\nTR10108|c0_g1_i1_coral  11.763540   4.206352  3.4474289   2.237121  2.2217582   5.246248\nTR10132|c0_g1_i1_coral  18.683270   2.523811 14.2822054   8.948482 17.0334798   2.098499\nVA_W_01_14 VA_W_01_18 VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_22\nTR10083|c0_g2_i1_coral   6.699293   7.732656   0.000000  4.5789684  3.1096023   6.888125\nTR10090|c0_g1_i1_coral   3.828167   1.104665   6.978382  0.6541383  5.4418040   9.392897\nTR10090|c0_g2_i1_coral   0.000000   3.866328   4.361489 13.0827667  6.9966052   3.130966\nTR10103|c0_g1_i1_coral   0.000000   3.866328   5.233787  0.6541383  6.9966052   2.504773\nTR10108|c0_g1_i1_coral   1.914084   4.418660   0.000000  9.1579367  3.1096023   3.757159\nTR10132|c0_g1_i1_coral   0.000000  10.494319   2.616893 13.0827667  0.7774006  13.150056\nVA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22\nTR10083|c0_g2_i1_coral   1.026752   4.334366   6.410680  0.0000000   3.848517  12.379823\nTR10090|c0_g1_i1_coral   2.566880   5.779155   3.945034  0.0000000   1.099576  12.379823\nTR10090|c0_g2_i1_coral   2.053504   7.705540  10.848842  0.0000000   2.748941   7.427894\nTR10103|c0_g1_i1_coral   3.593631   3.852770   2.958775  0.8118702   3.298729   2.475965\nTR10108|c0_g1_i1_coral   6.673887  20.227043   8.383196  1.6237403   5.497882  38.377452\nTR10132|c0_g1_i1_coral   2.566880   0.000000   0.000000  2.4356105   8.796611   2.475965\nVA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22\nTR10083|c0_g2_i1_coral   5.042735  2.5895188   0.000000 10.2466895   1.127608  0.9473408\nTR10090|c0_g1_i1_coral   0.000000  0.8631729   1.764014  1.5764138   1.127608  6.6313859\nTR10090|c0_g2_i1_coral   3.025641  0.8631729   0.000000  5.5174482   3.382824  2.8420225\nTR10103|c0_g1_i1_coral   8.068376  2.5895188   4.410036  4.7292413   0.000000  1.8946817\nTR10108|c0_g1_i1_coral   4.034188  6.9053834   5.292043  3.1528275   1.127608  1.8946817\nTR10132|c0_g1_i1_coral   9.076923 29.3478796  26.460214  0.7882069   6.765648  8.5260676\ngenes4heatmap\u0026lt;-resOrigin[resOrigin$pvalue \u0026lt;0.05 \u0026amp; !is.na(resOrigin$pvalue),]\nnames(genes4heatmap)\n[1] \"baseMean\"       \"log2FoldChange\" \"lfcSE\"          \"stat\"           \"pvalue\"\u003cbr\u003e\n[6] \"padj\"\u003cbr\u003e\nhead(genes4heatmap)\nlog2 fold change (MLE): Origin VA vs RI\nWald test p-value: Origin VA vs RI\nDataFrame with 6 rows and 6 columns\nbaseMean    log2FoldChange             lfcSE\n                  \nTR10090|c0_g2_i1_coral 3.65756460141883 0.943134641716211 0.474719393211504\nTR10487|c0_g1_i1_coral 6.57169761658039 0.727122969400975 0.363907404480523\nTR1069|c0_g1_i1_coral  4.14205241197446  4.49029083181354  1.98238112955566\nTR10748|c0_g1_i1_coral 2.93642320037697  1.78798410821023 0.559059118556594\nTR10773|c0_g1_i1_coral 3.02515761016279  1.75847746453697 0.659498298908414\nTR10773|c1_g1_i1_coral 2.32875934475272  1.29222344670373 0.632167692638483\nstat              pvalue               padj\n                     \nTR10090|c0_g2_i1_coral 1.98672027139201  0.0469534041233123  0.418043000342497\nTR10487|c0_g1_i1_coral 1.99809885824924  0.0457059435120877  0.412518610214263\nTR1069|c0_g1_i1_coral  2.26509966467448  0.0235065642111382  0.308653252532063\nTR10748|c0_g1_i1_coral 3.19820220950252 0.00138287275835452 0.0720091267133405\nTR10773|c0_g1_i1_coral 2.66638665702029 0.00766714549040745  0.173395444167676\nTR10773|c1_g1_i1_coral 2.04411497416195  0.0409421901910667                 NA\ndim(genes4heatmap)\n[1] 866   6\ndata4heatmap\u0026lt;-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]\ndim(data4heatmap)\n[1] 866  78\nhead(data4heatmap)\nRI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10090|c0_g2_i1_coral    0.00000   3.468731   2.447866   3.772786    0.00000   0.000000\nTR10487|c0_g1_i1_coral   10.94714   5.203096   8.567532   0.000000    0.00000   6.648979\nTR1069|c0_g1_i1_coral     0.00000   0.000000   0.000000   7.545573    3.45517   6.648979\nTR10748|c0_g1_i1_coral    0.00000   0.000000   2.447866   0.000000    0.00000   0.000000\nTR10773|c0_g1_i1_coral    0.00000   6.070279   2.447866   3.772786    0.00000   0.000000\nTR10773|c1_g1_i1_coral    0.00000   2.601548   4.895733   0.000000    0.00000   2.216326\nRI_B_03_14 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10090|c0_g2_i1_coral   0.000000   3.818995   4.809473  3.6845136          0   0.570426\nTR10487|c0_g1_i1_coral   9.010379   5.091993   2.404737  4.5032943          0   9.126817\nTR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000  0.0000000          0   0.000000\nTR10748|c0_g1_i1_coral   2.574394   0.000000   4.007894  2.8657328          0   1.711278\nTR10773|c0_g1_i1_coral   1.287197   0.000000   0.000000  0.4093904          0   0.000000\nTR10773|c1_g1_i1_coral   5.148788   1.272998   0.000000  0.0000000          0   0.285213\nRI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22 RI_B_07_14\nTR10090|c0_g2_i1_coral   0.000000  2.4538305          0   4.827992          0   3.734168\nTR10487|c0_g1_i1_coral   5.033716  0.8179435          0   2.758853          0   5.227835\nTR1069|c0_g1_i1_coral    0.000000  0.0000000          0   0.000000          0   0.000000\nTR10748|c0_g1_i1_coral   0.000000  0.0000000          0   0.000000          0   2.987334\nTR10773|c0_g1_i1_coral   0.000000  0.0000000          0   0.000000          0   0.000000\nTR10773|c1_g1_i1_coral   0.000000  0.8179435          0   0.000000          0   0.000000\nRI_B_07_18 RI_B_07_22 RI_W_01_18 RI_W_01_22 RI_W_02_14 RI_W_02_18\nTR10090|c0_g2_i1_coral   1.008773          0  12.257216   7.743994   4.651361  4.9947759\nTR10487|c0_g1_i1_coral   5.043864          0   8.652152   6.195195  16.279762  2.9968656\nTR1069|c0_g1_i1_coral    0.000000          0  10.094178   9.292792   0.000000  0.0000000\nTR10748|c0_g1_i1_coral   0.000000          0   0.000000   1.548799   4.651361  0.9989552\nTR10773|c0_g1_i1_coral   1.008773          0   7.931140  17.036786   0.000000 14.9843278\nTR10773|c1_g1_i1_coral   0.000000          0   5.047089  18.585585   0.000000 10.9885071\nRI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18\nTR10090|c0_g2_i1_coral   4.820288   3.044865   3.244480   1.581919    0.00000  7.4421015\nTR10487|c0_g1_i1_coral   1.205072   7.307675   7.209956  14.237275   17.10086  5.4575411\nTR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000   0.000000    0.00000  0.0000000\nTR10748|c0_g1_i1_coral   2.410144   2.435892   6.128462   4.218452    0.00000  0.4961401\nTR10773|c0_g1_i1_coral   6.025360  23.140971  12.617422  29.001856    0.00000  1.9845604\nTR10773|c1_g1_i1_coral   1.205072  10.352540   7.930951   7.382291    0.00000  1.4884203\nRI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22\nTR10090|c0_g2_i1_coral   3.614931  10.107037   1.381659   3.133041   1.336966   7.790833\nTR10487|c0_g1_i1_coral   4.066797   4.042815   0.000000   4.177388  13.369662  10.016785\nTR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000   0.000000   0.000000   0.000000\nTR10748|c0_g1_i1_coral   2.711198   1.010704   0.000000   3.133041   2.673932   2.225952\nTR10773|c0_g1_i1_coral   5.422396   2.021407   0.000000   5.221735   2.673932  10.016785\nTR10773|c1_g1_i1_coral   3.614931   3.032111   0.000000   1.044347   2.673932   3.338928\nRI_W_07_14 RI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10090|c0_g2_i1_coral   2.062318   1.118603   5.520462  10.204629   5.628251   5.690194\nTR10487|c0_g1_i1_coral   3.437196  13.423242   8.280694  10.204629   6.494135  10.242350\nTR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000  33.165045   9.524732  17.639603\nTR10748|c0_g1_i1_coral   2.749757   2.237207   0.000000   1.275579   1.298827   5.121175\nTR10773|c0_g1_i1_coral   1.374879   2.237207   0.000000   6.377893   4.762366   2.845097\nTR10773|c1_g1_i1_coral   2.062318   3.355810   2.760231   8.929051   6.061193   6.259214\nVA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14 VA_B_03_18 VA_B_03_22\nTR10090|c0_g2_i1_coral   0.000000     0.0000   0.000000  10.929073   4.301848  3.5605530\nTR10487|c0_g1_i1_coral   0.000000     0.0000  17.166789  10.929073   7.886721  7.8332166\nTR1069|c0_g1_i1_coral   86.631891    44.0504  40.771123   0.000000   0.000000  0.0000000\nTR10748|c0_g1_i1_coral  30.575962     0.0000   2.145849   3.643024   7.886721  3.5605530\nTR10773|c0_g1_i1_coral   1.698665     0.0000   6.437546   0.000000   1.433949  0.7121106\nTR10773|c1_g1_i1_coral   0.000000     2.5912   0.000000   0.000000   3.584873  0.7121106\nVA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10090|c0_g2_i1_coral  7.1085175   3.152383   0.000000   1.399244  0.5528859   5.441030\nTR10487|c0_g1_i1_coral 15.1055996   4.053064   8.285065   4.547541 17.1394640  21.764122\nTR1069|c0_g1_i1_coral   0.0000000   0.000000   0.000000   0.000000  0.0000000   0.000000\nTR10748|c0_g1_i1_coral  2.6656940   0.000000   8.802882   1.049433  1.6586578   3.627354\nTR10773|c0_g1_i1_coral  0.0000000   7.205447   8.285065   1.399244  1.1057719   0.000000\nTR10773|c1_g1_i1_coral  0.8885647   8.556469   2.589083   0.000000  4.4230875   0.000000\nVA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22\nTR10090|c0_g2_i1_coral   0.000000   4.206352   1.477470  1.4914137  15.552308   6.295498\nTR10487|c0_g1_i1_coral   0.000000   8.412704   2.462449  3.7285342   5.184103   8.393997\nTR1069|c0_g1_i1_coral   29.062864  16.825408   8.372327  0.0000000   0.000000   0.000000\nTR10748|c0_g1_i1_coral   6.919730   0.000000   2.954939  6.7113616   0.000000   4.196999\nTR10773|c0_g1_i1_coral   2.767892   0.000000   5.417388  1.4914137   2.962344   6.295498\nTR10773|c1_g1_i1_coral   0.000000   2.523811   2.462449  0.7457068   0.000000   4.196999\nVA_W_01_14 VA_W_01_18 VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_22\nTR10090|c0_g2_i1_coral   0.000000   3.866328  4.3614889  13.082767  6.9966052   3.130966\nTR10487|c0_g1_i1_coral   6.699293   4.970993  1.7445956  13.082767  3.1096023   9.392897\nTR1069|c0_g1_i1_coral    0.000000   0.000000  0.0000000   0.000000  0.0000000   0.000000\nTR10748|c0_g1_i1_coral   2.871126   2.209330  0.8722978   0.000000  0.7774006   3.757159\nTR10773|c0_g1_i1_coral   5.742251   1.104665  0.0000000   3.924830  1.5548011   2.504773\nTR10773|c1_g1_i1_coral   0.000000   3.866328  0.0000000   5.233107  0.7774006   2.504773\nVA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22\nTR10090|c0_g2_i1_coral   2.053504   7.705540 10.8488424    0.00000   2.748941   7.427894\nTR10487|c0_g1_i1_coral   3.593631  13.003099  1.4793876    0.00000   8.246822   8.665876\nTR1069|c0_g1_i1_coral    0.000000   0.000000  0.0000000    0.00000   0.000000   0.000000\nTR10748|c0_g1_i1_coral   6.673887   9.150329  7.3969380   14.61366   2.748941   9.903859\nTR10773|c0_g1_i1_coral   1.026752   0.000000  0.4931292    0.00000   0.000000   0.000000\nTR10773|c1_g1_i1_coral   1.026752   0.000000  0.0000000    0.00000   2.199153   1.237982\nVA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22\nTR10090|c0_g2_i1_coral   3.025641  0.8631729   0.000000   5.517448   3.382824   2.842023\nTR10487|c0_g1_i1_coral  10.085470  1.7263459  11.466093   7.093862   6.765648   3.789363\nTR1069|c0_g1_i1_coral    0.000000  0.0000000   0.000000   0.000000   0.000000   0.000000\nTR10748|c0_g1_i1_coral   2.017094  6.0422105   4.410036   0.000000   3.382824   1.894682\nTR10773|c0_g1_i1_coral   0.000000  1.7263459   0.000000   0.000000   0.000000   0.000000\nTR10773|c1_g1_i1_coral   2.017094  0.8631729   5.292043   0.000000   0.000000   0.000000\ntemp = as.matrix(rowMeans(data4heatmap))\nhead(temp)\n[,1]\nTR10090|c0_g2_i1_coral 3.657565\nTR10487|c0_g1_i1_coral 6.571698\nTR1069|c0_g1_i1_coral  4.142052\nTR10748|c0_g1_i1_coral 2.936423\nTR10773|c0_g1_i1_coral 3.025158\nTR10773|c1_g1_i1_coral 2.328759\nscaledmatrix\u0026lt;-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)\ndata4heatmapscaled = data4heatmap/scaledmatrix\nhead(data4heatmapscaled)\nRI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22\nTR10090|c0_g2_i1_coral   0.000000  0.9483717  0.6692613   1.031502  0.0000000  0.0000000\nTR10487|c0_g1_i1_coral   1.665801  0.7917431  1.3037015   0.000000  0.0000000  1.0117598\nTR1069|c0_g1_i1_coral    0.000000  0.0000000  0.0000000   1.821699  0.8341687  1.6052378\nTR10748|c0_g1_i1_coral   0.000000  0.0000000  0.8336218   0.000000  0.0000000  0.0000000\nTR10773|c0_g1_i1_coral   0.000000  2.0065992  0.8091699   1.247137  0.0000000  0.0000000\nTR10773|c1_g1_i1_coral   0.000000  1.1171391  2.1022922   0.000000  0.0000000  0.9517198\nRI_B_03_14 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14\nTR10090|c0_g2_i1_coral  0.0000000  1.0441360  1.3149387  1.0073680          0  0.1559579\nTR10487|c0_g1_i1_coral  1.3710885  0.7748368  0.3659232  0.6852559          0  1.3888066\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000  0.0000000          0  0.0000000\nTR10748|c0_g1_i1_coral  0.8767109  0.0000000  1.3648899  0.9759263          0  0.5827764\nTR10773|c0_g1_i1_coral  0.4254975  0.0000000  0.0000000  0.1353286          0  0.0000000\nTR10773|c1_g1_i1_coral  2.2109576  0.5466423  0.0000000  0.0000000          0  0.1224742\nRI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22 RI_B_07_14\nTR10090|c0_g2_i1_coral  0.0000000  0.6708919          0  1.3200019          0  1.0209437\nTR10487|c0_g1_i1_coral  0.7659688  0.1244646          0  0.4198082          0  0.7955075\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000          0  0.0000000          0  0.0000000\nTR10748|c0_g1_i1_coral  0.0000000  0.0000000          0  0.0000000          0  1.0173377\nTR10773|c0_g1_i1_coral  0.0000000  0.0000000          0  0.0000000          0  0.0000000\nTR10773|c1_g1_i1_coral  0.0000000  0.3512357          0  0.0000000          0  0.0000000\nRI_B_07_18 RI_B_07_22 RI_W_01_18 RI_W_01_22 RI_W_02_14 RI_W_02_18\nTR10090|c0_g2_i1_coral  0.2758045          0   3.351196  2.1172541   1.271710  1.3656016\nTR10487|c0_g1_i1_coral  0.7675130          0   1.316578  0.9427084   2.477254  0.4560261\nTR1069|c0_g1_i1_coral   0.0000000          0   2.436999  2.2435236   0.000000  0.0000000\nTR10748|c0_g1_i1_coral  0.0000000          0   0.000000  0.5274440   1.584023  0.3401946\nTR10773|c0_g1_i1_coral  0.3334612          0   2.621728  5.6317019   0.000000  4.9532387\nTR10773|c1_g1_i1_coral  0.0000000          0   2.167287  7.9808953   0.000000  4.7186100\nRI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18\nTR10090|c0_g2_i1_coral  1.3178956  0.8324842  0.8870602  0.4325062   0.000000  2.0347150\nTR10487|c0_g1_i1_coral  0.1833730  1.1119920  1.0971223  2.1664531   2.602198  0.8304614\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000  0.0000000   0.000000  0.0000000\nTR10748|c0_g1_i1_coral  0.8207755  0.8295438  2.0870501  1.4365953   0.000000  0.1689607\nTR10773|c0_g1_i1_coral  1.9917509  7.6495091  4.1708314  9.5868908   0.000000  0.6560188\nTR10773|c1_g1_i1_coral  0.5174739  4.4455171  3.4056552  3.1700530   0.000000  0.6391473\nRI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22\nTR10090|c0_g2_i1_coral  0.9883436  2.7633244  0.3777539  0.8565921  0.3655346  2.1300602\nTR10487|c0_g1_i1_coral  0.6188351  0.6151858  0.0000000  0.6356634  2.0344306  1.5242310\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000\nTR10748|c0_g1_i1_coral  0.9232995  0.3441955  0.0000000  1.0669582  0.9106087  0.7580488\nTR10773|c0_g1_i1_coral  1.7924342  0.6681991  0.0000000  1.7261033  0.8838986  3.3111614\nTR10773|c1_g1_i1_coral  1.5522989  1.3020286  0.0000000  0.4484564  1.1482219  1.4337799\nRI_W_07_14 RI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22\nTR10090|c0_g2_i1_coral  0.5638500  0.3058329   1.509327  2.7900066  1.5387973   1.555733\nTR10487|c0_g1_i1_coral  0.5230302  2.0425836   1.260054  1.5528148  0.9881975   1.558555\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000  8.0069110  2.2995199   4.258662\nTR10748|c0_g1_i1_coral  0.9364308  0.7618816   0.000000  0.4343988  0.4423160   1.744018\nTR10773|c0_g1_i1_coral  0.4544816  0.7395340   0.000000  2.1082846  1.5742538   0.940479\nTR10773|c1_g1_i1_coral  0.8855865  1.4410293   1.185280  3.8342522  2.6027562   2.687789\nVA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14 VA_B_03_18 VA_B_03_22\nTR10090|c0_g2_i1_coral  0.0000000   0.000000  0.0000000   2.988074  1.1761509  0.9734764\nTR10487|c0_g1_i1_coral  0.0000000   0.000000  2.6122305   1.663052  1.2001041  1.1919624\nTR1069|c0_g1_i1_coral  20.9152088  10.634922  9.8432176   0.000000  0.0000000  0.0000000\nTR10748|c0_g1_i1_coral 10.4126549   0.000000  0.7307695   1.240633  2.6858257  1.2125476\nTR10773|c0_g1_i1_coral  0.5615127   0.000000  2.1280034   0.000000  0.4740081  0.2353962\nTR10773|c1_g1_i1_coral  0.0000000   1.112696  0.0000000   0.000000  1.5393918  0.3057897\nVA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22\nTR10090|c0_g2_i1_coral  1.9435111  0.8618804   0.000000  0.3825615  0.1511623   1.487610\nTR10487|c0_g1_i1_coral  2.2985841  0.6167454   1.260719  0.6919888  2.6080725   3.311796\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000  0.0000000  0.0000000   0.000000\nTR10748|c0_g1_i1_coral  0.9078031  0.0000000   2.997825  0.3573847  0.5648565   1.235297\nTR10773|c0_g1_i1_coral  0.0000000  2.3818420   2.738722  0.4625357  0.3655254   0.000000\nTR10773|c1_g1_i1_coral  0.3815614  3.6742606   1.111786  0.0000000  1.8993321   0.000000\nVA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22\nTR10090|c0_g2_i1_coral  0.0000000   1.150042  0.4039490  0.4077614  4.2520938   1.721227\nTR10487|c0_g1_i1_coral  0.0000000   1.280142  0.3747052  0.5673624  0.7888529   1.277295\nTR1069|c0_g1_i1_coral   7.0165371   4.062094  2.0212992  0.0000000  0.0000000   0.000000\nTR10748|c0_g1_i1_coral  2.3565165   0.000000  1.0063056  2.2855567  0.0000000   1.429289\nTR10773|c0_g1_i1_coral  0.9149579   0.000000  1.7907788  0.4930036  0.9792364   2.081048\nTR10773|c1_g1_i1_coral  0.0000000   1.083758  1.0574082  0.3202164  0.0000000   1.802247\nVA_W_01_14 VA_W_01_18 VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_22\nTR10090|c0_g2_i1_coral  0.0000000  1.0570771  1.1924571   3.576907  1.9129136  0.8560247\nTR10487|c0_g1_i1_coral  1.0194159  0.7564245  0.2654711   1.990774  0.4731810  1.4292954\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000   0.000000  0.0000000  0.0000000\nTR10748|c0_g1_i1_coral  0.9777629  0.7523882  0.2970613   0.000000  0.2647441  1.2795019\nTR10773|c0_g1_i1_coral  1.8981660  0.3651595  0.0000000   1.297397  0.5139571  0.8279809\nTR10773|c1_g1_i1_coral  0.0000000  1.6602522  0.0000000   2.247165  0.3338261  1.0755824\nVA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22\nTR10090|c0_g2_i1_coral  0.5614402   2.106741  2.9661383   0.000000  0.7515768   2.030831\nTR10487|c0_g1_i1_coral  0.5468346   1.978651  0.2251150   0.000000  1.2548999   1.318666\nTR1069|c0_g1_i1_coral   0.0000000   0.000000  0.0000000   0.000000  0.0000000   0.000000\nTR10748|c0_g1_i1_coral  2.2727947   3.116148  2.5190299   4.976688  0.9361528   3.372763\nTR10773|c0_g1_i1_coral  0.3394044   0.000000  0.1630094   0.000000  0.0000000   0.000000\nTR10773|c1_g1_i1_coral  0.4409008   0.000000  0.0000000   0.000000  0.9443452   0.531606\nVA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22\nTR10090|c0_g2_i1_coral  0.8272283  0.2359966   0.000000   1.508503  0.9248843  0.7770259\nTR10487|c0_g1_i1_coral  1.5346826  0.2626940   1.744769   1.079457  1.0295131  0.5766186\nTR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000   0.000000  0.0000000  0.0000000\nTR10748|c0_g1_i1_coral  0.6869221  2.0576770   1.501839   0.000000  1.1520220  0.6452345\nTR10773|c0_g1_i1_coral  0.0000000  0.5706631   0.000000   0.000000  0.0000000  0.0000000\nTR10773|c1_g1_i1_coral  0.8661668  0.3706578   2.272473   0.000000  0.0000000  0.0000000\ndim(data4heatmapscaled)\n[1] 866  78\npairs.breaks \u0026lt;- seq(0, 3.0, by=0.1)\nlength(pairs.breaks)\n[1] 31\nmycol \u0026lt;- colorpanel(n=30, low=\"black\", high=\"yellow\")\npdf(file=\"XXXXXX_byrow.pdf\",7,7)\nheatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c(\"row\"), scale=\"none\", keysize=1, breaks=pairs.breaks, col=mycol, trace = \"none\", symkey = F, density.info = \"density\", colsep=c(24), sepcolor=c(\"white\"), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\ndev.off()\nRStudioGD\n2\npdf(file=\"XXXXXX_bycolumn.pdf\",7,7)\nheatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c(\"col\"), scale=\"none\", keysize=1, breaks=pairs.breaks, col=mycol, trace = \"none\", symkey = F, density.info = \"density\", colsep=c(24), sepcolor=c(\"white\"), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)\ndev.off()\nRStudioGD\n2\nplotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\"Origin\")\nplotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\"SymState\")\nplotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\"Temp\")\nplotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\"Origin_SymState\")\nplotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\"Origin_SymState_Temp\")\npdf(file=\"866_RIvsVA_PCAv2.pdf\",14, 14)\nplotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup=\"Origin\")\ndev.off()\npdf\n4\nprop.nullv3 \u0026lt;- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100*mean(x==0))\nbarplot(prop.nullv3, main=\"Percentage of null counts per sample\",\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    horiz=TRUE, cex.names=0.5, las=1, \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    horiz=TRUE, cex.names=0.5, las=1, \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003epdf(file=\"ResNullCountsv2.pdf\",14, 14)\nbarplot(prop.nullv3, main=\"Percentage of null counts per sample\",\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    horiz=TRUE, cex.names=0.5, las=1, \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    horiz=TRUE, cex.names=0.5, las=1, \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003edev.off()\npdf\n4\ndistsv2\u0026lt;-dist(t(assay(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),])))\nplot(hclust(distsv2))\nheatmap.2(as.matrix(distsv2), key=F, trace=\"none\",\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"      col=colorpanel(100, \u0026quot;black\u0026quot;, \u0026quot;white\u0026quot;),\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e      col=colorpanel(100, \"black\", \"white\"),\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"      margin=c(10, 10))\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e      margin=c(10, 10))\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003eres\u0026lt;-resOrigin\nAnnotations\u0026lt;-read.delim(\"Assembly_GoodCoralSymbiont_suffixed_totalannotated.txt\")\nhead(res)\nlog2 fold change (MLE): Origin VA vs RI\nWald test p-value: Origin VA vs RI\nDataFrame with 6 rows and 6 columns\nbaseMean      log2FoldChange             lfcSE\n                    \nTR10083|c0_g2_i1_coral   5.366479327573  -0.271242746289843 0.446224512750353\nTR10090|c0_g1_i1_coral 2.70436251126018    1.11251170496561 0.589306266446119\nTR10090|c0_g2_i1_coral 3.65756460141883   0.943134641716211 0.474719393211504\nTR10103|c0_g1_i1_coral  2.9127886212765  0.0386642676540012 0.456719612483097\nTR10108|c0_g1_i1_coral 6.31144026709945 -0.0179009694277555 0.461957599914831\nTR10132|c0_g1_i1_coral  9.2673703948612  -0.264990306479184 0.677607561801266\nstat             pvalue              padj\n                   \nTR10083|c0_g2_i1_coral  -0.607861600022842  0.543279269700006 0.865660813784253\nTR10090|c0_g1_i1_coral    1.88783281005095 0.0590483989759312                NA\nTR10090|c0_g2_i1_coral    1.98672027139201 0.0469534041233123 0.418043000342497\nTR10103|c0_g1_i1_coral  0.0846564644854882  0.932534507970102 0.984590265229257\nTR10108|c0_g1_i1_coral -0.0387502433795997  0.969089515041603 0.992149683420423\nTR10132|c0_g1_i1_coral  -0.391067516682912  0.695747330435347 0.921516804624192\nsum(res$log2FoldChange!=0)\n[1] 7659\ngenes\u0026lt;-as.integer(p.adjust(res$pvalue[res$log2FoldChange!=0 \u0026amp; !is.na(res$pvalue)], method=\"BH\")\u0026lt;.05)\nnames(genes)\u0026lt;-row.names(res[res$log2FoldChange!=0 \u0026amp; !is.na(res$pvalue),])\ntable(genes)\ngenes\n0    1\n7533  116\ngoterms\u0026lt;-strsplit(as.character(Annotations$GO), split=\" // \")\nnames(goterms)\u0026lt;-Annotations$ContigName\npwf\u0026lt;-nullp(genes, bias.data=Annotations[Annotations$ContigName%in%names(genes),\"ContigLength\"])\nWarning message:\nIn pcls(G) : initial point very close to some inequality constraints\nGOGOGO\u0026lt;-goseq(pwf,gene2cat=goterms)\nUsing manually entered categories.\nCalculating the p-values...\n'select()' returned 1:1 mapping between keys and columns\nGOGOGO$padj\u0026lt;-p.adjust(GOGOGO$over_represented_pvalue, method=\"fdr\")\nsum(GOGOGO$padj\u0026lt;0.2, na.rm=T)\n[1] 3\nsum(GOGOGO$padj\u0026lt;0.1, na.rm=T)\n[1] 2\nsum(GOGOGO$padj\u0026lt;0.05, na.rm=T)\n[1] 2\nGOGOGO[GOGOGO$padj\u0026lt;0.05,]\ncategory over_represented_pvalue under_represented_pvalue numDEInCat numInCat\n1326 GO:0006032            3.838364e-08                        1          5        8\n74   GO:0000272            8.726516e-07                        1          5       13\nterm ontology         padj\n1326         chitin catabolic process       BP 0.0003557012\n74   polysaccharide catabolic process       BP 0.0040434313\nGOTERM[[\"SOMEGOTERM\"]]\nNULL\nEnGOS\u0026lt;-GOGOGO$category[p.adjust(GOGOGO$over_represented_pvalue, method=\"BH\")\u0026lt;.05]\nfor(go in EnGOS){\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eprint(GOTERM[[go]])\u003c/li\u003e\n\u003cli\u003ecat(\"--------------------------------------\\n\")}\nGOID: GO:0006032\nTerm: chitin catabolic process\nOntology: BP\nDefinition: The chemical reactions and pathways resulting in the breakdown of\nchitin, a linear polysaccharide consisting of beta-(1-\u0026gt;4)-linked\nN-acetyl-D-glucosamine residues.\nSynonym: beta-1,4-linked N-acetylglucosamine catabolic process\nSynonym: beta-1,4-linked N-acetylglucosamine catabolism\nSynonym: chitin breakdown\nSynonym: chitin catabolism\nSynonym: chitin degradation\u003c/li\u003e\n\u003c/ul\u003e\n\u003chr\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGOID: GO:0000272\nTerm: polysaccharide catabolic process\nOntology: BP\nDefinition: The chemical reactions and pathways resulting in the breakdown of a\npolysaccharide, a polymer of many (typically more than 10) monosaccharide\nresidues linked glycosidically.\nSynonym: multicellular organismal polysaccharide catabolic process\nSynonym: polysaccharide breakdown\nSynonym: polysaccharide catabolism\nSynonym: polysaccharide degradation\nSynonym: GO:0044244\nSecondary: GO:0044244\u003c/h2\u003e\u003ca id=\"user-content-goid-go0000272term-polysaccharide-catabolic-processontology-bpdefinition-the-chemical-reactions-and-pathways-resulting-in-the-breakdown-of-apolysaccharide-a-polymer-of-many-typically-more-than-10-monosaccharideresidues-linked-glycosidicallysynonym-multicellular-organismal-polysaccharide-catabolic-processsynonym-polysaccharide-breakdownsynonym-polysaccharide-catabolismsynonym-polysaccharide-degradationsynonym-go0044244secondary-go0044244\" class=\"anchor\" aria-label=\"Permalink: GOID: GO:0000272\nTerm: polysaccharide catabolic process\nOntology: BP\nDefinition: The chemical reactions and pathways resulting in the breakdown of a\npolysaccharide, a polymer of many (typically more than 10) monosaccharide\nresidues linked glycosidically.\nSynonym: multicellular organismal polysaccharide catabolic process\nSynonym: polysaccharide breakdown\nSynonym: polysaccharide catabolism\nSynonym: polysaccharide degradation\nSynonym: GO:0044244\nSecondary: GO:0044244\" href=\"#goid-go0000272term-polysaccharide-catabolic-processontology-bpdefinition-the-chemical-reactions-and-pathways-resulting-in-the-breakdown-of-apolysaccharide-a-polymer-of-many-typically-more-than-10-monosaccharideresidues-linked-glycosidicallysynonym-multicellular-organismal-polysaccharide-catabolic-processsynonym-polysaccharide-breakdownsynonym-polysaccharide-catabolismsynonym-polysaccharide-degradationsynonym-go0044244secondary-go0044244\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cblockquote\u003e\n\u003cp dir=\"auto\"\u003enames(Annotations)\n[1] \"ContigName\"              \"ContigLength\"\u003cbr\u003e\n[3] \"topnrMatch\"              \"topnrEvalue\"\u003cbr\u003e\n[5] \"topnrAlignmentLength\"    \"topnrPercMatch\"\u003cbr\u003e\n[7] \"nobadwordnrMatch\"        \"nobadwordnrEvalue\"\u003cbr\u003e\n[9] \"nobadwordnrAlignLength\"  \"nobadwordnrPercentMatch\"\n[11] \"Uniprotmatch\"            \"X._identity\"\u003cbr\u003e\n[13] \"evalue\"                  \"ID\"\u003cbr\u003e\n[15] \"Description\"             \"KEGG\"\u003cbr\u003e\n[17] \"KEGGKO\"                  \"PFAM\"\u003cbr\u003e\n[19] \"GO\"                      \"GOCC\"\u003cbr\u003e\n[21] \"GOBP\"                    \"GOMF\"\u003cbr\u003e\n[23] \"Keywords\"\u003cbr\u003e\nhead(Annotations)\nContigName ContigLength\n1   TR10030|c0_g1_i1_sym          523\n2   TR10043|c0_g1_i1_sym          512\n3   TR10050|c0_g1_i1_sym          793\n4 TR10054|c0_g1_i1_coral          548\n5 TR10054|c0_g2_i1_coral          811\n6   TR10082|c0_g2_i1_sym         1176\ntopnrMatch\n1                                                                           gnl|BL_ORD_ID|101380155 OLP92908.1 Sodium/calcium exchanger 1 [Symbiodinium microadriaticum]\n2                                                             gnl|BL_ORD_ID|101372287 OLP84974.1 hypothetical protein AK812_SmicGene34099 [Symbiodinium microadriaticum]\n3                                                                                                                                                          No_sig_nr_hit\n4                                                                                     gnl|BL_ORD_ID|110384373 XP_020612383.1 titin-like isoform X4 [Orbicella faveolata]\n5                                                                                     gnl|BL_ORD_ID|110384422 XP_020612382.1 titin-like isoform X3 [Orbicella faveolata]\n6 gnl|BL_ORD_ID|14605113 XP_002785881.1 sugar transporter, putative [Perkinsus marinus ATCC 50983] EER17677.1 sugar transporter, putative [Perkinsus marinus ATCC 50983]\ntopnrEvalue topnrAlignmentLength topnrPercMatch\n1   5.74325e-73                  167           0.74\n2   5.42819e-23                   80           0.66\n3 No_sig_nr_hit        No_sig_nr_hit  No_sig_nr_hit\n4   8.93129e-73                  182           0.90\n5   1.31886e-95                  250           0.82\n6   1.07378e-49                  289           0.37\nnobadwordnrMatch\n1                                                                           gnl|BL_ORD_ID|101380155 OLP92908.1 Sodium/calcium exchanger 1 [Symbiodinium microadriaticum]\n2                                                                  gnl|BL_ORD_ID|51120734 XP_012166226.1 E3 ubiquitin-protein ligase RNF146, partial [Bombus terrestris]\n3                                                                                                                                                          No_sig_nr_hit\n4                                                                                     gnl|BL_ORD_ID|110384373 XP_020612383.1 titin-like isoform X4 [Orbicella faveolata]\n5                                                                                     gnl|BL_ORD_ID|110384422 XP_020612382.1 titin-like isoform X3 [Orbicella faveolata]\n6 gnl|BL_ORD_ID|14605113 XP_002785881.1 sugar transporter, putative [Perkinsus marinus ATCC 50983] EER17677.1 sugar transporter, putative [Perkinsus marinus ATCC 50983]\nnobadwordnrEvalue nobadwordnrAlignLength nobadwordnrPercentMatch\n1       5.74325e-73                    167                    0.74\n2        8.8118e-05                     58                    0.38\n3     No_sig_nr_hit          No_sig_nr_hit           No_sig_nr_hit\n4       8.93129e-73                    182                    0.90\n5       1.31886e-95                    250                    0.82\n6       1.07378e-49                    289                    0.37\nUniprotmatch   X.\u003cem\u003eidentity    evalue               ID\n1      A0A1Q9DCJ4        74.251  3.76e-73 A0A1Q9DCJ4_SYMMI\n2          C0HBT3        29.114  3.65e-05      RN146_SALSA\n3 No_Uniprotmatch No\u003c/em\u003e%_identity No_evalue            No_ID\n4      A0A2B4SVR7        77.473  8.77e-57 A0A2B4SVR7_STYPI\n5      A0A2B4SVR7        72.112  5.57e-76 A0A2B4SVR7_STYPI\n6          Q02334        21.402  2.12e-05      UGTP1_CAEEL\nDescription\n1                            Sodium/calcium exchanger 1 {ECO:0000313|EMBL:OLP92908.1}\n2                                                  E3 ubiquitin-protein ligase rnf146\n3                                                                      No_Description\n4 Golgi-associated plant pathogenesis-related protein 1 {ECO:0000313|EMBL:PFX34774.1}\n5 Golgi-associated plant pathogenesis-related protein 1 {ECO:0000313|EMBL:PFX34774.1}\n6                                                        UDP-galactose translocator 1\nKEGG    KEGGKO                                  PFAM\n1          No_KEGG No_KEGGKO PF00063; Myosin_head,  PF07647; SAM_2\n2   sasa:100380617    K15700                          PF02825; WWE\n3          No_KEGG No_KEGGKO                               No_PFAM\n4          No_KEGG No_KEGGKO                          PF00188; CAP\n5          No_KEGG No_KEGGKO                          PF00188; CAP\n6 cel:CELE_ZK370.7    K15272               PF04142; Nuc_sug_transp\nGO\n1                                                                                     GO:0016021 // GO:0016459 // GO:0005524 // GO:0003774 // GO:0003676\n2 GO:0005829 // GO:0005634 // GO:0072572 // GO:0061630 // GO:0004842 // GO:0008270 // GO:0090263 // GO:0051865 // GO:0070936 // GO:0006511 // GO:0016055\n3                                                                                                                                                  No_GO\n4                                                                                                                                             GO:0005576\n5                                                                                                                                             GO:0005576\n6                                                         GO:0060473 // GO:0005794 // GO:0005797 // GO:0030173 // GO:0015165 // GO:0015136 // GO:0008643\nGOCC\n1                                                                                                                                                                                                                                                   No_GOBP\n2 positive regulation of canonical Wnt signaling pathway (GO:0090263);protein autoubiquitination (GO:0051865);protein K48-linked ubiquitination (GO:0070936);ubiquitin-dependent protein catabolic process (GO:0006511);Wnt signaling pathway (GO:0016055);\n3                                                                                                                                                                                                                                                   No_GOCC\n4                                                                                                                                                                                                                                                   No_GOBP\n5                                                                                                                                                                                                                                                   No_GOBP\n6                                                                                                                                                                                                                      carbohydrate transport (GO:0008643);\nGOBP\n1                                                                                  ATP binding (GO:0005524);motor activity (GO:0003774);nucleic acid binding (GO:0003676);\n2 poly-ADP-D-ribose binding (GO:0072572);ubiquitin protein ligase activity (GO:0061630);ubiquitin-protein transferase activity (GO:0004842);zinc ion binding (GO:0008270);\n3                                                                                                                                                                  No_GOBP\n4                                                                                                                                                                  No_GOMF\n5                                                                                                                                                                  No_GOMF\n6                                 pyrimidine nucleotide-sugar transmembrane transporter activity (GO:0015165);sialic acid transmembrane transporter activity (GO:0015136);\nGOMF\n1                                                                         integral component of membrane (GO:0016021);myosin complex (GO:0016459);\n2                                                                                                       cytosol (GO:0005829);nucleus (GO:0005634);\n3                                                                                                                                          No_GOMF\n4                                                                                                               extracellular region (GO:0005576);\n5                                                                                                               extracellular region (GO:0005576);\n6 cortical granule (GO:0060473);Golgi apparatus (GO:0005794);Golgi medial cisterna (GO:0005797);integral component of Golgi membrane (GO:0030173);\nKeywords\n1 ATP-binding {ECO:0000256|SAAS:SAAS00875240};Coiled coil {ECO:0000256|SAM:Coils};Membrane {ECO:0000256|SAM:Phobius};Metal-binding {ECO:0000256|PROSITE-ProRule:PRU00175};Motor protein {ECO:0000256|SAAS:SAAS01033784};Myosin {ECO:0000256|SAAS:SAAS01033784};Nucleotide-binding {ECO:0000256|SAAS:SAAS00875240};Transmembrane {ECO:0000256|SAM:Phobius};Transmembrane helix {ECO:0000256|SAM:Phobius};Zinc {ECO:0000256|PROSITE-ProRule:PRU00175};Zinc-finger {ECO:0000256|PROSITE-ProRule:PRU00175}.;\n2                                                                                                                                                                                                                                                                                                                                                      Complete proteome;Cytoplasm;Metal-binding;Nucleus;Reference proteome;Transferase;Ubl conjugation pathway;Wnt signaling pathway;Zinc;Zinc-finger.;\n3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            No_keywords\n4                                                                                                                                                                                                                                                                                                                                                     Coiled coil {ECO:0000256|SAM:Coils};Complete proteome {ECO:0000313|Proteomes:UP000225706};Reference proteome {ECO:0000313|Proteomes:UP000225706}.;\n5                                                                                                                                                                                                                                                                                                                                                     Coiled coil {ECO:0000256|SAM:Coils};Complete proteome {ECO:0000313|Proteomes:UP000225706};Reference proteome {ECO:0000313|Proteomes:UP000225706}.;\n6                                                                                                                                                                                                                                                                                                                                                                                            Complete proteome;Membrane;Reference proteome;Sugar transport;Transmembrane;Transmembrane helix;Transport.;\ndim(genes4heatmap)\n[1] 866   6\u003c/p\u003e\n\u003c/blockquote\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":4,"text":"this script take the entries in the first column of table and renames (mv's) them to files with the nam$","anchor":"this-script-take-the-entries-in-the-first-column-of-table-and-renames-mvs-them-to-files-with-the-nam","htmlText":"this script take the entries in the first column of table and renames (mv's) them to files with the nam$"},{"level":1,"text":"print 'mv %s %s' %(cols[0], cols[1])","anchor":"print-mv-s-s-cols0-cols1","htmlText":"print 'mv %s %s' %(cols[0], cols[1])"},{"level":2,"text":"Counts of transcripts, etc.","anchor":"counts-of-transcripts-etc","htmlText":"Counts of transcripts, etc."},{"level":2,"text":"Stats based on ONLY LONGEST ISOFORM per 'GENE':","anchor":"stats-based-on-only-longest-isoform-per-gene","htmlText":"Stats based on ONLY LONGEST ISOFORM per 'GENE':"},{"level":4,"text":"PCAs","anchor":"pcas","htmlText":"PCAs"},{"level":1,"text":"individual labels","anchor":"individual-labels","htmlText":"individual labels"},{"level":1,"text":"population elipses","anchor":"population-elipses","htmlText":"population elipses"},{"level":4,"text":"Snapclust","anchor":"snapclust","htmlText":"Snapclust"},{"level":1,"text":"Sample\t\tOrigin\tSymState\tTemp  Origin_SymState Origin_SymState_Temp","anchor":"sampleoriginsymstatetemp--origin_symstate-origin_symstate_temp","htmlText":"Sample\t\tOrigin\tSymState\tTemp  Origin_SymState Origin_SymState_Temp"},{"level":1,"text":"RI_B_06_18\t\tRI\tB\t18  RI_B  RI_B_18","anchor":"ri_b_06_18rib18--ri_b--ri_b_18","htmlText":"RI_B_06_18\t\tRI\tB\t18  RI_B  RI_B_18"},{"level":1,"text":"VA_B_07_14\t\tVA\tB\t14  VA_B  VA_B_14","anchor":"va_b_07_14vab14--va_b--va_b_14","htmlText":"VA_B_07_14\t\tVA\tB\t14  VA_B  VA_B_14"},{"level":1,"text":"VA_W_07_22\t\tVA\tW\t22  VA_W  VA_W_22","anchor":"va_w_07_22vaw22--va_w--va_w_22","htmlText":"VA_W_07_22\t\tVA\tW\t22  VA_W  VA_W_22"},{"level":1,"text":"safest way to do this is in R","anchor":"safest-way-to-do-this-is-in-r","htmlText":"safest way to do this is in R"},{"level":1,"text":"how many genes have mean count \u003e=3","anchor":"how-many-genes-have-mean-count-3","htmlText":"how many genes have mean count \u0026gt;=3"},{"level":2,"text":"GOID: GO:0000272\nTerm: polysaccharide catabolic process\nOntology: BP\nDefinition: The chemical reactions and pathways resulting in the breakdown of a\npolysaccharide, a polymer of many (typically more than 10) monosaccharide\nresidues linked glycosidically.\nSynonym: multicellular organismal polysaccharide catabolic process\nSynonym: polysaccharide breakdown\nSynonym: polysaccharide catabolism\nSynonym: polysaccharide degradation\nSynonym: GO:0044244\nSecondary: GO:0044244","anchor":"goid-go0000272term-polysaccharide-catabolic-processontology-bpdefinition-the-chemical-reactions-and-pathways-resulting-in-the-breakdown-of-apolysaccharide-a-polymer-of-many-typically-more-than-10-monosaccharideresidues-linked-glycosidicallysynonym-multicellular-organismal-polysaccharide-catabolic-processsynonym-polysaccharide-breakdownsynonym-polysaccharide-catabolismsynonym-polysaccharide-degradationsynonym-go0044244secondary-go0044244","htmlText":"GOID: GO:0000272\nTerm: polysaccharide catabolic process\nOntology: BP\nDefinition: The chemical reactions and pathways resulting in the breakdown of a\npolysaccharide, a polymer of many (typically more than 10) monosaccharide\nresidues linked glycosidically.\nSynonym: multicellular organismal polysaccharide catabolic process\nSynonym: polysaccharide breakdown\nSynonym: polysaccharide catabolism\nSynonym: polysaccharide degradation\nSynonym: GO:0044244\nSecondary: GO:0044244"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Florenapr92%2F21SPStephGAdvanceGenomicslog"}}],"overviewFilesProcessingTime":110.10399899999999,"createFromTemplatePath":"/new?template_name=21SPStephGAdvanceGenomicslog\u0026template_owner=lorenapr92"}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":"https://github.dev/","enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":true,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":true}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cLeluw:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cLeluw:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cLeluw[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cLeluw:active{-webkit-transition:none;transition:none;}/*!sc*/
.cLeluw[data-inactive]{cursor:auto;}/*!sc*/
.cLeluw:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled [data-component=ButtonCounter],.cLeluw:disabled [data-component="leadingVisual"],.cLeluw:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cLeluw:focus{outline:solid 1px transparent;}}/*!sc*/
.cLeluw [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cLeluw[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cLeluw[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cLeluw[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cLeluw[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cLeluw[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cLeluw[data-block="block"]{width:100%;}/*!sc*/
.cLeluw[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cLeluw[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cLeluw [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cLeluw [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cLeluw [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cLeluw [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cLeluw:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cLeluw[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cLeluw[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cLeluw:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cLeluw:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cLeluw[data-size="small"][data-no-visuals]{margin-right:8px;height:28px;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,cLeluw,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/lorenapr92/21SPStephGAdvanceGenomicslog/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/lorenapr92/21SPStephGAdvanceGenomicslog/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <button hidden="">
                  </button>
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" aria-label="Add file" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R2hil5:" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-plus" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="react-directory-remove-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 pyyxt" id=":R2lil5:" tabindex="0" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Add file
                     </span>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </button>
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/lorenapr92/21SPStephGAdvanceGenomicslog/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           17 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/lorenapr92/21SPStephGAdvanceGenomicslog/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/lorenapr92/21SPStephGAdvanceGenomicslog/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/lorenapr92/21SPStephGAdvanceGenomicslog/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-label="Edit file" class="types__StyledButton-sc-ws60qy-0 cLeluw" data-component="IconButton" data-no-visuals="true" data-size="small" title="Edit file" type="button">
                    <svg aria-hidden="true" class="octicon octicon-pencil" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </button>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <p dir="auto">
                     #Stephanie Grizzard's README
#21SPStephGAdvanceGenomicsLog repository
                    </p>
                    <p dir="auto">
                     #Exercise2
cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG
mkdir data
mkdir exercises
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/exercises
cp /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day02/Exercise2.fast* ./
ls -alh
13M
gunzip -c Exercise2.fasta.gz &gt; Exercise2.fasta
ls -alh
37M
tar -zxvf Exercise2.fastq.tar.gz
ls -alh
62M
drwxrwxrwx 2 spere004 users  142 Jan 22 06:38 .
drwxrwxrwx 5 spere004 users  141 Jan 22 06:34 ..
-rwxrwxrwx 1 spere004 users  17M Jan 22 06:37 Exercise2.fasta
-rwxr-xr-x 1 spere004 users 4.3M Jan 22 06:35 Exercise2.fasta.gz
-rw-r--r-- 1 spere004 users  18M Sep  2  2015 Exercise2.fastq
-rwxr-xr-x 1 spere004 users 4.3M Jan 22 06:35 Exercise2.fastq.tar.gz
                    </p>
                    <p dir="auto">
                     [spere004@turing1 exercises]$ head -1 Exercise2.fasta
                    </p>
                    <blockquote>
                     <p dir="auto">
                      scaffold10078|size20675
[spere004@turing1 exercises]$ grep -c '&gt;' Exercise2.fasta
138
[spere004@turing1 exercises]$ head -1 Exercise2.fastq
@HS3:541:HAYTUADXX:1:1101:1297:1938 1:N:0:AGTTCC
[spere004@turing1 exercises]$ grep -c '@HS3' Exercise2.fastq
61304
[spere004@turing1 exercises]$ wc -l Exercise2.fastq
245216 Exercise2.fastq
cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/scripts/
                     </p>
                    </blockquote>
                    <p dir="auto">
                     [spere004@turing1 scripts]$ salloc
salloc: Pending job allocation 9268786
salloc: job 9268786 queued and waiting for resources
salloc: job 9268786 has been allocated resources
salloc: Granted job allocation 9268786
This session will be terminated in 7 days. If your application requires
a longer excution time, please use command "salloc -t N-0" where N is the
number of days that you need.
cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/exercises
                    </p>
                    <p dir="auto">
                     [spere004@coreV3-23-024 exercises]$ /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/scripts/avg_cov_len_fasta_advbioinf.py Exercise2.fasta
The total number of sequences is 138
The average sequence length is 126640
The total number of bases is 17476447
The minimum sequence length is 1122
The maximum sequence length is 562680
The N50 is 187037
Median Length = 92612
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 138
contigs &gt;= 1000bp = 138
contigs &gt;= 2000bp = 135
                    </p>
                    <p dir="auto">
                     stephanieperez@MacBook-Pro-2 ~/D/D/21sp_advgenomics&gt; cd 21SPStephGAdvancedGenomicsLog/
stephanieperez@MacBook-Pro-2 ~/D/D/2/21SPStephGAdvancedGenomicsLog&gt; ls
README.md
stephanieperez@MacBook-Pro-2 ~/D/D/2/21SPStephGAdvancedGenomicsLog&gt;  git add README.md
stephanieperez@MacBook-Pro-2 ~/D/D/2/21SPStephGAdvancedGenomicsLog&gt;
git commit -m 'updating with day02 exercises'
[main 552e36f] updating with day02 exercises
1 file changed, 81 insertions(+)
stephanieperez@MacBook-Pro-2 ~/D/D/2/21SPStephGAdvancedGenomicsLog&gt; git push -u origin main
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 8 threads
Compressing objects: 100% (2/2), done.
Writing objects: 100% (3/3), 1.37 KiB | 1.38 MiB/s, done.
Total 3 (delta 0), reused 0 (delta 0), pack-reused 0
To
                     <a href="https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog.git">
                      https://github.com/lorenapr92/21SPStephGAdvanceGenomicslog.git
                     </a>
                     e726f87..552e36f  main -&gt; main
Branch 'main' set up to track remote branch 'main' from 'origin'.
                    </p>
                    <p dir="auto">
                     #Homework2
cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/
emacs emacs StephFQCp.sh
inside shell script
#!/bin/bash -l
                    </p>
                    <p dir="auto">
                     #SBATCH -o stephfastqcope.txt
                     <br/>
                     #SBATCH -n 1
                     <br/>
                     #SBATCH --mail-user=
                     <a href="mailto:spere004@odu.edu">
                      spere004@odu.edu
                     </a>
                     <br/>
                     #SBATCH --mail-type=END
                     <br/>
                     #SBATCH --job-name=StephFQCp
                    </p>
                    <p dir="auto">
                     cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/originalfastqs/*.fastq.gz /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data
                    </p>
                    <p dir="auto">
                     sbatch StephFQCp.sh
#Submitted batch job 9268702
squeue -u spere004
JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
9268702      main StephFQC spere004  R       0:19      1 coreV3-23-044
                    </p>
                    <p dir="auto">
                     emacs StephGZfq.sh
inside shell script
#!/bin/bash -l
                    </p>
                    <p dir="auto">
                     #SBATCH -o stephgzfastqcopy.txt
                     <br/>
                     #SBATCH -n 1
                     <br/>
                     #SBATCH --mail-user=
                     <a href="mailto:spere004@odu.edu">
                      spere004@odu.edu
                     </a>
                     <br/>
                     #SBATCH --mail-type=END
                     <br/>
                     #SBATCH --job-name=StephGZfq
                    </p>
                    <p dir="auto">
                     gunzip /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/*.fastq.gz
                    </p>
                    <p dir="auto">
                     [spere004@turing1 data]$ sbatch StephGZfq.sh
Submitted batch job 9268787
[spere004@turing1 data]$ squeue -u spere004
JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
9268786      main       sh spere004  R       7:52      1 coreV3-23-024
9268787      main StephGZf spere004  R       0:25      1 coreV1-22-005
                    </p>
                    <p dir="auto">
                     [spere004@turing1 data]$ ls
HADB01-A_S17_L002_R1_001.fastq     HADB03-C_S51_L004_R1_001.fastq.gz  HADB05-F_S86_L006_R1_001.fastq.gz
HADB01-B_S18_L002_R1_001.fastq     HADB03-D_S52_L004_R1_001.fastq.gz  HADB05-G_S87_L006_R1_001.fastq.gz
HADB01-C_S19_L002_R1_001.fastq     HADB03-E_S53_L004_R1_001.fastq.gz  HADB05-H_S88_L006_R1_001.fastq.gz
HADB01-D_S20_L002_R1_001.fastq     HADB03-F_S54_L004_R1_001.fastq.gz  HADB05-I_S89_L006_R1_001.fastq.gz
HADB01-E_S21_L002_R1_001.fastq     HADB03-G_S55_L004_R1_001.fastq.gz  HADB05-J_S90_L006_R1_001.fastq.gz
HADB01-F_S22_L002_R1_001.fastq
                    </p>
                    <p dir="auto">
                     DAY03
                    </p>
                    <ol dir="auto">
                     <li>
                      <p dir="auto">
                       cp -r /cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03 ./
pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       mkdir fastq
mv *.fastq fastq/
ls
cd fastq/
ls
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       cp renamingtable_complete.txt /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/scripts
cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py ./
less renamer_advbioinf.py
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       module load python
python renamer_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/renamingtable_complete.txt /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/*.fastq
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       nano renamer_advbioinf.py
#!/usr/bin/env python
####usage renamer.py renamingtable
                      </p>
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      this script take the entries in the first column of table and renames (mv's) them to files with the nam$
                     </h4>
                     <a aria-label="Permalink: this script take the entries in the first column of table and renames (mv's) them to files with the nam$" class="anchor" href="#this-script-take-the-entries-in-the-first-column-of-table-and-renames-mvs-them-to-files-with-the-nam" id="user-content-this-script-take-the-entries-in-the-first-column-of-table-and-renames-mvs-them-to-files-with-the-nam">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     import sys
import os
                    </p>
                    <p dir="auto">
                     fin=open(sys.argv[1],'r')
linecount=0
for line in fin:
linecount+=1
if linecount&gt;=2:
cols=line.rstrip().split('\t')
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      print 'mv %s %s' %(cols[0], cols[1])
                     </h1>
                     <a aria-label="Permalink: print 'mv %s %s' %(cols[0], cols[1])" class="anchor" href="#print-mv-s-s-cols0-cols1" id="user-content-print-mv-s-s-cols0-cols1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="            os.popen('mv %s %s' %(cols[0], cols[1]))">
                     <pre class="notranslate"><code>            os.popen('mv %s %s' %(cols[0], cols[1]))
</code></pre>
                    </div>
                    <ol dir="auto" start="6">
                     <li>
                      nano Steph_Renamer.sh
#!/bin/bash -l
                     </li>
                    </ol>
                    <p dir="auto">
                     #SBATCH -o steph_renamercopy.txt
#SBATCH -n 1
#SBATCH --mail-user=
                     <a href="mailto:spere004@odu.edu">
                      spere004@odu.edu
                     </a>
                     #SBATCH --mail-type=END
#SBATCH --job-name=Steph_Renamer
                    </p>
                    <p dir="auto">
                     python /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/scripts/renamer_advbioinf.py renamingtable_complete.txt
                    </p>
                    <p dir="auto">
                     sbatch Steph_Renamer.sh
Submitted batch job 9271073
                    </p>
                    <ol dir="auto" start="7">
                     <li>
                      DONE
                     </li>
                    </ol>
                    <p dir="auto">
                     [spere004@turing1 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/scripts
10. cp /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Trimclipfilterstatsbatch_advbioinf.py ./
11. less Trimclipfilterstatsbatch_advbioinf.py
12. pwd
/cm/shared/courses/dbarshis/21AdvGenomics/assignments_exercises/day03
[spere004@turing1 day03]$ cp adapterlist_advbioinf.txt /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/
13. [spere004@turing1 fastq]$ nano Steph_Trimclipfilter.sh
#!/bin/bash -l
                    </p>
                    <p dir="auto">
                     #SBATCH -o steph_trimmerclip.txt
#SBATCH -n 4
#SBATCH --mail-user=
                     <a href="mailto:spere004@odu.edu">
                      spere004@odu.edu
                     </a>
                     #SBATCH --mail-type=END
#SBATCH --job-name=Steph_Trimmerclipfilter
                    </p>
                    <p dir="auto">
                     python /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/scripts/Trimclipfilterstatsbatch_advbioinf.py adapterlist_advbioinf.txt *.fastq
                    </p>
                    <p dir="auto">
                     [spere004@turing1 fastq]$ sbatch Steph_Trimclipfilter.sh
Submitted batch job 9271091
                    </p>
                    <p dir="auto">
                     Homework04
                    </p>
                    <p dir="auto">
                     1a.  [spere004@turing1 fastq]$ /cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py -h
Written by Peter Schafran
                     <a href="mailto:pscha005@odu.edu">
                      pscha005@odu.edu
                     </a>
                     5-Oct-2015
                    </p>
                    <p dir="auto">
                     This script takes a stats output file from fastx_clipper and converts it into a table.
                    </p>
                    <p dir="auto">
                     Usage: Schafran_trimstatstable.py [-c, -v, -h] inputfile.txt outputfile.txt
                    </p>
                    <p dir="auto">
                     Options (-c and -v must be listed separately to run together):
-h	Display this help message
-c	Use comma delimiter instead of tabs
-v	Verbose mode (print steps to stdout)
                    </p>
                    <p dir="auto">
                     1b.   pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/filteringstats
/cm/shared/courses/dbarshis/21AdvGenomics/scripts/Schafran_trimstatstable_advbioinf_clippedtrimmed.py trimclipstats.txt Steph_trimclipstatsout.txt
head -10 Steph_trimclipstatsout.txt
tail -n +2 Steph_trimclipstatsout.txt | head
1c.  tail -n +2 Steph_trimclipstatsout.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/Fulltrimclipstatstable.txt
                    </p>
                    <p dir="auto">
                     [spere004@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs
3. nano Steph_clippedtrimmedbowtie.sh
                    </p>
                    <p dir="auto">
                     #!/bin/bash -l
                    </p>
                    <p dir="auto">
                     #SBATCH -o steph_clippedtrimmedbowtie.txt
#SBATCH -n 1
#SBATCH --mail-user=
                     <a href="mailto:spere004@odu.edu">
                      spere004@odu.edu
                     </a>
                     #SBATCH --mail-type=END
#SBATCH --job-name=Steph_clippedtrimmedbowtie
                    </p>
                    <p dir="auto">
                     enable_lmod
module load bowtie2/2.4
for i in *_clippedtrimmed.fastq; do bowtie2 --rg-id ${i%_clippedtrimmed.fastq}
                     <br/>
                     --rg SM:${i%_clippedtrimmed.fastq}
                     <br/>
                     --very-sensitive -x /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/Apoc_hostsym -U $i \
                    </p>
                    <blockquote>
                     <p dir="auto">
                      ${i%_clippedtrimmedfilterd.fastq}.sam --no-unal -k 5; done
                     </p>
                    </blockquote>
                    <ol dir="auto" start="3">
                     <li>
                      sbatch Steph_clippedtrimmedbowtie.sh
Submitted batch job 9271724
                     </li>
                    </ol>
                    <p dir="auto">
                     Homework05
#Working with John. He copied his fastq files into my sandbox. Created a list with his fast names for shell script
pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs
ls *.fastq | head -32
nano Steph_trinity.sh
#! /bin/bash -l
                    </p>
                    <p dir="auto">
                     #SBATCH -o steph_trinity.txt
#SBATCH -n 32
#SBATCH -p himem
#SBATCH --mail-user=
                     <a href="mailto:spere004@odu.edu">
                      spere004@odu.edu
                     </a>
                     #SBATCH --mail-type=END
#SBATCH --job-name=Steph_trinity
                    </p>
                    <p dir="auto">
                     enable_lmod
module load container_env trinity
                    </p>
                    <p dir="auto">
                     crun Trinity --seqType fq --max_memory 768G --normalize_reads --single RI_B_05_18_clippedtrimmed.fastq,RI_B_05_22_clippedtrimmed.fastq,RI_B_06_14_clippedtrimmed.fastq,RI_B_06_22_clippedtrimmed.fastq,RI_W_05_18_clippedtrimmed.fastq,RI_W_05_22_clippedtrimmed.fastq,RI_W_06_14_clippedtrimmed.fastq,RI_W_06_22_clippedtrimmed.fastq,VA_B_05_18_clippedtrimmed.fastq,VA_B_05_22_clippedtrimmed.fastq,VA_B_06_14_clippedtrimmed.fastq,VA_B_06_22_clippedtrimmed.fastq,VA_W_05_18_clippedtrimmed.fastq,VA_W_05_22_clippedtrimmed.fastq,VA_W_06_14_clippedtrimmed.fastq,VA_W_06_22_clippedtrimmed.fastq,RI_B_02_14_clippedtrimmed.fastq,RI_B_02_18_clippedtrimmed.fastq,RI_B_02_22_clippedtrimmed.fastq,RI_B_09_SNP_clippedtrimmed.fastq,RI_W_02_14_clippedtrimmed.fastq,RI_W_02_18_clippedtrimmed.fastq,RI_W_02_22_clippedtrimmed.fastq,RI_W_09_SNP_clippedtrimmed.fastq,VA_B_02_14_clippedtrimmed.fastq,VA_B_02_18_clippedtrimmed.fastq,VA_B_02_22_clippedtrimmed.fastq,VA_B_08_SNP_clippedtrimmed.fastq,VA_W_02_14_clippedtrimmed.fastq,VA_W_02_18_clippedtrimmed.fastq,VA_W_02_22_clippedtrimmed.fastq,VA_W_09_SNP_clippedtrimmed.fastq --CPU 32
                    </p>
                    <p dir="auto">
                     sbatch Steph_trinity.sh
Submitted batch job 9272350
FAILED
sbatch Steph_trinity.sh
Submitted batch job 9272374
#w/o --normalized reads
                    </p>
                    <p dir="auto">
                     #Homework06
                    </p>
                    <ol dir="auto">
                     <li>
                      [spere004@turing1 QCFastqs]$ salloc
salloc: Pending job allocation 9272855
[spere004@coreV2-22-028 QCFastqs]$ cd djbtestassembly/
[spere004@coreV2-22-028 djbtestassembly]$ ls
Trinity.fasta
/cm/shared/apps/trinity/2.0.6/util/TrinityStats.pl Trinity.fasta
################################
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Counts of transcripts, etc.
                     </h2>
                     <a aria-label="Permalink: Counts of transcripts, etc." class="anchor" href="#counts-of-transcripts-etc" id="user-content-counts-of-transcripts-etc">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     ################################
Total trinity 'genes':	20980
Total trinity transcripts:	21992
Percent GC: 46.21
                    </p>
                    <p dir="auto">
                     ########################################
Stats based on ALL transcript contigs:
########################################
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Contig N10: 1178
Contig N20: 694
Contig N30: 514
Contig N40: 414
Contig N50: 347

Median contig length: 273
Average contig: 356.95
Total assembled bases: 7850006">
                     <pre class="notranslate"><code>Contig N10: 1178
Contig N20: 694
Contig N30: 514
Contig N40: 414
Contig N50: 347

Median contig length: 273
Average contig: 356.95
Total assembled bases: 7850006
</code></pre>
                    </div>
                    <p dir="auto">
                     #####################################################
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Stats based on ONLY LONGEST ISOFORM per 'GENE':
                     </h2>
                     <a aria-label="Permalink: Stats based on ONLY LONGEST ISOFORM per 'GENE':" class="anchor" href="#stats-based-on-only-longest-isoform-per-gene" id="user-content-stats-based-on-only-longest-isoform-per-gene">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     #####################################################
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Contig N10: 1027
Contig N20: 643
Contig N30: 486
Contig N40: 397
Contig N50: 336

Median contig length: 271
Average contig: 347.72
Total assembled bases: 7295061">
                     <pre class="notranslate"><code>Contig N10: 1027
Contig N20: 643
Contig N30: 486
Contig N40: 397
Contig N50: 336

Median contig length: 271
Average contig: 347.72
Total assembled bases: 7295061
</code></pre>
                    </div>
                    <ol dir="auto" start="2">
                     <li>
                      <p dir="auto">
                       [spere004@coreV2-22-028 scripts]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/scripts/python avg_cov_len_fasta_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta
The total number of sequences is 15079
The average sequence length is 876
The total number of bases is 13210470
The minimum sequence length is 500
The maximum sequence length is 10795
The N50 is 881
Median Length = 578
contigs &lt; 150bp = 0
contigs &gt;= 500bp = 15079
contigs &gt;= 1000bp = 3660
contigs &gt;= 2000bp = 536
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       [spere004@coreV2-22-028 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs
a) grep "overall alignment rate" steph_clippedtrimmedbowtie.txt
b) grep "aligned exactly 1 time" steph_clippedtrimmedbowtie.txt | cut -d " " -f 5
c) grep "aligned exactly 1 time" steph_clippedtrimmedbowtie.txt | cut -d "%" -f 1 | cut -d "(" -f 2
d) grep "aligned &gt;1 time" steph_clippedtrimmedbowtie.txt | cut -d "%" -f 1 | cut -d "(" -f 2
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       nano alignedstats_sg.txt
lane5_sg	2.03	1.11	235455.75	0.92
cat alignedstats_sg.txt &gt;&gt; /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/alignmentstatstable.txt
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       [spere004@coreV2-22-028 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data
mkdir testassembly
a) [spere004@coreV2-22-028 djbtestassembly]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs/djbtestassembly
mv Trinity.fasta /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/testassembly/
                      </p>
                     </li>
                    </ol>
                    <p dir="auto">
                     b) /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq
nano cleanup.sh
#!/bin/bash -l
                    </p>
                    <p dir="auto">
                     #SBATCH -o cleanup.txt
#SBATCH -n 1
#SBATCH --mail-user=
                     <a href="mailto:spere004@odu.edu">
                      spere004@odu.edu
                     </a>
                     #SBATCH --mail-type=END
#SBATCH --job-name=cleanup
                    </p>
                    <p dir="auto">
                     rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/originalfastqs
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/filteringstats
rm -r /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs/trinity_out_dir
                    </p>
                    <p dir="auto">
                     sbatch cleanup.sh
                    </p>
                    <ol dir="auto" start="6">
                     <li>
                      pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data
nano steph_blast.sh
#!/bin/bash -l
                     </li>
                    </ol>
                    <p dir="auto">
                     #SBATCH -o blastout.txt
#SBATCH -n 6
#SBATCH --mail-user=
                     <a href="mailto:spere004@odu.edu">
                      spere004@odu.edu
                     </a>
                     #SBATCH --mail-type=END
#SBATCH --job-name=steph_blast
                    </p>
                    <p dir="auto">
                     enable_lmod
module load container_env blast
blastx -query Trinity.fasta -db /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018 -out blastx.outfmt6 -evalue 1e-20 -num_threads 6 -max_target_seqs 1 -outfmt 6
                    </p>
                    <p dir="auto">
                     sbatch steph_blast.sh
Submitted batch job 9273231
                    </p>
                    <ol dir="auto" start="7">
                     <li>
                      <p dir="auto">
                       [spere004@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs
head -20 RI_B_05_18_clippedtrimmed.fastq.sam
@HD	VN:1.0	SO:unsorted
@SQ	SN:TR78|c0_g1_i1_coral	LN:1109
@SQ	SN:TR78|c0_g2_i1_coral	LN:1109
@SQ	SN:TR79|c0_g1_i1_coral	LN:610
@SQ	SN:TR79|c0_g2_i1_coral	LN:1549
@SQ	SN:TR87|c0_g1_i1_coral	LN:732
@SQ	SN:TR93|c0_g1_i1_coral	LN:550
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       grep -v '@' RI_B_05_18_clippedtrimmed.fastq.sam | head
K00188:59:HMTFHBBXX:6:1101:2788:1648	0	TR47986|c1_g3_i2_sym	9	255	51M	*	0	0	CNGGCAGTATGTGCAAAATTTATCTCCTGTTCACCTTTAAGTAACTAACCA	A#&lt;AFF-FJAFAJJJJJJFJJJFJ-FJJJFFJJFJAJJJJJJJJJFJJJJJ	AS:i:-1	XN:i:0	XM:i:1	XO:i:0	XG:i:0	NM:i:1	MD:Z:1G49	YT:Z:UU	RG:Z:RI_B_05_18
K00188:59:HMTFHBBXX:6:1101:16823:1666	16	TR47986|c1_g3_i2_sym	684	255	51M	*	0	0	TAAT
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       salloc
salloc: Pending job allocation 9273245
salloc: job 9273245 queued and waiting for resources
salloc: job 9273245 has been allocated resources
salloc: Granted job allocation 9273245
                      </p>
                     </li>
                    </ol>
                    <p dir="auto">
                     /cm/shared/courses/dbarshis/21AdvGenomics/scripts/get_explain_sam_flags_advbioinf.py RI_B_05_*.sam
RI_B_05_18_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
read reverse strand
not primary alignment
256 :
not primary alignment
16 :
read reverse strand
RI_B_05_22_clippedtrimmed.fastq.sam
['0', '272', '256', '16']
0 :
272 :
read reverse strand
not primary alignment
256 :
not primary alignment
16 :
read reverse strand
                    </p>
                    <ol dir="auto" start="10">
                     <li>
                      /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs
nano steph_SNPcall.sh
                     </li>
                    </ol>
                    <p dir="auto">
                     #!/bin/bash -l
                    </p>
                    <p dir="auto">
                     #SBATCH -o steph_SNPcall.txt
#SBATCH -n 1
#SBATCH --mail-user=
                     <a href="mailto:spere004@odu.edu">
                      spere004@odu.edu
                     </a>
                     #SBATCH --mail-type=END
#SBATCH --job-name=steph_SNPcall
                    </p>
                    <p dir="auto">
                     enable_lmod
module load samtools/1
for i in *.sam; do
                     <code>
                      samtools view -bS $i &gt; ${i%.sam}_UNSORTED.bam
                     </code>
                     ; done
                    </p>
                    <p dir="auto">
                     for i in *UNSORTED.bam; do samtools sort
                     <math-renderer class="js-inline-math" data-run-id="65e40d34dd7cab1ed98d9b508702929b" data-static-url="https://github.githubassets.com/static" style="display: inline">
                      $i &amp;gt; $
                     </math-renderer>
                     {i%_UNSORTED.bam}.bam
samtools index ${i%_UNSORTED.bam}.bam
done
                    </p>
                    <p dir="auto">
                     sbatch steph_SNPcall.sh
Submitted batch job 9273232
                    </p>
                    <p dir="auto">
                     #Homework07
                    </p>
                    <p dir="auto">
                     [spere004@turing1 data]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/testassembly/
                    </p>
                    <ol dir="auto">
                     <li>
                      nano blast parsing.sh
#!/bin/bash -l
                     </li>
                    </ol>
                    <p dir="auto">
                     #SBATCH -o blastparseout.txt
#SBATCH -n 1
#SBATCH --mail-user=
                     <a href="mailto:spere004@odu.edu">
                      spere004@odu.edu
                     </a>
                     #SBATCH --mail-type=END
#SBATCH --job-name=blastparsing
                    </p>
                    <p dir="auto">
                     /cm/shared/apps/trinity/2.0.6/util/analyze_blastPlus_topHit_coverage.pl blastx.outfmt6 Trinity.fasta /cm/shared/apps/blast/databases/uniprot_sprot_Sep2018.fasta
                    </p>
                    <p dir="auto">
                     sbatch blastparsing.sh
Submitted batch job 9276480
                    </p>
                    <p dir="auto">
                     cat blastparseout.txt
#hit_pct_cov_bin        count_in_bin    &gt;bin_below
100     143     143
90      52      195
80      77      272
70      87      359
60      110     469
50      133     602
40      219     821
30      395     1216
20      726     1942
10      674     2616
                    </p>
                    <ol dir="auto" start="2">
                     <li>
                      [spere004@turing1 QCFastqs]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs
ls *_UNSORTED.bam
RI_B_05_18_clippedtrimmed.fastq_UNSORTED.bam  VA_B_05_18_clippedtrimmed.fastq_UNSORTED.bam
RI_B_05_22_clippedtrimmed.fastq_UNSORTED.bam  VA_B_05_22_clippedtrimmed.fastq_UNSORTED.bam
RI_B_06_14_clippedtrimmed.fastq_UNSORTED.bam  VA_B_06_14_clippedtrimmed.fastq_UNSORTED.bam
RI_B_06_22_clippedtrimmed.fastq_UNSORTED.bam  VA_B_06_22_clippedtrimmed.fastq_UNSORTED.bam
RI_W_05_18_clippedtrimmed.fastq_UNSORTED.bam  VA_W_05_18_clippedtrimmed.fastq_UNSORTED.bam
RI_W_05_22_clippedtrimmed.fastq_UNSORTED.bam  VA_W_05_22_clippedtrimmed.fastq_UNSORTED.bam
RI_W_06_14_clippedtrimmed.fastq_UNSORTED.bam  VA_W_06_14_clippedtrimmed.fastq_UNSORTED.bam
RI_W_06_22_clippedtrimmed.fastq_UNSORTED.bam  VA_W_06_22_clippedtrimmed.fastq_UNSORTED.bam
nano rmunsorted.sh
#!/bin/bash -l
                     </li>
                    </ol>
                    <p dir="auto">
                     #SBATCH -o rmunsortout.txt
#SBATCH -n 1
#SBATCH --mail-user=
                     <a href="mailto:spere004@odu.edu">
                      spere004@odu.edu
                     </a>
                     #SBATCH --mail-type=END
#SBATCH --job-name=rmunsorted
                    </p>
                    <p dir="auto">
                     rm *_UNSORTED.bam
                    </p>
                    <p dir="auto">
                     sbatch rmunsorted.sh
Submitted batch job 9276496
ls *_UNSORTED.bam
ls: No match. ##all unsorted bam files are gone.
rm rmunsorted.txt ##unnecessary txt file to keep
                    </p>
                    <ol dir="auto" start="3">
                     <li>
                      /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs
nano steph_bayes.sh
#!/bin/bash -l
                     </li>
                    </ol>
                    <p dir="auto">
                     #SBATCH -o bayesout.txt
#SBATCH -n 1
                     <br/>
                     #SBATCH --mail-user=
                     <a href="mailto:spere004@odu.edu">
                      spere004@odu.edu
                     </a>
                     #SBATCH --mail-type=END
#SBATCH --job-name=steph_bayes
                    </p>
                    <p dir="auto">
                     enable_lmod
module load dDocent
freebayes --genotype-qualities -f /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/refassembly/15079_Apoc_hostsym.fasta *.bam &gt; SGmergedfastqs.vcf
                    </p>
                    <p dir="auto">
                     sbatch steph_bayes.sh
Submitted batch job 9276570
                    </p>
                    <p dir="auto">
                     ##Homework08
                    </p>
                    <ol dir="auto">
                     <li>
                      /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data
mkdir BAMs
mkdir SAMs
mkdir VCF
salloc
salloc: Pending job allocation 9278250
salloc: job 9278250 queued and waiting for resources
salloc: job 9278250 has been allocated resources
salloc: Granted job allocation 9278250
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/fastq/QCFastqs
mv *.sam /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/SAMs/
mv *.bam *.bam.bai /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/BAMs/
NO unfiltered vcfs
rm *.fastq
                     </li>
                    </ol>
                    <p dir="auto">
                     enable_lmod
module load dDocent
vcftools
                    </p>
                    <ol dir="auto" start="3">
                     <li>
                      <p dir="auto">
                       [spere004@turing1 VCF]$ pwd
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/VCF
cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/mergedfastq_HEAAstrangiaAssembly.vcf ./
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       [spere004@turing1 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly.vcf
                      </p>
                     </li>
                    </ol>
                    <p dir="auto">
                     VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009
                    </p>
                    <p dir="auto">
                     Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly.vcf
                    </p>
                    <p dir="auto">
                     After filtering, kept 40 out of 40 Individuals
After filtering, kept 1214003 out of a possible 1214003 Sites
Run Time = 10.00 seconds
                    </p>
                    <p dir="auto">
                     [spere004@turing1 VCF]$ /cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf SGmergedfastqs.vcf
                    </p>
                    <p dir="auto">
                     VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009
                    </p>
                    <p dir="auto">
                     Parameters as interpreted:
--vcf SGmergedfastqs.vcf
                    </p>
                    <p dir="auto">
                     After filtering, kept 16 out of 16 Individuals
After filtering, kept 130505 out of a possible 130505 Sites
Run Time = 1.00 seconds
                    </p>
                    <ol dir="auto" start="5">
                     <li>
                      cp /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/VCF/GoodCoralGenelistForVCFSubsetter.txt ./
[spere004@turing1 VCF]$ ls
GoodCoralGenelistForVCFSubsetter.txt  mergedfastq_HEAAstrangiaAssembly.vcf  out.log  SGmergedfastqs.vcf
                     </li>
                    </ol>
                    <p dir="auto">
                     grep -c "##contig" SGmergedfastqs.vcf
15079
                    </p>
                    <ol dir="auto" start="6">
                     <li>
                      <p dir="auto">
                       /cm/shared/courses/dbarshis/21AdvGenomics/scripts/21Sp_vcfsubsetter_advbioinf.py GoodCoralGenelistForVCFSubsetter.txt mergedfastq_HEAAstrangiaAssembly.vcf
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       [spere004@coreV3-23-031 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
                      </p>
                     </li>
                    </ol>
                    <p dir="auto">
                     VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009
                    </p>
                    <p dir="auto">
                     Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
                    </p>
                    <p dir="auto">
                     After filtering, kept 40 out of 40 Individuals
After filtering, kept 432676 out of a possible 432676 Sites
Run Time = 6.00 seconds
                    </p>
                    <p dir="auto">
                     --vcf mergedfastq_HEAAstrangiaAssembly.vcf
                    </p>
                    <p dir="auto">
                     VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009
                    </p>
                    <p dir="auto">
                     Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly.vcf
                    </p>
                    <p dir="auto">
                     After filtering, kept 40 out of 40 Individuals
After filtering, kept 1214003 out of a possible 1214003 Sites
Run Time = 5.00 seconds
                    </p>
                    <p dir="auto">
                     [spere004@coreV3-23-031 VCF]$ vcftools --vcf SGmergedfastqs.vcf
                    </p>
                    <p dir="auto">
                     VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009
                    </p>
                    <p dir="auto">
                     Parameters as interpreted:
--vcf SGmergedfastqs.vcf
                    </p>
                    <p dir="auto">
                     After filtering, kept 16 out of 16 Individuals
After filtering, kept 130505 out of a possible 130505 Sites
Run Time = 1.00 seconds
                    </p>
                    <ol dir="auto" start="8">
                     <li>
                      [spere004@coreV3-23-031 VCF]$ vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --recode --recode-INFO-all --out raw.g5mac3
                     </li>
                    </ol>
                    <p dir="auto">
                     VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009
                    </p>
                    <p dir="auto">
                     Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
--recode-INFO-all
--mac 3
--minQ 30
--max-missing 0.5
--out raw.g5mac3
--recode
                    </p>
                    <p dir="auto">
                     After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 99853 out of a possible 432676 Sites
Run Time = 31.00 seconds
                    </p>
                    <p dir="auto">
                     vcftools --vcf raw.g5mac3.recode.vcf --minDP 3 --recode --recode-INFO-all --out raw5.g5mac3dp3
                    </p>
                    <p dir="auto">
                     VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009
                    </p>
                    <p dir="auto">
                     Parameters as interpreted:
--vcf raw.g5mac3.recode.vcf
--recode-INFO-all
--minDP 3
--out raw5.g5mac3dp3
--recode
                    </p>
                    <p dir="auto">
                     After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 99853 out of a possible 99853 Sites
Run Time = 25.00 seconds
                    </p>
                    <p dir="auto">
                     vcftools --vcf raw5.g5mac3dp3.recode.vcf --missing-indv
                    </p>
                    <p dir="auto">
                     VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009
                    </p>
                    <p dir="auto">
                     Parameters as interpreted:
--vcf raw5.g5mac3dp3.recode.vcf
--missing-indv
                    </p>
                    <p dir="auto">
                     After filtering, kept 40 out of 40 Individuals
Outputting Individual Missingness
After filtering, kept 99853 out of a possible 99853 Sites
Run Time = 2.00 seconds
[spere004@coreV3-23-031 VCF]$ cat out.imiss
INDV	N_DATA	N_GENOTYPES_FILTERED	N_MISS	F_MISS
RI_W_06_merged	99853	0	67075	0.671737
RI_W_07_merged	99853	0	63917	0.640111
VA_B_03_merged	99853	0	56586	0.566693
RI_W_02_merged	99853	0	73105	0.732126
RI_W_04_merged	99853	0	71159	0.712638
VA_W_09_SNP_clipped	99853	0	19459	0.194876
RI_B_08_SNP_clipped	99853	0	88110	0.882397
VA_W_08_SNP_clipped	99853	0	83344	0.834667
VA_B_08_SNP_clipped	99853	0	94221	0.943597
VA_W_02_merged	99853	0	71050	0.711546
VA_B_07_merged	99853	0	62687	0.627793
RI_B_05_merged	99853	0	43945	0.440097
VA_W_06_merged	99853	0	65164	0.652599
VA_W_04_merged	99853	0	50911	0.509859
VA_W_01_merged	99853	0	67150	0.672489
VA_B_10_SNP_clipped	99853	0	82831	0.829529
VA_B_06_merged	99853	0	57724	0.57809
VA_W_05_merged	99853	0	66297	0.663946
RI_B_09_SNP_clipped	99853	0	85107	0.852323
VA_W_10_SNP_clipped	99853	0	83234	0.833565
RI_W_08_SNP_clipped	99853	0	77802	0.779165
RI_B_06_merged	99853	0	79766	0.798834
RI_W_10_SNP_clipped	99853	0	91323	0.914574
RI_B_04_merged	99853	0	46717	0.467858
VA_W_03_merged	99853	0	64016	0.641102
RI_B_07_merged	99853	0	70798	0.709022
RI_W_05_merged	99853	0	57117	0.572011
RI_W_09_SNP_clipped	99853	0	88492	0.886223
VA_B_01_merged	99853	0	47217	0.472865
VA_B_09_SNP_clipped	99853	0	79187	0.793036
RI_B_10_SNP_clipped	99853	0	82779	0.829009
RI_W_01_merged	99853	0	73169	0.732767
RI_B_01_merged	99853	0	69619	0.697215
VA_B_04_merged	99853	0	58190	0.582757
RI_B_02_merged	99853	0	87439	0.875677
RI_W_03_merged	99853	0	38760	0.388171
VA_B_02_merged	99853	0	78381	0.784964
VA_W_07_merged	99853	0	64592	0.646871
VA_B_05_merged	99853	0	54401	0.544811
RI_B_03_merged	99853	0	80369	0.804873
[spere004@coreV3-23-031 VCF]$ bash
[spere004@coreV3-23-031 VCF]$ mawk '!/IN/' out.imiss | cut -f5 &gt; totalmissing
[spere004@coreV3-23-031 VCF]$ gnuplot &lt;&lt; \EOF
                    </p>
                    <blockquote>
                     <p dir="auto">
                      set terminal dumb size 120, 30
set autoscale
unset label
set title "Histogram of % missing data per individual"
set ylabel "Number of Occurrences"
set xlabel "% of missing data"
#set yr [0:100000]
binwidth=0.01
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot 'totalmissing' using (bin($1,binwidth)):(1.0) smooth freq with boxes
pause -1
EOF
                     </p>
                    </blockquote>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="                                     Histogram of % missing data per individual">
                     <pre class="notranslate"><code>                                     Histogram of % missing data per individual
</code></pre>
                    </div>
                    <p dir="auto">
                     Number of Occurrences
3 ++----------+-----------+-----------+-----------+------------+---**
                     <em>
                      -----+-----------+-----------+----------++
+           +           +           +           +       'totalmissing' using (bin($1,binwidth)):(1.0) ****** +
|                                                                * *                                         |
|                                                                * *                                         |
|                                                                * *                                         |
|                                                                * *                                         |
2.5 ++                                                               * *                                        ++
|                                                                * *                                         |
|                                                                * *                                         |
|                                                                * *                                         |
|                                                                * *                                         |
|                                                                * *                                         |
2 ++                                                       **      * * ***  ******    ** ****   ****          ++
|                                                        **      * * * *  * *  *    ** ** *   *  *           |
|                                                        **      * * * *  * *  *    ** ** *   *  *           |
|                                                        **      * * * *  * *  *    ** ** *   *  *           |
|                                                        **      * * * *  * *  *    ** ** *   *  *           |
|                                                        **      * * * *  * *  *    ** ** *   *  *           |
1.5 ++                                                       **      * * * *  * *  *    ** ** *   *  *          ++
|                                                        **      * * * *  * *  *    ** ** *   *  *           |
|                                                        **      * * * *  * *  *    ** ** *   *  *           |
|                                                        **      * * * *  * *  *    ** ** *   *  *           |
|                                                        **      * * * *  * *  *    ** ** *   *  *           |
+           +           +           +           +        **  +   * * * * +
                     </em>
                     *  *    ** ** *   *  *           +
1 *********************************************************************************************************---++
0.1         0.2         0.3         0.4         0.5          0.6         0.7         0.8         0.9          1
% of missing data
                    </p>
                    <p dir="auto">
                     mawk '$5 &gt; 0.5' out.imiss | cut -f1 &gt; lowDP.indv
                    </p>
                    <p dir="auto">
                     --vcf raw5.g5mac3dp3.recode.vcf --remove lowDP.indv --recode --recode-INFO-all --out raw.g5mac3dplm
                    </p>
                    <p dir="auto">
                     VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009
                    </p>
                    <p dir="auto">
                     Parameters as interpreted:
--vcf raw5.g5mac3dp3.recode.vcf
--exclude lowDP.indv
--recode-INFO-all
--out raw.g5mac3dplm
--recode
                    </p>
                    <p dir="auto">
                     Excluding individuals in 'exclude' list
After filtering, kept 5 out of 40 Individuals
Outputting VCF file...
After filtering, kept 99853 out of a possible 99853 Sites
Run Time = 10.00 seconds
                    </p>
                    <p dir="auto">
                     #Homework09
/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/VCF
salloc
enable_lmod
module load dDocent
#second half run
/cm/shared/apps/vcftools/0.1.12b/bin/vcftools --vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf --max-missing 0.5 --mac 3 --minQ 30 --minDP 10 --max-alleles 2 --maf 0.015 --remove-indels --recode --recode-INFO-all --out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters
                    </p>
                    <p dir="auto">
                     VCFtools - v0.1.12b
(C) Adam Auton and Anthony Marcketta 2009
                    </p>
                    <p dir="auto">
                     Parameters as interpreted:
--vcf mergedfastq_HEAAstrangiaAssembly_subset.vcf
--recode-INFO-all
--mac 3
--maf 0.015
--max-alleles 2
--minDP 10
--minQ 30
--max-missing 0.5
--out 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters
--recode
--remove-indels
                    </p>
                    <p dir="auto">
                     After filtering, kept 40 out of 40 Individuals
Outputting VCF file...
After filtering, kept 1578 out of a possible 432676 Sites
Run Time = 6.00 seconds
                    </p>
                    <ol dir="auto" start="2">
                     <li>
                      <p dir="auto">
                       grep '#CHROM' mergedfastq_HEAAstrangiaAssembly_subset.vcf &gt; sampleIDs.txt
sed 's/\s+/\n/g' sampleIDs.txt &gt; sampleIDs1.txt
scp
                       <a href="mailto:spere004@turing.hpc.odu.edu">
                        spere004@turing.hpc.odu.edu
                       </a>
                       :/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/VCF/sampleIDs1.txt .
#Added the second column via excel and uploaded the file to the cluster again
scp sampleIDs1.txt
                       <a href="mailto:spere004@turing.hpc.odu.edu">
                        spere004@turing.hpc.odu.edu
                       </a>
                       :/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/VCF/sampleIDs2.txt
cp sampleIDs2.txt genepop.txt
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       /cm/shared/courses/dbarshis/21AdvGenomics/scripts/vcftogenepop_advbioinf.py 1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode.vcf genepop.txt
Indivs with genotypes in vcf file: RI_W_06_merged	RI_W_07_merged	VA_B_03_merged	RI_W_02_merged	RI_W_04_merged	VA_W_09_SNP_clipped	RI_B_08_SNP_clipped	VA_W_08_SNP_clipped	VA_B_08_SNP_clipped	VA_W_02_merged	VA_B_07_merged	RI_B_05_merged	VA_W_06_merged	VA_W_04_merged	VA_W_01_merged	VA_B_10_SNP_clipped	VA_B_06_merged	VA_W_05_merged	RI_B_09_SNP_clipped	VA_W_10_SNP_clipped	RI_W_08_SNP_clipped	RI_B_06_merged	RI_W_10_SNP_clipped	RI_B_04_merged	VA_W_03_merged	RI_B_07_merged	RI_W_05_merged	RI_W_09_SNP_clipped	VA_B_01_merged	VA_B_09_SNP_clipped	RI_B_10_SNP_clipped	RI_W_01_merged	RI_B_01_merged	VA_B_04_merged	RI_B_02_merged	RI_W_03_merged	VA_B_02_merged	VA_W_07_merged	VA_B_05_merged	RI_B_03_merged
44 1578 1578 1578 1578 40
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       scp
                       <a href="mailto:spere004@turing.hpc.odu.edu">
                        spere004@turing.hpc.odu.edu
                       </a>
                       :/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/VCF/1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen .
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       #Working on RStudio. Code below is r commands from day09 R script
                      </p>
                     </li>
                    </ol>
                    <blockquote>
                     <p dir="auto">
                      sum(is.na(datafile$tab))
[1] 35966
                     </p>
                    </blockquote>
                    <blockquote>
                     <p dir="auto">
                      datafile #shows info
/// GENIND OBJECT /////////
                     </p>
                    </blockquote>
                    <p dir="auto">
                     // 40 individuals; 1,578 loci; 3,156 alleles; size: 1.4 Mb
                    </p>
                    <p dir="auto">
                     // Basic content
@tab:  40 x 3156 matrix of allele counts
@loc.n.all: number of alleles per locus (range: 2-2)
@loc.fac: locus factor for the 3156 columns of @tab
@all.names: list of allele names for each locus
@ploidy: ploidy of each individual  (range: 2-2)
@type:  codom
@call: read.genepop(file = "1578_mergedfastq_HEAAstrangiaAssembly_subset_HEAFilters.recode_genepop.gen",
ncode = 2)
                    </p>
                    <p dir="auto">
                     // Optional content
@pop: population of each individual (group size range: 40-40)
                    </p>
                    <blockquote>
                     <p dir="auto">
                      YOURdata&lt;-scaleGen(datafile, NA.method='mean')
Warning message:
In .local(x, ...) : Some scaling values are null.
Corresponding alleles are removed.
X&lt;-YOURdata
Y&lt;-as.factor(substring(pop(datafile),1,4))
pca1 &lt;- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)
s.label(pca1$li)
s.class(pca1$li, pop(datafile))
col &lt;- c("blue","red", "green", "black")
s.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)
add.scatter.eig(pca1$eig[1:3], 3,1,2, posi="topright")
title("PCA of DJB_data\naxes 1-2")
a.clust&lt;-snapclust(datafile, k = 2)
class(a.clust)
[1] "snapclust" "list"
                      <br/>
                      names(a.clust)
[1] "group"     "ll"        "proba"     "converged" "n.iter"    "n.param"
                      <br/>
                      a.tab &lt;- table(pop(datafile), a.clust$group)
table.value(a.tab, col.labels = 1:2)
compoplot(a.clust)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     #plots were created and save on computer
                    </p>
                    <p dir="auto">
                     #Homework10
                    </p>
                    <ol dir="auto">
                     <li>
                      #Following code is R scrip ran on 2 files in day10 folder. Figures are saved on local computer.
                     </li>
                    </ol>
                    <p dir="auto">
                     library("ape")
library("pegas")
library("seqinr")
library("ggplot2")
library("adegenet")
setwd("/Users/stephanieperez/Documents/Data_analysis_class/21sp_advgenomics")
datafile&lt;-read.genepop("coral_66_cloneremoved_highoutliers.filtered1SNPper_genepop.gen", ncode=2)
datafile2&lt;-read.genepop("coral_279_cloneremoved_neutral.filtered1SNPper_genepop.gen", ncode=2)
                    </p>
                    <p dir="auto">
                     sum(is.na(datafile$tab))
datafile #shows info
                    </p>
                    <p dir="auto">
                     sum(is.na(datafile2$tab))
datafile2
                    </p>
                    <p dir="auto">
                     YOURdata&lt;-scaleGen(datafile, NA.method='mean')
X&lt;-YOURdata
Y&lt;-as.factor(substring(pop(datafile),1,4))
pca1 &lt;- dudi.pca(X,cent=F, scale=F, scannf=F, nf=3)
                    </p>
                    <p dir="auto">
                     YOUR2data&lt;-scaleGen(datafile2, NA.method='mean')
X2&lt;-YOUR2data
Y2&lt;-as.factor(substring(pop(datafile2),1,4))
pca12 &lt;- dudi.pca(X2,cent=F, scale=F, scannf=F, nf=3)
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      PCAs
                     </h4>
                     <a aria-label="Permalink: PCAs" class="anchor" href="#pcas" id="user-content-pcas">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      individual labels
                     </h1>
                     <a aria-label="Permalink: individual labels" class="anchor" href="#individual-labels" id="user-content-individual-labels">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     s.label(pca1$li)
s.label(pca12$li)
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      population elipses
                     </h1>
                     <a aria-label="Permalink: population elipses" class="anchor" href="#population-elipses" id="user-content-population-elipses">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     s.class(pca1$li, pop(datafile))
s.class(pca12$li, pop(datafile2))
                    </p>
                    <p dir="auto">
                     #color symbols, pop names
#pdf("YOURINITIALS_ColorPCA1v2.pdf")
col &lt;- c("blue","red", "green", "black")
s.class(pca1$li, Y,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)
add.scatter.eig(pca1$eig[1:3], 3,1,2, posi="topright")
title("PCA of DJB_data\naxes 1-2")
dev.off()
                    </p>
                    <p dir="auto">
                     col &lt;- c("blue","red", "green", "black")
s.class(pca12$li, Y2,xax=1,yax=2, col=transp(col,.6), axesell=F, cstar=0, cpoint=3, grid=FALSE, addaxes=TRUE)
add.scatter.eig(pca12$eig[1:3], 3,1,2, posi="topright")
title("PCA of DJB_data\naxes 1-2")
dev.off()
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Snapclust
                     </h4>
                     <a aria-label="Permalink: Snapclust" class="anchor" href="#snapclust" id="user-content-snapclust">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     a.clust&lt;-snapclust(datafile, k = 2)
class(a.clust)
names(a.clust)
a.tab &lt;- table(pop(datafile), a.clust$group)
table.value(a.tab, col.labels = 1:2)
compoplot(a.clust)
                    </p>
                    <p dir="auto">
                     a.clust1&lt;-snapclust(datafile2, k = 2)
class(a.clust1)
names(a.clust1)
a.tab &lt;- table(pop(datafile2), a.clust$group)
table.value(a.tab, col.labels = 1:2)
compoplot(a.clust1)
                    </p>
                    <ol dir="auto" start="2">
                     <li>
                      cd /cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/SAMs
nano countexpression.sh
#!/bin/bash -l
                     </li>
                    </ol>
                    <p dir="auto">
                     #SBATCH -o countexpression.txt
#SBATCH -n 1
                     <br/>
                     #SBATCH --mail-user=
                     <a href="mailto:spere004@odu.edu">
                      spere004@odu.edu
                     </a>
                     #SBATCH --mail-type=END
#SBATCH --job-name=countexpression
                    </p>
                    <p dir="auto">
                     /cm/shared/courses/dbarshis/21AdvGenomics/scripts/countxpression_SB_advbioinf.py *.sam
                    </p>
                    <p dir="auto">
                     sbatch countexpression.sh
                    </p>
                    <ol dir="auto" start="3">
                     <li>
                      salloc
salloc: Pending job allocation 9280866
salloc: job 9280866 queued and waiting for resources
salloc: job 9280866 has been allocated resources
salloc: Granted job allocation 9280866
                     </li>
                    </ol>
                    <p dir="auto">
                     mv match_counts.txt countsoutput.txt
                    </p>
                    <p dir="auto">
                     /cm/shared/courses/dbarshis/21AdvGenomics/scripts/ParseExpression2BigTable_advbioinf.py /cm/shared/courses/dbarshis/21AdvGenomics/classdata/Astrangia_poculata/host_genelist.txt StephFullCounts_summed.txt NoMatch *_counts.txt
Hits not matchedmatch_counts.txt=13381	RI_B_05_18_clippedtrimmed.fastq_counts.txt=1698	RI_B_05_22_clippedtrimmed.fastq_counts.txt=1698	RI_B_06_14_clippedtrimmed.fastq_counts.txt=1698	RI_B_06_22_clippedtrimmed.fastq_counts.txt=1698	RI_W_05_18_clippedtrimmed.fastq_counts.txt=1698	RI_W_05_22_clippedtrimmed.fastq_counts.txt=1698	RI_W_06_14_clippedtrimmed.fastq_counts.txt=1698	RI_W_06_22_clippedtrimmed.fastq_counts.txt=1698	VA_B_05_18_clippedtrimmed.fastq_counts.txt=1698	VA_B_05_22_clippedtrimmed.fastq_counts.txt=1698	VA_B_06_14_clippedtrimmed.fastq_counts.txt=1698	VA_B_06_22_clippedtrimmed.fastq_counts.txt=1698	VA_W_05_18_clippedtrimmed.fastq_counts.txt=1698	VA_W_05_22_clippedtrimmed.fastq_counts.txt=1698	VA_W_06_14_clippedtrimmed.fastq_counts.txt=1698	VA_W_06_22_clippedtrimmed.fastq_counts.txt=1698
                    </p>
                    <ol dir="auto" start="4">
                     <li>
                      <p dir="auto">
                       scp
                       <a href="mailto:spere004@turing.hpc.odu.edu">
                        spere004@turing.hpc.odu.edu
                       </a>
                       :/cm/shared/courses/dbarshis/21AdvGenomics/sandboxes/StephG/data/SAMs/StephFullCounts_summed.txt .
                       <a href="mailto:spere004@turing.hpc.odu.edu">
                        spere004@turing.hpc.odu.edu
                       </a>
                       's password:
StephFullCounts_summed.txt                                                                100%  860KB   5.2MB/s   00:00
                      </p>
                     </li>
                     <li>
                      <p dir="auto">
                       sed 's/_clippedtrimmed.fastq_counts.txt_UniqueTotReads//g' StephFullCounts_summed.txt &gt; NewStephfullcount_summed.txt
                      </p>
                     </li>
                    </ol>
                    <p dir="auto">
                     ##HOMEWORKDAY11
                    </p>
                    <blockquote>
                     <p dir="auto">
                      setwd("/Users/stephanieperez/Documents/Data_analysis_class/21sp_advgenomics")
getwd()
[1] "/Users/stephanieperez/Documents/Data_analysis_class/21sp_advgenomics"
countsTable &lt;- read.delim('djbFullCounts_summed.txt', header=TRUE, stringsAsFactors=TRUE, row.names=1)
head(countsTable)
RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10054|c0_g1_i1_coral          0          0          0          0          0          0
TR10054|c0_g2_i1_coral          0          1          6          0          0          0
TR10083|c0_g2_i1_coral          0          1          2          0          1          2
TR10090|c0_g1_i1_coral          0          3          1          0          4          0
TR10090|c0_g2_i1_coral          0          4          2          1          0          0
TR10103|c0_g1_i1_coral          0          1          4          0          0          4
RI_B_03_14 RI_B_03_18 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22
dim(countsTable)
[1] 13381    84
colSums(countsTable)
RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22 RI_B_03_14 RI_B_03_18
97406     207372     173843      47481      58036      84501     170330       5972
RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14 RI_B_05_18 RI_B_05_22 RI_B_06_14
199204     357124     544267      71600     630609     227313     225639      37012
colSums(countsTable)[order(colSums(countsTable))]
VA_W_02_22 VA_W_01_22 RI_B_03_18 RI_W_04_22 RI_B_07_22 RI_B_06_14 RI_W_04_14 RI_W_01_14
134        313       5972      10848      25369      37012      37579      39419
RI_B_02_14 RI_W_02_14 RI_B_02_18 RI_B_04_22 RI_B_06_22 VA_B_02_14 RI_B_02_22 VA_B_02_22
47481      54184      58036      71600      72254      79276      84501      85224
                     </p>
                    </blockquote>
                    <blockquote>
                     <p dir="auto">
                      LowCounts&lt;-c("VA_W_02_22","VA_W_01_22","RI_B_03_18")
names(countsTable)%in%LowCounts
[1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE
[15] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[29] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[43] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[57] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE FALSE
[71] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
                     </p>
                    </blockquote>
                    <blockquote>
                     <p dir="auto">
                      !(names(countsTable)%in%LowCounts)
[1]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[15]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[29]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[43]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[57]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE FALSE  TRUE
[71]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
countsTable&lt;-countsTable[,!(names(countsTable)%in%LowCounts)]
dim(countsTable)
[1] 13381    81
                     </p>
                    </blockquote>
                    <p dir="auto">
                     conds&lt;-data.frame("Sample"=names(countsTable))
                    </p>
                    <blockquote>
                     <p dir="auto">
                      conds$Origin&lt;-factor(substring(conds$Sample,1,2))
conds$SymState&lt;-factor(substring(conds$Sample,4,4))
conds$Temp&lt;-factor(substring(conds$Sample,9,10))
conds$Origin_SymState&lt;-factor(paste(conds$Origin,conds$SymState, sep="
                      <em>
                       "))
conds$Origin_SymState_Temp&lt;-factor(paste(conds$Origin,conds$SymState,conds$Temp, sep="
                      </em>
                      "))
                     </p>
                     <p dir="auto">
                      conds
Sample Origin SymState Temp Origin_SymState Origin_SymState_Temp
1  RI_B_01_14     RI        B   14            RI_B              RI_B_14
2  RI_B_01_18     RI        B   18            RI_B              RI_B_18
3  RI_B_01_22     RI        B   22            RI_B              RI_B_22
4  RI_B_02_14     RI        B   14            RI_B              RI_B_14
                     </p>
                    </blockquote>
                    <blockquote>
                     <p dir="auto">
                      prop.null &lt;- apply(countsTable, 2, function(x) 100*mean(x==0))
barplot(prop.null, main="Percentage of null counts per sample",
                     </p>
                    </blockquote>
                    <ul dir="auto">
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    horiz=TRUE, cex.names=0.5, las=1, ">
                       <pre class="notranslate"><code>    horiz=TRUE, cex.names=0.5, las=1, 
</code></pre>
                      </div>
                     </li>
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')">
                       <pre class="notranslate"><code>    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
</code></pre>
                      </div>
                     </li>
                    </ul>
                    <blockquote>
                     <p dir="auto">
                      pdf(file="OverallNullCounts.pdf",14, 14)
barplot(prop.null, main="Percentage of null counts per sample",
                     </p>
                    </blockquote>
                    <ul dir="auto">
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    horiz=TRUE, cex.names=0.5, las=1, ">
                       <pre class="notranslate"><code>    horiz=TRUE, cex.names=0.5, las=1, 
</code></pre>
                      </div>
                     </li>
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')">
                       <pre class="notranslate"><code>    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
</code></pre>
                      </div>
                     </li>
                    </ul>
                    <blockquote>
                     <p dir="auto">
                      dev.off()
RStudioGD
2
                     </p>
                    </blockquote>
                    <blockquote>
                     <p dir="auto">
                      means = apply(countsTable, 1, mean)
table(means&gt;=3)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     FALSE  TRUE
5838  7543
                    </p>
                    <blockquote>
                     <p dir="auto">
                      countsTable&lt;-countsTable[means&gt;=3,]
dim(countsTable)
[1] 7543   81
prop.nullv2 &lt;- apply(countsTable, 2, function(x) 100*mean(x==0))
pdf(file="Mean3NullCounts.pdf",14, 14)
barplot(prop.nullv2, main="Percentage of null counts per sample",
                     </p>
                    </blockquote>
                    <ul dir="auto">
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    horiz=TRUE, cex.names=0.5, las=1, ">
                       <pre class="notranslate"><code>    horiz=TRUE, cex.names=0.5, las=1, 
</code></pre>
                      </div>
                     </li>
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')">
                       <pre class="notranslate"><code>    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
</code></pre>
                      </div>
                     </li>
                    </ul>
                    <blockquote>
                     <p dir="auto">
                      dev.off()
RStudioGD
2
                     </p>
                    </blockquote>
                    <blockquote>
                     <p dir="auto">
                      dds &lt;- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=~ Origin_SymState_Temp)
dds &lt;- DESeq(dds)
estimating size factors
estimating dispersions
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
fitting model and testing
-- replacing outliers and refitting for 8 genes
-- DESeq argument 'minReplicatesForReplace' = 7
-- original counts are preserved in counts(dds)
estimating dispersions
fitting model and testing
vstCounts&lt;-varianceStabilizingTransformation(dds)
dists&lt;-dist(t(assay(vstCounts)))
plot(hclust(dists))
plotPCA(vstCounts, intgroup="Origin")
plotPCA(vstCounts, intgroup="SymState")
plotPCA(vstCounts, intgroup="Temp")
plotPCA(vstCounts, intgroup="Origin_SymState")
plotPCA(vstCounts, intgroup="Origin_SymState_Temp")
resultsNames(dds)
[1] "Intercept"                               "Origin_SymState_Temp_RI_B_18_vs_RI_B_14"
[3] "Origin_SymState_Temp_RI_B_22_vs_RI_B_14" "Origin_SymState_Temp_RI_W_14_vs_RI_B_14"
[5] "Origin_SymState_Temp_RI_W_18_vs_RI_B_14" "Origin_SymState_Temp_RI_W_22_vs_RI_B_14"
[7] "Origin_SymState_Temp_VA_B_14_vs_RI_B_14" "Origin_SymState_Temp_VA_B_18_vs_RI_B_14"
[9] "Origin_SymState_Temp_VA_B_22_vs_RI_B_14" "Origin_SymState_Temp_VA_W_14_vs_RI_B_14"
[11] "Origin_SymState_Temp_VA_W_18_vs_RI_B_14" "Origin_SymState_Temp_VA_W_22_vs_RI_B_14"
dim(dds)
[1] 7543   81
res &lt;- results(dds, contrast=c("Origin_SymState_Temp", "RI_B_18", "VA_B_18"))
head(res)
log2 fold change (MLE): Origin_SymState_Temp RI_B_18 vs VA_B_18
Wald test p-value: Origin_SymState_Temp RI_B_18 vs VA_B_18
DataFrame with 6 rows and 6 columns
baseMean     log2FoldChange             lfcSE
                   
TR10083|c0_g2_i1_coral 5.92149226796624 -0.124351582570554 0.881723031266384
TR10090|c0_g1_i1_coral   2.414668728522  0.403769276033225 0.959279168358701
TR10090|c0_g2_i1_coral 3.75861468695046  -0.83542301411573  0.86523878852701
TR10103|c0_g1_i1_coral 2.98376089398939 -0.801464007457239 0.848307534238223
TR10108|c0_g1_i1_coral 5.73440284734055 -0.526701449017958 0.837420233249731
TR10132|c0_g1_i1_coral 8.52629945810969   1.07157187389725  1.16012842796493
stat            pvalue              padj
                  
TR10083|c0_g2_i1_coral -0.141032476368405 0.887844286314782 0.996374898818957
TR10090|c0_g1_i1_coral  0.420909042280218 0.673821502118031 0.990719409569674
TR10090|c0_g2_i1_coral -0.965540409414564 0.334274206188308 0.964875734525325
TR10103|c0_g1_i1_coral -0.944780017988347 0.344771195342852 0.967913215005047
TR10108|c0_g1_i1_coral -0.628957156879308 0.529377105037453 0.987728806338776
TR10132|c0_g1_i1_coral  0.923666594203699 0.355659929898546 0.967913215005047
summary(res)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     out of 7542 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 3, 0.04%
LFC &lt; 0 (down)     : 5, 0.066%
outliers [1]       : 4, 0.053%
low counts [2]     : 1, 0.013%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results
                    </p>
                    <blockquote>
                     <p dir="auto">
                      res &lt;- results(dds, contrast=c("Origin_SymState_Temp", "RI_W_18", "VA_W_18"))
summary(res)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     out of 7542 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 22, 0.29%
LFC &lt; 0 (down)     : 38, 0.5%
outliers [1]       : 4, 0.053%
low counts [2]     : 3801, 50%
(mean count &lt; 6)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results
                    </p>
                    <blockquote>
                     <p dir="auto">
                      res &lt;- results(dds, contrast=c("Origin_SymState_Temp", "RI_B_14", "VA_B_14"))
summary(res)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     out of 7542 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 8, 0.11%
LFC &lt; 0 (down)     : 4, 0.053%
outliers [1]       : 4, 0.053%
low counts [2]     : 1, 0.013%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results
                    </p>
                    <blockquote>
                     <p dir="auto">
                      res &lt;- results(dds, contrast=c("Origin_SymState_Temp", "RI_W_14", "VA_W_14"))
summary(res)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     out of 7542 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 10, 0.13%
LFC &lt; 0 (down)     : 3, 0.04%
outliers [1]       : 4, 0.053%
low counts [2]     : 5263, 70%
(mean count &lt; 8)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results
                    </p>
                    <blockquote>
                     <p dir="auto">
                      res &lt;- results(dds, contrast=c("Origin_SymState_Temp", "RI_B_22", "VA_B_22"))
summary(res)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     out of 7542 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 8, 0.11%
LFC &lt; 0 (down)     : 2, 0.027%
outliers [1]       : 4, 0.053%
low counts [2]     : 1, 0.013%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results
                    </p>
                    <blockquote>
                     <p dir="auto">
                      res &lt;- results(dds, contrast=c("Origin_SymState_Temp", "RI_W_22", "VA_W_22"))
summary(res)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     out of 7542 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 11, 0.15%
LFC &lt; 0 (down)     : 5, 0.066%
outliers [1]       : 4, 0.053%
low counts [2]     : 146, 1.9%
(mean count &lt; 2)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results
                    </p>
                    <blockquote>
                     <p dir="auto">
                      head(res[order(res$padj),])
log2 fold change (MLE): Origin_SymState_Temp RI_W_22 vs VA_W_22
Wald test p-value: Origin_SymState_Temp RI_W_22 vs VA_W_22
DataFrame with 6 rows and 6 columns
baseMean    log2FoldChange             lfcSE
                  
TR42879|c0_g2_i1_coral  15.410578527665  17.0575331642567  3.13747844493701
TR13286|c1_g1_i1_coral 32.6102567902141 -3.38044295663573 0.648393226272248
TR47617|c0_g1_i1_coral  3.7465555529336  19.7261549909787  3.83515887756262
TR8327|c0_g3_i1_coral  8.73276616692518  19.7338447748926  4.06385316874438
TR29560|c0_g1_i1_coral 12.7955647888331  2.74246489048391 0.651667601875444
TR58072|c1_g1_i1_coral 53.8558310969574   1.3803927876847 0.330572470141824
stat               pvalue                 padj
                        
TR42879|c0_g2_i1_coral  5.43670130763213 5.42760110313753e-08 0.000401262549554957
TR13286|c1_g1_i1_coral -5.21356920409335 1.85241378350407e-07 0.000664534402813938
TR47617|c0_g1_i1_coral  5.14350399050882 2.69660923636117e-07 0.000664534402813938
TR8327|c0_g3_i1_coral   4.85594433545684  1.1981435873852e-06  0.00221446888538469
TR29560|c0_g1_i1_coral  4.20837998174426 2.57208081930306e-05   0.0365937404208763
TR58072|c1_g1_i1_coral  4.17576450662234 2.96986937001566e-05   0.0365937404208763
plotCounts(dds, "TR2367|c0_g1_i1_coral", intgroup="Origin_SymState_Temp")
plotCounts(dds, "TR2367|c0_g1_i1_coral", intgroup="Origin_SymState_Temp")
sum(res$padj &lt; 0.3, na.rm =T)
[1] 30
sum(res$padj &lt; 0.2, na.rm =T)
[1] 20
sum(res$padj &lt; 0.1, na.rm =T)
[1] 16
sum(res$pvalue &lt; 0.05, na.rm =T)
[1] 343
scaledcounts = counts(dds, normalized=T)
head(scaledcounts)
RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10083|c0_g2_i1_coral          0    0.84279   2.292953   0.000000   3.273656   4.312157
TR10090|c0_g1_i1_coral          0    2.52837   1.146477   0.000000  13.094624   0.000000
TR10090|c0_g2_i1_coral          0    3.37116   2.292953   3.553607   0.000000   0.000000
TR10103|c0_g1_i1_coral          0    0.84279   4.585907   0.000000   0.000000   8.624315...
genes4heatmap&lt;-res[res$pvalue &lt;0.05 &amp; !is.na(res$pvalue),]
names(genes4heatmap)
[1] "baseMean"       "log2FoldChange" "lfcSE"          "stat"           "pvalue"
                      <br/>
                      [6] "padj"
                      <br/>
                      head(genes4heatmap)
log2 fold change (MLE): Origin_SymState_Temp RI_W_22 vs VA_W_22
Wald test p-value: Origin_SymState_Temp RI_W_22 vs VA_W_22
DataFrame with 6 rows and 6 columns
baseMean    log2FoldChange             lfcSE
                  
TR10090|c0_g1_i1_coral   2.414668728522  -2.1602364606262  1.06658202772732
TR10450|c0_g1_i1_coral 3.06295343830589 -3.43890734797067  1.35815909767837
TR10768|c0_g1_i1_coral 4.11391026506041  1.64720448435771   0.8163169025316
TR10773|c0_g1_i1_coral 2.74299826965667  3.89664910540304  1.28628276466641
TR11771|c0_g1_i1_coral 2.86558169571988  2.26910497576441 0.996533942762988
TR11938|c0_g1_i2_coral 3.17991201457045  1.92891473796097  0.93224992931647
stat              pvalue              padj
                    
TR10090|c0_g1_i1_coral -2.02538239391605   0.042828113882863 0.996984539439689
TR10450|c0_g1_i1_coral -2.53203571941545   0.011340243841624 0.882509712853964
TR10768|c0_g1_i1_coral  2.01784929265745  0.0436069594078913 0.996984539439689
TR10773|c0_g1_i1_coral  3.02938763733929 0.00245050039091153 0.437812432964705
TR11771|c0_g1_i1_coral  2.27699717831296  0.0227863941200996 0.989064009775494
TR11938|c0_g1_i2_coral  2.06909614825636  0.0385370651149041 0.996984539439689
dim(genes4heatmap)
[1] 343   6
data4heatmap&lt;-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]
dim(data4heatmap)
[1] 343  81
head(data4heatmap)
RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10090|c0_g1_i1_coral    0.00000    2.52837   1.146477   0.000000   13.09462   0.000000
TR10450|c0_g1_i1_coral    0.00000    2.52837   0.000000   0.000000    0.00000   6.468236
TR10768|c0_g1_i1_coral   10.84813    4.21395   4.585907   7.107215    0.00000   8.624315
TR10773|c0_g1_i1_coral    0.00000    5.89953   2.292953   3.553607    0.00000   0.000000
TR11771|c0_g1_i1_coral    0.00000    2.52837   3.439430   0.000000    0.00000   0.000000
TR11938|c0_g1_i2_coral    0.00000    2.52837   0.000000   0.000000    0.00000   4.312157...
temp = as.matrix(rowMeans(data4heatmap))
head(temp)
[,1]
TR10090|c0_g1_i1_coral 2.414669
TR10450|c0_g1_i1_coral 3.062953
TR10768|c0_g1_i1_coral 4.113910
TR10773|c0_g1_i1_coral 2.742998
TR11771|c0_g1_i1_coral 2.865582
TR11938|c0_g1_i2_coral 3.179912
scaledmatrix&lt;-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)
data4heatmapscaled = data4heatmap/scaledmatrix
head(data4heatmapscaled)
RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10090|c0_g1_i1_coral   0.000000  1.0470877  0.4747967   0.000000   5.422948   0.000000
TR10450|c0_g1_i1_coral   0.000000  0.8254680  0.0000000   0.000000   0.000000   2.111764
TR10768|c0_g1_i1_coral   2.636938  1.0243174  1.1147318   1.727606   0.000000   2.096379
TR10773|c0_g1_i1_coral   0.000000  2.1507597  0.8359295   1.295519   0.000000   0.000000
TR11771|c0_g1_i1_coral   0.000000  0.8823235  1.2002554   0.000000   0.000000   0.000000
TR11938|c0_g1_i2_coral   0.000000  0.7951069  0.0000000   0.000000   0.000000   1.356062...
dim(data4heatmapscaled)
[1] 343  81
                     </p>
                    </blockquote>
                    <blockquote>
                     <p dir="auto">
                      pairs.breaks &lt;- seq(0, 3.0, by=0.1)
length(pairs.breaks)
[1] 31
mycol &lt;- colorpanel(n=30, low="black", high="yellow")
pdf(file="SGdata_byrow.pdf",14,7)
heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c("row"), scale="none", keysize=1, breaks=pairs.breaks, col=mycol, trace = "none", symkey = F, density.info = "density", colsep=c(24), sepcolor=c("white"), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
dev.off()
RStudioGD
2
pdf(file="SGdata_bycolumn.pdf",14,7)
heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c("col"), scale="none", keysize=1, breaks=pairs.breaks, col=mycol, trace = "none", symkey = F, density.info = "density", colsep=c(24), sepcolor=c("white"), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
dev.off()
RStudioGD
2
plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup="Origin")
prop.nullv3 &lt;- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100*mean(x==0))
pdf(file="ResNullCounts.pdf",14, 14)
barplot(prop.nullv3, main="Percentage of null counts per sample",
                     </p>
                    </blockquote>
                    <ul dir="auto">
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    horiz=TRUE, cex.names=0.5, las=1, ">
                       <pre class="notranslate"><code>    horiz=TRUE, cex.names=0.5, las=1, 
</code></pre>
                      </div>
                     </li>
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')">
                       <pre class="notranslate"><code>    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
</code></pre>
                      </div>
                     </li>
                    </ul>
                    <blockquote>
                     <p dir="auto">
                      dev.off()
RStudioGD
2
                     </p>
                    </blockquote>
                    <p dir="auto">
                     #Homeworkday12
                    </p>
                    <p dir="auto">
                     _B_03_14 RI_B_05_18 VA_W_07_14 VA_B_06_14 VA_W_07_22 VA_B_07_18 VA_W_02_18 VA_B_03_18
226784     227313     228792     232578     233899     234465     241804     251696
RI_B_07_14 VA_W_04_22 VA_B_01_14 VA_B_04_18 VA_B_04_22 VA_B_03_22 VA_W_03_14 VA_W_04_18
257730     261344     261824     264192     268290     271959     278000     280007
RI_W_03_14 VA_B_05_18 VA_W_04_14 RI_W_01_18 VA_W_03_22 RI_B_06_18 RI_W_04_18 VA_W_01_18
281297     288121     292041     292612     295382     296071     300578     321900
VA_B_05_14 VA_B_01_22 RI_W_07_14 RI_B_04_14 RI_W_03_22 VA_B_06_22 VA_W_05_18 RI_W_05_14
323715     328198     342795     357124     359053     362054     388532     392818
RI_W_03_18 RI_B_04_18 VA_B_01_18 RI_B_05_14
428532     544267     548448     630609
                    </p>
                    <blockquote>
                     <p dir="auto">
                      table(substring(names(countsTable),1,2))
                     </p>
                    </blockquote>
                    <p dir="auto">
                     RI VA
42 42
                    </p>
                    <blockquote>
                     <p dir="auto">
                      table(substring(names(countsTable),4,4))
                     </p>
                    </blockquote>
                    <p dir="auto">
                     B  W
42 42
                    </p>
                    <blockquote>
                     <p dir="auto">
                      table(substring(names(countsTable),9,10))
                     </p>
                    </blockquote>
                    <p dir="auto">
                     14 18 22
28 28 28
                    </p>
                    <blockquote>
                     <p dir="auto">
                      table(paste(substring(names(countsTable),1,2),substring(names(countsTable),4,4), sep="_"))
                     </p>
                    </blockquote>
                    <p dir="auto">
                     RI_B RI_W VA_B VA_W
21   21   21   21
                    </p>
                    <blockquote>
                     <p dir="auto">
                      table(paste(substring(names(countsTable),1,2),substring(names(countsTable),4,4), substring(names(countsTable),9,10), sep="_"))
                     </p>
                    </blockquote>
                    <p dir="auto">
                     RI_B_14 RI_B_18 RI_B_22 RI_W_14 RI_W_18 RI_W_22 VA_B_14 VA_B_18 VA_B_22 VA_W_14 VA_W_18
7       7       7       7       7       7       7       7       7       7       7
VA_W_22
7
                    </p>
                    <blockquote>
                     <p dir="auto">
                      LowCounts&lt;-c("VA_W_02_22","VA_W_01_22","RI_B_03_18", "RI_W_01_14", "RI_W_04_22", "VA_W_03_18")
names(countsTable)%in%LowCounts
[1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE
[15] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE
[29] FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[43] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
[57] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE FALSE
[71]  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
!(names(countsTable)%in%LowCounts)
[1]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[15]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[29]  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[43]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
[57]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE FALSE  TRUE
[71] FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
countsTable&lt;-countsTable[,!(names(countsTable)%in%LowCounts)]
dim(countsTable)
[1] 13381    78
                     </p>
                     <p dir="auto">
                      #make a table with conditions of each individual (e.g. "VA" and "RI")  There should be the same number of conditions described as there are samples in your data file, and in the same order.
#NOTE: It is absolutely critical that the columns of the count matrix and the rows of the column data (information about samples) are in the same order. DESeq2 will not make guesses as to which column of the count matrix belongs to which row of the column data, these must be provided to DESeq2 already in consistent order.
#Here's an example
                     </p>
                     <div class="markdown-heading" dir="auto">
                      <h1 class="heading-element" dir="auto" tabindex="-1">
                       Sample		Origin	SymState	Temp  Origin_SymState Origin_SymState_Temp
                      </h1>
                      <a aria-label="Permalink: Sample		Origin	SymState	Temp  Origin_SymState Origin_SymState_Temp" class="anchor" href="#sampleoriginsymstatetemp--origin_symstate-origin_symstate_temp" id="user-content-sampleoriginsymstatetemp--origin_symstate-origin_symstate_temp">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <div class="markdown-heading" dir="auto">
                      <h1 class="heading-element" dir="auto" tabindex="-1">
                       RI_B_06_18		RI	B	18  RI_B  RI_B_18
                      </h1>
                      <a aria-label="Permalink: RI_B_06_18		RI	B	18  RI_B  RI_B_18" class="anchor" href="#ri_b_06_18rib18--ri_b--ri_b_18" id="user-content-ri_b_06_18rib18--ri_b--ri_b_18">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <div class="markdown-heading" dir="auto">
                      <h1 class="heading-element" dir="auto" tabindex="-1">
                       VA_B_07_14		VA	B	14  VA_B  VA_B_14
                      </h1>
                      <a aria-label="Permalink: VA_B_07_14		VA	B	14  VA_B  VA_B_14" class="anchor" href="#va_b_07_14vab14--va_b--va_b_14" id="user-content-va_b_07_14vab14--va_b--va_b_14">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <div class="markdown-heading" dir="auto">
                      <h1 class="heading-element" dir="auto" tabindex="-1">
                       VA_W_07_22		VA	W	22  VA_W  VA_W_22
                      </h1>
                      <a aria-label="Permalink: VA_W_07_22		VA	W	22  VA_W  VA_W_22" class="anchor" href="#va_w_07_22vaw22--va_w--va_w_22" id="user-content-va_w_07_22vaw22--va_w--va_w_22">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <div class="markdown-heading" dir="auto">
                      <h1 class="heading-element" dir="auto" tabindex="-1">
                       safest way to do this is in R
                      </h1>
                      <a aria-label="Permalink: safest way to do this is in R" class="anchor" href="#safest-way-to-do-this-is-in-r" id="user-content-safest-way-to-do-this-is-in-r">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <p dir="auto">
                      conds&lt;-data.frame("Sample"=names(countsTable))
conds$Origin&lt;-factor(substring(conds$Sample,1,2))
conds$SymState&lt;-factor(substring(conds$Sample,4,4))
conds$Temp&lt;-factor(substring(conds$Sample,9,10))
conds$Origin_SymState&lt;-factor(paste(conds$Origin,conds$SymState, sep="
                      <em>
                       "))
conds$Origin_SymState_Temp&lt;-factor(paste(conds$Origin,conds$SymState,conds$Temp, sep="
                      </em>
                      "))
                     </p>
                     <p dir="auto">
                      ##Check null counts per sample
prop.null &lt;- apply(countsTable, 2, function(x) 100*mean(x==0))
                     </p>
                     <p dir="auto">
                      barplot(prop.null, main="Percentage of null counts per sample",
                     </p>
                    </blockquote>
                    <ul dir="auto">
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    horiz=TRUE, cex.names=0.5, las=1, ">
                       <pre class="notranslate"><code>    horiz=TRUE, cex.names=0.5, las=1, 
</code></pre>
                      </div>
                     </li>
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')">
                       <pre class="notranslate"><code>    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
</code></pre>
                      </div>
                     </li>
                    </ul>
                    <blockquote>
                     <p dir="auto">
                      pdf(file="OverallNullCountsv2.pdf",14, 14)
barplot(prop.null, main="Percentage of null counts per sample",
                     </p>
                    </blockquote>
                    <ul dir="auto">
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    horiz=TRUE, cex.names=0.5, las=1, ">
                       <pre class="notranslate"><code>    horiz=TRUE, cex.names=0.5, las=1, 
</code></pre>
                      </div>
                     </li>
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')">
                       <pre class="notranslate"><code>    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
</code></pre>
                      </div>
                     </li>
                    </ul>
                    <blockquote>
                     <p dir="auto">
                      dev.off()
RStudioGD
2
                     </p>
                     <div class="markdown-heading" dir="auto">
                      <h1 class="heading-element" dir="auto" tabindex="-1">
                       how many genes have mean count &gt;=3
                      </h1>
                      <a aria-label="Permalink: how many genes have mean count &gt;=3" class="anchor" href="#how-many-genes-have-mean-count-3" id="user-content-how-many-genes-have-mean-count-3">
                       <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                        </path>
                       </svg>
                      </a>
                     </div>
                     <p dir="auto">
                      means = apply(countsTable, 1, mean)
table(means&gt;=3)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     FALSE  TRUE
5721  7660
                    </p>
                    <blockquote>
                     <p dir="auto">
                      countsTable&lt;-countsTable[means&gt;=3,]
dim(countsTable)
[1] 7660   78
                     </p>
                     <p dir="auto">
                      prop.nullv2 &lt;- apply(countsTable, 2, function(x) 100*mean(x==0))
                     </p>
                     <p dir="auto">
                      barplot(prop.nullv2, main="Percentage of null counts per sample",
                     </p>
                    </blockquote>
                    <ul dir="auto">
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    horiz=TRUE, cex.names=0.5, las=1, ">
                       <pre class="notranslate"><code>    horiz=TRUE, cex.names=0.5, las=1, 
</code></pre>
                      </div>
                     </li>
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')">
                       <pre class="notranslate"><code>    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
</code></pre>
                      </div>
                     </li>
                    </ul>
                    <blockquote>
                     <p dir="auto">
                      pdf(file="Mean3NullCountsv2.pdf",14, 14)
barplot(prop.nullv2, main="Percentage of null counts per sample",
                     </p>
                    </blockquote>
                    <ul dir="auto">
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    horiz=TRUE, cex.names=0.5, las=1, ">
                       <pre class="notranslate"><code>    horiz=TRUE, cex.names=0.5, las=1, 
</code></pre>
                      </div>
                     </li>
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')">
                       <pre class="notranslate"><code>    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
</code></pre>
                      </div>
                     </li>
                    </ul>
                    <blockquote>
                     <p dir="auto">
                      dev.off()
RStudioGD
2
                     </p>
                     <p dir="auto">
                      #make count data sets
dds &lt;- DESeqDataSetFromMatrix(countData=countsTable, colData=conds, design=~ Origin + SymState + Temp + Origin:SymState)
                     </p>
                     <p dir="auto">
                      dds &lt;- DESeq(dds)
estimating size factors
estimating dispersions
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
fitting model and testing
-- replacing outliers and refitting for 14 genes
-- DESeq argument 'minReplicatesForReplace' = 7
-- original counts are preserved in counts(dds)
estimating dispersions
fitting model and testing
                     </p>
                     <p dir="auto">
                      dim(dds)
[1] 7660   78
                     </p>
                     <p dir="auto">
                      #transform counts to variance stablized counts
vstCounts&lt;-varianceStabilizingTransformation(dds)
                     </p>
                     <p dir="auto">
                      #plot cluster dendrogram across samples
dists&lt;-dist(t(assay(vstCounts)))
plot(hclust(dists))
heatmap.2(as.matrix(dists), key=F, trace="none",
                     </p>
                    </blockquote>
                    <ul dir="auto">
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='      col=colorpanel(100, "black", "white"),'>
                       <pre class="notranslate"><code>      col=colorpanel(100, "black", "white"),
</code></pre>
                      </div>
                     </li>
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="      margin=c(10, 10))">
                       <pre class="notranslate"><code>      margin=c(10, 10))
</code></pre>
                      </div>
                     </li>
                    </ul>
                    <blockquote>
                     <p dir="auto">
                      ###Figure out which contrast you want to examine (i.e. which two groups do you want to compare)
resultsNames(dds)
[1] "Intercept"          "Origin_VA_vs_RI"    "SymState_W_vs_B"    "Temp_18_vs_14"
                      <br/>
                      [5] "Temp_22_vs_14"      "OriginVA.SymStateW"
                     </p>
                     <p dir="auto">
                      res &lt;- results(dds)
head(res)
log2 fold change (MLE): OriginVA.SymStateW
Wald test p-value: OriginVA.SymStateW
DataFrame with 6 rows and 6 columns
baseMean     log2FoldChange             lfcSE
                   
TR10083|c0_g2_i1_coral   5.366479327573  0.107690151743452  0.64464619766983
TR10090|c0_g1_i1_coral 2.70436251126018 -0.247209127175208 0.831370968745152
TR10090|c0_g2_i1_coral 3.65756460141883 -0.976996511249959 0.654110304614643
TR10103|c0_g1_i1_coral  2.9127886212765 -0.172219518994785 0.641025633753981
TR10108|c0_g1_i1_coral 6.31144026709945  0.366046450639525 0.664400660631174
TR10132|c0_g1_i1_coral  9.2673703948612 -0.589356834713084 0.975809166670876
stat            pvalue              padj
                  
TR10083|c0_g2_i1_coral   0.16705310933767 0.867328260788591 0.999364127033592
TR10090|c0_g1_i1_coral -0.297351166289025 0.766198422057453  0.99520853313718
TR10090|c0_g2_i1_coral  -1.49362654029054  0.13527325854353 0.958806816980756
TR10103|c0_g1_i1_coral -0.268662452679515 0.788189448150222  0.99520853313718
TR10108|c0_g1_i1_coral  0.550942333940163 0.581673205867577 0.991106673557066
TR10132|c0_g1_i1_coral -0.603967307177248 0.545865378089277 0.990476522025752
summary(res)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     out of 7659 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 2, 0.026%
LFC &lt; 0 (down)     : 9, 0.12%
outliers [1]       : 10, 0.13%
low counts [2]     : 1, 0.013%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results
                    </p>
                    <blockquote>
                     <p dir="auto">
                      resSymState &lt;- results(dds, contrast=c("SymState", "B", "W"))
head(resSymState)
log2 fold change (MLE): SymState B vs W
Wald test p-value: SymState B vs W
DataFrame with 6 rows and 6 columns
baseMean     log2FoldChange             lfcSE
                   
TR10083|c0_g2_i1_coral   5.366479327573  0.304875236273871 0.463587635582715
TR10090|c0_g1_i1_coral 2.70436251126018 -0.340213618854429 0.624844624315529
TR10090|c0_g2_i1_coral 3.65756460141883   -1.0442719434015 0.488672538582417
TR10103|c0_g1_i1_coral  2.9127886212765 -0.316210568092073 0.469122505416545
TR10108|c0_g1_i1_coral 6.31144026709945   0.25797411425412 0.481341478797278
TR10132|c0_g1_i1_coral  9.2673703948612 -0.604956612553455 0.692218264872669
stat             pvalue              padj
                   
TR10083|c0_g2_i1_coral  0.657643157136089  0.510767456004599 0.986343106634585
TR10090|c0_g1_i1_coral -0.544477147782311  0.586113169806799 0.995961170585915
TR10090|c0_g2_i1_coral  -2.13695647074996 0.0326015322469065 0.816269739453607
TR10103|c0_g1_i1_coral -0.674046894875149  0.500281500123105 0.986343106634585
TR10108|c0_g1_i1_coral  0.535948231385995   0.59199432935739 0.995961170585915
TR10132|c0_g1_i1_coral -0.873939107435622  0.382151415850855 0.986343106634585
summary(resSymState)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     out of 7659 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 6, 0.078%
LFC &lt; 0 (down)     : 6, 0.078%
outliers [1]       : 10, 0.13%
low counts [2]     : 1, 0.013%
(mean count &lt; 1)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results
                    </p>
                    <blockquote>
                     <p dir="auto">
                      resOrigin &lt;- results(dds, contrast=c("Origin", "VA", "RI"))
head(resOrigin)
log2 fold change (MLE): Origin VA vs RI
Wald test p-value: Origin VA vs RI
DataFrame with 6 rows and 6 columns
baseMean      log2FoldChange             lfcSE
                    
TR10083|c0_g2_i1_coral   5.366479327573  -0.271242746289843 0.446224512750353
TR10090|c0_g1_i1_coral 2.70436251126018    1.11251170496561 0.589306266446119
TR10090|c0_g2_i1_coral 3.65756460141883   0.943134641716211 0.474719393211504
TR10103|c0_g1_i1_coral  2.9127886212765  0.0386642676540012 0.456719612483097
TR10108|c0_g1_i1_coral 6.31144026709945 -0.0179009694277555 0.461957599914831
TR10132|c0_g1_i1_coral  9.2673703948612  -0.264990306479184 0.677607561801266
stat             pvalue              padj
                   
TR10083|c0_g2_i1_coral  -0.607861600022842  0.543279269700006 0.865660813784253
TR10090|c0_g1_i1_coral    1.88783281005095 0.0590483989759312                NA
TR10090|c0_g2_i1_coral    1.98672027139201 0.0469534041233123 0.418043000342497
TR10103|c0_g1_i1_coral  0.0846564644854882  0.932534507970102 0.984590265229257
TR10108|c0_g1_i1_coral -0.0387502433795997  0.969089515041603 0.992149683420423
TR10132|c0_g1_i1_coral  -0.391067516682912  0.695747330435347 0.921516804624192
summary(resOrigin)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     out of 7659 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 73, 0.95%
LFC &lt; 0 (down)     : 104, 1.4%
outliers [1]       : 10, 0.13%
low counts [2]     : 594, 7.8%
(mean count &lt; 3)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results
                    </p>
                    <blockquote>
                     <p dir="auto">
                      res14v18 &lt;- results(dds, contrast=c("Temp", "14", "18"))
head(res14v18)
log2 fold change (MLE): Temp 14 vs 18
Wald test p-value: Temp 14 vs 18
DataFrame with 6 rows and 6 columns
baseMean     log2FoldChange             lfcSE
                   
TR10083|c0_g2_i1_coral   5.366479327573 0.0134105654377919 0.389675116366814
TR10090|c0_g1_i1_coral 2.70436251126018 -0.750040304511369 0.493719609600886
TR10090|c0_g2_i1_coral 3.65756460141883 -0.527849679316855 0.385227211070661
TR10103|c0_g1_i1_coral  2.9127886212765  0.144899722334601 0.377623543666657
TR10108|c0_g1_i1_coral 6.31144026709945 -0.117131615020316 0.401865058532759
TR10132|c0_g1_i1_coral  9.2673703948612  -1.16667257299433 0.592603491023266
stat             pvalue              padj
                   
TR10083|c0_g2_i1_coral 0.0344147339014795  0.972546434477265 0.994762640063651
TR10090|c0_g1_i1_coral  -1.51916247587915   0.12872160398251                NA
TR10090|c0_g2_i1_coral  -1.37022947535249   0.17061528001709                NA
TR10103|c0_g1_i1_coral  0.383714746510905  0.701189881652785                NA
TR10108|c0_g1_i1_coral -0.291470015950062  0.770691872645245 0.956296438232868
TR10132|c0_g1_i1_coral  -1.96872375992892 0.0489848213296003 0.440134464115494
summary(res14v18)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     out of 7659 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 31, 0.4%
LFC &lt; 0 (down)     : 111, 1.4%
outliers [1]       : 10, 0.13%
low counts [2]     : 2671, 35%
(mean count &lt; 4)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results
                    </p>
                    <blockquote>
                     <p dir="auto">
                      res14v22 &lt;- results(dds, contrast=c("Temp", "14", "22"))
head(res14v22)
log2 fold change (MLE): Temp 14 vs 22
Wald test p-value: Temp 14 vs 22
DataFrame with 6 rows and 6 columns
baseMean     log2FoldChange             lfcSE
                   
TR10083|c0_g2_i1_coral   5.366479327573 -0.381080048190283 0.396906053114453
TR10090|c0_g1_i1_coral 2.70436251126018 -0.420390942831918 0.517099767325245
TR10090|c0_g2_i1_coral 3.65756460141883 -0.149468056890158 0.407435526480227
TR10103|c0_g1_i1_coral  2.9127886212765  0.327505589102544 0.400037906787554
TR10108|c0_g1_i1_coral 6.31144026709945 -0.504680765262664 0.409220095704197
TR10132|c0_g1_i1_coral  9.2673703948612   -1.2400583742243 0.602190057747409
stat             pvalue              padj
                   
TR10083|c0_g2_i1_coral  -0.96012657201878  0.336991516539358  0.79600966973284
TR10090|c0_g1_i1_coral -0.812978402613553  0.416230440926914                NA
TR10090|c0_g2_i1_coral -0.366850819763779  0.713730297691035                NA
TR10103|c0_g1_i1_coral  0.818686388328871  0.412965364593821                NA
TR10108|c0_g1_i1_coral  -1.23327463768414  0.217473316972006 0.699820201183028
TR10132|c0_g1_i1_coral  -2.05924750545192 0.0394705344365933 0.387444149379081
summary(res14v22)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     out of 7659 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 64, 0.84%
LFC &lt; 0 (down)     : 97, 1.3%
outliers [1]       : 10, 0.13%
low counts [2]     : 2226, 29%
(mean count &lt; 4)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results
                    </p>
                    <blockquote>
                     <p dir="auto">
                      res18v22 &lt;- results(dds, contrast=c("Temp", "18", "22"))
head(res18v22)
log2 fold change (MLE): Temp 18 vs 22
Wald test p-value: Temp 18 vs 22
DataFrame with 6 rows and 6 columns
baseMean      log2FoldChange             lfcSE
                    
TR10083|c0_g2_i1_coral   5.366479327573  -0.394490613628075 0.395010013602601
TR10090|c0_g1_i1_coral 2.70436251126018   0.329649361679451 0.503429809209276
TR10090|c0_g2_i1_coral 3.65756460141883   0.378381622426696 0.396934638805837
TR10103|c0_g1_i1_coral  2.9127886212765   0.182605866767943 0.399213811730964
TR10108|c0_g1_i1_coral 6.31144026709945  -0.387549150242348 0.406735578634103
TR10132|c0_g1_i1_coral  9.2673703948612 -0.0733858012299701 0.596241474936225
stat            pvalue              padj
                  
TR10083|c0_g2_i1_coral -0.998685096689603 0.317947262436469 0.838585264651832
TR10090|c0_g1_i1_coral  0.654806997220173 0.512592024969438                NA
TR10090|c0_g2_i1_coral  0.953259265971454 0.340458724704261 0.847758501629442
TR10103|c0_g1_i1_coral  0.457413700132709  0.64737371896663                NA
TR10108|c0_g1_i1_coral -0.952828251572713 0.340677097558174 0.847758501629442
TR10132|c0_g1_i1_coral -0.123080671699029 0.902043216645959 0.991572144643207
summary(res18v22)
                     </p>
                    </blockquote>
                    <p dir="auto">
                     out of 7659 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 34, 0.44%
LFC &lt; 0 (down)     : 10, 0.13%
outliers [1]       : 10, 0.13%
low counts [2]     : 1337, 17%
(mean count &lt; 3)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results
                    </p>
                    <blockquote>
                     <p dir="auto">
                      sum(resOrigin$padj &lt; 0.1, na.rm =T)
[1] 177
sum(resOrigin$pvalue &lt; 0.05, na.rm =T)
[1] 866
scaledcounts = counts(dds, normalize=T)
head(scaledcounts)
RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10083|c0_g2_i1_coral          0  0.8671827   2.447866   0.000000   3.455170   4.432653
TR10090|c0_g1_i1_coral          0  2.6015481   1.223933   0.000000  13.820681   0.000000
TR10090|c0_g2_i1_coral          0  3.4687308   2.447866   3.772786   0.000000   0.000000
TR10103|c0_g1_i1_coral          0  0.8671827   4.895733   0.000000   0.000000   8.865306
TR10108|c0_g1_i1_coral          0  1.7343654   4.895733   0.000000   6.910341   0.000000
TR10132|c0_g1_i1_coral          0  6.0702789   1.223933  18.863931  51.827555   0.000000
RI_B_03_14 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10083|c0_g2_i1_coral  12.871970   1.272998  12.424473  8.1878079   0.000000   4.563408
TR10090|c0_g1_i1_coral   0.000000   0.000000   2.003947  4.0939039   6.765958   0.570426
TR10090|c0_g2_i1_coral   0.000000   3.818995   4.809473  3.6845136   0.000000   0.570426
TR10103|c0_g1_i1_coral   1.287197   3.818995   6.011842  3.2751232   0.000000   5.133834
TR10108|c0_g1_i1_coral  14.159167  15.275980   2.805526  4.9126847  16.914895   5.989473
TR10132|c0_g1_i1_coral   6.435985   2.545997   0.000000  0.8187808   3.382979   1.426065
RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22 RI_B_07_14
TR10083|c0_g2_i1_coral          0  2.4538305   19.55961   3.448566   18.78455  1.4936671
TR10090|c0_g1_i1_coral          0  0.0000000    0.00000   2.069140    0.00000  0.7468335
TR10090|c0_g2_i1_coral          0  2.4538305    0.00000   4.827992    0.00000  3.7341677
TR10103|c0_g1_i1_coral          0  4.0897176    0.00000   3.448566    0.00000  2.2405006
TR10108|c0_g1_i1_coral          0  0.8179435    6.51987  10.345698   40.69986  3.7341677
TR10132|c0_g1_i1_coral          0  1.6358870    0.00000   1.379426    0.00000  8.9620025
RI_B_07_18 RI_B_07_22 RI_W_01_18 RI_W_01_22 RI_W_02_14 RI_W_02_18
TR10083|c0_g2_i1_coral   8.070182   36.85316  10.815190   4.646396   0.000000  4.9947759
TR10090|c0_g1_i1_coral   2.017545    0.00000  12.257216   1.548799   0.000000  1.9979104
TR10090|c0_g2_i1_coral   1.008773    0.00000  12.257216   7.743994   4.651361  4.9947759
TR10103|c0_g1_i1_coral   2.017545    0.00000   7.931140   0.000000   0.000000  0.9989552
TR10108|c0_g1_i1_coral   1.008773    0.00000   2.884051   1.548799   9.302721  6.9926863
TR10132|c0_g1_i1_coral  44.386000   27.63987   1.442025   0.000000   0.000000  5.9937311
RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18
TR10083|c0_g2_i1_coral   3.615216   8.525621   5.407467   4.745758   12.82564  2.9768406
TR10090|c0_g1_i1_coral   0.000000   4.262810   1.081493   1.581919    0.00000  3.4729807
TR10090|c0_g2_i1_coral   4.820288   3.044865   3.244480   1.581919    0.00000  7.4421015
TR10103|c0_g1_i1_coral   3.615216   3.044865   3.244480   1.054613    0.00000  0.9922802
TR10108|c0_g1_i1_coral   2.410144   2.435892   4.686471   2.109226   21.37607 12.8996426
TR10132|c0_g1_i1_coral  44.587667   1.217946   1.081493   1.054613   29.92650  7.9382416
RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22
TR10083|c0_g2_i1_coral  3.1630643   1.010704   5.526636   1.044347   1.336966   5.564881
TR10090|c0_g1_i1_coral  0.9037326   8.085630   0.000000   0.000000   0.000000   4.451904
TR10090|c0_g2_i1_coral  3.6149306  10.107037   1.381659   3.133041   1.336966   7.790833
TR10103|c0_g1_i1_coral  4.0667969   2.021407   2.763318  10.443469   1.336966   2.225952
TR10108|c0_g1_i1_coral  5.4223959   2.021407   6.908296  11.487816   0.000000   5.564881
TR10132|c0_g1_i1_coral  1.8074653  10.107037  27.633182   5.221735   1.336966  42.293092
RI_W_07_14 RI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10083|c0_g2_i1_coral   4.124636   5.593017   6.900578   2.551157   1.731769   2.845097
TR10090|c0_g1_i1_coral   0.000000   0.000000   0.000000   7.653472   9.091789   1.707058
TR10090|c0_g2_i1_coral   2.062318   1.118603   5.520462  10.204629   5.628251   5.690194
TR10103|c0_g1_i1_coral   5.499514  13.423242   1.380116   3.826736   2.597654   2.845097
TR10108|c0_g1_i1_coral   0.000000   1.118603   5.520462   3.826736   9.091789   2.845097
TR10132|c0_g1_i1_coral   1.374879   2.237207  86.947282   1.275579   1.298827   0.000000
VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14 VA_B_03_18 VA_B_03_22
TR10083|c0_g2_i1_coral          0    14.2516  19.312637  16.029307  14.339493   7.833217
TR10090|c0_g1_i1_coral          0     0.0000   0.000000   5.828839   2.150924   7.833217
TR10090|c0_g2_i1_coral          0     0.0000   0.000000  10.929073   4.301848   3.560553
TR10103|c0_g1_i1_coral          0     0.0000   2.145849   2.185815   4.301848   3.560553
TR10108|c0_g1_i1_coral          0     0.0000   0.000000  14.572098  18.641340  21.363318
TR10132|c0_g1_i1_coral          0     0.0000   8.583394   9.471863  30.829909   8.545327
VA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10083|c0_g2_i1_coral   0.000000  0.9006809   6.731616   2.098865  3.3173156   1.813677
TR10090|c0_g1_i1_coral   0.000000  4.5034046   3.624716   4.897352  1.1057719   7.254707
TR10090|c0_g2_i1_coral   7.108517  3.1523832   0.000000   1.399244  0.5528859   5.441030
TR10103|c0_g1_i1_coral   4.442823  0.0000000   0.000000   0.000000  3.3173156   0.000000
TR10108|c0_g1_i1_coral   2.665694  4.9537450   3.624716   9.095083  5.5288593   0.000000
TR10132|c0_g1_i1_coral   0.000000  2.7020428   0.000000   0.000000  4.9759734   5.441030
VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22
TR10083|c0_g2_i1_coral   1.383946   0.000000  5.9098781   3.728534  0.7405861   2.098499
TR10090|c0_g1_i1_coral   8.303676   0.000000  0.4924898   0.000000  3.7029304   2.098499
TR10090|c0_g2_i1_coral   0.000000   4.206352  1.4774695   1.491414 15.5523076   6.295498
TR10103|c0_g1_i1_coral   5.535784   6.730163  3.9399187   1.491414  8.1464468   4.196999
TR10108|c0_g1_i1_coral  11.763540   4.206352  3.4474289   2.237121  2.2217582   5.246248
TR10132|c0_g1_i1_coral  18.683270   2.523811 14.2822054   8.948482 17.0334798   2.098499
VA_W_01_14 VA_W_01_18 VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_22
TR10083|c0_g2_i1_coral   6.699293   7.732656   0.000000  4.5789684  3.1096023   6.888125
TR10090|c0_g1_i1_coral   3.828167   1.104665   6.978382  0.6541383  5.4418040   9.392897
TR10090|c0_g2_i1_coral   0.000000   3.866328   4.361489 13.0827667  6.9966052   3.130966
TR10103|c0_g1_i1_coral   0.000000   3.866328   5.233787  0.6541383  6.9966052   2.504773
TR10108|c0_g1_i1_coral   1.914084   4.418660   0.000000  9.1579367  3.1096023   3.757159
TR10132|c0_g1_i1_coral   0.000000  10.494319   2.616893 13.0827667  0.7774006  13.150056
VA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22
TR10083|c0_g2_i1_coral   1.026752   4.334366   6.410680  0.0000000   3.848517  12.379823
TR10090|c0_g1_i1_coral   2.566880   5.779155   3.945034  0.0000000   1.099576  12.379823
TR10090|c0_g2_i1_coral   2.053504   7.705540  10.848842  0.0000000   2.748941   7.427894
TR10103|c0_g1_i1_coral   3.593631   3.852770   2.958775  0.8118702   3.298729   2.475965
TR10108|c0_g1_i1_coral   6.673887  20.227043   8.383196  1.6237403   5.497882  38.377452
TR10132|c0_g1_i1_coral   2.566880   0.000000   0.000000  2.4356105   8.796611   2.475965
VA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22
TR10083|c0_g2_i1_coral   5.042735  2.5895188   0.000000 10.2466895   1.127608  0.9473408
TR10090|c0_g1_i1_coral   0.000000  0.8631729   1.764014  1.5764138   1.127608  6.6313859
TR10090|c0_g2_i1_coral   3.025641  0.8631729   0.000000  5.5174482   3.382824  2.8420225
TR10103|c0_g1_i1_coral   8.068376  2.5895188   4.410036  4.7292413   0.000000  1.8946817
TR10108|c0_g1_i1_coral   4.034188  6.9053834   5.292043  3.1528275   1.127608  1.8946817
TR10132|c0_g1_i1_coral   9.076923 29.3478796  26.460214  0.7882069   6.765648  8.5260676
head(scaledcounts)
RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10083|c0_g2_i1_coral          0  0.8671827   2.447866   0.000000   3.455170   4.432653
TR10090|c0_g1_i1_coral          0  2.6015481   1.223933   0.000000  13.820681   0.000000
TR10090|c0_g2_i1_coral          0  3.4687308   2.447866   3.772786   0.000000   0.000000
TR10103|c0_g1_i1_coral          0  0.8671827   4.895733   0.000000   0.000000   8.865306
TR10108|c0_g1_i1_coral          0  1.7343654   4.895733   0.000000   6.910341   0.000000
TR10132|c0_g1_i1_coral          0  6.0702789   1.223933  18.863931  51.827555   0.000000
RI_B_03_14 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10083|c0_g2_i1_coral  12.871970   1.272998  12.424473  8.1878079   0.000000   4.563408
TR10090|c0_g1_i1_coral   0.000000   0.000000   2.003947  4.0939039   6.765958   0.570426
TR10090|c0_g2_i1_coral   0.000000   3.818995   4.809473  3.6845136   0.000000   0.570426
TR10103|c0_g1_i1_coral   1.287197   3.818995   6.011842  3.2751232   0.000000   5.133834
TR10108|c0_g1_i1_coral  14.159167  15.275980   2.805526  4.9126847  16.914895   5.989473
TR10132|c0_g1_i1_coral   6.435985   2.545997   0.000000  0.8187808   3.382979   1.426065
RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22 RI_B_07_14
TR10083|c0_g2_i1_coral          0  2.4538305   19.55961   3.448566   18.78455  1.4936671
TR10090|c0_g1_i1_coral          0  0.0000000    0.00000   2.069140    0.00000  0.7468335
TR10090|c0_g2_i1_coral          0  2.4538305    0.00000   4.827992    0.00000  3.7341677
TR10103|c0_g1_i1_coral          0  4.0897176    0.00000   3.448566    0.00000  2.2405006
TR10108|c0_g1_i1_coral          0  0.8179435    6.51987  10.345698   40.69986  3.7341677
TR10132|c0_g1_i1_coral          0  1.6358870    0.00000   1.379426    0.00000  8.9620025
RI_B_07_18 RI_B_07_22 RI_W_01_18 RI_W_01_22 RI_W_02_14 RI_W_02_18
TR10083|c0_g2_i1_coral   8.070182   36.85316  10.815190   4.646396   0.000000  4.9947759
TR10090|c0_g1_i1_coral   2.017545    0.00000  12.257216   1.548799   0.000000  1.9979104
TR10090|c0_g2_i1_coral   1.008773    0.00000  12.257216   7.743994   4.651361  4.9947759
TR10103|c0_g1_i1_coral   2.017545    0.00000   7.931140   0.000000   0.000000  0.9989552
TR10108|c0_g1_i1_coral   1.008773    0.00000   2.884051   1.548799   9.302721  6.9926863
TR10132|c0_g1_i1_coral  44.386000   27.63987   1.442025   0.000000   0.000000  5.9937311
RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18
TR10083|c0_g2_i1_coral   3.615216   8.525621   5.407467   4.745758   12.82564  2.9768406
TR10090|c0_g1_i1_coral   0.000000   4.262810   1.081493   1.581919    0.00000  3.4729807
TR10090|c0_g2_i1_coral   4.820288   3.044865   3.244480   1.581919    0.00000  7.4421015
TR10103|c0_g1_i1_coral   3.615216   3.044865   3.244480   1.054613    0.00000  0.9922802
TR10108|c0_g1_i1_coral   2.410144   2.435892   4.686471   2.109226   21.37607 12.8996426
TR10132|c0_g1_i1_coral  44.587667   1.217946   1.081493   1.054613   29.92650  7.9382416
RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22
TR10083|c0_g2_i1_coral  3.1630643   1.010704   5.526636   1.044347   1.336966   5.564881
TR10090|c0_g1_i1_coral  0.9037326   8.085630   0.000000   0.000000   0.000000   4.451904
TR10090|c0_g2_i1_coral  3.6149306  10.107037   1.381659   3.133041   1.336966   7.790833
TR10103|c0_g1_i1_coral  4.0667969   2.021407   2.763318  10.443469   1.336966   2.225952
TR10108|c0_g1_i1_coral  5.4223959   2.021407   6.908296  11.487816   0.000000   5.564881
TR10132|c0_g1_i1_coral  1.8074653  10.107037  27.633182   5.221735   1.336966  42.293092
RI_W_07_14 RI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10083|c0_g2_i1_coral   4.124636   5.593017   6.900578   2.551157   1.731769   2.845097
TR10090|c0_g1_i1_coral   0.000000   0.000000   0.000000   7.653472   9.091789   1.707058
TR10090|c0_g2_i1_coral   2.062318   1.118603   5.520462  10.204629   5.628251   5.690194
TR10103|c0_g1_i1_coral   5.499514  13.423242   1.380116   3.826736   2.597654   2.845097
TR10108|c0_g1_i1_coral   0.000000   1.118603   5.520462   3.826736   9.091789   2.845097
TR10132|c0_g1_i1_coral   1.374879   2.237207  86.947282   1.275579   1.298827   0.000000
VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14 VA_B_03_18 VA_B_03_22
TR10083|c0_g2_i1_coral          0    14.2516  19.312637  16.029307  14.339493   7.833217
TR10090|c0_g1_i1_coral          0     0.0000   0.000000   5.828839   2.150924   7.833217
TR10090|c0_g2_i1_coral          0     0.0000   0.000000  10.929073   4.301848   3.560553
TR10103|c0_g1_i1_coral          0     0.0000   2.145849   2.185815   4.301848   3.560553
TR10108|c0_g1_i1_coral          0     0.0000   0.000000  14.572098  18.641340  21.363318
TR10132|c0_g1_i1_coral          0     0.0000   8.583394   9.471863  30.829909   8.545327
VA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10083|c0_g2_i1_coral   0.000000  0.9006809   6.731616   2.098865  3.3173156   1.813677
TR10090|c0_g1_i1_coral   0.000000  4.5034046   3.624716   4.897352  1.1057719   7.254707
TR10090|c0_g2_i1_coral   7.108517  3.1523832   0.000000   1.399244  0.5528859   5.441030
TR10103|c0_g1_i1_coral   4.442823  0.0000000   0.000000   0.000000  3.3173156   0.000000
TR10108|c0_g1_i1_coral   2.665694  4.9537450   3.624716   9.095083  5.5288593   0.000000
TR10132|c0_g1_i1_coral   0.000000  2.7020428   0.000000   0.000000  4.9759734   5.441030
VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22
TR10083|c0_g2_i1_coral   1.383946   0.000000  5.9098781   3.728534  0.7405861   2.098499
TR10090|c0_g1_i1_coral   8.303676   0.000000  0.4924898   0.000000  3.7029304   2.098499
TR10090|c0_g2_i1_coral   0.000000   4.206352  1.4774695   1.491414 15.5523076   6.295498
TR10103|c0_g1_i1_coral   5.535784   6.730163  3.9399187   1.491414  8.1464468   4.196999
TR10108|c0_g1_i1_coral  11.763540   4.206352  3.4474289   2.237121  2.2217582   5.246248
TR10132|c0_g1_i1_coral  18.683270   2.523811 14.2822054   8.948482 17.0334798   2.098499
VA_W_01_14 VA_W_01_18 VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_22
TR10083|c0_g2_i1_coral   6.699293   7.732656   0.000000  4.5789684  3.1096023   6.888125
TR10090|c0_g1_i1_coral   3.828167   1.104665   6.978382  0.6541383  5.4418040   9.392897
TR10090|c0_g2_i1_coral   0.000000   3.866328   4.361489 13.0827667  6.9966052   3.130966
TR10103|c0_g1_i1_coral   0.000000   3.866328   5.233787  0.6541383  6.9966052   2.504773
TR10108|c0_g1_i1_coral   1.914084   4.418660   0.000000  9.1579367  3.1096023   3.757159
TR10132|c0_g1_i1_coral   0.000000  10.494319   2.616893 13.0827667  0.7774006  13.150056
VA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22
TR10083|c0_g2_i1_coral   1.026752   4.334366   6.410680  0.0000000   3.848517  12.379823
TR10090|c0_g1_i1_coral   2.566880   5.779155   3.945034  0.0000000   1.099576  12.379823
TR10090|c0_g2_i1_coral   2.053504   7.705540  10.848842  0.0000000   2.748941   7.427894
TR10103|c0_g1_i1_coral   3.593631   3.852770   2.958775  0.8118702   3.298729   2.475965
TR10108|c0_g1_i1_coral   6.673887  20.227043   8.383196  1.6237403   5.497882  38.377452
TR10132|c0_g1_i1_coral   2.566880   0.000000   0.000000  2.4356105   8.796611   2.475965
VA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22
TR10083|c0_g2_i1_coral   5.042735  2.5895188   0.000000 10.2466895   1.127608  0.9473408
TR10090|c0_g1_i1_coral   0.000000  0.8631729   1.764014  1.5764138   1.127608  6.6313859
TR10090|c0_g2_i1_coral   3.025641  0.8631729   0.000000  5.5174482   3.382824  2.8420225
TR10103|c0_g1_i1_coral   8.068376  2.5895188   4.410036  4.7292413   0.000000  1.8946817
TR10108|c0_g1_i1_coral   4.034188  6.9053834   5.292043  3.1528275   1.127608  1.8946817
TR10132|c0_g1_i1_coral   9.076923 29.3478796  26.460214  0.7882069   6.765648  8.5260676
genes4heatmap&lt;-resOrigin[resOrigin$pvalue &lt;0.05 &amp; !is.na(resOrigin$pvalue),]
names(genes4heatmap)
[1] "baseMean"       "log2FoldChange" "lfcSE"          "stat"           "pvalue"
                      <br/>
                      [6] "padj"
                      <br/>
                      head(genes4heatmap)
log2 fold change (MLE): Origin VA vs RI
Wald test p-value: Origin VA vs RI
DataFrame with 6 rows and 6 columns
baseMean    log2FoldChange             lfcSE
                  
TR10090|c0_g2_i1_coral 3.65756460141883 0.943134641716211 0.474719393211504
TR10487|c0_g1_i1_coral 6.57169761658039 0.727122969400975 0.363907404480523
TR1069|c0_g1_i1_coral  4.14205241197446  4.49029083181354  1.98238112955566
TR10748|c0_g1_i1_coral 2.93642320037697  1.78798410821023 0.559059118556594
TR10773|c0_g1_i1_coral 3.02515761016279  1.75847746453697 0.659498298908414
TR10773|c1_g1_i1_coral 2.32875934475272  1.29222344670373 0.632167692638483
stat              pvalue               padj
                     
TR10090|c0_g2_i1_coral 1.98672027139201  0.0469534041233123  0.418043000342497
TR10487|c0_g1_i1_coral 1.99809885824924  0.0457059435120877  0.412518610214263
TR1069|c0_g1_i1_coral  2.26509966467448  0.0235065642111382  0.308653252532063
TR10748|c0_g1_i1_coral 3.19820220950252 0.00138287275835452 0.0720091267133405
TR10773|c0_g1_i1_coral 2.66638665702029 0.00766714549040745  0.173395444167676
TR10773|c1_g1_i1_coral 2.04411497416195  0.0409421901910667                 NA
dim(genes4heatmap)
[1] 866   6
data4heatmap&lt;-scaledcounts[row.names(scaledcounts)%in%row.names(genes4heatmap),]
dim(data4heatmap)
[1] 866  78
head(data4heatmap)
RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10090|c0_g2_i1_coral    0.00000   3.468731   2.447866   3.772786    0.00000   0.000000
TR10487|c0_g1_i1_coral   10.94714   5.203096   8.567532   0.000000    0.00000   6.648979
TR1069|c0_g1_i1_coral     0.00000   0.000000   0.000000   7.545573    3.45517   6.648979
TR10748|c0_g1_i1_coral    0.00000   0.000000   2.447866   0.000000    0.00000   0.000000
TR10773|c0_g1_i1_coral    0.00000   6.070279   2.447866   3.772786    0.00000   0.000000
TR10773|c1_g1_i1_coral    0.00000   2.601548   4.895733   0.000000    0.00000   2.216326
RI_B_03_14 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10090|c0_g2_i1_coral   0.000000   3.818995   4.809473  3.6845136          0   0.570426
TR10487|c0_g1_i1_coral   9.010379   5.091993   2.404737  4.5032943          0   9.126817
TR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000  0.0000000          0   0.000000
TR10748|c0_g1_i1_coral   2.574394   0.000000   4.007894  2.8657328          0   1.711278
TR10773|c0_g1_i1_coral   1.287197   0.000000   0.000000  0.4093904          0   0.000000
TR10773|c1_g1_i1_coral   5.148788   1.272998   0.000000  0.0000000          0   0.285213
RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22 RI_B_07_14
TR10090|c0_g2_i1_coral   0.000000  2.4538305          0   4.827992          0   3.734168
TR10487|c0_g1_i1_coral   5.033716  0.8179435          0   2.758853          0   5.227835
TR1069|c0_g1_i1_coral    0.000000  0.0000000          0   0.000000          0   0.000000
TR10748|c0_g1_i1_coral   0.000000  0.0000000          0   0.000000          0   2.987334
TR10773|c0_g1_i1_coral   0.000000  0.0000000          0   0.000000          0   0.000000
TR10773|c1_g1_i1_coral   0.000000  0.8179435          0   0.000000          0   0.000000
RI_B_07_18 RI_B_07_22 RI_W_01_18 RI_W_01_22 RI_W_02_14 RI_W_02_18
TR10090|c0_g2_i1_coral   1.008773          0  12.257216   7.743994   4.651361  4.9947759
TR10487|c0_g1_i1_coral   5.043864          0   8.652152   6.195195  16.279762  2.9968656
TR1069|c0_g1_i1_coral    0.000000          0  10.094178   9.292792   0.000000  0.0000000
TR10748|c0_g1_i1_coral   0.000000          0   0.000000   1.548799   4.651361  0.9989552
TR10773|c0_g1_i1_coral   1.008773          0   7.931140  17.036786   0.000000 14.9843278
TR10773|c1_g1_i1_coral   0.000000          0   5.047089  18.585585   0.000000 10.9885071
RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18
TR10090|c0_g2_i1_coral   4.820288   3.044865   3.244480   1.581919    0.00000  7.4421015
TR10487|c0_g1_i1_coral   1.205072   7.307675   7.209956  14.237275   17.10086  5.4575411
TR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000   0.000000    0.00000  0.0000000
TR10748|c0_g1_i1_coral   2.410144   2.435892   6.128462   4.218452    0.00000  0.4961401
TR10773|c0_g1_i1_coral   6.025360  23.140971  12.617422  29.001856    0.00000  1.9845604
TR10773|c1_g1_i1_coral   1.205072  10.352540   7.930951   7.382291    0.00000  1.4884203
RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22
TR10090|c0_g2_i1_coral   3.614931  10.107037   1.381659   3.133041   1.336966   7.790833
TR10487|c0_g1_i1_coral   4.066797   4.042815   0.000000   4.177388  13.369662  10.016785
TR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000   0.000000   0.000000   0.000000
TR10748|c0_g1_i1_coral   2.711198   1.010704   0.000000   3.133041   2.673932   2.225952
TR10773|c0_g1_i1_coral   5.422396   2.021407   0.000000   5.221735   2.673932  10.016785
TR10773|c1_g1_i1_coral   3.614931   3.032111   0.000000   1.044347   2.673932   3.338928
RI_W_07_14 RI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10090|c0_g2_i1_coral   2.062318   1.118603   5.520462  10.204629   5.628251   5.690194
TR10487|c0_g1_i1_coral   3.437196  13.423242   8.280694  10.204629   6.494135  10.242350
TR1069|c0_g1_i1_coral    0.000000   0.000000   0.000000  33.165045   9.524732  17.639603
TR10748|c0_g1_i1_coral   2.749757   2.237207   0.000000   1.275579   1.298827   5.121175
TR10773|c0_g1_i1_coral   1.374879   2.237207   0.000000   6.377893   4.762366   2.845097
TR10773|c1_g1_i1_coral   2.062318   3.355810   2.760231   8.929051   6.061193   6.259214
VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14 VA_B_03_18 VA_B_03_22
TR10090|c0_g2_i1_coral   0.000000     0.0000   0.000000  10.929073   4.301848  3.5605530
TR10487|c0_g1_i1_coral   0.000000     0.0000  17.166789  10.929073   7.886721  7.8332166
TR1069|c0_g1_i1_coral   86.631891    44.0504  40.771123   0.000000   0.000000  0.0000000
TR10748|c0_g1_i1_coral  30.575962     0.0000   2.145849   3.643024   7.886721  3.5605530
TR10773|c0_g1_i1_coral   1.698665     0.0000   6.437546   0.000000   1.433949  0.7121106
TR10773|c1_g1_i1_coral   0.000000     2.5912   0.000000   0.000000   3.584873  0.7121106
VA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10090|c0_g2_i1_coral  7.1085175   3.152383   0.000000   1.399244  0.5528859   5.441030
TR10487|c0_g1_i1_coral 15.1055996   4.053064   8.285065   4.547541 17.1394640  21.764122
TR1069|c0_g1_i1_coral   0.0000000   0.000000   0.000000   0.000000  0.0000000   0.000000
TR10748|c0_g1_i1_coral  2.6656940   0.000000   8.802882   1.049433  1.6586578   3.627354
TR10773|c0_g1_i1_coral  0.0000000   7.205447   8.285065   1.399244  1.1057719   0.000000
TR10773|c1_g1_i1_coral  0.8885647   8.556469   2.589083   0.000000  4.4230875   0.000000
VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22
TR10090|c0_g2_i1_coral   0.000000   4.206352   1.477470  1.4914137  15.552308   6.295498
TR10487|c0_g1_i1_coral   0.000000   8.412704   2.462449  3.7285342   5.184103   8.393997
TR1069|c0_g1_i1_coral   29.062864  16.825408   8.372327  0.0000000   0.000000   0.000000
TR10748|c0_g1_i1_coral   6.919730   0.000000   2.954939  6.7113616   0.000000   4.196999
TR10773|c0_g1_i1_coral   2.767892   0.000000   5.417388  1.4914137   2.962344   6.295498
TR10773|c1_g1_i1_coral   0.000000   2.523811   2.462449  0.7457068   0.000000   4.196999
VA_W_01_14 VA_W_01_18 VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_22
TR10090|c0_g2_i1_coral   0.000000   3.866328  4.3614889  13.082767  6.9966052   3.130966
TR10487|c0_g1_i1_coral   6.699293   4.970993  1.7445956  13.082767  3.1096023   9.392897
TR1069|c0_g1_i1_coral    0.000000   0.000000  0.0000000   0.000000  0.0000000   0.000000
TR10748|c0_g1_i1_coral   2.871126   2.209330  0.8722978   0.000000  0.7774006   3.757159
TR10773|c0_g1_i1_coral   5.742251   1.104665  0.0000000   3.924830  1.5548011   2.504773
TR10773|c1_g1_i1_coral   0.000000   3.866328  0.0000000   5.233107  0.7774006   2.504773
VA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22
TR10090|c0_g2_i1_coral   2.053504   7.705540 10.8488424    0.00000   2.748941   7.427894
TR10487|c0_g1_i1_coral   3.593631  13.003099  1.4793876    0.00000   8.246822   8.665876
TR1069|c0_g1_i1_coral    0.000000   0.000000  0.0000000    0.00000   0.000000   0.000000
TR10748|c0_g1_i1_coral   6.673887   9.150329  7.3969380   14.61366   2.748941   9.903859
TR10773|c0_g1_i1_coral   1.026752   0.000000  0.4931292    0.00000   0.000000   0.000000
TR10773|c1_g1_i1_coral   1.026752   0.000000  0.0000000    0.00000   2.199153   1.237982
VA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22
TR10090|c0_g2_i1_coral   3.025641  0.8631729   0.000000   5.517448   3.382824   2.842023
TR10487|c0_g1_i1_coral  10.085470  1.7263459  11.466093   7.093862   6.765648   3.789363
TR1069|c0_g1_i1_coral    0.000000  0.0000000   0.000000   0.000000   0.000000   0.000000
TR10748|c0_g1_i1_coral   2.017094  6.0422105   4.410036   0.000000   3.382824   1.894682
TR10773|c0_g1_i1_coral   0.000000  1.7263459   0.000000   0.000000   0.000000   0.000000
TR10773|c1_g1_i1_coral   2.017094  0.8631729   5.292043   0.000000   0.000000   0.000000
temp = as.matrix(rowMeans(data4heatmap))
head(temp)
[,1]
TR10090|c0_g2_i1_coral 3.657565
TR10487|c0_g1_i1_coral 6.571698
TR1069|c0_g1_i1_coral  4.142052
TR10748|c0_g1_i1_coral 2.936423
TR10773|c0_g1_i1_coral 3.025158
TR10773|c1_g1_i1_coral 2.328759
scaledmatrix&lt;-matrix(data=temp, nrow=nrow(data4heatmap), ncol=ncol(data4heatmap), byrow=FALSE)
data4heatmapscaled = data4heatmap/scaledmatrix
head(data4heatmapscaled)
RI_B_01_14 RI_B_01_18 RI_B_01_22 RI_B_02_14 RI_B_02_18 RI_B_02_22
TR10090|c0_g2_i1_coral   0.000000  0.9483717  0.6692613   1.031502  0.0000000  0.0000000
TR10487|c0_g1_i1_coral   1.665801  0.7917431  1.3037015   0.000000  0.0000000  1.0117598
TR1069|c0_g1_i1_coral    0.000000  0.0000000  0.0000000   1.821699  0.8341687  1.6052378
TR10748|c0_g1_i1_coral   0.000000  0.0000000  0.8336218   0.000000  0.0000000  0.0000000
TR10773|c0_g1_i1_coral   0.000000  2.0065992  0.8091699   1.247137  0.0000000  0.0000000
TR10773|c1_g1_i1_coral   0.000000  1.1171391  2.1022922   0.000000  0.0000000  0.9517198
RI_B_03_14 RI_B_03_22 RI_B_04_14 RI_B_04_18 RI_B_04_22 RI_B_05_14
TR10090|c0_g2_i1_coral  0.0000000  1.0441360  1.3149387  1.0073680          0  0.1559579
TR10487|c0_g1_i1_coral  1.3710885  0.7748368  0.3659232  0.6852559          0  1.3888066
TR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000  0.0000000          0  0.0000000
TR10748|c0_g1_i1_coral  0.8767109  0.0000000  1.3648899  0.9759263          0  0.5827764
TR10773|c0_g1_i1_coral  0.4254975  0.0000000  0.0000000  0.1353286          0  0.0000000
TR10773|c1_g1_i1_coral  2.2109576  0.5466423  0.0000000  0.0000000          0  0.1224742
RI_B_05_18 RI_B_05_22 RI_B_06_14 RI_B_06_18 RI_B_06_22 RI_B_07_14
TR10090|c0_g2_i1_coral  0.0000000  0.6708919          0  1.3200019          0  1.0209437
TR10487|c0_g1_i1_coral  0.7659688  0.1244646          0  0.4198082          0  0.7955075
TR1069|c0_g1_i1_coral   0.0000000  0.0000000          0  0.0000000          0  0.0000000
TR10748|c0_g1_i1_coral  0.0000000  0.0000000          0  0.0000000          0  1.0173377
TR10773|c0_g1_i1_coral  0.0000000  0.0000000          0  0.0000000          0  0.0000000
TR10773|c1_g1_i1_coral  0.0000000  0.3512357          0  0.0000000          0  0.0000000
RI_B_07_18 RI_B_07_22 RI_W_01_18 RI_W_01_22 RI_W_02_14 RI_W_02_18
TR10090|c0_g2_i1_coral  0.2758045          0   3.351196  2.1172541   1.271710  1.3656016
TR10487|c0_g1_i1_coral  0.7675130          0   1.316578  0.9427084   2.477254  0.4560261
TR1069|c0_g1_i1_coral   0.0000000          0   2.436999  2.2435236   0.000000  0.0000000
TR10748|c0_g1_i1_coral  0.0000000          0   0.000000  0.5274440   1.584023  0.3401946
TR10773|c0_g1_i1_coral  0.3334612          0   2.621728  5.6317019   0.000000  4.9532387
TR10773|c1_g1_i1_coral  0.0000000          0   2.167287  7.9808953   0.000000  4.7186100
RI_W_02_22 RI_W_03_14 RI_W_03_18 RI_W_03_22 RI_W_04_14 RI_W_04_18
TR10090|c0_g2_i1_coral  1.3178956  0.8324842  0.8870602  0.4325062   0.000000  2.0347150
TR10487|c0_g1_i1_coral  0.1833730  1.1119920  1.0971223  2.1664531   2.602198  0.8304614
TR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000  0.0000000   0.000000  0.0000000
TR10748|c0_g1_i1_coral  0.8207755  0.8295438  2.0870501  1.4365953   0.000000  0.1689607
TR10773|c0_g1_i1_coral  1.9917509  7.6495091  4.1708314  9.5868908   0.000000  0.6560188
TR10773|c1_g1_i1_coral  0.5174739  4.4455171  3.4056552  3.1700530   0.000000  0.6391473
RI_W_05_14 RI_W_05_18 RI_W_05_22 RI_W_06_14 RI_W_06_18 RI_W_06_22
TR10090|c0_g2_i1_coral  0.9883436  2.7633244  0.3777539  0.8565921  0.3655346  2.1300602
TR10487|c0_g1_i1_coral  0.6188351  0.6151858  0.0000000  0.6356634  2.0344306  1.5242310
TR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000
TR10748|c0_g1_i1_coral  0.9232995  0.3441955  0.0000000  1.0669582  0.9106087  0.7580488
TR10773|c0_g1_i1_coral  1.7924342  0.6681991  0.0000000  1.7261033  0.8838986  3.3111614
TR10773|c1_g1_i1_coral  1.5522989  1.3020286  0.0000000  0.4484564  1.1482219  1.4337799
RI_W_07_14 RI_W_07_18 RI_W_07_22 VA_B_01_14 VA_B_01_18 VA_B_01_22
TR10090|c0_g2_i1_coral  0.5638500  0.3058329   1.509327  2.7900066  1.5387973   1.555733
TR10487|c0_g1_i1_coral  0.5230302  2.0425836   1.260054  1.5528148  0.9881975   1.558555
TR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000  8.0069110  2.2995199   4.258662
TR10748|c0_g1_i1_coral  0.9364308  0.7618816   0.000000  0.4343988  0.4423160   1.744018
TR10773|c0_g1_i1_coral  0.4544816  0.7395340   0.000000  2.1082846  1.5742538   0.940479
TR10773|c1_g1_i1_coral  0.8855865  1.4410293   1.185280  3.8342522  2.6027562   2.687789
VA_B_02_14 VA_B_02_18 VA_B_02_22 VA_B_03_14 VA_B_03_18 VA_B_03_22
TR10090|c0_g2_i1_coral  0.0000000   0.000000  0.0000000   2.988074  1.1761509  0.9734764
TR10487|c0_g1_i1_coral  0.0000000   0.000000  2.6122305   1.663052  1.2001041  1.1919624
TR1069|c0_g1_i1_coral  20.9152088  10.634922  9.8432176   0.000000  0.0000000  0.0000000
TR10748|c0_g1_i1_coral 10.4126549   0.000000  0.7307695   1.240633  2.6858257  1.2125476
TR10773|c0_g1_i1_coral  0.5615127   0.000000  2.1280034   0.000000  0.4740081  0.2353962
TR10773|c1_g1_i1_coral  0.0000000   1.112696  0.0000000   0.000000  1.5393918  0.3057897
VA_B_04_14 VA_B_04_18 VA_B_04_22 VA_B_05_14 VA_B_05_18 VA_B_05_22
TR10090|c0_g2_i1_coral  1.9435111  0.8618804   0.000000  0.3825615  0.1511623   1.487610
TR10487|c0_g1_i1_coral  2.2985841  0.6167454   1.260719  0.6919888  2.6080725   3.311796
TR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000  0.0000000  0.0000000   0.000000
TR10748|c0_g1_i1_coral  0.9078031  0.0000000   2.997825  0.3573847  0.5648565   1.235297
TR10773|c0_g1_i1_coral  0.0000000  2.3818420   2.738722  0.4625357  0.3655254   0.000000
TR10773|c1_g1_i1_coral  0.3815614  3.6742606   1.111786  0.0000000  1.8993321   0.000000
VA_B_06_14 VA_B_06_18 VA_B_06_22 VA_B_07_14 VA_B_07_18 VA_B_07_22
TR10090|c0_g2_i1_coral  0.0000000   1.150042  0.4039490  0.4077614  4.2520938   1.721227
TR10487|c0_g1_i1_coral  0.0000000   1.280142  0.3747052  0.5673624  0.7888529   1.277295
TR1069|c0_g1_i1_coral   7.0165371   4.062094  2.0212992  0.0000000  0.0000000   0.000000
TR10748|c0_g1_i1_coral  2.3565165   0.000000  1.0063056  2.2855567  0.0000000   1.429289
TR10773|c0_g1_i1_coral  0.9149579   0.000000  1.7907788  0.4930036  0.9792364   2.081048
TR10773|c1_g1_i1_coral  0.0000000   1.083758  1.0574082  0.3202164  0.0000000   1.802247
VA_W_01_14 VA_W_01_18 VA_W_02_14 VA_W_02_18 VA_W_03_14 VA_W_03_22
TR10090|c0_g2_i1_coral  0.0000000  1.0570771  1.1924571   3.576907  1.9129136  0.8560247
TR10487|c0_g1_i1_coral  1.0194159  0.7564245  0.2654711   1.990774  0.4731810  1.4292954
TR1069|c0_g1_i1_coral   0.0000000  0.0000000  0.0000000   0.000000  0.0000000  0.0000000
TR10748|c0_g1_i1_coral  0.9777629  0.7523882  0.2970613   0.000000  0.2647441  1.2795019
TR10773|c0_g1_i1_coral  1.8981660  0.3651595  0.0000000   1.297397  0.5139571  0.8279809
TR10773|c1_g1_i1_coral  0.0000000  1.6602522  0.0000000   2.247165  0.3338261  1.0755824
VA_W_04_14 VA_W_04_18 VA_W_04_22 VA_W_05_14 VA_W_05_18 VA_W_05_22
TR10090|c0_g2_i1_coral  0.5614402   2.106741  2.9661383   0.000000  0.7515768   2.030831
TR10487|c0_g1_i1_coral  0.5468346   1.978651  0.2251150   0.000000  1.2548999   1.318666
TR1069|c0_g1_i1_coral   0.0000000   0.000000  0.0000000   0.000000  0.0000000   0.000000
TR10748|c0_g1_i1_coral  2.2727947   3.116148  2.5190299   4.976688  0.9361528   3.372763
TR10773|c0_g1_i1_coral  0.3394044   0.000000  0.1630094   0.000000  0.0000000   0.000000
TR10773|c1_g1_i1_coral  0.4409008   0.000000  0.0000000   0.000000  0.9443452   0.531606
VA_W_06_14 VA_W_06_18 VA_W_06_22 VA_W_07_14 VA_W_07_18 VA_W_07_22
TR10090|c0_g2_i1_coral  0.8272283  0.2359966   0.000000   1.508503  0.9248843  0.7770259
TR10487|c0_g1_i1_coral  1.5346826  0.2626940   1.744769   1.079457  1.0295131  0.5766186
TR1069|c0_g1_i1_coral   0.0000000  0.0000000   0.000000   0.000000  0.0000000  0.0000000
TR10748|c0_g1_i1_coral  0.6869221  2.0576770   1.501839   0.000000  1.1520220  0.6452345
TR10773|c0_g1_i1_coral  0.0000000  0.5706631   0.000000   0.000000  0.0000000  0.0000000
TR10773|c1_g1_i1_coral  0.8661668  0.3706578   2.272473   0.000000  0.0000000  0.0000000
dim(data4heatmapscaled)
[1] 866  78
pairs.breaks &lt;- seq(0, 3.0, by=0.1)
length(pairs.breaks)
[1] 31
mycol &lt;- colorpanel(n=30, low="black", high="yellow")
pdf(file="XXXXXX_byrow.pdf",7,7)
heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=F, dendrogram = c("row"), scale="none", keysize=1, breaks=pairs.breaks, col=mycol, trace = "none", symkey = F, density.info = "density", colsep=c(24), sepcolor=c("white"), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
dev.off()
RStudioGD
2
pdf(file="XXXXXX_bycolumn.pdf",7,7)
heatmap.2(data.matrix(data4heatmapscaled), Rowv=T, Colv=T, dendrogram = c("col"), scale="none", keysize=1, breaks=pairs.breaks, col=mycol, trace = "none", symkey = F, density.info = "density", colsep=c(24), sepcolor=c("white"), sepwidth=c(.1,.1), margins=c(10,10), labRow=F)
dev.off()
RStudioGD
2
plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup="Origin")
plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup="SymState")
plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup="Temp")
plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup="Origin_SymState")
plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup="Origin_SymState_Temp")
pdf(file="866_RIvsVA_PCAv2.pdf",14, 14)
plotPCA(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),], intgroup="Origin")
dev.off()
pdf
4
prop.nullv3 &lt;- apply(countsTable[rownames(countsTable)%in%row.names(genes4heatmap),], 2, function(x) 100*mean(x==0))
barplot(prop.nullv3, main="Percentage of null counts per sample",
                     </p>
                    </blockquote>
                    <ul dir="auto">
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    horiz=TRUE, cex.names=0.5, las=1, ">
                       <pre class="notranslate"><code>    horiz=TRUE, cex.names=0.5, las=1, 
</code></pre>
                      </div>
                     </li>
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')">
                       <pre class="notranslate"><code>    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
</code></pre>
                      </div>
                     </li>
                    </ul>
                    <blockquote>
                     <p dir="auto">
                      pdf(file="ResNullCountsv2.pdf",14, 14)
barplot(prop.nullv3, main="Percentage of null counts per sample",
                     </p>
                    </blockquote>
                    <ul dir="auto">
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    horiz=TRUE, cex.names=0.5, las=1, ">
                       <pre class="notranslate"><code>    horiz=TRUE, cex.names=0.5, las=1, 
</code></pre>
                      </div>
                     </li>
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')">
                       <pre class="notranslate"><code>    col=conds$Origin_SymState_Temp, ylab='Samples', xlab='% of null counts')
</code></pre>
                      </div>
                     </li>
                    </ul>
                    <blockquote>
                     <p dir="auto">
                      dev.off()
pdf
4
distsv2&lt;-dist(t(assay(vstCounts[rownames(vstCounts)%in%row.names(genes4heatmap),])))
plot(hclust(distsv2))
heatmap.2(as.matrix(distsv2), key=F, trace="none",
                     </p>
                    </blockquote>
                    <ul dir="auto">
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='      col=colorpanel(100, "black", "white"),'>
                       <pre class="notranslate"><code>      col=colorpanel(100, "black", "white"),
</code></pre>
                      </div>
                     </li>
                     <li>
                      <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="      margin=c(10, 10))">
                       <pre class="notranslate"><code>      margin=c(10, 10))
</code></pre>
                      </div>
                     </li>
                    </ul>
                    <blockquote>
                     <p dir="auto">
                      res&lt;-resOrigin
Annotations&lt;-read.delim("Assembly_GoodCoralSymbiont_suffixed_totalannotated.txt")
head(res)
log2 fold change (MLE): Origin VA vs RI
Wald test p-value: Origin VA vs RI
DataFrame with 6 rows and 6 columns
baseMean      log2FoldChange             lfcSE
                    
TR10083|c0_g2_i1_coral   5.366479327573  -0.271242746289843 0.446224512750353
TR10090|c0_g1_i1_coral 2.70436251126018    1.11251170496561 0.589306266446119
TR10090|c0_g2_i1_coral 3.65756460141883   0.943134641716211 0.474719393211504
TR10103|c0_g1_i1_coral  2.9127886212765  0.0386642676540012 0.456719612483097
TR10108|c0_g1_i1_coral 6.31144026709945 -0.0179009694277555 0.461957599914831
TR10132|c0_g1_i1_coral  9.2673703948612  -0.264990306479184 0.677607561801266
stat             pvalue              padj
                   
TR10083|c0_g2_i1_coral  -0.607861600022842  0.543279269700006 0.865660813784253
TR10090|c0_g1_i1_coral    1.88783281005095 0.0590483989759312                NA
TR10090|c0_g2_i1_coral    1.98672027139201 0.0469534041233123 0.418043000342497
TR10103|c0_g1_i1_coral  0.0846564644854882  0.932534507970102 0.984590265229257
TR10108|c0_g1_i1_coral -0.0387502433795997  0.969089515041603 0.992149683420423
TR10132|c0_g1_i1_coral  -0.391067516682912  0.695747330435347 0.921516804624192
sum(res$log2FoldChange!=0)
[1] 7659
genes&lt;-as.integer(p.adjust(res$pvalue[res$log2FoldChange!=0 &amp; !is.na(res$pvalue)], method="BH")&lt;.05)
names(genes)&lt;-row.names(res[res$log2FoldChange!=0 &amp; !is.na(res$pvalue),])
table(genes)
genes
0    1
7533  116
goterms&lt;-strsplit(as.character(Annotations$GO), split=" // ")
names(goterms)&lt;-Annotations$ContigName
pwf&lt;-nullp(genes, bias.data=Annotations[Annotations$ContigName%in%names(genes),"ContigLength"])
Warning message:
In pcls(G) : initial point very close to some inequality constraints
GOGOGO&lt;-goseq(pwf,gene2cat=goterms)
Using manually entered categories.
Calculating the p-values...
'select()' returned 1:1 mapping between keys and columns
GOGOGO$padj&lt;-p.adjust(GOGOGO$over_represented_pvalue, method="fdr")
sum(GOGOGO$padj&lt;0.2, na.rm=T)
[1] 3
sum(GOGOGO$padj&lt;0.1, na.rm=T)
[1] 2
sum(GOGOGO$padj&lt;0.05, na.rm=T)
[1] 2
GOGOGO[GOGOGO$padj&lt;0.05,]
category over_represented_pvalue under_represented_pvalue numDEInCat numInCat
1326 GO:0006032            3.838364e-08                        1          5        8
74   GO:0000272            8.726516e-07                        1          5       13
term ontology         padj
1326         chitin catabolic process       BP 0.0003557012
74   polysaccharide catabolic process       BP 0.0040434313
GOTERM[["SOMEGOTERM"]]
NULL
EnGOS&lt;-GOGOGO$category[p.adjust(GOGOGO$over_represented_pvalue, method="BH")&lt;.05]
for(go in EnGOS){
                     </p>
                    </blockquote>
                    <ul dir="auto">
                     <li>
                      print(GOTERM[[go]])
                     </li>
                     <li>
                      cat("--------------------------------------\n")}
GOID: GO:0006032
Term: chitin catabolic process
Ontology: BP
Definition: The chemical reactions and pathways resulting in the breakdown of
chitin, a linear polysaccharide consisting of beta-(1-&gt;4)-linked
N-acetyl-D-glucosamine residues.
Synonym: beta-1,4-linked N-acetylglucosamine catabolic process
Synonym: beta-1,4-linked N-acetylglucosamine catabolism
Synonym: chitin breakdown
Synonym: chitin catabolism
Synonym: chitin degradation
                     </li>
                    </ul>
                    <hr/>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      GOID: GO:0000272
Term: polysaccharide catabolic process
Ontology: BP
Definition: The chemical reactions and pathways resulting in the breakdown of a
polysaccharide, a polymer of many (typically more than 10) monosaccharide
residues linked glycosidically.
Synonym: multicellular organismal polysaccharide catabolic process
Synonym: polysaccharide breakdown
Synonym: polysaccharide catabolism
Synonym: polysaccharide degradation
Synonym: GO:0044244
Secondary: GO:0044244
                     </h2>
                     <a aria-label="Permalink: GOID: GO:0000272
Term: polysaccharide catabolic process
Ontology: BP
Definition: The chemical reactions and pathways resulting in the breakdown of a
polysaccharide, a polymer of many (typically more than 10) monosaccharide
residues linked glycosidically.
Synonym: multicellular organismal polysaccharide catabolic process
Synonym: polysaccharide breakdown
Synonym: polysaccharide catabolism
Synonym: polysaccharide degradation
Synonym: GO:0044244
Secondary: GO:0044244" class="anchor" href="#goid-go0000272term-polysaccharide-catabolic-processontology-bpdefinition-the-chemical-reactions-and-pathways-resulting-in-the-breakdown-of-apolysaccharide-a-polymer-of-many-typically-more-than-10-monosaccharideresidues-linked-glycosidicallysynonym-multicellular-organismal-polysaccharide-catabolic-processsynonym-polysaccharide-breakdownsynonym-polysaccharide-catabolismsynonym-polysaccharide-degradationsynonym-go0044244secondary-go0044244" id="user-content-goid-go0000272term-polysaccharide-catabolic-processontology-bpdefinition-the-chemical-reactions-and-pathways-resulting-in-the-breakdown-of-apolysaccharide-a-polymer-of-many-typically-more-than-10-monosaccharideresidues-linked-glycosidicallysynonym-multicellular-organismal-polysaccharide-catabolic-processsynonym-polysaccharide-breakdownsynonym-polysaccharide-catabolismsynonym-polysaccharide-degradationsynonym-go0044244secondary-go0044244">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <blockquote>
                     <p dir="auto">
                      names(Annotations)
[1] "ContigName"              "ContigLength"
                      <br/>
                      [3] "topnrMatch"              "topnrEvalue"
                      <br/>
                      [5] "topnrAlignmentLength"    "topnrPercMatch"
                      <br/>
                      [7] "nobadwordnrMatch"        "nobadwordnrEvalue"
                      <br/>
                      [9] "nobadwordnrAlignLength"  "nobadwordnrPercentMatch"
[11] "Uniprotmatch"            "X._identity"
                      <br/>
                      [13] "evalue"                  "ID"
                      <br/>
                      [15] "Description"             "KEGG"
                      <br/>
                      [17] "KEGGKO"                  "PFAM"
                      <br/>
                      [19] "GO"                      "GOCC"
                      <br/>
                      [21] "GOBP"                    "GOMF"
                      <br/>
                      [23] "Keywords"
                      <br/>
                      head(Annotations)
ContigName ContigLength
1   TR10030|c0_g1_i1_sym          523
2   TR10043|c0_g1_i1_sym          512
3   TR10050|c0_g1_i1_sym          793
4 TR10054|c0_g1_i1_coral          548
5 TR10054|c0_g2_i1_coral          811
6   TR10082|c0_g2_i1_sym         1176
topnrMatch
1                                                                           gnl|BL_ORD_ID|101380155 OLP92908.1 Sodium/calcium exchanger 1 [Symbiodinium microadriaticum]
2                                                             gnl|BL_ORD_ID|101372287 OLP84974.1 hypothetical protein AK812_SmicGene34099 [Symbiodinium microadriaticum]
3                                                                                                                                                          No_sig_nr_hit
4                                                                                     gnl|BL_ORD_ID|110384373 XP_020612383.1 titin-like isoform X4 [Orbicella faveolata]
5                                                                                     gnl|BL_ORD_ID|110384422 XP_020612382.1 titin-like isoform X3 [Orbicella faveolata]
6 gnl|BL_ORD_ID|14605113 XP_002785881.1 sugar transporter, putative [Perkinsus marinus ATCC 50983] EER17677.1 sugar transporter, putative [Perkinsus marinus ATCC 50983]
topnrEvalue topnrAlignmentLength topnrPercMatch
1   5.74325e-73                  167           0.74
2   5.42819e-23                   80           0.66
3 No_sig_nr_hit        No_sig_nr_hit  No_sig_nr_hit
4   8.93129e-73                  182           0.90
5   1.31886e-95                  250           0.82
6   1.07378e-49                  289           0.37
nobadwordnrMatch
1                                                                           gnl|BL_ORD_ID|101380155 OLP92908.1 Sodium/calcium exchanger 1 [Symbiodinium microadriaticum]
2                                                                  gnl|BL_ORD_ID|51120734 XP_012166226.1 E3 ubiquitin-protein ligase RNF146, partial [Bombus terrestris]
3                                                                                                                                                          No_sig_nr_hit
4                                                                                     gnl|BL_ORD_ID|110384373 XP_020612383.1 titin-like isoform X4 [Orbicella faveolata]
5                                                                                     gnl|BL_ORD_ID|110384422 XP_020612382.1 titin-like isoform X3 [Orbicella faveolata]
6 gnl|BL_ORD_ID|14605113 XP_002785881.1 sugar transporter, putative [Perkinsus marinus ATCC 50983] EER17677.1 sugar transporter, putative [Perkinsus marinus ATCC 50983]
nobadwordnrEvalue nobadwordnrAlignLength nobadwordnrPercentMatch
1       5.74325e-73                    167                    0.74
2        8.8118e-05                     58                    0.38
3     No_sig_nr_hit          No_sig_nr_hit           No_sig_nr_hit
4       8.93129e-73                    182                    0.90
5       1.31886e-95                    250                    0.82
6       1.07378e-49                    289                    0.37
Uniprotmatch   X.
                      <em>
                       identity    evalue               ID
1      A0A1Q9DCJ4        74.251  3.76e-73 A0A1Q9DCJ4_SYMMI
2          C0HBT3        29.114  3.65e-05      RN146_SALSA
3 No_Uniprotmatch No
                      </em>
                      %_identity No_evalue            No_ID
4      A0A2B4SVR7        77.473  8.77e-57 A0A2B4SVR7_STYPI
5      A0A2B4SVR7        72.112  5.57e-76 A0A2B4SVR7_STYPI
6          Q02334        21.402  2.12e-05      UGTP1_CAEEL
Description
1                            Sodium/calcium exchanger 1 {ECO:0000313|EMBL:OLP92908.1}
2                                                  E3 ubiquitin-protein ligase rnf146
3                                                                      No_Description
4 Golgi-associated plant pathogenesis-related protein 1 {ECO:0000313|EMBL:PFX34774.1}
5 Golgi-associated plant pathogenesis-related protein 1 {ECO:0000313|EMBL:PFX34774.1}
6                                                        UDP-galactose translocator 1
KEGG    KEGGKO                                  PFAM
1          No_KEGG No_KEGGKO PF00063; Myosin_head,  PF07647; SAM_2
2   sasa:100380617    K15700                          PF02825; WWE
3          No_KEGG No_KEGGKO                               No_PFAM
4          No_KEGG No_KEGGKO                          PF00188; CAP
5          No_KEGG No_KEGGKO                          PF00188; CAP
6 cel:CELE_ZK370.7    K15272               PF04142; Nuc_sug_transp
GO
1                                                                                     GO:0016021 // GO:0016459 // GO:0005524 // GO:0003774 // GO:0003676
2 GO:0005829 // GO:0005634 // GO:0072572 // GO:0061630 // GO:0004842 // GO:0008270 // GO:0090263 // GO:0051865 // GO:0070936 // GO:0006511 // GO:0016055
3                                                                                                                                                  No_GO
4                                                                                                                                             GO:0005576
5                                                                                                                                             GO:0005576
6                                                         GO:0060473 // GO:0005794 // GO:0005797 // GO:0030173 // GO:0015165 // GO:0015136 // GO:0008643
GOCC
1                                                                                                                                                                                                                                                   No_GOBP
2 positive regulation of canonical Wnt signaling pathway (GO:0090263);protein autoubiquitination (GO:0051865);protein K48-linked ubiquitination (GO:0070936);ubiquitin-dependent protein catabolic process (GO:0006511);Wnt signaling pathway (GO:0016055);
3                                                                                                                                                                                                                                                   No_GOCC
4                                                                                                                                                                                                                                                   No_GOBP
5                                                                                                                                                                                                                                                   No_GOBP
6                                                                                                                                                                                                                      carbohydrate transport (GO:0008643);
GOBP
1                                                                                  ATP binding (GO:0005524);motor activity (GO:0003774);nucleic acid binding (GO:0003676);
2 poly-ADP-D-ribose binding (GO:0072572);ubiquitin protein ligase activity (GO:0061630);ubiquitin-protein transferase activity (GO:0004842);zinc ion binding (GO:0008270);
3                                                                                                                                                                  No_GOBP
4                                                                                                                                                                  No_GOMF
5                                                                                                                                                                  No_GOMF
6                                 pyrimidine nucleotide-sugar transmembrane transporter activity (GO:0015165);sialic acid transmembrane transporter activity (GO:0015136);
GOMF
1                                                                         integral component of membrane (GO:0016021);myosin complex (GO:0016459);
2                                                                                                       cytosol (GO:0005829);nucleus (GO:0005634);
3                                                                                                                                          No_GOMF
4                                                                                                               extracellular region (GO:0005576);
5                                                                                                               extracellular region (GO:0005576);
6 cortical granule (GO:0060473);Golgi apparatus (GO:0005794);Golgi medial cisterna (GO:0005797);integral component of Golgi membrane (GO:0030173);
Keywords
1 ATP-binding {ECO:0000256|SAAS:SAAS00875240};Coiled coil {ECO:0000256|SAM:Coils};Membrane {ECO:0000256|SAM:Phobius};Metal-binding {ECO:0000256|PROSITE-ProRule:PRU00175};Motor protein {ECO:0000256|SAAS:SAAS01033784};Myosin {ECO:0000256|SAAS:SAAS01033784};Nucleotide-binding {ECO:0000256|SAAS:SAAS00875240};Transmembrane {ECO:0000256|SAM:Phobius};Transmembrane helix {ECO:0000256|SAM:Phobius};Zinc {ECO:0000256|PROSITE-ProRule:PRU00175};Zinc-finger {ECO:0000256|PROSITE-ProRule:PRU00175}.;
2                                                                                                                                                                                                                                                                                                                                                      Complete proteome;Cytoplasm;Metal-binding;Nucleus;Reference proteome;Transferase;Ubl conjugation pathway;Wnt signaling pathway;Zinc;Zinc-finger.;
3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            No_keywords
4                                                                                                                                                                                                                                                                                                                                                     Coiled coil {ECO:0000256|SAM:Coils};Complete proteome {ECO:0000313|Proteomes:UP000225706};Reference proteome {ECO:0000313|Proteomes:UP000225706}.;
5                                                                                                                                                                                                                                                                                                                                                     Coiled coil {ECO:0000256|SAM:Coils};Complete proteome {ECO:0000313|Proteomes:UP000225706};Reference proteome {ECO:0000313|Proteomes:UP000225706}.;
6                                                                                                                                                                                                                                                                                                                                                                                            Complete proteome;Membrane;Reference proteome;Sugar transport;Transmembrane;Transmembrane helix;Transport.;
dim(genes4heatmap)
[1] 866   6
                     </p>
                    </blockquote>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" id="react-codespace-csrf" type="hidden" value="DbRGAQ_qWAN4O2NmToPu265Y2MK1ZKBp-ry8tHKPbvYabNmuk-EgY3t3mR8Y13MV2OL1IVXN2_OLb5gfdyzw1Q">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <div class="f4 my-3 color-fg-muted text-italic">
                 No description, website, or topics provided.
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/lorenapr92/21SPStephGAdvanceGenomicslog/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Florenapr92%2F21SPStephGAdvanceGenomicslog&amp;report=lorenapr92+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/lorenapr92/21SPStephGAdvanceGenomicslog/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/lorenapr92/packages?repo_name=21SPStephGAdvanceGenomicslog">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/lorenapr92/21SPStephGAdvanceGenomicslog/used_by_list">
               </include-fragment>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       Â© 2024 GitHub,Â Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You canât perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <style>
    .user-mention[href$="/ninenia"] {
        color: var(--color-user-mention-fg);
        background-color: var(--color-user-mention-bg);
        border-radius: 2px;
        margin-left: -2px;
        margin-right: -2px;
        padding: 0 2px;
      }
   </style>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
