<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light_v2-e830ce163aae.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark_v2-bc3c552285ea.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed_v2-191796a86a54.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast_v2-e0a0b6e42616.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind_v2-4fd40c9324e8.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind_v2-2f299c61eceb.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast_v2-05d723322773.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia_v2-5679d6101565.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia_v2-0ff66d76cc55.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","repository_suggester_elastic_search","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-bf4f9425a973.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-340b2bef927d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_github_catalyst_lib_index_-eccae9-1932eeecf006.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-318896a86f35.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-9fad2423070b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         matt-chung/wolbachia_transcriptomics_metaanalysis
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="FB70:6E100:5B3F76B:811AA92:66059933" data-turbo-transient="true" name="request-id">
         <meta content="22f598a296c639e8b750dc2cf1a20323836185b0d8b152a2e56d5461ad76529a" data-turbo-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6bnVsbCwicmVxdWVzdF9pZCI6IkZCNzA6NkUxMDA6NUIzRjc2Qjo4MTFBQTkyOjY2MDU5OTMzIiwidmlzaXRvcl9pZCI6IjU1NjM2ODU5OTEwODQ4MjkyODYiLCJyZWdpb25fZWRnZSI6ImlhZCIsInJlZ2lvbl9yZW5kZXIiOiJpYWQifQ==" data-turbo-transient="true" name="visitor-payload">
           <meta content="6ea171c66dd62a7f8ba855ed8fef93bef0d011cb82a5e4730a654a9957303969" data-turbo-transient="true" name="visitor-hmac">
            <meta content="repository:221776704" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="126681664" name="octolytics-actor-id">
               <meta content="ninenia" name="octolytics-actor-login">
                <meta content="57d6abdd600494be8ac2f82b25b957e4c0d029a4d648c10497790171de0492de" name="octolytics-actor-hash">
                 <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
                  <meta content="ninenia" name="user-login"/>
                  <link href="/sessions/sudo_modal" rel="sudo-modal"/>
                  <meta content="width=device-width" name="viewport"/>
                  <meta content="Contribute to matt-chung/wolbachia_transcriptomics_metaanalysis development by creating an account on GitHub." name="description"/>
                  <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
                  <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
                  <meta content="1401488693436528" property="fb:app_id"/>
                  <meta content="app-id=1477376905, app-argument=https://github.com/matt-chung/wolbachia_transcriptomics_metaanalysis" name="apple-itunes-app">
                   <meta content="https://opengraph.githubassets.com/5d138c768e07cf75495a7bf42797022adcf9bda124e77990aee412c2641550fc/matt-chung/wolbachia_transcriptomics_metaanalysis" name="twitter:image:src">
                    <meta content="@github" name="twitter:site">
                     <meta content="summary_large_image" name="twitter:card">
                      <meta content="matt-chung/wolbachia_transcriptomics_metaanalysis" name="twitter:title">
                       <meta content="Contribute to matt-chung/wolbachia_transcriptomics_metaanalysis development by creating an account on GitHub." name="twitter:description">
                        <meta content="https://opengraph.githubassets.com/5d138c768e07cf75495a7bf42797022adcf9bda124e77990aee412c2641550fc/matt-chung/wolbachia_transcriptomics_metaanalysis" property="og:image"/>
                        <meta content="Contribute to matt-chung/wolbachia_transcriptomics_metaanalysis development by creating an account on GitHub." property="og:image:alt"/>
                        <meta content="1200" property="og:image:width"/>
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="matt-chung/wolbachia_transcriptomics_metaanalysis" property="og:title"/>
                        <meta content="https://github.com/matt-chung/wolbachia_transcriptomics_metaanalysis" property="og:url"/>
                        <meta content="Contribute to matt-chung/wolbachia_transcriptomics_metaanalysis development by creating an account on GitHub." property="og:description"/>
                        <link data-refresh-url="/_alive" data-session-id="fae53153a7a7c8d08a43b9810520b9ce4325ce8b4e99a06c5fc3f52c85cc22fe" href="wss://alive.github.com/_sockets/u/126681664/ws?session=eyJ2IjoiVjMiLCJ1IjoxMjY2ODE2NjQsInMiOjEzMDQyODIxMTYsImMiOjQyMTQ5NTAyOTQsInQiOjE3MTE2NDI5MzZ9--53e517ae19777eaee8e35ea740eec0babcec54abfeef55121a10936e280bc35f" rel="shared-web-socket"/>
                        <link href="/assets-cdn/worker/socket-worker-9cc1149b224c.js" rel="shared-web-socket-src"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="all" name="keyboard-shortcuts-preference"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="a81c045767350c9f2e551b272402db2e99346c1d2710552f41e25d405666b5eb" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="3037d78571ecdebb88047af8e128a53bf5dc96ac8c533579876941ff0be8ae32" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/matt-chung/wolbachia_transcriptomics_metaanalysis git https://github.com/matt-chung/wolbachia_transcriptomics_metaanalysis.git" name="go-import"/>
                        <meta content="29619358" name="octolytics-dimension-user_id">
                         <meta content="matt-chung" name="octolytics-dimension-user_login">
                          <meta content="221776704" name="octolytics-dimension-repository_id">
                           <meta content="matt-chung/wolbachia_transcriptomics_metaanalysis" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="221776704" name="octolytics-dimension-repository_network_root_id">
                               <meta content="matt-chung/wolbachia_transcriptomics_metaanalysis" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/matt-chung/wolbachia_transcriptomics_metaanalysis" rel="canonical"/>
                                <meta content="logged-in env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <meta content="/windows-tile.png" name="msapplication-TileImage"/>
                                 <meta content="#ffffff" name="msapplication-TileColor"/>
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-in env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-in env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="p-3 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_allex_crc32_lib_crc32_esm_js-node_modules_github_mini-throttle_dist_deco-b38cad-748e74df23ce.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_clipboard-copy-element_dist_index_esm_js-node_modules_delegated-e-b37f7d-2f24d321a3fb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_command-palette_items_help-item_ts-app_assets_modules_github_comman-48ad9d-a9ed17f2ebff.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/command-palette-f620e569db0b.js" type="application/javascript">
    </script>
    <header class="AppHeader">
     <div class="AppHeader-globalBar pb-2 js-global-bar">
      <div class="AppHeader-globalBar-start">
       <deferred-side-panel data-url="/_side-panels/global">
        <include-fragment data-target="deferred-side-panel.fragment">
         <button aria-label="Open global navigation menu" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-bg-transparent p-0 color-fg-muted" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-e2c946e6-55c2-4815-85ea-6bb74edfc30d" data-view-component="true" id="dialog-show-dialog-e2c946e6-55c2-4815-85ea-6bb74edfc30d" type="button">
          <svg aria-hidden="true" class="octicon octicon-three-bars Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1 2.75A.75.75 0 0 1 1.75 2h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 2.75Zm0 5A.75.75 0 0 1 1.75 7h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 7.75ZM1.75 12h12.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Z">
           </path>
          </svg>
         </button>
         <dialog-helper>
          <dialog aria-describedby="dialog-e2c946e6-55c2-4815-85ea-6bb74edfc30d-description" aria-labelledby="dialog-e2c946e6-55c2-4815-85ea-6bb74edfc30d-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-left SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-e2c946e6-55c2-4815-85ea-6bb74edfc30d">
           <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title sr-only" id="dialog-e2c946e6-55c2-4815-85ea-6bb74edfc30d-title">
               Global navigation
              </h1>
              <div class="d-flex" data-view-component="true">
               <div class="AppHeader-logo position-relative" data-view-component="true">
                <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
                 <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
                 </path>
                </svg>
               </div>
              </div>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-e2c946e6-55c2-4815-85ea-6bb74edfc30d" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="dialog-e2c946e6-55c2-4815-85ea-6bb74edfc30d-title">
            <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
             <div class="d-flex flex-column mb-3" data-view-component="true">
              <nav aria-label="Site navigation" class="ActionList" data-view-component="true">
               <nav-list>
                <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"HOME","label":null}' data-hotkey="g d" data-view-component="true" href="/dashboard" id="item-4c583390-719c-4c4b-a236-95542f8d82f6">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-home" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M6.906.664a1.749 1.749 0 0 1 2.187 0l5.25 4.2c.415.332.657.835.657 1.367v7.019A1.75 1.75 0 0 1 13.25 15h-3.5a.75.75 0 0 1-.75-.75V9H7v5.25a.75.75 0 0 1-.75.75h-3.5A1.75 1.75 0 0 1 1 13.25V6.23c0-.531.242-1.034.657-1.366l5.25-4.2Zm1.25 1.171a.25.25 0 0 0-.312 0l-5.25 4.2a.25.25 0 0 0-.094.196v7.019c0 .138.112.25.25.25H5.5V8.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 .75.75v5.25h2.75a.25.25 0 0 0 .25-.25V6.23a.25.25 0 0 0-.094-.195Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Home
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"ISSUES","label":null}' data-hotkey="g i" data-view-component="true" href="/issues" id="item-48617676-90cb-480f-9567-21e6dfc018e3">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS","label":null}' data-hotkey="g p" data-view-component="true" href="/pulls" id="item-f18b9186-0990-41c7-a679-2d37b22d9aa0">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="projects" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROJECTS","label":null}' data-view-component="true" href="/projects" id="item-15bfc68f-6d27-4a10-95a6-f65c89f095ef">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"DISCUSSIONS","label":null}' data-view-component="true" href="/discussions" id="item-02157da5-9446-4ddd-bb70-0bb461ce6c8f">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Discussions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"CODESPACES","label":null}' data-view-component="true" href="https://github.com/codespaces" id="item-9a0dba3c-8d17-4d92-9a76-e8c869a48581">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Codespaces
                   </span>
                  </a>
                 </li>
                 <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"EXPLORE","label":null}' data-view-component="true" href="/explore" id="item-7282c345-2261-4037-bded-6ddc057d2249">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-telescope" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.184 1.143v-.001l1.422 2.464a1.75 1.75 0 0 1-.757 2.451L3.104 11.713a1.75 1.75 0 0 1-2.275-.702l-.447-.775a1.75 1.75 0 0 1 .53-2.32L11.682.573a1.748 1.748 0 0 1 2.502.57Zm-4.709 9.32h-.001l2.644 3.863a.75.75 0 1 1-1.238.848l-1.881-2.75v2.826a.75.75 0 0 1-1.5 0v-2.826l-1.881 2.75a.75.75 0 1 1-1.238-.848l2.049-2.992a.746.746 0 0 1 .293-.253l1.809-.87a.749.749 0 0 1 .944.252ZM9.436 3.92h-.001l-4.97 3.39.942 1.63 5.42-2.61Zm3.091-2.108h.001l-1.85 1.26 1.505 2.605 2.016-.97a.247.247 0 0 0 .13-.151.247.247 0 0 0-.022-.199l-1.422-2.464a.253.253 0 0 0-.161-.119.254.254 0 0 0-.197.038ZM1.756 9.157a.25.25 0 0 0-.075.33l.447.775a.25.25 0 0 0 .325.1l1.598-.769-.83-1.436-1.465 1Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Explore
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"MARKETPLACE","label":null}' data-view-component="true" href="/marketplace" id="item-870b225d-08a3-4e2a-8f9e-b9fff3739dc2">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Marketplace
                   </span>
                  </a>
                 </li>
                </ul>
               </nav-list>
              </nav>
              <div class="my-3 d-flex flex-justify-center height-full" data-view-component="true">
               <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                </circle>
                <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                </path>
               </svg>
              </div>
             </div>
             <div class="flex-1" data-view-component="true">
             </div>
             <div class="px-2" data-view-component="true">
              <p class="color-fg-subtle text-small text-light">
               © 2024 GitHub, Inc.
              </p>
              <div class="d-flex flex-wrap text-small text-light" data-view-component="true">
               <a class="Link mr-2" data-view-component="true" href="https://github.com/about" target="_blank">
                About
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.blog" target="_blank">
                Blog
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service" target="_blank">
                Terms
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement" target="_blank">
                Privacy
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.com/security" target="_blank">
                Security
               </a>
               <a class="Link mr-3" data-view-component="true" href="https://www.githubstatus.com/" target="_blank">
                Status
               </a>
              </div>
             </div>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </include-fragment>
       </deferred-side-panel>
       <a aria-label="Homepage " class="AppHeader-logo ml-2" data-analytics-event='{"category":"Header","action":"go to dashboard","label":"icon:logo"}' data-hotkey="g d" data-turbo="false" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github v-align-middle color-fg-default" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="AppHeader-context">
        <div class="AppHeader-context-compact">
         <button aria-expanded="false" aria-haspopup="dialog" aria-label="Page context: matt-chung / wolbachia_transcriptomics_metaanalysis" class="AppHeader-context-compact-trigger Truncate Button--secondary Button--medium Button box-shadow-none" data-show-dialog-id="context-region-dialog" data-view-component="true" id="dialog-show-context-region-dialog" type="button">
          <span class="Button-content">
           <span class="Button-label">
            <span class="AppHeader-context-compact-lead">
             <span class="AppHeader-context-compact-parentItem">
              matt-chung
             </span>
             <span class="no-wrap">
              /
             </span>
            </span>
            <strong class="AppHeader-context-compact-mainItem d-flex flex-items-center Truncate">
             <span class="Truncate-text">
              wolbachia_transcriptomics_metaanalysis
             </span>
            </strong>
           </span>
          </span>
         </button>
         <dialog-helper>
          <dialog aria-describedby="context-region-dialog-description" aria-labelledby="context-region-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-view-component="true" id="context-region-dialog">
           <div class="Overlay-header" data-view-component="true">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title" id="context-region-dialog-title">
               Navigate back to
              </h1>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="context-region-dialog" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="context-region-dialog-title">
            <div class="Overlay-body" data-view-component="true">
             <ul class="list-style-none" role="list">
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"matt-chung","screen_size":"compact"}' data-view-component="true" href="/matt-chung">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-person mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                  </path>
                 </svg>
                 matt-chung
                </span>
               </a>
              </li>
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"wolbachia_transcriptomics_metaanalysis","screen_size":"compact"}' data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-repo mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                  </path>
                 </svg>
                 wolbachia_transcriptomics_metaanalysis
                </span>
               </a>
              </li>
             </ul>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </div>
        <div class="AppHeader-context-full">
         <nav aria-label="Page context" role="navigation">
          <ul class="list-style-none" role="list">
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"matt-chung","screen_size":"full"}' data-hovercard-type="user" data-hovercard-url="/users/matt-chung/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" data-view-component="true" href="/matt-chung">
             <span class="AppHeader-context-item-label">
              matt-chung
             </span>
            </a>
            <span class="AppHeader-context-item-separator">
             /
            </span>
           </li>
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"wolbachia_transcriptomics_metaanalysis","screen_size":"full"}' data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis">
             <span class="AppHeader-context-item-label">
              wolbachia_transcriptomics_metaanalysis
             </span>
            </a>
           </li>
          </ul>
         </nav>
        </div>
       </div>
      </div>
      <div class="AppHeader-globalBar-end">
       <div class="AppHeader-search">
        <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;input type="hidden" value="_H6RNL903WFlZMv-EEq1Dwk4Z_6bWm9u2Ucsg__Xa-SsXZ28LfkCrtL-k9C13VdUPbSnFa9x2v8HPFI9UhPWDA" data-csrf="true" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="matt-chung" data-current-repository="matt-chung/wolbachia_transcriptomics_metaanalysis" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="6zB5MCP2HCTbtJbzoVSdOOMxmdo7JiPQMCYZNFA1w_2j-m4aBbBVUYwY5CpHUW0psZXbcx1B079cxFiftwvYsQ" data-header-redesign-enabled="true" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="true" data-max-custom-scopes="10" data-scope="repo:matt-chung/wolbachia_transcriptomics_metaanalysis">
         <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center height-auto color-bg-transparent border-0 color-fg-subtle mx-0" data-action="click:qbsearch-input#searchInputContainerClicked">
          <button aria-expanded="false" aria-haspopup="dialog" aria-label="Search or jump to…" class="AppHeader-button AppHeader-search-whenNarrow" data-action="click:qbsearch-input#handleExpand" type="button">
           <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
            </path>
           </svg>
          </button>
          <div class="AppHeader-search-whenRegular">
           <div class="AppHeader-search-wrap AppHeader-search-wrap--hasTrailing">
            <div class="AppHeader-search-control">
             <label aria-label="Search or jump to…" class="AppHeader-search-visual--leading" for="AppHeader-searchInput">
              <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
               </path>
              </svg>
             </label>
             <button aria-describedby="search-error-message-flash" class="AppHeader-searchButton form-control input-contrast text-left color-fg-subtle no-wrap" data-action="click:qbsearch-input#handleExpand" data-analytics-event='{"category":"Global navigation","action":"SEARCH","label":null}' data-hotkey="s,/" data-target="qbsearch-input.inputButton" type="button">
              <div class="overflow-hidden">
               <span data-target="qbsearch-input.inputButtonText" id="qb-input-query">
                Type
                <kbd class="AppHeader-search-kbd">
                 /
                </kbd>
                to search
               </span>
              </div>
             </button>
            </div>
            <button class="AppHeader-search-action--trailing js-activate-command-palette" data-analytics-event='{"category":"SiteHeaderComponent","action":"command_palette","label":"open command palette"}' id="AppHeader-commandPalette-button" type="button">
             <svg aria-hidden="true" class="octicon octicon-command-palette" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m6.354 8.04-4.773 4.773a.75.75 0 1 0 1.061 1.06L7.945 8.57a.75.75 0 0 0 0-1.06L2.642 2.206a.75.75 0 0 0-1.06 1.061L6.353 8.04ZM8.75 11.5a.75.75 0 0 0 0 1.5h5.5a.75.75 0 0 0 0-1.5h-5.5Z">
              </path>
             </svg>
            </button>
            <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="AppHeader-commandPalette-button" id="tooltip-b08ae270-71be-4584-baf1-abf48d584fa8" popover="manual">
             Command palette
            </tool-tip>
           </div>
          </div>
          <input class="js-site-search-type-field" name="type" type="hidden"/>
          <div class="Overlay--hidden" data-modal-dialog-overlay="">
           <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-medium Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
            <h1 class="sr-only" id="search-suggestions-dialog-header">
             Search code, repositories, users, issues, pull requests...
            </h1>
            <div class="Overlay-body Overlay-body--paddingNone">
             <div data-view-component="true">
              <div class="search-suggestions position-absolute width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                 <div class="FormControl FormControl--fullWidth">
                  <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                   Search
                  </label>
                  <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                   <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                    <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </span>
                   <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                    <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                    </div>
                    <div class="QueryBuilder-InputWrapper">
                     <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                     </div>
                     <input aria-describedby="validation-5cada75d-e585-42ee-baee-5a208f07c300" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                     </input>
                    </div>
                   </div>
                   <span class="sr-only" id="query-builder-test-clear">
                    Clear
                   </span>
                   <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                    <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <template id="search-icon">
                   <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-icon">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="file-code-icon">
                   <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                    </path>
                   </svg>
                  </template>
                  <template id="history-icon">
                   <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                    </path>
                   </svg>
                  </template>
                  <template id="repo-icon">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="bookmark-icon">
                   <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="plus-circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                    </path>
                   </svg>
                  </template>
                  <template id="trash-icon">
                   <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="team-icon">
                   <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="project-icon">
                   <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                    </path>
                   </svg>
                  </template>
                  <template id="pencil-icon">
                   <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                    </path>
                   </svg>
                  </template>
                  <template id="copilot-icon">
                   <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                    </path>
                    <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="workflow-icon">
                   <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="book-icon">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-review-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="codespaces-icon">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-discussion-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="organization-icon">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="rocket-icon">
                   <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="shield-check-icon">
                   <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="heart-icon">
                   <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="server-icon">
                   <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="globe-icon">
                   <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                    </path>
                   </svg>
                  </template>
                  <template id="issue-opened-icon">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-mobile-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="package-icon">
                   <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                    </path>
                   </svg>
                  </template>
                  <template id="credit-card-icon">
                   <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                    </path>
                    <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                    </path>
                   </svg>
                  </template>
                  <template id="play-icon">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </template>
                  <template id="gift-icon">
                   <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-square-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-desktop-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                    </path>
                   </svg>
                  </template>
                  <div class="position-relative">
                   <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                   </ul>
                  </div>
                  <div class="FormControl-inlineValidation" hidden="hidden" id="validation-5cada75d-e585-42ee-baee-5a208f07c300">
                   <span class="FormControl-inlineValidation--visual">
                    <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                     <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                     </path>
                    </svg>
                   </span>
                   <span>
                   </span>
                  </div>
                 </div>
                 <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                 </div>
                </query-builder>
               </form>
               <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                 Search syntax tips
                </a>
                <div class="d-flex flex-1">
                </div>
                <button class="Button--link Button--medium Button color-fg-accent text-normal ml-2" data-action="click:qbsearch-input#showFeedbackDialog" data-view-component="true" type="button">
                 <span class="Button-content">
                  <span class="Button-label">
                   Give feedback
                  </span>
                 </span>
                </button>
               </div>
              </div>
             </div>
            </div>
           </modal-dialog>
          </div>
         </div>
         <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
         </div>
         <div class="color-fg-default">
          <dialog-helper>
           <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
            <div class="Overlay-header" data-view-component="true">
             <div class="Overlay-headerContentWrap">
              <div class="Overlay-titleWrap">
               <h1 class="Overlay-title" id="feedback-dialog-title">
                Provide feedback
               </h1>
              </div>
              <div class="Overlay-actionWrap">
               <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
              </div>
             </div>
            </div>
            <scrollable-region data-labelled-by="feedback-dialog-title">
             <div class="Overlay-body" data-view-component="true">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
               <input name="authenticity_token" type="hidden" value="jJMytIMXovF8yVBet8LR-NbQobojpBI6OmfrlL_Ar4MeISuMS8GRa8zrXlqJ29AJC9tseXmE9G805_4uSZGqOA"/>
               <p>
                We read every piece of feedback, and take your input very seriously.
               </p>
               <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
               <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
               <label for="include_email" style="font-weight: normal">
                Include my email address so I can be contacted
               </label>
              </form>
             </div>
            </scrollable-region>
            <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
             <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
              Cancel
             </button>
             <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
              Submit feedback
             </button>
            </div>
           </dialog>
          </dialog-helper>
          <custom-scopes data-target="qbsearch-input.customScopesManager">
           <dialog-helper>
            <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
             <div class="Overlay-header Overlay-header--divided" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                 Saved searches
                </h1>
                <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                 Use saved searches to filter your results more quickly
                </h2>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="custom-scopes-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <div data-target="custom-scopes.customScopesModalDialogFlash">
               </div>
               <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                 <input name="authenticity_token" type="hidden" value="6T5GGM_q2lqpIbHevWv_4n_7IDteT4rwJcnHsWWfG_-lzCSil2uEg-JIDHP3vjLftqtELAhKyCuGkRYJethCqg">
                  <div data-target="custom-scopes.customScopesModalDialogFlash">
                  </div>
                  <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                  <div class="form-group">
                   <label for="custom_scope_name">
                    Name
                   </label>
                   <auto-check required="" src="/search/custom_scopes/check_name">
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                    <input data-csrf="true" type="hidden" value="dLfqFLpXfLVpq72PC4kVdA0liRUlk7FGPqpgqCkrgsds6qey7dXVtuq_9AxL20seu-KMkgtKUd5TCsJ6EURRDw">
                    </input>
                   </auto-check>
                  </div>
                  <div class="form-group">
                   <label for="custom_scope_query">
                    Query
                   </label>
                   <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                  </div>
                  <p class="text-small color-fg-muted">
                   To see all available qualifiers, see our
                   <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                    documentation
                   </a>
                   .
                  </p>
                 </input>
                </form>
               </div>
               <div data-target="custom-scopes.manageCustomScopesForm">
                <div data-target="custom-scopes.list">
                </div>
               </div>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
              <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
               Create saved search
              </button>
             </div>
            </dialog>
           </dialog-helper>
          </custom-scopes>
         </div>
        </qbsearch-input>
        <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="Wk3iWfkY8vl6B-EWFfRWoR_JkVjJsxWZAGrkc3RDaTSdGI174B80m53ce-vcTy2yz3FJkkwV06kPCbrW7NcIPg">
        </input>
       </div>
       <div class="AppHeader-actions position-relative">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="global-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="global-create-menu-button" popovertarget="global-create-menu-overlay" type="button">
           <span class="Button-content">
            <span class="Button-visual Button-leadingVisual">
             <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
              </path>
             </svg>
            </span>
            <span class="Button-label">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </span>
           </span>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="global-create-menu-button" id="tooltip-fdea6d5b-bfd5-421f-b252-9369f225c6ad" popover="manual">
           Create new...
          </tool-tip>
          <anchored-position align="end" anchor="global-create-menu-button" anchor-offset="normal" data-view-component="true" id="global-create-menu-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="global-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="global-create-menu-list" role="menu">
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-5c828fe4-585f-4d95-aec5-5093c40bcb1d" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New repository
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-ac7c68f9-768e-4782-aa49-d1123fb11f08" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Import repository
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-63c8dc73-93c7-466a-ad64-3fc7131aa110" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New codespace
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-65342781-1766-4ef8-894e-2d5e8298a02a" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New gist
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-b74dfd6e-6df6-4d7c-b0c0-235f6a020ec1" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New organization
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
        <a aria-labelledby="tooltip-e2a4a74d-f6f7-4573-a402-5ae4fa08ab41" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"ISSUES_HEADER","label":null}' data-view-component="true" href="/issues" id="icon-button-3e5600f2-557e-4d3c-ba22-9710b8d14406">
         <svg aria-hidden="true" class="octicon octicon-issue-opened Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-3e5600f2-557e-4d3c-ba22-9710b8d14406" id="tooltip-e2a4a74d-f6f7-4573-a402-5ae4fa08ab41" popover="manual">
         Issues
        </tool-tip>
        <a aria-labelledby="tooltip-98090864-7d48-4c11-93a0-087b0722dc0b" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS_HEADER","label":null}' data-view-component="true" href="/pulls" id="icon-button-6e1db7fa-90b0-424d-b880-3cc1625833f8">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-6e1db7fa-90b0-424d-b880-3cc1625833f8" id="tooltip-98090864-7d48-4c11-93a0-087b0722dc0b" popover="manual">
         Pull requests
        </tool-tip>
       </div>
       <notification-indicator class="js-socket-channel" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY0MjkzNn0=--0b43b2322d9d155a3ddd1ddd51f4717b3184b053672d34b7b2814df0a0e0c8a3" data-fetch-indicator-enabled="true" data-fetch-indicator-src="/notifications/indicator" data-header-redesign-enabled="true" data-indicator-mode="none" data-tooltip-global="You have unread notifications" data-tooltip-none="You have no unread notifications" data-tooltip-unavailable="Notifications are unavailable at the moment." data-view-component="true">
        <a aria-label="Notifications" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"NOTIFICATIONS_HEADER","label":null}' data-hotkey="g n" data-target="notification-indicator.link" data-view-component="true" href="/notifications" id="AppHeader-notifications-button">
         <svg aria-hidden="true" class="octicon octicon-inbox Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.8 2.06A1.75 1.75 0 0 1 4.41 1h7.18c.7 0 1.333.417 1.61 1.06l2.74 6.395c.04.093.06.194.06.295v4.5A1.75 1.75 0 0 1 14.25 15H1.75A1.75 1.75 0 0 1 0 13.25v-4.5c0-.101.02-.202.06-.295Zm1.61.44a.25.25 0 0 0-.23.152L1.887 8H4.75a.75.75 0 0 1 .6.3L6.625 10h2.75l1.275-1.7a.75.75 0 0 1 .6-.3h2.863L11.82 2.652a.25.25 0 0 0-.23-.152Zm10.09 7h-2.875l-1.275 1.7a.75.75 0 0 1-.6.3h-3.5a.75.75 0 0 1-.6-.3L4.375 9.5H1.5v3.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-target="notification-indicator.tooltip" data-type="description" data-view-component="true" for="AppHeader-notifications-button" id="tooltip-44713e84-adfa-4b66-8c21-1460e5729150" popover="manual">
         Notifications
        </tool-tip>
       </notification-indicator>
       <div class="AppHeader-user">
        <deferred-side-panel data-url="/_side-panels/user?react_global_nav=false&amp;repository_id=221776704">
         <include-fragment data-target="deferred-side-panel.fragment">
          <user-drawer-side-panel>
           <button aria-label="Open user account menu" class="AppHeader-logo Button--invisible Button--medium Button Button--invisible-noVisuals color-bg-transparent p-0" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-a019b52d-cd58-4725-8844-65d5fd546a20" data-view-component="true" id="dialog-show-dialog-a019b52d-cd58-4725-8844-65d5fd546a20" type="button">
            <span class="Button-content">
             <span class="Button-label">
              <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
             </span>
            </span>
           </button>
           <dialog-helper>
            <dialog aria-describedby="dialog-a019b52d-cd58-4725-8844-65d5fd546a20-description" aria-labelledby="dialog-a019b52d-cd58-4725-8844-65d5fd546a20-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-right SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-a019b52d-cd58-4725-8844-65d5fd546a20">
             <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title sr-only" id="dialog-a019b52d-cd58-4725-8844-65d5fd546a20-title">
                 Account menu
                </h1>
                <div class="d-flex" data-view-component="true">
                 <div class="AppHeader-logo position-relative" data-view-component="true">
                  <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
                 </div>
                 <div class="overflow-hidden d-flex width-full" data-view-component="true">
                  <div class="lh-condensed overflow-hidden d-flex flex-column flex-justify-center ml-2 f5 mr-auto width-full" data-view-component="true">
                   <span class="Truncate text-bold" data-view-component="true">
                    <span class="Truncate-text" data-view-component="true">
                     ninenia
                    </span>
                   </span>
                  </div>
                  <action-menu class="d-sm-none d-md-none d-lg-none" data-select-variant="none" data-view-component="true">
                   <focus-group direction="vertical" mnemonics="" retain="">
                    <button aria-controls="user-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="user-create-menu-button" popovertarget="user-create-menu-overlay" type="button">
                     <span class="Button-content">
                      <span class="Button-visual Button-leadingVisual">
                       <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                        </path>
                       </svg>
                      </span>
                      <span class="Button-label">
                       <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                        </path>
                       </svg>
                      </span>
                     </span>
                    </button>
                    <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="user-create-menu-button" id="tooltip-7dc1a2aa-424f-425c-8758-610d68cddece" popover="manual">
                     Create new...
                    </tool-tip>
                    <anchored-position align="end" anchor="user-create-menu-button" anchor-offset="normal" data-view-component="true" id="user-create-menu-overlay" popover="auto" side="outside-bottom">
                     <div class="Overlay Overlay--size-auto" data-view-component="true">
                      <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
                       <action-list>
                        <div data-view-component="true">
                         <ul aria-labelledby="user-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="user-create-menu-list" role="menu">
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-544dfb56-5f02-48e9-9cd7-d957aee2c372" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New repository
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-3d03e6d1-559b-4f1f-a128-39c54013e25f" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             Import repository
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-c6f0bfc0-817e-49c1-9eae-5e53aafa3aba" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                              </path>
                              <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New codespace
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-e60f9259-feb4-430f-8ca8-9217921088a2" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New gist
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-b833dd75-a814-4b43-8383-70229cf47262" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New organization
                            </span>
                           </a>
                          </li>
                         </ul>
                        </div>
                       </action-list>
                      </div>
                     </div>
                    </anchored-position>
                   </focus-group>
                  </action-menu>
                 </div>
                </div>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-a019b52d-cd58-4725-8844-65d5fd546a20" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="dialog-a019b52d-cd58-4725-8844-65d5fd546a20-title">
              <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
               <div class="d-flex flex-column mb-3" data-view-component="true">
                <nav aria-label="User navigation" class="ActionList" data-view-component="true">
                 <nav-list>
                  <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-6133fa35-4646-437b-a214-d63173ca13b2" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROFILE","label":null}' data-view-component="true" href="https://github.com/ninenia" id="item-c2e4bf29-b446-49cf-bf30-1d4de616b806">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-person" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your profile
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-066bbea4-2bbb-40fa-a2d3-6dbe51a11287" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_REPOSITORIES","label":null}' data-view-component="true" href="/ninenia?tab=repositories" id="item-01eb2ca7-163e-4457-ab8e-b7b8f9317af6">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your repositories
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_PROJECTS","label":null}' data-view-component="true" href="/ninenia?tab=projects" id="item-9bb66a1d-ff4c-4ddc-9371-cccf665444aa">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your projects
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-335e97e4-6b8e-44fe-8355-2fd40f9c49e2" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-04afe640-a9e6-4873-8cd3-6640b122d6c2" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_STARS","label":null}' data-view-component="true" href="/ninenia?tab=stars" id="item-a9399008-9d88-4484-a07c-053baed678bf">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-star" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your stars
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SPONSORS","label":null}' data-view-component="true" href="/sponsors/accounts" id="item-59233814-4d83-4dec-a3d6-05524db4d947">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your sponsors
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_GISTS","label":null}' data-view-component="true" href="https://gist.github.com/mine" id="item-dedebbee-6760-4ca9-9a8a-4ede7781321d">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your gists
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-3d4f7775-dcc4-44e8-94ac-aa1bab3c7589" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-6df6b9dd-82b7-4bdf-873c-7a30007c85c7" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SETTINGS","label":null}' data-view-component="true" href="/settings/profile" id="item-20badb09-6424-48c7-9891-d2aaaf3af1f9">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-gear" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Settings
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SUPPORT","label":null}' data-view-component="true" href="https://support.github.com" id="item-58aea033-3185-442e-a71c-f2f842d27a07">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Support
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"COMMUNITY","label":null}' data-view-component="true" href="https://community.github.com" id="item-586eedae-f569-4af2-a55b-2d1cd85fa765">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Community
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent" data-analytics-event='{"category":"Global navigation","action":"LOGOUT","label":null}' data-view-component="true" href="/logout" id="item-15763960-124f-4ab7-8908-a9dceb7395e0">
                     <span class="ActionListItem-label" data-view-component="true">
                      Sign out
                     </span>
                    </a>
                   </li>
                  </ul>
                 </nav-list>
                </nav>
               </div>
              </div>
             </scrollable-region>
            </dialog>
           </dialog-helper>
          </user-drawer-side-panel>
         </include-fragment>
        </deferred-side-panel>
       </div>
       <div class="position-absolute mt-2">
        <site-header-logged-in-user-menu>
        </site-header-logged-in-user-menu>
       </div>
      </div>
     </div>
     <div class="AppHeader-localBar">
      <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav" data-pjax="#js-repo-pjax-container" data-view-component="true">
       <ul class="UnderlineNav-body list-style-none" data-view-component="true">
        <li class="d-inline-flex" data-view-component="true">
         <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /matt-chung/wolbachia_transcriptomics_metaanalysis" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis" id="code-tab">
          <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
           </path>
          </svg>
          <span data-content="Code">
           Code
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /matt-chung/wolbachia_transcriptomics_metaanalysis/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/issues" id="issues-tab">
          <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
           </path>
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
           </path>
          </svg>
          <span data-content="Issues">
           Issues
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /matt-chung/wolbachia_transcriptomics_metaanalysis/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/pulls" id="pull-requests-tab">
          <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
           </path>
          </svg>
          <span data-content="Pull requests">
           Pull requests
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /matt-chung/wolbachia_transcriptomics_metaanalysis/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/actions" id="actions-tab">
          <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
           </path>
          </svg>
          <span data-content="Actions">
           Actions
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /matt-chung/wolbachia_transcriptomics_metaanalysis/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/projects" id="projects-tab">
          <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
           </path>
          </svg>
          <span data-content="Projects">
           Projects
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /matt-chung/wolbachia_transcriptomics_metaanalysis/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/security" id="security-tab">
          <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
           </path>
          </svg>
          <span data-content="Security">
           Security
          </span>
          <include-fragment accept="text/fragment+html" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/security/overall-count">
          </include-fragment>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /matt-chung/wolbachia_transcriptomics_metaanalysis/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/pulse" id="insights-tab">
          <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
           </path>
          </svg>
          <span data-content="Insights">
           Insights
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
       </ul>
       <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="action-menu-227ede85-41cd-465f-9621-f6717a6a4766-list" aria-haspopup="true" aria-labelledby="tooltip-a5be1977-791f-4717-b543-5d295b2d3cde" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-227ede85-41cd-465f-9621-f6717a6a4766-button" popovertarget="action-menu-227ede85-41cd-465f-9621-f6717a6a4766-overlay" type="button">
           <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
           </svg>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-227ede85-41cd-465f-9621-f6717a6a4766-button" id="tooltip-a5be1977-791f-4717-b543-5d295b2d3cde" popover="manual">
           Additional navigation options
          </tool-tip>
          <anchored-position align="start" anchor="action-menu-227ede85-41cd-465f-9621-f6717a6a4766-button" anchor-offset="normal" data-view-component="true" id="action-menu-227ede85-41cd-465f-9621-f6717a6a4766-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="action-menu-227ede85-41cd-465f-9621-f6717a6a4766-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-227ede85-41cd-465f-9621-f6717a6a4766-list" role="menu">
                <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis" id="item-8b27611a-32bb-4d3e-ac39-40423db5df58" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Code
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/issues" id="item-6a87746e-b340-45d9-8261-f14bc26b5696" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Issues
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/pulls" id="item-e5425914-1503-4a95-b51f-688846202063" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Pull requests
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/actions" id="item-4ebd9dfd-df5a-47fd-998b-01f99cc91328" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Actions
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/projects" id="item-02170459-2292-4fbf-a2b7-3ad82f89f439" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Projects
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/security" id="item-197f0625-e973-4443-bce4-f8e3f571387e" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Security
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/pulse" id="item-623ba75a-56b6-49f7-a695-bf49fcca3f6e" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Insights
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
       </div>
      </nav>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-2da05b02-4654-41bc-a0b3-d117816aaa4d" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-56a101cc-57d3-4ec1-b086-3cb87e90a465" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-56a101cc-57d3-4ec1-b086-3cb87e90a465" id="tooltip-2da05b02-4654-41bc-a0b3-d117816aaa4d" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <notification-shelf-watcher class="js-socket-channel" data-base-url="https://github.com/notifications/beta/shelf" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY0MjkzNn0=--0b43b2322d9d155a3ddd1ddd51f4717b3184b053672d34b7b2814df0a0e0c8a3" data-view-component="true">
   </notification-shelf-watcher>
   <div data-initial="" data-target="notification-shelf-watcher.placeholder" hidden="">
   </div>
   <details class="details-reset details-overlay details-overlay-dark js-command-palette-dialog" data-turbo-replace="" id="command-palette-pjax-container">
    <summary aria-label="Command palette trigger" tabindex="-1">
    </summary>
    <details-dialog aria-label="Command palette" class="command-palette-details-dialog d-flex flex-column flex-justify-center height-fit">
     <command-palette activation-hotkey="Mod+k,Mod+Alt+k" class="command-palette color-bg-default rounded-3 border color-shadow-small" command-mode-hotkey="Mod+Shift+K" data-action="
        command-palette-input-ready:command-palette#inputReady
        command-palette-page-stack-updated:command-palette#updateInputScope
        itemsUpdated:command-palette#itemsUpdated
        keydown:command-palette#onKeydown
        loadingStateChanged:command-palette#loadingStateChanged
        selectedItemChanged:command-palette#selectedItemChanged
        pageFetchError:command-palette#pageFetchError
      " return-to="/matt-chung/wolbachia_transcriptomics_metaanalysis" user-id="126681664">
      <command-palette-mode data-char="#" data-placeholder="Search issues and pull requests" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="#" data-placeholder="Search issues, pull requests, discussions, and projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="!" data-placeholder="Search projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a user, organization, or repository" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a repository" data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode data-char="/" data-placeholder="Search files" data-scope-types='["repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="?">
      </command-palette-mode>
      <command-palette-mode data-char="&gt;" data-placeholder="Run a command">
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode class="js-command-palette-default-mode" data-char="" data-placeholder="Search or jump to...">
      </command-palette-mode>
      <command-palette-input data-action="
          command-palette-input:command-palette#onInput
          command-palette-select:command-palette#onSelect
          command-palette-descope:command-palette#onDescope
          command-palette-cleared:command-palette#onInputClear
        " placeholder="Search or jump to...">
       <div class="js-search-icon d-flex flex-items-center mr-2" style="height: 26px">
        <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
         </path>
        </svg>
       </div>
       <div class="js-spinner d-flex flex-items-center mr-2 color-fg-muted" hidden="">
        <svg aria-label="Loading" class="anim-rotate" fill="none" height="16" viewbox="0 0 16 16" width="16">
         <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
         </circle>
         <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
         </path>
        </svg>
       </div>
       <command-palette-scope>
        <div class="color-fg-subtle" data-target="command-palette-scope.placeholder" hidden="">
         /
         <span class="text-semibold color-fg-default">
          ...
         </span>
         /
        </div>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDQ6VXNlcjI5NjE5MzU4" data-targets="command-palette-scope.tokens" data-text="matt-chung" data-type="owner" data-value="matt-chung" style="white-space:nowrap;line-height:20px;">
         matt-chung
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDEwOlJlcG9zaXRvcnkyMjE3NzY3MDQ=" data-targets="command-palette-scope.tokens" data-text="wolbachia_transcriptomics_metaanalysis" data-type="repository" data-value="wolbachia_transcriptomics_metaanalysis" style="white-space:nowrap;line-height:20px;">
         wolbachia_transcr...
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
       </command-palette-scope>
       <div class="command-palette-input-group flex-1 form-control border-0 box-shadow-none" style="z-index: 0">
        <div class="command-palette-typeahead position-absolute d-flex flex-items-center Truncate">
         <span class="typeahead-segment input-mirror" data-target="command-palette-input.mirror">
         </span>
         <span class="Truncate-text" data-target="command-palette-input.typeaheadText">
         </span>
         <span class="typeahead-segment" data-target="command-palette-input.typeaheadPlaceholder">
         </span>
        </div>
        <input aria-label="Hidden input for typeahead" class="js-overlay-input typeahead-input d-none" disabled="" tabindex="-1"/>
        <input aria-autocomplete="list" aria-controls="command-palette-page-stack" aria-expanded="false" aria-haspopup="listbox" aria-label="Command palette input" autocapitalize="off" autocomplete="off" autocorrect="off" class="js-input typeahead-input form-control border-0 box-shadow-none input-block width-full no-focus-indicator" data-action="
              input:command-palette-input#onInput
              keydown:command-palette-input#onKeydown
            " role="combobox" spellcheck="false" type="text"/>
       </div>
       <div class="position-relative d-inline-block" data-view-component="true">
        <button aria-keyshortcuts="Control+Backspace" class="btn-octicon command-palette-input-clear-button" data-action="click:command-palette-input#onClear keypress:command-palette-input#onClear" data-target="command-palette-input.clearButton" data-view-component="true" hidden="hidden" id="command-palette-clear-button" type="button">
         <svg aria-hidden="true" class="octicon octicon-x-circle-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
          </path>
         </svg>
        </button>
        <tool-tip class="sr-only position-absolute" data-direction="w" data-type="label" data-view-component="true" for="command-palette-clear-button" id="tooltip-dc888546-b1fc-4979-9a93-86e5d96948ca" popover="manual">
         Clear Command Palette
        </tool-tip>
       </div>
      </command-palette-input>
      <command-palette-page-stack data-action="command-palette-page-octicons-cached:command-palette-page-stack#cacheOcticons" data-default-scope-id="MDEwOlJlcG9zaXRvcnkyMjE3NzY3MDQ=" data-default-scope-type="Repository">
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search discussions
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           !
          </kbd>
          to search projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search teams
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='[""]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search people and organizations
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           &gt;
          </kbd>
          to activate command mode
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Go to your accessibility settings to change your keyboard shortcuts
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type author:@me to search your content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:pr to filter to pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:issue to filter to issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:project to filter to projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:open to filter to open content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="mx-3 my-2 flash flash-error d-flex flex-items-center" data-on-error="" data-scope-types="*">
        <div>
         <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
          </path>
         </svg>
        </div>
        <div class="px-2">
         We’ve encountered an error and some results aren't available at this time. Type a new search or try again later.
        </div>
       </command-palette-tip>
       <command-palette-tip class="h4 color-fg-default pl-3 pb-2 pt-3" data-match-mode="[^?]|^$" data-on-empty="" data-scope-types="*">
        No results matched your search
       </command-palette-tip>
       <div hidden="">
        <div data-octicon-id="arrow-right-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="arrow-right-color-fg-default" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-default" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="codespaces-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
          </path>
          <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="copy-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-copy color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
          </path>
          <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="dash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-dash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 7.75A.75.75 0 0 1 2.75 7h10a.75.75 0 0 1 0 1.5h-10A.75.75 0 0 1 2 7.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="file-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-file color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="gear-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-gear color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="lock-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-lock color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M4 4a4 4 0 0 1 8 0v2h.25c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 12.25 15h-8.5A1.75 1.75 0 0 1 2 13.25v-5.5C2 6.784 2.784 6 3.75 6H4Zm8.25 3.5h-8.5a.25.25 0 0 0-.25.25v5.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25ZM10.5 6V4a2.5 2.5 0 1 0-5 0v2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="moon-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-moon color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M9.598 1.591a.749.749 0 0 1 .785-.175 7.001 7.001 0 1 1-8.967 8.967.75.75 0 0 1 .961-.96 5.5 5.5 0 0 0 7.046-7.046.75.75 0 0 1 .175-.786Zm1.616 1.945a7 7 0 0 1-7.678 7.678 5.499 5.499 0 1 0 7.678-7.678Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="person-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-person color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="pencil-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-pencil color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="issue-opened-open" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-issue-opened open" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="git-pull-request-draft-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request-draft color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M3.25 1A2.25 2.25 0 0 1 4 5.372v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.251 2.251 0 0 1 3.25 1Zm9.5 14a2.25 2.25 0 1 1 0-4.5 2.25 2.25 0 0 1 0 4.5ZM2.5 3.25a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0ZM3.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm9.5 0a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM14 7.5a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm0-4.25a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="search-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sun-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sun color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8Zm0-1.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Zm5.657-8.157a.75.75 0 0 1 0 1.061l-1.061 1.06a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.06-1.06a.75.75 0 0 1 1.06 0Zm-9.193 9.193a.75.75 0 0 1 0 1.06l-1.06 1.061a.75.75 0 1 1-1.061-1.06l1.06-1.061a.75.75 0 0 1 1.061 0ZM8 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V.75A.75.75 0 0 1 8 0ZM3 8a.75.75 0 0 1-.75.75H.75a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 3 8Zm13 0a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 16 8Zm-8 5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 8 13Zm3.536-1.464a.75.75 0 0 1 1.06 0l1.061 1.06a.75.75 0 0 1-1.06 1.061l-1.061-1.06a.75.75 0 0 1 0-1.061ZM2.343 2.343a.75.75 0 0 1 1.061 0l1.06 1.061a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-1.06-1.06a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sync-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sync color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.705 8.005a.75.75 0 0 1 .834.656 5.5 5.5 0 0 0 9.592 2.97l-1.204-1.204a.25.25 0 0 1 .177-.427h3.646a.25.25 0 0 1 .25.25v3.646a.25.25 0 0 1-.427.177l-1.38-1.38A7.002 7.002 0 0 1 1.05 8.84a.75.75 0 0 1 .656-.834ZM8 2.5a5.487 5.487 0 0 0-4.131 1.869l1.204 1.204A.25.25 0 0 1 4.896 6H1.25A.25.25 0 0 1 1 5.75V2.104a.25.25 0 0 1 .427-.177l1.38 1.38A7.002 7.002 0 0 1 14.95 7.16a.75.75 0 0 1-1.49.178A5.5 5.5 0 0 0 8 2.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="trash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-trash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="key-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-key color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="comment-discussion-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-slash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell-slash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="paintbrush-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-paintbrush color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.134 1.535c.7-.509 1.416-.942 2.076-1.155.649-.21 1.463-.267 2.069.34.603.601.568 1.411.368 2.07-.202.668-.624 1.39-1.125 2.096-1.011 1.424-2.496 2.987-3.775 4.249-1.098 1.084-2.132 1.839-3.04 2.3a3.744 3.744 0 0 1-1.055 3.217c-.431.431-1.065.691-1.657.861-.614.177-1.294.287-1.914.357A21.151 21.151 0 0 1 .797 16H.743l.007-.75H.749L.742 16a.75.75 0 0 1-.743-.742l.743-.008-.742.007v-.054a21.25 21.25 0 0 1 .13-2.284c.067-.647.187-1.287.358-1.914.17-.591.43-1.226.86-1.657a3.746 3.746 0 0 1 3.227-1.054c.466-.893 1.225-1.907 2.314-2.982 1.271-1.255 2.833-2.75 4.245-3.777ZM1.62 13.089c-.051.464-.086.929-.104 1.395.466-.018.932-.053 1.396-.104a10.511 10.511 0 0 0 1.668-.309c.526-.151.856-.325 1.011-.48a2.25 2.25 0 1 0-3.182-3.182c-.155.155-.329.485-.48 1.01a10.515 10.515 0 0 0-.309 1.67Zm10.396-10.34c-1.224.89-2.605 2.189-3.822 3.384l1.718 1.718c1.21-1.205 2.51-2.597 3.387-3.833.47-.662.78-1.227.912-1.662.134-.444.032-.551.009-.575h-.001V1.78c-.014-.014-.113-.113-.548.027-.432.14-.995.462-1.655.942Zm-4.832 7.266-.001.001a9.859 9.859 0 0 0 1.63-1.142L7.155 7.216a9.7 9.7 0 0 0-1.161 1.607c.482.302.889.71 1.19 1.192Z">
          </path>
         </svg>
        </div>
        <command-palette-item-group data-default-priority="0" data-group-hint="" data-group-id="top" data-group-limits="{}" data-group-title="Top result">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="1" data-group-hint="Type &gt; to filter" data-group-id="commands" data-group-limits='{"static_items_page":50,"issue":50,"pull_request":50,"discussion":50}' data-group-title="Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="2" data-group-hint="Type &gt; to filter" data-group-id="global_commands" data-group-limits='{"issue":0,"pull_request":0,"discussion":0}' data-group-title="Global Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="3" data-group-hint="" data-group-id="this_page" data-group-limits="{}" data-group-title="This Page">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="4" data-group-hint="" data-group-id="files" data-group-limits="{}" data-group-title="Files">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="5" data-group-hint="" data-group-id="default" data-group-limits='{"static_items_page":50}' data-group-title="Default">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="6" data-group-hint="" data-group-id="pages" data-group-limits='{"repository":10}' data-group-title="Pages">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="7" data-group-hint="" data-group-id="access_policies" data-group-limits="{}" data-group-title="Access Policies">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="8" data-group-hint="" data-group-id="organizations" data-group-limits="{}" data-group-title="Organizations">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="9" data-group-hint="" data-group-id="repositories" data-group-limits="{}" data-group-title="Repositories">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="10" data-group-hint="Type # to filter" data-group-id="references" data-group-limits="{}" data-group-title="Issues, pull requests, and discussions">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="11" data-group-hint="" data-group-id="teams" data-group-limits="{}" data-group-title="Teams">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="12" data-group-hint="" data-group-id="users" data-group-limits="{}" data-group-title="Users">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="13" data-group-hint="" data-group-id="memex_projects" data-group-limits="{}" data-group-title="Projects">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="14" data-group-hint="" data-group-id="projects" data-group-limits="{}" data-group-title="Projects (classic)">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="15" data-group-hint="" data-group-id="footer" data-group-limits="{}" data-group-title="Footer">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="16" data-group-hint="" data-group-id="modes_help" data-group-limits="{}" data-group-title="Modes">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="17" data-group-hint="" data-group-id="filters_help" data-group-limits="{}" data-group-title="Use filters in issues, pull requests, discussions, and projects">
        </command-palette-item-group>
        <command-palette-page data-page-title="matt-chung" data-scope-id="MDQ6VXNlcjI5NjE5MzU4" data-scope-type="owner" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
        <command-palette-page data-page-title="wolbachia_transcriptomics_metaanalysis" data-scope-id="MDEwOlJlcG9zaXRvcnkyMjE3NzY3MDQ=" data-scope-type="repository" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
       </div>
       <command-palette-page data-is-root="">
       </command-palette-page>
       <command-palette-page data-page-title="matt-chung" data-scope-id="MDQ6VXNlcjI5NjE5MzU4" data-scope-type="owner">
       </command-palette-page>
       <command-palette-page data-page-title="wolbachia_transcriptomics_metaanalysis" data-scope-id="MDEwOlJlcG9zaXRvcnkyMjE3NzY3MDQ=" data-scope-type="repository">
       </command-palette-page>
      </command-palette-page-stack>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="search-links">
      </server-defined-provider>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="help">
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues
         </strong>
         and
         <strong>
          pull requests
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues, pull requests, discussions,
         </strong>
         and
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="@" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          organizations, repositories,
         </strong>
         and
         <strong>
          users
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          @
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="!" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          !
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="/" data-scope-types='["repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          files
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          /
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="&gt;">
        <span data-target="command-palette-help.titleElement">
         Activate
         <strong>
          command mode
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          &gt;
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:pr">
        <span data-target="command-palette-help.titleElement">
         Filter to pull requests
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:pr
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:issue">
        <span data-target="command-palette-help.titleElement">
         Filter to issues
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:issue
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:discussion" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:discussion
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:project" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to projects
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:project
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:open">
        <span data-target="command-palette-help.titleElement">
         Filter to open issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:open
         </kbd>
        </span>
       </command-palette-help>
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/commands" data-supported-modes="[]" data-supports-commands="" data-targets="command-palette.serverDefinedProviderElements" data-type="commands">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_page_navigation" data-supported-modes='[""]' data-supported-scope-types='["","owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to" data-supported-modes='["@","@"]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to_members_only" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_members_only_prefetched" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/files" data-supported-modes='["/"]' data-supported-scope-types='["repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="files">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/discussions" data-supported-modes='["#"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/projects" data-supported-modes='["#","!"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/recent_issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/teams" data-supported-modes='["@",""]' data-supported-scope-types='["owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/name_with_owner_repository" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
     </command-palette>
    </details-dialog>
   </details>
   <div class="position-fixed bottom-0 left-0 ml-5 mb-5 js-command-palette-toasts" style="z-index: 1000">
    <div class="Toast Toast--loading" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="Toast--spinner" height="18" viewbox="0 0 32 32" width="18">
       <path d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4" fill="#959da5">
       </path>
       <path d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z" fill="#ffffff">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--error" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-stop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M4.47.22A.749.749 0 0 1 5 0h6c.199 0 .389.079.53.22l4.25 4.25c.141.14.22.331.22.53v6a.749.749 0 0 1-.22.53l-4.25 4.25A.749.749 0 0 1 11 16H5a.749.749 0 0 1-.53-.22L.22 11.53A.749.749 0 0 1 0 11V5c0-.199.079-.389.22-.53Zm.84 1.28L1.5 5.31v5.38l3.81 3.81h5.38l3.81-3.81V5.31L10.69 1.5ZM8 4a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 4Zm0 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--warning" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--success" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-info" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.5 7.75A.75.75 0 0 1 7.25 7h1a.75.75 0 0 1 .75.75v2.75h.25a.75.75 0 0 1 0 1.5h-2a.75.75 0 0 1 0-1.5h.25v-2h-.25a.75.75 0 0 1-.75-.75ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
   </div>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <template class="js-user-list-create-dialog-template" data-label="Create list">
       <div class="Box-header">
        <h2 class="Box-title">
         Create list
        </h2>
       </div>
       <!-- '"` -->
       <!-- </textarea></xmp> -->
       <form accept-charset="UTF-8" action="/stars/ninenia/lists" class="Box-body d-flex flex-column p-3 js-user-list-form" data-turbo="false" method="post">
        <input autocomplete="off" name="authenticity_token" type="hidden" value="yign7Rywd9K84D-q8N6PaZmmKVjNAme6USMKkARfsZIaRI3JiWqhdV_dlc1jBhbPPKXiLUklTobPiS8lsI6nJQ">
         <p class="color-fg-subtle mb-3">
          Create a list to organize your starred repositories.
         </p>
         <input name="repository_id" type="hidden" value="{{ repositoryId }}"/>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <auto-check required="" src="/stars/ninenia/list-check?attr=name">
           <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
            <input aria-label="List name" autofocus="" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="32" maxlength="32" name="user_list[name]" placeholder="⭐️ Name this list" required="" type="text" value="{{ placeholderName }}"/>
           </text-expander>
           <input data-csrf="true" type="hidden" value="ekrnBRUcwRcpdiAlI_AbgUlXIAGhhWNFJLSvpGMEV9kaYrv02jKTc_89v2HA8t-3SL1-cA133X-IntVtgWzjdw">
           </input>
          </auto-check>
          <p class="note error position-relative js-user-list-error" hidden="">
           Name .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           32 remaining
          </p>
         </div>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
           <textarea aria-label="List description" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="160" maxlength="160" name="user_list[description]" placeholder="Write a description" style="height: 74px; min-height: 74px"></textarea>
          </text-expander>
          <p class="note error position-relative js-user-list-error" hidden="">
           Description .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           160 remaining
          </p>
         </div>
         <div class="js-user-list-base flash flash-error mx-0 mt-0 mb-2" data-generic-message="Unable to save your list at this time." data-view-component="true" hidden="hidden">
          .
         </div>
         <button class="Button--primary Button--medium Button Button--fullWidth mt-2" data-disable-invalid="true" data-submitting-message="Creating..." data-view-component="true" disabled="disabled" type="submit">
          <span class="Button-content">
           <span class="Button-label">
            Create
           </span>
          </span>
         </button>
         <p class="note mt-2 mb-0">
          <strong>
           Tip:
          </strong>
          type
          <code>
           :
          </code>
          to add emoji to the name or description.
         </p>
        </input>
       </form>
       <div class="Box-footer Box-row--gray text-small color-fg-muted d-flex flex-items-baseline py-2" data-view-component="true">
        <span aria-label="Feature Release Label: Beta" class="Label Label--success Label--inline px-2 mr-2" data-view-component="true" title="Feature Release Label: Beta">
         Beta
        </span>
        <span class="mr-1">
         Lists are currently in beta.
        </span>
        <a href="https://github.com//github/feedback/discussions/categories/lists">
         Share feedback and report bugs.
        </a>
       </div>
      </template>
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header">
       <div class="d-flex flex-wrap flex-justify-end mb-3 container-xl px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal" id="repo-title-component">
          <img alt="Owner avatar" class="avatar mr-2 d-none d-md-block avatar-user" height="24" src="https://avatars.githubusercontent.com/u/29619358?s=48&amp;v=4" width="24"/>
          <strong class="mr-2 flex-self-stretch d-none d-md-block" itemprop="name">
           <a class="color-fg-default" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/matt-chung/wolbachia_transcriptomics_metaanalysis">
            wolbachia_transcriptomics_metaanalysis
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1 d-none d-md-block">
           Public
          </span>
         </div>
         <div class="d-none d-md-block">
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <include-fragment loading="lazy" src="/notifications/221776704/watch_subscription?aria_id_prefix=repository-details&amp;button_block=false&amp;show_count=true">
            <div data-hide-on-error="">
             <button aria-label="Watching a repository" class="tooltipped tooltipped-n Button--secondary Button--small Button" data-view-component="true" disabled="disabled" type="button">
              <span class="Button-content">
               <span class="Button-visual Button-leadingVisual">
                <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                 </path>
                </svg>
               </span>
               <span class="Button-label">
                Watch
               </span>
              </span>
             </button>
            </div>
            <p data-show-on-error="" hidden="">
             <span class="d-flex pt-1">
              <svg aria-hidden="true" class="octicon octicon-alert mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
               </path>
              </svg>
              Couldn't load subscription status.
              <button class="Button--link Button--medium Button" data-retry-button="" data-view-component="true" type="button">
               <span class="Button-content">
                <span class="Button-label">
                 Retry
                </span>
               </span>
              </button>
             </span>
            </p>
           </include-fragment>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a class="btn-sm btn BtnGroup-item" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":221776704,"originating_url":"https://github.com/matt-chung/wolbachia_transcriptomics_metaanalysis","user_id":126681664}}' data-hydro-click-hmac="543dab41586939be3f70886c64bd8e853c7bd458dbaf25a0e45043768eede508" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/fork" icon="repo-forked" id="fork-button">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             Fork
             <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
              0
             </span>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="fork-button" id="tooltip-2c6d2275-b61e-44ce-a467-533225c7c2d8" popover="manual">
              Fork your own copy of matt-chung/wolbachia_transcriptomics_metaanalysis
             </tool-tip>
            </a>
            <details class="details-reset details-overlay BtnGroup-parent d-inline-block position-relative" data-view-component="true" group_item="true" id="my-forks-menu-221776704">
             <summary aria-label="See your forks of this repository" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
              <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
               </path>
              </svg>
             </summary>
             <details-menu class="SelectMenu right-0" role="menu" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/my_forks_menu_content?can_fork=true">
              <div class="SelectMenu-modal">
               <button aria-label="Close menu" class="SelectMenu-closeButton position-absolute right-0 m-2" data-toggle-for="my-forks-menu-221776704" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
               <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-785e4f">
                <div class="SelectMenu-list">
                 <include-fragment aria-label="Loading" class="SelectMenu-loading">
                  <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                   <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                   </circle>
                   <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                   </path>
                  </svg>
                 </include-fragment>
                </div>
               </div>
              </div>
             </details-menu>
            </details>
           </div>
          </li>
          <li>
           <template class="js-unstar-confirmation-dialog-template">
            <div class="Box-header">
             <h2 class="Box-title">
              Unstar this repository?
             </h2>
            </div>
            <div class="Box-body">
             <p class="mb-3">
              This will remove {{ repoNameWithOwner }} from the {{ listsWithCount }} that it's been added to.
             </p>
             <div class="form-actions">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="{{ confirmUrl }}" class="js-social-confirmation-form" data-turbo="false" method="post">
               <input name="authenticity_token" type="hidden" value="{{ confirmCsrfToken }}"/>
               <input name="confirm" type="hidden" value="true"/>
               <button class="btn-danger btn width-full" data-close-dialog="true" data-view-component="true" type="submit">
                Unstar
               </button>
              </form>
             </div>
            </div>
           </template>
           <div class="js-toggler-container js-social-container starring-container d-flex" data-view-component="true">
            <div class="starred BtnGroup flex-1 ml-0" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/matt-chung/wolbachia_transcriptomics_metaanalysis/unstar" class="js-social-form BtnGroup-parent flex-auto js-deferred-toggler-target" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="g70iZL6SIlYouIRr5MwoLnr2rxkZztjmAJEmWDrurDeQCfw6fKlD2wvOW0XDQLveSJoIem-dmbsOxiYB7pmMcA">
               <input class="js-confirm-csrf-token" data-csrf="true" type="hidden" value="H1Nesa5o97VdouS_nwSXHUNr8ATLnWUa2aNSOfJFdg4M54DvbFOWOH7UO5G4iATtcQdXZ73OJEfX9FJgJjJWSQ">
                <input name="context" type="hidden" value="repository"/>
                <button aria-label="Unstar this repository (1)" class="rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":221776704,"originating_url":"https://github.com/matt-chung/wolbachia_transcriptomics_metaanalysis","user_id":126681664}}' data-hydro-click-hmac="25bf062e03321c05e804f3e870a14c007d413d14ad174618dbbe8943410816aa" data-view-component="true" type="submit">
                 <svg aria-hidden="true" class="octicon octicon-star-fill starred-button-icon d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                  </path>
                 </svg>
                 <span class="d-inline" data-view-component="true">
                  Starred
                 </span>
                 <span aria-label="1 user starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-unstar" title="1">
                  1
                 </span>
                </button>
               </input>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-221776704-starred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-221776704-starred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-83e4b1">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
            <div class="unstarred BtnGroup ml-0 flex-1" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/matt-chung/wolbachia_transcriptomics_metaanalysis/star" class="js-social-form BtnGroup-parent flex-auto" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="t0GsaR-f_g8urCiT5rEhvvB9caGwD2jXfbAGtuPnjk2Fbotgvvh6J_bj3iROsq2npYmleEzBZNqFYzib6fYI-Q">
               <input name="context" type="hidden" value="repository"/>
               <button aria-label="Star this repository (1)" class="js-toggler-target rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":221776704,"originating_url":"https://github.com/matt-chung/wolbachia_transcriptomics_metaanalysis","user_id":126681664}}' data-hydro-click-hmac="21401791d6c54b4ae52799ec5357df3efe4f98f614ec4c1c0f8703a46ddd6bef" data-view-component="true" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
                <span class="d-inline" data-view-component="true">
                 Star
                </span>
                <span aria-label="1 user starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="1">
                 1
                </span>
               </button>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-221776704-unstarred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-221776704-unstarred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-550987">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div class="container-xl">
        <div data-turbo-replace="" id="responsive-meta-container">
         <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
          <div class="d-flex gap-2 mt-n3 mb-3 flex-wrap">
           <div class="d-flex flex-row gap-2">
            <notifications-list-subscription-form data-action="notifications-dialog-label-toggled:notifications-list-subscription-form#handleDialogLabelToggle">
             <details class="details-reset details-overlay details-overlay-dark" data-action="toggle:notifications-list-subscription-form#detailsToggled" data-target="notifications-list-subscription-form.details">
              <summary aria-label="Notifications" class="Button--iconOnly Button--secondary Button--medium Button p-0" data-ga-click="Repository, click Watch settings, action:files#disambiguate" data-hydro-click='{"event_type":"repository.click","payload":{"target":"WATCH_BUTTON","repository_id":221776704,"originating_url":"https://github.com/matt-chung/wolbachia_transcriptomics_metaanalysis","user_id":126681664}}' data-hydro-click-hmac="03105f419339635ff26b4a9a57569f5d017b27f9a6f5cda3ddb54be013082e04" data-view-component="true" id="notifications-button">
               <span class="Button-content">
                <span class="Button-label">
                 <span data-menu-button="">
                  <span data-target="notifications-list-subscription-form.unwatchButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.stopIgnoringButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-bell-slash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.watchButtonCopy">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                 </span>
                </span>
               </span>
              </summary>
              <details-dialog aria-label="Notification settings" class="anim-fade-in fast Box Box--overlay">
               <div data-target="notifications-list-subscription-form.menu">
                <div class="Box-header">
                 <button aria-label="Close dialog" class="Box-btn-octicon btn-octicon float-right" data-action="click:notifications-list-subscription-form#closeMenu" data-close-dialog="" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                 <h3 class="Box-title">
                  Notifications
                 </h3>
                </div>
                <div class="SelectMenu-list mb-0" role="menu">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitForm" data-target="notifications-list-subscription-form.form" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="kXwnSVTuB4NWNzg3n-t-zb2bJUcr_PxQKTmbfCRs25SJvCszSlFygrokDz1DC_i64r62Uv6mV4QLwmmIc91lDQ">
                   <input name="repository_id" type="hidden" value="221776704"/>
                   <button aria-checked="true" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="included">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Participating and @mentions
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Only receive notifications from this repository when participating or @mentioned.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="subscribed">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      All Activity
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Notified of all notifications on this repository.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="ignore">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Ignore
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Never be notified.
                     </div>
                    </div>
                   </button>
                  </input>
                 </form>
                 <button aria-checked="false" aria-haspopup="true" class="SelectMenu-item flex-items-start pr-3" data-action="click:notifications-list-subscription-form#openCustomDialog" data-target="notifications-list-subscription-form.customButton" role="menuitemradio" type="button">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                    </path>
                   </svg>
                  </span>
                  <div>
                   <div class="d-flex flex-items-start flex-justify-between">
                    <div class="f5 text-bold">
                     Custom
                    </div>
                    <div class="f5 pr-1">
                     <svg aria-hidden="true" class="octicon octicon-arrow-right" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </div>
                   <div class="text-small color-fg-muted text-normal pb-1">
                    Select events you want to be notified of in addition to participating and @mentions.
                   </div>
                  </div>
                 </button>
                 <div class="px-3 py-2 d-flex color-bg-subtle flex-items-center">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile SelectMenu-icon SelectMenu-icon--device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </span>
                  <span classname="text-small color-fg-muted text-normal pb-1">
                   Get push notifications on
                   <a href="https://apps.apple.com/app/apple-store/id1477376905?ct=watch-dropdown&amp;mt=8&amp;pt=524675" rel="noopener noreferrer" target="_blank">
                    iOS
                   </a>
                   or
                   <a href="https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dwatch-dropdown%26utm_medium%3Dweb%26utm_source%3Dgithub" rel="noopener noreferrer" target="_blank">
                    Android
                   </a>
                   .
                  </span>
                 </div>
                </div>
               </div>
               <div data-target="notifications-list-subscription-form.customDialog" hidden="">
                <header class="SelectMenu-header pb-0 border-bottom-0 px-2 px-sm-3">
                 <h1 class="f3 SelectMenu-title d-inline-flex">
                  <button aria-label="Return to menu" class="color-bg-default border-0 px-2 py-0 m-0 Link--secondary f5" data-action="click:notifications-list-subscription-form#closeCustomDialog" type="button">
                   <svg aria-hidden="true" class="octicon octicon-arrow-left" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z">
                    </path>
                   </svg>
                  </button>
                  Custom
                 </h1>
                </header>
                <div class="overflow-visible">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitCustomForm" data-target="notifications-list-subscription-form.customform" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="lt46lN4h2lovMQluAjBXWbgdklPwaGbMCosj30_wXMiOHjbuwJ6vW8MiPmTe0NEu5zgBRiUyzRgocNErGEHiUQ">
                   <input name="repository_id" type="hidden" value="221776704"/>
                   <fieldset>
                    <legend>
                     <div class="text-small color-fg-muted pt-0 pr-3 pb-3 pl-6 pl-sm-5 border-bottom mb-3">
                      Select events you want to be notified of in addition to participating and @mentions.
                     </div>
                    </legend>
                    <div data-target="notifications-list-subscription-form.labelInputs">
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Issue"/>
                      Issues
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="PullRequest"/>
                      Pull requests
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Release"/>
                      Releases
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input aria-describedby="Discussion-disabled" aria-disabled="true" data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Discussion"/>
                      Discussions
                     </label>
                     <div class="color-fg-muted" id="Discussion-disabled">
                      &gt;
                          Discussions are not enabled for this repository
                     </div>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="SecurityAlert"/>
                      Security alerts
                     </label>
                    </div>
                   </fieldset>
                   <div class="pt-2 pb-3 px-3 d-flex flex-justify-start flex-row-reverse">
                    <button class="Button--primary Button--medium Button ml-2" data-target="notifications-list-subscription-form.customSubmit" data-view-component="true" disabled="disabled" name="do" type="submit" value="custom">
                     <span class="Button-content">
                      <span class="Button-label">
                       Apply
                      </span>
                     </span>
                    </button>
                    <button class="Button--secondary Button--medium Button" data-action="click:notifications-list-subscription-form#resetForm" data-close-dialog="" data-view-component="true" type="button">
                     <span class="Button-content">
                      <span class="Button-label">
                       Cancel
                      </span>
                     </span>
                    </button>
                   </div>
                  </input>
                 </form>
                </div>
               </div>
              </details-dialog>
             </details>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="notifications-button" id="tooltip-0ffb761c-e90d-455d-b060-ff47fa53592a" popover="manual">
              Notification settings
             </tool-tip>
            </notifications-list-subscription-form>
            <div class="BtnGroup d-flex" data-view-component="true">
             <a aria-labelledby="tooltip-031d11c7-baeb-4b32-b753-3c5f447d9435" class="Button Button--iconOnly Button--secondary Button--medium" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":221776704,"originating_url":"https://github.com/matt-chung/wolbachia_transcriptomics_metaanalysis","user_id":126681664}}' data-hydro-click-hmac="543dab41586939be3f70886c64bd8e853c7bd458dbaf25a0e45043768eede508" data-view-component="true" group_item="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/fork" id="fork-icon-button" variant="small">
              <svg aria-hidden="true" class="octicon octicon-repo-forked Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
               </path>
              </svg>
             </a>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="fork-icon-button" id="tooltip-031d11c7-baeb-4b32-b753-3c5f447d9435" popover="manual">
              Fork your own copy of matt-chung/wolbachia_transcriptomics_metaanalysis
             </tool-tip>
            </div>
            <div class="js-toggler-container starring-container" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/matt-chung/wolbachia_transcriptomics_metaanalysis/unstar" class="starred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="yskQNqCzcsafEzjklaj9WvK6nRqE_1WTiT2-ZKKjYIfZfc5oYogTS7xl58qyJG6qwNY6efKsFM6Har49dtRAwA">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-f337686f-c1be-4046-85c3-5888d54bea89" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target starred-button-icon" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":221776704,"originating_url":"https://github.com/matt-chung/wolbachia_transcriptomics_metaanalysis","user_id":126681664}}' data-hydro-click-hmac="25bf062e03321c05e804f3e870a14c007d413d14ad174618dbbe8943410816aa" data-view-component="true" id="icon-button-b6df5f80-46a8-4996-a696-436f4b358248" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-b6df5f80-46a8-4996-a696-436f4b358248" id="tooltip-f337686f-c1be-4046-85c3-5888d54bea89" popover="manual">
                Unstar this repository
               </tool-tip>
              </input>
             </form>
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/matt-chung/wolbachia_transcriptomics_metaanalysis/star" class="unstarred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="_Q3_gl8DkBKQnKtZH4wSHSQakaRhAoG1Ng7lYAMvHuLPItiL_mQUOkjTXe63j54Ece5FfZ3MjbjO3dtNCT6YVg">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-45fab631-4a4b-4add-bab1-5705c691044c" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":221776704,"originating_url":"https://github.com/matt-chung/wolbachia_transcriptomics_metaanalysis","user_id":126681664}}' data-hydro-click-hmac="21401791d6c54b4ae52799ec5357df3efe4f98f614ec4c1c0f8703a46ddd6bef" data-view-component="true" id="icon-button-3c60ee0c-e3a8-4512-993a-cb539ffdc7d6" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-3c60ee0c-e3a8-4512-993a-cb539ffdc7d6" id="tooltip-45fab631-4a4b-4add-bab1-5705c691044c" popover="manual">
                Star this repository
               </tool-tip>
              </input>
             </form>
            </div>
           </div>
           <div class="d-flex flex-row gap-2">
           </div>
          </div>
          <h3 class="sr-only">
           License
          </h3>
          <div class="mb-2">
           <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/LICENSE">
            <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
             </path>
            </svg>
            MIT license
           </a>
          </div>
          <div class="mb-3">
           <ul aria-label="Repository details" class="d-flex flex-wrap mb-2 gap-2">
            <a class="Link--secondary no-underline d-block mr-2" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/stargazers" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              1
             </span>
             star
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/forks" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             forks
            </a>
            <div class="color-fg-muted mr-2" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-eye mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              1
             </span>
             watching
            </div>
            <a class="Link--secondary no-underline d-block mr-2" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/branches" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
              </path>
             </svg>
             <include-fragment src="/matt-chung/wolbachia_transcriptomics_metaanalysis/branch-count">
              <span>
               Branches
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/tags" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
              </path>
             </svg>
             <include-fragment src="/matt-chung/wolbachia_transcriptomics_metaanalysis/tag-count">
              <span>
               Tags
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/activity" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
              </path>
             </svg>
             <span>
              Activity
             </span>
            </a>
           </ul>
           <div class="mb-2 d-flex color-fg-muted">
            <div class="d-flex flex-items-center" style="height: 21px">
             <svg aria-hidden="true" class="octicon octicon-globe flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
              </path>
             </svg>
            </div>
            <span class="flex-auto min-width-0 width-fit">
             Public repository
            </span>
           </div>
          </div>
         </div>
        </div>
        <div class="border-bottom mx-xl-5">
        </div>
       </div>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <a class="d-none js-github-dev-shortcut" data-hotkey=".,Mod+Alt+." href="https://github.dev/">
         Open in github.dev
        </a>
        <a class="d-none js-github-dev-new-tab-shortcut" data-hotkey="Shift+.,Shift+&gt;,&gt;" href="https://github.dev/" rel="noopener noreferrer" target="_blank">
         Open in a new github.dev tab
        </a>
        <a class="d-none" data-hotkey=",,Mod+Alt+," href="/codespaces/new/matt-chung/wolbachia_transcriptomics_metaanalysis?resume=1" target="_blank">
         Open in codespace
        </a>
        <h1 class="sr-only">
         matt-chung/wolbachia_transcriptomics_metaanalysis
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/spoofed_commit_check/6fcdd77b7b8d96dc74e8d897d63be4f020dc8f40">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":true,"path":"/","repo":{"id":221776704,"defaultBranch":"master","name":"wolbachia_transcriptomics_metaanalysis","ownerLogin":"matt-chung","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2019-11-15T04:01:20.000+08:00","ownerAvatar":"https://avatars.githubusercontent.com/u/29619358?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":{"id":126681664,"login":"ninenia","userEmail":"2945518981@qq.com"},"refInfo":{"name":"master","listCacheKey":"v0:1573761685.0","canEdit":true,"refType":"branch","currentOid":"6fcdd77b7b8d96dc74e8d897d63be4f020dc8f40"},"tree":{"items":[{"name":"gff_map_files","path":"gff_map_files","contentType":"directory"},{"name":"images","path":"images","contentType":"directory"},{"name":"sup_tables","path":"sup_tables","contentType":"directory"},{"name":".gitattributes","path":".gitattributes","contentType":"file"},{"name":"LICENSE","path":"LICENSE","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":6,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/matt-chung/wolbachia_transcriptomics_metaanalysis/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":true,"httpUrl":"https://github.com/matt-chung/wolbachia_transcriptomics_metaanalysis.git","showCloneWarning":false,"sshUrl":"git@github.com:matt-chung/wolbachia_transcriptomics_metaanalysis.git","sshCertificatesRequired":false,"sshCertificatesAvailable":false,"ghCliUrl":"gh repo clone matt-chung/wolbachia_transcriptomics_metaanalysis","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"x-github-client://openRepo/https://github.com/matt-chung/wolbachia_transcriptomics_metaanalysis","showVisualStudioCloneButton":true,"visualStudioCloneUrl":"git-client://clone?repo=https%3A%2F%2Fgithub.com%2Fmatt-chung%2Fwolbachia_transcriptomics_metaanalysis","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/matt-chung/wolbachia_transcriptomics_metaanalysis/archive/refs/heads/master.zip"}},"repoPolicyInfo":{"allowed":true,"canBill":true,"changesWouldBeSafe":true,"disabledByBusiness":false,"disabledByOrganization":false,"hasIpAllowLists":false},"currentUserIsEnterpriseManaged":false,"enterpriseManagedBusinessName":null,"codespacesEnabled":true,"hasAccessToCodespaces":true},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"59","overviewFiles":[{"displayName":"README.md","repoName":"wolbachia_transcriptomics_metaanalysis","refName":"master","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eA meta-analysis of Wolbachia transcriptomics reveals a stage-specific Wolbachia transcriptional response shared across different hosts\u003c/h1\u003e\u003ca id=\"user-content-a-meta-analysis-of-wolbachia-transcriptomics-reveals-a-stage-specific-wolbachia-transcriptional-response-shared-across-different-hosts\" class=\"anchor\" aria-label=\"Permalink: A meta-analysis of Wolbachia transcriptomics reveals a stage-specific Wolbachia transcriptional response shared across different hosts\" href=\"#a-meta-analysis-of-wolbachia-transcriptomics-reveals-a-stage-specific-wolbachia-transcriptional-response-shared-across-different-hosts\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTable of Contents\u003c/h1\u003e\u003ca id=\"user-content-table-of-contents\" class=\"anchor\" aria-label=\"Permalink: Table of Contents\" href=\"#table-of-contents\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#setpaths\"\u003eSet software, input, and output directory paths\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#setpaths_software\"\u003eSoftware\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#setpaths_directories\"\u003eDirectories\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis\"\u003eDownload, align, and quantify Wolbachia sequencing data\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_createdirectories\"\u003eCreate directories\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_getreferences\"\u003eDownload reference files and create combined references\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_getfastq\"\u003eDownload FASTQ files\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_getfastq_sra\"\u003eSRA Datasets\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_getfastq_luck2015\"\u003eLuck et al 2015 Dataset\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_align\"\u003eAlign FASTQ files to combined host and Wolbachia reference while allowing for splicing\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_align_se\"\u003eSingle-End FASTQs\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_align_pe\"\u003ePaired-End FASTQs\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_wolbachiareads\"\u003eCreate a list of Wolbachia-mapping reads in all BAM files\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_subsetfastq\"\u003eCreate subsets of all FASTQ files that contain only Wolbachia-mapping reads\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_subsetfastq_se\"\u003eSingle-End FASTQs\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_subsetfastq_pe\"\u003ePaired-End FASTQs\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_realign\"\u003eAlign FASTQ files containing only Wolbachia-mapping reads to combined host and Wolbachia reference while disallowing splicing\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_realign_se\"\u003eSingle-End FASTQs\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_realign_pe\"\u003ePaired-End FASTQs\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_realign_sortbam\"\u003eSort BAM files\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_indexbam\"\u003eIndex BAM files\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_quant\"\u003eQuantify Wolbachia genes\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_gff2map\"\u003eConvert GFF to a mapping table\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_createcdsfasta\"\u003eCreate nucleotide coding sequence fasta files\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#bashanalysis_interproscan\"\u003eIdentify InterPro descriptions and GO terms for Wolbachia transcripts\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis\"\u003eTranscriptomics meta-analysis\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_counts\"\u003eCreate counts table\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_counts_setinputs\"\u003eSet R inputs\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_counts_sessioninfo\"\u003eView sessionInfo\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_counts_maketable\"\u003eCombine count files into a single table for each study\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_iprscan2geneinfo\"\u003eConvert InterProScan output to geneinfo file\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de\"\u003eDifferential expression and WGCNA analyses\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_setinputs\"\u003eSet R inputs\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_sessioninfo\"\u003eLoad R packages and view sessionInfo\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_loadfunctions\"\u003eLoad R functions\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_counts_readcounts\"\u003eProcess counts files and calculate TPM values\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_counts_readcounts\"\u003eRead counts table into a list\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_counts_removenonproteingenes\"\u003eExclude non-protein-coding genes from counts table\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_counts_removenoprobegenes\"\u003eExclude genes without Agilent SureSelect probes from counts table in Chung et al 2019 study\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_counts_countscolsum\"\u003ePrint the number of reads mapping to protein-coding genes for each sample in each study\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_counts_tpm\"\u003eCreate TPM table\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_keys\"\u003eCreate keys\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_keys_samples\"\u003eSample keys\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_keys_size\"\u003eSize keys\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_keys_hm1\"\u003eHeatmap log2TPM key\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_keys_hm2\"\u003eHeatmap z-score log2TPM key\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_keys_saturation\"\u003eAssess if samples have been sequenced to saturation using saturation curves\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_exclude\"\u003eExclude samples and/or studies from differential expression analyses due to inadequate sequencing depth\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_exclude\"\u003eIdentify the number of genes in each study that meets the minimum CPM threshold\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_de\"\u003eIdentify differentially expressed genes (FDR \u0026lt; 0.05) in each study\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_pca\"\u003eConduct principal component analyses\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_hc\"\u003eConduct hierarchical clustering analyses\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_heatmap\"\u003eCreate TPM and log2TPM heatmaps\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_selectanalysis\"\u003eChoose studies for basic pairwise analyses or WGCNA\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_pairwise\"\u003eConduct pairwise analyses on studies with two biological groups\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_pairwise_hm\"\u003eCreate z-score of log2TPM heatmaps with a column bar that shows up- and down-regulated genes\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_pairwise_fxnterm\"\u003eConduct functional term enrichment analysis on up- and down-regulated gene subsets\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_wgcna\"\u003eConduct WGCNA on time-course studies\u003c/a\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_wgcna_softpower\"\u003eIdentify WGCNA soft power values\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_wgcna_createmodules\"\u003eCreate WGCNA expression modules and merge similar modules\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_wgcna_heatmap\"\u003ePlot WGCNA-derived expression modules\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"#ranalysis_de_wgcna_fxnterm\"\u003eConduct functional term enrichment analysis on WGCNA expression modules\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ol\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSet software and directory paths \u003ca name=\"user-content-setpaths\"\u003e\u003c/a\u003e\u003c/h1\u003e\u003ca id=\"user-content-set-software-and-directory-paths-\" class=\"anchor\" aria-label=\"Permalink: Set software and directory paths \" href=\"#set-software-and-directory-paths-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFor rerunning analyses, all paths in this section must be set by the user.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSoftware \u003ca name=\"user-content-setpaths_software\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-software-\" class=\"anchor\" aria-label=\"Permalink: Software \" href=\"#software-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"JULIA_DEPOT_PATH=/home/mattchung/.julia\nPYTHON_LIB_PATH=/usr/local/packages/python-3.5/lib\n\nJULIA_BIN_DIR=/usr/local/bin\nPYTHON_BIN_DIR=/usr/local/bin\n\nFADU_BIN_DIR=/home/mattchung/scripts/FADU\nHISAT2_BIN_DIR=/usr/local/packages/hisat2-2.1.0\nINTERPROSCAN_BIN_DIR=/local/aberdeen2rw/julie/Matt_dir/packages/interproscan-5.34-73.0\nNCBIBLAST_BIN_DIR=/usr/local/packages/ncbi-blast-2.2.26/bin\nSAMTOOLS_BIN_DIR=/usr/local/packages/samtools-1.9/bin\nSEQTK_BIN_DIR=/usr/local/packages/seqtk-1.2/bin\nSRATOOLKIT_BIN_DIR=/usr/local/packages/sratoolkit-2.9.0/bin\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eJULIA_DEPOT_PATH=/home/mattchung/.julia\nPYTHON_LIB_PATH=/usr/local/packages/python-3.5/lib\n\nJULIA_BIN_DIR=/usr/local/bin\nPYTHON_BIN_DIR=/usr/local/bin\n\nFADU_BIN_DIR=/home/mattchung/scripts/FADU\nHISAT2_BIN_DIR=/usr/local/packages/hisat2-2.1.0\nINTERPROSCAN_BIN_DIR=/local/aberdeen2rw/julie/Matt_dir/packages/interproscan-5.34-73.0\nNCBIBLAST_BIN_DIR=/usr/local/packages/ncbi-blast-2.2.26/bin\nSAMTOOLS_BIN_DIR=/usr/local/packages/samtools-1.9/bin\nSEQTK_BIN_DIR=/usr/local/packages/seqtk-1.2/bin\nSRATOOLKIT_BIN_DIR=/usr/local/packages/sratoolkit-2.9.0/bin\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDirectories \u003ca name=\"user-content-setpaths_directories\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-directories-\" class=\"anchor\" aria-label=\"Permalink: Directories \" href=\"#directories-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"INPUTS_DIR=\nSCRIPTS_DIR=/home/mattchung/scripts\nREADS_DIR=/local/aberdeen2rw/julie/Matt_dir/wolbachia_metatranscriptome_analysis/\nREFERENCES_DIR=/local/aberdeen2rw/julie/Matt_dir/wolbachia_metatranscriptome_analysis/references/\nWORKING_DIR=/local/projects-t3/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eINPUTS_DIR=\nSCRIPTS_DIR=/home/mattchung/scripts\nREADS_DIR=/local/aberdeen2rw/julie/Matt_dir/wolbachia_metatranscriptome_analysis/\nREFERENCES_DIR=/local/aberdeen2rw/julie/Matt_dir/wolbachia_metatranscriptome_analysis/references/\nWORKING_DIR=/local/projects-t3/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDownload, align, and quantify Wolbachia sequencing data \u003ca name=\"user-content-bashanalysis\"\u003e\u003c/a\u003e\u003c/h1\u003e\u003ca id=\"user-content-download-align-and-quantify-wolbachia-sequencing-data-\" class=\"anchor\" aria-label=\"Permalink: Download, align, and quantify Wolbachia sequencing data \" href=\"#download-align-and-quantify-wolbachia-sequencing-data-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCreate directories (2019-05-30) \u003ca name=\"user-content-bashanalysis_createdirectories\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-create-directories-2019-05-30-\" class=\"anchor\" aria-label=\"Permalink: Create directories (2019-05-30) \" href=\"#create-directories-2019-05-30-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"mkdir -p \u0026quot;$REFERENCES_DIR\u0026quot;\n\nmkdir -p \u0026quot;$READS_DIR\u0026quot;/wOo_darby_2012/bam\nmkdir -p \u0026quot;$READS_DIR\u0026quot;/wMel_darby_2014/bam\nmkdir -p \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2014/bam\nmkdir -p \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/bam\nmkdir -p \u0026quot;$READS_DIR\u0026quot;/wMel_gutzwiller_2015/bam\nmkdir -p \u0026quot;$READS_DIR\u0026quot;/wBm_grote_2017/bam\nmkdir -p \u0026quot;$READS_DIR\u0026quot;/wBm_chung_2019/bam\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003emkdir -p \"$REFERENCES_DIR\"\n\nmkdir -p \"$READS_DIR\"/wOo_darby_2012/bam\nmkdir -p \"$READS_DIR\"/wMel_darby_2014/bam\nmkdir -p \"$READS_DIR\"/wDi_luck_2014/bam\nmkdir -p \"$READS_DIR\"/wDi_luck_2015/bam\nmkdir -p \"$READS_DIR\"/wMel_gutzwiller_2015/bam\nmkdir -p \"$READS_DIR\"/wBm_grote_2017/bam\nmkdir -p \"$READS_DIR\"/wBm_chung_2019/bam\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDownload reference files and create combined references (2019-05-30) \u003ca name=\"user-content-bashanalysis_getreferences\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-download-reference-files-and-create-combined-references-2019-05-30-\" class=\"anchor\" aria-label=\"Permalink: Download reference files and create combined references (2019-05-30) \" href=\"#download-reference-files-and-create-combined-references-2019-05-30-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe gff reference for wDi must be downloaded from this link: \u003ca href=\"http://rast.nmpdr.org/rast.cgi?page=JobDetails\u0026amp;job=53659\" rel=\"nofollow\"\u003ehttp://rast.nmpdr.org/rast.cgi?page=JobDetails\u0026amp;job=53659\u003c/a\u003e (login:guest, password:guest). All subsequent code will make use of this wDi.gff at this path: \"$REFERENCES_DIR\"/wDi.gff\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## wBm + B. malayi\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/wBm.fna.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_genomic.fna.gz\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/bmalayi.fna.gz ftp://ftp.wormbase.org/pub/wormbase/releases/WS270/species/b_malayi/PRJNA10729/b_malayi.PRJNA10729.WS270.genomic.fa.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/wBm.fna.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/bmalayi.fna.gz\ncat \u0026quot;$REFERENCES_DIR\u0026quot;/wBm.fna \u0026quot;$REFERENCES_DIR\u0026quot;/bmalayi.fna \u0026gt; \u0026quot;$REFERENCES_DIR\u0026quot;/wBm_bmalayi_combined.fna\n\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/wBm.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_genomic.gff.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/wBm.gff.gz\n\n## wDi + D. immitis\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/wDi.fna.gz http://nematodes.org/downloads/959nematodegenomes/blast/db2/Dirofilaria_immitis_wolbachia_2.2.fna.gz\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/dimmitis.fna.gz http://nematodes.org/downloads/959nematodegenomes/blast/db2/Dirofilaria_immitis_2.2.fna.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/wDi.fna.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/dimmitis.fna.gz\ncat \u0026quot;$REFERENCES_DIR\u0026quot;/wDi.fna \u0026quot;$REFERENCES_DIR\u0026quot;/dimmitis.fna \u0026gt; \u0026quot;$REFERENCES_DIR\u0026quot;/wDi_dimmitis_combined.fna\n\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/wBm.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_genomic.gff.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/wBm.gff.gz\n\n## wMel + D. melanogaster\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/wMel.fna.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/025/GCF_000008025.1_ASM802v1/GCF_000008025.1_ASM802v1_genomic.fna.gz\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/dmelanogaster.fna.gz ftp://ftp.flybase.net/releases/FB2019_02/dmel_r6.27/fasta/dmel-all-chromosome-r6.27.fasta.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/wMel.fna.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/dmelanogaster.fna.gz\ncat \u0026quot;$REFERENCES_DIR\u0026quot;/wMel.fna \u0026quot;$REFERENCES_DIR\u0026quot;/dmelanogaster.fna \u0026gt; \u0026quot;$REFERENCES_DIR\u0026quot;/wMel_dmelanogaster_combined.fna\n\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/wMel.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/025/GCF_000008025.1_ASM802v1/GCF_000008025.1_ASM802v1_genomic.gff.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/wMel.gff.gz\n\n## wOo + O. ochengi\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/wOo.fna.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/306/885/GCF_000306885.1_ASM30688v1/GCF_000306885.1_ASM30688v1_genomic.fna.gz\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/oochengi.fna.gz http://www.nematodes.org/downloads/959nematodegenomes/blast/db2/Onchocerca_ochengi_nuclear_assembly_nOo.2.0.fna.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/wOo.fna.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/oochengi.fna.gz\ncat \u0026quot;$REFERENCES_DIR\u0026quot;/wOo.fna \u0026quot;$REFERENCES_DIR\u0026quot;/oochengi.fna \u0026gt; \u0026quot;$REFERENCES_DIR\u0026quot;/wOo_oochengi_combined.fna\n\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/wOo.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/306/885/GCF_000306885.1_ASM30688v1/GCF_000306885.1_ASM30688v1_genomic.gff.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/wOo.gff.gz\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e## wBm + B. malayi\nwget -O \"$REFERENCES_DIR\"/wBm.fna.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_genomic.fna.gz\nwget -O \"$REFERENCES_DIR\"/bmalayi.fna.gz ftp://ftp.wormbase.org/pub/wormbase/releases/WS270/species/b_malayi/PRJNA10729/b_malayi.PRJNA10729.WS270.genomic.fa.gz\ngunzip \"$REFERENCES_DIR\"/wBm.fna.gz\ngunzip \"$REFERENCES_DIR\"/bmalayi.fna.gz\ncat \"$REFERENCES_DIR\"/wBm.fna \"$REFERENCES_DIR\"/bmalayi.fna \u0026gt; \"$REFERENCES_DIR\"/wBm_bmalayi_combined.fna\n\nwget -O \"$REFERENCES_DIR\"/wBm.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_genomic.gff.gz\ngunzip \"$REFERENCES_DIR\"/wBm.gff.gz\n\n## wDi + D. immitis\nwget -O \"$REFERENCES_DIR\"/wDi.fna.gz http://nematodes.org/downloads/959nematodegenomes/blast/db2/Dirofilaria_immitis_wolbachia_2.2.fna.gz\nwget -O \"$REFERENCES_DIR\"/dimmitis.fna.gz http://nematodes.org/downloads/959nematodegenomes/blast/db2/Dirofilaria_immitis_2.2.fna.gz\ngunzip \"$REFERENCES_DIR\"/wDi.fna.gz\ngunzip \"$REFERENCES_DIR\"/dimmitis.fna.gz\ncat \"$REFERENCES_DIR\"/wDi.fna \"$REFERENCES_DIR\"/dimmitis.fna \u0026gt; \"$REFERENCES_DIR\"/wDi_dimmitis_combined.fna\n\nwget -O \"$REFERENCES_DIR\"/wBm.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_genomic.gff.gz\ngunzip \"$REFERENCES_DIR\"/wBm.gff.gz\n\n## wMel + D. melanogaster\nwget -O \"$REFERENCES_DIR\"/wMel.fna.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/025/GCF_000008025.1_ASM802v1/GCF_000008025.1_ASM802v1_genomic.fna.gz\nwget -O \"$REFERENCES_DIR\"/dmelanogaster.fna.gz ftp://ftp.flybase.net/releases/FB2019_02/dmel_r6.27/fasta/dmel-all-chromosome-r6.27.fasta.gz\ngunzip \"$REFERENCES_DIR\"/wMel.fna.gz\ngunzip \"$REFERENCES_DIR\"/dmelanogaster.fna.gz\ncat \"$REFERENCES_DIR\"/wMel.fna \"$REFERENCES_DIR\"/dmelanogaster.fna \u0026gt; \"$REFERENCES_DIR\"/wMel_dmelanogaster_combined.fna\n\nwget -O \"$REFERENCES_DIR\"/wMel.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/025/GCF_000008025.1_ASM802v1/GCF_000008025.1_ASM802v1_genomic.gff.gz\ngunzip \"$REFERENCES_DIR\"/wMel.gff.gz\n\n## wOo + O. ochengi\nwget -O \"$REFERENCES_DIR\"/wOo.fna.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/306/885/GCF_000306885.1_ASM30688v1/GCF_000306885.1_ASM30688v1_genomic.fna.gz\nwget -O \"$REFERENCES_DIR\"/oochengi.fna.gz http://www.nematodes.org/downloads/959nematodegenomes/blast/db2/Onchocerca_ochengi_nuclear_assembly_nOo.2.0.fna.gz\ngunzip \"$REFERENCES_DIR\"/wOo.fna.gz\ngunzip \"$REFERENCES_DIR\"/oochengi.fna.gz\ncat \"$REFERENCES_DIR\"/wOo.fna \"$REFERENCES_DIR\"/oochengi.fna \u0026gt; \"$REFERENCES_DIR\"/wOo_oochengi_combined.fna\n\nwget -O \"$REFERENCES_DIR\"/wOo.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/306/885/GCF_000306885.1_ASM30688v1/GCF_000306885.1_ASM30688v1_genomic.gff.gz\ngunzip \"$REFERENCES_DIR\"/wOo.gff.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDownload FASTQ files (2019-05-30) \u003ca name=\"user-content-bashanalysis_getfastq\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-download-fastq-files-2019-05-30-\" class=\"anchor\" aria-label=\"Permalink: Download FASTQ files (2019-05-30) \" href=\"#download-fastq-files-2019-05-30-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAll FASTQ files from 6 of the studies were downloaded from the SRA. Because the Wolbachia reads were not included in the FASTQ files uploaded for the Luck 2015 study, NEB uploaded them onto the IGS FTP site (ftp://nsflgt@ftp.igs.umaryland.edu; password: drosophila) for download to this directory: \"$READS_DIR\"/wDi_luck_2015\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSRA Datasets \u003ca name=\"user-content-bashanalysis_getfastq_sra\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-sra-datasets-\" class=\"anchor\" aria-label=\"Permalink: SRA Datasets \" href=\"#sra-datasets-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## wOo Darby 2012\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wOo_darby_2012_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wOo_darby_2012\n\n## wMel Darby 2014\nSRR_ID_LIST=/wMel_darby_2014_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_darby_2014\n\n## wDi Luck 2014\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wDi_luck_2014_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2014\n\n## wMel Gutzwiller 2015\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wMel_gutzwiller_2015_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_gutzwiller_2015\n\n## wBm Grote 2017\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wBm_grote_2017_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_grote_2017\n\n## wBm Chung 2019\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wBm_chung_2019_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_chung_2019\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e## wOo Darby 2012\nSRR_ID_LIST=\"$INPUTS_DIR\"/wOo_darby_2012_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wOo_darby_2012\n\n## wMel Darby 2014\nSRR_ID_LIST=/wMel_darby_2014_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wMel_darby_2014\n\n## wDi Luck 2014\nSRR_ID_LIST=\"$INPUTS_DIR\"/wDi_luck_2014_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wDi_luck_2014\n\n## wMel Gutzwiller 2015\nSRR_ID_LIST=\"$INPUTS_DIR\"/wMel_gutzwiller_2015_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wMel_gutzwiller_2015\n\n## wBm Grote 2017\nSRR_ID_LIST=\"$INPUTS_DIR\"/wBm_grote_2017_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wBm_grote_2017\n\n## wBm Chung 2019\nSRR_ID_LIST=\"$INPUTS_DIR\"/wBm_chung_2019_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wBm_chung_2019\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-1\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"while read Line\ndo\n\nSRR_ID=\u0026quot;$(echo $Line | awk -F \u0026quot;\\t\u0026quot; '{print $1}')\u0026quot;\n\nqsub -P jdhotopp-lab -l mem_free=2G -N fastq_dump -wd \u0026quot;$OUTPUT_DIR\u0026quot; -b y \u0026quot;$SRATOOLKIT_BIN_DIR\u0026quot;/fastq-dump --split-files \u0026quot;$SRR_ID\u0026quot; -O \u0026quot;$OUTPUT_DIR\u0026quot;\n\ndone \u0026lt; \u0026quot;$SRR_ID_LIST\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003ewhile read Line\ndo\n\nSRR_ID=\"$(echo $Line | awk -F \"\\t\" '{print $1}')\"\n\nqsub -P jdhotopp-lab -l mem_free=2G -N fastq_dump -wd \"$OUTPUT_DIR\" -b y \"$SRATOOLKIT_BIN_DIR\"/fastq-dump --split-files \"$SRR_ID\" -O \"$OUTPUT_DIR\"\n\ndone \u0026lt; \"$SRR_ID_LIST\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eLuck et al 2015 Dataset \u003ca name=\"user-content-bashanalysis_getfastq_luck2015\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-luck-et-al-2015-dataset-\" class=\"anchor\" aria-label=\"Permalink: Luck et al 2015 Dataset \" href=\"#luck-et-al-2015-dataset-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-1\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## wDi Luck 2015\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wDi_luck_2015_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e## wDi Luck 2015\nSRR_ID_LIST=\"$INPUTS_DIR\"/wDi_luck_2015_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wDi_luck_2015\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-2\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"zcat \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/Galaxy1-[Di_Female_Intestine_2].fastqsanger.zip \u0026gt; \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/SRR1974179.fastq\nzcat \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/Galaxy2-[Di_Female_Intestine_1].fastqsanger.zip \u0026gt; \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/SRR1974178.fastq\nzcat \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/Galaxy3-[Di_Female_Body_Wall_2].fastqsanger.zip \u0026gt; \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/SRR1974175.fastq\nzcat \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/Galaxy4-[Di_Female_Uterus_1].fastqsanger.zip \u0026gt; \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/SRR1974180.fastq\nzcat \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/Galaxy5-[Di_Female_Head_2].fastqsanger.zip \u0026gt; \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/SRR1974177.fastq\nzcat \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/Galaxy6-[Di_Female_Uterus_2].fastqsanger.zip \u0026gt; \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/SRR1974181.fastq\nzcat \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/Galaxy7-[Di_Female_Head_1].fastqsanger.zip \u0026gt; \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/SRR1974176.fastq\nzcat \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/Galaxy8-[Di_Female_Body_Wall_1].fastqsanger.zip \u0026gt; \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/SRR1974174.fastq\nzcat \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/Galaxy9-[Di_Male_Testes].fastq.zip \u0026gt; \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/SRR1974185.fastq\nzcat \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/Galaxy10-[Di_Male_Body_Wall_1].fastq.zip \u0026gt; \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/SRR1974182.fastq\nzcat \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/Galaxy11-[Di_Male_Body_Wall_2] (1).fastq.zip \u0026gt; \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/SRR1974183.fastq\nzcat \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/Galaxy12-[Di_Male_Intestine] (1).fastq.zip \u0026gt; \u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/SRR1974184.fastq\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003ezcat \"$READS_DIR\"/wDi_luck_2015/Galaxy1-[Di_Female_Intestine_2].fastqsanger.zip \u0026gt; \"$READS_DIR\"/wDi_luck_2015/SRR1974179.fastq\nzcat \"$READS_DIR\"/wDi_luck_2015/Galaxy2-[Di_Female_Intestine_1].fastqsanger.zip \u0026gt; \"$READS_DIR\"/wDi_luck_2015/SRR1974178.fastq\nzcat \"$READS_DIR\"/wDi_luck_2015/Galaxy3-[Di_Female_Body_Wall_2].fastqsanger.zip \u0026gt; \"$READS_DIR\"/wDi_luck_2015/SRR1974175.fastq\nzcat \"$READS_DIR\"/wDi_luck_2015/Galaxy4-[Di_Female_Uterus_1].fastqsanger.zip \u0026gt; \"$READS_DIR\"/wDi_luck_2015/SRR1974180.fastq\nzcat \"$READS_DIR\"/wDi_luck_2015/Galaxy5-[Di_Female_Head_2].fastqsanger.zip \u0026gt; \"$READS_DIR\"/wDi_luck_2015/SRR1974177.fastq\nzcat \"$READS_DIR\"/wDi_luck_2015/Galaxy6-[Di_Female_Uterus_2].fastqsanger.zip \u0026gt; \"$READS_DIR\"/wDi_luck_2015/SRR1974181.fastq\nzcat \"$READS_DIR\"/wDi_luck_2015/Galaxy7-[Di_Female_Head_1].fastqsanger.zip \u0026gt; \"$READS_DIR\"/wDi_luck_2015/SRR1974176.fastq\nzcat \"$READS_DIR\"/wDi_luck_2015/Galaxy8-[Di_Female_Body_Wall_1].fastqsanger.zip \u0026gt; \"$READS_DIR\"/wDi_luck_2015/SRR1974174.fastq\nzcat \"$READS_DIR\"/wDi_luck_2015/Galaxy9-[Di_Male_Testes].fastq.zip \u0026gt; \"$READS_DIR\"/wDi_luck_2015/SRR1974185.fastq\nzcat \"$READS_DIR\"/wDi_luck_2015/Galaxy10-[Di_Male_Body_Wall_1].fastq.zip \u0026gt; \"$READS_DIR\"/wDi_luck_2015/SRR1974182.fastq\nzcat \"$READS_DIR\"/wDi_luck_2015/Galaxy11-[Di_Male_Body_Wall_2] (1).fastq.zip \u0026gt; \"$READS_DIR\"/wDi_luck_2015/SRR1974183.fastq\nzcat \"$READS_DIR\"/wDi_luck_2015/Galaxy12-[Di_Male_Intestine] (1).fastq.zip \u0026gt; \"$READS_DIR\"/wDi_luck_2015/SRR1974184.fastq\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAlign FASTQ files to combined host and Wolbachia reference while allowing for splicing (2019-05-30)  \u003ca name=\"user-content-bashanalysis_align\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-align-fastq-files-to-combined-host-and-wolbachia-reference-while-allowing-for-splicing-2019-05-30--\" class=\"anchor\" aria-label=\"Permalink: Align FASTQ files to combined host and Wolbachia reference while allowing for splicing (2019-05-30)  \" href=\"#align-fastq-files-to-combined-host-and-wolbachia-reference-while-allowing-for-splicing-2019-05-30--\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSingle-End FASTQs \u003ca name=\"user-content-bashanalysis_align_se\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-single-end-fastqs-\" class=\"anchor\" aria-label=\"Permalink: Single-End FASTQs \" href=\"#single-end-fastqs-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-2\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"THREADS=4\n\n## wOo Darby 2012\nREF_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wOo_oochengi_combined.fna\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wOo_oochengi_2012_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wOo_oochengi_2012/bam\nSTRANDEDNESS=\u0026quot;\u0026quot;\nPREFIX=wOo_darby_2012\n\n## wDi Luck 2014\nREF_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wDi_dimmitis_combined.fna\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wDi_luck_2014_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2014/bam\nSTRANDEDNESS=\u0026quot;\u0026quot;\nPREFIX=wDi_luck_2014\n\n## wDi Luck 2015\nREF_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wDi_dimmitis_combined.fna\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wDi_luck_2015_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/bam\nSTRANDEDNESS=\u0026quot;\u0026quot;\nPREFIX=wDi_luck_2015\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eTHREADS=4\n\n## wOo Darby 2012\nREF_FNA=\"$REFERENCES_DIR\"/wOo_oochengi_combined.fna\nSRR_ID_LIST=\"$INPUTS_DIR\"/wOo_oochengi_2012_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wOo_oochengi_2012/bam\nSTRANDEDNESS=\"\"\nPREFIX=wOo_darby_2012\n\n## wDi Luck 2014\nREF_FNA=\"$REFERENCES_DIR\"/wDi_dimmitis_combined.fna\nSRR_ID_LIST=\"$INPUTS_DIR\"/wDi_luck_2014_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wDi_luck_2014/bam\nSTRANDEDNESS=\"\"\nPREFIX=wDi_luck_2014\n\n## wDi Luck 2015\nREF_FNA=\"$REFERENCES_DIR\"/wDi_dimmitis_combined.fna\nSRR_ID_LIST=\"$INPUTS_DIR\"/wDi_luck_2015_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wDi_luck_2015/bam\nSTRANDEDNESS=\"\"\nPREFIX=wDi_luck_2015\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-3\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-3\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"while read SRR\ndo\n\techo \u0026quot;\u0026quot;$HISAT2_BIN_DIR\u0026quot;/hisat2 -p \u0026quot;$THREADS\u0026quot; -x \u0026quot;$REF_FNA\u0026quot; \u0026quot;$STRANDEDNESS\u0026quot; -U \u0026quot;$READS_DIR\u0026quot;/\u0026quot;$PREFIX\u0026quot;/\u0026quot;$SRR\u0026quot;_1.fastq | \u0026quot;$SAMTOOLS_BIN_DIR\u0026quot;/samtools view -bhSo \u0026quot;$OUTPUT_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;.bam -\u0026quot; | qsub -q threaded.q  -pe thread \u0026quot;$THREADS\u0026quot; -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd \u0026quot;$OUTPUT_DIR\u0026quot;\ndone \u0026lt; \u0026quot;$SRR_ID_LIST\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003ewhile read SRR\ndo\n\techo \"\"$HISAT2_BIN_DIR\"/hisat2 -p \"$THREADS\" -x \"$REF_FNA\" \"$STRANDEDNESS\" -U \"$READS_DIR\"/\"$PREFIX\"/\"$SRR\"_1.fastq | \"$SAMTOOLS_BIN_DIR\"/samtools view -bhSo \"$OUTPUT_DIR\"/\"$SRR\".bam -\" | qsub -q threaded.q  -pe thread \"$THREADS\" -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd \"$OUTPUT_DIR\"\ndone \u0026lt; \"$SRR_ID_LIST\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePaired-End FASTQs \u003ca name=\"user-content-bashanalysis_align_pe\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-paired-end-fastqs-\" class=\"anchor\" aria-label=\"Permalink: Paired-End FASTQs \" href=\"#paired-end-fastqs-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-3\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-3\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"THREADS=4\n\n## wMel Darby 2014\nREF_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wMel_dmelanogaster_combined.fna\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wMel_darby_2014_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_darby_2014/bam\nSTRANDEDNESS=\u0026quot;--rna-strandness FR\u0026quot;\nPREFIX=wMel_darby_2014\n\n## wMel Gutzwiller 2015\nREF_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wMel_dmelanogaster_combined.fna\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wMel_gutzwiller_2015_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_gutzwiller_2015/bam\nSTRANDEDNESS=\u0026quot;--rna-strandness RF\u0026quot;\nPREFIX=wMel_gutzwiller_2015\n\n## wBm Grote 2017\nREF_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wBm_bmalayi_combined.fna\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wBm_grote_2017_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_grote_2017/bam\nSTRANDEDNESS=\u0026quot;\u0026quot;\nPREFIX=wBm_grote_2017\n\n## wBm Chung 2019\nREF_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wBm_bmalayi_combined.fna\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wBm_chung_2019_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_chung_2019/bam\nSTRANDEDNESS=\u0026quot;--rna-strandness RF\u0026quot;\nPREFIX=wBm_chung_2019\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eTHREADS=4\n\n## wMel Darby 2014\nREF_FNA=\"$REFERENCES_DIR\"/wMel_dmelanogaster_combined.fna\nSRR_ID_LIST=\"$INPUTS_DIR\"/wMel_darby_2014_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wMel_darby_2014/bam\nSTRANDEDNESS=\"--rna-strandness FR\"\nPREFIX=wMel_darby_2014\n\n## wMel Gutzwiller 2015\nREF_FNA=\"$REFERENCES_DIR\"/wMel_dmelanogaster_combined.fna\nSRR_ID_LIST=\"$INPUTS_DIR\"/wMel_gutzwiller_2015_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wMel_gutzwiller_2015/bam\nSTRANDEDNESS=\"--rna-strandness RF\"\nPREFIX=wMel_gutzwiller_2015\n\n## wBm Grote 2017\nREF_FNA=\"$REFERENCES_DIR\"/wBm_bmalayi_combined.fna\nSRR_ID_LIST=\"$INPUTS_DIR\"/wBm_grote_2017_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wBm_grote_2017/bam\nSTRANDEDNESS=\"\"\nPREFIX=wBm_grote_2017\n\n## wBm Chung 2019\nREF_FNA=\"$REFERENCES_DIR\"/wBm_bmalayi_combined.fna\nSRR_ID_LIST=\"$INPUTS_DIR\"/wBm_chung_2019_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wBm_chung_2019/bam\nSTRANDEDNESS=\"--rna-strandness RF\"\nPREFIX=wBm_chung_2019\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-4\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-4\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"while read srr\ndo\n\techo \u0026quot;\u0026quot;$HISAT2_BIN_DIR\u0026quot;/hisat2 -p \u0026quot;$THREADS\u0026quot; -x \u0026quot;$REF_FNA\u0026quot; \u0026quot;$STRANDEDNESS\u0026quot; -1 \u0026quot;$READS_DIR\u0026quot;/\u0026quot;$PREFIX\u0026quot;/\u0026quot;$SRR\u0026quot;_1.fastq  -2 \u0026quot;$READS_DIR\u0026quot;/\u0026quot;$PREFIX\u0026quot;/\u0026quot;$SRR\u0026quot;_2.fastq | \u0026quot;$SAMTOOLS_BIN_DIR\u0026quot;/samtools view -bhSo \u0026quot;$OUTPUT_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;.bam -\u0026quot; | qsub -q threaded.q  -pe thread \u0026quot;$THREADS\u0026quot; -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd \u0026quot;$OUTPUT_DIR\u0026quot;\ndone \u0026lt; \u0026quot;$SRR_ID_LIST\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003ewhile read srr\ndo\n\techo \"\"$HISAT2_BIN_DIR\"/hisat2 -p \"$THREADS\" -x \"$REF_FNA\" \"$STRANDEDNESS\" -1 \"$READS_DIR\"/\"$PREFIX\"/\"$SRR\"_1.fastq  -2 \"$READS_DIR\"/\"$PREFIX\"/\"$SRR\"_2.fastq | \"$SAMTOOLS_BIN_DIR\"/samtools view -bhSo \"$OUTPUT_DIR\"/\"$SRR\".bam -\" | qsub -q threaded.q  -pe thread \"$THREADS\" -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd \"$OUTPUT_DIR\"\ndone \u0026lt; \"$SRR_ID_LIST\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCreate a list of Wolbachia-mapping reads in all BAM files (2019-05-30) \u003ca name=\"user-content-bashanalysis_wolbachiareads\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-create-a-list-of-wolbachia-mapping-reads-in-all-bam-files-2019-05-30-\" class=\"anchor\" aria-label=\"Permalink: Create a list of Wolbachia-mapping reads in all BAM files (2019-05-30) \" href=\"#create-a-list-of-wolbachia-mapping-reads-in-all-bam-files-2019-05-30-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-4\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-4\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## wOo Darby 2012\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wOo_darby_2012/bam\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wOo_darby_2012_srr.list\nCONTIG=NC_018267.1\n\n## wMel Darby 2014\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_darby_2014/bam\nSRR_ID_LIST=/wMel_darby_2014_srr.list\nCONTIG=NC_002978.6\n\n## wDi Luck 2014\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2014/bam\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wDi_luck_2014_srr.list\nCONTIG=wDi22.scaf1\n\n## wDi Luck 2015\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/bam\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wDi_luck_2015_srr.list\nCONTIG=wDi22.scaf1\n\n## wMel Gutzwiller 2015\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_gutzwiller_2015/bam\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wMel_gutzwiller_2015_srr.list\nCONTIG=NC_002978.6\n\n## wBm Grote 2017\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_grote_2017/bam\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wBm_grote_2017_srr.list\nCONTIG=NC_006833.1\n\n## wBm Chung 2019\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_chung_2019/bam\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wBm_chung_2019_srr.list\nCONTIG=NC_006833.1\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e## wOo Darby 2012\nBAM_DIR=\"$READS_DIR\"/wOo_darby_2012/bam\nSRR_ID_LIST=\"$INPUTS_DIR\"/wOo_darby_2012_srr.list\nCONTIG=NC_018267.1\n\n## wMel Darby 2014\nBAM_DIR=\"$READS_DIR\"/wMel_darby_2014/bam\nSRR_ID_LIST=/wMel_darby_2014_srr.list\nCONTIG=NC_002978.6\n\n## wDi Luck 2014\nBAM_DIR=\"$READS_DIR\"/wDi_luck_2014/bam\nSRR_ID_LIST=\"$INPUTS_DIR\"/wDi_luck_2014_srr.list\nCONTIG=wDi22.scaf1\n\n## wDi Luck 2015\nBAM_DIR=\"$READS_DIR\"/wDi_luck_2015/bam\nSRR_ID_LIST=\"$INPUTS_DIR\"/wDi_luck_2015_srr.list\nCONTIG=wDi22.scaf1\n\n## wMel Gutzwiller 2015\nBAM_DIR=\"$READS_DIR\"/wMel_gutzwiller_2015/bam\nSRR_ID_LIST=\"$INPUTS_DIR\"/wMel_gutzwiller_2015_srr.list\nCONTIG=NC_002978.6\n\n## wBm Grote 2017\nBAM_DIR=\"$READS_DIR\"/wBm_grote_2017/bam\nSRR_ID_LIST=\"$INPUTS_DIR\"/wBm_grote_2017_srr.list\nCONTIG=NC_006833.1\n\n## wBm Chung 2019\nBAM_DIR=\"$READS_DIR\"/wBm_chung_2019/bam\nSRR_ID_LIST=\"$INPUTS_DIR\"/wBm_chung_2019_srr.list\nCONTIG=NC_006833.1\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-5\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-5\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"while read SRR\ndo\n\t\u0026quot;#SAMTOOLS_BIN_DIR\u0026quot;/samtools view \u0026quot;$BAM_DIR\u0026quot;/\u0026quot;SRR\u0026quot;.bam | grep \u0026quot;$PREFIX\u0026quot; | awk '{print $1}' | sort -n | uniq \u0026gt; \u0026quot;$BAM_DIR\u0026quot;/\u0026quot;SRR\u0026quot;.target_reads.list\ndone \u0026lt; \u0026quot;$SRR_ID_LIST\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003ewhile read SRR\ndo\n\t\"#SAMTOOLS_BIN_DIR\"/samtools view \"$BAM_DIR\"/\"SRR\".bam | grep \"$PREFIX\" | awk '{print $1}' | sort -n | uniq \u0026gt; \"$BAM_DIR\"/\"SRR\".target_reads.list\ndone \u0026lt; \"$SRR_ID_LIST\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCreate subsets of all FASTQ files that contain only Wolbachia-mapping reads (2019-05-30) \u003ca name=\"user-content-bashanalysis_subsetfastq\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-create-subsets-of-all-fastq-files-that-contain-only-wolbachia-mapping-reads-2019-05-30-\" class=\"anchor\" aria-label=\"Permalink: Create subsets of all FASTQ files that contain only Wolbachia-mapping reads (2019-05-30) \" href=\"#create-subsets-of-all-fastq-files-that-contain-only-wolbachia-mapping-reads-2019-05-30-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSingle-End FASTQs \u003ca name=\"user-content-bashanalysis_subsetfastq_se\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-single-end-fastqs--1\" class=\"anchor\" aria-label=\"Permalink: Single-End FASTQs \" href=\"#single-end-fastqs--1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-5\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-5\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## wOo Darby 2012\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wOo_oochengi_2012_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wOo_oochengi_2012\n\n## wDi Luck 2014\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wDi_luck_2014_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;//wDi_luck_2014\n\n## wDi Luck 2015\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wDi_luck_2015_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e## wOo Darby 2012\nSRR_ID_LIST=\"$INPUTS_DIR\"/wOo_oochengi_2012_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wOo_oochengi_2012\n\n## wDi Luck 2014\nSRR_ID_LIST=\"$INPUTS_DIR\"/wDi_luck_2014_srr.list\nOUTPUT_DIR=\"$READS_DIR\"//wDi_luck_2014\n\n## wDi Luck 2015\nSRR_ID_LIST=\"$INPUTS_DIR\"/wDi_luck_2015_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wDi_luck_2015\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-6\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-6\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"while read SRR\ndo\n\techo -e \u0026quot;\u0026quot;$SEQTK_BIN_DIR\u0026quot;/seqtk subseq \u0026quot;$OUTPUT_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;_1.fastq \u0026quot;$OUTPUT_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;.target_reads.list \u0026gt; \u0026quot;$OUTPUT_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;_1.subset.fastq\u0026quot; | qsub -P jdhotopp-lab -l mem_free=2G -N seqtk -wd \u0026quot;$OUTPUT_DIR\u0026quot;\ndone \u0026lt; \u0026quot;$SRR_ID_LIST\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003ewhile read SRR\ndo\n\techo -e \"\"$SEQTK_BIN_DIR\"/seqtk subseq \"$OUTPUT_DIR\"/\"$SRR\"_1.fastq \"$OUTPUT_DIR\"/\"$SRR\".target_reads.list \u0026gt; \"$OUTPUT_DIR\"/\"$SRR\"_1.subset.fastq\" | qsub -P jdhotopp-lab -l mem_free=2G -N seqtk -wd \"$OUTPUT_DIR\"\ndone \u0026lt; \"$SRR_ID_LIST\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePaired-End FASTQs \u003ca name=\"user-content-bashanalysis_subsetfastq_pe\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-paired-end-fastqs--1\" class=\"anchor\" aria-label=\"Permalink: Paired-End FASTQs \" href=\"#paired-end-fastqs--1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-6\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-6\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## wMel Darby 2014\nSRR_ID_LIST=/wMel_darby_2014_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_darby_2014\n\n## wMel Gutzwiller 2015\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wMel_gutzwiller_2015_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_gutzwiller_2015\n\n## wBm Grote 2017\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wBm_grote_2017_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_grote_2017\n\n## wBm Chung 2019\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wBm_chung_2019_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_chung_2019\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e## wMel Darby 2014\nSRR_ID_LIST=/wMel_darby_2014_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wMel_darby_2014\n\n## wMel Gutzwiller 2015\nSRR_ID_LIST=\"$INPUTS_DIR\"/wMel_gutzwiller_2015_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wMel_gutzwiller_2015\n\n## wBm Grote 2017\nSRR_ID_LIST=\"$INPUTS_DIR\"/wBm_grote_2017_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wBm_grote_2017\n\n## wBm Chung 2019\nSRR_ID_LIST=\"$INPUTS_DIR\"/wBm_chung_2019_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wBm_chung_2019\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-7\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-7\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"while read SRR\ndo\n\n\techo -e \u0026quot;\u0026quot;$SEQTK_BIN_DIR\u0026quot;/seqtk subseq \u0026quot;$OUTPUT_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;_1.fastq \u0026quot;$OUTPUT_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;.target_reads.list \u0026gt; \u0026quot;$OUTPUT_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;_1.subset.fastq\u0026quot; | qsub -P jdhotopp-lab -l mem_free=2G -N seqtk -wd \u0026quot;$OUTPUT_DIR\u0026quot;\n\n\techo -e \u0026quot;\u0026quot;$SEQTK_BIN_DIR\u0026quot;/seqtk subseq \u0026quot;$OUTPUT_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;_2.fastq \u0026quot;$OUTPUT_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;.target_reads.list \u0026gt; \u0026quot;$OUTPUT_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;_2.subset.fastq\u0026quot; | qsub -P jdhotopp-lab -l mem_free=2G -N seqtk -wd \u0026quot;$OUTPUT_DIR\u0026quot;\n\ndone \u0026lt; \u0026quot;$SRR_ID_LIST\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003ewhile read SRR\ndo\n\n\techo -e \"\"$SEQTK_BIN_DIR\"/seqtk subseq \"$OUTPUT_DIR\"/\"$SRR\"_1.fastq \"$OUTPUT_DIR\"/\"$SRR\".target_reads.list \u0026gt; \"$OUTPUT_DIR\"/\"$SRR\"_1.subset.fastq\" | qsub -P jdhotopp-lab -l mem_free=2G -N seqtk -wd \"$OUTPUT_DIR\"\n\n\techo -e \"\"$SEQTK_BIN_DIR\"/seqtk subseq \"$OUTPUT_DIR\"/\"$SRR\"_2.fastq \"$OUTPUT_DIR\"/\"$SRR\".target_reads.list \u0026gt; \"$OUTPUT_DIR\"/\"$SRR\"_2.subset.fastq\" | qsub -P jdhotopp-lab -l mem_free=2G -N seqtk -wd \"$OUTPUT_DIR\"\n\ndone \u0026lt; \"$SRR_ID_LIST\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAlign FASTQ files containing only Wolbachia-mapping reads to combined host and Wolbachia reference while disallowing splicing (2019-06-01) \u003ca name=\"user-content-bashanalysis_realign\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-align-fastq-files-containing-only-wolbachia-mapping-reads-to-combined-host-and-wolbachia-reference-while-disallowing-splicing-2019-06-01-\" class=\"anchor\" aria-label=\"Permalink: Align FASTQ files containing only Wolbachia-mapping reads to combined host and Wolbachia reference while disallowing splicing (2019-06-01) \" href=\"#align-fastq-files-containing-only-wolbachia-mapping-reads-to-combined-host-and-wolbachia-reference-while-disallowing-splicing-2019-06-01-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSingle-End FASTQs \u003ca name=\"user-content-bashanalysis_realign_se\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-single-end-fastqs--2\" class=\"anchor\" aria-label=\"Permalink: Single-End FASTQs \" href=\"#single-end-fastqs--2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-7\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-7\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"THREADS=4\n\n## wOo Darby 2012\nREF_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wOo_oochengi_combined.fna\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wOo_oochengi_2012_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wOo_oochengi_2012/bam\nSTRANDEDNESS=\u0026quot;\u0026quot;\nPREFIX=wOo_darby_2012\n\n## wDi Luck 2014\nREF_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wDi_dimmitis_combined.fna\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wDi_luck_2014_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2014/bam\nSTRANDEDNESS=\u0026quot;\u0026quot;\nPREFIX=wDi_luck_2014\n\n## wDi Luck 2015\nREF_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wDi_dimmitis_combined.fna\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wDi_luck_2015_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/bam\nSTRANDEDNESS=\u0026quot;\u0026quot;\nPREFIX=wDi_luck_2015\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eTHREADS=4\n\n## wOo Darby 2012\nREF_FNA=\"$REFERENCES_DIR\"/wOo_oochengi_combined.fna\nSRR_ID_LIST=\"$INPUTS_DIR\"/wOo_oochengi_2012_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wOo_oochengi_2012/bam\nSTRANDEDNESS=\"\"\nPREFIX=wOo_darby_2012\n\n## wDi Luck 2014\nREF_FNA=\"$REFERENCES_DIR\"/wDi_dimmitis_combined.fna\nSRR_ID_LIST=\"$INPUTS_DIR\"/wDi_luck_2014_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wDi_luck_2014/bam\nSTRANDEDNESS=\"\"\nPREFIX=wDi_luck_2014\n\n## wDi Luck 2015\nREF_FNA=\"$REFERENCES_DIR\"/wDi_dimmitis_combined.fna\nSRR_ID_LIST=\"$INPUTS_DIR\"/wDi_luck_2015_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wDi_luck_2015/bam\nSTRANDEDNESS=\"\"\nPREFIX=wDi_luck_2015\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-8\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-8\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"while read SRR\ndo\n\techo -e \u0026quot;\u0026quot;$HISAT2_BIN_DIR\u0026quot;/hisat2 -p \u0026quot;$THREADS\u0026quot; -x \u0026quot;$REF_FNA\u0026quot; \u0026quot;$STRANDEDNESS\u0026quot; -U \u0026quot;$READS_DIR\u0026quot;/\u0026quot;$PREFIX\u0026quot;/\u0026quot;$SRR\u0026quot;_1.subset.fastq --no-spliced-alignment -X 1000 | \u0026quot;$SAMTOOLS_BIN_DIR\u0026quot;/samtools view -bhSo \u0026quot;$OUTPUT_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;.bam -\u0026quot; | qsub -q threaded.q  -pe thread \u0026quot;$THREADS\u0026quot; -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd \u0026quot;$OUTPUT_DIR\u0026quot;\ndone \u0026lt; \u0026quot;$SRR_ID_LIST\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003ewhile read SRR\ndo\n\techo -e \"\"$HISAT2_BIN_DIR\"/hisat2 -p \"$THREADS\" -x \"$REF_FNA\" \"$STRANDEDNESS\" -U \"$READS_DIR\"/\"$PREFIX\"/\"$SRR\"_1.subset.fastq --no-spliced-alignment -X 1000 | \"$SAMTOOLS_BIN_DIR\"/samtools view -bhSo \"$OUTPUT_DIR\"/\"$SRR\".bam -\" | qsub -q threaded.q  -pe thread \"$THREADS\" -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd \"$OUTPUT_DIR\"\ndone \u0026lt; \"$SRR_ID_LIST\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePaired-End FASTQs \u003ca name=\"user-content-bashanalysis_realign_pe\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-paired-end-fastqs--2\" class=\"anchor\" aria-label=\"Permalink: Paired-End FASTQs \" href=\"#paired-end-fastqs--2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-8\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-8\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"THREADS=4\n\n## wMel Darby 2014\nREF_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wMel_dmelanogaster_combined.fna\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wMel_darby_2014_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_darby_2014/bam\nSTRANDEDNESS=\u0026quot;--rna-strandness FR\u0026quot;\nPREFIX=wMel_darby_2014\n\n## wMel Gutzwiller 2015\nREF_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wMel_dmelanogaster_combined.fna\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wMel_gutzwiller_2015_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_gutzwiller_2015/bam\nSTRANDEDNESS=\u0026quot;--rna-strandness RF\u0026quot;\nPREFIX=wMel_gutzwiller_2015\n\n## wBm Grote 2017\nREF_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wBm_bmalayi_combined.fna\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wBm_grote_2017_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_grote_2017/bam\nSTRANDEDNESS=\u0026quot;\u0026quot;\nPREFIX=wBm_grote_2017\n\n## wBm Chung 2019\nREF_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wBm_bmalayi_combined.fna\nSRR_ID_LIST=\u0026quot;$INPUTS_DIR\u0026quot;/wBm_chung_2019_srr.list\nOUTPUT_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_chung_2019/bam\nSTRANDEDNESS=\u0026quot;--rna-strandness RF\u0026quot;\nPREFIX=wBm_chung_2019\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eTHREADS=4\n\n## wMel Darby 2014\nREF_FNA=\"$REFERENCES_DIR\"/wMel_dmelanogaster_combined.fna\nSRR_ID_LIST=\"$INPUTS_DIR\"/wMel_darby_2014_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wMel_darby_2014/bam\nSTRANDEDNESS=\"--rna-strandness FR\"\nPREFIX=wMel_darby_2014\n\n## wMel Gutzwiller 2015\nREF_FNA=\"$REFERENCES_DIR\"/wMel_dmelanogaster_combined.fna\nSRR_ID_LIST=\"$INPUTS_DIR\"/wMel_gutzwiller_2015_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wMel_gutzwiller_2015/bam\nSTRANDEDNESS=\"--rna-strandness RF\"\nPREFIX=wMel_gutzwiller_2015\n\n## wBm Grote 2017\nREF_FNA=\"$REFERENCES_DIR\"/wBm_bmalayi_combined.fna\nSRR_ID_LIST=\"$INPUTS_DIR\"/wBm_grote_2017_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wBm_grote_2017/bam\nSTRANDEDNESS=\"\"\nPREFIX=wBm_grote_2017\n\n## wBm Chung 2019\nREF_FNA=\"$REFERENCES_DIR\"/wBm_bmalayi_combined.fna\nSRR_ID_LIST=\"$INPUTS_DIR\"/wBm_chung_2019_srr.list\nOUTPUT_DIR=\"$READS_DIR\"/wBm_chung_2019/bam\nSTRANDEDNESS=\"--rna-strandness RF\"\nPREFIX=wBm_chung_2019\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-9\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-9\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"while read SRR\ndo\n\techo -e \u0026quot;\u0026quot;$HISAT2_BIN_DIR\u0026quot;/hisat2 -p \u0026quot;$THREADS\u0026quot; -x \u0026quot;$REF_FNA\u0026quot; \u0026quot;$STRANDEDNESS\u0026quot; -1 \u0026quot;$READS_DIR\u0026quot;/\u0026quot;$PREFIX\u0026quot;/\u0026quot;$SRR\u0026quot;_1.subset.fastq  -2 \u0026quot;$READS_DIR\u0026quot;/\u0026quot;$PREFIX\u0026quot;/\u0026quot;$SRR\u0026quot;_2.subset.fastq --no-spliced-alignment -X 1000 | \u0026quot;$SAMTOOLS_BIN_DIR\u0026quot;/samtools view -bhSo \u0026quot;$OUTPUT_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;.bam -\u0026quot; | qsub -q threaded.q  -pe thread \u0026quot;$THREADS\u0026quot; -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd \u0026quot;$OUTPUT_DIR\u0026quot;\ndone \u0026lt; \u0026quot;$SRR_ID_LIST\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003ewhile read SRR\ndo\n\techo -e \"\"$HISAT2_BIN_DIR\"/hisat2 -p \"$THREADS\" -x \"$REF_FNA\" \"$STRANDEDNESS\" -1 \"$READS_DIR\"/\"$PREFIX\"/\"$SRR\"_1.subset.fastq  -2 \"$READS_DIR\"/\"$PREFIX\"/\"$SRR\"_2.subset.fastq --no-spliced-alignment -X 1000 | \"$SAMTOOLS_BIN_DIR\"/samtools view -bhSo \"$OUTPUT_DIR\"/\"$SRR\".bam -\" | qsub -q threaded.q  -pe thread \"$THREADS\" -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd \"$OUTPUT_DIR\"\ndone \u0026lt; \"$SRR_ID_LIST\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSort BAM files (2019-06-01) \u003ca name=\"user-content-bashanalysis_sortbam\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-sort-bam-files-2019-06-01-\" class=\"anchor\" aria-label=\"Permalink: Sort BAM files (2019-06-01) \" href=\"#sort-bam-files-2019-06-01-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-9\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-9\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"THREADS=4\n\n## wOo Darby 2012\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wOo_darby_2012/bam\n\n## wMel Darby 2014\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_darby_2014/bam\n\n## wDi Luck 2014\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2014/bam\n\n## wDi Luck 2015\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/bam\n\n## wMel Gutzwiller 2015\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_gutzwiller_2015/bam\n\n## wBm Grote 2017\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_grote_2017/bam\n\n## wBm Chung 2019\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_chung_2019/bam\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eTHREADS=4\n\n## wOo Darby 2012\nBAM_DIR=\"$READS_DIR\"/wOo_darby_2012/bam\n\n## wMel Darby 2014\nBAM_DIR=\"$READS_DIR\"/wMel_darby_2014/bam\n\n## wDi Luck 2014\nBAM_DIR=\"$READS_DIR\"/wDi_luck_2014/bam\n\n## wDi Luck 2015\nBAM_DIR=\"$READS_DIR\"/wDi_luck_2015/bam\n\n## wMel Gutzwiller 2015\nBAM_DIR=\"$READS_DIR\"/wMel_gutzwiller_2015/bam\n\n## wBm Grote 2017\nBAM_DIR=\"$READS_DIR\"/wBm_grote_2017/bam\n\n## wBm Chung 2019\nBAM_DIR=\"$READS_DIR\"/wBm_chung_2019/bam\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-10\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-10\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"while read SRR\ndo\n\techo -e \u0026quot;\u0026quot;$SAMTOOLS_BIN_DIR\u0026quot;/samtools sort -@ \u0026quot;$THREADS\u0026quot; -o \u0026quot;$BAM_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;.sortedbyposition.bam \u0026quot;$BAM_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;.bam\u0026quot; | qsub -q threaded.q -pe thread \u0026quot;$THREADS\u0026quot; -P jdhotopp-lab -l mem_free=2G -N samtools -wd \u0026quot;$BAM_DIR\u0026quot;\ndone \u0026lt; \u0026quot;$SRR_ID_LIST\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003ewhile read SRR\ndo\n\techo -e \"\"$SAMTOOLS_BIN_DIR\"/samtools sort -@ \"$THREADS\" -o \"$BAM_DIR\"/\"$SRR\".sortedbyposition.bam \"$BAM_DIR\"/\"$SRR\".bam\" | qsub -q threaded.q -pe thread \"$THREADS\" -P jdhotopp-lab -l mem_free=2G -N samtools -wd \"$BAM_DIR\"\ndone \u0026lt; \"$SRR_ID_LIST\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIndex BAM files (2019-06-01) \u003ca name=\"user-content-bashanalysis_indexbam\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-index-bam-files-2019-06-01-\" class=\"anchor\" aria-label=\"Permalink: Index BAM files (2019-06-01) \" href=\"#index-bam-files-2019-06-01-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-10\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-10\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"THREADS=4\n\n## wOo Darby 2012\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wOo_darby_2012/bam\n\n## wMel Darby 2014\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_darby_2014/bam\n\n## wDi Luck 2014\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2014/bam\n\n## wDi Luck 2015\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/bam\n\n## wMel Gutzwiller 2015\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_gutzwiller_2015/bam\n\n## wBm Grote 2017\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_grote_2017/bam\n\n## wBm Chung 2019\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_chung_2019/bam\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eTHREADS=4\n\n## wOo Darby 2012\nBAM_DIR=\"$READS_DIR\"/wOo_darby_2012/bam\n\n## wMel Darby 2014\nBAM_DIR=\"$READS_DIR\"/wMel_darby_2014/bam\n\n## wDi Luck 2014\nBAM_DIR=\"$READS_DIR\"/wDi_luck_2014/bam\n\n## wDi Luck 2015\nBAM_DIR=\"$READS_DIR\"/wDi_luck_2015/bam\n\n## wMel Gutzwiller 2015\nBAM_DIR=\"$READS_DIR\"/wMel_gutzwiller_2015/bam\n\n## wBm Grote 2017\nBAM_DIR=\"$READS_DIR\"/wBm_grote_2017/bam\n\n## wBm Chung 2019\nBAM_DIR=\"$READS_DIR\"/wBm_chung_2019/bam\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-11\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-11\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"while read SRR\ndo\n\techo -e \u0026quot;\u0026quot;$SAMTOOLS_BIN_DIR\u0026quot;/samtools index -@ \u0026quot;$THREADS\u0026quot; \u0026quot;$BAM_DIR\u0026quot;/\u0026quot;$SRR\u0026quot;.sortedbyposition.bam\u0026quot; | qsub -q threaded.q -pe thread \u0026quot;$THREADS\u0026quot; -P jdhotopp-lab -l mem_free=2G -N samtools -wd \u0026quot;$BAM_DIR\u0026quot;\ndone \u0026lt; \u0026quot;$SRR_ID_LIST\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003ewhile read SRR\ndo\n\techo -e \"\"$SAMTOOLS_BIN_DIR\"/samtools index -@ \"$THREADS\" \"$BAM_DIR\"/\"$SRR\".sortedbyposition.bam\" | qsub -q threaded.q -pe thread \"$THREADS\" -P jdhotopp-lab -l mem_free=2G -N samtools -wd \"$BAM_DIR\"\ndone \u0026lt; \"$SRR_ID_LIST\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eQuantify Wolbachia genes (2019-06-01) \u003ca name=\"user-content-bashanalysis_quant\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-quantify-wolbachia-genes-2019-06-01-\" class=\"anchor\" aria-label=\"Permalink: Quantify Wolbachia genes (2019-06-01) \" href=\"#quantify-wolbachia-genes-2019-06-01-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-11\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-11\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"FEAT_TYPE=\u0026quot;CDS\u0026quot;\nATTR_TYPE=\u0026quot;ID\u0026quot;\n\n## wOo Darby 2012\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wOo_darby_2012/bam\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wOo.gff\nSTRANDED=\u0026quot;no\u0026quot;\n\n## wMel Darby 2014\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_darby_2014/bam\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wMel.gff\nSTRANDED=\u0026quot;yes\u0026quot;\n\n## wDi Luck 2014\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2014/bam\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wDi.gff\nSTRANDED=\u0026quot;no\u0026quot;\n\n## wDi Luck 2015\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wDi_luck_2015/bam\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wDi.gff\nSTRANDED=\u0026quot;no\u0026quot;\n\n## wMel Gutzwiller 2015\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wMel_gutzwiller_2015/bam\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wMel.gff\nSTRANDED=\u0026quot;reverse\u0026quot;\n\n## wBm Grote 2017\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_grote_2017/bam\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wBm.gff\nSTRANDED=\u0026quot;no\u0026quot;\n\n## wBm Chung 2019\nBAM_DIR=\u0026quot;$READS_DIR\u0026quot;/wBm_chung_2019/bam\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wBm.gff\nSTRANDED=\u0026quot;reverse\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eFEAT_TYPE=\"CDS\"\nATTR_TYPE=\"ID\"\n\n## wOo Darby 2012\nBAM_DIR=\"$READS_DIR\"/wOo_darby_2012/bam\nGFF3=\"$REFERENCES_DIR\"/wOo.gff\nSTRANDED=\"no\"\n\n## wMel Darby 2014\nBAM_DIR=\"$READS_DIR\"/wMel_darby_2014/bam\nGFF3=\"$REFERENCES_DIR\"/wMel.gff\nSTRANDED=\"yes\"\n\n## wDi Luck 2014\nBAM_DIR=\"$READS_DIR\"/wDi_luck_2014/bam\nGFF3=\"$REFERENCES_DIR\"/wDi.gff\nSTRANDED=\"no\"\n\n## wDi Luck 2015\nBAM_DIR=\"$READS_DIR\"/wDi_luck_2015/bam\nGFF3=\"$REFERENCES_DIR\"/wDi.gff\nSTRANDED=\"no\"\n\n## wMel Gutzwiller 2015\nBAM_DIR=\"$READS_DIR\"/wMel_gutzwiller_2015/bam\nGFF3=\"$REFERENCES_DIR\"/wMel.gff\nSTRANDED=\"reverse\"\n\n## wBm Grote 2017\nBAM_DIR=\"$READS_DIR\"/wBm_grote_2017/bam\nGFF3=\"$REFERENCES_DIR\"/wBm.gff\nSTRANDED=\"no\"\n\n## wBm Chung 2019\nBAM_DIR=\"$READS_DIR\"/wBm_chung_2019/bam\nGFF3=\"$REFERENCES_DIR\"/wBm.gff\nSTRANDED=\"reverse\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-12\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-12\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"mkdir -p \u0026quot;$WORKING_DIR\u0026quot;/fadu\nfor BAM in $(find \u0026quot;$BAM_DIR\u0026quot; -name \u0026quot;*sortedbyposition.bam\u0026quot;)\ndo\n\n\techo -e \u0026quot;export JULIA_DEPOT_PATH=\u0026quot;$JULIA_DEPOT_PATH\u0026quot;\\n\u0026quot;$JULIA_BIN_DIR\u0026quot;/julia \u0026quot;$FADU_BIN_DIR\u0026quot;/fadu.jl -g \u0026quot;$GFF3\u0026quot; -b \u0026quot;$BAM\u0026quot; -o \u0026quot;$WORKING_DIR\u0026quot;/fadu -s \u0026quot;$STRANDED\u0026quot; -f \u0026quot;$FEAT_TYPE\u0026quot; -a \u0026quot;$ATTR_TYPE\u0026quot;\u0026quot; | qsub -P jdhotopp-lab -l mem_free=5G -N fadu -wd \u0026quot;$WORKING_DIR\u0026quot;/fadu\n\ndone\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003emkdir -p \"$WORKING_DIR\"/fadu\nfor BAM in $(find \"$BAM_DIR\" -name \"*sortedbyposition.bam\")\ndo\n\n\techo -e \"export JULIA_DEPOT_PATH=\"$JULIA_DEPOT_PATH\"\\n\"$JULIA_BIN_DIR\"/julia \"$FADU_BIN_DIR\"/fadu.jl -g \"$GFF3\" -b \"$BAM\" -o \"$WORKING_DIR\"/fadu -s \"$STRANDED\" -f \"$FEAT_TYPE\" -a \"$ATTR_TYPE\"\" | qsub -P jdhotopp-lab -l mem_free=5G -N fadu -wd \"$WORKING_DIR\"/fadu\n\ndone\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConvert GFF to a mapping table (2019-06-07) \u003ca name=\"user-content-bashanalysis_gff2map\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-convert-gff-to-a-mapping-table-2019-06-07-\" class=\"anchor\" aria-label=\"Permalink: Convert GFF to a mapping table (2019-06-07) \" href=\"#convert-gff-to-a-mapping-table-2019-06-07-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-12\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-12\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## wBm\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wBm.gff\n\n## wDi\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wDi.gff\n\n## wMel\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wMel.gff\n\n## wOo\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wOo.gff\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e## wBm\nGFF3=\"$REFERENCES_DIR\"/wBm.gff\n\n## wDi\nGFF3=\"$REFERENCES_DIR\"/wDi.gff\n\n## wMel\nGFF3=\"$REFERENCES_DIR\"/wMel.gff\n\n## wOo\nGFF3=\"$REFERENCES_DIR\"/wOo.gff\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-13\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-13\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\u0026quot;$R_BIN_DIR\u0026quot;/Rscript \u0026quot;$SCRIPTS_DIR\u0026quot;/gff3_to_map.R \u0026quot;$GFF_PATH\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e\"$R_BIN_DIR\"/Rscript \"$SCRIPTS_DIR\"/gff3_to_map.R \"$GFF_PATH\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCreate nucleotide coding sequence fasta files (2019-06-07) \u003ca name=\"user-content-bashanalysis_createcdsfasta\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-create-nucleotide-coding-sequence-fasta-files-2019-06-07-\" class=\"anchor\" aria-label=\"Permalink: Create nucleotide coding sequence fasta files (2019-06-07) \" href=\"#create-nucleotide-coding-sequence-fasta-files-2019-06-07-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-13\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-13\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"FEAT_TYPE=\u0026quot;CDS\u0026quot;\nATTR_TYPE=\u0026quot;ID\u0026quot;\n\n## wBm\nFNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wBm.fna\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wBm.gff\n\n## wDi\nFNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wDi.fna\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wDi.gff\n\n## wMel\nFNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wMel.fna\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wMel.gff\n\n## wOo\nFNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wOo.fna\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wOo.gff\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eFEAT_TYPE=\"CDS\"\nATTR_TYPE=\"ID\"\n\n## wBm\nFNA=\"$REFERENCES_DIR\"/wBm.fna\nGFF3=\"$REFERENCES_DIR\"/wBm.gff\n\n## wDi\nFNA=\"$REFERENCES_DIR\"/wDi.fna\nGFF3=\"$REFERENCES_DIR\"/wDi.gff\n\n## wMel\nFNA=\"$REFERENCES_DIR\"/wMel.fna\nGFF3=\"$REFERENCES_DIR\"/wMel.gff\n\n## wOo\nFNA=\"$REFERENCES_DIR\"/wOo.fna\nGFF3=\"$REFERENCES_DIR\"/wOo.gff\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-14\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-14\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\u0026quot;$SCRIPTS_DIR\u0026quot;/createnuccdsfasta.sh -n \u0026quot;$FNA\u0026quot; -g \u0026quot;$GFF3\u0026quot; -f \u0026quot;$FEAT_TYPE\u0026quot; -i \u0026quot;$ATTR_TYPE\u0026quot; \u0026gt; \u0026quot;$(echo -e \u0026quot;$FNA\u0026quot; | sed \u0026quot;s/[.]fna$/.cds.fna/g\u0026quot;)\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e\"$SCRIPTS_DIR\"/createnuccdsfasta.sh -n \"$FNA\" -g \"$GFF3\" -f \"$FEAT_TYPE\" -i \"$ATTR_TYPE\" \u0026gt; \"$(echo -e \"$FNA\" | sed \"s/[.]fna$/.cds.fna/g\")\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIdentify InterPro descriptions and GO terms for Wolbachia transcripts (2019-06-07) \u003ca name=\"user-content-bashanalysis_interproscan\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-identify-interpro-descriptions-and-go-terms-for-wolbachia-transcripts-2019-06-07-\" class=\"anchor\" aria-label=\"Permalink: Identify InterPro descriptions and GO terms for Wolbachia transcripts (2019-06-07) \" href=\"#identify-interpro-descriptions-and-go-terms-for-wolbachia-transcripts-2019-06-07-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-14\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-14\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"SEQ_TYPE=n\nTHREADS=4\n\n## wBm\nCDS_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wBm.cds.fna\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wBm.gff\n\n## wDi\nCDS_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wDi.cds.fna\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wDi.gff\n\n## wMel\nCDS_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wMel.cds.fna\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wMel.gff\n\n## wOo\nCDS_FNA=\u0026quot;$REFERENCES_DIR\u0026quot;/wOo.cds.fna\nGFF3=\u0026quot;$REFERENCES_DIR\u0026quot;/wOo.gff\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eSEQ_TYPE=n\nTHREADS=4\n\n## wBm\nCDS_FNA=\"$REFERENCES_DIR\"/wBm.cds.fna\nGFF3=\"$REFERENCES_DIR\"/wBm.gff\n\n## wDi\nCDS_FNA=\"$REFERENCES_DIR\"/wDi.cds.fna\nGFF3=\"$REFERENCES_DIR\"/wDi.gff\n\n## wMel\nCDS_FNA=\"$REFERENCES_DIR\"/wMel.cds.fna\nGFF3=\"$REFERENCES_DIR\"/wMel.gff\n\n## wOo\nCDS_FNA=\"$REFERENCES_DIR\"/wOo.cds.fna\nGFF3=\"$REFERENCES_DIR\"/wOo.gff\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-15\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-15\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"echo -e \u0026quot;export LD_LIBRARY_PATH=\u0026quot;$PYTHON_LIB_PATH\u0026quot;:\u0026quot;$LD_LIBRARY_PATH\u0026quot;\\n\u0026quot;$INTERPROSCAN_BIN_DIR\u0026quot;/interproscan.sh -i \u0026quot;$CDS_FNA\u0026quot; -f tsv -o \u0026quot;$CDS_FNA\u0026quot;.interproscan.tsv --seqtype \u0026quot;$SEQ_TYPE\u0026quot; --goterms --iprlookup\u0026quot; | qsub -P jdhotopp-lab -q threaded.q  -pe thread \u0026quot;$THREADS\u0026quot; -l mem_free=20G -N interproscan -wd \u0026quot;$(dirname \u0026quot;$CDS_FNA\u0026quot;)\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eecho -e \"export LD_LIBRARY_PATH=\"$PYTHON_LIB_PATH\":\"$LD_LIBRARY_PATH\"\\n\"$INTERPROSCAN_BIN_DIR\"/interproscan.sh -i \"$CDS_FNA\" -f tsv -o \"$CDS_FNA\".interproscan.tsv --seqtype \"$SEQ_TYPE\" --goterms --iprlookup\" | qsub -P jdhotopp-lab -q threaded.q  -pe thread \"$THREADS\" -l mem_free=20G -N interproscan -wd \"$(dirname \"$CDS_FNA\")\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTranscriptomics meta-analysis \u003ca name=\"user-content-ranalysis\"\u003e\u003c/a\u003e\u003c/h1\u003e\u003ca id=\"user-content-transcriptomics-meta-analysis-\" class=\"anchor\" aria-label=\"Permalink: Transcriptomics meta-analysis \" href=\"#transcriptomics-meta-analysis-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCreate counts table \u003ca name=\"user-content-ranalysis_counts\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-create-counts-table-\" class=\"anchor\" aria-label=\"Permalink: Create counts table \" href=\"#create-counts-table-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSet R inputs \u003ca name=\"user-content-ranalysis_setinputs\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-set-r-inputs-\" class=\"anchor\" aria-label=\"Permalink: Set R inputs \" href=\"#set-r-inputs-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eR inputs from bash should be:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSAMPLE_MAP.PATH = \"$INPUTS_DIR\"/study_sample_map.tsv.txt\u003cbr\u003e\nRAW_COUNTS.DIR = \"$WORKING_DIR\"/fadu\u003cbr\u003e\nOUTPUT.DIR = \"$WORKING_DIR\"\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"SAMPLE_MAP.PATH  \u0026lt;- \u0026quot;Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/study_sample_map.tsv.txt\u0026quot;\nRAW_COUNTS.DIR \u0026lt;- \u0026quot;Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/fadu/\u0026quot;\nOUTPUT.DIR \u0026lt;- \u0026quot;Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/\u0026quot;\"\u003e\u003cpre lang=\"{R}\" class=\"notranslate\"\u003e\u003ccode\u003eSAMPLE_MAP.PATH  \u0026lt;- \"Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/study_sample_map.tsv.txt\"\nRAW_COUNTS.DIR \u0026lt;- \"Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/fadu/\"\nOUTPUT.DIR \u0026lt;- \"Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eView sessionInfo \u003ca name=\"user-content-ranalysis_sessioninfo\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-view-sessioninfo-\" class=\"anchor\" aria-label=\"Permalink: View sessionInfo \" href=\"#view-sessioninfo-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"sessionInfo()\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003esessionInfo()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCombine count files into a single table for each study \u003ca name=\"user-content-ranalysis_counts_maketable\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-combine-count-files-into-a-single-table-for-each-study-\" class=\"anchor\" aria-label=\"Permalink: Combine count files into a single table for each study \" href=\"#combine-count-files-into-a-single-table-for-each-study-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"sample_map \u0026lt;- read.delim(SAMPLE_MAP.PATH,header = T,sep = \u0026quot;\\t\u0026quot;)\ncounts.files \u0026lt;- list.files(RAW_COUNTS.DIR,\n                           full.names = T,\n                           pattern=\u0026quot;.*counts.txt\u0026quot;)\n\ncounts.list \u0026lt;- list(wOo_darby_2012=\u0026quot;\u0026quot;,\n                    wMel_darby_2014=\u0026quot;\u0026quot;,\n                    wDi_luck_2014=\u0026quot;\u0026quot;,\n                    wDi_luck_2015=\u0026quot;\u0026quot;,\n                    wMel_gutzwiller_2015=\u0026quot;\u0026quot;,\n                    wBm_grote_2017=\u0026quot;\u0026quot;,\n                    wBm_chung_2019=\u0026quot;\u0026quot;)\n\n\nfor(i in 1:length(counts.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(counts.list)[i],]\n  samples \u0026lt;- unique(sample_map.subset$sample_identifier)\n  \n  genes \u0026lt;- read.delim(grep(sample_map.subset$sra_id[i],counts.files,value = T),header = T)[,1]\n  counts.list[[i]] \u0026lt;- as.data.frame(matrix(0,\n                                           nrow = length(genes),\n                                           ncol = length(samples)))\n  rownames(counts.list[[i]]) \u0026lt;- genes\n  colnames(counts.list[[i]]) \u0026lt;- samples\n                                    \n  for(j in 1:ncol(counts.list[[i]])){\n    srr \u0026lt;- sample_map.subset$sra_id[sample_map.subset$sample_identifier == colnames(counts.list[[i]])[j]]\n    while(length(srr) \u0026gt; 0){\n      raw_counts.file \u0026lt;- read.delim(paste0(RAW_COUNTS.DIR,\u0026quot;/\u0026quot;,srr[1],\u0026quot;.sortedbyposition.counts.txt\u0026quot;),\n                                    header = T)\n      raw_counts.file \u0026lt;- raw_counts.file[match(rownames(counts.list[[i]]),raw_counts.file[,1]),]\n      counts.list[[i]][,j] \u0026lt;- counts.list[[i]][,j] + raw_counts.file$counts\n      \n      srr \u0026lt;- srr[-1]\n    }\n  }\n  write.table(counts.list[[i]],\n              paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.list)[i],\u0026quot;_counts.tsv\u0026quot;),\n              row.names = T,\n              col.names = T,\n              quote = F,\n              sep = \u0026quot;\\t\u0026quot;)\n}\"\u003e\u003cpre lang=\"{R}\" class=\"notranslate\"\u003e\u003ccode\u003esample_map \u0026lt;- read.delim(SAMPLE_MAP.PATH,header = T,sep = \"\\t\")\ncounts.files \u0026lt;- list.files(RAW_COUNTS.DIR,\n                           full.names = T,\n                           pattern=\".*counts.txt\")\n\ncounts.list \u0026lt;- list(wOo_darby_2012=\"\",\n                    wMel_darby_2014=\"\",\n                    wDi_luck_2014=\"\",\n                    wDi_luck_2015=\"\",\n                    wMel_gutzwiller_2015=\"\",\n                    wBm_grote_2017=\"\",\n                    wBm_chung_2019=\"\")\n\n\nfor(i in 1:length(counts.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(counts.list)[i],]\n  samples \u0026lt;- unique(sample_map.subset$sample_identifier)\n  \n  genes \u0026lt;- read.delim(grep(sample_map.subset$sra_id[i],counts.files,value = T),header = T)[,1]\n  counts.list[[i]] \u0026lt;- as.data.frame(matrix(0,\n                                           nrow = length(genes),\n                                           ncol = length(samples)))\n  rownames(counts.list[[i]]) \u0026lt;- genes\n  colnames(counts.list[[i]]) \u0026lt;- samples\n                                    \n  for(j in 1:ncol(counts.list[[i]])){\n    srr \u0026lt;- sample_map.subset$sra_id[sample_map.subset$sample_identifier == colnames(counts.list[[i]])[j]]\n    while(length(srr) \u0026gt; 0){\n      raw_counts.file \u0026lt;- read.delim(paste0(RAW_COUNTS.DIR,\"/\",srr[1],\".sortedbyposition.counts.txt\"),\n                                    header = T)\n      raw_counts.file \u0026lt;- raw_counts.file[match(rownames(counts.list[[i]]),raw_counts.file[,1]),]\n      counts.list[[i]][,j] \u0026lt;- counts.list[[i]][,j] + raw_counts.file$counts\n      \n      srr \u0026lt;- srr[-1]\n    }\n  }\n  write.table(counts.list[[i]],\n              paste0(OUTPUT.DIR,\"/\",names(counts.list)[i],\"_counts.tsv\"),\n              row.names = T,\n              col.names = T,\n              quote = F,\n              sep = \"\\t\")\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConvert InterProScan output to geneinfo file (2019-06-07) \u003ca name=\"user-content-ranalysis_iprscan2geneinfo\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-convert-interproscan-output-to-geneinfo-file-2019-06-07-\" class=\"anchor\" aria-label=\"Permalink: Convert InterProScan output to geneinfo file (2019-06-07) \" href=\"#convert-interproscan-output-to-geneinfo-file-2019-06-07-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInput Sets:\u003c/h4\u003e\u003ca id=\"user-content-input-sets-15\" class=\"anchor\" aria-label=\"Permalink: Input Sets:\" href=\"#input-sets-15\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"SEQ_TYPE=n\nTHREADS=4\n\n## wBm\nINTERPROSCAN_OUTPUT=\u0026quot;$REFERENCES_DIR\u0026quot;/wBm.cds.fna.interproscan.tsv\nCOUNTS=\u0026quot;$WORKING_DIR\u0026quot;/wBm_chung_2019_counts.tsv\n\n## wDi\nINTERPROSCAN_OUTPUT=\u0026quot;$REFERENCES_DIR\u0026quot;/wDi.cds.fna.interproscan.tsv\nCOUNTS=\u0026quot;$WORKING_DIR\u0026quot;/wDi_luck_2015_counts.tsv\n\n## wMel\nINTERPROSCAN_OUTPUT=\u0026quot;$REFERENCES_DIR\u0026quot;/wMel.cds.fna.interproscan.tsv\nCOUNTS=\u0026quot;$WORKING_DIR\u0026quot;/wMel_darby_2014_counts.tsv\n\n## wOo\nINTERPROSCAN_OUTPUT=\u0026quot;$REFERENCES_DIR\u0026quot;/wOo.cds.fna.interproscan.tsv\nCOUNTS=\u0026quot;$WORKING_DIR\u0026quot;/wOo_darby_2012_counts.tsv\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eSEQ_TYPE=n\nTHREADS=4\n\n## wBm\nINTERPROSCAN_OUTPUT=\"$REFERENCES_DIR\"/wBm.cds.fna.interproscan.tsv\nCOUNTS=\"$WORKING_DIR\"/wBm_chung_2019_counts.tsv\n\n## wDi\nINTERPROSCAN_OUTPUT=\"$REFERENCES_DIR\"/wDi.cds.fna.interproscan.tsv\nCOUNTS=\"$WORKING_DIR\"/wDi_luck_2015_counts.tsv\n\n## wMel\nINTERPROSCAN_OUTPUT=\"$REFERENCES_DIR\"/wMel.cds.fna.interproscan.tsv\nCOUNTS=\"$WORKING_DIR\"/wMel_darby_2014_counts.tsv\n\n## wOo\nINTERPROSCAN_OUTPUT=\"$REFERENCES_DIR\"/wOo.cds.fna.interproscan.tsv\nCOUNTS=\"$WORKING_DIR\"/wOo_darby_2012_counts.tsv\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-16\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-16\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\u0026quot;$R_BIN_DIR\u0026quot;/Rscript \u0026quot;$SCRIPTS_DIR\u0026quot;/interproscan2geneinfo.R \u0026quot;$INTERPROSCAN_OUTPUT\u0026quot; \u0026quot;$COUNTS\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e\"$R_BIN_DIR\"/Rscript \"$SCRIPTS_DIR\"/interproscan2geneinfo.R \"$INTERPROSCAN_OUTPUT\" \"$COUNTS\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDifferential expression and WGCNA analyses (2019-06-07) \u003ca name=\"user-content-ranalysis_iprscan2geneinfo\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-differential-expression-and-wgcna-analyses-2019-06-07-\" class=\"anchor\" aria-label=\"Permalink: Differential expression and WGCNA analyses (2019-06-07) \" href=\"#differential-expression-and-wgcna-analyses-2019-06-07-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSet R inputs \u003ca name=\"user-content-ranalysis_de_setinputs\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-set-r-inputs--1\" class=\"anchor\" aria-label=\"Permalink: Set R inputs \" href=\"#set-r-inputs--1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eR inputs from bash should be:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSAMPLE_MAP.PATH = \"$INPUTS_DIR\"/study_sample_map.tsv.txt\u003cbr\u003e\nCOUNTS.DIR = \"$WORKING_DIR\"\u003cbr\u003e\nRAW_COUNTS.DIR = \"$WORKING_DIR\"/fadu\u003cbr\u003e\nGFF3MAP.DIR = \"$REFERENCES_DIR\"\u003cbr\u003e\nOUTPUT.DIR = \"$WORKING_DIR\"\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"SAMPLE_MAP.PATH  \u0026lt;- \u0026quot;Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/study_sample_map.tsv.txt\u0026quot;\nCOUNTS.DIR \u0026lt;- \u0026quot;Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/\u0026quot;\nGENEINFO.DIR \u0026lt;- \u0026quot;Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/references\u0026quot;\nGFF3MAP.DIR \u0026lt;- \u0026quot;Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/references\u0026quot;\nRAW_COUNTS.DIR \u0026lt;- \u0026quot;Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/fadu\u0026quot;\nOUTPUT.DIR \u0026lt;- \u0026quot;Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/\u0026quot;\"\u003e\u003cpre lang=\"{R}\" class=\"notranslate\"\u003e\u003ccode\u003eSAMPLE_MAP.PATH  \u0026lt;- \"Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/study_sample_map.tsv.txt\"\nCOUNTS.DIR \u0026lt;- \"Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/\"\nGENEINFO.DIR \u0026lt;- \"Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/references\"\nGFF3MAP.DIR \u0026lt;- \"Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/references\"\nRAW_COUNTS.DIR \u0026lt;- \"Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/fadu\"\nOUTPUT.DIR \u0026lt;- \"Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eLoad R packages and view sessionInfo \u003ca name=\"user-content-ranalysis_de_sessioninfo\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-load-r-packages-and-view-sessioninfo-\" class=\"anchor\" aria-label=\"Permalink: Load R packages and view sessionInfo \" href=\"#load-r-packages-and-view-sessioninfo-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"library(cowplot)\nlibrary(edgeR)\nlibrary(emdbook)\nlibrary(FactoMineR)\nlibrary(dendextend)\nlibrary(ggdendro)\nlibrary(ggplot2)\nlibrary(gplots)\nlibrary(gridExtra)\nlibrary(gtools)\nlibrary(pvclust)\nlibrary(reshape2)\nlibrary(vegan)\nlibrary(WGCNA)\n\nsessionInfo()\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003elibrary(cowplot)\nlibrary(edgeR)\nlibrary(emdbook)\nlibrary(FactoMineR)\nlibrary(dendextend)\nlibrary(ggdendro)\nlibrary(ggplot2)\nlibrary(gplots)\nlibrary(gridExtra)\nlibrary(gtools)\nlibrary(pvclust)\nlibrary(reshape2)\nlibrary(vegan)\nlibrary(WGCNA)\n\nsessionInfo()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"R version 3.5.1 (2018-07-02)\nPlatform: x86_64-w64-mingw32/x64 (64-bit)\nRunning under: Windows 7 x64 (build 7601) Service Pack 1\n\nMatrix products: default\n\nlocale:\n[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252\n[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    \n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] WGCNA_1.66            fastcluster_1.1.25    dynamicTreeCut_1.63-1 vegan_2.5-4           lattice_0.20-35       permute_0.9-4        \n [7] reshape2_1.4.3        pvclust_2.0-0         gtools_3.8.1          gridExtra_2.3         gplots_3.0.1.1        ggdendro_0.1-20      \n[13] dendextend_1.9.0      FactoMineR_1.41       emdbook_1.3.11        edgeR_3.24.3          limma_3.38.3          cowplot_0.9.4        \n[19] ggplot2_3.2.1        \n\nloaded via a namespace (and not attached):\n  [1] colorspace_1.4-1      class_7.3-14          modeltools_0.2-22     mclust_5.4.3          htmlTable_1.13.1      base64enc_0.1-3      \n  [7] rstudioapi_0.10       flexmix_2.3-15        bit64_0.9-7           AnnotationDbi_1.44.0  mvtnorm_1.0-10        codetools_0.2-15     \n [13] splines_3.5.1         leaps_3.0             doParallel_1.0.14     impute_1.56.0         robustbase_0.93-4     knitr_1.22           \n [19] Formula_1.2-3         cluster_2.0.7-1       kernlab_0.9-27        GO.db_3.7.0           rrcov_1.4-7           compiler_3.5.1       \n [25] backports_1.1.3       assertthat_0.2.1      Matrix_1.2-14         lazyeval_0.2.2        acepack_1.4.1         htmltools_0.3.6      \n [31] tools_3.5.1           coda_0.19-2           gtable_0.3.0          glue_1.3.1            dplyr_0.8.3           Rcpp_1.0.1           \n [37] bbmle_1.0.20          Biobase_2.42.0        trimcluster_0.1-2.1   gdata_2.18.0          preprocessCore_1.44.0 nlme_3.1-137         \n [43] iterators_1.0.10      fpc_2.1-11.1          xfun_0.6              stringr_1.4.0         DEoptimR_1.0-8        MASS_7.3-50          \n [49] scales_1.0.0          parallel_3.5.1        RColorBrewer_1.1-2    yaml_2.2.0            memoise_1.1.0         rpart_4.1-13         \n [55] latticeExtra_0.6-28   stringi_1.4.3         RSQLite_2.1.1         S4Vectors_0.20.1      pcaPP_1.9-73          foreach_1.4.4        \n [61] checkmate_1.9.1       caTools_1.17.1.2      BiocGenerics_0.28.0   rlang_0.4.0           pkgconfig_2.0.2       prabclus_2.2-7       \n [67] bitops_1.0-6          matrixStats_0.54.0    purrr_0.3.2           htmlwidgets_1.3       bit_1.1-14            tidyselect_0.2.5     \n [73] robust_0.4-18         plyr_1.8.4            magrittr_1.5          R6_2.4.0              IRanges_2.16.0        Hmisc_4.2-0          \n [79] fit.models_0.5-14     DBI_1.0.0             pillar_1.3.1          whisker_0.3-2         foreign_0.8-70        withr_2.1.2          \n [85] mgcv_1.8-24           survival_2.42-3       scatterplot3d_0.3-41  nnet_7.3-12           tibble_2.1.1          crayon_1.3.4         \n [91] KernSmooth_2.23-15    viridis_0.5.1         locfit_1.5-9.1        grid_3.5.1            data.table_1.12.2     blob_1.1.1           \n [97] digest_0.6.18         diptest_0.75-7        flashClust_1.01-2     numDeriv_2016.8-1     stats4_3.5.1          munsell_0.5.0        \n[103] viridisLite_0.3.0\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003eR version 3.5.1 (2018-07-02)\nPlatform: x86_64-w64-mingw32/x64 (64-bit)\nRunning under: Windows 7 x64 (build 7601) Service Pack 1\n\nMatrix products: default\n\nlocale:\n[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252\n[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    \n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] WGCNA_1.66            fastcluster_1.1.25    dynamicTreeCut_1.63-1 vegan_2.5-4           lattice_0.20-35       permute_0.9-4        \n [7] reshape2_1.4.3        pvclust_2.0-0         gtools_3.8.1          gridExtra_2.3         gplots_3.0.1.1        ggdendro_0.1-20      \n[13] dendextend_1.9.0      FactoMineR_1.41       emdbook_1.3.11        edgeR_3.24.3          limma_3.38.3          cowplot_0.9.4        \n[19] ggplot2_3.2.1        \n\nloaded via a namespace (and not attached):\n  [1] colorspace_1.4-1      class_7.3-14          modeltools_0.2-22     mclust_5.4.3          htmlTable_1.13.1      base64enc_0.1-3      \n  [7] rstudioapi_0.10       flexmix_2.3-15        bit64_0.9-7           AnnotationDbi_1.44.0  mvtnorm_1.0-10        codetools_0.2-15     \n [13] splines_3.5.1         leaps_3.0             doParallel_1.0.14     impute_1.56.0         robustbase_0.93-4     knitr_1.22           \n [19] Formula_1.2-3         cluster_2.0.7-1       kernlab_0.9-27        GO.db_3.7.0           rrcov_1.4-7           compiler_3.5.1       \n [25] backports_1.1.3       assertthat_0.2.1      Matrix_1.2-14         lazyeval_0.2.2        acepack_1.4.1         htmltools_0.3.6      \n [31] tools_3.5.1           coda_0.19-2           gtable_0.3.0          glue_1.3.1            dplyr_0.8.3           Rcpp_1.0.1           \n [37] bbmle_1.0.20          Biobase_2.42.0        trimcluster_0.1-2.1   gdata_2.18.0          preprocessCore_1.44.0 nlme_3.1-137         \n [43] iterators_1.0.10      fpc_2.1-11.1          xfun_0.6              stringr_1.4.0         DEoptimR_1.0-8        MASS_7.3-50          \n [49] scales_1.0.0          parallel_3.5.1        RColorBrewer_1.1-2    yaml_2.2.0            memoise_1.1.0         rpart_4.1-13         \n [55] latticeExtra_0.6-28   stringi_1.4.3         RSQLite_2.1.1         S4Vectors_0.20.1      pcaPP_1.9-73          foreach_1.4.4        \n [61] checkmate_1.9.1       caTools_1.17.1.2      BiocGenerics_0.28.0   rlang_0.4.0           pkgconfig_2.0.2       prabclus_2.2-7       \n [67] bitops_1.0-6          matrixStats_0.54.0    purrr_0.3.2           htmlwidgets_1.3       bit_1.1-14            tidyselect_0.2.5     \n [73] robust_0.4-18         plyr_1.8.4            magrittr_1.5          R6_2.4.0              IRanges_2.16.0        Hmisc_4.2-0          \n [79] fit.models_0.5-14     DBI_1.0.0             pillar_1.3.1          whisker_0.3-2         foreign_0.8-70        withr_2.1.2          \n [85] mgcv_1.8-24           survival_2.42-3       scatterplot3d_0.3-41  nnet_7.3-12           tibble_2.1.1          crayon_1.3.4         \n [91] KernSmooth_2.23-15    viridis_0.5.1         locfit_1.5-9.1        grid_3.5.1            data.table_1.12.2     blob_1.1.1           \n [97] digest_0.6.18         diptest_0.75-7        flashClust_1.01-2     numDeriv_2016.8-1     stats4_3.5.1          munsell_0.5.0        \n[103] viridisLite_0.3.0\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eLoad R functions \u003ca name=\"user-content-ranalysis_de_loadfunctions\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-load-r-functions-\" class=\"anchor\" aria-label=\"Permalink: Load R functions \" href=\"#load-r-functions-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"g_legend\u0026lt;-function(a.gplot){ \n  tmp \u0026lt;- ggplot_gtable(ggplot_build(a.gplot)) \n  leg \u0026lt;- which(sapply(tmp$grobs, function(x) x$name) == \u0026quot;guide-box\u0026quot;) \n  legend \u0026lt;- tmp$grobs[[leg]] \n  return(legend)\n} \n\nget_dendro_structure \u0026lt;- function(result){\n  structure \u0026lt;- hang.dendrogram(as.dendrogram(result$hclust))\n  structure \u0026lt;- capture.output(str(structure))\n  structure \u0026lt;- structure[grepl(\u0026quot;leaf\u0026quot;, structure)]\n  structure \u0026lt;- as.numeric(as.character(substr(structure, regexpr(\u0026quot;h=\u0026quot;, structure ) + 3, regexpr(\u0026quot;  )\u0026quot;, structure))))\n  return(structure)\n}\nget_dendro_data \u0026lt;- function(result){\n  dendro.data \u0026lt;- dendro_data(result$hclust)\n  dendro.data \u0026lt;- dendro.data$segments[which(dendro.data$segments$y == dendro.data$segments$yend),]\n  for(i in 1:nrow(dendro.data)){\n    dendro.data$minx[i] \u0026lt;- min(c(dendro.data$x[i], dendro.data$xend[i]))\n  }\n  dendro.data \u0026lt;- dendro.data[order(as.numeric(as.character(dendro.data$y)), as.numeric(as.character(dendro.data$minx))),]\n  return(dendro.data)\n}\nget_dendro_bootstraps \u0026lt;- function(dendro_data){\n  bootstrap.positions \u0026lt;- as.data.frame(matrix(nrow = length(dendro_data$y[duplicated(dendro_data$y)]),\n                                              ncol = 2))\n  for(i in 1:length(dendro_data$y[duplicated(dendro_data$y)])){\n    dendro_data.subset \u0026lt;- dendro_data[which(dendro_data$y == dendro_data$y[duplicated(dendro_data$y)][i]),]\n    bootstrap.positions[i,1] \u0026lt;- unique(dendro_data.subset$x)\n    bootstrap.positions[i,2] \u0026lt;- unique(dendro_data.subset$y)\n  }\n  return(bootstrap.positions)\n}\nget_heatmap_separators \u0026lt;- function(vector){\n  sep \u0026lt;- c()\n  for(i in 2:length(unique(vector))){\n    sep[length(sep) + 1] \u0026lt;- min(which(vector == unique(vector)[i])) - 1\n  }\n  return(sep)\n}\nformalize_study_names \u0026lt;- function(vector){\n  vector \u0026lt;- gsub(\u0026quot;wOo_darby_2012\u0026quot;, \u0026quot;Darby et al., 2012\u0026quot;,vector)\n  vector \u0026lt;- gsub(\u0026quot;wMel_darby_2014\u0026quot;, \u0026quot;Darby et al., 2014\u0026quot;,vector)\n  vector \u0026lt;- gsub(\u0026quot;wDi_luck_2014\u0026quot;, \u0026quot;Luck et al., 2014\u0026quot;,vector)\n  vector \u0026lt;- gsub(\u0026quot;wDi_luck_2015\u0026quot;, \u0026quot;Luck et al., 2015\u0026quot;,vector)\n  vector \u0026lt;- gsub(\u0026quot;wMel_gutzwiller_2015\u0026quot;, \u0026quot;Gutzwiller et al., 2015\u0026quot;,vector)\n  vector \u0026lt;- gsub(\u0026quot;wBm_grote_2017\u0026quot;, \u0026quot;Grote et al., 2017\u0026quot;,vector)\n  vector \u0026lt;- gsub(\u0026quot;wBm_chung_2019\u0026quot;, \u0026quot;Chung et al., 2019\u0026quot;,vector)\n  vector \u0026lt;- factor(vector,levels=c(\u0026quot;Darby et al., 2012\u0026quot;,\n                                   \u0026quot;Darby et al., 2014\u0026quot;,\n                                   \u0026quot;Luck et al., 2014\u0026quot;,\n                                   \u0026quot;Luck et al., 2015\u0026quot;,\n                                   \u0026quot;Gutzwiller et al., 2015\u0026quot;,\n                                   \u0026quot;Grote et al., 2017\u0026quot;,\n                                   \u0026quot;Chung et al., 2019\u0026quot;))\n  return(vector)\n}\nfind_soft_power \u0026lt;- function(sft){\n  df \u0026lt;- as.data.frame(cbind(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2]))\n  y \u0026lt;- -sign(sft$fitIndices[,3])*sft$fitIndices[,2]\n  dy \u0026lt;- diff(y) \n  softpower \u0026lt;- which(abs(diff(y)) \u0026lt; 0.05)[1]\n  return(softpower)\n}\neigengene_invert_id \u0026lt;- function(tpm.de.wgcna, mergedColors, mergedMEs){\n  tpm.de.wgcna$invert \u0026lt;- T\n  tpm.de.wgcna$module \u0026lt;- mergedColors\n  for(i in 1:nrow(tpm.de.wgcna)){\n    if(cor(t(tpm.de.wgcna[i,1:(ncol(tpm.de.wgcna) - 2)]), mergedMEs[,which(colnames(mergedMEs) == paste0(\u0026quot;ME\u0026quot;,tpm.de.wgcna$module[i]))], method = \u0026quot;pearson\u0026quot;) \u0026gt; 0){\n      tpm.de.wgcna$invert[i] \u0026lt;- F\n    }\n  }\n  return(tpm.de.wgcna)\n}\nwgcna_heatmap_reorder \u0026lt;- function(tpm.de.wgcna){\n  clusters \u0026lt;- as.data.frame(table(tpm.de.wgcna$module))\n  clusters \u0026lt;- clusters[order(-clusters[,2]),1]\n  \n  tpm.de.wgcna.reordered \u0026lt;- as.data.frame(matrix(nrow = 0,\n                                                 ncol = ncol(tpm.de.wgcna)))\n  for(i in 1:length(clusters)){\n    tpm.de.wgcna.reordered \u0026lt;- as.data.frame(rbind(tpm.de.wgcna.reordered,\n                                                  tpm.de.wgcna[tpm.de.wgcna$module == clusters[i] \u0026amp; tpm.de.wgcna$invert == F,],\n                                                  tpm.de.wgcna[tpm.de.wgcna$module == clusters[i] \u0026amp; tpm.de.wgcna$invert == T,]))\n  }\n  return(tpm.de.wgcna.reordered)\n}\nfunctionaltermenrichment \u0026lt;- function(genes, geneinfo){\n  for(i in 1:ncol(geneinfo)){geneinfo[,i] \u0026lt;- as.character(geneinfo[,i])}\n  geneinfo$interpro_description[which(is.na(geneinfo$interpro_description))] \u0026lt;- \u0026quot;No InterPro entry\u0026quot;\n  geneinfo$go_biologicalprocess[which(is.na(geneinfo$go_biologicalprocess))] \u0026lt;- \u0026quot;No GO terms for biological process\u0026quot;\n  geneinfo$go_cellularcomponent[which(is.na(geneinfo$go_cellularcomponent))] \u0026lt;- \u0026quot;No GO terms for cellular component\u0026quot;\n  geneinfo$go_molecularfunction[which(is.na(geneinfo$go_molecularfunction))] \u0026lt;- \u0026quot;No GO terms for molecular function\u0026quot;\n  \n  functionalterms.list \u0026lt;- list(ipr=as.data.frame(table(unlist(strsplit(paste(geneinfo$interpro_description, collapse = \u0026quot;|\u0026quot;),  split = \u0026quot;[|]\u0026quot;)))),\n                               gobio=as.data.frame(table(unlist(strsplit(paste(geneinfo$go_biologicalprocess, collapse = \u0026quot;|\u0026quot;),  split = \u0026quot;[|]\u0026quot;)))),\n                               gocell=as.data.frame(table(unlist(strsplit(paste(geneinfo$go_cellularcomponent, collapse = \u0026quot;|\u0026quot;),  split = \u0026quot;[|]\u0026quot;)))),\n                               gomol=as.data.frame(table(unlist(strsplit(paste(geneinfo$go_molecularfunction, collapse = \u0026quot;|\u0026quot;),  split = \u0026quot;[|]\u0026quot;)))))\n  \n  geneinfo.subset \u0026lt;- geneinfo[geneinfo$gene %in% genes,]\n  term \u0026lt;- c()\n  clusteroccurences \u0026lt;- c()\n  genomeoccurences \u0026lt;- c()\n  pvalue \u0026lt;- c()\n  correctedpvalue \u0026lt;- c()\n  oddsratio \u0026lt;- c()\n\n  functionalterms.list.subset \u0026lt;- list(ipr=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$interpro_description, collapse = \u0026quot;|\u0026quot;),  split = \u0026quot;[|]\u0026quot;)))),\n                                      gobio=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$go_biologicalprocess, collapse = \u0026quot;|\u0026quot;),  split = \u0026quot;[|]\u0026quot;)))),\n                                      gocell=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$go_cellularcomponent, collapse = \u0026quot;|\u0026quot;),  split = \u0026quot;[|]\u0026quot;)))),\n                                      gomol=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$go_molecularfunction, collapse = \u0026quot;|\u0026quot;),  split = \u0026quot;[|]\u0026quot;)))))\n  \n  for(i in 1:length(functionalterms.list)){\n    for(j in 1:nrow(functionalterms.list[[i]])){\n      freq.all \u0026lt;- functionalterms.list[[i]][j,2]\n      freq.subset \u0026lt;- ifelse(functionalterms.list[[i]][j,1] %in% functionalterms.list.subset[[i]][,1],\n                            functionalterms.list.subset[[i]][functionalterms.list.subset[[i]][,1] == as.character(functionalterms.list[[i]][j,1]),2],\n                            0)\n      genes.all \u0026lt;- nrow(geneinfo)\n      genes.subset \u0026lt;- nrow(geneinfo.subset)\n\n      fisherexact.matrix \u0026lt;- matrix(c(freq.subset, freq.all - freq.subset,\n                                     genes.subset - freq.subset, genes.all - genes.subset - freq.all + freq.subset),\n                                   nrow = 2,\n                                   ncol = 2)\n      fisher.test \u0026lt;- fisher.test(fisherexact.matrix)\n      \n      term[length(term) + 1] \u0026lt;- as.character(functionalterms.list[[i]][j,1])\n      clusteroccurences[length(clusteroccurences) + 1] \u0026lt;- as.numeric(as.character(freq.subset))\n      genomeoccurences[length(genomeoccurences) + 1] \u0026lt;- as.numeric(as.character(freq.all))\n      pvalue[length(pvalue) + 1] \u0026lt;- as.numeric(as.character(fisher.test$p.value))\n      correctedpvalue[length(correctedpvalue) + 1] \u0026lt;- p.adjust(as.numeric(as.character(fisher.test$p.value)), method = \u0026quot;fdr\u0026quot;, n = nrow(functionalterms.list[[i]]))\n      oddsratio[length(oddsratio) + 1] \u0026lt;- as.numeric(as.character(fisher.test$estimate))\n    }\n  }\n  \n  terms.df \u0026lt;- as.data.frame(cbind(term,\n                                  clusteroccurences,\n                                  genomeoccurences,\n                                  pvalue,\n                                  correctedpvalue,\n                                  oddsratio))\n  terms.df \u0026lt;- terms.df[order(as.numeric(as.character(terms.df$pvalue))),]\n  return(terms.df)\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003eg_legend\u0026lt;-function(a.gplot){ \n  tmp \u0026lt;- ggplot_gtable(ggplot_build(a.gplot)) \n  leg \u0026lt;- which(sapply(tmp$grobs, function(x) x$name) == \"guide-box\") \n  legend \u0026lt;- tmp$grobs[[leg]] \n  return(legend)\n} \n\nget_dendro_structure \u0026lt;- function(result){\n  structure \u0026lt;- hang.dendrogram(as.dendrogram(result$hclust))\n  structure \u0026lt;- capture.output(str(structure))\n  structure \u0026lt;- structure[grepl(\"leaf\", structure)]\n  structure \u0026lt;- as.numeric(as.character(substr(structure, regexpr(\"h=\", structure ) + 3, regexpr(\"  )\", structure))))\n  return(structure)\n}\nget_dendro_data \u0026lt;- function(result){\n  dendro.data \u0026lt;- dendro_data(result$hclust)\n  dendro.data \u0026lt;- dendro.data$segments[which(dendro.data$segments$y == dendro.data$segments$yend),]\n  for(i in 1:nrow(dendro.data)){\n    dendro.data$minx[i] \u0026lt;- min(c(dendro.data$x[i], dendro.data$xend[i]))\n  }\n  dendro.data \u0026lt;- dendro.data[order(as.numeric(as.character(dendro.data$y)), as.numeric(as.character(dendro.data$minx))),]\n  return(dendro.data)\n}\nget_dendro_bootstraps \u0026lt;- function(dendro_data){\n  bootstrap.positions \u0026lt;- as.data.frame(matrix(nrow = length(dendro_data$y[duplicated(dendro_data$y)]),\n                                              ncol = 2))\n  for(i in 1:length(dendro_data$y[duplicated(dendro_data$y)])){\n    dendro_data.subset \u0026lt;- dendro_data[which(dendro_data$y == dendro_data$y[duplicated(dendro_data$y)][i]),]\n    bootstrap.positions[i,1] \u0026lt;- unique(dendro_data.subset$x)\n    bootstrap.positions[i,2] \u0026lt;- unique(dendro_data.subset$y)\n  }\n  return(bootstrap.positions)\n}\nget_heatmap_separators \u0026lt;- function(vector){\n  sep \u0026lt;- c()\n  for(i in 2:length(unique(vector))){\n    sep[length(sep) + 1] \u0026lt;- min(which(vector == unique(vector)[i])) - 1\n  }\n  return(sep)\n}\nformalize_study_names \u0026lt;- function(vector){\n  vector \u0026lt;- gsub(\"wOo_darby_2012\", \"Darby et al., 2012\",vector)\n  vector \u0026lt;- gsub(\"wMel_darby_2014\", \"Darby et al., 2014\",vector)\n  vector \u0026lt;- gsub(\"wDi_luck_2014\", \"Luck et al., 2014\",vector)\n  vector \u0026lt;- gsub(\"wDi_luck_2015\", \"Luck et al., 2015\",vector)\n  vector \u0026lt;- gsub(\"wMel_gutzwiller_2015\", \"Gutzwiller et al., 2015\",vector)\n  vector \u0026lt;- gsub(\"wBm_grote_2017\", \"Grote et al., 2017\",vector)\n  vector \u0026lt;- gsub(\"wBm_chung_2019\", \"Chung et al., 2019\",vector)\n  vector \u0026lt;- factor(vector,levels=c(\"Darby et al., 2012\",\n                                   \"Darby et al., 2014\",\n                                   \"Luck et al., 2014\",\n                                   \"Luck et al., 2015\",\n                                   \"Gutzwiller et al., 2015\",\n                                   \"Grote et al., 2017\",\n                                   \"Chung et al., 2019\"))\n  return(vector)\n}\nfind_soft_power \u0026lt;- function(sft){\n  df \u0026lt;- as.data.frame(cbind(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2]))\n  y \u0026lt;- -sign(sft$fitIndices[,3])*sft$fitIndices[,2]\n  dy \u0026lt;- diff(y) \n  softpower \u0026lt;- which(abs(diff(y)) \u0026lt; 0.05)[1]\n  return(softpower)\n}\neigengene_invert_id \u0026lt;- function(tpm.de.wgcna, mergedColors, mergedMEs){\n  tpm.de.wgcna$invert \u0026lt;- T\n  tpm.de.wgcna$module \u0026lt;- mergedColors\n  for(i in 1:nrow(tpm.de.wgcna)){\n    if(cor(t(tpm.de.wgcna[i,1:(ncol(tpm.de.wgcna) - 2)]), mergedMEs[,which(colnames(mergedMEs) == paste0(\"ME\",tpm.de.wgcna$module[i]))], method = \"pearson\") \u0026gt; 0){\n      tpm.de.wgcna$invert[i] \u0026lt;- F\n    }\n  }\n  return(tpm.de.wgcna)\n}\nwgcna_heatmap_reorder \u0026lt;- function(tpm.de.wgcna){\n  clusters \u0026lt;- as.data.frame(table(tpm.de.wgcna$module))\n  clusters \u0026lt;- clusters[order(-clusters[,2]),1]\n  \n  tpm.de.wgcna.reordered \u0026lt;- as.data.frame(matrix(nrow = 0,\n                                                 ncol = ncol(tpm.de.wgcna)))\n  for(i in 1:length(clusters)){\n    tpm.de.wgcna.reordered \u0026lt;- as.data.frame(rbind(tpm.de.wgcna.reordered,\n                                                  tpm.de.wgcna[tpm.de.wgcna$module == clusters[i] \u0026amp; tpm.de.wgcna$invert == F,],\n                                                  tpm.de.wgcna[tpm.de.wgcna$module == clusters[i] \u0026amp; tpm.de.wgcna$invert == T,]))\n  }\n  return(tpm.de.wgcna.reordered)\n}\nfunctionaltermenrichment \u0026lt;- function(genes, geneinfo){\n  for(i in 1:ncol(geneinfo)){geneinfo[,i] \u0026lt;- as.character(geneinfo[,i])}\n  geneinfo$interpro_description[which(is.na(geneinfo$interpro_description))] \u0026lt;- \"No InterPro entry\"\n  geneinfo$go_biologicalprocess[which(is.na(geneinfo$go_biologicalprocess))] \u0026lt;- \"No GO terms for biological process\"\n  geneinfo$go_cellularcomponent[which(is.na(geneinfo$go_cellularcomponent))] \u0026lt;- \"No GO terms for cellular component\"\n  geneinfo$go_molecularfunction[which(is.na(geneinfo$go_molecularfunction))] \u0026lt;- \"No GO terms for molecular function\"\n  \n  functionalterms.list \u0026lt;- list(ipr=as.data.frame(table(unlist(strsplit(paste(geneinfo$interpro_description, collapse = \"|\"),  split = \"[|]\")))),\n                               gobio=as.data.frame(table(unlist(strsplit(paste(geneinfo$go_biologicalprocess, collapse = \"|\"),  split = \"[|]\")))),\n                               gocell=as.data.frame(table(unlist(strsplit(paste(geneinfo$go_cellularcomponent, collapse = \"|\"),  split = \"[|]\")))),\n                               gomol=as.data.frame(table(unlist(strsplit(paste(geneinfo$go_molecularfunction, collapse = \"|\"),  split = \"[|]\")))))\n  \n  geneinfo.subset \u0026lt;- geneinfo[geneinfo$gene %in% genes,]\n  term \u0026lt;- c()\n  clusteroccurences \u0026lt;- c()\n  genomeoccurences \u0026lt;- c()\n  pvalue \u0026lt;- c()\n  correctedpvalue \u0026lt;- c()\n  oddsratio \u0026lt;- c()\n\n  functionalterms.list.subset \u0026lt;- list(ipr=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$interpro_description, collapse = \"|\"),  split = \"[|]\")))),\n                                      gobio=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$go_biologicalprocess, collapse = \"|\"),  split = \"[|]\")))),\n                                      gocell=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$go_cellularcomponent, collapse = \"|\"),  split = \"[|]\")))),\n                                      gomol=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$go_molecularfunction, collapse = \"|\"),  split = \"[|]\")))))\n  \n  for(i in 1:length(functionalterms.list)){\n    for(j in 1:nrow(functionalterms.list[[i]])){\n      freq.all \u0026lt;- functionalterms.list[[i]][j,2]\n      freq.subset \u0026lt;- ifelse(functionalterms.list[[i]][j,1] %in% functionalterms.list.subset[[i]][,1],\n                            functionalterms.list.subset[[i]][functionalterms.list.subset[[i]][,1] == as.character(functionalterms.list[[i]][j,1]),2],\n                            0)\n      genes.all \u0026lt;- nrow(geneinfo)\n      genes.subset \u0026lt;- nrow(geneinfo.subset)\n\n      fisherexact.matrix \u0026lt;- matrix(c(freq.subset, freq.all - freq.subset,\n                                     genes.subset - freq.subset, genes.all - genes.subset - freq.all + freq.subset),\n                                   nrow = 2,\n                                   ncol = 2)\n      fisher.test \u0026lt;- fisher.test(fisherexact.matrix)\n      \n      term[length(term) + 1] \u0026lt;- as.character(functionalterms.list[[i]][j,1])\n      clusteroccurences[length(clusteroccurences) + 1] \u0026lt;- as.numeric(as.character(freq.subset))\n      genomeoccurences[length(genomeoccurences) + 1] \u0026lt;- as.numeric(as.character(freq.all))\n      pvalue[length(pvalue) + 1] \u0026lt;- as.numeric(as.character(fisher.test$p.value))\n      correctedpvalue[length(correctedpvalue) + 1] \u0026lt;- p.adjust(as.numeric(as.character(fisher.test$p.value)), method = \"fdr\", n = nrow(functionalterms.list[[i]]))\n      oddsratio[length(oddsratio) + 1] \u0026lt;- as.numeric(as.character(fisher.test$estimate))\n    }\n  }\n  \n  terms.df \u0026lt;- as.data.frame(cbind(term,\n                                  clusteroccurences,\n                                  genomeoccurences,\n                                  pvalue,\n                                  correctedpvalue,\n                                  oddsratio))\n  terms.df \u0026lt;- terms.df[order(as.numeric(as.character(terms.df$pvalue))),]\n  return(terms.df)\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eProcess counts files and calculate TPM values  \u003ca name=\"user-content-ranalysis_de_counts_readcounts\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-process-counts-files-and-calculate-tpm-values--\" class=\"anchor\" aria-label=\"Permalink: Process counts files and calculate TPM values  \" href=\"#process-counts-files-and-calculate-tpm-values--\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRead counts table into a list \u003ca name=\"user-content-ranalysis_de_counts_readcounts\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-read-counts-table-into-a-list-\" class=\"anchor\" aria-label=\"Permalink: Read counts table into a list \" href=\"#read-counts-table-into-a-list-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"sample_map \u0026lt;- read.delim(SAMPLE_MAP.PATH)\n\ncounts.list \u0026lt;- list(wOo_darby_2012=\u0026quot;\u0026quot;,\n                    wMel_darby_2014=\u0026quot;\u0026quot;,\n                    wDi_luck_2014=\u0026quot;\u0026quot;,\n                    wDi_luck_2015=\u0026quot;\u0026quot;,\n                    wMel_gutzwiller_2015=\u0026quot;\u0026quot;,\n                    wBm_grote_2017=\u0026quot;\u0026quot;,\n                    wBm_chung_2019=\u0026quot;\u0026quot;)\nfor(i in 1:length(counts.list)){\n  counts.list[[i]] \u0026lt;- read.delim(paste0(COUNTS.DIR,\u0026quot;/\u0026quot;,names(counts.list)[i],\u0026quot;_counts.tsv\u0026quot;),\n                                 row.names = 1)\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003esample_map \u0026lt;- read.delim(SAMPLE_MAP.PATH)\n\ncounts.list \u0026lt;- list(wOo_darby_2012=\"\",\n                    wMel_darby_2014=\"\",\n                    wDi_luck_2014=\"\",\n                    wDi_luck_2015=\"\",\n                    wMel_gutzwiller_2015=\"\",\n                    wBm_grote_2017=\"\",\n                    wBm_chung_2019=\"\")\nfor(i in 1:length(counts.list)){\n  counts.list[[i]] \u0026lt;- read.delim(paste0(COUNTS.DIR,\"/\",names(counts.list)[i],\"_counts.tsv\"),\n                                 row.names = 1)\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eExclude non-protein-coding genes from counts table \u003ca name=\"user-content-ranalysis_de_counts_removenonproteingenes\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-exclude-non-protein-coding-genes-from-counts-table-\" class=\"anchor\" aria-label=\"Permalink: Exclude non-protein-coding genes from counts table \" href=\"#exclude-non-protein-coding-genes-from-counts-table-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(counts.list)){\n  species \u0026lt;- substr(names(counts.list)[i],1,regexpr(\u0026quot;_\u0026quot;,names(counts.list[i]))-1)\n  if(species != \u0026quot;wDi\u0026quot;){\n    map \u0026lt;- read.delim(paste0(GFF3MAP.DIR,\u0026quot;/\u0026quot;,species,\u0026quot;.gff.map\u0026quot;),\n                      header = T,\n                      sep = \u0026quot;\\t\u0026quot;)\n    map \u0026lt;- map[map$gene_biotype == \u0026quot;protein_coding\u0026quot; \u0026amp; !is.na(map$gene_biotype),]\n    map$ID \u0026lt;- gsub(\u0026quot;.*[|]cds\u0026quot;,\u0026quot;cds\u0026quot;,map$ID)\n    counts.list[[i]] \u0026lt;- counts.list[[i]][rownames(counts.list[[i]]) %in% map$ID,]\n  }\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(counts.list)){\n  species \u0026lt;- substr(names(counts.list)[i],1,regexpr(\"_\",names(counts.list[i]))-1)\n  if(species != \"wDi\"){\n    map \u0026lt;- read.delim(paste0(GFF3MAP.DIR,\"/\",species,\".gff.map\"),\n                      header = T,\n                      sep = \"\\t\")\n    map \u0026lt;- map[map$gene_biotype == \"protein_coding\" \u0026amp; !is.na(map$gene_biotype),]\n    map$ID \u0026lt;- gsub(\".*[|]cds\",\"cds\",map$ID)\n    counts.list[[i]] \u0026lt;- counts.list[[i]][rownames(counts.list[[i]]) %in% map$ID,]\n  }\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eExclude genes without Agilent SureSelect probes from counts table in Chung et al 2019 study \u003ca name=\"user-content-ranalysis_de_counts_removenoprobegenes\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-exclude-genes-without-agilent-sureselect-probes-from-counts-table-in-chung-et-al-2019-study-\" class=\"anchor\" aria-label=\"Permalink: Exclude genes without Agilent SureSelect probes from counts table in Chung et al 2019 study \" href=\"#exclude-genes-without-agilent-sureselect-probes-from-counts-table-in-chung-et-al-2019-study-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"exclude.genes \u0026lt;- c(\u0026quot;cds28\u0026quot;,\u0026quot;cds31\u0026quot;,\u0026quot;cds63\u0026quot;,\u0026quot;cds64\u0026quot;,\u0026quot;cds74\u0026quot;,\u0026quot;cds99\u0026quot;,\u0026quot;cds118\u0026quot;,\u0026quot;cds127\u0026quot;,\u0026quot;cds135\u0026quot;,\u0026quot;cds168\u0026quot;,\u0026quot;cds169\u0026quot;,\u0026quot;cds170\u0026quot;,\u0026quot;cds171\u0026quot;,\u0026quot;cds185\u0026quot;,\u0026quot;cds240\u0026quot;,\u0026quot;cds245\u0026quot;,\u0026quot;cds246\u0026quot;,\u0026quot;cds307\u0026quot;,\u0026quot;cds309\u0026quot;,\u0026quot;cds321\u0026quot;,\u0026quot;cds322\u0026quot;,\u0026quot;cds324\u0026quot;,\u0026quot;cds328\u0026quot;,\u0026quot;cds354\u0026quot;,\u0026quot;cds359\u0026quot;,\u0026quot;cds360\u0026quot;,\u0026quot;cds363\u0026quot;,\u0026quot;cds371\u0026quot;,\u0026quot;cds372\u0026quot;,\u0026quot;cds380\u0026quot;,\u0026quot;cds426\u0026quot;,\u0026quot;cds427\u0026quot;,\u0026quot;cds431\u0026quot;,\u0026quot;cds444\u0026quot;,\u0026quot;cds452\u0026quot;,\u0026quot;cds453\u0026quot;,\u0026quot;cds466\u0026quot;,\u0026quot;cds490\u0026quot;,\u0026quot;cds491\u0026quot;,\u0026quot;cds492\u0026quot;,\u0026quot;cds493\u0026quot;,\u0026quot;cds500\u0026quot;,\u0026quot;cds543\u0026quot;,\u0026quot;cds573\u0026quot;,\u0026quot;cds581\u0026quot;,\u0026quot;cds607\u0026quot;,\u0026quot;cds608\u0026quot;,\u0026quot;cds614\u0026quot;,\u0026quot;cds622\u0026quot;,\u0026quot;cds635\u0026quot;,\u0026quot;cds636\u0026quot;,\u0026quot;cds638\u0026quot;,\u0026quot;cds643\u0026quot;,\u0026quot;cds657\u0026quot;,\u0026quot;cds658\u0026quot;,\u0026quot;cds733\u0026quot;,\u0026quot;cds734\u0026quot;,\u0026quot;cds737\u0026quot;,\u0026quot;cds746\u0026quot;,\u0026quot;cds750\u0026quot;,\u0026quot;cds774\u0026quot;,\u0026quot;cds775\u0026quot;,\u0026quot;cds784\u0026quot;,\u0026quot;cds785\u0026quot;,\u0026quot;cds787\u0026quot;,\u0026quot;cds805\u0026quot;,\u0026quot;cds816\u0026quot;,\u0026quot;cds817\u0026quot;,\u0026quot;cds822\u0026quot;,\u0026quot;cds828\u0026quot;,\u0026quot;cds851\u0026quot;,\u0026quot;cds853\u0026quot;,\u0026quot;cds870\u0026quot;,\u0026quot;cds871\u0026quot;,\u0026quot;cds872\u0026quot;,\u0026quot;cds874\u0026quot;,\u0026quot;cds886\u0026quot;,\u0026quot;cds909\u0026quot;,\u0026quot;cds910\u0026quot;,\u0026quot;cds932\u0026quot;,\u0026quot;cds957\u0026quot;,\u0026quot;cds963\u0026quot;,\u0026quot;cds965\u0026quot;,\u0026quot;cds978\u0026quot;,\u0026quot;cds979\u0026quot;)\n\ncounts.list$wBm_chung_2019 \u0026lt;- counts.list$wBm_chung_2019[!(rownames(counts.list$wBm_chung_2019) %in% exclude.genes),]\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003eexclude.genes \u0026lt;- c(\"cds28\",\"cds31\",\"cds63\",\"cds64\",\"cds74\",\"cds99\",\"cds118\",\"cds127\",\"cds135\",\"cds168\",\"cds169\",\"cds170\",\"cds171\",\"cds185\",\"cds240\",\"cds245\",\"cds246\",\"cds307\",\"cds309\",\"cds321\",\"cds322\",\"cds324\",\"cds328\",\"cds354\",\"cds359\",\"cds360\",\"cds363\",\"cds371\",\"cds372\",\"cds380\",\"cds426\",\"cds427\",\"cds431\",\"cds444\",\"cds452\",\"cds453\",\"cds466\",\"cds490\",\"cds491\",\"cds492\",\"cds493\",\"cds500\",\"cds543\",\"cds573\",\"cds581\",\"cds607\",\"cds608\",\"cds614\",\"cds622\",\"cds635\",\"cds636\",\"cds638\",\"cds643\",\"cds657\",\"cds658\",\"cds733\",\"cds734\",\"cds737\",\"cds746\",\"cds750\",\"cds774\",\"cds775\",\"cds784\",\"cds785\",\"cds787\",\"cds805\",\"cds816\",\"cds817\",\"cds822\",\"cds828\",\"cds851\",\"cds853\",\"cds870\",\"cds871\",\"cds872\",\"cds874\",\"cds886\",\"cds909\",\"cds910\",\"cds932\",\"cds957\",\"cds963\",\"cds965\",\"cds978\",\"cds979\")\n\ncounts.list$wBm_chung_2019 \u0026lt;- counts.list$wBm_chung_2019[!(rownames(counts.list$wBm_chung_2019) %in% exclude.genes),]\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePrint the number of reads mapping to protein-coding genes for each sample in each study \u003ca name=\"user-content-ranalysis_de_counts_countscolsum\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-print-the-number-of-reads-mapping-to-protein-coding-genes-for-each-sample-in-each-study-\" class=\"anchor\" aria-label=\"Permalink: Print the number of reads mapping to protein-coding genes for each sample in each study \" href=\"#print-the-number-of-reads-mapping-to-protein-coding-genes-for-each-sample-in-each-study-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(counts.list)){\n  print(names(counts.list)[i])\n  print(colSums(counts.list[[i]]))\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(counts.list)){\n  print(names(counts.list)[i])\n  print(colSums(counts.list[[i]]))\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[1] \u0026quot;wOo_darby_2012\u0026quot;\n  darby_woo_adultmale_a   darby_woo_adultmale_b darby_woo_adultfemale_a darby_woo_adultfemale_b \n               26067.08                83692.20               146642.65                80723.69 \n[1] \u0026quot;wMel_darby_2014\u0026quot;\n  darby_wmel_doxy_a   darby_wmel_doxy_b   darby_wmel_doxy_c darby_wmel_nodoxy_c darby_wmel_nodoxy_a darby_wmel_nodoxy_b \n          120344.75            72768.46            73974.65            64291.14            42960.91            82329.35 \n[1] \u0026quot;wDi_luck_2014\u0026quot;\n         luck_wdi_L3_a          luck_wdi_L4_a   luck_wdi_adultmale_a luck_wdi_adultfemale_a          luck_wdi_MF_a \n                495.32                1238.56                3788.02               12840.55               27118.30 \n[1] \u0026quot;wDi_luck_2015\u0026quot;\n luck_wdi_adultfemale_bodywall_a  luck_wdi_adultfemale_bodywall_b      luck_wdi_adultfemale_head_a      luck_wdi_adultfemale_head_b \n                         6691.13                          8436.58                          4045.55                          6019.27 \nluck_wdi_adultfemale_intestine_a luck_wdi_adultfemale_intestine_b    luck_wdi_adultfemale_uterus_a    luck_wdi_adultfemale_uterus_b \n                           14.34                            11.18                          1785.09                           188.19 \n   luck_wdi_adultmale_bodywall_a    luck_wdi_adultmale_bodywall_b   luck_wdi_adultmale_intestine_a      luck_wdi_adultmale_testis_a \n                        14444.09                         10506.72                            36.70                            13.34 \n[1] \u0026quot;wMel_gutzwiller_2015\u0026quot;\n                 gutzwiller_wmel_embryo0to2hr_a                  gutzwiller_wmel_embryo0to2hr_b                  gutzwiller_wmel_embryo2to4hr_a \n                                      114790.66                                        85945.65                                       110702.61 \n                 gutzwiller_wmel_embryo2to4hr_b                gutzwiller_wmel_embryo10to12hr_a                gutzwiller_wmel_embryo10to12hr_b \n                                       53392.96                                       188273.14                                       134967.66 \n               gutzwiller_wmel_embryo12to14hr_a                gutzwiller_wmel_embryo12to14hr_b                gutzwiller_wmel_embryo18to20hr_a \n                                      137773.04                                        40542.10                                       177066.53 \n               gutzwiller_wmel_embryo18to20hr_b                gutzwiller_wmel_embryo22to24hr_a                gutzwiller_wmel_embryo22to24hr_b \n                                      148100.78                                       192603.87                                        18862.07 \n                           gutzwiller_wmel_L1_a                            gutzwiller_wmel_L1_b                            gutzwiller_wmel_L2_a \n                                      153643.74                                       128799.29                                        65018.60 \n                           gutzwiller_wmel_L2_b               gutzwiller_wmel_L3_12hrpostmolt_a               gutzwiller_wmel_L3_12hrpostmolt_b \n                                       55155.07                                        72087.91                                        45411.64 \n        gutzwiller_wmel_L3_darkbluegutPS_1to2_a         gutzwiller_wmel_L3_darkbluegutPS_1to2_b        gutzwiller_wmel_L3_lightbluegutPS_3to6_a \n                                      122280.34                                        90161.56                                       152077.10 \n       gutzwiller_wmel_L3_lightbluegutPS_3to6_b            gutzwiller_wmel_L3_cleargutPS_7to9_a            gutzwiller_wmel_L3_cleargutPS_7to9_b \n                                      112893.91                                       241950.76                                        76163.54 \n                          gutzwiller_wmel_WPP_a                           gutzwiller_wmel_WPP_b             gutzwiller_wmel_pupae_12hrpostWPP_a \n                                      178456.20                                       115149.63                                       162560.29 \n            gutzwiller_wmel_pupae_12hrpostWPP_b             gutzwiller_wmel_pupae_24hrpostWPP_a             gutzwiller_wmel_pupae_24hrpostWPP_b \n                                       92488.90                                       112776.08                                        48500.17 \n           gutzwiller_wmel_pupae_2dayspostWPP_a            gutzwiller_wmel_pupae_2dayspostWPP_b            gutzwiller_wmel_pupae_3dayspostWPP_a \n                                      106883.82                                        60639.46                                       203627.16 \n           gutzwiller_wmel_pupae_3dayspostWPP_b            gutzwiller_wmel_pupae_4dayspostWPP_a            gutzwiller_wmel_pupae_4dayspostWPP_b \n                                       76254.69                                       105137.01                                        95285.08 \n gutzwiller_wmel_adultfemale_1dayposteclosion_a  gutzwiller_wmel_adultfemale_1dayposteclosion_b  gutzwiller_wmel_adultfemale_5dayposteclosion_a \n                                      254985.87                                       140491.93                                       221664.24 \n gutzwiller_wmel_adultfemale_5dayposteclosion_b  gutzwiller_wmel_adultfemale_5dayposteclosion_c  gutzwiller_wmel_adultfemale_5dayposteclosion_d \n                                      111503.67                                       103197.92                                       180473.06 \n gutzwiller_wmel_adultfemale_5dayposteclosion_f gutzwiller_wmel_adultfemale_30dayposteclosion_a gutzwiller_wmel_adultfemale_30dayposteclosion_b \n                                      195493.19                                       341687.94                                       244695.15 \n   gutzwiller_wmel_adultmale_1dayposteclosion_a    gutzwiller_wmel_adultmale_1dayposteclosion_b    gutzwiller_wmel_adultmale_5dayposteclosion_a \n                                      225874.52                                       137847.49                                       417013.04 \n   gutzwiller_wmel_adultmale_5dayposteclosion_b    gutzwiller_wmel_adultmale_5dayposteclosion_c    gutzwiller_wmel_adultmale_5dayposteclosion_d \n                                      136930.51                                       328285.55                                       222262.97 \n   gutzwiller_wmel_adultmale_5dayposteclosion_i    gutzwiller_wmel_adultmale_5dayposteclosion_j   gutzwiller_wmel_adultmale_30dayposteclosion_a \n                                       88320.71                                        93761.47                                       325975.98 \n  gutzwiller_wmel_adultmale_30dayposteclosion_b \n                                      321449.11 \n[1] \u0026quot;wBm_grote_2017\u0026quot;\n                  grote_wbm_L4_a                   grote_wbm_L4_b   grote_wbm_30dpi_immaturemale_a   grote_wbm_30dpi_immaturemale_b \n                        22374.59                          6178.84                         12843.80                          9650.02 \ngrote_wbm_30dpi_immaturefemale_a grote_wbm_30dpi_immaturefemale_b   grote_wbm_42dpi_immaturemale_a   grote_wbm_42dpi_immaturemale_b \n                         9080.46                         14064.82                         13019.66                          8636.27 \ngrote_wbm_42dpi_immaturefemale_a grote_wbm_42dpi_immaturefemale_b     grote_wbm_120dpi_adultmale_a     grote_wbm_120dpi_adultmale_b \n                         6721.62                         13460.39                         16643.58                          8373.13 \n  grote_wbm_120dpi_adultfemale_a   grote_wbm_120dpi_adultfemale_b \n                        31289.42                         19151.90 \n[1] \u0026quot;wBm_chung_2019\u0026quot;\n               chung_wbm_vector18hpi_a                chung_wbm_vector18hpi_b                 chung_wbm_vector4dpi_a \n                             481351.25                              625028.83                              240693.73 \n                chung_wbm_vector4dpi_b                 chung_wbm_vector8dpi_a                 chung_wbm_vector8dpi_b \n                             777934.24                             3218498.46                             3987678.06 \n                chung_wbm_mammal1dpi_a                 chung_wbm_mammal1dpi_b                 chung_wbm_mammal2dpi_a \n                             159809.14                              103367.39                               53272.07 \n                chung_wbm_mammal2dpi_b                 chung_wbm_mammal3dpi_a                 chung_wbm_mammal3dpi_b \n                             123961.25                              109225.39                              122560.27 \n                chung_wbm_mammal4dpi_a                 chung_wbm_mammal4dpi_b                 chung_wbm_mammal8dpi_a \n                             120910.97                              121522.47                              284943.40 \n                chung_wbm_mammal8dpi_b   chung_wbm_mammal20dpi_immaturemale_a   chung_wbm_mammal20dpi_immaturemale_b \n                             221120.86                              108117.16                               78688.09 \nchung_wbm_mammal24dpi_immaturefemale_a chung_wbm_mammal24dpi_immaturefemale_b                  chung_wbm_adultmale_a \n                           35276890.78                              192206.48                               23790.09 \n                 chung_wbm_adultmale_b                chung_wbm_adultfemale_a                chung_wbm_adultfemale_b \n                              43446.49                            18840207.11                               10561.16 \n               chung_wbm_adultfemale_c                chung_wbm_adultfemale_d                     chung_wbm_embryo_a \n                              13947.67                            24661817.10                                8525.69 \n                    chung_wbm_embryo_b                 chung_wbm_immatureMF_a                 chung_wbm_immatureMF_b \n                              14704.98                               65069.68                               57976.58 \n                  chung_wbm_matureMF_a                   chung_wbm_matureMF_b \n                              46554.09                               51856.71 \n\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003e[1] \"wOo_darby_2012\"\n  darby_woo_adultmale_a   darby_woo_adultmale_b darby_woo_adultfemale_a darby_woo_adultfemale_b \n               26067.08                83692.20               146642.65                80723.69 \n[1] \"wMel_darby_2014\"\n  darby_wmel_doxy_a   darby_wmel_doxy_b   darby_wmel_doxy_c darby_wmel_nodoxy_c darby_wmel_nodoxy_a darby_wmel_nodoxy_b \n          120344.75            72768.46            73974.65            64291.14            42960.91            82329.35 \n[1] \"wDi_luck_2014\"\n         luck_wdi_L3_a          luck_wdi_L4_a   luck_wdi_adultmale_a luck_wdi_adultfemale_a          luck_wdi_MF_a \n                495.32                1238.56                3788.02               12840.55               27118.30 \n[1] \"wDi_luck_2015\"\n luck_wdi_adultfemale_bodywall_a  luck_wdi_adultfemale_bodywall_b      luck_wdi_adultfemale_head_a      luck_wdi_adultfemale_head_b \n                         6691.13                          8436.58                          4045.55                          6019.27 \nluck_wdi_adultfemale_intestine_a luck_wdi_adultfemale_intestine_b    luck_wdi_adultfemale_uterus_a    luck_wdi_adultfemale_uterus_b \n                           14.34                            11.18                          1785.09                           188.19 \n   luck_wdi_adultmale_bodywall_a    luck_wdi_adultmale_bodywall_b   luck_wdi_adultmale_intestine_a      luck_wdi_adultmale_testis_a \n                        14444.09                         10506.72                            36.70                            13.34 \n[1] \"wMel_gutzwiller_2015\"\n                 gutzwiller_wmel_embryo0to2hr_a                  gutzwiller_wmel_embryo0to2hr_b                  gutzwiller_wmel_embryo2to4hr_a \n                                      114790.66                                        85945.65                                       110702.61 \n                 gutzwiller_wmel_embryo2to4hr_b                gutzwiller_wmel_embryo10to12hr_a                gutzwiller_wmel_embryo10to12hr_b \n                                       53392.96                                       188273.14                                       134967.66 \n               gutzwiller_wmel_embryo12to14hr_a                gutzwiller_wmel_embryo12to14hr_b                gutzwiller_wmel_embryo18to20hr_a \n                                      137773.04                                        40542.10                                       177066.53 \n               gutzwiller_wmel_embryo18to20hr_b                gutzwiller_wmel_embryo22to24hr_a                gutzwiller_wmel_embryo22to24hr_b \n                                      148100.78                                       192603.87                                        18862.07 \n                           gutzwiller_wmel_L1_a                            gutzwiller_wmel_L1_b                            gutzwiller_wmel_L2_a \n                                      153643.74                                       128799.29                                        65018.60 \n                           gutzwiller_wmel_L2_b               gutzwiller_wmel_L3_12hrpostmolt_a               gutzwiller_wmel_L3_12hrpostmolt_b \n                                       55155.07                                        72087.91                                        45411.64 \n        gutzwiller_wmel_L3_darkbluegutPS_1to2_a         gutzwiller_wmel_L3_darkbluegutPS_1to2_b        gutzwiller_wmel_L3_lightbluegutPS_3to6_a \n                                      122280.34                                        90161.56                                       152077.10 \n       gutzwiller_wmel_L3_lightbluegutPS_3to6_b            gutzwiller_wmel_L3_cleargutPS_7to9_a            gutzwiller_wmel_L3_cleargutPS_7to9_b \n                                      112893.91                                       241950.76                                        76163.54 \n                          gutzwiller_wmel_WPP_a                           gutzwiller_wmel_WPP_b             gutzwiller_wmel_pupae_12hrpostWPP_a \n                                      178456.20                                       115149.63                                       162560.29 \n            gutzwiller_wmel_pupae_12hrpostWPP_b             gutzwiller_wmel_pupae_24hrpostWPP_a             gutzwiller_wmel_pupae_24hrpostWPP_b \n                                       92488.90                                       112776.08                                        48500.17 \n           gutzwiller_wmel_pupae_2dayspostWPP_a            gutzwiller_wmel_pupae_2dayspostWPP_b            gutzwiller_wmel_pupae_3dayspostWPP_a \n                                      106883.82                                        60639.46                                       203627.16 \n           gutzwiller_wmel_pupae_3dayspostWPP_b            gutzwiller_wmel_pupae_4dayspostWPP_a            gutzwiller_wmel_pupae_4dayspostWPP_b \n                                       76254.69                                       105137.01                                        95285.08 \n gutzwiller_wmel_adultfemale_1dayposteclosion_a  gutzwiller_wmel_adultfemale_1dayposteclosion_b  gutzwiller_wmel_adultfemale_5dayposteclosion_a \n                                      254985.87                                       140491.93                                       221664.24 \n gutzwiller_wmel_adultfemale_5dayposteclosion_b  gutzwiller_wmel_adultfemale_5dayposteclosion_c  gutzwiller_wmel_adultfemale_5dayposteclosion_d \n                                      111503.67                                       103197.92                                       180473.06 \n gutzwiller_wmel_adultfemale_5dayposteclosion_f gutzwiller_wmel_adultfemale_30dayposteclosion_a gutzwiller_wmel_adultfemale_30dayposteclosion_b \n                                      195493.19                                       341687.94                                       244695.15 \n   gutzwiller_wmel_adultmale_1dayposteclosion_a    gutzwiller_wmel_adultmale_1dayposteclosion_b    gutzwiller_wmel_adultmale_5dayposteclosion_a \n                                      225874.52                                       137847.49                                       417013.04 \n   gutzwiller_wmel_adultmale_5dayposteclosion_b    gutzwiller_wmel_adultmale_5dayposteclosion_c    gutzwiller_wmel_adultmale_5dayposteclosion_d \n                                      136930.51                                       328285.55                                       222262.97 \n   gutzwiller_wmel_adultmale_5dayposteclosion_i    gutzwiller_wmel_adultmale_5dayposteclosion_j   gutzwiller_wmel_adultmale_30dayposteclosion_a \n                                       88320.71                                        93761.47                                       325975.98 \n  gutzwiller_wmel_adultmale_30dayposteclosion_b \n                                      321449.11 \n[1] \"wBm_grote_2017\"\n                  grote_wbm_L4_a                   grote_wbm_L4_b   grote_wbm_30dpi_immaturemale_a   grote_wbm_30dpi_immaturemale_b \n                        22374.59                          6178.84                         12843.80                          9650.02 \ngrote_wbm_30dpi_immaturefemale_a grote_wbm_30dpi_immaturefemale_b   grote_wbm_42dpi_immaturemale_a   grote_wbm_42dpi_immaturemale_b \n                         9080.46                         14064.82                         13019.66                          8636.27 \ngrote_wbm_42dpi_immaturefemale_a grote_wbm_42dpi_immaturefemale_b     grote_wbm_120dpi_adultmale_a     grote_wbm_120dpi_adultmale_b \n                         6721.62                         13460.39                         16643.58                          8373.13 \n  grote_wbm_120dpi_adultfemale_a   grote_wbm_120dpi_adultfemale_b \n                        31289.42                         19151.90 \n[1] \"wBm_chung_2019\"\n               chung_wbm_vector18hpi_a                chung_wbm_vector18hpi_b                 chung_wbm_vector4dpi_a \n                             481351.25                              625028.83                              240693.73 \n                chung_wbm_vector4dpi_b                 chung_wbm_vector8dpi_a                 chung_wbm_vector8dpi_b \n                             777934.24                             3218498.46                             3987678.06 \n                chung_wbm_mammal1dpi_a                 chung_wbm_mammal1dpi_b                 chung_wbm_mammal2dpi_a \n                             159809.14                              103367.39                               53272.07 \n                chung_wbm_mammal2dpi_b                 chung_wbm_mammal3dpi_a                 chung_wbm_mammal3dpi_b \n                             123961.25                              109225.39                              122560.27 \n                chung_wbm_mammal4dpi_a                 chung_wbm_mammal4dpi_b                 chung_wbm_mammal8dpi_a \n                             120910.97                              121522.47                              284943.40 \n                chung_wbm_mammal8dpi_b   chung_wbm_mammal20dpi_immaturemale_a   chung_wbm_mammal20dpi_immaturemale_b \n                             221120.86                              108117.16                               78688.09 \nchung_wbm_mammal24dpi_immaturefemale_a chung_wbm_mammal24dpi_immaturefemale_b                  chung_wbm_adultmale_a \n                           35276890.78                              192206.48                               23790.09 \n                 chung_wbm_adultmale_b                chung_wbm_adultfemale_a                chung_wbm_adultfemale_b \n                              43446.49                            18840207.11                               10561.16 \n               chung_wbm_adultfemale_c                chung_wbm_adultfemale_d                     chung_wbm_embryo_a \n                              13947.67                            24661817.10                                8525.69 \n                    chung_wbm_embryo_b                 chung_wbm_immatureMF_a                 chung_wbm_immatureMF_b \n                              14704.98                               65069.68                               57976.58 \n                  chung_wbm_matureMF_a                   chung_wbm_matureMF_b \n                              46554.09                               51856.71 \n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCreate TPM table \u003ca name=\"user-content-ranalysis_de_counts_tpm\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-create-tpm-table-\" class=\"anchor\" aria-label=\"Permalink: Create TPM table \" href=\"#create-tpm-table-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"tpm.list \u0026lt;- counts.list\n\nfor(i in 1:length(tpm.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(counts.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]\n  srr \u0026lt;- sample_map.subset$sra_id[sample_map.subset$sample_identifier == colnames(counts.list[[i]])[1]]\n  raw_counts.file \u0026lt;- read.delim(paste0(RAW_COUNTS.DIR,\u0026quot;/\u0026quot;,srr[1],\u0026quot;.sortedbyposition.counts.txt\u0026quot;),\n                                      header = T)\n  genelength \u0026lt;- raw_counts.file$uniq_len[match(rownames(tpm.list[[i]]),raw_counts.file$featureID)]\n  \n  for(j in 1:ncol(tpm.list[[i]])){\n    tpm.list[[i]][,j] \u0026lt;- tpm.list[[i]][,j]/genelength\n    tpm.list[[i]][,j] \u0026lt;- tpm.list[[i]][,j]/(sum(tpm.list[[i]][,j])/1000000)\n  }\n  \n  write.table(tpm.list[[i]],\n            paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.list)[i],\u0026quot;_tpm.tsv\u0026quot;),\n            row.names = T,\n            col.names = T,\n            quote = F,\n            sep = \u0026quot;\\t\u0026quot;)\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003etpm.list \u0026lt;- counts.list\n\nfor(i in 1:length(tpm.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(counts.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]\n  srr \u0026lt;- sample_map.subset$sra_id[sample_map.subset$sample_identifier == colnames(counts.list[[i]])[1]]\n  raw_counts.file \u0026lt;- read.delim(paste0(RAW_COUNTS.DIR,\"/\",srr[1],\".sortedbyposition.counts.txt\"),\n                                      header = T)\n  genelength \u0026lt;- raw_counts.file$uniq_len[match(rownames(tpm.list[[i]]),raw_counts.file$featureID)]\n  \n  for(j in 1:ncol(tpm.list[[i]])){\n    tpm.list[[i]][,j] \u0026lt;- tpm.list[[i]][,j]/genelength\n    tpm.list[[i]][,j] \u0026lt;- tpm.list[[i]][,j]/(sum(tpm.list[[i]][,j])/1000000)\n  }\n  \n  write.table(tpm.list[[i]],\n            paste0(OUTPUT.DIR,\"/\",names(tpm.list)[i],\"_tpm.tsv\"),\n            row.names = T,\n            col.names = T,\n            quote = F,\n            sep = \"\\t\")\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCreate keys \u003ca name=\"user-content-ranalysis_de_keys\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-create-keys-\" class=\"anchor\" aria-label=\"Permalink: Create keys \" href=\"#create-keys-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSample keys \u003ca name=\"user-content-ranalysis_de_keys_samples\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-sample-keys-\" class=\"anchor\" aria-label=\"Permalink: Sample keys \" href=\"#sample-keys-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(counts.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(counts.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  legend.plot \u0026lt;- ggplot()+\n    geom_point(aes(x=groups[,2], y=seq(1,length(groups[,2]),1), color = groups[,2]), show.legend = T, size = 2)+\n    scale_color_manual(values = as.character(unique(groups[,3])))+\n    guides(colour = guide_legend(title = \u0026quot;Samples\u0026quot;, title.position = \u0026quot;top\u0026quot;, nrow = 6))+\n    theme_bw()+\n    theme(legend.position=\u0026quot;top\u0026quot;,legend.title.align=0.5)\n  \n  sample.legend \u0026lt;- g_legend(legend.plot)\n  \n  pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.list)[i],\u0026quot;_samplekey.pdf\u0026quot;),\n      height=2,\n      width=10)\n  grid.arrange(sample.legend)\n  dev.off()\n  \n  grid.arrange(sample.legend)\n}\"\u003e\u003cpre lang=\"{R,fig.height=2,fig.width=10}\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(counts.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(counts.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  legend.plot \u0026lt;- ggplot()+\n    geom_point(aes(x=groups[,2], y=seq(1,length(groups[,2]),1), color = groups[,2]), show.legend = T, size = 2)+\n    scale_color_manual(values = as.character(unique(groups[,3])))+\n    guides(colour = guide_legend(title = \"Samples\", title.position = \"top\", nrow = 6))+\n    theme_bw()+\n    theme(legend.position=\"top\",legend.title.align=0.5)\n  \n  sample.legend \u0026lt;- g_legend(legend.plot)\n  \n  pdf(paste0(OUTPUT.DIR,\"/\",names(counts.list)[i],\"_samplekey.pdf\"),\n      height=2,\n      width=10)\n  grid.arrange(sample.legend)\n  dev.off()\n  \n  grid.arrange(sample.legend)\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewOo, Darby et al 2012\u003c/h5\u003e\u003ca id=\"user-content-woo-darby-et-al-2012\" class=\"anchor\" aria-label=\"Permalink: wOo, Darby et al 2012\" href=\"#woo-darby-et-al-2012\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_samplekey.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_samplekey.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Darby et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wmel-darby-et-al-2014\" class=\"anchor\" aria-label=\"Permalink: wMel, Darby et al 2014\" href=\"#wmel-darby-et-al-2014\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_samplekey.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_samplekey.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2014\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2014\" href=\"#wdi-luck-et-al-2014\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_samplekey.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_samplekey.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2015\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2015\" href=\"#wdi-luck-et-al-2015\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_samplekey.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_samplekey.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Gutzwiller et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wmel-gutzwiller-et-al-2015\" class=\"anchor\" aria-label=\"Permalink: wMel, Gutzwiller et al 2015\" href=\"#wmel-gutzwiller-et-al-2015\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_samplekey.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_samplekey.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Grote et al 2017\u003c/h5\u003e\u003ca id=\"user-content-wbm-grote-et-al-2017\" class=\"anchor\" aria-label=\"Permalink: wBm, Grote et al 2017\" href=\"#wbm-grote-et-al-2017\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_samplekey.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_samplekey.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Chung et al 2019\u003c/h5\u003e\u003ca id=\"user-content-wbm-chung-et-al-2019\" class=\"anchor\" aria-label=\"Permalink: wBm, Chung et al 2019\" href=\"#wbm-chung-et-al-2019\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_samplekey.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_samplekey.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSize key  \u003ca name=\"user-content-ranalysis_de_keys_size\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-size-key--\" class=\"anchor\" aria-label=\"Permalink: Size key  \" href=\"#size-key--\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"max_lib_sizes \u0026lt;- c()\nmin_lib_sizes \u0026lt;- c()\nfor(i in 1:length(counts.list)){\n  max_lib_sizes \u0026lt;- c(max_lib_sizes,max(colSums(counts.list[[i]])))\n  min_lib_sizes \u0026lt;- c(min_lib_sizes,min(colSums(counts.list[[i]])))\n}\nsize_limits \u0026lt;- c(1,10^ceiling(log10(max(max_lib_sizes))))\nsize_breaks \u0026lt;- lseq(10^floor(log10(min(min_lib_sizes))),10^ceiling(log10(max(max_lib_sizes))),8)\nsize.plot \u0026lt;- ggplot()+\n  geom_point(aes(x=seq(1,length(colSums(counts.list[[i]]))), y=colSums(counts.list[[i]]),size=colSums(counts.list[[i]])))+\n  scale_size_continuous(limits=size_limits,breaks=size_breaks ,trans = \u0026quot;log\u0026quot;)+\n  guides(size = guide_legend(title = \u0026quot;Reads Mapping to Wolbachia Genes\u0026quot;, title.position = \u0026quot;top\u0026quot;, nrow = 2))+\n  theme_bw()+\n  theme(legend.position=\u0026quot;top\u0026quot;,legend.title.align=0.5)\n  \nsize.legend \u0026lt;- g_legend(size.plot)\n\npdf(paste0(OUTPUT.DIR,\u0026quot;/sizekey.pdf\u0026quot;),\n    height=2,\n    width=10)\ngrid.arrange(size.legend)\ndev.off()\n\npng(paste0(OUTPUT.DIR,\u0026quot;/sizekey.png\u0026quot;),\n    height=2,\n    width=10,\n    units = \u0026quot;in\u0026quot;,res=300)\ngrid.arrange(size.legend)\ndev.off()\n\ngrid.arrange(size.legend)\"\u003e\u003cpre lang=\"{R,fig.height=2,fig.width=10}\" class=\"notranslate\"\u003e\u003ccode\u003emax_lib_sizes \u0026lt;- c()\nmin_lib_sizes \u0026lt;- c()\nfor(i in 1:length(counts.list)){\n  max_lib_sizes \u0026lt;- c(max_lib_sizes,max(colSums(counts.list[[i]])))\n  min_lib_sizes \u0026lt;- c(min_lib_sizes,min(colSums(counts.list[[i]])))\n}\nsize_limits \u0026lt;- c(1,10^ceiling(log10(max(max_lib_sizes))))\nsize_breaks \u0026lt;- lseq(10^floor(log10(min(min_lib_sizes))),10^ceiling(log10(max(max_lib_sizes))),8)\nsize.plot \u0026lt;- ggplot()+\n  geom_point(aes(x=seq(1,length(colSums(counts.list[[i]]))), y=colSums(counts.list[[i]]),size=colSums(counts.list[[i]])))+\n  scale_size_continuous(limits=size_limits,breaks=size_breaks ,trans = \"log\")+\n  guides(size = guide_legend(title = \"Reads Mapping to Wolbachia Genes\", title.position = \"top\", nrow = 2))+\n  theme_bw()+\n  theme(legend.position=\"top\",legend.title.align=0.5)\n  \nsize.legend \u0026lt;- g_legend(size.plot)\n\npdf(paste0(OUTPUT.DIR,\"/sizekey.pdf\"),\n    height=2,\n    width=10)\ngrid.arrange(size.legend)\ndev.off()\n\npng(paste0(OUTPUT.DIR,\"/sizekey.png\"),\n    height=2,\n    width=10,\n    units = \"in\",res=300)\ngrid.arrange(size.legend)\ndev.off()\n\ngrid.arrange(size.legend)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/sizekey.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/sizekey.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHeatmap log2TPM key \u003ca name=\"user-content-ranalysis_de_keys_hm1\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-heatmap-log2tpm-key-\" class=\"anchor\" aria-label=\"Permalink: Heatmap log2TPM key \" href=\"#heatmap-log2tpm-key-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"hmcol1 \u0026lt;- colorRampPalette(c(\u0026quot;#FFFFFF\u0026quot;,\u0026quot;#FDE0D2\u0026quot;,\u0026quot;#FABAA1\u0026quot;,\u0026quot;#F69274\u0026quot;, \u0026quot;#F16B4E\u0026quot;,\u0026quot;#EF3D2D\u0026quot;,\u0026quot;#590A16\u0026quot;,\u0026quot;#A51E23\u0026quot;,\u0026quot;#650C16\u0026quot;))(20)\nhmcol2 \u0026lt;- colorRampPalette(c(\u0026quot;navyblue\u0026quot;,\u0026quot;white\u0026quot;,\u0026quot;firebrick3\u0026quot;))(12)\n\nhmcolor1.plot \u0026lt;- ggplot() + \n  geom_raster(aes(x=seq(0,10,0.5), y=seq(0,10,0.5), fill = seq(0,10,0.5)))+\n  scale_fill_gradientn(name = \u0026quot;log2TPM\u0026quot;,\n                       colours=hmcol1,\n                       breaks=c(0,5,10))+\n  theme(legend.position=\u0026quot;bottom\u0026quot;)+\n  guides(fill = guide_colorbar(title.position = \u0026quot;top\u0026quot;))\n\nsample.legend \u0026lt;- g_legend(hmcolor1.plot)\n\npdf(paste0(OUTPUT.DIR,\u0026quot;/hmcolor1key.pdf\u0026quot;),\n    height=2,\n    width=7)\ngrid.arrange(sample.legend)\ndev.off()\n\npng(paste0(OUTPUT.DIR,\u0026quot;/hmcolor1key.png\u0026quot;),\n    height=2,\n    width=7,\n    units = \u0026quot;in\u0026quot;,res=300)\ngrid.arrange(sample.legend)\ndev.off()\n\ngrid.arrange(sample.legend)\"\u003e\u003cpre lang=\"{R,fig.height=2,fig.width=7}\" class=\"notranslate\"\u003e\u003ccode\u003ehmcol1 \u0026lt;- colorRampPalette(c(\"#FFFFFF\",\"#FDE0D2\",\"#FABAA1\",\"#F69274\", \"#F16B4E\",\"#EF3D2D\",\"#590A16\",\"#A51E23\",\"#650C16\"))(20)\nhmcol2 \u0026lt;- colorRampPalette(c(\"navyblue\",\"white\",\"firebrick3\"))(12)\n\nhmcolor1.plot \u0026lt;- ggplot() + \n  geom_raster(aes(x=seq(0,10,0.5), y=seq(0,10,0.5), fill = seq(0,10,0.5)))+\n  scale_fill_gradientn(name = \"log2TPM\",\n                       colours=hmcol1,\n                       breaks=c(0,5,10))+\n  theme(legend.position=\"bottom\")+\n  guides(fill = guide_colorbar(title.position = \"top\"))\n\nsample.legend \u0026lt;- g_legend(hmcolor1.plot)\n\npdf(paste0(OUTPUT.DIR,\"/hmcolor1key.pdf\"),\n    height=2,\n    width=7)\ngrid.arrange(sample.legend)\ndev.off()\n\npng(paste0(OUTPUT.DIR,\"/hmcolor1key.png\"),\n    height=2,\n    width=7,\n    units = \"in\",res=300)\ngrid.arrange(sample.legend)\ndev.off()\n\ngrid.arrange(sample.legend)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/hmcolor1key.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/hmcolor1key.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eHeatmap z-score log2TPM key \u003ca name=\"user-content-ranalysis_de_keys_hm2\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-heatmap-z-score-log2tpm-key-\" class=\"anchor\" aria-label=\"Permalink: Heatmap z-score log2TPM key \" href=\"#heatmap-z-score-log2tpm-key-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"hmcolor2.plot \u0026lt;- ggplot() + \n  geom_raster(aes(x=seq(-3,3,0.5), y=seq(-3,3,0.5), fill = seq(-3,3,0.5)))+\n  scale_fill_gradientn(name = \u0026quot;z-score of log2TPM\u0026quot;,\n                       colours=hmcol2,\n                       breaks=c(-3,0,3))+\n  theme(legend.position=\u0026quot;bottom\u0026quot;)+\n  guides(fill = guide_colorbar(title.position = \u0026quot;top\u0026quot;))\n\nheat.legend \u0026lt;- g_legend(hmcolor2.plot)\n\npdf(paste0(OUTPUT.DIR,\u0026quot;/hmcolor2key.pdf\u0026quot;),\n    height=2,\n    width=7)\ngrid.arrange(heat.legend)\ndev.off()\n\npng(paste0(OUTPUT.DIR,\u0026quot;/hmcolor2key.png\u0026quot;),\n    height=2,\n    width=7,\n    units = \u0026quot;in\u0026quot;,res=300)\ngrid.arrange(heat.legend)\ndev.off()\n\ngrid.arrange(heat.legend)\"\u003e\u003cpre lang=\"{R,fig.height=2,fig.width=7}\" class=\"notranslate\"\u003e\u003ccode\u003ehmcolor2.plot \u0026lt;- ggplot() + \n  geom_raster(aes(x=seq(-3,3,0.5), y=seq(-3,3,0.5), fill = seq(-3,3,0.5)))+\n  scale_fill_gradientn(name = \"z-score of log2TPM\",\n                       colours=hmcol2,\n                       breaks=c(-3,0,3))+\n  theme(legend.position=\"bottom\")+\n  guides(fill = guide_colorbar(title.position = \"top\"))\n\nheat.legend \u0026lt;- g_legend(hmcolor2.plot)\n\npdf(paste0(OUTPUT.DIR,\"/hmcolor2key.pdf\"),\n    height=2,\n    width=7)\ngrid.arrange(heat.legend)\ndev.off()\n\npng(paste0(OUTPUT.DIR,\"/hmcolor2key.png\"),\n    height=2,\n    width=7,\n    units = \"in\",res=300)\ngrid.arrange(heat.legend)\ndev.off()\n\ngrid.arrange(heat.legend)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/hmcolor2key.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/hmcolor2key.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAssess if samples have been sequenced to saturation using saturation curves \u003ca name=\"user-content-ranalysis_de_keys_saturation\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-assess-if-samples-have-been-sequenced-to-saturation-using-saturation-curves-\" class=\"anchor\" aria-label=\"Permalink: Assess if samples have been sequenced to saturation using saturation curves \" href=\"#assess-if-samples-have-been-sequenced-to-saturation-using-saturation-curves-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(tpm.list)){\n  rarefy.counts \u0026lt;- ceiling(counts.list[[i]])\n  rarefy.counts \u0026lt;- rarefy.counts[,colSums(rarefy.counts) != 0,]\n  \n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(counts.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]\n  \n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n\n  raremax \u0026lt;- round(min(rowSums(t(rarefy.counts))[rowSums(t(rarefy.counts)) != 0]),0)\n  srare \u0026lt;- rarefy(t(rarefy.counts),raremax)\n  raremax \u0026lt;- ifelse(raremax \u0026lt; 10, 10, raremax)\n  \n  rarefy.raw.df \u0026lt;- rarecurve(t(rarefy.counts), step = round(raremax/10,0), sample = raremax)\n  \n  rarefy.df \u0026lt;- as.data.frame(matrix(nrow = 0,\n                                    ncol = 5))\n  rarefy.points.df \u0026lt;- rarefy.df\n  for(j in 1:length(rarefy.raw.df)){\n    steps \u0026lt;- as.numeric(gsub(\u0026quot;N\u0026quot;,\u0026quot;\u0026quot;,names(rarefy.raw.df[[j]])))\n    detected_genes \u0026lt;- as.numeric(rarefy.raw.df[[j]])\n    rarefy.df \u0026lt;- as.data.frame(rbind(rarefy.df,\n                                     cbind(as.numeric(steps),\n                                           as.numeric(detected_genes),\n                                           as.character(groups[j,1]),\n                                           as.character(groups[j,2]),\n                                           groups[j,3])))\n    rarefy.points.df \u0026lt;- as.data.frame(rbind(rarefy.points.df,\n                                     cbind(as.numeric(max(steps)),\n                                           as.numeric(max(detected_genes)),\n                                           as.character(groups[j,1]),\n                                           as.character(groups[j,2]),\n                                           groups[j,3])))\n    \n  }\n  rarefy.plot \u0026lt;- ggplot()+\n    geom_line(mapping=aes(x=as.numeric(as.character(rarefy.df[,1])), y=as.numeric(as.character(rarefy.df[,2])),group=rarefy.df[,3],color=rarefy.df[,4]))+\n    #geom_point(mapping=aes(x=as.numeric(as.character(rarefy.df[,1])), y=as.numeric(as.character(rarefy.df[,2])),group=rarefy.df[,3],color=rarefy.df[,4]))+\n    geom_point(mapping=aes(x=as.numeric(as.character(rarefy.points.df[,1])), y=as.numeric(as.character(rarefy.points.df[,2])),group=rarefy.points.df[,3],color=rarefy.points.df[,4]),size = 3)+\n    guides(colour = F,shape = F)+\n    scale_color_manual(values = levels(groups[,3]))+\n    labs(x=\u0026quot;reads mapping to protein-coding genes\u0026quot;, y=\u0026quot;genes detected\u0026quot;, color = \u0026quot;Sample\u0026quot;)+\n    #coord_cartesian(xlim=c(0,100000))+\n    theme_bw()\n  \n  pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.list)[i],\u0026quot;_rarefaction.pdf\u0026quot;),\n      height=5,\n      width=5)\n  print(rarefy.plot)\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.list)[i],\u0026quot;_rarefaction.png\u0026quot;),\n      height=5,\n      width=5,\n      units = \u0026quot;in\u0026quot;,res=300)\n  print(rarefy.plot)\n  dev.off()\n  \n  print(rarefy.plot)\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(tpm.list)){\n  rarefy.counts \u0026lt;- ceiling(counts.list[[i]])\n  rarefy.counts \u0026lt;- rarefy.counts[,colSums(rarefy.counts) != 0,]\n  \n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(counts.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]\n  \n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n\n  raremax \u0026lt;- round(min(rowSums(t(rarefy.counts))[rowSums(t(rarefy.counts)) != 0]),0)\n  srare \u0026lt;- rarefy(t(rarefy.counts),raremax)\n  raremax \u0026lt;- ifelse(raremax \u0026lt; 10, 10, raremax)\n  \n  rarefy.raw.df \u0026lt;- rarecurve(t(rarefy.counts), step = round(raremax/10,0), sample = raremax)\n  \n  rarefy.df \u0026lt;- as.data.frame(matrix(nrow = 0,\n                                    ncol = 5))\n  rarefy.points.df \u0026lt;- rarefy.df\n  for(j in 1:length(rarefy.raw.df)){\n    steps \u0026lt;- as.numeric(gsub(\"N\",\"\",names(rarefy.raw.df[[j]])))\n    detected_genes \u0026lt;- as.numeric(rarefy.raw.df[[j]])\n    rarefy.df \u0026lt;- as.data.frame(rbind(rarefy.df,\n                                     cbind(as.numeric(steps),\n                                           as.numeric(detected_genes),\n                                           as.character(groups[j,1]),\n                                           as.character(groups[j,2]),\n                                           groups[j,3])))\n    rarefy.points.df \u0026lt;- as.data.frame(rbind(rarefy.points.df,\n                                     cbind(as.numeric(max(steps)),\n                                           as.numeric(max(detected_genes)),\n                                           as.character(groups[j,1]),\n                                           as.character(groups[j,2]),\n                                           groups[j,3])))\n    \n  }\n  rarefy.plot \u0026lt;- ggplot()+\n    geom_line(mapping=aes(x=as.numeric(as.character(rarefy.df[,1])), y=as.numeric(as.character(rarefy.df[,2])),group=rarefy.df[,3],color=rarefy.df[,4]))+\n    #geom_point(mapping=aes(x=as.numeric(as.character(rarefy.df[,1])), y=as.numeric(as.character(rarefy.df[,2])),group=rarefy.df[,3],color=rarefy.df[,4]))+\n    geom_point(mapping=aes(x=as.numeric(as.character(rarefy.points.df[,1])), y=as.numeric(as.character(rarefy.points.df[,2])),group=rarefy.points.df[,3],color=rarefy.points.df[,4]),size = 3)+\n    guides(colour = F,shape = F)+\n    scale_color_manual(values = levels(groups[,3]))+\n    labs(x=\"reads mapping to protein-coding genes\", y=\"genes detected\", color = \"Sample\")+\n    #coord_cartesian(xlim=c(0,100000))+\n    theme_bw()\n  \n  pdf(paste0(OUTPUT.DIR,\"/\",names(counts.list)[i],\"_rarefaction.pdf\"),\n      height=5,\n      width=5)\n  print(rarefy.plot)\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\"/\",names(counts.list)[i],\"_rarefaction.png\"),\n      height=5,\n      width=5,\n      units = \"in\",res=300)\n  print(rarefy.plot)\n  dev.off()\n  \n  print(rarefy.plot)\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewOo, Darby et al 2012\u003c/h5\u003e\u003ca id=\"user-content-woo-darby-et-al-2012-1\" class=\"anchor\" aria-label=\"Permalink: wOo, Darby et al 2012\" href=\"#woo-darby-et-al-2012-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_rarefaction.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_rarefaction.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Darby et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wmel-darby-et-al-2014-1\" class=\"anchor\" aria-label=\"Permalink: wMel, Darby et al 2014\" href=\"#wmel-darby-et-al-2014-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_rarefaction.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_rarefaction.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2014-1\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2014\" href=\"#wdi-luck-et-al-2014-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_rarefaction.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_rarefaction.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2015-1\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2015\" href=\"#wdi-luck-et-al-2015-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_rarefaction.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_rarefaction.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Gutzwiller et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wmel-gutzwiller-et-al-2015-1\" class=\"anchor\" aria-label=\"Permalink: wMel, Gutzwiller et al 2015\" href=\"#wmel-gutzwiller-et-al-2015-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_rarefaction.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_rarefaction.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Grote et al 2017\u003c/h5\u003e\u003ca id=\"user-content-wbm-grote-et-al-2017-1\" class=\"anchor\" aria-label=\"Permalink: wBm, Grote et al 2017\" href=\"#wbm-grote-et-al-2017-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_rarefaction.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_rarefaction.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Chung et al 2019\u003c/h5\u003e\u003ca id=\"user-content-wbm-chung-et-al-2019-1\" class=\"anchor\" aria-label=\"Permalink: wBm, Chung et al 2019\" href=\"#wbm-chung-et-al-2019-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_rarefaction.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_rarefaction.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eExclude samples and/or studies from differential expression analyses due to inadequate sequencing depth \u003ca name=\"user-content-ranalysis_de_exclude\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-exclude-samples-andor-studies-from-differential-expression-analyses-due-to-inadequate-sequencing-depth-\" class=\"anchor\" aria-label=\"Permalink: Exclude samples and/or studies from differential expression analyses due to inadequate sequencing depth \" href=\"#exclude-samples-andor-studies-from-differential-expression-analyses-due-to-inadequate-sequencing-depth-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAll samples from Luck et al 2014 were excluded due to having inadequate replicates and sequencing depth for differential expression analyses. All samples with \u0026lt;2000 gene counts from Luck et al 2015 are excluded due to have inadequate sequencing depth (refer to saturation curves).\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"counts.kept.list \u0026lt;- counts.list\ntpm.kept.list \u0026lt;- tpm.list\n\ncounts.edgeR.list \u0026lt;- counts.list\ntpm.edgeR.list \u0026lt;- tpm.list\n\ncounts.edgeR.list[[which(names(counts.edgeR.list) == \u0026quot;wDi_luck_2014\u0026quot;)]] \u0026lt;- NULL\ntpm.edgeR.list[[which(names(tpm.edgeR.list) == \u0026quot;wDi_luck_2014\u0026quot;)]] \u0026lt;- NULL\n\ncounts.kept.list[[which(names(counts.kept.list) == \u0026quot;wDi_luck_2015\u0026quot;)]] \u0026lt;- counts.kept.list[[which(names(counts.kept.list) == \u0026quot;wDi_luck_2015\u0026quot;)]][colSums(counts.kept.list[[which(names(counts.kept.list) == \u0026quot;wDi_luck_2015\u0026quot;)]]) \u0026gt; 2000]\ntpm.kept.list[[which(names(tpm.kept.list) == \u0026quot;wDi_luck_2015\u0026quot;)]] \u0026lt;- tpm.kept.list[[which(names(tpm.kept.list) == \u0026quot;wDi_luck_2015\u0026quot;)]][,colnames(tpm.kept.list[[which(names(tpm.kept.list) == \u0026quot;wDi_luck_2015\u0026quot;)]]) %in% colnames(counts.kept.list[[which(names(counts.kept.list) == \u0026quot;wDi_luck_2015\u0026quot;)]])]\n\ncounts.edgeR.list[[which(names(counts.edgeR.list) == \u0026quot;wDi_luck_2015\u0026quot;)]] \u0026lt;- counts.edgeR.list[[which(names(counts.edgeR.list) == \u0026quot;wDi_luck_2015\u0026quot;)]][colSums(counts.edgeR.list[[which(names(counts.edgeR.list) == \u0026quot;wDi_luck_2015\u0026quot;)]]) \u0026gt; 2000]\ntpm.edgeR.list[[which(names(tpm.edgeR.list) == \u0026quot;wDi_luck_2015\u0026quot;)]] \u0026lt;- tpm.edgeR.list[[which(names(tpm.edgeR.list) == \u0026quot;wDi_luck_2015\u0026quot;)]][,colnames(tpm.edgeR.list[[which(names(tpm.edgeR.list) == \u0026quot;wDi_luck_2015\u0026quot;)]]) %in% colnames(counts.edgeR.list[[which(names(counts.edgeR.list) == \u0026quot;wDi_luck_2015\u0026quot;)]])]\n\nqlf.list \u0026lt;- counts.edgeR.list\"\u003e\u003cpre lang=\"{R}\" class=\"notranslate\"\u003e\u003ccode\u003ecounts.kept.list \u0026lt;- counts.list\ntpm.kept.list \u0026lt;- tpm.list\n\ncounts.edgeR.list \u0026lt;- counts.list\ntpm.edgeR.list \u0026lt;- tpm.list\n\ncounts.edgeR.list[[which(names(counts.edgeR.list) == \"wDi_luck_2014\")]] \u0026lt;- NULL\ntpm.edgeR.list[[which(names(tpm.edgeR.list) == \"wDi_luck_2014\")]] \u0026lt;- NULL\n\ncounts.kept.list[[which(names(counts.kept.list) == \"wDi_luck_2015\")]] \u0026lt;- counts.kept.list[[which(names(counts.kept.list) == \"wDi_luck_2015\")]][colSums(counts.kept.list[[which(names(counts.kept.list) == \"wDi_luck_2015\")]]) \u0026gt; 2000]\ntpm.kept.list[[which(names(tpm.kept.list) == \"wDi_luck_2015\")]] \u0026lt;- tpm.kept.list[[which(names(tpm.kept.list) == \"wDi_luck_2015\")]][,colnames(tpm.kept.list[[which(names(tpm.kept.list) == \"wDi_luck_2015\")]]) %in% colnames(counts.kept.list[[which(names(counts.kept.list) == \"wDi_luck_2015\")]])]\n\ncounts.edgeR.list[[which(names(counts.edgeR.list) == \"wDi_luck_2015\")]] \u0026lt;- counts.edgeR.list[[which(names(counts.edgeR.list) == \"wDi_luck_2015\")]][colSums(counts.edgeR.list[[which(names(counts.edgeR.list) == \"wDi_luck_2015\")]]) \u0026gt; 2000]\ntpm.edgeR.list[[which(names(tpm.edgeR.list) == \"wDi_luck_2015\")]] \u0026lt;- tpm.edgeR.list[[which(names(tpm.edgeR.list) == \"wDi_luck_2015\")]][,colnames(tpm.edgeR.list[[which(names(tpm.edgeR.list) == \"wDi_luck_2015\")]]) %in% colnames(counts.edgeR.list[[which(names(counts.edgeR.list) == \"wDi_luck_2015\")]])]\n\nqlf.list \u0026lt;- counts.edgeR.list\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIdentify the number of genes in each study that meets the minimum CPM threshold \u003ca name=\"user-content-ranalysis_de_exclude\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-identify-the-number-of-genes-in-each-study-that-meets-the-minimum-cpm-threshold-\" class=\"anchor\" aria-label=\"Permalink: Identify the number of genes in each study that meets the minimum CPM threshold \" href=\"#identify-the-number-of-genes-in-each-study-that-meets-the-minimum-cpm-threshold-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(counts.kept.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(counts.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- groups[groups[,1] %in% colnames(counts.kept.list[[i]]),]\n  groups \u0026lt;- unique(groups)\n\n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n\n  cpm.cutoff \u0026lt;- 5/min(colSums(counts.kept.list[[i]])) * 1000000\n\n  y \u0026lt;- DGEList(counts = counts.kept.list[[i]], group = groups[,2])\n  y \u0026lt;- calcNormFactors(y)\n  keep \u0026lt;- rowSums(cpm(y) \u0026gt;= cpm.cutoff) \u0026gt;= min(table(groups[,2]))\n\n  keep.df \u0026lt;- as.data.frame(table(keep))\n  print(paste0(names(counts.kept.list)[i],\u0026quot;: \u0026quot;,keep.df[keep.df[,1] == F,2],\u0026quot; genes excluded using edgeR filter, \u0026quot;,keep.df[keep.df[,1] == T,2], \u0026quot; genes kept\u0026quot; ))\n\n  counts.kept.list[[which(names(counts.kept.list) == names(counts.list)[i])]] \u0026lt;- counts.kept.list[[which(names(counts.kept.list) == names(counts.list)[i])]][keep,]\n  tpm.kept.list[[which(names(tpm.kept.list) == names(tpm.list)[i])]] \u0026lt;- tpm.kept.list[[which(names(tpm.kept.list) == names(tpm.list)[i])]][keep,]\n}\"\u003e\u003cpre lang=\"{R}\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(counts.kept.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(counts.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- groups[groups[,1] %in% colnames(counts.kept.list[[i]]),]\n  groups \u0026lt;- unique(groups)\n\n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n\n  cpm.cutoff \u0026lt;- 5/min(colSums(counts.kept.list[[i]])) * 1000000\n\n  y \u0026lt;- DGEList(counts = counts.kept.list[[i]], group = groups[,2])\n  y \u0026lt;- calcNormFactors(y)\n  keep \u0026lt;- rowSums(cpm(y) \u0026gt;= cpm.cutoff) \u0026gt;= min(table(groups[,2]))\n\n  keep.df \u0026lt;- as.data.frame(table(keep))\n  print(paste0(names(counts.kept.list)[i],\": \",keep.df[keep.df[,1] == F,2],\" genes excluded using edgeR filter, \",keep.df[keep.df[,1] == T,2], \" genes kept\" ))\n\n  counts.kept.list[[which(names(counts.kept.list) == names(counts.list)[i])]] \u0026lt;- counts.kept.list[[which(names(counts.kept.list) == names(counts.list)[i])]][keep,]\n  tpm.kept.list[[which(names(tpm.kept.list) == names(tpm.list)[i])]] \u0026lt;- tpm.kept.list[[which(names(tpm.kept.list) == names(tpm.list)[i])]][keep,]\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[1] \u0026quot;wOo_darby_2012: 91 genes excluded using edgeR filter, 560 genes kept\u0026quot;\n[1] \u0026quot;wMel_darby_2014: 219 genes excluded using edgeR filter, 867 genes kept\u0026quot;\n[1] \u0026quot;wDi_luck_2014: 853 genes excluded using edgeR filter, 18 genes kept\u0026quot;\n[1] \u0026quot;wDi_luck_2015: 546 genes excluded using edgeR filter, 325 genes kept\u0026quot;\n[1] \u0026quot;wMel_gutzwiller_2015: 305 genes excluded using edgeR filter, 781 genes kept\u0026quot;\n[1] \u0026quot;wBm_grote_2017: 414 genes excluded using edgeR filter, 425 genes kept\u0026quot;\n[1] \u0026quot;wBm_chung_2019: 206 genes excluded using edgeR filter, 548 genes kept\u0026quot;\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003e[1] \"wOo_darby_2012: 91 genes excluded using edgeR filter, 560 genes kept\"\n[1] \"wMel_darby_2014: 219 genes excluded using edgeR filter, 867 genes kept\"\n[1] \"wDi_luck_2014: 853 genes excluded using edgeR filter, 18 genes kept\"\n[1] \"wDi_luck_2015: 546 genes excluded using edgeR filter, 325 genes kept\"\n[1] \"wMel_gutzwiller_2015: 305 genes excluded using edgeR filter, 781 genes kept\"\n[1] \"wBm_grote_2017: 414 genes excluded using edgeR filter, 425 genes kept\"\n[1] \"wBm_chung_2019: 206 genes excluded using edgeR filter, 548 genes kept\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIdentify differentially expressed genes (FDR \u0026lt; 0.05) in each study \u003ca name=\"user-content-ranalysis_de_de\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-identify-differentially-expressed-genes-fdr--005-in-each-study-\" class=\"anchor\" aria-label=\"Permalink: Identify differentially expressed genes (FDR \u0026lt; 0.05) in each study \" href=\"#identify-differentially-expressed-genes-fdr--005-in-each-study-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"FDRcutoff \u0026lt;- 0.05\n\nfor(i in 1:length(counts.edgeR.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(counts.edgeR.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(counts.edgeR.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n\n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n\n  cpm.cutoff \u0026lt;- 5/min(colSums(counts.edgeR.list[[i]])) * 1000000\n\n  y \u0026lt;- DGEList(counts = counts.edgeR.list[[i]], group = groups[,2])\n  y \u0026lt;- calcNormFactors(y)\n  keep \u0026lt;- rowSums(cpm(y) \u0026gt;= cpm.cutoff) \u0026gt;= min(table(groups[,2]))\n\n  y \u0026lt;- y[keep, , keep.lib.sizes = F]\n  design \u0026lt;- model.matrix(~groups[,2])\n  y \u0026lt;- estimateDisp(y , design)\n  fit \u0026lt;- glmQLFit(y, design)\n  qlf \u0026lt;- glmQLFTest(fit, coef = 2:ncol(fit))\n\n  FDR \u0026lt;- as.data.frame(p.adjust(qlf$table$PValue, method=\u0026quot;BH\u0026quot;))\n  rownames(FDR) \u0026lt;- rownames(qlf$table)\n  FDR.genes \u0026lt;- rownames(FDR[FDR[,1] \u0026lt; FDRcutoff, , drop = F])\n  print(paste0(names(counts.edgeR.list)[i],\u0026quot;: \u0026quot;,length(FDR.genes),\u0026quot; DE genes\u0026quot; ))\n  \n  qlf.list[[i]] \u0026lt;- as.data.frame(qlf$table)\n  counts.edgeR.list[[i]] \u0026lt;- counts.edgeR.list[[i]][rownames(counts.edgeR.list[[i]]) %in% FDR.genes,]\n  tpm.edgeR.list[[i]] \u0026lt;- tpm.edgeR.list[[i]][rownames(tpm.edgeR.list[[i]]) %in% FDR.genes,]\n  \n  write.table(tpm.edgeR.list[[i]],\n            paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.edgeR.list)[i],\u0026quot;_de_tpm.tsv\u0026quot;),\n            row.names = T,\n            col.names = T,\n            quote = F,\n            sep = \u0026quot;\\t\u0026quot;)\n}\"\u003e\u003cpre lang=\"{R}\" class=\"notranslate\"\u003e\u003ccode\u003eFDRcutoff \u0026lt;- 0.05\n\nfor(i in 1:length(counts.edgeR.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(counts.edgeR.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(counts.edgeR.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n\n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n\n  cpm.cutoff \u0026lt;- 5/min(colSums(counts.edgeR.list[[i]])) * 1000000\n\n  y \u0026lt;- DGEList(counts = counts.edgeR.list[[i]], group = groups[,2])\n  y \u0026lt;- calcNormFactors(y)\n  keep \u0026lt;- rowSums(cpm(y) \u0026gt;= cpm.cutoff) \u0026gt;= min(table(groups[,2]))\n\n  y \u0026lt;- y[keep, , keep.lib.sizes = F]\n  design \u0026lt;- model.matrix(~groups[,2])\n  y \u0026lt;- estimateDisp(y , design)\n  fit \u0026lt;- glmQLFit(y, design)\n  qlf \u0026lt;- glmQLFTest(fit, coef = 2:ncol(fit))\n\n  FDR \u0026lt;- as.data.frame(p.adjust(qlf$table$PValue, method=\"BH\"))\n  rownames(FDR) \u0026lt;- rownames(qlf$table)\n  FDR.genes \u0026lt;- rownames(FDR[FDR[,1] \u0026lt; FDRcutoff, , drop = F])\n  print(paste0(names(counts.edgeR.list)[i],\": \",length(FDR.genes),\" DE genes\" ))\n  \n  qlf.list[[i]] \u0026lt;- as.data.frame(qlf$table)\n  counts.edgeR.list[[i]] \u0026lt;- counts.edgeR.list[[i]][rownames(counts.edgeR.list[[i]]) %in% FDR.genes,]\n  tpm.edgeR.list[[i]] \u0026lt;- tpm.edgeR.list[[i]][rownames(tpm.edgeR.list[[i]]) %in% FDR.genes,]\n  \n  write.table(tpm.edgeR.list[[i]],\n            paste0(OUTPUT.DIR,\"/\",names(tpm.edgeR.list)[i],\"_de_tpm.tsv\"),\n            row.names = T,\n            col.names = T,\n            quote = F,\n            sep = \"\\t\")\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"[1] \u0026quot;wOo_darby_2012: 0 DE genes\u0026quot;\n[1] \u0026quot;wMel_darby_2014: 120 DE genes\u0026quot;\n[1] \u0026quot;wDi_luck_2015: 36 DE genes\u0026quot;\n[1] \u0026quot;wMel_gutzwiller_2015: 473 DE genes\u0026quot;\n[1] \u0026quot;wBm_grote_2017: 94 DE genes\u0026quot;\n[1] \u0026quot;wBm_chung_2019: 373 DE genes\u0026quot;\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003e[1] \"wOo_darby_2012: 0 DE genes\"\n[1] \"wMel_darby_2014: 120 DE genes\"\n[1] \"wDi_luck_2015: 36 DE genes\"\n[1] \"wMel_gutzwiller_2015: 473 DE genes\"\n[1] \"wBm_grote_2017: 94 DE genes\"\n[1] \"wBm_chung_2019: 373 DE genes\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConduct principal component analyses \u003ca name=\"user-content-ranalysis_de_pca\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-conduct-principal-component-analyses-\" class=\"anchor\" aria-label=\"Permalink: Conduct principal component analyses \" href=\"#conduct-principal-component-analyses-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAll genes \u003ca name=\"user-content-ranalysis_de_pca_all\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-all-genes-\" class=\"anchor\" aria-label=\"Permalink: All genes \" href=\"#all-genes-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(tpm.list)){ \n\tsample_map.subset \u0026lt;- sample_map[sample_map$study == names(counts.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]\n\n\tgroups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n\t                            as.character(sample_map.subset$sample_group),\n\t                            as.character(sample_map.subset$color)))\n\tgroups \u0026lt;- unique(groups)\n\n\tgroups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n\tgroups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n\tgroups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n\n\tpca.df \u0026lt;- t(scale(t(log2(tpm.list[[i]] + 1))))\n\tpca.df[is.na(pca.df)] \u0026lt;- 0\n\tpca.df \u0026lt;- pca.df[rowSums(pca.df == 0) != ncol(pca.df),]\n\tpca \u0026lt;- PCA(as.data.frame(scale(t(pca.df))), graph = FALSE, ncp = ncol(counts.list[[i]]) - 1)\n\n\tpca.plot \u0026lt;- ggplot()+\n\t  geom_point(aes(x=pca$ind$coord[,1], y=pca$ind$coord[,2], color = groups[,2],size = colSums(counts.list[[i]])))+\n\t  labs(col = \u0026quot;Samples\u0026quot;, size = \u0026quot;Reads Mapped\\nto Features\u0026quot;, \n\t       x = paste0(\u0026quot;PC1 (\u0026quot;, round(pca$eig[1,2],1), \u0026quot;%)\u0026quot;), \n\t       y = paste0(\u0026quot;PC2 (\u0026quot;, round(pca$eig[2,2],1), \u0026quot;%)\u0026quot;))+\n\t  guides(color = F,size = F)+\n\t  # guides(colour = guide_legend(ncol = 2))+\n\t  scale_color_manual(values = levels(groups[,3]))+\n\t  scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = \u0026quot;log\u0026quot;)+\n\t  theme_bw()\n\n\tpdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.list)[i],\u0026quot;_pca_allgenes.pdf\u0026quot;),\n\t  height=5,\n\t  width=5)\n\tprint(pca.plot)\n\tdev.off()\n\t\n\tpng(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.list)[i],\u0026quot;_pca_allgenes.png\u0026quot;),\n\t  height=5,\n\t  width=5,\n\t  units = \u0026quot;in\u0026quot;,res=300)\n\tprint(pca.plot)\n\tdev.off()\n\n\tprint(pca.plot)\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(tpm.list)){ \n\tsample_map.subset \u0026lt;- sample_map[sample_map$study == names(counts.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]\n\n\tgroups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n\t                            as.character(sample_map.subset$sample_group),\n\t                            as.character(sample_map.subset$color)))\n\tgroups \u0026lt;- unique(groups)\n\n\tgroups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n\tgroups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n\tgroups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n\n\tpca.df \u0026lt;- t(scale(t(log2(tpm.list[[i]] + 1))))\n\tpca.df[is.na(pca.df)] \u0026lt;- 0\n\tpca.df \u0026lt;- pca.df[rowSums(pca.df == 0) != ncol(pca.df),]\n\tpca \u0026lt;- PCA(as.data.frame(scale(t(pca.df))), graph = FALSE, ncp = ncol(counts.list[[i]]) - 1)\n\n\tpca.plot \u0026lt;- ggplot()+\n\t  geom_point(aes(x=pca$ind$coord[,1], y=pca$ind$coord[,2], color = groups[,2],size = colSums(counts.list[[i]])))+\n\t  labs(col = \"Samples\", size = \"Reads Mapped\\nto Features\", \n\t       x = paste0(\"PC1 (\", round(pca$eig[1,2],1), \"%)\"), \n\t       y = paste0(\"PC2 (\", round(pca$eig[2,2],1), \"%)\"))+\n\t  guides(color = F,size = F)+\n\t  # guides(colour = guide_legend(ncol = 2))+\n\t  scale_color_manual(values = levels(groups[,3]))+\n\t  scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = \"log\")+\n\t  theme_bw()\n\n\tpdf(paste0(OUTPUT.DIR,\"/\",names(counts.list)[i],\"_pca_allgenes.pdf\"),\n\t  height=5,\n\t  width=5)\n\tprint(pca.plot)\n\tdev.off()\n\t\n\tpng(paste0(OUTPUT.DIR,\"/\",names(counts.list)[i],\"_pca_allgenes.png\"),\n\t  height=5,\n\t  width=5,\n\t  units = \"in\",res=300)\n\tprint(pca.plot)\n\tdev.off()\n\n\tprint(pca.plot)\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewOo, Darby et al 2012\u003c/h5\u003e\u003ca id=\"user-content-woo-darby-et-al-2012-2\" class=\"anchor\" aria-label=\"Permalink: wOo, Darby et al 2012\" href=\"#woo-darby-et-al-2012-2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_pca_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_pca_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Darby et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wmel-darby-et-al-2014-2\" class=\"anchor\" aria-label=\"Permalink: wMel, Darby et al 2014\" href=\"#wmel-darby-et-al-2014-2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_pca_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_pca_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2014-2\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2014\" href=\"#wdi-luck-et-al-2014-2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_pca_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_pca_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2015-2\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2015\" href=\"#wdi-luck-et-al-2015-2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_pca_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_pca_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Gutzwiller et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wmel-gutzwiller-et-al-2015-2\" class=\"anchor\" aria-label=\"Permalink: wMel, Gutzwiller et al 2015\" href=\"#wmel-gutzwiller-et-al-2015-2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_pca_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_pca_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Grote et al 2017\u003c/h5\u003e\u003ca id=\"user-content-wbm-grote-et-al-2017-2\" class=\"anchor\" aria-label=\"Permalink: wBm, Grote et al 2017\" href=\"#wbm-grote-et-al-2017-2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_pca_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_pca_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Chung et al 2019\u003c/h5\u003e\u003ca id=\"user-content-wbm-chung-et-al-2019-2\" class=\"anchor\" aria-label=\"Permalink: wBm, Chung et al 2019\" href=\"#wbm-chung-et-al-2019-2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_pca_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_pca_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eKept genes \u003ca name=\"user-content-ranalysis_de_pca_kept\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-kept-genes-\" class=\"anchor\" aria-label=\"Permalink: Kept genes \" href=\"#kept-genes-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(tpm.kept.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.kept.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(tpm.kept.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  if(nrow(tpm.kept.list[[i]]) \u0026gt; 0){\n    pca.df \u0026lt;- t(scale(t(log2(tpm.kept.list[[i]] + 1))))\n    pca.df \u0026lt;- pca.df[rowSums(pca.df == 0) != ncol(pca.df),]\n    pca \u0026lt;- PCA(as.data.frame(scale(t(pca.df))), graph = FALSE, ncp = ncol(tpm.kept.list[[i]]) - 1)\n    \n    pca.plot \u0026lt;- ggplot()+\n      geom_point(aes(x=pca$ind$coord[,1], y=pca$ind$coord[,2], color = groups[,2],size = colSums(counts.kept.list[[i]])))+\n      labs(col = \u0026quot;Samples\u0026quot;, size = \u0026quot;Reads Mapped\\nto Features\u0026quot;, \n           x = paste0(\u0026quot;PC1 (\u0026quot;, round(pca$eig[1,2],1), \u0026quot;%)\u0026quot;), \n           y = paste0(\u0026quot;PC2 (\u0026quot;, round(pca$eig[2,2],1), \u0026quot;%)\u0026quot;))+\n      guides(color = F,size = F)+\n      # guides(colour = guide_legend(ncol = 2))+\n      scale_color_manual(values = levels(groups[,3]))+\n      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = \u0026quot;log\u0026quot;)+\n      theme_bw()\n    \n    pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.list)[i],\u0026quot;_pca_keptgenes.pdf\u0026quot;),\n          height=5,\n          width=5)\n      print(pca.plot)\n    dev.off()\n      \n    png(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.list)[i],\u0026quot;_pca_keptgenes.png\u0026quot;),\n  \t    height=5,\n  \t    width=5,\n  \t    units = \u0026quot;in\u0026quot;,res=300)\n  \t  print(pca.plot)\n  \tdev.off()\n    \n    print(pca.plot)\n  }\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(tpm.kept.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.kept.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(tpm.kept.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  if(nrow(tpm.kept.list[[i]]) \u0026gt; 0){\n    pca.df \u0026lt;- t(scale(t(log2(tpm.kept.list[[i]] + 1))))\n    pca.df \u0026lt;- pca.df[rowSums(pca.df == 0) != ncol(pca.df),]\n    pca \u0026lt;- PCA(as.data.frame(scale(t(pca.df))), graph = FALSE, ncp = ncol(tpm.kept.list[[i]]) - 1)\n    \n    pca.plot \u0026lt;- ggplot()+\n      geom_point(aes(x=pca$ind$coord[,1], y=pca$ind$coord[,2], color = groups[,2],size = colSums(counts.kept.list[[i]])))+\n      labs(col = \"Samples\", size = \"Reads Mapped\\nto Features\", \n           x = paste0(\"PC1 (\", round(pca$eig[1,2],1), \"%)\"), \n           y = paste0(\"PC2 (\", round(pca$eig[2,2],1), \"%)\"))+\n      guides(color = F,size = F)+\n      # guides(colour = guide_legend(ncol = 2))+\n      scale_color_manual(values = levels(groups[,3]))+\n      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = \"log\")+\n      theme_bw()\n    \n    pdf(paste0(OUTPUT.DIR,\"/\",names(counts.list)[i],\"_pca_keptgenes.pdf\"),\n          height=5,\n          width=5)\n      print(pca.plot)\n    dev.off()\n      \n    png(paste0(OUTPUT.DIR,\"/\",names(counts.list)[i],\"_pca_keptgenes.png\"),\n  \t    height=5,\n  \t    width=5,\n  \t    units = \"in\",res=300)\n  \t  print(pca.plot)\n  \tdev.off()\n    \n    print(pca.plot)\n  }\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewOo, Darby et al 2012\u003c/h5\u003e\u003ca id=\"user-content-woo-darby-et-al-2012-3\" class=\"anchor\" aria-label=\"Permalink: wOo, Darby et al 2012\" href=\"#woo-darby-et-al-2012-3\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_pca_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_pca_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Darby et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wmel-darby-et-al-2014-3\" class=\"anchor\" aria-label=\"Permalink: wMel, Darby et al 2014\" href=\"#wmel-darby-et-al-2014-3\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_pca_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_pca_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2014-3\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2014\" href=\"#wdi-luck-et-al-2014-3\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_pca_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_pca_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2015-3\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2015\" href=\"#wdi-luck-et-al-2015-3\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_pca_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_pca_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Gutzwiller et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wmel-gutzwiller-et-al-2015-3\" class=\"anchor\" aria-label=\"Permalink: wMel, Gutzwiller et al 2015\" href=\"#wmel-gutzwiller-et-al-2015-3\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_pca_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_pca_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Grote et al 2017\u003c/h5\u003e\u003ca id=\"user-content-wbm-grote-et-al-2017-3\" class=\"anchor\" aria-label=\"Permalink: wBm, Grote et al 2017\" href=\"#wbm-grote-et-al-2017-3\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_pca_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_pca_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Chung et al 2019\u003c/h5\u003e\u003ca id=\"user-content-wbm-chung-et-al-2019-3\" class=\"anchor\" aria-label=\"Permalink: wBm, Chung et al 2019\" href=\"#wbm-chung-et-al-2019-3\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_pca_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_pca_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDifferentially expressed genes \u003ca name=\"user-content-ranalysis_de_pca_de\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-differentially-expressed-genes-\" class=\"anchor\" aria-label=\"Permalink: Differentially expressed genes \" href=\"#differentially-expressed-genes-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(tpm.edgeR.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.edgeR.list)[i]\u0026amp; sample_map$sample_identifier %in% colnames(tpm.edgeR.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  if(nrow(tpm.edgeR.list[[i]]) \u0026gt; 0){\n    pca.df \u0026lt;- t(scale(t(log2(tpm.edgeR.list[[i]] + 1))))\n    pca.df \u0026lt;- pca.df[rowSums(pca.df == 0) != ncol(pca.df),]\n    pca \u0026lt;- PCA(as.data.frame(scale(t(pca.df))), graph = FALSE, ncp = ncol(tpm.edgeR.list[[i]]) - 1)\n    \n    pca.plot \u0026lt;- ggplot()+\n      geom_point(aes(x=pca$ind$coord[,1], y=pca$ind$coord[,2], color = groups[,2],size = colSums(counts.edgeR.list[[i]])))+\n      labs(col = \u0026quot;Samples\u0026quot;, size = \u0026quot;Reads Mapped\\nto Features\u0026quot;, \n           x = paste0(\u0026quot;PC1 (\u0026quot;, round(pca$eig[1,2],1), \u0026quot;%)\u0026quot;), \n           y = paste0(\u0026quot;PC2 (\u0026quot;, round(pca$eig[2,2],1), \u0026quot;%)\u0026quot;))+\n      guides(color = F,size = F)+\n      # guides(colour = guide_legend(ncol = 2))+\n      scale_color_manual(values = levels(groups[,3]))+\n      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = \u0026quot;log\u0026quot;)+\n      theme_bw()\n    \n    pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.list)[i],\u0026quot;_pca_degenes.pdf\u0026quot;),\n        height=5,\n        width=5)\n      print(pca.plot)\n    dev.off()\n    \n    png(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.list)[i],\u0026quot;_pca_degenes.png\u0026quot;),\n        height=5,\n        width=5,\n        units = \u0026quot;in\u0026quot;,res=300)\n      print(pca.plot)\n    dev.off()\n    \n    print(pca.plot)\n  }\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(tpm.edgeR.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.edgeR.list)[i]\u0026amp; sample_map$sample_identifier %in% colnames(tpm.edgeR.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  if(nrow(tpm.edgeR.list[[i]]) \u0026gt; 0){\n    pca.df \u0026lt;- t(scale(t(log2(tpm.edgeR.list[[i]] + 1))))\n    pca.df \u0026lt;- pca.df[rowSums(pca.df == 0) != ncol(pca.df),]\n    pca \u0026lt;- PCA(as.data.frame(scale(t(pca.df))), graph = FALSE, ncp = ncol(tpm.edgeR.list[[i]]) - 1)\n    \n    pca.plot \u0026lt;- ggplot()+\n      geom_point(aes(x=pca$ind$coord[,1], y=pca$ind$coord[,2], color = groups[,2],size = colSums(counts.edgeR.list[[i]])))+\n      labs(col = \"Samples\", size = \"Reads Mapped\\nto Features\", \n           x = paste0(\"PC1 (\", round(pca$eig[1,2],1), \"%)\"), \n           y = paste0(\"PC2 (\", round(pca$eig[2,2],1), \"%)\"))+\n      guides(color = F,size = F)+\n      # guides(colour = guide_legend(ncol = 2))+\n      scale_color_manual(values = levels(groups[,3]))+\n      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = \"log\")+\n      theme_bw()\n    \n    pdf(paste0(OUTPUT.DIR,\"/\",names(counts.list)[i],\"_pca_degenes.pdf\"),\n        height=5,\n        width=5)\n      print(pca.plot)\n    dev.off()\n    \n    png(paste0(OUTPUT.DIR,\"/\",names(counts.list)[i],\"_pca_degenes.png\"),\n        height=5,\n        width=5,\n        units = \"in\",res=300)\n      print(pca.plot)\n    dev.off()\n    \n    print(pca.plot)\n  }\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Darby et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wmel-darby-et-al-2014-4\" class=\"anchor\" aria-label=\"Permalink: wMel, Darby et al 2014\" href=\"#wmel-darby-et-al-2014-4\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_pca_degenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_pca_degenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2015-4\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2015\" href=\"#wdi-luck-et-al-2015-4\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_pca_degenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_pca_degenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Gutzwiller et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wmel-gutzwiller-et-al-2015-4\" class=\"anchor\" aria-label=\"Permalink: wMel, Gutzwiller et al 2015\" href=\"#wmel-gutzwiller-et-al-2015-4\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_pca_degenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_pca_degenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Grote et al 2017\u003c/h5\u003e\u003ca id=\"user-content-wbm-grote-et-al-2017-4\" class=\"anchor\" aria-label=\"Permalink: wBm, Grote et al 2017\" href=\"#wbm-grote-et-al-2017-4\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_pca_degenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_pca_degenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Chung et al 2019\u003c/h5\u003e\u003ca id=\"user-content-wbm-chung-et-al-2019-4\" class=\"anchor\" aria-label=\"Permalink: wBm, Chung et al 2019\" href=\"#wbm-chung-et-al-2019-4\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_pca_degenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_pca_degenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConduct hierarchical clustering analyses \u003ca name=\"user-content-ranalysis_de_hc\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-conduct-hierarchical-clustering-analyses-\" class=\"anchor\" aria-label=\"Permalink: Conduct hierarchical clustering analyses \" href=\"#conduct-hierarchical-clustering-analyses-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAll genes \u003ca name=\"user-content-ranalysis_de_hc_all\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-all-genes--1\" class=\"anchor\" aria-label=\"Permalink: All genes \" href=\"#all-genes--1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(tpm.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.list)[i]\u0026amp; sample_map$sample_identifier %in% colnames(tpm.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  dendrogram \u0026lt;- as.data.frame(t(scale(t(log2(tpm.list[[i]] + 1)))))\n  dendrogram \u0026lt;- dendrogram[rowSums(is.na(dendrogram)) == 0,]\n  result \u0026lt;- pvclust(dendrogram, method.dist=\u0026quot;cor\u0026quot;, method.hclust=\u0026quot;average\u0026quot;, nboot=100, quiet=T)\n  \n  structure \u0026lt;- get_dendro_structure(result)\n  dendro.data \u0026lt;- get_dendro_data(result)\n  bootstrap.positions \u0026lt;- get_dendro_bootstraps(dendro.data)\n    \n  points.df \u0026lt;- as.data.frame(cbind(seq(1,length(structure),1),\n                                   structure))\n  dendrogroups \u0026lt;- groups[,2][result$hclust$order]\n  dendrocol \u0026lt;- groups[,3][result$hclust$order]\n  dendrosize \u0026lt;- colSums(counts.list[[i]])[result$hclust$order]\n  \n  for(j in 1:length(unique(groups[,1]))){\n    result$hclust$labels \u0026lt;- gsub(groups[,1][j],groups[,2][j],result$hclust$labels)\n  }\n  \n  dendrogram.plot \u0026lt;- ggdendrogram(hang.dendrogram(as.dendrogram(result$hclust)), theme_dendro = T)+\n    geom_point(aes(x=seq(1,length(structure)), y = structure, color = dendrogroups, size = dendrosize))+\n    scale_color_manual(values = levels(groups[,3]))+\n    labs(x = \u0026quot;\u0026quot;, y = \u0026quot;\u0026quot;, col = \u0026quot;Samples\u0026quot;, size = \u0026quot;Reads Mapped\\nto Features\u0026quot;)+\n    guides(colour = guide_legend(ncol = 2))+\n    theme_minimal()+      \n    scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = \u0026quot;log\u0026quot;)+\n    theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust = 1),\n          axis.text.y = element_blank(),\n          panel.grid.major = element_blank(),\n          panel.grid.minor = element_blank(),\n          legend.position=\u0026quot;none\u0026quot;)\n  \n  for(j in 1:length(result$edges$bp)){\n    text \u0026lt;- round(result$edges$bp[j] * 100,0)\n    dendrogram.plot \u0026lt;- dendrogram.plot + annotate(\u0026quot;text\u0026quot;, label = text, x=bootstrap.positions[j,1] + 0.4, y=bootstrap.positions[j,2] + 0.03, size = 3)\n  }\n  \n  pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.list)[i],\u0026quot;_dendrogram_allgenes.pdf\u0026quot;),\n      width = 11,\n      height = 8)\n    print(dendrogram.plot)\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.list)[i],\u0026quot;_dendrogram_allgenes.png\u0026quot;),\n      width = 11,\n      height = 8,\n      units = \u0026quot;in\u0026quot;,res=300)\n    print(dendrogram.plot)\n  dev.off()\n  \n  print(dendrogram.plot)\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(tpm.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.list)[i]\u0026amp; sample_map$sample_identifier %in% colnames(tpm.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  dendrogram \u0026lt;- as.data.frame(t(scale(t(log2(tpm.list[[i]] + 1)))))\n  dendrogram \u0026lt;- dendrogram[rowSums(is.na(dendrogram)) == 0,]\n  result \u0026lt;- pvclust(dendrogram, method.dist=\"cor\", method.hclust=\"average\", nboot=100, quiet=T)\n  \n  structure \u0026lt;- get_dendro_structure(result)\n  dendro.data \u0026lt;- get_dendro_data(result)\n  bootstrap.positions \u0026lt;- get_dendro_bootstraps(dendro.data)\n    \n  points.df \u0026lt;- as.data.frame(cbind(seq(1,length(structure),1),\n                                   structure))\n  dendrogroups \u0026lt;- groups[,2][result$hclust$order]\n  dendrocol \u0026lt;- groups[,3][result$hclust$order]\n  dendrosize \u0026lt;- colSums(counts.list[[i]])[result$hclust$order]\n  \n  for(j in 1:length(unique(groups[,1]))){\n    result$hclust$labels \u0026lt;- gsub(groups[,1][j],groups[,2][j],result$hclust$labels)\n  }\n  \n  dendrogram.plot \u0026lt;- ggdendrogram(hang.dendrogram(as.dendrogram(result$hclust)), theme_dendro = T)+\n    geom_point(aes(x=seq(1,length(structure)), y = structure, color = dendrogroups, size = dendrosize))+\n    scale_color_manual(values = levels(groups[,3]))+\n    labs(x = \"\", y = \"\", col = \"Samples\", size = \"Reads Mapped\\nto Features\")+\n    guides(colour = guide_legend(ncol = 2))+\n    theme_minimal()+      \n    scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = \"log\")+\n    theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust = 1),\n          axis.text.y = element_blank(),\n          panel.grid.major = element_blank(),\n          panel.grid.minor = element_blank(),\n          legend.position=\"none\")\n  \n  for(j in 1:length(result$edges$bp)){\n    text \u0026lt;- round(result$edges$bp[j] * 100,0)\n    dendrogram.plot \u0026lt;- dendrogram.plot + annotate(\"text\", label = text, x=bootstrap.positions[j,1] + 0.4, y=bootstrap.positions[j,2] + 0.03, size = 3)\n  }\n  \n  pdf(paste0(OUTPUT.DIR,\"/\",names(counts.list)[i],\"_dendrogram_allgenes.pdf\"),\n      width = 11,\n      height = 8)\n    print(dendrogram.plot)\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\"/\",names(counts.list)[i],\"_dendrogram_allgenes.png\"),\n      width = 11,\n      height = 8,\n      units = \"in\",res=300)\n    print(dendrogram.plot)\n  dev.off()\n  \n  print(dendrogram.plot)\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewOo, Darby et al 2012\u003c/h5\u003e\u003ca id=\"user-content-woo-darby-et-al-2012-4\" class=\"anchor\" aria-label=\"Permalink: wOo, Darby et al 2012\" href=\"#woo-darby-et-al-2012-4\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_dendrogram_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_dendrogram_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Darby et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wmel-darby-et-al-2014-5\" class=\"anchor\" aria-label=\"Permalink: wMel, Darby et al 2014\" href=\"#wmel-darby-et-al-2014-5\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_dendrogram_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_dendrogram_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2014-4\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2014\" href=\"#wdi-luck-et-al-2014-4\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_dendrogram_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_dendrogram_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2015-5\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2015\" href=\"#wdi-luck-et-al-2015-5\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_dendrogram_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_dendrogram_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Gutzwiller et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wmel-gutzwiller-et-al-2015-5\" class=\"anchor\" aria-label=\"Permalink: wMel, Gutzwiller et al 2015\" href=\"#wmel-gutzwiller-et-al-2015-5\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_dendrogram_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_dendrogram_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Grote et al 2017\u003c/h5\u003e\u003ca id=\"user-content-wbm-grote-et-al-2017-5\" class=\"anchor\" aria-label=\"Permalink: wBm, Grote et al 2017\" href=\"#wbm-grote-et-al-2017-5\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_dendrogram_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_dendrogram_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Chung et al 2019\u003c/h5\u003e\u003ca id=\"user-content-wbm-chung-et-al-2019-5\" class=\"anchor\" aria-label=\"Permalink: wBm, Chung et al 2019\" href=\"#wbm-chung-et-al-2019-5\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_dendrogram_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_dendrogram_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eKept genes \u003ca name=\"user-content-ranalysis_de_hc_kept\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-kept-genes--1\" class=\"anchor\" aria-label=\"Permalink: Kept genes \" href=\"#kept-genes--1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(tpm.kept.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.kept.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(tpm.kept.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  dendrogram \u0026lt;- as.data.frame(t(scale(t(log2(tpm.kept.list[[i]]) + 1))))\n  if(nrow(dendrogram) \u0026gt; 0){\n    result \u0026lt;- pvclust(dendrogram, method.dist=\u0026quot;cor\u0026quot;, method.hclust=\u0026quot;average\u0026quot;, nboot=100, quiet=T)\n    \n    structure \u0026lt;- get_dendro_structure(result)\n    dendro.data \u0026lt;- get_dendro_data(result)\n    bootstrap.positions \u0026lt;- get_dendro_bootstraps(dendro.data)\n      \n    points.df \u0026lt;- as.data.frame(cbind(seq(1,length(structure),1),\n                                     structure))\n    dendrogroups \u0026lt;- groups[,2][result$hclust$order]\n    dendrocol \u0026lt;- groups[,3][result$hclust$order]\n    dendrosize \u0026lt;- colSums(counts.kept.list[[i]])[result$hclust$order]\n    \n    for(j in 1:length(unique(groups[,1]))){\n      result$hclust$labels \u0026lt;- gsub(groups[,1][j],groups[,2][j],result$hclust$labels)\n    }\n    \n    dendrogram.plot \u0026lt;- ggdendrogram(hang.dendrogram(as.dendrogram(result$hclust)), theme_dendro = T)+\n      geom_point(aes(x=seq(1,length(structure)), y = structure, color = dendrogroups, size = dendrosize))+\n      scale_color_manual(values = levels(groups[,3]))+\n      labs(x = \u0026quot;\u0026quot;, y = \u0026quot;\u0026quot;, col = \u0026quot;Samples\u0026quot;, size = \u0026quot;Reads Mapped\\nto Features\u0026quot;)+\n      guides(colour = guide_legend(ncol = 2))+\n      theme_minimal()+\n      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = \u0026quot;log\u0026quot;)+\n      theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust = 1),\n            axis.text.y = element_blank(),\n            panel.grid.major = element_blank(),\n            panel.grid.minor = element_blank(),\n            legend.position=\u0026quot;none\u0026quot;)\n    \n    for(j in 1:length(result$edges$bp)){\n      text \u0026lt;- round(result$edges$bp[j] * 100,0)\n      dendrogram.plot \u0026lt;- dendrogram.plot + annotate(\u0026quot;text\u0026quot;, label = text, x=bootstrap.positions[j,1] + 0.4, y=bootstrap.positions[j,2] + 0.03, size = 3)\n    }\n    pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.kept.list)[i],\u0026quot;_dendrogram_keptgenes.pdf\u0026quot;),\n        width = 11,\n        height = 8)\n      print(dendrogram.plot)\n    dev.off()\n    \n    png(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.kept.list)[i],\u0026quot;_dendrogram_keptgenes.png\u0026quot;),\n        width = 11,\n        height = 8,\n        units = \u0026quot;in\u0026quot;,res=300)\n      print(dendrogram.plot)\n    dev.off()\n    \n    print(dendrogram.plot)\n  }\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(tpm.kept.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.kept.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(tpm.kept.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  dendrogram \u0026lt;- as.data.frame(t(scale(t(log2(tpm.kept.list[[i]]) + 1))))\n  if(nrow(dendrogram) \u0026gt; 0){\n    result \u0026lt;- pvclust(dendrogram, method.dist=\"cor\", method.hclust=\"average\", nboot=100, quiet=T)\n    \n    structure \u0026lt;- get_dendro_structure(result)\n    dendro.data \u0026lt;- get_dendro_data(result)\n    bootstrap.positions \u0026lt;- get_dendro_bootstraps(dendro.data)\n      \n    points.df \u0026lt;- as.data.frame(cbind(seq(1,length(structure),1),\n                                     structure))\n    dendrogroups \u0026lt;- groups[,2][result$hclust$order]\n    dendrocol \u0026lt;- groups[,3][result$hclust$order]\n    dendrosize \u0026lt;- colSums(counts.kept.list[[i]])[result$hclust$order]\n    \n    for(j in 1:length(unique(groups[,1]))){\n      result$hclust$labels \u0026lt;- gsub(groups[,1][j],groups[,2][j],result$hclust$labels)\n    }\n    \n    dendrogram.plot \u0026lt;- ggdendrogram(hang.dendrogram(as.dendrogram(result$hclust)), theme_dendro = T)+\n      geom_point(aes(x=seq(1,length(structure)), y = structure, color = dendrogroups, size = dendrosize))+\n      scale_color_manual(values = levels(groups[,3]))+\n      labs(x = \"\", y = \"\", col = \"Samples\", size = \"Reads Mapped\\nto Features\")+\n      guides(colour = guide_legend(ncol = 2))+\n      theme_minimal()+\n      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = \"log\")+\n      theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust = 1),\n            axis.text.y = element_blank(),\n            panel.grid.major = element_blank(),\n            panel.grid.minor = element_blank(),\n            legend.position=\"none\")\n    \n    for(j in 1:length(result$edges$bp)){\n      text \u0026lt;- round(result$edges$bp[j] * 100,0)\n      dendrogram.plot \u0026lt;- dendrogram.plot + annotate(\"text\", label = text, x=bootstrap.positions[j,1] + 0.4, y=bootstrap.positions[j,2] + 0.03, size = 3)\n    }\n    pdf(paste0(OUTPUT.DIR,\"/\",names(counts.kept.list)[i],\"_dendrogram_keptgenes.pdf\"),\n        width = 11,\n        height = 8)\n      print(dendrogram.plot)\n    dev.off()\n    \n    png(paste0(OUTPUT.DIR,\"/\",names(counts.kept.list)[i],\"_dendrogram_keptgenes.png\"),\n        width = 11,\n        height = 8,\n        units = \"in\",res=300)\n      print(dendrogram.plot)\n    dev.off()\n    \n    print(dendrogram.plot)\n  }\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewOo, Darby et al 2012\u003c/h5\u003e\u003ca id=\"user-content-woo-darby-et-al-2012-5\" class=\"anchor\" aria-label=\"Permalink: wOo, Darby et al 2012\" href=\"#woo-darby-et-al-2012-5\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_dendrogram_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_dendrogram_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Darby et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wmel-darby-et-al-2014-6\" class=\"anchor\" aria-label=\"Permalink: wMel, Darby et al 2014\" href=\"#wmel-darby-et-al-2014-6\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_dendrogram_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_dendrogram_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2014-5\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2014\" href=\"#wdi-luck-et-al-2014-5\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_dendrogram_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_dendrogram_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2015-6\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2015\" href=\"#wdi-luck-et-al-2015-6\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_dendrogram_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_dendrogram_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Gutzwiller et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wmel-gutzwiller-et-al-2015-6\" class=\"anchor\" aria-label=\"Permalink: wMel, Gutzwiller et al 2015\" href=\"#wmel-gutzwiller-et-al-2015-6\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_dendrogram_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_dendrogram_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Grote et al 2017\u003c/h5\u003e\u003ca id=\"user-content-wbm-grote-et-al-2017-6\" class=\"anchor\" aria-label=\"Permalink: wBm, Grote et al 2017\" href=\"#wbm-grote-et-al-2017-6\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_dendrogram_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_dendrogram_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Chung et al 2019\u003c/h5\u003e\u003ca id=\"user-content-wbm-chung-et-al-2019-6\" class=\"anchor\" aria-label=\"Permalink: wBm, Chung et al 2019\" href=\"#wbm-chung-et-al-2019-6\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_dendrogram_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_dendrogram_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDifferentially expressed genes \u003ca name=\"user-content-ranalysis_de_hc_de\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-differentially-expressed-genes--1\" class=\"anchor\" aria-label=\"Permalink: Differentially expressed genes \" href=\"#differentially-expressed-genes--1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(tpm.edgeR.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.edgeR.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(tpm.edgeR.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  dendrogram \u0026lt;- as.data.frame(t(scale(t(log2(tpm.edgeR.list[[i]]) + 1))))\n  if(nrow(dendrogram) \u0026gt; 0){\n    result \u0026lt;- pvclust(dendrogram, method.dist=\u0026quot;cor\u0026quot;, method.hclust=\u0026quot;average\u0026quot;, nboot=100, quiet=T)\n    \n    structure \u0026lt;- get_dendro_structure(result)\n    dendro.data \u0026lt;- get_dendro_data(result)\n    bootstrap.positions \u0026lt;- get_dendro_bootstraps(dendro.data)\n      \n    points.df \u0026lt;- as.data.frame(cbind(seq(1,length(structure),1),\n                                     structure))\n    dendrogroups \u0026lt;- groups[,2][result$hclust$order]\n    dendrocol \u0026lt;- groups[,3][result$hclust$order]\n    dendrosize \u0026lt;- colSums(counts.edgeR.list[[i]])[result$hclust$order]\n    \n    for(j in 1:length(unique(groups[,1]))){\n      result$hclust$labels \u0026lt;- gsub(groups[,1][j],groups[,2][j],result$hclust$labels)\n    }\n    \n    dendrogram.plot \u0026lt;- ggdendrogram(hang.dendrogram(as.dendrogram(result$hclust)), theme_dendro = T)+\n      geom_point(aes(x=seq(1,length(structure)), y = structure, color = dendrogroups, size = dendrosize))+\n      scale_color_manual(values = levels(groups[,3]))+\n      labs(x = \u0026quot;\u0026quot;, y = \u0026quot;\u0026quot;, col = \u0026quot;Samples\u0026quot;, size = \u0026quot;Reads Mapped\\nto Features\u0026quot;)+\n      guides(colour = guide_legend(ncol = 2))+\n      theme_minimal()+\n      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = \u0026quot;log\u0026quot;)+\n      theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust = 1),\n            axis.text.y = element_blank(),\n            panel.grid.major = element_blank(),\n            panel.grid.minor = element_blank(),\n            legend.position=\u0026quot;none\u0026quot;)\n    \n    for(j in 1:length(result$edges$bp)){\n      text \u0026lt;- round(result$edges$bp[j] * 100,0)\n      dendrogram.plot \u0026lt;- dendrogram.plot + annotate(\u0026quot;text\u0026quot;, label = text, x=bootstrap.positions[j,1] + 0.4, y=bootstrap.positions[j,2] + 0.03, size = 3)\n    }\n    \n    pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.edgeR.list)[i],\u0026quot;_dendrogram_degenes.pdf\u0026quot;),\n        width = 11,\n        height = 8)\n      print(dendrogram.plot)\n    dev.off()\n    \n    png(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(counts.edgeR.list)[i],\u0026quot;_dendrogram_degenes.png\u0026quot;),\n        width = 11,\n        height = 8,\n        units = \u0026quot;in\u0026quot;,res=300)\n      print(dendrogram.plot)\n    dev.off()\n    \n    print(dendrogram.plot)\n  }\n}\"\u003e\u003cpre lang=\"{R,fig.height=8,fig.width=11}\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(tpm.edgeR.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.edgeR.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(tpm.edgeR.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  dendrogram \u0026lt;- as.data.frame(t(scale(t(log2(tpm.edgeR.list[[i]]) + 1))))\n  if(nrow(dendrogram) \u0026gt; 0){\n    result \u0026lt;- pvclust(dendrogram, method.dist=\"cor\", method.hclust=\"average\", nboot=100, quiet=T)\n    \n    structure \u0026lt;- get_dendro_structure(result)\n    dendro.data \u0026lt;- get_dendro_data(result)\n    bootstrap.positions \u0026lt;- get_dendro_bootstraps(dendro.data)\n      \n    points.df \u0026lt;- as.data.frame(cbind(seq(1,length(structure),1),\n                                     structure))\n    dendrogroups \u0026lt;- groups[,2][result$hclust$order]\n    dendrocol \u0026lt;- groups[,3][result$hclust$order]\n    dendrosize \u0026lt;- colSums(counts.edgeR.list[[i]])[result$hclust$order]\n    \n    for(j in 1:length(unique(groups[,1]))){\n      result$hclust$labels \u0026lt;- gsub(groups[,1][j],groups[,2][j],result$hclust$labels)\n    }\n    \n    dendrogram.plot \u0026lt;- ggdendrogram(hang.dendrogram(as.dendrogram(result$hclust)), theme_dendro = T)+\n      geom_point(aes(x=seq(1,length(structure)), y = structure, color = dendrogroups, size = dendrosize))+\n      scale_color_manual(values = levels(groups[,3]))+\n      labs(x = \"\", y = \"\", col = \"Samples\", size = \"Reads Mapped\\nto Features\")+\n      guides(colour = guide_legend(ncol = 2))+\n      theme_minimal()+\n      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = \"log\")+\n      theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust = 1),\n            axis.text.y = element_blank(),\n            panel.grid.major = element_blank(),\n            panel.grid.minor = element_blank(),\n            legend.position=\"none\")\n    \n    for(j in 1:length(result$edges$bp)){\n      text \u0026lt;- round(result$edges$bp[j] * 100,0)\n      dendrogram.plot \u0026lt;- dendrogram.plot + annotate(\"text\", label = text, x=bootstrap.positions[j,1] + 0.4, y=bootstrap.positions[j,2] + 0.03, size = 3)\n    }\n    \n    pdf(paste0(OUTPUT.DIR,\"/\",names(counts.edgeR.list)[i],\"_dendrogram_degenes.pdf\"),\n        width = 11,\n        height = 8)\n      print(dendrogram.plot)\n    dev.off()\n    \n    png(paste0(OUTPUT.DIR,\"/\",names(counts.edgeR.list)[i],\"_dendrogram_degenes.png\"),\n        width = 11,\n        height = 8,\n        units = \"in\",res=300)\n      print(dendrogram.plot)\n    dev.off()\n    \n    print(dendrogram.plot)\n  }\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Darby et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wmel-darby-et-al-2014-7\" class=\"anchor\" aria-label=\"Permalink: wMel, Darby et al 2014\" href=\"#wmel-darby-et-al-2014-7\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_dendrogram_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_dendrogram_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2015-7\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2015\" href=\"#wdi-luck-et-al-2015-7\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_dendrogram_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_dendrogram_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Gutzwiller et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wmel-gutzwiller-et-al-2015-7\" class=\"anchor\" aria-label=\"Permalink: wMel, Gutzwiller et al 2015\" href=\"#wmel-gutzwiller-et-al-2015-7\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_dendrogram_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_dendrogram_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Grote et al 2017\u003c/h5\u003e\u003ca id=\"user-content-wbm-grote-et-al-2017-7\" class=\"anchor\" aria-label=\"Permalink: wBm, Grote et al 2017\" href=\"#wbm-grote-et-al-2017-7\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_dendrogram_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_dendrogram_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Chung et al 2019\u003c/h5\u003e\u003ca id=\"user-content-wbm-chung-et-al-2019-7\" class=\"anchor\" aria-label=\"Permalink: wBm, Chung et al 2019\" href=\"#wbm-chung-et-al-2019-7\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_dendrogram_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_dendrogram_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCreate TPM and log2TPM heatmaps \u003ca name=\"user-content-ranalysis_de_heatmap\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-create-tpm-and-log2tpm-heatmaps-\" class=\"anchor\" aria-label=\"Permalink: Create TPM and log2TPM heatmaps \" href=\"#create-tpm-and-log2tpm-heatmaps-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAll genes \u003ca name=\"user-content-ranalysis_de_heatmap_all\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-all-genes--2\" class=\"anchor\" aria-label=\"Permalink: All genes \" href=\"#all-genes--2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(tpm.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(tpm.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  heatmap1.df \u0026lt;- as.data.frame(log2(tpm.list[[i]]+1))\n  heatmap2.df \u0026lt;- as.data.frame(t(scale(t(log2(tpm.list[[i]] + 1)))))\n  \n  colcol \u0026lt;- as.character(groups[,3])\n  \n  #rowdendro \u0026lt;- pvclust(t(heatmap1.df), method.dist=\u0026quot;cor\u0026quot;, method.hclust=\u0026quot;average\u0026quot;, nboot=100)\n  coldendro \u0026lt;- pvclust(heatmap2.df, method.dist=\u0026quot;cor\u0026quot;, method.hclust=\u0026quot;average\u0026quot;, nboot=100, quiet=T)\n  \n  heatmap2.df[is.na(heatmap2.df)] \u0026lt;- 0\n  \n  pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.list)[i],\u0026quot;_heatmap_allgenes.pdf\u0026quot;),\n      width = 8,\n      height = 10)\n  heatmap.2(as.matrix(heatmap1.df),\n            col=hmcol1,\n            trace=\u0026quot;none\u0026quot;,\n            labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap1.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(0,10,by=.5),\n            symkey=F,\n            dendrogram = \u0026quot;both\u0026quot;)\n  \n  heatmap.2(as.matrix(heatmap2.df),\n            col=hmcol2,\n            trace=\u0026quot;none\u0026quot;,\n            labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap2.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(-3,3,by=.5),\n            symkey=F,\n            dendrogram = \u0026quot;both\u0026quot;)\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.list)[i],\u0026quot;_heatmap_allgenes.png\u0026quot;),\n      width = 8,\n      height = 10,\n      units = \u0026quot;in\u0026quot;,res=300)\n  heatmap.2(as.matrix(heatmap1.df),\n            col=hmcol1,\n            trace=\u0026quot;none\u0026quot;,\n            labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap1.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(0,10,by=.5),\n            symkey=F,\n            dendrogram = \u0026quot;both\u0026quot;)\n  \n  heatmap.2(as.matrix(heatmap2.df),\n            col=hmcol2,\n            trace=\u0026quot;none\u0026quot;,\n            labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap2.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(-3,3,by=.5),\n            symkey=F,\n            dendrogram = \u0026quot;both\u0026quot;)\n  dev.off()\n  \n  heatmap.2(as.matrix(heatmap1.df),\n            col=hmcol1,\n            trace=\u0026quot;none\u0026quot;,\n            labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap1.df)),\n            labCol=groups[,2],\n           #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(0,10,by=.5),\n            symkey=F,\n            dendrogram = \u0026quot;both\u0026quot;)\n  \n  heatmap.2(as.matrix(heatmap2.df),\n            col=hmcol2,\n            trace=\u0026quot;none\u0026quot;,\n            labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap2.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(-3,3,by=.5),\n            symkey=F,\n            dendrogram = \u0026quot;both\u0026quot;)\n}\"\u003e\u003cpre lang=\"{R,fig.height=10,fig.width=8}\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(tpm.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(tpm.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  heatmap1.df \u0026lt;- as.data.frame(log2(tpm.list[[i]]+1))\n  heatmap2.df \u0026lt;- as.data.frame(t(scale(t(log2(tpm.list[[i]] + 1)))))\n  \n  colcol \u0026lt;- as.character(groups[,3])\n  \n  #rowdendro \u0026lt;- pvclust(t(heatmap1.df), method.dist=\"cor\", method.hclust=\"average\", nboot=100)\n  coldendro \u0026lt;- pvclust(heatmap2.df, method.dist=\"cor\", method.hclust=\"average\", nboot=100, quiet=T)\n  \n  heatmap2.df[is.na(heatmap2.df)] \u0026lt;- 0\n  \n  pdf(paste0(OUTPUT.DIR,\"/\",names(tpm.list)[i],\"_heatmap_allgenes.pdf\"),\n      width = 8,\n      height = 10)\n  heatmap.2(as.matrix(heatmap1.df),\n            col=hmcol1,\n            trace=\"none\",\n            labRow=vector(mode = \"character\", length = nrow(heatmap1.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(0,10,by=.5),\n            symkey=F,\n            dendrogram = \"both\")\n  \n  heatmap.2(as.matrix(heatmap2.df),\n            col=hmcol2,\n            trace=\"none\",\n            labRow=vector(mode = \"character\", length = nrow(heatmap2.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(-3,3,by=.5),\n            symkey=F,\n            dendrogram = \"both\")\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\"/\",names(tpm.list)[i],\"_heatmap_allgenes.png\"),\n      width = 8,\n      height = 10,\n      units = \"in\",res=300)\n  heatmap.2(as.matrix(heatmap1.df),\n            col=hmcol1,\n            trace=\"none\",\n            labRow=vector(mode = \"character\", length = nrow(heatmap1.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(0,10,by=.5),\n            symkey=F,\n            dendrogram = \"both\")\n  \n  heatmap.2(as.matrix(heatmap2.df),\n            col=hmcol2,\n            trace=\"none\",\n            labRow=vector(mode = \"character\", length = nrow(heatmap2.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(-3,3,by=.5),\n            symkey=F,\n            dendrogram = \"both\")\n  dev.off()\n  \n  heatmap.2(as.matrix(heatmap1.df),\n            col=hmcol1,\n            trace=\"none\",\n            labRow=vector(mode = \"character\", length = nrow(heatmap1.df)),\n            labCol=groups[,2],\n           #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(0,10,by=.5),\n            symkey=F,\n            dendrogram = \"both\")\n  \n  heatmap.2(as.matrix(heatmap2.df),\n            col=hmcol2,\n            trace=\"none\",\n            labRow=vector(mode = \"character\", length = nrow(heatmap2.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(-3,3,by=.5),\n            symkey=F,\n            dendrogram = \"both\")\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewOo, Darby et al 2012\u003c/h5\u003e\u003ca id=\"user-content-woo-darby-et-al-2012-6\" class=\"anchor\" aria-label=\"Permalink: wOo, Darby et al 2012\" href=\"#woo-darby-et-al-2012-6\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_heatmap_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_heatmap_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Darby et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wmel-darby-et-al-2014-8\" class=\"anchor\" aria-label=\"Permalink: wMel, Darby et al 2014\" href=\"#wmel-darby-et-al-2014-8\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_heatmap_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_heatmap_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2014-6\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2014\" href=\"#wdi-luck-et-al-2014-6\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_heatmap_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_heatmap_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2015-8\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2015\" href=\"#wdi-luck-et-al-2015-8\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_heatmap_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_heatmap_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Gutzwiller et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wmel-gutzwiller-et-al-2015-8\" class=\"anchor\" aria-label=\"Permalink: wMel, Gutzwiller et al 2015\" href=\"#wmel-gutzwiller-et-al-2015-8\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_heatmap_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_heatmap_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Grote et al 2017\u003c/h5\u003e\u003ca id=\"user-content-wbm-grote-et-al-2017-8\" class=\"anchor\" aria-label=\"Permalink: wBm, Grote et al 2017\" href=\"#wbm-grote-et-al-2017-8\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_heatmap_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_heatmap_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Chung et al 2019\u003c/h5\u003e\u003ca id=\"user-content-wbm-chung-et-al-2019-8\" class=\"anchor\" aria-label=\"Permalink: wBm, Chung et al 2019\" href=\"#wbm-chung-et-al-2019-8\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_heatmap_allgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_heatmap_allgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eKept genes \u003ca name=\"user-content-ranalysis_de_heatmap_kept\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-kept-genes--2\" class=\"anchor\" aria-label=\"Permalink: Kept genes \" href=\"#kept-genes--2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(tpm.kept.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.kept.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(tpm.kept.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  heatmap1.df \u0026lt;- as.data.frame(log2(tpm.kept.list[[i]]+1))\n  heatmap2.df \u0026lt;- as.data.frame(t(scale(t(log2(tpm.kept.list[[i]] + 1)))))\n  \n  colcol \u0026lt;- as.character(groups[,3])\n  \n  #rowdendro \u0026lt;- pvclust(t(heatmap1.df), method.dist=\u0026quot;cor\u0026quot;, method.hclust=\u0026quot;average\u0026quot;, nboot=100)\n  coldendro \u0026lt;- pvclust(heatmap2.df, method.dist=\u0026quot;cor\u0026quot;, method.hclust=\u0026quot;average\u0026quot;, nboot=100, quiet=T)\n  \n  heatmap2.df[is.na(heatmap2.df)] \u0026lt;- 0\n  \n  pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.kept.list)[i],\u0026quot;_heatmap_keptgenes.pdf\u0026quot;),\n      width = 8,\n      height = 10)\n  heatmap.2(as.matrix(heatmap1.df),\n            col=hmcol1,\n            trace=\u0026quot;none\u0026quot;,\n            labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap1.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(0,10,by=.5),\n            symkey=F,\n            dendrogram = \u0026quot;both\u0026quot;)\n  \n  heatmap.2(as.matrix(heatmap2.df),\n            col=hmcol2,\n            trace=\u0026quot;none\u0026quot;,\n            labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap2.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(-3,3,by=.5),\n            symkey=F,\n            dendrogram = \u0026quot;both\u0026quot;)\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.kept.list)[i],\u0026quot;_heatmap_keptgenes.png\u0026quot;),\n      width = 8,\n      height = 10,\n      units = \u0026quot;in\u0026quot;,res=300)\n  heatmap.2(as.matrix(heatmap1.df),\n            col=hmcol1,\n            trace=\u0026quot;none\u0026quot;,\n            labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap1.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(0,10,by=.5),\n            symkey=F,\n            dendrogram = \u0026quot;both\u0026quot;)\n  \n  heatmap.2(as.matrix(heatmap2.df),\n            col=hmcol2,\n            trace=\u0026quot;none\u0026quot;,\n            labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap2.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(-3,3,by=.5),\n            symkey=F,\n            dendrogram = \u0026quot;both\u0026quot;)\n  dev.off()\n  \n  heatmap.2(as.matrix(heatmap1.df),\n            col=hmcol1,\n            trace=\u0026quot;none\u0026quot;,\n            labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap1.df)),\n            labCol=groups[,2],\n           #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(0,10,by=.5),\n            symkey=F,\n            dendrogram = \u0026quot;both\u0026quot;)\n  \n  heatmap.2(as.matrix(heatmap2.df),\n            col=hmcol2,\n            trace=\u0026quot;none\u0026quot;,\n            labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap2.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(-3,3,by=.5),\n            symkey=F,\n            dendrogram = \u0026quot;both\u0026quot;)\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(tpm.kept.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.kept.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(tpm.kept.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  heatmap1.df \u0026lt;- as.data.frame(log2(tpm.kept.list[[i]]+1))\n  heatmap2.df \u0026lt;- as.data.frame(t(scale(t(log2(tpm.kept.list[[i]] + 1)))))\n  \n  colcol \u0026lt;- as.character(groups[,3])\n  \n  #rowdendro \u0026lt;- pvclust(t(heatmap1.df), method.dist=\"cor\", method.hclust=\"average\", nboot=100)\n  coldendro \u0026lt;- pvclust(heatmap2.df, method.dist=\"cor\", method.hclust=\"average\", nboot=100, quiet=T)\n  \n  heatmap2.df[is.na(heatmap2.df)] \u0026lt;- 0\n  \n  pdf(paste0(OUTPUT.DIR,\"/\",names(tpm.kept.list)[i],\"_heatmap_keptgenes.pdf\"),\n      width = 8,\n      height = 10)\n  heatmap.2(as.matrix(heatmap1.df),\n            col=hmcol1,\n            trace=\"none\",\n            labRow=vector(mode = \"character\", length = nrow(heatmap1.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(0,10,by=.5),\n            symkey=F,\n            dendrogram = \"both\")\n  \n  heatmap.2(as.matrix(heatmap2.df),\n            col=hmcol2,\n            trace=\"none\",\n            labRow=vector(mode = \"character\", length = nrow(heatmap2.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(-3,3,by=.5),\n            symkey=F,\n            dendrogram = \"both\")\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\"/\",names(tpm.kept.list)[i],\"_heatmap_keptgenes.png\"),\n      width = 8,\n      height = 10,\n      units = \"in\",res=300)\n  heatmap.2(as.matrix(heatmap1.df),\n            col=hmcol1,\n            trace=\"none\",\n            labRow=vector(mode = \"character\", length = nrow(heatmap1.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(0,10,by=.5),\n            symkey=F,\n            dendrogram = \"both\")\n  \n  heatmap.2(as.matrix(heatmap2.df),\n            col=hmcol2,\n            trace=\"none\",\n            labRow=vector(mode = \"character\", length = nrow(heatmap2.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(-3,3,by=.5),\n            symkey=F,\n            dendrogram = \"both\")\n  dev.off()\n  \n  heatmap.2(as.matrix(heatmap1.df),\n            col=hmcol1,\n            trace=\"none\",\n            labRow=vector(mode = \"character\", length = nrow(heatmap1.df)),\n            labCol=groups[,2],\n           #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(0,10,by=.5),\n            symkey=F,\n            dendrogram = \"both\")\n  \n  heatmap.2(as.matrix(heatmap2.df),\n            col=hmcol2,\n            trace=\"none\",\n            labRow=vector(mode = \"character\", length = nrow(heatmap2.df)),\n            labCol=groups[,2],\n            #Rowv = as.dendrogram(rowdendro),\n            Colv = as.dendrogram(coldendro),\n            ColSideColors=colcol,\n            lwid=c(1,4),\n            lhei = c(1,5),\n            breaks = seq(-3,3,by=.5),\n            symkey=F,\n            dendrogram = \"both\")\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewOo, Darby et al 2012\u003c/h5\u003e\u003ca id=\"user-content-woo-darby-et-al-2012-7\" class=\"anchor\" aria-label=\"Permalink: wOo, Darby et al 2012\" href=\"#woo-darby-et-al-2012-7\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_heatmap_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_heatmap_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Darby et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wmel-darby-et-al-2014-9\" class=\"anchor\" aria-label=\"Permalink: wMel, Darby et al 2014\" href=\"#wmel-darby-et-al-2014-9\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_heatmap_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_heatmap_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2014-7\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2014\" href=\"#wdi-luck-et-al-2014-7\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_heatmap_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_heatmap_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2015-9\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2015\" href=\"#wdi-luck-et-al-2015-9\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_heatmap_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_heatmap_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Gutzwiller et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wmel-gutzwiller-et-al-2015-9\" class=\"anchor\" aria-label=\"Permalink: wMel, Gutzwiller et al 2015\" href=\"#wmel-gutzwiller-et-al-2015-9\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_heatmap_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_heatmap_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Grote et al 2017\u003c/h5\u003e\u003ca id=\"user-content-wbm-grote-et-al-2017-9\" class=\"anchor\" aria-label=\"Permalink: wBm, Grote et al 2017\" href=\"#wbm-grote-et-al-2017-9\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_heatmap_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_heatmap_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Chung et al 2019\u003c/h5\u003e\u003ca id=\"user-content-wbm-chung-et-al-2019-9\" class=\"anchor\" aria-label=\"Permalink: wBm, Chung et al 2019\" href=\"#wbm-chung-et-al-2019-9\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_heatmap_keptgenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_heatmap_keptgenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDifferentially expressed genes \u003ca name=\"user-content-ranalysis_de_heatmap_de\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-differentially-expressed-genes--2\" class=\"anchor\" aria-label=\"Permalink: Differentially expressed genes \" href=\"#differentially-expressed-genes--2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(tpm.edgeR.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.edgeR.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(tpm.edgeR.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  heatmap1.df \u0026lt;- as.data.frame(log2(tpm.edgeR.list[[i]]+1))\n  heatmap2.df \u0026lt;- as.data.frame(t(scale(t(log2(tpm.edgeR.list[[i]] + 1)))))\n  \n  colcol \u0026lt;- as.character(groups[,3])\n  \n  if(nrow(tpm.edgeR.list[[i]]) \u0026gt; 0){\n    #rowdendro \u0026lt;- pvclust(t(heatmap1.df), method.dist=\u0026quot;cor\u0026quot;, method.hclust=\u0026quot;average\u0026quot;, nboot=100)\n    coldendro \u0026lt;- pvclust(heatmap2.df, method.dist=\u0026quot;cor\u0026quot;, method.hclust=\u0026quot;average\u0026quot;, nboot=100, quiet=T)\n    \n    heatmap2.df[is.na(heatmap2.df)] \u0026lt;- 0\n    \n    pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.edgeR.list)[i],\u0026quot;_heatmap_degenes.pdf\u0026quot;),\n        width = 8,\n        height = 10)\n    heatmap.2(as.matrix(heatmap1.df),\n              col=hmcol1,\n              trace=\u0026quot;none\u0026quot;,\n              labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap1.df)),\n              labCol=groups[,2],\n              #Rowv = as.dendrogram(rowdendro),\n              Colv = as.dendrogram(coldendro),\n              ColSideColors=colcol,\n              lwid=c(1,4),\n              lhei = c(1,5),\n              breaks = seq(0,10,by=.5),\n              symkey=F,\n              dendrogram = \u0026quot;both\u0026quot;)\n    \n    heatmap.2(as.matrix(heatmap2.df),\n              col=hmcol2,\n              trace=\u0026quot;none\u0026quot;,\n              labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap2.df)),\n              labCol=groups[,2],\n              #Rowv = as.dendrogram(rowdendro),\n              Colv = as.dendrogram(coldendro),\n              ColSideColors=colcol,\n              lwid=c(1,4),\n              lhei = c(1,5),\n              breaks = seq(-3,3,by=.5),\n              symkey=F,\n              dendrogram = \u0026quot;both\u0026quot;)\n    dev.off()\n    \n    png(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.edgeR.list)[i],\u0026quot;_heatmap_keptgenes.png\u0026quot;),\n        width = 8,\n        height = 10,\n        units = \u0026quot;in\u0026quot;,res=300)\n    heatmap.2(as.matrix(heatmap1.df),\n              col=hmcol1,\n              trace=\u0026quot;none\u0026quot;,\n              labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap1.df)),\n              labCol=groups[,2],\n              #Rowv = as.dendrogram(rowdendro),\n              Colv = as.dendrogram(coldendro),\n              ColSideColors=colcol,\n              lwid=c(1,4),\n              lhei = c(1,5),\n              breaks = seq(0,10,by=.5),\n              symkey=F,\n              dendrogram = \u0026quot;both\u0026quot;)\n    \n    heatmap.2(as.matrix(heatmap2.df),\n              col=hmcol2,\n              trace=\u0026quot;none\u0026quot;,\n              labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap2.df)),\n              labCol=groups[,2],\n              #Rowv = as.dendrogram(rowdendro),\n              Colv = as.dendrogram(coldendro),\n              ColSideColors=colcol,\n              lwid=c(1,4),\n              lhei = c(1,5),\n              breaks = seq(-3,3,by=.5),\n              symkey=F,\n              dendrogram = \u0026quot;both\u0026quot;)\n    dev.off()\n    \n    heatmap.2(as.matrix(heatmap1.df),\n              col=hmcol1,\n              trace=\u0026quot;none\u0026quot;,\n              labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap1.df)),\n              labCol=groups[,2],\n             #Rowv = as.dendrogram(rowdendro),\n              Colv = as.dendrogram(coldendro),\n              ColSideColors=colcol,\n              lwid=c(1,4),\n              lhei = c(1,5),\n              breaks = seq(0,10,by=.5),\n              symkey=F,\n              dendrogram = \u0026quot;both\u0026quot;)\n    \n    heatmap.2(as.matrix(heatmap2.df),\n              col=hmcol2,\n              trace=\u0026quot;none\u0026quot;,\n              labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(heatmap2.df)),\n              labCol=groups[,2],\n              #Rowv = as.dendrogram(rowdendro),\n              Colv = as.dendrogram(coldendro),\n              ColSideColors=colcol,\n              lwid=c(1,4),\n              lhei = c(1,5),\n              breaks = seq(-3,3,by=.5),\n              symkey=F,\n              dendrogram = \u0026quot;both\u0026quot;)\n  }\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(tpm.edgeR.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.edgeR.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(tpm.edgeR.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n  \n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  heatmap1.df \u0026lt;- as.data.frame(log2(tpm.edgeR.list[[i]]+1))\n  heatmap2.df \u0026lt;- as.data.frame(t(scale(t(log2(tpm.edgeR.list[[i]] + 1)))))\n  \n  colcol \u0026lt;- as.character(groups[,3])\n  \n  if(nrow(tpm.edgeR.list[[i]]) \u0026gt; 0){\n    #rowdendro \u0026lt;- pvclust(t(heatmap1.df), method.dist=\"cor\", method.hclust=\"average\", nboot=100)\n    coldendro \u0026lt;- pvclust(heatmap2.df, method.dist=\"cor\", method.hclust=\"average\", nboot=100, quiet=T)\n    \n    heatmap2.df[is.na(heatmap2.df)] \u0026lt;- 0\n    \n    pdf(paste0(OUTPUT.DIR,\"/\",names(tpm.edgeR.list)[i],\"_heatmap_degenes.pdf\"),\n        width = 8,\n        height = 10)\n    heatmap.2(as.matrix(heatmap1.df),\n              col=hmcol1,\n              trace=\"none\",\n              labRow=vector(mode = \"character\", length = nrow(heatmap1.df)),\n              labCol=groups[,2],\n              #Rowv = as.dendrogram(rowdendro),\n              Colv = as.dendrogram(coldendro),\n              ColSideColors=colcol,\n              lwid=c(1,4),\n              lhei = c(1,5),\n              breaks = seq(0,10,by=.5),\n              symkey=F,\n              dendrogram = \"both\")\n    \n    heatmap.2(as.matrix(heatmap2.df),\n              col=hmcol2,\n              trace=\"none\",\n              labRow=vector(mode = \"character\", length = nrow(heatmap2.df)),\n              labCol=groups[,2],\n              #Rowv = as.dendrogram(rowdendro),\n              Colv = as.dendrogram(coldendro),\n              ColSideColors=colcol,\n              lwid=c(1,4),\n              lhei = c(1,5),\n              breaks = seq(-3,3,by=.5),\n              symkey=F,\n              dendrogram = \"both\")\n    dev.off()\n    \n    png(paste0(OUTPUT.DIR,\"/\",names(tpm.edgeR.list)[i],\"_heatmap_keptgenes.png\"),\n        width = 8,\n        height = 10,\n        units = \"in\",res=300)\n    heatmap.2(as.matrix(heatmap1.df),\n              col=hmcol1,\n              trace=\"none\",\n              labRow=vector(mode = \"character\", length = nrow(heatmap1.df)),\n              labCol=groups[,2],\n              #Rowv = as.dendrogram(rowdendro),\n              Colv = as.dendrogram(coldendro),\n              ColSideColors=colcol,\n              lwid=c(1,4),\n              lhei = c(1,5),\n              breaks = seq(0,10,by=.5),\n              symkey=F,\n              dendrogram = \"both\")\n    \n    heatmap.2(as.matrix(heatmap2.df),\n              col=hmcol2,\n              trace=\"none\",\n              labRow=vector(mode = \"character\", length = nrow(heatmap2.df)),\n              labCol=groups[,2],\n              #Rowv = as.dendrogram(rowdendro),\n              Colv = as.dendrogram(coldendro),\n              ColSideColors=colcol,\n              lwid=c(1,4),\n              lhei = c(1,5),\n              breaks = seq(-3,3,by=.5),\n              symkey=F,\n              dendrogram = \"both\")\n    dev.off()\n    \n    heatmap.2(as.matrix(heatmap1.df),\n              col=hmcol1,\n              trace=\"none\",\n              labRow=vector(mode = \"character\", length = nrow(heatmap1.df)),\n              labCol=groups[,2],\n             #Rowv = as.dendrogram(rowdendro),\n              Colv = as.dendrogram(coldendro),\n              ColSideColors=colcol,\n              lwid=c(1,4),\n              lhei = c(1,5),\n              breaks = seq(0,10,by=.5),\n              symkey=F,\n              dendrogram = \"both\")\n    \n    heatmap.2(as.matrix(heatmap2.df),\n              col=hmcol2,\n              trace=\"none\",\n              labRow=vector(mode = \"character\", length = nrow(heatmap2.df)),\n              labCol=groups[,2],\n              #Rowv = as.dendrogram(rowdendro),\n              Colv = as.dendrogram(coldendro),\n              ColSideColors=colcol,\n              lwid=c(1,4),\n              lhei = c(1,5),\n              breaks = seq(-3,3,by=.5),\n              symkey=F,\n              dendrogram = \"both\")\n  }\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Darby et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wmel-darby-et-al-2014-10\" class=\"anchor\" aria-label=\"Permalink: wMel, Darby et al 2014\" href=\"#wmel-darby-et-al-2014-10\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_heatmap_degenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_heatmap_degenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2015-10\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2015\" href=\"#wdi-luck-et-al-2015-10\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_heatmap_degenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_heatmap_degenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Gutzwiller et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wmel-gutzwiller-et-al-2015-10\" class=\"anchor\" aria-label=\"Permalink: wMel, Gutzwiller et al 2015\" href=\"#wmel-gutzwiller-et-al-2015-10\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_heatmap_degenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_heatmap_degenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Grote et al 2017\u003c/h5\u003e\u003ca id=\"user-content-wbm-grote-et-al-2017-10\" class=\"anchor\" aria-label=\"Permalink: wBm, Grote et al 2017\" href=\"#wbm-grote-et-al-2017-10\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_heatmap_degenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_heatmap_degenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Chung et al 2019\u003c/h5\u003e\u003ca id=\"user-content-wbm-chung-et-al-2019-10\" class=\"anchor\" aria-label=\"Permalink: wBm, Chung et al 2019\" href=\"#wbm-chung-et-al-2019-10\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_heatmap_degenes.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_heatmap_degenes.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eChoose studies for basic pairwise analyses or WGCNA \u003ca name=\"user-content-ranalysis_de_selectanalysis\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-choose-studies-for-basic-pairwise-analyses-or-wgcna-\" class=\"anchor\" aria-label=\"Permalink: Choose studies for basic pairwise analyses or WGCNA \" href=\"#choose-studies-for-basic-pairwise-analyses-or-wgcna-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pairwise.studies \u0026lt;- c(\u0026quot;wMel_darby_2014\u0026quot;)\nwgcna.studies \u0026lt;- c(\u0026quot;wDi_luck_2015\u0026quot;,\u0026quot;wMel_gutzwiller_2015\u0026quot;,\u0026quot;wBm_grote_2017\u0026quot;,\u0026quot;wBm_chung_2019\u0026quot;)\"\u003e\u003cpre lang=\"{R}\" class=\"notranslate\"\u003e\u003ccode\u003epairwise.studies \u0026lt;- c(\"wMel_darby_2014\")\nwgcna.studies \u0026lt;- c(\"wDi_luck_2015\",\"wMel_gutzwiller_2015\",\"wBm_grote_2017\",\"wBm_chung_2019\")\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConduct pairwise analyses on studies with two biological groups \u003ca name=\"user-content-ranalysis_de_pairwise\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-conduct-pairwise-analyses-on-studies-with-two-biological-groups-\" class=\"anchor\" aria-label=\"Permalink: Conduct pairwise analyses on studies with two biological groups \" href=\"#conduct-pairwise-analyses-on-studies-with-two-biological-groups-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCreate z-score of log2TPM heatmaps with a column bar that shows up- and down-regulated genes \u003ca name=\"user-content-ranalysis_de_pairwise_hm\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-create-z-score-of-log2tpm-heatmaps-with-a-column-bar-that-shows-up--and-down-regulated-genes-\" class=\"anchor\" aria-label=\"Permalink: Create z-score of log2TPM heatmaps with a column bar that shows up- and down-regulated genes \" href=\"#create-z-score-of-log2tpm-heatmaps-with-a-column-bar-that-shows-up--and-down-regulated-genes-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(pairwise.studies)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == pairwise.studies[i],]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n\n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  qlf \u0026lt;- as.data.frame(qlf.list[names(qlf.list) == pairwise.studies[i]])\n  qlf \u0026lt;- qlf[order(qlf[,1],decreasing = T),]\n  qlf \u0026lt;- qlf[rownames(qlf) %in% rownames(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]])),]\n  \n  log2tpm.de \u0026lt;- log2(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]])+1)\n  zscore.log2tpm.de \u0026lt;- as.data.frame(t(scale(t(log2tpm.de))))\n  zscore.log2tpm.de \u0026lt;- zscore.log2tpm.de[match(rownames(qlf),rownames(zscore.log2tpm.de)),]\n  \n  hmcol1 \u0026lt;- colorRampPalette(c(\u0026quot;navyblue\u0026quot;,\u0026quot;white\u0026quot;,\u0026quot;firebrick3\u0026quot;))(12)\n  hmcol2 \u0026lt;- colorRampPalette(c(\u0026quot;#FFFFFF\u0026quot;,\u0026quot;#FDE0D2\u0026quot;,\u0026quot;#FABAA1\u0026quot;,\u0026quot;#F69274\u0026quot;, \u0026quot;#F16B4E\u0026quot;,\u0026quot;#EF3D2D\u0026quot;,\u0026quot;#590A16\u0026quot;,\u0026quot;#A51E23\u0026quot;,\u0026quot;#650C16\u0026quot;))(30)\n  rowcol1 \u0026lt;- ifelse(qlf[,1] \u0026gt; 0,\u0026quot;red\u0026quot;,\u0026quot;blue\u0026quot;)\n  colcol \u0026lt;- as.character(groups[,3])\n  rowsep \u0026lt;- get_heatmap_separators(rowcol1)\n  colsep \u0026lt;- get_heatmap_separators(colcol)\n  \n  coldendro \u0026lt;- pvclust(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]]), method.dist=\u0026quot;cor\u0026quot;, method.hclust=\u0026quot;average\u0026quot;, nboot=100, quiet=T)\n\n  pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,pairwise.studies[i],\u0026quot;_pairwise_zscorelog2tpm_heatmap.pdf\u0026quot;),\n      width = 5,\n      height = 10)\n    \n  heatmap.2(as.matrix(zscore.log2tpm.de),\n              col=hmcol1,\n              trace=\u0026quot;none\u0026quot;,\n              labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(zscore.log2tpm.de)),\n              Rowv = F,\n              Colv = as.dendrogram(coldendro),\n              RowSideColors=rowcol1,\n              ColSideColors=colcol,\n              lhei = c(2,8),\n              breaks = seq(-3,3,by=.5),\n              rowsep = rowsep,\n              colsep = colsep,\n              symkey=F,\n              dendrogram = \u0026quot;column\u0026quot;)\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,pairwise.studies[i],\u0026quot;_pairwise_zscorelog2tpm_heatmap.png\u0026quot;),\n      width = 5,\n      height = 10,\n      units = \u0026quot;in\u0026quot;,res=300)\n    \n  heatmap.2(as.matrix(zscore.log2tpm.de),\n              col=hmcol1,\n              trace=\u0026quot;none\u0026quot;,\n              labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(zscore.log2tpm.de)),\n              Rowv = F,\n              Colv = as.dendrogram(coldendro),\n              RowSideColors=rowcol1,\n              ColSideColors=colcol,\n              lhei = c(2,8),\n              breaks = seq(-3,3,by=.5),\n              rowsep = rowsep,\n              colsep = colsep,\n              symkey=F,\n              dendrogram = \u0026quot;column\u0026quot;)\n  dev.off()\n  \n  heatmap.2(as.matrix(zscore.log2tpm.de),\n              col=hmcol1,\n              trace=\u0026quot;none\u0026quot;,\n              labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(zscore.log2tpm.de)),\n              Rowv = F,\n              Colv = as.dendrogram(coldendro),\n              RowSideColors=rowcol1,\n              ColSideColors=colcol,\n              lhei = c(2,8),\n              breaks = seq(-3,3,by=.5),\n              rowsep = rowsep,\n              colsep = colsep,\n              symkey=F,\n              dendrogram = \u0026quot;column\u0026quot;)\n  \n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(pairwise.studies)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == pairwise.studies[i],]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n\n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  qlf \u0026lt;- as.data.frame(qlf.list[names(qlf.list) == pairwise.studies[i]])\n  qlf \u0026lt;- qlf[order(qlf[,1],decreasing = T),]\n  qlf \u0026lt;- qlf[rownames(qlf) %in% rownames(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]])),]\n  \n  log2tpm.de \u0026lt;- log2(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]])+1)\n  zscore.log2tpm.de \u0026lt;- as.data.frame(t(scale(t(log2tpm.de))))\n  zscore.log2tpm.de \u0026lt;- zscore.log2tpm.de[match(rownames(qlf),rownames(zscore.log2tpm.de)),]\n  \n  hmcol1 \u0026lt;- colorRampPalette(c(\"navyblue\",\"white\",\"firebrick3\"))(12)\n  hmcol2 \u0026lt;- colorRampPalette(c(\"#FFFFFF\",\"#FDE0D2\",\"#FABAA1\",\"#F69274\", \"#F16B4E\",\"#EF3D2D\",\"#590A16\",\"#A51E23\",\"#650C16\"))(30)\n  rowcol1 \u0026lt;- ifelse(qlf[,1] \u0026gt; 0,\"red\",\"blue\")\n  colcol \u0026lt;- as.character(groups[,3])\n  rowsep \u0026lt;- get_heatmap_separators(rowcol1)\n  colsep \u0026lt;- get_heatmap_separators(colcol)\n  \n  coldendro \u0026lt;- pvclust(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]]), method.dist=\"cor\", method.hclust=\"average\", nboot=100, quiet=T)\n\n  pdf(paste0(OUTPUT.DIR,\"/\",pairwise.studies[i],\"_pairwise_zscorelog2tpm_heatmap.pdf\"),\n      width = 5,\n      height = 10)\n    \n  heatmap.2(as.matrix(zscore.log2tpm.de),\n              col=hmcol1,\n              trace=\"none\",\n              labRow=vector(mode = \"character\", length = nrow(zscore.log2tpm.de)),\n              Rowv = F,\n              Colv = as.dendrogram(coldendro),\n              RowSideColors=rowcol1,\n              ColSideColors=colcol,\n              lhei = c(2,8),\n              breaks = seq(-3,3,by=.5),\n              rowsep = rowsep,\n              colsep = colsep,\n              symkey=F,\n              dendrogram = \"column\")\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\"/\",pairwise.studies[i],\"_pairwise_zscorelog2tpm_heatmap.png\"),\n      width = 5,\n      height = 10,\n      units = \"in\",res=300)\n    \n  heatmap.2(as.matrix(zscore.log2tpm.de),\n              col=hmcol1,\n              trace=\"none\",\n              labRow=vector(mode = \"character\", length = nrow(zscore.log2tpm.de)),\n              Rowv = F,\n              Colv = as.dendrogram(coldendro),\n              RowSideColors=rowcol1,\n              ColSideColors=colcol,\n              lhei = c(2,8),\n              breaks = seq(-3,3,by=.5),\n              rowsep = rowsep,\n              colsep = colsep,\n              symkey=F,\n              dendrogram = \"column\")\n  dev.off()\n  \n  heatmap.2(as.matrix(zscore.log2tpm.de),\n              col=hmcol1,\n              trace=\"none\",\n              labRow=vector(mode = \"character\", length = nrow(zscore.log2tpm.de)),\n              Rowv = F,\n              Colv = as.dendrogram(coldendro),\n              RowSideColors=rowcol1,\n              ColSideColors=colcol,\n              lhei = c(2,8),\n              breaks = seq(-3,3,by=.5),\n              rowsep = rowsep,\n              colsep = colsep,\n              symkey=F,\n              dendrogram = \"column\")\n  \n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Darby et al 2014\u003c/h5\u003e\u003ca id=\"user-content-wmel-darby-et-al-2014-11\" class=\"anchor\" aria-label=\"Permalink: wMel, Darby et al 2014\" href=\"#wmel-darby-et-al-2014-11\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_pairwise_zscorelog2tpm_heatmap.pdf\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_pairwise_zscorelog2tpm_heatmap.pdf\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConduct functional term enrichment analysis on up- and down-regulated gene subsets \u003ca name=\"user-content-ranalysis_de_pairwise_fxnterm\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-conduct-functional-term-enrichment-analysis-on-up--and-down-regulated-gene-subsets-\" class=\"anchor\" aria-label=\"Permalink: Conduct functional term enrichment analysis on up- and down-regulated gene subsets \" href=\"#conduct-functional-term-enrichment-analysis-on-up--and-down-regulated-gene-subsets-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"terms.pairwise.list \u0026lt;- list()\nterms.sig.pairwise.list \u0026lt;- terms.pairwise.list\n\nfor(i in 1:length(pairwise.studies)){\n  strain \u0026lt;- substr(pairwise.studies,1,regexpr(\u0026quot;_\u0026quot;,pairwise.studies)-1)\n  geneinfo \u0026lt;- read.delim(paste0(GENEINFO.DIR,\u0026quot;/\u0026quot;,strain,\u0026quot;.cds.fna.interproscan.geneinfo.tsv\u0026quot;))\n  \n  qlf \u0026lt;- as.data.frame(qlf.list[names(qlf.list) == pairwise.studies[i]])\n  qlf \u0026lt;- qlf[order(qlf[,1],decreasing = T),]\n  qlf \u0026lt;- qlf[rownames(qlf) %in% rownames(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]])),]\n  \n  terms.colnames \u0026lt;- c(\u0026quot;term\u0026quot;,\u0026quot;clusteroccurences\u0026quot;,\u0026quot;genomeoccurences\u0026quot;,\u0026quot;pvalue\u0026quot;,\u0026quot;correctedpvalue\u0026quot;,\u0026quot;oddsratio\u0026quot;,\u0026quot;group\u0026quot;)\n  terms.pairwise \u0026lt;- as.data.frame(matrix(nrow = 0,\n                                         ncol = 7))\n  colnames(terms.pairwise) \u0026lt;- terms.colnames\n  \n  terms.pairwise.up \u0026lt;- as.data.frame(cbind(functionaltermenrichment(rownames(qlf)[qlf[,1] \u0026gt; 0],geneinfo),\n                                           \u0026quot;upregulated\u0026quot;))\n  terms.pairwise.down \u0026lt;- as.data.frame(cbind(functionaltermenrichment(rownames(qlf)[qlf[,1] \u0026lt; 0],geneinfo),\n                                       \u0026quot;downregulated\u0026quot;))\n  \n  colnames(terms.pairwise.up) \u0026lt;- terms.colnames\n  colnames(terms.pairwise.down) \u0026lt;- terms.colnames\n  terms.pairwise \u0026lt;- as.data.frame(rbind(terms.pairwise.up,\n                                     terms.pairwise.down))\n  \n  for(j in 2:6){\n    terms.pairwise[,j] \u0026lt;- as.numeric(as.character(terms.pairwise[,j]))\n  }\n  \n  terms.pairwise.list[[i]] \u0026lt;- terms.pairwise\n  terms.sig.pairwise.list[[i]] \u0026lt;- terms.pairwise[terms.pairwise$correctedpvalue \u0026lt; 0.05,]\n  \n  names(terms.pairwise.list)[i] \u0026lt;- pairwise.studies[i]\n  \n  write.table(terms.pairwise.list[[i]],\n            paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(terms.pairwise.list)[i],\u0026quot;_functionaltermenrichment.tsv\u0026quot;),\n            row.names = T,\n            col.names = T,\n            quote = F,\n            sep = \u0026quot;\\t\u0026quot;)\n}\"\u003e\u003cpre lang=\"{R}\" class=\"notranslate\"\u003e\u003ccode\u003eterms.pairwise.list \u0026lt;- list()\nterms.sig.pairwise.list \u0026lt;- terms.pairwise.list\n\nfor(i in 1:length(pairwise.studies)){\n  strain \u0026lt;- substr(pairwise.studies,1,regexpr(\"_\",pairwise.studies)-1)\n  geneinfo \u0026lt;- read.delim(paste0(GENEINFO.DIR,\"/\",strain,\".cds.fna.interproscan.geneinfo.tsv\"))\n  \n  qlf \u0026lt;- as.data.frame(qlf.list[names(qlf.list) == pairwise.studies[i]])\n  qlf \u0026lt;- qlf[order(qlf[,1],decreasing = T),]\n  qlf \u0026lt;- qlf[rownames(qlf) %in% rownames(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]])),]\n  \n  terms.colnames \u0026lt;- c(\"term\",\"clusteroccurences\",\"genomeoccurences\",\"pvalue\",\"correctedpvalue\",\"oddsratio\",\"group\")\n  terms.pairwise \u0026lt;- as.data.frame(matrix(nrow = 0,\n                                         ncol = 7))\n  colnames(terms.pairwise) \u0026lt;- terms.colnames\n  \n  terms.pairwise.up \u0026lt;- as.data.frame(cbind(functionaltermenrichment(rownames(qlf)[qlf[,1] \u0026gt; 0],geneinfo),\n                                           \"upregulated\"))\n  terms.pairwise.down \u0026lt;- as.data.frame(cbind(functionaltermenrichment(rownames(qlf)[qlf[,1] \u0026lt; 0],geneinfo),\n                                       \"downregulated\"))\n  \n  colnames(terms.pairwise.up) \u0026lt;- terms.colnames\n  colnames(terms.pairwise.down) \u0026lt;- terms.colnames\n  terms.pairwise \u0026lt;- as.data.frame(rbind(terms.pairwise.up,\n                                     terms.pairwise.down))\n  \n  for(j in 2:6){\n    terms.pairwise[,j] \u0026lt;- as.numeric(as.character(terms.pairwise[,j]))\n  }\n  \n  terms.pairwise.list[[i]] \u0026lt;- terms.pairwise\n  terms.sig.pairwise.list[[i]] \u0026lt;- terms.pairwise[terms.pairwise$correctedpvalue \u0026lt; 0.05,]\n  \n  names(terms.pairwise.list)[i] \u0026lt;- pairwise.studies[i]\n  \n  write.table(terms.pairwise.list[[i]],\n            paste0(OUTPUT.DIR,\"/\",names(terms.pairwise.list)[i],\"_functionaltermenrichment.tsv\"),\n            row.names = T,\n            col.names = T,\n            quote = F,\n            sep = \"\\t\")\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConduct WGCNA on time-course studies \u003ca name=\"user-content-ranalysis_de_wgcna\"\u003e\u003c/a\u003e\u003c/h3\u003e\u003ca id=\"user-content-conduct-wgcna-on-time-course-studies-\" class=\"anchor\" aria-label=\"Permalink: Conduct WGCNA on time-course studies \" href=\"#conduct-wgcna-on-time-course-studies-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eIdentify WGCNA soft power values \u003ca name=\"user-content-ranalysis_de_wgcna_softpower\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-identify-wgcna-soft-power-values-\" class=\"anchor\" aria-label=\"Permalink: Identify WGCNA soft power values \" href=\"#identify-wgcna-soft-power-values-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"tpm.wgcna.list \u0026lt;- tpm.edgeR.list\nfor(i in 1:length(tpm.wgcna.list)){\n  tpm.wgcna.list[[i]] \u0026lt;- as.data.frame(t(tpm.edgeR.list[[i]]))\n}\ntpm.wgcna.list \u0026lt;- tpm.wgcna.list[names(tpm.wgcna.list) %in% wgcna.studies]\nsoftpower.list \u0026lt;- tpm.wgcna.list\n\nfor(i in 1:length(tpm.wgcna.list)){\n  powers \u0026lt;- c(1:20)\n  sft \u0026lt;- pickSoftThreshold(tpm.wgcna.list[[i]], powerVector = powers, verbose = 5)\n  softpower.list[[i]] \u0026lt;- find_soft_power(sft)\n  \n  text.color \u0026lt;- rep(\u0026quot;black\u0026quot;,length(sft$fitIndices[,1]))\n  text.color[which(sft$fitIndices[,1] == softpower.list[[i]])] \u0026lt;- \u0026quot;red\u0026quot;\n  \n  scale_independence.plot \u0026lt;- ggplot()+\n    geom_text(mapping = aes(x = sft$fitIndices[,1], y = -sign(sft$fitIndices[,3])*sft$fitIndices[,2], label=sft$fitIndices[,1]), color = text.color)+\n    labs(title = \u0026quot;scale independence\u0026quot;, x = \u0026quot;soft threshold (power)\u0026quot;, y = \u0026quot;scale free topology model fit, signed R^2\u0026quot;)+\n    theme_bw()\n  \n  mean_connectivity.plot \u0026lt;- ggplot()+\n    geom_text(mapping = aes(x = sft$fitIndices[,1], y = sft$fitIndices[,5], label=sft$fitIndices[,1]), color = text.color)+\n    labs(title = \u0026quot;mean connectivity\u0026quot;, x = \u0026quot;soft threshold (power)\u0026quot;, y = \u0026quot;mean connectivity\u0026quot;)+\n    theme_bw()\n  \n  wgcna_soft_power_plots \u0026lt;- list(scale_independence.plot, mean_connectivity.plot)\n  lay \u0026lt;- rbind(c(1,2))\n  pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.wgcna.list)[i],\u0026quot;_wgcna_soft_power_plot.pdf\u0026quot;),\n      width = 8, \n      height = 4)\n  grid.arrange(grobs = wgcna_soft_power_plots,\n               widths = c(5,5),\n               heights = c(5),\n               layout_matrix = lay)\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.wgcna.list)[i],\u0026quot;_wgcna_soft_power_plot.png\u0026quot;),\n      width = 8, \n      height = 4,\n      units = \u0026quot;in\u0026quot;,res=300)\n  grid.arrange(grobs = wgcna_soft_power_plots,\n               widths = c(5,5),\n               heights = c(5),\n               layout_matrix = lay)\n  dev.off()\n  \n  grid.arrange(grobs = wgcna_soft_power_plots,\n               widths = c(5,5),\n               heights = c(5),\n               layout_matrix = lay)\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003etpm.wgcna.list \u0026lt;- tpm.edgeR.list\nfor(i in 1:length(tpm.wgcna.list)){\n  tpm.wgcna.list[[i]] \u0026lt;- as.data.frame(t(tpm.edgeR.list[[i]]))\n}\ntpm.wgcna.list \u0026lt;- tpm.wgcna.list[names(tpm.wgcna.list) %in% wgcna.studies]\nsoftpower.list \u0026lt;- tpm.wgcna.list\n\nfor(i in 1:length(tpm.wgcna.list)){\n  powers \u0026lt;- c(1:20)\n  sft \u0026lt;- pickSoftThreshold(tpm.wgcna.list[[i]], powerVector = powers, verbose = 5)\n  softpower.list[[i]] \u0026lt;- find_soft_power(sft)\n  \n  text.color \u0026lt;- rep(\"black\",length(sft$fitIndices[,1]))\n  text.color[which(sft$fitIndices[,1] == softpower.list[[i]])] \u0026lt;- \"red\"\n  \n  scale_independence.plot \u0026lt;- ggplot()+\n    geom_text(mapping = aes(x = sft$fitIndices[,1], y = -sign(sft$fitIndices[,3])*sft$fitIndices[,2], label=sft$fitIndices[,1]), color = text.color)+\n    labs(title = \"scale independence\", x = \"soft threshold (power)\", y = \"scale free topology model fit, signed R^2\")+\n    theme_bw()\n  \n  mean_connectivity.plot \u0026lt;- ggplot()+\n    geom_text(mapping = aes(x = sft$fitIndices[,1], y = sft$fitIndices[,5], label=sft$fitIndices[,1]), color = text.color)+\n    labs(title = \"mean connectivity\", x = \"soft threshold (power)\", y = \"mean connectivity\")+\n    theme_bw()\n  \n  wgcna_soft_power_plots \u0026lt;- list(scale_independence.plot, mean_connectivity.plot)\n  lay \u0026lt;- rbind(c(1,2))\n  pdf(paste0(OUTPUT.DIR,\"/\",names(tpm.wgcna.list)[i],\"_wgcna_soft_power_plot.pdf\"),\n      width = 8, \n      height = 4)\n  grid.arrange(grobs = wgcna_soft_power_plots,\n               widths = c(5,5),\n               heights = c(5),\n               layout_matrix = lay)\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\"/\",names(tpm.wgcna.list)[i],\"_wgcna_soft_power_plot.png\"),\n      width = 8, \n      height = 4,\n      units = \"in\",res=300)\n  grid.arrange(grobs = wgcna_soft_power_plots,\n               widths = c(5,5),\n               heights = c(5),\n               layout_matrix = lay)\n  dev.off()\n  \n  grid.arrange(grobs = wgcna_soft_power_plots,\n               widths = c(5,5),\n               heights = c(5),\n               layout_matrix = lay)\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2015-11\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2015\" href=\"#wdi-luck-et-al-2015-11\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_wgcna_soft_power_plot.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_wgcna_soft_power_plot.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Gutzwiller et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wmel-gutzwiller-et-al-2015-11\" class=\"anchor\" aria-label=\"Permalink: wMel, Gutzwiller et al 2015\" href=\"#wmel-gutzwiller-et-al-2015-11\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_wgcna_soft_power_plot.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_wgcna_soft_power_plot.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Grote et al 2017\u003c/h5\u003e\u003ca id=\"user-content-wbm-grote-et-al-2017-11\" class=\"anchor\" aria-label=\"Permalink: wBm, Grote et al 2017\" href=\"#wbm-grote-et-al-2017-11\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_wgcna_soft_power_plot.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_wgcna_soft_power_plot.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Chung et al 2019\u003c/h5\u003e\u003ca id=\"user-content-wbm-chung-et-al-2019-11\" class=\"anchor\" aria-label=\"Permalink: wBm, Chung et al 2019\" href=\"#wbm-chung-et-al-2019-11\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_wgcna_soft_power_plot.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_wgcna_soft_power_plot.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCreate WGCNA expression modules and merge similar modules \u003ca name=\"user-content-ranalysis_de_wgcna_createmodules\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-create-wgcna-expression-modules-and-merge-similar-modules-\" class=\"anchor\" aria-label=\"Permalink: Create WGCNA expression modules and merge similar modules \" href=\"#create-wgcna-expression-modules-and-merge-similar-modules-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(tpm.wgcna.list)){\n  adjacency \u0026lt;- adjacency(tpm.wgcna.list[[i]], power = softpower.list[[i]])\n  TOM \u0026lt;- TOMsimilarity(adjacency)\n  dissTOM \u0026lt;- 1-TOM\n  geneTree \u0026lt;- hclust(as.dist(dissTOM), method = \u0026quot;average\u0026quot;);\n  \n  minModuleSize \u0026lt;- 1\n  dynamicMods \u0026lt;- cutreeDynamic(dendro = geneTree, distM = dissTOM,\n                               deepSplit = 2, pamRespectsDendro = FALSE,\n                               minClusterSize = minModuleSize)\n  \n  dynamicColors = labels2colors(dynamicMods)\n  MEList = moduleEigengenes(tpm.wgcna.list[[i]], colors = dynamicColors)\n  MEs = MEList$eigengenes\n  MEDiss = 1-cor(MEs, use = \u0026quot;pairwise.complete.obs\u0026quot;)\n  METree = hclust(as.dist(MEDiss), method = \u0026quot;average\u0026quot;)\n  \n  MEDissThres = 0.25\n  \n  pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.wgcna.list)[i],\u0026quot;_wgcna_merge_eigengenes_plot.pdf\u0026quot;),\n      width = 8,\n      height = 5)\n  plot(METree, main = \u0026quot;clustering of module eigengenes\u0026quot;,\n       xlab = \u0026quot;\u0026quot;, sub = \u0026quot;\u0026quot;)\n  \n  abline(h=MEDissThres, col = \u0026quot;red\u0026quot;)\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.wgcna.list)[i],\u0026quot;_wgcna_merge_eigengenes_plot.png\u0026quot;),\n      width = 8,\n      height = 5,\n      units = \u0026quot;in\u0026quot;,res=300)\n  plot(METree, main = \u0026quot;clustering of module eigengenes\u0026quot;,\n       xlab = \u0026quot;\u0026quot;, sub = \u0026quot;\u0026quot;)\n  abline(h=MEDissThres, col = \u0026quot;red\u0026quot;)\n  dev.off()\n  \n  plot(METree, main = \u0026quot;clustering of module eigengenes\u0026quot;,\n       xlab = \u0026quot;\u0026quot;, sub = \u0026quot;\u0026quot;)\n  abline(h=MEDissThres, col = \u0026quot;red\u0026quot;)\n  \n  merge = mergeCloseModules(tpm.wgcna.list[[i]], dynamicColors, cutHeight = MEDissThres, verbose = 3)\n  mergedColors = merge$colors\n  mergedMEs = merge$newMEs\n  \n  pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.wgcna.list)[i],\u0026quot;_wgcna_eigengene_dendrogram_plot.pdf\u0026quot;),\n      width = 8,\n      height = 5)\n  plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors), c(\u0026quot;dynamic tree cut\u0026quot;, \u0026quot;merged dynamic\u0026quot;),\n                      dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05)\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.wgcna.list)[i],\u0026quot;_wgcna_eigengene_dendrogram_plot.png\u0026quot;),\n      width = 8,\n      height = 5,\n      units = \u0026quot;in\u0026quot;,res=300)\n  plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors), c(\u0026quot;dynamic tree cut\u0026quot;, \u0026quot;merged dynamic\u0026quot;),\n                      dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05)\n  dev.off()\n  \n  plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors), c(\u0026quot;dynamic tree cut\u0026quot;, \u0026quot;merged dynamic\u0026quot;),\n                      dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05)\n  \n  tpm.wgcna.list[[i]] \u0026lt;- eigengene_invert_id(as.data.frame(t(tpm.wgcna.list[[i]])), mergedColors, mergedMEs)\n  tpm.wgcna.list[[i]] \u0026lt;- wgcna_heatmap_reorder(tpm.wgcna.list[[i]])\n  \n  write.table(tpm.wgcna.list[[i]],\n              paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.wgcna.list)[i],\u0026quot;_tpm_de_modules.tsv\u0026quot;),\n              row.names = T,\n              col.names = T,\n              quote = F,\n              sep = \u0026quot;\\t\u0026quot;)\n}\"\u003e\u003cpre lang=\"{r\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(tpm.wgcna.list)){\n  adjacency \u0026lt;- adjacency(tpm.wgcna.list[[i]], power = softpower.list[[i]])\n  TOM \u0026lt;- TOMsimilarity(adjacency)\n  dissTOM \u0026lt;- 1-TOM\n  geneTree \u0026lt;- hclust(as.dist(dissTOM), method = \"average\");\n  \n  minModuleSize \u0026lt;- 1\n  dynamicMods \u0026lt;- cutreeDynamic(dendro = geneTree, distM = dissTOM,\n                               deepSplit = 2, pamRespectsDendro = FALSE,\n                               minClusterSize = minModuleSize)\n  \n  dynamicColors = labels2colors(dynamicMods)\n  MEList = moduleEigengenes(tpm.wgcna.list[[i]], colors = dynamicColors)\n  MEs = MEList$eigengenes\n  MEDiss = 1-cor(MEs, use = \"pairwise.complete.obs\")\n  METree = hclust(as.dist(MEDiss), method = \"average\")\n  \n  MEDissThres = 0.25\n  \n  pdf(paste0(OUTPUT.DIR,\"/\",names(tpm.wgcna.list)[i],\"_wgcna_merge_eigengenes_plot.pdf\"),\n      width = 8,\n      height = 5)\n  plot(METree, main = \"clustering of module eigengenes\",\n       xlab = \"\", sub = \"\")\n  \n  abline(h=MEDissThres, col = \"red\")\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\"/\",names(tpm.wgcna.list)[i],\"_wgcna_merge_eigengenes_plot.png\"),\n      width = 8,\n      height = 5,\n      units = \"in\",res=300)\n  plot(METree, main = \"clustering of module eigengenes\",\n       xlab = \"\", sub = \"\")\n  abline(h=MEDissThres, col = \"red\")\n  dev.off()\n  \n  plot(METree, main = \"clustering of module eigengenes\",\n       xlab = \"\", sub = \"\")\n  abline(h=MEDissThres, col = \"red\")\n  \n  merge = mergeCloseModules(tpm.wgcna.list[[i]], dynamicColors, cutHeight = MEDissThres, verbose = 3)\n  mergedColors = merge$colors\n  mergedMEs = merge$newMEs\n  \n  pdf(paste0(OUTPUT.DIR,\"/\",names(tpm.wgcna.list)[i],\"_wgcna_eigengene_dendrogram_plot.pdf\"),\n      width = 8,\n      height = 5)\n  plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors), c(\"dynamic tree cut\", \"merged dynamic\"),\n                      dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05)\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\"/\",names(tpm.wgcna.list)[i],\"_wgcna_eigengene_dendrogram_plot.png\"),\n      width = 8,\n      height = 5,\n      units = \"in\",res=300)\n  plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors), c(\"dynamic tree cut\", \"merged dynamic\"),\n                      dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05)\n  dev.off()\n  \n  plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors), c(\"dynamic tree cut\", \"merged dynamic\"),\n                      dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05)\n  \n  tpm.wgcna.list[[i]] \u0026lt;- eigengene_invert_id(as.data.frame(t(tpm.wgcna.list[[i]])), mergedColors, mergedMEs)\n  tpm.wgcna.list[[i]] \u0026lt;- wgcna_heatmap_reorder(tpm.wgcna.list[[i]])\n  \n  write.table(tpm.wgcna.list[[i]],\n              paste0(OUTPUT.DIR,\"/\",names(tpm.wgcna.list)[i],\"_tpm_de_modules.tsv\"),\n              row.names = T,\n              col.names = T,\n              quote = F,\n              sep = \"\\t\")\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2015-12\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2015\" href=\"#wdi-luck-et-al-2015-12\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_wgcna_eigengene_dendrogram_plot.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_wgcna_eigengene_dendrogram_plot.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_wgcna_merge_eigengenes_plot.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_wgcna_merge_eigengenes_plot.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Gutzwiller et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wmel-gutzwiller-et-al-2015-12\" class=\"anchor\" aria-label=\"Permalink: wMel, Gutzwiller et al 2015\" href=\"#wmel-gutzwiller-et-al-2015-12\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_wgcna_eigengene_dendrogram_plot.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_wgcna_eigengene_dendrogram_plot.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_wgcna_merge_eigengenes_plot.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_wgcna_merge_eigengenes_plot.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Grote et al 2017\u003c/h5\u003e\u003ca id=\"user-content-wbm-grote-et-al-2017-12\" class=\"anchor\" aria-label=\"Permalink: wBm, Grote et al 2017\" href=\"#wbm-grote-et-al-2017-12\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_wgcna_eigengene_dendrogram_plot.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_wgcna_eigengene_dendrogram_plot.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_wgcna_merge_eigengenes_plot.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_wgcna_merge_eigengenes_plot.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Chung et al 2019\u003c/h5\u003e\u003ca id=\"user-content-wbm-chung-et-al-2019-12\" class=\"anchor\" aria-label=\"Permalink: wBm, Chung et al 2019\" href=\"#wbm-chung-et-al-2019-12\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_wgcna_eigengene_dendrogram_plot.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_wgcna_eigengene_dendrogram_plot.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_wgcna_merge_eigengenes_plot.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_wgcna_merge_eigengenes_plot.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePlot WGCNA-derived expression modules \u003ca name=\"user-content-ranalysis_de_wgcna_heatmap\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-plot-wgcna-derived-expression-modules-\" class=\"anchor\" aria-label=\"Permalink: Plot WGCNA-derived expression modules \" href=\"#plot-wgcna-derived-expression-modules-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for(i in 1:length(tpm.wgcna.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.wgcna.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(tpm.wgcna.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n\n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  log2tpm.de \u0026lt;- log2(tpm.wgcna.list[[i]][,1:(ncol(tpm.wgcna.list[[i]]) - 2)] + 1)\n  zscore.log2tpm.de \u0026lt;- as.data.frame(t(scale(t(log2tpm.de))))\n  \n  hmcol1 \u0026lt;- colorRampPalette(c(\u0026quot;navyblue\u0026quot;,\u0026quot;white\u0026quot;,\u0026quot;firebrick3\u0026quot;))(12)\n  hmcol2 \u0026lt;- colorRampPalette(c(\u0026quot;#FFFFFF\u0026quot;,\u0026quot;#FDE0D2\u0026quot;,\u0026quot;#FABAA1\u0026quot;,\u0026quot;#F69274\u0026quot;, \u0026quot;#F16B4E\u0026quot;,\u0026quot;#EF3D2D\u0026quot;,\u0026quot;#590A16\u0026quot;,\u0026quot;#A51E23\u0026quot;,\u0026quot;#650C16\u0026quot;))(30)\n  rowcol1 \u0026lt;- tpm.wgcna.list[[i]]$module\n  colcol \u0026lt;- as.character(groups[,3])\n  rowsep \u0026lt;- get_heatmap_separators(rowcol1)\n  colsep \u0026lt;- get_heatmap_separators(colcol)\n  \n  rowcol2 \u0026lt;- tpm.wgcna.list[[i]]$invert\n  rowcol2[rowcol2 == F] \u0026lt;- \u0026quot;grey\u0026quot;\n  rowcol2[rowcol2 == T] \u0026lt;- \u0026quot;black\u0026quot;\n  \n  coldendro.df \u0026lt;- as.data.frame(tpm.kept.list[names(tpm.kept.list) == names(tpm.wgcna.list)[i]])\n  coldendro.df \u0026lt;- t(scale(t(log2(coldendro.df + 1))))\n  coldendro \u0026lt;- pvclust(coldendro.df, method.dist=\u0026quot;cor\u0026quot;, method.hclust=\u0026quot;average\u0026quot;, nboot=100, quiet=T)\n\n  if(names(tpm.wgcna.list)[i] == \u0026quot;wMel_gutzwiller_2015\u0026quot;){\n    coldendro \u0026lt;- rotate(as.dendrogram(coldendro), c(30:50,55:54,52:53,51,11:29,1:10))\n  }\n\n  pdf(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.wgcna.list)[i],\u0026quot;_wgcna_zscorelog2tpm_heatmap.pdf\u0026quot;),\n      width = 5,\n      height = 10)\n  heatmap.2(as.matrix(zscore.log2tpm.de),\n                col=hmcol1,\n                trace=\u0026quot;none\u0026quot;,\n                labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(zscore.log2tpm.de)),\n                Rowv = F,\n                Colv = as.dendrogram(coldendro),\n                RowSideColors=rowcol1,\n                ColSideColors=colcol,\n                lhei = c(2,8),\n                breaks = seq(-3,3,by=.5),\n                rowsep = rowsep,\n                colsep = colsep,\n                dendrogram = \u0026quot;col\u0026quot;)\n  heatmap.2(as.matrix(zscore.log2tpm.de),\n                col=hmcol1,\n                trace=\u0026quot;none\u0026quot;,\n                labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(zscore.log2tpm.de)),\n                Rowv = F,\n                Colv = as.dendrogram(coldendro),\n                RowSideColors=rowcol2,\n                ColSideColors=colcol,\n                lhei = c(2,8),\n                breaks = seq(-3,3,by=.5),\n                rowsep = rowsep,\n                colsep = colsep,\n                dendrogram = \u0026quot;col\u0026quot;)\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(tpm.wgcna.list)[i],\u0026quot;_wgcna_zscorelog2tpm_heatmap.png\u0026quot;),\n      width = 5,\n      height = 10,\n      units = \u0026quot;in\u0026quot;,res=300)\n  heatmap.2(as.matrix(zscore.log2tpm.de),\n                col=hmcol1,\n                trace=\u0026quot;none\u0026quot;,\n                labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(zscore.log2tpm.de)),\n                Rowv = F,\n                Colv = as.dendrogram(coldendro),\n                RowSideColors=rowcol1,\n                ColSideColors=colcol,\n                lhei = c(2,8),\n                breaks = seq(-3,3,by=.5),\n                rowsep = rowsep,\n                colsep = colsep,\n                dendrogram = \u0026quot;col\u0026quot;)\n  heatmap.2(as.matrix(zscore.log2tpm.de),\n                col=hmcol1,\n                trace=\u0026quot;none\u0026quot;,\n                labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(zscore.log2tpm.de)),\n                Rowv = F,\n                Colv = as.dendrogram(coldendro),\n                RowSideColors=rowcol2,\n                ColSideColors=colcol,\n                lhei = c(2,8),\n                breaks = seq(-3,3,by=.5),\n                rowsep = rowsep,\n                colsep = colsep,\n                dendrogram = \u0026quot;col\u0026quot;)\n  dev.off()\n  \n  heatmap.2(as.matrix(zscore.log2tpm.de),\n                col=hmcol1,\n                trace=\u0026quot;none\u0026quot;,\n                labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(zscore.log2tpm.de)),\n                Rowv = F,\n                Colv = as.dendrogram(coldendro),\n                RowSideColors=rowcol1,\n                ColSideColors=colcol,\n                lhei = c(2,8),\n                breaks = seq(-3,3,by=.5),\n                rowsep = rowsep,\n                colsep = colsep,\n                dendrogram = \u0026quot;col\u0026quot;)\n  heatmap.2(as.matrix(zscore.log2tpm.de),\n                col=hmcol1,\n                trace=\u0026quot;none\u0026quot;,\n                labRow=vector(mode = \u0026quot;character\u0026quot;, length = nrow(zscore.log2tpm.de)),\n                Rowv = F,\n                Colv = as.dendrogram(coldendro),\n                RowSideColors=rowcol2,\n                ColSideColors=colcol,\n                lhei = c(2,8),\n                breaks = seq(-3,3,by=.5),\n                rowsep = rowsep,\n                colsep = colsep,\n                dendrogram = \u0026quot;col\u0026quot;)\n}\"\u003e\u003cpre lang=\"{R,\" class=\"notranslate\"\u003e\u003ccode\u003efor(i in 1:length(tpm.wgcna.list)){\n  sample_map.subset \u0026lt;- sample_map[sample_map$study == names(tpm.wgcna.list)[i] \u0026amp; sample_map$sample_identifier %in% colnames(tpm.wgcna.list[[i]]),]\n  groups \u0026lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),\n                                as.character(sample_map.subset$sample_group),\n                                as.character(sample_map.subset$color)))\n  groups \u0026lt;- unique(groups)\n\n  groups[,1] \u0026lt;- factor(groups[,1],levels=groups[,1])\n  groups[,2] \u0026lt;- factor(groups[,2],levels=unique(groups[,2]))\n  groups[,3] \u0026lt;- factor(groups[,3],levels=unique(groups[,3]))\n  \n  log2tpm.de \u0026lt;- log2(tpm.wgcna.list[[i]][,1:(ncol(tpm.wgcna.list[[i]]) - 2)] + 1)\n  zscore.log2tpm.de \u0026lt;- as.data.frame(t(scale(t(log2tpm.de))))\n  \n  hmcol1 \u0026lt;- colorRampPalette(c(\"navyblue\",\"white\",\"firebrick3\"))(12)\n  hmcol2 \u0026lt;- colorRampPalette(c(\"#FFFFFF\",\"#FDE0D2\",\"#FABAA1\",\"#F69274\", \"#F16B4E\",\"#EF3D2D\",\"#590A16\",\"#A51E23\",\"#650C16\"))(30)\n  rowcol1 \u0026lt;- tpm.wgcna.list[[i]]$module\n  colcol \u0026lt;- as.character(groups[,3])\n  rowsep \u0026lt;- get_heatmap_separators(rowcol1)\n  colsep \u0026lt;- get_heatmap_separators(colcol)\n  \n  rowcol2 \u0026lt;- tpm.wgcna.list[[i]]$invert\n  rowcol2[rowcol2 == F] \u0026lt;- \"grey\"\n  rowcol2[rowcol2 == T] \u0026lt;- \"black\"\n  \n  coldendro.df \u0026lt;- as.data.frame(tpm.kept.list[names(tpm.kept.list) == names(tpm.wgcna.list)[i]])\n  coldendro.df \u0026lt;- t(scale(t(log2(coldendro.df + 1))))\n  coldendro \u0026lt;- pvclust(coldendro.df, method.dist=\"cor\", method.hclust=\"average\", nboot=100, quiet=T)\n\n  if(names(tpm.wgcna.list)[i] == \"wMel_gutzwiller_2015\"){\n    coldendro \u0026lt;- rotate(as.dendrogram(coldendro), c(30:50,55:54,52:53,51,11:29,1:10))\n  }\n\n  pdf(paste0(OUTPUT.DIR,\"/\",names(tpm.wgcna.list)[i],\"_wgcna_zscorelog2tpm_heatmap.pdf\"),\n      width = 5,\n      height = 10)\n  heatmap.2(as.matrix(zscore.log2tpm.de),\n                col=hmcol1,\n                trace=\"none\",\n                labRow=vector(mode = \"character\", length = nrow(zscore.log2tpm.de)),\n                Rowv = F,\n                Colv = as.dendrogram(coldendro),\n                RowSideColors=rowcol1,\n                ColSideColors=colcol,\n                lhei = c(2,8),\n                breaks = seq(-3,3,by=.5),\n                rowsep = rowsep,\n                colsep = colsep,\n                dendrogram = \"col\")\n  heatmap.2(as.matrix(zscore.log2tpm.de),\n                col=hmcol1,\n                trace=\"none\",\n                labRow=vector(mode = \"character\", length = nrow(zscore.log2tpm.de)),\n                Rowv = F,\n                Colv = as.dendrogram(coldendro),\n                RowSideColors=rowcol2,\n                ColSideColors=colcol,\n                lhei = c(2,8),\n                breaks = seq(-3,3,by=.5),\n                rowsep = rowsep,\n                colsep = colsep,\n                dendrogram = \"col\")\n  dev.off()\n  \n  png(paste0(OUTPUT.DIR,\"/\",names(tpm.wgcna.list)[i],\"_wgcna_zscorelog2tpm_heatmap.png\"),\n      width = 5,\n      height = 10,\n      units = \"in\",res=300)\n  heatmap.2(as.matrix(zscore.log2tpm.de),\n                col=hmcol1,\n                trace=\"none\",\n                labRow=vector(mode = \"character\", length = nrow(zscore.log2tpm.de)),\n                Rowv = F,\n                Colv = as.dendrogram(coldendro),\n                RowSideColors=rowcol1,\n                ColSideColors=colcol,\n                lhei = c(2,8),\n                breaks = seq(-3,3,by=.5),\n                rowsep = rowsep,\n                colsep = colsep,\n                dendrogram = \"col\")\n  heatmap.2(as.matrix(zscore.log2tpm.de),\n                col=hmcol1,\n                trace=\"none\",\n                labRow=vector(mode = \"character\", length = nrow(zscore.log2tpm.de)),\n                Rowv = F,\n                Colv = as.dendrogram(coldendro),\n                RowSideColors=rowcol2,\n                ColSideColors=colcol,\n                lhei = c(2,8),\n                breaks = seq(-3,3,by=.5),\n                rowsep = rowsep,\n                colsep = colsep,\n                dendrogram = \"col\")\n  dev.off()\n  \n  heatmap.2(as.matrix(zscore.log2tpm.de),\n                col=hmcol1,\n                trace=\"none\",\n                labRow=vector(mode = \"character\", length = nrow(zscore.log2tpm.de)),\n                Rowv = F,\n                Colv = as.dendrogram(coldendro),\n                RowSideColors=rowcol1,\n                ColSideColors=colcol,\n                lhei = c(2,8),\n                breaks = seq(-3,3,by=.5),\n                rowsep = rowsep,\n                colsep = colsep,\n                dendrogram = \"col\")\n  heatmap.2(as.matrix(zscore.log2tpm.de),\n                col=hmcol1,\n                trace=\"none\",\n                labRow=vector(mode = \"character\", length = nrow(zscore.log2tpm.de)),\n                Rowv = F,\n                Colv = as.dendrogram(coldendro),\n                RowSideColors=rowcol2,\n                ColSideColors=colcol,\n                lhei = c(2,8),\n                breaks = seq(-3,3,by=.5),\n                rowsep = rowsep,\n                colsep = colsep,\n                dendrogram = \"col\")\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewDi, Luck et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wdi-luck-et-al-2015-13\" class=\"anchor\" aria-label=\"Permalink: wDi, Luck et al 2015\" href=\"#wdi-luck-et-al-2015-13\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_wgcna_zscorelog2tpm_heatmap_bar1.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_wgcna_zscorelog2tpm_heatmap_bar1.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_wgcna_zscorelog2tpm_heatmap_bar2.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_wgcna_zscorelog2tpm_heatmap_bar2.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewMel, Gutzwiller et al 2015\u003c/h5\u003e\u003ca id=\"user-content-wmel-gutzwiller-et-al-2015-13\" class=\"anchor\" aria-label=\"Permalink: wMel, Gutzwiller et al 2015\" href=\"#wmel-gutzwiller-et-al-2015-13\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_wgcna_zscorelog2tpm_heatmap_bar1.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_wgcna_zscorelog2tpm_heatmap_bar1.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_wgcna_zscorelog2tpm_heatmap_bar2.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_wgcna_zscorelog2tpm_heatmap_bar2.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Grote et al 2017\u003c/h5\u003e\u003ca id=\"user-content-wbm-grote-et-al-2017-13\" class=\"anchor\" aria-label=\"Permalink: wBm, Grote et al 2017\" href=\"#wbm-grote-et-al-2017-13\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_wgcna_zscorelog2tpm_heatmap_bar1.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_wgcna_zscorelog2tpm_heatmap_bar1.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_wgcna_zscorelog2tpm_heatmap_bar2.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_wgcna_zscorelog2tpm_heatmap_bar2.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch5 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ewBm, Chung et al 2019\u003c/h5\u003e\u003ca id=\"user-content-wbm-chung-et-al-2019-13\" class=\"anchor\" aria-label=\"Permalink: wBm, Chung et al 2019\" href=\"#wbm-chung-et-al-2019-13\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_wgcna_zscorelog2tpm_heatmap_bar1.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_wgcna_zscorelog2tpm_heatmap_bar1.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_wgcna_zscorelog2tpm_heatmap_bar2.png\"\u003e\u003cimg src=\"/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_wgcna_zscorelog2tpm_heatmap_bar2.png\" alt=\"Image description\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eConduct functional term enrichment analysis on WGCNA expression modules \u003ca name=\"user-content-ranalysis_de_wgcna_fxnterm\"\u003e\u003c/a\u003e\u003c/h4\u003e\u003ca id=\"user-content-conduct-functional-term-enrichment-analysis-on-wgcna-expression-modules-\" class=\"anchor\" aria-label=\"Permalink: Conduct functional term enrichment analysis on WGCNA expression modules \" href=\"#conduct-functional-term-enrichment-analysis-on-wgcna-expression-modules-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"terms.wgcna.list \u0026lt;- tpm.wgcna.list\nterms.sig.wgcna.list \u0026lt;- tpm.wgcna.list\n\nfor(i in 1:length(tpm.wgcna.list)){\n  strain \u0026lt;- substr(names(tpm.wgcna.list)[i],1,regexpr(\u0026quot;_\u0026quot;,names(tpm.wgcna.list)[i])-1)\n  geneinfo \u0026lt;- read.delim(paste0(GENEINFO.DIR,\u0026quot;/\u0026quot;,strain,\u0026quot;.cds.fna.interproscan.geneinfo.tsv\u0026quot;))\n  \n  terms.colnames \u0026lt;- c(\u0026quot;term\u0026quot;,\u0026quot;clusteroccurences\u0026quot;,\u0026quot;genomeoccurences\u0026quot;,\u0026quot;pvalue\u0026quot;,\u0026quot;correctedpvalue\u0026quot;,\u0026quot;oddsratio\u0026quot;,\u0026quot;module\u0026quot;,\u0026quot;invert\u0026quot;)\n  terms.wgcna \u0026lt;- as.data.frame(matrix(nrow = 0,\n                                      ncol = 8))\n  colnames(terms.wgcna) \u0026lt;- terms.colnames\n  for(j in 1:length(unique(tpm.wgcna.list[[i]]$module))){\n    terms.wgcna.f \u0026lt;- as.data.frame(cbind(functionaltermenrichment(rownames(tpm.wgcna.list[[i]])[tpm.wgcna.list[[i]]$module == unique(tpm.wgcna.list[[i]]$module)[j] \u0026amp; tpm.wgcna.list[[i]]$invert == F],geneinfo),unique(tpm.wgcna.list[[i]]$module)[j],F))\n    terms.wgcna.t \u0026lt;- as.data.frame(cbind(functionaltermenrichment(rownames(tpm.wgcna.list[[i]])[tpm.wgcna.list[[i]]$module == unique(tpm.wgcna.list[[i]]$module)[j] \u0026amp; tpm.wgcna.list[[i]]$invert == T],geneinfo),unique(tpm.wgcna.list[[i]]$module)[j],T))\n    \n    colnames(terms.wgcna.f) \u0026lt;- terms.colnames\n    colnames(terms.wgcna.t) \u0026lt;- terms.colnames\n    terms.wgcna \u0026lt;- as.data.frame(rbind(terms.wgcna,\n                                       terms.wgcna.f, \n                                       terms.wgcna.t))\n  }\n  \n  for(j in 2:6){\n    terms.wgcna[,j] \u0026lt;- as.numeric(as.character(terms.wgcna[,j]))\n  }\n  \n  terms.wgcna.list[[i]] \u0026lt;- terms.wgcna\n  terms.sig.wgcna.list[[i]] \u0026lt;- terms.wgcna[terms.wgcna$correctedpvalue \u0026lt; 0.05,]\n  \n  write.table(terms.wgcna.list[[i]],\n            paste0(OUTPUT.DIR,\u0026quot;/\u0026quot;,names(terms.wgcna.list)[i],\u0026quot;_functionaltermenrichment.tsv\u0026quot;),\n            row.names = T,\n            col.names = T,\n            quote = F,\n            sep = \u0026quot;\\t\u0026quot;)\n}\"\u003e\u003cpre lang=\"{R}\" class=\"notranslate\"\u003e\u003ccode\u003eterms.wgcna.list \u0026lt;- tpm.wgcna.list\nterms.sig.wgcna.list \u0026lt;- tpm.wgcna.list\n\nfor(i in 1:length(tpm.wgcna.list)){\n  strain \u0026lt;- substr(names(tpm.wgcna.list)[i],1,regexpr(\"_\",names(tpm.wgcna.list)[i])-1)\n  geneinfo \u0026lt;- read.delim(paste0(GENEINFO.DIR,\"/\",strain,\".cds.fna.interproscan.geneinfo.tsv\"))\n  \n  terms.colnames \u0026lt;- c(\"term\",\"clusteroccurences\",\"genomeoccurences\",\"pvalue\",\"correctedpvalue\",\"oddsratio\",\"module\",\"invert\")\n  terms.wgcna \u0026lt;- as.data.frame(matrix(nrow = 0,\n                                      ncol = 8))\n  colnames(terms.wgcna) \u0026lt;- terms.colnames\n  for(j in 1:length(unique(tpm.wgcna.list[[i]]$module))){\n    terms.wgcna.f \u0026lt;- as.data.frame(cbind(functionaltermenrichment(rownames(tpm.wgcna.list[[i]])[tpm.wgcna.list[[i]]$module == unique(tpm.wgcna.list[[i]]$module)[j] \u0026amp; tpm.wgcna.list[[i]]$invert == F],geneinfo),unique(tpm.wgcna.list[[i]]$module)[j],F))\n    terms.wgcna.t \u0026lt;- as.data.frame(cbind(functionaltermenrichment(rownames(tpm.wgcna.list[[i]])[tpm.wgcna.list[[i]]$module == unique(tpm.wgcna.list[[i]]$module)[j] \u0026amp; tpm.wgcna.list[[i]]$invert == T],geneinfo),unique(tpm.wgcna.list[[i]]$module)[j],T))\n    \n    colnames(terms.wgcna.f) \u0026lt;- terms.colnames\n    colnames(terms.wgcna.t) \u0026lt;- terms.colnames\n    terms.wgcna \u0026lt;- as.data.frame(rbind(terms.wgcna,\n                                       terms.wgcna.f, \n                                       terms.wgcna.t))\n  }\n  \n  for(j in 2:6){\n    terms.wgcna[,j] \u0026lt;- as.numeric(as.character(terms.wgcna[,j]))\n  }\n  \n  terms.wgcna.list[[i]] \u0026lt;- terms.wgcna\n  terms.sig.wgcna.list[[i]] \u0026lt;- terms.wgcna[terms.wgcna$correctedpvalue \u0026lt; 0.05,]\n  \n  write.table(terms.wgcna.list[[i]],\n            paste0(OUTPUT.DIR,\"/\",names(terms.wgcna.list)[i],\"_functionaltermenrichment.tsv\"),\n            row.names = T,\n            col.names = T,\n            quote = F,\n            sep = \"\\t\")\n}\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDownload reference files (2019-12-13) \u003ca name=\"user-content-coreanalysis_getreferences\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-download-reference-files-2019-12-13-\" class=\"anchor\" aria-label=\"Permalink: Download reference files (2019-12-13) \" href=\"#download-reference-files-2019-12-13-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe faa reference for wDi must be downloaded from this link: \u003ca href=\"http://rast.nmpdr.org/rast.cgi?page=JobDetails\u0026amp;job=53659\" rel=\"nofollow\"\u003ehttp://rast.nmpdr.org/rast.cgi?page=JobDetails\u0026amp;job=53659\u003c/a\u003e (login:guest, password:guest). All subsequent code will make use of this wDi.faa at this path: \"$REFERENCES_DIR\"/wDi.faa\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-17\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-17\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## wBm\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/wBm.faa.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_translated_cds.faa.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/wBm.faa.gz\n\n## wMel\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/wMel.faa.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/025/GCF_000008025.1_ASM802v1/GCF_000008025.1_ASM802v1_translated_cds.faa.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/wMel.faa.gz\n\n## wOo\nwget -O \u0026quot;$REFERENCES_DIR\u0026quot;/wOo.faa.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/306/885/GCF_000306885.1_ASM30688v1/GCF_000306885.1_ASM30688v1_translated_cds.faa.gz\ngunzip \u0026quot;$REFERENCES_DIR\u0026quot;/wOo.faa.gz\n\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003e## wBm\nwget -O \"$REFERENCES_DIR\"/wBm.faa.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_translated_cds.faa.gz\ngunzip \"$REFERENCES_DIR\"/wBm.faa.gz\n\n## wMel\nwget -O \"$REFERENCES_DIR\"/wMel.faa.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/025/GCF_000008025.1_ASM802v1/GCF_000008025.1_ASM802v1_translated_cds.faa.gz\ngunzip \"$REFERENCES_DIR\"/wMel.faa.gz\n\n## wOo\nwget -O \"$REFERENCES_DIR\"/wOo.faa.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/306/885/GCF_000306885.1_ASM30688v1/GCF_000306885.1_ASM30688v1_translated_cds.faa.gz\ngunzip \"$REFERENCES_DIR\"/wOo.faa.gz\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCreate BLAST database with the amino acid sequences of the four Wolbachia genomes (2019-12-13) \u003ca name=\"user-content-coreanalysis_createblastdb\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-create-blast-database-with-the-amino-acid-sequences-of-the-four-wolbachia-genomes-2019-12-13-\" class=\"anchor\" aria-label=\"Permalink: Create BLAST database with the amino acid sequences of the four Wolbachia genomes (2019-12-13) \" href=\"#create-blast-database-with-the-amino-acid-sequences-of-the-four-wolbachia-genomes-2019-12-13-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInputs:\u003c/h4\u003e\u003ca id=\"user-content-inputs\" class=\"anchor\" aria-label=\"Permalink: Inputs:\" href=\"#inputs\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"DB_FAA=\u0026quot;$REFERENCES_DIR\u0026quot;/combined_wolbachia_reference.faa\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eDB_FAA=\"$REFERENCES_DIR\"/combined_wolbachia_reference.faa\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-18\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-18\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cat \u0026quot;$REFERENCES_DIR\u0026quot;/wBm.faa \u0026quot;$REFERENCES_DIR\u0026quot;/wDi.faa \u0026quot;$REFERENCES_DIR\u0026quot;/wMel.faa \u0026quot;$REFERENCES_DIR\u0026quot;/wOo.faa \u0026gt; \u0026quot;$DB_FAA\u0026quot;\n$NCBIBLAST_BIN_DIR/formatdb -t \u0026quot;$(echo \u0026quot;$DB_FAA\u0026quot; | sed \u0026quot;s/[.]faa$//g\u0026quot;)\u0026quot; -i \u0026quot;$DB_FAA\u0026quot;\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003ecat \"$REFERENCES_DIR\"/wBm.faa \"$REFERENCES_DIR\"/wDi.faa \"$REFERENCES_DIR\"/wMel.faa \"$REFERENCES_DIR\"/wOo.faa \u0026gt; \"$DB_FAA\"\n$NCBIBLAST_BIN_DIR/formatdb -t \"$(echo \"$DB_FAA\" | sed \"s/[.]faa$//g\")\" -i \"$DB_FAA\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRun self-BLASTP search against the protein sequences of the four Wolbachia strains (2019-12-13) \u003ca name=\"user-content-coreanalysis_createblastdb\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-run-self-blastp-search-against-the-protein-sequences-of-the-four-wolbachia-strains-2019-12-13-\" class=\"anchor\" aria-label=\"Permalink: Run self-BLASTP search against the protein sequences of the four Wolbachia strains (2019-12-13) \" href=\"#run-self-blastp-search-against-the-protein-sequences-of-the-four-wolbachia-strains-2019-12-13-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInputs:\u003c/h4\u003e\u003ca id=\"user-content-inputs-1\" class=\"anchor\" aria-label=\"Permalink: Inputs:\" href=\"#inputs-1\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"BLAST_DB=\u0026quot;$REFERENCES_DIR\u0026quot;/combined_wolbachia_reference.faa\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eBLAST_DB=\"$REFERENCES_DIR\"/combined_wolbachia_reference.faa\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCommands:\u003c/h4\u003e\u003ca id=\"user-content-commands-19\" class=\"anchor\" aria-label=\"Permalink: Commands:\" href=\"#commands-19\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"qsub -P jdhotopp-lab -l mem_free=5G -N blastp -wd \u0026quot;$(dirname \u0026quot;$BLAST_DB\u0026quot;)\u0026quot; -b y $NCBIBLAST_BIN_DIR/blastall -p blastp -i \u0026quot;$BLAST_DB\u0026quot; -d \u0026quot;$BLAST_DB\u0026quot; -e 1e-5 -m 8 -o \u0026quot;$(dirname \u0026quot;$BLAST_DB\u0026quot;)\u0026quot;/blastp_raw.out\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eqsub -P jdhotopp-lab -l mem_free=5G -N blastp -wd \"$(dirname \"$BLAST_DB\")\" -b y $NCBIBLAST_BIN_DIR/blastall -p blastp -i \"$BLAST_DB\" -d \"$BLAST_DB\" -e 1e-5 -m 8 -o \"$(dirname \"$BLAST_DB\")\"/blastp_raw.out\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRun PanOCT to identify core Wolbachia genes (2019-12-13) \u003ca name=\"user-content-coreanalysis_createblastdb\"\u003e\u003c/a\u003e\u003c/h2\u003e\u003ca id=\"user-content-run-panoct-to-identify-core-wolbachia-genes-2019-12-13-\" class=\"anchor\" aria-label=\"Permalink: Run PanOCT to identify core Wolbachia genes (2019-12-13) \" href=\"#run-panoct-to-identify-core-wolbachia-genes-2019-12-13-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInputs:\u003c/h4\u003e\u003ca id=\"user-content-inputs-2\" class=\"anchor\" aria-label=\"Permalink: Inputs:\" href=\"#inputs-2\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"FAA=\u0026quot;$REFERENCES_DIR\u0026quot;/combined_wolbachia_reference.faa\nBLAST_OUTPUT=\u0026quot;$(dirname \u0026quot;$BLAST_DB\u0026quot;)\u0026quot;/blastp_raw.out\nOUTPUT_DIR=\u0026quot;$WORKING_DIR\u0026quot;/panoct\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003eFAA=\"$REFERENCES_DIR\"/combined_wolbachia_reference.faa\nBLAST_OUTPUT=\"$(dirname \"$BLAST_DB\")\"/blastp_raw.out\nOUTPUT_DIR=\"$WORKING_DIR\"/panoct\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch4 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInputs:\u003c/h4\u003e\u003ca id=\"user-content-inputs-3\" class=\"anchor\" aria-label=\"Permalink: Inputs:\" href=\"#inputs-3\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"mkdir -p \u0026quot;$OUTPUT_DIR\u0026quot;\n\n\"\u003e\u003cpre lang=\"{bash,\" class=\"notranslate\"\u003e\u003ccode\u003emkdir -p \"$OUTPUT_DIR\"\n\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"A meta-analysis of Wolbachia transcriptomics reveals a stage-specific Wolbachia transcriptional response shared across different hosts","anchor":"a-meta-analysis-of-wolbachia-transcriptomics-reveals-a-stage-specific-wolbachia-transcriptional-response-shared-across-different-hosts","htmlText":"A meta-analysis of Wolbachia transcriptomics reveals a stage-specific Wolbachia transcriptional response shared across different hosts"},{"level":1,"text":"Table of Contents","anchor":"table-of-contents","htmlText":"Table of Contents"},{"level":1,"text":"Set software and directory paths ","anchor":"set-software-and-directory-paths-","htmlText":"Set software and directory paths "},{"level":2,"text":"Software ","anchor":"software-","htmlText":"Software "},{"level":2,"text":"Directories ","anchor":"directories-","htmlText":"Directories "},{"level":1,"text":"Download, align, and quantify Wolbachia sequencing data ","anchor":"download-align-and-quantify-wolbachia-sequencing-data-","htmlText":"Download, align, and quantify Wolbachia sequencing data "},{"level":2,"text":"Create directories (2019-05-30) ","anchor":"create-directories-2019-05-30-","htmlText":"Create directories (2019-05-30) "},{"level":2,"text":"Download reference files and create combined references (2019-05-30) ","anchor":"download-reference-files-and-create-combined-references-2019-05-30-","htmlText":"Download reference files and create combined references (2019-05-30) "},{"level":4,"text":"Commands:","anchor":"commands","htmlText":"Commands:"},{"level":2,"text":"Download FASTQ files (2019-05-30) ","anchor":"download-fastq-files-2019-05-30-","htmlText":"Download FASTQ files (2019-05-30) "},{"level":3,"text":"SRA Datasets ","anchor":"sra-datasets-","htmlText":"SRA Datasets "},{"level":4,"text":"Input Sets:","anchor":"input-sets","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-1","htmlText":"Commands:"},{"level":3,"text":"Luck et al 2015 Dataset ","anchor":"luck-et-al-2015-dataset-","htmlText":"Luck et al 2015 Dataset "},{"level":4,"text":"Input Sets:","anchor":"input-sets-1","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-2","htmlText":"Commands:"},{"level":2,"text":"Align FASTQ files to combined host and Wolbachia reference while allowing for splicing (2019-05-30)  ","anchor":"align-fastq-files-to-combined-host-and-wolbachia-reference-while-allowing-for-splicing-2019-05-30--","htmlText":"Align FASTQ files to combined host and Wolbachia reference while allowing for splicing (2019-05-30)  "},{"level":3,"text":"Single-End FASTQs ","anchor":"single-end-fastqs-","htmlText":"Single-End FASTQs "},{"level":4,"text":"Input Sets:","anchor":"input-sets-2","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-3","htmlText":"Commands:"},{"level":3,"text":"Paired-End FASTQs ","anchor":"paired-end-fastqs-","htmlText":"Paired-End FASTQs "},{"level":4,"text":"Input Sets:","anchor":"input-sets-3","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-4","htmlText":"Commands:"},{"level":2,"text":"Create a list of Wolbachia-mapping reads in all BAM files (2019-05-30) ","anchor":"create-a-list-of-wolbachia-mapping-reads-in-all-bam-files-2019-05-30-","htmlText":"Create a list of Wolbachia-mapping reads in all BAM files (2019-05-30) "},{"level":4,"text":"Input Sets:","anchor":"input-sets-4","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-5","htmlText":"Commands:"},{"level":2,"text":"Create subsets of all FASTQ files that contain only Wolbachia-mapping reads (2019-05-30) ","anchor":"create-subsets-of-all-fastq-files-that-contain-only-wolbachia-mapping-reads-2019-05-30-","htmlText":"Create subsets of all FASTQ files that contain only Wolbachia-mapping reads (2019-05-30) "},{"level":3,"text":"Single-End FASTQs ","anchor":"single-end-fastqs--1","htmlText":"Single-End FASTQs "},{"level":4,"text":"Input Sets:","anchor":"input-sets-5","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-6","htmlText":"Commands:"},{"level":3,"text":"Paired-End FASTQs ","anchor":"paired-end-fastqs--1","htmlText":"Paired-End FASTQs "},{"level":4,"text":"Input Sets:","anchor":"input-sets-6","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-7","htmlText":"Commands:"},{"level":2,"text":"Align FASTQ files containing only Wolbachia-mapping reads to combined host and Wolbachia reference while disallowing splicing (2019-06-01) ","anchor":"align-fastq-files-containing-only-wolbachia-mapping-reads-to-combined-host-and-wolbachia-reference-while-disallowing-splicing-2019-06-01-","htmlText":"Align FASTQ files containing only Wolbachia-mapping reads to combined host and Wolbachia reference while disallowing splicing (2019-06-01) "},{"level":3,"text":"Single-End FASTQs ","anchor":"single-end-fastqs--2","htmlText":"Single-End FASTQs "},{"level":4,"text":"Input Sets:","anchor":"input-sets-7","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-8","htmlText":"Commands:"},{"level":3,"text":"Paired-End FASTQs ","anchor":"paired-end-fastqs--2","htmlText":"Paired-End FASTQs "},{"level":4,"text":"Input Sets:","anchor":"input-sets-8","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-9","htmlText":"Commands:"},{"level":2,"text":"Sort BAM files (2019-06-01) ","anchor":"sort-bam-files-2019-06-01-","htmlText":"Sort BAM files (2019-06-01) "},{"level":4,"text":"Input Sets:","anchor":"input-sets-9","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-10","htmlText":"Commands:"},{"level":2,"text":"Index BAM files (2019-06-01) ","anchor":"index-bam-files-2019-06-01-","htmlText":"Index BAM files (2019-06-01) "},{"level":4,"text":"Input Sets:","anchor":"input-sets-10","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-11","htmlText":"Commands:"},{"level":2,"text":"Quantify Wolbachia genes (2019-06-01) ","anchor":"quantify-wolbachia-genes-2019-06-01-","htmlText":"Quantify Wolbachia genes (2019-06-01) "},{"level":4,"text":"Input Sets:","anchor":"input-sets-11","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-12","htmlText":"Commands:"},{"level":2,"text":"Convert GFF to a mapping table (2019-06-07) ","anchor":"convert-gff-to-a-mapping-table-2019-06-07-","htmlText":"Convert GFF to a mapping table (2019-06-07) "},{"level":4,"text":"Input Sets:","anchor":"input-sets-12","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-13","htmlText":"Commands:"},{"level":2,"text":"Create nucleotide coding sequence fasta files (2019-06-07) ","anchor":"create-nucleotide-coding-sequence-fasta-files-2019-06-07-","htmlText":"Create nucleotide coding sequence fasta files (2019-06-07) "},{"level":4,"text":"Input Sets:","anchor":"input-sets-13","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-14","htmlText":"Commands:"},{"level":2,"text":"Identify InterPro descriptions and GO terms for Wolbachia transcripts (2019-06-07) ","anchor":"identify-interpro-descriptions-and-go-terms-for-wolbachia-transcripts-2019-06-07-","htmlText":"Identify InterPro descriptions and GO terms for Wolbachia transcripts (2019-06-07) "},{"level":4,"text":"Input Sets:","anchor":"input-sets-14","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-15","htmlText":"Commands:"},{"level":1,"text":"Transcriptomics meta-analysis ","anchor":"transcriptomics-meta-analysis-","htmlText":"Transcriptomics meta-analysis "},{"level":2,"text":"Create counts table ","anchor":"create-counts-table-","htmlText":"Create counts table "},{"level":3,"text":"Set R inputs ","anchor":"set-r-inputs-","htmlText":"Set R inputs "},{"level":3,"text":"View sessionInfo ","anchor":"view-sessioninfo-","htmlText":"View sessionInfo "},{"level":3,"text":"Combine count files into a single table for each study ","anchor":"combine-count-files-into-a-single-table-for-each-study-","htmlText":"Combine count files into a single table for each study "},{"level":2,"text":"Convert InterProScan output to geneinfo file (2019-06-07) ","anchor":"convert-interproscan-output-to-geneinfo-file-2019-06-07-","htmlText":"Convert InterProScan output to geneinfo file (2019-06-07) "},{"level":4,"text":"Input Sets:","anchor":"input-sets-15","htmlText":"Input Sets:"},{"level":4,"text":"Commands:","anchor":"commands-16","htmlText":"Commands:"},{"level":2,"text":"Differential expression and WGCNA analyses (2019-06-07) ","anchor":"differential-expression-and-wgcna-analyses-2019-06-07-","htmlText":"Differential expression and WGCNA analyses (2019-06-07) "},{"level":3,"text":"Set R inputs ","anchor":"set-r-inputs--1","htmlText":"Set R inputs "},{"level":3,"text":"Load R packages and view sessionInfo ","anchor":"load-r-packages-and-view-sessioninfo-","htmlText":"Load R packages and view sessionInfo "},{"level":3,"text":"Load R functions ","anchor":"load-r-functions-","htmlText":"Load R functions "},{"level":3,"text":"Process counts files and calculate TPM values  ","anchor":"process-counts-files-and-calculate-tpm-values--","htmlText":"Process counts files and calculate TPM values  "},{"level":4,"text":"Read counts table into a list ","anchor":"read-counts-table-into-a-list-","htmlText":"Read counts table into a list "},{"level":4,"text":"Exclude non-protein-coding genes from counts table ","anchor":"exclude-non-protein-coding-genes-from-counts-table-","htmlText":"Exclude non-protein-coding genes from counts table "},{"level":4,"text":"Exclude genes without Agilent SureSelect probes from counts table in Chung et al 2019 study ","anchor":"exclude-genes-without-agilent-sureselect-probes-from-counts-table-in-chung-et-al-2019-study-","htmlText":"Exclude genes without Agilent SureSelect probes from counts table in Chung et al 2019 study "},{"level":4,"text":"Print the number of reads mapping to protein-coding genes for each sample in each study ","anchor":"print-the-number-of-reads-mapping-to-protein-coding-genes-for-each-sample-in-each-study-","htmlText":"Print the number of reads mapping to protein-coding genes for each sample in each study "},{"level":4,"text":"Create TPM table ","anchor":"create-tpm-table-","htmlText":"Create TPM table "},{"level":3,"text":"Create keys ","anchor":"create-keys-","htmlText":"Create keys "},{"level":4,"text":"Sample keys ","anchor":"sample-keys-","htmlText":"Sample keys "},{"level":5,"text":"wOo, Darby et al 2012","anchor":"woo-darby-et-al-2012","htmlText":"wOo, Darby et al 2012"},{"level":5,"text":"wMel, Darby et al 2014","anchor":"wmel-darby-et-al-2014","htmlText":"wMel, Darby et al 2014"},{"level":5,"text":"wDi, Luck et al 2014","anchor":"wdi-luck-et-al-2014","htmlText":"wDi, Luck et al 2014"},{"level":5,"text":"wDi, Luck et al 2015","anchor":"wdi-luck-et-al-2015","htmlText":"wDi, Luck et al 2015"},{"level":5,"text":"wMel, Gutzwiller et al 2015","anchor":"wmel-gutzwiller-et-al-2015","htmlText":"wMel, Gutzwiller et al 2015"},{"level":5,"text":"wBm, Grote et al 2017","anchor":"wbm-grote-et-al-2017","htmlText":"wBm, Grote et al 2017"},{"level":5,"text":"wBm, Chung et al 2019","anchor":"wbm-chung-et-al-2019","htmlText":"wBm, Chung et al 2019"},{"level":4,"text":"Size key  ","anchor":"size-key--","htmlText":"Size key  "},{"level":4,"text":"Heatmap log2TPM key ","anchor":"heatmap-log2tpm-key-","htmlText":"Heatmap log2TPM key "},{"level":4,"text":"Heatmap z-score log2TPM key ","anchor":"heatmap-z-score-log2tpm-key-","htmlText":"Heatmap z-score log2TPM key "},{"level":3,"text":"Assess if samples have been sequenced to saturation using saturation curves ","anchor":"assess-if-samples-have-been-sequenced-to-saturation-using-saturation-curves-","htmlText":"Assess if samples have been sequenced to saturation using saturation curves "},{"level":5,"text":"wOo, Darby et al 2012","anchor":"woo-darby-et-al-2012-1","htmlText":"wOo, Darby et al 2012"},{"level":5,"text":"wMel, Darby et al 2014","anchor":"wmel-darby-et-al-2014-1","htmlText":"wMel, Darby et al 2014"},{"level":5,"text":"wDi, Luck et al 2014","anchor":"wdi-luck-et-al-2014-1","htmlText":"wDi, Luck et al 2014"},{"level":5,"text":"wDi, Luck et al 2015","anchor":"wdi-luck-et-al-2015-1","htmlText":"wDi, Luck et al 2015"},{"level":5,"text":"wMel, Gutzwiller et al 2015","anchor":"wmel-gutzwiller-et-al-2015-1","htmlText":"wMel, Gutzwiller et al 2015"},{"level":5,"text":"wBm, Grote et al 2017","anchor":"wbm-grote-et-al-2017-1","htmlText":"wBm, Grote et al 2017"},{"level":5,"text":"wBm, Chung et al 2019","anchor":"wbm-chung-et-al-2019-1","htmlText":"wBm, Chung et al 2019"},{"level":3,"text":"Exclude samples and/or studies from differential expression analyses due to inadequate sequencing depth ","anchor":"exclude-samples-andor-studies-from-differential-expression-analyses-due-to-inadequate-sequencing-depth-","htmlText":"Exclude samples and/or studies from differential expression analyses due to inadequate sequencing depth "},{"level":3,"text":"Identify the number of genes in each study that meets the minimum CPM threshold ","anchor":"identify-the-number-of-genes-in-each-study-that-meets-the-minimum-cpm-threshold-","htmlText":"Identify the number of genes in each study that meets the minimum CPM threshold "},{"level":3,"text":"Identify differentially expressed genes (FDR \u003c 0.05) in each study ","anchor":"identify-differentially-expressed-genes-fdr--005-in-each-study-","htmlText":"Identify differentially expressed genes (FDR \u0026lt; 0.05) in each study "},{"level":3,"text":"Conduct principal component analyses ","anchor":"conduct-principal-component-analyses-","htmlText":"Conduct principal component analyses "},{"level":4,"text":"All genes ","anchor":"all-genes-","htmlText":"All genes "},{"level":5,"text":"wOo, Darby et al 2012","anchor":"woo-darby-et-al-2012-2","htmlText":"wOo, Darby et al 2012"},{"level":5,"text":"wMel, Darby et al 2014","anchor":"wmel-darby-et-al-2014-2","htmlText":"wMel, Darby et al 2014"},{"level":5,"text":"wDi, Luck et al 2014","anchor":"wdi-luck-et-al-2014-2","htmlText":"wDi, Luck et al 2014"},{"level":5,"text":"wDi, Luck et al 2015","anchor":"wdi-luck-et-al-2015-2","htmlText":"wDi, Luck et al 2015"},{"level":5,"text":"wMel, Gutzwiller et al 2015","anchor":"wmel-gutzwiller-et-al-2015-2","htmlText":"wMel, Gutzwiller et al 2015"},{"level":5,"text":"wBm, Grote et al 2017","anchor":"wbm-grote-et-al-2017-2","htmlText":"wBm, Grote et al 2017"},{"level":5,"text":"wBm, Chung et al 2019","anchor":"wbm-chung-et-al-2019-2","htmlText":"wBm, Chung et al 2019"},{"level":4,"text":"Kept genes ","anchor":"kept-genes-","htmlText":"Kept genes "},{"level":5,"text":"wOo, Darby et al 2012","anchor":"woo-darby-et-al-2012-3","htmlText":"wOo, Darby et al 2012"},{"level":5,"text":"wMel, Darby et al 2014","anchor":"wmel-darby-et-al-2014-3","htmlText":"wMel, Darby et al 2014"},{"level":5,"text":"wDi, Luck et al 2014","anchor":"wdi-luck-et-al-2014-3","htmlText":"wDi, Luck et al 2014"},{"level":5,"text":"wDi, Luck et al 2015","anchor":"wdi-luck-et-al-2015-3","htmlText":"wDi, Luck et al 2015"},{"level":5,"text":"wMel, Gutzwiller et al 2015","anchor":"wmel-gutzwiller-et-al-2015-3","htmlText":"wMel, Gutzwiller et al 2015"},{"level":5,"text":"wBm, Grote et al 2017","anchor":"wbm-grote-et-al-2017-3","htmlText":"wBm, Grote et al 2017"},{"level":5,"text":"wBm, Chung et al 2019","anchor":"wbm-chung-et-al-2019-3","htmlText":"wBm, Chung et al 2019"},{"level":4,"text":"Differentially expressed genes ","anchor":"differentially-expressed-genes-","htmlText":"Differentially expressed genes "},{"level":5,"text":"wMel, Darby et al 2014","anchor":"wmel-darby-et-al-2014-4","htmlText":"wMel, Darby et al 2014"},{"level":5,"text":"wDi, Luck et al 2015","anchor":"wdi-luck-et-al-2015-4","htmlText":"wDi, Luck et al 2015"},{"level":5,"text":"wMel, Gutzwiller et al 2015","anchor":"wmel-gutzwiller-et-al-2015-4","htmlText":"wMel, Gutzwiller et al 2015"},{"level":5,"text":"wBm, Grote et al 2017","anchor":"wbm-grote-et-al-2017-4","htmlText":"wBm, Grote et al 2017"},{"level":5,"text":"wBm, Chung et al 2019","anchor":"wbm-chung-et-al-2019-4","htmlText":"wBm, Chung et al 2019"},{"level":3,"text":"Conduct hierarchical clustering analyses ","anchor":"conduct-hierarchical-clustering-analyses-","htmlText":"Conduct hierarchical clustering analyses "},{"level":4,"text":"All genes ","anchor":"all-genes--1","htmlText":"All genes "},{"level":5,"text":"wOo, Darby et al 2012","anchor":"woo-darby-et-al-2012-4","htmlText":"wOo, Darby et al 2012"},{"level":5,"text":"wMel, Darby et al 2014","anchor":"wmel-darby-et-al-2014-5","htmlText":"wMel, Darby et al 2014"},{"level":5,"text":"wDi, Luck et al 2014","anchor":"wdi-luck-et-al-2014-4","htmlText":"wDi, Luck et al 2014"},{"level":5,"text":"wDi, Luck et al 2015","anchor":"wdi-luck-et-al-2015-5","htmlText":"wDi, Luck et al 2015"},{"level":5,"text":"wMel, Gutzwiller et al 2015","anchor":"wmel-gutzwiller-et-al-2015-5","htmlText":"wMel, Gutzwiller et al 2015"},{"level":5,"text":"wBm, Grote et al 2017","anchor":"wbm-grote-et-al-2017-5","htmlText":"wBm, Grote et al 2017"},{"level":5,"text":"wBm, Chung et al 2019","anchor":"wbm-chung-et-al-2019-5","htmlText":"wBm, Chung et al 2019"},{"level":4,"text":"Kept genes ","anchor":"kept-genes--1","htmlText":"Kept genes "},{"level":5,"text":"wOo, Darby et al 2012","anchor":"woo-darby-et-al-2012-5","htmlText":"wOo, Darby et al 2012"},{"level":5,"text":"wMel, Darby et al 2014","anchor":"wmel-darby-et-al-2014-6","htmlText":"wMel, Darby et al 2014"},{"level":5,"text":"wDi, Luck et al 2014","anchor":"wdi-luck-et-al-2014-5","htmlText":"wDi, Luck et al 2014"},{"level":5,"text":"wDi, Luck et al 2015","anchor":"wdi-luck-et-al-2015-6","htmlText":"wDi, Luck et al 2015"},{"level":5,"text":"wMel, Gutzwiller et al 2015","anchor":"wmel-gutzwiller-et-al-2015-6","htmlText":"wMel, Gutzwiller et al 2015"},{"level":5,"text":"wBm, Grote et al 2017","anchor":"wbm-grote-et-al-2017-6","htmlText":"wBm, Grote et al 2017"},{"level":5,"text":"wBm, Chung et al 2019","anchor":"wbm-chung-et-al-2019-6","htmlText":"wBm, Chung et al 2019"},{"level":4,"text":"Differentially expressed genes ","anchor":"differentially-expressed-genes--1","htmlText":"Differentially expressed genes "},{"level":5,"text":"wMel, Darby et al 2014","anchor":"wmel-darby-et-al-2014-7","htmlText":"wMel, Darby et al 2014"},{"level":5,"text":"wDi, Luck et al 2015","anchor":"wdi-luck-et-al-2015-7","htmlText":"wDi, Luck et al 2015"},{"level":5,"text":"wMel, Gutzwiller et al 2015","anchor":"wmel-gutzwiller-et-al-2015-7","htmlText":"wMel, Gutzwiller et al 2015"},{"level":5,"text":"wBm, Grote et al 2017","anchor":"wbm-grote-et-al-2017-7","htmlText":"wBm, Grote et al 2017"},{"level":5,"text":"wBm, Chung et al 2019","anchor":"wbm-chung-et-al-2019-7","htmlText":"wBm, Chung et al 2019"},{"level":3,"text":"Create TPM and log2TPM heatmaps ","anchor":"create-tpm-and-log2tpm-heatmaps-","htmlText":"Create TPM and log2TPM heatmaps "},{"level":4,"text":"All genes ","anchor":"all-genes--2","htmlText":"All genes "},{"level":5,"text":"wOo, Darby et al 2012","anchor":"woo-darby-et-al-2012-6","htmlText":"wOo, Darby et al 2012"},{"level":5,"text":"wMel, Darby et al 2014","anchor":"wmel-darby-et-al-2014-8","htmlText":"wMel, Darby et al 2014"},{"level":5,"text":"wDi, Luck et al 2014","anchor":"wdi-luck-et-al-2014-6","htmlText":"wDi, Luck et al 2014"},{"level":5,"text":"wDi, Luck et al 2015","anchor":"wdi-luck-et-al-2015-8","htmlText":"wDi, Luck et al 2015"},{"level":5,"text":"wMel, Gutzwiller et al 2015","anchor":"wmel-gutzwiller-et-al-2015-8","htmlText":"wMel, Gutzwiller et al 2015"},{"level":5,"text":"wBm, Grote et al 2017","anchor":"wbm-grote-et-al-2017-8","htmlText":"wBm, Grote et al 2017"},{"level":5,"text":"wBm, Chung et al 2019","anchor":"wbm-chung-et-al-2019-8","htmlText":"wBm, Chung et al 2019"},{"level":4,"text":"Kept genes ","anchor":"kept-genes--2","htmlText":"Kept genes "},{"level":5,"text":"wOo, Darby et al 2012","anchor":"woo-darby-et-al-2012-7","htmlText":"wOo, Darby et al 2012"},{"level":5,"text":"wMel, Darby et al 2014","anchor":"wmel-darby-et-al-2014-9","htmlText":"wMel, Darby et al 2014"},{"level":5,"text":"wDi, Luck et al 2014","anchor":"wdi-luck-et-al-2014-7","htmlText":"wDi, Luck et al 2014"},{"level":5,"text":"wDi, Luck et al 2015","anchor":"wdi-luck-et-al-2015-9","htmlText":"wDi, Luck et al 2015"},{"level":5,"text":"wMel, Gutzwiller et al 2015","anchor":"wmel-gutzwiller-et-al-2015-9","htmlText":"wMel, Gutzwiller et al 2015"},{"level":5,"text":"wBm, Grote et al 2017","anchor":"wbm-grote-et-al-2017-9","htmlText":"wBm, Grote et al 2017"},{"level":5,"text":"wBm, Chung et al 2019","anchor":"wbm-chung-et-al-2019-9","htmlText":"wBm, Chung et al 2019"},{"level":4,"text":"Differentially expressed genes ","anchor":"differentially-expressed-genes--2","htmlText":"Differentially expressed genes "},{"level":5,"text":"wMel, Darby et al 2014","anchor":"wmel-darby-et-al-2014-10","htmlText":"wMel, Darby et al 2014"},{"level":5,"text":"wDi, Luck et al 2015","anchor":"wdi-luck-et-al-2015-10","htmlText":"wDi, Luck et al 2015"},{"level":5,"text":"wMel, Gutzwiller et al 2015","anchor":"wmel-gutzwiller-et-al-2015-10","htmlText":"wMel, Gutzwiller et al 2015"},{"level":5,"text":"wBm, Grote et al 2017","anchor":"wbm-grote-et-al-2017-10","htmlText":"wBm, Grote et al 2017"},{"level":5,"text":"wBm, Chung et al 2019","anchor":"wbm-chung-et-al-2019-10","htmlText":"wBm, Chung et al 2019"},{"level":3,"text":"Choose studies for basic pairwise analyses or WGCNA ","anchor":"choose-studies-for-basic-pairwise-analyses-or-wgcna-","htmlText":"Choose studies for basic pairwise analyses or WGCNA "},{"level":3,"text":"Conduct pairwise analyses on studies with two biological groups ","anchor":"conduct-pairwise-analyses-on-studies-with-two-biological-groups-","htmlText":"Conduct pairwise analyses on studies with two biological groups "},{"level":4,"text":"Create z-score of log2TPM heatmaps with a column bar that shows up- and down-regulated genes ","anchor":"create-z-score-of-log2tpm-heatmaps-with-a-column-bar-that-shows-up--and-down-regulated-genes-","htmlText":"Create z-score of log2TPM heatmaps with a column bar that shows up- and down-regulated genes "},{"level":5,"text":"wMel, Darby et al 2014","anchor":"wmel-darby-et-al-2014-11","htmlText":"wMel, Darby et al 2014"},{"level":4,"text":"Conduct functional term enrichment analysis on up- and down-regulated gene subsets ","anchor":"conduct-functional-term-enrichment-analysis-on-up--and-down-regulated-gene-subsets-","htmlText":"Conduct functional term enrichment analysis on up- and down-regulated gene subsets "},{"level":3,"text":"Conduct WGCNA on time-course studies ","anchor":"conduct-wgcna-on-time-course-studies-","htmlText":"Conduct WGCNA on time-course studies "},{"level":4,"text":"Identify WGCNA soft power values ","anchor":"identify-wgcna-soft-power-values-","htmlText":"Identify WGCNA soft power values "},{"level":5,"text":"wDi, Luck et al 2015","anchor":"wdi-luck-et-al-2015-11","htmlText":"wDi, Luck et al 2015"},{"level":5,"text":"wMel, Gutzwiller et al 2015","anchor":"wmel-gutzwiller-et-al-2015-11","htmlText":"wMel, Gutzwiller et al 2015"},{"level":5,"text":"wBm, Grote et al 2017","anchor":"wbm-grote-et-al-2017-11","htmlText":"wBm, Grote et al 2017"},{"level":5,"text":"wBm, Chung et al 2019","anchor":"wbm-chung-et-al-2019-11","htmlText":"wBm, Chung et al 2019"},{"level":4,"text":"Create WGCNA expression modules and merge similar modules ","anchor":"create-wgcna-expression-modules-and-merge-similar-modules-","htmlText":"Create WGCNA expression modules and merge similar modules "},{"level":5,"text":"wDi, Luck et al 2015","anchor":"wdi-luck-et-al-2015-12","htmlText":"wDi, Luck et al 2015"},{"level":5,"text":"wMel, Gutzwiller et al 2015","anchor":"wmel-gutzwiller-et-al-2015-12","htmlText":"wMel, Gutzwiller et al 2015"},{"level":5,"text":"wBm, Grote et al 2017","anchor":"wbm-grote-et-al-2017-12","htmlText":"wBm, Grote et al 2017"},{"level":5,"text":"wBm, Chung et al 2019","anchor":"wbm-chung-et-al-2019-12","htmlText":"wBm, Chung et al 2019"},{"level":4,"text":"Plot WGCNA-derived expression modules ","anchor":"plot-wgcna-derived-expression-modules-","htmlText":"Plot WGCNA-derived expression modules "},{"level":5,"text":"wDi, Luck et al 2015","anchor":"wdi-luck-et-al-2015-13","htmlText":"wDi, Luck et al 2015"},{"level":5,"text":"wMel, Gutzwiller et al 2015","anchor":"wmel-gutzwiller-et-al-2015-13","htmlText":"wMel, Gutzwiller et al 2015"},{"level":5,"text":"wBm, Grote et al 2017","anchor":"wbm-grote-et-al-2017-13","htmlText":"wBm, Grote et al 2017"},{"level":5,"text":"wBm, Chung et al 2019","anchor":"wbm-chung-et-al-2019-13","htmlText":"wBm, Chung et al 2019"},{"level":4,"text":"Conduct functional term enrichment analysis on WGCNA expression modules ","anchor":"conduct-functional-term-enrichment-analysis-on-wgcna-expression-modules-","htmlText":"Conduct functional term enrichment analysis on WGCNA expression modules "},{"level":2,"text":"Download reference files (2019-12-13) ","anchor":"download-reference-files-2019-12-13-","htmlText":"Download reference files (2019-12-13) "},{"level":4,"text":"Commands:","anchor":"commands-17","htmlText":"Commands:"},{"level":2,"text":"Create BLAST database with the amino acid sequences of the four Wolbachia genomes (2019-12-13) ","anchor":"create-blast-database-with-the-amino-acid-sequences-of-the-four-wolbachia-genomes-2019-12-13-","htmlText":"Create BLAST database with the amino acid sequences of the four Wolbachia genomes (2019-12-13) "},{"level":4,"text":"Inputs:","anchor":"inputs","htmlText":"Inputs:"},{"level":4,"text":"Commands:","anchor":"commands-18","htmlText":"Commands:"},{"level":2,"text":"Run self-BLASTP search against the protein sequences of the four Wolbachia strains (2019-12-13) ","anchor":"run-self-blastp-search-against-the-protein-sequences-of-the-four-wolbachia-strains-2019-12-13-","htmlText":"Run self-BLASTP search against the protein sequences of the four Wolbachia strains (2019-12-13) "},{"level":4,"text":"Inputs:","anchor":"inputs-1","htmlText":"Inputs:"},{"level":4,"text":"Commands:","anchor":"commands-19","htmlText":"Commands:"},{"level":2,"text":"Run PanOCT to identify core Wolbachia genes (2019-12-13) ","anchor":"run-panoct-to-identify-core-wolbachia-genes-2019-12-13-","htmlText":"Run PanOCT to identify core Wolbachia genes (2019-12-13) "},{"level":4,"text":"Inputs:","anchor":"inputs-2","htmlText":"Inputs:"},{"level":4,"text":"Inputs:","anchor":"inputs-3","htmlText":"Inputs:"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fmatt-chung%2Fwolbachia_transcriptomics_metaanalysis"}},{"displayName":"LICENSE","repoName":"wolbachia_transcriptomics_metaanalysis","refName":"master","path":"LICENSE","preferredFileType":"license","tabName":"MIT","richText":null,"loaded":false,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":null,"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fmatt-chung%2Fwolbachia_transcriptomics_metaanalysis"}}],"overviewFilesProcessingTime":189.316554,"createFromTemplatePath":"/new?template_name=wolbachia_transcriptomics_metaanalysis\u0026template_owner=matt-chung"}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":"https://github.dev/","enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":true,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":true}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
.bhqztV{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.bhqztV:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.bhqztV span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.bhqztV:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.bhqztV:focus{outline:2px solid transparent;}/*!sc*/
.bhqztV:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.bhqztV:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.bhqztV span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.bhqztV::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:transparent;border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.bhqztV::after{background-color:transparent;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,bhqztV,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cLeluw:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cLeluw:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cLeluw[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cLeluw:active{-webkit-transition:none;transition:none;}/*!sc*/
.cLeluw[data-inactive]{cursor:auto;}/*!sc*/
.cLeluw:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled [data-component=ButtonCounter],.cLeluw:disabled [data-component="leadingVisual"],.cLeluw:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cLeluw:focus{outline:solid 1px transparent;}}/*!sc*/
.cLeluw [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cLeluw[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cLeluw[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cLeluw[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cLeluw[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cLeluw[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cLeluw[data-block="block"]{width:100%;}/*!sc*/
.cLeluw[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cLeluw[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cLeluw [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cLeluw [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cLeluw [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cLeluw [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cLeluw:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cLeluw[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cLeluw[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cLeluw:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cLeluw:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cLeluw[data-size="small"][data-no-visuals]{margin-right:8px;height:28px;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,cLeluw,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="master branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        master
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <button hidden="">
                  </button>
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" aria-label="Add file" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R2hil5:" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-plus" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="react-directory-remove-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 pyyxt" id=":R2lil5:" tabindex="0" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Add file
                     </span>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </button>
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/commits/master/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           59 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/commits/master/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="gff_map_files, (Directory)" class="Link--primary" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/tree/master/gff_map_files" title="gff_map_files">
                          gff_map_files
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="gff_map_files, (Directory)" class="Link--primary" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/tree/master/gff_map_files" title="gff_map_files">
                          gff_map_files
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="images, (Directory)" class="Link--primary" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/tree/master/images" title="images">
                          images
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="images, (Directory)" class="Link--primary" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/tree/master/images" title="images">
                          images
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="sup_tables, (Directory)" class="Link--primary" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/tree/master/sup_tables" title="sup_tables">
                          sup_tables
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="sup_tables, (Directory)" class="Link--primary" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/tree/master/sup_tables" title="sup_tables">
                          sup_tables
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitattributes, (File)" class="Link--primary" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/.gitattributes" title=".gitattributes">
                          .gitattributes
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitattributes, (File)" class="Link--primary" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/.gitattributes" title=".gitattributes">
                          .gitattributes
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="LICENSE, (File)" class="Link--primary" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/LICENSE" title="LICENSE">
                          LICENSE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-5">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a class="Link__StyledLink-sc-14289xe-0 bhqztV" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-law" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0" data-component="text" data-content="MIT license">
                        MIT license
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-label="Edit file" class="types__StyledButton-sc-ws60qy-0 cLeluw" data-component="IconButton" data-no-visuals="true" data-size="small" title="Edit file" type="button">
                    <svg aria-hidden="true" class="octicon octicon-pencil" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </button>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      A meta-analysis of Wolbachia transcriptomics reveals a stage-specific Wolbachia transcriptional response shared across different hosts
                     </h1>
                     <a aria-label="Permalink: A meta-analysis of Wolbachia transcriptomics reveals a stage-specific Wolbachia transcriptional response shared across different hosts" class="anchor" href="#a-meta-analysis-of-wolbachia-transcriptomics-reveals-a-stage-specific-wolbachia-transcriptional-response-shared-across-different-hosts" id="user-content-a-meta-analysis-of-wolbachia-transcriptomics-reveals-a-stage-specific-wolbachia-transcriptional-response-shared-across-different-hosts">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Table of Contents
                     </h1>
                     <a aria-label="Permalink: Table of Contents" class="anchor" href="#table-of-contents" id="user-content-table-of-contents">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <a href="#setpaths">
                       Set software, input, and output directory paths
                      </a>
                      <ol dir="auto">
                       <li>
                        <a href="#setpaths_software">
                         Software
                        </a>
                       </li>
                       <li>
                        <a href="#setpaths_directories">
                         Directories
                        </a>
                       </li>
                      </ol>
                     </li>
                     <li>
                      <a href="#bashanalysis">
                       Download, align, and quantify Wolbachia sequencing data
                      </a>
                      <ol dir="auto">
                       <li>
                        <a href="#bashanalysis_createdirectories">
                         Create directories
                        </a>
                       </li>
                       <li>
                        <a href="#bashanalysis_getreferences">
                         Download reference files and create combined references
                        </a>
                       </li>
                       <li>
                        <a href="#bashanalysis_getfastq">
                         Download FASTQ files
                        </a>
                        <ol dir="auto">
                         <li>
                          <a href="#bashanalysis_getfastq_sra">
                           SRA Datasets
                          </a>
                         </li>
                         <li>
                          <a href="#bashanalysis_getfastq_luck2015">
                           Luck et al 2015 Dataset
                          </a>
                         </li>
                        </ol>
                       </li>
                       <li>
                        <a href="#bashanalysis_align">
                         Align FASTQ files to combined host and Wolbachia reference while allowing for splicing
                        </a>
                        <ol dir="auto">
                         <li>
                          <a href="#bashanalysis_align_se">
                           Single-End FASTQs
                          </a>
                         </li>
                         <li>
                          <a href="#bashanalysis_align_pe">
                           Paired-End FASTQs
                          </a>
                         </li>
                        </ol>
                       </li>
                       <li>
                        <a href="#bashanalysis_wolbachiareads">
                         Create a list of Wolbachia-mapping reads in all BAM files
                        </a>
                       </li>
                       <li>
                        <a href="#bashanalysis_subsetfastq">
                         Create subsets of all FASTQ files that contain only Wolbachia-mapping reads
                        </a>
                        <ol dir="auto">
                         <li>
                          <a href="#bashanalysis_subsetfastq_se">
                           Single-End FASTQs
                          </a>
                         </li>
                         <li>
                          <a href="#bashanalysis_subsetfastq_pe">
                           Paired-End FASTQs
                          </a>
                         </li>
                        </ol>
                       </li>
                       <li>
                        <a href="#bashanalysis_realign">
                         Align FASTQ files containing only Wolbachia-mapping reads to combined host and Wolbachia reference while disallowing splicing
                        </a>
                        <ol dir="auto">
                         <li>
                          <a href="#bashanalysis_realign_se">
                           Single-End FASTQs
                          </a>
                         </li>
                         <li>
                          <a href="#bashanalysis_realign_pe">
                           Paired-End FASTQs
                          </a>
                         </li>
                        </ol>
                       </li>
                       <li>
                        <a href="#bashanalysis_realign_sortbam">
                         Sort BAM files
                        </a>
                       </li>
                       <li>
                        <a href="#bashanalysis_indexbam">
                         Index BAM files
                        </a>
                       </li>
                       <li>
                        <a href="#bashanalysis_quant">
                         Quantify Wolbachia genes
                        </a>
                       </li>
                       <li>
                        <a href="#bashanalysis_gff2map">
                         Convert GFF to a mapping table
                        </a>
                       </li>
                       <li>
                        <a href="#bashanalysis_createcdsfasta">
                         Create nucleotide coding sequence fasta files
                        </a>
                       </li>
                       <li>
                        <a href="#bashanalysis_interproscan">
                         Identify InterPro descriptions and GO terms for Wolbachia transcripts
                        </a>
                       </li>
                      </ol>
                     </li>
                     <li>
                      <a href="#ranalysis">
                       Transcriptomics meta-analysis
                      </a>
                      <ol dir="auto">
                       <li>
                        <a href="#ranalysis_counts">
                         Create counts table
                        </a>
                        <ol dir="auto">
                         <li>
                          <a href="#ranalysis_counts_setinputs">
                           Set R inputs
                          </a>
                         </li>
                         <li>
                          <a href="#ranalysis_counts_sessioninfo">
                           View sessionInfo
                          </a>
                         </li>
                         <li>
                          <a href="#ranalysis_counts_maketable">
                           Combine count files into a single table for each study
                          </a>
                         </li>
                        </ol>
                       </li>
                       <li>
                        <a href="#ranalysis_iprscan2geneinfo">
                         Convert InterProScan output to geneinfo file
                        </a>
                       </li>
                       <li>
                        <a href="#ranalysis_de">
                         Differential expression and WGCNA analyses
                        </a>
                        <ol dir="auto">
                         <li>
                          <a href="#ranalysis_de_setinputs">
                           Set R inputs
                          </a>
                         </li>
                         <li>
                          <a href="#ranalysis_de_sessioninfo">
                           Load R packages and view sessionInfo
                          </a>
                         </li>
                         <li>
                          <a href="#ranalysis_de_loadfunctions">
                           Load R functions
                          </a>
                         </li>
                         <li>
                          <a href="#ranalysis_de_counts_readcounts">
                           Process counts files and calculate TPM values
                          </a>
                          <ol dir="auto">
                           <li>
                            <a href="#ranalysis_de_counts_readcounts">
                             Read counts table into a list
                            </a>
                           </li>
                           <li>
                            <a href="#ranalysis_de_counts_removenonproteingenes">
                             Exclude non-protein-coding genes from counts table
                            </a>
                           </li>
                           <li>
                            <a href="#ranalysis_de_counts_removenoprobegenes">
                             Exclude genes without Agilent SureSelect probes from counts table in Chung et al 2019 study
                            </a>
                           </li>
                           <li>
                            <a href="#ranalysis_de_counts_countscolsum">
                             Print the number of reads mapping to protein-coding genes for each sample in each study
                            </a>
                           </li>
                           <li>
                            <a href="#ranalysis_de_counts_tpm">
                             Create TPM table
                            </a>
                           </li>
                          </ol>
                         </li>
                         <li>
                          <a href="#ranalysis_de_keys">
                           Create keys
                          </a>
                          <ol dir="auto">
                           <li>
                            <a href="#ranalysis_de_keys_samples">
                             Sample keys
                            </a>
                           </li>
                           <li>
                            <a href="#ranalysis_de_keys_size">
                             Size keys
                            </a>
                           </li>
                           <li>
                            <a href="#ranalysis_de_keys_hm1">
                             Heatmap log2TPM key
                            </a>
                           </li>
                           <li>
                            <a href="#ranalysis_de_keys_hm2">
                             Heatmap z-score log2TPM key
                            </a>
                           </li>
                          </ol>
                         </li>
                         <li>
                          <a href="#ranalysis_de_keys_saturation">
                           Assess if samples have been sequenced to saturation using saturation curves
                          </a>
                         </li>
                         <li>
                          <a href="#ranalysis_de_exclude">
                           Exclude samples and/or studies from differential expression analyses due to inadequate sequencing depth
                          </a>
                         </li>
                         <li>
                          <a href="#ranalysis_de_exclude">
                           Identify the number of genes in each study that meets the minimum CPM threshold
                          </a>
                         </li>
                        </ol>
                       </li>
                       <li>
                        <a href="#ranalysis_de_de">
                         Identify differentially expressed genes (FDR &lt; 0.05) in each study
                        </a>
                       </li>
                       <li>
                        <a href="#ranalysis_de_pca">
                         Conduct principal component analyses
                        </a>
                       </li>
                       <li>
                        <a href="#ranalysis_de_hc">
                         Conduct hierarchical clustering analyses
                        </a>
                       </li>
                       <li>
                        <a href="#ranalysis_de_heatmap">
                         Create TPM and log2TPM heatmaps
                        </a>
                       </li>
                       <li>
                        <a href="#ranalysis_de_selectanalysis">
                         Choose studies for basic pairwise analyses or WGCNA
                        </a>
                       </li>
                       <li>
                        <a href="#ranalysis_de_pairwise">
                         Conduct pairwise analyses on studies with two biological groups
                        </a>
                        <ol dir="auto">
                         <li>
                          <a href="#ranalysis_de_pairwise_hm">
                           Create z-score of log2TPM heatmaps with a column bar that shows up- and down-regulated genes
                          </a>
                         </li>
                         <li>
                          <a href="#ranalysis_de_pairwise_fxnterm">
                           Conduct functional term enrichment analysis on up- and down-regulated gene subsets
                          </a>
                         </li>
                        </ol>
                       </li>
                       <li>
                        <a href="#ranalysis_de_wgcna">
                         Conduct WGCNA on time-course studies
                        </a>
                        <ol dir="auto">
                         <li>
                          <a href="#ranalysis_de_wgcna_softpower">
                           Identify WGCNA soft power values
                          </a>
                         </li>
                         <li>
                          <a href="#ranalysis_de_wgcna_createmodules">
                           Create WGCNA expression modules and merge similar modules
                          </a>
                         </li>
                         <li>
                          <a href="#ranalysis_de_wgcna_heatmap">
                           Plot WGCNA-derived expression modules
                          </a>
                         </li>
                         <li>
                          <a href="#ranalysis_de_wgcna_fxnterm">
                           Conduct functional term enrichment analysis on WGCNA expression modules
                          </a>
                         </li>
                        </ol>
                       </li>
                      </ol>
                     </li>
                    </ol>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Set software and directory paths
                      <a name="user-content-setpaths">
                      </a>
                     </h1>
                     <a aria-label="Permalink: Set software and directory paths " class="anchor" href="#set-software-and-directory-paths-" id="user-content-set-software-and-directory-paths-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     For rerunning analyses, all paths in this section must be set by the user.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Software
                      <a name="user-content-setpaths_software">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Software " class="anchor" href="#software-" id="user-content-software-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="JULIA_DEPOT_PATH=/home/mattchung/.julia
PYTHON_LIB_PATH=/usr/local/packages/python-3.5/lib

JULIA_BIN_DIR=/usr/local/bin
PYTHON_BIN_DIR=/usr/local/bin

FADU_BIN_DIR=/home/mattchung/scripts/FADU
HISAT2_BIN_DIR=/usr/local/packages/hisat2-2.1.0
INTERPROSCAN_BIN_DIR=/local/aberdeen2rw/julie/Matt_dir/packages/interproscan-5.34-73.0
NCBIBLAST_BIN_DIR=/usr/local/packages/ncbi-blast-2.2.26/bin
SAMTOOLS_BIN_DIR=/usr/local/packages/samtools-1.9/bin
SEQTK_BIN_DIR=/usr/local/packages/seqtk-1.2/bin
SRATOOLKIT_BIN_DIR=/usr/local/packages/sratoolkit-2.9.0/bin">
                     <pre class="notranslate" lang="{bash,"><code>JULIA_DEPOT_PATH=/home/mattchung/.julia
PYTHON_LIB_PATH=/usr/local/packages/python-3.5/lib

JULIA_BIN_DIR=/usr/local/bin
PYTHON_BIN_DIR=/usr/local/bin

FADU_BIN_DIR=/home/mattchung/scripts/FADU
HISAT2_BIN_DIR=/usr/local/packages/hisat2-2.1.0
INTERPROSCAN_BIN_DIR=/local/aberdeen2rw/julie/Matt_dir/packages/interproscan-5.34-73.0
NCBIBLAST_BIN_DIR=/usr/local/packages/ncbi-blast-2.2.26/bin
SAMTOOLS_BIN_DIR=/usr/local/packages/samtools-1.9/bin
SEQTK_BIN_DIR=/usr/local/packages/seqtk-1.2/bin
SRATOOLKIT_BIN_DIR=/usr/local/packages/sratoolkit-2.9.0/bin
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Directories
                      <a name="user-content-setpaths_directories">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Directories " class="anchor" href="#directories-" id="user-content-directories-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="INPUTS_DIR=
SCRIPTS_DIR=/home/mattchung/scripts
READS_DIR=/local/aberdeen2rw/julie/Matt_dir/wolbachia_metatranscriptome_analysis/
REFERENCES_DIR=/local/aberdeen2rw/julie/Matt_dir/wolbachia_metatranscriptome_analysis/references/
WORKING_DIR=/local/projects-t3/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/">
                     <pre class="notranslate" lang="{bash,"><code>INPUTS_DIR=
SCRIPTS_DIR=/home/mattchung/scripts
READS_DIR=/local/aberdeen2rw/julie/Matt_dir/wolbachia_metatranscriptome_analysis/
REFERENCES_DIR=/local/aberdeen2rw/julie/Matt_dir/wolbachia_metatranscriptome_analysis/references/
WORKING_DIR=/local/projects-t3/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Download, align, and quantify Wolbachia sequencing data
                      <a name="user-content-bashanalysis">
                      </a>
                     </h1>
                     <a aria-label="Permalink: Download, align, and quantify Wolbachia sequencing data " class="anchor" href="#download-align-and-quantify-wolbachia-sequencing-data-" id="user-content-download-align-and-quantify-wolbachia-sequencing-data-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Create directories (2019-05-30)
                      <a name="user-content-bashanalysis_createdirectories">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Create directories (2019-05-30) " class="anchor" href="#create-directories-2019-05-30-" id="user-content-create-directories-2019-05-30-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='mkdir -p "$REFERENCES_DIR"

mkdir -p "$READS_DIR"/wOo_darby_2012/bam
mkdir -p "$READS_DIR"/wMel_darby_2014/bam
mkdir -p "$READS_DIR"/wDi_luck_2014/bam
mkdir -p "$READS_DIR"/wDi_luck_2015/bam
mkdir -p "$READS_DIR"/wMel_gutzwiller_2015/bam
mkdir -p "$READS_DIR"/wBm_grote_2017/bam
mkdir -p "$READS_DIR"/wBm_chung_2019/bam'>
                     <pre class="notranslate" lang="{bash,"><code>mkdir -p "$REFERENCES_DIR"

mkdir -p "$READS_DIR"/wOo_darby_2012/bam
mkdir -p "$READS_DIR"/wMel_darby_2014/bam
mkdir -p "$READS_DIR"/wDi_luck_2014/bam
mkdir -p "$READS_DIR"/wDi_luck_2015/bam
mkdir -p "$READS_DIR"/wMel_gutzwiller_2015/bam
mkdir -p "$READS_DIR"/wBm_grote_2017/bam
mkdir -p "$READS_DIR"/wBm_chung_2019/bam
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Download reference files and create combined references (2019-05-30)
                      <a name="user-content-bashanalysis_getreferences">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Download reference files and create combined references (2019-05-30) " class="anchor" href="#download-reference-files-and-create-combined-references-2019-05-30-" id="user-content-download-reference-files-and-create-combined-references-2019-05-30-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The gff reference for wDi must be downloaded from this link:
                     <a href="http://rast.nmpdr.org/rast.cgi?page=JobDetails&amp;job=53659" rel="nofollow">
                      http://rast.nmpdr.org/rast.cgi?page=JobDetails&amp;job=53659
                     </a>
                     (login:guest, password:guest). All subsequent code will make use of this wDi.gff at this path: "$REFERENCES_DIR"/wDi.gff
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands" id="user-content-commands">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='## wBm + B. malayi
wget -O "$REFERENCES_DIR"/wBm.fna.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_genomic.fna.gz
wget -O "$REFERENCES_DIR"/bmalayi.fna.gz ftp://ftp.wormbase.org/pub/wormbase/releases/WS270/species/b_malayi/PRJNA10729/b_malayi.PRJNA10729.WS270.genomic.fa.gz
gunzip "$REFERENCES_DIR"/wBm.fna.gz
gunzip "$REFERENCES_DIR"/bmalayi.fna.gz
cat "$REFERENCES_DIR"/wBm.fna "$REFERENCES_DIR"/bmalayi.fna &gt; "$REFERENCES_DIR"/wBm_bmalayi_combined.fna

wget -O "$REFERENCES_DIR"/wBm.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_genomic.gff.gz
gunzip "$REFERENCES_DIR"/wBm.gff.gz

## wDi + D. immitis
wget -O "$REFERENCES_DIR"/wDi.fna.gz http://nematodes.org/downloads/959nematodegenomes/blast/db2/Dirofilaria_immitis_wolbachia_2.2.fna.gz
wget -O "$REFERENCES_DIR"/dimmitis.fna.gz http://nematodes.org/downloads/959nematodegenomes/blast/db2/Dirofilaria_immitis_2.2.fna.gz
gunzip "$REFERENCES_DIR"/wDi.fna.gz
gunzip "$REFERENCES_DIR"/dimmitis.fna.gz
cat "$REFERENCES_DIR"/wDi.fna "$REFERENCES_DIR"/dimmitis.fna &gt; "$REFERENCES_DIR"/wDi_dimmitis_combined.fna

wget -O "$REFERENCES_DIR"/wBm.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_genomic.gff.gz
gunzip "$REFERENCES_DIR"/wBm.gff.gz

## wMel + D. melanogaster
wget -O "$REFERENCES_DIR"/wMel.fna.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/025/GCF_000008025.1_ASM802v1/GCF_000008025.1_ASM802v1_genomic.fna.gz
wget -O "$REFERENCES_DIR"/dmelanogaster.fna.gz ftp://ftp.flybase.net/releases/FB2019_02/dmel_r6.27/fasta/dmel-all-chromosome-r6.27.fasta.gz
gunzip "$REFERENCES_DIR"/wMel.fna.gz
gunzip "$REFERENCES_DIR"/dmelanogaster.fna.gz
cat "$REFERENCES_DIR"/wMel.fna "$REFERENCES_DIR"/dmelanogaster.fna &gt; "$REFERENCES_DIR"/wMel_dmelanogaster_combined.fna

wget -O "$REFERENCES_DIR"/wMel.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/025/GCF_000008025.1_ASM802v1/GCF_000008025.1_ASM802v1_genomic.gff.gz
gunzip "$REFERENCES_DIR"/wMel.gff.gz

## wOo + O. ochengi
wget -O "$REFERENCES_DIR"/wOo.fna.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/306/885/GCF_000306885.1_ASM30688v1/GCF_000306885.1_ASM30688v1_genomic.fna.gz
wget -O "$REFERENCES_DIR"/oochengi.fna.gz http://www.nematodes.org/downloads/959nematodegenomes/blast/db2/Onchocerca_ochengi_nuclear_assembly_nOo.2.0.fna.gz
gunzip "$REFERENCES_DIR"/wOo.fna.gz
gunzip "$REFERENCES_DIR"/oochengi.fna.gz
cat "$REFERENCES_DIR"/wOo.fna "$REFERENCES_DIR"/oochengi.fna &gt; "$REFERENCES_DIR"/wOo_oochengi_combined.fna

wget -O "$REFERENCES_DIR"/wOo.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/306/885/GCF_000306885.1_ASM30688v1/GCF_000306885.1_ASM30688v1_genomic.gff.gz
gunzip "$REFERENCES_DIR"/wOo.gff.gz'>
                     <pre class="notranslate" lang="{bash,"><code>## wBm + B. malayi
wget -O "$REFERENCES_DIR"/wBm.fna.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_genomic.fna.gz
wget -O "$REFERENCES_DIR"/bmalayi.fna.gz ftp://ftp.wormbase.org/pub/wormbase/releases/WS270/species/b_malayi/PRJNA10729/b_malayi.PRJNA10729.WS270.genomic.fa.gz
gunzip "$REFERENCES_DIR"/wBm.fna.gz
gunzip "$REFERENCES_DIR"/bmalayi.fna.gz
cat "$REFERENCES_DIR"/wBm.fna "$REFERENCES_DIR"/bmalayi.fna &gt; "$REFERENCES_DIR"/wBm_bmalayi_combined.fna

wget -O "$REFERENCES_DIR"/wBm.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_genomic.gff.gz
gunzip "$REFERENCES_DIR"/wBm.gff.gz

## wDi + D. immitis
wget -O "$REFERENCES_DIR"/wDi.fna.gz http://nematodes.org/downloads/959nematodegenomes/blast/db2/Dirofilaria_immitis_wolbachia_2.2.fna.gz
wget -O "$REFERENCES_DIR"/dimmitis.fna.gz http://nematodes.org/downloads/959nematodegenomes/blast/db2/Dirofilaria_immitis_2.2.fna.gz
gunzip "$REFERENCES_DIR"/wDi.fna.gz
gunzip "$REFERENCES_DIR"/dimmitis.fna.gz
cat "$REFERENCES_DIR"/wDi.fna "$REFERENCES_DIR"/dimmitis.fna &gt; "$REFERENCES_DIR"/wDi_dimmitis_combined.fna

wget -O "$REFERENCES_DIR"/wBm.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_genomic.gff.gz
gunzip "$REFERENCES_DIR"/wBm.gff.gz

## wMel + D. melanogaster
wget -O "$REFERENCES_DIR"/wMel.fna.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/025/GCF_000008025.1_ASM802v1/GCF_000008025.1_ASM802v1_genomic.fna.gz
wget -O "$REFERENCES_DIR"/dmelanogaster.fna.gz ftp://ftp.flybase.net/releases/FB2019_02/dmel_r6.27/fasta/dmel-all-chromosome-r6.27.fasta.gz
gunzip "$REFERENCES_DIR"/wMel.fna.gz
gunzip "$REFERENCES_DIR"/dmelanogaster.fna.gz
cat "$REFERENCES_DIR"/wMel.fna "$REFERENCES_DIR"/dmelanogaster.fna &gt; "$REFERENCES_DIR"/wMel_dmelanogaster_combined.fna

wget -O "$REFERENCES_DIR"/wMel.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/025/GCF_000008025.1_ASM802v1/GCF_000008025.1_ASM802v1_genomic.gff.gz
gunzip "$REFERENCES_DIR"/wMel.gff.gz

## wOo + O. ochengi
wget -O "$REFERENCES_DIR"/wOo.fna.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/306/885/GCF_000306885.1_ASM30688v1/GCF_000306885.1_ASM30688v1_genomic.fna.gz
wget -O "$REFERENCES_DIR"/oochengi.fna.gz http://www.nematodes.org/downloads/959nematodegenomes/blast/db2/Onchocerca_ochengi_nuclear_assembly_nOo.2.0.fna.gz
gunzip "$REFERENCES_DIR"/wOo.fna.gz
gunzip "$REFERENCES_DIR"/oochengi.fna.gz
cat "$REFERENCES_DIR"/wOo.fna "$REFERENCES_DIR"/oochengi.fna &gt; "$REFERENCES_DIR"/wOo_oochengi_combined.fna

wget -O "$REFERENCES_DIR"/wOo.gff.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/306/885/GCF_000306885.1_ASM30688v1/GCF_000306885.1_ASM30688v1_genomic.gff.gz
gunzip "$REFERENCES_DIR"/wOo.gff.gz
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Download FASTQ files (2019-05-30)
                      <a name="user-content-bashanalysis_getfastq">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Download FASTQ files (2019-05-30) " class="anchor" href="#download-fastq-files-2019-05-30-" id="user-content-download-fastq-files-2019-05-30-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     All FASTQ files from 6 of the studies were downloaded from the SRA. Because the Wolbachia reads were not included in the FASTQ files uploaded for the Luck 2015 study, NEB uploaded them onto the IGS FTP site (ftp://nsflgt@ftp.igs.umaryland.edu; password: drosophila) for download to this directory: "$READS_DIR"/wDi_luck_2015
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      SRA Datasets
                      <a name="user-content-bashanalysis_getfastq_sra">
                      </a>
                     </h3>
                     <a aria-label="Permalink: SRA Datasets " class="anchor" href="#sra-datasets-" id="user-content-sra-datasets-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets" id="user-content-input-sets">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='## wOo Darby 2012
SRR_ID_LIST="$INPUTS_DIR"/wOo_darby_2012_srr.list
OUTPUT_DIR="$READS_DIR"/wOo_darby_2012

## wMel Darby 2014
SRR_ID_LIST=/wMel_darby_2014_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_darby_2014

## wDi Luck 2014
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2014_srr.list
OUTPUT_DIR="$READS_DIR"/wDi_luck_2014

## wMel Gutzwiller 2015
SRR_ID_LIST="$INPUTS_DIR"/wMel_gutzwiller_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_gutzwiller_2015

## wBm Grote 2017
SRR_ID_LIST="$INPUTS_DIR"/wBm_grote_2017_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_grote_2017

## wBm Chung 2019
SRR_ID_LIST="$INPUTS_DIR"/wBm_chung_2019_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_chung_2019'>
                     <pre class="notranslate" lang="{bash,"><code>## wOo Darby 2012
SRR_ID_LIST="$INPUTS_DIR"/wOo_darby_2012_srr.list
OUTPUT_DIR="$READS_DIR"/wOo_darby_2012

## wMel Darby 2014
SRR_ID_LIST=/wMel_darby_2014_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_darby_2014

## wDi Luck 2014
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2014_srr.list
OUTPUT_DIR="$READS_DIR"/wDi_luck_2014

## wMel Gutzwiller 2015
SRR_ID_LIST="$INPUTS_DIR"/wMel_gutzwiller_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_gutzwiller_2015

## wBm Grote 2017
SRR_ID_LIST="$INPUTS_DIR"/wBm_grote_2017_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_grote_2017

## wBm Chung 2019
SRR_ID_LIST="$INPUTS_DIR"/wBm_chung_2019_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_chung_2019
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-1" id="user-content-commands-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="while read Line
do

SRR_ID=&quot;$(echo $Line | awk -F &quot;\t&quot; '{print $1}')&quot;

qsub -P jdhotopp-lab -l mem_free=2G -N fastq_dump -wd &quot;$OUTPUT_DIR&quot; -b y &quot;$SRATOOLKIT_BIN_DIR&quot;/fastq-dump --split-files &quot;$SRR_ID&quot; -O &quot;$OUTPUT_DIR&quot;

done &lt; &quot;$SRR_ID_LIST&quot;">
                     <pre class="notranslate" lang="{bash,"><code>while read Line
do

SRR_ID="$(echo $Line | awk -F "\t" '{print $1}')"

qsub -P jdhotopp-lab -l mem_free=2G -N fastq_dump -wd "$OUTPUT_DIR" -b y "$SRATOOLKIT_BIN_DIR"/fastq-dump --split-files "$SRR_ID" -O "$OUTPUT_DIR"

done &lt; "$SRR_ID_LIST"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Luck et al 2015 Dataset
                      <a name="user-content-bashanalysis_getfastq_luck2015">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Luck et al 2015 Dataset " class="anchor" href="#luck-et-al-2015-dataset-" id="user-content-luck-et-al-2015-dataset-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-1" id="user-content-input-sets-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='## wDi Luck 2015
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wDi_luck_2015'>
                     <pre class="notranslate" lang="{bash,"><code>## wDi Luck 2015
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wDi_luck_2015
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-2" id="user-content-commands-2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='zcat "$READS_DIR"/wDi_luck_2015/Galaxy1-[Di_Female_Intestine_2].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974179.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy2-[Di_Female_Intestine_1].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974178.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy3-[Di_Female_Body_Wall_2].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974175.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy4-[Di_Female_Uterus_1].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974180.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy5-[Di_Female_Head_2].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974177.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy6-[Di_Female_Uterus_2].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974181.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy7-[Di_Female_Head_1].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974176.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy8-[Di_Female_Body_Wall_1].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974174.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy9-[Di_Male_Testes].fastq.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974185.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy10-[Di_Male_Body_Wall_1].fastq.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974182.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy11-[Di_Male_Body_Wall_2] (1).fastq.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974183.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy12-[Di_Male_Intestine] (1).fastq.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974184.fastq'>
                     <pre class="notranslate" lang="{bash,"><code>zcat "$READS_DIR"/wDi_luck_2015/Galaxy1-[Di_Female_Intestine_2].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974179.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy2-[Di_Female_Intestine_1].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974178.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy3-[Di_Female_Body_Wall_2].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974175.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy4-[Di_Female_Uterus_1].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974180.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy5-[Di_Female_Head_2].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974177.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy6-[Di_Female_Uterus_2].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974181.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy7-[Di_Female_Head_1].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974176.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy8-[Di_Female_Body_Wall_1].fastqsanger.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974174.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy9-[Di_Male_Testes].fastq.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974185.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy10-[Di_Male_Body_Wall_1].fastq.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974182.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy11-[Di_Male_Body_Wall_2] (1).fastq.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974183.fastq
zcat "$READS_DIR"/wDi_luck_2015/Galaxy12-[Di_Male_Intestine] (1).fastq.zip &gt; "$READS_DIR"/wDi_luck_2015/SRR1974184.fastq
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Align FASTQ files to combined host and Wolbachia reference while allowing for splicing (2019-05-30)
                      <a name="user-content-bashanalysis_align">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Align FASTQ files to combined host and Wolbachia reference while allowing for splicing (2019-05-30)  " class="anchor" href="#align-fastq-files-to-combined-host-and-wolbachia-reference-while-allowing-for-splicing-2019-05-30--" id="user-content-align-fastq-files-to-combined-host-and-wolbachia-reference-while-allowing-for-splicing-2019-05-30--">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Single-End FASTQs
                      <a name="user-content-bashanalysis_align_se">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Single-End FASTQs " class="anchor" href="#single-end-fastqs-" id="user-content-single-end-fastqs-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-2" id="user-content-input-sets-2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='THREADS=4

## wOo Darby 2012
REF_FNA="$REFERENCES_DIR"/wOo_oochengi_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wOo_oochengi_2012_srr.list
OUTPUT_DIR="$READS_DIR"/wOo_oochengi_2012/bam
STRANDEDNESS=""
PREFIX=wOo_darby_2012

## wDi Luck 2014
REF_FNA="$REFERENCES_DIR"/wDi_dimmitis_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2014_srr.list
OUTPUT_DIR="$READS_DIR"/wDi_luck_2014/bam
STRANDEDNESS=""
PREFIX=wDi_luck_2014

## wDi Luck 2015
REF_FNA="$REFERENCES_DIR"/wDi_dimmitis_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wDi_luck_2015/bam
STRANDEDNESS=""
PREFIX=wDi_luck_2015'>
                     <pre class="notranslate" lang="{bash,"><code>THREADS=4

## wOo Darby 2012
REF_FNA="$REFERENCES_DIR"/wOo_oochengi_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wOo_oochengi_2012_srr.list
OUTPUT_DIR="$READS_DIR"/wOo_oochengi_2012/bam
STRANDEDNESS=""
PREFIX=wOo_darby_2012

## wDi Luck 2014
REF_FNA="$REFERENCES_DIR"/wDi_dimmitis_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2014_srr.list
OUTPUT_DIR="$READS_DIR"/wDi_luck_2014/bam
STRANDEDNESS=""
PREFIX=wDi_luck_2014

## wDi Luck 2015
REF_FNA="$REFERENCES_DIR"/wDi_dimmitis_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wDi_luck_2015/bam
STRANDEDNESS=""
PREFIX=wDi_luck_2015
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-3" id="user-content-commands-3">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='while read SRR
do
	echo ""$HISAT2_BIN_DIR"/hisat2 -p "$THREADS" -x "$REF_FNA" "$STRANDEDNESS" -U "$READS_DIR"/"$PREFIX"/"$SRR"_1.fastq | "$SAMTOOLS_BIN_DIR"/samtools view -bhSo "$OUTPUT_DIR"/"$SRR".bam -" | qsub -q threaded.q  -pe thread "$THREADS" -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd "$OUTPUT_DIR"
done &lt; "$SRR_ID_LIST"'>
                     <pre class="notranslate" lang="{bash,"><code>while read SRR
do
	echo ""$HISAT2_BIN_DIR"/hisat2 -p "$THREADS" -x "$REF_FNA" "$STRANDEDNESS" -U "$READS_DIR"/"$PREFIX"/"$SRR"_1.fastq | "$SAMTOOLS_BIN_DIR"/samtools view -bhSo "$OUTPUT_DIR"/"$SRR".bam -" | qsub -q threaded.q  -pe thread "$THREADS" -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd "$OUTPUT_DIR"
done &lt; "$SRR_ID_LIST"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Paired-End FASTQs
                      <a name="user-content-bashanalysis_align_pe">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Paired-End FASTQs " class="anchor" href="#paired-end-fastqs-" id="user-content-paired-end-fastqs-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-3" id="user-content-input-sets-3">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='THREADS=4

## wMel Darby 2014
REF_FNA="$REFERENCES_DIR"/wMel_dmelanogaster_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wMel_darby_2014_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_darby_2014/bam
STRANDEDNESS="--rna-strandness FR"
PREFIX=wMel_darby_2014

## wMel Gutzwiller 2015
REF_FNA="$REFERENCES_DIR"/wMel_dmelanogaster_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wMel_gutzwiller_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_gutzwiller_2015/bam
STRANDEDNESS="--rna-strandness RF"
PREFIX=wMel_gutzwiller_2015

## wBm Grote 2017
REF_FNA="$REFERENCES_DIR"/wBm_bmalayi_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wBm_grote_2017_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_grote_2017/bam
STRANDEDNESS=""
PREFIX=wBm_grote_2017

## wBm Chung 2019
REF_FNA="$REFERENCES_DIR"/wBm_bmalayi_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wBm_chung_2019_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_chung_2019/bam
STRANDEDNESS="--rna-strandness RF"
PREFIX=wBm_chung_2019'>
                     <pre class="notranslate" lang="{bash,"><code>THREADS=4

## wMel Darby 2014
REF_FNA="$REFERENCES_DIR"/wMel_dmelanogaster_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wMel_darby_2014_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_darby_2014/bam
STRANDEDNESS="--rna-strandness FR"
PREFIX=wMel_darby_2014

## wMel Gutzwiller 2015
REF_FNA="$REFERENCES_DIR"/wMel_dmelanogaster_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wMel_gutzwiller_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_gutzwiller_2015/bam
STRANDEDNESS="--rna-strandness RF"
PREFIX=wMel_gutzwiller_2015

## wBm Grote 2017
REF_FNA="$REFERENCES_DIR"/wBm_bmalayi_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wBm_grote_2017_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_grote_2017/bam
STRANDEDNESS=""
PREFIX=wBm_grote_2017

## wBm Chung 2019
REF_FNA="$REFERENCES_DIR"/wBm_bmalayi_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wBm_chung_2019_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_chung_2019/bam
STRANDEDNESS="--rna-strandness RF"
PREFIX=wBm_chung_2019
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-4" id="user-content-commands-4">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='while read srr
do
	echo ""$HISAT2_BIN_DIR"/hisat2 -p "$THREADS" -x "$REF_FNA" "$STRANDEDNESS" -1 "$READS_DIR"/"$PREFIX"/"$SRR"_1.fastq  -2 "$READS_DIR"/"$PREFIX"/"$SRR"_2.fastq | "$SAMTOOLS_BIN_DIR"/samtools view -bhSo "$OUTPUT_DIR"/"$SRR".bam -" | qsub -q threaded.q  -pe thread "$THREADS" -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd "$OUTPUT_DIR"
done &lt; "$SRR_ID_LIST"'>
                     <pre class="notranslate" lang="{bash,"><code>while read srr
do
	echo ""$HISAT2_BIN_DIR"/hisat2 -p "$THREADS" -x "$REF_FNA" "$STRANDEDNESS" -1 "$READS_DIR"/"$PREFIX"/"$SRR"_1.fastq  -2 "$READS_DIR"/"$PREFIX"/"$SRR"_2.fastq | "$SAMTOOLS_BIN_DIR"/samtools view -bhSo "$OUTPUT_DIR"/"$SRR".bam -" | qsub -q threaded.q  -pe thread "$THREADS" -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd "$OUTPUT_DIR"
done &lt; "$SRR_ID_LIST"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Create a list of Wolbachia-mapping reads in all BAM files (2019-05-30)
                      <a name="user-content-bashanalysis_wolbachiareads">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Create a list of Wolbachia-mapping reads in all BAM files (2019-05-30) " class="anchor" href="#create-a-list-of-wolbachia-mapping-reads-in-all-bam-files-2019-05-30-" id="user-content-create-a-list-of-wolbachia-mapping-reads-in-all-bam-files-2019-05-30-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-4" id="user-content-input-sets-4">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='## wOo Darby 2012
BAM_DIR="$READS_DIR"/wOo_darby_2012/bam
SRR_ID_LIST="$INPUTS_DIR"/wOo_darby_2012_srr.list
CONTIG=NC_018267.1

## wMel Darby 2014
BAM_DIR="$READS_DIR"/wMel_darby_2014/bam
SRR_ID_LIST=/wMel_darby_2014_srr.list
CONTIG=NC_002978.6

## wDi Luck 2014
BAM_DIR="$READS_DIR"/wDi_luck_2014/bam
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2014_srr.list
CONTIG=wDi22.scaf1

## wDi Luck 2015
BAM_DIR="$READS_DIR"/wDi_luck_2015/bam
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2015_srr.list
CONTIG=wDi22.scaf1

## wMel Gutzwiller 2015
BAM_DIR="$READS_DIR"/wMel_gutzwiller_2015/bam
SRR_ID_LIST="$INPUTS_DIR"/wMel_gutzwiller_2015_srr.list
CONTIG=NC_002978.6

## wBm Grote 2017
BAM_DIR="$READS_DIR"/wBm_grote_2017/bam
SRR_ID_LIST="$INPUTS_DIR"/wBm_grote_2017_srr.list
CONTIG=NC_006833.1

## wBm Chung 2019
BAM_DIR="$READS_DIR"/wBm_chung_2019/bam
SRR_ID_LIST="$INPUTS_DIR"/wBm_chung_2019_srr.list
CONTIG=NC_006833.1'>
                     <pre class="notranslate" lang="{bash,"><code>## wOo Darby 2012
BAM_DIR="$READS_DIR"/wOo_darby_2012/bam
SRR_ID_LIST="$INPUTS_DIR"/wOo_darby_2012_srr.list
CONTIG=NC_018267.1

## wMel Darby 2014
BAM_DIR="$READS_DIR"/wMel_darby_2014/bam
SRR_ID_LIST=/wMel_darby_2014_srr.list
CONTIG=NC_002978.6

## wDi Luck 2014
BAM_DIR="$READS_DIR"/wDi_luck_2014/bam
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2014_srr.list
CONTIG=wDi22.scaf1

## wDi Luck 2015
BAM_DIR="$READS_DIR"/wDi_luck_2015/bam
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2015_srr.list
CONTIG=wDi22.scaf1

## wMel Gutzwiller 2015
BAM_DIR="$READS_DIR"/wMel_gutzwiller_2015/bam
SRR_ID_LIST="$INPUTS_DIR"/wMel_gutzwiller_2015_srr.list
CONTIG=NC_002978.6

## wBm Grote 2017
BAM_DIR="$READS_DIR"/wBm_grote_2017/bam
SRR_ID_LIST="$INPUTS_DIR"/wBm_grote_2017_srr.list
CONTIG=NC_006833.1

## wBm Chung 2019
BAM_DIR="$READS_DIR"/wBm_chung_2019/bam
SRR_ID_LIST="$INPUTS_DIR"/wBm_chung_2019_srr.list
CONTIG=NC_006833.1
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-5" id="user-content-commands-5">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="while read SRR
do
	&quot;#SAMTOOLS_BIN_DIR&quot;/samtools view &quot;$BAM_DIR&quot;/&quot;SRR&quot;.bam | grep &quot;$PREFIX&quot; | awk '{print $1}' | sort -n | uniq &gt; &quot;$BAM_DIR&quot;/&quot;SRR&quot;.target_reads.list
done &lt; &quot;$SRR_ID_LIST&quot;">
                     <pre class="notranslate" lang="{bash,"><code>while read SRR
do
	"#SAMTOOLS_BIN_DIR"/samtools view "$BAM_DIR"/"SRR".bam | grep "$PREFIX" | awk '{print $1}' | sort -n | uniq &gt; "$BAM_DIR"/"SRR".target_reads.list
done &lt; "$SRR_ID_LIST"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Create subsets of all FASTQ files that contain only Wolbachia-mapping reads (2019-05-30)
                      <a name="user-content-bashanalysis_subsetfastq">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Create subsets of all FASTQ files that contain only Wolbachia-mapping reads (2019-05-30) " class="anchor" href="#create-subsets-of-all-fastq-files-that-contain-only-wolbachia-mapping-reads-2019-05-30-" id="user-content-create-subsets-of-all-fastq-files-that-contain-only-wolbachia-mapping-reads-2019-05-30-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Single-End FASTQs
                      <a name="user-content-bashanalysis_subsetfastq_se">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Single-End FASTQs " class="anchor" href="#single-end-fastqs--1" id="user-content-single-end-fastqs--1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-5" id="user-content-input-sets-5">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='## wOo Darby 2012
SRR_ID_LIST="$INPUTS_DIR"/wOo_oochengi_2012_srr.list
OUTPUT_DIR="$READS_DIR"/wOo_oochengi_2012

## wDi Luck 2014
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2014_srr.list
OUTPUT_DIR="$READS_DIR"//wDi_luck_2014

## wDi Luck 2015
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wDi_luck_2015'>
                     <pre class="notranslate" lang="{bash,"><code>## wOo Darby 2012
SRR_ID_LIST="$INPUTS_DIR"/wOo_oochengi_2012_srr.list
OUTPUT_DIR="$READS_DIR"/wOo_oochengi_2012

## wDi Luck 2014
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2014_srr.list
OUTPUT_DIR="$READS_DIR"//wDi_luck_2014

## wDi Luck 2015
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wDi_luck_2015
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-6" id="user-content-commands-6">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='while read SRR
do
	echo -e ""$SEQTK_BIN_DIR"/seqtk subseq "$OUTPUT_DIR"/"$SRR"_1.fastq "$OUTPUT_DIR"/"$SRR".target_reads.list &gt; "$OUTPUT_DIR"/"$SRR"_1.subset.fastq" | qsub -P jdhotopp-lab -l mem_free=2G -N seqtk -wd "$OUTPUT_DIR"
done &lt; "$SRR_ID_LIST"'>
                     <pre class="notranslate" lang="{bash,"><code>while read SRR
do
	echo -e ""$SEQTK_BIN_DIR"/seqtk subseq "$OUTPUT_DIR"/"$SRR"_1.fastq "$OUTPUT_DIR"/"$SRR".target_reads.list &gt; "$OUTPUT_DIR"/"$SRR"_1.subset.fastq" | qsub -P jdhotopp-lab -l mem_free=2G -N seqtk -wd "$OUTPUT_DIR"
done &lt; "$SRR_ID_LIST"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Paired-End FASTQs
                      <a name="user-content-bashanalysis_subsetfastq_pe">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Paired-End FASTQs " class="anchor" href="#paired-end-fastqs--1" id="user-content-paired-end-fastqs--1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-6" id="user-content-input-sets-6">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='## wMel Darby 2014
SRR_ID_LIST=/wMel_darby_2014_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_darby_2014

## wMel Gutzwiller 2015
SRR_ID_LIST="$INPUTS_DIR"/wMel_gutzwiller_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_gutzwiller_2015

## wBm Grote 2017
SRR_ID_LIST="$INPUTS_DIR"/wBm_grote_2017_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_grote_2017

## wBm Chung 2019
SRR_ID_LIST="$INPUTS_DIR"/wBm_chung_2019_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_chung_2019'>
                     <pre class="notranslate" lang="{bash,"><code>## wMel Darby 2014
SRR_ID_LIST=/wMel_darby_2014_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_darby_2014

## wMel Gutzwiller 2015
SRR_ID_LIST="$INPUTS_DIR"/wMel_gutzwiller_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_gutzwiller_2015

## wBm Grote 2017
SRR_ID_LIST="$INPUTS_DIR"/wBm_grote_2017_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_grote_2017

## wBm Chung 2019
SRR_ID_LIST="$INPUTS_DIR"/wBm_chung_2019_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_chung_2019
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-7" id="user-content-commands-7">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='while read SRR
do

	echo -e ""$SEQTK_BIN_DIR"/seqtk subseq "$OUTPUT_DIR"/"$SRR"_1.fastq "$OUTPUT_DIR"/"$SRR".target_reads.list &gt; "$OUTPUT_DIR"/"$SRR"_1.subset.fastq" | qsub -P jdhotopp-lab -l mem_free=2G -N seqtk -wd "$OUTPUT_DIR"

	echo -e ""$SEQTK_BIN_DIR"/seqtk subseq "$OUTPUT_DIR"/"$SRR"_2.fastq "$OUTPUT_DIR"/"$SRR".target_reads.list &gt; "$OUTPUT_DIR"/"$SRR"_2.subset.fastq" | qsub -P jdhotopp-lab -l mem_free=2G -N seqtk -wd "$OUTPUT_DIR"

done &lt; "$SRR_ID_LIST"'>
                     <pre class="notranslate" lang="{bash,"><code>while read SRR
do

	echo -e ""$SEQTK_BIN_DIR"/seqtk subseq "$OUTPUT_DIR"/"$SRR"_1.fastq "$OUTPUT_DIR"/"$SRR".target_reads.list &gt; "$OUTPUT_DIR"/"$SRR"_1.subset.fastq" | qsub -P jdhotopp-lab -l mem_free=2G -N seqtk -wd "$OUTPUT_DIR"

	echo -e ""$SEQTK_BIN_DIR"/seqtk subseq "$OUTPUT_DIR"/"$SRR"_2.fastq "$OUTPUT_DIR"/"$SRR".target_reads.list &gt; "$OUTPUT_DIR"/"$SRR"_2.subset.fastq" | qsub -P jdhotopp-lab -l mem_free=2G -N seqtk -wd "$OUTPUT_DIR"

done &lt; "$SRR_ID_LIST"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Align FASTQ files containing only Wolbachia-mapping reads to combined host and Wolbachia reference while disallowing splicing (2019-06-01)
                      <a name="user-content-bashanalysis_realign">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Align FASTQ files containing only Wolbachia-mapping reads to combined host and Wolbachia reference while disallowing splicing (2019-06-01) " class="anchor" href="#align-fastq-files-containing-only-wolbachia-mapping-reads-to-combined-host-and-wolbachia-reference-while-disallowing-splicing-2019-06-01-" id="user-content-align-fastq-files-containing-only-wolbachia-mapping-reads-to-combined-host-and-wolbachia-reference-while-disallowing-splicing-2019-06-01-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Single-End FASTQs
                      <a name="user-content-bashanalysis_realign_se">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Single-End FASTQs " class="anchor" href="#single-end-fastqs--2" id="user-content-single-end-fastqs--2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-7" id="user-content-input-sets-7">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='THREADS=4

## wOo Darby 2012
REF_FNA="$REFERENCES_DIR"/wOo_oochengi_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wOo_oochengi_2012_srr.list
OUTPUT_DIR="$READS_DIR"/wOo_oochengi_2012/bam
STRANDEDNESS=""
PREFIX=wOo_darby_2012

## wDi Luck 2014
REF_FNA="$REFERENCES_DIR"/wDi_dimmitis_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2014_srr.list
OUTPUT_DIR="$READS_DIR"/wDi_luck_2014/bam
STRANDEDNESS=""
PREFIX=wDi_luck_2014

## wDi Luck 2015
REF_FNA="$REFERENCES_DIR"/wDi_dimmitis_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wDi_luck_2015/bam
STRANDEDNESS=""
PREFIX=wDi_luck_2015'>
                     <pre class="notranslate" lang="{bash,"><code>THREADS=4

## wOo Darby 2012
REF_FNA="$REFERENCES_DIR"/wOo_oochengi_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wOo_oochengi_2012_srr.list
OUTPUT_DIR="$READS_DIR"/wOo_oochengi_2012/bam
STRANDEDNESS=""
PREFIX=wOo_darby_2012

## wDi Luck 2014
REF_FNA="$REFERENCES_DIR"/wDi_dimmitis_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2014_srr.list
OUTPUT_DIR="$READS_DIR"/wDi_luck_2014/bam
STRANDEDNESS=""
PREFIX=wDi_luck_2014

## wDi Luck 2015
REF_FNA="$REFERENCES_DIR"/wDi_dimmitis_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wDi_luck_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wDi_luck_2015/bam
STRANDEDNESS=""
PREFIX=wDi_luck_2015
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-8" id="user-content-commands-8">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='while read SRR
do
	echo -e ""$HISAT2_BIN_DIR"/hisat2 -p "$THREADS" -x "$REF_FNA" "$STRANDEDNESS" -U "$READS_DIR"/"$PREFIX"/"$SRR"_1.subset.fastq --no-spliced-alignment -X 1000 | "$SAMTOOLS_BIN_DIR"/samtools view -bhSo "$OUTPUT_DIR"/"$SRR".bam -" | qsub -q threaded.q  -pe thread "$THREADS" -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd "$OUTPUT_DIR"
done &lt; "$SRR_ID_LIST"'>
                     <pre class="notranslate" lang="{bash,"><code>while read SRR
do
	echo -e ""$HISAT2_BIN_DIR"/hisat2 -p "$THREADS" -x "$REF_FNA" "$STRANDEDNESS" -U "$READS_DIR"/"$PREFIX"/"$SRR"_1.subset.fastq --no-spliced-alignment -X 1000 | "$SAMTOOLS_BIN_DIR"/samtools view -bhSo "$OUTPUT_DIR"/"$SRR".bam -" | qsub -q threaded.q  -pe thread "$THREADS" -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd "$OUTPUT_DIR"
done &lt; "$SRR_ID_LIST"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Paired-End FASTQs
                      <a name="user-content-bashanalysis_realign_pe">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Paired-End FASTQs " class="anchor" href="#paired-end-fastqs--2" id="user-content-paired-end-fastqs--2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-8" id="user-content-input-sets-8">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='THREADS=4

## wMel Darby 2014
REF_FNA="$REFERENCES_DIR"/wMel_dmelanogaster_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wMel_darby_2014_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_darby_2014/bam
STRANDEDNESS="--rna-strandness FR"
PREFIX=wMel_darby_2014

## wMel Gutzwiller 2015
REF_FNA="$REFERENCES_DIR"/wMel_dmelanogaster_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wMel_gutzwiller_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_gutzwiller_2015/bam
STRANDEDNESS="--rna-strandness RF"
PREFIX=wMel_gutzwiller_2015

## wBm Grote 2017
REF_FNA="$REFERENCES_DIR"/wBm_bmalayi_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wBm_grote_2017_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_grote_2017/bam
STRANDEDNESS=""
PREFIX=wBm_grote_2017

## wBm Chung 2019
REF_FNA="$REFERENCES_DIR"/wBm_bmalayi_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wBm_chung_2019_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_chung_2019/bam
STRANDEDNESS="--rna-strandness RF"
PREFIX=wBm_chung_2019'>
                     <pre class="notranslate" lang="{bash,"><code>THREADS=4

## wMel Darby 2014
REF_FNA="$REFERENCES_DIR"/wMel_dmelanogaster_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wMel_darby_2014_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_darby_2014/bam
STRANDEDNESS="--rna-strandness FR"
PREFIX=wMel_darby_2014

## wMel Gutzwiller 2015
REF_FNA="$REFERENCES_DIR"/wMel_dmelanogaster_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wMel_gutzwiller_2015_srr.list
OUTPUT_DIR="$READS_DIR"/wMel_gutzwiller_2015/bam
STRANDEDNESS="--rna-strandness RF"
PREFIX=wMel_gutzwiller_2015

## wBm Grote 2017
REF_FNA="$REFERENCES_DIR"/wBm_bmalayi_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wBm_grote_2017_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_grote_2017/bam
STRANDEDNESS=""
PREFIX=wBm_grote_2017

## wBm Chung 2019
REF_FNA="$REFERENCES_DIR"/wBm_bmalayi_combined.fna
SRR_ID_LIST="$INPUTS_DIR"/wBm_chung_2019_srr.list
OUTPUT_DIR="$READS_DIR"/wBm_chung_2019/bam
STRANDEDNESS="--rna-strandness RF"
PREFIX=wBm_chung_2019
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-9" id="user-content-commands-9">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='while read SRR
do
	echo -e ""$HISAT2_BIN_DIR"/hisat2 -p "$THREADS" -x "$REF_FNA" "$STRANDEDNESS" -1 "$READS_DIR"/"$PREFIX"/"$SRR"_1.subset.fastq  -2 "$READS_DIR"/"$PREFIX"/"$SRR"_2.subset.fastq --no-spliced-alignment -X 1000 | "$SAMTOOLS_BIN_DIR"/samtools view -bhSo "$OUTPUT_DIR"/"$SRR".bam -" | qsub -q threaded.q  -pe thread "$THREADS" -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd "$OUTPUT_DIR"
done &lt; "$SRR_ID_LIST"'>
                     <pre class="notranslate" lang="{bash,"><code>while read SRR
do
	echo -e ""$HISAT2_BIN_DIR"/hisat2 -p "$THREADS" -x "$REF_FNA" "$STRANDEDNESS" -1 "$READS_DIR"/"$PREFIX"/"$SRR"_1.subset.fastq  -2 "$READS_DIR"/"$PREFIX"/"$SRR"_2.subset.fastq --no-spliced-alignment -X 1000 | "$SAMTOOLS_BIN_DIR"/samtools view -bhSo "$OUTPUT_DIR"/"$SRR".bam -" | qsub -q threaded.q  -pe thread "$THREADS" -P jdhotopp-lab -l mem_free=20G -N hisat2 -wd "$OUTPUT_DIR"
done &lt; "$SRR_ID_LIST"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Sort BAM files (2019-06-01)
                      <a name="user-content-bashanalysis_sortbam">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Sort BAM files (2019-06-01) " class="anchor" href="#sort-bam-files-2019-06-01-" id="user-content-sort-bam-files-2019-06-01-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-9" id="user-content-input-sets-9">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='THREADS=4

## wOo Darby 2012
BAM_DIR="$READS_DIR"/wOo_darby_2012/bam

## wMel Darby 2014
BAM_DIR="$READS_DIR"/wMel_darby_2014/bam

## wDi Luck 2014
BAM_DIR="$READS_DIR"/wDi_luck_2014/bam

## wDi Luck 2015
BAM_DIR="$READS_DIR"/wDi_luck_2015/bam

## wMel Gutzwiller 2015
BAM_DIR="$READS_DIR"/wMel_gutzwiller_2015/bam

## wBm Grote 2017
BAM_DIR="$READS_DIR"/wBm_grote_2017/bam

## wBm Chung 2019
BAM_DIR="$READS_DIR"/wBm_chung_2019/bam'>
                     <pre class="notranslate" lang="{bash,"><code>THREADS=4

## wOo Darby 2012
BAM_DIR="$READS_DIR"/wOo_darby_2012/bam

## wMel Darby 2014
BAM_DIR="$READS_DIR"/wMel_darby_2014/bam

## wDi Luck 2014
BAM_DIR="$READS_DIR"/wDi_luck_2014/bam

## wDi Luck 2015
BAM_DIR="$READS_DIR"/wDi_luck_2015/bam

## wMel Gutzwiller 2015
BAM_DIR="$READS_DIR"/wMel_gutzwiller_2015/bam

## wBm Grote 2017
BAM_DIR="$READS_DIR"/wBm_grote_2017/bam

## wBm Chung 2019
BAM_DIR="$READS_DIR"/wBm_chung_2019/bam
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-10" id="user-content-commands-10">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='while read SRR
do
	echo -e ""$SAMTOOLS_BIN_DIR"/samtools sort -@ "$THREADS" -o "$BAM_DIR"/"$SRR".sortedbyposition.bam "$BAM_DIR"/"$SRR".bam" | qsub -q threaded.q -pe thread "$THREADS" -P jdhotopp-lab -l mem_free=2G -N samtools -wd "$BAM_DIR"
done &lt; "$SRR_ID_LIST"'>
                     <pre class="notranslate" lang="{bash,"><code>while read SRR
do
	echo -e ""$SAMTOOLS_BIN_DIR"/samtools sort -@ "$THREADS" -o "$BAM_DIR"/"$SRR".sortedbyposition.bam "$BAM_DIR"/"$SRR".bam" | qsub -q threaded.q -pe thread "$THREADS" -P jdhotopp-lab -l mem_free=2G -N samtools -wd "$BAM_DIR"
done &lt; "$SRR_ID_LIST"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Index BAM files (2019-06-01)
                      <a name="user-content-bashanalysis_indexbam">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Index BAM files (2019-06-01) " class="anchor" href="#index-bam-files-2019-06-01-" id="user-content-index-bam-files-2019-06-01-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-10" id="user-content-input-sets-10">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='THREADS=4

## wOo Darby 2012
BAM_DIR="$READS_DIR"/wOo_darby_2012/bam

## wMel Darby 2014
BAM_DIR="$READS_DIR"/wMel_darby_2014/bam

## wDi Luck 2014
BAM_DIR="$READS_DIR"/wDi_luck_2014/bam

## wDi Luck 2015
BAM_DIR="$READS_DIR"/wDi_luck_2015/bam

## wMel Gutzwiller 2015
BAM_DIR="$READS_DIR"/wMel_gutzwiller_2015/bam

## wBm Grote 2017
BAM_DIR="$READS_DIR"/wBm_grote_2017/bam

## wBm Chung 2019
BAM_DIR="$READS_DIR"/wBm_chung_2019/bam'>
                     <pre class="notranslate" lang="{bash,"><code>THREADS=4

## wOo Darby 2012
BAM_DIR="$READS_DIR"/wOo_darby_2012/bam

## wMel Darby 2014
BAM_DIR="$READS_DIR"/wMel_darby_2014/bam

## wDi Luck 2014
BAM_DIR="$READS_DIR"/wDi_luck_2014/bam

## wDi Luck 2015
BAM_DIR="$READS_DIR"/wDi_luck_2015/bam

## wMel Gutzwiller 2015
BAM_DIR="$READS_DIR"/wMel_gutzwiller_2015/bam

## wBm Grote 2017
BAM_DIR="$READS_DIR"/wBm_grote_2017/bam

## wBm Chung 2019
BAM_DIR="$READS_DIR"/wBm_chung_2019/bam
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-11" id="user-content-commands-11">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='while read SRR
do
	echo -e ""$SAMTOOLS_BIN_DIR"/samtools index -@ "$THREADS" "$BAM_DIR"/"$SRR".sortedbyposition.bam" | qsub -q threaded.q -pe thread "$THREADS" -P jdhotopp-lab -l mem_free=2G -N samtools -wd "$BAM_DIR"
done &lt; "$SRR_ID_LIST"'>
                     <pre class="notranslate" lang="{bash,"><code>while read SRR
do
	echo -e ""$SAMTOOLS_BIN_DIR"/samtools index -@ "$THREADS" "$BAM_DIR"/"$SRR".sortedbyposition.bam" | qsub -q threaded.q -pe thread "$THREADS" -P jdhotopp-lab -l mem_free=2G -N samtools -wd "$BAM_DIR"
done &lt; "$SRR_ID_LIST"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Quantify Wolbachia genes (2019-06-01)
                      <a name="user-content-bashanalysis_quant">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Quantify Wolbachia genes (2019-06-01) " class="anchor" href="#quantify-wolbachia-genes-2019-06-01-" id="user-content-quantify-wolbachia-genes-2019-06-01-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-11" id="user-content-input-sets-11">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='FEAT_TYPE="CDS"
ATTR_TYPE="ID"

## wOo Darby 2012
BAM_DIR="$READS_DIR"/wOo_darby_2012/bam
GFF3="$REFERENCES_DIR"/wOo.gff
STRANDED="no"

## wMel Darby 2014
BAM_DIR="$READS_DIR"/wMel_darby_2014/bam
GFF3="$REFERENCES_DIR"/wMel.gff
STRANDED="yes"

## wDi Luck 2014
BAM_DIR="$READS_DIR"/wDi_luck_2014/bam
GFF3="$REFERENCES_DIR"/wDi.gff
STRANDED="no"

## wDi Luck 2015
BAM_DIR="$READS_DIR"/wDi_luck_2015/bam
GFF3="$REFERENCES_DIR"/wDi.gff
STRANDED="no"

## wMel Gutzwiller 2015
BAM_DIR="$READS_DIR"/wMel_gutzwiller_2015/bam
GFF3="$REFERENCES_DIR"/wMel.gff
STRANDED="reverse"

## wBm Grote 2017
BAM_DIR="$READS_DIR"/wBm_grote_2017/bam
GFF3="$REFERENCES_DIR"/wBm.gff
STRANDED="no"

## wBm Chung 2019
BAM_DIR="$READS_DIR"/wBm_chung_2019/bam
GFF3="$REFERENCES_DIR"/wBm.gff
STRANDED="reverse"'>
                     <pre class="notranslate" lang="{bash,"><code>FEAT_TYPE="CDS"
ATTR_TYPE="ID"

## wOo Darby 2012
BAM_DIR="$READS_DIR"/wOo_darby_2012/bam
GFF3="$REFERENCES_DIR"/wOo.gff
STRANDED="no"

## wMel Darby 2014
BAM_DIR="$READS_DIR"/wMel_darby_2014/bam
GFF3="$REFERENCES_DIR"/wMel.gff
STRANDED="yes"

## wDi Luck 2014
BAM_DIR="$READS_DIR"/wDi_luck_2014/bam
GFF3="$REFERENCES_DIR"/wDi.gff
STRANDED="no"

## wDi Luck 2015
BAM_DIR="$READS_DIR"/wDi_luck_2015/bam
GFF3="$REFERENCES_DIR"/wDi.gff
STRANDED="no"

## wMel Gutzwiller 2015
BAM_DIR="$READS_DIR"/wMel_gutzwiller_2015/bam
GFF3="$REFERENCES_DIR"/wMel.gff
STRANDED="reverse"

## wBm Grote 2017
BAM_DIR="$READS_DIR"/wBm_grote_2017/bam
GFF3="$REFERENCES_DIR"/wBm.gff
STRANDED="no"

## wBm Chung 2019
BAM_DIR="$READS_DIR"/wBm_chung_2019/bam
GFF3="$REFERENCES_DIR"/wBm.gff
STRANDED="reverse"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-12" id="user-content-commands-12">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='mkdir -p "$WORKING_DIR"/fadu
for BAM in $(find "$BAM_DIR" -name "*sortedbyposition.bam")
do

	echo -e "export JULIA_DEPOT_PATH="$JULIA_DEPOT_PATH"\n"$JULIA_BIN_DIR"/julia "$FADU_BIN_DIR"/fadu.jl -g "$GFF3" -b "$BAM" -o "$WORKING_DIR"/fadu -s "$STRANDED" -f "$FEAT_TYPE" -a "$ATTR_TYPE"" | qsub -P jdhotopp-lab -l mem_free=5G -N fadu -wd "$WORKING_DIR"/fadu

done'>
                     <pre class="notranslate" lang="{bash,"><code>mkdir -p "$WORKING_DIR"/fadu
for BAM in $(find "$BAM_DIR" -name "*sortedbyposition.bam")
do

	echo -e "export JULIA_DEPOT_PATH="$JULIA_DEPOT_PATH"\n"$JULIA_BIN_DIR"/julia "$FADU_BIN_DIR"/fadu.jl -g "$GFF3" -b "$BAM" -o "$WORKING_DIR"/fadu -s "$STRANDED" -f "$FEAT_TYPE" -a "$ATTR_TYPE"" | qsub -P jdhotopp-lab -l mem_free=5G -N fadu -wd "$WORKING_DIR"/fadu

done
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Convert GFF to a mapping table (2019-06-07)
                      <a name="user-content-bashanalysis_gff2map">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Convert GFF to a mapping table (2019-06-07) " class="anchor" href="#convert-gff-to-a-mapping-table-2019-06-07-" id="user-content-convert-gff-to-a-mapping-table-2019-06-07-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-12" id="user-content-input-sets-12">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='## wBm
GFF3="$REFERENCES_DIR"/wBm.gff

## wDi
GFF3="$REFERENCES_DIR"/wDi.gff

## wMel
GFF3="$REFERENCES_DIR"/wMel.gff

## wOo
GFF3="$REFERENCES_DIR"/wOo.gff'>
                     <pre class="notranslate" lang="{bash,"><code>## wBm
GFF3="$REFERENCES_DIR"/wBm.gff

## wDi
GFF3="$REFERENCES_DIR"/wDi.gff

## wMel
GFF3="$REFERENCES_DIR"/wMel.gff

## wOo
GFF3="$REFERENCES_DIR"/wOo.gff
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-13" id="user-content-commands-13">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='"$R_BIN_DIR"/Rscript "$SCRIPTS_DIR"/gff3_to_map.R "$GFF_PATH"'>
                     <pre class="notranslate" lang="{bash,"><code>"$R_BIN_DIR"/Rscript "$SCRIPTS_DIR"/gff3_to_map.R "$GFF_PATH"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Create nucleotide coding sequence fasta files (2019-06-07)
                      <a name="user-content-bashanalysis_createcdsfasta">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Create nucleotide coding sequence fasta files (2019-06-07) " class="anchor" href="#create-nucleotide-coding-sequence-fasta-files-2019-06-07-" id="user-content-create-nucleotide-coding-sequence-fasta-files-2019-06-07-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-13" id="user-content-input-sets-13">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='FEAT_TYPE="CDS"
ATTR_TYPE="ID"

## wBm
FNA="$REFERENCES_DIR"/wBm.fna
GFF3="$REFERENCES_DIR"/wBm.gff

## wDi
FNA="$REFERENCES_DIR"/wDi.fna
GFF3="$REFERENCES_DIR"/wDi.gff

## wMel
FNA="$REFERENCES_DIR"/wMel.fna
GFF3="$REFERENCES_DIR"/wMel.gff

## wOo
FNA="$REFERENCES_DIR"/wOo.fna
GFF3="$REFERENCES_DIR"/wOo.gff'>
                     <pre class="notranslate" lang="{bash,"><code>FEAT_TYPE="CDS"
ATTR_TYPE="ID"

## wBm
FNA="$REFERENCES_DIR"/wBm.fna
GFF3="$REFERENCES_DIR"/wBm.gff

## wDi
FNA="$REFERENCES_DIR"/wDi.fna
GFF3="$REFERENCES_DIR"/wDi.gff

## wMel
FNA="$REFERENCES_DIR"/wMel.fna
GFF3="$REFERENCES_DIR"/wMel.gff

## wOo
FNA="$REFERENCES_DIR"/wOo.fna
GFF3="$REFERENCES_DIR"/wOo.gff
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-14" id="user-content-commands-14">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='"$SCRIPTS_DIR"/createnuccdsfasta.sh -n "$FNA" -g "$GFF3" -f "$FEAT_TYPE" -i "$ATTR_TYPE" &gt; "$(echo -e "$FNA" | sed "s/[.]fna$/.cds.fna/g")"'>
                     <pre class="notranslate" lang="{bash,"><code>"$SCRIPTS_DIR"/createnuccdsfasta.sh -n "$FNA" -g "$GFF3" -f "$FEAT_TYPE" -i "$ATTR_TYPE" &gt; "$(echo -e "$FNA" | sed "s/[.]fna$/.cds.fna/g")"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Identify InterPro descriptions and GO terms for Wolbachia transcripts (2019-06-07)
                      <a name="user-content-bashanalysis_interproscan">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Identify InterPro descriptions and GO terms for Wolbachia transcripts (2019-06-07) " class="anchor" href="#identify-interpro-descriptions-and-go-terms-for-wolbachia-transcripts-2019-06-07-" id="user-content-identify-interpro-descriptions-and-go-terms-for-wolbachia-transcripts-2019-06-07-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-14" id="user-content-input-sets-14">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='SEQ_TYPE=n
THREADS=4

## wBm
CDS_FNA="$REFERENCES_DIR"/wBm.cds.fna
GFF3="$REFERENCES_DIR"/wBm.gff

## wDi
CDS_FNA="$REFERENCES_DIR"/wDi.cds.fna
GFF3="$REFERENCES_DIR"/wDi.gff

## wMel
CDS_FNA="$REFERENCES_DIR"/wMel.cds.fna
GFF3="$REFERENCES_DIR"/wMel.gff

## wOo
CDS_FNA="$REFERENCES_DIR"/wOo.cds.fna
GFF3="$REFERENCES_DIR"/wOo.gff'>
                     <pre class="notranslate" lang="{bash,"><code>SEQ_TYPE=n
THREADS=4

## wBm
CDS_FNA="$REFERENCES_DIR"/wBm.cds.fna
GFF3="$REFERENCES_DIR"/wBm.gff

## wDi
CDS_FNA="$REFERENCES_DIR"/wDi.cds.fna
GFF3="$REFERENCES_DIR"/wDi.gff

## wMel
CDS_FNA="$REFERENCES_DIR"/wMel.cds.fna
GFF3="$REFERENCES_DIR"/wMel.gff

## wOo
CDS_FNA="$REFERENCES_DIR"/wOo.cds.fna
GFF3="$REFERENCES_DIR"/wOo.gff
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-15" id="user-content-commands-15">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='echo -e "export LD_LIBRARY_PATH="$PYTHON_LIB_PATH":"$LD_LIBRARY_PATH"\n"$INTERPROSCAN_BIN_DIR"/interproscan.sh -i "$CDS_FNA" -f tsv -o "$CDS_FNA".interproscan.tsv --seqtype "$SEQ_TYPE" --goterms --iprlookup" | qsub -P jdhotopp-lab -q threaded.q  -pe thread "$THREADS" -l mem_free=20G -N interproscan -wd "$(dirname "$CDS_FNA")"'>
                     <pre class="notranslate" lang="{bash,"><code>echo -e "export LD_LIBRARY_PATH="$PYTHON_LIB_PATH":"$LD_LIBRARY_PATH"\n"$INTERPROSCAN_BIN_DIR"/interproscan.sh -i "$CDS_FNA" -f tsv -o "$CDS_FNA".interproscan.tsv --seqtype "$SEQ_TYPE" --goterms --iprlookup" | qsub -P jdhotopp-lab -q threaded.q  -pe thread "$THREADS" -l mem_free=20G -N interproscan -wd "$(dirname "$CDS_FNA")"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Transcriptomics meta-analysis
                      <a name="user-content-ranalysis">
                      </a>
                     </h1>
                     <a aria-label="Permalink: Transcriptomics meta-analysis " class="anchor" href="#transcriptomics-meta-analysis-" id="user-content-transcriptomics-meta-analysis-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Create counts table
                      <a name="user-content-ranalysis_counts">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Create counts table " class="anchor" href="#create-counts-table-" id="user-content-create-counts-table-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Set R inputs
                      <a name="user-content-ranalysis_setinputs">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Set R inputs " class="anchor" href="#set-r-inputs-" id="user-content-set-r-inputs-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     R inputs from bash should be:
                    </p>
                    <p dir="auto">
                     SAMPLE_MAP.PATH = "$INPUTS_DIR"/study_sample_map.tsv.txt
                     <br/>
                     RAW_COUNTS.DIR = "$WORKING_DIR"/fadu
                     <br/>
                     OUTPUT.DIR = "$WORKING_DIR"
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='SAMPLE_MAP.PATH  &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/study_sample_map.tsv.txt"
RAW_COUNTS.DIR &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/fadu/"
OUTPUT.DIR &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/"'>
                     <pre class="notranslate" lang="{R}"><code>SAMPLE_MAP.PATH  &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/study_sample_map.tsv.txt"
RAW_COUNTS.DIR &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/fadu/"
OUTPUT.DIR &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      View sessionInfo
                      <a name="user-content-ranalysis_sessioninfo">
                      </a>
                     </h3>
                     <a aria-label="Permalink: View sessionInfo " class="anchor" href="#view-sessioninfo-" id="user-content-view-sessioninfo-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="sessionInfo()">
                     <pre class="notranslate" lang="{R,"><code>sessionInfo()
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Combine count files into a single table for each study
                      <a name="user-content-ranalysis_counts_maketable">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Combine count files into a single table for each study " class="anchor" href="#combine-count-files-into-a-single-table-for-each-study-" id="user-content-combine-count-files-into-a-single-table-for-each-study-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='sample_map &lt;- read.delim(SAMPLE_MAP.PATH,header = T,sep = "\t")
counts.files &lt;- list.files(RAW_COUNTS.DIR,
                           full.names = T,
                           pattern=".*counts.txt")

counts.list &lt;- list(wOo_darby_2012="",
                    wMel_darby_2014="",
                    wDi_luck_2014="",
                    wDi_luck_2015="",
                    wMel_gutzwiller_2015="",
                    wBm_grote_2017="",
                    wBm_chung_2019="")


for(i in 1:length(counts.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(counts.list)[i],]
  samples &lt;- unique(sample_map.subset$sample_identifier)
  
  genes &lt;- read.delim(grep(sample_map.subset$sra_id[i],counts.files,value = T),header = T)[,1]
  counts.list[[i]] &lt;- as.data.frame(matrix(0,
                                           nrow = length(genes),
                                           ncol = length(samples)))
  rownames(counts.list[[i]]) &lt;- genes
  colnames(counts.list[[i]]) &lt;- samples
                                    
  for(j in 1:ncol(counts.list[[i]])){
    srr &lt;- sample_map.subset$sra_id[sample_map.subset$sample_identifier == colnames(counts.list[[i]])[j]]
    while(length(srr) &gt; 0){
      raw_counts.file &lt;- read.delim(paste0(RAW_COUNTS.DIR,"/",srr[1],".sortedbyposition.counts.txt"),
                                    header = T)
      raw_counts.file &lt;- raw_counts.file[match(rownames(counts.list[[i]]),raw_counts.file[,1]),]
      counts.list[[i]][,j] &lt;- counts.list[[i]][,j] + raw_counts.file$counts
      
      srr &lt;- srr[-1]
    }
  }
  write.table(counts.list[[i]],
              paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_counts.tsv"),
              row.names = T,
              col.names = T,
              quote = F,
              sep = "\t")
}'>
                     <pre class="notranslate" lang="{R}"><code>sample_map &lt;- read.delim(SAMPLE_MAP.PATH,header = T,sep = "\t")
counts.files &lt;- list.files(RAW_COUNTS.DIR,
                           full.names = T,
                           pattern=".*counts.txt")

counts.list &lt;- list(wOo_darby_2012="",
                    wMel_darby_2014="",
                    wDi_luck_2014="",
                    wDi_luck_2015="",
                    wMel_gutzwiller_2015="",
                    wBm_grote_2017="",
                    wBm_chung_2019="")


for(i in 1:length(counts.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(counts.list)[i],]
  samples &lt;- unique(sample_map.subset$sample_identifier)
  
  genes &lt;- read.delim(grep(sample_map.subset$sra_id[i],counts.files,value = T),header = T)[,1]
  counts.list[[i]] &lt;- as.data.frame(matrix(0,
                                           nrow = length(genes),
                                           ncol = length(samples)))
  rownames(counts.list[[i]]) &lt;- genes
  colnames(counts.list[[i]]) &lt;- samples
                                    
  for(j in 1:ncol(counts.list[[i]])){
    srr &lt;- sample_map.subset$sra_id[sample_map.subset$sample_identifier == colnames(counts.list[[i]])[j]]
    while(length(srr) &gt; 0){
      raw_counts.file &lt;- read.delim(paste0(RAW_COUNTS.DIR,"/",srr[1],".sortedbyposition.counts.txt"),
                                    header = T)
      raw_counts.file &lt;- raw_counts.file[match(rownames(counts.list[[i]]),raw_counts.file[,1]),]
      counts.list[[i]][,j] &lt;- counts.list[[i]][,j] + raw_counts.file$counts
      
      srr &lt;- srr[-1]
    }
  }
  write.table(counts.list[[i]],
              paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_counts.tsv"),
              row.names = T,
              col.names = T,
              quote = F,
              sep = "\t")
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Convert InterProScan output to geneinfo file (2019-06-07)
                      <a name="user-content-ranalysis_iprscan2geneinfo">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Convert InterProScan output to geneinfo file (2019-06-07) " class="anchor" href="#convert-interproscan-output-to-geneinfo-file-2019-06-07-" id="user-content-convert-interproscan-output-to-geneinfo-file-2019-06-07-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Input Sets:
                     </h4>
                     <a aria-label="Permalink: Input Sets:" class="anchor" href="#input-sets-15" id="user-content-input-sets-15">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='SEQ_TYPE=n
THREADS=4

## wBm
INTERPROSCAN_OUTPUT="$REFERENCES_DIR"/wBm.cds.fna.interproscan.tsv
COUNTS="$WORKING_DIR"/wBm_chung_2019_counts.tsv

## wDi
INTERPROSCAN_OUTPUT="$REFERENCES_DIR"/wDi.cds.fna.interproscan.tsv
COUNTS="$WORKING_DIR"/wDi_luck_2015_counts.tsv

## wMel
INTERPROSCAN_OUTPUT="$REFERENCES_DIR"/wMel.cds.fna.interproscan.tsv
COUNTS="$WORKING_DIR"/wMel_darby_2014_counts.tsv

## wOo
INTERPROSCAN_OUTPUT="$REFERENCES_DIR"/wOo.cds.fna.interproscan.tsv
COUNTS="$WORKING_DIR"/wOo_darby_2012_counts.tsv'>
                     <pre class="notranslate" lang="{bash,"><code>SEQ_TYPE=n
THREADS=4

## wBm
INTERPROSCAN_OUTPUT="$REFERENCES_DIR"/wBm.cds.fna.interproscan.tsv
COUNTS="$WORKING_DIR"/wBm_chung_2019_counts.tsv

## wDi
INTERPROSCAN_OUTPUT="$REFERENCES_DIR"/wDi.cds.fna.interproscan.tsv
COUNTS="$WORKING_DIR"/wDi_luck_2015_counts.tsv

## wMel
INTERPROSCAN_OUTPUT="$REFERENCES_DIR"/wMel.cds.fna.interproscan.tsv
COUNTS="$WORKING_DIR"/wMel_darby_2014_counts.tsv

## wOo
INTERPROSCAN_OUTPUT="$REFERENCES_DIR"/wOo.cds.fna.interproscan.tsv
COUNTS="$WORKING_DIR"/wOo_darby_2012_counts.tsv
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-16" id="user-content-commands-16">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='"$R_BIN_DIR"/Rscript "$SCRIPTS_DIR"/interproscan2geneinfo.R "$INTERPROSCAN_OUTPUT" "$COUNTS"'>
                     <pre class="notranslate" lang="{bash,"><code>"$R_BIN_DIR"/Rscript "$SCRIPTS_DIR"/interproscan2geneinfo.R "$INTERPROSCAN_OUTPUT" "$COUNTS"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Differential expression and WGCNA analyses (2019-06-07)
                      <a name="user-content-ranalysis_iprscan2geneinfo">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Differential expression and WGCNA analyses (2019-06-07) " class="anchor" href="#differential-expression-and-wgcna-analyses-2019-06-07-" id="user-content-differential-expression-and-wgcna-analyses-2019-06-07-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Set R inputs
                      <a name="user-content-ranalysis_de_setinputs">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Set R inputs " class="anchor" href="#set-r-inputs--1" id="user-content-set-r-inputs--1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     R inputs from bash should be:
                    </p>
                    <p dir="auto">
                     SAMPLE_MAP.PATH = "$INPUTS_DIR"/study_sample_map.tsv.txt
                     <br/>
                     COUNTS.DIR = "$WORKING_DIR"
                     <br/>
                     RAW_COUNTS.DIR = "$WORKING_DIR"/fadu
                     <br/>
                     GFF3MAP.DIR = "$REFERENCES_DIR"
                     <br/>
                     OUTPUT.DIR = "$WORKING_DIR"
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='SAMPLE_MAP.PATH  &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/study_sample_map.tsv.txt"
COUNTS.DIR &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/"
GENEINFO.DIR &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/references"
GFF3MAP.DIR &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/references"
RAW_COUNTS.DIR &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/fadu"
OUTPUT.DIR &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/"'>
                     <pre class="notranslate" lang="{R}"><code>SAMPLE_MAP.PATH  &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/study_sample_map.tsv.txt"
COUNTS.DIR &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/"
GENEINFO.DIR &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/references"
GFF3MAP.DIR &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/references"
RAW_COUNTS.DIR &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/fadu"
OUTPUT.DIR &lt;- "Z:/EBMAL/mchung_dir/wolbachia_metatranscriptome_analysis/"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Load R packages and view sessionInfo
                      <a name="user-content-ranalysis_de_sessioninfo">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Load R packages and view sessionInfo " class="anchor" href="#load-r-packages-and-view-sessioninfo-" id="user-content-load-r-packages-and-view-sessioninfo-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="library(cowplot)
library(edgeR)
library(emdbook)
library(FactoMineR)
library(dendextend)
library(ggdendro)
library(ggplot2)
library(gplots)
library(gridExtra)
library(gtools)
library(pvclust)
library(reshape2)
library(vegan)
library(WGCNA)

sessionInfo()">
                     <pre class="notranslate" lang="{R,"><code>library(cowplot)
library(edgeR)
library(emdbook)
library(FactoMineR)
library(dendextend)
library(ggdendro)
library(ggplot2)
library(gplots)
library(gridExtra)
library(gtools)
library(pvclust)
library(reshape2)
library(vegan)
library(WGCNA)

sessionInfo()
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="R version 3.5.1 (2018-07-02)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] WGCNA_1.66            fastcluster_1.1.25    dynamicTreeCut_1.63-1 vegan_2.5-4           lattice_0.20-35       permute_0.9-4        
 [7] reshape2_1.4.3        pvclust_2.0-0         gtools_3.8.1          gridExtra_2.3         gplots_3.0.1.1        ggdendro_0.1-20      
[13] dendextend_1.9.0      FactoMineR_1.41       emdbook_1.3.11        edgeR_3.24.3          limma_3.38.3          cowplot_0.9.4        
[19] ggplot2_3.2.1        

loaded via a namespace (and not attached):
  [1] colorspace_1.4-1      class_7.3-14          modeltools_0.2-22     mclust_5.4.3          htmlTable_1.13.1      base64enc_0.1-3      
  [7] rstudioapi_0.10       flexmix_2.3-15        bit64_0.9-7           AnnotationDbi_1.44.0  mvtnorm_1.0-10        codetools_0.2-15     
 [13] splines_3.5.1         leaps_3.0             doParallel_1.0.14     impute_1.56.0         robustbase_0.93-4     knitr_1.22           
 [19] Formula_1.2-3         cluster_2.0.7-1       kernlab_0.9-27        GO.db_3.7.0           rrcov_1.4-7           compiler_3.5.1       
 [25] backports_1.1.3       assertthat_0.2.1      Matrix_1.2-14         lazyeval_0.2.2        acepack_1.4.1         htmltools_0.3.6      
 [31] tools_3.5.1           coda_0.19-2           gtable_0.3.0          glue_1.3.1            dplyr_0.8.3           Rcpp_1.0.1           
 [37] bbmle_1.0.20          Biobase_2.42.0        trimcluster_0.1-2.1   gdata_2.18.0          preprocessCore_1.44.0 nlme_3.1-137         
 [43] iterators_1.0.10      fpc_2.1-11.1          xfun_0.6              stringr_1.4.0         DEoptimR_1.0-8        MASS_7.3-50          
 [49] scales_1.0.0          parallel_3.5.1        RColorBrewer_1.1-2    yaml_2.2.0            memoise_1.1.0         rpart_4.1-13         
 [55] latticeExtra_0.6-28   stringi_1.4.3         RSQLite_2.1.1         S4Vectors_0.20.1      pcaPP_1.9-73          foreach_1.4.4        
 [61] checkmate_1.9.1       caTools_1.17.1.2      BiocGenerics_0.28.0   rlang_0.4.0           pkgconfig_2.0.2       prabclus_2.2-7       
 [67] bitops_1.0-6          matrixStats_0.54.0    purrr_0.3.2           htmlwidgets_1.3       bit_1.1-14            tidyselect_0.2.5     
 [73] robust_0.4-18         plyr_1.8.4            magrittr_1.5          R6_2.4.0              IRanges_2.16.0        Hmisc_4.2-0          
 [79] fit.models_0.5-14     DBI_1.0.0             pillar_1.3.1          whisker_0.3-2         foreign_0.8-70        withr_2.1.2          
 [85] mgcv_1.8-24           survival_2.42-3       scatterplot3d_0.3-41  nnet_7.3-12           tibble_2.1.1          crayon_1.3.4         
 [91] KernSmooth_2.23-15    viridis_0.5.1         locfit_1.5-9.1        grid_3.5.1            data.table_1.12.2     blob_1.1.1           
 [97] digest_0.6.18         diptest_0.75-7        flashClust_1.01-2     numDeriv_2016.8-1     stats4_3.5.1          munsell_0.5.0        
[103] viridisLite_0.3.0">
                     <pre class="notranslate" lang="{R,"><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] WGCNA_1.66            fastcluster_1.1.25    dynamicTreeCut_1.63-1 vegan_2.5-4           lattice_0.20-35       permute_0.9-4        
 [7] reshape2_1.4.3        pvclust_2.0-0         gtools_3.8.1          gridExtra_2.3         gplots_3.0.1.1        ggdendro_0.1-20      
[13] dendextend_1.9.0      FactoMineR_1.41       emdbook_1.3.11        edgeR_3.24.3          limma_3.38.3          cowplot_0.9.4        
[19] ggplot2_3.2.1        

loaded via a namespace (and not attached):
  [1] colorspace_1.4-1      class_7.3-14          modeltools_0.2-22     mclust_5.4.3          htmlTable_1.13.1      base64enc_0.1-3      
  [7] rstudioapi_0.10       flexmix_2.3-15        bit64_0.9-7           AnnotationDbi_1.44.0  mvtnorm_1.0-10        codetools_0.2-15     
 [13] splines_3.5.1         leaps_3.0             doParallel_1.0.14     impute_1.56.0         robustbase_0.93-4     knitr_1.22           
 [19] Formula_1.2-3         cluster_2.0.7-1       kernlab_0.9-27        GO.db_3.7.0           rrcov_1.4-7           compiler_3.5.1       
 [25] backports_1.1.3       assertthat_0.2.1      Matrix_1.2-14         lazyeval_0.2.2        acepack_1.4.1         htmltools_0.3.6      
 [31] tools_3.5.1           coda_0.19-2           gtable_0.3.0          glue_1.3.1            dplyr_0.8.3           Rcpp_1.0.1           
 [37] bbmle_1.0.20          Biobase_2.42.0        trimcluster_0.1-2.1   gdata_2.18.0          preprocessCore_1.44.0 nlme_3.1-137         
 [43] iterators_1.0.10      fpc_2.1-11.1          xfun_0.6              stringr_1.4.0         DEoptimR_1.0-8        MASS_7.3-50          
 [49] scales_1.0.0          parallel_3.5.1        RColorBrewer_1.1-2    yaml_2.2.0            memoise_1.1.0         rpart_4.1-13         
 [55] latticeExtra_0.6-28   stringi_1.4.3         RSQLite_2.1.1         S4Vectors_0.20.1      pcaPP_1.9-73          foreach_1.4.4        
 [61] checkmate_1.9.1       caTools_1.17.1.2      BiocGenerics_0.28.0   rlang_0.4.0           pkgconfig_2.0.2       prabclus_2.2-7       
 [67] bitops_1.0-6          matrixStats_0.54.0    purrr_0.3.2           htmlwidgets_1.3       bit_1.1-14            tidyselect_0.2.5     
 [73] robust_0.4-18         plyr_1.8.4            magrittr_1.5          R6_2.4.0              IRanges_2.16.0        Hmisc_4.2-0          
 [79] fit.models_0.5-14     DBI_1.0.0             pillar_1.3.1          whisker_0.3-2         foreign_0.8-70        withr_2.1.2          
 [85] mgcv_1.8-24           survival_2.42-3       scatterplot3d_0.3-41  nnet_7.3-12           tibble_2.1.1          crayon_1.3.4         
 [91] KernSmooth_2.23-15    viridis_0.5.1         locfit_1.5-9.1        grid_3.5.1            data.table_1.12.2     blob_1.1.1           
 [97] digest_0.6.18         diptest_0.75-7        flashClust_1.01-2     numDeriv_2016.8-1     stats4_3.5.1          munsell_0.5.0        
[103] viridisLite_0.3.0
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Load R functions
                      <a name="user-content-ranalysis_de_loadfunctions">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Load R functions " class="anchor" href="#load-r-functions-" id="user-content-load-r-functions-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='g_legend&lt;-function(a.gplot){ 
  tmp &lt;- ggplot_gtable(ggplot_build(a.gplot)) 
  leg &lt;- which(sapply(tmp$grobs, function(x) x$name) == "guide-box") 
  legend &lt;- tmp$grobs[[leg]] 
  return(legend)
} 

get_dendro_structure &lt;- function(result){
  structure &lt;- hang.dendrogram(as.dendrogram(result$hclust))
  structure &lt;- capture.output(str(structure))
  structure &lt;- structure[grepl("leaf", structure)]
  structure &lt;- as.numeric(as.character(substr(structure, regexpr("h=", structure ) + 3, regexpr("  )", structure))))
  return(structure)
}
get_dendro_data &lt;- function(result){
  dendro.data &lt;- dendro_data(result$hclust)
  dendro.data &lt;- dendro.data$segments[which(dendro.data$segments$y == dendro.data$segments$yend),]
  for(i in 1:nrow(dendro.data)){
    dendro.data$minx[i] &lt;- min(c(dendro.data$x[i], dendro.data$xend[i]))
  }
  dendro.data &lt;- dendro.data[order(as.numeric(as.character(dendro.data$y)), as.numeric(as.character(dendro.data$minx))),]
  return(dendro.data)
}
get_dendro_bootstraps &lt;- function(dendro_data){
  bootstrap.positions &lt;- as.data.frame(matrix(nrow = length(dendro_data$y[duplicated(dendro_data$y)]),
                                              ncol = 2))
  for(i in 1:length(dendro_data$y[duplicated(dendro_data$y)])){
    dendro_data.subset &lt;- dendro_data[which(dendro_data$y == dendro_data$y[duplicated(dendro_data$y)][i]),]
    bootstrap.positions[i,1] &lt;- unique(dendro_data.subset$x)
    bootstrap.positions[i,2] &lt;- unique(dendro_data.subset$y)
  }
  return(bootstrap.positions)
}
get_heatmap_separators &lt;- function(vector){
  sep &lt;- c()
  for(i in 2:length(unique(vector))){
    sep[length(sep) + 1] &lt;- min(which(vector == unique(vector)[i])) - 1
  }
  return(sep)
}
formalize_study_names &lt;- function(vector){
  vector &lt;- gsub("wOo_darby_2012", "Darby et al., 2012",vector)
  vector &lt;- gsub("wMel_darby_2014", "Darby et al., 2014",vector)
  vector &lt;- gsub("wDi_luck_2014", "Luck et al., 2014",vector)
  vector &lt;- gsub("wDi_luck_2015", "Luck et al., 2015",vector)
  vector &lt;- gsub("wMel_gutzwiller_2015", "Gutzwiller et al., 2015",vector)
  vector &lt;- gsub("wBm_grote_2017", "Grote et al., 2017",vector)
  vector &lt;- gsub("wBm_chung_2019", "Chung et al., 2019",vector)
  vector &lt;- factor(vector,levels=c("Darby et al., 2012",
                                   "Darby et al., 2014",
                                   "Luck et al., 2014",
                                   "Luck et al., 2015",
                                   "Gutzwiller et al., 2015",
                                   "Grote et al., 2017",
                                   "Chung et al., 2019"))
  return(vector)
}
find_soft_power &lt;- function(sft){
  df &lt;- as.data.frame(cbind(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2]))
  y &lt;- -sign(sft$fitIndices[,3])*sft$fitIndices[,2]
  dy &lt;- diff(y) 
  softpower &lt;- which(abs(diff(y)) &lt; 0.05)[1]
  return(softpower)
}
eigengene_invert_id &lt;- function(tpm.de.wgcna, mergedColors, mergedMEs){
  tpm.de.wgcna$invert &lt;- T
  tpm.de.wgcna$module &lt;- mergedColors
  for(i in 1:nrow(tpm.de.wgcna)){
    if(cor(t(tpm.de.wgcna[i,1:(ncol(tpm.de.wgcna) - 2)]), mergedMEs[,which(colnames(mergedMEs) == paste0("ME",tpm.de.wgcna$module[i]))], method = "pearson") &gt; 0){
      tpm.de.wgcna$invert[i] &lt;- F
    }
  }
  return(tpm.de.wgcna)
}
wgcna_heatmap_reorder &lt;- function(tpm.de.wgcna){
  clusters &lt;- as.data.frame(table(tpm.de.wgcna$module))
  clusters &lt;- clusters[order(-clusters[,2]),1]
  
  tpm.de.wgcna.reordered &lt;- as.data.frame(matrix(nrow = 0,
                                                 ncol = ncol(tpm.de.wgcna)))
  for(i in 1:length(clusters)){
    tpm.de.wgcna.reordered &lt;- as.data.frame(rbind(tpm.de.wgcna.reordered,
                                                  tpm.de.wgcna[tpm.de.wgcna$module == clusters[i] &amp; tpm.de.wgcna$invert == F,],
                                                  tpm.de.wgcna[tpm.de.wgcna$module == clusters[i] &amp; tpm.de.wgcna$invert == T,]))
  }
  return(tpm.de.wgcna.reordered)
}
functionaltermenrichment &lt;- function(genes, geneinfo){
  for(i in 1:ncol(geneinfo)){geneinfo[,i] &lt;- as.character(geneinfo[,i])}
  geneinfo$interpro_description[which(is.na(geneinfo$interpro_description))] &lt;- "No InterPro entry"
  geneinfo$go_biologicalprocess[which(is.na(geneinfo$go_biologicalprocess))] &lt;- "No GO terms for biological process"
  geneinfo$go_cellularcomponent[which(is.na(geneinfo$go_cellularcomponent))] &lt;- "No GO terms for cellular component"
  geneinfo$go_molecularfunction[which(is.na(geneinfo$go_molecularfunction))] &lt;- "No GO terms for molecular function"
  
  functionalterms.list &lt;- list(ipr=as.data.frame(table(unlist(strsplit(paste(geneinfo$interpro_description, collapse = "|"),  split = "[|]")))),
                               gobio=as.data.frame(table(unlist(strsplit(paste(geneinfo$go_biologicalprocess, collapse = "|"),  split = "[|]")))),
                               gocell=as.data.frame(table(unlist(strsplit(paste(geneinfo$go_cellularcomponent, collapse = "|"),  split = "[|]")))),
                               gomol=as.data.frame(table(unlist(strsplit(paste(geneinfo$go_molecularfunction, collapse = "|"),  split = "[|]")))))
  
  geneinfo.subset &lt;- geneinfo[geneinfo$gene %in% genes,]
  term &lt;- c()
  clusteroccurences &lt;- c()
  genomeoccurences &lt;- c()
  pvalue &lt;- c()
  correctedpvalue &lt;- c()
  oddsratio &lt;- c()

  functionalterms.list.subset &lt;- list(ipr=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$interpro_description, collapse = "|"),  split = "[|]")))),
                                      gobio=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$go_biologicalprocess, collapse = "|"),  split = "[|]")))),
                                      gocell=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$go_cellularcomponent, collapse = "|"),  split = "[|]")))),
                                      gomol=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$go_molecularfunction, collapse = "|"),  split = "[|]")))))
  
  for(i in 1:length(functionalterms.list)){
    for(j in 1:nrow(functionalterms.list[[i]])){
      freq.all &lt;- functionalterms.list[[i]][j,2]
      freq.subset &lt;- ifelse(functionalterms.list[[i]][j,1] %in% functionalterms.list.subset[[i]][,1],
                            functionalterms.list.subset[[i]][functionalterms.list.subset[[i]][,1] == as.character(functionalterms.list[[i]][j,1]),2],
                            0)
      genes.all &lt;- nrow(geneinfo)
      genes.subset &lt;- nrow(geneinfo.subset)

      fisherexact.matrix &lt;- matrix(c(freq.subset, freq.all - freq.subset,
                                     genes.subset - freq.subset, genes.all - genes.subset - freq.all + freq.subset),
                                   nrow = 2,
                                   ncol = 2)
      fisher.test &lt;- fisher.test(fisherexact.matrix)
      
      term[length(term) + 1] &lt;- as.character(functionalterms.list[[i]][j,1])
      clusteroccurences[length(clusteroccurences) + 1] &lt;- as.numeric(as.character(freq.subset))
      genomeoccurences[length(genomeoccurences) + 1] &lt;- as.numeric(as.character(freq.all))
      pvalue[length(pvalue) + 1] &lt;- as.numeric(as.character(fisher.test$p.value))
      correctedpvalue[length(correctedpvalue) + 1] &lt;- p.adjust(as.numeric(as.character(fisher.test$p.value)), method = "fdr", n = nrow(functionalterms.list[[i]]))
      oddsratio[length(oddsratio) + 1] &lt;- as.numeric(as.character(fisher.test$estimate))
    }
  }
  
  terms.df &lt;- as.data.frame(cbind(term,
                                  clusteroccurences,
                                  genomeoccurences,
                                  pvalue,
                                  correctedpvalue,
                                  oddsratio))
  terms.df &lt;- terms.df[order(as.numeric(as.character(terms.df$pvalue))),]
  return(terms.df)
}'>
                     <pre class="notranslate" lang="{R,"><code>g_legend&lt;-function(a.gplot){ 
  tmp &lt;- ggplot_gtable(ggplot_build(a.gplot)) 
  leg &lt;- which(sapply(tmp$grobs, function(x) x$name) == "guide-box") 
  legend &lt;- tmp$grobs[[leg]] 
  return(legend)
} 

get_dendro_structure &lt;- function(result){
  structure &lt;- hang.dendrogram(as.dendrogram(result$hclust))
  structure &lt;- capture.output(str(structure))
  structure &lt;- structure[grepl("leaf", structure)]
  structure &lt;- as.numeric(as.character(substr(structure, regexpr("h=", structure ) + 3, regexpr("  )", structure))))
  return(structure)
}
get_dendro_data &lt;- function(result){
  dendro.data &lt;- dendro_data(result$hclust)
  dendro.data &lt;- dendro.data$segments[which(dendro.data$segments$y == dendro.data$segments$yend),]
  for(i in 1:nrow(dendro.data)){
    dendro.data$minx[i] &lt;- min(c(dendro.data$x[i], dendro.data$xend[i]))
  }
  dendro.data &lt;- dendro.data[order(as.numeric(as.character(dendro.data$y)), as.numeric(as.character(dendro.data$minx))),]
  return(dendro.data)
}
get_dendro_bootstraps &lt;- function(dendro_data){
  bootstrap.positions &lt;- as.data.frame(matrix(nrow = length(dendro_data$y[duplicated(dendro_data$y)]),
                                              ncol = 2))
  for(i in 1:length(dendro_data$y[duplicated(dendro_data$y)])){
    dendro_data.subset &lt;- dendro_data[which(dendro_data$y == dendro_data$y[duplicated(dendro_data$y)][i]),]
    bootstrap.positions[i,1] &lt;- unique(dendro_data.subset$x)
    bootstrap.positions[i,2] &lt;- unique(dendro_data.subset$y)
  }
  return(bootstrap.positions)
}
get_heatmap_separators &lt;- function(vector){
  sep &lt;- c()
  for(i in 2:length(unique(vector))){
    sep[length(sep) + 1] &lt;- min(which(vector == unique(vector)[i])) - 1
  }
  return(sep)
}
formalize_study_names &lt;- function(vector){
  vector &lt;- gsub("wOo_darby_2012", "Darby et al., 2012",vector)
  vector &lt;- gsub("wMel_darby_2014", "Darby et al., 2014",vector)
  vector &lt;- gsub("wDi_luck_2014", "Luck et al., 2014",vector)
  vector &lt;- gsub("wDi_luck_2015", "Luck et al., 2015",vector)
  vector &lt;- gsub("wMel_gutzwiller_2015", "Gutzwiller et al., 2015",vector)
  vector &lt;- gsub("wBm_grote_2017", "Grote et al., 2017",vector)
  vector &lt;- gsub("wBm_chung_2019", "Chung et al., 2019",vector)
  vector &lt;- factor(vector,levels=c("Darby et al., 2012",
                                   "Darby et al., 2014",
                                   "Luck et al., 2014",
                                   "Luck et al., 2015",
                                   "Gutzwiller et al., 2015",
                                   "Grote et al., 2017",
                                   "Chung et al., 2019"))
  return(vector)
}
find_soft_power &lt;- function(sft){
  df &lt;- as.data.frame(cbind(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2]))
  y &lt;- -sign(sft$fitIndices[,3])*sft$fitIndices[,2]
  dy &lt;- diff(y) 
  softpower &lt;- which(abs(diff(y)) &lt; 0.05)[1]
  return(softpower)
}
eigengene_invert_id &lt;- function(tpm.de.wgcna, mergedColors, mergedMEs){
  tpm.de.wgcna$invert &lt;- T
  tpm.de.wgcna$module &lt;- mergedColors
  for(i in 1:nrow(tpm.de.wgcna)){
    if(cor(t(tpm.de.wgcna[i,1:(ncol(tpm.de.wgcna) - 2)]), mergedMEs[,which(colnames(mergedMEs) == paste0("ME",tpm.de.wgcna$module[i]))], method = "pearson") &gt; 0){
      tpm.de.wgcna$invert[i] &lt;- F
    }
  }
  return(tpm.de.wgcna)
}
wgcna_heatmap_reorder &lt;- function(tpm.de.wgcna){
  clusters &lt;- as.data.frame(table(tpm.de.wgcna$module))
  clusters &lt;- clusters[order(-clusters[,2]),1]
  
  tpm.de.wgcna.reordered &lt;- as.data.frame(matrix(nrow = 0,
                                                 ncol = ncol(tpm.de.wgcna)))
  for(i in 1:length(clusters)){
    tpm.de.wgcna.reordered &lt;- as.data.frame(rbind(tpm.de.wgcna.reordered,
                                                  tpm.de.wgcna[tpm.de.wgcna$module == clusters[i] &amp; tpm.de.wgcna$invert == F,],
                                                  tpm.de.wgcna[tpm.de.wgcna$module == clusters[i] &amp; tpm.de.wgcna$invert == T,]))
  }
  return(tpm.de.wgcna.reordered)
}
functionaltermenrichment &lt;- function(genes, geneinfo){
  for(i in 1:ncol(geneinfo)){geneinfo[,i] &lt;- as.character(geneinfo[,i])}
  geneinfo$interpro_description[which(is.na(geneinfo$interpro_description))] &lt;- "No InterPro entry"
  geneinfo$go_biologicalprocess[which(is.na(geneinfo$go_biologicalprocess))] &lt;- "No GO terms for biological process"
  geneinfo$go_cellularcomponent[which(is.na(geneinfo$go_cellularcomponent))] &lt;- "No GO terms for cellular component"
  geneinfo$go_molecularfunction[which(is.na(geneinfo$go_molecularfunction))] &lt;- "No GO terms for molecular function"
  
  functionalterms.list &lt;- list(ipr=as.data.frame(table(unlist(strsplit(paste(geneinfo$interpro_description, collapse = "|"),  split = "[|]")))),
                               gobio=as.data.frame(table(unlist(strsplit(paste(geneinfo$go_biologicalprocess, collapse = "|"),  split = "[|]")))),
                               gocell=as.data.frame(table(unlist(strsplit(paste(geneinfo$go_cellularcomponent, collapse = "|"),  split = "[|]")))),
                               gomol=as.data.frame(table(unlist(strsplit(paste(geneinfo$go_molecularfunction, collapse = "|"),  split = "[|]")))))
  
  geneinfo.subset &lt;- geneinfo[geneinfo$gene %in% genes,]
  term &lt;- c()
  clusteroccurences &lt;- c()
  genomeoccurences &lt;- c()
  pvalue &lt;- c()
  correctedpvalue &lt;- c()
  oddsratio &lt;- c()

  functionalterms.list.subset &lt;- list(ipr=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$interpro_description, collapse = "|"),  split = "[|]")))),
                                      gobio=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$go_biologicalprocess, collapse = "|"),  split = "[|]")))),
                                      gocell=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$go_cellularcomponent, collapse = "|"),  split = "[|]")))),
                                      gomol=as.data.frame(table(unlist(strsplit(paste(geneinfo.subset$go_molecularfunction, collapse = "|"),  split = "[|]")))))
  
  for(i in 1:length(functionalterms.list)){
    for(j in 1:nrow(functionalterms.list[[i]])){
      freq.all &lt;- functionalterms.list[[i]][j,2]
      freq.subset &lt;- ifelse(functionalterms.list[[i]][j,1] %in% functionalterms.list.subset[[i]][,1],
                            functionalterms.list.subset[[i]][functionalterms.list.subset[[i]][,1] == as.character(functionalterms.list[[i]][j,1]),2],
                            0)
      genes.all &lt;- nrow(geneinfo)
      genes.subset &lt;- nrow(geneinfo.subset)

      fisherexact.matrix &lt;- matrix(c(freq.subset, freq.all - freq.subset,
                                     genes.subset - freq.subset, genes.all - genes.subset - freq.all + freq.subset),
                                   nrow = 2,
                                   ncol = 2)
      fisher.test &lt;- fisher.test(fisherexact.matrix)
      
      term[length(term) + 1] &lt;- as.character(functionalterms.list[[i]][j,1])
      clusteroccurences[length(clusteroccurences) + 1] &lt;- as.numeric(as.character(freq.subset))
      genomeoccurences[length(genomeoccurences) + 1] &lt;- as.numeric(as.character(freq.all))
      pvalue[length(pvalue) + 1] &lt;- as.numeric(as.character(fisher.test$p.value))
      correctedpvalue[length(correctedpvalue) + 1] &lt;- p.adjust(as.numeric(as.character(fisher.test$p.value)), method = "fdr", n = nrow(functionalterms.list[[i]]))
      oddsratio[length(oddsratio) + 1] &lt;- as.numeric(as.character(fisher.test$estimate))
    }
  }
  
  terms.df &lt;- as.data.frame(cbind(term,
                                  clusteroccurences,
                                  genomeoccurences,
                                  pvalue,
                                  correctedpvalue,
                                  oddsratio))
  terms.df &lt;- terms.df[order(as.numeric(as.character(terms.df$pvalue))),]
  return(terms.df)
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Process counts files and calculate TPM values
                      <a name="user-content-ranalysis_de_counts_readcounts">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Process counts files and calculate TPM values  " class="anchor" href="#process-counts-files-and-calculate-tpm-values--" id="user-content-process-counts-files-and-calculate-tpm-values--">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Read counts table into a list
                      <a name="user-content-ranalysis_de_counts_readcounts">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Read counts table into a list " class="anchor" href="#read-counts-table-into-a-list-" id="user-content-read-counts-table-into-a-list-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='sample_map &lt;- read.delim(SAMPLE_MAP.PATH)

counts.list &lt;- list(wOo_darby_2012="",
                    wMel_darby_2014="",
                    wDi_luck_2014="",
                    wDi_luck_2015="",
                    wMel_gutzwiller_2015="",
                    wBm_grote_2017="",
                    wBm_chung_2019="")
for(i in 1:length(counts.list)){
  counts.list[[i]] &lt;- read.delim(paste0(COUNTS.DIR,"/",names(counts.list)[i],"_counts.tsv"),
                                 row.names = 1)
}'>
                     <pre class="notranslate" lang="{R,"><code>sample_map &lt;- read.delim(SAMPLE_MAP.PATH)

counts.list &lt;- list(wOo_darby_2012="",
                    wMel_darby_2014="",
                    wDi_luck_2014="",
                    wDi_luck_2015="",
                    wMel_gutzwiller_2015="",
                    wBm_grote_2017="",
                    wBm_chung_2019="")
for(i in 1:length(counts.list)){
  counts.list[[i]] &lt;- read.delim(paste0(COUNTS.DIR,"/",names(counts.list)[i],"_counts.tsv"),
                                 row.names = 1)
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Exclude non-protein-coding genes from counts table
                      <a name="user-content-ranalysis_de_counts_removenonproteingenes">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Exclude non-protein-coding genes from counts table " class="anchor" href="#exclude-non-protein-coding-genes-from-counts-table-" id="user-content-exclude-non-protein-coding-genes-from-counts-table-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(counts.list)){
  species &lt;- substr(names(counts.list)[i],1,regexpr("_",names(counts.list[i]))-1)
  if(species != "wDi"){
    map &lt;- read.delim(paste0(GFF3MAP.DIR,"/",species,".gff.map"),
                      header = T,
                      sep = "\t")
    map &lt;- map[map$gene_biotype == "protein_coding" &amp; !is.na(map$gene_biotype),]
    map$ID &lt;- gsub(".*[|]cds","cds",map$ID)
    counts.list[[i]] &lt;- counts.list[[i]][rownames(counts.list[[i]]) %in% map$ID,]
  }
}'>
                     <pre class="notranslate" lang="{R,"><code>for(i in 1:length(counts.list)){
  species &lt;- substr(names(counts.list)[i],1,regexpr("_",names(counts.list[i]))-1)
  if(species != "wDi"){
    map &lt;- read.delim(paste0(GFF3MAP.DIR,"/",species,".gff.map"),
                      header = T,
                      sep = "\t")
    map &lt;- map[map$gene_biotype == "protein_coding" &amp; !is.na(map$gene_biotype),]
    map$ID &lt;- gsub(".*[|]cds","cds",map$ID)
    counts.list[[i]] &lt;- counts.list[[i]][rownames(counts.list[[i]]) %in% map$ID,]
  }
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Exclude genes without Agilent SureSelect probes from counts table in Chung et al 2019 study
                      <a name="user-content-ranalysis_de_counts_removenoprobegenes">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Exclude genes without Agilent SureSelect probes from counts table in Chung et al 2019 study " class="anchor" href="#exclude-genes-without-agilent-sureselect-probes-from-counts-table-in-chung-et-al-2019-study-" id="user-content-exclude-genes-without-agilent-sureselect-probes-from-counts-table-in-chung-et-al-2019-study-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='exclude.genes &lt;- c("cds28","cds31","cds63","cds64","cds74","cds99","cds118","cds127","cds135","cds168","cds169","cds170","cds171","cds185","cds240","cds245","cds246","cds307","cds309","cds321","cds322","cds324","cds328","cds354","cds359","cds360","cds363","cds371","cds372","cds380","cds426","cds427","cds431","cds444","cds452","cds453","cds466","cds490","cds491","cds492","cds493","cds500","cds543","cds573","cds581","cds607","cds608","cds614","cds622","cds635","cds636","cds638","cds643","cds657","cds658","cds733","cds734","cds737","cds746","cds750","cds774","cds775","cds784","cds785","cds787","cds805","cds816","cds817","cds822","cds828","cds851","cds853","cds870","cds871","cds872","cds874","cds886","cds909","cds910","cds932","cds957","cds963","cds965","cds978","cds979")

counts.list$wBm_chung_2019 &lt;- counts.list$wBm_chung_2019[!(rownames(counts.list$wBm_chung_2019) %in% exclude.genes),]'>
                     <pre class="notranslate" lang="{R,"><code>exclude.genes &lt;- c("cds28","cds31","cds63","cds64","cds74","cds99","cds118","cds127","cds135","cds168","cds169","cds170","cds171","cds185","cds240","cds245","cds246","cds307","cds309","cds321","cds322","cds324","cds328","cds354","cds359","cds360","cds363","cds371","cds372","cds380","cds426","cds427","cds431","cds444","cds452","cds453","cds466","cds490","cds491","cds492","cds493","cds500","cds543","cds573","cds581","cds607","cds608","cds614","cds622","cds635","cds636","cds638","cds643","cds657","cds658","cds733","cds734","cds737","cds746","cds750","cds774","cds775","cds784","cds785","cds787","cds805","cds816","cds817","cds822","cds828","cds851","cds853","cds870","cds871","cds872","cds874","cds886","cds909","cds910","cds932","cds957","cds963","cds965","cds978","cds979")

counts.list$wBm_chung_2019 &lt;- counts.list$wBm_chung_2019[!(rownames(counts.list$wBm_chung_2019) %in% exclude.genes),]
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Print the number of reads mapping to protein-coding genes for each sample in each study
                      <a name="user-content-ranalysis_de_counts_countscolsum">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Print the number of reads mapping to protein-coding genes for each sample in each study " class="anchor" href="#print-the-number-of-reads-mapping-to-protein-coding-genes-for-each-sample-in-each-study-" id="user-content-print-the-number-of-reads-mapping-to-protein-coding-genes-for-each-sample-in-each-study-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for(i in 1:length(counts.list)){
  print(names(counts.list)[i])
  print(colSums(counts.list[[i]]))
}">
                     <pre class="notranslate" lang="{R,"><code>for(i in 1:length(counts.list)){
  print(names(counts.list)[i])
  print(colSums(counts.list[[i]]))
}
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='[1] "wOo_darby_2012"
  darby_woo_adultmale_a   darby_woo_adultmale_b darby_woo_adultfemale_a darby_woo_adultfemale_b 
               26067.08                83692.20               146642.65                80723.69 
[1] "wMel_darby_2014"
  darby_wmel_doxy_a   darby_wmel_doxy_b   darby_wmel_doxy_c darby_wmel_nodoxy_c darby_wmel_nodoxy_a darby_wmel_nodoxy_b 
          120344.75            72768.46            73974.65            64291.14            42960.91            82329.35 
[1] "wDi_luck_2014"
         luck_wdi_L3_a          luck_wdi_L4_a   luck_wdi_adultmale_a luck_wdi_adultfemale_a          luck_wdi_MF_a 
                495.32                1238.56                3788.02               12840.55               27118.30 
[1] "wDi_luck_2015"
 luck_wdi_adultfemale_bodywall_a  luck_wdi_adultfemale_bodywall_b      luck_wdi_adultfemale_head_a      luck_wdi_adultfemale_head_b 
                         6691.13                          8436.58                          4045.55                          6019.27 
luck_wdi_adultfemale_intestine_a luck_wdi_adultfemale_intestine_b    luck_wdi_adultfemale_uterus_a    luck_wdi_adultfemale_uterus_b 
                           14.34                            11.18                          1785.09                           188.19 
   luck_wdi_adultmale_bodywall_a    luck_wdi_adultmale_bodywall_b   luck_wdi_adultmale_intestine_a      luck_wdi_adultmale_testis_a 
                        14444.09                         10506.72                            36.70                            13.34 
[1] "wMel_gutzwiller_2015"
                 gutzwiller_wmel_embryo0to2hr_a                  gutzwiller_wmel_embryo0to2hr_b                  gutzwiller_wmel_embryo2to4hr_a 
                                      114790.66                                        85945.65                                       110702.61 
                 gutzwiller_wmel_embryo2to4hr_b                gutzwiller_wmel_embryo10to12hr_a                gutzwiller_wmel_embryo10to12hr_b 
                                       53392.96                                       188273.14                                       134967.66 
               gutzwiller_wmel_embryo12to14hr_a                gutzwiller_wmel_embryo12to14hr_b                gutzwiller_wmel_embryo18to20hr_a 
                                      137773.04                                        40542.10                                       177066.53 
               gutzwiller_wmel_embryo18to20hr_b                gutzwiller_wmel_embryo22to24hr_a                gutzwiller_wmel_embryo22to24hr_b 
                                      148100.78                                       192603.87                                        18862.07 
                           gutzwiller_wmel_L1_a                            gutzwiller_wmel_L1_b                            gutzwiller_wmel_L2_a 
                                      153643.74                                       128799.29                                        65018.60 
                           gutzwiller_wmel_L2_b               gutzwiller_wmel_L3_12hrpostmolt_a               gutzwiller_wmel_L3_12hrpostmolt_b 
                                       55155.07                                        72087.91                                        45411.64 
        gutzwiller_wmel_L3_darkbluegutPS_1to2_a         gutzwiller_wmel_L3_darkbluegutPS_1to2_b        gutzwiller_wmel_L3_lightbluegutPS_3to6_a 
                                      122280.34                                        90161.56                                       152077.10 
       gutzwiller_wmel_L3_lightbluegutPS_3to6_b            gutzwiller_wmel_L3_cleargutPS_7to9_a            gutzwiller_wmel_L3_cleargutPS_7to9_b 
                                      112893.91                                       241950.76                                        76163.54 
                          gutzwiller_wmel_WPP_a                           gutzwiller_wmel_WPP_b             gutzwiller_wmel_pupae_12hrpostWPP_a 
                                      178456.20                                       115149.63                                       162560.29 
            gutzwiller_wmel_pupae_12hrpostWPP_b             gutzwiller_wmel_pupae_24hrpostWPP_a             gutzwiller_wmel_pupae_24hrpostWPP_b 
                                       92488.90                                       112776.08                                        48500.17 
           gutzwiller_wmel_pupae_2dayspostWPP_a            gutzwiller_wmel_pupae_2dayspostWPP_b            gutzwiller_wmel_pupae_3dayspostWPP_a 
                                      106883.82                                        60639.46                                       203627.16 
           gutzwiller_wmel_pupae_3dayspostWPP_b            gutzwiller_wmel_pupae_4dayspostWPP_a            gutzwiller_wmel_pupae_4dayspostWPP_b 
                                       76254.69                                       105137.01                                        95285.08 
 gutzwiller_wmel_adultfemale_1dayposteclosion_a  gutzwiller_wmel_adultfemale_1dayposteclosion_b  gutzwiller_wmel_adultfemale_5dayposteclosion_a 
                                      254985.87                                       140491.93                                       221664.24 
 gutzwiller_wmel_adultfemale_5dayposteclosion_b  gutzwiller_wmel_adultfemale_5dayposteclosion_c  gutzwiller_wmel_adultfemale_5dayposteclosion_d 
                                      111503.67                                       103197.92                                       180473.06 
 gutzwiller_wmel_adultfemale_5dayposteclosion_f gutzwiller_wmel_adultfemale_30dayposteclosion_a gutzwiller_wmel_adultfemale_30dayposteclosion_b 
                                      195493.19                                       341687.94                                       244695.15 
   gutzwiller_wmel_adultmale_1dayposteclosion_a    gutzwiller_wmel_adultmale_1dayposteclosion_b    gutzwiller_wmel_adultmale_5dayposteclosion_a 
                                      225874.52                                       137847.49                                       417013.04 
   gutzwiller_wmel_adultmale_5dayposteclosion_b    gutzwiller_wmel_adultmale_5dayposteclosion_c    gutzwiller_wmel_adultmale_5dayposteclosion_d 
                                      136930.51                                       328285.55                                       222262.97 
   gutzwiller_wmel_adultmale_5dayposteclosion_i    gutzwiller_wmel_adultmale_5dayposteclosion_j   gutzwiller_wmel_adultmale_30dayposteclosion_a 
                                       88320.71                                        93761.47                                       325975.98 
  gutzwiller_wmel_adultmale_30dayposteclosion_b 
                                      321449.11 
[1] "wBm_grote_2017"
                  grote_wbm_L4_a                   grote_wbm_L4_b   grote_wbm_30dpi_immaturemale_a   grote_wbm_30dpi_immaturemale_b 
                        22374.59                          6178.84                         12843.80                          9650.02 
grote_wbm_30dpi_immaturefemale_a grote_wbm_30dpi_immaturefemale_b   grote_wbm_42dpi_immaturemale_a   grote_wbm_42dpi_immaturemale_b 
                         9080.46                         14064.82                         13019.66                          8636.27 
grote_wbm_42dpi_immaturefemale_a grote_wbm_42dpi_immaturefemale_b     grote_wbm_120dpi_adultmale_a     grote_wbm_120dpi_adultmale_b 
                         6721.62                         13460.39                         16643.58                          8373.13 
  grote_wbm_120dpi_adultfemale_a   grote_wbm_120dpi_adultfemale_b 
                        31289.42                         19151.90 
[1] "wBm_chung_2019"
               chung_wbm_vector18hpi_a                chung_wbm_vector18hpi_b                 chung_wbm_vector4dpi_a 
                             481351.25                              625028.83                              240693.73 
                chung_wbm_vector4dpi_b                 chung_wbm_vector8dpi_a                 chung_wbm_vector8dpi_b 
                             777934.24                             3218498.46                             3987678.06 
                chung_wbm_mammal1dpi_a                 chung_wbm_mammal1dpi_b                 chung_wbm_mammal2dpi_a 
                             159809.14                              103367.39                               53272.07 
                chung_wbm_mammal2dpi_b                 chung_wbm_mammal3dpi_a                 chung_wbm_mammal3dpi_b 
                             123961.25                              109225.39                              122560.27 
                chung_wbm_mammal4dpi_a                 chung_wbm_mammal4dpi_b                 chung_wbm_mammal8dpi_a 
                             120910.97                              121522.47                              284943.40 
                chung_wbm_mammal8dpi_b   chung_wbm_mammal20dpi_immaturemale_a   chung_wbm_mammal20dpi_immaturemale_b 
                             221120.86                              108117.16                               78688.09 
chung_wbm_mammal24dpi_immaturefemale_a chung_wbm_mammal24dpi_immaturefemale_b                  chung_wbm_adultmale_a 
                           35276890.78                              192206.48                               23790.09 
                 chung_wbm_adultmale_b                chung_wbm_adultfemale_a                chung_wbm_adultfemale_b 
                              43446.49                            18840207.11                               10561.16 
               chung_wbm_adultfemale_c                chung_wbm_adultfemale_d                     chung_wbm_embryo_a 
                              13947.67                            24661817.10                                8525.69 
                    chung_wbm_embryo_b                 chung_wbm_immatureMF_a                 chung_wbm_immatureMF_b 
                              14704.98                               65069.68                               57976.58 
                  chung_wbm_matureMF_a                   chung_wbm_matureMF_b 
                              46554.09                               51856.71 
'>
                     <pre class="notranslate" lang="{R,"><code>[1] "wOo_darby_2012"
  darby_woo_adultmale_a   darby_woo_adultmale_b darby_woo_adultfemale_a darby_woo_adultfemale_b 
               26067.08                83692.20               146642.65                80723.69 
[1] "wMel_darby_2014"
  darby_wmel_doxy_a   darby_wmel_doxy_b   darby_wmel_doxy_c darby_wmel_nodoxy_c darby_wmel_nodoxy_a darby_wmel_nodoxy_b 
          120344.75            72768.46            73974.65            64291.14            42960.91            82329.35 
[1] "wDi_luck_2014"
         luck_wdi_L3_a          luck_wdi_L4_a   luck_wdi_adultmale_a luck_wdi_adultfemale_a          luck_wdi_MF_a 
                495.32                1238.56                3788.02               12840.55               27118.30 
[1] "wDi_luck_2015"
 luck_wdi_adultfemale_bodywall_a  luck_wdi_adultfemale_bodywall_b      luck_wdi_adultfemale_head_a      luck_wdi_adultfemale_head_b 
                         6691.13                          8436.58                          4045.55                          6019.27 
luck_wdi_adultfemale_intestine_a luck_wdi_adultfemale_intestine_b    luck_wdi_adultfemale_uterus_a    luck_wdi_adultfemale_uterus_b 
                           14.34                            11.18                          1785.09                           188.19 
   luck_wdi_adultmale_bodywall_a    luck_wdi_adultmale_bodywall_b   luck_wdi_adultmale_intestine_a      luck_wdi_adultmale_testis_a 
                        14444.09                         10506.72                            36.70                            13.34 
[1] "wMel_gutzwiller_2015"
                 gutzwiller_wmel_embryo0to2hr_a                  gutzwiller_wmel_embryo0to2hr_b                  gutzwiller_wmel_embryo2to4hr_a 
                                      114790.66                                        85945.65                                       110702.61 
                 gutzwiller_wmel_embryo2to4hr_b                gutzwiller_wmel_embryo10to12hr_a                gutzwiller_wmel_embryo10to12hr_b 
                                       53392.96                                       188273.14                                       134967.66 
               gutzwiller_wmel_embryo12to14hr_a                gutzwiller_wmel_embryo12to14hr_b                gutzwiller_wmel_embryo18to20hr_a 
                                      137773.04                                        40542.10                                       177066.53 
               gutzwiller_wmel_embryo18to20hr_b                gutzwiller_wmel_embryo22to24hr_a                gutzwiller_wmel_embryo22to24hr_b 
                                      148100.78                                       192603.87                                        18862.07 
                           gutzwiller_wmel_L1_a                            gutzwiller_wmel_L1_b                            gutzwiller_wmel_L2_a 
                                      153643.74                                       128799.29                                        65018.60 
                           gutzwiller_wmel_L2_b               gutzwiller_wmel_L3_12hrpostmolt_a               gutzwiller_wmel_L3_12hrpostmolt_b 
                                       55155.07                                        72087.91                                        45411.64 
        gutzwiller_wmel_L3_darkbluegutPS_1to2_a         gutzwiller_wmel_L3_darkbluegutPS_1to2_b        gutzwiller_wmel_L3_lightbluegutPS_3to6_a 
                                      122280.34                                        90161.56                                       152077.10 
       gutzwiller_wmel_L3_lightbluegutPS_3to6_b            gutzwiller_wmel_L3_cleargutPS_7to9_a            gutzwiller_wmel_L3_cleargutPS_7to9_b 
                                      112893.91                                       241950.76                                        76163.54 
                          gutzwiller_wmel_WPP_a                           gutzwiller_wmel_WPP_b             gutzwiller_wmel_pupae_12hrpostWPP_a 
                                      178456.20                                       115149.63                                       162560.29 
            gutzwiller_wmel_pupae_12hrpostWPP_b             gutzwiller_wmel_pupae_24hrpostWPP_a             gutzwiller_wmel_pupae_24hrpostWPP_b 
                                       92488.90                                       112776.08                                        48500.17 
           gutzwiller_wmel_pupae_2dayspostWPP_a            gutzwiller_wmel_pupae_2dayspostWPP_b            gutzwiller_wmel_pupae_3dayspostWPP_a 
                                      106883.82                                        60639.46                                       203627.16 
           gutzwiller_wmel_pupae_3dayspostWPP_b            gutzwiller_wmel_pupae_4dayspostWPP_a            gutzwiller_wmel_pupae_4dayspostWPP_b 
                                       76254.69                                       105137.01                                        95285.08 
 gutzwiller_wmel_adultfemale_1dayposteclosion_a  gutzwiller_wmel_adultfemale_1dayposteclosion_b  gutzwiller_wmel_adultfemale_5dayposteclosion_a 
                                      254985.87                                       140491.93                                       221664.24 
 gutzwiller_wmel_adultfemale_5dayposteclosion_b  gutzwiller_wmel_adultfemale_5dayposteclosion_c  gutzwiller_wmel_adultfemale_5dayposteclosion_d 
                                      111503.67                                       103197.92                                       180473.06 
 gutzwiller_wmel_adultfemale_5dayposteclosion_f gutzwiller_wmel_adultfemale_30dayposteclosion_a gutzwiller_wmel_adultfemale_30dayposteclosion_b 
                                      195493.19                                       341687.94                                       244695.15 
   gutzwiller_wmel_adultmale_1dayposteclosion_a    gutzwiller_wmel_adultmale_1dayposteclosion_b    gutzwiller_wmel_adultmale_5dayposteclosion_a 
                                      225874.52                                       137847.49                                       417013.04 
   gutzwiller_wmel_adultmale_5dayposteclosion_b    gutzwiller_wmel_adultmale_5dayposteclosion_c    gutzwiller_wmel_adultmale_5dayposteclosion_d 
                                      136930.51                                       328285.55                                       222262.97 
   gutzwiller_wmel_adultmale_5dayposteclosion_i    gutzwiller_wmel_adultmale_5dayposteclosion_j   gutzwiller_wmel_adultmale_30dayposteclosion_a 
                                       88320.71                                        93761.47                                       325975.98 
  gutzwiller_wmel_adultmale_30dayposteclosion_b 
                                      321449.11 
[1] "wBm_grote_2017"
                  grote_wbm_L4_a                   grote_wbm_L4_b   grote_wbm_30dpi_immaturemale_a   grote_wbm_30dpi_immaturemale_b 
                        22374.59                          6178.84                         12843.80                          9650.02 
grote_wbm_30dpi_immaturefemale_a grote_wbm_30dpi_immaturefemale_b   grote_wbm_42dpi_immaturemale_a   grote_wbm_42dpi_immaturemale_b 
                         9080.46                         14064.82                         13019.66                          8636.27 
grote_wbm_42dpi_immaturefemale_a grote_wbm_42dpi_immaturefemale_b     grote_wbm_120dpi_adultmale_a     grote_wbm_120dpi_adultmale_b 
                         6721.62                         13460.39                         16643.58                          8373.13 
  grote_wbm_120dpi_adultfemale_a   grote_wbm_120dpi_adultfemale_b 
                        31289.42                         19151.90 
[1] "wBm_chung_2019"
               chung_wbm_vector18hpi_a                chung_wbm_vector18hpi_b                 chung_wbm_vector4dpi_a 
                             481351.25                              625028.83                              240693.73 
                chung_wbm_vector4dpi_b                 chung_wbm_vector8dpi_a                 chung_wbm_vector8dpi_b 
                             777934.24                             3218498.46                             3987678.06 
                chung_wbm_mammal1dpi_a                 chung_wbm_mammal1dpi_b                 chung_wbm_mammal2dpi_a 
                             159809.14                              103367.39                               53272.07 
                chung_wbm_mammal2dpi_b                 chung_wbm_mammal3dpi_a                 chung_wbm_mammal3dpi_b 
                             123961.25                              109225.39                              122560.27 
                chung_wbm_mammal4dpi_a                 chung_wbm_mammal4dpi_b                 chung_wbm_mammal8dpi_a 
                             120910.97                              121522.47                              284943.40 
                chung_wbm_mammal8dpi_b   chung_wbm_mammal20dpi_immaturemale_a   chung_wbm_mammal20dpi_immaturemale_b 
                             221120.86                              108117.16                               78688.09 
chung_wbm_mammal24dpi_immaturefemale_a chung_wbm_mammal24dpi_immaturefemale_b                  chung_wbm_adultmale_a 
                           35276890.78                              192206.48                               23790.09 
                 chung_wbm_adultmale_b                chung_wbm_adultfemale_a                chung_wbm_adultfemale_b 
                              43446.49                            18840207.11                               10561.16 
               chung_wbm_adultfemale_c                chung_wbm_adultfemale_d                     chung_wbm_embryo_a 
                              13947.67                            24661817.10                                8525.69 
                    chung_wbm_embryo_b                 chung_wbm_immatureMF_a                 chung_wbm_immatureMF_b 
                              14704.98                               65069.68                               57976.58 
                  chung_wbm_matureMF_a                   chung_wbm_matureMF_b 
                              46554.09                               51856.71 

</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Create TPM table
                      <a name="user-content-ranalysis_de_counts_tpm">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Create TPM table " class="anchor" href="#create-tpm-table-" id="user-content-create-tpm-table-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='tpm.list &lt;- counts.list

for(i in 1:length(tpm.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(counts.list)[i] &amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]
  srr &lt;- sample_map.subset$sra_id[sample_map.subset$sample_identifier == colnames(counts.list[[i]])[1]]
  raw_counts.file &lt;- read.delim(paste0(RAW_COUNTS.DIR,"/",srr[1],".sortedbyposition.counts.txt"),
                                      header = T)
  genelength &lt;- raw_counts.file$uniq_len[match(rownames(tpm.list[[i]]),raw_counts.file$featureID)]
  
  for(j in 1:ncol(tpm.list[[i]])){
    tpm.list[[i]][,j] &lt;- tpm.list[[i]][,j]/genelength
    tpm.list[[i]][,j] &lt;- tpm.list[[i]][,j]/(sum(tpm.list[[i]][,j])/1000000)
  }
  
  write.table(tpm.list[[i]],
            paste0(OUTPUT.DIR,"/",names(tpm.list)[i],"_tpm.tsv"),
            row.names = T,
            col.names = T,
            quote = F,
            sep = "\t")
}'>
                     <pre class="notranslate" lang="{R,"><code>tpm.list &lt;- counts.list

for(i in 1:length(tpm.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(counts.list)[i] &amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]
  srr &lt;- sample_map.subset$sra_id[sample_map.subset$sample_identifier == colnames(counts.list[[i]])[1]]
  raw_counts.file &lt;- read.delim(paste0(RAW_COUNTS.DIR,"/",srr[1],".sortedbyposition.counts.txt"),
                                      header = T)
  genelength &lt;- raw_counts.file$uniq_len[match(rownames(tpm.list[[i]]),raw_counts.file$featureID)]
  
  for(j in 1:ncol(tpm.list[[i]])){
    tpm.list[[i]][,j] &lt;- tpm.list[[i]][,j]/genelength
    tpm.list[[i]][,j] &lt;- tpm.list[[i]][,j]/(sum(tpm.list[[i]][,j])/1000000)
  }
  
  write.table(tpm.list[[i]],
            paste0(OUTPUT.DIR,"/",names(tpm.list)[i],"_tpm.tsv"),
            row.names = T,
            col.names = T,
            quote = F,
            sep = "\t")
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Create keys
                      <a name="user-content-ranalysis_de_keys">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Create keys " class="anchor" href="#create-keys-" id="user-content-create-keys-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Sample keys
                      <a name="user-content-ranalysis_de_keys_samples">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Sample keys " class="anchor" href="#sample-keys-" id="user-content-sample-keys-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(counts.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(counts.list)[i] &amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  legend.plot &lt;- ggplot()+
    geom_point(aes(x=groups[,2], y=seq(1,length(groups[,2]),1), color = groups[,2]), show.legend = T, size = 2)+
    scale_color_manual(values = as.character(unique(groups[,3])))+
    guides(colour = guide_legend(title = "Samples", title.position = "top", nrow = 6))+
    theme_bw()+
    theme(legend.position="top",legend.title.align=0.5)
  
  sample.legend &lt;- g_legend(legend.plot)
  
  pdf(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_samplekey.pdf"),
      height=2,
      width=10)
  grid.arrange(sample.legend)
  dev.off()
  
  grid.arrange(sample.legend)
}'>
                     <pre class="notranslate" lang="{R,fig.height=2,fig.width=10}"><code>for(i in 1:length(counts.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(counts.list)[i] &amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  legend.plot &lt;- ggplot()+
    geom_point(aes(x=groups[,2], y=seq(1,length(groups[,2]),1), color = groups[,2]), show.legend = T, size = 2)+
    scale_color_manual(values = as.character(unique(groups[,3])))+
    guides(colour = guide_legend(title = "Samples", title.position = "top", nrow = 6))+
    theme_bw()+
    theme(legend.position="top",legend.title.align=0.5)
  
  sample.legend &lt;- g_legend(legend.plot)
  
  pdf(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_samplekey.pdf"),
      height=2,
      width=10)
  grid.arrange(sample.legend)
  dev.off()
  
  grid.arrange(sample.legend)
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wOo, Darby et al 2012
                     </h5>
                     <a aria-label="Permalink: wOo, Darby et al 2012" class="anchor" href="#woo-darby-et-al-2012" id="user-content-woo-darby-et-al-2012">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_samplekey.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_samplekey.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Darby et al 2014
                     </h5>
                     <a aria-label="Permalink: wMel, Darby et al 2014" class="anchor" href="#wmel-darby-et-al-2014" id="user-content-wmel-darby-et-al-2014">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_samplekey.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_samplekey.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2014
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2014" class="anchor" href="#wdi-luck-et-al-2014" id="user-content-wdi-luck-et-al-2014">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_samplekey.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_samplekey.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2015
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2015" class="anchor" href="#wdi-luck-et-al-2015" id="user-content-wdi-luck-et-al-2015">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_samplekey.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_samplekey.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Gutzwiller et al 2015
                     </h5>
                     <a aria-label="Permalink: wMel, Gutzwiller et al 2015" class="anchor" href="#wmel-gutzwiller-et-al-2015" id="user-content-wmel-gutzwiller-et-al-2015">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_samplekey.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_samplekey.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Grote et al 2017
                     </h5>
                     <a aria-label="Permalink: wBm, Grote et al 2017" class="anchor" href="#wbm-grote-et-al-2017" id="user-content-wbm-grote-et-al-2017">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_samplekey.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_samplekey.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Chung et al 2019
                     </h5>
                     <a aria-label="Permalink: wBm, Chung et al 2019" class="anchor" href="#wbm-chung-et-al-2019" id="user-content-wbm-chung-et-al-2019">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_samplekey.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_samplekey.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Size key
                      <a name="user-content-ranalysis_de_keys_size">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Size key  " class="anchor" href="#size-key--" id="user-content-size-key--">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='max_lib_sizes &lt;- c()
min_lib_sizes &lt;- c()
for(i in 1:length(counts.list)){
  max_lib_sizes &lt;- c(max_lib_sizes,max(colSums(counts.list[[i]])))
  min_lib_sizes &lt;- c(min_lib_sizes,min(colSums(counts.list[[i]])))
}
size_limits &lt;- c(1,10^ceiling(log10(max(max_lib_sizes))))
size_breaks &lt;- lseq(10^floor(log10(min(min_lib_sizes))),10^ceiling(log10(max(max_lib_sizes))),8)
size.plot &lt;- ggplot()+
  geom_point(aes(x=seq(1,length(colSums(counts.list[[i]]))), y=colSums(counts.list[[i]]),size=colSums(counts.list[[i]])))+
  scale_size_continuous(limits=size_limits,breaks=size_breaks ,trans = "log")+
  guides(size = guide_legend(title = "Reads Mapping to Wolbachia Genes", title.position = "top", nrow = 2))+
  theme_bw()+
  theme(legend.position="top",legend.title.align=0.5)
  
size.legend &lt;- g_legend(size.plot)

pdf(paste0(OUTPUT.DIR,"/sizekey.pdf"),
    height=2,
    width=10)
grid.arrange(size.legend)
dev.off()

png(paste0(OUTPUT.DIR,"/sizekey.png"),
    height=2,
    width=10,
    units = "in",res=300)
grid.arrange(size.legend)
dev.off()

grid.arrange(size.legend)'>
                     <pre class="notranslate" lang="{R,fig.height=2,fig.width=10}"><code>max_lib_sizes &lt;- c()
min_lib_sizes &lt;- c()
for(i in 1:length(counts.list)){
  max_lib_sizes &lt;- c(max_lib_sizes,max(colSums(counts.list[[i]])))
  min_lib_sizes &lt;- c(min_lib_sizes,min(colSums(counts.list[[i]])))
}
size_limits &lt;- c(1,10^ceiling(log10(max(max_lib_sizes))))
size_breaks &lt;- lseq(10^floor(log10(min(min_lib_sizes))),10^ceiling(log10(max(max_lib_sizes))),8)
size.plot &lt;- ggplot()+
  geom_point(aes(x=seq(1,length(colSums(counts.list[[i]]))), y=colSums(counts.list[[i]]),size=colSums(counts.list[[i]])))+
  scale_size_continuous(limits=size_limits,breaks=size_breaks ,trans = "log")+
  guides(size = guide_legend(title = "Reads Mapping to Wolbachia Genes", title.position = "top", nrow = 2))+
  theme_bw()+
  theme(legend.position="top",legend.title.align=0.5)
  
size.legend &lt;- g_legend(size.plot)

pdf(paste0(OUTPUT.DIR,"/sizekey.pdf"),
    height=2,
    width=10)
grid.arrange(size.legend)
dev.off()

png(paste0(OUTPUT.DIR,"/sizekey.png"),
    height=2,
    width=10,
    units = "in",res=300)
grid.arrange(size.legend)
dev.off()

grid.arrange(size.legend)
</code></pre>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/sizekey.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/sizekey.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Heatmap log2TPM key
                      <a name="user-content-ranalysis_de_keys_hm1">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Heatmap log2TPM key " class="anchor" href="#heatmap-log2tpm-key-" id="user-content-heatmap-log2tpm-key-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='hmcol1 &lt;- colorRampPalette(c("#FFFFFF","#FDE0D2","#FABAA1","#F69274", "#F16B4E","#EF3D2D","#590A16","#A51E23","#650C16"))(20)
hmcol2 &lt;- colorRampPalette(c("navyblue","white","firebrick3"))(12)

hmcolor1.plot &lt;- ggplot() + 
  geom_raster(aes(x=seq(0,10,0.5), y=seq(0,10,0.5), fill = seq(0,10,0.5)))+
  scale_fill_gradientn(name = "log2TPM",
                       colours=hmcol1,
                       breaks=c(0,5,10))+
  theme(legend.position="bottom")+
  guides(fill = guide_colorbar(title.position = "top"))

sample.legend &lt;- g_legend(hmcolor1.plot)

pdf(paste0(OUTPUT.DIR,"/hmcolor1key.pdf"),
    height=2,
    width=7)
grid.arrange(sample.legend)
dev.off()

png(paste0(OUTPUT.DIR,"/hmcolor1key.png"),
    height=2,
    width=7,
    units = "in",res=300)
grid.arrange(sample.legend)
dev.off()

grid.arrange(sample.legend)'>
                     <pre class="notranslate" lang="{R,fig.height=2,fig.width=7}"><code>hmcol1 &lt;- colorRampPalette(c("#FFFFFF","#FDE0D2","#FABAA1","#F69274", "#F16B4E","#EF3D2D","#590A16","#A51E23","#650C16"))(20)
hmcol2 &lt;- colorRampPalette(c("navyblue","white","firebrick3"))(12)

hmcolor1.plot &lt;- ggplot() + 
  geom_raster(aes(x=seq(0,10,0.5), y=seq(0,10,0.5), fill = seq(0,10,0.5)))+
  scale_fill_gradientn(name = "log2TPM",
                       colours=hmcol1,
                       breaks=c(0,5,10))+
  theme(legend.position="bottom")+
  guides(fill = guide_colorbar(title.position = "top"))

sample.legend &lt;- g_legend(hmcolor1.plot)

pdf(paste0(OUTPUT.DIR,"/hmcolor1key.pdf"),
    height=2,
    width=7)
grid.arrange(sample.legend)
dev.off()

png(paste0(OUTPUT.DIR,"/hmcolor1key.png"),
    height=2,
    width=7,
    units = "in",res=300)
grid.arrange(sample.legend)
dev.off()

grid.arrange(sample.legend)
</code></pre>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/hmcolor1key.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/hmcolor1key.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Heatmap z-score log2TPM key
                      <a name="user-content-ranalysis_de_keys_hm2">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Heatmap z-score log2TPM key " class="anchor" href="#heatmap-z-score-log2tpm-key-" id="user-content-heatmap-z-score-log2tpm-key-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='hmcolor2.plot &lt;- ggplot() + 
  geom_raster(aes(x=seq(-3,3,0.5), y=seq(-3,3,0.5), fill = seq(-3,3,0.5)))+
  scale_fill_gradientn(name = "z-score of log2TPM",
                       colours=hmcol2,
                       breaks=c(-3,0,3))+
  theme(legend.position="bottom")+
  guides(fill = guide_colorbar(title.position = "top"))

heat.legend &lt;- g_legend(hmcolor2.plot)

pdf(paste0(OUTPUT.DIR,"/hmcolor2key.pdf"),
    height=2,
    width=7)
grid.arrange(heat.legend)
dev.off()

png(paste0(OUTPUT.DIR,"/hmcolor2key.png"),
    height=2,
    width=7,
    units = "in",res=300)
grid.arrange(heat.legend)
dev.off()

grid.arrange(heat.legend)'>
                     <pre class="notranslate" lang="{R,fig.height=2,fig.width=7}"><code>hmcolor2.plot &lt;- ggplot() + 
  geom_raster(aes(x=seq(-3,3,0.5), y=seq(-3,3,0.5), fill = seq(-3,3,0.5)))+
  scale_fill_gradientn(name = "z-score of log2TPM",
                       colours=hmcol2,
                       breaks=c(-3,0,3))+
  theme(legend.position="bottom")+
  guides(fill = guide_colorbar(title.position = "top"))

heat.legend &lt;- g_legend(hmcolor2.plot)

pdf(paste0(OUTPUT.DIR,"/hmcolor2key.pdf"),
    height=2,
    width=7)
grid.arrange(heat.legend)
dev.off()

png(paste0(OUTPUT.DIR,"/hmcolor2key.png"),
    height=2,
    width=7,
    units = "in",res=300)
grid.arrange(heat.legend)
dev.off()

grid.arrange(heat.legend)
</code></pre>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/hmcolor2key.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/hmcolor2key.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Assess if samples have been sequenced to saturation using saturation curves
                      <a name="user-content-ranalysis_de_keys_saturation">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Assess if samples have been sequenced to saturation using saturation curves " class="anchor" href="#assess-if-samples-have-been-sequenced-to-saturation-using-saturation-curves-" id="user-content-assess-if-samples-have-been-sequenced-to-saturation-using-saturation-curves-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(tpm.list)){
  rarefy.counts &lt;- ceiling(counts.list[[i]])
  rarefy.counts &lt;- rarefy.counts[,colSums(rarefy.counts) != 0,]
  
  sample_map.subset &lt;- sample_map[sample_map$study == names(counts.list)[i] &amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]
  
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  

  raremax &lt;- round(min(rowSums(t(rarefy.counts))[rowSums(t(rarefy.counts)) != 0]),0)
  srare &lt;- rarefy(t(rarefy.counts),raremax)
  raremax &lt;- ifelse(raremax &lt; 10, 10, raremax)
  
  rarefy.raw.df &lt;- rarecurve(t(rarefy.counts), step = round(raremax/10,0), sample = raremax)
  
  rarefy.df &lt;- as.data.frame(matrix(nrow = 0,
                                    ncol = 5))
  rarefy.points.df &lt;- rarefy.df
  for(j in 1:length(rarefy.raw.df)){
    steps &lt;- as.numeric(gsub("N","",names(rarefy.raw.df[[j]])))
    detected_genes &lt;- as.numeric(rarefy.raw.df[[j]])
    rarefy.df &lt;- as.data.frame(rbind(rarefy.df,
                                     cbind(as.numeric(steps),
                                           as.numeric(detected_genes),
                                           as.character(groups[j,1]),
                                           as.character(groups[j,2]),
                                           groups[j,3])))
    rarefy.points.df &lt;- as.data.frame(rbind(rarefy.points.df,
                                     cbind(as.numeric(max(steps)),
                                           as.numeric(max(detected_genes)),
                                           as.character(groups[j,1]),
                                           as.character(groups[j,2]),
                                           groups[j,3])))
    
  }
  rarefy.plot &lt;- ggplot()+
    geom_line(mapping=aes(x=as.numeric(as.character(rarefy.df[,1])), y=as.numeric(as.character(rarefy.df[,2])),group=rarefy.df[,3],color=rarefy.df[,4]))+
    #geom_point(mapping=aes(x=as.numeric(as.character(rarefy.df[,1])), y=as.numeric(as.character(rarefy.df[,2])),group=rarefy.df[,3],color=rarefy.df[,4]))+
    geom_point(mapping=aes(x=as.numeric(as.character(rarefy.points.df[,1])), y=as.numeric(as.character(rarefy.points.df[,2])),group=rarefy.points.df[,3],color=rarefy.points.df[,4]),size = 3)+
    guides(colour = F,shape = F)+
    scale_color_manual(values = levels(groups[,3]))+
    labs(x="reads mapping to protein-coding genes", y="genes detected", color = "Sample")+
    #coord_cartesian(xlim=c(0,100000))+
    theme_bw()
  
  pdf(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_rarefaction.pdf"),
      height=5,
      width=5)
  print(rarefy.plot)
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_rarefaction.png"),
      height=5,
      width=5,
      units = "in",res=300)
  print(rarefy.plot)
  dev.off()
  
  print(rarefy.plot)
}'>
                     <pre class="notranslate" lang="{R,"><code>for(i in 1:length(tpm.list)){
  rarefy.counts &lt;- ceiling(counts.list[[i]])
  rarefy.counts &lt;- rarefy.counts[,colSums(rarefy.counts) != 0,]
  
  sample_map.subset &lt;- sample_map[sample_map$study == names(counts.list)[i] &amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]
  
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  

  raremax &lt;- round(min(rowSums(t(rarefy.counts))[rowSums(t(rarefy.counts)) != 0]),0)
  srare &lt;- rarefy(t(rarefy.counts),raremax)
  raremax &lt;- ifelse(raremax &lt; 10, 10, raremax)
  
  rarefy.raw.df &lt;- rarecurve(t(rarefy.counts), step = round(raremax/10,0), sample = raremax)
  
  rarefy.df &lt;- as.data.frame(matrix(nrow = 0,
                                    ncol = 5))
  rarefy.points.df &lt;- rarefy.df
  for(j in 1:length(rarefy.raw.df)){
    steps &lt;- as.numeric(gsub("N","",names(rarefy.raw.df[[j]])))
    detected_genes &lt;- as.numeric(rarefy.raw.df[[j]])
    rarefy.df &lt;- as.data.frame(rbind(rarefy.df,
                                     cbind(as.numeric(steps),
                                           as.numeric(detected_genes),
                                           as.character(groups[j,1]),
                                           as.character(groups[j,2]),
                                           groups[j,3])))
    rarefy.points.df &lt;- as.data.frame(rbind(rarefy.points.df,
                                     cbind(as.numeric(max(steps)),
                                           as.numeric(max(detected_genes)),
                                           as.character(groups[j,1]),
                                           as.character(groups[j,2]),
                                           groups[j,3])))
    
  }
  rarefy.plot &lt;- ggplot()+
    geom_line(mapping=aes(x=as.numeric(as.character(rarefy.df[,1])), y=as.numeric(as.character(rarefy.df[,2])),group=rarefy.df[,3],color=rarefy.df[,4]))+
    #geom_point(mapping=aes(x=as.numeric(as.character(rarefy.df[,1])), y=as.numeric(as.character(rarefy.df[,2])),group=rarefy.df[,3],color=rarefy.df[,4]))+
    geom_point(mapping=aes(x=as.numeric(as.character(rarefy.points.df[,1])), y=as.numeric(as.character(rarefy.points.df[,2])),group=rarefy.points.df[,3],color=rarefy.points.df[,4]),size = 3)+
    guides(colour = F,shape = F)+
    scale_color_manual(values = levels(groups[,3]))+
    labs(x="reads mapping to protein-coding genes", y="genes detected", color = "Sample")+
    #coord_cartesian(xlim=c(0,100000))+
    theme_bw()
  
  pdf(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_rarefaction.pdf"),
      height=5,
      width=5)
  print(rarefy.plot)
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_rarefaction.png"),
      height=5,
      width=5,
      units = "in",res=300)
  print(rarefy.plot)
  dev.off()
  
  print(rarefy.plot)
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wOo, Darby et al 2012
                     </h5>
                     <a aria-label="Permalink: wOo, Darby et al 2012" class="anchor" href="#woo-darby-et-al-2012-1" id="user-content-woo-darby-et-al-2012-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_rarefaction.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_rarefaction.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Darby et al 2014
                     </h5>
                     <a aria-label="Permalink: wMel, Darby et al 2014" class="anchor" href="#wmel-darby-et-al-2014-1" id="user-content-wmel-darby-et-al-2014-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_rarefaction.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_rarefaction.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2014
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2014" class="anchor" href="#wdi-luck-et-al-2014-1" id="user-content-wdi-luck-et-al-2014-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_rarefaction.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_rarefaction.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2015
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2015" class="anchor" href="#wdi-luck-et-al-2015-1" id="user-content-wdi-luck-et-al-2015-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_rarefaction.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_rarefaction.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Gutzwiller et al 2015
                     </h5>
                     <a aria-label="Permalink: wMel, Gutzwiller et al 2015" class="anchor" href="#wmel-gutzwiller-et-al-2015-1" id="user-content-wmel-gutzwiller-et-al-2015-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_rarefaction.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_rarefaction.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Grote et al 2017
                     </h5>
                     <a aria-label="Permalink: wBm, Grote et al 2017" class="anchor" href="#wbm-grote-et-al-2017-1" id="user-content-wbm-grote-et-al-2017-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_rarefaction.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_rarefaction.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Chung et al 2019
                     </h5>
                     <a aria-label="Permalink: wBm, Chung et al 2019" class="anchor" href="#wbm-chung-et-al-2019-1" id="user-content-wbm-chung-et-al-2019-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_rarefaction.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_rarefaction.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Exclude samples and/or studies from differential expression analyses due to inadequate sequencing depth
                      <a name="user-content-ranalysis_de_exclude">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Exclude samples and/or studies from differential expression analyses due to inadequate sequencing depth " class="anchor" href="#exclude-samples-andor-studies-from-differential-expression-analyses-due-to-inadequate-sequencing-depth-" id="user-content-exclude-samples-andor-studies-from-differential-expression-analyses-due-to-inadequate-sequencing-depth-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     All samples from Luck et al 2014 were excluded due to having inadequate replicates and sequencing depth for differential expression analyses. All samples with &lt;2000 gene counts from Luck et al 2015 are excluded due to have inadequate sequencing depth (refer to saturation curves).
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts.kept.list &lt;- counts.list
tpm.kept.list &lt;- tpm.list

counts.edgeR.list &lt;- counts.list
tpm.edgeR.list &lt;- tpm.list

counts.edgeR.list[[which(names(counts.edgeR.list) == "wDi_luck_2014")]] &lt;- NULL
tpm.edgeR.list[[which(names(tpm.edgeR.list) == "wDi_luck_2014")]] &lt;- NULL

counts.kept.list[[which(names(counts.kept.list) == "wDi_luck_2015")]] &lt;- counts.kept.list[[which(names(counts.kept.list) == "wDi_luck_2015")]][colSums(counts.kept.list[[which(names(counts.kept.list) == "wDi_luck_2015")]]) &gt; 2000]
tpm.kept.list[[which(names(tpm.kept.list) == "wDi_luck_2015")]] &lt;- tpm.kept.list[[which(names(tpm.kept.list) == "wDi_luck_2015")]][,colnames(tpm.kept.list[[which(names(tpm.kept.list) == "wDi_luck_2015")]]) %in% colnames(counts.kept.list[[which(names(counts.kept.list) == "wDi_luck_2015")]])]

counts.edgeR.list[[which(names(counts.edgeR.list) == "wDi_luck_2015")]] &lt;- counts.edgeR.list[[which(names(counts.edgeR.list) == "wDi_luck_2015")]][colSums(counts.edgeR.list[[which(names(counts.edgeR.list) == "wDi_luck_2015")]]) &gt; 2000]
tpm.edgeR.list[[which(names(tpm.edgeR.list) == "wDi_luck_2015")]] &lt;- tpm.edgeR.list[[which(names(tpm.edgeR.list) == "wDi_luck_2015")]][,colnames(tpm.edgeR.list[[which(names(tpm.edgeR.list) == "wDi_luck_2015")]]) %in% colnames(counts.edgeR.list[[which(names(counts.edgeR.list) == "wDi_luck_2015")]])]

qlf.list &lt;- counts.edgeR.list'>
                     <pre class="notranslate" lang="{R}"><code>counts.kept.list &lt;- counts.list
tpm.kept.list &lt;- tpm.list

counts.edgeR.list &lt;- counts.list
tpm.edgeR.list &lt;- tpm.list

counts.edgeR.list[[which(names(counts.edgeR.list) == "wDi_luck_2014")]] &lt;- NULL
tpm.edgeR.list[[which(names(tpm.edgeR.list) == "wDi_luck_2014")]] &lt;- NULL

counts.kept.list[[which(names(counts.kept.list) == "wDi_luck_2015")]] &lt;- counts.kept.list[[which(names(counts.kept.list) == "wDi_luck_2015")]][colSums(counts.kept.list[[which(names(counts.kept.list) == "wDi_luck_2015")]]) &gt; 2000]
tpm.kept.list[[which(names(tpm.kept.list) == "wDi_luck_2015")]] &lt;- tpm.kept.list[[which(names(tpm.kept.list) == "wDi_luck_2015")]][,colnames(tpm.kept.list[[which(names(tpm.kept.list) == "wDi_luck_2015")]]) %in% colnames(counts.kept.list[[which(names(counts.kept.list) == "wDi_luck_2015")]])]

counts.edgeR.list[[which(names(counts.edgeR.list) == "wDi_luck_2015")]] &lt;- counts.edgeR.list[[which(names(counts.edgeR.list) == "wDi_luck_2015")]][colSums(counts.edgeR.list[[which(names(counts.edgeR.list) == "wDi_luck_2015")]]) &gt; 2000]
tpm.edgeR.list[[which(names(tpm.edgeR.list) == "wDi_luck_2015")]] &lt;- tpm.edgeR.list[[which(names(tpm.edgeR.list) == "wDi_luck_2015")]][,colnames(tpm.edgeR.list[[which(names(tpm.edgeR.list) == "wDi_luck_2015")]]) %in% colnames(counts.edgeR.list[[which(names(counts.edgeR.list) == "wDi_luck_2015")]])]

qlf.list &lt;- counts.edgeR.list
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Identify the number of genes in each study that meets the minimum CPM threshold
                      <a name="user-content-ranalysis_de_exclude">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Identify the number of genes in each study that meets the minimum CPM threshold " class="anchor" href="#identify-the-number-of-genes-in-each-study-that-meets-the-minimum-cpm-threshold-" id="user-content-identify-the-number-of-genes-in-each-study-that-meets-the-minimum-cpm-threshold-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(counts.kept.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(counts.list)[i] &amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- groups[groups[,1] %in% colnames(counts.kept.list[[i]]),]
  groups &lt;- unique(groups)

  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))

  cpm.cutoff &lt;- 5/min(colSums(counts.kept.list[[i]])) * 1000000

  y &lt;- DGEList(counts = counts.kept.list[[i]], group = groups[,2])
  y &lt;- calcNormFactors(y)
  keep &lt;- rowSums(cpm(y) &gt;= cpm.cutoff) &gt;= min(table(groups[,2]))

  keep.df &lt;- as.data.frame(table(keep))
  print(paste0(names(counts.kept.list)[i],": ",keep.df[keep.df[,1] == F,2]," genes excluded using edgeR filter, ",keep.df[keep.df[,1] == T,2], " genes kept" ))

  counts.kept.list[[which(names(counts.kept.list) == names(counts.list)[i])]] &lt;- counts.kept.list[[which(names(counts.kept.list) == names(counts.list)[i])]][keep,]
  tpm.kept.list[[which(names(tpm.kept.list) == names(tpm.list)[i])]] &lt;- tpm.kept.list[[which(names(tpm.kept.list) == names(tpm.list)[i])]][keep,]
}'>
                     <pre class="notranslate" lang="{R}"><code>for(i in 1:length(counts.kept.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(counts.list)[i] &amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- groups[groups[,1] %in% colnames(counts.kept.list[[i]]),]
  groups &lt;- unique(groups)

  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))

  cpm.cutoff &lt;- 5/min(colSums(counts.kept.list[[i]])) * 1000000

  y &lt;- DGEList(counts = counts.kept.list[[i]], group = groups[,2])
  y &lt;- calcNormFactors(y)
  keep &lt;- rowSums(cpm(y) &gt;= cpm.cutoff) &gt;= min(table(groups[,2]))

  keep.df &lt;- as.data.frame(table(keep))
  print(paste0(names(counts.kept.list)[i],": ",keep.df[keep.df[,1] == F,2]," genes excluded using edgeR filter, ",keep.df[keep.df[,1] == T,2], " genes kept" ))

  counts.kept.list[[which(names(counts.kept.list) == names(counts.list)[i])]] &lt;- counts.kept.list[[which(names(counts.kept.list) == names(counts.list)[i])]][keep,]
  tpm.kept.list[[which(names(tpm.kept.list) == names(tpm.list)[i])]] &lt;- tpm.kept.list[[which(names(tpm.kept.list) == names(tpm.list)[i])]][keep,]
}
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='[1] "wOo_darby_2012: 91 genes excluded using edgeR filter, 560 genes kept"
[1] "wMel_darby_2014: 219 genes excluded using edgeR filter, 867 genes kept"
[1] "wDi_luck_2014: 853 genes excluded using edgeR filter, 18 genes kept"
[1] "wDi_luck_2015: 546 genes excluded using edgeR filter, 325 genes kept"
[1] "wMel_gutzwiller_2015: 305 genes excluded using edgeR filter, 781 genes kept"
[1] "wBm_grote_2017: 414 genes excluded using edgeR filter, 425 genes kept"
[1] "wBm_chung_2019: 206 genes excluded using edgeR filter, 548 genes kept"'>
                     <pre class="notranslate" lang="{R,"><code>[1] "wOo_darby_2012: 91 genes excluded using edgeR filter, 560 genes kept"
[1] "wMel_darby_2014: 219 genes excluded using edgeR filter, 867 genes kept"
[1] "wDi_luck_2014: 853 genes excluded using edgeR filter, 18 genes kept"
[1] "wDi_luck_2015: 546 genes excluded using edgeR filter, 325 genes kept"
[1] "wMel_gutzwiller_2015: 305 genes excluded using edgeR filter, 781 genes kept"
[1] "wBm_grote_2017: 414 genes excluded using edgeR filter, 425 genes kept"
[1] "wBm_chung_2019: 206 genes excluded using edgeR filter, 548 genes kept"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Identify differentially expressed genes (FDR &lt; 0.05) in each study
                      <a name="user-content-ranalysis_de_de">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Identify differentially expressed genes (FDR &lt; 0.05) in each study " class="anchor" href="#identify-differentially-expressed-genes-fdr--005-in-each-study-" id="user-content-identify-differentially-expressed-genes-fdr--005-in-each-study-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='FDRcutoff &lt;- 0.05

for(i in 1:length(counts.edgeR.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(counts.edgeR.list)[i] &amp; sample_map$sample_identifier %in% colnames(counts.edgeR.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)

  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))

  cpm.cutoff &lt;- 5/min(colSums(counts.edgeR.list[[i]])) * 1000000

  y &lt;- DGEList(counts = counts.edgeR.list[[i]], group = groups[,2])
  y &lt;- calcNormFactors(y)
  keep &lt;- rowSums(cpm(y) &gt;= cpm.cutoff) &gt;= min(table(groups[,2]))

  y &lt;- y[keep, , keep.lib.sizes = F]
  design &lt;- model.matrix(~groups[,2])
  y &lt;- estimateDisp(y , design)
  fit &lt;- glmQLFit(y, design)
  qlf &lt;- glmQLFTest(fit, coef = 2:ncol(fit))

  FDR &lt;- as.data.frame(p.adjust(qlf$table$PValue, method="BH"))
  rownames(FDR) &lt;- rownames(qlf$table)
  FDR.genes &lt;- rownames(FDR[FDR[,1] &lt; FDRcutoff, , drop = F])
  print(paste0(names(counts.edgeR.list)[i],": ",length(FDR.genes)," DE genes" ))
  
  qlf.list[[i]] &lt;- as.data.frame(qlf$table)
  counts.edgeR.list[[i]] &lt;- counts.edgeR.list[[i]][rownames(counts.edgeR.list[[i]]) %in% FDR.genes,]
  tpm.edgeR.list[[i]] &lt;- tpm.edgeR.list[[i]][rownames(tpm.edgeR.list[[i]]) %in% FDR.genes,]
  
  write.table(tpm.edgeR.list[[i]],
            paste0(OUTPUT.DIR,"/",names(tpm.edgeR.list)[i],"_de_tpm.tsv"),
            row.names = T,
            col.names = T,
            quote = F,
            sep = "\t")
}'>
                     <pre class="notranslate" lang="{R}"><code>FDRcutoff &lt;- 0.05

for(i in 1:length(counts.edgeR.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(counts.edgeR.list)[i] &amp; sample_map$sample_identifier %in% colnames(counts.edgeR.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)

  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))

  cpm.cutoff &lt;- 5/min(colSums(counts.edgeR.list[[i]])) * 1000000

  y &lt;- DGEList(counts = counts.edgeR.list[[i]], group = groups[,2])
  y &lt;- calcNormFactors(y)
  keep &lt;- rowSums(cpm(y) &gt;= cpm.cutoff) &gt;= min(table(groups[,2]))

  y &lt;- y[keep, , keep.lib.sizes = F]
  design &lt;- model.matrix(~groups[,2])
  y &lt;- estimateDisp(y , design)
  fit &lt;- glmQLFit(y, design)
  qlf &lt;- glmQLFTest(fit, coef = 2:ncol(fit))

  FDR &lt;- as.data.frame(p.adjust(qlf$table$PValue, method="BH"))
  rownames(FDR) &lt;- rownames(qlf$table)
  FDR.genes &lt;- rownames(FDR[FDR[,1] &lt; FDRcutoff, , drop = F])
  print(paste0(names(counts.edgeR.list)[i],": ",length(FDR.genes)," DE genes" ))
  
  qlf.list[[i]] &lt;- as.data.frame(qlf$table)
  counts.edgeR.list[[i]] &lt;- counts.edgeR.list[[i]][rownames(counts.edgeR.list[[i]]) %in% FDR.genes,]
  tpm.edgeR.list[[i]] &lt;- tpm.edgeR.list[[i]][rownames(tpm.edgeR.list[[i]]) %in% FDR.genes,]
  
  write.table(tpm.edgeR.list[[i]],
            paste0(OUTPUT.DIR,"/",names(tpm.edgeR.list)[i],"_de_tpm.tsv"),
            row.names = T,
            col.names = T,
            quote = F,
            sep = "\t")
}
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='[1] "wOo_darby_2012: 0 DE genes"
[1] "wMel_darby_2014: 120 DE genes"
[1] "wDi_luck_2015: 36 DE genes"
[1] "wMel_gutzwiller_2015: 473 DE genes"
[1] "wBm_grote_2017: 94 DE genes"
[1] "wBm_chung_2019: 373 DE genes"'>
                     <pre class="notranslate" lang="{R,"><code>[1] "wOo_darby_2012: 0 DE genes"
[1] "wMel_darby_2014: 120 DE genes"
[1] "wDi_luck_2015: 36 DE genes"
[1] "wMel_gutzwiller_2015: 473 DE genes"
[1] "wBm_grote_2017: 94 DE genes"
[1] "wBm_chung_2019: 373 DE genes"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Conduct principal component analyses
                      <a name="user-content-ranalysis_de_pca">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Conduct principal component analyses " class="anchor" href="#conduct-principal-component-analyses-" id="user-content-conduct-principal-component-analyses-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      All genes
                      <a name="user-content-ranalysis_de_pca_all">
                      </a>
                     </h4>
                     <a aria-label="Permalink: All genes " class="anchor" href="#all-genes-" id="user-content-all-genes-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(tpm.list)){ 
	sample_map.subset &lt;- sample_map[sample_map$study == names(counts.list)[i] &amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]

	groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
	                            as.character(sample_map.subset$sample_group),
	                            as.character(sample_map.subset$color)))
	groups &lt;- unique(groups)

	groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
	groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
	groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))

	pca.df &lt;- t(scale(t(log2(tpm.list[[i]] + 1))))
	pca.df[is.na(pca.df)] &lt;- 0
	pca.df &lt;- pca.df[rowSums(pca.df == 0) != ncol(pca.df),]
	pca &lt;- PCA(as.data.frame(scale(t(pca.df))), graph = FALSE, ncp = ncol(counts.list[[i]]) - 1)

	pca.plot &lt;- ggplot()+
	  geom_point(aes(x=pca$ind$coord[,1], y=pca$ind$coord[,2], color = groups[,2],size = colSums(counts.list[[i]])))+
	  labs(col = "Samples", size = "Reads Mapped\nto Features", 
	       x = paste0("PC1 (", round(pca$eig[1,2],1), "%)"), 
	       y = paste0("PC2 (", round(pca$eig[2,2],1), "%)"))+
	  guides(color = F,size = F)+
	  # guides(colour = guide_legend(ncol = 2))+
	  scale_color_manual(values = levels(groups[,3]))+
	  scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = "log")+
	  theme_bw()

	pdf(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_pca_allgenes.pdf"),
	  height=5,
	  width=5)
	print(pca.plot)
	dev.off()
	
	png(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_pca_allgenes.png"),
	  height=5,
	  width=5,
	  units = "in",res=300)
	print(pca.plot)
	dev.off()

	print(pca.plot)
}'>
                     <pre class="notranslate" lang="{R,"><code>for(i in 1:length(tpm.list)){ 
	sample_map.subset &lt;- sample_map[sample_map$study == names(counts.list)[i] &amp; sample_map$sample_identifier %in% colnames(counts.list[[i]]),]

	groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
	                            as.character(sample_map.subset$sample_group),
	                            as.character(sample_map.subset$color)))
	groups &lt;- unique(groups)

	groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
	groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
	groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))

	pca.df &lt;- t(scale(t(log2(tpm.list[[i]] + 1))))
	pca.df[is.na(pca.df)] &lt;- 0
	pca.df &lt;- pca.df[rowSums(pca.df == 0) != ncol(pca.df),]
	pca &lt;- PCA(as.data.frame(scale(t(pca.df))), graph = FALSE, ncp = ncol(counts.list[[i]]) - 1)

	pca.plot &lt;- ggplot()+
	  geom_point(aes(x=pca$ind$coord[,1], y=pca$ind$coord[,2], color = groups[,2],size = colSums(counts.list[[i]])))+
	  labs(col = "Samples", size = "Reads Mapped\nto Features", 
	       x = paste0("PC1 (", round(pca$eig[1,2],1), "%)"), 
	       y = paste0("PC2 (", round(pca$eig[2,2],1), "%)"))+
	  guides(color = F,size = F)+
	  # guides(colour = guide_legend(ncol = 2))+
	  scale_color_manual(values = levels(groups[,3]))+
	  scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = "log")+
	  theme_bw()

	pdf(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_pca_allgenes.pdf"),
	  height=5,
	  width=5)
	print(pca.plot)
	dev.off()
	
	png(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_pca_allgenes.png"),
	  height=5,
	  width=5,
	  units = "in",res=300)
	print(pca.plot)
	dev.off()

	print(pca.plot)
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wOo, Darby et al 2012
                     </h5>
                     <a aria-label="Permalink: wOo, Darby et al 2012" class="anchor" href="#woo-darby-et-al-2012-2" id="user-content-woo-darby-et-al-2012-2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_pca_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_pca_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Darby et al 2014
                     </h5>
                     <a aria-label="Permalink: wMel, Darby et al 2014" class="anchor" href="#wmel-darby-et-al-2014-2" id="user-content-wmel-darby-et-al-2014-2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_pca_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_pca_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2014
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2014" class="anchor" href="#wdi-luck-et-al-2014-2" id="user-content-wdi-luck-et-al-2014-2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_pca_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_pca_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2015
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2015" class="anchor" href="#wdi-luck-et-al-2015-2" id="user-content-wdi-luck-et-al-2015-2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_pca_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_pca_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Gutzwiller et al 2015
                     </h5>
                     <a aria-label="Permalink: wMel, Gutzwiller et al 2015" class="anchor" href="#wmel-gutzwiller-et-al-2015-2" id="user-content-wmel-gutzwiller-et-al-2015-2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_pca_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_pca_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Grote et al 2017
                     </h5>
                     <a aria-label="Permalink: wBm, Grote et al 2017" class="anchor" href="#wbm-grote-et-al-2017-2" id="user-content-wbm-grote-et-al-2017-2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_pca_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_pca_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Chung et al 2019
                     </h5>
                     <a aria-label="Permalink: wBm, Chung et al 2019" class="anchor" href="#wbm-chung-et-al-2019-2" id="user-content-wbm-chung-et-al-2019-2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_pca_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_pca_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Kept genes
                      <a name="user-content-ranalysis_de_pca_kept">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Kept genes " class="anchor" href="#kept-genes-" id="user-content-kept-genes-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(tpm.kept.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.kept.list)[i] &amp; sample_map$sample_identifier %in% colnames(tpm.kept.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  if(nrow(tpm.kept.list[[i]]) &gt; 0){
    pca.df &lt;- t(scale(t(log2(tpm.kept.list[[i]] + 1))))
    pca.df &lt;- pca.df[rowSums(pca.df == 0) != ncol(pca.df),]
    pca &lt;- PCA(as.data.frame(scale(t(pca.df))), graph = FALSE, ncp = ncol(tpm.kept.list[[i]]) - 1)
    
    pca.plot &lt;- ggplot()+
      geom_point(aes(x=pca$ind$coord[,1], y=pca$ind$coord[,2], color = groups[,2],size = colSums(counts.kept.list[[i]])))+
      labs(col = "Samples", size = "Reads Mapped\nto Features", 
           x = paste0("PC1 (", round(pca$eig[1,2],1), "%)"), 
           y = paste0("PC2 (", round(pca$eig[2,2],1), "%)"))+
      guides(color = F,size = F)+
      # guides(colour = guide_legend(ncol = 2))+
      scale_color_manual(values = levels(groups[,3]))+
      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = "log")+
      theme_bw()
    
    pdf(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_pca_keptgenes.pdf"),
          height=5,
          width=5)
      print(pca.plot)
    dev.off()
      
    png(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_pca_keptgenes.png"),
  	    height=5,
  	    width=5,
  	    units = "in",res=300)
  	  print(pca.plot)
  	dev.off()
    
    print(pca.plot)
  }
}'>
                     <pre class="notranslate" lang="{R,"><code>for(i in 1:length(tpm.kept.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.kept.list)[i] &amp; sample_map$sample_identifier %in% colnames(tpm.kept.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  if(nrow(tpm.kept.list[[i]]) &gt; 0){
    pca.df &lt;- t(scale(t(log2(tpm.kept.list[[i]] + 1))))
    pca.df &lt;- pca.df[rowSums(pca.df == 0) != ncol(pca.df),]
    pca &lt;- PCA(as.data.frame(scale(t(pca.df))), graph = FALSE, ncp = ncol(tpm.kept.list[[i]]) - 1)
    
    pca.plot &lt;- ggplot()+
      geom_point(aes(x=pca$ind$coord[,1], y=pca$ind$coord[,2], color = groups[,2],size = colSums(counts.kept.list[[i]])))+
      labs(col = "Samples", size = "Reads Mapped\nto Features", 
           x = paste0("PC1 (", round(pca$eig[1,2],1), "%)"), 
           y = paste0("PC2 (", round(pca$eig[2,2],1), "%)"))+
      guides(color = F,size = F)+
      # guides(colour = guide_legend(ncol = 2))+
      scale_color_manual(values = levels(groups[,3]))+
      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = "log")+
      theme_bw()
    
    pdf(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_pca_keptgenes.pdf"),
          height=5,
          width=5)
      print(pca.plot)
    dev.off()
      
    png(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_pca_keptgenes.png"),
  	    height=5,
  	    width=5,
  	    units = "in",res=300)
  	  print(pca.plot)
  	dev.off()
    
    print(pca.plot)
  }
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wOo, Darby et al 2012
                     </h5>
                     <a aria-label="Permalink: wOo, Darby et al 2012" class="anchor" href="#woo-darby-et-al-2012-3" id="user-content-woo-darby-et-al-2012-3">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_pca_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_pca_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Darby et al 2014
                     </h5>
                     <a aria-label="Permalink: wMel, Darby et al 2014" class="anchor" href="#wmel-darby-et-al-2014-3" id="user-content-wmel-darby-et-al-2014-3">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_pca_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_pca_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2014
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2014" class="anchor" href="#wdi-luck-et-al-2014-3" id="user-content-wdi-luck-et-al-2014-3">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_pca_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_pca_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2015
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2015" class="anchor" href="#wdi-luck-et-al-2015-3" id="user-content-wdi-luck-et-al-2015-3">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_pca_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_pca_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Gutzwiller et al 2015
                     </h5>
                     <a aria-label="Permalink: wMel, Gutzwiller et al 2015" class="anchor" href="#wmel-gutzwiller-et-al-2015-3" id="user-content-wmel-gutzwiller-et-al-2015-3">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_pca_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_pca_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Grote et al 2017
                     </h5>
                     <a aria-label="Permalink: wBm, Grote et al 2017" class="anchor" href="#wbm-grote-et-al-2017-3" id="user-content-wbm-grote-et-al-2017-3">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_pca_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_pca_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Chung et al 2019
                     </h5>
                     <a aria-label="Permalink: wBm, Chung et al 2019" class="anchor" href="#wbm-chung-et-al-2019-3" id="user-content-wbm-chung-et-al-2019-3">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_pca_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_pca_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Differentially expressed genes
                      <a name="user-content-ranalysis_de_pca_de">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Differentially expressed genes " class="anchor" href="#differentially-expressed-genes-" id="user-content-differentially-expressed-genes-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(tpm.edgeR.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.edgeR.list)[i]&amp; sample_map$sample_identifier %in% colnames(tpm.edgeR.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  if(nrow(tpm.edgeR.list[[i]]) &gt; 0){
    pca.df &lt;- t(scale(t(log2(tpm.edgeR.list[[i]] + 1))))
    pca.df &lt;- pca.df[rowSums(pca.df == 0) != ncol(pca.df),]
    pca &lt;- PCA(as.data.frame(scale(t(pca.df))), graph = FALSE, ncp = ncol(tpm.edgeR.list[[i]]) - 1)
    
    pca.plot &lt;- ggplot()+
      geom_point(aes(x=pca$ind$coord[,1], y=pca$ind$coord[,2], color = groups[,2],size = colSums(counts.edgeR.list[[i]])))+
      labs(col = "Samples", size = "Reads Mapped\nto Features", 
           x = paste0("PC1 (", round(pca$eig[1,2],1), "%)"), 
           y = paste0("PC2 (", round(pca$eig[2,2],1), "%)"))+
      guides(color = F,size = F)+
      # guides(colour = guide_legend(ncol = 2))+
      scale_color_manual(values = levels(groups[,3]))+
      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = "log")+
      theme_bw()
    
    pdf(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_pca_degenes.pdf"),
        height=5,
        width=5)
      print(pca.plot)
    dev.off()
    
    png(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_pca_degenes.png"),
        height=5,
        width=5,
        units = "in",res=300)
      print(pca.plot)
    dev.off()
    
    print(pca.plot)
  }
}'>
                     <pre class="notranslate" lang="{R,"><code>for(i in 1:length(tpm.edgeR.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.edgeR.list)[i]&amp; sample_map$sample_identifier %in% colnames(tpm.edgeR.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  if(nrow(tpm.edgeR.list[[i]]) &gt; 0){
    pca.df &lt;- t(scale(t(log2(tpm.edgeR.list[[i]] + 1))))
    pca.df &lt;- pca.df[rowSums(pca.df == 0) != ncol(pca.df),]
    pca &lt;- PCA(as.data.frame(scale(t(pca.df))), graph = FALSE, ncp = ncol(tpm.edgeR.list[[i]]) - 1)
    
    pca.plot &lt;- ggplot()+
      geom_point(aes(x=pca$ind$coord[,1], y=pca$ind$coord[,2], color = groups[,2],size = colSums(counts.edgeR.list[[i]])))+
      labs(col = "Samples", size = "Reads Mapped\nto Features", 
           x = paste0("PC1 (", round(pca$eig[1,2],1), "%)"), 
           y = paste0("PC2 (", round(pca$eig[2,2],1), "%)"))+
      guides(color = F,size = F)+
      # guides(colour = guide_legend(ncol = 2))+
      scale_color_manual(values = levels(groups[,3]))+
      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = "log")+
      theme_bw()
    
    pdf(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_pca_degenes.pdf"),
        height=5,
        width=5)
      print(pca.plot)
    dev.off()
    
    png(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_pca_degenes.png"),
        height=5,
        width=5,
        units = "in",res=300)
      print(pca.plot)
    dev.off()
    
    print(pca.plot)
  }
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Darby et al 2014
                     </h5>
                     <a aria-label="Permalink: wMel, Darby et al 2014" class="anchor" href="#wmel-darby-et-al-2014-4" id="user-content-wmel-darby-et-al-2014-4">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_pca_degenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_pca_degenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2015
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2015" class="anchor" href="#wdi-luck-et-al-2015-4" id="user-content-wdi-luck-et-al-2015-4">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_pca_degenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_pca_degenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Gutzwiller et al 2015
                     </h5>
                     <a aria-label="Permalink: wMel, Gutzwiller et al 2015" class="anchor" href="#wmel-gutzwiller-et-al-2015-4" id="user-content-wmel-gutzwiller-et-al-2015-4">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_pca_degenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_pca_degenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Grote et al 2017
                     </h5>
                     <a aria-label="Permalink: wBm, Grote et al 2017" class="anchor" href="#wbm-grote-et-al-2017-4" id="user-content-wbm-grote-et-al-2017-4">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_pca_degenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_pca_degenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Chung et al 2019
                     </h5>
                     <a aria-label="Permalink: wBm, Chung et al 2019" class="anchor" href="#wbm-chung-et-al-2019-4" id="user-content-wbm-chung-et-al-2019-4">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_pca_degenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_pca_degenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Conduct hierarchical clustering analyses
                      <a name="user-content-ranalysis_de_hc">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Conduct hierarchical clustering analyses " class="anchor" href="#conduct-hierarchical-clustering-analyses-" id="user-content-conduct-hierarchical-clustering-analyses-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      All genes
                      <a name="user-content-ranalysis_de_hc_all">
                      </a>
                     </h4>
                     <a aria-label="Permalink: All genes " class="anchor" href="#all-genes--1" id="user-content-all-genes--1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(tpm.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.list)[i]&amp; sample_map$sample_identifier %in% colnames(tpm.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  dendrogram &lt;- as.data.frame(t(scale(t(log2(tpm.list[[i]] + 1)))))
  dendrogram &lt;- dendrogram[rowSums(is.na(dendrogram)) == 0,]
  result &lt;- pvclust(dendrogram, method.dist="cor", method.hclust="average", nboot=100, quiet=T)
  
  structure &lt;- get_dendro_structure(result)
  dendro.data &lt;- get_dendro_data(result)
  bootstrap.positions &lt;- get_dendro_bootstraps(dendro.data)
    
  points.df &lt;- as.data.frame(cbind(seq(1,length(structure),1),
                                   structure))
  dendrogroups &lt;- groups[,2][result$hclust$order]
  dendrocol &lt;- groups[,3][result$hclust$order]
  dendrosize &lt;- colSums(counts.list[[i]])[result$hclust$order]
  
  for(j in 1:length(unique(groups[,1]))){
    result$hclust$labels &lt;- gsub(groups[,1][j],groups[,2][j],result$hclust$labels)
  }
  
  dendrogram.plot &lt;- ggdendrogram(hang.dendrogram(as.dendrogram(result$hclust)), theme_dendro = T)+
    geom_point(aes(x=seq(1,length(structure)), y = structure, color = dendrogroups, size = dendrosize))+
    scale_color_manual(values = levels(groups[,3]))+
    labs(x = "", y = "", col = "Samples", size = "Reads Mapped\nto Features")+
    guides(colour = guide_legend(ncol = 2))+
    theme_minimal()+      
    scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = "log")+
    theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust = 1),
          axis.text.y = element_blank(),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          legend.position="none")
  
  for(j in 1:length(result$edges$bp)){
    text &lt;- round(result$edges$bp[j] * 100,0)
    dendrogram.plot &lt;- dendrogram.plot + annotate("text", label = text, x=bootstrap.positions[j,1] + 0.4, y=bootstrap.positions[j,2] + 0.03, size = 3)
  }
  
  pdf(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_dendrogram_allgenes.pdf"),
      width = 11,
      height = 8)
    print(dendrogram.plot)
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_dendrogram_allgenes.png"),
      width = 11,
      height = 8,
      units = "in",res=300)
    print(dendrogram.plot)
  dev.off()
  
  print(dendrogram.plot)
}'>
                     <pre class="notranslate" lang="{R,"><code>for(i in 1:length(tpm.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.list)[i]&amp; sample_map$sample_identifier %in% colnames(tpm.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  dendrogram &lt;- as.data.frame(t(scale(t(log2(tpm.list[[i]] + 1)))))
  dendrogram &lt;- dendrogram[rowSums(is.na(dendrogram)) == 0,]
  result &lt;- pvclust(dendrogram, method.dist="cor", method.hclust="average", nboot=100, quiet=T)
  
  structure &lt;- get_dendro_structure(result)
  dendro.data &lt;- get_dendro_data(result)
  bootstrap.positions &lt;- get_dendro_bootstraps(dendro.data)
    
  points.df &lt;- as.data.frame(cbind(seq(1,length(structure),1),
                                   structure))
  dendrogroups &lt;- groups[,2][result$hclust$order]
  dendrocol &lt;- groups[,3][result$hclust$order]
  dendrosize &lt;- colSums(counts.list[[i]])[result$hclust$order]
  
  for(j in 1:length(unique(groups[,1]))){
    result$hclust$labels &lt;- gsub(groups[,1][j],groups[,2][j],result$hclust$labels)
  }
  
  dendrogram.plot &lt;- ggdendrogram(hang.dendrogram(as.dendrogram(result$hclust)), theme_dendro = T)+
    geom_point(aes(x=seq(1,length(structure)), y = structure, color = dendrogroups, size = dendrosize))+
    scale_color_manual(values = levels(groups[,3]))+
    labs(x = "", y = "", col = "Samples", size = "Reads Mapped\nto Features")+
    guides(colour = guide_legend(ncol = 2))+
    theme_minimal()+      
    scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = "log")+
    theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust = 1),
          axis.text.y = element_blank(),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          legend.position="none")
  
  for(j in 1:length(result$edges$bp)){
    text &lt;- round(result$edges$bp[j] * 100,0)
    dendrogram.plot &lt;- dendrogram.plot + annotate("text", label = text, x=bootstrap.positions[j,1] + 0.4, y=bootstrap.positions[j,2] + 0.03, size = 3)
  }
  
  pdf(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_dendrogram_allgenes.pdf"),
      width = 11,
      height = 8)
    print(dendrogram.plot)
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(counts.list)[i],"_dendrogram_allgenes.png"),
      width = 11,
      height = 8,
      units = "in",res=300)
    print(dendrogram.plot)
  dev.off()
  
  print(dendrogram.plot)
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wOo, Darby et al 2012
                     </h5>
                     <a aria-label="Permalink: wOo, Darby et al 2012" class="anchor" href="#woo-darby-et-al-2012-4" id="user-content-woo-darby-et-al-2012-4">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_dendrogram_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_dendrogram_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Darby et al 2014
                     </h5>
                     <a aria-label="Permalink: wMel, Darby et al 2014" class="anchor" href="#wmel-darby-et-al-2014-5" id="user-content-wmel-darby-et-al-2014-5">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_dendrogram_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_dendrogram_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2014
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2014" class="anchor" href="#wdi-luck-et-al-2014-4" id="user-content-wdi-luck-et-al-2014-4">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_dendrogram_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_dendrogram_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2015
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2015" class="anchor" href="#wdi-luck-et-al-2015-5" id="user-content-wdi-luck-et-al-2015-5">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_dendrogram_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_dendrogram_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Gutzwiller et al 2015
                     </h5>
                     <a aria-label="Permalink: wMel, Gutzwiller et al 2015" class="anchor" href="#wmel-gutzwiller-et-al-2015-5" id="user-content-wmel-gutzwiller-et-al-2015-5">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_dendrogram_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_dendrogram_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Grote et al 2017
                     </h5>
                     <a aria-label="Permalink: wBm, Grote et al 2017" class="anchor" href="#wbm-grote-et-al-2017-5" id="user-content-wbm-grote-et-al-2017-5">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_dendrogram_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_dendrogram_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Chung et al 2019
                     </h5>
                     <a aria-label="Permalink: wBm, Chung et al 2019" class="anchor" href="#wbm-chung-et-al-2019-5" id="user-content-wbm-chung-et-al-2019-5">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_dendrogram_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_dendrogram_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Kept genes
                      <a name="user-content-ranalysis_de_hc_kept">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Kept genes " class="anchor" href="#kept-genes--1" id="user-content-kept-genes--1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(tpm.kept.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.kept.list)[i] &amp; sample_map$sample_identifier %in% colnames(tpm.kept.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  dendrogram &lt;- as.data.frame(t(scale(t(log2(tpm.kept.list[[i]]) + 1))))
  if(nrow(dendrogram) &gt; 0){
    result &lt;- pvclust(dendrogram, method.dist="cor", method.hclust="average", nboot=100, quiet=T)
    
    structure &lt;- get_dendro_structure(result)
    dendro.data &lt;- get_dendro_data(result)
    bootstrap.positions &lt;- get_dendro_bootstraps(dendro.data)
      
    points.df &lt;- as.data.frame(cbind(seq(1,length(structure),1),
                                     structure))
    dendrogroups &lt;- groups[,2][result$hclust$order]
    dendrocol &lt;- groups[,3][result$hclust$order]
    dendrosize &lt;- colSums(counts.kept.list[[i]])[result$hclust$order]
    
    for(j in 1:length(unique(groups[,1]))){
      result$hclust$labels &lt;- gsub(groups[,1][j],groups[,2][j],result$hclust$labels)
    }
    
    dendrogram.plot &lt;- ggdendrogram(hang.dendrogram(as.dendrogram(result$hclust)), theme_dendro = T)+
      geom_point(aes(x=seq(1,length(structure)), y = structure, color = dendrogroups, size = dendrosize))+
      scale_color_manual(values = levels(groups[,3]))+
      labs(x = "", y = "", col = "Samples", size = "Reads Mapped\nto Features")+
      guides(colour = guide_legend(ncol = 2))+
      theme_minimal()+
      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = "log")+
      theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust = 1),
            axis.text.y = element_blank(),
            panel.grid.major = element_blank(),
            panel.grid.minor = element_blank(),
            legend.position="none")
    
    for(j in 1:length(result$edges$bp)){
      text &lt;- round(result$edges$bp[j] * 100,0)
      dendrogram.plot &lt;- dendrogram.plot + annotate("text", label = text, x=bootstrap.positions[j,1] + 0.4, y=bootstrap.positions[j,2] + 0.03, size = 3)
    }
    pdf(paste0(OUTPUT.DIR,"/",names(counts.kept.list)[i],"_dendrogram_keptgenes.pdf"),
        width = 11,
        height = 8)
      print(dendrogram.plot)
    dev.off()
    
    png(paste0(OUTPUT.DIR,"/",names(counts.kept.list)[i],"_dendrogram_keptgenes.png"),
        width = 11,
        height = 8,
        units = "in",res=300)
      print(dendrogram.plot)
    dev.off()
    
    print(dendrogram.plot)
  }
}'>
                     <pre class="notranslate" lang="{R,"><code>for(i in 1:length(tpm.kept.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.kept.list)[i] &amp; sample_map$sample_identifier %in% colnames(tpm.kept.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  dendrogram &lt;- as.data.frame(t(scale(t(log2(tpm.kept.list[[i]]) + 1))))
  if(nrow(dendrogram) &gt; 0){
    result &lt;- pvclust(dendrogram, method.dist="cor", method.hclust="average", nboot=100, quiet=T)
    
    structure &lt;- get_dendro_structure(result)
    dendro.data &lt;- get_dendro_data(result)
    bootstrap.positions &lt;- get_dendro_bootstraps(dendro.data)
      
    points.df &lt;- as.data.frame(cbind(seq(1,length(structure),1),
                                     structure))
    dendrogroups &lt;- groups[,2][result$hclust$order]
    dendrocol &lt;- groups[,3][result$hclust$order]
    dendrosize &lt;- colSums(counts.kept.list[[i]])[result$hclust$order]
    
    for(j in 1:length(unique(groups[,1]))){
      result$hclust$labels &lt;- gsub(groups[,1][j],groups[,2][j],result$hclust$labels)
    }
    
    dendrogram.plot &lt;- ggdendrogram(hang.dendrogram(as.dendrogram(result$hclust)), theme_dendro = T)+
      geom_point(aes(x=seq(1,length(structure)), y = structure, color = dendrogroups, size = dendrosize))+
      scale_color_manual(values = levels(groups[,3]))+
      labs(x = "", y = "", col = "Samples", size = "Reads Mapped\nto Features")+
      guides(colour = guide_legend(ncol = 2))+
      theme_minimal()+
      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = "log")+
      theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust = 1),
            axis.text.y = element_blank(),
            panel.grid.major = element_blank(),
            panel.grid.minor = element_blank(),
            legend.position="none")
    
    for(j in 1:length(result$edges$bp)){
      text &lt;- round(result$edges$bp[j] * 100,0)
      dendrogram.plot &lt;- dendrogram.plot + annotate("text", label = text, x=bootstrap.positions[j,1] + 0.4, y=bootstrap.positions[j,2] + 0.03, size = 3)
    }
    pdf(paste0(OUTPUT.DIR,"/",names(counts.kept.list)[i],"_dendrogram_keptgenes.pdf"),
        width = 11,
        height = 8)
      print(dendrogram.plot)
    dev.off()
    
    png(paste0(OUTPUT.DIR,"/",names(counts.kept.list)[i],"_dendrogram_keptgenes.png"),
        width = 11,
        height = 8,
        units = "in",res=300)
      print(dendrogram.plot)
    dev.off()
    
    print(dendrogram.plot)
  }
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wOo, Darby et al 2012
                     </h5>
                     <a aria-label="Permalink: wOo, Darby et al 2012" class="anchor" href="#woo-darby-et-al-2012-5" id="user-content-woo-darby-et-al-2012-5">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_dendrogram_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_dendrogram_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Darby et al 2014
                     </h5>
                     <a aria-label="Permalink: wMel, Darby et al 2014" class="anchor" href="#wmel-darby-et-al-2014-6" id="user-content-wmel-darby-et-al-2014-6">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_dendrogram_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_dendrogram_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2014
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2014" class="anchor" href="#wdi-luck-et-al-2014-5" id="user-content-wdi-luck-et-al-2014-5">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_dendrogram_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_dendrogram_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2015
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2015" class="anchor" href="#wdi-luck-et-al-2015-6" id="user-content-wdi-luck-et-al-2015-6">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_dendrogram_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_dendrogram_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Gutzwiller et al 2015
                     </h5>
                     <a aria-label="Permalink: wMel, Gutzwiller et al 2015" class="anchor" href="#wmel-gutzwiller-et-al-2015-6" id="user-content-wmel-gutzwiller-et-al-2015-6">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_dendrogram_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_dendrogram_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Grote et al 2017
                     </h5>
                     <a aria-label="Permalink: wBm, Grote et al 2017" class="anchor" href="#wbm-grote-et-al-2017-6" id="user-content-wbm-grote-et-al-2017-6">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_dendrogram_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_dendrogram_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Chung et al 2019
                     </h5>
                     <a aria-label="Permalink: wBm, Chung et al 2019" class="anchor" href="#wbm-chung-et-al-2019-6" id="user-content-wbm-chung-et-al-2019-6">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_dendrogram_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_dendrogram_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Differentially expressed genes
                      <a name="user-content-ranalysis_de_hc_de">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Differentially expressed genes " class="anchor" href="#differentially-expressed-genes--1" id="user-content-differentially-expressed-genes--1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(tpm.edgeR.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.edgeR.list)[i] &amp; sample_map$sample_identifier %in% colnames(tpm.edgeR.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  dendrogram &lt;- as.data.frame(t(scale(t(log2(tpm.edgeR.list[[i]]) + 1))))
  if(nrow(dendrogram) &gt; 0){
    result &lt;- pvclust(dendrogram, method.dist="cor", method.hclust="average", nboot=100, quiet=T)
    
    structure &lt;- get_dendro_structure(result)
    dendro.data &lt;- get_dendro_data(result)
    bootstrap.positions &lt;- get_dendro_bootstraps(dendro.data)
      
    points.df &lt;- as.data.frame(cbind(seq(1,length(structure),1),
                                     structure))
    dendrogroups &lt;- groups[,2][result$hclust$order]
    dendrocol &lt;- groups[,3][result$hclust$order]
    dendrosize &lt;- colSums(counts.edgeR.list[[i]])[result$hclust$order]
    
    for(j in 1:length(unique(groups[,1]))){
      result$hclust$labels &lt;- gsub(groups[,1][j],groups[,2][j],result$hclust$labels)
    }
    
    dendrogram.plot &lt;- ggdendrogram(hang.dendrogram(as.dendrogram(result$hclust)), theme_dendro = T)+
      geom_point(aes(x=seq(1,length(structure)), y = structure, color = dendrogroups, size = dendrosize))+
      scale_color_manual(values = levels(groups[,3]))+
      labs(x = "", y = "", col = "Samples", size = "Reads Mapped\nto Features")+
      guides(colour = guide_legend(ncol = 2))+
      theme_minimal()+
      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = "log")+
      theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust = 1),
            axis.text.y = element_blank(),
            panel.grid.major = element_blank(),
            panel.grid.minor = element_blank(),
            legend.position="none")
    
    for(j in 1:length(result$edges$bp)){
      text &lt;- round(result$edges$bp[j] * 100,0)
      dendrogram.plot &lt;- dendrogram.plot + annotate("text", label = text, x=bootstrap.positions[j,1] + 0.4, y=bootstrap.positions[j,2] + 0.03, size = 3)
    }
    
    pdf(paste0(OUTPUT.DIR,"/",names(counts.edgeR.list)[i],"_dendrogram_degenes.pdf"),
        width = 11,
        height = 8)
      print(dendrogram.plot)
    dev.off()
    
    png(paste0(OUTPUT.DIR,"/",names(counts.edgeR.list)[i],"_dendrogram_degenes.png"),
        width = 11,
        height = 8,
        units = "in",res=300)
      print(dendrogram.plot)
    dev.off()
    
    print(dendrogram.plot)
  }
}'>
                     <pre class="notranslate" lang="{R,fig.height=8,fig.width=11}"><code>for(i in 1:length(tpm.edgeR.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.edgeR.list)[i] &amp; sample_map$sample_identifier %in% colnames(tpm.edgeR.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  dendrogram &lt;- as.data.frame(t(scale(t(log2(tpm.edgeR.list[[i]]) + 1))))
  if(nrow(dendrogram) &gt; 0){
    result &lt;- pvclust(dendrogram, method.dist="cor", method.hclust="average", nboot=100, quiet=T)
    
    structure &lt;- get_dendro_structure(result)
    dendro.data &lt;- get_dendro_data(result)
    bootstrap.positions &lt;- get_dendro_bootstraps(dendro.data)
      
    points.df &lt;- as.data.frame(cbind(seq(1,length(structure),1),
                                     structure))
    dendrogroups &lt;- groups[,2][result$hclust$order]
    dendrocol &lt;- groups[,3][result$hclust$order]
    dendrosize &lt;- colSums(counts.edgeR.list[[i]])[result$hclust$order]
    
    for(j in 1:length(unique(groups[,1]))){
      result$hclust$labels &lt;- gsub(groups[,1][j],groups[,2][j],result$hclust$labels)
    }
    
    dendrogram.plot &lt;- ggdendrogram(hang.dendrogram(as.dendrogram(result$hclust)), theme_dendro = T)+
      geom_point(aes(x=seq(1,length(structure)), y = structure, color = dendrogroups, size = dendrosize))+
      scale_color_manual(values = levels(groups[,3]))+
      labs(x = "", y = "", col = "Samples", size = "Reads Mapped\nto Features")+
      guides(colour = guide_legend(ncol = 2))+
      theme_minimal()+
      scale_size_continuous(limits=size_limits,breaks=size_breaks,trans = "log")+
      theme(axis.text.x = element_text(angle = 90, vjust = 0, hjust = 1),
            axis.text.y = element_blank(),
            panel.grid.major = element_blank(),
            panel.grid.minor = element_blank(),
            legend.position="none")
    
    for(j in 1:length(result$edges$bp)){
      text &lt;- round(result$edges$bp[j] * 100,0)
      dendrogram.plot &lt;- dendrogram.plot + annotate("text", label = text, x=bootstrap.positions[j,1] + 0.4, y=bootstrap.positions[j,2] + 0.03, size = 3)
    }
    
    pdf(paste0(OUTPUT.DIR,"/",names(counts.edgeR.list)[i],"_dendrogram_degenes.pdf"),
        width = 11,
        height = 8)
      print(dendrogram.plot)
    dev.off()
    
    png(paste0(OUTPUT.DIR,"/",names(counts.edgeR.list)[i],"_dendrogram_degenes.png"),
        width = 11,
        height = 8,
        units = "in",res=300)
      print(dendrogram.plot)
    dev.off()
    
    print(dendrogram.plot)
  }
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Darby et al 2014
                     </h5>
                     <a aria-label="Permalink: wMel, Darby et al 2014" class="anchor" href="#wmel-darby-et-al-2014-7" id="user-content-wmel-darby-et-al-2014-7">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_dendrogram_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_dendrogram_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2015
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2015" class="anchor" href="#wdi-luck-et-al-2015-7" id="user-content-wdi-luck-et-al-2015-7">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_dendrogram_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_dendrogram_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Gutzwiller et al 2015
                     </h5>
                     <a aria-label="Permalink: wMel, Gutzwiller et al 2015" class="anchor" href="#wmel-gutzwiller-et-al-2015-7" id="user-content-wmel-gutzwiller-et-al-2015-7">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_dendrogram_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_dendrogram_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Grote et al 2017
                     </h5>
                     <a aria-label="Permalink: wBm, Grote et al 2017" class="anchor" href="#wbm-grote-et-al-2017-7" id="user-content-wbm-grote-et-al-2017-7">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_dendrogram_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_dendrogram_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Chung et al 2019
                     </h5>
                     <a aria-label="Permalink: wBm, Chung et al 2019" class="anchor" href="#wbm-chung-et-al-2019-7" id="user-content-wbm-chung-et-al-2019-7">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_dendrogram_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_dendrogram_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Create TPM and log2TPM heatmaps
                      <a name="user-content-ranalysis_de_heatmap">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Create TPM and log2TPM heatmaps " class="anchor" href="#create-tpm-and-log2tpm-heatmaps-" id="user-content-create-tpm-and-log2tpm-heatmaps-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      All genes
                      <a name="user-content-ranalysis_de_heatmap_all">
                      </a>
                     </h4>
                     <a aria-label="Permalink: All genes " class="anchor" href="#all-genes--2" id="user-content-all-genes--2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(tpm.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.list)[i] &amp; sample_map$sample_identifier %in% colnames(tpm.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  heatmap1.df &lt;- as.data.frame(log2(tpm.list[[i]]+1))
  heatmap2.df &lt;- as.data.frame(t(scale(t(log2(tpm.list[[i]] + 1)))))
  
  colcol &lt;- as.character(groups[,3])
  
  #rowdendro &lt;- pvclust(t(heatmap1.df), method.dist="cor", method.hclust="average", nboot=100)
  coldendro &lt;- pvclust(heatmap2.df, method.dist="cor", method.hclust="average", nboot=100, quiet=T)
  
  heatmap2.df[is.na(heatmap2.df)] &lt;- 0
  
  pdf(paste0(OUTPUT.DIR,"/",names(tpm.list)[i],"_heatmap_allgenes.pdf"),
      width = 8,
      height = 10)
  heatmap.2(as.matrix(heatmap1.df),
            col=hmcol1,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap1.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(0,10,by=.5),
            symkey=F,
            dendrogram = "both")
  
  heatmap.2(as.matrix(heatmap2.df),
            col=hmcol2,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap2.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(-3,3,by=.5),
            symkey=F,
            dendrogram = "both")
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(tpm.list)[i],"_heatmap_allgenes.png"),
      width = 8,
      height = 10,
      units = "in",res=300)
  heatmap.2(as.matrix(heatmap1.df),
            col=hmcol1,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap1.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(0,10,by=.5),
            symkey=F,
            dendrogram = "both")
  
  heatmap.2(as.matrix(heatmap2.df),
            col=hmcol2,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap2.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(-3,3,by=.5),
            symkey=F,
            dendrogram = "both")
  dev.off()
  
  heatmap.2(as.matrix(heatmap1.df),
            col=hmcol1,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap1.df)),
            labCol=groups[,2],
           #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(0,10,by=.5),
            symkey=F,
            dendrogram = "both")
  
  heatmap.2(as.matrix(heatmap2.df),
            col=hmcol2,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap2.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(-3,3,by=.5),
            symkey=F,
            dendrogram = "both")
}'>
                     <pre class="notranslate" lang="{R,fig.height=10,fig.width=8}"><code>for(i in 1:length(tpm.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.list)[i] &amp; sample_map$sample_identifier %in% colnames(tpm.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  heatmap1.df &lt;- as.data.frame(log2(tpm.list[[i]]+1))
  heatmap2.df &lt;- as.data.frame(t(scale(t(log2(tpm.list[[i]] + 1)))))
  
  colcol &lt;- as.character(groups[,3])
  
  #rowdendro &lt;- pvclust(t(heatmap1.df), method.dist="cor", method.hclust="average", nboot=100)
  coldendro &lt;- pvclust(heatmap2.df, method.dist="cor", method.hclust="average", nboot=100, quiet=T)
  
  heatmap2.df[is.na(heatmap2.df)] &lt;- 0
  
  pdf(paste0(OUTPUT.DIR,"/",names(tpm.list)[i],"_heatmap_allgenes.pdf"),
      width = 8,
      height = 10)
  heatmap.2(as.matrix(heatmap1.df),
            col=hmcol1,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap1.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(0,10,by=.5),
            symkey=F,
            dendrogram = "both")
  
  heatmap.2(as.matrix(heatmap2.df),
            col=hmcol2,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap2.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(-3,3,by=.5),
            symkey=F,
            dendrogram = "both")
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(tpm.list)[i],"_heatmap_allgenes.png"),
      width = 8,
      height = 10,
      units = "in",res=300)
  heatmap.2(as.matrix(heatmap1.df),
            col=hmcol1,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap1.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(0,10,by=.5),
            symkey=F,
            dendrogram = "both")
  
  heatmap.2(as.matrix(heatmap2.df),
            col=hmcol2,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap2.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(-3,3,by=.5),
            symkey=F,
            dendrogram = "both")
  dev.off()
  
  heatmap.2(as.matrix(heatmap1.df),
            col=hmcol1,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap1.df)),
            labCol=groups[,2],
           #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(0,10,by=.5),
            symkey=F,
            dendrogram = "both")
  
  heatmap.2(as.matrix(heatmap2.df),
            col=hmcol2,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap2.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(-3,3,by=.5),
            symkey=F,
            dendrogram = "both")
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wOo, Darby et al 2012
                     </h5>
                     <a aria-label="Permalink: wOo, Darby et al 2012" class="anchor" href="#woo-darby-et-al-2012-6" id="user-content-woo-darby-et-al-2012-6">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_heatmap_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_heatmap_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Darby et al 2014
                     </h5>
                     <a aria-label="Permalink: wMel, Darby et al 2014" class="anchor" href="#wmel-darby-et-al-2014-8" id="user-content-wmel-darby-et-al-2014-8">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_heatmap_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_heatmap_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2014
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2014" class="anchor" href="#wdi-luck-et-al-2014-6" id="user-content-wdi-luck-et-al-2014-6">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_heatmap_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_heatmap_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2015
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2015" class="anchor" href="#wdi-luck-et-al-2015-8" id="user-content-wdi-luck-et-al-2015-8">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_heatmap_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_heatmap_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Gutzwiller et al 2015
                     </h5>
                     <a aria-label="Permalink: wMel, Gutzwiller et al 2015" class="anchor" href="#wmel-gutzwiller-et-al-2015-8" id="user-content-wmel-gutzwiller-et-al-2015-8">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_heatmap_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_heatmap_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Grote et al 2017
                     </h5>
                     <a aria-label="Permalink: wBm, Grote et al 2017" class="anchor" href="#wbm-grote-et-al-2017-8" id="user-content-wbm-grote-et-al-2017-8">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_heatmap_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_heatmap_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Chung et al 2019
                     </h5>
                     <a aria-label="Permalink: wBm, Chung et al 2019" class="anchor" href="#wbm-chung-et-al-2019-8" id="user-content-wbm-chung-et-al-2019-8">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_heatmap_allgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_heatmap_allgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Kept genes
                      <a name="user-content-ranalysis_de_heatmap_kept">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Kept genes " class="anchor" href="#kept-genes--2" id="user-content-kept-genes--2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(tpm.kept.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.kept.list)[i] &amp; sample_map$sample_identifier %in% colnames(tpm.kept.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  heatmap1.df &lt;- as.data.frame(log2(tpm.kept.list[[i]]+1))
  heatmap2.df &lt;- as.data.frame(t(scale(t(log2(tpm.kept.list[[i]] + 1)))))
  
  colcol &lt;- as.character(groups[,3])
  
  #rowdendro &lt;- pvclust(t(heatmap1.df), method.dist="cor", method.hclust="average", nboot=100)
  coldendro &lt;- pvclust(heatmap2.df, method.dist="cor", method.hclust="average", nboot=100, quiet=T)
  
  heatmap2.df[is.na(heatmap2.df)] &lt;- 0
  
  pdf(paste0(OUTPUT.DIR,"/",names(tpm.kept.list)[i],"_heatmap_keptgenes.pdf"),
      width = 8,
      height = 10)
  heatmap.2(as.matrix(heatmap1.df),
            col=hmcol1,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap1.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(0,10,by=.5),
            symkey=F,
            dendrogram = "both")
  
  heatmap.2(as.matrix(heatmap2.df),
            col=hmcol2,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap2.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(-3,3,by=.5),
            symkey=F,
            dendrogram = "both")
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(tpm.kept.list)[i],"_heatmap_keptgenes.png"),
      width = 8,
      height = 10,
      units = "in",res=300)
  heatmap.2(as.matrix(heatmap1.df),
            col=hmcol1,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap1.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(0,10,by=.5),
            symkey=F,
            dendrogram = "both")
  
  heatmap.2(as.matrix(heatmap2.df),
            col=hmcol2,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap2.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(-3,3,by=.5),
            symkey=F,
            dendrogram = "both")
  dev.off()
  
  heatmap.2(as.matrix(heatmap1.df),
            col=hmcol1,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap1.df)),
            labCol=groups[,2],
           #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(0,10,by=.5),
            symkey=F,
            dendrogram = "both")
  
  heatmap.2(as.matrix(heatmap2.df),
            col=hmcol2,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap2.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(-3,3,by=.5),
            symkey=F,
            dendrogram = "both")
}'>
                     <pre class="notranslate" lang="{R,"><code>for(i in 1:length(tpm.kept.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.kept.list)[i] &amp; sample_map$sample_identifier %in% colnames(tpm.kept.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  heatmap1.df &lt;- as.data.frame(log2(tpm.kept.list[[i]]+1))
  heatmap2.df &lt;- as.data.frame(t(scale(t(log2(tpm.kept.list[[i]] + 1)))))
  
  colcol &lt;- as.character(groups[,3])
  
  #rowdendro &lt;- pvclust(t(heatmap1.df), method.dist="cor", method.hclust="average", nboot=100)
  coldendro &lt;- pvclust(heatmap2.df, method.dist="cor", method.hclust="average", nboot=100, quiet=T)
  
  heatmap2.df[is.na(heatmap2.df)] &lt;- 0
  
  pdf(paste0(OUTPUT.DIR,"/",names(tpm.kept.list)[i],"_heatmap_keptgenes.pdf"),
      width = 8,
      height = 10)
  heatmap.2(as.matrix(heatmap1.df),
            col=hmcol1,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap1.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(0,10,by=.5),
            symkey=F,
            dendrogram = "both")
  
  heatmap.2(as.matrix(heatmap2.df),
            col=hmcol2,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap2.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(-3,3,by=.5),
            symkey=F,
            dendrogram = "both")
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(tpm.kept.list)[i],"_heatmap_keptgenes.png"),
      width = 8,
      height = 10,
      units = "in",res=300)
  heatmap.2(as.matrix(heatmap1.df),
            col=hmcol1,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap1.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(0,10,by=.5),
            symkey=F,
            dendrogram = "both")
  
  heatmap.2(as.matrix(heatmap2.df),
            col=hmcol2,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap2.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(-3,3,by=.5),
            symkey=F,
            dendrogram = "both")
  dev.off()
  
  heatmap.2(as.matrix(heatmap1.df),
            col=hmcol1,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap1.df)),
            labCol=groups[,2],
           #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(0,10,by=.5),
            symkey=F,
            dendrogram = "both")
  
  heatmap.2(as.matrix(heatmap2.df),
            col=hmcol2,
            trace="none",
            labRow=vector(mode = "character", length = nrow(heatmap2.df)),
            labCol=groups[,2],
            #Rowv = as.dendrogram(rowdendro),
            Colv = as.dendrogram(coldendro),
            ColSideColors=colcol,
            lwid=c(1,4),
            lhei = c(1,5),
            breaks = seq(-3,3,by=.5),
            symkey=F,
            dendrogram = "both")
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wOo, Darby et al 2012
                     </h5>
                     <a aria-label="Permalink: wOo, Darby et al 2012" class="anchor" href="#woo-darby-et-al-2012-7" id="user-content-woo-darby-et-al-2012-7">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wOo_darby_2012_heatmap_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wOo_darby_2012_heatmap_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Darby et al 2014
                     </h5>
                     <a aria-label="Permalink: wMel, Darby et al 2014" class="anchor" href="#wmel-darby-et-al-2014-9" id="user-content-wmel-darby-et-al-2014-9">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_heatmap_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_heatmap_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2014
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2014" class="anchor" href="#wdi-luck-et-al-2014-7" id="user-content-wdi-luck-et-al-2014-7">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2014_heatmap_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2014_heatmap_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2015
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2015" class="anchor" href="#wdi-luck-et-al-2015-9" id="user-content-wdi-luck-et-al-2015-9">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_heatmap_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_heatmap_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Gutzwiller et al 2015
                     </h5>
                     <a aria-label="Permalink: wMel, Gutzwiller et al 2015" class="anchor" href="#wmel-gutzwiller-et-al-2015-9" id="user-content-wmel-gutzwiller-et-al-2015-9">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_heatmap_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_heatmap_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Grote et al 2017
                     </h5>
                     <a aria-label="Permalink: wBm, Grote et al 2017" class="anchor" href="#wbm-grote-et-al-2017-9" id="user-content-wbm-grote-et-al-2017-9">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_heatmap_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_heatmap_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Chung et al 2019
                     </h5>
                     <a aria-label="Permalink: wBm, Chung et al 2019" class="anchor" href="#wbm-chung-et-al-2019-9" id="user-content-wbm-chung-et-al-2019-9">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_heatmap_keptgenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_heatmap_keptgenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Differentially expressed genes
                      <a name="user-content-ranalysis_de_heatmap_de">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Differentially expressed genes " class="anchor" href="#differentially-expressed-genes--2" id="user-content-differentially-expressed-genes--2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(tpm.edgeR.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.edgeR.list)[i] &amp; sample_map$sample_identifier %in% colnames(tpm.edgeR.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  heatmap1.df &lt;- as.data.frame(log2(tpm.edgeR.list[[i]]+1))
  heatmap2.df &lt;- as.data.frame(t(scale(t(log2(tpm.edgeR.list[[i]] + 1)))))
  
  colcol &lt;- as.character(groups[,3])
  
  if(nrow(tpm.edgeR.list[[i]]) &gt; 0){
    #rowdendro &lt;- pvclust(t(heatmap1.df), method.dist="cor", method.hclust="average", nboot=100)
    coldendro &lt;- pvclust(heatmap2.df, method.dist="cor", method.hclust="average", nboot=100, quiet=T)
    
    heatmap2.df[is.na(heatmap2.df)] &lt;- 0
    
    pdf(paste0(OUTPUT.DIR,"/",names(tpm.edgeR.list)[i],"_heatmap_degenes.pdf"),
        width = 8,
        height = 10)
    heatmap.2(as.matrix(heatmap1.df),
              col=hmcol1,
              trace="none",
              labRow=vector(mode = "character", length = nrow(heatmap1.df)),
              labCol=groups[,2],
              #Rowv = as.dendrogram(rowdendro),
              Colv = as.dendrogram(coldendro),
              ColSideColors=colcol,
              lwid=c(1,4),
              lhei = c(1,5),
              breaks = seq(0,10,by=.5),
              symkey=F,
              dendrogram = "both")
    
    heatmap.2(as.matrix(heatmap2.df),
              col=hmcol2,
              trace="none",
              labRow=vector(mode = "character", length = nrow(heatmap2.df)),
              labCol=groups[,2],
              #Rowv = as.dendrogram(rowdendro),
              Colv = as.dendrogram(coldendro),
              ColSideColors=colcol,
              lwid=c(1,4),
              lhei = c(1,5),
              breaks = seq(-3,3,by=.5),
              symkey=F,
              dendrogram = "both")
    dev.off()
    
    png(paste0(OUTPUT.DIR,"/",names(tpm.edgeR.list)[i],"_heatmap_keptgenes.png"),
        width = 8,
        height = 10,
        units = "in",res=300)
    heatmap.2(as.matrix(heatmap1.df),
              col=hmcol1,
              trace="none",
              labRow=vector(mode = "character", length = nrow(heatmap1.df)),
              labCol=groups[,2],
              #Rowv = as.dendrogram(rowdendro),
              Colv = as.dendrogram(coldendro),
              ColSideColors=colcol,
              lwid=c(1,4),
              lhei = c(1,5),
              breaks = seq(0,10,by=.5),
              symkey=F,
              dendrogram = "both")
    
    heatmap.2(as.matrix(heatmap2.df),
              col=hmcol2,
              trace="none",
              labRow=vector(mode = "character", length = nrow(heatmap2.df)),
              labCol=groups[,2],
              #Rowv = as.dendrogram(rowdendro),
              Colv = as.dendrogram(coldendro),
              ColSideColors=colcol,
              lwid=c(1,4),
              lhei = c(1,5),
              breaks = seq(-3,3,by=.5),
              symkey=F,
              dendrogram = "both")
    dev.off()
    
    heatmap.2(as.matrix(heatmap1.df),
              col=hmcol1,
              trace="none",
              labRow=vector(mode = "character", length = nrow(heatmap1.df)),
              labCol=groups[,2],
             #Rowv = as.dendrogram(rowdendro),
              Colv = as.dendrogram(coldendro),
              ColSideColors=colcol,
              lwid=c(1,4),
              lhei = c(1,5),
              breaks = seq(0,10,by=.5),
              symkey=F,
              dendrogram = "both")
    
    heatmap.2(as.matrix(heatmap2.df),
              col=hmcol2,
              trace="none",
              labRow=vector(mode = "character", length = nrow(heatmap2.df)),
              labCol=groups[,2],
              #Rowv = as.dendrogram(rowdendro),
              Colv = as.dendrogram(coldendro),
              ColSideColors=colcol,
              lwid=c(1,4),
              lhei = c(1,5),
              breaks = seq(-3,3,by=.5),
              symkey=F,
              dendrogram = "both")
  }
}'>
                     <pre class="notranslate" lang="{R,"><code>for(i in 1:length(tpm.edgeR.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.edgeR.list)[i] &amp; sample_map$sample_identifier %in% colnames(tpm.edgeR.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)
  
  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  heatmap1.df &lt;- as.data.frame(log2(tpm.edgeR.list[[i]]+1))
  heatmap2.df &lt;- as.data.frame(t(scale(t(log2(tpm.edgeR.list[[i]] + 1)))))
  
  colcol &lt;- as.character(groups[,3])
  
  if(nrow(tpm.edgeR.list[[i]]) &gt; 0){
    #rowdendro &lt;- pvclust(t(heatmap1.df), method.dist="cor", method.hclust="average", nboot=100)
    coldendro &lt;- pvclust(heatmap2.df, method.dist="cor", method.hclust="average", nboot=100, quiet=T)
    
    heatmap2.df[is.na(heatmap2.df)] &lt;- 0
    
    pdf(paste0(OUTPUT.DIR,"/",names(tpm.edgeR.list)[i],"_heatmap_degenes.pdf"),
        width = 8,
        height = 10)
    heatmap.2(as.matrix(heatmap1.df),
              col=hmcol1,
              trace="none",
              labRow=vector(mode = "character", length = nrow(heatmap1.df)),
              labCol=groups[,2],
              #Rowv = as.dendrogram(rowdendro),
              Colv = as.dendrogram(coldendro),
              ColSideColors=colcol,
              lwid=c(1,4),
              lhei = c(1,5),
              breaks = seq(0,10,by=.5),
              symkey=F,
              dendrogram = "both")
    
    heatmap.2(as.matrix(heatmap2.df),
              col=hmcol2,
              trace="none",
              labRow=vector(mode = "character", length = nrow(heatmap2.df)),
              labCol=groups[,2],
              #Rowv = as.dendrogram(rowdendro),
              Colv = as.dendrogram(coldendro),
              ColSideColors=colcol,
              lwid=c(1,4),
              lhei = c(1,5),
              breaks = seq(-3,3,by=.5),
              symkey=F,
              dendrogram = "both")
    dev.off()
    
    png(paste0(OUTPUT.DIR,"/",names(tpm.edgeR.list)[i],"_heatmap_keptgenes.png"),
        width = 8,
        height = 10,
        units = "in",res=300)
    heatmap.2(as.matrix(heatmap1.df),
              col=hmcol1,
              trace="none",
              labRow=vector(mode = "character", length = nrow(heatmap1.df)),
              labCol=groups[,2],
              #Rowv = as.dendrogram(rowdendro),
              Colv = as.dendrogram(coldendro),
              ColSideColors=colcol,
              lwid=c(1,4),
              lhei = c(1,5),
              breaks = seq(0,10,by=.5),
              symkey=F,
              dendrogram = "both")
    
    heatmap.2(as.matrix(heatmap2.df),
              col=hmcol2,
              trace="none",
              labRow=vector(mode = "character", length = nrow(heatmap2.df)),
              labCol=groups[,2],
              #Rowv = as.dendrogram(rowdendro),
              Colv = as.dendrogram(coldendro),
              ColSideColors=colcol,
              lwid=c(1,4),
              lhei = c(1,5),
              breaks = seq(-3,3,by=.5),
              symkey=F,
              dendrogram = "both")
    dev.off()
    
    heatmap.2(as.matrix(heatmap1.df),
              col=hmcol1,
              trace="none",
              labRow=vector(mode = "character", length = nrow(heatmap1.df)),
              labCol=groups[,2],
             #Rowv = as.dendrogram(rowdendro),
              Colv = as.dendrogram(coldendro),
              ColSideColors=colcol,
              lwid=c(1,4),
              lhei = c(1,5),
              breaks = seq(0,10,by=.5),
              symkey=F,
              dendrogram = "both")
    
    heatmap.2(as.matrix(heatmap2.df),
              col=hmcol2,
              trace="none",
              labRow=vector(mode = "character", length = nrow(heatmap2.df)),
              labCol=groups[,2],
              #Rowv = as.dendrogram(rowdendro),
              Colv = as.dendrogram(coldendro),
              ColSideColors=colcol,
              lwid=c(1,4),
              lhei = c(1,5),
              breaks = seq(-3,3,by=.5),
              symkey=F,
              dendrogram = "both")
  }
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Darby et al 2014
                     </h5>
                     <a aria-label="Permalink: wMel, Darby et al 2014" class="anchor" href="#wmel-darby-et-al-2014-10" id="user-content-wmel-darby-et-al-2014-10">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_heatmap_degenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_heatmap_degenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2015
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2015" class="anchor" href="#wdi-luck-et-al-2015-10" id="user-content-wdi-luck-et-al-2015-10">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_heatmap_degenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_heatmap_degenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Gutzwiller et al 2015
                     </h5>
                     <a aria-label="Permalink: wMel, Gutzwiller et al 2015" class="anchor" href="#wmel-gutzwiller-et-al-2015-10" id="user-content-wmel-gutzwiller-et-al-2015-10">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_heatmap_degenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_heatmap_degenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Grote et al 2017
                     </h5>
                     <a aria-label="Permalink: wBm, Grote et al 2017" class="anchor" href="#wbm-grote-et-al-2017-10" id="user-content-wbm-grote-et-al-2017-10">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_heatmap_degenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_heatmap_degenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Chung et al 2019
                     </h5>
                     <a aria-label="Permalink: wBm, Chung et al 2019" class="anchor" href="#wbm-chung-et-al-2019-10" id="user-content-wbm-chung-et-al-2019-10">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_heatmap_degenes.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_heatmap_degenes.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Choose studies for basic pairwise analyses or WGCNA
                      <a name="user-content-ranalysis_de_selectanalysis">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Choose studies for basic pairwise analyses or WGCNA " class="anchor" href="#choose-studies-for-basic-pairwise-analyses-or-wgcna-" id="user-content-choose-studies-for-basic-pairwise-analyses-or-wgcna-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pairwise.studies &lt;- c("wMel_darby_2014")
wgcna.studies &lt;- c("wDi_luck_2015","wMel_gutzwiller_2015","wBm_grote_2017","wBm_chung_2019")'>
                     <pre class="notranslate" lang="{R}"><code>pairwise.studies &lt;- c("wMel_darby_2014")
wgcna.studies &lt;- c("wDi_luck_2015","wMel_gutzwiller_2015","wBm_grote_2017","wBm_chung_2019")
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Conduct pairwise analyses on studies with two biological groups
                      <a name="user-content-ranalysis_de_pairwise">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Conduct pairwise analyses on studies with two biological groups " class="anchor" href="#conduct-pairwise-analyses-on-studies-with-two-biological-groups-" id="user-content-conduct-pairwise-analyses-on-studies-with-two-biological-groups-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Create z-score of log2TPM heatmaps with a column bar that shows up- and down-regulated genes
                      <a name="user-content-ranalysis_de_pairwise_hm">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Create z-score of log2TPM heatmaps with a column bar that shows up- and down-regulated genes " class="anchor" href="#create-z-score-of-log2tpm-heatmaps-with-a-column-bar-that-shows-up--and-down-regulated-genes-" id="user-content-create-z-score-of-log2tpm-heatmaps-with-a-column-bar-that-shows-up--and-down-regulated-genes-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(pairwise.studies)){
  sample_map.subset &lt;- sample_map[sample_map$study == pairwise.studies[i],]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)

  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  qlf &lt;- as.data.frame(qlf.list[names(qlf.list) == pairwise.studies[i]])
  qlf &lt;- qlf[order(qlf[,1],decreasing = T),]
  qlf &lt;- qlf[rownames(qlf) %in% rownames(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]])),]
  
  log2tpm.de &lt;- log2(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]])+1)
  zscore.log2tpm.de &lt;- as.data.frame(t(scale(t(log2tpm.de))))
  zscore.log2tpm.de &lt;- zscore.log2tpm.de[match(rownames(qlf),rownames(zscore.log2tpm.de)),]
  
  hmcol1 &lt;- colorRampPalette(c("navyblue","white","firebrick3"))(12)
  hmcol2 &lt;- colorRampPalette(c("#FFFFFF","#FDE0D2","#FABAA1","#F69274", "#F16B4E","#EF3D2D","#590A16","#A51E23","#650C16"))(30)
  rowcol1 &lt;- ifelse(qlf[,1] &gt; 0,"red","blue")
  colcol &lt;- as.character(groups[,3])
  rowsep &lt;- get_heatmap_separators(rowcol1)
  colsep &lt;- get_heatmap_separators(colcol)
  
  coldendro &lt;- pvclust(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]]), method.dist="cor", method.hclust="average", nboot=100, quiet=T)

  pdf(paste0(OUTPUT.DIR,"/",pairwise.studies[i],"_pairwise_zscorelog2tpm_heatmap.pdf"),
      width = 5,
      height = 10)
    
  heatmap.2(as.matrix(zscore.log2tpm.de),
              col=hmcol1,
              trace="none",
              labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
              Rowv = F,
              Colv = as.dendrogram(coldendro),
              RowSideColors=rowcol1,
              ColSideColors=colcol,
              lhei = c(2,8),
              breaks = seq(-3,3,by=.5),
              rowsep = rowsep,
              colsep = colsep,
              symkey=F,
              dendrogram = "column")
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",pairwise.studies[i],"_pairwise_zscorelog2tpm_heatmap.png"),
      width = 5,
      height = 10,
      units = "in",res=300)
    
  heatmap.2(as.matrix(zscore.log2tpm.de),
              col=hmcol1,
              trace="none",
              labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
              Rowv = F,
              Colv = as.dendrogram(coldendro),
              RowSideColors=rowcol1,
              ColSideColors=colcol,
              lhei = c(2,8),
              breaks = seq(-3,3,by=.5),
              rowsep = rowsep,
              colsep = colsep,
              symkey=F,
              dendrogram = "column")
  dev.off()
  
  heatmap.2(as.matrix(zscore.log2tpm.de),
              col=hmcol1,
              trace="none",
              labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
              Rowv = F,
              Colv = as.dendrogram(coldendro),
              RowSideColors=rowcol1,
              ColSideColors=colcol,
              lhei = c(2,8),
              breaks = seq(-3,3,by=.5),
              rowsep = rowsep,
              colsep = colsep,
              symkey=F,
              dendrogram = "column")
  
}'>
                     <pre class="notranslate" lang="{R,"><code>for(i in 1:length(pairwise.studies)){
  sample_map.subset &lt;- sample_map[sample_map$study == pairwise.studies[i],]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)

  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  qlf &lt;- as.data.frame(qlf.list[names(qlf.list) == pairwise.studies[i]])
  qlf &lt;- qlf[order(qlf[,1],decreasing = T),]
  qlf &lt;- qlf[rownames(qlf) %in% rownames(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]])),]
  
  log2tpm.de &lt;- log2(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]])+1)
  zscore.log2tpm.de &lt;- as.data.frame(t(scale(t(log2tpm.de))))
  zscore.log2tpm.de &lt;- zscore.log2tpm.de[match(rownames(qlf),rownames(zscore.log2tpm.de)),]
  
  hmcol1 &lt;- colorRampPalette(c("navyblue","white","firebrick3"))(12)
  hmcol2 &lt;- colorRampPalette(c("#FFFFFF","#FDE0D2","#FABAA1","#F69274", "#F16B4E","#EF3D2D","#590A16","#A51E23","#650C16"))(30)
  rowcol1 &lt;- ifelse(qlf[,1] &gt; 0,"red","blue")
  colcol &lt;- as.character(groups[,3])
  rowsep &lt;- get_heatmap_separators(rowcol1)
  colsep &lt;- get_heatmap_separators(colcol)
  
  coldendro &lt;- pvclust(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]]), method.dist="cor", method.hclust="average", nboot=100, quiet=T)

  pdf(paste0(OUTPUT.DIR,"/",pairwise.studies[i],"_pairwise_zscorelog2tpm_heatmap.pdf"),
      width = 5,
      height = 10)
    
  heatmap.2(as.matrix(zscore.log2tpm.de),
              col=hmcol1,
              trace="none",
              labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
              Rowv = F,
              Colv = as.dendrogram(coldendro),
              RowSideColors=rowcol1,
              ColSideColors=colcol,
              lhei = c(2,8),
              breaks = seq(-3,3,by=.5),
              rowsep = rowsep,
              colsep = colsep,
              symkey=F,
              dendrogram = "column")
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",pairwise.studies[i],"_pairwise_zscorelog2tpm_heatmap.png"),
      width = 5,
      height = 10,
      units = "in",res=300)
    
  heatmap.2(as.matrix(zscore.log2tpm.de),
              col=hmcol1,
              trace="none",
              labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
              Rowv = F,
              Colv = as.dendrogram(coldendro),
              RowSideColors=rowcol1,
              ColSideColors=colcol,
              lhei = c(2,8),
              breaks = seq(-3,3,by=.5),
              rowsep = rowsep,
              colsep = colsep,
              symkey=F,
              dendrogram = "column")
  dev.off()
  
  heatmap.2(as.matrix(zscore.log2tpm.de),
              col=hmcol1,
              trace="none",
              labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
              Rowv = F,
              Colv = as.dendrogram(coldendro),
              RowSideColors=rowcol1,
              ColSideColors=colcol,
              lhei = c(2,8),
              breaks = seq(-3,3,by=.5),
              rowsep = rowsep,
              colsep = colsep,
              symkey=F,
              dendrogram = "column")
  
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Darby et al 2014
                     </h5>
                     <a aria-label="Permalink: wMel, Darby et al 2014" class="anchor" href="#wmel-darby-et-al-2014-11" id="user-content-wmel-darby-et-al-2014-11">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_darby_2014_pairwise_zscorelog2tpm_heatmap.pdf" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_darby_2014_pairwise_zscorelog2tpm_heatmap.pdf" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Conduct functional term enrichment analysis on up- and down-regulated gene subsets
                      <a name="user-content-ranalysis_de_pairwise_fxnterm">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Conduct functional term enrichment analysis on up- and down-regulated gene subsets " class="anchor" href="#conduct-functional-term-enrichment-analysis-on-up--and-down-regulated-gene-subsets-" id="user-content-conduct-functional-term-enrichment-analysis-on-up--and-down-regulated-gene-subsets-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='terms.pairwise.list &lt;- list()
terms.sig.pairwise.list &lt;- terms.pairwise.list

for(i in 1:length(pairwise.studies)){
  strain &lt;- substr(pairwise.studies,1,regexpr("_",pairwise.studies)-1)
  geneinfo &lt;- read.delim(paste0(GENEINFO.DIR,"/",strain,".cds.fna.interproscan.geneinfo.tsv"))
  
  qlf &lt;- as.data.frame(qlf.list[names(qlf.list) == pairwise.studies[i]])
  qlf &lt;- qlf[order(qlf[,1],decreasing = T),]
  qlf &lt;- qlf[rownames(qlf) %in% rownames(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]])),]
  
  terms.colnames &lt;- c("term","clusteroccurences","genomeoccurences","pvalue","correctedpvalue","oddsratio","group")
  terms.pairwise &lt;- as.data.frame(matrix(nrow = 0,
                                         ncol = 7))
  colnames(terms.pairwise) &lt;- terms.colnames
  
  terms.pairwise.up &lt;- as.data.frame(cbind(functionaltermenrichment(rownames(qlf)[qlf[,1] &gt; 0],geneinfo),
                                           "upregulated"))
  terms.pairwise.down &lt;- as.data.frame(cbind(functionaltermenrichment(rownames(qlf)[qlf[,1] &lt; 0],geneinfo),
                                       "downregulated"))
  
  colnames(terms.pairwise.up) &lt;- terms.colnames
  colnames(terms.pairwise.down) &lt;- terms.colnames
  terms.pairwise &lt;- as.data.frame(rbind(terms.pairwise.up,
                                     terms.pairwise.down))
  
  for(j in 2:6){
    terms.pairwise[,j] &lt;- as.numeric(as.character(terms.pairwise[,j]))
  }
  
  terms.pairwise.list[[i]] &lt;- terms.pairwise
  terms.sig.pairwise.list[[i]] &lt;- terms.pairwise[terms.pairwise$correctedpvalue &lt; 0.05,]
  
  names(terms.pairwise.list)[i] &lt;- pairwise.studies[i]
  
  write.table(terms.pairwise.list[[i]],
            paste0(OUTPUT.DIR,"/",names(terms.pairwise.list)[i],"_functionaltermenrichment.tsv"),
            row.names = T,
            col.names = T,
            quote = F,
            sep = "\t")
}'>
                     <pre class="notranslate" lang="{R}"><code>terms.pairwise.list &lt;- list()
terms.sig.pairwise.list &lt;- terms.pairwise.list

for(i in 1:length(pairwise.studies)){
  strain &lt;- substr(pairwise.studies,1,regexpr("_",pairwise.studies)-1)
  geneinfo &lt;- read.delim(paste0(GENEINFO.DIR,"/",strain,".cds.fna.interproscan.geneinfo.tsv"))
  
  qlf &lt;- as.data.frame(qlf.list[names(qlf.list) == pairwise.studies[i]])
  qlf &lt;- qlf[order(qlf[,1],decreasing = T),]
  qlf &lt;- qlf[rownames(qlf) %in% rownames(as.data.frame(tpm.edgeR.list[names(tpm.edgeR.list) == pairwise.studies[i]])),]
  
  terms.colnames &lt;- c("term","clusteroccurences","genomeoccurences","pvalue","correctedpvalue","oddsratio","group")
  terms.pairwise &lt;- as.data.frame(matrix(nrow = 0,
                                         ncol = 7))
  colnames(terms.pairwise) &lt;- terms.colnames
  
  terms.pairwise.up &lt;- as.data.frame(cbind(functionaltermenrichment(rownames(qlf)[qlf[,1] &gt; 0],geneinfo),
                                           "upregulated"))
  terms.pairwise.down &lt;- as.data.frame(cbind(functionaltermenrichment(rownames(qlf)[qlf[,1] &lt; 0],geneinfo),
                                       "downregulated"))
  
  colnames(terms.pairwise.up) &lt;- terms.colnames
  colnames(terms.pairwise.down) &lt;- terms.colnames
  terms.pairwise &lt;- as.data.frame(rbind(terms.pairwise.up,
                                     terms.pairwise.down))
  
  for(j in 2:6){
    terms.pairwise[,j] &lt;- as.numeric(as.character(terms.pairwise[,j]))
  }
  
  terms.pairwise.list[[i]] &lt;- terms.pairwise
  terms.sig.pairwise.list[[i]] &lt;- terms.pairwise[terms.pairwise$correctedpvalue &lt; 0.05,]
  
  names(terms.pairwise.list)[i] &lt;- pairwise.studies[i]
  
  write.table(terms.pairwise.list[[i]],
            paste0(OUTPUT.DIR,"/",names(terms.pairwise.list)[i],"_functionaltermenrichment.tsv"),
            row.names = T,
            col.names = T,
            quote = F,
            sep = "\t")
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Conduct WGCNA on time-course studies
                      <a name="user-content-ranalysis_de_wgcna">
                      </a>
                     </h3>
                     <a aria-label="Permalink: Conduct WGCNA on time-course studies " class="anchor" href="#conduct-wgcna-on-time-course-studies-" id="user-content-conduct-wgcna-on-time-course-studies-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Identify WGCNA soft power values
                      <a name="user-content-ranalysis_de_wgcna_softpower">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Identify WGCNA soft power values " class="anchor" href="#identify-wgcna-soft-power-values-" id="user-content-identify-wgcna-soft-power-values-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='tpm.wgcna.list &lt;- tpm.edgeR.list
for(i in 1:length(tpm.wgcna.list)){
  tpm.wgcna.list[[i]] &lt;- as.data.frame(t(tpm.edgeR.list[[i]]))
}
tpm.wgcna.list &lt;- tpm.wgcna.list[names(tpm.wgcna.list) %in% wgcna.studies]
softpower.list &lt;- tpm.wgcna.list

for(i in 1:length(tpm.wgcna.list)){
  powers &lt;- c(1:20)
  sft &lt;- pickSoftThreshold(tpm.wgcna.list[[i]], powerVector = powers, verbose = 5)
  softpower.list[[i]] &lt;- find_soft_power(sft)
  
  text.color &lt;- rep("black",length(sft$fitIndices[,1]))
  text.color[which(sft$fitIndices[,1] == softpower.list[[i]])] &lt;- "red"
  
  scale_independence.plot &lt;- ggplot()+
    geom_text(mapping = aes(x = sft$fitIndices[,1], y = -sign(sft$fitIndices[,3])*sft$fitIndices[,2], label=sft$fitIndices[,1]), color = text.color)+
    labs(title = "scale independence", x = "soft threshold (power)", y = "scale free topology model fit, signed R^2")+
    theme_bw()
  
  mean_connectivity.plot &lt;- ggplot()+
    geom_text(mapping = aes(x = sft$fitIndices[,1], y = sft$fitIndices[,5], label=sft$fitIndices[,1]), color = text.color)+
    labs(title = "mean connectivity", x = "soft threshold (power)", y = "mean connectivity")+
    theme_bw()
  
  wgcna_soft_power_plots &lt;- list(scale_independence.plot, mean_connectivity.plot)
  lay &lt;- rbind(c(1,2))
  pdf(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_soft_power_plot.pdf"),
      width = 8, 
      height = 4)
  grid.arrange(grobs = wgcna_soft_power_plots,
               widths = c(5,5),
               heights = c(5),
               layout_matrix = lay)
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_soft_power_plot.png"),
      width = 8, 
      height = 4,
      units = "in",res=300)
  grid.arrange(grobs = wgcna_soft_power_plots,
               widths = c(5,5),
               heights = c(5),
               layout_matrix = lay)
  dev.off()
  
  grid.arrange(grobs = wgcna_soft_power_plots,
               widths = c(5,5),
               heights = c(5),
               layout_matrix = lay)
}'>
                     <pre class="notranslate" lang="{R,"><code>tpm.wgcna.list &lt;- tpm.edgeR.list
for(i in 1:length(tpm.wgcna.list)){
  tpm.wgcna.list[[i]] &lt;- as.data.frame(t(tpm.edgeR.list[[i]]))
}
tpm.wgcna.list &lt;- tpm.wgcna.list[names(tpm.wgcna.list) %in% wgcna.studies]
softpower.list &lt;- tpm.wgcna.list

for(i in 1:length(tpm.wgcna.list)){
  powers &lt;- c(1:20)
  sft &lt;- pickSoftThreshold(tpm.wgcna.list[[i]], powerVector = powers, verbose = 5)
  softpower.list[[i]] &lt;- find_soft_power(sft)
  
  text.color &lt;- rep("black",length(sft$fitIndices[,1]))
  text.color[which(sft$fitIndices[,1] == softpower.list[[i]])] &lt;- "red"
  
  scale_independence.plot &lt;- ggplot()+
    geom_text(mapping = aes(x = sft$fitIndices[,1], y = -sign(sft$fitIndices[,3])*sft$fitIndices[,2], label=sft$fitIndices[,1]), color = text.color)+
    labs(title = "scale independence", x = "soft threshold (power)", y = "scale free topology model fit, signed R^2")+
    theme_bw()
  
  mean_connectivity.plot &lt;- ggplot()+
    geom_text(mapping = aes(x = sft$fitIndices[,1], y = sft$fitIndices[,5], label=sft$fitIndices[,1]), color = text.color)+
    labs(title = "mean connectivity", x = "soft threshold (power)", y = "mean connectivity")+
    theme_bw()
  
  wgcna_soft_power_plots &lt;- list(scale_independence.plot, mean_connectivity.plot)
  lay &lt;- rbind(c(1,2))
  pdf(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_soft_power_plot.pdf"),
      width = 8, 
      height = 4)
  grid.arrange(grobs = wgcna_soft_power_plots,
               widths = c(5,5),
               heights = c(5),
               layout_matrix = lay)
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_soft_power_plot.png"),
      width = 8, 
      height = 4,
      units = "in",res=300)
  grid.arrange(grobs = wgcna_soft_power_plots,
               widths = c(5,5),
               heights = c(5),
               layout_matrix = lay)
  dev.off()
  
  grid.arrange(grobs = wgcna_soft_power_plots,
               widths = c(5,5),
               heights = c(5),
               layout_matrix = lay)
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2015
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2015" class="anchor" href="#wdi-luck-et-al-2015-11" id="user-content-wdi-luck-et-al-2015-11">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_wgcna_soft_power_plot.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_wgcna_soft_power_plot.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Gutzwiller et al 2015
                     </h5>
                     <a aria-label="Permalink: wMel, Gutzwiller et al 2015" class="anchor" href="#wmel-gutzwiller-et-al-2015-11" id="user-content-wmel-gutzwiller-et-al-2015-11">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_wgcna_soft_power_plot.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_wgcna_soft_power_plot.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Grote et al 2017
                     </h5>
                     <a aria-label="Permalink: wBm, Grote et al 2017" class="anchor" href="#wbm-grote-et-al-2017-11" id="user-content-wbm-grote-et-al-2017-11">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_wgcna_soft_power_plot.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_wgcna_soft_power_plot.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Chung et al 2019
                     </h5>
                     <a aria-label="Permalink: wBm, Chung et al 2019" class="anchor" href="#wbm-chung-et-al-2019-11" id="user-content-wbm-chung-et-al-2019-11">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_wgcna_soft_power_plot.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_wgcna_soft_power_plot.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Create WGCNA expression modules and merge similar modules
                      <a name="user-content-ranalysis_de_wgcna_createmodules">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Create WGCNA expression modules and merge similar modules " class="anchor" href="#create-wgcna-expression-modules-and-merge-similar-modules-" id="user-content-create-wgcna-expression-modules-and-merge-similar-modules-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(tpm.wgcna.list)){
  adjacency &lt;- adjacency(tpm.wgcna.list[[i]], power = softpower.list[[i]])
  TOM &lt;- TOMsimilarity(adjacency)
  dissTOM &lt;- 1-TOM
  geneTree &lt;- hclust(as.dist(dissTOM), method = "average");
  
  minModuleSize &lt;- 1
  dynamicMods &lt;- cutreeDynamic(dendro = geneTree, distM = dissTOM,
                               deepSplit = 2, pamRespectsDendro = FALSE,
                               minClusterSize = minModuleSize)
  
  dynamicColors = labels2colors(dynamicMods)
  MEList = moduleEigengenes(tpm.wgcna.list[[i]], colors = dynamicColors)
  MEs = MEList$eigengenes
  MEDiss = 1-cor(MEs, use = "pairwise.complete.obs")
  METree = hclust(as.dist(MEDiss), method = "average")
  
  MEDissThres = 0.25
  
  pdf(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_merge_eigengenes_plot.pdf"),
      width = 8,
      height = 5)
  plot(METree, main = "clustering of module eigengenes",
       xlab = "", sub = "")
  
  abline(h=MEDissThres, col = "red")
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_merge_eigengenes_plot.png"),
      width = 8,
      height = 5,
      units = "in",res=300)
  plot(METree, main = "clustering of module eigengenes",
       xlab = "", sub = "")
  abline(h=MEDissThres, col = "red")
  dev.off()
  
  plot(METree, main = "clustering of module eigengenes",
       xlab = "", sub = "")
  abline(h=MEDissThres, col = "red")
  
  merge = mergeCloseModules(tpm.wgcna.list[[i]], dynamicColors, cutHeight = MEDissThres, verbose = 3)
  mergedColors = merge$colors
  mergedMEs = merge$newMEs
  
  pdf(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_eigengene_dendrogram_plot.pdf"),
      width = 8,
      height = 5)
  plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors), c("dynamic tree cut", "merged dynamic"),
                      dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05)
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_eigengene_dendrogram_plot.png"),
      width = 8,
      height = 5,
      units = "in",res=300)
  plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors), c("dynamic tree cut", "merged dynamic"),
                      dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05)
  dev.off()
  
  plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors), c("dynamic tree cut", "merged dynamic"),
                      dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05)
  
  tpm.wgcna.list[[i]] &lt;- eigengene_invert_id(as.data.frame(t(tpm.wgcna.list[[i]])), mergedColors, mergedMEs)
  tpm.wgcna.list[[i]] &lt;- wgcna_heatmap_reorder(tpm.wgcna.list[[i]])
  
  write.table(tpm.wgcna.list[[i]],
              paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_tpm_de_modules.tsv"),
              row.names = T,
              col.names = T,
              quote = F,
              sep = "\t")
}'>
                     <pre class="notranslate" lang="{r"><code>for(i in 1:length(tpm.wgcna.list)){
  adjacency &lt;- adjacency(tpm.wgcna.list[[i]], power = softpower.list[[i]])
  TOM &lt;- TOMsimilarity(adjacency)
  dissTOM &lt;- 1-TOM
  geneTree &lt;- hclust(as.dist(dissTOM), method = "average");
  
  minModuleSize &lt;- 1
  dynamicMods &lt;- cutreeDynamic(dendro = geneTree, distM = dissTOM,
                               deepSplit = 2, pamRespectsDendro = FALSE,
                               minClusterSize = minModuleSize)
  
  dynamicColors = labels2colors(dynamicMods)
  MEList = moduleEigengenes(tpm.wgcna.list[[i]], colors = dynamicColors)
  MEs = MEList$eigengenes
  MEDiss = 1-cor(MEs, use = "pairwise.complete.obs")
  METree = hclust(as.dist(MEDiss), method = "average")
  
  MEDissThres = 0.25
  
  pdf(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_merge_eigengenes_plot.pdf"),
      width = 8,
      height = 5)
  plot(METree, main = "clustering of module eigengenes",
       xlab = "", sub = "")
  
  abline(h=MEDissThres, col = "red")
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_merge_eigengenes_plot.png"),
      width = 8,
      height = 5,
      units = "in",res=300)
  plot(METree, main = "clustering of module eigengenes",
       xlab = "", sub = "")
  abline(h=MEDissThres, col = "red")
  dev.off()
  
  plot(METree, main = "clustering of module eigengenes",
       xlab = "", sub = "")
  abline(h=MEDissThres, col = "red")
  
  merge = mergeCloseModules(tpm.wgcna.list[[i]], dynamicColors, cutHeight = MEDissThres, verbose = 3)
  mergedColors = merge$colors
  mergedMEs = merge$newMEs
  
  pdf(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_eigengene_dendrogram_plot.pdf"),
      width = 8,
      height = 5)
  plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors), c("dynamic tree cut", "merged dynamic"),
                      dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05)
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_eigengene_dendrogram_plot.png"),
      width = 8,
      height = 5,
      units = "in",res=300)
  plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors), c("dynamic tree cut", "merged dynamic"),
                      dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05)
  dev.off()
  
  plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors), c("dynamic tree cut", "merged dynamic"),
                      dendroLabels = FALSE, hang = 0.03, addGuide = TRUE, guideHang = 0.05)
  
  tpm.wgcna.list[[i]] &lt;- eigengene_invert_id(as.data.frame(t(tpm.wgcna.list[[i]])), mergedColors, mergedMEs)
  tpm.wgcna.list[[i]] &lt;- wgcna_heatmap_reorder(tpm.wgcna.list[[i]])
  
  write.table(tpm.wgcna.list[[i]],
              paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_tpm_de_modules.tsv"),
              row.names = T,
              col.names = T,
              quote = F,
              sep = "\t")
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2015
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2015" class="anchor" href="#wdi-luck-et-al-2015-12" id="user-content-wdi-luck-et-al-2015-12">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_wgcna_eigengene_dendrogram_plot.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_wgcna_eigengene_dendrogram_plot.png" style="max-width: 100%;"/>
                     </a>
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_wgcna_merge_eigengenes_plot.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_wgcna_merge_eigengenes_plot.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Gutzwiller et al 2015
                     </h5>
                     <a aria-label="Permalink: wMel, Gutzwiller et al 2015" class="anchor" href="#wmel-gutzwiller-et-al-2015-12" id="user-content-wmel-gutzwiller-et-al-2015-12">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_wgcna_eigengene_dendrogram_plot.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_wgcna_eigengene_dendrogram_plot.png" style="max-width: 100%;"/>
                     </a>
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_wgcna_merge_eigengenes_plot.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_wgcna_merge_eigengenes_plot.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Grote et al 2017
                     </h5>
                     <a aria-label="Permalink: wBm, Grote et al 2017" class="anchor" href="#wbm-grote-et-al-2017-12" id="user-content-wbm-grote-et-al-2017-12">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_wgcna_eigengene_dendrogram_plot.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_wgcna_eigengene_dendrogram_plot.png" style="max-width: 100%;"/>
                     </a>
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_wgcna_merge_eigengenes_plot.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_wgcna_merge_eigengenes_plot.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Chung et al 2019
                     </h5>
                     <a aria-label="Permalink: wBm, Chung et al 2019" class="anchor" href="#wbm-chung-et-al-2019-12" id="user-content-wbm-chung-et-al-2019-12">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_wgcna_eigengene_dendrogram_plot.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_wgcna_eigengene_dendrogram_plot.png" style="max-width: 100%;"/>
                     </a>
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_wgcna_merge_eigengenes_plot.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_wgcna_merge_eigengenes_plot.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Plot WGCNA-derived expression modules
                      <a name="user-content-ranalysis_de_wgcna_heatmap">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Plot WGCNA-derived expression modules " class="anchor" href="#plot-wgcna-derived-expression-modules-" id="user-content-plot-wgcna-derived-expression-modules-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='for(i in 1:length(tpm.wgcna.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.wgcna.list)[i] &amp; sample_map$sample_identifier %in% colnames(tpm.wgcna.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)

  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  log2tpm.de &lt;- log2(tpm.wgcna.list[[i]][,1:(ncol(tpm.wgcna.list[[i]]) - 2)] + 1)
  zscore.log2tpm.de &lt;- as.data.frame(t(scale(t(log2tpm.de))))
  
  hmcol1 &lt;- colorRampPalette(c("navyblue","white","firebrick3"))(12)
  hmcol2 &lt;- colorRampPalette(c("#FFFFFF","#FDE0D2","#FABAA1","#F69274", "#F16B4E","#EF3D2D","#590A16","#A51E23","#650C16"))(30)
  rowcol1 &lt;- tpm.wgcna.list[[i]]$module
  colcol &lt;- as.character(groups[,3])
  rowsep &lt;- get_heatmap_separators(rowcol1)
  colsep &lt;- get_heatmap_separators(colcol)
  
  rowcol2 &lt;- tpm.wgcna.list[[i]]$invert
  rowcol2[rowcol2 == F] &lt;- "grey"
  rowcol2[rowcol2 == T] &lt;- "black"
  
  coldendro.df &lt;- as.data.frame(tpm.kept.list[names(tpm.kept.list) == names(tpm.wgcna.list)[i]])
  coldendro.df &lt;- t(scale(t(log2(coldendro.df + 1))))
  coldendro &lt;- pvclust(coldendro.df, method.dist="cor", method.hclust="average", nboot=100, quiet=T)

  if(names(tpm.wgcna.list)[i] == "wMel_gutzwiller_2015"){
    coldendro &lt;- rotate(as.dendrogram(coldendro), c(30:50,55:54,52:53,51,11:29,1:10))
  }

  pdf(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_zscorelog2tpm_heatmap.pdf"),
      width = 5,
      height = 10)
  heatmap.2(as.matrix(zscore.log2tpm.de),
                col=hmcol1,
                trace="none",
                labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
                Rowv = F,
                Colv = as.dendrogram(coldendro),
                RowSideColors=rowcol1,
                ColSideColors=colcol,
                lhei = c(2,8),
                breaks = seq(-3,3,by=.5),
                rowsep = rowsep,
                colsep = colsep,
                dendrogram = "col")
  heatmap.2(as.matrix(zscore.log2tpm.de),
                col=hmcol1,
                trace="none",
                labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
                Rowv = F,
                Colv = as.dendrogram(coldendro),
                RowSideColors=rowcol2,
                ColSideColors=colcol,
                lhei = c(2,8),
                breaks = seq(-3,3,by=.5),
                rowsep = rowsep,
                colsep = colsep,
                dendrogram = "col")
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_zscorelog2tpm_heatmap.png"),
      width = 5,
      height = 10,
      units = "in",res=300)
  heatmap.2(as.matrix(zscore.log2tpm.de),
                col=hmcol1,
                trace="none",
                labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
                Rowv = F,
                Colv = as.dendrogram(coldendro),
                RowSideColors=rowcol1,
                ColSideColors=colcol,
                lhei = c(2,8),
                breaks = seq(-3,3,by=.5),
                rowsep = rowsep,
                colsep = colsep,
                dendrogram = "col")
  heatmap.2(as.matrix(zscore.log2tpm.de),
                col=hmcol1,
                trace="none",
                labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
                Rowv = F,
                Colv = as.dendrogram(coldendro),
                RowSideColors=rowcol2,
                ColSideColors=colcol,
                lhei = c(2,8),
                breaks = seq(-3,3,by=.5),
                rowsep = rowsep,
                colsep = colsep,
                dendrogram = "col")
  dev.off()
  
  heatmap.2(as.matrix(zscore.log2tpm.de),
                col=hmcol1,
                trace="none",
                labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
                Rowv = F,
                Colv = as.dendrogram(coldendro),
                RowSideColors=rowcol1,
                ColSideColors=colcol,
                lhei = c(2,8),
                breaks = seq(-3,3,by=.5),
                rowsep = rowsep,
                colsep = colsep,
                dendrogram = "col")
  heatmap.2(as.matrix(zscore.log2tpm.de),
                col=hmcol1,
                trace="none",
                labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
                Rowv = F,
                Colv = as.dendrogram(coldendro),
                RowSideColors=rowcol2,
                ColSideColors=colcol,
                lhei = c(2,8),
                breaks = seq(-3,3,by=.5),
                rowsep = rowsep,
                colsep = colsep,
                dendrogram = "col")
}'>
                     <pre class="notranslate" lang="{R,"><code>for(i in 1:length(tpm.wgcna.list)){
  sample_map.subset &lt;- sample_map[sample_map$study == names(tpm.wgcna.list)[i] &amp; sample_map$sample_identifier %in% colnames(tpm.wgcna.list[[i]]),]
  groups &lt;- as.data.frame(cbind(as.character(sample_map.subset$sample_identifier),
                                as.character(sample_map.subset$sample_group),
                                as.character(sample_map.subset$color)))
  groups &lt;- unique(groups)

  groups[,1] &lt;- factor(groups[,1],levels=groups[,1])
  groups[,2] &lt;- factor(groups[,2],levels=unique(groups[,2]))
  groups[,3] &lt;- factor(groups[,3],levels=unique(groups[,3]))
  
  log2tpm.de &lt;- log2(tpm.wgcna.list[[i]][,1:(ncol(tpm.wgcna.list[[i]]) - 2)] + 1)
  zscore.log2tpm.de &lt;- as.data.frame(t(scale(t(log2tpm.de))))
  
  hmcol1 &lt;- colorRampPalette(c("navyblue","white","firebrick3"))(12)
  hmcol2 &lt;- colorRampPalette(c("#FFFFFF","#FDE0D2","#FABAA1","#F69274", "#F16B4E","#EF3D2D","#590A16","#A51E23","#650C16"))(30)
  rowcol1 &lt;- tpm.wgcna.list[[i]]$module
  colcol &lt;- as.character(groups[,3])
  rowsep &lt;- get_heatmap_separators(rowcol1)
  colsep &lt;- get_heatmap_separators(colcol)
  
  rowcol2 &lt;- tpm.wgcna.list[[i]]$invert
  rowcol2[rowcol2 == F] &lt;- "grey"
  rowcol2[rowcol2 == T] &lt;- "black"
  
  coldendro.df &lt;- as.data.frame(tpm.kept.list[names(tpm.kept.list) == names(tpm.wgcna.list)[i]])
  coldendro.df &lt;- t(scale(t(log2(coldendro.df + 1))))
  coldendro &lt;- pvclust(coldendro.df, method.dist="cor", method.hclust="average", nboot=100, quiet=T)

  if(names(tpm.wgcna.list)[i] == "wMel_gutzwiller_2015"){
    coldendro &lt;- rotate(as.dendrogram(coldendro), c(30:50,55:54,52:53,51,11:29,1:10))
  }

  pdf(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_zscorelog2tpm_heatmap.pdf"),
      width = 5,
      height = 10)
  heatmap.2(as.matrix(zscore.log2tpm.de),
                col=hmcol1,
                trace="none",
                labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
                Rowv = F,
                Colv = as.dendrogram(coldendro),
                RowSideColors=rowcol1,
                ColSideColors=colcol,
                lhei = c(2,8),
                breaks = seq(-3,3,by=.5),
                rowsep = rowsep,
                colsep = colsep,
                dendrogram = "col")
  heatmap.2(as.matrix(zscore.log2tpm.de),
                col=hmcol1,
                trace="none",
                labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
                Rowv = F,
                Colv = as.dendrogram(coldendro),
                RowSideColors=rowcol2,
                ColSideColors=colcol,
                lhei = c(2,8),
                breaks = seq(-3,3,by=.5),
                rowsep = rowsep,
                colsep = colsep,
                dendrogram = "col")
  dev.off()
  
  png(paste0(OUTPUT.DIR,"/",names(tpm.wgcna.list)[i],"_wgcna_zscorelog2tpm_heatmap.png"),
      width = 5,
      height = 10,
      units = "in",res=300)
  heatmap.2(as.matrix(zscore.log2tpm.de),
                col=hmcol1,
                trace="none",
                labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
                Rowv = F,
                Colv = as.dendrogram(coldendro),
                RowSideColors=rowcol1,
                ColSideColors=colcol,
                lhei = c(2,8),
                breaks = seq(-3,3,by=.5),
                rowsep = rowsep,
                colsep = colsep,
                dendrogram = "col")
  heatmap.2(as.matrix(zscore.log2tpm.de),
                col=hmcol1,
                trace="none",
                labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
                Rowv = F,
                Colv = as.dendrogram(coldendro),
                RowSideColors=rowcol2,
                ColSideColors=colcol,
                lhei = c(2,8),
                breaks = seq(-3,3,by=.5),
                rowsep = rowsep,
                colsep = colsep,
                dendrogram = "col")
  dev.off()
  
  heatmap.2(as.matrix(zscore.log2tpm.de),
                col=hmcol1,
                trace="none",
                labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
                Rowv = F,
                Colv = as.dendrogram(coldendro),
                RowSideColors=rowcol1,
                ColSideColors=colcol,
                lhei = c(2,8),
                breaks = seq(-3,3,by=.5),
                rowsep = rowsep,
                colsep = colsep,
                dendrogram = "col")
  heatmap.2(as.matrix(zscore.log2tpm.de),
                col=hmcol1,
                trace="none",
                labRow=vector(mode = "character", length = nrow(zscore.log2tpm.de)),
                Rowv = F,
                Colv = as.dendrogram(coldendro),
                RowSideColors=rowcol2,
                ColSideColors=colcol,
                lhei = c(2,8),
                breaks = seq(-3,3,by=.5),
                rowsep = rowsep,
                colsep = colsep,
                dendrogram = "col")
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wDi, Luck et al 2015
                     </h5>
                     <a aria-label="Permalink: wDi, Luck et al 2015" class="anchor" href="#wdi-luck-et-al-2015-13" id="user-content-wdi-luck-et-al-2015-13">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_wgcna_zscorelog2tpm_heatmap_bar1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_wgcna_zscorelog2tpm_heatmap_bar1.png" style="max-width: 100%;"/>
                     </a>
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wDi_luck_2015_wgcna_zscorelog2tpm_heatmap_bar2.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wDi_luck_2015_wgcna_zscorelog2tpm_heatmap_bar2.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wMel, Gutzwiller et al 2015
                     </h5>
                     <a aria-label="Permalink: wMel, Gutzwiller et al 2015" class="anchor" href="#wmel-gutzwiller-et-al-2015-13" id="user-content-wmel-gutzwiller-et-al-2015-13">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_wgcna_zscorelog2tpm_heatmap_bar1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_wgcna_zscorelog2tpm_heatmap_bar1.png" style="max-width: 100%;"/>
                     </a>
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wMel_gutzwiller_2015_wgcna_zscorelog2tpm_heatmap_bar2.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wMel_gutzwiller_2015_wgcna_zscorelog2tpm_heatmap_bar2.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Grote et al 2017
                     </h5>
                     <a aria-label="Permalink: wBm, Grote et al 2017" class="anchor" href="#wbm-grote-et-al-2017-13" id="user-content-wbm-grote-et-al-2017-13">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_wgcna_zscorelog2tpm_heatmap_bar1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_wgcna_zscorelog2tpm_heatmap_bar1.png" style="max-width: 100%;"/>
                     </a>
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_grote_2017_wgcna_zscorelog2tpm_heatmap_bar2.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_grote_2017_wgcna_zscorelog2tpm_heatmap_bar2.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h5 class="heading-element" dir="auto" tabindex="-1">
                      wBm, Chung et al 2019
                     </h5>
                     <a aria-label="Permalink: wBm, Chung et al 2019" class="anchor" href="#wbm-chung-et-al-2019-13" id="user-content-wbm-chung-et-al-2019-13">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_wgcna_zscorelog2tpm_heatmap_bar1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_wgcna_zscorelog2tpm_heatmap_bar1.png" style="max-width: 100%;"/>
                     </a>
                     <a href="/matt-chung/wolbachia_transcriptomics_metaanalysis/blob/master/images/wBm_chung_2019_wgcna_zscorelog2tpm_heatmap_bar2.png" rel="noopener noreferrer" target="_blank">
                      <img alt="Image description" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/raw/master/images/wBm_chung_2019_wgcna_zscorelog2tpm_heatmap_bar2.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Conduct functional term enrichment analysis on WGCNA expression modules
                      <a name="user-content-ranalysis_de_wgcna_fxnterm">
                      </a>
                     </h4>
                     <a aria-label="Permalink: Conduct functional term enrichment analysis on WGCNA expression modules " class="anchor" href="#conduct-functional-term-enrichment-analysis-on-wgcna-expression-modules-" id="user-content-conduct-functional-term-enrichment-analysis-on-wgcna-expression-modules-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='terms.wgcna.list &lt;- tpm.wgcna.list
terms.sig.wgcna.list &lt;- tpm.wgcna.list

for(i in 1:length(tpm.wgcna.list)){
  strain &lt;- substr(names(tpm.wgcna.list)[i],1,regexpr("_",names(tpm.wgcna.list)[i])-1)
  geneinfo &lt;- read.delim(paste0(GENEINFO.DIR,"/",strain,".cds.fna.interproscan.geneinfo.tsv"))
  
  terms.colnames &lt;- c("term","clusteroccurences","genomeoccurences","pvalue","correctedpvalue","oddsratio","module","invert")
  terms.wgcna &lt;- as.data.frame(matrix(nrow = 0,
                                      ncol = 8))
  colnames(terms.wgcna) &lt;- terms.colnames
  for(j in 1:length(unique(tpm.wgcna.list[[i]]$module))){
    terms.wgcna.f &lt;- as.data.frame(cbind(functionaltermenrichment(rownames(tpm.wgcna.list[[i]])[tpm.wgcna.list[[i]]$module == unique(tpm.wgcna.list[[i]]$module)[j] &amp; tpm.wgcna.list[[i]]$invert == F],geneinfo),unique(tpm.wgcna.list[[i]]$module)[j],F))
    terms.wgcna.t &lt;- as.data.frame(cbind(functionaltermenrichment(rownames(tpm.wgcna.list[[i]])[tpm.wgcna.list[[i]]$module == unique(tpm.wgcna.list[[i]]$module)[j] &amp; tpm.wgcna.list[[i]]$invert == T],geneinfo),unique(tpm.wgcna.list[[i]]$module)[j],T))
    
    colnames(terms.wgcna.f) &lt;- terms.colnames
    colnames(terms.wgcna.t) &lt;- terms.colnames
    terms.wgcna &lt;- as.data.frame(rbind(terms.wgcna,
                                       terms.wgcna.f, 
                                       terms.wgcna.t))
  }
  
  for(j in 2:6){
    terms.wgcna[,j] &lt;- as.numeric(as.character(terms.wgcna[,j]))
  }
  
  terms.wgcna.list[[i]] &lt;- terms.wgcna
  terms.sig.wgcna.list[[i]] &lt;- terms.wgcna[terms.wgcna$correctedpvalue &lt; 0.05,]
  
  write.table(terms.wgcna.list[[i]],
            paste0(OUTPUT.DIR,"/",names(terms.wgcna.list)[i],"_functionaltermenrichment.tsv"),
            row.names = T,
            col.names = T,
            quote = F,
            sep = "\t")
}'>
                     <pre class="notranslate" lang="{R}"><code>terms.wgcna.list &lt;- tpm.wgcna.list
terms.sig.wgcna.list &lt;- tpm.wgcna.list

for(i in 1:length(tpm.wgcna.list)){
  strain &lt;- substr(names(tpm.wgcna.list)[i],1,regexpr("_",names(tpm.wgcna.list)[i])-1)
  geneinfo &lt;- read.delim(paste0(GENEINFO.DIR,"/",strain,".cds.fna.interproscan.geneinfo.tsv"))
  
  terms.colnames &lt;- c("term","clusteroccurences","genomeoccurences","pvalue","correctedpvalue","oddsratio","module","invert")
  terms.wgcna &lt;- as.data.frame(matrix(nrow = 0,
                                      ncol = 8))
  colnames(terms.wgcna) &lt;- terms.colnames
  for(j in 1:length(unique(tpm.wgcna.list[[i]]$module))){
    terms.wgcna.f &lt;- as.data.frame(cbind(functionaltermenrichment(rownames(tpm.wgcna.list[[i]])[tpm.wgcna.list[[i]]$module == unique(tpm.wgcna.list[[i]]$module)[j] &amp; tpm.wgcna.list[[i]]$invert == F],geneinfo),unique(tpm.wgcna.list[[i]]$module)[j],F))
    terms.wgcna.t &lt;- as.data.frame(cbind(functionaltermenrichment(rownames(tpm.wgcna.list[[i]])[tpm.wgcna.list[[i]]$module == unique(tpm.wgcna.list[[i]]$module)[j] &amp; tpm.wgcna.list[[i]]$invert == T],geneinfo),unique(tpm.wgcna.list[[i]]$module)[j],T))
    
    colnames(terms.wgcna.f) &lt;- terms.colnames
    colnames(terms.wgcna.t) &lt;- terms.colnames
    terms.wgcna &lt;- as.data.frame(rbind(terms.wgcna,
                                       terms.wgcna.f, 
                                       terms.wgcna.t))
  }
  
  for(j in 2:6){
    terms.wgcna[,j] &lt;- as.numeric(as.character(terms.wgcna[,j]))
  }
  
  terms.wgcna.list[[i]] &lt;- terms.wgcna
  terms.sig.wgcna.list[[i]] &lt;- terms.wgcna[terms.wgcna$correctedpvalue &lt; 0.05,]
  
  write.table(terms.wgcna.list[[i]],
            paste0(OUTPUT.DIR,"/",names(terms.wgcna.list)[i],"_functionaltermenrichment.tsv"),
            row.names = T,
            col.names = T,
            quote = F,
            sep = "\t")
}
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Download reference files (2019-12-13)
                      <a name="user-content-coreanalysis_getreferences">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Download reference files (2019-12-13) " class="anchor" href="#download-reference-files-2019-12-13-" id="user-content-download-reference-files-2019-12-13-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The faa reference for wDi must be downloaded from this link:
                     <a href="http://rast.nmpdr.org/rast.cgi?page=JobDetails&amp;job=53659" rel="nofollow">
                      http://rast.nmpdr.org/rast.cgi?page=JobDetails&amp;job=53659
                     </a>
                     (login:guest, password:guest). All subsequent code will make use of this wDi.faa at this path: "$REFERENCES_DIR"/wDi.faa
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-17" id="user-content-commands-17">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='## wBm
wget -O "$REFERENCES_DIR"/wBm.faa.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_translated_cds.faa.gz
gunzip "$REFERENCES_DIR"/wBm.faa.gz

## wMel
wget -O "$REFERENCES_DIR"/wMel.faa.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/025/GCF_000008025.1_ASM802v1/GCF_000008025.1_ASM802v1_translated_cds.faa.gz
gunzip "$REFERENCES_DIR"/wMel.faa.gz

## wOo
wget -O "$REFERENCES_DIR"/wOo.faa.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/306/885/GCF_000306885.1_ASM30688v1/GCF_000306885.1_ASM30688v1_translated_cds.faa.gz
gunzip "$REFERENCES_DIR"/wOo.faa.gz
'>
                     <pre class="notranslate" lang="{bash,"><code>## wBm
wget -O "$REFERENCES_DIR"/wBm.faa.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/385/GCF_000008385.1_ASM838v1/GCF_000008385.1_ASM838v1_translated_cds.faa.gz
gunzip "$REFERENCES_DIR"/wBm.faa.gz

## wMel
wget -O "$REFERENCES_DIR"/wMel.faa.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/008/025/GCF_000008025.1_ASM802v1/GCF_000008025.1_ASM802v1_translated_cds.faa.gz
gunzip "$REFERENCES_DIR"/wMel.faa.gz

## wOo
wget -O "$REFERENCES_DIR"/wOo.faa.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/306/885/GCF_000306885.1_ASM30688v1/GCF_000306885.1_ASM30688v1_translated_cds.faa.gz
gunzip "$REFERENCES_DIR"/wOo.faa.gz

</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Create BLAST database with the amino acid sequences of the four Wolbachia genomes (2019-12-13)
                      <a name="user-content-coreanalysis_createblastdb">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Create BLAST database with the amino acid sequences of the four Wolbachia genomes (2019-12-13) " class="anchor" href="#create-blast-database-with-the-amino-acid-sequences-of-the-four-wolbachia-genomes-2019-12-13-" id="user-content-create-blast-database-with-the-amino-acid-sequences-of-the-four-wolbachia-genomes-2019-12-13-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Inputs:
                     </h4>
                     <a aria-label="Permalink: Inputs:" class="anchor" href="#inputs" id="user-content-inputs">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='DB_FAA="$REFERENCES_DIR"/combined_wolbachia_reference.faa'>
                     <pre class="notranslate" lang="{bash,"><code>DB_FAA="$REFERENCES_DIR"/combined_wolbachia_reference.faa
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-18" id="user-content-commands-18">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='cat "$REFERENCES_DIR"/wBm.faa "$REFERENCES_DIR"/wDi.faa "$REFERENCES_DIR"/wMel.faa "$REFERENCES_DIR"/wOo.faa &gt; "$DB_FAA"
$NCBIBLAST_BIN_DIR/formatdb -t "$(echo "$DB_FAA" | sed "s/[.]faa$//g")" -i "$DB_FAA"'>
                     <pre class="notranslate" lang="{bash,"><code>cat "$REFERENCES_DIR"/wBm.faa "$REFERENCES_DIR"/wDi.faa "$REFERENCES_DIR"/wMel.faa "$REFERENCES_DIR"/wOo.faa &gt; "$DB_FAA"
$NCBIBLAST_BIN_DIR/formatdb -t "$(echo "$DB_FAA" | sed "s/[.]faa$//g")" -i "$DB_FAA"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Run self-BLASTP search against the protein sequences of the four Wolbachia strains (2019-12-13)
                      <a name="user-content-coreanalysis_createblastdb">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Run self-BLASTP search against the protein sequences of the four Wolbachia strains (2019-12-13) " class="anchor" href="#run-self-blastp-search-against-the-protein-sequences-of-the-four-wolbachia-strains-2019-12-13-" id="user-content-run-self-blastp-search-against-the-protein-sequences-of-the-four-wolbachia-strains-2019-12-13-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Inputs:
                     </h4>
                     <a aria-label="Permalink: Inputs:" class="anchor" href="#inputs-1" id="user-content-inputs-1">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='BLAST_DB="$REFERENCES_DIR"/combined_wolbachia_reference.faa'>
                     <pre class="notranslate" lang="{bash,"><code>BLAST_DB="$REFERENCES_DIR"/combined_wolbachia_reference.faa
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Commands:
                     </h4>
                     <a aria-label="Permalink: Commands:" class="anchor" href="#commands-19" id="user-content-commands-19">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='qsub -P jdhotopp-lab -l mem_free=5G -N blastp -wd "$(dirname "$BLAST_DB")" -b y $NCBIBLAST_BIN_DIR/blastall -p blastp -i "$BLAST_DB" -d "$BLAST_DB" -e 1e-5 -m 8 -o "$(dirname "$BLAST_DB")"/blastp_raw.out'>
                     <pre class="notranslate" lang="{bash,"><code>qsub -P jdhotopp-lab -l mem_free=5G -N blastp -wd "$(dirname "$BLAST_DB")" -b y $NCBIBLAST_BIN_DIR/blastall -p blastp -i "$BLAST_DB" -d "$BLAST_DB" -e 1e-5 -m 8 -o "$(dirname "$BLAST_DB")"/blastp_raw.out
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Run PanOCT to identify core Wolbachia genes (2019-12-13)
                      <a name="user-content-coreanalysis_createblastdb">
                      </a>
                     </h2>
                     <a aria-label="Permalink: Run PanOCT to identify core Wolbachia genes (2019-12-13) " class="anchor" href="#run-panoct-to-identify-core-wolbachia-genes-2019-12-13-" id="user-content-run-panoct-to-identify-core-wolbachia-genes-2019-12-13-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Inputs:
                     </h4>
                     <a aria-label="Permalink: Inputs:" class="anchor" href="#inputs-2" id="user-content-inputs-2">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='FAA="$REFERENCES_DIR"/combined_wolbachia_reference.faa
BLAST_OUTPUT="$(dirname "$BLAST_DB")"/blastp_raw.out
OUTPUT_DIR="$WORKING_DIR"/panoct'>
                     <pre class="notranslate" lang="{bash,"><code>FAA="$REFERENCES_DIR"/combined_wolbachia_reference.faa
BLAST_OUTPUT="$(dirname "$BLAST_DB")"/blastp_raw.out
OUTPUT_DIR="$WORKING_DIR"/panoct
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h4 class="heading-element" dir="auto" tabindex="-1">
                      Inputs:
                     </h4>
                     <a aria-label="Permalink: Inputs:" class="anchor" href="#inputs-3" id="user-content-inputs-3">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='mkdir -p "$OUTPUT_DIR"

'>
                     <pre class="notranslate" lang="{bash,"><code>mkdir -p "$OUTPUT_DIR"


</code></pre>
                    </div>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" id="react-codespace-csrf" type="hidden" value="vSkMwoEYAFdB05-5rEYHn2hwtRdW_tg8BC5kMBc23MWq8ZNtHRN4N0KfZcD6EppRHsqY9LZXo6Z1_UCbEpVC5g">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <div class="f4 my-3 color-fg-muted text-italic">
                 No description, website, or topics provided.
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <h3 class="sr-only">
                 License
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:license"}' href="#MIT-1-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-law mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8.75.75V2h.985c.304 0 .603.08.867.231l1.29.736c.038.022.08.033.124.033h2.234a.75.75 0 0 1 0 1.5h-.427l2.111 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.006.005-.01.01-.045.04c-.21.176-.441.327-.686.45C14.556 10.78 13.88 11 13 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L12.178 4.5h-.162c-.305 0-.604-.079-.868-.231l-1.29-.736a.245.245 0 0 0-.124-.033H8.75V13h2.5a.75.75 0 0 1 0 1.5h-6.5a.75.75 0 0 1 0-1.5h2.5V3.5h-.984a.245.245 0 0 0-.124.033l-1.289.737c-.265.15-.564.23-.869.23h-.162l2.112 4.692a.75.75 0 0 1-.154.838l-.53-.53.529.531-.001.002-.002.002-.006.006-.016.015-.045.04c-.21.176-.441.327-.686.45C4.556 10.78 3.88 11 3 11a4.498 4.498 0 0 1-2.023-.454 3.544 3.544 0 0 1-.686-.45l-.045-.04-.016-.015-.006-.006-.004-.004v-.001a.75.75 0 0 1-.154-.838L2.178 4.5H1.75a.75.75 0 0 1 0-1.5h2.234a.249.249 0 0 0 .125-.033l1.288-.737c.265-.15.564-.23.869-.23h.984V.75a.75.75 0 0 1 1.5 0Zm2.945 8.477c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L13 6.327Zm-10 0c.285.135.718.273 1.305.273s1.02-.138 1.305-.273L3 6.327Z">
                   </path>
                  </svg>
                  MIT license
                 </a>
                </div>
                <include-fragment src="/matt-chung/wolbachia_transcriptomics_metaanalysis/hovercards/citation/sidebar_partial?tree_name=master">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  star
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fmatt-chung%2Fwolbachia_transcriptomics_metaanalysis&amp;report=matt-chung+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/matt-chung/wolbachia_transcriptomics_metaanalysis/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/matt-chung/packages?repo_name=wolbachia_transcriptomics_metaanalysis">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/matt-chung/wolbachia_transcriptomics_metaanalysis/used_by_list">
               </include-fragment>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <style>
    .user-mention[href$="/ninenia"] {
        color: var(--color-user-mention-fg);
        background-color: var(--color-user-mention-bg);
        border-radius: 2px;
        margin-left: -2px;
        margin-right: -2px;
        padding: 0 2px;
      }
   </style>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
