<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light_v2-e830ce163aae.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark_v2-bc3c552285ea.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed_v2-191796a86a54.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast_v2-e0a0b6e42616.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind_v2-4fd40c9324e8.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind_v2-2f299c61eceb.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast_v2-05d723322773.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia_v2-5679d6101565.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia_v2-0ff66d76cc55.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","repository_suggester_elastic_search","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-4c826ff1ab15.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-ba9bd5e82e26.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-4eced6de5532.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         QuentinRougemont/coho_ldscp_genomics: scripts to reproduce our landscape genomics results in coho salmon
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="8F80:1D249F:15C25E7:1F32F79:6605E1F9" data-turbo-transient="true" name="request-id">
         <meta content="22f598a296c639e8b750dc2cf1a20323836185b0d8b152a2e56d5461ad76529a" data-turbo-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6bnVsbCwicmVxdWVzdF9pZCI6IjhGODA6MUQyNDlGOjE1QzI1RTc6MUYzMkY3OTo2NjA1RTFGOSIsInZpc2l0b3JfaWQiOiI1NTYzNjg1OTkxMDg0ODI5Mjg2IiwicmVnaW9uX2VkZ2UiOiJpYWQiLCJyZWdpb25fcmVuZGVyIjoiaWFkIn0=" data-turbo-transient="true" name="visitor-payload">
           <meta content="956b80f3194a4d23acb169bfc1492b821cd618298261d177f069254d52ca10c8" data-turbo-transient="true" name="visitor-hmac">
            <meta content="repository:379816251" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="126681664" name="octolytics-actor-id">
               <meta content="ninenia" name="octolytics-actor-login">
                <meta content="57d6abdd600494be8ac2f82b25b957e4c0d029a4d648c10497790171de0492de" name="octolytics-actor-hash">
                 <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
                  <meta content="ninenia" name="user-login"/>
                  <link href="/sessions/sudo_modal" rel="sudo-modal"/>
                  <meta content="width=device-width" name="viewport"/>
                  <meta content="scripts to reproduce our landscape genomics results in coho salmon - QuentinRougemont/coho_ldscp_genomics" name="description"/>
                  <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
                  <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
                  <meta content="1401488693436528" property="fb:app_id"/>
                  <meta content="app-id=1477376905, app-argument=https://github.com/QuentinRougemont/coho_ldscp_genomics" name="apple-itunes-app">
                   <meta content="https://opengraph.githubassets.com/679607d3ab4d2f615b64d08d27d2f89b292152064538bcc26a58b29627191ff8/QuentinRougemont/coho_ldscp_genomics" name="twitter:image:src">
                    <meta content="@github" name="twitter:site">
                     <meta content="summary_large_image" name="twitter:card">
                      <meta content="QuentinRougemont/coho_ldscp_genomics: scripts to reproduce our landscape genomics results in coho salmon" name="twitter:title">
                       <meta content="scripts to reproduce our landscape genomics results in coho salmon - QuentinRougemont/coho_ldscp_genomics" name="twitter:description">
                        <meta content="https://opengraph.githubassets.com/679607d3ab4d2f615b64d08d27d2f89b292152064538bcc26a58b29627191ff8/QuentinRougemont/coho_ldscp_genomics" property="og:image"/>
                        <meta content="scripts to reproduce our landscape genomics results in coho salmon - QuentinRougemont/coho_ldscp_genomics" property="og:image:alt"/>
                        <meta content="1200" property="og:image:width"/>
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="QuentinRougemont/coho_ldscp_genomics: scripts to reproduce our landscape genomics results in coho salmon" property="og:title"/>
                        <meta content="https://github.com/QuentinRougemont/coho_ldscp_genomics" property="og:url"/>
                        <meta content="scripts to reproduce our landscape genomics results in coho salmon - QuentinRougemont/coho_ldscp_genomics" property="og:description"/>
                        <link data-refresh-url="/_alive" data-session-id="fae53153a7a7c8d08a43b9810520b9ce4325ce8b4e99a06c5fc3f52c85cc22fe" href="wss://alive.github.com/_sockets/u/126681664/ws?session=eyJ2IjoiVjMiLCJ1IjoxMjY2ODE2NjQsInMiOjEzMDQyODIxMTYsImMiOjE3ODkzNDE0NiwidCI6MTcxMTY2MTU2NX0=--bce3c49b6fc61b50366ba0891d71888520cb243d4a630da7f6c30e99b5259362" rel="shared-web-socket"/>
                        <link href="/assets-cdn/worker/socket-worker-9cc1149b224c.js" rel="shared-web-socket-src"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="all" name="keyboard-shortcuts-preference"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="ce7ac010b2a2d244f4e22532a2864ada4189093671a7a5533bb73137af2ff3fe" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="af60a253d00dc707d466f3bb6e8cc7535995d8a78f2d1be3507d205c4ae531a3" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/QuentinRougemont/coho_ldscp_genomics git https://github.com/QuentinRougemont/coho_ldscp_genomics.git" name="go-import"/>
                        <meta content="17588888" name="octolytics-dimension-user_id">
                         <meta content="QuentinRougemont" name="octolytics-dimension-user_login">
                          <meta content="379816251" name="octolytics-dimension-repository_id">
                           <meta content="QuentinRougemont/coho_ldscp_genomics" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="379816251" name="octolytics-dimension-repository_network_root_id">
                               <meta content="QuentinRougemont/coho_ldscp_genomics" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/QuentinRougemont/coho_ldscp_genomics" rel="canonical"/>
                                <meta content="logged-in env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <meta content="/windows-tile.png" name="msapplication-TileImage"/>
                                 <meta content="#ffffff" name="msapplication-TileColor"/>
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-in env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-in env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="p-3 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-1c70f65f7356.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_allex_crc32_lib_crc32_esm_js-node_modules_github_mini-throttle_dist_deco-b38cad-748e74df23ce.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_clipboard-copy-element_dist_index_esm_js-node_modules_delegated-e-b37f7d-2f24d321a3fb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_command-palette_items_help-item_ts-app_assets_modules_github_comman-48ad9d-a9ed17f2ebff.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/command-palette-f620e569db0b.js" type="application/javascript">
    </script>
    <header class="AppHeader">
     <div class="AppHeader-globalBar pb-2 js-global-bar">
      <div class="AppHeader-globalBar-start">
       <deferred-side-panel data-url="/_side-panels/global">
        <include-fragment data-target="deferred-side-panel.fragment">
         <button aria-label="Open global navigation menu" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-bg-transparent p-0 color-fg-muted" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-b5474fcc-e1b8-4887-86eb-637efa9f4065" data-view-component="true" id="dialog-show-dialog-b5474fcc-e1b8-4887-86eb-637efa9f4065" type="button">
          <svg aria-hidden="true" class="octicon octicon-three-bars Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1 2.75A.75.75 0 0 1 1.75 2h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 2.75Zm0 5A.75.75 0 0 1 1.75 7h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 7.75ZM1.75 12h12.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Z">
           </path>
          </svg>
         </button>
         <dialog-helper>
          <dialog aria-describedby="dialog-b5474fcc-e1b8-4887-86eb-637efa9f4065-description" aria-labelledby="dialog-b5474fcc-e1b8-4887-86eb-637efa9f4065-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-left SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-b5474fcc-e1b8-4887-86eb-637efa9f4065">
           <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title sr-only" id="dialog-b5474fcc-e1b8-4887-86eb-637efa9f4065-title">
               Global navigation
              </h1>
              <div class="d-flex" data-view-component="true">
               <div class="AppHeader-logo position-relative" data-view-component="true">
                <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
                 <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
                 </path>
                </svg>
               </div>
              </div>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-b5474fcc-e1b8-4887-86eb-637efa9f4065" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="dialog-b5474fcc-e1b8-4887-86eb-637efa9f4065-title">
            <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
             <div class="d-flex flex-column mb-3" data-view-component="true">
              <nav aria-label="Site navigation" class="ActionList" data-view-component="true">
               <nav-list>
                <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"HOME","label":null}' data-hotkey="g d" data-view-component="true" href="/dashboard" id="item-205d353a-7f09-4116-83b7-98dd05cf8513">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-home" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M6.906.664a1.749 1.749 0 0 1 2.187 0l5.25 4.2c.415.332.657.835.657 1.367v7.019A1.75 1.75 0 0 1 13.25 15h-3.5a.75.75 0 0 1-.75-.75V9H7v5.25a.75.75 0 0 1-.75.75h-3.5A1.75 1.75 0 0 1 1 13.25V6.23c0-.531.242-1.034.657-1.366l5.25-4.2Zm1.25 1.171a.25.25 0 0 0-.312 0l-5.25 4.2a.25.25 0 0 0-.094.196v7.019c0 .138.112.25.25.25H5.5V8.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 .75.75v5.25h2.75a.25.25 0 0 0 .25-.25V6.23a.25.25 0 0 0-.094-.195Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Home
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"ISSUES","label":null}' data-hotkey="g i" data-view-component="true" href="/issues" id="item-7a7c7ab4-6dbd-48ea-bade-f891b1472bc3">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS","label":null}' data-hotkey="g p" data-view-component="true" href="/pulls" id="item-bb64309a-0395-48ec-90bc-5302ec814abc">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="projects" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROJECTS","label":null}' data-view-component="true" href="/projects" id="item-8fd54e5c-bdd2-4606-a277-d5fb250cb420">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"DISCUSSIONS","label":null}' data-view-component="true" href="/discussions" id="item-05fb2d8a-b5ec-4ce2-bf1d-54f569135dd3">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Discussions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"CODESPACES","label":null}' data-view-component="true" href="https://github.com/codespaces" id="item-b0b34ba1-5afb-4cc5-9b77-7da5b27cc17d">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Codespaces
                   </span>
                  </a>
                 </li>
                 <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"EXPLORE","label":null}' data-view-component="true" href="/explore" id="item-264a42a0-a18f-49b4-a2cb-88189d9cd393">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-telescope" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.184 1.143v-.001l1.422 2.464a1.75 1.75 0 0 1-.757 2.451L3.104 11.713a1.75 1.75 0 0 1-2.275-.702l-.447-.775a1.75 1.75 0 0 1 .53-2.32L11.682.573a1.748 1.748 0 0 1 2.502.57Zm-4.709 9.32h-.001l2.644 3.863a.75.75 0 1 1-1.238.848l-1.881-2.75v2.826a.75.75 0 0 1-1.5 0v-2.826l-1.881 2.75a.75.75 0 1 1-1.238-.848l2.049-2.992a.746.746 0 0 1 .293-.253l1.809-.87a.749.749 0 0 1 .944.252ZM9.436 3.92h-.001l-4.97 3.39.942 1.63 5.42-2.61Zm3.091-2.108h.001l-1.85 1.26 1.505 2.605 2.016-.97a.247.247 0 0 0 .13-.151.247.247 0 0 0-.022-.199l-1.422-2.464a.253.253 0 0 0-.161-.119.254.254 0 0 0-.197.038ZM1.756 9.157a.25.25 0 0 0-.075.33l.447.775a.25.25 0 0 0 .325.1l1.598-.769-.83-1.436-1.465 1Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Explore
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"MARKETPLACE","label":null}' data-view-component="true" href="/marketplace" id="item-7ae15a50-a648-481c-8752-55e0f0d6635c">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Marketplace
                   </span>
                  </a>
                 </li>
                </ul>
               </nav-list>
              </nav>
              <div class="my-3 d-flex flex-justify-center height-full" data-view-component="true">
               <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                </circle>
                <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                </path>
               </svg>
              </div>
             </div>
             <div class="flex-1" data-view-component="true">
             </div>
             <div class="px-2" data-view-component="true">
              <p class="color-fg-subtle text-small text-light">
               © 2024 GitHub, Inc.
              </p>
              <div class="d-flex flex-wrap text-small text-light" data-view-component="true">
               <a class="Link mr-2" data-view-component="true" href="https://github.com/about" target="_blank">
                About
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.blog" target="_blank">
                Blog
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service" target="_blank">
                Terms
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement" target="_blank">
                Privacy
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.com/security" target="_blank">
                Security
               </a>
               <a class="Link mr-3" data-view-component="true" href="https://www.githubstatus.com/" target="_blank">
                Status
               </a>
              </div>
             </div>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </include-fragment>
       </deferred-side-panel>
       <a aria-label="Homepage " class="AppHeader-logo ml-2" data-analytics-event='{"category":"Header","action":"go to dashboard","label":"icon:logo"}' data-hotkey="g d" data-turbo="false" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github v-align-middle color-fg-default" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="AppHeader-context">
        <div class="AppHeader-context-compact">
         <button aria-expanded="false" aria-haspopup="dialog" aria-label="Page context: QuentinRougemont / coho_ldscp_genomics" class="AppHeader-context-compact-trigger Truncate Button--secondary Button--medium Button box-shadow-none" data-show-dialog-id="context-region-dialog" data-view-component="true" id="dialog-show-context-region-dialog" type="button">
          <span class="Button-content">
           <span class="Button-label">
            <span class="AppHeader-context-compact-lead">
             <span class="AppHeader-context-compact-parentItem">
              QuentinRougemont
             </span>
             <span class="no-wrap">
              /
             </span>
            </span>
            <strong class="AppHeader-context-compact-mainItem d-flex flex-items-center Truncate">
             <span class="Truncate-text">
              coho_ldscp_genomics
             </span>
            </strong>
           </span>
          </span>
         </button>
         <dialog-helper>
          <dialog aria-describedby="context-region-dialog-description" aria-labelledby="context-region-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-view-component="true" id="context-region-dialog">
           <div class="Overlay-header" data-view-component="true">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title" id="context-region-dialog-title">
               Navigate back to
              </h1>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="context-region-dialog" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="context-region-dialog-title">
            <div class="Overlay-body" data-view-component="true">
             <ul class="list-style-none" role="list">
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"QuentinRougemont","screen_size":"compact"}' data-view-component="true" href="/QuentinRougemont">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-person mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                  </path>
                 </svg>
                 QuentinRougemont
                </span>
               </a>
              </li>
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"coho_ldscp_genomics","screen_size":"compact"}' data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-repo mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                  </path>
                 </svg>
                 coho_ldscp_genomics
                </span>
               </a>
              </li>
             </ul>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </div>
        <div class="AppHeader-context-full">
         <nav aria-label="Page context" role="navigation">
          <ul class="list-style-none" role="list">
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"QuentinRougemont","screen_size":"full"}' data-hovercard-type="user" data-hovercard-url="/users/QuentinRougemont/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" data-view-component="true" href="/QuentinRougemont">
             <span class="AppHeader-context-item-label">
              QuentinRougemont
             </span>
            </a>
            <span class="AppHeader-context-item-separator">
             /
            </span>
           </li>
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"coho_ldscp_genomics","screen_size":"full"}' data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics">
             <span class="AppHeader-context-item-label">
              coho_ldscp_genomics
             </span>
            </a>
           </li>
          </ul>
         </nav>
        </div>
       </div>
      </div>
      <div class="AppHeader-globalBar-end">
       <div class="AppHeader-search">
        <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;input type="hidden" value="1rZI0KYwo1KEzVA-8MrkzfdYZm-g7tOyxysPsVTvU5aGlURYNL18nTNXCBBVXQaWw9SmhJTFZiMZUHEP-Svufg" data-csrf="true" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="QuentinRougemont" data-current-repository="QuentinRougemont/coho_ldscp_genomics" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="Ss1vMAanKXy0i8xick4KP0dGOZOwko-hqaV564RA9qYCB3gaIOFgCeMnvruUS_ouFeJ7Opb1f87FRzhAY37t6g" data-header-redesign-enabled="true" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="true" data-max-custom-scopes="10" data-scope="repo:QuentinRougemont/coho_ldscp_genomics">
         <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center height-auto color-bg-transparent border-0 color-fg-subtle mx-0" data-action="click:qbsearch-input#searchInputContainerClicked">
          <button aria-expanded="false" aria-haspopup="dialog" aria-label="Search or jump to…" class="AppHeader-button AppHeader-search-whenNarrow" data-action="click:qbsearch-input#handleExpand" type="button">
           <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
            </path>
           </svg>
          </button>
          <div class="AppHeader-search-whenRegular">
           <div class="AppHeader-search-wrap AppHeader-search-wrap--hasTrailing">
            <div class="AppHeader-search-control">
             <label aria-label="Search or jump to…" class="AppHeader-search-visual--leading" for="AppHeader-searchInput">
              <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
               </path>
              </svg>
             </label>
             <button aria-describedby="search-error-message-flash" class="AppHeader-searchButton form-control input-contrast text-left color-fg-subtle no-wrap" data-action="click:qbsearch-input#handleExpand" data-analytics-event='{"category":"Global navigation","action":"SEARCH","label":null}' data-hotkey="s,/" data-target="qbsearch-input.inputButton" type="button">
              <div class="overflow-hidden">
               <span data-target="qbsearch-input.inputButtonText" id="qb-input-query">
                Type
                <kbd class="AppHeader-search-kbd">
                 /
                </kbd>
                to search
               </span>
              </div>
             </button>
            </div>
            <button class="AppHeader-search-action--trailing js-activate-command-palette" data-analytics-event='{"category":"SiteHeaderComponent","action":"command_palette","label":"open command palette"}' id="AppHeader-commandPalette-button" type="button">
             <svg aria-hidden="true" class="octicon octicon-command-palette" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m6.354 8.04-4.773 4.773a.75.75 0 1 0 1.061 1.06L7.945 8.57a.75.75 0 0 0 0-1.06L2.642 2.206a.75.75 0 0 0-1.06 1.061L6.353 8.04ZM8.75 11.5a.75.75 0 0 0 0 1.5h5.5a.75.75 0 0 0 0-1.5h-5.5Z">
              </path>
             </svg>
            </button>
            <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="AppHeader-commandPalette-button" id="tooltip-ca397bfd-fba4-44eb-b54b-ac15f8f81133" popover="manual">
             Command palette
            </tool-tip>
           </div>
          </div>
          <input class="js-site-search-type-field" name="type" type="hidden"/>
          <div class="Overlay--hidden" data-modal-dialog-overlay="">
           <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-medium Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
            <h1 class="sr-only" id="search-suggestions-dialog-header">
             Search code, repositories, users, issues, pull requests...
            </h1>
            <div class="Overlay-body Overlay-body--paddingNone">
             <div data-view-component="true">
              <div class="search-suggestions position-absolute width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                 <div class="FormControl FormControl--fullWidth">
                  <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                   Search
                  </label>
                  <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                   <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                    <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </span>
                   <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                    <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                    </div>
                    <div class="QueryBuilder-InputWrapper">
                     <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                     </div>
                     <input aria-describedby="validation-6c1f6214-052f-4cf3-8ca6-7b6737868e14" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                     </input>
                    </div>
                   </div>
                   <span class="sr-only" id="query-builder-test-clear">
                    Clear
                   </span>
                   <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                    <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <template id="search-icon">
                   <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-icon">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="file-code-icon">
                   <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                    </path>
                   </svg>
                  </template>
                  <template id="history-icon">
                   <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                    </path>
                   </svg>
                  </template>
                  <template id="repo-icon">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="bookmark-icon">
                   <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="plus-circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                    </path>
                   </svg>
                  </template>
                  <template id="trash-icon">
                   <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="team-icon">
                   <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="project-icon">
                   <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                    </path>
                   </svg>
                  </template>
                  <template id="pencil-icon">
                   <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                    </path>
                   </svg>
                  </template>
                  <template id="copilot-icon">
                   <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                    </path>
                    <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="workflow-icon">
                   <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="book-icon">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-review-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="codespaces-icon">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-discussion-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="organization-icon">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="rocket-icon">
                   <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="shield-check-icon">
                   <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="heart-icon">
                   <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="server-icon">
                   <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="globe-icon">
                   <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                    </path>
                   </svg>
                  </template>
                  <template id="issue-opened-icon">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-mobile-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="package-icon">
                   <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                    </path>
                   </svg>
                  </template>
                  <template id="credit-card-icon">
                   <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                    </path>
                    <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                    </path>
                   </svg>
                  </template>
                  <template id="play-icon">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </template>
                  <template id="gift-icon">
                   <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-square-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-desktop-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                    </path>
                   </svg>
                  </template>
                  <div class="position-relative">
                   <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                   </ul>
                  </div>
                  <div class="FormControl-inlineValidation" hidden="hidden" id="validation-6c1f6214-052f-4cf3-8ca6-7b6737868e14">
                   <span class="FormControl-inlineValidation--visual">
                    <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                     <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                     </path>
                    </svg>
                   </span>
                   <span>
                   </span>
                  </div>
                 </div>
                 <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                 </div>
                </query-builder>
               </form>
               <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                 Search syntax tips
                </a>
                <div class="d-flex flex-1">
                </div>
                <button class="Button--link Button--medium Button color-fg-accent text-normal ml-2" data-action="click:qbsearch-input#showFeedbackDialog" data-view-component="true" type="button">
                 <span class="Button-content">
                  <span class="Button-label">
                   Give feedback
                  </span>
                 </span>
                </button>
               </div>
              </div>
             </div>
            </div>
           </modal-dialog>
          </div>
         </div>
         <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
         </div>
         <div class="color-fg-default">
          <dialog-helper>
           <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
            <div class="Overlay-header" data-view-component="true">
             <div class="Overlay-headerContentWrap">
              <div class="Overlay-titleWrap">
               <h1 class="Overlay-title" id="feedback-dialog-title">
                Provide feedback
               </h1>
              </div>
              <div class="Overlay-actionWrap">
               <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
              </div>
             </div>
            </div>
            <scrollable-region data-labelled-by="feedback-dialog-title">
             <div class="Overlay-body" data-view-component="true">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
               <input name="authenticity_token" type="hidden" value="t5f895AtKOwqKpnnhXHEM4wxtuLCyo76kceY9hCx22clJeXPWPsbdpoIl-O7aMXCUTp7IZjqaK-fR41M5uDe3A"/>
               <p>
                We read every piece of feedback, and take your input very seriously.
               </p>
               <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
               <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
               <label for="include_email" style="font-weight: normal">
                Include my email address so I can be contacted
               </label>
              </form>
             </div>
            </scrollable-region>
            <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
             <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
              Cancel
             </button>
             <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
              Submit feedback
             </button>
            </div>
           </dialog>
          </dialog-helper>
          <custom-scopes data-target="qbsearch-input.customScopesManager">
           <dialog-helper>
            <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
             <div class="Overlay-header Overlay-header--divided" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                 Saved searches
                </h1>
                <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                 Use saved searches to filter your results more quickly
                </h2>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="custom-scopes-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <div data-target="custom-scopes.customScopesModalDialogFlash">
               </div>
               <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                 <input name="authenticity_token" type="hidden" value="lGWJk6LQk6f3zvaX6kqgK2Ut1klSc77Yw_0R-rtHcwnYl-sp-lHNfrynSzqgn20WrH2yXgR2_ANgpcBCpAAqXA">
                  <div data-target="custom-scopes.customScopesModalDialogFlash">
                  </div>
                  <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                  <div class="form-group">
                   <label for="custom_scope_name">
                    Name
                   </label>
                   <auto-check required="" src="/search/custom_scopes/check_name">
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                    <input data-csrf="true" type="hidden" value="Mj6cTvT6ZYAa0-J9ZmBk4nZybTSCQmBn7Mw9BiDp1B8qY9Hoo3jMg5nHq_4mMjqIwLVos6ybgP-BbJ_UGIYH1w">
                    </input>
                   </auto-check>
                  </div>
                  <div class="form-group">
                   <label for="custom_scope_query">
                    Query
                   </label>
                   <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                  </div>
                  <p class="text-small color-fg-muted">
                   To see all available qualifiers, see our
                   <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                    documentation
                   </a>
                   .
                  </p>
                 </input>
                </form>
               </div>
               <div data-target="custom-scopes.manageCustomScopesForm">
                <div data-target="custom-scopes.list">
                </div>
               </div>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
              <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
               Create saved search
              </button>
             </div>
            </dialog>
           </dialog-helper>
          </custom-scopes>
         </div>
        </qbsearch-input>
        <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="f0fquOERhC85IEs71KR4T9RzkS1HFlJfSPhvApvgsWW4EoWa-BZCTd770cYdHwNcBMtJ58KwlG9HmzGnA3TQbw">
        </input>
       </div>
       <div class="AppHeader-actions position-relative">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="global-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="global-create-menu-button" popovertarget="global-create-menu-overlay" type="button">
           <span class="Button-content">
            <span class="Button-visual Button-leadingVisual">
             <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
              </path>
             </svg>
            </span>
            <span class="Button-label">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </span>
           </span>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="global-create-menu-button" id="tooltip-1fecb7da-35ad-426d-b3d0-662e4d9593a9" popover="manual">
           Create new...
          </tool-tip>
          <anchored-position align="end" anchor="global-create-menu-button" anchor-offset="normal" data-view-component="true" id="global-create-menu-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="global-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="global-create-menu-list" role="menu">
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-287674fe-76e5-4cee-a4b5-723f7c0fa4e5" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New repository
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-8f826feb-6af9-4bd4-acbb-caa6fcb926f8" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Import repository
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-5cea43fd-7272-45b9-b31e-69a60c26cd60" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New codespace
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-e7241385-4ed1-4159-b13f-60ea2deb869e" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New gist
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-0a4ddb5a-0662-47ce-9eb3-a10ae5ee8fae" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New organization
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
        <a aria-labelledby="tooltip-da55e67e-237c-4954-bb1e-0e84a8290e38" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"ISSUES_HEADER","label":null}' data-view-component="true" href="/issues" id="icon-button-82e59e97-d0c5-466a-9d2c-0bb68ac612ba">
         <svg aria-hidden="true" class="octicon octicon-issue-opened Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-82e59e97-d0c5-466a-9d2c-0bb68ac612ba" id="tooltip-da55e67e-237c-4954-bb1e-0e84a8290e38" popover="manual">
         Issues
        </tool-tip>
        <a aria-labelledby="tooltip-1c1495f0-b868-44b7-839d-c9a96c9f1c1e" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS_HEADER","label":null}' data-view-component="true" href="/pulls" id="icon-button-13161966-1d48-490c-acb8-2105dbd40cbc">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-13161966-1d48-490c-acb8-2105dbd40cbc" id="tooltip-1c1495f0-b868-44b7-839d-c9a96c9f1c1e" popover="manual">
         Pull requests
        </tool-tip>
       </div>
       <notification-indicator class="js-socket-channel" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY2MTU2NX0=--5e4cf12ce3c87b506a515e42d4302604f0f931308680554ad2c6cbf5631417e8" data-fetch-indicator-enabled="true" data-fetch-indicator-src="/notifications/indicator" data-header-redesign-enabled="true" data-indicator-mode="none" data-tooltip-global="You have unread notifications" data-tooltip-none="You have no unread notifications" data-tooltip-unavailable="Notifications are unavailable at the moment." data-view-component="true">
        <a aria-label="Notifications" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"NOTIFICATIONS_HEADER","label":null}' data-hotkey="g n" data-target="notification-indicator.link" data-view-component="true" href="/notifications" id="AppHeader-notifications-button">
         <svg aria-hidden="true" class="octicon octicon-inbox Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.8 2.06A1.75 1.75 0 0 1 4.41 1h7.18c.7 0 1.333.417 1.61 1.06l2.74 6.395c.04.093.06.194.06.295v4.5A1.75 1.75 0 0 1 14.25 15H1.75A1.75 1.75 0 0 1 0 13.25v-4.5c0-.101.02-.202.06-.295Zm1.61.44a.25.25 0 0 0-.23.152L1.887 8H4.75a.75.75 0 0 1 .6.3L6.625 10h2.75l1.275-1.7a.75.75 0 0 1 .6-.3h2.863L11.82 2.652a.25.25 0 0 0-.23-.152Zm10.09 7h-2.875l-1.275 1.7a.75.75 0 0 1-.6.3h-3.5a.75.75 0 0 1-.6-.3L4.375 9.5H1.5v3.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-target="notification-indicator.tooltip" data-type="description" data-view-component="true" for="AppHeader-notifications-button" id="tooltip-617ea5f2-bad7-4cfc-ada7-0cff625040d7" popover="manual">
         Notifications
        </tool-tip>
       </notification-indicator>
       <div class="AppHeader-user">
        <deferred-side-panel data-url="/_side-panels/user?react_global_nav=false&amp;repository_id=379816251">
         <include-fragment data-target="deferred-side-panel.fragment">
          <user-drawer-side-panel>
           <button aria-label="Open user account menu" class="AppHeader-logo Button--invisible Button--medium Button Button--invisible-noVisuals color-bg-transparent p-0" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-adf4327e-96fb-4c57-a44b-5c3f4b407cca" data-view-component="true" id="dialog-show-dialog-adf4327e-96fb-4c57-a44b-5c3f4b407cca" type="button">
            <span class="Button-content">
             <span class="Button-label">
              <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
             </span>
            </span>
           </button>
           <dialog-helper>
            <dialog aria-describedby="dialog-adf4327e-96fb-4c57-a44b-5c3f4b407cca-description" aria-labelledby="dialog-adf4327e-96fb-4c57-a44b-5c3f4b407cca-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-right SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-adf4327e-96fb-4c57-a44b-5c3f4b407cca">
             <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title sr-only" id="dialog-adf4327e-96fb-4c57-a44b-5c3f4b407cca-title">
                 Account menu
                </h1>
                <div class="d-flex" data-view-component="true">
                 <div class="AppHeader-logo position-relative" data-view-component="true">
                  <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
                 </div>
                 <div class="overflow-hidden d-flex width-full" data-view-component="true">
                  <div class="lh-condensed overflow-hidden d-flex flex-column flex-justify-center ml-2 f5 mr-auto width-full" data-view-component="true">
                   <span class="Truncate text-bold" data-view-component="true">
                    <span class="Truncate-text" data-view-component="true">
                     ninenia
                    </span>
                   </span>
                  </div>
                  <action-menu class="d-sm-none d-md-none d-lg-none" data-select-variant="none" data-view-component="true">
                   <focus-group direction="vertical" mnemonics="" retain="">
                    <button aria-controls="user-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="user-create-menu-button" popovertarget="user-create-menu-overlay" type="button">
                     <span class="Button-content">
                      <span class="Button-visual Button-leadingVisual">
                       <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                        </path>
                       </svg>
                      </span>
                      <span class="Button-label">
                       <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                        </path>
                       </svg>
                      </span>
                     </span>
                    </button>
                    <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="user-create-menu-button" id="tooltip-61fdad39-e745-4a7b-966d-99c228fe1d9f" popover="manual">
                     Create new...
                    </tool-tip>
                    <anchored-position align="end" anchor="user-create-menu-button" anchor-offset="normal" data-view-component="true" id="user-create-menu-overlay" popover="auto" side="outside-bottom">
                     <div class="Overlay Overlay--size-auto" data-view-component="true">
                      <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
                       <action-list>
                        <div data-view-component="true">
                         <ul aria-labelledby="user-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="user-create-menu-list" role="menu">
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-2ba3b205-e704-4d67-b2e7-610b86d3cbeb" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New repository
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-cc4ca747-65e5-48a0-ae3e-71769ccc7d17" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             Import repository
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-8794d36d-c781-49c2-b732-a6c2da6a8a61" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                              </path>
                              <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New codespace
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-662e2a92-52c0-431d-9972-3599fd9c8f12" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New gist
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-2b4ab77a-55e3-4869-a549-b8444a189f0b" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New organization
                            </span>
                           </a>
                          </li>
                         </ul>
                        </div>
                       </action-list>
                      </div>
                     </div>
                    </anchored-position>
                   </focus-group>
                  </action-menu>
                 </div>
                </div>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-adf4327e-96fb-4c57-a44b-5c3f4b407cca" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="dialog-adf4327e-96fb-4c57-a44b-5c3f4b407cca-title">
              <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
               <div class="d-flex flex-column mb-3" data-view-component="true">
                <nav aria-label="User navigation" class="ActionList" data-view-component="true">
                 <nav-list>
                  <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-9008310b-5d59-4dc3-9dc2-91df1babd06c" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROFILE","label":null}' data-view-component="true" href="https://github.com/ninenia" id="item-1d1923d9-4d1b-4290-968d-be6ff41fe1e5">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-person" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your profile
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-fa3b2df7-699d-4a18-91db-31e95a189386" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_REPOSITORIES","label":null}' data-view-component="true" href="/ninenia?tab=repositories" id="item-c114b1b4-f36c-4e69-827d-b2d6ade54169">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your repositories
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_PROJECTS","label":null}' data-view-component="true" href="/ninenia?tab=projects" id="item-78681669-a5bf-42d4-a5be-20cce43fe5e3">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your projects
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-63565c23-300d-480b-8f16-ce40d33c8d7f" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-e7e8a02d-6079-414c-b2eb-01431805f9bd" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_STARS","label":null}' data-view-component="true" href="/ninenia?tab=stars" id="item-4e29c7dd-1929-43ae-9357-c843e27096d2">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-star" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your stars
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SPONSORS","label":null}' data-view-component="true" href="/sponsors/accounts" id="item-49b283de-7135-4f7c-bd4e-a3d4f8f887a2">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your sponsors
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_GISTS","label":null}' data-view-component="true" href="https://gist.github.com/mine" id="item-40c3d3fa-cb00-4446-8ac3-e8714a886629">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your gists
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-67a02dc9-32a9-49a9-906c-ee4ec0f9f720" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-1224c5e9-eee4-4b29-9a57-e5917f166769" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SETTINGS","label":null}' data-view-component="true" href="/settings/profile" id="item-aa0854fd-433a-457c-b557-f85ddfb0aa0f">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-gear" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Settings
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SUPPORT","label":null}' data-view-component="true" href="https://support.github.com" id="item-66f3686a-77e8-442f-92b5-da5b371815e3">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Support
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"COMMUNITY","label":null}' data-view-component="true" href="https://community.github.com" id="item-fd9819bc-5d42-42fe-ba44-e2c1ca400384">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Community
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent" data-analytics-event='{"category":"Global navigation","action":"LOGOUT","label":null}' data-view-component="true" href="/logout" id="item-0cb86af5-bcb6-49c0-9b20-4829698accc1">
                     <span class="ActionListItem-label" data-view-component="true">
                      Sign out
                     </span>
                    </a>
                   </li>
                  </ul>
                 </nav-list>
                </nav>
               </div>
              </div>
             </scrollable-region>
            </dialog>
           </dialog-helper>
          </user-drawer-side-panel>
         </include-fragment>
        </deferred-side-panel>
       </div>
       <div class="position-absolute mt-2">
        <site-header-logged-in-user-menu>
        </site-header-logged-in-user-menu>
       </div>
      </div>
     </div>
     <div class="AppHeader-localBar">
      <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav" data-pjax="#js-repo-pjax-container" data-view-component="true">
       <ul class="UnderlineNav-body list-style-none" data-view-component="true">
        <li class="d-inline-flex" data-view-component="true">
         <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /QuentinRougemont/coho_ldscp_genomics" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics" id="code-tab">
          <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
           </path>
          </svg>
          <span data-content="Code">
           Code
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /QuentinRougemont/coho_ldscp_genomics/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/issues" id="issues-tab">
          <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
           </path>
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
           </path>
          </svg>
          <span data-content="Issues">
           Issues
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /QuentinRougemont/coho_ldscp_genomics/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/pulls" id="pull-requests-tab">
          <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
           </path>
          </svg>
          <span data-content="Pull requests">
           Pull requests
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /QuentinRougemont/coho_ldscp_genomics/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/actions" id="actions-tab">
          <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
           </path>
          </svg>
          <span data-content="Actions">
           Actions
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /QuentinRougemont/coho_ldscp_genomics/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/projects" id="projects-tab">
          <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
           </path>
          </svg>
          <span data-content="Projects">
           Projects
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /QuentinRougemont/coho_ldscp_genomics/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/security" id="security-tab">
          <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
           </path>
          </svg>
          <span data-content="Security">
           Security
          </span>
          <include-fragment accept="text/fragment+html" src="/QuentinRougemont/coho_ldscp_genomics/security/overall-count">
          </include-fragment>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /QuentinRougemont/coho_ldscp_genomics/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/pulse" id="insights-tab">
          <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
           </path>
          </svg>
          <span data-content="Insights">
           Insights
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
       </ul>
       <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="action-menu-e755f939-4d7c-4b1d-b837-1ff2e7ef92b4-list" aria-haspopup="true" aria-labelledby="tooltip-eb8f72b4-8424-4d0c-ac94-e45d28f4b91c" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-e755f939-4d7c-4b1d-b837-1ff2e7ef92b4-button" popovertarget="action-menu-e755f939-4d7c-4b1d-b837-1ff2e7ef92b4-overlay" type="button">
           <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
           </svg>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-e755f939-4d7c-4b1d-b837-1ff2e7ef92b4-button" id="tooltip-eb8f72b4-8424-4d0c-ac94-e45d28f4b91c" popover="manual">
           Additional navigation options
          </tool-tip>
          <anchored-position align="start" anchor="action-menu-e755f939-4d7c-4b1d-b837-1ff2e7ef92b4-button" anchor-offset="normal" data-view-component="true" id="action-menu-e755f939-4d7c-4b1d-b837-1ff2e7ef92b4-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="action-menu-e755f939-4d7c-4b1d-b837-1ff2e7ef92b4-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-e755f939-4d7c-4b1d-b837-1ff2e7ef92b4-list" role="menu">
                <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics" id="item-28f692e9-b3af-4ec5-9cde-6a2e5dc2a1aa" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Code
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/issues" id="item-73d10688-ed89-436c-a374-3d603a43375b" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Issues
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/pulls" id="item-738afe93-0307-4569-b327-df130bdaa3cc" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Pull requests
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/actions" id="item-4fec4bc8-ff40-4ab1-b0dc-679e515cc82c" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Actions
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/projects" id="item-77ddb6a9-772b-4241-b263-87c1f652af63" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Projects
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/security" id="item-65d63715-ef3d-4fd2-89f0-ad71df5e40ea" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Security
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/pulse" id="item-92944de0-95c4-43ac-a2ac-4331d598e245" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Insights
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
       </div>
      </nav>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-bb6aba10-113c-4ddb-9687-14be926b2bc5" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-7b673abe-5418-442f-a48b-d4f07dce909e" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-7b673abe-5418-442f-a48b-d4f07dce909e" id="tooltip-bb6aba10-113c-4ddb-9687-14be926b2bc5" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <notification-shelf-watcher class="js-socket-channel" data-base-url="https://github.com/notifications/beta/shelf" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY2MTU2NX0=--5e4cf12ce3c87b506a515e42d4302604f0f931308680554ad2c6cbf5631417e8" data-view-component="true">
   </notification-shelf-watcher>
   <div data-initial="" data-target="notification-shelf-watcher.placeholder" hidden="">
   </div>
   <details class="details-reset details-overlay details-overlay-dark js-command-palette-dialog" data-turbo-replace="" id="command-palette-pjax-container">
    <summary aria-label="Command palette trigger" tabindex="-1">
    </summary>
    <details-dialog aria-label="Command palette" class="command-palette-details-dialog d-flex flex-column flex-justify-center height-fit">
     <command-palette activation-hotkey="Mod+k,Mod+Alt+k" class="command-palette color-bg-default rounded-3 border color-shadow-small" command-mode-hotkey="Mod+Shift+K" data-action="
        command-palette-input-ready:command-palette#inputReady
        command-palette-page-stack-updated:command-palette#updateInputScope
        itemsUpdated:command-palette#itemsUpdated
        keydown:command-palette#onKeydown
        loadingStateChanged:command-palette#loadingStateChanged
        selectedItemChanged:command-palette#selectedItemChanged
        pageFetchError:command-palette#pageFetchError
      " return-to="/QuentinRougemont/coho_ldscp_genomics" user-id="126681664">
      <command-palette-mode data-char="#" data-placeholder="Search issues and pull requests" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="#" data-placeholder="Search issues, pull requests, discussions, and projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="!" data-placeholder="Search projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a user, organization, or repository" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a repository" data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode data-char="/" data-placeholder="Search files" data-scope-types='["repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="?">
      </command-palette-mode>
      <command-palette-mode data-char="&gt;" data-placeholder="Run a command">
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode class="js-command-palette-default-mode" data-char="" data-placeholder="Search or jump to...">
      </command-palette-mode>
      <command-palette-input data-action="
          command-palette-input:command-palette#onInput
          command-palette-select:command-palette#onSelect
          command-palette-descope:command-palette#onDescope
          command-palette-cleared:command-palette#onInputClear
        " placeholder="Search or jump to...">
       <div class="js-search-icon d-flex flex-items-center mr-2" style="height: 26px">
        <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
         </path>
        </svg>
       </div>
       <div class="js-spinner d-flex flex-items-center mr-2 color-fg-muted" hidden="">
        <svg aria-label="Loading" class="anim-rotate" fill="none" height="16" viewbox="0 0 16 16" width="16">
         <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
         </circle>
         <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
         </path>
        </svg>
       </div>
       <command-palette-scope>
        <div class="color-fg-subtle" data-target="command-palette-scope.placeholder" hidden="">
         /
         <span class="text-semibold color-fg-default">
          ...
         </span>
         /
        </div>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDQ6VXNlcjE3NTg4ODg4" data-targets="command-palette-scope.tokens" data-text="QuentinRougemont" data-type="owner" data-value="QuentinRougemont" style="white-space:nowrap;line-height:20px;">
         QuentinRougemont
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDEwOlJlcG9zaXRvcnkzNzk4MTYyNTE=" data-targets="command-palette-scope.tokens" data-text="coho_ldscp_genomics" data-type="repository" data-value="coho_ldscp_genomics" style="white-space:nowrap;line-height:20px;">
         coho_ldscp_genomics
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
       </command-palette-scope>
       <div class="command-palette-input-group flex-1 form-control border-0 box-shadow-none" style="z-index: 0">
        <div class="command-palette-typeahead position-absolute d-flex flex-items-center Truncate">
         <span class="typeahead-segment input-mirror" data-target="command-palette-input.mirror">
         </span>
         <span class="Truncate-text" data-target="command-palette-input.typeaheadText">
         </span>
         <span class="typeahead-segment" data-target="command-palette-input.typeaheadPlaceholder">
         </span>
        </div>
        <input aria-label="Hidden input for typeahead" class="js-overlay-input typeahead-input d-none" disabled="" tabindex="-1"/>
        <input aria-autocomplete="list" aria-controls="command-palette-page-stack" aria-expanded="false" aria-haspopup="listbox" aria-label="Command palette input" autocapitalize="off" autocomplete="off" autocorrect="off" class="js-input typeahead-input form-control border-0 box-shadow-none input-block width-full no-focus-indicator" data-action="
              input:command-palette-input#onInput
              keydown:command-palette-input#onKeydown
            " role="combobox" spellcheck="false" type="text"/>
       </div>
       <div class="position-relative d-inline-block" data-view-component="true">
        <button aria-keyshortcuts="Control+Backspace" class="btn-octicon command-palette-input-clear-button" data-action="click:command-palette-input#onClear keypress:command-palette-input#onClear" data-target="command-palette-input.clearButton" data-view-component="true" hidden="hidden" id="command-palette-clear-button" type="button">
         <svg aria-hidden="true" class="octicon octicon-x-circle-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
          </path>
         </svg>
        </button>
        <tool-tip class="sr-only position-absolute" data-direction="w" data-type="label" data-view-component="true" for="command-palette-clear-button" id="tooltip-6097cbf0-9d01-443b-9a78-c48d08f77325" popover="manual">
         Clear Command Palette
        </tool-tip>
       </div>
      </command-palette-input>
      <command-palette-page-stack data-action="command-palette-page-octicons-cached:command-palette-page-stack#cacheOcticons" data-default-scope-id="MDEwOlJlcG9zaXRvcnkzNzk4MTYyNTE=" data-default-scope-type="Repository">
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search discussions
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           !
          </kbd>
          to search projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search teams
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='[""]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search people and organizations
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           &gt;
          </kbd>
          to activate command mode
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Go to your accessibility settings to change your keyboard shortcuts
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type author:@me to search your content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:pr to filter to pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:issue to filter to issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:project to filter to projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:open to filter to open content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="mx-3 my-2 flash flash-error d-flex flex-items-center" data-on-error="" data-scope-types="*">
        <div>
         <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
          </path>
         </svg>
        </div>
        <div class="px-2">
         We’ve encountered an error and some results aren't available at this time. Type a new search or try again later.
        </div>
       </command-palette-tip>
       <command-palette-tip class="h4 color-fg-default pl-3 pb-2 pt-3" data-match-mode="[^?]|^$" data-on-empty="" data-scope-types="*">
        No results matched your search
       </command-palette-tip>
       <div hidden="">
        <div data-octicon-id="arrow-right-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="arrow-right-color-fg-default" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-default" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="codespaces-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
          </path>
          <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="copy-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-copy color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
          </path>
          <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="dash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-dash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 7.75A.75.75 0 0 1 2.75 7h10a.75.75 0 0 1 0 1.5h-10A.75.75 0 0 1 2 7.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="file-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-file color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="gear-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-gear color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="lock-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-lock color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M4 4a4 4 0 0 1 8 0v2h.25c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 12.25 15h-8.5A1.75 1.75 0 0 1 2 13.25v-5.5C2 6.784 2.784 6 3.75 6H4Zm8.25 3.5h-8.5a.25.25 0 0 0-.25.25v5.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25ZM10.5 6V4a2.5 2.5 0 1 0-5 0v2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="moon-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-moon color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M9.598 1.591a.749.749 0 0 1 .785-.175 7.001 7.001 0 1 1-8.967 8.967.75.75 0 0 1 .961-.96 5.5 5.5 0 0 0 7.046-7.046.75.75 0 0 1 .175-.786Zm1.616 1.945a7 7 0 0 1-7.678 7.678 5.499 5.499 0 1 0 7.678-7.678Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="person-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-person color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="pencil-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-pencil color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="issue-opened-open" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-issue-opened open" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="git-pull-request-draft-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request-draft color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M3.25 1A2.25 2.25 0 0 1 4 5.372v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.251 2.251 0 0 1 3.25 1Zm9.5 14a2.25 2.25 0 1 1 0-4.5 2.25 2.25 0 0 1 0 4.5ZM2.5 3.25a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0ZM3.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm9.5 0a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM14 7.5a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm0-4.25a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="search-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sun-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sun color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8Zm0-1.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Zm5.657-8.157a.75.75 0 0 1 0 1.061l-1.061 1.06a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.06-1.06a.75.75 0 0 1 1.06 0Zm-9.193 9.193a.75.75 0 0 1 0 1.06l-1.06 1.061a.75.75 0 1 1-1.061-1.06l1.06-1.061a.75.75 0 0 1 1.061 0ZM8 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V.75A.75.75 0 0 1 8 0ZM3 8a.75.75 0 0 1-.75.75H.75a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 3 8Zm13 0a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 16 8Zm-8 5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 8 13Zm3.536-1.464a.75.75 0 0 1 1.06 0l1.061 1.06a.75.75 0 0 1-1.06 1.061l-1.061-1.06a.75.75 0 0 1 0-1.061ZM2.343 2.343a.75.75 0 0 1 1.061 0l1.06 1.061a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-1.06-1.06a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sync-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sync color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.705 8.005a.75.75 0 0 1 .834.656 5.5 5.5 0 0 0 9.592 2.97l-1.204-1.204a.25.25 0 0 1 .177-.427h3.646a.25.25 0 0 1 .25.25v3.646a.25.25 0 0 1-.427.177l-1.38-1.38A7.002 7.002 0 0 1 1.05 8.84a.75.75 0 0 1 .656-.834ZM8 2.5a5.487 5.487 0 0 0-4.131 1.869l1.204 1.204A.25.25 0 0 1 4.896 6H1.25A.25.25 0 0 1 1 5.75V2.104a.25.25 0 0 1 .427-.177l1.38 1.38A7.002 7.002 0 0 1 14.95 7.16a.75.75 0 0 1-1.49.178A5.5 5.5 0 0 0 8 2.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="trash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-trash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="key-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-key color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="comment-discussion-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-slash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell-slash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="paintbrush-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-paintbrush color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.134 1.535c.7-.509 1.416-.942 2.076-1.155.649-.21 1.463-.267 2.069.34.603.601.568 1.411.368 2.07-.202.668-.624 1.39-1.125 2.096-1.011 1.424-2.496 2.987-3.775 4.249-1.098 1.084-2.132 1.839-3.04 2.3a3.744 3.744 0 0 1-1.055 3.217c-.431.431-1.065.691-1.657.861-.614.177-1.294.287-1.914.357A21.151 21.151 0 0 1 .797 16H.743l.007-.75H.749L.742 16a.75.75 0 0 1-.743-.742l.743-.008-.742.007v-.054a21.25 21.25 0 0 1 .13-2.284c.067-.647.187-1.287.358-1.914.17-.591.43-1.226.86-1.657a3.746 3.746 0 0 1 3.227-1.054c.466-.893 1.225-1.907 2.314-2.982 1.271-1.255 2.833-2.75 4.245-3.777ZM1.62 13.089c-.051.464-.086.929-.104 1.395.466-.018.932-.053 1.396-.104a10.511 10.511 0 0 0 1.668-.309c.526-.151.856-.325 1.011-.48a2.25 2.25 0 1 0-3.182-3.182c-.155.155-.329.485-.48 1.01a10.515 10.515 0 0 0-.309 1.67Zm10.396-10.34c-1.224.89-2.605 2.189-3.822 3.384l1.718 1.718c1.21-1.205 2.51-2.597 3.387-3.833.47-.662.78-1.227.912-1.662.134-.444.032-.551.009-.575h-.001V1.78c-.014-.014-.113-.113-.548.027-.432.14-.995.462-1.655.942Zm-4.832 7.266-.001.001a9.859 9.859 0 0 0 1.63-1.142L7.155 7.216a9.7 9.7 0 0 0-1.161 1.607c.482.302.889.71 1.19 1.192Z">
          </path>
         </svg>
        </div>
        <command-palette-item-group data-default-priority="0" data-group-hint="" data-group-id="top" data-group-limits="{}" data-group-title="Top result">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="1" data-group-hint="Type &gt; to filter" data-group-id="commands" data-group-limits='{"static_items_page":50,"issue":50,"pull_request":50,"discussion":50}' data-group-title="Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="2" data-group-hint="Type &gt; to filter" data-group-id="global_commands" data-group-limits='{"issue":0,"pull_request":0,"discussion":0}' data-group-title="Global Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="3" data-group-hint="" data-group-id="this_page" data-group-limits="{}" data-group-title="This Page">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="4" data-group-hint="" data-group-id="files" data-group-limits="{}" data-group-title="Files">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="5" data-group-hint="" data-group-id="default" data-group-limits='{"static_items_page":50}' data-group-title="Default">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="6" data-group-hint="" data-group-id="pages" data-group-limits='{"repository":10}' data-group-title="Pages">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="7" data-group-hint="" data-group-id="access_policies" data-group-limits="{}" data-group-title="Access Policies">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="8" data-group-hint="" data-group-id="organizations" data-group-limits="{}" data-group-title="Organizations">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="9" data-group-hint="" data-group-id="repositories" data-group-limits="{}" data-group-title="Repositories">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="10" data-group-hint="Type # to filter" data-group-id="references" data-group-limits="{}" data-group-title="Issues, pull requests, and discussions">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="11" data-group-hint="" data-group-id="teams" data-group-limits="{}" data-group-title="Teams">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="12" data-group-hint="" data-group-id="users" data-group-limits="{}" data-group-title="Users">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="13" data-group-hint="" data-group-id="memex_projects" data-group-limits="{}" data-group-title="Projects">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="14" data-group-hint="" data-group-id="projects" data-group-limits="{}" data-group-title="Projects (classic)">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="15" data-group-hint="" data-group-id="footer" data-group-limits="{}" data-group-title="Footer">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="16" data-group-hint="" data-group-id="modes_help" data-group-limits="{}" data-group-title="Modes">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="17" data-group-hint="" data-group-id="filters_help" data-group-limits="{}" data-group-title="Use filters in issues, pull requests, discussions, and projects">
        </command-palette-item-group>
        <command-palette-page data-page-title="QuentinRougemont" data-scope-id="MDQ6VXNlcjE3NTg4ODg4" data-scope-type="owner" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
        <command-palette-page data-page-title="coho_ldscp_genomics" data-scope-id="MDEwOlJlcG9zaXRvcnkzNzk4MTYyNTE=" data-scope-type="repository" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
       </div>
       <command-palette-page data-is-root="">
       </command-palette-page>
       <command-palette-page data-page-title="QuentinRougemont" data-scope-id="MDQ6VXNlcjE3NTg4ODg4" data-scope-type="owner">
       </command-palette-page>
       <command-palette-page data-page-title="coho_ldscp_genomics" data-scope-id="MDEwOlJlcG9zaXRvcnkzNzk4MTYyNTE=" data-scope-type="repository">
       </command-palette-page>
      </command-palette-page-stack>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="search-links">
      </server-defined-provider>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="help">
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues
         </strong>
         and
         <strong>
          pull requests
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues, pull requests, discussions,
         </strong>
         and
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="@" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          organizations, repositories,
         </strong>
         and
         <strong>
          users
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          @
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="!" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          !
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="/" data-scope-types='["repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          files
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          /
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="&gt;">
        <span data-target="command-palette-help.titleElement">
         Activate
         <strong>
          command mode
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          &gt;
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:pr">
        <span data-target="command-palette-help.titleElement">
         Filter to pull requests
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:pr
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:issue">
        <span data-target="command-palette-help.titleElement">
         Filter to issues
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:issue
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:discussion" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:discussion
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:project" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to projects
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:project
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:open">
        <span data-target="command-palette-help.titleElement">
         Filter to open issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:open
         </kbd>
        </span>
       </command-palette-help>
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/commands" data-supported-modes="[]" data-supports-commands="" data-targets="command-palette.serverDefinedProviderElements" data-type="commands">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_page_navigation" data-supported-modes='[""]' data-supported-scope-types='["","owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to" data-supported-modes='["@","@"]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to_members_only" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_members_only_prefetched" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/files" data-supported-modes='["/"]' data-supported-scope-types='["repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="files">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/discussions" data-supported-modes='["#"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/projects" data-supported-modes='["#","!"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/recent_issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/teams" data-supported-modes='["@",""]' data-supported-scope-types='["owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/name_with_owner_repository" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
     </command-palette>
    </details-dialog>
   </details>
   <div class="position-fixed bottom-0 left-0 ml-5 mb-5 js-command-palette-toasts" style="z-index: 1000">
    <div class="Toast Toast--loading" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="Toast--spinner" height="18" viewbox="0 0 32 32" width="18">
       <path d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4" fill="#959da5">
       </path>
       <path d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z" fill="#ffffff">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--error" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-stop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M4.47.22A.749.749 0 0 1 5 0h6c.199 0 .389.079.53.22l4.25 4.25c.141.14.22.331.22.53v6a.749.749 0 0 1-.22.53l-4.25 4.25A.749.749 0 0 1 11 16H5a.749.749 0 0 1-.53-.22L.22 11.53A.749.749 0 0 1 0 11V5c0-.199.079-.389.22-.53Zm.84 1.28L1.5 5.31v5.38l3.81 3.81h5.38l3.81-3.81V5.31L10.69 1.5ZM8 4a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 4Zm0 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--warning" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--success" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-info" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.5 7.75A.75.75 0 0 1 7.25 7h1a.75.75 0 0 1 .75.75v2.75h.25a.75.75 0 0 1 0 1.5h-2a.75.75 0 0 1 0-1.5h.25v-2h-.25a.75.75 0 0 1-.75-.75ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
   </div>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <template class="js-user-list-create-dialog-template" data-label="Create list">
       <div class="Box-header">
        <h2 class="Box-title">
         Create list
        </h2>
       </div>
       <!-- '"` -->
       <!-- </textarea></xmp> -->
       <form accept-charset="UTF-8" action="/stars/ninenia/lists" class="Box-body d-flex flex-column p-3 js-user-list-form" data-turbo="false" method="post">
        <input autocomplete="off" name="authenticity_token" type="hidden" value="WCJxPN9Ov0u3oPfD3LVmUCTvdD9fGjCA_QocHlnQHa6ITtsYSpRp7FSdXaRPbf_2gey_Sts9GbxjoDmr7QELGQ">
         <p class="color-fg-subtle mb-3">
          Create a list to organize your starred repositories.
         </p>
         <input name="repository_id" type="hidden" value="{{ repositoryId }}"/>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <auto-check required="" src="/stars/ninenia/list-check?attr=name">
           <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
            <input aria-label="List name" autofocus="" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="32" maxlength="32" name="user_list[name]" placeholder="⭐️ Name this list" required="" type="text" value="{{ placeholderName }}"/>
           </text-expander>
           <input data-csrf="true" type="hidden" value="2iAnd18ZmoyWYyZAt1WEFg0QbYPTXfqGdfSneq2i3KO6CHuGkDfI6EAouQRUV0AgDPoz8n-vRLzZ3t2zT8poDQ">
           </input>
          </auto-check>
          <p class="note error position-relative js-user-list-error" hidden="">
           Name .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           32 remaining
          </p>
         </div>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
           <textarea aria-label="List description" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="160" maxlength="160" name="user_list[description]" placeholder="Write a description" style="height: 74px; min-height: 74px"></textarea>
          </text-expander>
          <p class="note error position-relative js-user-list-error" hidden="">
           Description .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           160 remaining
          </p>
         </div>
         <div class="js-user-list-base flash flash-error mx-0 mt-0 mb-2" data-generic-message="Unable to save your list at this time." data-view-component="true" hidden="hidden">
          .
         </div>
         <button class="Button--primary Button--medium Button Button--fullWidth mt-2" data-disable-invalid="true" data-submitting-message="Creating..." data-view-component="true" disabled="disabled" type="submit">
          <span class="Button-content">
           <span class="Button-label">
            Create
           </span>
          </span>
         </button>
         <p class="note mt-2 mb-0">
          <strong>
           Tip:
          </strong>
          type
          <code>
           :
          </code>
          to add emoji to the name or description.
         </p>
        </input>
       </form>
       <div class="Box-footer Box-row--gray text-small color-fg-muted d-flex flex-items-baseline py-2" data-view-component="true">
        <span aria-label="Feature Release Label: Beta" class="Label Label--success Label--inline px-2 mr-2" data-view-component="true" title="Feature Release Label: Beta">
         Beta
        </span>
        <span class="mr-1">
         Lists are currently in beta.
        </span>
        <a href="https://github.com//github/feedback/discussions/categories/lists">
         Share feedback and report bugs.
        </a>
       </div>
      </template>
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header">
       <div class="d-flex flex-wrap flex-justify-end mb-3 container-xl px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal" id="repo-title-component">
          <img alt="Owner avatar" class="avatar mr-2 d-none d-md-block avatar-user" height="24" src="https://avatars.githubusercontent.com/u/17588888?s=48&amp;v=4" width="24"/>
          <strong class="mr-2 flex-self-stretch d-none d-md-block" itemprop="name">
           <a class="color-fg-default" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/QuentinRougemont/coho_ldscp_genomics">
            coho_ldscp_genomics
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1 d-none d-md-block">
           Public
          </span>
         </div>
         <div class="d-none d-md-block">
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <include-fragment loading="lazy" src="/notifications/379816251/watch_subscription?aria_id_prefix=repository-details&amp;button_block=false&amp;show_count=true">
            <div data-hide-on-error="">
             <button aria-label="Watching a repository" class="tooltipped tooltipped-n Button--secondary Button--small Button" data-view-component="true" disabled="disabled" type="button">
              <span class="Button-content">
               <span class="Button-visual Button-leadingVisual">
                <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                 </path>
                </svg>
               </span>
               <span class="Button-label">
                Watch
               </span>
              </span>
             </button>
            </div>
            <p data-show-on-error="" hidden="">
             <span class="d-flex pt-1">
              <svg aria-hidden="true" class="octicon octicon-alert mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
               </path>
              </svg>
              Couldn't load subscription status.
              <button class="Button--link Button--medium Button" data-retry-button="" data-view-component="true" type="button">
               <span class="Button-content">
                <span class="Button-label">
                 Retry
                </span>
               </span>
              </button>
             </span>
            </p>
           </include-fragment>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a class="btn-sm btn BtnGroup-item" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":379816251,"originating_url":"https://github.com/QuentinRougemont/coho_ldscp_genomics","user_id":126681664}}' data-hydro-click-hmac="84664cc8111f2a8115f5454cc30fadb6df4d2e1c56cb75a8797953bf0321ed4c" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/fork" icon="repo-forked" id="fork-button">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             Fork
             <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="1">
              1
             </span>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="fork-button" id="tooltip-aca9f404-0235-4d27-8e0a-b97a9253f41d" popover="manual">
              Fork your own copy of QuentinRougemont/coho_ldscp_genomics
             </tool-tip>
            </a>
            <details class="details-reset details-overlay BtnGroup-parent d-inline-block position-relative" data-view-component="true" group_item="true" id="my-forks-menu-379816251">
             <summary aria-label="See your forks of this repository" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
              <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
               </path>
              </svg>
             </summary>
             <details-menu class="SelectMenu right-0" role="menu" src="/QuentinRougemont/coho_ldscp_genomics/my_forks_menu_content?can_fork=true">
              <div class="SelectMenu-modal">
               <button aria-label="Close menu" class="SelectMenu-closeButton position-absolute right-0 m-2" data-toggle-for="my-forks-menu-379816251" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
               <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-ac0047">
                <div class="SelectMenu-list">
                 <include-fragment aria-label="Loading" class="SelectMenu-loading">
                  <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                   <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                   </circle>
                   <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                   </path>
                  </svg>
                 </include-fragment>
                </div>
               </div>
              </div>
             </details-menu>
            </details>
           </div>
          </li>
          <li>
           <template class="js-unstar-confirmation-dialog-template">
            <div class="Box-header">
             <h2 class="Box-title">
              Unstar this repository?
             </h2>
            </div>
            <div class="Box-body">
             <p class="mb-3">
              This will remove {{ repoNameWithOwner }} from the {{ listsWithCount }} that it's been added to.
             </p>
             <div class="form-actions">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="{{ confirmUrl }}" class="js-social-confirmation-form" data-turbo="false" method="post">
               <input name="authenticity_token" type="hidden" value="{{ confirmCsrfToken }}"/>
               <input name="confirm" type="hidden" value="true"/>
               <button class="btn-danger btn width-full" data-close-dialog="true" data-view-component="true" type="submit">
                Unstar
               </button>
              </form>
             </div>
            </div>
           </template>
           <div class="js-toggler-container js-social-container starring-container d-flex" data-view-component="true">
            <div class="starred BtnGroup flex-1 ml-0" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/QuentinRougemont/coho_ldscp_genomics/unstar" class="js-social-form BtnGroup-parent flex-auto js-deferred-toggler-target" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="vvSfOL9wbdfwOfa2snsIpbyU2qnFgkceIm4NHFWMv01uqhpboolHLz1xdvJBn--PWUTmJzDIWDFNotJiLU1wSA">
               <input class="js-confirm-csrf-token" data-csrf="true" type="hidden" value="8_MCg3A3Tq0YNTVHIl6xDU9x-8X7-BKOwh_GBcs5wv0jrYfgbc5kVdV9tQPRulYnqqHHSw6yDaGt0xl7s_gN-A">
                <input name="context" type="hidden" value="repository"/>
                <button aria-label="Unstar this repository (0)" class="rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":379816251,"originating_url":"https://github.com/QuentinRougemont/coho_ldscp_genomics","user_id":126681664}}' data-hydro-click-hmac="dd12f1758eadab5d2887b21c40eca2b8a70ec4f05aeb3ba184369bf032a01b98" data-view-component="true" type="submit">
                 <svg aria-hidden="true" class="octicon octicon-star-fill starred-button-icon d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                  </path>
                 </svg>
                 <span class="d-inline" data-view-component="true">
                  Starred
                 </span>
                 <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-unstar" title="0">
                  0
                 </span>
                </button>
               </input>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-379816251-starred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/QuentinRougemont/coho_ldscp_genomics/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-379816251-starred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-dc7927">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
            <div class="unstarred BtnGroup ml-0 flex-1" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/QuentinRougemont/coho_ldscp_genomics/star" class="js-social-form BtnGroup-parent flex-auto" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="3EcFQLiYs9hxnYpmzC9s1LTNAeJfG4lVaoXNaC3eiI6NA7Rt6xlbiHjh1Su-i864rsloK8A8A11LrQ5ZwYTT5Q">
               <input name="context" type="hidden" value="repository"/>
               <button aria-label="Star this repository (0)" class="js-toggler-target rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":379816251,"originating_url":"https://github.com/QuentinRougemont/coho_ldscp_genomics","user_id":126681664}}' data-hydro-click-hmac="ff48562ed73404d4617e44f742ec36e053981337d82081da302be4343b9d9a70" data-view-component="true" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
                <span class="d-inline" data-view-component="true">
                 Star
                </span>
                <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
                 0
                </span>
               </button>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-379816251-unstarred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/QuentinRougemont/coho_ldscp_genomics/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-379816251-unstarred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-9578b0">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div class="container-xl">
        <div data-turbo-replace="" id="responsive-meta-container">
         <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
          <div class="d-flex gap-2 mt-n3 mb-3 flex-wrap">
           <div class="d-flex flex-row gap-2">
            <notifications-list-subscription-form data-action="notifications-dialog-label-toggled:notifications-list-subscription-form#handleDialogLabelToggle">
             <details class="details-reset details-overlay details-overlay-dark" data-action="toggle:notifications-list-subscription-form#detailsToggled" data-target="notifications-list-subscription-form.details">
              <summary aria-label="Notifications" class="Button--iconOnly Button--secondary Button--medium Button p-0" data-ga-click="Repository, click Watch settings, action:files#disambiguate" data-hydro-click='{"event_type":"repository.click","payload":{"target":"WATCH_BUTTON","repository_id":379816251,"originating_url":"https://github.com/QuentinRougemont/coho_ldscp_genomics","user_id":126681664}}' data-hydro-click-hmac="0f769806c75de9655daed1feffe48ccdb540b3ab559a8e93f39e09ed9c458087" data-view-component="true" id="notifications-button">
               <span class="Button-content">
                <span class="Button-label">
                 <span data-menu-button="">
                  <span data-target="notifications-list-subscription-form.unwatchButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.stopIgnoringButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-bell-slash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.watchButtonCopy">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                 </span>
                </span>
               </span>
              </summary>
              <details-dialog aria-label="Notification settings" class="anim-fade-in fast Box Box--overlay">
               <div data-target="notifications-list-subscription-form.menu">
                <div class="Box-header">
                 <button aria-label="Close dialog" class="Box-btn-octicon btn-octicon float-right" data-action="click:notifications-list-subscription-form#closeMenu" data-close-dialog="" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                 <h3 class="Box-title">
                  Notifications
                 </h3>
                </div>
                <div class="SelectMenu-list mb-0" role="menu">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitForm" data-target="notifications-list-subscription-form.form" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="SZpZrL_eOqldFR3LAUQRlkS3sbvVWD5eDBSpUol6s9NRWlXWoWFPqLEGKsHdpJfhG5IirgAClYou71um3ssNSg">
                   <input name="repository_id" type="hidden" value="379816251"/>
                   <button aria-checked="true" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="included">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Participating and @mentions
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Only receive notifications from this repository when participating or @mentioned.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="subscribed">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      All Activity
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Notified of all notifications on this repository.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="ignore">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Ignore
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Never be notified.
                     </div>
                    </div>
                   </button>
                  </input>
                 </form>
                 <button aria-checked="false" aria-haspopup="true" class="SelectMenu-item flex-items-start pr-3" data-action="click:notifications-list-subscription-form#openCustomDialog" data-target="notifications-list-subscription-form.customButton" role="menuitemradio" type="button">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                    </path>
                   </svg>
                  </span>
                  <div>
                   <div class="d-flex flex-items-start flex-justify-between">
                    <div class="f5 text-bold">
                     Custom
                    </div>
                    <div class="f5 pr-1">
                     <svg aria-hidden="true" class="octicon octicon-arrow-right" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </div>
                   <div class="text-small color-fg-muted text-normal pb-1">
                    Select events you want to be notified of in addition to participating and @mentions.
                   </div>
                  </div>
                 </button>
                 <div class="px-3 py-2 d-flex color-bg-subtle flex-items-center">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile SelectMenu-icon SelectMenu-icon--device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </span>
                  <span classname="text-small color-fg-muted text-normal pb-1">
                   Get push notifications on
                   <a href="https://apps.apple.com/app/apple-store/id1477376905?ct=watch-dropdown&amp;mt=8&amp;pt=524675" rel="noopener noreferrer" target="_blank">
                    iOS
                   </a>
                   or
                   <a href="https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dwatch-dropdown%26utm_medium%3Dweb%26utm_source%3Dgithub" rel="noopener noreferrer" target="_blank">
                    Android
                   </a>
                   .
                  </span>
                 </div>
                </div>
               </div>
               <div data-target="notifications-list-subscription-form.customDialog" hidden="">
                <header class="SelectMenu-header pb-0 border-bottom-0 px-2 px-sm-3">
                 <h1 class="f3 SelectMenu-title d-inline-flex">
                  <button aria-label="Return to menu" class="color-bg-default border-0 px-2 py-0 m-0 Link--secondary f5" data-action="click:notifications-list-subscription-form#closeCustomDialog" type="button">
                   <svg aria-hidden="true" class="octicon octicon-arrow-left" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z">
                    </path>
                   </svg>
                  </button>
                  Custom
                 </h1>
                </header>
                <div class="overflow-visible">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitCustomForm" data-target="notifications-list-subscription-form.customform" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="4cOmZaTTdBs6g4ARXkpUUD2mTlPLEvOdhZ3Gnzj9wyv5A6ofumwBGtaQtxuCqtInYoPdRh5IWEmnZjRrb0x9sg">
                   <input name="repository_id" type="hidden" value="379816251"/>
                   <fieldset>
                    <legend>
                     <div class="text-small color-fg-muted pt-0 pr-3 pb-3 pl-6 pl-sm-5 border-bottom mb-3">
                      Select events you want to be notified of in addition to participating and @mentions.
                     </div>
                    </legend>
                    <div data-target="notifications-list-subscription-form.labelInputs">
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Issue"/>
                      Issues
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="PullRequest"/>
                      Pull requests
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Release"/>
                      Releases
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input aria-describedby="Discussion-disabled" aria-disabled="true" data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Discussion"/>
                      Discussions
                     </label>
                     <div class="color-fg-muted" id="Discussion-disabled">
                      &gt;
                          Discussions are not enabled for this repository
                     </div>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="SecurityAlert"/>
                      Security alerts
                     </label>
                    </div>
                   </fieldset>
                   <div class="pt-2 pb-3 px-3 d-flex flex-justify-start flex-row-reverse">
                    <button class="Button--primary Button--medium Button ml-2" data-target="notifications-list-subscription-form.customSubmit" data-view-component="true" disabled="disabled" name="do" type="submit" value="custom">
                     <span class="Button-content">
                      <span class="Button-label">
                       Apply
                      </span>
                     </span>
                    </button>
                    <button class="Button--secondary Button--medium Button" data-action="click:notifications-list-subscription-form#resetForm" data-close-dialog="" data-view-component="true" type="button">
                     <span class="Button-content">
                      <span class="Button-label">
                       Cancel
                      </span>
                     </span>
                    </button>
                   </div>
                  </input>
                 </form>
                </div>
               </div>
              </details-dialog>
             </details>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="notifications-button" id="tooltip-3ee7096d-5e30-4bc8-80d1-9f597deb9d03" popover="manual">
              Notification settings
             </tool-tip>
            </notifications-list-subscription-form>
            <div class="BtnGroup d-flex" data-view-component="true">
             <a aria-labelledby="tooltip-0144206f-5f8c-47d6-8fd4-4480e1312bda" class="Button Button--iconOnly Button--secondary Button--medium" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":379816251,"originating_url":"https://github.com/QuentinRougemont/coho_ldscp_genomics","user_id":126681664}}' data-hydro-click-hmac="84664cc8111f2a8115f5454cc30fadb6df4d2e1c56cb75a8797953bf0321ed4c" data-view-component="true" group_item="true" href="/QuentinRougemont/coho_ldscp_genomics/fork" id="fork-icon-button" variant="small">
              <svg aria-hidden="true" class="octicon octicon-repo-forked Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
               </path>
              </svg>
             </a>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="fork-icon-button" id="tooltip-0144206f-5f8c-47d6-8fd4-4480e1312bda" popover="manual">
              Fork your own copy of QuentinRougemont/coho_ldscp_genomics
             </tool-tip>
            </div>
            <div class="js-toggler-container starring-container" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/QuentinRougemont/coho_ldscp_genomics/unstar" class="starred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="zKkA6Bwr8OHyN8cbG5Al8tmUUCi_6WQMs6yq26fry0Ec94WLAdLaGT9_R1_odMLYPERspkqjeyPcYHWl3yoERA">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-cf848fd3-6b0d-4cdc-8670-721aa3004416" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target starred-button-icon" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":379816251,"originating_url":"https://github.com/QuentinRougemont/coho_ldscp_genomics","user_id":126681664}}' data-hydro-click-hmac="dd12f1758eadab5d2887b21c40eca2b8a70ec4f05aeb3ba184369bf032a01b98" data-view-component="true" id="icon-button-409f0e6d-ccbb-4e08-b957-7cea12b9222f" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-409f0e6d-ccbb-4e08-b957-7cea12b9222f" id="tooltip-cf848fd3-6b0d-4cdc-8670-721aa3004416" popover="manual">
                Unstar this repository
               </tool-tip>
              </input>
             </form>
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/QuentinRougemont/coho_ldscp_genomics/star" class="unstarred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="gsMNYOUg5avLem-D5sB94CQqKZnQcVZddxTbcP5FblrTh7xNtqEN-8IGMM6UZN-MPi5AUE9W3FVWPBhBEh81MQ">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-3d11cdbf-30c7-4771-9592-facd49f818b3" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":379816251,"originating_url":"https://github.com/QuentinRougemont/coho_ldscp_genomics","user_id":126681664}}' data-hydro-click-hmac="ff48562ed73404d4617e44f742ec36e053981337d82081da302be4343b9d9a70" data-view-component="true" id="icon-button-4a5e2e17-6f96-4335-b3f6-27aad1412cad" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-4a5e2e17-6f96-4335-b3f6-27aad1412cad" id="tooltip-3d11cdbf-30c7-4771-9592-facd49f818b3" popover="manual">
                Star this repository
               </tool-tip>
              </input>
             </form>
            </div>
           </div>
           <div class="d-flex flex-row gap-2">
           </div>
          </div>
          <p class="f4 mb-3 color-fg-muted">
           scripts to reproduce our landscape genomics results in coho salmon
          </p>
          <div class="mb-3">
           <ul aria-label="Repository details" class="d-flex flex-wrap mb-2 gap-2">
            <a class="Link--secondary no-underline d-block mr-2" href="/QuentinRougemont/coho_ldscp_genomics/stargazers" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             stars
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/QuentinRougemont/coho_ldscp_genomics/forks" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              1
             </span>
             fork
            </a>
            <div class="color-fg-muted mr-2" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-eye mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              3
             </span>
             watching
            </div>
            <a class="Link--secondary no-underline d-block mr-2" href="/QuentinRougemont/coho_ldscp_genomics/branches" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
              </path>
             </svg>
             <include-fragment src="/QuentinRougemont/coho_ldscp_genomics/branch-count">
              <span>
               Branches
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/QuentinRougemont/coho_ldscp_genomics/tags" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
              </path>
             </svg>
             <include-fragment src="/QuentinRougemont/coho_ldscp_genomics/tag-count">
              <span>
               Tags
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/QuentinRougemont/coho_ldscp_genomics/activity" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
              </path>
             </svg>
             <span>
              Activity
             </span>
            </a>
           </ul>
           <div class="mb-2 d-flex color-fg-muted">
            <div class="d-flex flex-items-center" style="height: 21px">
             <svg aria-hidden="true" class="octicon octicon-globe flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
              </path>
             </svg>
            </div>
            <span class="flex-auto min-width-0 width-fit">
             Public repository
            </span>
           </div>
          </div>
         </div>
        </div>
        <div class="border-bottom mx-xl-5">
        </div>
       </div>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <a class="d-none js-github-dev-shortcut" data-hotkey=".,Mod+Alt+." href="https://github.dev/">
         Open in github.dev
        </a>
        <a class="d-none js-github-dev-new-tab-shortcut" data-hotkey="Shift+.,Shift+&gt;,&gt;" href="https://github.dev/" rel="noopener noreferrer" target="_blank">
         Open in a new github.dev tab
        </a>
        <a class="d-none" data-hotkey=",,Mod+Alt+," href="/codespaces/new/QuentinRougemont/coho_ldscp_genomics?resume=1" target="_blank">
         Open in codespace
        </a>
        <h1 class="sr-only">
         QuentinRougemont/coho_ldscp_genomics
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/QuentinRougemont/coho_ldscp_genomics/spoofed_commit_check/a5efe93ab7689abe31583a03126d5f777b512a36">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-e8df95-53b8f38a33de.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_Dialog_js-node_modules_primer_react_lib-esm_-114c24-1306e57fe74e.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-d06085a838f3.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":true,"path":"/","repo":{"id":379816251,"defaultBranch":"main","name":"coho_ldscp_genomics","ownerLogin":"QuentinRougemont","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2021-06-24T13:43:02.000+08:00","ownerAvatar":"https://avatars.githubusercontent.com/u/17588888?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":{"id":126681664,"login":"ninenia","userEmail":"2945518981@qq.com"},"refInfo":{"name":"main","listCacheKey":"v0:1624513382.9701602","canEdit":true,"refType":"branch","currentOid":"a5efe93ab7689abe31583a03126d5f777b512a36"},"tree":{"items":[{"name":"00-scripts","path":"00-scripts","contentType":"directory"},{"name":"01-info","path":"01-info","contentType":"directory"},{"name":"02-data","path":"02-data","contentType":"directory"},{"name":"pictures","path":"pictures","contentType":"directory"},{"name":".gitignore","path":".gitignore","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":6,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/QuentinRougemont/coho_ldscp_genomics/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":true,"httpUrl":"https://github.com/QuentinRougemont/coho_ldscp_genomics.git","showCloneWarning":false,"sshUrl":"git@github.com:QuentinRougemont/coho_ldscp_genomics.git","sshCertificatesRequired":false,"sshCertificatesAvailable":false,"ghCliUrl":"gh repo clone QuentinRougemont/coho_ldscp_genomics","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"x-github-client://openRepo/https://github.com/QuentinRougemont/coho_ldscp_genomics","showVisualStudioCloneButton":true,"visualStudioCloneUrl":"git-client://clone?repo=https%3A%2F%2Fgithub.com%2FQuentinRougemont%2Fcoho_ldscp_genomics","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/QuentinRougemont/coho_ldscp_genomics/archive/refs/heads/main.zip"}},"repoPolicyInfo":{"allowed":true,"canBill":true,"changesWouldBeSafe":true,"disabledByBusiness":false,"disabledByOrganization":false,"hasIpAllowLists":false},"currentUserIsEnterpriseManaged":false,"enterpriseManagedBusinessName":null,"codespacesEnabled":true,"hasAccessToCodespaces":true},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"58","overviewFiles":[{"displayName":"README.md","repoName":"coho_ldscp_genomics","refName":"main","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003elandscape_genomics_coho\u003c/h1\u003e\u003ca id=\"user-content-landscape_genomics_coho\" class=\"anchor\" aria-label=\"Permalink: landscape_genomics_coho\" href=\"#landscape_genomics_coho\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003escripts to reproduce our landscape genomics results in coho salmon\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSee paper: Long distance migration is a major factor driving local adaptation at continental scale in a Pacific Salmon by Rougemont et al. available on \u003ca href=\"https://www.biorxiv.org/content/10.1101/2021.11.22.469546v1\" rel=\"nofollow\"\u003eBiorXiv\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eall necessary steps are highlighted below:\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDependancies:\u003c/h1\u003e\u003ca id=\"user-content-dependancies\" class=\"anchor\" aria-label=\"Permalink: Dependancies:\" href=\"#dependancies\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAlignment and SNP calling:\n\u003cstrong\u003ebwa\u003c/strong\u003e software avialble \u003ca href=\"https://sourceforge.net/projects/bio-bwa/files/\" rel=\"nofollow\"\u003ehere\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003efastp\u003c/strong\u003e software availalbe \u003ca href=\"https://github.com/OpenGene/fastp\"\u003ehere\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSamtools\u003c/strong\u003e software available \u003ca href=\"http://www.htslib.org/\" rel=\"nofollow\"\u003ehere\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003e\u003ca href=\"http://www.htslib.org/download/\" rel=\"nofollow\"\u003ehtslib\u003c/a\u003e\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003ecutadapt\u003c/strong\u003e \u003ca href=\"https://github.com/marcelm/cutadapt\"\u003ecutadapt\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003estacks\u003c/strong\u003e available \u003ca href=\"http://catchenlab.life.illinois.edu/stacks/\" rel=\"nofollow\"\u003ehere\u003c/a\u003e (see details of installation on the website)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003efor filtration:\n\u003cstrong\u003ebcftools\u003c/strong\u003e software available \u003ca href=\"http://www.htslib.org/download/\" rel=\"nofollow\"\u003ehere\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003evcftools\u003c/strong\u003e available \u003ca href=\"http://vcftools.sourceforge.net/\" rel=\"nofollow\"\u003ehere\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eANGSD, R \u0026amp; python\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eI like to use the following fonction in my \u003cstrong\u003e.bashrc :\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"function strata () { cat \u0026lt;(grep \u0026quot;CHR\u0026quot; \u0026quot;$1\u0026quot; ) |\\\n    cut -f 10- |\\                                  \n    perl -pe 's/\\t/\\n/g' | \\\n              sed 's/_/\\t/g' | awk '{print $1\u0026quot;\\t\u0026quot;$1\u0026quot;_\u0026quot;$2}' \u0026gt; \u0026quot;$2\u0026quot;; }\"\u003e\u003cpre\u003e\u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e \u003cspan class=\"pl-en\"\u003estrata\u003c/span\u003e () { cat \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\u0026lt;(\u003c/span\u003egrep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCHR\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e)\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e\\\n    cut -f 10- \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\ \u003c/span\u003e                                 \n    perl -pe \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/\\t/\\n/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e \\\n              sed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/_/\\t/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e awk \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e{print $1\"\\t\"$1\"_\"$2}\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$2\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e }\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDATA:\u003c/h1\u003e\u003ca id=\"user-content-data\" class=\"anchor\" aria-label=\"Permalink: DATA:\" href=\"#data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003epart of the RAW data are deposited on NCBI \u003ca href=\"https://www.ncbi.nlm.nih.gov/bioproject/PRJNA647050\" rel=\"nofollow\"\u003ePRJNA647050\u003c/a\u003e. The remaining samples should be available soon\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRAW vcf\u003c/strong\u003e and \u003cstrong\u003efiltered vcf\u003c/strong\u003e have been depostied on dryad (link to come soon)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe \u003cstrong\u003eallele frequency file\u003c/strong\u003e used in the RDA will be available on dryad\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eOther intermediary file (e.g. LFMM or hierfstat input) can be easily constructed with commands below\u003c/p\u003e\n\u003cp dir=\"auto\"\u003emetadata are available for GEA in the folder \u003ccode\u003e02-data/env\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eadditional metadata (individual name, river abbreviation, Latitude, Longitude, region of sampling, etc) can be fond in \u003ccode\u003e01-info\u003c/code\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eSteps :\u003c/h1\u003e\u003ca id=\"user-content-steps-\" class=\"anchor\" aria-label=\"Permalink: Steps :\" href=\"#steps-\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e1. Aligned Read and Call SNPs\u003c/h2\u003e\u003ca id=\"user-content-1-aligned-read-and-call-snps\" class=\"anchor\" aria-label=\"Permalink: 1. Aligned Read and Call SNPs\" href=\"#1-aligned-read-and-call-snps\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eUse \u003ca href=\"https://github.com/QuentinRougemont/stacks_v2_workflow\"\u003estacks_workflow\u003c/a\u003e to perform the following:\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003edemultiplexing and trimming\u003c/li\u003e\n\u003cli\u003ealign read,\u003c/li\u003e\n\u003cli\u003erun stacks\u003c/li\u003e\n\u003cli\u003e(filter vcf)\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003enote: I don't update stacks workflow any more and recommand to use the one of Eric Normandeau available \u003ca href=\"https://github.com/enormandeau/stacks_workflow\"\u003ehere\u003c/a\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eUse ANSGD to produce likelihood file (used to compute Fst and PBS latter)\u003c/li\u003e\n\u003c/ul\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2. Filter vcf and perform quality checks\u003c/h2\u003e\u003ca id=\"user-content-2-filter-vcf-and-perform-quality-checks\" class=\"anchor\" aria-label=\"Permalink: 2. Filter vcf and perform quality checks\" href=\"#2-filter-vcf-and-perform-quality-checks\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo reproduce the filtering you can access the raw vcf from dryad\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.1. Filtering vcf\u003c/h3\u003e\u003ca id=\"user-content-21-filtering-vcf\" class=\"anchor\" aria-label=\"Permalink: 2.1. Filtering vcf\" href=\"#21-filtering-vcf\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003euse mostly vcftools and bash to filter vcf.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n############################# Filtering RAW VCF ###########################\n\n# in stacks populations was run requiring:\n# SNP Genotyped in : \u0026gt;70% of the samples\n#                    \u0026gt;70% of the population\n\n\nvcftools \\\n      --gzvcf populations.snps.vcf.gz\n        --remove blacklisted_individuals.txt\n        --recode-INFO-all\n        --mac 1\n        --max-meanDP 120\n        --min-meanDP 10\n        --out populations.DP10_120\n        --recode\n\n\nvcftools \\\n        --vcf populations.10_120.recode.vcf\n        --recode-INFO-all\n        --max-missing 0.85\n        --out populations.10_120.miss85\n        --recode  \n\n# compute allele frequency and keep one SNP per loci:  \n\ninput=\u0026quot;populations.10_120.miss85.recode.vcf\u0026quot;  \n\nvcftools \\\n        --vcf $input\\\n        --freq --out out  \n\nsed 1d out.frq | 's/:/\\t/g' |cut -f 1,2,7,8 out.frq \u0026gt; frq2  \n\n#then I use a horrible bash script to create a whitelist of independant SNPs. \n#A script in Eric Normandeau's [workflow](https://github.com/enormandeau/stacks_workflow/blob/master/00-scripts/utility_scripts/extract_snp_with_max_maf.py) should do the same more efficiently.\n\n#compute missing rate per individuals  \nvcftools --vcf $input --missing-indv --out individuals\n\n#we remove individuals with too many missing data with awk\nawk '$5\u0026gt;0.20 {print $1}'  individuals.imiss \u0026gt; blacklisted.individuals.txt\n\n#remove individuals with too much missing data and keep one SNP per locus:  \nvcftools \\\n     --vcf $input\\\n        --remove blacklisted.individuals.txt\\\n        --recode-INFO-all\\\n\t--mac 1 \\\n        --out populations.mac1\\\n        --recode\\\n        --snps snp_id.txt\\\n\n#then perform a last quality filter:\nvcftools \\\n    --vcf populations.mac1.recode.vcf \\\n    --max-missing 0.95 \\\n    --out populations.mac1miss0.95\\\n    --min-meanDP 10 --max-meanDP 120\\\n    --recode \\\n    --recode-INFO-all\n\n#We don't want to filter on HWE since we may be interested in finding markers under selection \n#Moreover a global HWE filter is meaningless in highly structured populations\n\nstrata populations.mac1miss0.95.recode.vcf\nbgzip populations.mac1miss0.95.recode.vcf; tabix -p vcf populations.mac1miss0.95.recode.vcf.gz\n\n#then we need to rename part of the population HOP into HOD:\ncut -f 1 strata.txt |sed '2576,2607s/HOP/HOD/g' \u0026gt; new.ind.tmp\ncut -f 1 strata.txt \u0026gt; ind.tmp\n\npaste ind.tmp new.ind.tmp \u0026gt; new_sample_name.txt\nrm *tmp \n\ninput=populations.mac1miss0.95.recode.vcf.gz\n\n#then we rename individuals in the vcf \nbcftools reheader --samples new_sample_name.txt -o ${input%.recode.vcf.gz}.renamed.vcf.gz $input\n\n#this is our final vcf ready for estimating genetic diversity and structure \n#see below for filtering for GEA\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e############################ Filtering RAW VCF ###########################\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e in stacks populations was run requiring:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e SNP Genotyped in : \u0026gt;70% of the samples\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e                    \u0026gt;70% of the population\u003c/span\u003e\n\n\nvcftools \\\n      --gzvcf populations.snps.vcf.gz\n        --remove blacklisted_individuals.txt\n        --recode-INFO-all\n        --mac 1\n        --max-meanDP 120\n        --min-meanDP 10\n        --out populations.DP10_120\n        --recode\n\n\nvcftools \\\n        --vcf populations.10_120.recode.vcf\n        --recode-INFO-all\n        --max-missing 0.85\n        --out populations.10_120.miss85\n        --recode  \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e compute allele frequency and keep one SNP per loci:  \u003c/span\u003e\n\ninput=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epopulations.10_120.miss85.recode.vcf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e  \n\nvcftools \\\n        --vcf \u003cspan class=\"pl-smi\"\u003e$input\u003c/span\u003e\\\n        --freq --out out  \n\nsed 1d out.frq \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/:/\\t/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003ecut -f 1,2,7,8 out.frq \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e frq2  \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ethen I use a horrible bash script to create a whitelist of independant SNPs. \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eA script in Eric Normandeau's [workflow](https://github.com/enormandeau/stacks_workflow/blob/master/00-scripts/utility_scripts/extract_snp_with_max_maf.py) should do the same more efficiently.\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ecompute missing rate per individuals  \u003c/span\u003e\nvcftools --vcf \u003cspan class=\"pl-smi\"\u003e$input\u003c/span\u003e --missing-indv --out individuals\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ewe remove individuals with too many missing data with awk\u003c/span\u003e\nawk \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e$5\u0026gt;0.20 {print $1}\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e  individuals.imiss \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e blacklisted.individuals.txt\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eremove individuals with too much missing data and keep one SNP per locus:  \u003c/span\u003e\nvcftools \\\n     --vcf \u003cspan class=\"pl-smi\"\u003e$input\u003c/span\u003e\\\n        --remove blacklisted.individuals.txt\\\n        --recode-INFO-all\\\n\t--mac 1 \\\n        --out populations.mac1\\\n        --recode\\\n        --snps snp_id.txt\\\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ethen perform a last quality filter:\u003c/span\u003e\nvcftools \\\n    --vcf populations.mac1.recode.vcf \\\n    --max-missing 0.95 \\\n    --out populations.mac1miss0.95\\\n    --min-meanDP 10 --max-meanDP 120\\\n    --recode \\\n    --recode-INFO-all\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eWe don't want to filter on HWE since we may be interested in finding markers under selection \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eMoreover a global HWE filter is meaningless in highly structured populations\u003c/span\u003e\n\nstrata populations.mac1miss0.95.recode.vcf\nbgzip populations.mac1miss0.95.recode.vcf\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e tabix -p vcf populations.mac1miss0.95.recode.vcf.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ethen we need to rename part of the population HOP into HOD:\u003c/span\u003e\ncut -f 1 strata.txt \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003esed \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e2576,2607s/HOP/HOD/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e new.ind.tmp\ncut -f 1 strata.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e ind.tmp\n\npaste ind.tmp new.ind.tmp \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e new_sample_name.txt\nrm \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003etmp \n\ninput=populations.mac1miss0.95.recode.vcf.gz\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ethen we rename individuals in the vcf \u003c/span\u003e\nbcftools reheader --samples new_sample_name.txt -o \u003cspan class=\"pl-smi\"\u003e${input\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.recode.vcf.gz}\u003c/span\u003e.renamed.vcf.gz \u003cspan class=\"pl-smi\"\u003e$input\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ethis is our final vcf ready for estimating genetic diversity and structure \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003esee below for filtering for GEA\u003c/span\u003e\n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e2.2. Quality check\u003c/h3\u003e\u003ca id=\"user-content-22-quality-check\" class=\"anchor\" aria-label=\"Permalink: 2.2. Quality check\" href=\"#22-quality-check\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eworking on the same cleaned file (populations.mac1miss0.95.renamed.vcf)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"* compute depth of sequencing, missing rate, genotyping rate  \n\nuse:\n\t* vcftools site-mean-depth \n\t* vcftools missing-indv  \n\n\n* verifiy the absence of a pattern in missing data :\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e* compute depth of sequencing, missing rate, genotyping rate  \n\nuse:\n\t* vcftools site-mean-depth \n\t* vcftools missing-indv  \n\n\n* verifiy the absence of a pattern in missing data :\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"plink --file populations.mac1miss0.95.renamed.vcf \\\n    --allow-extra-chr \\\n    --mds-plot 4  --cluster missing  \\\n    --out ibm_mds_to_plot\"\u003e\u003cpre\u003eplink --file populations.mac1miss0.95.renamed.vcf \\\n    --allow-extra-chr \\\n    --mds-plot 4  --cluster missing  \\\n    --out ibm_mds_to_plot\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ethen plot it using:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"Rscript ./00-scripts/quality_check/plot_mds.R\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eRscript\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e00\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003escripts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003equality_check\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eplot_mds.R\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ewe obtained the following plot:\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/ibm_mds_point.git.png\"\u003e\u003cimg src=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/ibm_mds_point.git.png\" alt=\"example_graph\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ewe see that missing data are low and that these missing data do not show a particular pattern of structure.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e3. Compute genetic diversity and plot it\u003c/h2\u003e\u003ca id=\"user-content-3-compute-genetic-diversity-and-plot-it\" class=\"anchor\" aria-label=\"Permalink: 3. Compute genetic diversity and plot it\" href=\"#3-compute-genetic-diversity-and-plot-it\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003efirst convert vcf into hierfstat and other usefull input\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"vcf=$1 #vcf here\n\n#check compression\nif file --mime-type \u0026quot;$vcf\u0026quot; | grep -q gzip$; then\n   echo \u0026quot;$vcf is gzipped\u0026quot;\n   gunzip \u0026quot;$vcf\u0026quot;\n   INPUT=${vcf%}.gz\nelse\n   echo \u0026quot;$vcf is not gzipped\u0026quot;\n   INPUT=$vcf\nfi\n\n#extract population and individuals\nstrata $INPUT strata.txt ; cut -f 1 strata.txt \u0026gt; pop.tmp\n\nvcf2geno $INPUT\ngeno2lfmm ${INPUT%.vcf}.geno\nsnp=${INPUT%.vcf}.vcfsnp\nINPUT=${INPUT%.vcf}.geno\n\nsed -e 's/2/22/g' $INPUT |\\\n    sed -e 's/9/NA/g' -e 's/1/12/g' -e 's/0/11/g' \u0026gt; out.tmp\npaste pop out.tmp \u0026gt; hierfstat.data.tmp\necho \u0026quot;loc\u0026quot; \u0026gt; loc.tmp\n\ncut -d \u0026quot; \u0026quot; -f 3 $snp |perl -pe \u0026quot;s/\\n/\\t/g\u0026quot; \u0026gt; snp_id.tmp\npaste loc.tmp snp_id.tmp \u0026gt; loc.tmp\ncat loc.tmp hierfstat.data.tmp \u0026gt; hierfstat.data.txt\nrm \\*tmp\n\"\u003e\u003cpre\u003evcf=\u003cspan class=\"pl-smi\"\u003e$1\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003evcf here\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003echeck compression\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e file --mime-type \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$vcf\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e grep -q gzip$\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e \u003cspan class=\"pl-k\"\u003ethen\u003c/span\u003e\n   \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$vcf\u003c/span\u003e is gzipped\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n   gunzip \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$vcf\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n   INPUT=\u003cspan class=\"pl-smi\"\u003e${vcf\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e}\u003c/span\u003e.gz\n\u003cspan class=\"pl-k\"\u003eelse\u003c/span\u003e\n   \u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e$vcf\u003c/span\u003e is not gzipped\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n   INPUT=\u003cspan class=\"pl-smi\"\u003e$vcf\u003c/span\u003e\n\u003cspan class=\"pl-k\"\u003efi\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eextract population and individuals\u003c/span\u003e\nstrata \u003cspan class=\"pl-smi\"\u003e$INPUT\u003c/span\u003e strata.txt \u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e cut -f 1 strata.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e pop.tmp\n\nvcf2geno \u003cspan class=\"pl-smi\"\u003e$INPUT\u003c/span\u003e\ngeno2lfmm \u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.vcf}\u003c/span\u003e.geno\nsnp=\u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.vcf}\u003c/span\u003e.vcfsnp\nINPUT=\u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.vcf}\u003c/span\u003e.geno\n\nsed -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/2/22/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e$INPUT\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003e\\\n    sed -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/9/NA/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/1/12/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e -e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/0/11/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e out.tmp\npaste pop out.tmp \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e hierfstat.data.tmp\n\u003cspan class=\"pl-c1\"\u003eecho\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eloc\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e loc.tmp\n\ncut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f 3 \u003cspan class=\"pl-smi\"\u003e$snp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e|\u003c/span\u003eperl -pe \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003es/\\n/\\t/g\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e snp_id.tmp\npaste loc.tmp snp_id.tmp \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e loc.tmp\ncat loc.tmp hierfstat.data.tmp \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e hierfstat.data.txt\nrm \u003cspan class=\"pl-cce\"\u003e\\*\u003c/span\u003etmp\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ethen use R to compute basic statistics (Hs, Ho, Fis, Bst, in hierfstat)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ethe scripts:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"00-scripts/diversity/00.vcf2hierfstats.sh\n00-scripts/diversity/01.hierfstats.R\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e00-scripts/diversity/00.vcf2hierfstats.sh\n00-scripts/diversity/01.hierfstats.R\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eshould procude all the results.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe second script run on a cluster with \u0026gt;50Gb of RAM  :\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"Rscript ./00-scripts/diversity/01.hierfstats.R\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eRscript\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e00\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003escripts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ediversity\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e01.\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ehierfstats.R\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThen we will perform plots of the correlation between the distance to the southernmost site and Bst and Hs statistics.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003ePlotting diversity\u003c/h3\u003e\u003ca id=\"user-content-plotting-diversity\" class=\"anchor\" aria-label=\"Permalink: Plotting diversity\" href=\"#plotting-diversity\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRun:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"Rscript ./00-scripts/diversity/02.plot_ho_betaST_distance.R\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eRscript\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e00\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003escripts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ediversity\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e02.\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eplot_ho_betaST_distance.R\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eon the command line or alternatively in Rstudio,\u003cbr\u003e\nthis will test for correlation among diversity and distance and produce the following graph:\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/figureS07.git.png\"\u003e\u003cimg src=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/figureS07.git.png\" alt=\"example_graph\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4. Perform PCA and VAE analyses\u003c/h2\u003e\u003ca id=\"user-content-4-perform-pca-and-vae-analyses\" class=\"anchor\" aria-label=\"Permalink: 4. Perform PCA and VAE analyses\" href=\"#4-perform-pca-and-vae-analyses\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4a. PCA\u003c/h3\u003e\u003ca id=\"user-content-4a-pca\" class=\"anchor\" aria-label=\"Permalink: 4a. PCA\" href=\"#4a-pca\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePCA was performed on individuals using \u003ca href=\"https://github.com/QuentinRougemont/utility_scripts/blob/master/02.PCA/pca_on_vcf.R\"\u003ethis script\u003c/a\u003e and populations allele frequency using \u003ca href=\"https://github.com/QuentinRougemont/utility_scripts/blob/master/02.PCA/pca_on_freq.R\"\u003ethis script\u003c/a\u003e.\nI also made an IBS plot using plink with this \u003ca href=\"https://github.com/QuentinRougemont/utility_scripts/blob/master/07.random_scripts/plink_cluster_IBS.sh\"\u003esimple script\u003c/a\u003e.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTo plot and fit to the Coho salmon data here's a simple modification of the script.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\nfile=\u0026quot;plink.frq.strat.renamed.gz\u0026quot;\n\n#load libs\nlibs \u0026lt;- c('dplyr','reshape2','ade4','data.table', 'factoextra', 'magrittr','ggsci')\ninvisible(lapply(libs, library, character.only = TRUE))\n\nfile \u0026lt;- paste0(\u0026quot;zcat \u0026quot;,file) \nfreq \u0026lt;- fread(file)\nfreq2 \u0026lt;- dplyr::select(freq,SNP,CLST,MAF) \nfreq3 \u0026lt;- reshape2::dcast(freq2,SNP~CLST)\nfreq3 \u0026lt;- freq3[,-1] \nfreq4 \u0026lt;- t(freq3)\npop \u0026lt;- unique(freq$CLST)   \npop \u0026lt;- data.frame(pop) %\u0026gt;% set_colnames(.,c(\u0026quot;POP\u0026quot;))\n\n\n#perform PCA\npca1 \u0026lt;- dudi.pca(freq4,scale=FALSE,scannf=FALSE)\n\n#load strata\nstrata \u0026lt;- read.table(\u0026quot;pop_ind_region_latitute_corrected3.txt\u0026quot;) \nstrata \u0026lt;- select(strata, V2,V6) %\u0026gt;% \n    set_colnames(.,c(\u0026quot;POP\u0026quot;,\u0026quot;REGION\u0026quot;))\nstrata \u0026lt;- unique(strata)\nstrata \u0026lt;- strata[order(strata$POP),]  #reorder because population DRA has been renamed into WAC\n\n#ensure that the order in strata will match order of the freq file:\nstrata = left_join(pop, strata)\n\n#personalizing the colors:\nmyColors \u0026lt;- c(\u0026quot;blue\u0026quot;,\u0026quot;orange\u0026quot;,\u0026quot;red\u0026quot;,\u0026quot;darkviolet\u0026quot;,\u0026quot;chocolate4\u0026quot;, \u0026quot;springgreen4\u0026quot;,\u0026quot;green\u0026quot;)\nnames(myColors) \u0026lt;- levels(strata$REGION)\ncolScale \u0026lt;- scale_colour_manual(name = \u0026quot;REGION\u0026quot;,values = myColors)\n\n#plot with river label:\np \u0026lt;- fviz_pca_ind(pca1, label=\u0026quot;none\u0026quot;, pointsize = 0.0) +\n    geom_text(aes(label=strata$POP, \n        colour=factor(strata$REGION)),\n        size = 2 )\n#p \u0026lt;- p + theme_minimal() + theme(legend.position = \u0026quot;none\u0026quot;)  + colScale\n#p \u0026lt;- p + scale_color_igv() + theme_minimal() + theme(legend.position = \u0026quot;none\u0026quot;)\np \u0026lt;- p +theme(legend.text = element_text(size = 10, face = \u0026quot;bold\u0026quot;)) +\n    theme(legend.justification=c(1,0), legend.position=c(1,0)) + \n    colScale + \n    ggtitle(\u0026quot;PCA on allele frequencies\u0026quot;) + \n     theme(plot.title = element_text(lineheight=.8, face=\u0026quot;bold\u0026quot; , hjust = 0.5))\n\n    \npdf(file=\u0026quot;pca_on_freq_population.pdf\u0026quot;)\np\ndev.off()\n\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eplink.frq.strat.renamed.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eload libs\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003elibs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003edplyr\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ereshape2\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eade4\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003edata.table\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003efactoextra\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emagrittr\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eggsci\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003einvisible\u003c/span\u003e(lapply(\u003cspan class=\"pl-smi\"\u003elibs\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003elibrary\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003echaracter.only\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e paste0(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ezcat \u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e) \n\u003cspan class=\"pl-smi\"\u003efreq\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-smi\"\u003efile\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003efreq2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003efreq\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSNP\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eCLST\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eMAF\u003c/span\u003e) \n\u003cspan class=\"pl-smi\"\u003efreq3\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003ereshape2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003edcast(\u003cspan class=\"pl-smi\"\u003efreq2\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eSNP\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCLST\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003efreq3\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efreq3\u003c/span\u003e[,\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e] \n\u003cspan class=\"pl-smi\"\u003efreq4\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e t(\u003cspan class=\"pl-smi\"\u003efreq3\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003epop\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e unique(\u003cspan class=\"pl-smi\"\u003efreq\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCLST\u003c/span\u003e)   \n\u003cspan class=\"pl-smi\"\u003epop\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003epop\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% set_colnames(\u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e,c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePOP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eperform PCA\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003epca1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e dudi.pca(\u003cspan class=\"pl-smi\"\u003efreq4\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003escannf\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eload strata\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003estrata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epop_ind_region_latitute_corrected3.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \n\u003cspan class=\"pl-smi\"\u003estrata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e select(\u003cspan class=\"pl-smi\"\u003estrata\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eV6\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n    set_colnames(\u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e,c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePOP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eREGION\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003estrata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e unique(\u003cspan class=\"pl-smi\"\u003estrata\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003estrata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003estrata\u003c/span\u003e[order(\u003cspan class=\"pl-smi\"\u003estrata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePOP\u003c/span\u003e),]  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ereorder because population DRA has been renamed into WAC\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eensure that the order in strata will match order of the freq file:\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003estrata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e left_join(\u003cspan class=\"pl-smi\"\u003epop\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003estrata\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003epersonalizing the colors:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emyColors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorange\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkviolet\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003echocolate4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003espringgreen4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egreen\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nnames(\u003cspan class=\"pl-smi\"\u003emyColors\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e levels(\u003cspan class=\"pl-smi\"\u003estrata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eREGION\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ecolScale\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e scale_colour_manual(\u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eREGION\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003evalues\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emyColors\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eplot with river label:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fviz_pca_ind(\u003cspan class=\"pl-smi\"\u003epca1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003epointsize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.0\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    geom_text(aes(\u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003estrata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePOP\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-k\"\u003efactor\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003estrata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eREGION\u003c/span\u003e)),\n        \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e )\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ep \u0026lt;- p + theme_minimal() + theme(legend.position = \"none\")  + colScale\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ep \u0026lt;- p + scale_color_igv() + theme_minimal() + theme(legend.position = \"none\")\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003etheme(\u003cspan class=\"pl-v\"\u003elegend.text\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eface\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebold\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    theme(\u003cspan class=\"pl-v\"\u003elegend.justification\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003elegend.position\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n    \u003cspan class=\"pl-smi\"\u003ecolScale\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n    ggtitle(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePCA on allele frequencies\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n     theme(\u003cspan class=\"pl-v\"\u003eplot.title\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003elineheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.8\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eface\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebold\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e , \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e))\n\n    \npdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epca_on_freq_population.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ep\u003c/span\u003e\ndev.off()\n\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eyou should obtained an image as follows\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/pca_on_freq_population.git.png\"\u003e\u003cimg src=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/pca_on_freq_population.git.png\" alt=\"example_graph\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ein addition it is possible to test for population structure using a PCA at the individual level or an IBS plot using a similar command to the one use for IBM:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"plink --file  populations.mac1miss0.95.renamed.vcf \\\n    --allow-extra-chr \\\n    --mds-plot 4  --cluster\\\n    --out mds_to_plot\"\u003e\u003cpre\u003eplink --file  populations.mac1miss0.95.renamed.vcf \\\n    --allow-extra-chr \\\n    --mds-plot 4  --cluster\\\n    --out mds_to_plot\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRun this very simple script (similar to the one used for IBM):\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"Rscript 00-scripts/pca_ibs/plot_mds.R\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eRscript\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e00\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003escripts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003epca_ibs\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eplot_mds.R\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ethen you'll obtain the following graph:\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/mds_point.git.png\"\u003e\u003cimg src=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/mds_point.git.png\" alt=\"example_graph\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis highlights particularly well the continuity of our data, where the axis of a PCA are highly correlated with latitude and longitude (see our paper and our previous one).\u003cbr\u003e\nInterestingly, we see that the Russian samples falls within Alaska and is not much divergent.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch3 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e4b. Running VAE:\u003c/h3\u003e\u003ca id=\"user-content-4b-running-vae\" class=\"anchor\" aria-label=\"Permalink: 4b. Running VAE:\" href=\"#4b-running-vae\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eInstall vae available \u003ca href=\"https://github.com/kr-colab/popvae\"\u003ehere\u003c/a\u003e.\nDue to its stochasticity, you'll obtained slightly different results than I did.\nFor instance depanding on the parameters used, you may obtained a distinction of nearly all river in separate cluster (Figure 2 of our paper)\nIn other case, you'll obtained a continuous distribution of the genotype entierely correlated with latitude.\nThese two results reflect the existence of fine scale structure (recent postglacial homing) and of strong IBD following the postglacial expansion\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\n#install VAE and activate\n#install tensorflow as well!\n\n#create a working directory:\nmkdir VAE\ncd VAE\n#copy the vcf \nln -s ../02-data/populations.mac1miss0.95.renamed.vcf.gz .\n# create folder to store the results\nmkdir results\n\n# run popvae with default parameters\npopvae.py --infile populations.mac1miss0.95.renamed.vcf.gz --out results/coho --seed 12345\n\n# more advanced use:\nmkdir results2\npopvae.py --infile  populations.mac1miss0.95.renamed.vcf.gz\\\n    --out results2/coho \\ \n    --seed 12345 \\\n    --search_network_sizes \\ \n    --train_prop 0.90\n\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003einstall VAE and activate\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003einstall tensorflow as well!\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ecreate a working directory:\u003c/span\u003e\nmkdir VAE\n\u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e VAE\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ecopy the vcf \u003c/span\u003e\nln -s ../02-data/populations.mac1miss0.95.renamed.vcf.gz \u003cspan class=\"pl-c1\"\u003e.\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e create folder to store the results\u003c/span\u003e\nmkdir results\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e run popvae with default parameters\u003c/span\u003e\npopvae.py --infile populations.mac1miss0.95.renamed.vcf.gz --out results/coho --seed 12345\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e more advanced use:\u003c/span\u003e\nmkdir results2\npopvae.py --infile  populations.mac1miss0.95.renamed.vcf.gz\\\n    --out results2/coho \u003cspan class=\"pl-cce\"\u003e\\ \u003c/span\u003e\n    --seed 12345 \\\n    --search_network_sizes \u003cspan class=\"pl-cce\"\u003e\\ \u003c/span\u003e\n    --train_prop 0.90\n\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eadvice\u003c/strong\u003e: try different run to see how results can differ\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThen plot the results in R. This bits of code should to the work:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(dplyr);library(cowplot);library(ggplot2);library(tidyr);library(ggsci)\n\npd \u0026lt;- read.table(\u0026quot;coho_latent_coords.txt\u0026quot;,T)\nnames(pd)[1:2] \u0026lt;- c(\u0026quot;LD1\u0026quot;,\u0026quot;LD2\u0026quot;)\nmeta \u0026lt;- read.table(\u0026quot;01-info/metadata\u0026quot;,T, sep=\u0026quot;\\t\u0026quot; ) %\u0026gt;% select(POP, Latitude, Longitude, Region)\npd1 \u0026lt;- separate(pd,sampleID,c(\u0026quot;POP\u0026quot;,\u0026quot;ind\u0026quot;),\u0026quot;_\u0026quot;)\npd \u0026lt;- merge(pd1, meta ) #.x=\u0026quot;POP\u0026quot;,by.y=\u0026quot;POP_ID\u0026quot;)\n\n############################################################################################### \n# test correlation  with geography:\nsummary(lm(pd$LD1 ~ pd$Latitude))\nsummary(lm(pd$LD2 ~ pd$Latitude))\nsummary(lm(pd$LD2 ~ pd$Longitude))\nsummary(lm(pd$LD1 ~ pd$Longitude))\nsummary(lm(pd$LD1 ~ pd$Latitude + pd$Longitude ))\nsummary(lm(pd$LD1 ~ pd$Latitude + pd$Longitude + pd$Latitude * pd$Longitude ))\n\n# set colors:\nmyColors \u0026lt;- c(\u0026quot;blue\u0026quot;,\u0026quot;orange\u0026quot;,\u0026quot;red\u0026quot;,\u0026quot;darkviolet\u0026quot;,\u0026quot;chocolate4\u0026quot;, \u0026quot;springgreen4\u0026quot;,\u0026quot;green\u0026quot;)\nnames(myColors) \u0026lt;- levels(pd$Region)\ncolScale \u0026lt;- scale_colour_manual(name = \u0026quot;Region\u0026quot;,values = myColors)\n\n# plot the 2 ld\np \u0026lt;- pd %\u0026gt;% \n  dplyr::rename_at(\u0026quot;POP\u0026quot;,~\u0026quot;river\u0026quot;) %\u0026gt;%\n  #separate(REGION2,c(\u0026quot;Rivers\u0026quot;,NA), sep =\u0026quot;-\u0026quot; ) #)\n  #ggplot(aes(x=LD1,y=LD2, col= river))+\n  ggplot(aes(x = LD1, y = LD2, col = Region))+\n  colScale +\n  geom_point(size=1) + theme_classic()\n\nhaida \u0026lt;- dplyr::filter(pd, Region==\u0026quot;HaidaGwaii\u0026quot;) %\u0026gt;% \n  dplyr::rename_at(\u0026quot;POP\u0026quot;,~\u0026quot;river\u0026quot;) %\u0026gt;%\n  ggplot(aes(x=LD1,y=LD2, col=river))+ #labs(title = \u0026quot;B\u0026quot;) +\n  geom_text(aes ( label= river), size=3.5) + theme_classic() +\n  theme(legend.position = \u0026quot;none\u0026quot;)  \nh \u0026lt;- haida + scale_color_igv()\n\nall \u0026lt;- ggdraw(p + theme_half_open(12)) +\n    draw_plot(h , .62, .62, .35, .35) +\n        draw_plot_label(\n         c(\u0026quot;A\u0026quot;, \u0026quot;B\u0026quot;),\n         c(0, 0.45),\n         c(1, 0.95),\n         size = 12)\n\npdf(file=\u0026quot;vae_LD1_LD2.pdf\u0026quot;)\nall \ndev.off()\n\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e);library(\u003cspan class=\"pl-smi\"\u003ecowplot\u003c/span\u003e);library(\u003cspan class=\"pl-smi\"\u003eggplot2\u003c/span\u003e);library(\u003cspan class=\"pl-smi\"\u003etidyr\u003c/span\u003e);library(\u003cspan class=\"pl-smi\"\u003eggsci\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecoho_latent_coords.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)\nnames(\u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e)[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLD1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLD2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003emeta\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e01-info/metadata\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e ) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% select(\u003cspan class=\"pl-smi\"\u003ePOP\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eLatitude\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eLongitude\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eRegion\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003epd1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e separate(\u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003esampleID\u003c/span\u003e,c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePOP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eind\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e merge(\u003cspan class=\"pl-smi\"\u003epd1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003emeta\u003c/span\u003e ) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e.x=\"POP\",by.y=\"POP_ID\")\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e############################################################################################## \u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e test correlation  with geography:\u003c/span\u003e\nsummary(lm(\u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLD1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLatitude\u003c/span\u003e))\nsummary(lm(\u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLD2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLatitude\u003c/span\u003e))\nsummary(lm(\u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLD2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLongitude\u003c/span\u003e))\nsummary(lm(\u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLD1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLongitude\u003c/span\u003e))\nsummary(lm(\u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLD1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLatitude\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLongitude\u003c/span\u003e ))\nsummary(lm(\u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLD1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLatitude\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLongitude\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLatitude\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLongitude\u003c/span\u003e ))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e set colors:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emyColors\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eorange\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkviolet\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003echocolate4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003espringgreen4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egreen\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\nnames(\u003cspan class=\"pl-smi\"\u003emyColors\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e levels(\u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ecolScale\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e scale_colour_manual(\u003cspan class=\"pl-v\"\u003ename\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eRegion\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003evalues\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emyColors\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e plot the 2 ld\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003erename_at(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePOP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eriver\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eseparate(REGION2,c(\"Rivers\",NA), sep =\"-\" ) #)\u003c/span\u003e\n  \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eggplot(aes(x=LD1,y=LD2, col= river))+\u003c/span\u003e\n  ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eLD1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eLD2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eRegion\u003c/span\u003e))\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \u003cspan class=\"pl-smi\"\u003ecolScale\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_point(\u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e theme_classic()\n\n\u003cspan class=\"pl-smi\"\u003ehaida\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003efilter(\u003cspan class=\"pl-smi\"\u003epd\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eRegion\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eHaidaGwaii\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n  \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003erename_at(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePOP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eriver\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLD1\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eLD2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eriver\u003c/span\u003e))\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003elabs(title = \"B\") +\u003c/span\u003e\n  geom_text(aes ( \u003cspan class=\"pl-v\"\u003elabel\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eriver\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e3.5\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e theme_classic() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  theme(\u003cspan class=\"pl-v\"\u003elegend.position\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enone\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)  \n\u003cspan class=\"pl-smi\"\u003eh\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ehaida\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e scale_color_igv()\n\n\u003cspan class=\"pl-smi\"\u003eall\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e ggdraw(\u003cspan class=\"pl-smi\"\u003ep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e theme_half_open(\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    draw_plot(\u003cspan class=\"pl-smi\"\u003eh\u003c/span\u003e , \u003cspan class=\"pl-smi\"\u003e.62\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003e.62\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003e.35\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003e.35\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n        draw_plot_label(\n         c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n         c(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.45\u003c/span\u003e),\n         c(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e0.95\u003c/span\u003e),\n         \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e)\n\npdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evae_LD1_LD2.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eall\u003c/span\u003e \ndev.off()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ehere's an example with many cluster:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/Fig2.git.png\"\u003e\u003cimg src=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/Fig2.git.png\" alt=\"example_graph\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ethis is the graph I obtained at first run of the VAE and which is in our manuscript. Almost all sampling localities have been separated in discrete cluster).\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePopulations further away in space are also further away on the graph.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn the panel B/C I zoomed in the location of HaidaGwaii because there are few samples there. We can see that we are able to finely discriminate all samples and that those very closed in space are very closed in the Graph.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe same results could be obtained by running many separate PCA on subset of data. Here only one run was necessary.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAt the same time there is strong IBD in our data.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ehere's an example with pure geography (IBD):\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/vae_LD1_LD2.png\"\u003e\u003cimg src=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/vae_LD1_LD2.png\" alt=\"example_graph\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eaccordingly we obtain a strong correlation with latitude:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/vae_LD1_Latitude.png\"\u003e\u003cimg src=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/vae_LD1_Latitude.png\" alt=\"example_graph\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5. GEA Analyses:\u003c/h2\u003e\u003ca id=\"user-content-5-gea-analyses\" class=\"anchor\" aria-label=\"Permalink: 5. GEA Analyses:\" href=\"#5-gea-analyses\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.1. Extract environmental data\u003c/h2\u003e\u003ca id=\"user-content-51-extract-environmental-data\" class=\"anchor\" aria-label=\"Permalink: 5.1. Extract environmental data\" href=\"#51-extract-environmental-data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe environmental data are located in the \u003ccode\u003e01-info\u003c/code\u003e folder with two text file:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003emetadata\u003c/li\u003e\n\u003cli\u003ebioclimatic_var\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eData included the following:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eDistance to the spawning site (km)\u003c/li\u003e\n\u003cli\u003eElevation (m)\u003c/li\u003e\n\u003cli\u003eWatershed areas\u003c/li\u003e\n\u003cli\u003eClimate data (precipitation and temperature from WorldClim)\u003c/li\u003e\n\u003cli\u003eGeological data (extracted from the USGS database)  (included in \u003ccode\u003e02-data/env/geology\u003c/code\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eClimate data include the mean, min, max range, sdt of 19 variables associated to precipitation and temperature.\nThese are obviously redundant and we summarized them and decorelate the variable using a PCA.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe following simple code available at \u003ccode\u003e00-scripts/gea/00.pca_of_env.R\u003c/code\u003e \u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"## load libs\nlibs \u0026lt;- c('dplyr','ade4','factoextra','vegan','cowplot')\ninvisible(lapply(libs, library, character.only = TRUE))\n\n###### DOWNLOAD ENV DATA\nmetadata \u0026lt;- read.table(\u0026quot;01-info/metadata\u0026quot;, h = T, sep = \u0026quot;\\t\u0026quot;)\n\nbioclim \u0026lt;- read.table(\u0026quot;01-info/bioclimatic_var\u0026quot;, h = T, sep = \u0026quot;\\t\u0026quot; )\n#remove BNV and SAI:\nbioclim \u0026lt;- bioclim %\u0026gt;%filter(POP !=\u0026quot;SAI\u0026quot; \u0026amp; POP !=\u0026quot;BNV\u0026quot;)\n\n####### PERFORM PCA ON ENVT VAR #################################################\n#will do a PCA on the dataaset that contains only climatic variables:\n#separate temperature and precipitation and keep only significant axis or only first axis\nX.temp \u0026lt;- dudi.pca(df = bioclim[, 2:56], center = T, scale = T, scannf = FALSE) #nf = 3)\nX.prec  \u0026lt;- dudi.pca(df = bioclim[, 57:96], center = T, scale = T, scannf = FALSE) # nf = 3)\n\n## singificance of axis\neig.val \u0026lt;- get_eigenvalue(X.temp) #first we get the eigen value in an easy form\neig.val$eig \u0026lt;- eig.val$variance.percent/100 #percent\nexpected \u0026lt;- bstick(length(eig.val$eig) )\nsignif \u0026lt;- eig.val$eig \u0026gt; expected #get signifcicant axis #3axes sont signif\nsignif\n#4 axis are significant\n\neig.val.p \u0026lt;- get_eigenvalue(X.prec) #first we get the eigen value in an easy form\neig.val.p$eig \u0026lt;- eig.val.p$variance.percent/100 #percent\nexpected \u0026lt;- bstick(length(eig.val.p$eig) )\nsignif.p \u0026lt;- eig.val.p$eig \u0026gt; expected #get signifcicant axis #3axes sont signif\nsignif.p\n#3 axis are signficant\n\n#visualisation of individuals cos^2:\nX.temp \u0026lt;- dudi.pca(df = bioclim[, 2:53], center = T, scale = T, scannf = FALSE, nf = 4)\nX.prec  \u0026lt;- dudi.pca(df = bioclim[, 54:95], center = T, scale = T, scannf = FALSE,  nf = 3)\n\nfviz_pca_ind(X.prec, col.ind=\u0026quot;cos2\u0026quot;, geom = \u0026quot;point\u0026quot;) +\n    scale_color_gradient2(low=\u0026quot;white\u0026quot;, mid=\u0026quot;blue\u0026quot;,\n    high=\u0026quot;red\u0026quot;, midpoint=0.6)+ theme_minimal()\n\np12 \u0026lt;- fviz_pca_var(X.prec, col.var=\u0026quot;steelblue\u0026quot;)+\n    theme_minimal()  +  ggtitle(\u0026quot;Precipitation PC axis 12\\nvariables\u0026quot;) + \n     theme(plot.title = element_text(lineheight=.8, face=\u0026quot;bold\u0026quot;, hjust = 0.5))\n\np23 \u0026lt;- fviz_pca_var(X.prec, axes = c(2,3) , col.var=\u0026quot;steelblue\u0026quot;)+\n    theme_minimal() +  ggtitle(\u0026quot;Precipitation PC axis 23\\nvariables\u0026quot;) + \n     theme(plot.title = element_text(lineheight=.8, face=\u0026quot;bold\u0026quot;, hjust = 0.5))\n\nt12 \u0026lt;- fviz_pca_var(X.temp, col.var=\u0026quot;steelblue\u0026quot;)+\n    theme_minimal() +  ggtitle(\u0026quot;Temperature PC axis 12\\nvariables\u0026quot;) + \n     theme(plot.title = element_text(lineheight=.8, face=\u0026quot;bold\u0026quot; , hjust = 0.5))\n\nt34 \u0026lt;- fviz_pca_var(X.temp, axes = c(3,4) , col.var=\u0026quot;steelblue\u0026quot;)+\n    theme_minimal() +  ggtitle(\u0026quot;Temperature PC axis 34\\nvariables\u0026quot;) + \n     theme(plot.title = element_text(lineheight=.8, face=\u0026quot;bold\u0026quot; , hjust = 0.5))\n\n\nt_eig  \u0026lt;- fviz_eig(X.temp, addlabels = TRUE, ylim = c(0, 50)) + \n    ggtitle(\u0026quot;Temperature eigenvalues\u0026quot;) + \n     theme(plot.title = element_text(lineheight=.8, face=\u0026quot;bold\u0026quot; , hjust = 0.5))\n\n\np_eig  \u0026lt;- fviz_eig(X.prec, addlabels = TRUE, ylim = c(0, 50)) + \n    ggtitle(\u0026quot;Precipitation eigenvalues\u0026quot;) + \n     theme(plot.title = element_text(lineheight=.8, face=\u0026quot;bold\u0026quot; , hjust = 0.5))\n\npdf(file = \u0026quot;temperature_and_precipiration.pca.pdf\u0026quot;, 18,12)\nplot_grid(t12, t34, t_eig, p12, p23,p_eig, \n    labels = \u0026quot;auto\u0026quot;, ncol = 3)\ndev.off()\n\n#We will keep all three axis (significant, but could also work on the first only:\nprec.ind \u0026lt;- get_pca_ind(X.prec)\ntemp.ind \u0026lt;- get_pca_ind(X.temp)\ncolnames(temp.ind$coord) \u0026lt;- c(\u0026quot;Temp1\u0026quot;,\u0026quot;Temp2\u0026quot;,\u0026quot;Temp3\u0026quot;,\u0026quot;Temp4\u0026quot;)\ncolnames(prec.ind$coord) \u0026lt;- c(\u0026quot;Prec1\u0026quot;,\u0026quot;Prec2\u0026quot;,\u0026quot;Prec3\u0026quot;)\n\nprec = cbind(bioclim$POP, prec.ind$coord)\ntemp = cbind(bioclim$POP, temp.ind$coord)\n\nwrite.table(temp, \u0026quot;02.data/env/temperature_coordinates_pca.txt\u0026quot; , quote = F, row.names = F )\nwrite.table(prec, \u0026quot;02.data/env/precipitation_coordinates_pca.txt\u0026quot; , quote = F, row.names = F )\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# load libs\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003elibs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003edplyr\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eade4\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003efactoextra\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003evegan\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecowplot\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003einvisible\u003c/span\u003e(lapply(\u003cspan class=\"pl-smi\"\u003elibs\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003elibrary\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003echaracter.only\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e))\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e##### DOWNLOAD ENV DATA\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e01-info/metadata\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eh\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003ebioclim\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e01-info/bioclimatic_var\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eh\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eremove BNV and SAI:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ebioclim\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ebioclim\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%filter(\u003cspan class=\"pl-smi\"\u003ePOP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSAI\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ePOP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBNV\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e###### PERFORM PCA ON ENVT VAR #################################################\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ewill do a PCA on the dataaset that contains only climatic variables:\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eseparate temperature and precipitation and keep only significant axis or only first axis\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eX.temp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e dudi.pca(\u003cspan class=\"pl-v\"\u003edf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ebioclim\u003c/span\u003e[, \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e56\u003c/span\u003e], \u003cspan class=\"pl-v\"\u003ecenter\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escannf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003enf = 3)\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eX.prec\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e dudi.pca(\u003cspan class=\"pl-v\"\u003edf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ebioclim\u003c/span\u003e[, \u003cspan class=\"pl-c1\"\u003e57\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e96\u003c/span\u003e], \u003cspan class=\"pl-v\"\u003ecenter\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escannf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e nf = 3)\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# singificance of axis\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eeig.val\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e get_eigenvalue(\u003cspan class=\"pl-smi\"\u003eX.temp\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003efirst we get the eigen value in an easy form\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eeig.val\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eeig\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eeig.val\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003evariance.percent\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003epercent\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eexpected\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e bstick(length(\u003cspan class=\"pl-smi\"\u003eeig.val\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eeig\u003c/span\u003e) )\n\u003cspan class=\"pl-smi\"\u003esignif\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eeig.val\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eeig\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eexpected\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eget signifcicant axis #3axes sont signif\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esignif\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4 axis are significant\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eeig.val.p\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e get_eigenvalue(\u003cspan class=\"pl-smi\"\u003eX.prec\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003efirst we get the eigen value in an easy form\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eeig.val.p\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eeig\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eeig.val.p\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003evariance.percent\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003epercent\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eexpected\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e bstick(length(\u003cspan class=\"pl-smi\"\u003eeig.val.p\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eeig\u003c/span\u003e) )\n\u003cspan class=\"pl-smi\"\u003esignif.p\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eeig.val.p\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eeig\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eexpected\u003c/span\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eget signifcicant axis #3axes sont signif\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003esignif.p\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3 axis are signficant\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003evisualisation of individuals cos^2:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eX.temp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e dudi.pca(\u003cspan class=\"pl-v\"\u003edf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ebioclim\u003c/span\u003e[, \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e53\u003c/span\u003e], \u003cspan class=\"pl-v\"\u003ecenter\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escannf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eX.prec\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e dudi.pca(\u003cspan class=\"pl-v\"\u003edf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ebioclim\u003c/span\u003e[, \u003cspan class=\"pl-c1\"\u003e54\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e95\u003c/span\u003e], \u003cspan class=\"pl-v\"\u003ecenter\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003escannf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e,  \u003cspan class=\"pl-v\"\u003enf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e)\n\nfviz_pca_ind(\u003cspan class=\"pl-smi\"\u003eX.prec\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol.ind\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecos2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egeom\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epoint\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    scale_color_gradient2(\u003cspan class=\"pl-v\"\u003elow\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ewhite\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emid\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003ehigh\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emidpoint\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.6\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e theme_minimal()\n\n\u003cspan class=\"pl-smi\"\u003ep12\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fviz_pca_var(\u003cspan class=\"pl-smi\"\u003eX.prec\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol.var\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esteelblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    theme_minimal()  \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e  ggtitle(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePrecipitation PC axis 12\u003cspan class=\"pl-cce\"\u003e\\n\u003c/span\u003evariables\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n     theme(\u003cspan class=\"pl-v\"\u003eplot.title\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003elineheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.8\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eface\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebold\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003ep23\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fviz_pca_var(\u003cspan class=\"pl-smi\"\u003eX.prec\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eaxes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e) , \u003cspan class=\"pl-v\"\u003ecol.var\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esteelblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    theme_minimal() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e  ggtitle(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePrecipitation PC axis 23\u003cspan class=\"pl-cce\"\u003e\\n\u003c/span\u003evariables\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n     theme(\u003cspan class=\"pl-v\"\u003eplot.title\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003elineheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.8\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eface\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebold\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003et12\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fviz_pca_var(\u003cspan class=\"pl-smi\"\u003eX.temp\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecol.var\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esteelblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    theme_minimal() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e  ggtitle(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemperature PC axis 12\u003cspan class=\"pl-cce\"\u003e\\n\u003c/span\u003evariables\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n     theme(\u003cspan class=\"pl-v\"\u003eplot.title\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003elineheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.8\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eface\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebold\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e , \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003et34\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fviz_pca_var(\u003cspan class=\"pl-smi\"\u003eX.temp\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eaxes\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e) , \u003cspan class=\"pl-v\"\u003ecol.var\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esteelblue\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    theme_minimal() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e  ggtitle(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemperature PC axis 34\u003cspan class=\"pl-cce\"\u003e\\n\u003c/span\u003evariables\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n     theme(\u003cspan class=\"pl-v\"\u003eplot.title\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003elineheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.8\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eface\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebold\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e , \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e))\n\n\n\u003cspan class=\"pl-smi\"\u003et_eig\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fviz_eig(\u003cspan class=\"pl-smi\"\u003eX.temp\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eaddlabels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylim\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e50\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n    ggtitle(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemperature eigenvalues\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n     theme(\u003cspan class=\"pl-v\"\u003eplot.title\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003elineheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.8\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eface\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebold\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e , \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e))\n\n\n\u003cspan class=\"pl-smi\"\u003ep_eig\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fviz_eig(\u003cspan class=\"pl-smi\"\u003eX.prec\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eaddlabels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylim\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e50\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n    ggtitle(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePrecipitation eigenvalues\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \n     theme(\u003cspan class=\"pl-v\"\u003eplot.title\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003elineheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.8\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eface\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ebold\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e , \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e))\n\npdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etemperature_and_precipiration.pca.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e18\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e12\u003c/span\u003e)\nplot_grid(\u003cspan class=\"pl-smi\"\u003et12\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003et34\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003et_eig\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ep12\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ep23\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ep_eig\u003c/span\u003e, \n    \u003cspan class=\"pl-v\"\u003elabels\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eauto\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003encol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e)\ndev.off()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eWe will keep all three axis (significant, but could also work on the first only:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eprec.ind\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e get_pca_ind(\u003cspan class=\"pl-smi\"\u003eX.prec\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003etemp.ind\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e get_pca_ind(\u003cspan class=\"pl-smi\"\u003eX.temp\u003c/span\u003e)\ncolnames(\u003cspan class=\"pl-smi\"\u003etemp.ind\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecoord\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemp1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemp2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemp3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemp4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ncolnames(\u003cspan class=\"pl-smi\"\u003eprec.ind\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecoord\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePrec1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePrec2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePrec3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003eprec\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e cbind(\u003cspan class=\"pl-smi\"\u003ebioclim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePOP\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eprec.ind\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecoord\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003etemp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e cbind(\u003cspan class=\"pl-smi\"\u003ebioclim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePOP\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etemp.ind\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecoord\u003c/span\u003e)\n\nwrite.table(\u003cspan class=\"pl-smi\"\u003etemp\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e02.data/env/temperature_coordinates_pca.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e , \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e )\nwrite.table(\u003cspan class=\"pl-smi\"\u003eprec\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e02.data/env/precipitation_coordinates_pca.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e , \u003cspan class=\"pl-v\"\u003equote\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erow.names\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e )\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ethis will create the two input-files\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003etemperature_coordinates_pca.txt\u003c/li\u003e\n\u003cli\u003eprecipitation_coordinates_pca.txt\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003elocated in \u003ccode\u003e02.data/env/\u003c/code\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThis will be used in the GEA\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ewe can check that the data are not correlated using a simple correlation plot in R for instance.\nThese few lines of code will do the job:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"\nlibrary(magrittr)\nlibrary(dplyr)\nlibrary(corrplot)\n\n###### DOWNLOAD ENV DATA\nmetadata \u0026lt;- read.table(\u0026quot;01-info/metadata\u0026quot;, h = T, sep = \u0026quot;\\t\u0026quot;)\nmetadata\u0026lt;- metadata %\u0026gt;%filter(POP !=\u0026quot;SAI\u0026quot; \u0026amp; POP !=\u0026quot;BNV\u0026quot;)\n\ngeol \u0026lt;- read.table(\u0026quot;02-data/env/geology\u0026quot;, T)\n#remove BNV and SAI:\ngeol \u0026lt;- geol %\u0026gt;%filter(POP !=\u0026quot;SAI\u0026quot; \u0026amp; POP !=\u0026quot;BNV\u0026quot;)\n#remove BNV and SAI:\n\n#replace altitude of zero by 1\nmetadata$elevation[metadata$elevation == 0.00000 ] \u0026lt;- 1\n\n#now takes into accont the standardized elevation * distance interaction\n#standardiztation function:\nrange01 \u0026lt;- function(x){(x-min(x))/(max(x)-min(x))}\nmetadata$normalized_distance =  metadata$elevation * metadata$dist_max_km\nmetadata$normalized_distance \u0026lt;- range01(metadata$normalized_distance)\n\n####### PERFORM PCA ON ENVT VAR #################################################\ntemp \u0026lt;- read.table(\u0026quot;02-data/env/temperature_coordinates_pca.txt\u0026quot;, T)\nprec \u0026lt;- read.table(\u0026quot;02-data/env/precipitation_coordinates_pca.txt\u0026quot;, T)\n\nenv1 \u0026lt;- dplyr::select(metadata, POP, Latitude, normalized_distance, Region)\nenv1 \u0026lt;- merge(env1, temp)\nenv1 \u0026lt;- merge(env1, prec)\nenv1 \u0026lt;- merge(env1, geol)\n\n#remplacer avec sed directement dans metadata, geology et bioclim\nenv1 \u0026lt;- env1[order(env1$POP),]\n\n#choose the variable we want to work with:\nenv \u0026lt;- select(env1, -POP, -Region)\n\n#rename for readiblity of the plot:\ncolnames(env) \u0026lt;- c(\u0026quot;Latitude\u0026quot;,\u0026quot;normalized_distance\u0026quot;, \n    \u0026quot;TemperaturePC1\u0026quot;, \u0026quot;TemperaturePC2\u0026quot;, \u0026quot;TemperaturePC3\u0026quot;, \u0026quot;TemperaturePC4\u0026quot;,\n    \u0026quot;PrecipitationPC1\u0026quot;, \u0026quot;PrecipitationPC2\u0026quot;, \u0026quot;PrecipitationPC3\u0026quot;, \n    \u0026quot;geology\u0026quot;)\n\n#again write the correlation \nmat \u0026lt;- round(cor(env),2)\n\npdf(file=\u0026quot;corrplot.pdf\u0026quot;)\ncorrplot(mat, method = \u0026quot;number\u0026quot;, type = \u0026quot;lower\u0026quot;)\ndev.off()\n\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003emagrittr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003ecorrplot\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e##### DOWNLOAD ENV DATA\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e01-info/metadata\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eh\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-cce\"\u003e\\t\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%filter(\u003cspan class=\"pl-smi\"\u003ePOP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSAI\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ePOP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBNV\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003egeol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e02-data/env/geology\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eremove BNV and SAI:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003egeol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egeol\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%filter(\u003cspan class=\"pl-smi\"\u003ePOP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSAI\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026amp;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ePOP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e!=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eBNV\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eremove BNV and SAI:\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ereplace altitude of zero by 1\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eelevation\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eelevation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.00000\u003c/span\u003e ] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003enow takes into accont the standardized elevation * distance interaction\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003estandardiztation function:\u003c/span\u003e\n\u003cspan class=\"pl-en\"\u003erange01\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e){(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003emin(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e))\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e(max(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003emin(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e))}\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-v\"\u003enormalized_distance\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e  \u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eelevation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003edist_max_km\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enormalized_distance\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e range01(\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enormalized_distance\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e###### PERFORM PCA ON ENVT VAR #################################################\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etemp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e02-data/env/temperature_coordinates_pca.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eprec\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e read.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e02-data/env/precipitation_coordinates_pca.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003eT\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eenv1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003emetadata\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ePOP\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eLatitude\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003enormalized_distance\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eRegion\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eenv1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e merge(\u003cspan class=\"pl-smi\"\u003eenv1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etemp\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eenv1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e merge(\u003cspan class=\"pl-smi\"\u003eenv1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eprec\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eenv1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e merge(\u003cspan class=\"pl-smi\"\u003eenv1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003egeol\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eremplacer avec sed directement dans metadata, geology et bioclim\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eenv1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eenv1\u003c/span\u003e[order(\u003cspan class=\"pl-smi\"\u003eenv1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePOP\u003c/span\u003e),]\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003echoose the variable we want to work with:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eenv\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e select(\u003cspan class=\"pl-smi\"\u003eenv1\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePOP\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRegion\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003erename for readiblity of the plot:\u003c/span\u003e\ncolnames(\u003cspan class=\"pl-smi\"\u003eenv\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLatitude\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enormalized_distance\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemperaturePC1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemperaturePC2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemperaturePC3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemperaturePC4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePrecipitationPC1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePrecipitationPC2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePrecipitationPC3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003egeology\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eagain write the correlation \u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003emat\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e round(cor(\u003cspan class=\"pl-smi\"\u003eenv\u003c/span\u003e),\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\n\npdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecorrplot.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ncorrplot(\u003cspan class=\"pl-smi\"\u003emat\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003enumber\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003elower\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\ndev.off()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eand produce the following graph:\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/corrplot.png\"\u003e\u003cimg src=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/corrplot.png\" alt=\"example_graph\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.2. Prepare genomics data\u003c/h2\u003e\u003ca id=\"user-content-52-prepare-genomics-data\" class=\"anchor\" aria-label=\"Permalink: 5.2. Prepare genomics data\" href=\"#52-prepare-genomics-data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe first need to filter the data to exclude 2 populations, namely BNV and SAI for which environmental data are not available.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-shell notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" #################################################################################\n#                           preparing data for RDA and GEA\n#################################################################################\n\nmkdir GEA\ncd GEA\n\n#we create a file containing population SAI and BNV that we want to exclude\ngrep \u0026quot;SAI\\|BNV\u0026quot; ../strata \u0026gt; SAI_BNV.txt \n\nvcftools --gzvcf ../populations.24_09.missing0.92mac15.indep.clean2.recode.vcf.gz \\\n    --remove SAI_BNV.txt \\\n    --mac 15  \\\n    --max-missing 0.95 \\\n    --min-meanDP 10 --max-meanDP 120 \\\n    --out population_for_GEA \\\n    --recode \\\n    --recode-INFO-all\n\ninput=population_for_GEA.recode.vcf\nstrata $input strata.txt\nbgzip $input\n\ninput=$input.gz\n\ncut -f 2 strata.txt \u0026gt; ind.tmp; cp ind.tmp new.ind.tmp \n\n#in this file we need to rename the 32 first individuals from HOP into HOD as the \u0026quot;HOP\u0026quot; population contains individual sample from both \u0026quot;HOP\u0026quot; and \u0026quot;HOD\u0026quot;\nsed -i '2536,2567s/HOP/HOD/g' new.ind.tmp\nsed -i 's/DRA/WAC/g' new.ind.tmp \n\npaste ind.tmp new.ind.tmp \u0026gt; new_sample_name.txt \n\n#then we rename individuals in the vcf \nbcftools reheader --samples new_sample_name.txt -o ${input%.recode.vcf.gz}.renamed.vcf.gz $input \n\ngunzip ${input%.recode.vcf.gz}.renamed.vcf.gz\n\nINPUT=\u0026quot;${input%.recode.vcf.gz}.renamed.vcf\u0026quot;\n\n#create usefull intput for LFMM:\nvcf2geno $INPUT \ngeno2lfmm ${INPUT%.vcf}.geno \nsed -i 's/9/NA/g' ${INPUT%.vcf}.lfmm\n\n#save space\ngzip ${INPUT%.vcf}.lfmm\ngzip ${INPUT%.vcf}.geno\n\n\nplink --vcf $INPUT \\\n    --allow-extra-chr \\\n    --out ${INPUT%.recode.vcf} \\\n    --recode\n    --recode\n\n#prepare a cluster file\nrm strata.txt\nstrata $INPUT strata.txt\ncut -d \u0026quot;_\u0026quot; -f2 new.ind.tmp \u0026gt; col2\ncut -d \u0026quot;_\u0026quot; -f1 new.ind.tmp \u0026gt; col1\npaste col1 col2 col1 \u0026gt; cluster.dat\n\n#now compute allele frequency for RDA:\nplink --file  ${INPUT%.recode.vcf}  \\\n    --allow-extra-chr --freq \\\n    --within cluster.dat \n\ngzip plink.frq.strat\n\nrm *tmp col*\n#now we can run the RDA and LFMM\n\n\n################################################################################\n#           prepare the data by excluding the Thompson samples                 #\n################################################################################\n# Since the Thompson samples populations are highly divergent they may be driving part of the signal, we will therefore replicate the analyses by removing them from the samples.\n\nmkdir no_thompson\ncd no_thompson\n\nawk -F\u0026quot;\\t\u0026quot; '$10==\u0026quot;Thompson\u0026quot; {print $0}' ../metadata \u0026gt; thompson\nawk -F\u0026quot;\\t\u0026quot; '$10==\u0026quot;Thompson\u0026quot; {print $1}' ../metadata \u0026gt; thompson.pop #to keep only thompson rivers\n\ngrep -Ff thompson.pop ../newnames  \u0026gt; individus.thompson #to extract individuals\n\ninput=population_for_GEA.renamed.vcf.gz\noutput=population_for_GEA.renamed.nothompson\n\nvcftools --gzvcf ../$input \\\n    --remove individus.thompson \\\n    --mac 15  \\\n    --max-missing 0.95 \\\n    --min-meanDP 10 --max-meanDP 120 \\\n    --out $output \\\n    --recode \\\n    --recode-INFO-all\n\n\nstrata $output.recode.vcf strata.txt\n\nINPUT=\u0026quot;${output}.recode.vcf\u0026quot;\n\nvcf2geno $INPUT \ngeno2lfmm ${INPUT%.vcf}.geno \nsed -i 's/9/NA/g' ${INPUT%.vcf}.lfmm\n\n# save space\n\ngzip ${INPUT%.vcf}.lfmm\ngzip ${INPUT%.vcf}.geno\n\nplink --vcf $INPUT \\\n    --allow-extra-chr \\\n    --out ${INPUT%.recode.vcf} \\\n    --recode\n    --recode\n\n# prepare a cluster file\n\nstrata $INPUT strata.txt\ncut -f1 strata.txt \u0026gt; col1\ncut -d \u0026quot;\\_\u0026quot; -f2 strata.txt \u0026gt; col2\npaste col1 col2 col1 \u0026gt; cluster.dat\n\n# now compute allele frequency for RDA\n\nplink --file  ${INPUT%.recode.vcf} \\\n    --allow-extra-chr --freq \\\n    --within cluster.dat \n\ngzip plink.frq.strat\n\nrm \\*tmp col\\* \n#now we can run the RDA and LFMM\n\n\"\u003e\u003cpre\u003e \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e################################################################################\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e                           preparing data for RDA and GEA\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e################################################################################\u003c/span\u003e\n\nmkdir GEA\n\u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e GEA\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ewe create a file containing population SAI and BNV that we want to exclude\u003c/span\u003e\ngrep \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSAI\\|BNV\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e ../strata \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e SAI_BNV.txt \n\nvcftools --gzvcf ../populations.24_09.missing0.92mac15.indep.clean2.recode.vcf.gz \\\n    --remove SAI_BNV.txt \\\n    --mac 15  \\\n    --max-missing 0.95 \\\n    --min-meanDP 10 --max-meanDP 120 \\\n    --out population_for_GEA \\\n    --recode \\\n    --recode-INFO-all\n\ninput=population_for_GEA.recode.vcf\nstrata \u003cspan class=\"pl-smi\"\u003e$input\u003c/span\u003e strata.txt\nbgzip \u003cspan class=\"pl-smi\"\u003e$input\u003c/span\u003e\n\ninput=\u003cspan class=\"pl-smi\"\u003e$input\u003c/span\u003e.gz\n\ncut -f 2 strata.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e ind.tmp\u003cspan class=\"pl-k\"\u003e;\u003c/span\u003e cp ind.tmp new.ind.tmp \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ein this file we need to rename the 32 first individuals from HOP into HOD as the \"HOP\" population contains individual sample from both \"HOP\" and \"HOD\"\u003c/span\u003e\nsed -i \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e2536,2567s/HOP/HOD/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e new.ind.tmp\nsed -i \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/DRA/WAC/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e new.ind.tmp \n\npaste ind.tmp new.ind.tmp \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e new_sample_name.txt \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ethen we rename individuals in the vcf \u003c/span\u003e\nbcftools reheader --samples new_sample_name.txt -o \u003cspan class=\"pl-smi\"\u003e${input\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.recode.vcf.gz}\u003c/span\u003e.renamed.vcf.gz \u003cspan class=\"pl-smi\"\u003e$input\u003c/span\u003e \n\ngunzip \u003cspan class=\"pl-smi\"\u003e${input\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.recode.vcf.gz}\u003c/span\u003e.renamed.vcf.gz\n\nINPUT=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e${input\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.recode.vcf.gz}\u003c/span\u003e.renamed.vcf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ecreate usefull intput for LFMM:\u003c/span\u003e\nvcf2geno \u003cspan class=\"pl-smi\"\u003e$INPUT\u003c/span\u003e \ngeno2lfmm \u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.vcf}\u003c/span\u003e.geno \nsed -i \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/9/NA/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.vcf}\u003c/span\u003e.lfmm\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003esave space\u003c/span\u003e\ngzip \u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.vcf}\u003c/span\u003e.lfmm\ngzip \u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.vcf}\u003c/span\u003e.geno\n\n\nplink --vcf \u003cspan class=\"pl-smi\"\u003e$INPUT\u003c/span\u003e \\\n    --allow-extra-chr \\\n    --out \u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.recode.vcf}\u003c/span\u003e \\\n    --recode\n    --recode\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eprepare a cluster file\u003c/span\u003e\nrm strata.txt\nstrata \u003cspan class=\"pl-smi\"\u003e$INPUT\u003c/span\u003e strata.txt\ncut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f2 new.ind.tmp \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e col2\ncut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f1 new.ind.tmp \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e col1\npaste col1 col2 col1 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e cluster.dat\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003enow compute allele frequency for RDA:\u003c/span\u003e\nplink --file  \u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.recode.vcf}\u003c/span\u003e  \\\n    --allow-extra-chr --freq \\\n    --within cluster.dat \n\ngzip plink.frq.strat\n\nrm \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003etmp col\u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003enow we can run the RDA and LFMM\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e###############################################################################\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e           prepare the data by excluding the Thompson samples                 #\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e###############################################################################\u003c/span\u003e\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Since the Thompson samples populations are highly divergent they may be driving part of the signal, we will therefore replicate the analyses by removing them from the samples.\u003c/span\u003e\n\nmkdir no_thompson\n\u003cspan class=\"pl-c1\"\u003ecd\u003c/span\u003e no_thompson\n\nawk -F\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\\t\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e$10==\"Thompson\" {print $0}\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e ../metadata \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e thompson\nawk -F\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\\t\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e$10==\"Thompson\" {print $1}\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e ../metadata \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e thompson.pop \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eto keep only thompson rivers\u003c/span\u003e\n\ngrep -Ff thompson.pop ../newnames  \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e individus.thompson \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eto extract individuals\u003c/span\u003e\n\ninput=population_for_GEA.renamed.vcf.gz\noutput=population_for_GEA.renamed.nothompson\n\nvcftools --gzvcf ../\u003cspan class=\"pl-smi\"\u003e$input\u003c/span\u003e \\\n    --remove individus.thompson \\\n    --mac 15  \\\n    --max-missing 0.95 \\\n    --min-meanDP 10 --max-meanDP 120 \\\n    --out \u003cspan class=\"pl-smi\"\u003e$output\u003c/span\u003e \\\n    --recode \\\n    --recode-INFO-all\n\n\nstrata \u003cspan class=\"pl-smi\"\u003e$output\u003c/span\u003e.recode.vcf strata.txt\n\nINPUT=\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e${output}\u003c/span\u003e.recode.vcf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n\nvcf2geno \u003cspan class=\"pl-smi\"\u003e$INPUT\u003c/span\u003e \ngeno2lfmm \u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.vcf}\u003c/span\u003e.geno \nsed -i \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003es/9/NA/g\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.vcf}\u003c/span\u003e.lfmm\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e save space\u003c/span\u003e\n\ngzip \u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.vcf}\u003c/span\u003e.lfmm\ngzip \u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.vcf}\u003c/span\u003e.geno\n\nplink --vcf \u003cspan class=\"pl-smi\"\u003e$INPUT\u003c/span\u003e \\\n    --allow-extra-chr \\\n    --out \u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.recode.vcf}\u003c/span\u003e \\\n    --recode\n    --recode\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e prepare a cluster file\u003c/span\u003e\n\nstrata \u003cspan class=\"pl-smi\"\u003e$INPUT\u003c/span\u003e strata.txt\ncut -f1 strata.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e col1\ncut -d \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\\_\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e -f2 strata.txt \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e col2\npaste col1 col2 col1 \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e cluster.dat\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e now compute allele frequency for RDA\u003c/span\u003e\n\nplink --file  \u003cspan class=\"pl-smi\"\u003e${INPUT\u003cspan class=\"pl-k\"\u003e%\u003c/span\u003e.recode.vcf}\u003c/span\u003e \\\n    --allow-extra-chr --freq \\\n    --within cluster.dat \n\ngzip plink.frq.strat\n\nrm \u003cspan class=\"pl-cce\"\u003e\\*\u003c/span\u003etmp col\u003cspan class=\"pl-cce\"\u003e\\*\u003c/span\u003e \n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003enow we can run the RDA and LFMM\u003c/span\u003e\n\n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.3. Running the GEAs\u003c/h2\u003e\u003ca id=\"user-content-53-running-the-geas\" class=\"anchor\" aria-label=\"Permalink: 5.3. Running the GEAs\" href=\"#53-running-the-geas\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.3.1 LFMM\u003c/h2\u003e\u003ca id=\"user-content-531-lfmm\" class=\"anchor\" aria-label=\"Permalink: 5.3.1 LFMM\" href=\"#531-lfmm\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eLFMM will use a mixed model analyses to identify outliers\u003c/p\u003e\n\u003cp dir=\"auto\"\u003esimply use the script in\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"./00-scripts/gea/01.lfmm_step1.R \"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e00\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003escripts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egea\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e01.\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003elfmm_step1.R\u003c/span\u003e \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ethis will run rather rapidly with approximately 20-30Gb of RAM\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.3.2 RDA\u003c/h2\u003e\u003ca id=\"user-content-532-rda\" class=\"anchor\" aria-label=\"Permalink: 5.3.2 RDA\" href=\"#532-rda\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNext we run the RDA in 2 steps\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e**Significance testing **\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eFirst we need to test the significance of environmental data and of the RDA axes.\nThis is done through an ANOVA like permutation test of the RDA using the \u003ccode\u003eanova.cca\u003c/code\u003e function in R\nThe \u003ccode\u003eordiR2step\u003c/code\u003e function also allows to test the best model and choose significant variable of the RDA through a set of permutation test.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eRun the script :\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"./00-scripts/gea/02.RDA_significance_testing.R\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e00\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003escripts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egea\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e02.\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRDA_significance_testing.R\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eit takes a few hours (~one night) on a cluster\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003e\u003cstrong\u003eidentify outliers:\u003c/strong\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003erun the script :\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"Rscript 00-scripts/gea/03.RDA_identify_outliers.R\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eRscript\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e00\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003escripts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egea\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e03.\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eRDA_identify_outliers.R\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ethis will write the number of outlier in a file as well as other useful data such as the number of outliers in each chromosomes.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eWe used the approach of Forester et al. (2018) to identify outliers.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e5.3.3. Create a RDA Figure and LFMM plot:\u003c/h2\u003e\u003ca id=\"user-content-533-create-a-rda-figure-and-lfmm-plot\" class=\"anchor\" aria-label=\"Permalink: 5.3.3. Create a RDA Figure and LFMM plot:\" href=\"#533-create-a-rda-figure-and-lfmm-plot\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eTo reproduce the Figure 3 simply run:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"Rscript 00-scripts/gea/Figure3.R\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003eRscript\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e00\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003escripts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egea\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFigure3.R\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis should automatically produce the Figure3 from our manuscript\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e6. Looking for parallelism\u003c/h2\u003e\u003ca id=\"user-content-6-looking-for-parallelism\" class=\"anchor\" aria-label=\"Permalink: 6. Looking for parallelism\" href=\"#6-looking-for-parallelism\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRun the Rscript\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"00-scripts/gea/05.Figure4.R \"\u003e\u003cpre\u003e\u003cspan class=\"pl-c1\"\u003e00\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003escripts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egea\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e05.\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFigure4.R\u003c/span\u003e \u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis should automatically produce the Figure4 from our manuscript as well as some statisticall tests\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eSee explanation in our methods of the manuscript\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eI'll document all of this later.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e7. plotting LFMM results:\u003c/h2\u003e\u003ca id=\"user-content-7-plotting-lfmm-results\" class=\"anchor\" aria-label=\"Permalink: 7. plotting LFMM results:\" href=\"#7-plotting-lfmm-results\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ethis can be done with CMplot\nsee script 00-scripts/gea/LFMM_plot_FigureS10.R or simple code below:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"## load libs\nlibs \u0026lt;- c('dplyr','data.table', 'magrittr', 'cowplot','CMplot', 'gridGraphics')\ninvisible(lapply(libs, library, character.only = TRUE))\n\npval_loc \u0026lt;- fread(\u0026quot;zcat adjust_pvaluesBH_lfkmm_K.20.txt.gz\u0026quot;)\n\n#we will remove scaffold to plot only the 30 chromosome + the 8 dups.\npvalsub \u0026lt;- pval_loc %\u0026gt;% filter(!(grepl(\u0026quot;scaff\u0026quot;, V1)))\npvalsub$V1\u0026lt;- gsub(\u0026quot;Okis\u0026quot;,\u0026quot;\u0026quot;,pvalsub$V1)\npvalsub$V1\u0026lt;- gsub(\u0026quot;LG\u0026quot;,\u0026quot;\u0026quot;,pvalsub$V1)\npvalsub$V1\u0026lt;- gsub(\u0026quot;_hom\u0026quot;,\u0026quot;\u0026quot;,pvalsub$V1)\n\n#temperature\ntempplot \u0026lt;- select(pvalsub,V3,V1,V2,Temperature1, Temperature2, Temperature3, Temperature4) %\u0026gt;% \n    set_colnames(.,c(\u0026quot;SNP\u0026quot;,\u0026quot;Chromosome\u0026quot;,\u0026quot;Position\u0026quot;,\n    \u0026quot;TemperaturePCaxis1\u0026quot;,\u0026quot;TemperaturePCaxis2\u0026quot;,\u0026quot;TemperaturePCaxis3\u0026quot;,\u0026quot;TemperaturePCaxis4\u0026quot;))\ntemplot \u0026lt;- as.data.frame(tempplot)\n\n#precipitation\nprecplot \u0026lt;- select(pvalsub,V3,V1,V2,Precipitation1, Precipitation2, Precipitation3) %\u0026gt;% \n    set_colnames(.,c(\u0026quot;SNP\u0026quot;,\u0026quot;Chromosome\u0026quot;,\u0026quot;Position\u0026quot;,\n    \u0026quot;PrecipitationPCaxis1\u0026quot;,\u0026quot;PrecipitationPCaxis2\u0026quot;,\u0026quot;PrecipitationPCaxis3\u0026quot;))\nprecplot \u0026lt;- as.data.frame(precplot)\n\n#geology\ngeolplot \u0026lt;- select(pvalsub,V3,V1,V2,Geology) %\u0026gt;% \n    set_colnames(.,c(\u0026quot;SNP\u0026quot;,\u0026quot;Chromosome\u0026quot;,\u0026quot;Position\u0026quot;,\u0026quot;Geology\u0026quot;))\ngeolplot \u0026lt;- as.data.frame(geolplot)\n\n\nCMplot(geolplot, plot.type=\u0026quot;m\u0026quot;, LOG10=TRUE, ylim=NULL, threshold=NULL ,\n       chr.den.col = c(\u0026quot;darkgreen\u0026quot;, \u0026quot;yellow\u0026quot;, \u0026quot;red\u0026quot;),\n       signal.pch = c(19,19),file=\u0026quot;pdf\u0026quot;,memo=\u0026quot;\u0026quot;,dpi=300,\n       file.output = F,\n       verbose = TRUE,\n       width=14,height=6, chr.labels.angle=45)\n\np3 \u0026lt;- recordPlot()\nCMplot(templot, plot.type=\u0026quot;m\u0026quot;, multracks = TRUE, LOG10=TRUE, ylim=NULL, threshold=NULL, mar = c(5,8,3,3),\n       chr.den.col = c(\u0026quot;darkgreen\u0026quot;, \u0026quot;yellow\u0026quot;, \u0026quot;red\u0026quot;),\n       signal.pch = c(19,19),file=\u0026quot;pdf\u0026quot;,memo=\u0026quot;\u0026quot;,dpi=300,\n       file.output = F,\n       verbose = TRUE,\n       width=14,height=6, chr.labels.angle=45)\np1 \u0026lt;- recordPlot()\n\n\nCMplot(precplot, plot.type=\u0026quot;m\u0026quot;, multracks = TRUE, LOG10=TRUE, ylim=NULL, threshold=NULL, mar = c(5,8,3,3),\n       chr.den.col = c(\u0026quot;darkgreen\u0026quot;, \u0026quot;yellow\u0026quot;, \u0026quot;red\u0026quot;),\n       signal.pch = c(19,19),file=\u0026quot;pdf\u0026quot;,memo=\u0026quot;\u0026quot;,dpi=300,\n       file.output = F,\n       verbose = TRUE,\n       width=14,height=6, chr.labels.angle=45)\np2 \u0026lt;- recordPlot()\n\n#export final figure\npdf(file=\u0026quot;FigureS10.pdf\u0026quot;,15,20)\nplot_grid(p1, p2,p3, \n    ncol = 1, \n    align = \u0026quot;v\u0026quot;, \n    labels=c(\u0026quot;A - Temperature\u0026quot;, \u0026quot;B - Precipitation\u0026quot;, \u0026quot;C - Geology\u0026quot;) )\ndev.off()\n\"\u003e\u003cpre\u003e\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e# load libs\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003elibs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003edplyr\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003edata.table\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003emagrittr\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003ecowplot\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003eCMplot\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003egridGraphics\u003cspan class=\"pl-pds\"\u003e'\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-k\"\u003einvisible\u003c/span\u003e(lapply(\u003cspan class=\"pl-smi\"\u003elibs\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003elibrary\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003echaracter.only\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003epval_loc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ezcat adjust_pvaluesBH_lfkmm_K.20.txt.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003ewe will remove scaffold to plot only the 30 chromosome + the 8 dups.\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003epvalsub\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epval_loc\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% filter(\u003cspan class=\"pl-k\"\u003e!\u003c/span\u003e(grepl(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003escaff\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eV1\u003c/span\u003e)))\n\u003cspan class=\"pl-smi\"\u003epvalsub\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e gsub(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOkis\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003epvalsub\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV1\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003epvalsub\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e gsub(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eLG\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003epvalsub\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV1\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003epvalsub\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e gsub(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e_hom\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003epvalsub\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV1\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003etemperature\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etempplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e select(\u003cspan class=\"pl-smi\"\u003epvalsub\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eV3\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eV1\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eTemperature1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eTemperature2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eTemperature3\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eTemperature4\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n    set_colnames(\u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e,c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eChromosome\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePosition\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemperaturePCaxis1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemperaturePCaxis2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemperaturePCaxis3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTemperaturePCaxis4\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003etemplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.data.frame(\u003cspan class=\"pl-smi\"\u003etempplot\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eprecipitation\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eprecplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e select(\u003cspan class=\"pl-smi\"\u003epvalsub\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eV3\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eV1\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ePrecipitation1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ePrecipitation2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ePrecipitation3\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n    set_colnames(\u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e,c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eChromosome\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePosition\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePrecipitationPCaxis1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePrecipitationPCaxis2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePrecipitationPCaxis3\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eprecplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.data.frame(\u003cspan class=\"pl-smi\"\u003eprecplot\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003egeology\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003egeolplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e select(\u003cspan class=\"pl-smi\"\u003epvalsub\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eV3\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eV1\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eV2\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003eGeology\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n    set_colnames(\u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e,c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eChromosome\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePosition\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eGeology\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003egeolplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e as.data.frame(\u003cspan class=\"pl-smi\"\u003egeolplot\u003c/span\u003e)\n\n\nCMplot(\u003cspan class=\"pl-smi\"\u003egeolplot\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eplot.type\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003em\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eLOG10\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eNULL\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ethreshold\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eNULL\u003c/span\u003e ,\n       \u003cspan class=\"pl-v\"\u003echr.den.col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgreen\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eyellow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n       \u003cspan class=\"pl-v\"\u003esignal.pch\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e19\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e19\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ememo\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edpi\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e,\n       \u003cspan class=\"pl-v\"\u003efile.output\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\n       \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n       \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e14\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003echr.labels.angle\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e45\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003ep3\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e recordPlot()\nCMplot(\u003cspan class=\"pl-smi\"\u003etemplot\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eplot.type\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003em\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emultracks\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eLOG10\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eNULL\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ethreshold\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eNULL\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emar\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e),\n       \u003cspan class=\"pl-v\"\u003echr.den.col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgreen\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eyellow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n       \u003cspan class=\"pl-v\"\u003esignal.pch\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e19\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e19\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ememo\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edpi\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e,\n       \u003cspan class=\"pl-v\"\u003efile.output\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\n       \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n       \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e14\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003echr.labels.angle\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e45\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ep1\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e recordPlot()\n\n\nCMplot(\u003cspan class=\"pl-smi\"\u003eprecplot\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eplot.type\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003em\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emultracks\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eLOG10\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eylim\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eNULL\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ethreshold\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eNULL\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emar\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e8\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e),\n       \u003cspan class=\"pl-v\"\u003echr.den.col\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003edarkgreen\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eyellow\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ered\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n       \u003cspan class=\"pl-v\"\u003esignal.pch\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c(\u003cspan class=\"pl-c1\"\u003e19\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e19\u003c/span\u003e),\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ememo\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edpi\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e,\n       \u003cspan class=\"pl-v\"\u003efile.output\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\n       \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n       \u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e14\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003echr.labels.angle\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e45\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ep2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e recordPlot()\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003eexport final figure\u003c/span\u003e\npdf(\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFigureS10.pdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e15\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e)\nplot_grid(\u003cspan class=\"pl-smi\"\u003ep1\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ep2\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003ep3\u003c/span\u003e, \n    \u003cspan class=\"pl-v\"\u003encol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \n    \u003cspan class=\"pl-v\"\u003ealign\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ev\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n    \u003cspan class=\"pl-v\"\u003elabels\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eA - Temperature\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eB - Precipitation\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eC - Geology\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) )\ndev.off()\n\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e8. PBS analyses\u003c/h2\u003e\u003ca id=\"user-content-8-pbs-analyses\" class=\"anchor\" aria-label=\"Permalink: 8. PBS analyses\" href=\"#8-pbs-analyses\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cstrong\u003eANGSD analyses:\u003c/strong\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThe major reason for using ANGSD was to compute the PBS score from Yi et al. 2010 (although it can be computed from the vcf directly)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAll steps to run ANGSD are found \u003ca href=\"https://github.com/QuentinRougemont/utility_scripts/tree/master/00.ANGSD\"\u003ehere\u003c/a\u003e. They necessitate to start from the bam files.\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003cstrong\u003eimportant note\u003c/strong\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003ebefore running Fst or PBS I used a three-species outgroup sequence for proper folding made of :\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"https://www.ncbi.nlm.nih.gov/bioproject/PRJNA287458\" rel=\"nofollow\"\u003eAtlantic salmon\u003c/a\u003e n = 5 individuals\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"https://www.ncbi.nlm.nih.gov/bioproject/PRJNA386519\" rel=\"nofollow\"\u003eRainbow trout\u003c/a\u003e n = 5 individuals and\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"https://www.ncbi.nlm.nih.gov/bioproject/PRJNA530256\" rel=\"nofollow\"\u003eSockeye salmon\u003c/a\u003e n = 3 individuals\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eWhole genome sequences were:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eprocessed with \u003ca href=\"https://github.com/QuentinRougemont/gatk_haplotype/blob/master/gatk4/01_scripts/01_fastp.sh\"\u003efastp\u003c/a\u003e\u003c/li\u003e\n\u003cli\u003emapped to the coho salmon genome using \u003ca href=\"https://github.com/QuentinRougemont/gatk_haplotype/blob/master/gatk4/01_scripts/02_bwa_mem2_align_reads_PE.sh\"\u003ebwa-mem\u003c/a\u003e.\u003c/li\u003e\n\u003cli\u003emarked for duplicates with \u003ca href=\"https://github.com/QuentinRougemont/gatk_haplotype/blob/master/gatk4/01_scripts/03_remove_duplicates.sh\"\u003epicard\u003c/a\u003e\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThe resulting bam were processed to reconstruct an ancestral sequence with \u003ca href=\"https://github.com/QuentinRougemont/utility_scripts/blob/master/00.ANGSD/ancestral_seq.sh\"\u003eancestral_seq.sh\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThen Simply follow the scripts in the order of their number:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/QuentinRougemont/utility_scripts/blob/master/00.ANGSD/00_angsd_bam_to_saf.sh\"\u003e00_angsd_bam_to_saf.sh\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/QuentinRougemont/utility_scripts/blob/master/00.ANGSD/01_1dsfs.sh\"\u003e01_1dsfs.sh\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/QuentinRougemont/utility_scripts/blob/master/00.ANGSD/02_2dsfs.sh\"\u003e02_2dsfs.sh\u003c/a\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"4\" dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/QuentinRougemont/utility_scripts/blob/master/00.ANGSD/04.pairwise_fst_sliding_window.sh\"\u003e04.pairwise_fst_sliding_window.sh\u003c/a\u003e #to get only Fst\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003col start=\"5\" dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/QuentinRougemont/utility_scripts/blob/master/00.ANGSD/05_PBS_sliding_windows.sh\"\u003e05_PBS_sliding_windows.sh\u003c/a\u003e #to get Fst and PBS\u003c/li\u003e\n\u003c/ol\u003e\n\u003c/li\u003e\n\u003cli\u003e\n\u003cp dir=\"auto\"\u003eProduce graph: I used \u003ca href=\"https://github.com/YinLiLin/CMplot\"\u003eCMplot\u003c/a\u003e to make all manhattan plot.\u003c/p\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eThe used is straightforward:\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(data.table)\nlibrary(dplyr)\nlibrary(magrittr)\nlibrary(CMplot)\nrm(list=ls())\nALB_LCA \u0026lt;- fread(\u0026quot;zcat all_ALB_LCA_SAI.OkisLG01.pbs.50.txt.gz\u0026quot;) #1 Thompson\nPOR_SHK \u0026lt;- fread(\u0026quot;zcat all_POR_SHK_SAI.OkisLG01.pbs.50.txt.gz\u0026quot;) #2 alaska\nSFK_GOO \u0026lt;- fread(\u0026quot;zcat all_SFK_GOO_SAI.OkisLG01.pbs.50.txt.gz\u0026quot;) #3 alkaska #2\nSUS_OON \u0026lt;- fread(\u0026quot;zcat all_SUS_OON_SAI.OkisLG01.pbs.50.txt.gz\u0026quot;) #4 BC North\n\nALB_LCA  \u0026lt;- dplyr::select(ALB_LCA, -Fst02, -Fst12, -nSite, -V3, -PBS1, -PBS2)\nPOR_SHK  \u0026lt;- dplyr::select(POR_SHK, -Fst02, -Fst12, -nSite, -V3, -PBS1, -PBS2)\nSFK_GOO  \u0026lt;- dplyr::select(SFK_GOO, -Fst02, -Fst12, -nSite, -V3, -PBS1, -PBS2)\nSUS_OON  \u0026lt;- dplyr::select(SUS_OON, -Fst02, -Fst12, -nSite, -V3, -PBS1, -PBS2)\n\n### Effectuer un manathan plots:\nALB_LCA2 \u0026lt;- dplyr::select(ALB_LCA, chr, midPos, PBS0)\nPOR_SHK2 \u0026lt;- dplyr::select(POR_SHK, chr, midPos, PBS0)\nSFK_GOO2 \u0026lt;- dplyr::select(SFK_GOO, chr, midPos, PBS0)\nSUS_OON2 \u0026lt;- dplyr::select(SUS_OON, chr, midPos, PBS0)\n\nALB_POR \u0026lt;- full_join(ALB_LCA2, POR_SHK2, by=c(\u0026quot;chr\u0026quot;,\u0026quot;midPos\u0026quot;))\nALB_POR_SFK \u0026lt;- full_join(ALB_POR, SFK_GOO2, by=c(\u0026quot;chr\u0026quot;,\u0026quot;midPos\u0026quot;))\nALB_POR_SFK_SUS \u0026lt;- full_join(ALB_POR_SFK, SUS_OON2, by=c(\u0026quot;chr\u0026quot;,\u0026quot;midPos\u0026quot;))\n\ndistplot \u0026lt;- ALB_POR_SFK_SUS\ndistplot$SNP \u0026lt;- seq(1:nrow(distplot) )\n\ndistplot \u0026lt;- dplyr::select(distplot, SNP,chr,midPos,\n     PBS0.x, PBS0.y, PBS0.x.x, PBS0.y.y) %\u0026gt;% \n     set_colnames(.,c(\u0026quot;SNP\u0026quot;,\u0026quot;Chromosome\u0026quot;,\u0026quot;Position\u0026quot;,\u0026quot;ALB\u0026quot;,\u0026quot;POR\u0026quot;,\u0026quot;SFK\u0026quot;,\u0026quot;SUS\u0026quot;))\n\ndistplot \u0026lt;- filter(distplot, !grepl(\u0026quot;scaf\u0026quot;, distplot$Chromosome))\n#distplot \u0026lt;- filter(distplot, !grepl(\u0026quot;hom\u0026quot;, distplot$Chromosome))\ndistplot$Chromosome \u0026lt;- gsub(\u0026quot;Okis\u0026quot;,\u0026quot;\u0026quot;,distplot$Chromosome)\ndistplot[distplot\u0026lt;0] \u0026lt;- 0\n\nCMplot(distplot,type=\u0026quot;p\u0026quot;,plot.type=\u0026quot;m\u0026quot;, multracks=TRUE,\n    LOG10=F,threshold=NULL,memo=\u0026quot;\u0026quot;,\n    ylab=\u0026quot;PBS\u0026quot;,cex=0.6,cex.lab=0.8,\n    file.output=TRUE,file=\u0026quot;pdf\u0026quot;,dpi=300,\n    verbose=TRUE,width=20,height=4,chr.labels.angle=45)\n\n################## FST CMPLOT ##################################################\nALB\\_LCA2 \u0026lt;- dplyr::select(ALB\\_LCA, chr, midPos, Fst01)\nPOR\\_SHK2 \u0026lt;- dplyr::select(POR\\_SHK, chr, midPos, Fst01)\nSFK\\_GOO2 \u0026lt;- dplyr::select(SFK\\_GOO, chr, midPos, Fst01)\nSUS\\_OON2 \u0026lt;- dplyr::select(SUS\\_OON, chr, midPos, Fst01)\n\nALB\\_POR \u0026lt;- full\\_join(ALB\\_LCA2, POR\\_SHK2, by=c(\u0026quot;chr\u0026quot;,\u0026quot;midPos\u0026quot;))\nALB\\_POR\\_SFK \u0026lt;- full\\_join(ALB\\_POR, SFK\\_GOO2, by=c(\u0026quot;chr\u0026quot;,\u0026quot;midPos\u0026quot;))\nALB\\_POR\\_SFK\\_SUS \u0026lt;- full\\_join(ALB\\_POR\\_SFK, SUS\\_OON2, by=c(\u0026quot;chr\u0026quot;,\u0026quot;midPos\u0026quot;))\n\ndistplot \u0026lt;- ALB\\_POR\\_SFK\\_SUS\ndistplot$SNP \u0026lt;- seq(1:nrow(distplot) )\n\ndistplot \u0026lt;- dplyr::select(distplot, SNP,chr,midPos,\n            Fst01.x, Fst01.y, Fst01.x.x, Fst01.y.y ) %\u0026gt;% \n\tset_colnames(.,c(\u0026quot;SNP\u0026quot;,\u0026quot;Chromosome\u0026quot;,\u0026quot;Position\u0026quot; ,\u0026quot;ALB\u0026quot;,\u0026quot;POR\u0026quot;,\u0026quot;SFK\u0026quot;,\u0026quot;SUS\u0026quot;))\n\ndistplot \u0026lt;- filter(distplot, !grepl(\u0026quot;scaf\u0026quot;, distplot$Chromosome))\ndistplot$Chromosome \u0026lt;- gsub(\u0026quot;Okis\u0026quot;,\u0026quot;\u0026quot;,distplot$Chromosome)\ndistplot[distplot\u0026lt;0] \u0026lt;- 0\n\nCMplot(distplot,type=\u0026quot;p\u0026quot;,plot.type=\u0026quot;m\u0026quot;, multracks=TRUE,\n       LOG10=F,threshold=NULL,memo=\u0026quot;\u0026quot;,\n       ylab=\u0026quot;Fst\u0026quot;,cex=0.6,cex.lab=0.8,\n       file.output=TRUE,file=\u0026quot;pdf\u0026quot;,dpi=300,\n       verbose=TRUE,width=20,height=4,chr.labels.angle=45)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003edata.table\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003edplyr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003emagrittr\u003c/span\u003e)\nlibrary(\u003cspan class=\"pl-smi\"\u003eCMplot\u003c/span\u003e)\nrm(\u003cspan class=\"pl-v\"\u003elist\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003els())\n\u003cspan class=\"pl-smi\"\u003eALB_LCA\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ezcat all_ALB_LCA_SAI.OkisLG01.pbs.50.txt.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e1 Thompson\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ePOR_SHK\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ezcat all_POR_SHK_SAI.OkisLG01.pbs.50.txt.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e2 alaska\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eSFK_GOO\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ezcat all_SFK_GOO_SAI.OkisLG01.pbs.50.txt.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e3 alkaska #2\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eSUS_OON\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e fread(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ezcat all_SUS_OON_SAI.OkisLG01.pbs.50.txt.gz\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e4 BC North\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003eALB_LCA\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eALB_LCA\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFst02\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFst12\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enSite\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV3\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePBS1\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePBS2\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ePOR_SHK\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003ePOR_SHK\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFst02\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFst12\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enSite\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV3\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePBS1\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePBS2\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eSFK_GOO\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eSFK_GOO\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFst02\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFst12\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enSite\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV3\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePBS1\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePBS2\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eSUS_OON\u003c/span\u003e  \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eSUS_OON\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFst02\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eFst12\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003enSite\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eV3\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePBS1\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ePBS2\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e## Effectuer un manathan plots:\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eALB_LCA2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eALB_LCA\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003echr\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003emidPos\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ePBS0\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ePOR_SHK2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003ePOR_SHK\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003echr\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003emidPos\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ePBS0\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eSFK_GOO2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eSFK_GOO\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003echr\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003emidPos\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ePBS0\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eSUS_OON2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eSUS_OON\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003echr\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003emidPos\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ePBS0\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eALB_POR\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e full_join(\u003cspan class=\"pl-smi\"\u003eALB_LCA2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ePOR_SHK2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003echr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emidPos\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eALB_POR_SFK\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e full_join(\u003cspan class=\"pl-smi\"\u003eALB_POR\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSFK_GOO2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003echr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emidPos\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eALB_POR_SFK_SUS\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e full_join(\u003cspan class=\"pl-smi\"\u003eALB_POR_SFK\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSUS_OON2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003echr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emidPos\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eALB_POR_SFK_SUS\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e seq(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003enrow(\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e) )\n\n\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSNP\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003echr\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003emidPos\u003c/span\u003e,\n     \u003cspan class=\"pl-smi\"\u003ePBS0.x\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ePBS0.y\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ePBS0.x.x\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ePBS0.y.y\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n     set_colnames(\u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e,c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eChromosome\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePosition\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eALB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePOR\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSFK\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSUS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e filter(\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e!\u003c/span\u003egrepl(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003escaf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eChromosome\u003c/span\u003e))\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003edistplot \u0026lt;- filter(distplot, !grepl(\"hom\", distplot$Chromosome))\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eChromosome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e gsub(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOkis\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eChromosome\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e\n\nCMplot(\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ep\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eplot.type\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003em\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emultracks\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eLOG10\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ethreshold\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eNULL\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ememo\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePBS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.6\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecex.lab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.8\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003efile.output\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edpi\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003echr.labels.angle\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e45\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e################# FST CMPLOT ##################################################\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003eALB\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_LCA2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eALB\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_LCA\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003echr\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003emidPos\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eFst01\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ePOR\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_SHK2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003ePOR\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_SHK\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003echr\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003emidPos\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eFst01\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eSFK\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_GOO2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eSFK\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_GOO\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003echr\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003emidPos\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eFst01\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eSUS\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_OON2\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003eSUS\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_OON\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003echr\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003emidPos\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eFst01\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003eALB\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_POR\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efull\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_join\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eALB\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_LCA2\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ePOR\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_SHK2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003echr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emidPos\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eALB\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_POR\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_SFK\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efull\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_join\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eALB\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_POR\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSFK\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_GOO2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003echr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emidPos\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003eALB\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_POR\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_SFK\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_SUS\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efull\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_join\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003eALB\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_POR\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_SFK\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSUS\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_OON2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eby\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003ec(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003echr\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emidPos\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eALB\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_POR\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_SFK\u003c/span\u003e\\\u003cspan class=\"pl-ii\"\u003e_SUS\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eSNP\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e seq(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003enrow(\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e) )\n\n\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-e\"\u003edplyr\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eselect(\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eSNP\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003echr\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003emidPos\u003c/span\u003e,\n            \u003cspan class=\"pl-smi\"\u003eFst01.x\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eFst01.y\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eFst01.x.x\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eFst01.y.y\u003c/span\u003e ) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n\tset_colnames(\u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e,c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNP\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eChromosome\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePosition\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e ,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eALB\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePOR\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSFK\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSUS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e filter(\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e!\u003c/span\u003egrepl(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003escaf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eChromosome\u003c/span\u003e))\n\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eChromosome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e gsub(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eOkis\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eChromosome\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e\u0026lt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e] \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e\n\nCMplot(\u003cspan class=\"pl-smi\"\u003edistplot\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003etype\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ep\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eplot.type\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003em\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emultracks\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n       \u003cspan class=\"pl-v\"\u003eLOG10\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eF\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ethreshold\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eNULL\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ememo\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n       \u003cspan class=\"pl-v\"\u003eylab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eFst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.6\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecex.lab\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0.8\u003c/span\u003e,\n       \u003cspan class=\"pl-v\"\u003efile.output\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003efile\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003epdf\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edpi\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e300\u003c/span\u003e,\n       \u003cspan class=\"pl-v\"\u003everbose\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ewidth\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e20\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003eheight\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003echr.labels.angle\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e45\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e9. Association between recombination and outliers\u003c/h2\u003e\u003ca id=\"user-content-9-association-between-recombination-and-outliers\" class=\"anchor\" aria-label=\"Permalink: 9. Association between recombination and outliers\" href=\"#9-association-between-recombination-and-outliers\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003e\u003cstrong\u003eestimate recombination\u003c/strong\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eRecombination was estimated using 30X whole Genome data from this \u003ca href=\"https://www.biorxiv.org/content/10.1101/732750v2\" rel=\"nofollow\"\u003epreprint\u003c/a\u003e.\u003cbr\u003e\n(note that the WGS data were removed from the published article and will be published later)\u003cbr\u003e\nLDhat was used with this \u003ca href=\"https://github.com/QuentinRougemont/LDhat_workflow\"\u003epipeline\u003c/a\u003e. To reproduce these estimate simply use the pipeline.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eRho was estimated for each population separately\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThen results were summarized into 250 kb windows and each populations were concatenated together\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe scripts \u003ca href=\"https://github.com/QuentinRougemont/LDhat_workflow/blob/master/02-scripts/06.run_slidingwindows.sh\"\u003ehere\u003c/a\u003e should enables to summarize the data with a bit of modifications depending on your data\u003c/p\u003e\n\u003col start=\"2\" dir=\"auto\"\u003e\n\u003cli\u003e\u003cstrong\u003eidentify outliers\u003c/strong\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eoutliers were identified in the steps above for the RDA\u003c/p\u003e\n\u003col start=\"3\" dir=\"auto\"\u003e\n\u003cli\u003e\u003cstrong\u003etest for how outlier (GEA and Fst/PBS) are influenced by recombination:\u003c/strong\u003e\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003eLook at the script\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"00-scripts/recombination/association_recombination_GEA.R\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e00-scripts/recombination/association_recombination_GEA.R\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThis will allow to perform statistical test:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"* wilcoxon test of difference in mean recombination rate between outliers vs neutral region\n* mixed models test \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e* wilcoxon test of difference in mean recombination rate between outliers vs neutral region\n* mixed models test \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eA plot like the one in FigS15 will be produced.\n\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/FigS15.git.png\"\u003e\u003cimg src=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/FigS15.git.png\" alt=\"example_graph\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHere we only have the population scale recombination rate (rho = 4\u003cem\u003eNe\u003c/em\u003er)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eIn addition it is possible to explore this by considering either:\u003cbr\u003e\n* shared outliers between LFMM \u0026amp; RDA\u003cbr\u003e\n* RDA only outliers\u003cbr\u003e\n* LFMM only outliers\u003cbr\u003e\n* outliers that fall into areas of residual tetraploidy can be investigated separately\nsince there's less variance in recombination in these region\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"There was no difference in recombination when considering the few outliers on region of residual tetraploidy.  \n    Obviously, removing these region increased the strength of the signal   \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eThere was no difference in recombination when considering the few outliers on region of residual tetraploidy.  \n    Obviously, removing these region increased the strength of the signal   \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e10. looking for candidate\u003c/h2\u003e\u003ca id=\"user-content-10-looking-for-candidate\" class=\"anchor\" aria-label=\"Permalink: 10. looking for candidate\" href=\"#10-looking-for-candidate\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eNo GO enrichment instead I only use SNPeff and look for meaningfull outliers.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eSNPeff can be found \u003ca href=\"https://pcingola.github.io/SnpEff/\" rel=\"nofollow\"\u003ehere\u003c/a\u003e and was run with standard parameters\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003e11. Other stuff\u003c/h2\u003e\u003ca id=\"user-content-11-other-stuff\" class=\"anchor\" aria-label=\"Permalink: 11. Other stuff\" href=\"#11-other-stuff\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003ebelow is a zoomable map for BC/Thompson/HaidaGwaii that include all sample sites used here.\u003c/li\u003e\n\u003cli\u003emay be usefull for users\u003c/li\u003e\n\u003cli\u003eThis map was obtained with very simple code in R\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/sampling_map.git.png\"\u003e\u003cimg src=\"https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/sampling_map.git.png\" alt=\"example_graph\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"landscape_genomics_coho","anchor":"landscape_genomics_coho","htmlText":"landscape_genomics_coho"},{"level":1,"text":"Dependancies:","anchor":"dependancies","htmlText":"Dependancies:"},{"level":1,"text":"DATA:","anchor":"data","htmlText":"DATA:"},{"level":1,"text":"Steps :","anchor":"steps-","htmlText":"Steps :"},{"level":2,"text":"1. Aligned Read and Call SNPs","anchor":"1-aligned-read-and-call-snps","htmlText":"1. Aligned Read and Call SNPs"},{"level":2,"text":"2. Filter vcf and perform quality checks","anchor":"2-filter-vcf-and-perform-quality-checks","htmlText":"2. Filter vcf and perform quality checks"},{"level":3,"text":"2.1. Filtering vcf","anchor":"21-filtering-vcf","htmlText":"2.1. Filtering vcf"},{"level":3,"text":"2.2. Quality check","anchor":"22-quality-check","htmlText":"2.2. Quality check"},{"level":2,"text":"3. Compute genetic diversity and plot it","anchor":"3-compute-genetic-diversity-and-plot-it","htmlText":"3. Compute genetic diversity and plot it"},{"level":3,"text":"Plotting diversity","anchor":"plotting-diversity","htmlText":"Plotting diversity"},{"level":2,"text":"4. Perform PCA and VAE analyses","anchor":"4-perform-pca-and-vae-analyses","htmlText":"4. Perform PCA and VAE analyses"},{"level":3,"text":"4a. PCA","anchor":"4a-pca","htmlText":"4a. PCA"},{"level":3,"text":"4b. Running VAE:","anchor":"4b-running-vae","htmlText":"4b. Running VAE:"},{"level":2,"text":"5. GEA Analyses:","anchor":"5-gea-analyses","htmlText":"5. GEA Analyses:"},{"level":2,"text":"5.1. Extract environmental data","anchor":"51-extract-environmental-data","htmlText":"5.1. Extract environmental data"},{"level":2,"text":"5.2. Prepare genomics data","anchor":"52-prepare-genomics-data","htmlText":"5.2. Prepare genomics data"},{"level":2,"text":"5.3. Running the GEAs","anchor":"53-running-the-geas","htmlText":"5.3. Running the GEAs"},{"level":2,"text":"5.3.1 LFMM","anchor":"531-lfmm","htmlText":"5.3.1 LFMM"},{"level":2,"text":"5.3.2 RDA","anchor":"532-rda","htmlText":"5.3.2 RDA"},{"level":2,"text":"5.3.3. Create a RDA Figure and LFMM plot:","anchor":"533-create-a-rda-figure-and-lfmm-plot","htmlText":"5.3.3. Create a RDA Figure and LFMM plot:"},{"level":2,"text":"6. Looking for parallelism","anchor":"6-looking-for-parallelism","htmlText":"6. Looking for parallelism"},{"level":2,"text":"7. plotting LFMM results:","anchor":"7-plotting-lfmm-results","htmlText":"7. plotting LFMM results:"},{"level":2,"text":"8. PBS analyses","anchor":"8-pbs-analyses","htmlText":"8. PBS analyses"},{"level":2,"text":"9. Association between recombination and outliers","anchor":"9-association-between-recombination-and-outliers","htmlText":"9. Association between recombination and outliers"},{"level":2,"text":"10. looking for candidate","anchor":"10-looking-for-candidate","htmlText":"10. looking for candidate"},{"level":2,"text":"11. Other stuff","anchor":"11-other-stuff","htmlText":"11. Other stuff"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2FQuentinRougemont%2Fcoho_ldscp_genomics"}}],"overviewFilesProcessingTime":124.10946,"createFromTemplatePath":"/new?template_name=coho_ldscp_genomics\u0026template_owner=QuentinRougemont"}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":"https://github.dev/","enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":true,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":true}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cLeluw:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cLeluw:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cLeluw[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cLeluw:active{-webkit-transition:none;transition:none;}/*!sc*/
.cLeluw[data-inactive]{cursor:auto;}/*!sc*/
.cLeluw:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled [data-component=ButtonCounter],.cLeluw:disabled [data-component="leadingVisual"],.cLeluw:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cLeluw:focus{outline:solid 1px transparent;}}/*!sc*/
.cLeluw [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cLeluw[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cLeluw[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cLeluw[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cLeluw[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cLeluw[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cLeluw[data-block="block"]{width:100%;}/*!sc*/
.cLeluw[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cLeluw[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cLeluw [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cLeluw [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cLeluw [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cLeluw [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cLeluw:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cLeluw[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cLeluw[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cLeluw:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cLeluw:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cLeluw[data-size="small"][data-no-visuals]{margin-right:8px;height:28px;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,cLeluw,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="main branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        main
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/QuentinRougemont/coho_ldscp_genomics/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/QuentinRougemont/coho_ldscp_genomics/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/QuentinRougemont/coho_ldscp_genomics/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/QuentinRougemont/coho_ldscp_genomics/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <button hidden="">
                  </button>
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" aria-label="Add file" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R2hil5:" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-plus" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="react-directory-remove-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 pyyxt" id=":R2lil5:" tabindex="0" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Add file
                     </span>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </button>
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/QuentinRougemont/coho_ldscp_genomics/commits/main/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           58 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/QuentinRougemont/coho_ldscp_genomics/commits/main/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="00-scripts, (Directory)" class="Link--primary" href="/QuentinRougemont/coho_ldscp_genomics/tree/main/00-scripts" title="00-scripts">
                          00-scripts
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="00-scripts, (Directory)" class="Link--primary" href="/QuentinRougemont/coho_ldscp_genomics/tree/main/00-scripts" title="00-scripts">
                          00-scripts
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="01-info, (Directory)" class="Link--primary" href="/QuentinRougemont/coho_ldscp_genomics/tree/main/01-info" title="01-info">
                          01-info
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="01-info, (Directory)" class="Link--primary" href="/QuentinRougemont/coho_ldscp_genomics/tree/main/01-info" title="01-info">
                          01-info
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="02-data, (Directory)" class="Link--primary" href="/QuentinRougemont/coho_ldscp_genomics/tree/main/02-data" title="02-data">
                          02-data
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="02-data, (Directory)" class="Link--primary" href="/QuentinRougemont/coho_ldscp_genomics/tree/main/02-data" title="02-data">
                          02-data
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="pictures, (Directory)" class="Link--primary" href="/QuentinRougemont/coho_ldscp_genomics/tree/main/pictures" title="pictures">
                          pictures
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="pictures, (Directory)" class="Link--primary" href="/QuentinRougemont/coho_ldscp_genomics/tree/main/pictures" title="pictures">
                          pictures
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/QuentinRougemont/coho_ldscp_genomics/blob/main/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/QuentinRougemont/coho_ldscp_genomics/blob/main/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-5">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/QuentinRougemont/coho_ldscp_genomics/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/QuentinRougemont/coho_ldscp_genomics/blob/main/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-label="Edit file" class="types__StyledButton-sc-ws60qy-0 cLeluw" data-component="IconButton" data-no-visuals="true" data-size="small" title="Edit file" type="button">
                    <svg aria-hidden="true" class="octicon octicon-pencil" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </button>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      landscape_genomics_coho
                     </h1>
                     <a aria-label="Permalink: landscape_genomics_coho" class="anchor" href="#landscape_genomics_coho" id="user-content-landscape_genomics_coho">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     scripts to reproduce our landscape genomics results in coho salmon
                    </p>
                    <p dir="auto">
                     See paper: Long distance migration is a major factor driving local adaptation at continental scale in a Pacific Salmon by Rougemont et al. available on
                     <a href="https://www.biorxiv.org/content/10.1101/2021.11.22.469546v1" rel="nofollow">
                      BiorXiv
                     </a>
                    </p>
                    <p dir="auto">
                     all necessary steps are highlighted below:
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Dependancies:
                     </h1>
                     <a aria-label="Permalink: Dependancies:" class="anchor" href="#dependancies" id="user-content-dependancies">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Alignment and SNP calling:
                     <strong>
                      bwa
                     </strong>
                     software avialble
                     <a href="https://sourceforge.net/projects/bio-bwa/files/" rel="nofollow">
                      here
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      fastp
                     </strong>
                     software availalbe
                     <a href="https://github.com/OpenGene/fastp">
                      here
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Samtools
                     </strong>
                     software available
                     <a href="http://www.htslib.org/" rel="nofollow">
                      here
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      <a href="http://www.htslib.org/download/" rel="nofollow">
                       htslib
                      </a>
                     </strong>
                    </p>
                    <p dir="auto">
                     <strong>
                      cutadapt
                     </strong>
                     <a href="https://github.com/marcelm/cutadapt">
                      cutadapt
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      stacks
                     </strong>
                     available
                     <a href="http://catchenlab.life.illinois.edu/stacks/" rel="nofollow">
                      here
                     </a>
                     (see details of installation on the website)
                    </p>
                    <p dir="auto">
                     for filtration:
                     <strong>
                      bcftools
                     </strong>
                     software available
                     <a href="http://www.htslib.org/download/" rel="nofollow">
                      here
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      vcftools
                     </strong>
                     available
                     <a href="http://vcftools.sourceforge.net/" rel="nofollow">
                      here
                     </a>
                    </p>
                    <p dir="auto">
                     ANGSD, R &amp; python
                    </p>
                    <p dir="auto">
                     I like to use the following fonction in my
                     <strong>
                      .bashrc :
                     </strong>
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="function strata () { cat &lt;(grep &quot;CHR&quot; &quot;$1&quot; ) |\
    cut -f 10- |\                                  
    perl -pe 's/\t/\n/g' | \
              sed 's/_/\t/g' | awk '{print $1&quot;\t&quot;$1&quot;_&quot;$2}' &gt; &quot;$2&quot;; }" dir="auto">
                     <pre><span class="pl-k">function</span> <span class="pl-en">strata</span> () { cat <span class="pl-s"><span class="pl-pds">&lt;(</span>grep <span class="pl-s"><span class="pl-pds">"</span>CHR<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$1</span><span class="pl-pds">"</span></span> <span class="pl-pds">)</span></span> <span class="pl-k">|</span>\
    cut -f 10- <span class="pl-k">|</span><span class="pl-cce">\ </span>                                 
    perl -pe <span class="pl-s"><span class="pl-pds">'</span>s/\t/\n/g<span class="pl-pds">'</span></span> <span class="pl-k">|</span> \
              sed <span class="pl-s"><span class="pl-pds">'</span>s/_/\t/g<span class="pl-pds">'</span></span> <span class="pl-k">|</span> awk <span class="pl-s"><span class="pl-pds">'</span>{print $1"\t"$1"_"$2}<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$2</span><span class="pl-pds">"</span></span><span class="pl-k">;</span> }</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      DATA:
                     </h1>
                     <a aria-label="Permalink: DATA:" class="anchor" href="#data" id="user-content-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     part of the RAW data are deposited on NCBI
                     <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA647050" rel="nofollow">
                      PRJNA647050
                     </a>
                     . The remaining samples should be available soon
                    </p>
                    <p dir="auto">
                     <strong>
                      RAW vcf
                     </strong>
                     and
                     <strong>
                      filtered vcf
                     </strong>
                     have been depostied on dryad (link to come soon)
                    </p>
                    <p dir="auto">
                     The
                     <strong>
                      allele frequency file
                     </strong>
                     used in the RDA will be available on dryad
                    </p>
                    <p dir="auto">
                     Other intermediary file (e.g. LFMM or hierfstat input) can be easily constructed with commands below
                    </p>
                    <p dir="auto">
                     metadata are available for GEA in the folder
                     <code>
                      02-data/env
                     </code>
                    </p>
                    <p dir="auto">
                     additional metadata (individual name, river abbreviation, Latitude, Longitude, region of sampling, etc) can be fond in
                     <code>
                      01-info
                     </code>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Steps :
                     </h1>
                     <a aria-label="Permalink: Steps :" class="anchor" href="#steps-" id="user-content-steps-">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      1. Aligned Read and Call SNPs
                     </h2>
                     <a aria-label="Permalink: 1. Aligned Read and Call SNPs" class="anchor" href="#1-aligned-read-and-call-snps" id="user-content-1-aligned-read-and-call-snps">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      Use
                      <a href="https://github.com/QuentinRougemont/stacks_v2_workflow">
                       stacks_workflow
                      </a>
                      to perform the following:
                      <ul dir="auto">
                       <li>
                        demultiplexing and trimming
                       </li>
                       <li>
                        align read,
                       </li>
                       <li>
                        run stacks
                       </li>
                       <li>
                        (filter vcf)
                       </li>
                      </ul>
                     </li>
                    </ul>
                    <p dir="auto">
                     note: I don't update stacks workflow any more and recommand to use the one of Eric Normandeau available
                     <a href="https://github.com/enormandeau/stacks_workflow">
                      here
                     </a>
                    </p>
                    <ul dir="auto">
                     <li>
                      Use ANSGD to produce likelihood file (used to compute Fst and PBS latter)
                     </li>
                    </ul>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      2. Filter vcf and perform quality checks
                     </h2>
                     <a aria-label="Permalink: 2. Filter vcf and perform quality checks" class="anchor" href="#2-filter-vcf-and-perform-quality-checks" id="user-content-2-filter-vcf-and-perform-quality-checks">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     To reproduce the filtering you can access the raw vcf from dryad
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      2.1. Filtering vcf
                     </h3>
                     <a aria-label="Permalink: 2.1. Filtering vcf" class="anchor" href="#21-filtering-vcf" id="user-content-21-filtering-vcf">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     use mostly vcftools and bash to filter vcf.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
############################# Filtering RAW VCF ###########################

# in stacks populations was run requiring:
# SNP Genotyped in : &gt;70% of the samples
#                    &gt;70% of the population


vcftools \
      --gzvcf populations.snps.vcf.gz
        --remove blacklisted_individuals.txt
        --recode-INFO-all
        --mac 1
        --max-meanDP 120
        --min-meanDP 10
        --out populations.DP10_120
        --recode


vcftools \
        --vcf populations.10_120.recode.vcf
        --recode-INFO-all
        --max-missing 0.85
        --out populations.10_120.miss85
        --recode  

# compute allele frequency and keep one SNP per loci:  

input=&quot;populations.10_120.miss85.recode.vcf&quot;  

vcftools \
        --vcf $input\
        --freq --out out  

sed 1d out.frq | 's/:/\t/g' |cut -f 1,2,7,8 out.frq &gt; frq2  

#then I use a horrible bash script to create a whitelist of independant SNPs. 
#A script in Eric Normandeau's [workflow](https://github.com/enormandeau/stacks_workflow/blob/master/00-scripts/utility_scripts/extract_snp_with_max_maf.py) should do the same more efficiently.

#compute missing rate per individuals  
vcftools --vcf $input --missing-indv --out individuals

#we remove individuals with too many missing data with awk
awk '$5&gt;0.20 {print $1}'  individuals.imiss &gt; blacklisted.individuals.txt

#remove individuals with too much missing data and keep one SNP per locus:  
vcftools \
     --vcf $input\
        --remove blacklisted.individuals.txt\
        --recode-INFO-all\
	--mac 1 \
        --out populations.mac1\
        --recode\
        --snps snp_id.txt\

#then perform a last quality filter:
vcftools \
    --vcf populations.mac1.recode.vcf \
    --max-missing 0.95 \
    --out populations.mac1miss0.95\
    --min-meanDP 10 --max-meanDP 120\
    --recode \
    --recode-INFO-all

#We don't want to filter on HWE since we may be interested in finding markers under selection 
#Moreover a global HWE filter is meaningless in highly structured populations

strata populations.mac1miss0.95.recode.vcf
bgzip populations.mac1miss0.95.recode.vcf; tabix -p vcf populations.mac1miss0.95.recode.vcf.gz

#then we need to rename part of the population HOP into HOD:
cut -f 1 strata.txt |sed '2576,2607s/HOP/HOD/g' &gt; new.ind.tmp
cut -f 1 strata.txt &gt; ind.tmp

paste ind.tmp new.ind.tmp &gt; new_sample_name.txt
rm *tmp 

input=populations.mac1miss0.95.recode.vcf.gz

#then we rename individuals in the vcf 
bcftools reheader --samples new_sample_name.txt -o ${input%.recode.vcf.gz}.renamed.vcf.gz $input

#this is our final vcf ready for estimating genetic diversity and structure 
#see below for filtering for GEA
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>############################ Filtering RAW VCF ###########################</span>

<span class="pl-c"><span class="pl-c">#</span> in stacks populations was run requiring:</span>
<span class="pl-c"><span class="pl-c">#</span> SNP Genotyped in : &gt;70% of the samples</span>
<span class="pl-c"><span class="pl-c">#</span>                    &gt;70% of the population</span>


vcftools \
      --gzvcf populations.snps.vcf.gz
        --remove blacklisted_individuals.txt
        --recode-INFO-all
        --mac 1
        --max-meanDP 120
        --min-meanDP 10
        --out populations.DP10_120
        --recode


vcftools \
        --vcf populations.10_120.recode.vcf
        --recode-INFO-all
        --max-missing 0.85
        --out populations.10_120.miss85
        --recode  

<span class="pl-c"><span class="pl-c">#</span> compute allele frequency and keep one SNP per loci:  </span>

input=<span class="pl-s"><span class="pl-pds">"</span>populations.10_120.miss85.recode.vcf<span class="pl-pds">"</span></span>  

vcftools \
        --vcf <span class="pl-smi">$input</span>\
        --freq --out out  

sed 1d out.frq <span class="pl-k">|</span> <span class="pl-s"><span class="pl-pds">'</span>s/:/\t/g<span class="pl-pds">'</span></span> <span class="pl-k">|</span>cut -f 1,2,7,8 out.frq <span class="pl-k">&gt;</span> frq2  

<span class="pl-c"><span class="pl-c">#</span>then I use a horrible bash script to create a whitelist of independant SNPs. </span>
<span class="pl-c"><span class="pl-c">#</span>A script in Eric Normandeau's [workflow](https://github.com/enormandeau/stacks_workflow/blob/master/00-scripts/utility_scripts/extract_snp_with_max_maf.py) should do the same more efficiently.</span>

<span class="pl-c"><span class="pl-c">#</span>compute missing rate per individuals  </span>
vcftools --vcf <span class="pl-smi">$input</span> --missing-indv --out individuals

<span class="pl-c"><span class="pl-c">#</span>we remove individuals with too many missing data with awk</span>
awk <span class="pl-s"><span class="pl-pds">'</span>$5&gt;0.20 {print $1}<span class="pl-pds">'</span></span>  individuals.imiss <span class="pl-k">&gt;</span> blacklisted.individuals.txt

<span class="pl-c"><span class="pl-c">#</span>remove individuals with too much missing data and keep one SNP per locus:  </span>
vcftools \
     --vcf <span class="pl-smi">$input</span>\
        --remove blacklisted.individuals.txt\
        --recode-INFO-all\
	--mac 1 \
        --out populations.mac1\
        --recode\
        --snps snp_id.txt\

<span class="pl-c"><span class="pl-c">#</span>then perform a last quality filter:</span>
vcftools \
    --vcf populations.mac1.recode.vcf \
    --max-missing 0.95 \
    --out populations.mac1miss0.95\
    --min-meanDP 10 --max-meanDP 120\
    --recode \
    --recode-INFO-all

<span class="pl-c"><span class="pl-c">#</span>We don't want to filter on HWE since we may be interested in finding markers under selection </span>
<span class="pl-c"><span class="pl-c">#</span>Moreover a global HWE filter is meaningless in highly structured populations</span>

strata populations.mac1miss0.95.recode.vcf
bgzip populations.mac1miss0.95.recode.vcf<span class="pl-k">;</span> tabix -p vcf populations.mac1miss0.95.recode.vcf.gz

<span class="pl-c"><span class="pl-c">#</span>then we need to rename part of the population HOP into HOD:</span>
cut -f 1 strata.txt <span class="pl-k">|</span>sed <span class="pl-s"><span class="pl-pds">'</span>2576,2607s/HOP/HOD/g<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> new.ind.tmp
cut -f 1 strata.txt <span class="pl-k">&gt;</span> ind.tmp

paste ind.tmp new.ind.tmp <span class="pl-k">&gt;</span> new_sample_name.txt
rm <span class="pl-k">*</span>tmp 

input=populations.mac1miss0.95.recode.vcf.gz

<span class="pl-c"><span class="pl-c">#</span>then we rename individuals in the vcf </span>
bcftools reheader --samples new_sample_name.txt -o <span class="pl-smi">${input<span class="pl-k">%</span>.recode.vcf.gz}</span>.renamed.vcf.gz <span class="pl-smi">$input</span>

<span class="pl-c"><span class="pl-c">#</span>this is our final vcf ready for estimating genetic diversity and structure </span>
<span class="pl-c"><span class="pl-c">#</span>see below for filtering for GEA</span>
</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      2.2. Quality check
                     </h3>
                     <a aria-label="Permalink: 2.2. Quality check" class="anchor" href="#22-quality-check" id="user-content-22-quality-check">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     working on the same cleaned file (populations.mac1miss0.95.renamed.vcf)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="* compute depth of sequencing, missing rate, genotyping rate  

use:
	* vcftools site-mean-depth 
	* vcftools missing-indv  


* verifiy the absence of a pattern in missing data :">
                     <pre class="notranslate"><code>* compute depth of sequencing, missing rate, genotyping rate  

use:
	* vcftools site-mean-depth 
	* vcftools missing-indv  


* verifiy the absence of a pattern in missing data :
</code></pre>
                    </div>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="plink --file populations.mac1miss0.95.renamed.vcf \
    --allow-extra-chr \
    --mds-plot 4  --cluster missing  \
    --out ibm_mds_to_plot" dir="auto">
                     <pre>plink --file populations.mac1miss0.95.renamed.vcf \
    --allow-extra-chr \
    --mds-plot 4  --cluster missing  \
    --out ibm_mds_to_plot</pre>
                    </div>
                    <p dir="auto">
                     then plot it using:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Rscript ./00-scripts/quality_check/plot_mds.R" dir="auto">
                     <pre><span class="pl-smi">Rscript</span> <span class="pl-smi">.</span><span class="pl-k">/</span><span class="pl-c1">00</span><span class="pl-k">-</span><span class="pl-smi">scripts</span><span class="pl-k">/</span><span class="pl-smi">quality_check</span><span class="pl-k">/</span><span class="pl-smi">plot_mds.R</span></pre>
                    </div>
                    <p dir="auto">
                     we obtained the following plot:
                     <a href="https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/ibm_mds_point.git.png" rel="noopener noreferrer" target="_blank">
                      <img alt="example_graph" src="https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/ibm_mds_point.git.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     we see that missing data are low and that these missing data do not show a particular pattern of structure.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      3. Compute genetic diversity and plot it
                     </h2>
                     <a aria-label="Permalink: 3. Compute genetic diversity and plot it" class="anchor" href="#3-compute-genetic-diversity-and-plot-it" id="user-content-3-compute-genetic-diversity-and-plot-it">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     first convert vcf into hierfstat and other usefull input
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="vcf=$1 #vcf here

#check compression
if file --mime-type &quot;$vcf&quot; | grep -q gzip$; then
   echo &quot;$vcf is gzipped&quot;
   gunzip &quot;$vcf&quot;
   INPUT=${vcf%}.gz
else
   echo &quot;$vcf is not gzipped&quot;
   INPUT=$vcf
fi

#extract population and individuals
strata $INPUT strata.txt ; cut -f 1 strata.txt &gt; pop.tmp

vcf2geno $INPUT
geno2lfmm ${INPUT%.vcf}.geno
snp=${INPUT%.vcf}.vcfsnp
INPUT=${INPUT%.vcf}.geno

sed -e 's/2/22/g' $INPUT |\
    sed -e 's/9/NA/g' -e 's/1/12/g' -e 's/0/11/g' &gt; out.tmp
paste pop out.tmp &gt; hierfstat.data.tmp
echo &quot;loc&quot; &gt; loc.tmp

cut -d &quot; &quot; -f 3 $snp |perl -pe &quot;s/\n/\t/g&quot; &gt; snp_id.tmp
paste loc.tmp snp_id.tmp &gt; loc.tmp
cat loc.tmp hierfstat.data.tmp &gt; hierfstat.data.txt
rm \*tmp
" dir="auto">
                     <pre>vcf=<span class="pl-smi">$1</span> <span class="pl-c"><span class="pl-c">#</span>vcf here</span>

<span class="pl-c"><span class="pl-c">#</span>check compression</span>
<span class="pl-k">if</span> file --mime-type <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$vcf</span><span class="pl-pds">"</span></span> <span class="pl-k">|</span> grep -q gzip$<span class="pl-k">;</span> <span class="pl-k">then</span>
   <span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$vcf</span> is gzipped<span class="pl-pds">"</span></span>
   gunzip <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$vcf</span><span class="pl-pds">"</span></span>
   INPUT=<span class="pl-smi">${vcf<span class="pl-k">%</span>}</span>.gz
<span class="pl-k">else</span>
   <span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">$vcf</span> is not gzipped<span class="pl-pds">"</span></span>
   INPUT=<span class="pl-smi">$vcf</span>
<span class="pl-k">fi</span>

<span class="pl-c"><span class="pl-c">#</span>extract population and individuals</span>
strata <span class="pl-smi">$INPUT</span> strata.txt <span class="pl-k">;</span> cut -f 1 strata.txt <span class="pl-k">&gt;</span> pop.tmp

vcf2geno <span class="pl-smi">$INPUT</span>
geno2lfmm <span class="pl-smi">${INPUT<span class="pl-k">%</span>.vcf}</span>.geno
snp=<span class="pl-smi">${INPUT<span class="pl-k">%</span>.vcf}</span>.vcfsnp
INPUT=<span class="pl-smi">${INPUT<span class="pl-k">%</span>.vcf}</span>.geno

sed -e <span class="pl-s"><span class="pl-pds">'</span>s/2/22/g<span class="pl-pds">'</span></span> <span class="pl-smi">$INPUT</span> <span class="pl-k">|</span>\
    sed -e <span class="pl-s"><span class="pl-pds">'</span>s/9/NA/g<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>s/1/12/g<span class="pl-pds">'</span></span> -e <span class="pl-s"><span class="pl-pds">'</span>s/0/11/g<span class="pl-pds">'</span></span> <span class="pl-k">&gt;</span> out.tmp
paste pop out.tmp <span class="pl-k">&gt;</span> hierfstat.data.tmp
<span class="pl-c1">echo</span> <span class="pl-s"><span class="pl-pds">"</span>loc<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> loc.tmp

cut -d <span class="pl-s"><span class="pl-pds">"</span> <span class="pl-pds">"</span></span> -f 3 <span class="pl-smi">$snp</span> <span class="pl-k">|</span>perl -pe <span class="pl-s"><span class="pl-pds">"</span>s/\n/\t/g<span class="pl-pds">"</span></span> <span class="pl-k">&gt;</span> snp_id.tmp
paste loc.tmp snp_id.tmp <span class="pl-k">&gt;</span> loc.tmp
cat loc.tmp hierfstat.data.tmp <span class="pl-k">&gt;</span> hierfstat.data.txt
rm <span class="pl-cce">\*</span>tmp
</pre>
                    </div>
                    <p dir="auto">
                     then use R to compute basic statistics (Hs, Ho, Fis, Bst, in hierfstat)
                    </p>
                    <p dir="auto">
                     the scripts:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="00-scripts/diversity/00.vcf2hierfstats.sh
00-scripts/diversity/01.hierfstats.R">
                     <pre class="notranslate"><code>00-scripts/diversity/00.vcf2hierfstats.sh
00-scripts/diversity/01.hierfstats.R
</code></pre>
                    </div>
                    <p dir="auto">
                     should procude all the results.
                    </p>
                    <p dir="auto">
                     The second script run on a cluster with &gt;50Gb of RAM  :
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Rscript ./00-scripts/diversity/01.hierfstats.R" dir="auto">
                     <pre><span class="pl-smi">Rscript</span> <span class="pl-smi">.</span><span class="pl-k">/</span><span class="pl-c1">00</span><span class="pl-k">-</span><span class="pl-smi">scripts</span><span class="pl-k">/</span><span class="pl-smi">diversity</span><span class="pl-k">/</span><span class="pl-c1">01.</span><span class="pl-smi">hierfstats.R</span></pre>
                    </div>
                    <p dir="auto">
                     Then we will perform plots of the correlation between the distance to the southernmost site and Bst and Hs statistics.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      Plotting diversity
                     </h3>
                     <a aria-label="Permalink: Plotting diversity" class="anchor" href="#plotting-diversity" id="user-content-plotting-diversity">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Run:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Rscript ./00-scripts/diversity/02.plot_ho_betaST_distance.R" dir="auto">
                     <pre><span class="pl-smi">Rscript</span> <span class="pl-smi">.</span><span class="pl-k">/</span><span class="pl-c1">00</span><span class="pl-k">-</span><span class="pl-smi">scripts</span><span class="pl-k">/</span><span class="pl-smi">diversity</span><span class="pl-k">/</span><span class="pl-c1">02.</span><span class="pl-smi">plot_ho_betaST_distance.R</span></pre>
                    </div>
                    <p dir="auto">
                     on the command line or alternatively in Rstudio,
                     <br/>
                     this will test for correlation among diversity and distance and produce the following graph:
                     <a href="https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/figureS07.git.png" rel="noopener noreferrer" target="_blank">
                      <img alt="example_graph" src="https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/figureS07.git.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      4. Perform PCA and VAE analyses
                     </h2>
                     <a aria-label="Permalink: 4. Perform PCA and VAE analyses" class="anchor" href="#4-perform-pca-and-vae-analyses" id="user-content-4-perform-pca-and-vae-analyses">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4a. PCA
                     </h3>
                     <a aria-label="Permalink: 4a. PCA" class="anchor" href="#4a-pca" id="user-content-4a-pca">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     PCA was performed on individuals using
                     <a href="https://github.com/QuentinRougemont/utility_scripts/blob/master/02.PCA/pca_on_vcf.R">
                      this script
                     </a>
                     and populations allele frequency using
                     <a href="https://github.com/QuentinRougemont/utility_scripts/blob/master/02.PCA/pca_on_freq.R">
                      this script
                     </a>
                     .
I also made an IBS plot using plink with this
                     <a href="https://github.com/QuentinRougemont/utility_scripts/blob/master/07.random_scripts/plink_cluster_IBS.sh">
                      simple script
                     </a>
                     .
                    </p>
                    <p dir="auto">
                     To plot and fit to the Coho salmon data here's a simple modification of the script.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
file=&quot;plink.frq.strat.renamed.gz&quot;

#load libs
libs &lt;- c('dplyr','reshape2','ade4','data.table', 'factoextra', 'magrittr','ggsci')
invisible(lapply(libs, library, character.only = TRUE))

file &lt;- paste0(&quot;zcat &quot;,file) 
freq &lt;- fread(file)
freq2 &lt;- dplyr::select(freq,SNP,CLST,MAF) 
freq3 &lt;- reshape2::dcast(freq2,SNP~CLST)
freq3 &lt;- freq3[,-1] 
freq4 &lt;- t(freq3)
pop &lt;- unique(freq$CLST)   
pop &lt;- data.frame(pop) %&gt;% set_colnames(.,c(&quot;POP&quot;))


#perform PCA
pca1 &lt;- dudi.pca(freq4,scale=FALSE,scannf=FALSE)

#load strata
strata &lt;- read.table(&quot;pop_ind_region_latitute_corrected3.txt&quot;) 
strata &lt;- select(strata, V2,V6) %&gt;% 
    set_colnames(.,c(&quot;POP&quot;,&quot;REGION&quot;))
strata &lt;- unique(strata)
strata &lt;- strata[order(strata$POP),]  #reorder because population DRA has been renamed into WAC

#ensure that the order in strata will match order of the freq file:
strata = left_join(pop, strata)

#personalizing the colors:
myColors &lt;- c(&quot;blue&quot;,&quot;orange&quot;,&quot;red&quot;,&quot;darkviolet&quot;,&quot;chocolate4&quot;, &quot;springgreen4&quot;,&quot;green&quot;)
names(myColors) &lt;- levels(strata$REGION)
colScale &lt;- scale_colour_manual(name = &quot;REGION&quot;,values = myColors)

#plot with river label:
p &lt;- fviz_pca_ind(pca1, label=&quot;none&quot;, pointsize = 0.0) +
    geom_text(aes(label=strata$POP, 
        colour=factor(strata$REGION)),
        size = 2 )
#p &lt;- p + theme_minimal() + theme(legend.position = &quot;none&quot;)  + colScale
#p &lt;- p + scale_color_igv() + theme_minimal() + theme(legend.position = &quot;none&quot;)
p &lt;- p +theme(legend.text = element_text(size = 10, face = &quot;bold&quot;)) +
    theme(legend.justification=c(1,0), legend.position=c(1,0)) + 
    colScale + 
    ggtitle(&quot;PCA on allele frequencies&quot;) + 
     theme(plot.title = element_text(lineheight=.8, face=&quot;bold&quot; , hjust = 0.5))

    
pdf(file=&quot;pca_on_freq_population.pdf&quot;)
p
dev.off()

" dir="auto">
                     <pre><span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>plink.frq.strat.renamed.gz<span class="pl-pds">"</span></span>

<span class="pl-c"><span class="pl-c">#</span>load libs</span>
<span class="pl-smi">libs</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">'</span>dplyr<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>reshape2<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>ade4<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>data.table<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>factoextra<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>magrittr<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>ggsci<span class="pl-pds">'</span></span>)
<span class="pl-k">invisible</span>(lapply(<span class="pl-smi">libs</span>, <span class="pl-smi">library</span>, <span class="pl-v">character.only</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>))

<span class="pl-smi">file</span> <span class="pl-k">&lt;-</span> paste0(<span class="pl-s"><span class="pl-pds">"</span>zcat <span class="pl-pds">"</span></span>,<span class="pl-smi">file</span>) 
<span class="pl-smi">freq</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-smi">file</span>)
<span class="pl-smi">freq2</span> <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">freq</span>,<span class="pl-smi">SNP</span>,<span class="pl-smi">CLST</span>,<span class="pl-smi">MAF</span>) 
<span class="pl-smi">freq3</span> <span class="pl-k">&lt;-</span> <span class="pl-e">reshape2</span><span class="pl-k">::</span>dcast(<span class="pl-smi">freq2</span>,<span class="pl-smi">SNP</span><span class="pl-k">~</span><span class="pl-smi">CLST</span>)
<span class="pl-smi">freq3</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">freq3</span>[,<span class="pl-k">-</span><span class="pl-c1">1</span>] 
<span class="pl-smi">freq4</span> <span class="pl-k">&lt;-</span> t(<span class="pl-smi">freq3</span>)
<span class="pl-smi">pop</span> <span class="pl-k">&lt;-</span> unique(<span class="pl-smi">freq</span><span class="pl-k">$</span><span class="pl-smi">CLST</span>)   
<span class="pl-smi">pop</span> <span class="pl-k">&lt;-</span> <span class="pl-k">data.frame</span>(<span class="pl-smi">pop</span>) %<span class="pl-k">&gt;</span>% set_colnames(<span class="pl-smi">.</span>,c(<span class="pl-s"><span class="pl-pds">"</span>POP<span class="pl-pds">"</span></span>))


<span class="pl-c"><span class="pl-c">#</span>perform PCA</span>
<span class="pl-smi">pca1</span> <span class="pl-k">&lt;-</span> dudi.pca(<span class="pl-smi">freq4</span>,<span class="pl-v">scale</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>,<span class="pl-v">scannf</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>)

<span class="pl-c"><span class="pl-c">#</span>load strata</span>
<span class="pl-smi">strata</span> <span class="pl-k">&lt;-</span> read.table(<span class="pl-s"><span class="pl-pds">"</span>pop_ind_region_latitute_corrected3.txt<span class="pl-pds">"</span></span>) 
<span class="pl-smi">strata</span> <span class="pl-k">&lt;-</span> select(<span class="pl-smi">strata</span>, <span class="pl-smi">V2</span>,<span class="pl-smi">V6</span>) %<span class="pl-k">&gt;</span>% 
    set_colnames(<span class="pl-smi">.</span>,c(<span class="pl-s"><span class="pl-pds">"</span>POP<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>REGION<span class="pl-pds">"</span></span>))
<span class="pl-smi">strata</span> <span class="pl-k">&lt;-</span> unique(<span class="pl-smi">strata</span>)
<span class="pl-smi">strata</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">strata</span>[order(<span class="pl-smi">strata</span><span class="pl-k">$</span><span class="pl-smi">POP</span>),]  <span class="pl-c"><span class="pl-c">#</span>reorder because population DRA has been renamed into WAC</span>

<span class="pl-c"><span class="pl-c">#</span>ensure that the order in strata will match order of the freq file:</span>
<span class="pl-v">strata</span> <span class="pl-k">=</span> left_join(<span class="pl-smi">pop</span>, <span class="pl-smi">strata</span>)

<span class="pl-c"><span class="pl-c">#</span>personalizing the colors:</span>
<span class="pl-smi">myColors</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>blue<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>orange<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>darkviolet<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>chocolate4<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>springgreen4<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>green<span class="pl-pds">"</span></span>)
names(<span class="pl-smi">myColors</span>) <span class="pl-k">&lt;-</span> levels(<span class="pl-smi">strata</span><span class="pl-k">$</span><span class="pl-smi">REGION</span>)
<span class="pl-smi">colScale</span> <span class="pl-k">&lt;-</span> scale_colour_manual(<span class="pl-v">name</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>REGION<span class="pl-pds">"</span></span>,<span class="pl-v">values</span> <span class="pl-k">=</span> <span class="pl-smi">myColors</span>)

<span class="pl-c"><span class="pl-c">#</span>plot with river label:</span>
<span class="pl-smi">p</span> <span class="pl-k">&lt;-</span> fviz_pca_ind(<span class="pl-smi">pca1</span>, <span class="pl-v">label</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>, <span class="pl-v">pointsize</span> <span class="pl-k">=</span> <span class="pl-c1">0.0</span>) <span class="pl-k">+</span>
    geom_text(aes(<span class="pl-v">label</span><span class="pl-k">=</span><span class="pl-smi">strata</span><span class="pl-k">$</span><span class="pl-smi">POP</span>, 
        <span class="pl-v">colour</span><span class="pl-k">=</span><span class="pl-k">factor</span>(<span class="pl-smi">strata</span><span class="pl-k">$</span><span class="pl-smi">REGION</span>)),
        <span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">2</span> )
<span class="pl-c"><span class="pl-c">#</span>p &lt;- p + theme_minimal() + theme(legend.position = "none")  + colScale</span>
<span class="pl-c"><span class="pl-c">#</span>p &lt;- p + scale_color_igv() + theme_minimal() + theme(legend.position = "none")</span>
<span class="pl-smi">p</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">p</span> <span class="pl-k">+</span>theme(<span class="pl-v">legend.text</span> <span class="pl-k">=</span> element_text(<span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">10</span>, <span class="pl-v">face</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>bold<span class="pl-pds">"</span></span>)) <span class="pl-k">+</span>
    theme(<span class="pl-v">legend.justification</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">0</span>), <span class="pl-v">legend.position</span><span class="pl-k">=</span>c(<span class="pl-c1">1</span>,<span class="pl-c1">0</span>)) <span class="pl-k">+</span> 
    <span class="pl-smi">colScale</span> <span class="pl-k">+</span> 
    ggtitle(<span class="pl-s"><span class="pl-pds">"</span>PCA on allele frequencies<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> 
     theme(<span class="pl-v">plot.title</span> <span class="pl-k">=</span> element_text(<span class="pl-v">lineheight</span><span class="pl-k">=</span><span class="pl-smi">.8</span>, <span class="pl-v">face</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>bold<span class="pl-pds">"</span></span> , <span class="pl-v">hjust</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>))

    
pdf(<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>pca_on_freq_population.pdf<span class="pl-pds">"</span></span>)
<span class="pl-smi">p</span>
dev.off()

</pre>
                    </div>
                    <p dir="auto">
                     you should obtained an image as follows
                     <a href="https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/pca_on_freq_population.git.png" rel="noopener noreferrer" target="_blank">
                      <img alt="example_graph" src="https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/pca_on_freq_population.git.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     in addition it is possible to test for population structure using a PCA at the individual level or an IBS plot using a similar command to the one use for IBM:
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="plink --file  populations.mac1miss0.95.renamed.vcf \
    --allow-extra-chr \
    --mds-plot 4  --cluster\
    --out mds_to_plot" dir="auto">
                     <pre>plink --file  populations.mac1miss0.95.renamed.vcf \
    --allow-extra-chr \
    --mds-plot 4  --cluster\
    --out mds_to_plot</pre>
                    </div>
                    <p dir="auto">
                     Run this very simple script (similar to the one used for IBM):
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Rscript 00-scripts/pca_ibs/plot_mds.R" dir="auto">
                     <pre><span class="pl-smi">Rscript</span> <span class="pl-c1">00</span><span class="pl-k">-</span><span class="pl-smi">scripts</span><span class="pl-k">/</span><span class="pl-smi">pca_ibs</span><span class="pl-k">/</span><span class="pl-smi">plot_mds.R</span></pre>
                    </div>
                    <p dir="auto">
                     then you'll obtain the following graph:
                     <a href="https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/mds_point.git.png" rel="noopener noreferrer" target="_blank">
                      <img alt="example_graph" src="https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/mds_point.git.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     This highlights particularly well the continuity of our data, where the axis of a PCA are highly correlated with latitude and longitude (see our paper and our previous one).
                     <br/>
                     Interestingly, we see that the Russian samples falls within Alaska and is not much divergent.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h3 class="heading-element" dir="auto" tabindex="-1">
                      4b. Running VAE:
                     </h3>
                     <a aria-label="Permalink: 4b. Running VAE:" class="anchor" href="#4b-running-vae" id="user-content-4b-running-vae">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Install vae available
                     <a href="https://github.com/kr-colab/popvae">
                      here
                     </a>
                     .
Due to its stochasticity, you'll obtained slightly different results than I did.
For instance depanding on the parameters used, you may obtained a distinction of nearly all river in separate cluster (Figure 2 of our paper)
In other case, you'll obtained a continuous distribution of the genotype entierely correlated with latitude.
These two results reflect the existence of fine scale structure (recent postglacial homing) and of strong IBD following the postglacial expansion
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
#install VAE and activate
#install tensorflow as well!

#create a working directory:
mkdir VAE
cd VAE
#copy the vcf 
ln -s ../02-data/populations.mac1miss0.95.renamed.vcf.gz .
# create folder to store the results
mkdir results

# run popvae with default parameters
popvae.py --infile populations.mac1miss0.95.renamed.vcf.gz --out results/coho --seed 12345

# more advanced use:
mkdir results2
popvae.py --infile  populations.mac1miss0.95.renamed.vcf.gz\
    --out results2/coho \ 
    --seed 12345 \
    --search_network_sizes \ 
    --train_prop 0.90

" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span>install VAE and activate</span>
<span class="pl-c"><span class="pl-c">#</span>install tensorflow as well!</span>

<span class="pl-c"><span class="pl-c">#</span>create a working directory:</span>
mkdir VAE
<span class="pl-c1">cd</span> VAE
<span class="pl-c"><span class="pl-c">#</span>copy the vcf </span>
ln -s ../02-data/populations.mac1miss0.95.renamed.vcf.gz <span class="pl-c1">.</span>
<span class="pl-c"><span class="pl-c">#</span> create folder to store the results</span>
mkdir results

<span class="pl-c"><span class="pl-c">#</span> run popvae with default parameters</span>
popvae.py --infile populations.mac1miss0.95.renamed.vcf.gz --out results/coho --seed 12345

<span class="pl-c"><span class="pl-c">#</span> more advanced use:</span>
mkdir results2
popvae.py --infile  populations.mac1miss0.95.renamed.vcf.gz\
    --out results2/coho <span class="pl-cce">\ </span>
    --seed 12345 \
    --search_network_sizes <span class="pl-cce">\ </span>
    --train_prop 0.90

</pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      advice
                     </strong>
                     : try different run to see how results can differ
                    </p>
                    <p dir="auto">
                     Then plot the results in R. This bits of code should to the work:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(dplyr);library(cowplot);library(ggplot2);library(tidyr);library(ggsci)

pd &lt;- read.table("coho_latent_coords.txt",T)
names(pd)[1:2] &lt;- c("LD1","LD2")
meta &lt;- read.table("01-info/metadata",T, sep="\t" ) %&gt;% select(POP, Latitude, Longitude, Region)
pd1 &lt;- separate(pd,sampleID,c("POP","ind"),"_")
pd &lt;- merge(pd1, meta ) #.x="POP",by.y="POP_ID")

############################################################################################### 
# test correlation  with geography:
summary(lm(pd$LD1 ~ pd$Latitude))
summary(lm(pd$LD2 ~ pd$Latitude))
summary(lm(pd$LD2 ~ pd$Longitude))
summary(lm(pd$LD1 ~ pd$Longitude))
summary(lm(pd$LD1 ~ pd$Latitude + pd$Longitude ))
summary(lm(pd$LD1 ~ pd$Latitude + pd$Longitude + pd$Latitude * pd$Longitude ))

# set colors:
myColors &lt;- c("blue","orange","red","darkviolet","chocolate4", "springgreen4","green")
names(myColors) &lt;- levels(pd$Region)
colScale &lt;- scale_colour_manual(name = "Region",values = myColors)

# plot the 2 ld
p &lt;- pd %&gt;% 
  dplyr::rename_at("POP",~"river") %&gt;%
  #separate(REGION2,c("Rivers",NA), sep ="-" ) #)
  #ggplot(aes(x=LD1,y=LD2, col= river))+
  ggplot(aes(x = LD1, y = LD2, col = Region))+
  colScale +
  geom_point(size=1) + theme_classic()

haida &lt;- dplyr::filter(pd, Region=="HaidaGwaii") %&gt;% 
  dplyr::rename_at("POP",~"river") %&gt;%
  ggplot(aes(x=LD1,y=LD2, col=river))+ #labs(title = "B") +
  geom_text(aes ( label= river), size=3.5) + theme_classic() +
  theme(legend.position = "none")  
h &lt;- haida + scale_color_igv()

all &lt;- ggdraw(p + theme_half_open(12)) +
    draw_plot(h , .62, .62, .35, .35) +
        draw_plot_label(
         c("A", "B"),
         c(0, 0.45),
         c(1, 0.95),
         size = 12)

pdf(file="vae_LD1_LD2.pdf")
all 
dev.off()
' dir="auto">
                     <pre>library(<span class="pl-smi">dplyr</span>);library(<span class="pl-smi">cowplot</span>);library(<span class="pl-smi">ggplot2</span>);library(<span class="pl-smi">tidyr</span>);library(<span class="pl-smi">ggsci</span>)

<span class="pl-smi">pd</span> <span class="pl-k">&lt;-</span> read.table(<span class="pl-s"><span class="pl-pds">"</span>coho_latent_coords.txt<span class="pl-pds">"</span></span>,<span class="pl-c1">T</span>)
names(<span class="pl-smi">pd</span>)[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">2</span>] <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>LD1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>LD2<span class="pl-pds">"</span></span>)
<span class="pl-smi">meta</span> <span class="pl-k">&lt;-</span> read.table(<span class="pl-s"><span class="pl-pds">"</span>01-info/metadata<span class="pl-pds">"</span></span>,<span class="pl-c1">T</span>, <span class="pl-v">sep</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\t</span><span class="pl-pds">"</span></span> ) %<span class="pl-k">&gt;</span>% select(<span class="pl-smi">POP</span>, <span class="pl-smi">Latitude</span>, <span class="pl-smi">Longitude</span>, <span class="pl-smi">Region</span>)
<span class="pl-smi">pd1</span> <span class="pl-k">&lt;-</span> separate(<span class="pl-smi">pd</span>,<span class="pl-smi">sampleID</span>,c(<span class="pl-s"><span class="pl-pds">"</span>POP<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>ind<span class="pl-pds">"</span></span>),<span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span>)
<span class="pl-smi">pd</span> <span class="pl-k">&lt;-</span> merge(<span class="pl-smi">pd1</span>, <span class="pl-smi">meta</span> ) <span class="pl-c"><span class="pl-c">#</span>.x="POP",by.y="POP_ID")</span>

<span class="pl-c"><span class="pl-c">#</span>############################################################################################## </span>
<span class="pl-c"><span class="pl-c">#</span> test correlation  with geography:</span>
summary(lm(<span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">LD1</span> <span class="pl-k">~</span> <span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">Latitude</span>))
summary(lm(<span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">LD2</span> <span class="pl-k">~</span> <span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">Latitude</span>))
summary(lm(<span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">LD2</span> <span class="pl-k">~</span> <span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">Longitude</span>))
summary(lm(<span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">LD1</span> <span class="pl-k">~</span> <span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">Longitude</span>))
summary(lm(<span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">LD1</span> <span class="pl-k">~</span> <span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">Latitude</span> <span class="pl-k">+</span> <span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">Longitude</span> ))
summary(lm(<span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">LD1</span> <span class="pl-k">~</span> <span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">Latitude</span> <span class="pl-k">+</span> <span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">Longitude</span> <span class="pl-k">+</span> <span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">Latitude</span> <span class="pl-k">*</span> <span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">Longitude</span> ))

<span class="pl-c"><span class="pl-c">#</span> set colors:</span>
<span class="pl-smi">myColors</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>blue<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>orange<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>darkviolet<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>chocolate4<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>springgreen4<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>green<span class="pl-pds">"</span></span>)
names(<span class="pl-smi">myColors</span>) <span class="pl-k">&lt;-</span> levels(<span class="pl-smi">pd</span><span class="pl-k">$</span><span class="pl-smi">Region</span>)
<span class="pl-smi">colScale</span> <span class="pl-k">&lt;-</span> scale_colour_manual(<span class="pl-v">name</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Region<span class="pl-pds">"</span></span>,<span class="pl-v">values</span> <span class="pl-k">=</span> <span class="pl-smi">myColors</span>)

<span class="pl-c"><span class="pl-c">#</span> plot the 2 ld</span>
<span class="pl-smi">p</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">pd</span> %<span class="pl-k">&gt;</span>% 
  <span class="pl-e">dplyr</span><span class="pl-k">::</span>rename_at(<span class="pl-s"><span class="pl-pds">"</span>POP<span class="pl-pds">"</span></span>,<span class="pl-k">~</span><span class="pl-s"><span class="pl-pds">"</span>river<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>%
  <span class="pl-c"><span class="pl-c">#</span>separate(REGION2,c("Rivers",NA), sep ="-" ) #)</span>
  <span class="pl-c"><span class="pl-c">#</span>ggplot(aes(x=LD1,y=LD2, col= river))+</span>
  ggplot(aes(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">LD1</span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">LD2</span>, <span class="pl-v">col</span> <span class="pl-k">=</span> <span class="pl-smi">Region</span>))<span class="pl-k">+</span>
  <span class="pl-smi">colScale</span> <span class="pl-k">+</span>
  geom_point(<span class="pl-v">size</span><span class="pl-k">=</span><span class="pl-c1">1</span>) <span class="pl-k">+</span> theme_classic()

<span class="pl-smi">haida</span> <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>filter(<span class="pl-smi">pd</span>, <span class="pl-smi">Region</span><span class="pl-k">==</span><span class="pl-s"><span class="pl-pds">"</span>HaidaGwaii<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>% 
  <span class="pl-e">dplyr</span><span class="pl-k">::</span>rename_at(<span class="pl-s"><span class="pl-pds">"</span>POP<span class="pl-pds">"</span></span>,<span class="pl-k">~</span><span class="pl-s"><span class="pl-pds">"</span>river<span class="pl-pds">"</span></span>) %<span class="pl-k">&gt;</span>%
  ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">LD1</span>,<span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-smi">LD2</span>, <span class="pl-v">col</span><span class="pl-k">=</span><span class="pl-smi">river</span>))<span class="pl-k">+</span> <span class="pl-c"><span class="pl-c">#</span>labs(title = "B") +</span>
  geom_text(aes ( <span class="pl-v">label</span><span class="pl-k">=</span> <span class="pl-smi">river</span>), <span class="pl-v">size</span><span class="pl-k">=</span><span class="pl-c1">3.5</span>) <span class="pl-k">+</span> theme_classic() <span class="pl-k">+</span>
  theme(<span class="pl-v">legend.position</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>none<span class="pl-pds">"</span></span>)  
<span class="pl-smi">h</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">haida</span> <span class="pl-k">+</span> scale_color_igv()

<span class="pl-smi">all</span> <span class="pl-k">&lt;-</span> ggdraw(<span class="pl-smi">p</span> <span class="pl-k">+</span> theme_half_open(<span class="pl-c1">12</span>)) <span class="pl-k">+</span>
    draw_plot(<span class="pl-smi">h</span> , <span class="pl-smi">.62</span>, <span class="pl-smi">.62</span>, <span class="pl-smi">.35</span>, <span class="pl-smi">.35</span>) <span class="pl-k">+</span>
        draw_plot_label(
         c(<span class="pl-s"><span class="pl-pds">"</span>A<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>B<span class="pl-pds">"</span></span>),
         c(<span class="pl-c1">0</span>, <span class="pl-c1">0.45</span>),
         c(<span class="pl-c1">1</span>, <span class="pl-c1">0.95</span>),
         <span class="pl-v">size</span> <span class="pl-k">=</span> <span class="pl-c1">12</span>)

pdf(<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>vae_LD1_LD2.pdf<span class="pl-pds">"</span></span>)
<span class="pl-smi">all</span> 
dev.off()
</pre>
                    </div>
                    <p dir="auto">
                     here's an example with many cluster:
                    </p>
                    <p dir="auto">
                     <a href="https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/Fig2.git.png" rel="noopener noreferrer" target="_blank">
                      <img alt="example_graph" src="https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/Fig2.git.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     this is the graph I obtained at first run of the VAE and which is in our manuscript. Almost all sampling localities have been separated in discrete cluster).
                    </p>
                    <p dir="auto">
                     Populations further away in space are also further away on the graph.
                    </p>
                    <p dir="auto">
                     In the panel B/C I zoomed in the location of HaidaGwaii because there are few samples there. We can see that we are able to finely discriminate all samples and that those very closed in space are very closed in the Graph.
                    </p>
                    <p dir="auto">
                     The same results could be obtained by running many separate PCA on subset of data. Here only one run was necessary.
                    </p>
                    <p dir="auto">
                     At the same time there is strong IBD in our data.
                    </p>
                    <p dir="auto">
                     here's an example with pure geography (IBD):
                    </p>
                    <p dir="auto">
                     <a href="https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/vae_LD1_LD2.png" rel="noopener noreferrer" target="_blank">
                      <img alt="example_graph" src="https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/vae_LD1_LD2.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     accordingly we obtain a strong correlation with latitude:
                    </p>
                    <p dir="auto">
                     <a href="https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/vae_LD1_Latitude.png" rel="noopener noreferrer" target="_blank">
                      <img alt="example_graph" src="https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/vae_LD1_Latitude.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5. GEA Analyses:
                     </h2>
                     <a aria-label="Permalink: 5. GEA Analyses:" class="anchor" href="#5-gea-analyses" id="user-content-5-gea-analyses">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.1. Extract environmental data
                     </h2>
                     <a aria-label="Permalink: 5.1. Extract environmental data" class="anchor" href="#51-extract-environmental-data" id="user-content-51-extract-environmental-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The environmental data are located in the
                     <code>
                      01-info
                     </code>
                     folder with two text file:
                    </p>
                    <ul dir="auto">
                     <li>
                      metadata
                     </li>
                     <li>
                      bioclimatic_var
                     </li>
                    </ul>
                    <p dir="auto">
                     Data included the following:
                    </p>
                    <ul dir="auto">
                     <li>
                      Distance to the spawning site (km)
                     </li>
                     <li>
                      Elevation (m)
                     </li>
                     <li>
                      Watershed areas
                     </li>
                     <li>
                      Climate data (precipitation and temperature from WorldClim)
                     </li>
                     <li>
                      Geological data (extracted from the USGS database)  (included in
                      <code>
                       02-data/env/geology
                      </code>
                     </li>
                    </ul>
                    <p dir="auto">
                     Climate data include the mean, min, max range, sdt of 19 variables associated to precipitation and temperature.
These are obviously redundant and we summarized them and decorelate the variable using a PCA.
                    </p>
                    <p dir="auto">
                     The following simple code available at
                     <code>
                      00-scripts/gea/00.pca_of_env.R
                     </code>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## load libs
libs &lt;- c('dplyr','ade4','factoextra','vegan','cowplot')
invisible(lapply(libs, library, character.only = TRUE))

###### DOWNLOAD ENV DATA
metadata &lt;- read.table(&quot;01-info/metadata&quot;, h = T, sep = &quot;\t&quot;)

bioclim &lt;- read.table(&quot;01-info/bioclimatic_var&quot;, h = T, sep = &quot;\t&quot; )
#remove BNV and SAI:
bioclim &lt;- bioclim %&gt;%filter(POP !=&quot;SAI&quot; &amp; POP !=&quot;BNV&quot;)

####### PERFORM PCA ON ENVT VAR #################################################
#will do a PCA on the dataaset that contains only climatic variables:
#separate temperature and precipitation and keep only significant axis or only first axis
X.temp &lt;- dudi.pca(df = bioclim[, 2:56], center = T, scale = T, scannf = FALSE) #nf = 3)
X.prec  &lt;- dudi.pca(df = bioclim[, 57:96], center = T, scale = T, scannf = FALSE) # nf = 3)

## singificance of axis
eig.val &lt;- get_eigenvalue(X.temp) #first we get the eigen value in an easy form
eig.val$eig &lt;- eig.val$variance.percent/100 #percent
expected &lt;- bstick(length(eig.val$eig) )
signif &lt;- eig.val$eig &gt; expected #get signifcicant axis #3axes sont signif
signif
#4 axis are significant

eig.val.p &lt;- get_eigenvalue(X.prec) #first we get the eigen value in an easy form
eig.val.p$eig &lt;- eig.val.p$variance.percent/100 #percent
expected &lt;- bstick(length(eig.val.p$eig) )
signif.p &lt;- eig.val.p$eig &gt; expected #get signifcicant axis #3axes sont signif
signif.p
#3 axis are signficant

#visualisation of individuals cos^2:
X.temp &lt;- dudi.pca(df = bioclim[, 2:53], center = T, scale = T, scannf = FALSE, nf = 4)
X.prec  &lt;- dudi.pca(df = bioclim[, 54:95], center = T, scale = T, scannf = FALSE,  nf = 3)

fviz_pca_ind(X.prec, col.ind=&quot;cos2&quot;, geom = &quot;point&quot;) +
    scale_color_gradient2(low=&quot;white&quot;, mid=&quot;blue&quot;,
    high=&quot;red&quot;, midpoint=0.6)+ theme_minimal()

p12 &lt;- fviz_pca_var(X.prec, col.var=&quot;steelblue&quot;)+
    theme_minimal()  +  ggtitle(&quot;Precipitation PC axis 12\nvariables&quot;) + 
     theme(plot.title = element_text(lineheight=.8, face=&quot;bold&quot;, hjust = 0.5))

p23 &lt;- fviz_pca_var(X.prec, axes = c(2,3) , col.var=&quot;steelblue&quot;)+
    theme_minimal() +  ggtitle(&quot;Precipitation PC axis 23\nvariables&quot;) + 
     theme(plot.title = element_text(lineheight=.8, face=&quot;bold&quot;, hjust = 0.5))

t12 &lt;- fviz_pca_var(X.temp, col.var=&quot;steelblue&quot;)+
    theme_minimal() +  ggtitle(&quot;Temperature PC axis 12\nvariables&quot;) + 
     theme(plot.title = element_text(lineheight=.8, face=&quot;bold&quot; , hjust = 0.5))

t34 &lt;- fviz_pca_var(X.temp, axes = c(3,4) , col.var=&quot;steelblue&quot;)+
    theme_minimal() +  ggtitle(&quot;Temperature PC axis 34\nvariables&quot;) + 
     theme(plot.title = element_text(lineheight=.8, face=&quot;bold&quot; , hjust = 0.5))


t_eig  &lt;- fviz_eig(X.temp, addlabels = TRUE, ylim = c(0, 50)) + 
    ggtitle(&quot;Temperature eigenvalues&quot;) + 
     theme(plot.title = element_text(lineheight=.8, face=&quot;bold&quot; , hjust = 0.5))


p_eig  &lt;- fviz_eig(X.prec, addlabels = TRUE, ylim = c(0, 50)) + 
    ggtitle(&quot;Precipitation eigenvalues&quot;) + 
     theme(plot.title = element_text(lineheight=.8, face=&quot;bold&quot; , hjust = 0.5))

pdf(file = &quot;temperature_and_precipiration.pca.pdf&quot;, 18,12)
plot_grid(t12, t34, t_eig, p12, p23,p_eig, 
    labels = &quot;auto&quot;, ncol = 3)
dev.off()

#We will keep all three axis (significant, but could also work on the first only:
prec.ind &lt;- get_pca_ind(X.prec)
temp.ind &lt;- get_pca_ind(X.temp)
colnames(temp.ind$coord) &lt;- c(&quot;Temp1&quot;,&quot;Temp2&quot;,&quot;Temp3&quot;,&quot;Temp4&quot;)
colnames(prec.ind$coord) &lt;- c(&quot;Prec1&quot;,&quot;Prec2&quot;,&quot;Prec3&quot;)

prec = cbind(bioclim$POP, prec.ind$coord)
temp = cbind(bioclim$POP, temp.ind$coord)

write.table(temp, &quot;02.data/env/temperature_coordinates_pca.txt&quot; , quote = F, row.names = F )
write.table(prec, &quot;02.data/env/precipitation_coordinates_pca.txt&quot; , quote = F, row.names = F )
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># load libs</span>
<span class="pl-smi">libs</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">'</span>dplyr<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>ade4<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>factoextra<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>vegan<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>cowplot<span class="pl-pds">'</span></span>)
<span class="pl-k">invisible</span>(lapply(<span class="pl-smi">libs</span>, <span class="pl-smi">library</span>, <span class="pl-v">character.only</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>))

<span class="pl-c"><span class="pl-c">#</span>##### DOWNLOAD ENV DATA</span>
<span class="pl-smi">metadata</span> <span class="pl-k">&lt;-</span> read.table(<span class="pl-s"><span class="pl-pds">"</span>01-info/metadata<span class="pl-pds">"</span></span>, <span class="pl-v">h</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\t</span><span class="pl-pds">"</span></span>)

<span class="pl-smi">bioclim</span> <span class="pl-k">&lt;-</span> read.table(<span class="pl-s"><span class="pl-pds">"</span>01-info/bioclimatic_var<span class="pl-pds">"</span></span>, <span class="pl-v">h</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\t</span><span class="pl-pds">"</span></span> )
<span class="pl-c"><span class="pl-c">#</span>remove BNV and SAI:</span>
<span class="pl-smi">bioclim</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">bioclim</span> %<span class="pl-k">&gt;</span>%filter(<span class="pl-smi">POP</span> <span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">"</span>SAI<span class="pl-pds">"</span></span> <span class="pl-k">&amp;</span> <span class="pl-smi">POP</span> <span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">"</span>BNV<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span>###### PERFORM PCA ON ENVT VAR #################################################</span>
<span class="pl-c"><span class="pl-c">#</span>will do a PCA on the dataaset that contains only climatic variables:</span>
<span class="pl-c"><span class="pl-c">#</span>separate temperature and precipitation and keep only significant axis or only first axis</span>
<span class="pl-smi">X.temp</span> <span class="pl-k">&lt;-</span> dudi.pca(<span class="pl-v">df</span> <span class="pl-k">=</span> <span class="pl-smi">bioclim</span>[, <span class="pl-c1">2</span><span class="pl-k">:</span><span class="pl-c1">56</span>], <span class="pl-v">center</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>, <span class="pl-v">scale</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>, <span class="pl-v">scannf</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>) <span class="pl-c"><span class="pl-c">#</span>nf = 3)</span>
<span class="pl-smi">X.prec</span>  <span class="pl-k">&lt;-</span> dudi.pca(<span class="pl-v">df</span> <span class="pl-k">=</span> <span class="pl-smi">bioclim</span>[, <span class="pl-c1">57</span><span class="pl-k">:</span><span class="pl-c1">96</span>], <span class="pl-v">center</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>, <span class="pl-v">scale</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>, <span class="pl-v">scannf</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>) <span class="pl-c"><span class="pl-c">#</span> nf = 3)</span>

<span class="pl-c"><span class="pl-c">#</span># singificance of axis</span>
<span class="pl-smi">eig.val</span> <span class="pl-k">&lt;-</span> get_eigenvalue(<span class="pl-smi">X.temp</span>) <span class="pl-c"><span class="pl-c">#</span>first we get the eigen value in an easy form</span>
<span class="pl-smi">eig.val</span><span class="pl-k">$</span><span class="pl-smi">eig</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">eig.val</span><span class="pl-k">$</span><span class="pl-smi">variance.percent</span><span class="pl-k">/</span><span class="pl-c1">100</span> <span class="pl-c"><span class="pl-c">#</span>percent</span>
<span class="pl-smi">expected</span> <span class="pl-k">&lt;-</span> bstick(length(<span class="pl-smi">eig.val</span><span class="pl-k">$</span><span class="pl-smi">eig</span>) )
<span class="pl-smi">signif</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">eig.val</span><span class="pl-k">$</span><span class="pl-smi">eig</span> <span class="pl-k">&gt;</span> <span class="pl-smi">expected</span> <span class="pl-c"><span class="pl-c">#</span>get signifcicant axis #3axes sont signif</span>
<span class="pl-smi">signif</span>
<span class="pl-c"><span class="pl-c">#</span>4 axis are significant</span>

<span class="pl-smi">eig.val.p</span> <span class="pl-k">&lt;-</span> get_eigenvalue(<span class="pl-smi">X.prec</span>) <span class="pl-c"><span class="pl-c">#</span>first we get the eigen value in an easy form</span>
<span class="pl-smi">eig.val.p</span><span class="pl-k">$</span><span class="pl-smi">eig</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">eig.val.p</span><span class="pl-k">$</span><span class="pl-smi">variance.percent</span><span class="pl-k">/</span><span class="pl-c1">100</span> <span class="pl-c"><span class="pl-c">#</span>percent</span>
<span class="pl-smi">expected</span> <span class="pl-k">&lt;-</span> bstick(length(<span class="pl-smi">eig.val.p</span><span class="pl-k">$</span><span class="pl-smi">eig</span>) )
<span class="pl-smi">signif.p</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">eig.val.p</span><span class="pl-k">$</span><span class="pl-smi">eig</span> <span class="pl-k">&gt;</span> <span class="pl-smi">expected</span> <span class="pl-c"><span class="pl-c">#</span>get signifcicant axis #3axes sont signif</span>
<span class="pl-smi">signif.p</span>
<span class="pl-c"><span class="pl-c">#</span>3 axis are signficant</span>

<span class="pl-c"><span class="pl-c">#</span>visualisation of individuals cos^2:</span>
<span class="pl-smi">X.temp</span> <span class="pl-k">&lt;-</span> dudi.pca(<span class="pl-v">df</span> <span class="pl-k">=</span> <span class="pl-smi">bioclim</span>[, <span class="pl-c1">2</span><span class="pl-k">:</span><span class="pl-c1">53</span>], <span class="pl-v">center</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>, <span class="pl-v">scale</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>, <span class="pl-v">scannf</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>, <span class="pl-v">nf</span> <span class="pl-k">=</span> <span class="pl-c1">4</span>)
<span class="pl-smi">X.prec</span>  <span class="pl-k">&lt;-</span> dudi.pca(<span class="pl-v">df</span> <span class="pl-k">=</span> <span class="pl-smi">bioclim</span>[, <span class="pl-c1">54</span><span class="pl-k">:</span><span class="pl-c1">95</span>], <span class="pl-v">center</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>, <span class="pl-v">scale</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>, <span class="pl-v">scannf</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>,  <span class="pl-v">nf</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>)

fviz_pca_ind(<span class="pl-smi">X.prec</span>, <span class="pl-v">col.ind</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>cos2<span class="pl-pds">"</span></span>, <span class="pl-v">geom</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>point<span class="pl-pds">"</span></span>) <span class="pl-k">+</span>
    scale_color_gradient2(<span class="pl-v">low</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>white<span class="pl-pds">"</span></span>, <span class="pl-v">mid</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>blue<span class="pl-pds">"</span></span>,
    <span class="pl-v">high</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>, <span class="pl-v">midpoint</span><span class="pl-k">=</span><span class="pl-c1">0.6</span>)<span class="pl-k">+</span> theme_minimal()

<span class="pl-smi">p12</span> <span class="pl-k">&lt;-</span> fviz_pca_var(<span class="pl-smi">X.prec</span>, <span class="pl-v">col.var</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>steelblue<span class="pl-pds">"</span></span>)<span class="pl-k">+</span>
    theme_minimal()  <span class="pl-k">+</span>  ggtitle(<span class="pl-s"><span class="pl-pds">"</span>Precipitation PC axis 12<span class="pl-cce">\n</span>variables<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> 
     theme(<span class="pl-v">plot.title</span> <span class="pl-k">=</span> element_text(<span class="pl-v">lineheight</span><span class="pl-k">=</span><span class="pl-smi">.8</span>, <span class="pl-v">face</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>bold<span class="pl-pds">"</span></span>, <span class="pl-v">hjust</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>))

<span class="pl-smi">p23</span> <span class="pl-k">&lt;-</span> fviz_pca_var(<span class="pl-smi">X.prec</span>, <span class="pl-v">axes</span> <span class="pl-k">=</span> c(<span class="pl-c1">2</span>,<span class="pl-c1">3</span>) , <span class="pl-v">col.var</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>steelblue<span class="pl-pds">"</span></span>)<span class="pl-k">+</span>
    theme_minimal() <span class="pl-k">+</span>  ggtitle(<span class="pl-s"><span class="pl-pds">"</span>Precipitation PC axis 23<span class="pl-cce">\n</span>variables<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> 
     theme(<span class="pl-v">plot.title</span> <span class="pl-k">=</span> element_text(<span class="pl-v">lineheight</span><span class="pl-k">=</span><span class="pl-smi">.8</span>, <span class="pl-v">face</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>bold<span class="pl-pds">"</span></span>, <span class="pl-v">hjust</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>))

<span class="pl-smi">t12</span> <span class="pl-k">&lt;-</span> fviz_pca_var(<span class="pl-smi">X.temp</span>, <span class="pl-v">col.var</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>steelblue<span class="pl-pds">"</span></span>)<span class="pl-k">+</span>
    theme_minimal() <span class="pl-k">+</span>  ggtitle(<span class="pl-s"><span class="pl-pds">"</span>Temperature PC axis 12<span class="pl-cce">\n</span>variables<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> 
     theme(<span class="pl-v">plot.title</span> <span class="pl-k">=</span> element_text(<span class="pl-v">lineheight</span><span class="pl-k">=</span><span class="pl-smi">.8</span>, <span class="pl-v">face</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>bold<span class="pl-pds">"</span></span> , <span class="pl-v">hjust</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>))

<span class="pl-smi">t34</span> <span class="pl-k">&lt;-</span> fviz_pca_var(<span class="pl-smi">X.temp</span>, <span class="pl-v">axes</span> <span class="pl-k">=</span> c(<span class="pl-c1">3</span>,<span class="pl-c1">4</span>) , <span class="pl-v">col.var</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>steelblue<span class="pl-pds">"</span></span>)<span class="pl-k">+</span>
    theme_minimal() <span class="pl-k">+</span>  ggtitle(<span class="pl-s"><span class="pl-pds">"</span>Temperature PC axis 34<span class="pl-cce">\n</span>variables<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> 
     theme(<span class="pl-v">plot.title</span> <span class="pl-k">=</span> element_text(<span class="pl-v">lineheight</span><span class="pl-k">=</span><span class="pl-smi">.8</span>, <span class="pl-v">face</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>bold<span class="pl-pds">"</span></span> , <span class="pl-v">hjust</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>))


<span class="pl-smi">t_eig</span>  <span class="pl-k">&lt;-</span> fviz_eig(<span class="pl-smi">X.temp</span>, <span class="pl-v">addlabels</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">ylim</span> <span class="pl-k">=</span> c(<span class="pl-c1">0</span>, <span class="pl-c1">50</span>)) <span class="pl-k">+</span> 
    ggtitle(<span class="pl-s"><span class="pl-pds">"</span>Temperature eigenvalues<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> 
     theme(<span class="pl-v">plot.title</span> <span class="pl-k">=</span> element_text(<span class="pl-v">lineheight</span><span class="pl-k">=</span><span class="pl-smi">.8</span>, <span class="pl-v">face</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>bold<span class="pl-pds">"</span></span> , <span class="pl-v">hjust</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>))


<span class="pl-smi">p_eig</span>  <span class="pl-k">&lt;-</span> fviz_eig(<span class="pl-smi">X.prec</span>, <span class="pl-v">addlabels</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">ylim</span> <span class="pl-k">=</span> c(<span class="pl-c1">0</span>, <span class="pl-c1">50</span>)) <span class="pl-k">+</span> 
    ggtitle(<span class="pl-s"><span class="pl-pds">"</span>Precipitation eigenvalues<span class="pl-pds">"</span></span>) <span class="pl-k">+</span> 
     theme(<span class="pl-v">plot.title</span> <span class="pl-k">=</span> element_text(<span class="pl-v">lineheight</span><span class="pl-k">=</span><span class="pl-smi">.8</span>, <span class="pl-v">face</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>bold<span class="pl-pds">"</span></span> , <span class="pl-v">hjust</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>))

pdf(<span class="pl-v">file</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>temperature_and_precipiration.pca.pdf<span class="pl-pds">"</span></span>, <span class="pl-c1">18</span>,<span class="pl-c1">12</span>)
plot_grid(<span class="pl-smi">t12</span>, <span class="pl-smi">t34</span>, <span class="pl-smi">t_eig</span>, <span class="pl-smi">p12</span>, <span class="pl-smi">p23</span>,<span class="pl-smi">p_eig</span>, 
    <span class="pl-v">labels</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>auto<span class="pl-pds">"</span></span>, <span class="pl-v">ncol</span> <span class="pl-k">=</span> <span class="pl-c1">3</span>)
dev.off()

<span class="pl-c"><span class="pl-c">#</span>We will keep all three axis (significant, but could also work on the first only:</span>
<span class="pl-smi">prec.ind</span> <span class="pl-k">&lt;-</span> get_pca_ind(<span class="pl-smi">X.prec</span>)
<span class="pl-smi">temp.ind</span> <span class="pl-k">&lt;-</span> get_pca_ind(<span class="pl-smi">X.temp</span>)
colnames(<span class="pl-smi">temp.ind</span><span class="pl-k">$</span><span class="pl-smi">coord</span>) <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>Temp1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Temp2<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Temp3<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Temp4<span class="pl-pds">"</span></span>)
colnames(<span class="pl-smi">prec.ind</span><span class="pl-k">$</span><span class="pl-smi">coord</span>) <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>Prec1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Prec2<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Prec3<span class="pl-pds">"</span></span>)

<span class="pl-v">prec</span> <span class="pl-k">=</span> cbind(<span class="pl-smi">bioclim</span><span class="pl-k">$</span><span class="pl-smi">POP</span>, <span class="pl-smi">prec.ind</span><span class="pl-k">$</span><span class="pl-smi">coord</span>)
<span class="pl-v">temp</span> <span class="pl-k">=</span> cbind(<span class="pl-smi">bioclim</span><span class="pl-k">$</span><span class="pl-smi">POP</span>, <span class="pl-smi">temp.ind</span><span class="pl-k">$</span><span class="pl-smi">coord</span>)

write.table(<span class="pl-smi">temp</span>, <span class="pl-s"><span class="pl-pds">"</span>02.data/env/temperature_coordinates_pca.txt<span class="pl-pds">"</span></span> , <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span> )
write.table(<span class="pl-smi">prec</span>, <span class="pl-s"><span class="pl-pds">"</span>02.data/env/precipitation_coordinates_pca.txt<span class="pl-pds">"</span></span> , <span class="pl-v">quote</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>, <span class="pl-v">row.names</span> <span class="pl-k">=</span> <span class="pl-c1">F</span> )
</pre>
                    </div>
                    <p dir="auto">
                     this will create the two input-files
                    </p>
                    <ul dir="auto">
                     <li>
                      temperature_coordinates_pca.txt
                     </li>
                     <li>
                      precipitation_coordinates_pca.txt
                     </li>
                    </ul>
                    <p dir="auto">
                     located in
                     <code>
                      02.data/env/
                     </code>
                    </p>
                    <p dir="auto">
                     This will be used in the GEA
                    </p>
                    <p dir="auto">
                     we can check that the data are not correlated using a simple correlation plot in R for instance.
These few lines of code will do the job:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='
library(magrittr)
library(dplyr)
library(corrplot)

###### DOWNLOAD ENV DATA
metadata &lt;- read.table("01-info/metadata", h = T, sep = "\t")
metadata&lt;- metadata %&gt;%filter(POP !="SAI" &amp; POP !="BNV")

geol &lt;- read.table("02-data/env/geology", T)
#remove BNV and SAI:
geol &lt;- geol %&gt;%filter(POP !="SAI" &amp; POP !="BNV")
#remove BNV and SAI:

#replace altitude of zero by 1
metadata$elevation[metadata$elevation == 0.00000 ] &lt;- 1

#now takes into accont the standardized elevation * distance interaction
#standardiztation function:
range01 &lt;- function(x){(x-min(x))/(max(x)-min(x))}
metadata$normalized_distance =  metadata$elevation * metadata$dist_max_km
metadata$normalized_distance &lt;- range01(metadata$normalized_distance)

####### PERFORM PCA ON ENVT VAR #################################################
temp &lt;- read.table("02-data/env/temperature_coordinates_pca.txt", T)
prec &lt;- read.table("02-data/env/precipitation_coordinates_pca.txt", T)

env1 &lt;- dplyr::select(metadata, POP, Latitude, normalized_distance, Region)
env1 &lt;- merge(env1, temp)
env1 &lt;- merge(env1, prec)
env1 &lt;- merge(env1, geol)

#remplacer avec sed directement dans metadata, geology et bioclim
env1 &lt;- env1[order(env1$POP),]

#choose the variable we want to work with:
env &lt;- select(env1, -POP, -Region)

#rename for readiblity of the plot:
colnames(env) &lt;- c("Latitude","normalized_distance", 
    "TemperaturePC1", "TemperaturePC2", "TemperaturePC3", "TemperaturePC4",
    "PrecipitationPC1", "PrecipitationPC2", "PrecipitationPC3", 
    "geology")

#again write the correlation 
mat &lt;- round(cor(env),2)

pdf(file="corrplot.pdf")
corrplot(mat, method = "number", type = "lower")
dev.off()
' dir="auto">
                     <pre>library(<span class="pl-smi">magrittr</span>)
library(<span class="pl-smi">dplyr</span>)
library(<span class="pl-smi">corrplot</span>)

<span class="pl-c"><span class="pl-c">#</span>##### DOWNLOAD ENV DATA</span>
<span class="pl-smi">metadata</span> <span class="pl-k">&lt;-</span> read.table(<span class="pl-s"><span class="pl-pds">"</span>01-info/metadata<span class="pl-pds">"</span></span>, <span class="pl-v">h</span> <span class="pl-k">=</span> <span class="pl-c1">T</span>, <span class="pl-v">sep</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span><span class="pl-cce">\t</span><span class="pl-pds">"</span></span>)
<span class="pl-smi">metadata</span><span class="pl-k">&lt;-</span> <span class="pl-smi">metadata</span> %<span class="pl-k">&gt;</span>%filter(<span class="pl-smi">POP</span> <span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">"</span>SAI<span class="pl-pds">"</span></span> <span class="pl-k">&amp;</span> <span class="pl-smi">POP</span> <span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">"</span>BNV<span class="pl-pds">"</span></span>)

<span class="pl-smi">geol</span> <span class="pl-k">&lt;-</span> read.table(<span class="pl-s"><span class="pl-pds">"</span>02-data/env/geology<span class="pl-pds">"</span></span>, <span class="pl-c1">T</span>)
<span class="pl-c"><span class="pl-c">#</span>remove BNV and SAI:</span>
<span class="pl-smi">geol</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">geol</span> %<span class="pl-k">&gt;</span>%filter(<span class="pl-smi">POP</span> <span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">"</span>SAI<span class="pl-pds">"</span></span> <span class="pl-k">&amp;</span> <span class="pl-smi">POP</span> <span class="pl-k">!=</span><span class="pl-s"><span class="pl-pds">"</span>BNV<span class="pl-pds">"</span></span>)
<span class="pl-c"><span class="pl-c">#</span>remove BNV and SAI:</span>

<span class="pl-c"><span class="pl-c">#</span>replace altitude of zero by 1</span>
<span class="pl-smi">metadata</span><span class="pl-k">$</span><span class="pl-smi">elevation</span>[<span class="pl-smi">metadata</span><span class="pl-k">$</span><span class="pl-smi">elevation</span> <span class="pl-k">==</span> <span class="pl-c1">0.00000</span> ] <span class="pl-k">&lt;-</span> <span class="pl-c1">1</span>

<span class="pl-c"><span class="pl-c">#</span>now takes into accont the standardized elevation * distance interaction</span>
<span class="pl-c"><span class="pl-c">#</span>standardiztation function:</span>
<span class="pl-en">range01</span> <span class="pl-k">&lt;-</span> <span class="pl-k">function</span>(<span class="pl-smi">x</span>){(<span class="pl-smi">x</span><span class="pl-k">-</span>min(<span class="pl-smi">x</span>))<span class="pl-k">/</span>(max(<span class="pl-smi">x</span>)<span class="pl-k">-</span>min(<span class="pl-smi">x</span>))}
<span class="pl-smi">metadata</span><span class="pl-k">$</span><span class="pl-v">normalized_distance</span> <span class="pl-k">=</span>  <span class="pl-smi">metadata</span><span class="pl-k">$</span><span class="pl-smi">elevation</span> <span class="pl-k">*</span> <span class="pl-smi">metadata</span><span class="pl-k">$</span><span class="pl-smi">dist_max_km</span>
<span class="pl-smi">metadata</span><span class="pl-k">$</span><span class="pl-smi">normalized_distance</span> <span class="pl-k">&lt;-</span> range01(<span class="pl-smi">metadata</span><span class="pl-k">$</span><span class="pl-smi">normalized_distance</span>)

<span class="pl-c"><span class="pl-c">#</span>###### PERFORM PCA ON ENVT VAR #################################################</span>
<span class="pl-smi">temp</span> <span class="pl-k">&lt;-</span> read.table(<span class="pl-s"><span class="pl-pds">"</span>02-data/env/temperature_coordinates_pca.txt<span class="pl-pds">"</span></span>, <span class="pl-c1">T</span>)
<span class="pl-smi">prec</span> <span class="pl-k">&lt;-</span> read.table(<span class="pl-s"><span class="pl-pds">"</span>02-data/env/precipitation_coordinates_pca.txt<span class="pl-pds">"</span></span>, <span class="pl-c1">T</span>)

<span class="pl-smi">env1</span> <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">metadata</span>, <span class="pl-smi">POP</span>, <span class="pl-smi">Latitude</span>, <span class="pl-smi">normalized_distance</span>, <span class="pl-smi">Region</span>)
<span class="pl-smi">env1</span> <span class="pl-k">&lt;-</span> merge(<span class="pl-smi">env1</span>, <span class="pl-smi">temp</span>)
<span class="pl-smi">env1</span> <span class="pl-k">&lt;-</span> merge(<span class="pl-smi">env1</span>, <span class="pl-smi">prec</span>)
<span class="pl-smi">env1</span> <span class="pl-k">&lt;-</span> merge(<span class="pl-smi">env1</span>, <span class="pl-smi">geol</span>)

<span class="pl-c"><span class="pl-c">#</span>remplacer avec sed directement dans metadata, geology et bioclim</span>
<span class="pl-smi">env1</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">env1</span>[order(<span class="pl-smi">env1</span><span class="pl-k">$</span><span class="pl-smi">POP</span>),]

<span class="pl-c"><span class="pl-c">#</span>choose the variable we want to work with:</span>
<span class="pl-smi">env</span> <span class="pl-k">&lt;-</span> select(<span class="pl-smi">env1</span>, <span class="pl-k">-</span><span class="pl-smi">POP</span>, <span class="pl-k">-</span><span class="pl-smi">Region</span>)

<span class="pl-c"><span class="pl-c">#</span>rename for readiblity of the plot:</span>
colnames(<span class="pl-smi">env</span>) <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">"</span>Latitude<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>normalized_distance<span class="pl-pds">"</span></span>, 
    <span class="pl-s"><span class="pl-pds">"</span>TemperaturePC1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>TemperaturePC2<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>TemperaturePC3<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>TemperaturePC4<span class="pl-pds">"</span></span>,
    <span class="pl-s"><span class="pl-pds">"</span>PrecipitationPC1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>PrecipitationPC2<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>PrecipitationPC3<span class="pl-pds">"</span></span>, 
    <span class="pl-s"><span class="pl-pds">"</span>geology<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span>again write the correlation </span>
<span class="pl-smi">mat</span> <span class="pl-k">&lt;-</span> round(cor(<span class="pl-smi">env</span>),<span class="pl-c1">2</span>)

pdf(<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>corrplot.pdf<span class="pl-pds">"</span></span>)
corrplot(<span class="pl-smi">mat</span>, <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>number<span class="pl-pds">"</span></span>, <span class="pl-v">type</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>lower<span class="pl-pds">"</span></span>)
dev.off()
</pre>
                    </div>
                    <p dir="auto">
                     and produce the following graph:
                     <a href="https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/corrplot.png" rel="noopener noreferrer" target="_blank">
                      <img alt="example_graph" src="https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/corrplot.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.2. Prepare genomics data
                     </h2>
                     <a aria-label="Permalink: 5.2. Prepare genomics data" class="anchor" href="#52-prepare-genomics-data" id="user-content-52-prepare-genomics-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We first need to filter the data to exclude 2 populations, namely BNV and SAI for which environmental data are not available.
                    </p>
                    <div class="highlight highlight-source-shell notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" #################################################################################
#                           preparing data for RDA and GEA
#################################################################################

mkdir GEA
cd GEA

#we create a file containing population SAI and BNV that we want to exclude
grep &quot;SAI\|BNV&quot; ../strata &gt; SAI_BNV.txt 

vcftools --gzvcf ../populations.24_09.missing0.92mac15.indep.clean2.recode.vcf.gz \
    --remove SAI_BNV.txt \
    --mac 15  \
    --max-missing 0.95 \
    --min-meanDP 10 --max-meanDP 120 \
    --out population_for_GEA \
    --recode \
    --recode-INFO-all

input=population_for_GEA.recode.vcf
strata $input strata.txt
bgzip $input

input=$input.gz

cut -f 2 strata.txt &gt; ind.tmp; cp ind.tmp new.ind.tmp 

#in this file we need to rename the 32 first individuals from HOP into HOD as the &quot;HOP&quot; population contains individual sample from both &quot;HOP&quot; and &quot;HOD&quot;
sed -i '2536,2567s/HOP/HOD/g' new.ind.tmp
sed -i 's/DRA/WAC/g' new.ind.tmp 

paste ind.tmp new.ind.tmp &gt; new_sample_name.txt 

#then we rename individuals in the vcf 
bcftools reheader --samples new_sample_name.txt -o ${input%.recode.vcf.gz}.renamed.vcf.gz $input 

gunzip ${input%.recode.vcf.gz}.renamed.vcf.gz

INPUT=&quot;${input%.recode.vcf.gz}.renamed.vcf&quot;

#create usefull intput for LFMM:
vcf2geno $INPUT 
geno2lfmm ${INPUT%.vcf}.geno 
sed -i 's/9/NA/g' ${INPUT%.vcf}.lfmm

#save space
gzip ${INPUT%.vcf}.lfmm
gzip ${INPUT%.vcf}.geno


plink --vcf $INPUT \
    --allow-extra-chr \
    --out ${INPUT%.recode.vcf} \
    --recode
    --recode

#prepare a cluster file
rm strata.txt
strata $INPUT strata.txt
cut -d &quot;_&quot; -f2 new.ind.tmp &gt; col2
cut -d &quot;_&quot; -f1 new.ind.tmp &gt; col1
paste col1 col2 col1 &gt; cluster.dat

#now compute allele frequency for RDA:
plink --file  ${INPUT%.recode.vcf}  \
    --allow-extra-chr --freq \
    --within cluster.dat 

gzip plink.frq.strat

rm *tmp col*
#now we can run the RDA and LFMM


################################################################################
#           prepare the data by excluding the Thompson samples                 #
################################################################################
# Since the Thompson samples populations are highly divergent they may be driving part of the signal, we will therefore replicate the analyses by removing them from the samples.

mkdir no_thompson
cd no_thompson

awk -F&quot;\t&quot; '$10==&quot;Thompson&quot; {print $0}' ../metadata &gt; thompson
awk -F&quot;\t&quot; '$10==&quot;Thompson&quot; {print $1}' ../metadata &gt; thompson.pop #to keep only thompson rivers

grep -Ff thompson.pop ../newnames  &gt; individus.thompson #to extract individuals

input=population_for_GEA.renamed.vcf.gz
output=population_for_GEA.renamed.nothompson

vcftools --gzvcf ../$input \
    --remove individus.thompson \
    --mac 15  \
    --max-missing 0.95 \
    --min-meanDP 10 --max-meanDP 120 \
    --out $output \
    --recode \
    --recode-INFO-all


strata $output.recode.vcf strata.txt

INPUT=&quot;${output}.recode.vcf&quot;

vcf2geno $INPUT 
geno2lfmm ${INPUT%.vcf}.geno 
sed -i 's/9/NA/g' ${INPUT%.vcf}.lfmm

# save space

gzip ${INPUT%.vcf}.lfmm
gzip ${INPUT%.vcf}.geno

plink --vcf $INPUT \
    --allow-extra-chr \
    --out ${INPUT%.recode.vcf} \
    --recode
    --recode

# prepare a cluster file

strata $INPUT strata.txt
cut -f1 strata.txt &gt; col1
cut -d &quot;\_&quot; -f2 strata.txt &gt; col2
paste col1 col2 col1 &gt; cluster.dat

# now compute allele frequency for RDA

plink --file  ${INPUT%.recode.vcf} \
    --allow-extra-chr --freq \
    --within cluster.dat 

gzip plink.frq.strat

rm \*tmp col\* 
#now we can run the RDA and LFMM

" dir="auto">
                     <pre> <span class="pl-c"><span class="pl-c">#</span>################################################################################</span>
<span class="pl-c"><span class="pl-c">#</span>                           preparing data for RDA and GEA</span>
<span class="pl-c"><span class="pl-c">#</span>################################################################################</span>

mkdir GEA
<span class="pl-c1">cd</span> GEA

<span class="pl-c"><span class="pl-c">#</span>we create a file containing population SAI and BNV that we want to exclude</span>
grep <span class="pl-s"><span class="pl-pds">"</span>SAI\|BNV<span class="pl-pds">"</span></span> ../strata <span class="pl-k">&gt;</span> SAI_BNV.txt 

vcftools --gzvcf ../populations.24_09.missing0.92mac15.indep.clean2.recode.vcf.gz \
    --remove SAI_BNV.txt \
    --mac 15  \
    --max-missing 0.95 \
    --min-meanDP 10 --max-meanDP 120 \
    --out population_for_GEA \
    --recode \
    --recode-INFO-all

input=population_for_GEA.recode.vcf
strata <span class="pl-smi">$input</span> strata.txt
bgzip <span class="pl-smi">$input</span>

input=<span class="pl-smi">$input</span>.gz

cut -f 2 strata.txt <span class="pl-k">&gt;</span> ind.tmp<span class="pl-k">;</span> cp ind.tmp new.ind.tmp 

<span class="pl-c"><span class="pl-c">#</span>in this file we need to rename the 32 first individuals from HOP into HOD as the "HOP" population contains individual sample from both "HOP" and "HOD"</span>
sed -i <span class="pl-s"><span class="pl-pds">'</span>2536,2567s/HOP/HOD/g<span class="pl-pds">'</span></span> new.ind.tmp
sed -i <span class="pl-s"><span class="pl-pds">'</span>s/DRA/WAC/g<span class="pl-pds">'</span></span> new.ind.tmp 

paste ind.tmp new.ind.tmp <span class="pl-k">&gt;</span> new_sample_name.txt 

<span class="pl-c"><span class="pl-c">#</span>then we rename individuals in the vcf </span>
bcftools reheader --samples new_sample_name.txt -o <span class="pl-smi">${input<span class="pl-k">%</span>.recode.vcf.gz}</span>.renamed.vcf.gz <span class="pl-smi">$input</span> 

gunzip <span class="pl-smi">${input<span class="pl-k">%</span>.recode.vcf.gz}</span>.renamed.vcf.gz

INPUT=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">${input<span class="pl-k">%</span>.recode.vcf.gz}</span>.renamed.vcf<span class="pl-pds">"</span></span>

<span class="pl-c"><span class="pl-c">#</span>create usefull intput for LFMM:</span>
vcf2geno <span class="pl-smi">$INPUT</span> 
geno2lfmm <span class="pl-smi">${INPUT<span class="pl-k">%</span>.vcf}</span>.geno 
sed -i <span class="pl-s"><span class="pl-pds">'</span>s/9/NA/g<span class="pl-pds">'</span></span> <span class="pl-smi">${INPUT<span class="pl-k">%</span>.vcf}</span>.lfmm

<span class="pl-c"><span class="pl-c">#</span>save space</span>
gzip <span class="pl-smi">${INPUT<span class="pl-k">%</span>.vcf}</span>.lfmm
gzip <span class="pl-smi">${INPUT<span class="pl-k">%</span>.vcf}</span>.geno


plink --vcf <span class="pl-smi">$INPUT</span> \
    --allow-extra-chr \
    --out <span class="pl-smi">${INPUT<span class="pl-k">%</span>.recode.vcf}</span> \
    --recode
    --recode

<span class="pl-c"><span class="pl-c">#</span>prepare a cluster file</span>
rm strata.txt
strata <span class="pl-smi">$INPUT</span> strata.txt
cut -d <span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span> -f2 new.ind.tmp <span class="pl-k">&gt;</span> col2
cut -d <span class="pl-s"><span class="pl-pds">"</span>_<span class="pl-pds">"</span></span> -f1 new.ind.tmp <span class="pl-k">&gt;</span> col1
paste col1 col2 col1 <span class="pl-k">&gt;</span> cluster.dat

<span class="pl-c"><span class="pl-c">#</span>now compute allele frequency for RDA:</span>
plink --file  <span class="pl-smi">${INPUT<span class="pl-k">%</span>.recode.vcf}</span>  \
    --allow-extra-chr --freq \
    --within cluster.dat 

gzip plink.frq.strat

rm <span class="pl-k">*</span>tmp col<span class="pl-k">*</span>
<span class="pl-c"><span class="pl-c">#</span>now we can run the RDA and LFMM</span>


<span class="pl-c"><span class="pl-c">#</span>###############################################################################</span>
<span class="pl-c"><span class="pl-c">#</span>           prepare the data by excluding the Thompson samples                 #</span>
<span class="pl-c"><span class="pl-c">#</span>###############################################################################</span>
<span class="pl-c"><span class="pl-c">#</span> Since the Thompson samples populations are highly divergent they may be driving part of the signal, we will therefore replicate the analyses by removing them from the samples.</span>

mkdir no_thompson
<span class="pl-c1">cd</span> no_thompson

awk -F<span class="pl-s"><span class="pl-pds">"</span>\t<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">'</span>$10=="Thompson" {print $0}<span class="pl-pds">'</span></span> ../metadata <span class="pl-k">&gt;</span> thompson
awk -F<span class="pl-s"><span class="pl-pds">"</span>\t<span class="pl-pds">"</span></span> <span class="pl-s"><span class="pl-pds">'</span>$10=="Thompson" {print $1}<span class="pl-pds">'</span></span> ../metadata <span class="pl-k">&gt;</span> thompson.pop <span class="pl-c"><span class="pl-c">#</span>to keep only thompson rivers</span>

grep -Ff thompson.pop ../newnames  <span class="pl-k">&gt;</span> individus.thompson <span class="pl-c"><span class="pl-c">#</span>to extract individuals</span>

input=population_for_GEA.renamed.vcf.gz
output=population_for_GEA.renamed.nothompson

vcftools --gzvcf ../<span class="pl-smi">$input</span> \
    --remove individus.thompson \
    --mac 15  \
    --max-missing 0.95 \
    --min-meanDP 10 --max-meanDP 120 \
    --out <span class="pl-smi">$output</span> \
    --recode \
    --recode-INFO-all


strata <span class="pl-smi">$output</span>.recode.vcf strata.txt

INPUT=<span class="pl-s"><span class="pl-pds">"</span><span class="pl-smi">${output}</span>.recode.vcf<span class="pl-pds">"</span></span>

vcf2geno <span class="pl-smi">$INPUT</span> 
geno2lfmm <span class="pl-smi">${INPUT<span class="pl-k">%</span>.vcf}</span>.geno 
sed -i <span class="pl-s"><span class="pl-pds">'</span>s/9/NA/g<span class="pl-pds">'</span></span> <span class="pl-smi">${INPUT<span class="pl-k">%</span>.vcf}</span>.lfmm

<span class="pl-c"><span class="pl-c">#</span> save space</span>

gzip <span class="pl-smi">${INPUT<span class="pl-k">%</span>.vcf}</span>.lfmm
gzip <span class="pl-smi">${INPUT<span class="pl-k">%</span>.vcf}</span>.geno

plink --vcf <span class="pl-smi">$INPUT</span> \
    --allow-extra-chr \
    --out <span class="pl-smi">${INPUT<span class="pl-k">%</span>.recode.vcf}</span> \
    --recode
    --recode

<span class="pl-c"><span class="pl-c">#</span> prepare a cluster file</span>

strata <span class="pl-smi">$INPUT</span> strata.txt
cut -f1 strata.txt <span class="pl-k">&gt;</span> col1
cut -d <span class="pl-s"><span class="pl-pds">"</span>\_<span class="pl-pds">"</span></span> -f2 strata.txt <span class="pl-k">&gt;</span> col2
paste col1 col2 col1 <span class="pl-k">&gt;</span> cluster.dat

<span class="pl-c"><span class="pl-c">#</span> now compute allele frequency for RDA</span>

plink --file  <span class="pl-smi">${INPUT<span class="pl-k">%</span>.recode.vcf}</span> \
    --allow-extra-chr --freq \
    --within cluster.dat 

gzip plink.frq.strat

rm <span class="pl-cce">\*</span>tmp col<span class="pl-cce">\*</span> 
<span class="pl-c"><span class="pl-c">#</span>now we can run the RDA and LFMM</span>

</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.3. Running the GEAs
                     </h2>
                     <a aria-label="Permalink: 5.3. Running the GEAs" class="anchor" href="#53-running-the-geas" id="user-content-53-running-the-geas">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.3.1 LFMM
                     </h2>
                     <a aria-label="Permalink: 5.3.1 LFMM" class="anchor" href="#531-lfmm" id="user-content-531-lfmm">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     LFMM will use a mixed model analyses to identify outliers
                    </p>
                    <p dir="auto">
                     simply use the script in
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="./00-scripts/gea/01.lfmm_step1.R " dir="auto">
                     <pre><span class="pl-smi">.</span><span class="pl-k">/</span><span class="pl-c1">00</span><span class="pl-k">-</span><span class="pl-smi">scripts</span><span class="pl-k">/</span><span class="pl-smi">gea</span><span class="pl-k">/</span><span class="pl-c1">01.</span><span class="pl-smi">lfmm_step1.R</span> </pre>
                    </div>
                    <p dir="auto">
                     this will run rather rapidly with approximately 20-30Gb of RAM
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.3.2 RDA
                     </h2>
                     <a aria-label="Permalink: 5.3.2 RDA" class="anchor" href="#532-rda" id="user-content-532-rda">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Next we run the RDA in 2 steps
                    </p>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto">
                       <li>
                        **Significance testing **
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <p dir="auto">
                     First we need to test the significance of environmental data and of the RDA axes.
This is done through an ANOVA like permutation test of the RDA using the
                     <code>
                      anova.cca
                     </code>
                     function in R
The
                     <code>
                      ordiR2step
                     </code>
                     function also allows to test the best model and choose significant variable of the RDA through a set of permutation test.
                    </p>
                    <p dir="auto">
                     Run the script :
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="./00-scripts/gea/02.RDA_significance_testing.R" dir="auto">
                     <pre><span class="pl-smi">.</span><span class="pl-k">/</span><span class="pl-c1">00</span><span class="pl-k">-</span><span class="pl-smi">scripts</span><span class="pl-k">/</span><span class="pl-smi">gea</span><span class="pl-k">/</span><span class="pl-c1">02.</span><span class="pl-smi">RDA_significance_testing.R</span></pre>
                    </div>
                    <p dir="auto">
                     it takes a few hours (~one night) on a cluster
                    </p>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto" start="2">
                       <li>
                        <strong>
                         identify outliers:
                        </strong>
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <p dir="auto">
                     run the script :
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Rscript 00-scripts/gea/03.RDA_identify_outliers.R" dir="auto">
                     <pre><span class="pl-smi">Rscript</span> <span class="pl-c1">00</span><span class="pl-k">-</span><span class="pl-smi">scripts</span><span class="pl-k">/</span><span class="pl-smi">gea</span><span class="pl-k">/</span><span class="pl-c1">03.</span><span class="pl-smi">RDA_identify_outliers.R</span></pre>
                    </div>
                    <p dir="auto">
                     this will write the number of outlier in a file as well as other useful data such as the number of outliers in each chromosomes.
                    </p>
                    <p dir="auto">
                     We used the approach of Forester et al. (2018) to identify outliers.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      5.3.3. Create a RDA Figure and LFMM plot:
                     </h2>
                     <a aria-label="Permalink: 5.3.3. Create a RDA Figure and LFMM plot:" class="anchor" href="#533-create-a-rda-figure-and-lfmm-plot" id="user-content-533-create-a-rda-figure-and-lfmm-plot">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     To reproduce the Figure 3 simply run:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Rscript 00-scripts/gea/Figure3.R" dir="auto">
                     <pre><span class="pl-smi">Rscript</span> <span class="pl-c1">00</span><span class="pl-k">-</span><span class="pl-smi">scripts</span><span class="pl-k">/</span><span class="pl-smi">gea</span><span class="pl-k">/</span><span class="pl-smi">Figure3.R</span></pre>
                    </div>
                    <p dir="auto">
                     This should automatically produce the Figure3 from our manuscript
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      6. Looking for parallelism
                     </h2>
                     <a aria-label="Permalink: 6. Looking for parallelism" class="anchor" href="#6-looking-for-parallelism" id="user-content-6-looking-for-parallelism">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     Run the Rscript
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="00-scripts/gea/05.Figure4.R " dir="auto">
                     <pre><span class="pl-c1">00</span><span class="pl-k">-</span><span class="pl-smi">scripts</span><span class="pl-k">/</span><span class="pl-smi">gea</span><span class="pl-k">/</span><span class="pl-c1">05.</span><span class="pl-smi">Figure4.R</span> </pre>
                    </div>
                    <p dir="auto">
                     This should automatically produce the Figure4 from our manuscript as well as some statisticall tests
                    </p>
                    <p dir="auto">
                     See explanation in our methods of the manuscript
                    </p>
                    <p dir="auto">
                     I'll document all of this later.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      7. plotting LFMM results:
                     </h2>
                     <a aria-label="Permalink: 7. plotting LFMM results:" class="anchor" href="#7-plotting-lfmm-results" id="user-content-7-plotting-lfmm-results">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     this can be done with CMplot
see script 00-scripts/gea/LFMM_plot_FigureS10.R or simple code below:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## load libs
libs &lt;- c('dplyr','data.table', 'magrittr', 'cowplot','CMplot', 'gridGraphics')
invisible(lapply(libs, library, character.only = TRUE))

pval_loc &lt;- fread(&quot;zcat adjust_pvaluesBH_lfkmm_K.20.txt.gz&quot;)

#we will remove scaffold to plot only the 30 chromosome + the 8 dups.
pvalsub &lt;- pval_loc %&gt;% filter(!(grepl(&quot;scaff&quot;, V1)))
pvalsub$V1&lt;- gsub(&quot;Okis&quot;,&quot;&quot;,pvalsub$V1)
pvalsub$V1&lt;- gsub(&quot;LG&quot;,&quot;&quot;,pvalsub$V1)
pvalsub$V1&lt;- gsub(&quot;_hom&quot;,&quot;&quot;,pvalsub$V1)

#temperature
tempplot &lt;- select(pvalsub,V3,V1,V2,Temperature1, Temperature2, Temperature3, Temperature4) %&gt;% 
    set_colnames(.,c(&quot;SNP&quot;,&quot;Chromosome&quot;,&quot;Position&quot;,
    &quot;TemperaturePCaxis1&quot;,&quot;TemperaturePCaxis2&quot;,&quot;TemperaturePCaxis3&quot;,&quot;TemperaturePCaxis4&quot;))
templot &lt;- as.data.frame(tempplot)

#precipitation
precplot &lt;- select(pvalsub,V3,V1,V2,Precipitation1, Precipitation2, Precipitation3) %&gt;% 
    set_colnames(.,c(&quot;SNP&quot;,&quot;Chromosome&quot;,&quot;Position&quot;,
    &quot;PrecipitationPCaxis1&quot;,&quot;PrecipitationPCaxis2&quot;,&quot;PrecipitationPCaxis3&quot;))
precplot &lt;- as.data.frame(precplot)

#geology
geolplot &lt;- select(pvalsub,V3,V1,V2,Geology) %&gt;% 
    set_colnames(.,c(&quot;SNP&quot;,&quot;Chromosome&quot;,&quot;Position&quot;,&quot;Geology&quot;))
geolplot &lt;- as.data.frame(geolplot)


CMplot(geolplot, plot.type=&quot;m&quot;, LOG10=TRUE, ylim=NULL, threshold=NULL ,
       chr.den.col = c(&quot;darkgreen&quot;, &quot;yellow&quot;, &quot;red&quot;),
       signal.pch = c(19,19),file=&quot;pdf&quot;,memo=&quot;&quot;,dpi=300,
       file.output = F,
       verbose = TRUE,
       width=14,height=6, chr.labels.angle=45)

p3 &lt;- recordPlot()
CMplot(templot, plot.type=&quot;m&quot;, multracks = TRUE, LOG10=TRUE, ylim=NULL, threshold=NULL, mar = c(5,8,3,3),
       chr.den.col = c(&quot;darkgreen&quot;, &quot;yellow&quot;, &quot;red&quot;),
       signal.pch = c(19,19),file=&quot;pdf&quot;,memo=&quot;&quot;,dpi=300,
       file.output = F,
       verbose = TRUE,
       width=14,height=6, chr.labels.angle=45)
p1 &lt;- recordPlot()


CMplot(precplot, plot.type=&quot;m&quot;, multracks = TRUE, LOG10=TRUE, ylim=NULL, threshold=NULL, mar = c(5,8,3,3),
       chr.den.col = c(&quot;darkgreen&quot;, &quot;yellow&quot;, &quot;red&quot;),
       signal.pch = c(19,19),file=&quot;pdf&quot;,memo=&quot;&quot;,dpi=300,
       file.output = F,
       verbose = TRUE,
       width=14,height=6, chr.labels.angle=45)
p2 &lt;- recordPlot()

#export final figure
pdf(file=&quot;FigureS10.pdf&quot;,15,20)
plot_grid(p1, p2,p3, 
    ncol = 1, 
    align = &quot;v&quot;, 
    labels=c(&quot;A - Temperature&quot;, &quot;B - Precipitation&quot;, &quot;C - Geology&quot;) )
dev.off()
" dir="auto">
                     <pre><span class="pl-c"><span class="pl-c">#</span># load libs</span>
<span class="pl-smi">libs</span> <span class="pl-k">&lt;-</span> c(<span class="pl-s"><span class="pl-pds">'</span>dplyr<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>data.table<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>magrittr<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>cowplot<span class="pl-pds">'</span></span>,<span class="pl-s"><span class="pl-pds">'</span>CMplot<span class="pl-pds">'</span></span>, <span class="pl-s"><span class="pl-pds">'</span>gridGraphics<span class="pl-pds">'</span></span>)
<span class="pl-k">invisible</span>(lapply(<span class="pl-smi">libs</span>, <span class="pl-smi">library</span>, <span class="pl-v">character.only</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>))

<span class="pl-smi">pval_loc</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>zcat adjust_pvaluesBH_lfkmm_K.20.txt.gz<span class="pl-pds">"</span></span>)

<span class="pl-c"><span class="pl-c">#</span>we will remove scaffold to plot only the 30 chromosome + the 8 dups.</span>
<span class="pl-smi">pvalsub</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">pval_loc</span> %<span class="pl-k">&gt;</span>% filter(<span class="pl-k">!</span>(grepl(<span class="pl-s"><span class="pl-pds">"</span>scaff<span class="pl-pds">"</span></span>, <span class="pl-smi">V1</span>)))
<span class="pl-smi">pvalsub</span><span class="pl-k">$</span><span class="pl-smi">V1</span><span class="pl-k">&lt;-</span> gsub(<span class="pl-s"><span class="pl-pds">"</span>Okis<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>,<span class="pl-smi">pvalsub</span><span class="pl-k">$</span><span class="pl-smi">V1</span>)
<span class="pl-smi">pvalsub</span><span class="pl-k">$</span><span class="pl-smi">V1</span><span class="pl-k">&lt;-</span> gsub(<span class="pl-s"><span class="pl-pds">"</span>LG<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>,<span class="pl-smi">pvalsub</span><span class="pl-k">$</span><span class="pl-smi">V1</span>)
<span class="pl-smi">pvalsub</span><span class="pl-k">$</span><span class="pl-smi">V1</span><span class="pl-k">&lt;-</span> gsub(<span class="pl-s"><span class="pl-pds">"</span>_hom<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>,<span class="pl-smi">pvalsub</span><span class="pl-k">$</span><span class="pl-smi">V1</span>)

<span class="pl-c"><span class="pl-c">#</span>temperature</span>
<span class="pl-smi">tempplot</span> <span class="pl-k">&lt;-</span> select(<span class="pl-smi">pvalsub</span>,<span class="pl-smi">V3</span>,<span class="pl-smi">V1</span>,<span class="pl-smi">V2</span>,<span class="pl-smi">Temperature1</span>, <span class="pl-smi">Temperature2</span>, <span class="pl-smi">Temperature3</span>, <span class="pl-smi">Temperature4</span>) %<span class="pl-k">&gt;</span>% 
    set_colnames(<span class="pl-smi">.</span>,c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Chromosome<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Position<span class="pl-pds">"</span></span>,
    <span class="pl-s"><span class="pl-pds">"</span>TemperaturePCaxis1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>TemperaturePCaxis2<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>TemperaturePCaxis3<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>TemperaturePCaxis4<span class="pl-pds">"</span></span>))
<span class="pl-smi">templot</span> <span class="pl-k">&lt;-</span> as.data.frame(<span class="pl-smi">tempplot</span>)

<span class="pl-c"><span class="pl-c">#</span>precipitation</span>
<span class="pl-smi">precplot</span> <span class="pl-k">&lt;-</span> select(<span class="pl-smi">pvalsub</span>,<span class="pl-smi">V3</span>,<span class="pl-smi">V1</span>,<span class="pl-smi">V2</span>,<span class="pl-smi">Precipitation1</span>, <span class="pl-smi">Precipitation2</span>, <span class="pl-smi">Precipitation3</span>) %<span class="pl-k">&gt;</span>% 
    set_colnames(<span class="pl-smi">.</span>,c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Chromosome<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Position<span class="pl-pds">"</span></span>,
    <span class="pl-s"><span class="pl-pds">"</span>PrecipitationPCaxis1<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>PrecipitationPCaxis2<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>PrecipitationPCaxis3<span class="pl-pds">"</span></span>))
<span class="pl-smi">precplot</span> <span class="pl-k">&lt;-</span> as.data.frame(<span class="pl-smi">precplot</span>)

<span class="pl-c"><span class="pl-c">#</span>geology</span>
<span class="pl-smi">geolplot</span> <span class="pl-k">&lt;-</span> select(<span class="pl-smi">pvalsub</span>,<span class="pl-smi">V3</span>,<span class="pl-smi">V1</span>,<span class="pl-smi">V2</span>,<span class="pl-smi">Geology</span>) %<span class="pl-k">&gt;</span>% 
    set_colnames(<span class="pl-smi">.</span>,c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Chromosome<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Position<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Geology<span class="pl-pds">"</span></span>))
<span class="pl-smi">geolplot</span> <span class="pl-k">&lt;-</span> as.data.frame(<span class="pl-smi">geolplot</span>)


CMplot(<span class="pl-smi">geolplot</span>, <span class="pl-v">plot.type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>m<span class="pl-pds">"</span></span>, <span class="pl-v">LOG10</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">ylim</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">threshold</span><span class="pl-k">=</span><span class="pl-c1">NULL</span> ,
       <span class="pl-v">chr.den.col</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>darkgreen<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>yellow<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>),
       <span class="pl-v">signal.pch</span> <span class="pl-k">=</span> c(<span class="pl-c1">19</span>,<span class="pl-c1">19</span>),<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>pdf<span class="pl-pds">"</span></span>,<span class="pl-v">memo</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>,<span class="pl-v">dpi</span><span class="pl-k">=</span><span class="pl-c1">300</span>,
       <span class="pl-v">file.output</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,
       <span class="pl-v">verbose</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
       <span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">14</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">6</span>, <span class="pl-v">chr.labels.angle</span><span class="pl-k">=</span><span class="pl-c1">45</span>)

<span class="pl-smi">p3</span> <span class="pl-k">&lt;-</span> recordPlot()
CMplot(<span class="pl-smi">templot</span>, <span class="pl-v">plot.type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>m<span class="pl-pds">"</span></span>, <span class="pl-v">multracks</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">LOG10</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">ylim</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">threshold</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">mar</span> <span class="pl-k">=</span> c(<span class="pl-c1">5</span>,<span class="pl-c1">8</span>,<span class="pl-c1">3</span>,<span class="pl-c1">3</span>),
       <span class="pl-v">chr.den.col</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>darkgreen<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>yellow<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>),
       <span class="pl-v">signal.pch</span> <span class="pl-k">=</span> c(<span class="pl-c1">19</span>,<span class="pl-c1">19</span>),<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>pdf<span class="pl-pds">"</span></span>,<span class="pl-v">memo</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>,<span class="pl-v">dpi</span><span class="pl-k">=</span><span class="pl-c1">300</span>,
       <span class="pl-v">file.output</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,
       <span class="pl-v">verbose</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
       <span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">14</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">6</span>, <span class="pl-v">chr.labels.angle</span><span class="pl-k">=</span><span class="pl-c1">45</span>)
<span class="pl-smi">p1</span> <span class="pl-k">&lt;-</span> recordPlot()


CMplot(<span class="pl-smi">precplot</span>, <span class="pl-v">plot.type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>m<span class="pl-pds">"</span></span>, <span class="pl-v">multracks</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>, <span class="pl-v">LOG10</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>, <span class="pl-v">ylim</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">threshold</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>, <span class="pl-v">mar</span> <span class="pl-k">=</span> c(<span class="pl-c1">5</span>,<span class="pl-c1">8</span>,<span class="pl-c1">3</span>,<span class="pl-c1">3</span>),
       <span class="pl-v">chr.den.col</span> <span class="pl-k">=</span> c(<span class="pl-s"><span class="pl-pds">"</span>darkgreen<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>yellow<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>red<span class="pl-pds">"</span></span>),
       <span class="pl-v">signal.pch</span> <span class="pl-k">=</span> c(<span class="pl-c1">19</span>,<span class="pl-c1">19</span>),<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>pdf<span class="pl-pds">"</span></span>,<span class="pl-v">memo</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>,<span class="pl-v">dpi</span><span class="pl-k">=</span><span class="pl-c1">300</span>,
       <span class="pl-v">file.output</span> <span class="pl-k">=</span> <span class="pl-c1">F</span>,
       <span class="pl-v">verbose</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
       <span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">14</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">6</span>, <span class="pl-v">chr.labels.angle</span><span class="pl-k">=</span><span class="pl-c1">45</span>)
<span class="pl-smi">p2</span> <span class="pl-k">&lt;-</span> recordPlot()

<span class="pl-c"><span class="pl-c">#</span>export final figure</span>
pdf(<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>FigureS10.pdf<span class="pl-pds">"</span></span>,<span class="pl-c1">15</span>,<span class="pl-c1">20</span>)
plot_grid(<span class="pl-smi">p1</span>, <span class="pl-smi">p2</span>,<span class="pl-smi">p3</span>, 
    <span class="pl-v">ncol</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>, 
    <span class="pl-v">align</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>v<span class="pl-pds">"</span></span>, 
    <span class="pl-v">labels</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>A - Temperature<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>B - Precipitation<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>C - Geology<span class="pl-pds">"</span></span>) )
dev.off()
</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      8. PBS analyses
                     </h2>
                     <a aria-label="Permalink: 8. PBS analyses" class="anchor" href="#8-pbs-analyses" id="user-content-8-pbs-analyses">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      <strong>
                       ANGSD analyses:
                      </strong>
                     </li>
                    </ul>
                    <p dir="auto">
                     The major reason for using ANGSD was to compute the PBS score from Yi et al. 2010 (although it can be computed from the vcf directly)
                    </p>
                    <p dir="auto">
                     All steps to run ANGSD are found
                     <a href="https://github.com/QuentinRougemont/utility_scripts/tree/master/00.ANGSD">
                      here
                     </a>
                     . They necessitate to start from the bam files.
                    </p>
                    <ul dir="auto">
                     <li>
                      <strong>
                       important note
                      </strong>
                     </li>
                    </ul>
                    <p dir="auto">
                     before running Fst or PBS I used a three-species outgroup sequence for proper folding made of :
                    </p>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto">
                       <li>
                        <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA287458" rel="nofollow">
                         Atlantic salmon
                        </a>
                        n = 5 individuals
                       </li>
                      </ol>
                     </li>
                     <li>
                      <ol dir="auto" start="2">
                       <li>
                        <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA386519" rel="nofollow">
                         Rainbow trout
                        </a>
                        n = 5 individuals and
                       </li>
                      </ol>
                     </li>
                     <li>
                      <ol dir="auto" start="3">
                       <li>
                        <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA530256" rel="nofollow">
                         Sockeye salmon
                        </a>
                        n = 3 individuals
                       </li>
                      </ol>
                     </li>
                    </ul>
                    <p dir="auto">
                     Whole genome sequences were:
                    </p>
                    <ul dir="auto">
                     <li>
                      processed with
                      <a href="https://github.com/QuentinRougemont/gatk_haplotype/blob/master/gatk4/01_scripts/01_fastp.sh">
                       fastp
                      </a>
                     </li>
                     <li>
                      mapped to the coho salmon genome using
                      <a href="https://github.com/QuentinRougemont/gatk_haplotype/blob/master/gatk4/01_scripts/02_bwa_mem2_align_reads_PE.sh">
                       bwa-mem
                      </a>
                      .
                     </li>
                     <li>
                      marked for duplicates with
                      <a href="https://github.com/QuentinRougemont/gatk_haplotype/blob/master/gatk4/01_scripts/03_remove_duplicates.sh">
                       picard
                      </a>
                     </li>
                    </ul>
                    <p dir="auto">
                     The resulting bam were processed to reconstruct an ancestral sequence with
                     <a href="https://github.com/QuentinRougemont/utility_scripts/blob/master/00.ANGSD/ancestral_seq.sh">
                      ancestral_seq.sh
                     </a>
                    </p>
                    <p dir="auto">
                     Then Simply follow the scripts in the order of their number:
                    </p>
                    <ul dir="auto">
                     <li>
                      <ol dir="auto">
                       <li>
                        <a href="https://github.com/QuentinRougemont/utility_scripts/blob/master/00.ANGSD/00_angsd_bam_to_saf.sh">
                         00_angsd_bam_to_saf.sh
                        </a>
                       </li>
                      </ol>
                     </li>
                     <li>
                      <ol dir="auto" start="2">
                       <li>
                        <a href="https://github.com/QuentinRougemont/utility_scripts/blob/master/00.ANGSD/01_1dsfs.sh">
                         01_1dsfs.sh
                        </a>
                       </li>
                      </ol>
                     </li>
                     <li>
                      <ol dir="auto" start="3">
                       <li>
                        <a href="https://github.com/QuentinRougemont/utility_scripts/blob/master/00.ANGSD/02_2dsfs.sh">
                         02_2dsfs.sh
                        </a>
                       </li>
                      </ol>
                     </li>
                     <li>
                      <ol dir="auto" start="4">
                       <li>
                        <a href="https://github.com/QuentinRougemont/utility_scripts/blob/master/00.ANGSD/04.pairwise_fst_sliding_window.sh">
                         04.pairwise_fst_sliding_window.sh
                        </a>
                        #to get only Fst
                       </li>
                      </ol>
                     </li>
                     <li>
                      <ol dir="auto" start="5">
                       <li>
                        <a href="https://github.com/QuentinRougemont/utility_scripts/blob/master/00.ANGSD/05_PBS_sliding_windows.sh">
                         05_PBS_sliding_windows.sh
                        </a>
                        #to get Fst and PBS
                       </li>
                      </ol>
                     </li>
                     <li>
                      <p dir="auto">
                       Produce graph: I used
                       <a href="https://github.com/YinLiLin/CMplot">
                        CMplot
                       </a>
                       to make all manhattan plot.
                      </p>
                     </li>
                    </ul>
                    <p dir="auto">
                     The used is straightforward:
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(data.table)
library(dplyr)
library(magrittr)
library(CMplot)
rm(list=ls())
ALB_LCA &lt;- fread("zcat all_ALB_LCA_SAI.OkisLG01.pbs.50.txt.gz") #1 Thompson
POR_SHK &lt;- fread("zcat all_POR_SHK_SAI.OkisLG01.pbs.50.txt.gz") #2 alaska
SFK_GOO &lt;- fread("zcat all_SFK_GOO_SAI.OkisLG01.pbs.50.txt.gz") #3 alkaska #2
SUS_OON &lt;- fread("zcat all_SUS_OON_SAI.OkisLG01.pbs.50.txt.gz") #4 BC North

ALB_LCA  &lt;- dplyr::select(ALB_LCA, -Fst02, -Fst12, -nSite, -V3, -PBS1, -PBS2)
POR_SHK  &lt;- dplyr::select(POR_SHK, -Fst02, -Fst12, -nSite, -V3, -PBS1, -PBS2)
SFK_GOO  &lt;- dplyr::select(SFK_GOO, -Fst02, -Fst12, -nSite, -V3, -PBS1, -PBS2)
SUS_OON  &lt;- dplyr::select(SUS_OON, -Fst02, -Fst12, -nSite, -V3, -PBS1, -PBS2)

### Effectuer un manathan plots:
ALB_LCA2 &lt;- dplyr::select(ALB_LCA, chr, midPos, PBS0)
POR_SHK2 &lt;- dplyr::select(POR_SHK, chr, midPos, PBS0)
SFK_GOO2 &lt;- dplyr::select(SFK_GOO, chr, midPos, PBS0)
SUS_OON2 &lt;- dplyr::select(SUS_OON, chr, midPos, PBS0)

ALB_POR &lt;- full_join(ALB_LCA2, POR_SHK2, by=c("chr","midPos"))
ALB_POR_SFK &lt;- full_join(ALB_POR, SFK_GOO2, by=c("chr","midPos"))
ALB_POR_SFK_SUS &lt;- full_join(ALB_POR_SFK, SUS_OON2, by=c("chr","midPos"))

distplot &lt;- ALB_POR_SFK_SUS
distplot$SNP &lt;- seq(1:nrow(distplot) )

distplot &lt;- dplyr::select(distplot, SNP,chr,midPos,
     PBS0.x, PBS0.y, PBS0.x.x, PBS0.y.y) %&gt;% 
     set_colnames(.,c("SNP","Chromosome","Position","ALB","POR","SFK","SUS"))

distplot &lt;- filter(distplot, !grepl("scaf", distplot$Chromosome))
#distplot &lt;- filter(distplot, !grepl("hom", distplot$Chromosome))
distplot$Chromosome &lt;- gsub("Okis","",distplot$Chromosome)
distplot[distplot&lt;0] &lt;- 0

CMplot(distplot,type="p",plot.type="m", multracks=TRUE,
    LOG10=F,threshold=NULL,memo="",
    ylab="PBS",cex=0.6,cex.lab=0.8,
    file.output=TRUE,file="pdf",dpi=300,
    verbose=TRUE,width=20,height=4,chr.labels.angle=45)

################## FST CMPLOT ##################################################
ALB\_LCA2 &lt;- dplyr::select(ALB\_LCA, chr, midPos, Fst01)
POR\_SHK2 &lt;- dplyr::select(POR\_SHK, chr, midPos, Fst01)
SFK\_GOO2 &lt;- dplyr::select(SFK\_GOO, chr, midPos, Fst01)
SUS\_OON2 &lt;- dplyr::select(SUS\_OON, chr, midPos, Fst01)

ALB\_POR &lt;- full\_join(ALB\_LCA2, POR\_SHK2, by=c("chr","midPos"))
ALB\_POR\_SFK &lt;- full\_join(ALB\_POR, SFK\_GOO2, by=c("chr","midPos"))
ALB\_POR\_SFK\_SUS &lt;- full\_join(ALB\_POR\_SFK, SUS\_OON2, by=c("chr","midPos"))

distplot &lt;- ALB\_POR\_SFK\_SUS
distplot$SNP &lt;- seq(1:nrow(distplot) )

distplot &lt;- dplyr::select(distplot, SNP,chr,midPos,
            Fst01.x, Fst01.y, Fst01.x.x, Fst01.y.y ) %&gt;% 
	set_colnames(.,c("SNP","Chromosome","Position" ,"ALB","POR","SFK","SUS"))

distplot &lt;- filter(distplot, !grepl("scaf", distplot$Chromosome))
distplot$Chromosome &lt;- gsub("Okis","",distplot$Chromosome)
distplot[distplot&lt;0] &lt;- 0

CMplot(distplot,type="p",plot.type="m", multracks=TRUE,
       LOG10=F,threshold=NULL,memo="",
       ylab="Fst",cex=0.6,cex.lab=0.8,
       file.output=TRUE,file="pdf",dpi=300,
       verbose=TRUE,width=20,height=4,chr.labels.angle=45)' dir="auto">
                     <pre>library(<span class="pl-smi">data.table</span>)
library(<span class="pl-smi">dplyr</span>)
library(<span class="pl-smi">magrittr</span>)
library(<span class="pl-smi">CMplot</span>)
rm(<span class="pl-v">list</span><span class="pl-k">=</span>ls())
<span class="pl-smi">ALB_LCA</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>zcat all_ALB_LCA_SAI.OkisLG01.pbs.50.txt.gz<span class="pl-pds">"</span></span>) <span class="pl-c"><span class="pl-c">#</span>1 Thompson</span>
<span class="pl-smi">POR_SHK</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>zcat all_POR_SHK_SAI.OkisLG01.pbs.50.txt.gz<span class="pl-pds">"</span></span>) <span class="pl-c"><span class="pl-c">#</span>2 alaska</span>
<span class="pl-smi">SFK_GOO</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>zcat all_SFK_GOO_SAI.OkisLG01.pbs.50.txt.gz<span class="pl-pds">"</span></span>) <span class="pl-c"><span class="pl-c">#</span>3 alkaska #2</span>
<span class="pl-smi">SUS_OON</span> <span class="pl-k">&lt;-</span> fread(<span class="pl-s"><span class="pl-pds">"</span>zcat all_SUS_OON_SAI.OkisLG01.pbs.50.txt.gz<span class="pl-pds">"</span></span>) <span class="pl-c"><span class="pl-c">#</span>4 BC North</span>

<span class="pl-smi">ALB_LCA</span>  <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">ALB_LCA</span>, <span class="pl-k">-</span><span class="pl-smi">Fst02</span>, <span class="pl-k">-</span><span class="pl-smi">Fst12</span>, <span class="pl-k">-</span><span class="pl-smi">nSite</span>, <span class="pl-k">-</span><span class="pl-smi">V3</span>, <span class="pl-k">-</span><span class="pl-smi">PBS1</span>, <span class="pl-k">-</span><span class="pl-smi">PBS2</span>)
<span class="pl-smi">POR_SHK</span>  <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">POR_SHK</span>, <span class="pl-k">-</span><span class="pl-smi">Fst02</span>, <span class="pl-k">-</span><span class="pl-smi">Fst12</span>, <span class="pl-k">-</span><span class="pl-smi">nSite</span>, <span class="pl-k">-</span><span class="pl-smi">V3</span>, <span class="pl-k">-</span><span class="pl-smi">PBS1</span>, <span class="pl-k">-</span><span class="pl-smi">PBS2</span>)
<span class="pl-smi">SFK_GOO</span>  <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">SFK_GOO</span>, <span class="pl-k">-</span><span class="pl-smi">Fst02</span>, <span class="pl-k">-</span><span class="pl-smi">Fst12</span>, <span class="pl-k">-</span><span class="pl-smi">nSite</span>, <span class="pl-k">-</span><span class="pl-smi">V3</span>, <span class="pl-k">-</span><span class="pl-smi">PBS1</span>, <span class="pl-k">-</span><span class="pl-smi">PBS2</span>)
<span class="pl-smi">SUS_OON</span>  <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">SUS_OON</span>, <span class="pl-k">-</span><span class="pl-smi">Fst02</span>, <span class="pl-k">-</span><span class="pl-smi">Fst12</span>, <span class="pl-k">-</span><span class="pl-smi">nSite</span>, <span class="pl-k">-</span><span class="pl-smi">V3</span>, <span class="pl-k">-</span><span class="pl-smi">PBS1</span>, <span class="pl-k">-</span><span class="pl-smi">PBS2</span>)

<span class="pl-c"><span class="pl-c">#</span>## Effectuer un manathan plots:</span>
<span class="pl-smi">ALB_LCA2</span> <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">ALB_LCA</span>, <span class="pl-smi">chr</span>, <span class="pl-smi">midPos</span>, <span class="pl-smi">PBS0</span>)
<span class="pl-smi">POR_SHK2</span> <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">POR_SHK</span>, <span class="pl-smi">chr</span>, <span class="pl-smi">midPos</span>, <span class="pl-smi">PBS0</span>)
<span class="pl-smi">SFK_GOO2</span> <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">SFK_GOO</span>, <span class="pl-smi">chr</span>, <span class="pl-smi">midPos</span>, <span class="pl-smi">PBS0</span>)
<span class="pl-smi">SUS_OON2</span> <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">SUS_OON</span>, <span class="pl-smi">chr</span>, <span class="pl-smi">midPos</span>, <span class="pl-smi">PBS0</span>)

<span class="pl-smi">ALB_POR</span> <span class="pl-k">&lt;-</span> full_join(<span class="pl-smi">ALB_LCA2</span>, <span class="pl-smi">POR_SHK2</span>, <span class="pl-v">by</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>chr<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>midPos<span class="pl-pds">"</span></span>))
<span class="pl-smi">ALB_POR_SFK</span> <span class="pl-k">&lt;-</span> full_join(<span class="pl-smi">ALB_POR</span>, <span class="pl-smi">SFK_GOO2</span>, <span class="pl-v">by</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>chr<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>midPos<span class="pl-pds">"</span></span>))
<span class="pl-smi">ALB_POR_SFK_SUS</span> <span class="pl-k">&lt;-</span> full_join(<span class="pl-smi">ALB_POR_SFK</span>, <span class="pl-smi">SUS_OON2</span>, <span class="pl-v">by</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>chr<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>midPos<span class="pl-pds">"</span></span>))

<span class="pl-smi">distplot</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">ALB_POR_SFK_SUS</span>
<span class="pl-smi">distplot</span><span class="pl-k">$</span><span class="pl-smi">SNP</span> <span class="pl-k">&lt;-</span> seq(<span class="pl-c1">1</span><span class="pl-k">:</span>nrow(<span class="pl-smi">distplot</span>) )

<span class="pl-smi">distplot</span> <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">distplot</span>, <span class="pl-smi">SNP</span>,<span class="pl-smi">chr</span>,<span class="pl-smi">midPos</span>,
     <span class="pl-smi">PBS0.x</span>, <span class="pl-smi">PBS0.y</span>, <span class="pl-smi">PBS0.x.x</span>, <span class="pl-smi">PBS0.y.y</span>) %<span class="pl-k">&gt;</span>% 
     set_colnames(<span class="pl-smi">.</span>,c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Chromosome<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Position<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>ALB<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>POR<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>SFK<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>SUS<span class="pl-pds">"</span></span>))

<span class="pl-smi">distplot</span> <span class="pl-k">&lt;-</span> filter(<span class="pl-smi">distplot</span>, <span class="pl-k">!</span>grepl(<span class="pl-s"><span class="pl-pds">"</span>scaf<span class="pl-pds">"</span></span>, <span class="pl-smi">distplot</span><span class="pl-k">$</span><span class="pl-smi">Chromosome</span>))
<span class="pl-c"><span class="pl-c">#</span>distplot &lt;- filter(distplot, !grepl("hom", distplot$Chromosome))</span>
<span class="pl-smi">distplot</span><span class="pl-k">$</span><span class="pl-smi">Chromosome</span> <span class="pl-k">&lt;-</span> gsub(<span class="pl-s"><span class="pl-pds">"</span>Okis<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>,<span class="pl-smi">distplot</span><span class="pl-k">$</span><span class="pl-smi">Chromosome</span>)
<span class="pl-smi">distplot</span>[<span class="pl-smi">distplot</span><span class="pl-k">&lt;</span><span class="pl-c1">0</span>] <span class="pl-k">&lt;-</span> <span class="pl-c1">0</span>

CMplot(<span class="pl-smi">distplot</span>,<span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>p<span class="pl-pds">"</span></span>,<span class="pl-v">plot.type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>m<span class="pl-pds">"</span></span>, <span class="pl-v">multracks</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>,
    <span class="pl-v">LOG10</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">threshold</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>,<span class="pl-v">memo</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>,
    <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>PBS<span class="pl-pds">"</span></span>,<span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">0.6</span>,<span class="pl-v">cex.lab</span><span class="pl-k">=</span><span class="pl-c1">0.8</span>,
    <span class="pl-v">file.output</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>,<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>pdf<span class="pl-pds">"</span></span>,<span class="pl-v">dpi</span><span class="pl-k">=</span><span class="pl-c1">300</span>,
    <span class="pl-v">verbose</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>,<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">20</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">4</span>,<span class="pl-v">chr.labels.angle</span><span class="pl-k">=</span><span class="pl-c1">45</span>)

<span class="pl-c"><span class="pl-c">#</span>################# FST CMPLOT ##################################################</span>
<span class="pl-smi">ALB</span>\<span class="pl-ii">_LCA2</span> <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">ALB</span>\<span class="pl-ii">_LCA</span>, <span class="pl-smi">chr</span>, <span class="pl-smi">midPos</span>, <span class="pl-smi">Fst01</span>)
<span class="pl-smi">POR</span>\<span class="pl-ii">_SHK2</span> <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">POR</span>\<span class="pl-ii">_SHK</span>, <span class="pl-smi">chr</span>, <span class="pl-smi">midPos</span>, <span class="pl-smi">Fst01</span>)
<span class="pl-smi">SFK</span>\<span class="pl-ii">_GOO2</span> <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">SFK</span>\<span class="pl-ii">_GOO</span>, <span class="pl-smi">chr</span>, <span class="pl-smi">midPos</span>, <span class="pl-smi">Fst01</span>)
<span class="pl-smi">SUS</span>\<span class="pl-ii">_OON2</span> <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">SUS</span>\<span class="pl-ii">_OON</span>, <span class="pl-smi">chr</span>, <span class="pl-smi">midPos</span>, <span class="pl-smi">Fst01</span>)

<span class="pl-smi">ALB</span>\<span class="pl-ii">_POR</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">full</span>\<span class="pl-ii">_join</span>(<span class="pl-smi">ALB</span>\<span class="pl-ii">_LCA2</span>, <span class="pl-smi">POR</span>\<span class="pl-ii">_SHK2</span>, <span class="pl-v">by</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>chr<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>midPos<span class="pl-pds">"</span></span>))
<span class="pl-smi">ALB</span>\<span class="pl-ii">_POR</span>\<span class="pl-ii">_SFK</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">full</span>\<span class="pl-ii">_join</span>(<span class="pl-smi">ALB</span>\<span class="pl-ii">_POR</span>, <span class="pl-smi">SFK</span>\<span class="pl-ii">_GOO2</span>, <span class="pl-v">by</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>chr<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>midPos<span class="pl-pds">"</span></span>))
<span class="pl-smi">ALB</span>\<span class="pl-ii">_POR</span>\<span class="pl-ii">_SFK</span>\<span class="pl-ii">_SUS</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">full</span>\<span class="pl-ii">_join</span>(<span class="pl-smi">ALB</span>\<span class="pl-ii">_POR</span>\<span class="pl-ii">_SFK</span>, <span class="pl-smi">SUS</span>\<span class="pl-ii">_OON2</span>, <span class="pl-v">by</span><span class="pl-k">=</span>c(<span class="pl-s"><span class="pl-pds">"</span>chr<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>midPos<span class="pl-pds">"</span></span>))

<span class="pl-smi">distplot</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">ALB</span>\<span class="pl-ii">_POR</span>\<span class="pl-ii">_SFK</span>\<span class="pl-ii">_SUS</span>
<span class="pl-smi">distplot</span><span class="pl-k">$</span><span class="pl-smi">SNP</span> <span class="pl-k">&lt;-</span> seq(<span class="pl-c1">1</span><span class="pl-k">:</span>nrow(<span class="pl-smi">distplot</span>) )

<span class="pl-smi">distplot</span> <span class="pl-k">&lt;-</span> <span class="pl-e">dplyr</span><span class="pl-k">::</span>select(<span class="pl-smi">distplot</span>, <span class="pl-smi">SNP</span>,<span class="pl-smi">chr</span>,<span class="pl-smi">midPos</span>,
            <span class="pl-smi">Fst01.x</span>, <span class="pl-smi">Fst01.y</span>, <span class="pl-smi">Fst01.x.x</span>, <span class="pl-smi">Fst01.y.y</span> ) %<span class="pl-k">&gt;</span>% 
	set_colnames(<span class="pl-smi">.</span>,c(<span class="pl-s"><span class="pl-pds">"</span>SNP<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Chromosome<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>Position<span class="pl-pds">"</span></span> ,<span class="pl-s"><span class="pl-pds">"</span>ALB<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>POR<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>SFK<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span>SUS<span class="pl-pds">"</span></span>))

<span class="pl-smi">distplot</span> <span class="pl-k">&lt;-</span> filter(<span class="pl-smi">distplot</span>, <span class="pl-k">!</span>grepl(<span class="pl-s"><span class="pl-pds">"</span>scaf<span class="pl-pds">"</span></span>, <span class="pl-smi">distplot</span><span class="pl-k">$</span><span class="pl-smi">Chromosome</span>))
<span class="pl-smi">distplot</span><span class="pl-k">$</span><span class="pl-smi">Chromosome</span> <span class="pl-k">&lt;-</span> gsub(<span class="pl-s"><span class="pl-pds">"</span>Okis<span class="pl-pds">"</span></span>,<span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>,<span class="pl-smi">distplot</span><span class="pl-k">$</span><span class="pl-smi">Chromosome</span>)
<span class="pl-smi">distplot</span>[<span class="pl-smi">distplot</span><span class="pl-k">&lt;</span><span class="pl-c1">0</span>] <span class="pl-k">&lt;-</span> <span class="pl-c1">0</span>

CMplot(<span class="pl-smi">distplot</span>,<span class="pl-v">type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>p<span class="pl-pds">"</span></span>,<span class="pl-v">plot.type</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>m<span class="pl-pds">"</span></span>, <span class="pl-v">multracks</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>,
       <span class="pl-v">LOG10</span><span class="pl-k">=</span><span class="pl-c1">F</span>,<span class="pl-v">threshold</span><span class="pl-k">=</span><span class="pl-c1">NULL</span>,<span class="pl-v">memo</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span><span class="pl-pds">"</span></span>,
       <span class="pl-v">ylab</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>Fst<span class="pl-pds">"</span></span>,<span class="pl-v">cex</span><span class="pl-k">=</span><span class="pl-c1">0.6</span>,<span class="pl-v">cex.lab</span><span class="pl-k">=</span><span class="pl-c1">0.8</span>,
       <span class="pl-v">file.output</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>,<span class="pl-v">file</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>pdf<span class="pl-pds">"</span></span>,<span class="pl-v">dpi</span><span class="pl-k">=</span><span class="pl-c1">300</span>,
       <span class="pl-v">verbose</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>,<span class="pl-v">width</span><span class="pl-k">=</span><span class="pl-c1">20</span>,<span class="pl-v">height</span><span class="pl-k">=</span><span class="pl-c1">4</span>,<span class="pl-v">chr.labels.angle</span><span class="pl-k">=</span><span class="pl-c1">45</span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      9. Association between recombination and outliers
                     </h2>
                     <a aria-label="Permalink: 9. Association between recombination and outliers" class="anchor" href="#9-association-between-recombination-and-outliers" id="user-content-9-association-between-recombination-and-outliers">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ol dir="auto">
                     <li>
                      <strong>
                       estimate recombination
                      </strong>
                     </li>
                    </ol>
                    <p dir="auto">
                     Recombination was estimated using 30X whole Genome data from this
                     <a href="https://www.biorxiv.org/content/10.1101/732750v2" rel="nofollow">
                      preprint
                     </a>
                     .
                     <br/>
                     (note that the WGS data were removed from the published article and will be published later)
                     <br/>
                     LDhat was used with this
                     <a href="https://github.com/QuentinRougemont/LDhat_workflow">
                      pipeline
                     </a>
                     . To reproduce these estimate simply use the pipeline.
                    </p>
                    <p dir="auto">
                     Rho was estimated for each population separately
                    </p>
                    <p dir="auto">
                     Then results were summarized into 250 kb windows and each populations were concatenated together
                    </p>
                    <p dir="auto">
                     The scripts
                     <a href="https://github.com/QuentinRougemont/LDhat_workflow/blob/master/02-scripts/06.run_slidingwindows.sh">
                      here
                     </a>
                     should enables to summarize the data with a bit of modifications depending on your data
                    </p>
                    <ol dir="auto" start="2">
                     <li>
                      <strong>
                       identify outliers
                      </strong>
                     </li>
                    </ol>
                    <p dir="auto">
                     outliers were identified in the steps above for the RDA
                    </p>
                    <ol dir="auto" start="3">
                     <li>
                      <strong>
                       test for how outlier (GEA and Fst/PBS) are influenced by recombination:
                      </strong>
                     </li>
                    </ol>
                    <p dir="auto">
                     Look at the script
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="00-scripts/recombination/association_recombination_GEA.R">
                     <pre class="notranslate"><code>00-scripts/recombination/association_recombination_GEA.R
</code></pre>
                    </div>
                    <p dir="auto">
                     This will allow to perform statistical test:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="* wilcoxon test of difference in mean recombination rate between outliers vs neutral region
* mixed models test ">
                     <pre class="notranslate"><code>* wilcoxon test of difference in mean recombination rate between outliers vs neutral region
* mixed models test 
</code></pre>
                    </div>
                    <p dir="auto">
                     A plot like the one in FigS15 will be produced.
                     <a href="https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/FigS15.git.png" rel="noopener noreferrer" target="_blank">
                      <img alt="example_graph" src="https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/FigS15.git.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     Here we only have the population scale recombination rate (rho = 4
                     <em>
                      Ne
                     </em>
                     r)
                    </p>
                    <p dir="auto">
                     In addition it is possible to explore this by considering either:
                     <br/>
                     * shared outliers between LFMM &amp; RDA
                     <br/>
                     * RDA only outliers
                     <br/>
                     * LFMM only outliers
                     <br/>
                     * outliers that fall into areas of residual tetraploidy can be investigated separately
since there's less variance in recombination in these region
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="There was no difference in recombination when considering the few outliers on region of residual tetraploidy.  
    Obviously, removing these region increased the strength of the signal   ">
                     <pre class="notranslate"><code>There was no difference in recombination when considering the few outliers on region of residual tetraploidy.  
    Obviously, removing these region increased the strength of the signal   
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      10. looking for candidate
                     </h2>
                     <a aria-label="Permalink: 10. looking for candidate" class="anchor" href="#10-looking-for-candidate" id="user-content-10-looking-for-candidate">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      No GO enrichment instead I only use SNPeff and look for meaningfull outliers.
                     </li>
                    </ul>
                    <p dir="auto">
                     SNPeff can be found
                     <a href="https://pcingola.github.io/SnpEff/" rel="nofollow">
                      here
                     </a>
                     and was run with standard parameters
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      11. Other stuff
                     </h2>
                     <a aria-label="Permalink: 11. Other stuff" class="anchor" href="#11-other-stuff" id="user-content-11-other-stuff">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <ul dir="auto">
                     <li>
                      below is a zoomable map for BC/Thompson/HaidaGwaii that include all sample sites used here.
                     </li>
                     <li>
                      may be usefull for users
                     </li>
                     <li>
                      This map was obtained with very simple code in R
                     </li>
                    </ul>
                    <p dir="auto">
                     <a href="https://github.com/QuentinRougemont/coho_ldscp_genomics/blob/main/pictures/sampling_map.git.png" rel="noopener noreferrer" target="_blank">
                      <img alt="example_graph" src="https://github.com/QuentinRougemont/coho_ldscp_genomics/raw/main/pictures/sampling_map.git.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" id="react-codespace-csrf" type="hidden" value="JBMg9L0BYDOGUsLcp7vzE3ufAdbTwGvIjNffuxUk4kszy79bIQoYU4UeOKXx727dDSUsNTNpEFL9BPsQEId8aA">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 scripts to reproduce our landscape genomics results in coho salmon
                </p>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/QuentinRougemont/coho_ldscp_genomics/hovercards/citation/sidebar_partial?tree_name=main">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   3
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  fork
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2FQuentinRougemont%2Fcoho_ldscp_genomics&amp;report=QuentinRougemont+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/QuentinRougemont/coho_ldscp_genomics/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/QuentinRougemont/packages?repo_name=coho_ldscp_genomics">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/QuentinRougemont/coho_ldscp_genomics/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                Languages
               </h2>
               <div class="mb-2">
                <span class="Progress" data-view-component="true">
                 <span aria-label="R 99.2" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#198CE7 !important;;width: 99.2%;">
                 </span>
                 <span aria-label="Shell 0.8" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#89e051 !important;;width: 0.8%;">
                 </span>
                </span>
               </div>
               <ul class="list-style-none">
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/QuentinRougemont/coho_ldscp_genomics/search?l=r">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#198CE7;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   R
                  </span>
                  <span>
                   99.2%
                  </span>
                 </a>
                </li>
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/QuentinRougemont/coho_ldscp_genomics/search?l=shell">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#89e051;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   Shell
                  </span>
                  <span>
                   0.8%
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <style>
    .user-mention[href$="/ninenia"] {
        color: var(--color-user-mention-fg);
        background-color: var(--color-user-mention-bg);
        border-radius: 2px;
        margin-left: -2px;
        margin-right: -2px;
        padding: 0 2px;
      }
   </style>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
