<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light_v2-e830ce163aae.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark_v2-bc3c552285ea.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed_v2-191796a86a54.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast_v2-e0a0b6e42616.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind_v2-4fd40c9324e8.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind_v2-2f299c61eceb.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast_v2-05d723322773.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia_v2-5679d6101565.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia_v2-0ff66d76cc55.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","repository_suggester_elastic_search","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-bf4f9425a973.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-ad941a96f009.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-31df4ccdeed4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-340b2bef927d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-5d7607113ea3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-b2cd2ed93da5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_github_catalyst_lib_index_-eccae9-1932eeecf006.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-318896a86f35.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-9fad2423070b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-f1bd820e03da.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         stemangiola/tidybulk: Brings bulk and pseudobulk transcriptomics to the tidyverse
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="FB70:6E100:57AF7BE:7C195E5:66058E30" data-turbo-transient="true" name="request-id">
         <meta content="22f598a296c639e8b750dc2cf1a20323836185b0d8b152a2e56d5461ad76529a" data-turbo-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6bnVsbCwicmVxdWVzdF9pZCI6IkZCNzA6NkUxMDA6NTdBRjdCRTo3QzE5NUU1OjY2MDU4RTMwIiwidmlzaXRvcl9pZCI6IjU1NjM2ODU5OTEwODQ4MjkyODYiLCJyZWdpb25fZWRnZSI6ImlhZCIsInJlZ2lvbl9yZW5kZXIiOiJpYWQifQ==" data-turbo-transient="true" name="visitor-payload">
           <meta content="c67f804a0adb8c22cf8f4d74479de624128b37f8ba8402b03808dbdec1625ac1" data-turbo-transient="true" name="visitor-hmac">
            <meta content="repository:205042805" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="126681664" name="octolytics-actor-id">
               <meta content="ninenia" name="octolytics-actor-login">
                <meta content="57d6abdd600494be8ac2f82b25b957e4c0d029a4d648c10497790171de0492de" name="octolytics-actor-hash">
                 <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
                  <meta content="ninenia" name="user-login"/>
                  <link href="/sessions/sudo_modal" rel="sudo-modal"/>
                  <meta content="width=device-width" name="viewport"/>
                  <meta content="Brings bulk and pseudobulk transcriptomics to the tidyverse - stemangiola/tidybulk" name="description"/>
                  <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
                  <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
                  <meta content="1401488693436528" property="fb:app_id"/>
                  <meta content="app-id=1477376905, app-argument=https://github.com/stemangiola/tidybulk" name="apple-itunes-app">
                   <meta content="https://repository-images.githubusercontent.com/205042805/b511fc00-7364-11ea-978e-63ab63f7142a" name="twitter:image:src">
                    <meta content="@github" name="twitter:site">
                     <meta content="summary_large_image" name="twitter:card">
                      <meta content="stemangiola/tidybulk: Brings bulk and pseudobulk transcriptomics to the tidyverse" name="twitter:title">
                       <meta content="Brings bulk and pseudobulk transcriptomics to the tidyverse - stemangiola/tidybulk" name="twitter:description">
                        <meta content="https://repository-images.githubusercontent.com/205042805/b511fc00-7364-11ea-978e-63ab63f7142a" property="og:image"/>
                        <meta content="Brings bulk and pseudobulk transcriptomics to the tidyverse - stemangiola/tidybulk" property="og:image:alt"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="stemangiola/tidybulk: Brings bulk and pseudobulk transcriptomics to the tidyverse" property="og:title"/>
                        <meta content="https://github.com/stemangiola/tidybulk" property="og:url"/>
                        <meta content="Brings bulk and pseudobulk transcriptomics to the tidyverse - stemangiola/tidybulk" property="og:description"/>
                        <link data-refresh-url="/_alive" data-session-id="fae53153a7a7c8d08a43b9810520b9ce4325ce8b4e99a06c5fc3f52c85cc22fe" href="wss://alive.github.com/_sockets/u/126681664/ws?session=eyJ2IjoiVjMiLCJ1IjoxMjY2ODE2NjQsInMiOjEzMDQyODIxMTYsImMiOjE3MzQ2OTg0NzUsInQiOjE3MTE2NDAxMTd9--8fe613d09f0f190dcc8102fad925915314b33a96c469a5ce9eff0bbbb407a0bb" rel="shared-web-socket"/>
                        <link href="/assets-cdn/worker/socket-worker-9cc1149b224c.js" rel="shared-web-socket-src"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="all" name="keyboard-shortcuts-preference"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="a81c045767350c9f2e551b272402db2e99346c1d2710552f41e25d405666b5eb" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="3037d78571ecdebb88047af8e128a53bf5dc96ac8c533579876941ff0be8ae32" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/stemangiola/tidybulk git https://github.com/stemangiola/tidybulk.git" name="go-import"/>
                        <meta content="7232890" name="octolytics-dimension-user_id">
                         <meta content="stemangiola" name="octolytics-dimension-user_login">
                          <meta content="205042805" name="octolytics-dimension-repository_id">
                           <meta content="stemangiola/tidybulk" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="205042805" name="octolytics-dimension-repository_network_root_id">
                               <meta content="stemangiola/tidybulk" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/stemangiola/tidybulk" rel="canonical"/>
                                <meta content="logged-in env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <meta content="/windows-tile.png" name="msapplication-TileImage"/>
                                 <meta content="#ffffff" name="msapplication-TileColor"/>
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-in env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-in env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="p-3 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_allex_crc32_lib_crc32_esm_js-node_modules_github_mini-throttle_dist_deco-b38cad-748e74df23ce.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_clipboard-copy-element_dist_index_esm_js-node_modules_delegated-e-b37f7d-2f24d321a3fb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_command-palette_items_help-item_ts-app_assets_modules_github_comman-48ad9d-a9ed17f2ebff.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/command-palette-f620e569db0b.js" type="application/javascript">
    </script>
    <header class="AppHeader">
     <div class="AppHeader-globalBar pb-2 js-global-bar">
      <div class="AppHeader-globalBar-start">
       <deferred-side-panel data-url="/_side-panels/global">
        <include-fragment data-target="deferred-side-panel.fragment">
         <button aria-label="Open global navigation menu" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-bg-transparent p-0 color-fg-muted" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-94f175c4-49f1-48a0-b5d0-b32c78b790e5" data-view-component="true" id="dialog-show-dialog-94f175c4-49f1-48a0-b5d0-b32c78b790e5" type="button">
          <svg aria-hidden="true" class="octicon octicon-three-bars Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1 2.75A.75.75 0 0 1 1.75 2h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 2.75Zm0 5A.75.75 0 0 1 1.75 7h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 7.75ZM1.75 12h12.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Z">
           </path>
          </svg>
         </button>
         <dialog-helper>
          <dialog aria-describedby="dialog-94f175c4-49f1-48a0-b5d0-b32c78b790e5-description" aria-labelledby="dialog-94f175c4-49f1-48a0-b5d0-b32c78b790e5-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-left SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-94f175c4-49f1-48a0-b5d0-b32c78b790e5">
           <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title sr-only" id="dialog-94f175c4-49f1-48a0-b5d0-b32c78b790e5-title">
               Global navigation
              </h1>
              <div class="d-flex" data-view-component="true">
               <div class="AppHeader-logo position-relative" data-view-component="true">
                <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
                 <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
                 </path>
                </svg>
               </div>
              </div>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-94f175c4-49f1-48a0-b5d0-b32c78b790e5" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="dialog-94f175c4-49f1-48a0-b5d0-b32c78b790e5-title">
            <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
             <div class="d-flex flex-column mb-3" data-view-component="true">
              <nav aria-label="Site navigation" class="ActionList" data-view-component="true">
               <nav-list>
                <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"HOME","label":null}' data-hotkey="g d" data-view-component="true" href="/dashboard" id="item-69575daa-86b0-4da3-93ba-a343b86327be">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-home" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M6.906.664a1.749 1.749 0 0 1 2.187 0l5.25 4.2c.415.332.657.835.657 1.367v7.019A1.75 1.75 0 0 1 13.25 15h-3.5a.75.75 0 0 1-.75-.75V9H7v5.25a.75.75 0 0 1-.75.75h-3.5A1.75 1.75 0 0 1 1 13.25V6.23c0-.531.242-1.034.657-1.366l5.25-4.2Zm1.25 1.171a.25.25 0 0 0-.312 0l-5.25 4.2a.25.25 0 0 0-.094.196v7.019c0 .138.112.25.25.25H5.5V8.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 .75.75v5.25h2.75a.25.25 0 0 0 .25-.25V6.23a.25.25 0 0 0-.094-.195Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Home
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"ISSUES","label":null}' data-hotkey="g i" data-view-component="true" href="/issues" id="item-8dd27482-db57-4b36-9856-ed74fb517ae8">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS","label":null}' data-hotkey="g p" data-view-component="true" href="/pulls" id="item-c9d7b499-94cb-4f3d-b478-0240404b712d">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="projects" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROJECTS","label":null}' data-view-component="true" href="/projects" id="item-5b13e6d1-214d-4977-ac4e-d3a22f50ecb1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"DISCUSSIONS","label":null}' data-view-component="true" href="/discussions" id="item-c5a506f8-f550-4457-a089-d94768ae69a3">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Discussions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"CODESPACES","label":null}' data-view-component="true" href="https://github.com/codespaces" id="item-c2b6ff2f-b41e-4256-984f-8ec477ab4500">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Codespaces
                   </span>
                  </a>
                 </li>
                 <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"EXPLORE","label":null}' data-view-component="true" href="/explore" id="item-5e9243ac-592f-4bb2-8823-7abf307c53d1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-telescope" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.184 1.143v-.001l1.422 2.464a1.75 1.75 0 0 1-.757 2.451L3.104 11.713a1.75 1.75 0 0 1-2.275-.702l-.447-.775a1.75 1.75 0 0 1 .53-2.32L11.682.573a1.748 1.748 0 0 1 2.502.57Zm-4.709 9.32h-.001l2.644 3.863a.75.75 0 1 1-1.238.848l-1.881-2.75v2.826a.75.75 0 0 1-1.5 0v-2.826l-1.881 2.75a.75.75 0 1 1-1.238-.848l2.049-2.992a.746.746 0 0 1 .293-.253l1.809-.87a.749.749 0 0 1 .944.252ZM9.436 3.92h-.001l-4.97 3.39.942 1.63 5.42-2.61Zm3.091-2.108h.001l-1.85 1.26 1.505 2.605 2.016-.97a.247.247 0 0 0 .13-.151.247.247 0 0 0-.022-.199l-1.422-2.464a.253.253 0 0 0-.161-.119.254.254 0 0 0-.197.038ZM1.756 9.157a.25.25 0 0 0-.075.33l.447.775a.25.25 0 0 0 .325.1l1.598-.769-.83-1.436-1.465 1Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Explore
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"MARKETPLACE","label":null}' data-view-component="true" href="/marketplace" id="item-1fe88d1b-8c8e-4e62-95fc-e80f2940eae1">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Marketplace
                   </span>
                  </a>
                 </li>
                </ul>
               </nav-list>
              </nav>
              <div class="my-3 d-flex flex-justify-center height-full" data-view-component="true">
               <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                </circle>
                <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                </path>
               </svg>
              </div>
             </div>
             <div class="flex-1" data-view-component="true">
             </div>
             <div class="px-2" data-view-component="true">
              <p class="color-fg-subtle text-small text-light">
               © 2024 GitHub, Inc.
              </p>
              <div class="d-flex flex-wrap text-small text-light" data-view-component="true">
               <a class="Link mr-2" data-view-component="true" href="https://github.com/about" target="_blank">
                About
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.blog" target="_blank">
                Blog
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service" target="_blank">
                Terms
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement" target="_blank">
                Privacy
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.com/security" target="_blank">
                Security
               </a>
               <a class="Link mr-3" data-view-component="true" href="https://www.githubstatus.com/" target="_blank">
                Status
               </a>
              </div>
             </div>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </include-fragment>
       </deferred-side-panel>
       <a aria-label="Homepage " class="AppHeader-logo ml-2" data-analytics-event='{"category":"Header","action":"go to dashboard","label":"icon:logo"}' data-hotkey="g d" data-turbo="false" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github v-align-middle color-fg-default" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="AppHeader-context">
        <div class="AppHeader-context-compact">
         <button aria-expanded="false" aria-haspopup="dialog" aria-label="Page context: stemangiola / tidybulk" class="AppHeader-context-compact-trigger Truncate Button--secondary Button--medium Button box-shadow-none" data-show-dialog-id="context-region-dialog" data-view-component="true" id="dialog-show-context-region-dialog" type="button">
          <span class="Button-content">
           <span class="Button-label">
            <span class="AppHeader-context-compact-lead">
             <span class="AppHeader-context-compact-parentItem">
              stemangiola
             </span>
             <span class="no-wrap">
              /
             </span>
            </span>
            <strong class="AppHeader-context-compact-mainItem d-flex flex-items-center Truncate">
             <span class="Truncate-text">
              tidybulk
             </span>
            </strong>
           </span>
          </span>
         </button>
         <dialog-helper>
          <dialog aria-describedby="context-region-dialog-description" aria-labelledby="context-region-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-view-component="true" id="context-region-dialog">
           <div class="Overlay-header" data-view-component="true">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title" id="context-region-dialog-title">
               Navigate back to
              </h1>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="context-region-dialog" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="context-region-dialog-title">
            <div class="Overlay-body" data-view-component="true">
             <ul class="list-style-none" role="list">
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"stemangiola","screen_size":"compact"}' data-view-component="true" href="/stemangiola">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-person mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                  </path>
                 </svg>
                 stemangiola
                </span>
               </a>
              </li>
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"tidybulk","screen_size":"compact"}' data-view-component="true" href="/stemangiola/tidybulk">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-repo mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                  </path>
                 </svg>
                 tidybulk
                </span>
               </a>
              </li>
             </ul>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </div>
        <div class="AppHeader-context-full">
         <nav aria-label="Page context" role="navigation">
          <ul class="list-style-none" role="list">
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"stemangiola","screen_size":"full"}' data-hovercard-type="user" data-hovercard-url="/users/stemangiola/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" data-view-component="true" href="/stemangiola">
             <span class="AppHeader-context-item-label">
              stemangiola
             </span>
            </a>
            <span class="AppHeader-context-item-separator">
             /
            </span>
           </li>
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"tidybulk","screen_size":"full"}' data-view-component="true" href="/stemangiola/tidybulk">
             <span class="AppHeader-context-item-label">
              tidybulk
             </span>
            </a>
           </li>
          </ul>
         </nav>
        </div>
       </div>
      </div>
      <div class="AppHeader-globalBar-end">
       <div class="AppHeader-search">
        <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;input type="hidden" value="kIXoQIzuHzhlc5yBaQ5oqixFfd2gY3wEVAoF4lZZ08HApuTIHmPA99LpxK_MmYrxGMm9NpRIyZWKcXtc-51uKQ" data-csrf="true" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="stemangiola" data-current-repository="stemangiola/tidybulk" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="fdPiQ1khocFW-p2JJACtczaZjD8scDZffWdlf2JpSB01GfVpf2fotAFW71DCBV1iZD3OlgoXxjARhSTUhVdTUQ" data-header-redesign-enabled="true" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="true" data-max-custom-scopes="10" data-scope="repo:stemangiola/tidybulk">
         <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center height-auto color-bg-transparent border-0 color-fg-subtle mx-0" data-action="click:qbsearch-input#searchInputContainerClicked">
          <button aria-expanded="false" aria-haspopup="dialog" aria-label="Search or jump to…" class="AppHeader-button AppHeader-search-whenNarrow" data-action="click:qbsearch-input#handleExpand" type="button">
           <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
            </path>
           </svg>
          </button>
          <div class="AppHeader-search-whenRegular">
           <div class="AppHeader-search-wrap AppHeader-search-wrap--hasTrailing">
            <div class="AppHeader-search-control">
             <label aria-label="Search or jump to…" class="AppHeader-search-visual--leading" for="AppHeader-searchInput">
              <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
               </path>
              </svg>
             </label>
             <button aria-describedby="search-error-message-flash" class="AppHeader-searchButton form-control input-contrast text-left color-fg-subtle no-wrap" data-action="click:qbsearch-input#handleExpand" data-analytics-event='{"category":"Global navigation","action":"SEARCH","label":null}' data-hotkey="s,/" data-target="qbsearch-input.inputButton" type="button">
              <div class="overflow-hidden">
               <span data-target="qbsearch-input.inputButtonText" id="qb-input-query">
                Type
                <kbd class="AppHeader-search-kbd">
                 /
                </kbd>
                to search
               </span>
              </div>
             </button>
            </div>
            <button class="AppHeader-search-action--trailing js-activate-command-palette" data-analytics-event='{"category":"SiteHeaderComponent","action":"command_palette","label":"open command palette"}' id="AppHeader-commandPalette-button" type="button">
             <svg aria-hidden="true" class="octicon octicon-command-palette" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m6.354 8.04-4.773 4.773a.75.75 0 1 0 1.061 1.06L7.945 8.57a.75.75 0 0 0 0-1.06L2.642 2.206a.75.75 0 0 0-1.06 1.061L6.353 8.04ZM8.75 11.5a.75.75 0 0 0 0 1.5h5.5a.75.75 0 0 0 0-1.5h-5.5Z">
              </path>
             </svg>
            </button>
            <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="AppHeader-commandPalette-button" id="tooltip-3781188d-f3bc-4048-9bb8-bccf39e68c81" popover="manual">
             Command palette
            </tool-tip>
           </div>
          </div>
          <input class="js-site-search-type-field" name="type" type="hidden"/>
          <div class="Overlay--hidden" data-modal-dialog-overlay="">
           <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-medium Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
            <h1 class="sr-only" id="search-suggestions-dialog-header">
             Search code, repositories, users, issues, pull requests...
            </h1>
            <div class="Overlay-body Overlay-body--paddingNone">
             <div data-view-component="true">
              <div class="search-suggestions position-absolute width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                 <div class="FormControl FormControl--fullWidth">
                  <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                   Search
                  </label>
                  <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                   <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                    <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </span>
                   <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                    <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                    </div>
                    <div class="QueryBuilder-InputWrapper">
                     <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                     </div>
                     <input aria-describedby="validation-5b4d81a7-cbf6-4b55-9108-79a55f238d50" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                     </input>
                    </div>
                   </div>
                   <span class="sr-only" id="query-builder-test-clear">
                    Clear
                   </span>
                   <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                    <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <template id="search-icon">
                   <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-icon">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="file-code-icon">
                   <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                    </path>
                   </svg>
                  </template>
                  <template id="history-icon">
                   <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                    </path>
                   </svg>
                  </template>
                  <template id="repo-icon">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="bookmark-icon">
                   <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="plus-circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                    </path>
                   </svg>
                  </template>
                  <template id="trash-icon">
                   <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="team-icon">
                   <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="project-icon">
                   <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                    </path>
                   </svg>
                  </template>
                  <template id="pencil-icon">
                   <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                    </path>
                   </svg>
                  </template>
                  <template id="copilot-icon">
                   <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                    </path>
                    <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="workflow-icon">
                   <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="book-icon">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-review-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="codespaces-icon">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-discussion-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="organization-icon">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="rocket-icon">
                   <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="shield-check-icon">
                   <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="heart-icon">
                   <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="server-icon">
                   <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="globe-icon">
                   <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                    </path>
                   </svg>
                  </template>
                  <template id="issue-opened-icon">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-mobile-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="package-icon">
                   <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                    </path>
                   </svg>
                  </template>
                  <template id="credit-card-icon">
                   <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                    </path>
                    <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                    </path>
                   </svg>
                  </template>
                  <template id="play-icon">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </template>
                  <template id="gift-icon">
                   <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-square-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-desktop-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                    </path>
                   </svg>
                  </template>
                  <div class="position-relative">
                   <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                   </ul>
                  </div>
                  <div class="FormControl-inlineValidation" hidden="hidden" id="validation-5b4d81a7-cbf6-4b55-9108-79a55f238d50">
                   <span class="FormControl-inlineValidation--visual">
                    <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                     <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                     </path>
                    </svg>
                   </span>
                   <span>
                   </span>
                  </div>
                 </div>
                 <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                 </div>
                </query-builder>
               </form>
               <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                 Search syntax tips
                </a>
                <div class="d-flex flex-1">
                </div>
                <button class="Button--link Button--medium Button color-fg-accent text-normal ml-2" data-action="click:qbsearch-input#showFeedbackDialog" data-view-component="true" type="button">
                 <span class="Button-content">
                  <span class="Button-label">
                   Give feedback
                  </span>
                 </span>
                </button>
               </div>
              </div>
             </div>
            </div>
           </modal-dialog>
          </div>
         </div>
         <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
         </div>
         <div class="color-fg-default">
          <dialog-helper>
           <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
            <div class="Overlay-header" data-view-component="true">
             <div class="Overlay-headerContentWrap">
              <div class="Overlay-titleWrap">
               <h1 class="Overlay-title" id="feedback-dialog-title">
                Provide feedback
               </h1>
              </div>
              <div class="Overlay-actionWrap">
               <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
              </div>
             </div>
            </div>
            <scrollable-region data-labelled-by="feedback-dialog-title">
             <div class="Overlay-body" data-view-component="true">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
               <input name="authenticity_token" type="hidden" value="tzkqGpL4ZSaasPEgblSXHS2N7Gzi2OGolJ_MEYppJrwlizMiWi5WvCqS_yRQTZbs8IYhr7j4B_2aH9mrfDgjBw"/>
               <p>
                We read every piece of feedback, and take your input very seriously.
               </p>
               <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
               <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
               <label for="include_email" style="font-weight: normal">
                Include my email address so I can be contacted
               </label>
              </form>
             </div>
            </scrollable-region>
            <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
             <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
              Cancel
             </button>
             <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
              Submit feedback
             </button>
            </div>
           </dialog>
          </dialog-helper>
          <custom-scopes data-target="qbsearch-input.customScopesManager">
           <dialog-helper>
            <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
             <div class="Overlay-header Overlay-header--divided" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                 Saved searches
                </h1>
                <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                 Use saved searches to filter your results more quickly
                </h2>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="custom-scopes-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <div data-target="custom-scopes.customScopesModalDialogFlash">
               </div>
               <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                 <input name="authenticity_token" type="hidden" value="WKzY9fBZ4LTmZv7Lrl1piinMmgDWKqMNE7jCtympqbgUXrpPqNi-ba0PQ2bkiKS34Jz-F4Av4daw4BMPNu7w7Q">
                  <div data-target="custom-scopes.customScopesModalDialogFlash">
                  </div>
                  <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                  <div class="form-group">
                   <label for="custom_scope_name">
                    Name
                   </label>
                   <auto-check required="" src="/search/custom_scopes/check_name">
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                    <input data-csrf="true" type="hidden" value="q7jVqhYT9BUqsuk1Y6TQNS7tGJoJVq9ew4QzLirLJeOz5ZgMQZFdFqmmoLYj9o5fmCodHSePT8auJJH8EqT2Kw">
                    </input>
                   </auto-check>
                  </div>
                  <div class="form-group">
                   <label for="custom_scope_query">
                    Query
                   </label>
                   <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                  </div>
                  <p class="text-small color-fg-muted">
                   To see all available qualifiers, see our
                   <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                    documentation
                   </a>
                   .
                  </p>
                 </input>
                </form>
               </div>
               <div data-target="custom-scopes.manageCustomScopesForm">
                <div data-target="custom-scopes.list">
                </div>
               </div>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
              <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
               Create saved search
              </button>
             </div>
            </dialog>
           </dialog-helper>
          </custom-scopes>
         </div>
        </qbsearch-input>
        <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="yJmOUV6M-Q4PWiPMRxiwkT7IZjkG1_SOiKb_JZteI8cPzOFzR4s_bOiBuTGOo8uC7nC-84NxMr6HxaGAA8pCzQ">
        </input>
       </div>
       <div class="AppHeader-actions position-relative">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="global-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="global-create-menu-button" popovertarget="global-create-menu-overlay" type="button">
           <span class="Button-content">
            <span class="Button-visual Button-leadingVisual">
             <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
              </path>
             </svg>
            </span>
            <span class="Button-label">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </span>
           </span>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="global-create-menu-button" id="tooltip-c4ad218e-21a2-44d6-87ac-b06920ed7662" popover="manual">
           Create new...
          </tool-tip>
          <anchored-position align="end" anchor="global-create-menu-button" anchor-offset="normal" data-view-component="true" id="global-create-menu-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="global-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="global-create-menu-list" role="menu">
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-a2f45eae-af66-433b-b7c0-41d18d9c52d5" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New repository
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-c7911b5d-75f4-4626-8560-10383a180273" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Import repository
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-56aca225-59d1-40b7-8384-846bf029f510" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New codespace
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-62891e4a-7626-4be4-9960-f401a90bcfc5" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New gist
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-74a96f05-5808-4c84-8370-399c9af4ac2a" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New organization
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
        <a aria-labelledby="tooltip-9b10236d-7ef8-4189-b19b-d97e4dfdc182" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"ISSUES_HEADER","label":null}' data-view-component="true" href="/issues" id="icon-button-2827f1f5-9b78-4961-881c-11ede4f659f8">
         <svg aria-hidden="true" class="octicon octicon-issue-opened Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-2827f1f5-9b78-4961-881c-11ede4f659f8" id="tooltip-9b10236d-7ef8-4189-b19b-d97e4dfdc182" popover="manual">
         Issues
        </tool-tip>
        <a aria-labelledby="tooltip-184cab8e-02ae-4be8-b2a1-5f7d64f48f8a" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS_HEADER","label":null}' data-view-component="true" href="/pulls" id="icon-button-201e29d3-ad3c-476d-aa66-fda1ff018235">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-201e29d3-ad3c-476d-aa66-fda1ff018235" id="tooltip-184cab8e-02ae-4be8-b2a1-5f7d64f48f8a" popover="manual">
         Pull requests
        </tool-tip>
       </div>
       <notification-indicator class="js-socket-channel" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY0MDExN30=--4145f0d212f0e1143d18c0210820ecb490355da2d3640aa864d55c8e55f7e93b" data-fetch-indicator-enabled="true" data-fetch-indicator-src="/notifications/indicator" data-header-redesign-enabled="true" data-indicator-mode="none" data-tooltip-global="You have unread notifications" data-tooltip-none="You have no unread notifications" data-tooltip-unavailable="Notifications are unavailable at the moment." data-view-component="true">
        <a aria-label="Notifications" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"NOTIFICATIONS_HEADER","label":null}' data-hotkey="g n" data-target="notification-indicator.link" data-view-component="true" href="/notifications" id="AppHeader-notifications-button">
         <svg aria-hidden="true" class="octicon octicon-inbox Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.8 2.06A1.75 1.75 0 0 1 4.41 1h7.18c.7 0 1.333.417 1.61 1.06l2.74 6.395c.04.093.06.194.06.295v4.5A1.75 1.75 0 0 1 14.25 15H1.75A1.75 1.75 0 0 1 0 13.25v-4.5c0-.101.02-.202.06-.295Zm1.61.44a.25.25 0 0 0-.23.152L1.887 8H4.75a.75.75 0 0 1 .6.3L6.625 10h2.75l1.275-1.7a.75.75 0 0 1 .6-.3h2.863L11.82 2.652a.25.25 0 0 0-.23-.152Zm10.09 7h-2.875l-1.275 1.7a.75.75 0 0 1-.6.3h-3.5a.75.75 0 0 1-.6-.3L4.375 9.5H1.5v3.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-target="notification-indicator.tooltip" data-type="description" data-view-component="true" for="AppHeader-notifications-button" id="tooltip-b8cfb4a0-071e-4534-b2b9-80dd1ae19c23" popover="manual">
         Notifications
        </tool-tip>
       </notification-indicator>
       <div class="AppHeader-user">
        <deferred-side-panel data-url="/_side-panels/user?react_global_nav=false&amp;repository_id=205042805">
         <include-fragment data-target="deferred-side-panel.fragment">
          <user-drawer-side-panel>
           <button aria-label="Open user account menu" class="AppHeader-logo Button--invisible Button--medium Button Button--invisible-noVisuals color-bg-transparent p-0" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-0678624e-2342-4561-9969-990c7b06d005" data-view-component="true" id="dialog-show-dialog-0678624e-2342-4561-9969-990c7b06d005" type="button">
            <span class="Button-content">
             <span class="Button-label">
              <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
             </span>
            </span>
           </button>
           <dialog-helper>
            <dialog aria-describedby="dialog-0678624e-2342-4561-9969-990c7b06d005-description" aria-labelledby="dialog-0678624e-2342-4561-9969-990c7b06d005-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-right SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-0678624e-2342-4561-9969-990c7b06d005">
             <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title sr-only" id="dialog-0678624e-2342-4561-9969-990c7b06d005-title">
                 Account menu
                </h1>
                <div class="d-flex" data-view-component="true">
                 <div class="AppHeader-logo position-relative" data-view-component="true">
                  <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
                 </div>
                 <div class="overflow-hidden d-flex width-full" data-view-component="true">
                  <div class="lh-condensed overflow-hidden d-flex flex-column flex-justify-center ml-2 f5 mr-auto width-full" data-view-component="true">
                   <span class="Truncate text-bold" data-view-component="true">
                    <span class="Truncate-text" data-view-component="true">
                     ninenia
                    </span>
                   </span>
                  </div>
                  <action-menu class="d-sm-none d-md-none d-lg-none" data-select-variant="none" data-view-component="true">
                   <focus-group direction="vertical" mnemonics="" retain="">
                    <button aria-controls="user-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="user-create-menu-button" popovertarget="user-create-menu-overlay" type="button">
                     <span class="Button-content">
                      <span class="Button-visual Button-leadingVisual">
                       <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                        </path>
                       </svg>
                      </span>
                      <span class="Button-label">
                       <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                        </path>
                       </svg>
                      </span>
                     </span>
                    </button>
                    <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="user-create-menu-button" id="tooltip-a05ce933-9d65-450b-b886-5503b86cf2ed" popover="manual">
                     Create new...
                    </tool-tip>
                    <anchored-position align="end" anchor="user-create-menu-button" anchor-offset="normal" data-view-component="true" id="user-create-menu-overlay" popover="auto" side="outside-bottom">
                     <div class="Overlay Overlay--size-auto" data-view-component="true">
                      <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
                       <action-list>
                        <div data-view-component="true">
                         <ul aria-labelledby="user-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="user-create-menu-list" role="menu">
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-525c2a28-725a-4fd6-b7f8-4e7385403d9c" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New repository
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-70770875-9505-4e8c-93b5-a7e5c03f3455" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             Import repository
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-e077f2b2-b1cd-4326-abe7-001ca9a2651d" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                              </path>
                              <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New codespace
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-bad4ffbb-6bf5-4b9c-8363-db1542619b53" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New gist
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-497d08b7-85ba-47eb-8bf9-0f193604ae50" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New organization
                            </span>
                           </a>
                          </li>
                         </ul>
                        </div>
                       </action-list>
                      </div>
                     </div>
                    </anchored-position>
                   </focus-group>
                  </action-menu>
                 </div>
                </div>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-0678624e-2342-4561-9969-990c7b06d005" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="dialog-0678624e-2342-4561-9969-990c7b06d005-title">
              <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
               <div class="d-flex flex-column mb-3" data-view-component="true">
                <nav aria-label="User navigation" class="ActionList" data-view-component="true">
                 <nav-list>
                  <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-9f9253da-6666-41d5-9446-f5bd687d53e9" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROFILE","label":null}' data-view-component="true" href="https://github.com/ninenia" id="item-8f08e12d-afb0-4086-b661-941312d3f1d9">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-person" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your profile
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-34e68a42-eb94-4cc5-9edc-f48180c59b35" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_REPOSITORIES","label":null}' data-view-component="true" href="/ninenia?tab=repositories" id="item-8dc380a6-45e8-4fb6-84dd-69f3cff27908">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your repositories
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_PROJECTS","label":null}' data-view-component="true" href="/ninenia?tab=projects" id="item-40793a6a-5215-4d74-b20e-20f8aa02eeb8">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your projects
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-daaee420-4da1-443c-83cc-9de60bc4985e" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-54c52a3b-124b-4e03-bff8-66cad73b2f27" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_STARS","label":null}' data-view-component="true" href="/ninenia?tab=stars" id="item-bfbcc53c-490d-4ebb-a5cb-21eed44fb138">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-star" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your stars
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SPONSORS","label":null}' data-view-component="true" href="/sponsors/accounts" id="item-d65af8d0-2ceb-41c7-9d1c-521e15549117">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your sponsors
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_GISTS","label":null}' data-view-component="true" href="https://gist.github.com/mine" id="item-24474c0b-5875-46ad-a825-a7fb8bd708de">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your gists
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-de290998-3fb0-4e5f-b840-4eba6714a04f" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-fe399386-4f92-4642-840d-f6689f268ade" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SETTINGS","label":null}' data-view-component="true" href="/settings/profile" id="item-4bf859d2-40d6-435c-946d-699340da424f">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-gear" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Settings
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SUPPORT","label":null}' data-view-component="true" href="https://support.github.com" id="item-2417268d-c8b2-4624-82a0-4a768e8d98bc">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Support
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"COMMUNITY","label":null}' data-view-component="true" href="https://community.github.com" id="item-2b137dc8-4162-44ac-a46b-93632114a617">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Community
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent" data-analytics-event='{"category":"Global navigation","action":"LOGOUT","label":null}' data-view-component="true" href="/logout" id="item-0c38c56d-0ffa-4bc4-ae5a-3756deeaf6c8">
                     <span class="ActionListItem-label" data-view-component="true">
                      Sign out
                     </span>
                    </a>
                   </li>
                  </ul>
                 </nav-list>
                </nav>
               </div>
              </div>
             </scrollable-region>
            </dialog>
           </dialog-helper>
          </user-drawer-side-panel>
         </include-fragment>
        </deferred-side-panel>
       </div>
       <div class="position-absolute mt-2">
        <site-header-logged-in-user-menu>
        </site-header-logged-in-user-menu>
       </div>
      </div>
     </div>
     <div class="AppHeader-localBar">
      <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav" data-pjax="#js-repo-pjax-container" data-view-component="true">
       <ul class="UnderlineNav-body list-style-none" data-view-component="true">
        <li class="d-inline-flex" data-view-component="true">
         <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /stemangiola/tidybulk" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/stemangiola/tidybulk" id="code-tab">
          <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
           </path>
          </svg>
          <span data-content="Code">
           Code
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /stemangiola/tidybulk/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/stemangiola/tidybulk/issues" id="issues-tab">
          <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
           </path>
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
           </path>
          </svg>
          <span data-content="Issues">
           Issues
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="issues-repo-tab-count" title="12">
           12
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /stemangiola/tidybulk/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/stemangiola/tidybulk/pulls" id="pull-requests-tab">
          <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
           </path>
          </svg>
          <span data-content="Pull requests">
           Pull requests
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="pull-requests-repo-tab-count" title="4">
           4
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /stemangiola/tidybulk/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/stemangiola/tidybulk/actions" id="actions-tab">
          <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
           </path>
          </svg>
          <span data-content="Actions">
           Actions
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /stemangiola/tidybulk/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/stemangiola/tidybulk/projects" id="projects-tab">
          <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
           </path>
          </svg>
          <span data-content="Projects">
           Projects
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /stemangiola/tidybulk/security" data-tab-item="i5security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/stemangiola/tidybulk/security" id="security-tab">
          <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
           </path>
          </svg>
          <span data-content="Security">
           Security
          </span>
          <include-fragment accept="text/fragment+html" src="/stemangiola/tidybulk/security/overall-count">
          </include-fragment>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /stemangiola/tidybulk/pulse" data-tab-item="i6insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/stemangiola/tidybulk/pulse" id="insights-tab">
          <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
           </path>
          </svg>
          <span data-content="Insights">
           Insights
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
       </ul>
       <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="action-menu-93b82449-a7bc-486c-ab82-6923c214e346-list" aria-haspopup="true" aria-labelledby="tooltip-e142b5f9-6c71-472f-937e-8caf0f9dff6b" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-93b82449-a7bc-486c-ab82-6923c214e346-button" popovertarget="action-menu-93b82449-a7bc-486c-ab82-6923c214e346-overlay" type="button">
           <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
           </svg>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-93b82449-a7bc-486c-ab82-6923c214e346-button" id="tooltip-e142b5f9-6c71-472f-937e-8caf0f9dff6b" popover="manual">
           Additional navigation options
          </tool-tip>
          <anchored-position align="start" anchor="action-menu-93b82449-a7bc-486c-ab82-6923c214e346-button" anchor-offset="normal" data-view-component="true" id="action-menu-93b82449-a7bc-486c-ab82-6923c214e346-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="action-menu-93b82449-a7bc-486c-ab82-6923c214e346-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-93b82449-a7bc-486c-ab82-6923c214e346-list" role="menu">
                <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/stemangiola/tidybulk" id="item-b4f8be08-3d53-4ab3-84b9-244c9ff9e1ec" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Code
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/stemangiola/tidybulk/issues" id="item-d82c25ed-af41-48c7-8ec7-279c1429271c" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Issues
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/stemangiola/tidybulk/pulls" id="item-3d16e058-6e8c-45df-b7a7-8787b2fe39b6" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Pull requests
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/stemangiola/tidybulk/actions" id="item-7b446240-9a6b-438c-b756-4aee5a68b12d" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Actions
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/stemangiola/tidybulk/projects" id="item-b23a02d7-d2d4-4c55-bd00-87bf124536eb" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Projects
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i5security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/stemangiola/tidybulk/security" id="item-ada6a1ce-fb6b-4887-957a-1569a3d0f4fa" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Security
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i6insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/stemangiola/tidybulk/pulse" id="item-63bd3c59-3f1b-4e4c-af1a-8c892ccd90b6" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Insights
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
       </div>
      </nav>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-f55016a8-2ee0-415c-a378-7e2002ce517e" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-0268f54c-6331-4ff0-b917-26c2f138dfac" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-0268f54c-6331-4ff0-b917-26c2f138dfac" id="tooltip-f55016a8-2ee0-415c-a378-7e2002ce517e" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <notification-shelf-watcher class="js-socket-channel" data-base-url="https://github.com/notifications/beta/shelf" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTY0MDExN30=--4145f0d212f0e1143d18c0210820ecb490355da2d3640aa864d55c8e55f7e93b" data-view-component="true">
   </notification-shelf-watcher>
   <div data-initial="" data-target="notification-shelf-watcher.placeholder" hidden="">
   </div>
   <details class="details-reset details-overlay details-overlay-dark js-command-palette-dialog" data-turbo-replace="" id="command-palette-pjax-container">
    <summary aria-label="Command palette trigger" tabindex="-1">
    </summary>
    <details-dialog aria-label="Command palette" class="command-palette-details-dialog d-flex flex-column flex-justify-center height-fit">
     <command-palette activation-hotkey="Mod+k,Mod+Alt+k" class="command-palette color-bg-default rounded-3 border color-shadow-small" command-mode-hotkey="Mod+Shift+K" data-action="
        command-palette-input-ready:command-palette#inputReady
        command-palette-page-stack-updated:command-palette#updateInputScope
        itemsUpdated:command-palette#itemsUpdated
        keydown:command-palette#onKeydown
        loadingStateChanged:command-palette#loadingStateChanged
        selectedItemChanged:command-palette#selectedItemChanged
        pageFetchError:command-palette#pageFetchError
      " return-to="/stemangiola/tidybulk" user-id="126681664">
      <command-palette-mode data-char="#" data-placeholder="Search issues and pull requests" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="#" data-placeholder="Search issues, pull requests, discussions, and projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="!" data-placeholder="Search projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a user, organization, or repository" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a repository" data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode data-char="/" data-placeholder="Search files" data-scope-types='["repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="?">
      </command-palette-mode>
      <command-palette-mode data-char="&gt;" data-placeholder="Run a command">
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode class="js-command-palette-default-mode" data-char="" data-placeholder="Search or jump to...">
      </command-palette-mode>
      <command-palette-input data-action="
          command-palette-input:command-palette#onInput
          command-palette-select:command-palette#onSelect
          command-palette-descope:command-palette#onDescope
          command-palette-cleared:command-palette#onInputClear
        " placeholder="Search or jump to...">
       <div class="js-search-icon d-flex flex-items-center mr-2" style="height: 26px">
        <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
         </path>
        </svg>
       </div>
       <div class="js-spinner d-flex flex-items-center mr-2 color-fg-muted" hidden="">
        <svg aria-label="Loading" class="anim-rotate" fill="none" height="16" viewbox="0 0 16 16" width="16">
         <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
         </circle>
         <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
         </path>
        </svg>
       </div>
       <command-palette-scope>
        <div class="color-fg-subtle" data-target="command-palette-scope.placeholder" hidden="">
         /
         <span class="text-semibold color-fg-default">
          ...
         </span>
         /
        </div>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDQ6VXNlcjcyMzI4OTA=" data-targets="command-palette-scope.tokens" data-text="stemangiola" data-type="owner" data-value="stemangiola" style="white-space:nowrap;line-height:20px;">
         stemangiola
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDEwOlJlcG9zaXRvcnkyMDUwNDI4MDU=" data-targets="command-palette-scope.tokens" data-text="tidybulk" data-type="repository" data-value="tidybulk" style="white-space:nowrap;line-height:20px;">
         tidybulk
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
       </command-palette-scope>
       <div class="command-palette-input-group flex-1 form-control border-0 box-shadow-none" style="z-index: 0">
        <div class="command-palette-typeahead position-absolute d-flex flex-items-center Truncate">
         <span class="typeahead-segment input-mirror" data-target="command-palette-input.mirror">
         </span>
         <span class="Truncate-text" data-target="command-palette-input.typeaheadText">
         </span>
         <span class="typeahead-segment" data-target="command-palette-input.typeaheadPlaceholder">
         </span>
        </div>
        <input aria-label="Hidden input for typeahead" class="js-overlay-input typeahead-input d-none" disabled="" tabindex="-1"/>
        <input aria-autocomplete="list" aria-controls="command-palette-page-stack" aria-expanded="false" aria-haspopup="listbox" aria-label="Command palette input" autocapitalize="off" autocomplete="off" autocorrect="off" class="js-input typeahead-input form-control border-0 box-shadow-none input-block width-full no-focus-indicator" data-action="
              input:command-palette-input#onInput
              keydown:command-palette-input#onKeydown
            " role="combobox" spellcheck="false" type="text"/>
       </div>
       <div class="position-relative d-inline-block" data-view-component="true">
        <button aria-keyshortcuts="Control+Backspace" class="btn-octicon command-palette-input-clear-button" data-action="click:command-palette-input#onClear keypress:command-palette-input#onClear" data-target="command-palette-input.clearButton" data-view-component="true" hidden="hidden" id="command-palette-clear-button" type="button">
         <svg aria-hidden="true" class="octicon octicon-x-circle-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
          </path>
         </svg>
        </button>
        <tool-tip class="sr-only position-absolute" data-direction="w" data-type="label" data-view-component="true" for="command-palette-clear-button" id="tooltip-751e467d-2aed-446b-a032-0734952264c6" popover="manual">
         Clear Command Palette
        </tool-tip>
       </div>
      </command-palette-input>
      <command-palette-page-stack data-action="command-palette-page-octicons-cached:command-palette-page-stack#cacheOcticons" data-default-scope-id="MDEwOlJlcG9zaXRvcnkyMDUwNDI4MDU=" data-default-scope-type="Repository">
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search discussions
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           !
          </kbd>
          to search projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search teams
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='[""]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search people and organizations
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           &gt;
          </kbd>
          to activate command mode
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Go to your accessibility settings to change your keyboard shortcuts
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type author:@me to search your content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:pr to filter to pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:issue to filter to issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:project to filter to projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:open to filter to open content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="mx-3 my-2 flash flash-error d-flex flex-items-center" data-on-error="" data-scope-types="*">
        <div>
         <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
          </path>
         </svg>
        </div>
        <div class="px-2">
         We’ve encountered an error and some results aren't available at this time. Type a new search or try again later.
        </div>
       </command-palette-tip>
       <command-palette-tip class="h4 color-fg-default pl-3 pb-2 pt-3" data-match-mode="[^?]|^$" data-on-empty="" data-scope-types="*">
        No results matched your search
       </command-palette-tip>
       <div hidden="">
        <div data-octicon-id="arrow-right-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="arrow-right-color-fg-default" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-default" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="codespaces-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
          </path>
          <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="copy-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-copy color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
          </path>
          <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="dash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-dash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 7.75A.75.75 0 0 1 2.75 7h10a.75.75 0 0 1 0 1.5h-10A.75.75 0 0 1 2 7.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="file-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-file color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="gear-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-gear color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="lock-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-lock color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M4 4a4 4 0 0 1 8 0v2h.25c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 12.25 15h-8.5A1.75 1.75 0 0 1 2 13.25v-5.5C2 6.784 2.784 6 3.75 6H4Zm8.25 3.5h-8.5a.25.25 0 0 0-.25.25v5.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25ZM10.5 6V4a2.5 2.5 0 1 0-5 0v2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="moon-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-moon color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M9.598 1.591a.749.749 0 0 1 .785-.175 7.001 7.001 0 1 1-8.967 8.967.75.75 0 0 1 .961-.96 5.5 5.5 0 0 0 7.046-7.046.75.75 0 0 1 .175-.786Zm1.616 1.945a7 7 0 0 1-7.678 7.678 5.499 5.499 0 1 0 7.678-7.678Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="person-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-person color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="pencil-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-pencil color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="issue-opened-open" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-issue-opened open" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="git-pull-request-draft-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request-draft color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M3.25 1A2.25 2.25 0 0 1 4 5.372v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.251 2.251 0 0 1 3.25 1Zm9.5 14a2.25 2.25 0 1 1 0-4.5 2.25 2.25 0 0 1 0 4.5ZM2.5 3.25a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0ZM3.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm9.5 0a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM14 7.5a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm0-4.25a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="search-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sun-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sun color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8Zm0-1.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Zm5.657-8.157a.75.75 0 0 1 0 1.061l-1.061 1.06a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.06-1.06a.75.75 0 0 1 1.06 0Zm-9.193 9.193a.75.75 0 0 1 0 1.06l-1.06 1.061a.75.75 0 1 1-1.061-1.06l1.06-1.061a.75.75 0 0 1 1.061 0ZM8 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V.75A.75.75 0 0 1 8 0ZM3 8a.75.75 0 0 1-.75.75H.75a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 3 8Zm13 0a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 16 8Zm-8 5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 8 13Zm3.536-1.464a.75.75 0 0 1 1.06 0l1.061 1.06a.75.75 0 0 1-1.06 1.061l-1.061-1.06a.75.75 0 0 1 0-1.061ZM2.343 2.343a.75.75 0 0 1 1.061 0l1.06 1.061a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-1.06-1.06a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sync-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sync color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.705 8.005a.75.75 0 0 1 .834.656 5.5 5.5 0 0 0 9.592 2.97l-1.204-1.204a.25.25 0 0 1 .177-.427h3.646a.25.25 0 0 1 .25.25v3.646a.25.25 0 0 1-.427.177l-1.38-1.38A7.002 7.002 0 0 1 1.05 8.84a.75.75 0 0 1 .656-.834ZM8 2.5a5.487 5.487 0 0 0-4.131 1.869l1.204 1.204A.25.25 0 0 1 4.896 6H1.25A.25.25 0 0 1 1 5.75V2.104a.25.25 0 0 1 .427-.177l1.38 1.38A7.002 7.002 0 0 1 14.95 7.16a.75.75 0 0 1-1.49.178A5.5 5.5 0 0 0 8 2.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="trash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-trash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="key-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-key color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="comment-discussion-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-slash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell-slash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="paintbrush-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-paintbrush color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.134 1.535c.7-.509 1.416-.942 2.076-1.155.649-.21 1.463-.267 2.069.34.603.601.568 1.411.368 2.07-.202.668-.624 1.39-1.125 2.096-1.011 1.424-2.496 2.987-3.775 4.249-1.098 1.084-2.132 1.839-3.04 2.3a3.744 3.744 0 0 1-1.055 3.217c-.431.431-1.065.691-1.657.861-.614.177-1.294.287-1.914.357A21.151 21.151 0 0 1 .797 16H.743l.007-.75H.749L.742 16a.75.75 0 0 1-.743-.742l.743-.008-.742.007v-.054a21.25 21.25 0 0 1 .13-2.284c.067-.647.187-1.287.358-1.914.17-.591.43-1.226.86-1.657a3.746 3.746 0 0 1 3.227-1.054c.466-.893 1.225-1.907 2.314-2.982 1.271-1.255 2.833-2.75 4.245-3.777ZM1.62 13.089c-.051.464-.086.929-.104 1.395.466-.018.932-.053 1.396-.104a10.511 10.511 0 0 0 1.668-.309c.526-.151.856-.325 1.011-.48a2.25 2.25 0 1 0-3.182-3.182c-.155.155-.329.485-.48 1.01a10.515 10.515 0 0 0-.309 1.67Zm10.396-10.34c-1.224.89-2.605 2.189-3.822 3.384l1.718 1.718c1.21-1.205 2.51-2.597 3.387-3.833.47-.662.78-1.227.912-1.662.134-.444.032-.551.009-.575h-.001V1.78c-.014-.014-.113-.113-.548.027-.432.14-.995.462-1.655.942Zm-4.832 7.266-.001.001a9.859 9.859 0 0 0 1.63-1.142L7.155 7.216a9.7 9.7 0 0 0-1.161 1.607c.482.302.889.71 1.19 1.192Z">
          </path>
         </svg>
        </div>
        <command-palette-item-group data-default-priority="0" data-group-hint="" data-group-id="top" data-group-limits="{}" data-group-title="Top result">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="1" data-group-hint="Type &gt; to filter" data-group-id="commands" data-group-limits='{"static_items_page":50,"issue":50,"pull_request":50,"discussion":50}' data-group-title="Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="2" data-group-hint="Type &gt; to filter" data-group-id="global_commands" data-group-limits='{"issue":0,"pull_request":0,"discussion":0}' data-group-title="Global Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="3" data-group-hint="" data-group-id="this_page" data-group-limits="{}" data-group-title="This Page">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="4" data-group-hint="" data-group-id="files" data-group-limits="{}" data-group-title="Files">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="5" data-group-hint="" data-group-id="default" data-group-limits='{"static_items_page":50}' data-group-title="Default">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="6" data-group-hint="" data-group-id="pages" data-group-limits='{"repository":10}' data-group-title="Pages">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="7" data-group-hint="" data-group-id="access_policies" data-group-limits="{}" data-group-title="Access Policies">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="8" data-group-hint="" data-group-id="organizations" data-group-limits="{}" data-group-title="Organizations">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="9" data-group-hint="" data-group-id="repositories" data-group-limits="{}" data-group-title="Repositories">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="10" data-group-hint="Type # to filter" data-group-id="references" data-group-limits="{}" data-group-title="Issues, pull requests, and discussions">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="11" data-group-hint="" data-group-id="teams" data-group-limits="{}" data-group-title="Teams">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="12" data-group-hint="" data-group-id="users" data-group-limits="{}" data-group-title="Users">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="13" data-group-hint="" data-group-id="memex_projects" data-group-limits="{}" data-group-title="Projects">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="14" data-group-hint="" data-group-id="projects" data-group-limits="{}" data-group-title="Projects (classic)">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="15" data-group-hint="" data-group-id="footer" data-group-limits="{}" data-group-title="Footer">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="16" data-group-hint="" data-group-id="modes_help" data-group-limits="{}" data-group-title="Modes">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="17" data-group-hint="" data-group-id="filters_help" data-group-limits="{}" data-group-title="Use filters in issues, pull requests, discussions, and projects">
        </command-palette-item-group>
        <command-palette-page data-page-title="stemangiola" data-scope-id="MDQ6VXNlcjcyMzI4OTA=" data-scope-type="owner" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
        <command-palette-page data-page-title="tidybulk" data-scope-id="MDEwOlJlcG9zaXRvcnkyMDUwNDI4MDU=" data-scope-type="repository" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
       </div>
       <command-palette-page data-is-root="">
       </command-palette-page>
       <command-palette-page data-page-title="stemangiola" data-scope-id="MDQ6VXNlcjcyMzI4OTA=" data-scope-type="owner">
       </command-palette-page>
       <command-palette-page data-page-title="tidybulk" data-scope-id="MDEwOlJlcG9zaXRvcnkyMDUwNDI4MDU=" data-scope-type="repository">
       </command-palette-page>
      </command-palette-page-stack>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="search-links">
      </server-defined-provider>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="help">
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues
         </strong>
         and
         <strong>
          pull requests
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues, pull requests, discussions,
         </strong>
         and
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="@" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          organizations, repositories,
         </strong>
         and
         <strong>
          users
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          @
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="!" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          !
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="/" data-scope-types='["repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          files
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          /
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="&gt;">
        <span data-target="command-palette-help.titleElement">
         Activate
         <strong>
          command mode
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          &gt;
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:pr">
        <span data-target="command-palette-help.titleElement">
         Filter to pull requests
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:pr
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:issue">
        <span data-target="command-palette-help.titleElement">
         Filter to issues
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:issue
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:discussion" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:discussion
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:project" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to projects
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:project
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:open">
        <span data-target="command-palette-help.titleElement">
         Filter to open issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:open
         </kbd>
        </span>
       </command-palette-help>
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/commands" data-supported-modes="[]" data-supports-commands="" data-targets="command-palette.serverDefinedProviderElements" data-type="commands">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_page_navigation" data-supported-modes='[""]' data-supported-scope-types='["","owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to" data-supported-modes='["@","@"]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to_members_only" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_members_only_prefetched" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/files" data-supported-modes='["/"]' data-supported-scope-types='["repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="files">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/discussions" data-supported-modes='["#"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/projects" data-supported-modes='["#","!"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/recent_issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/teams" data-supported-modes='["@",""]' data-supported-scope-types='["owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/name_with_owner_repository" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
     </command-palette>
    </details-dialog>
   </details>
   <div class="position-fixed bottom-0 left-0 ml-5 mb-5 js-command-palette-toasts" style="z-index: 1000">
    <div class="Toast Toast--loading" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="Toast--spinner" height="18" viewbox="0 0 32 32" width="18">
       <path d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4" fill="#959da5">
       </path>
       <path d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z" fill="#ffffff">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--error" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-stop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M4.47.22A.749.749 0 0 1 5 0h6c.199 0 .389.079.53.22l4.25 4.25c.141.14.22.331.22.53v6a.749.749 0 0 1-.22.53l-4.25 4.25A.749.749 0 0 1 11 16H5a.749.749 0 0 1-.53-.22L.22 11.53A.749.749 0 0 1 0 11V5c0-.199.079-.389.22-.53Zm.84 1.28L1.5 5.31v5.38l3.81 3.81h5.38l3.81-3.81V5.31L10.69 1.5ZM8 4a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 4Zm0 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--warning" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--success" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-info" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.5 7.75A.75.75 0 0 1 7.25 7h1a.75.75 0 0 1 .75.75v2.75h.25a.75.75 0 0 1 0 1.5h-2a.75.75 0 0 1 0-1.5h.25v-2h-.25a.75.75 0 0 1-.75-.75ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
   </div>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <template class="js-user-list-create-dialog-template" data-label="Create list">
       <div class="Box-header">
        <h2 class="Box-title">
         Create list
        </h2>
       </div>
       <!-- '"` -->
       <!-- </textarea></xmp> -->
       <form accept-charset="UTF-8" action="/stars/ninenia/lists" class="Box-body d-flex flex-column p-3 js-user-list-form" data-turbo="false" method="post">
        <input autocomplete="off" name="authenticity_token" type="hidden" value="EPrwLTnHpEH_vnnjRF0eOv2vT7nVgFK0g-FQxvIExl7AlloJrB1y5hyD04TXhYecWKyEzFGne4gdS3VzRtXQ6Q">
         <p class="color-fg-subtle mb-3">
          Create a list to organize your starred repositories.
         </p>
         <input name="repository_id" type="hidden" value="{{ repositoryId }}"/>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <auto-check required="" src="/stars/ninenia/list-check?attr=name">
           <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
            <input aria-label="List name" autofocus="" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="32" maxlength="32" name="user_list[name]" placeholder="⭐️ Name this list" required="" type="text" value="{{ placeholderName }}"/>
           </text-expander>
           <input data-csrf="true" type="hidden" value="iMZOyUI39o8AgaQ1Tz7J061-yh60p5ihYZgWyTdbGZTo7hI4jRmk69bKO3GsPA3lrJSUbxhVJpvNsmwA1TOtOg">
           </input>
          </auto-check>
          <p class="note error position-relative js-user-list-error" hidden="">
           Name .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           32 remaining
          </p>
         </div>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
           <textarea aria-label="List description" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="160" maxlength="160" name="user_list[description]" placeholder="Write a description" style="height: 74px; min-height: 74px"></textarea>
          </text-expander>
          <p class="note error position-relative js-user-list-error" hidden="">
           Description .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           160 remaining
          </p>
         </div>
         <div class="js-user-list-base flash flash-error mx-0 mt-0 mb-2" data-generic-message="Unable to save your list at this time." data-view-component="true" hidden="hidden">
          .
         </div>
         <button class="Button--primary Button--medium Button Button--fullWidth mt-2" data-disable-invalid="true" data-submitting-message="Creating..." data-view-component="true" disabled="disabled" type="submit">
          <span class="Button-content">
           <span class="Button-label">
            Create
           </span>
          </span>
         </button>
         <p class="note mt-2 mb-0">
          <strong>
           Tip:
          </strong>
          type
          <code>
           :
          </code>
          to add emoji to the name or description.
         </p>
        </input>
       </form>
       <div class="Box-footer Box-row--gray text-small color-fg-muted d-flex flex-items-baseline py-2" data-view-component="true">
        <span aria-label="Feature Release Label: Beta" class="Label Label--success Label--inline px-2 mr-2" data-view-component="true" title="Feature Release Label: Beta">
         Beta
        </span>
        <span class="mr-1">
         Lists are currently in beta.
        </span>
        <a href="https://github.com//github/feedback/discussions/categories/lists">
         Share feedback and report bugs.
        </a>
       </div>
      </template>
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header">
       <div class="d-flex flex-wrap flex-justify-end mb-3 container-xl px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal" id="repo-title-component">
          <svg aria-hidden="true" class="octicon octicon-repo-template color-fg-muted mr-2 d-none d-sm-none d-md-block" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M13.25 8a.75.75 0 0 1 .75.75v4.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-.75a.75.75 0 0 1 0-1.5h.75v-.25a.75.75 0 0 1 .75-.75ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2ZM2.75 8a.75.75 0 0 1 .75.75v.268c.083-.012.166-.018.25-.018h.5a.75.75 0 0 1 0 1.5h-.5a.25.25 0 0 0-.25.25v.75c0 .28.114.532.3.714a.75.75 0 1 1-1.05 1.072A2.495 2.495 0 0 1 2 11.5V8.75A.75.75 0 0 1 2.75 8ZM11 .75a.75.75 0 0 1 .75-.75h1.5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V1.5h-.75A.75.75 0 0 1 11 .75Zm-5 0A.75.75 0 0 1 6.75 0h2.5a.75.75 0 0 1 0 1.5h-2.5A.75.75 0 0 1 6 .75Zm0 9A.75.75 0 0 1 6.75 9h2.5a.75.75 0 0 1 0 1.5h-2.5A.75.75 0 0 1 6 9.75ZM4.992.662a.75.75 0 0 1-.636.848c-.436.063-.783.41-.846.846a.751.751 0 0 1-1.485-.212A2.501 2.501 0 0 1 4.144.025a.75.75 0 0 1 .848.637ZM2.75 4a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 2.75 4Zm10.5 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5a.75.75 0 0 1 .75-.75Z">
           </path>
          </svg>
          <strong class="mr-2 flex-self-stretch d-none d-md-block" itemprop="name">
           <a class="color-fg-default" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/stemangiola/tidybulk">
            tidybulk
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1 d-none d-md-block">
           Public template
          </span>
         </div>
         <div class="d-none d-md-block">
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <include-fragment loading="lazy" src="/notifications/205042805/watch_subscription?aria_id_prefix=repository-details&amp;button_block=false&amp;show_count=true">
            <div data-hide-on-error="">
             <button aria-label="Watching a repository" class="tooltipped tooltipped-n Button--secondary Button--small Button" data-view-component="true" disabled="disabled" type="button">
              <span class="Button-content">
               <span class="Button-visual Button-leadingVisual">
                <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                 </path>
                </svg>
               </span>
               <span class="Button-label">
                Watch
               </span>
              </span>
             </button>
            </div>
            <p data-show-on-error="" hidden="">
             <span class="d-flex pt-1">
              <svg aria-hidden="true" class="octicon octicon-alert mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
               </path>
              </svg>
              Couldn't load subscription status.
              <button class="Button--link Button--medium Button" data-retry-button="" data-view-component="true" type="button">
               <span class="Button-content">
                <span class="Button-label">
                 Retry
                </span>
               </span>
              </button>
             </span>
            </p>
           </include-fragment>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a class="btn-sm btn BtnGroup-item" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":205042805,"originating_url":"https://github.com/stemangiola/tidybulk","user_id":126681664}}' data-hydro-click-hmac="c79f4d3c6888fbbc462b7ad892004101d366411187030dee0b3759ec4bc43eec" data-view-component="true" href="/stemangiola/tidybulk/fork" icon="repo-forked" id="fork-button">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             Fork
             <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="23">
              23
             </span>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="fork-button" id="tooltip-151f3971-9d53-4a76-99b9-e28dc591c20e" popover="manual">
              Fork your own copy of stemangiola/tidybulk
             </tool-tip>
            </a>
            <details class="details-reset details-overlay BtnGroup-parent d-inline-block position-relative" data-view-component="true" group_item="true" id="my-forks-menu-205042805">
             <summary aria-label="See your forks of this repository" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
              <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
               </path>
              </svg>
             </summary>
             <details-menu class="SelectMenu right-0" role="menu" src="/stemangiola/tidybulk/my_forks_menu_content?can_fork=true">
              <div class="SelectMenu-modal">
               <button aria-label="Close menu" class="SelectMenu-closeButton position-absolute right-0 m-2" data-toggle-for="my-forks-menu-205042805" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
               <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-0680f3">
                <div class="SelectMenu-list">
                 <include-fragment aria-label="Loading" class="SelectMenu-loading">
                  <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                   <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                   </circle>
                   <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                   </path>
                  </svg>
                 </include-fragment>
                </div>
               </div>
              </div>
             </details-menu>
            </details>
           </div>
          </li>
          <li>
           <template class="js-unstar-confirmation-dialog-template">
            <div class="Box-header">
             <h2 class="Box-title">
              Unstar this repository?
             </h2>
            </div>
            <div class="Box-body">
             <p class="mb-3">
              This will remove {{ repoNameWithOwner }} from the {{ listsWithCount }} that it's been added to.
             </p>
             <div class="form-actions">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="{{ confirmUrl }}" class="js-social-confirmation-form" data-turbo="false" method="post">
               <input name="authenticity_token" type="hidden" value="{{ confirmCsrfToken }}"/>
               <input name="confirm" type="hidden" value="true"/>
               <button class="btn-danger btn width-full" data-close-dialog="true" data-view-component="true" type="submit">
                Unstar
               </button>
              </form>
             </div>
            </div>
           </template>
           <div class="js-toggler-container js-social-container starring-container d-flex" data-view-component="true">
            <div class="starred BtnGroup flex-1 ml-0" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/stemangiola/tidybulk/unstar" class="js-social-form BtnGroup-parent flex-auto js-deferred-toggler-target" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="JI_Bv40bld7kKGKZGuC-kdjyswS-NjL59bbfnn-xGgl_9sZhtw5oBEzvDdkvXdvItg7rL467R039qZeZB2eRcA">
               <input class="js-confirm-csrf-token" data-csrf="true" type="hidden" value="as86yOxaPi-bsXJ20I9gjFPAY2_xrQIkY_PkahxY4WUxtj0W1k_D9TN2HTblMgXVPTw7RMEgd5Br7KxtZI5qHA">
                <input name="context" type="hidden" value="repository"/>
                <button aria-label="Unstar this repository (158)" class="rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":205042805,"originating_url":"https://github.com/stemangiola/tidybulk","user_id":126681664}}' data-hydro-click-hmac="e112d49ba7015332c0b3326f78578f1c95a8c61901e1c6a2ae930dca017641df" data-view-component="true" type="submit">
                 <svg aria-hidden="true" class="octicon octicon-star-fill starred-button-icon d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                  </path>
                 </svg>
                 <span class="d-inline" data-view-component="true">
                  Starred
                 </span>
                 <span aria-label="158 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-unstar" title="158">
                  158
                 </span>
                </button>
               </input>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-205042805-starred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/stemangiola/tidybulk/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-205042805-starred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-4bd1d6">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
            <div class="unstarred BtnGroup ml-0 flex-1" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/stemangiola/tidybulk/star" class="js-social-form BtnGroup-parent flex-auto" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="hdPr6K0_zN1_pXdE_xlO-ETvDp4TT19XE2j3WG73AE4VNaHHVgIsSMYNGXMCyJ7w7HqGR-9mb_NBfZUdC04pAQ">
               <input name="context" type="hidden" value="repository"/>
               <button aria-label="Star this repository (158)" class="js-toggler-target rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":205042805,"originating_url":"https://github.com/stemangiola/tidybulk","user_id":126681664}}' data-hydro-click-hmac="77ebefe882c7eca8dc2b36096c96b069a568a55552db00ce705c8d9fea1ed49f" data-view-component="true" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
                <span class="d-inline" data-view-component="true">
                 Star
                </span>
                <span aria-label="158 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="158">
                 158
                </span>
               </button>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-205042805-unstarred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/stemangiola/tidybulk/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-205042805-unstarred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-c28ea5">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
           </div>
          </li>
          <li>
           <details class="details-overlay details-reset position-relative" data-view-component="true">
            <summary class="btn-primary btn-sm btn ml-0" data-view-component="true" role="button">
             <span class="flex-items-center">
              Use this template
              <span class="dropdown-caret ml-1">
              </span>
             </span>
            </summary>
            <div data-view-component="true">
             <ul class="dropdown-menu dropdown-menu-sw" style="width: fit-content;">
              <li class="d-block d-md-none dropdown-divider" role="none">
              </li>
              <li>
               <a class="dropdown-item" href="/new?template_name=tidybulk&amp;template_owner=stemangiola">
                Create a new repository
               </a>
              </li>
              <li>
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="/codespaces" data-turbo="false" method="post" target="_blank">
                <input autocomplete="off" name="authenticity_token" type="hidden" value="vx8-59rG8KGaXZjiYCWr4BePAK1hfErtXV_mfNUNu4qox6FIRs2IwZkRYps2cTYuYTUtToHVMXcsjMLX0K4lqQ">
                 <input autocomplete="off" name="codespace[template_repository_id]" type="hidden" value="205042805">
                  <input autocomplete="off" name="codespace[repository_id]" type="hidden" value="205042805">
                   <input autocomplete="off" name="codespace[ref]" type="hidden" value="master">
                    <button class="dropdown-item btn-link" data-hydro-click='{"event_type":"codespace_create.click","payload":{"ref":"master","repository_id":205042805,"target":"USE_TEMPLATE_DROPDOWN","user_id":126681664,"originating_url":"https://github.com/stemangiola/tidybulk"}}' data-hydro-click-hmac="d0e4ffeed55a5b2e7e6e5552f01c48a4f7dbc89ee24a8ef87913737fd03cc436" name="button" type="submit">
                     Open in a codespace
                    </button>
                   </input>
                  </input>
                 </input>
                </input>
               </form>
              </li>
             </ul>
            </div>
           </details>
          </li>
         </ul>
        </div>
       </div>
       <div class="container-xl">
        <div data-turbo-replace="" id="responsive-meta-container">
         <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
          <div class="d-flex gap-2 mt-n3 mb-3 flex-wrap">
           <div class="d-flex flex-row gap-2">
            <notifications-list-subscription-form data-action="notifications-dialog-label-toggled:notifications-list-subscription-form#handleDialogLabelToggle">
             <details class="details-reset details-overlay details-overlay-dark" data-action="toggle:notifications-list-subscription-form#detailsToggled" data-target="notifications-list-subscription-form.details">
              <summary aria-label="Notifications" class="Button--iconOnly Button--secondary Button--medium Button p-0" data-ga-click="Repository, click Watch settings, action:files#disambiguate" data-hydro-click='{"event_type":"repository.click","payload":{"target":"WATCH_BUTTON","repository_id":205042805,"originating_url":"https://github.com/stemangiola/tidybulk","user_id":126681664}}' data-hydro-click-hmac="b4eb509ea1b4ca9b7897d31db4a5d1ee2a56f2123b8ff4da6e28e91eda1eb9db" data-view-component="true" id="notifications-button">
               <span class="Button-content">
                <span class="Button-label">
                 <span data-menu-button="">
                  <span data-target="notifications-list-subscription-form.unwatchButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.stopIgnoringButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-bell-slash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.watchButtonCopy">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                 </span>
                </span>
               </span>
              </summary>
              <details-dialog aria-label="Notification settings" class="anim-fade-in fast Box Box--overlay">
               <div data-target="notifications-list-subscription-form.menu">
                <div class="Box-header">
                 <button aria-label="Close dialog" class="Box-btn-octicon btn-octicon float-right" data-action="click:notifications-list-subscription-form#closeMenu" data-close-dialog="" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                 <h3 class="Box-title">
                  Notifications
                 </h3>
                </div>
                <div class="SelectMenu-list mb-0" role="menu">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitForm" data-target="notifications-list-subscription-form.form" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="JrXhhGUs0quqcf7i0oAnkStvGfq8h_UvdezINk7aVR4-de3-e5OnqkZiyegOYKHmdEqK72ndXvtXFzrCGWvrhw"/>
                  <input name="repository_id" type="hidden" value="205042805"/>
                  <button aria-checked="true" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="included">
                   <span class="f5">
                    <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                     </path>
                    </svg>
                   </span>
                   <div>
                    <div class="f5 text-bold">
                     Participating and @mentions
                    </div>
                    <div class="text-small color-fg-muted text-normal pb-1">
                     Only receive notifications from this repository when participating or @mentioned.
                    </div>
                   </div>
                  </button>
                  <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="subscribed">
                   <span class="f5">
                    <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                     </path>
                    </svg>
                   </span>
                   <div>
                    <div class="f5 text-bold">
                     All Activity
                    </div>
                    <div class="text-small color-fg-muted text-normal pb-1">
                     Notified of all notifications on this repository.
                    </div>
                   </div>
                  </button>
                  <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="ignore">
                   <span class="f5">
                    <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                     </path>
                    </svg>
                   </span>
                   <div>
                    <div class="f5 text-bold">
                     Ignore
                    </div>
                    <div class="text-small color-fg-muted text-normal pb-1">
                     Never be notified.
                    </div>
                   </div>
                  </button>
                 </form>
                 <button aria-checked="false" aria-haspopup="true" class="SelectMenu-item flex-items-start pr-3" data-action="click:notifications-list-subscription-form#openCustomDialog" data-target="notifications-list-subscription-form.customButton" role="menuitemradio" type="button">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                    </path>
                   </svg>
                  </span>
                  <div>
                   <div class="d-flex flex-items-start flex-justify-between">
                    <div class="f5 text-bold">
                     Custom
                    </div>
                    <div class="f5 pr-1">
                     <svg aria-hidden="true" class="octicon octicon-arrow-right" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </div>
                   <div class="text-small color-fg-muted text-normal pb-1">
                    Select events you want to be notified of in addition to participating and @mentions.
                   </div>
                  </div>
                 </button>
                 <div class="px-3 py-2 d-flex color-bg-subtle flex-items-center">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile SelectMenu-icon SelectMenu-icon--device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </span>
                  <span classname="text-small color-fg-muted text-normal pb-1">
                   Get push notifications on
                   <a href="https://apps.apple.com/app/apple-store/id1477376905?ct=watch-dropdown&amp;mt=8&amp;pt=524675" rel="noopener noreferrer" target="_blank">
                    iOS
                   </a>
                   or
                   <a href="https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dwatch-dropdown%26utm_medium%3Dweb%26utm_source%3Dgithub" rel="noopener noreferrer" target="_blank">
                    Android
                   </a>
                   .
                  </span>
                 </div>
                </div>
               </div>
               <div data-target="notifications-list-subscription-form.customDialog" hidden="">
                <header class="SelectMenu-header pb-0 border-bottom-0 px-2 px-sm-3">
                 <h1 class="f3 SelectMenu-title d-inline-flex">
                  <button aria-label="Return to menu" class="color-bg-default border-0 px-2 py-0 m-0 Link--secondary f5" data-action="click:notifications-list-subscription-form#closeCustomDialog" type="button">
                   <svg aria-hidden="true" class="octicon octicon-arrow-left" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z">
                    </path>
                   </svg>
                  </button>
                  Custom
                 </h1>
                </header>
                <div class="overflow-visible">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitCustomForm" data-target="notifications-list-subscription-form.customform" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="CRPFLGqpXbNerrGDrO1bk9euUgm907QYjGCinDHmWlcR08lWdBYosrK9holwDd3kiIvBHGiJH8yum1BoZlfkzg">
                   <input name="repository_id" type="hidden" value="205042805"/>
                   <fieldset>
                    <legend>
                     <div class="text-small color-fg-muted pt-0 pr-3 pb-3 pl-6 pl-sm-5 border-bottom mb-3">
                      Select events you want to be notified of in addition to participating and @mentions.
                     </div>
                    </legend>
                    <div data-target="notifications-list-subscription-form.labelInputs">
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Issue"/>
                      Issues
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="PullRequest"/>
                      Pull requests
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Release"/>
                      Releases
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input aria-describedby="Discussion-disabled" aria-disabled="true" data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Discussion"/>
                      Discussions
                     </label>
                     <div class="color-fg-muted" id="Discussion-disabled">
                      &gt;
                          Discussions are not enabled for this repository
                     </div>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="SecurityAlert"/>
                      Security alerts
                     </label>
                    </div>
                   </fieldset>
                   <div class="pt-2 pb-3 px-3 d-flex flex-justify-start flex-row-reverse">
                    <button class="Button--primary Button--medium Button ml-2" data-target="notifications-list-subscription-form.customSubmit" data-view-component="true" disabled="disabled" name="do" type="submit" value="custom">
                     <span class="Button-content">
                      <span class="Button-label">
                       Apply
                      </span>
                     </span>
                    </button>
                    <button class="Button--secondary Button--medium Button" data-action="click:notifications-list-subscription-form#resetForm" data-close-dialog="" data-view-component="true" type="button">
                     <span class="Button-content">
                      <span class="Button-label">
                       Cancel
                      </span>
                     </span>
                    </button>
                   </div>
                  </input>
                 </form>
                </div>
               </div>
              </details-dialog>
             </details>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="notifications-button" id="tooltip-834f63a4-4ff5-419e-9644-c1951e2b7cf1" popover="manual">
              Notification settings
             </tool-tip>
            </notifications-list-subscription-form>
            <div class="BtnGroup d-flex" data-view-component="true">
             <a aria-labelledby="tooltip-1990d4ec-4b7d-4452-9df9-db0008e20a16" class="Button Button--iconOnly Button--secondary Button--medium" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":205042805,"originating_url":"https://github.com/stemangiola/tidybulk","user_id":126681664}}' data-hydro-click-hmac="c79f4d3c6888fbbc462b7ad892004101d366411187030dee0b3759ec4bc43eec" data-view-component="true" group_item="true" href="/stemangiola/tidybulk/fork" id="fork-icon-button" variant="small">
              <svg aria-hidden="true" class="octicon octicon-repo-forked Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
               </path>
              </svg>
             </a>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="fork-icon-button" id="tooltip-1990d4ec-4b7d-4452-9df9-db0008e20a16" popover="manual">
              Fork your own copy of stemangiola/tidybulk
             </tool-tip>
            </div>
            <div class="js-toggler-container starring-container" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/stemangiola/tidybulk/unstar" class="starred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="S-0nBOgsNASyfZcPjmUWTpBHmXFbCu5UmSfZToknXacQlCDa0jnJ3hq6-E-72HMX_rvBWmuHm-CROJFJ8fHW3g">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-6414ab98-c018-42cc-97ba-a793c266a760" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target starred-button-icon" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":205042805,"originating_url":"https://github.com/stemangiola/tidybulk","user_id":126681664}}' data-hydro-click-hmac="e112d49ba7015332c0b3326f78578f1c95a8c61901e1c6a2ae930dca017641df" data-view-component="true" id="icon-button-a822b84c-8f0e-4673-b413-116481949f47" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-a822b84c-8f0e-4673-b413-116481949f47" id="tooltip-6414ab98-c018-42cc-97ba-a793c266a760" popover="manual">
                Unstar this repository
               </tool-tip>
              </input>
             </form>
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/stemangiola/tidybulk/star" class="unstarred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="Qfhp5PFtzLvyzVviuiuKY3P7jpiT8jEvul0-llwTCZfRHiPLClAsLktlNdVH-lpr224GQW_bAYvoSFzTOaog2A">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-87893d2c-ad91-45bc-a976-1a288155cc72" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":205042805,"originating_url":"https://github.com/stemangiola/tidybulk","user_id":126681664}}' data-hydro-click-hmac="77ebefe882c7eca8dc2b36096c96b069a568a55552db00ce705c8d9fea1ed49f" data-view-component="true" id="icon-button-cff22cf9-88a0-4e70-bff9-5bc456205abc" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-cff22cf9-88a0-4e70-bff9-5bc456205abc" id="tooltip-87893d2c-ad91-45bc-a976-1a288155cc72" popover="manual">
                Star this repository
               </tool-tip>
              </input>
             </form>
            </div>
           </div>
           <div class="d-flex flex-row gap-2">
            <details class="details-overlay details-reset position-relative" data-view-component="true">
             <summary class="btn-primary btn ml-0" data-view-component="true" role="button">
              <span class="flex-items-center">
               Use this template
               <span class="dropdown-caret ml-1">
               </span>
              </span>
             </summary>
             <div data-view-component="true">
              <ul class="dropdown-menu dropdown-menu-sw" style="width: fit-content;">
               <li class="d-block d-md-none dropdown-divider" role="none">
               </li>
               <li>
                <a class="dropdown-item" href="/new?template_name=tidybulk&amp;template_owner=stemangiola">
                 Create a new repository
                </a>
               </li>
               <li>
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="/codespaces" data-turbo="false" method="post" target="_blank">
                 <input autocomplete="off" name="authenticity_token" type="hidden" value="ESn-P9O1P3MHx0h-ggJfaUrrO3y2rrLU7fSRDsNY1xMG8WGQT75HEwSLsgfUVsKnPFEWn1YHyU6cJ7WlxvtJMA">
                  <input autocomplete="off" name="codespace[template_repository_id]" type="hidden" value="205042805">
                   <input autocomplete="off" name="codespace[repository_id]" type="hidden" value="205042805">
                    <input autocomplete="off" name="codespace[ref]" type="hidden" value="master">
                     <button class="dropdown-item btn-link" data-hydro-click='{"event_type":"codespace_create.click","payload":{"ref":"master","repository_id":205042805,"target":"USE_TEMPLATE_DROPDOWN","user_id":126681664,"originating_url":"https://github.com/stemangiola/tidybulk"}}' data-hydro-click-hmac="d0e4ffeed55a5b2e7e6e5552f01c48a4f7dbc89ee24a8ef87913737fd03cc436" name="button" type="submit">
                      Open in a codespace
                     </button>
                    </input>
                   </input>
                  </input>
                 </input>
                </form>
               </li>
              </ul>
             </div>
            </details>
           </div>
          </div>
          <p class="f4 mb-3 color-fg-muted">
           Brings bulk and pseudobulk transcriptomics to the tidyverse
          </p>
          <div class="mb-3">
           <ul aria-label="Repository details" class="d-flex flex-wrap mb-2 gap-2">
            <a class="Link--secondary no-underline d-block mr-2" href="/stemangiola/tidybulk/stargazers" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              158
             </span>
             stars
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/stemangiola/tidybulk/forks" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              23
             </span>
             forks
            </a>
            <div class="color-fg-muted mr-2" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-eye mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              5
             </span>
             watching
            </div>
            <a class="Link--secondary no-underline d-block mr-2" href="/stemangiola/tidybulk/branches" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
              </path>
             </svg>
             <include-fragment src="/stemangiola/tidybulk/branch-count">
              <span>
               Branches
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/stemangiola/tidybulk/tags" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
              </path>
             </svg>
             <include-fragment src="/stemangiola/tidybulk/tag-count">
              <span>
               Tags
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/stemangiola/tidybulk/activity" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
              </path>
             </svg>
             <span>
              Activity
             </span>
            </a>
           </ul>
           <div class="mb-2 d-flex color-fg-muted">
            <div class="d-flex flex-items-center" style="height: 21px">
             <svg aria-hidden="true" class="octicon octicon-repo-template flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M13.25 8a.75.75 0 0 1 .75.75v4.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-.75a.75.75 0 0 1 0-1.5h.75v-.25a.75.75 0 0 1 .75-.75ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2ZM2.75 8a.75.75 0 0 1 .75.75v.268c.083-.012.166-.018.25-.018h.5a.75.75 0 0 1 0 1.5h-.5a.25.25 0 0 0-.25.25v.75c0 .28.114.532.3.714a.75.75 0 1 1-1.05 1.072A2.495 2.495 0 0 1 2 11.5V8.75A.75.75 0 0 1 2.75 8ZM11 .75a.75.75 0 0 1 .75-.75h1.5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V1.5h-.75A.75.75 0 0 1 11 .75Zm-5 0A.75.75 0 0 1 6.75 0h2.5a.75.75 0 0 1 0 1.5h-2.5A.75.75 0 0 1 6 .75Zm0 9A.75.75 0 0 1 6.75 9h2.5a.75.75 0 0 1 0 1.5h-2.5A.75.75 0 0 1 6 9.75ZM4.992.662a.75.75 0 0 1-.636.848c-.436.063-.783.41-.846.846a.751.751 0 0 1-1.485-.212A2.501 2.501 0 0 1 4.144.025a.75.75 0 0 1 .848.637ZM2.75 4a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 2.75 4Zm10.5 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5a.75.75 0 0 1 .75-.75Z">
              </path>
             </svg>
            </div>
            <span class="flex-auto min-width-0 width-fit">
             Public template repository
            </span>
           </div>
          </div>
         </div>
        </div>
        <div class="border-bottom mx-xl-5">
        </div>
       </div>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <a class="d-none js-github-dev-shortcut" data-hotkey=".,Mod+Alt+." href="https://github.dev/">
         Open in github.dev
        </a>
        <a class="d-none js-github-dev-new-tab-shortcut" data-hotkey="Shift+.,Shift+&gt;,&gt;" href="https://github.dev/" rel="noopener noreferrer" target="_blank">
         Open in a new github.dev tab
        </a>
        <a class="d-none" data-hotkey=",,Mod+Alt+," href="/codespaces/new/stemangiola/tidybulk?resume=1" target="_blank">
         Open in codespace
        </a>
        <h1 class="sr-only">
         stemangiola/tidybulk
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/stemangiola/tidybulk/spoofed_commit_check/58d34393d0dd970f663d0ba31b0248c0fad50462">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-81fbc7d814d2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-8a9f668f1de0.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-11980ca85c50.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":true,"path":"/","repo":{"id":205042805,"defaultBranch":"master","name":"tidybulk","ownerLogin":"stemangiola","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2019-08-29T07:46:30.000+08:00","ownerAvatar":"https://avatars.githubusercontent.com/u/7232890?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":{"id":126681664,"login":"ninenia","userEmail":"2945518981@qq.com"},"refInfo":{"name":"master","listCacheKey":"v0:1707710079.0","canEdit":true,"refType":"branch","currentOid":"58d34393d0dd970f663d0ba31b0248c0fad50462"},"tree":{"items":[{"name":".github","path":".github","contentType":"directory"},{"name":"R","path":"R","contentType":"directory"},{"name":"data","path":"data","contentType":"directory"},{"name":"dev","path":"dev","contentType":"directory"},{"name":"inst","path":"inst","contentType":"directory"},{"name":"man","path":"man","contentType":"directory"},{"name":"tests","path":"tests","contentType":"directory"},{"name":"vignettes","path":"vignettes","contentType":"directory"},{"name":".Rbuildignore","path":".Rbuildignore","contentType":"file"},{"name":".coveralls.yml","path":".coveralls.yml","contentType":"file"},{"name":".gitignore","path":".gitignore","contentType":"file"},{"name":".travis.yml","path":".travis.yml","contentType":"file"},{"name":"DESCRIPTION","path":"DESCRIPTION","contentType":"file"},{"name":"NAMESPACE","path":"NAMESPACE","contentType":"file"},{"name":"README.Rmd","path":"README.Rmd","contentType":"file"},{"name":"README.md","path":"README.md","contentType":"file"},{"name":"_pkgdown.yml","path":"_pkgdown.yml","contentType":"file"},{"name":"codecov.yml","path":"codecov.yml","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":18,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/stemangiola/tidybulk/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":true,"httpUrl":"https://github.com/stemangiola/tidybulk.git","showCloneWarning":false,"sshUrl":"git@github.com:stemangiola/tidybulk.git","sshCertificatesRequired":false,"sshCertificatesAvailable":false,"ghCliUrl":"gh repo clone stemangiola/tidybulk","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"x-github-client://openRepo/https://github.com/stemangiola/tidybulk","showVisualStudioCloneButton":true,"visualStudioCloneUrl":"git-client://clone?repo=https%3A%2F%2Fgithub.com%2Fstemangiola%2Ftidybulk","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/stemangiola/tidybulk/archive/refs/heads/master.zip"}},"repoPolicyInfo":{"allowed":true,"canBill":true,"changesWouldBeSafe":true,"disabledByBusiness":false,"disabledByOrganization":false,"hasIpAllowLists":false},"currentUserIsEnterpriseManaged":false,"enterpriseManagedBusinessName":null,"codespacesEnabled":true,"hasAccessToCodespaces":true},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"1,436","overviewFiles":[{"displayName":"README.md","repoName":"tidybulk","refName":"master","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003etidybulk - part of tidyTranscriptomics\u003c/h1\u003e\u003ca id=\"user-content-tidybulk---part-of-tidytranscriptomics\" class=\"anchor\" aria-label=\"Permalink: tidybulk - part of tidyTranscriptomics\" href=\"#tidybulk---part-of-tidytranscriptomics\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\n\u003cp dir=\"auto\"\u003e\u003ca href=\"https://www.tidyverse.org/lifecycle/#maturing\" rel=\"nofollow\"\u003e\u003cimg src=\"https://camo.githubusercontent.com/ee7536b3f1f530e91563303a6cdb32710af61731eb2f482dcb1af6b7d1bb42b2/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f6c6966656379636c652d6d61747572696e672d626c75652e737667\" alt=\"Lifecycle:maturing\" data-canonical-src=\"https://img.shields.io/badge/lifecycle-maturing-blue.svg\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\n\u003ca href=\"https://github.com/stemangiola/tidybulk/actions\"\u003e\u003cimg src=\"https://github.com/stemangiola/tidybulk/workflows/R-CMD-check-bioc/badge.svg\" alt=\"R build status\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eBrings transcriptomics to the tidyverse!\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eThe code is released under the version 3 of the GNU General Public\nLicense.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/stemangiola/tidybulk/blob/master/man/figures/logo.png\"\u003e\u003cimg src=\"/stemangiola/tidybulk/raw/master/man/figures/logo.png\" width=\"120px\" height=\"139px\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ewebsite:\n\u003ca href=\"http://stemangiola.github.io/tidybulk/\" rel=\"nofollow\"\u003estemangiola.github.io/tidybulk/\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePlease have a look also to\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/stemangiola/tidySummarizedExperiment\"\u003etidySummarizedExperiment\u003c/a\u003e\nfor bulk data tidy representation\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/stemangiola/tidySingleCellExperiment\"\u003etidySingleCellExperiment\u003c/a\u003e\nfor single-cell data tidy representation\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/stemangiola/tidyseurat\"\u003etidyseurat\u003c/a\u003e for\nsingle-cell data tidy representation\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/stemangiola/tidyHeatmap\"\u003etidyHeatmap\u003c/a\u003e for\nheatmaps produced with tidy principles analysis and manipulation\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/stemangiola/nanny\"\u003enanny\u003c/a\u003e for tidy high-level\ndata analysis and manipulation\u003c/li\u003e\n\u003cli\u003e\u003ca href=\"https://github.com/stemangiola/tidygate\"\u003etidygate\u003c/a\u003e for adding\ncustom gate information to your tibble\u003c/li\u003e\n\u003c/ul\u003e\n\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/stemangiola/tidybulk/blob/master/man/figures/new_SE_usage-01.png\"\u003e\u003cimg src=\"/stemangiola/tidybulk/raw/master/man/figures/new_SE_usage-01.png\" width=\"800px\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFunctions/utilities available\u003c/h2\u003e\u003ca id=\"user-content-functionsutilities-available\" class=\"anchor\" aria-label=\"Permalink: Functions/utilities available\" href=\"#functionsutilities-available\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003eFunction\u003c/th\u003e\n\u003cth\u003eDescription\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003eaggregate_duplicates\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eAggregate abundance and annotation of duplicated transcripts in a robust way\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003eidentify_abundant\u003c/code\u003e \u003ccode\u003ekeep_abundant\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eidentify or keep the abundant genes\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003ekeep_variable\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eFilter for top variable features\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003escale_abundance\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eScale (normalise) abundance for RNA sequencing depth\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003ereduce_dimensions\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003ePerform dimensionality reduction (PCA, MDS, tSNE, UMAP)\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003ecluster_elements\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eLabels elements with cluster identity (kmeans, SNN)\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003eremove_redundancy\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eFilter out elements with highly correlated features\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003eadjust_abundance\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eRemove known unwanted variation (Combat)\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003etest_differential_abundance\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eDifferential transcript abundance testing (DESeq2, edgeR, voom)\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003edeconvolve_cellularity\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eEstimated tissue composition (Cibersort, llsr, epic, xCell, mcp_counter, quantiseq\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003etest_differential_cellularity\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eDifferential cell-type abundance testing\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003etest_stratification_cellularity\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eEstimate Kaplan-Meier survival differences\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003etest_gene_enrichment\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eGene enrichment analyses (EGSEA)\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003etest_gene_overrepresentation\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eGene enrichment on list of transcript names (no rank)\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003etest_gene_rank\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eGene enrichment on list of transcript (GSEA)\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003eimpute_missing_abundance\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eImpute abundance for missing data points using sample groupings\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003eUtilities\u003c/th\u003e\n\u003cth\u003eDescription\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003eget_bibliography\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eGet the bibliography of your workflow\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003etidybulk\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eadd tidybulk attributes to a tibble object\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003etidybulk_SAM_BAM\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eConvert SAM BAM files into tidybulk tibble\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003epivot_sample\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eSelect sample-wise columns/information\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003epivot_transcript\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eSelect transcript-wise columns/information\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003erotate_dimensions\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eRotate two dimensions of a degree\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003eensembl_to_symbol\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eAdd gene symbol from ensembl IDs\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003esymbol_to_entrez\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eAdd entrez ID from gene symbol\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003e\u003ccode\u003edescribe_transcript\u003c/code\u003e\u003c/td\u003e\n\u003ctd\u003eAdd gene description from gene symbol\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cp dir=\"auto\"\u003eAll functions are directly compatibble with \u003ccode\u003eSummarizedExperiment\u003c/code\u003e\nobject.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eInstallation\u003c/h2\u003e\u003ca id=\"user-content-installation\" class=\"anchor\" aria-label=\"Permalink: Installation\" href=\"#installation\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFrom Bioconductor\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"BiocManager::install(\u0026quot;tidybulk\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-e\"\u003eBiocManager\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003einstall(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etidybulk\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFrom Github\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"devtools::install_github(\u0026quot;stemangiola/tidybulk\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-e\"\u003edevtools\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003einstall_github(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003estemangiola/tidybulk\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch1 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eData\u003c/h1\u003e\u003ca id=\"user-content-data\" class=\"anchor\" aria-label=\"Permalink: Data\" href=\"#data\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe will use a \u003ccode\u003eSummarizedExperiment\u003c/code\u003e object\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## # A SummarizedExperiment-tibble abstraction: 408,624 × 48\n## # �[90mFeatures=8513 | Samples=48 | Assays=count�[0m\n##    .feature .sample    count Cell.type time  condition batch factor_of_interest\n##    \u0026lt;chr\u0026gt;    \u0026lt;chr\u0026gt;      \u0026lt;dbl\u0026gt; \u0026lt;fct\u0026gt;     \u0026lt;fct\u0026gt; \u0026lt;lgl\u0026gt;     \u0026lt;fct\u0026gt; \u0026lt;lgl\u0026gt;             \n##  1 A1BG     SRR1740034   153 b_cell    0 d   TRUE      0     TRUE              \n##  2 A1BG-AS1 SRR1740034    83 b_cell    0 d   TRUE      0     TRUE              \n##  3 AAAS     SRR1740034   868 b_cell    0 d   TRUE      0     TRUE              \n##  4 AACS     SRR1740034   222 b_cell    0 d   TRUE      0     TRUE              \n##  5 AAGAB    SRR1740034   590 b_cell    0 d   TRUE      0     TRUE              \n##  6 AAMDC    SRR1740034    48 b_cell    0 d   TRUE      0     TRUE              \n##  7 AAMP     SRR1740034  1257 b_cell    0 d   TRUE      0     TRUE              \n##  8 AANAT    SRR1740034   284 b_cell    0 d   TRUE      0     TRUE              \n##  9 AAR2     SRR1740034   379 b_cell    0 d   TRUE      0     TRUE              \n## 10 AARS2    SRR1740034   685 b_cell    0 d   TRUE      0     TRUE              \n## # … with 40 more rows\n## # ℹ Use `print(n = ...)` to see more rows\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## # A SummarizedExperiment-tibble abstraction: 408,624 × 48\n## # �[90mFeatures=8513 | Samples=48 | Assays=count�[0m\n##    .feature .sample    count Cell.type time  condition batch factor_of_interest\n##    \u0026lt;chr\u0026gt;    \u0026lt;chr\u0026gt;      \u0026lt;dbl\u0026gt; \u0026lt;fct\u0026gt;     \u0026lt;fct\u0026gt; \u0026lt;lgl\u0026gt;     \u0026lt;fct\u0026gt; \u0026lt;lgl\u0026gt;             \n##  1 A1BG     SRR1740034   153 b_cell    0 d   TRUE      0     TRUE              \n##  2 A1BG-AS1 SRR1740034    83 b_cell    0 d   TRUE      0     TRUE              \n##  3 AAAS     SRR1740034   868 b_cell    0 d   TRUE      0     TRUE              \n##  4 AACS     SRR1740034   222 b_cell    0 d   TRUE      0     TRUE              \n##  5 AAGAB    SRR1740034   590 b_cell    0 d   TRUE      0     TRUE              \n##  6 AAMDC    SRR1740034    48 b_cell    0 d   TRUE      0     TRUE              \n##  7 AAMP     SRR1740034  1257 b_cell    0 d   TRUE      0     TRUE              \n##  8 AANAT    SRR1740034   284 b_cell    0 d   TRUE      0     TRUE              \n##  9 AAR2     SRR1740034   379 b_cell    0 d   TRUE      0     TRUE              \n## 10 AARS2    SRR1740034   685 b_cell    0 d   TRUE      0     TRUE              \n## # … with 40 more rows\n## # ℹ Use `print(n = ...)` to see more rows\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eLoading \u003ccode\u003etidySummarizedExperiment\u003c/code\u003e will automatically abstract this\nobject as \u003ccode\u003etibble\u003c/code\u003e, so we can display it and manipulate it with tidy\ntools. Although it looks different, and more tools (tidyverse) are\navailable to us, this object is in fact a \u003ccode\u003eSummarizedExperiment\u003c/code\u003e object.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"class(counts_SE)\"\u003e\u003cpre\u003eclass(\u003cspan class=\"pl-smi\"\u003ecounts_SE\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## [1] \u0026quot;SummarizedExperiment\u0026quot;\n## attr(,\u0026quot;package\u0026quot;)\n## [1] \u0026quot;SummarizedExperiment\u0026quot;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## [1] \"SummarizedExperiment\"\n## attr(,\"package\")\n## [1] \"SummarizedExperiment\"\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eGet the bibliography of your workflow\u003c/h2\u003e\u003ca id=\"user-content-get-the-bibliography-of-your-workflow\" class=\"anchor\" aria-label=\"Permalink: Get the bibliography of your workflow\" href=\"#get-the-bibliography-of-your-workflow\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFirst of all, you can cite all articles utilised within your workflow\nautomatically from any tidybulk tibble\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE %\u0026gt;%   get_bibliography()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%   get_bibliography()\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAggregate duplicated \u003ccode\u003etranscripts\u003c/code\u003e\u003c/h2\u003e\u003ca id=\"user-content-aggregate-duplicated-transcripts\" class=\"anchor\" aria-label=\"Permalink: Aggregate duplicated transcripts\" href=\"#aggregate-duplicated-transcripts\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003etidybulk provide the \u003ccode\u003eaggregate_duplicates\u003c/code\u003e function to aggregate\nduplicated transcripts (e.g., isoforms, ensembl). For example, we often\nhave to convert ensembl symbols to gene/transcript symbol, but in doing\nso we have to deal with duplicates. \u003ccode\u003eaggregate_duplicates\u003c/code\u003e takes a\ntibble and column names (as symbols; for \u003ccode\u003esample\u003c/code\u003e, \u003ccode\u003etranscript\u003c/code\u003e and\n\u003ccode\u003ecount\u003c/code\u003e) as arguments and returns a tibble with transcripts with the\nsame name aggregated. All the rest of the columns are appended, and\nfactors and boolean are appended as characters.\u003c/p\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eTidyTranscriptomics\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"rowData(counts_SE)$gene_name = rownames(counts_SE)\ncounts_SE.aggr = counts_SE %\u0026gt;% aggregate_duplicates(.transcript = gene_name)\"\u003e\u003cpre\u003erowData(\u003cspan class=\"pl-smi\"\u003ecounts_SE\u003c/span\u003e)\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-v\"\u003egene_name\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e rownames(\u003cspan class=\"pl-smi\"\u003ecounts_SE\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003ecounts_SE.aggr\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecounts_SE\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% aggregate_duplicates(\u003cspan class=\"pl-v\"\u003e.transcript\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003egene_name\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eStandard procedure (comparative purpose)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"temp = data.frame(\n    symbol = dge_list$genes$symbol,\n    dge_list$counts\n)\ndge_list.nr \u0026lt;- by(temp, temp$symbol,\n    function(df)\n        if(length(df[1,1])\u0026gt;0)\n            matrixStats:::colSums(as.matrix(df[,-1]))\n)\ndge_list.nr \u0026lt;- do.call(\u0026quot;rbind\u0026quot;, dge_list.nr)\ncolnames(dge_list.nr) \u0026lt;- colnames(dge_list)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003etemp\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003edata.frame\u003c/span\u003e(\n    \u003cspan class=\"pl-v\"\u003esymbol\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edge_list\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egenes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esymbol\u003c/span\u003e,\n    \u003cspan class=\"pl-smi\"\u003edge_list\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e\n)\n\u003cspan class=\"pl-smi\"\u003edge_list.nr\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e by(\u003cspan class=\"pl-smi\"\u003etemp\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etemp\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esymbol\u003c/span\u003e,\n    \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e)\n        \u003cspan class=\"pl-k\"\u003eif\u003c/span\u003e(length(\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e])\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e)\n            \u003cspan class=\"pl-e\"\u003ematrixStats\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:::\u003c/span\u003ecolSums(as.matrix(\u003cspan class=\"pl-smi\"\u003edf\u003c/span\u003e[,\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]))\n)\n\u003cspan class=\"pl-smi\"\u003edge_list.nr\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e do.call(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003erbind\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003edge_list.nr\u003c/span\u003e)\ncolnames(\u003cspan class=\"pl-smi\"\u003edge_list.nr\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e colnames(\u003cspan class=\"pl-smi\"\u003edge_list\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eScale \u003ccode\u003ecounts\u003c/code\u003e\u003c/h2\u003e\u003ca id=\"user-content-scale-counts\" class=\"anchor\" aria-label=\"Permalink: Scale counts\" href=\"#scale-counts\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe may want to compensate for sequencing depth, scaling the transcript\nabundance (e.g., with TMM algorithm, Robinson and Oshlack\ndoi.org/10.1186/gb-2010-11-3-r25). \u003ccode\u003escale_abundance\u003c/code\u003e takes a tibble,\ncolumn names (as symbols; for \u003ccode\u003esample\u003c/code\u003e, \u003ccode\u003etranscript\u003c/code\u003e and \u003ccode\u003ecount\u003c/code\u003e) and a\nmethod as arguments and returns a tibble with additional columns with\nscaled data as \u003ccode\u003e\u0026lt;NAME OF COUNT COLUMN\u0026gt;_scaled\u003c/code\u003e.\u003c/p\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eTidyTranscriptomics\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm = counts_SE.aggr %\u0026gt;% identify_abundant(factor_of_interest = condition) %\u0026gt;% scale_abundance()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_SE.norm\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecounts_SE.aggr\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% identify_abundant(\u003cspan class=\"pl-v\"\u003efactor_of_interest\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003econdition\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% scale_abundance()\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eStandard procedure (comparative purpose)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(edgeR)\n\ndgList \u0026lt;- DGEList(count_m=x,group=group)\nkeep \u0026lt;- filterByExpr(dgList)\ndgList \u0026lt;- dgList[keep,,keep.lib.sizes=FALSE]\n[...]\ndgList \u0026lt;- calcNormFactors(dgList, method=\u0026quot;TMM\u0026quot;)\nnorm_counts.table \u0026lt;- cpm(dgList)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eedgeR\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e DGEList(\u003cspan class=\"pl-v\"\u003ecount_m\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003egroup\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ekeep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e filterByExpr(\u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ekeep\u003c/span\u003e,,\u003cspan class=\"pl-v\"\u003ekeep.lib.sizes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e]\n[\u003cspan class=\"pl-k\"\u003e...\u003c/span\u003e]\n\u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e calcNormFactors(\u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eTMM\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003enorm_counts.table\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e cpm(\u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe can easily plot the scaled density to check the scaling outcome. On\nthe x axis we have the log scaled counts, on the y axes we have the\ndensity, data is grouped by sample and coloured by cell type.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm %\u0026gt;%\n    ggplot(aes(count_scaled + 1, group=sample, color=`Cell.type`)) +\n    geom_density() +\n    scale_x_log10() +\n    my_theme\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE.norm\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-smi\"\u003ecount_scaled\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003egroup\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Cell.type`\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    geom_density() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    scale_x_log10() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n    \u003cspan class=\"pl-smi\"\u003emy_theme\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/stemangiola/tidybulk/blob/master/man/figures/plot_normalise-1.png\"\u003e\u003cimg src=\"/stemangiola/tidybulk/raw/master/man/figures/plot_normalise-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFilter \u003ccode\u003evariable transcripts\u003c/code\u003e\u003c/h2\u003e\u003ca id=\"user-content-filter-variable-transcripts\" class=\"anchor\" aria-label=\"Permalink: Filter variable transcripts\" href=\"#filter-variable-transcripts\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe may want to identify and filter variable transcripts.\u003c/p\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eTidyTranscriptomics\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.variable = counts_SE.norm %\u0026gt;% keep_variable()\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_SE.norm.variable\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecounts_SE.norm\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% keep_variable()\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eStandard procedure (comparative purpose)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(edgeR)\n\nx = norm_counts.table\n\ns \u0026lt;- rowMeans((x-rowMeans(x))^2)\no \u0026lt;- order(s,decreasing=TRUE)\nx \u0026lt;- x[o[1L:top],,drop=FALSE]\n\nnorm_counts.table = norm_counts.table[rownames(x)]\n\nnorm_counts.table$cell_type = tibble_counts[\n    match(\n        tibble_counts$sample,\n        rownames(norm_counts.table)\n    ),\n    \u0026quot;Cell.type\u0026quot;\n]\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eedgeR\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enorm_counts.table\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003es\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e rowMeans((\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e-\u003c/span\u003erowMeans(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e))\u003cspan class=\"pl-k\"\u003e^\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eo\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e order(\u003cspan class=\"pl-smi\"\u003es\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003edecreasing\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003eo\u003c/span\u003e[\u003cspan class=\"pl-c1\"\u003e1L\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003etop\u003c/span\u003e],,\u003cspan class=\"pl-v\"\u003edrop\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e]\n\n\u003cspan class=\"pl-v\"\u003enorm_counts.table\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003enorm_counts.table\u003c/span\u003e[rownames(\u003cspan class=\"pl-smi\"\u003ex\u003c/span\u003e)]\n\n\u003cspan class=\"pl-smi\"\u003enorm_counts.table\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-v\"\u003ecell_type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etibble_counts\u003c/span\u003e[\n    match(\n        \u003cspan class=\"pl-smi\"\u003etibble_counts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e,\n        rownames(\u003cspan class=\"pl-smi\"\u003enorm_counts.table\u003c/span\u003e)\n    ),\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCell.type\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n]\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eReduce \u003ccode\u003edimensions\u003c/code\u003e\u003c/h2\u003e\u003ca id=\"user-content-reduce-dimensions\" class=\"anchor\" aria-label=\"Permalink: Reduce dimensions\" href=\"#reduce-dimensions\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe may want to reduce the dimensions of our data, for example using PCA\nor MDS algorithms. \u003ccode\u003ereduce_dimensions\u003c/code\u003e takes a tibble, column names (as\nsymbols; for \u003ccode\u003esample\u003c/code\u003e, \u003ccode\u003etranscript\u003c/code\u003e and \u003ccode\u003ecount\u003c/code\u003e) and a method (e.g., MDS\nor PCA) as arguments and returns a tibble with additional columns for\nthe reduced dimensions.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eMDS\u003c/strong\u003e (Robinson et al., 10.1093/bioinformatics/btp616)\u003c/p\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eTidyTranscriptomics\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.MDS =\n  counts_SE.norm %\u0026gt;%\n  reduce_dimensions(method=\u0026quot;MDS\u0026quot;, .dims = 6)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_SE.norm.MDS\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\n  \u003cspan class=\"pl-smi\"\u003ecounts_SE.norm\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  reduce_dimensions(\u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eMDS\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003e.dims\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eStandard procedure (comparative purpose)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(limma)\n\ncount_m_log = log(count_m + 1)\ncmds = limma::plotMDS(ndim = .dims, plot = FALSE)\n\ncmds = cmds %$% \n    cmdscale.out %\u0026gt;%\n    setNames(sprintf(\u0026quot;Dim%s\u0026quot;, 1:6))\n\ncmds$cell_type = tibble_counts[\n    match(tibble_counts$sample, rownames(cmds)),\n    \u0026quot;Cell.type\u0026quot;\n]\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003elimma\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003ecount_m_log\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e log(\u003cspan class=\"pl-smi\"\u003ecount_m\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003ecmds\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-e\"\u003elimma\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eplotMDS(\u003cspan class=\"pl-v\"\u003endim\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.dims\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eplot\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003ecmds\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecmds\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e% \n    \u003cspan class=\"pl-smi\"\u003ecmdscale.out\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    setNames(sprintf(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDim%s\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e))\n\n\u003cspan class=\"pl-smi\"\u003ecmds\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-v\"\u003ecell_type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etibble_counts\u003c/span\u003e[\n    match(\u003cspan class=\"pl-smi\"\u003etibble_counts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e, rownames(\u003cspan class=\"pl-smi\"\u003ecmds\u003c/span\u003e)),\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCell.type\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n]\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOn the x and y axes axis we have the reduced dimensions 1 to 3, data is\ncoloured by cell type.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.MDS %\u0026gt;% pivot_sample()  %\u0026gt;% select(contains(\u0026quot;Dim\u0026quot;), everything())\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.MDS\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% pivot_sample()  %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% select(contains(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDim\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), everything())\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## # A tibble: 48 × 14\n##      Dim1   Dim2   Dim3     Dim4    Dim5    Dim6 .sample   Cell.…¹ time  condi…²\n##     \u0026lt;dbl\u0026gt;  \u0026lt;dbl\u0026gt;  \u0026lt;dbl\u0026gt;    \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt; \u0026lt;chr\u0026gt;     \u0026lt;fct\u0026gt;   \u0026lt;fct\u0026gt; \u0026lt;lgl\u0026gt;  \n##  1 -1.46   0.220 -1.68   0.0553   0.0658 -0.126  SRR17400… b_cell  0 d   TRUE   \n##  2 -1.46   0.226 -1.71   0.0300   0.0454 -0.137  SRR17400… b_cell  1 d   TRUE   \n##  3 -1.44   0.193 -1.60   0.0890   0.0503 -0.121  SRR17400… b_cell  3 d   TRUE   \n##  4 -1.44   0.198 -1.67   0.0891   0.0543 -0.110  SRR17400… b_cell  7 d   TRUE   \n##  5  0.243 -1.42   0.182  0.00642 -0.503  -0.131  SRR17400… dendri… 0 d   FALSE  \n##  6  0.191 -1.42   0.195  0.0180  -0.457  -0.130  SRR17400… dendri… 1 d   FALSE  \n##  7  0.257 -1.42   0.152  0.0130  -0.582  -0.0927 SRR17400… dendri… 3 d   FALSE  \n##  8  0.162 -1.43   0.189  0.0232  -0.452  -0.109  SRR17400… dendri… 7 d   FALSE  \n##  9  0.516 -1.47   0.240 -0.251    0.457  -0.119  SRR17400… monocy… 0 d   FALSE  \n## 10  0.514 -1.41   0.231 -0.219    0.458  -0.131  SRR17400… monocy… 1 d   FALSE  \n## # … with 38 more rows, 4 more variables: batch \u0026lt;fct\u0026gt;, factor_of_interest \u0026lt;lgl\u0026gt;,\n## #   TMM \u0026lt;dbl\u0026gt;, multiplier \u0026lt;dbl\u0026gt;, and abbreviated variable names ¹​Cell.type,\n## #   ²​condition\n## # ℹ Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## # A tibble: 48 × 14\n##      Dim1   Dim2   Dim3     Dim4    Dim5    Dim6 .sample   Cell.…¹ time  condi…²\n##     \u0026lt;dbl\u0026gt;  \u0026lt;dbl\u0026gt;  \u0026lt;dbl\u0026gt;    \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt; \u0026lt;chr\u0026gt;     \u0026lt;fct\u0026gt;   \u0026lt;fct\u0026gt; \u0026lt;lgl\u0026gt;  \n##  1 -1.46   0.220 -1.68   0.0553   0.0658 -0.126  SRR17400… b_cell  0 d   TRUE   \n##  2 -1.46   0.226 -1.71   0.0300   0.0454 -0.137  SRR17400… b_cell  1 d   TRUE   \n##  3 -1.44   0.193 -1.60   0.0890   0.0503 -0.121  SRR17400… b_cell  3 d   TRUE   \n##  4 -1.44   0.198 -1.67   0.0891   0.0543 -0.110  SRR17400… b_cell  7 d   TRUE   \n##  5  0.243 -1.42   0.182  0.00642 -0.503  -0.131  SRR17400… dendri… 0 d   FALSE  \n##  6  0.191 -1.42   0.195  0.0180  -0.457  -0.130  SRR17400… dendri… 1 d   FALSE  \n##  7  0.257 -1.42   0.152  0.0130  -0.582  -0.0927 SRR17400… dendri… 3 d   FALSE  \n##  8  0.162 -1.43   0.189  0.0232  -0.452  -0.109  SRR17400… dendri… 7 d   FALSE  \n##  9  0.516 -1.47   0.240 -0.251    0.457  -0.119  SRR17400… monocy… 0 d   FALSE  \n## 10  0.514 -1.41   0.231 -0.219    0.458  -0.131  SRR17400… monocy… 1 d   FALSE  \n## # … with 38 more rows, 4 more variables: batch \u0026lt;fct\u0026gt;, factor_of_interest \u0026lt;lgl\u0026gt;,\n## #   TMM \u0026lt;dbl\u0026gt;, multiplier \u0026lt;dbl\u0026gt;, and abbreviated variable names ¹​Cell.type,\n## #   ²​condition\n## # ℹ Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.MDS %\u0026gt;%\n    pivot_sample() %\u0026gt;%\n  GGally::ggpairs(columns = 6:(6+5), ggplot2::aes(colour=`Cell.type`))\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.MDS\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    pivot_sample() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  \u003cspan class=\"pl-e\"\u003eGGally\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eggpairs(\u003cspan class=\"pl-v\"\u003ecolumns\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e(\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e), \u003cspan class=\"pl-e\"\u003eggplot2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eaes(\u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Cell.type`\u003c/span\u003e))\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/stemangiola/tidybulk/blob/master/man/figures/plot_mds-1.png\"\u003e\u003cimg src=\"/stemangiola/tidybulk/raw/master/man/figures/plot_mds-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003ePCA\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eTidyTranscriptomics\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.PCA =\n  counts_SE.norm %\u0026gt;%\n  reduce_dimensions(method=\u0026quot;PCA\u0026quot;, .dims = 6)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_SE.norm.PCA\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\n  \u003cspan class=\"pl-smi\"\u003ecounts_SE.norm\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  reduce_dimensions(\u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePCA\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003e.dims\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eStandard procedure (comparative purpose)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"count_m_log = log(count_m + 1)\npc = count_m_log %\u0026gt;% prcomp(scale = TRUE)\nvariance = pc$sdev^2\nvariance = (variance / sum(variance))[1:6]\npc$cell_type = counts[\n    match(counts$sample, rownames(pc)),\n    \u0026quot;Cell.type\u0026quot;\n]\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecount_m_log\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e log(\u003cspan class=\"pl-smi\"\u003ecount_m\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003epc\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecount_m_log\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% prcomp(\u003cspan class=\"pl-v\"\u003escale\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003evariance\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epc\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esdev\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e^\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e\n\u003cspan class=\"pl-v\"\u003evariance\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e (\u003cspan class=\"pl-smi\"\u003evariance\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e sum(\u003cspan class=\"pl-smi\"\u003evariance\u003c/span\u003e))[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e6\u003c/span\u003e]\n\u003cspan class=\"pl-smi\"\u003epc\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-v\"\u003ecell_type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e[\n    match(\u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e, rownames(\u003cspan class=\"pl-smi\"\u003epc\u003c/span\u003e)),\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCell.type\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n]\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eOn the x and y axes axis we have the reduced dimensions 1 to 3, data is\ncoloured by cell type.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.PCA %\u0026gt;% pivot_sample() %\u0026gt;% select(contains(\u0026quot;PC\u0026quot;), everything())\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.PCA\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% pivot_sample() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% select(contains(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ePC\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), everything())\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## # A tibble: 48 × 14\n##        PC1   PC2    PC3     PC4    PC5   PC6 .sample Cell.…¹ time  condi…² batch\n##      \u0026lt;dbl\u0026gt; \u0026lt;dbl\u0026gt;  \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;  \u0026lt;dbl\u0026gt; \u0026lt;dbl\u0026gt; \u0026lt;chr\u0026gt;   \u0026lt;fct\u0026gt;   \u0026lt;fct\u0026gt; \u0026lt;lgl\u0026gt;   \u0026lt;fct\u0026gt;\n##  1 -12.6   -2.52 -14.9  -0.424  -0.592 -1.22 SRR174… b_cell  0 d   TRUE    0    \n##  2 -12.6   -2.57 -15.2  -0.140  -0.388 -1.30 SRR174… b_cell  1 d   TRUE    1    \n##  3 -12.6   -2.41 -14.5  -0.714  -0.344 -1.10 SRR174… b_cell  3 d   TRUE    1    \n##  4 -12.5   -2.34 -14.9  -0.816  -0.427 -1.00 SRR174… b_cell  7 d   TRUE    1    \n##  5   0.189 13.0    1.66 -0.0269  4.64  -1.35 SRR174… dendri… 0 d   FALSE   0    \n##  6  -0.293 12.9    1.76 -0.0727  4.21  -1.28 SRR174… dendri… 1 d   FALSE   0    \n##  7   0.407 13.0    1.42 -0.0529  5.37  -1.01 SRR174… dendri… 3 d   FALSE   1    \n##  8  -0.620 13.0    1.73 -0.201   4.17  -1.07 SRR174… dendri… 7 d   FALSE   0    \n##  9   2.56  13.5    2.32  2.03   -4.32  -1.22 SRR174… monocy… 0 d   FALSE   1    \n## 10   2.65  13.1    2.21  1.80   -4.29  -1.30 SRR174… monocy… 1 d   FALSE   1    \n## # … with 38 more rows, 3 more variables: factor_of_interest \u0026lt;lgl\u0026gt;, TMM \u0026lt;dbl\u0026gt;,\n## #   multiplier \u0026lt;dbl\u0026gt;, and abbreviated variable names ¹​Cell.type, ²​condition\n## # ℹ Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## # A tibble: 48 × 14\n##        PC1   PC2    PC3     PC4    PC5   PC6 .sample Cell.…¹ time  condi…² batch\n##      \u0026lt;dbl\u0026gt; \u0026lt;dbl\u0026gt;  \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;  \u0026lt;dbl\u0026gt; \u0026lt;dbl\u0026gt; \u0026lt;chr\u0026gt;   \u0026lt;fct\u0026gt;   \u0026lt;fct\u0026gt; \u0026lt;lgl\u0026gt;   \u0026lt;fct\u0026gt;\n##  1 -12.6   -2.52 -14.9  -0.424  -0.592 -1.22 SRR174… b_cell  0 d   TRUE    0    \n##  2 -12.6   -2.57 -15.2  -0.140  -0.388 -1.30 SRR174… b_cell  1 d   TRUE    1    \n##  3 -12.6   -2.41 -14.5  -0.714  -0.344 -1.10 SRR174… b_cell  3 d   TRUE    1    \n##  4 -12.5   -2.34 -14.9  -0.816  -0.427 -1.00 SRR174… b_cell  7 d   TRUE    1    \n##  5   0.189 13.0    1.66 -0.0269  4.64  -1.35 SRR174… dendri… 0 d   FALSE   0    \n##  6  -0.293 12.9    1.76 -0.0727  4.21  -1.28 SRR174… dendri… 1 d   FALSE   0    \n##  7   0.407 13.0    1.42 -0.0529  5.37  -1.01 SRR174… dendri… 3 d   FALSE   1    \n##  8  -0.620 13.0    1.73 -0.201   4.17  -1.07 SRR174… dendri… 7 d   FALSE   0    \n##  9   2.56  13.5    2.32  2.03   -4.32  -1.22 SRR174… monocy… 0 d   FALSE   1    \n## 10   2.65  13.1    2.21  1.80   -4.29  -1.30 SRR174… monocy… 1 d   FALSE   1    \n## # … with 38 more rows, 3 more variables: factor_of_interest \u0026lt;lgl\u0026gt;, TMM \u0026lt;dbl\u0026gt;,\n## #   multiplier \u0026lt;dbl\u0026gt;, and abbreviated variable names ¹​Cell.type, ²​condition\n## # ℹ Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.PCA %\u0026gt;%\n     pivot_sample() %\u0026gt;%\n  GGally::ggpairs(columns = 11:13, ggplot2::aes(colour=`Cell.type`))\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.PCA\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n     pivot_sample() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  \u003cspan class=\"pl-e\"\u003eGGally\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eggpairs(\u003cspan class=\"pl-v\"\u003ecolumns\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e11\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e13\u003c/span\u003e, \u003cspan class=\"pl-e\"\u003eggplot2\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eaes(\u003cspan class=\"pl-v\"\u003ecolour\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Cell.type`\u003c/span\u003e))\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/stemangiola/tidybulk/blob/master/man/figures/plot_pca-1.png\"\u003e\u003cimg src=\"/stemangiola/tidybulk/raw/master/man/figures/plot_pca-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003etSNE\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eTidyTranscriptomics\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.tSNE =\n    breast_tcga_mini_SE %\u0026gt;%\n    identify_abundant() %\u0026gt;%\n    reduce_dimensions(\n        method = \u0026quot;tSNE\u0026quot;,\n        perplexity=10,\n        pca_scale =TRUE\n    )\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_SE.norm.tSNE\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\n    \u003cspan class=\"pl-smi\"\u003ebreast_tcga_mini_SE\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    identify_abundant() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    reduce_dimensions(\n        \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etSNE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003eperplexity\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003epca_scale\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e\n    )\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eStandard procedure (comparative purpose)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"count_m_log = log(count_m + 1)\n\ntsne = Rtsne::Rtsne(\n    t(count_m_log),\n    perplexity=10,\n        pca_scale =TRUE\n)$Y\ntsne$cell_type = tibble_counts[\n    match(tibble_counts$sample, rownames(tsne)),\n    \u0026quot;Cell.type\u0026quot;\n]\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecount_m_log\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e log(\u003cspan class=\"pl-smi\"\u003ecount_m\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003etsne\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-e\"\u003eRtsne\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eRtsne(\n    t(\u003cspan class=\"pl-smi\"\u003ecount_m_log\u003c/span\u003e),\n    \u003cspan class=\"pl-v\"\u003eperplexity\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e10\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003epca_scale\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e\n)\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eY\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003etsne\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-v\"\u003ecell_type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etibble_counts\u003c/span\u003e[\n    match(\u003cspan class=\"pl-smi\"\u003etibble_counts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e, rownames(\u003cspan class=\"pl-smi\"\u003etsne\u003c/span\u003e)),\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCell.type\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n]\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePlot\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.tSNE %\u0026gt;%\n    pivot_sample() %\u0026gt;%\n    select(contains(\u0026quot;tSNE\u0026quot;), everything()) \"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.tSNE\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    pivot_sample() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    select(contains(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etSNE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), everything()) \u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## # A tibble: 251 × 4\n##     tSNE1  tSNE2 .sample                      Call \n##     \u0026lt;dbl\u0026gt;  \u0026lt;dbl\u0026gt; \u0026lt;chr\u0026gt;                        \u0026lt;fct\u0026gt;\n##  1  -5.25  10.2  TCGA-A1-A0SD-01A-11R-A115-07 LumA \n##  2   6.41   2.79 TCGA-A1-A0SF-01A-11R-A144-07 LumA \n##  3  -9.28   6.63 TCGA-A1-A0SG-01A-11R-A144-07 LumA \n##  4  -1.76   4.82 TCGA-A1-A0SH-01A-11R-A084-07 LumA \n##  5  -1.41  12.2  TCGA-A1-A0SI-01A-11R-A144-07 LumB \n##  6  -1.89  -3.60 TCGA-A1-A0SJ-01A-11R-A084-07 LumA \n##  7  18.5  -13.4  TCGA-A1-A0SK-01A-12R-A084-07 Basal\n##  8  -4.03 -10.4  TCGA-A1-A0SM-01A-11R-A084-07 LumA \n##  9  -2.84 -10.8  TCGA-A1-A0SN-01A-11R-A144-07 LumB \n## 10 -19.3    5.03 TCGA-A1-A0SQ-01A-21R-A144-07 LumA \n## # … with 241 more rows\n## # ℹ Use `print(n = ...)` to see more rows\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## # A tibble: 251 × 4\n##     tSNE1  tSNE2 .sample                      Call \n##     \u0026lt;dbl\u0026gt;  \u0026lt;dbl\u0026gt; \u0026lt;chr\u0026gt;                        \u0026lt;fct\u0026gt;\n##  1  -5.25  10.2  TCGA-A1-A0SD-01A-11R-A115-07 LumA \n##  2   6.41   2.79 TCGA-A1-A0SF-01A-11R-A144-07 LumA \n##  3  -9.28   6.63 TCGA-A1-A0SG-01A-11R-A144-07 LumA \n##  4  -1.76   4.82 TCGA-A1-A0SH-01A-11R-A084-07 LumA \n##  5  -1.41  12.2  TCGA-A1-A0SI-01A-11R-A144-07 LumB \n##  6  -1.89  -3.60 TCGA-A1-A0SJ-01A-11R-A084-07 LumA \n##  7  18.5  -13.4  TCGA-A1-A0SK-01A-12R-A084-07 Basal\n##  8  -4.03 -10.4  TCGA-A1-A0SM-01A-11R-A084-07 LumA \n##  9  -2.84 -10.8  TCGA-A1-A0SN-01A-11R-A144-07 LumB \n## 10 -19.3    5.03 TCGA-A1-A0SQ-01A-21R-A144-07 LumA \n## # … with 241 more rows\n## # ℹ Use `print(n = ...)` to see more rows\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.tSNE %\u0026gt;%\n    pivot_sample() %\u0026gt;%\n    ggplot(aes(x = `tSNE1`, y = `tSNE2`, color=Call)) + geom_point() + my_theme\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.tSNE\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    pivot_sample() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e`tSNE1`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e`tSNE2`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCall\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_point() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emy_theme\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/stemangiola/tidybulk/blob/master/man/figures/unnamed-chunk-14-1.png\"\u003e\u003cimg src=\"/stemangiola/tidybulk/raw/master/man/figures/unnamed-chunk-14-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eRotate \u003ccode\u003edimensions\u003c/code\u003e\u003c/h2\u003e\u003ca id=\"user-content-rotate-dimensions\" class=\"anchor\" aria-label=\"Permalink: Rotate dimensions\" href=\"#rotate-dimensions\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe may want to rotate the reduced dimensions (or any two numeric columns\nreally) of our data, of a set angle. \u003ccode\u003erotate_dimensions\u003c/code\u003e takes a tibble,\ncolumn names (as symbols; for \u003ccode\u003esample\u003c/code\u003e, \u003ccode\u003etranscript\u003c/code\u003e and \u003ccode\u003ecount\u003c/code\u003e) and an\nangle as arguments and returns a tibble with additional columns for the\nrotated dimensions. The rotated dimensions will be added to the original\ndata set as \u003ccode\u003e\u0026lt;NAME OF DIMENSION\u0026gt; rotated \u0026lt;ANGLE\u0026gt;\u003c/code\u003e by default, or as\nspecified in the input arguments.\u003c/p\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eTidyTranscriptomics\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.MDS.rotated =\n  counts_SE.norm.MDS %\u0026gt;%\n    rotate_dimensions(`Dim1`, `Dim2`, rotation_degrees = 45, action=\u0026quot;get\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_SE.norm.MDS.rotated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\n  \u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.MDS\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    rotate_dimensions(\u003cspan class=\"pl-smi\"\u003e`Dim1`\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003e`Dim2`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003erotation_degrees\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e45\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eaction\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eget\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eStandard procedure (comparative purpose)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"rotation = function(m, d) {\n    r = d * pi / 180\n    ((bind_rows(\n        c(`1` = cos(r), `2` = -sin(r)),\n        c(`1` = sin(r), `2` = cos(r))\n    ) %\u0026gt;% as_matrix) %*% m)\n}\nmds_r = pca %\u0026gt;% rotation(rotation_degrees)\nmds_r$cell_type = counts[\n    match(counts$sample, rownames(mds_r)),\n    \u0026quot;Cell.type\u0026quot;\n]\"\u003e\u003cpre\u003e\u003cspan class=\"pl-en\"\u003erotation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003efunction\u003c/span\u003e(\u003cspan class=\"pl-smi\"\u003em\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e) {\n    \u003cspan class=\"pl-v\"\u003er\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ed\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e*\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003epi\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e180\u003c/span\u003e\n    ((bind_rows(\n        c(\u003cspan class=\"pl-smi\"\u003e`1`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e cos(\u003cspan class=\"pl-smi\"\u003er\u003c/span\u003e), \u003cspan class=\"pl-smi\"\u003e`2`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003esin(\u003cspan class=\"pl-smi\"\u003er\u003c/span\u003e)),\n        c(\u003cspan class=\"pl-smi\"\u003e`1`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e sin(\u003cspan class=\"pl-smi\"\u003er\u003c/span\u003e), \u003cspan class=\"pl-smi\"\u003e`2`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e cos(\u003cspan class=\"pl-smi\"\u003er\u003c/span\u003e))\n    ) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \u003cspan class=\"pl-smi\"\u003eas_matrix\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e%*%\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003em\u003c/span\u003e)\n}\n\u003cspan class=\"pl-v\"\u003emds_r\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003epca\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% rotation(\u003cspan class=\"pl-smi\"\u003erotation_degrees\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003emds_r\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-v\"\u003ecell_type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e[\n    match(\u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e, rownames(\u003cspan class=\"pl-smi\"\u003emds_r\u003c/span\u003e)),\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCell.type\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n]\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eOriginal\u003c/strong\u003e On the x and y axes axis we have the first two reduced\ndimensions, data is coloured by cell type.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.MDS.rotated %\u0026gt;%\n    ggplot(aes(x=`Dim1`, y=`Dim2`, color=`Cell.type` )) +\n  geom_point() +\n  my_theme\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.MDS.rotated\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Dim1`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Dim2`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Cell.type`\u003c/span\u003e )) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_point() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \u003cspan class=\"pl-smi\"\u003emy_theme\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/stemangiola/tidybulk/blob/master/man/figures/plot_rotate_1-1.png\"\u003e\u003cimg src=\"/stemangiola/tidybulk/raw/master/man/figures/plot_rotate_1-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eRotated\u003c/strong\u003e On the x and y axes axis we have the first two reduced\ndimensions rotated of 45 degrees, data is coloured by cell type.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.MDS.rotated %\u0026gt;%\n    pivot_sample() %\u0026gt;%\n    ggplot(aes(x=`Dim1_rotated_45`, y=`Dim2_rotated_45`, color=`Cell.type` )) +\n  geom_point() +\n  my_theme\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.MDS.rotated\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    pivot_sample() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Dim1_rotated_45`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Dim2_rotated_45`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Cell.type`\u003c/span\u003e )) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_point() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \u003cspan class=\"pl-smi\"\u003emy_theme\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/stemangiola/tidybulk/blob/master/man/figures/plot_rotate_2-1.png\"\u003e\u003cimg src=\"/stemangiola/tidybulk/raw/master/man/figures/plot_rotate_2-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTest \u003ccode\u003edifferential abundance\u003c/code\u003e\u003c/h2\u003e\u003ca id=\"user-content-test-differential-abundance\" class=\"anchor\" aria-label=\"Permalink: Test differential abundance\" href=\"#test-differential-abundance\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe may want to test for differential transcription between sample-wise\nfactors of interest (e.g., with edgeR). \u003ccode\u003etest_differential_abundance\u003c/code\u003e\ntakes a tibble, column names (as symbols; for \u003ccode\u003esample\u003c/code\u003e, \u003ccode\u003etranscript\u003c/code\u003e and\n\u003ccode\u003ecount\u003c/code\u003e) and a formula representing the desired linear model as\narguments and returns a tibble with additional columns for the\nstatistics from the hypothesis test (e.g., log fold change, p-value and\nfalse discovery rate).\u003c/p\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eTidyTranscriptomics\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.de =\n    counts_SE %\u0026gt;%\n    test_differential_abundance( ~ condition, action=\u0026quot;get\u0026quot;)\ncounts_SE.de\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_SE.de\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\n    \u003cspan class=\"pl-smi\"\u003ecounts_SE\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    test_differential_abundance( \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003econdition\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eaction\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eget\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ecounts_SE.de\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eStandard procedure (comparative purpose)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(edgeR)\n\ndgList \u0026lt;- DGEList(counts=counts_m,group=group)\nkeep \u0026lt;- filterByExpr(dgList)\ndgList \u0026lt;- dgList[keep,,keep.lib.sizes=FALSE]\ndgList \u0026lt;- calcNormFactors(dgList)\ndesign \u0026lt;- model.matrix(~group)\ndgList \u0026lt;- estimateDisp(dgList,design)\nfit \u0026lt;- glmQLFit(dgList,design)\nqlf \u0026lt;- glmQLFTest(fit,coef=2)\ntopTags(qlf, n=Inf)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eedgeR\u003c/span\u003e)\n\n\u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e DGEList(\u003cspan class=\"pl-v\"\u003ecounts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecounts_m\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003egroup\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ekeep\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e filterByExpr(\u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e[\u003cspan class=\"pl-smi\"\u003ekeep\u003c/span\u003e,,\u003cspan class=\"pl-v\"\u003ekeep.lib.sizes\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e]\n\u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e calcNormFactors(\u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003edesign\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e model.matrix(\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003egroup\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e estimateDisp(\u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003edesign\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003efit\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e glmQLFit(\u003cspan class=\"pl-smi\"\u003edgList\u003c/span\u003e,\u003cspan class=\"pl-smi\"\u003edesign\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eqlf\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e glmQLFTest(\u003cspan class=\"pl-smi\"\u003efit\u003c/span\u003e,\u003cspan class=\"pl-v\"\u003ecoef\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e)\ntopTags(\u003cspan class=\"pl-smi\"\u003eqlf\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003en\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eInf\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe functon \u003ccode\u003etest_differential_abundance\u003c/code\u003e operated with contrasts too.\nThe constrasts hve the name of the design matrix (generally\n\u0026lt;NAME_COLUMN_COVARIATE\u0026gt;\u0026lt;VALUES_OF_COVARIATE\u0026gt;)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.de =\n    counts_SE %\u0026gt;%\n    identify_abundant(factor_of_interest = condition) %\u0026gt;%\n    test_differential_abundance(\n        ~ 0 + condition,                  \n        .contrasts = c( \u0026quot;conditionTRUE - conditionFALSE\u0026quot;),\n        action=\u0026quot;get\u0026quot;\n    )\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_SE.de\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\n    \u003cspan class=\"pl-smi\"\u003ecounts_SE\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    identify_abundant(\u003cspan class=\"pl-v\"\u003efactor_of_interest\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003econdition\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    test_differential_abundance(\n        \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003econdition\u003c/span\u003e,                  \n        \u003cspan class=\"pl-v\"\u003e.contrasts\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e c( \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003econditionTRUE - conditionFALSE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e),\n        \u003cspan class=\"pl-v\"\u003eaction\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eget\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n    )\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eAdjust \u003ccode\u003ecounts\u003c/code\u003e\u003c/h2\u003e\u003ca id=\"user-content-adjust-counts\" class=\"anchor\" aria-label=\"Permalink: Adjust counts\" href=\"#adjust-counts\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe may want to adjust \u003ccode\u003ecounts\u003c/code\u003e for (known) unwanted variation.\n\u003ccode\u003eadjust_abundance\u003c/code\u003e takes as arguments a tibble, column names (as\nsymbols; for \u003ccode\u003esample\u003c/code\u003e, \u003ccode\u003etranscript\u003c/code\u003e and \u003ccode\u003ecount\u003c/code\u003e) and a formula\nrepresenting the desired linear model where the first covariate is the\nfactor of interest and the second covariate is the unwanted variation,\nand returns a tibble with additional columns for the adjusted counts as\n\u003ccode\u003e\u0026lt;COUNT COLUMN\u0026gt;_adjusted\u003c/code\u003e. At the moment just an unwanted covariated is\nallowed at a time.\u003c/p\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eTidyTranscriptomics\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.adj =\n    counts_SE.norm %\u0026gt;% adjust_abundance(    ~ factor_of_interest + batch)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_SE.norm.adj\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\n    \u003cspan class=\"pl-smi\"\u003ecounts_SE.norm\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% adjust_abundance(    \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efactor_of_interest\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ebatch\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eStandard procedure (comparative purpose)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(sva)\n\ncount_m_log = log(count_m + 1)\n\ndesign =\n        model.matrix(\n            object = ~ factor_of_interest + batch,\n            data = annotation\n        )\n\ncount_m_log.sva =\n    ComBat(\n            batch = design[,2],\n            mod = design,\n            ...\n        )\n\ncount_m_log.sva = ceiling(exp(count_m_log.sva) -1)\ncount_m_log.sva$cell_type = counts[\n    match(counts$sample, rownames(count_m_log.sva)),\n    \u0026quot;Cell.type\u0026quot;\n]\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003esva\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003ecount_m_log\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e log(\u003cspan class=\"pl-smi\"\u003ecount_m\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003edesign\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\n        model.matrix(\n            \u003cspan class=\"pl-v\"\u003eobject\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efactor_of_interest\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ebatch\u003c/span\u003e,\n            \u003cspan class=\"pl-v\"\u003edata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eannotation\u003c/span\u003e\n        )\n\n\u003cspan class=\"pl-v\"\u003ecount_m_log.sva\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\n    ComBat(\n            \u003cspan class=\"pl-v\"\u003ebatch\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edesign\u003c/span\u003e[,\u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e],\n            \u003cspan class=\"pl-v\"\u003emod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003edesign\u003c/span\u003e,\n            \u003cspan class=\"pl-k\"\u003e...\u003c/span\u003e\n        )\n\n\u003cspan class=\"pl-v\"\u003ecount_m_log.sva\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e ceiling(exp(\u003cspan class=\"pl-smi\"\u003ecount_m_log.sva\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003ecount_m_log.sva\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-v\"\u003ecell_type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e[\n    match(\u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e, rownames(\u003cspan class=\"pl-smi\"\u003ecount_m_log.sva\u003c/span\u003e)),\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCell.type\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n]\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDeconvolve \u003ccode\u003eCell type composition\u003c/code\u003e\u003c/h2\u003e\u003ca id=\"user-content-deconvolve-cell-type-composition\" class=\"anchor\" aria-label=\"Permalink: Deconvolve Cell type composition\" href=\"#deconvolve-cell-type-composition\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe may want to infer the cell type composition of our samples (with the\nalgorithm Cibersort; Newman et al., 10.1038/nmeth.3337).\n\u003ccode\u003edeconvolve_cellularity\u003c/code\u003e takes as arguments a tibble, column names (as\nsymbols; for \u003ccode\u003esample\u003c/code\u003e, \u003ccode\u003etranscript\u003c/code\u003e and \u003ccode\u003ecount\u003c/code\u003e) and returns a tibble\nwith additional columns for the adjusted cell type proportions.\u003c/p\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eTidyTranscriptomics\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.cibersort =\n    counts_SE %\u0026gt;%\n    deconvolve_cellularity(action=\u0026quot;get\u0026quot;, cores=1, prefix = \u0026quot;cibersort__\u0026quot;) \"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_SE.cibersort\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\n    \u003cspan class=\"pl-smi\"\u003ecounts_SE\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    deconvolve_cellularity(\u003cspan class=\"pl-v\"\u003eaction\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eget\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecores\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eprefix\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecibersort__\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e) \u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## \n## The downloaded binary packages are in\n##  /var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T//RtmpIi5KN6/downloaded_packages\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## \n## The downloaded binary packages are in\n##  /var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T//RtmpIi5KN6/downloaded_packages\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eStandard procedure (comparative purpose)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"source(‘CIBERSORT.R’)\ncount_m %\u0026gt;% write.table(\u0026quot;mixture_file.txt\u0026quot;)\nresults \u0026lt;- CIBERSORT(\n    \u0026quot;sig_matrix_file.txt\u0026quot;,\n    \u0026quot;mixture_file.txt\u0026quot;,\n    perm=100, QN=TRUE\n)\nresults$cell_type = tibble_counts[\n    match(tibble_counts$sample, rownames(results)),\n    \u0026quot;Cell.type\u0026quot;\n]\"\u003e\u003cpre\u003esource(‘\u003cspan class=\"pl-smi\"\u003eCIBERSORT.R\u003c/span\u003e’)\n\u003cspan class=\"pl-smi\"\u003ecount_m\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% write.table(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emixture_file.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-smi\"\u003eresults\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026lt;-\u003c/span\u003e CIBERSORT(\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003esig_matrix_file.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003emixture_file.txt\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eperm\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e100\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eQN\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e\n)\n\u003cspan class=\"pl-smi\"\u003eresults\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-v\"\u003ecell_type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etibble_counts\u003c/span\u003e[\n    match(\u003cspan class=\"pl-smi\"\u003etibble_counts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e, rownames(\u003cspan class=\"pl-smi\"\u003eresults\u003c/span\u003e)),\n    \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCell.type\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n]\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWith the new annotated data frame, we can plot the distributions of cell\ntypes across samples, and compare them with the nominal cell type labels\nto check for the purity of isolation. On the x axis we have the cell\ntypes inferred by Cibersort, on the y axis we have the inferred\nproportions. The data is facetted and coloured by nominal cell types\n(annotation given by the researcher after FACS sorting).\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.cibersort %\u0026gt;%\n    pivot_longer(\n        names_to= \u0026quot;Cell_type_inferred\u0026quot;, \n        values_to = \u0026quot;proportion\u0026quot;, \n        names_prefix =\u0026quot;cibersort__\u0026quot;, \n        cols=contains(\u0026quot;cibersort__\u0026quot;)\n    ) %\u0026gt;%\n  ggplot(aes(x=`Cell_type_inferred`, y=proportion, fill=`Cell.type`)) +\n  geom_boxplot() +\n  facet_wrap(~`Cell.type`) +\n  my_theme +\n  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5), aspect.ratio=1/5)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE.cibersort\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    pivot_longer(\n        \u003cspan class=\"pl-v\"\u003enames_to\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCell_type_inferred\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003evalues_to\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eproportion\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003enames_prefix\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecibersort__\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \n        \u003cspan class=\"pl-v\"\u003ecols\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003econtains(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecibersort__\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n    ) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Cell_type_inferred`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eproportion\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003efill\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Cell.type`\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_boxplot() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  facet_wrap(\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Cell.type`\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \u003cspan class=\"pl-smi\"\u003emy_theme\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  theme(\u003cspan class=\"pl-v\"\u003eaxis.text.x\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e element_text(\u003cspan class=\"pl-v\"\u003eangle\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e90\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ehjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003evjust\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e0.5\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003easpect.ratio\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e/\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e5\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/stemangiola/tidybulk/blob/master/man/figures/plot_cibersort-1.png\"\u003e\u003cimg src=\"/stemangiola/tidybulk/raw/master/man/figures/plot_cibersort-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eTest differential cell-type abundance\u003c/h2\u003e\u003ca id=\"user-content-test-differential-cell-type-abundance\" class=\"anchor\" aria-label=\"Permalink: Test differential cell-type abundance\" href=\"#test-differential-cell-type-abundance\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe can also perform a statistical test on the differential cell-type\nabundance across conditions\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"    counts_SE %\u0026gt;%\n    test_differential_cellularity(. ~ condition )\"\u003e\u003cpre\u003e    \u003cspan class=\"pl-smi\"\u003ecounts_SE\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    test_differential_cellularity(\u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003econdition\u003c/span\u003e )\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## \n## The downloaded binary packages are in\n##  /var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T//RtmpIi5KN6/downloaded_packages\n\n## # A tibble: 22 × 7\n##    .cell_type                  cell_t…¹ estim…² estim…³ std.e…⁴ stati…⁵ p.valu…⁶\n##    \u0026lt;chr\u0026gt;                       \u0026lt;list\u0026gt;     \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;    \u0026lt;dbl\u0026gt;\n##  1 cibersort.B.cells.naive     \u0026lt;tibble\u0026gt;   -2.94   2.25    0.367   6.13  8.77e-10\n##  2 cibersort.B.cells.memory    \u0026lt;tibble\u0026gt;   -4.86   1.48    0.436   3.40  6.77e- 4\n##  3 cibersort.Plasma.cells      \u0026lt;tibble\u0026gt;   -5.33  -0.487   0.507  -0.960 3.37e- 1\n##  4 cibersort.T.cells.CD8       \u0026lt;tibble\u0026gt;   -2.33   0.924   0.475   1.94  5.18e- 2\n##  5 cibersort.T.cells.CD4.naive \u0026lt;tibble\u0026gt;   -2.83  -0.620   0.531  -1.17  2.43e- 1\n##  6 cibersort.T.cells.CD4.memo… \u0026lt;tibble\u0026gt;   -2.46   0.190   0.500   0.380 7.04e- 1\n##  7 cibersort.T.cells.CD4.memo… \u0026lt;tibble\u0026gt;   -3.67   2.23    0.427   5.22  1.80e- 7\n##  8 cibersort.T.cells.follicul… \u0026lt;tibble\u0026gt;   -5.68  -0.217   0.507  -0.427 6.69e- 1\n##  9 cibersort.T.cells.regulato… \u0026lt;tibble\u0026gt;   -5.04   1.94    0.360   5.39  6.86e- 8\n## 10 cibersort.T.cells.gamma.de… \u0026lt;tibble\u0026gt;   -4.78  -0.250   0.514  -0.486 6.27e- 1\n## # … with 12 more rows, and abbreviated variable names ¹​cell_type_proportions,\n## #   ²​`estimate_(Intercept)`, ³​estimate_conditionTRUE, ⁴​std.error_conditionTRUE,\n## #   ⁵​statistic_conditionTRUE, ⁶​p.value_conditionTRUE\n## # ℹ Use `print(n = ...)` to see more rows\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## \n## The downloaded binary packages are in\n##  /var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T//RtmpIi5KN6/downloaded_packages\n\n## # A tibble: 22 × 7\n##    .cell_type                  cell_t…¹ estim…² estim…³ std.e…⁴ stati…⁵ p.valu…⁶\n##    \u0026lt;chr\u0026gt;                       \u0026lt;list\u0026gt;     \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;    \u0026lt;dbl\u0026gt;\n##  1 cibersort.B.cells.naive     \u0026lt;tibble\u0026gt;   -2.94   2.25    0.367   6.13  8.77e-10\n##  2 cibersort.B.cells.memory    \u0026lt;tibble\u0026gt;   -4.86   1.48    0.436   3.40  6.77e- 4\n##  3 cibersort.Plasma.cells      \u0026lt;tibble\u0026gt;   -5.33  -0.487   0.507  -0.960 3.37e- 1\n##  4 cibersort.T.cells.CD8       \u0026lt;tibble\u0026gt;   -2.33   0.924   0.475   1.94  5.18e- 2\n##  5 cibersort.T.cells.CD4.naive \u0026lt;tibble\u0026gt;   -2.83  -0.620   0.531  -1.17  2.43e- 1\n##  6 cibersort.T.cells.CD4.memo… \u0026lt;tibble\u0026gt;   -2.46   0.190   0.500   0.380 7.04e- 1\n##  7 cibersort.T.cells.CD4.memo… \u0026lt;tibble\u0026gt;   -3.67   2.23    0.427   5.22  1.80e- 7\n##  8 cibersort.T.cells.follicul… \u0026lt;tibble\u0026gt;   -5.68  -0.217   0.507  -0.427 6.69e- 1\n##  9 cibersort.T.cells.regulato… \u0026lt;tibble\u0026gt;   -5.04   1.94    0.360   5.39  6.86e- 8\n## 10 cibersort.T.cells.gamma.de… \u0026lt;tibble\u0026gt;   -4.78  -0.250   0.514  -0.486 6.27e- 1\n## # … with 12 more rows, and abbreviated variable names ¹​cell_type_proportions,\n## #   ²​`estimate_(Intercept)`, ³​estimate_conditionTRUE, ⁴​std.error_conditionTRUE,\n## #   ⁵​statistic_conditionTRUE, ⁶​p.value_conditionTRUE\n## # ℹ Use `print(n = ...)` to see more rows\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe can also perform regression analysis with censored data (coxph).\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"    # Add survival data\n\ncounts_SE_survival = \n    counts_SE %\u0026gt;%\n    nest(data = -sample) %\u0026gt;%\n        mutate(\n            days = sample(1:1000, size = n()),\n            dead = sample(c(0,1), size = n(), replace = TRUE)\n        ) %\u0026gt;%\n    unnest(data) \n\n# Test\ncounts_SE_survival %\u0026gt;%\n    test_differential_cellularity(survival::Surv(days, dead) ~ .)\"\u003e\u003cpre\u003e    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Add survival data\u003c/span\u003e\n\n\u003cspan class=\"pl-v\"\u003ecounts_SE_survival\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \n    \u003cspan class=\"pl-smi\"\u003ecounts_SE\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    nest(\u003cspan class=\"pl-v\"\u003edata\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n        mutate(\n            \u003cspan class=\"pl-v\"\u003edays\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e sample(\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e1000\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e n()),\n            \u003cspan class=\"pl-v\"\u003edead\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e sample(c(\u003cspan class=\"pl-c1\"\u003e0\u003c/span\u003e,\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e), \u003cspan class=\"pl-v\"\u003esize\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e n(), \u003cspan class=\"pl-v\"\u003ereplace\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e)\n        ) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    unnest(\u003cspan class=\"pl-smi\"\u003edata\u003c/span\u003e) \n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Test\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ecounts_SE_survival\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    test_differential_cellularity(\u003cspan class=\"pl-e\"\u003esurvival\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eSurv(\u003cspan class=\"pl-smi\"\u003edays\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003edead\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## # A tibble: 22 × 6\n##    .cell_type                           cell_t…¹ estim…² std.e…³ stati…⁴ p.value\n##    \u0026lt;chr\u0026gt;                                \u0026lt;list\u0026gt;     \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;\n##  1 cibersort.B.cells.naive              \u0026lt;tibble\u0026gt;  -0.224   0.415  -0.540  0.589 \n##  2 cibersort.B.cells.memory             \u0026lt;tibble\u0026gt;   0.510   0.346   1.48   0.140 \n##  3 cibersort.Plasma.cells               \u0026lt;tibble\u0026gt;   0.892   0.449   1.99   0.0467\n##  4 cibersort.T.cells.CD8                \u0026lt;tibble\u0026gt;   0.531   0.639   0.831  0.406 \n##  5 cibersort.T.cells.CD4.naive          \u0026lt;tibble\u0026gt;   0.112   0.386   0.290  0.772 \n##  6 cibersort.T.cells.CD4.memory.resting \u0026lt;tibble\u0026gt;   0.498   0.540   0.921  0.357 \n##  7 cibersort.T.cells.CD4.memory.activa… \u0026lt;tibble\u0026gt;   2.37    0.939   2.52   0.0117\n##  8 cibersort.T.cells.follicular.helper  \u0026lt;tibble\u0026gt;  -0.544   0.421  -1.29   0.197 \n##  9 cibersort.T.cells.regulatory..Tregs. \u0026lt;tibble\u0026gt;   1.59    0.656   2.42   0.0157\n## 10 cibersort.T.cells.gamma.delta        \u0026lt;tibble\u0026gt;   0.510   0.688   0.741  0.459 \n## # … with 12 more rows, and abbreviated variable names ¹​cell_type_proportions,\n## #   ²​estimate, ³​std.error, ⁴​statistic\n## # ℹ Use `print(n = ...)` to see more rows\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## # A tibble: 22 × 6\n##    .cell_type                           cell_t…¹ estim…² std.e…³ stati…⁴ p.value\n##    \u0026lt;chr\u0026gt;                                \u0026lt;list\u0026gt;     \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;\n##  1 cibersort.B.cells.naive              \u0026lt;tibble\u0026gt;  -0.224   0.415  -0.540  0.589 \n##  2 cibersort.B.cells.memory             \u0026lt;tibble\u0026gt;   0.510   0.346   1.48   0.140 \n##  3 cibersort.Plasma.cells               \u0026lt;tibble\u0026gt;   0.892   0.449   1.99   0.0467\n##  4 cibersort.T.cells.CD8                \u0026lt;tibble\u0026gt;   0.531   0.639   0.831  0.406 \n##  5 cibersort.T.cells.CD4.naive          \u0026lt;tibble\u0026gt;   0.112   0.386   0.290  0.772 \n##  6 cibersort.T.cells.CD4.memory.resting \u0026lt;tibble\u0026gt;   0.498   0.540   0.921  0.357 \n##  7 cibersort.T.cells.CD4.memory.activa… \u0026lt;tibble\u0026gt;   2.37    0.939   2.52   0.0117\n##  8 cibersort.T.cells.follicular.helper  \u0026lt;tibble\u0026gt;  -0.544   0.421  -1.29   0.197 \n##  9 cibersort.T.cells.regulatory..Tregs. \u0026lt;tibble\u0026gt;   1.59    0.656   2.42   0.0157\n## 10 cibersort.T.cells.gamma.delta        \u0026lt;tibble\u0026gt;   0.510   0.688   0.741  0.459 \n## # … with 12 more rows, and abbreviated variable names ¹​cell_type_proportions,\n## #   ²​estimate, ³​std.error, ⁴​statistic\n## # ℹ Use `print(n = ...)` to see more rows\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe can also perform test of Kaplan-Meier curves.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_stratified = \n    counts_SE_survival %\u0026gt;%\n\n    # Test\n    test_stratification_cellularity(\n        survival::Surv(days, dead) ~ .,\n        sample, transcript, count\n    )\n\ncounts_stratified\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_stratified\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \n    \u003cspan class=\"pl-smi\"\u003ecounts_SE_survival\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n\n    \u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Test\u003c/span\u003e\n    test_stratification_cellularity(\n        \u003cspan class=\"pl-e\"\u003esurvival\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e::\u003c/span\u003eSurv(\u003cspan class=\"pl-smi\"\u003edays\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003edead\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e.\u003c/span\u003e,\n        \u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003etranscript\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003ecount\u003c/span\u003e\n    )\n\n\u003cspan class=\"pl-smi\"\u003ecounts_stratified\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## # A tibble: 22 × 6\n##    .cell_type                         cell_t…¹ .low_…² .high…³ pvalue plot      \n##    \u0026lt;chr\u0026gt;                              \u0026lt;list\u0026gt;     \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;  \u0026lt;dbl\u0026gt; \u0026lt;list\u0026gt;    \n##  1 cibersort.B.cells.naive            \u0026lt;tibble\u0026gt;   14.4     7.56  0.506 \u0026lt;ggsrvplt\u0026gt;\n##  2 cibersort.B.cells.memory           \u0026lt;tibble\u0026gt;   17.2     4.77  0.500 \u0026lt;ggsrvplt\u0026gt;\n##  3 cibersort.Plasma.cells             \u0026lt;tibble\u0026gt;   13.3     8.73  0.903 \u0026lt;ggsrvplt\u0026gt;\n##  4 cibersort.T.cells.CD8              \u0026lt;tibble\u0026gt;   13.9     8.06  0.369 \u0026lt;ggsrvplt\u0026gt;\n##  5 cibersort.T.cells.CD4.naive        \u0026lt;tibble\u0026gt;   12.8     9.15  0.407 \u0026lt;ggsrvplt\u0026gt;\n##  6 cibersort.T.cells.CD4.memory.rest… \u0026lt;tibble\u0026gt;    7.65   14.4   0.105 \u0026lt;ggsrvplt\u0026gt;\n##  7 cibersort.T.cells.CD4.memory.acti… \u0026lt;tibble\u0026gt;   15.7     6.26  0.392 \u0026lt;ggsrvplt\u0026gt;\n##  8 cibersort.T.cells.follicular.help… \u0026lt;tibble\u0026gt;   17.1     4.88  0.949 \u0026lt;ggsrvplt\u0026gt;\n##  9 cibersort.T.cells.regulatory..Tre… \u0026lt;tibble\u0026gt;   13.7     8.35  0.771 \u0026lt;ggsrvplt\u0026gt;\n## 10 cibersort.T.cells.gamma.delta      \u0026lt;tibble\u0026gt;   16.2     5.76  0.379 \u0026lt;ggsrvplt\u0026gt;\n## # … with 12 more rows, and abbreviated variable names ¹​cell_type_proportions,\n## #   ²​.low_cellularity_expected, ³​.high_cellularity_expected\n## # ℹ Use `print(n = ...)` to see more rows\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## # A tibble: 22 × 6\n##    .cell_type                         cell_t…¹ .low_…² .high…³ pvalue plot      \n##    \u0026lt;chr\u0026gt;                              \u0026lt;list\u0026gt;     \u0026lt;dbl\u0026gt;   \u0026lt;dbl\u0026gt;  \u0026lt;dbl\u0026gt; \u0026lt;list\u0026gt;    \n##  1 cibersort.B.cells.naive            \u0026lt;tibble\u0026gt;   14.4     7.56  0.506 \u0026lt;ggsrvplt\u0026gt;\n##  2 cibersort.B.cells.memory           \u0026lt;tibble\u0026gt;   17.2     4.77  0.500 \u0026lt;ggsrvplt\u0026gt;\n##  3 cibersort.Plasma.cells             \u0026lt;tibble\u0026gt;   13.3     8.73  0.903 \u0026lt;ggsrvplt\u0026gt;\n##  4 cibersort.T.cells.CD8              \u0026lt;tibble\u0026gt;   13.9     8.06  0.369 \u0026lt;ggsrvplt\u0026gt;\n##  5 cibersort.T.cells.CD4.naive        \u0026lt;tibble\u0026gt;   12.8     9.15  0.407 \u0026lt;ggsrvplt\u0026gt;\n##  6 cibersort.T.cells.CD4.memory.rest… \u0026lt;tibble\u0026gt;    7.65   14.4   0.105 \u0026lt;ggsrvplt\u0026gt;\n##  7 cibersort.T.cells.CD4.memory.acti… \u0026lt;tibble\u0026gt;   15.7     6.26  0.392 \u0026lt;ggsrvplt\u0026gt;\n##  8 cibersort.T.cells.follicular.help… \u0026lt;tibble\u0026gt;   17.1     4.88  0.949 \u0026lt;ggsrvplt\u0026gt;\n##  9 cibersort.T.cells.regulatory..Tre… \u0026lt;tibble\u0026gt;   13.7     8.35  0.771 \u0026lt;ggsrvplt\u0026gt;\n## 10 cibersort.T.cells.gamma.delta      \u0026lt;tibble\u0026gt;   16.2     5.76  0.379 \u0026lt;ggsrvplt\u0026gt;\n## # … with 12 more rows, and abbreviated variable names ¹​cell_type_proportions,\n## #   ²​.low_cellularity_expected, ³​.high_cellularity_expected\n## # ℹ Use `print(n = ...)` to see more rows\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePlot Kaplan-Meier curves\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_stratified$plot[[1]]\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_stratified\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eplot\u003c/span\u003e[[\u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e]]\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/stemangiola/tidybulk/blob/master/man/figures/unnamed-chunk-19-1.png\"\u003e\u003cimg src=\"/stemangiola/tidybulk/raw/master/man/figures/unnamed-chunk-19-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eCluster \u003ccode\u003esamples\u003c/code\u003e\u003c/h2\u003e\u003ca id=\"user-content-cluster-samples\" class=\"anchor\" aria-label=\"Permalink: Cluster samples\" href=\"#cluster-samples\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe may want to cluster our data (e.g., using k-means sample-wise).\n\u003ccode\u003ecluster_elements\u003c/code\u003e takes as arguments a tibble, column names (as\nsymbols; for \u003ccode\u003esample\u003c/code\u003e, \u003ccode\u003etranscript\u003c/code\u003e and \u003ccode\u003ecount\u003c/code\u003e) and returns a tibble\nwith additional columns for the cluster annotation. At the moment only\nk-means clustering is supported, the plan is to introduce more\nclustering methods.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003ek-means\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eTidyTranscriptomics\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.cluster = counts_SE.norm.MDS %\u0026gt;%\n  cluster_elements(method=\u0026quot;kmeans\u0026quot;, centers = 2, action=\u0026quot;get\u0026quot; )\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_SE.norm.cluster\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.MDS\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  cluster_elements(\u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ekmeans\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecenters\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e2\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eaction\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eget\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eStandard procedure (comparative purpose)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"count_m_log = log(count_m + 1)\n\nk = kmeans(count_m_log, iter.max = 1000, ...)\ncluster = k$cluster\n\ncluster$cell_type = tibble_counts[\n    match(tibble_counts$sample, rownames(cluster)),\n    c(\u0026quot;Cell.type\u0026quot;, \u0026quot;Dim1\u0026quot;, \u0026quot;Dim2\u0026quot;)\n]\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecount_m_log\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e log(\u003cspan class=\"pl-smi\"\u003ecount_m\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003ek\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e kmeans(\u003cspan class=\"pl-smi\"\u003ecount_m_log\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eiter.max\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e1000\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e...\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003ecluster\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ek\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e\n\n\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-v\"\u003ecell_type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etibble_counts\u003c/span\u003e[\n    match(\u003cspan class=\"pl-smi\"\u003etibble_counts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e, rownames(\u003cspan class=\"pl-smi\"\u003ecluster\u003c/span\u003e)),\n    c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCell.type\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDim1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDim2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n]\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe can add cluster annotation to the MDS dimension reduced data set and\nplot.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\" counts_SE.norm.cluster %\u0026gt;%\n    ggplot(aes(x=`Dim1`, y=`Dim2`, color=`cluster_kmeans`)) +\n  geom_point() +\n  my_theme\"\u003e\u003cpre\u003e \u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.cluster\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Dim1`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Dim2`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`cluster_kmeans`\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_point() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \u003cspan class=\"pl-smi\"\u003emy_theme\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/stemangiola/tidybulk/blob/master/man/figures/plot_cluster-1.png\"\u003e\u003cimg src=\"/stemangiola/tidybulk/raw/master/man/figures/plot_cluster-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSNN\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMatrix package (v1.3-3) causes an error with Seurat::FindNeighbors used\nin this method. We are trying to solve this issue. At the moment this\noption in unaviable.\u003c/p\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eTidyTranscriptomics\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.SNN =\n    counts_SE.norm.tSNE %\u0026gt;%\n    cluster_elements(method = \u0026quot;SNN\u0026quot;)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_SE.norm.SNN\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\n    \u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.tSNE\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    cluster_elements(\u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eSNN\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eStandard procedure (comparative purpose)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(Seurat)\n\nsnn = CreateSeuratObject(count_m)\nsnn = ScaleData(\n    snn, display.progress = TRUE,\n    num.cores=4, do.par = TRUE\n)\nsnn = FindVariableFeatures(snn, selection.method = \u0026quot;vst\u0026quot;)\nsnn = FindVariableFeatures(snn, selection.method = \u0026quot;vst\u0026quot;)\nsnn = RunPCA(snn, npcs = 30)\nsnn = FindNeighbors(snn)\nsnn = FindClusters(snn, method = \u0026quot;igraph\u0026quot;, ...)\nsnn = snn[[\u0026quot;seurat_clusters\u0026quot;]]\n\nsnn$cell_type = tibble_counts[\n    match(tibble_counts$sample, rownames(snn)),\n    c(\u0026quot;Cell.type\u0026quot;, \u0026quot;Dim1\u0026quot;, \u0026quot;Dim2\u0026quot;)\n]\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003eSeurat\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003esnn\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e CreateSeuratObject(\u003cspan class=\"pl-smi\"\u003ecount_m\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003esnn\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e ScaleData(\n    \u003cspan class=\"pl-smi\"\u003esnn\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edisplay.progress\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003enum.cores\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-c1\"\u003e4\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003edo.par\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e\n)\n\u003cspan class=\"pl-v\"\u003esnn\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e FindVariableFeatures(\u003cspan class=\"pl-smi\"\u003esnn\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eselection.method\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003esnn\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e FindVariableFeatures(\u003cspan class=\"pl-smi\"\u003esnn\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003eselection.method\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003evst\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003esnn\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e RunPCA(\u003cspan class=\"pl-smi\"\u003esnn\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003enpcs\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e30\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003esnn\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e FindNeighbors(\u003cspan class=\"pl-smi\"\u003esnn\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003esnn\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e FindClusters(\u003cspan class=\"pl-smi\"\u003esnn\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eigraph\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e...\u003c/span\u003e)\n\u003cspan class=\"pl-v\"\u003esnn\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003esnn\u003c/span\u003e[[\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eseurat_clusters\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e]]\n\n\u003cspan class=\"pl-smi\"\u003esnn\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-v\"\u003ecell_type\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etibble_counts\u003c/span\u003e[\n    match(\u003cspan class=\"pl-smi\"\u003etibble_counts\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e$\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e, rownames(\u003cspan class=\"pl-smi\"\u003esnn\u003c/span\u003e)),\n    c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCell.type\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDim1\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eDim2\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)\n]\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003c/div\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.SNN %\u0026gt;%\n    pivot_sample() %\u0026gt;%\n    select(contains(\u0026quot;tSNE\u0026quot;), everything()) \n\ncounts_SE.norm.SNN %\u0026gt;%\n    pivot_sample() %\u0026gt;%\n    gather(source, Call, c(\u0026quot;cluster_SNN\u0026quot;, \u0026quot;Call\u0026quot;)) %\u0026gt;%\n    distinct() %\u0026gt;%\n    ggplot(aes(x = `tSNE1`, y = `tSNE2`, color=Call)) + geom_point() + facet_grid(~source) + my_theme\n\n\n# Do differential transcription between clusters\ncounts_SE.norm.SNN %\u0026gt;%\n    mutate(factor_of_interest = `cluster_SNN` == 3) %\u0026gt;%\n    test_differential_abundance(\n    ~ factor_of_interest,\n    action=\u0026quot;get\u0026quot;\n   )\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.SNN\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    pivot_sample() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    select(contains(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003etSNE\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e), everything()) \n\n\u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.SNN\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    pivot_sample() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    gather(\u003cspan class=\"pl-smi\"\u003esource\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003eCall\u003c/span\u003e, c(\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecluster_SNN\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e, \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eCall\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e)) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    distinct() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e`tSNE1`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e`tSNE2`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003eCall\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e geom_point() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e facet_grid(\u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003esource\u003c/span\u003e) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003emy_theme\u003c/span\u003e\n\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Do differential transcription between clusters\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.SNN\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    mutate(\u003cspan class=\"pl-v\"\u003efactor_of_interest\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e`cluster_SNN`\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e==\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003e3\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    test_differential_abundance(\n    \u003cspan class=\"pl-k\"\u003e~\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003efactor_of_interest\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eaction\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003eget\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e\n   )\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eDrop \u003ccode\u003eredundant\u003c/code\u003e transcripts\u003c/h2\u003e\u003ca id=\"user-content-drop-redundant-transcripts\" class=\"anchor\" aria-label=\"Permalink: Drop redundant transcripts\" href=\"#drop-redundant-transcripts\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe may want to remove redundant elements from the original data set\n(e.g., samples or transcripts), for example if we want to define\ncell-type specific signatures with low sample redundancy.\n\u003ccode\u003eremove_redundancy\u003c/code\u003e takes as arguments a tibble, column names (as\nsymbols; for \u003ccode\u003esample\u003c/code\u003e, \u003ccode\u003etranscript\u003c/code\u003e and \u003ccode\u003ecount\u003c/code\u003e) and returns a tibble\nwith redundant elements removed (e.g., samples). Two redundancy\nestimation approaches are supported:\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eremoval of highly correlated clusters of elements (keeping a\nrepresentative) with method=“correlation”\u003c/li\u003e\n\u003cli\u003eremoval of most proximal element pairs in a reduced dimensional\nspace.\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eApproach 1\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eTidyTranscriptomics\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.non_redundant =\n    counts_SE.norm.MDS %\u0026gt;%\n  remove_redundancy(    method = \u0026quot;correlation\u0026quot; )\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_SE.norm.non_redundant\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\n    \u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.MDS\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  remove_redundancy(    \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ecorrelation\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e )\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## \n## The downloaded binary packages are in\n##  /var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T//RtmpIi5KN6/downloaded_packages\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## \n## The downloaded binary packages are in\n##  /var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T//RtmpIi5KN6/downloaded_packages\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003cp dir=\"auto\"\u003eStandard procedure (comparative purpose)\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"library(widyr)\n\n.data.correlated =\n    pairwise_cor(\n        counts,\n        sample,\n        transcript,\n        rc,\n        sort = TRUE,\n        diag = FALSE,\n        upper = FALSE\n    ) %\u0026gt;%\n    filter(correlation \u0026gt; correlation_threshold) %\u0026gt;%\n    distinct(item1) %\u0026gt;%\n    rename(!!.element := item1)\n\n# Return non redudant data frame\ncounts %\u0026gt;% anti_join(.data.correlated) %\u0026gt;%\n    spread(sample, rc, - transcript) %\u0026gt;%\n    left_join(annotation)\"\u003e\u003cpre\u003elibrary(\u003cspan class=\"pl-smi\"\u003ewidyr\u003c/span\u003e)\n\n\u003cspan class=\"pl-v\"\u003e.data.correlated\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\n    pairwise_cor(\n        \u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e,\n        \u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e,\n        \u003cspan class=\"pl-smi\"\u003etranscript\u003c/span\u003e,\n        \u003cspan class=\"pl-smi\"\u003erc\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003esort\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003ediag\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e,\n        \u003cspan class=\"pl-v\"\u003eupper\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e\n    ) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    filter(\u003cspan class=\"pl-smi\"\u003ecorrelation\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003ecorrelation_threshold\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    distinct(\u003cspan class=\"pl-smi\"\u003eitem1\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    rename(\u003cspan class=\"pl-k\"\u003e!!\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e.element\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e:\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003eitem1\u003c/span\u003e)\n\n\u003cspan class=\"pl-c\"\u003e\u003cspan class=\"pl-c\"\u003e#\u003c/span\u003e Return non redudant data frame\u003c/span\u003e\n\u003cspan class=\"pl-smi\"\u003ecounts\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% anti_join(\u003cspan class=\"pl-smi\"\u003e.data.correlated\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    spread(\u003cspan class=\"pl-smi\"\u003esample\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003erc\u003c/span\u003e, \u003cspan class=\"pl-k\"\u003e-\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003etranscript\u003c/span\u003e) %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    left_join(\u003cspan class=\"pl-smi\"\u003eannotation\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003c/div\u003e\n\u003cdiv dir=\"auto\"\u003e\n\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe can visualise how the reduced redundancy with the reduced dimentions\nlook like\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.non_redundant %\u0026gt;%\n    pivot_sample() %\u0026gt;%\n    ggplot(aes(x=`Dim1`, y=`Dim2`, color=`Cell.type`)) +\n  geom_point() +\n  my_theme\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.non_redundant\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    pivot_sample() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Dim1`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Dim2`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Cell.type`\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_point() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \u003cspan class=\"pl-smi\"\u003emy_theme\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/stemangiola/tidybulk/blob/master/man/figures/plot_drop-1.png\"\u003e\u003cimg src=\"/stemangiola/tidybulk/raw/master/man/figures/plot_drop-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eApproach 2\u003c/strong\u003e\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.non_redundant =\n    counts_SE.norm.MDS %\u0026gt;%\n  remove_redundancy(\n    method = \u0026quot;reduced_dimensions\u0026quot;,\n    Dim_a_column = `Dim1`,\n    Dim_b_column = `Dim2`\n  )\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts_SE.norm.non_redundant\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\n    \u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.MDS\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n  remove_redundancy(\n    \u003cspan class=\"pl-v\"\u003emethod\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ereduced_dimensions\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eDim_a_column\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e`Dim1`\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eDim_b_column\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-smi\"\u003e`Dim2`\u003c/span\u003e\n  )\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe can visualise MDS reduced dimensions of the samples with the closest\npair removed.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE.norm.non_redundant %\u0026gt;%\n    pivot_sample() %\u0026gt;%\n    ggplot(aes(x=`Dim1`, y=`Dim2`, color=`Cell.type`)) +\n  geom_point() +\n  my_theme\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE.norm.non_redundant\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    pivot_sample() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e%\n    ggplot(aes(\u003cspan class=\"pl-v\"\u003ex\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Dim1`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ey\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Dim2`\u003c/span\u003e, \u003cspan class=\"pl-v\"\u003ecolor\u003c/span\u003e\u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e\u003cspan class=\"pl-smi\"\u003e`Cell.type`\u003c/span\u003e)) \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  geom_point() \u003cspan class=\"pl-k\"\u003e+\u003c/span\u003e\n  \u003cspan class=\"pl-smi\"\u003emy_theme\u003c/span\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003ca target=\"_blank\" rel=\"noopener noreferrer\" href=\"/stemangiola/tidybulk/blob/master/man/figures/plot_drop2-1.png\"\u003e\u003cimg src=\"/stemangiola/tidybulk/raw/master/man/figures/plot_drop2-1.png\" alt=\"\" style=\"max-width: 100%;\"\u003e\u003c/a\u003e\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eOther useful wrappers\u003c/h2\u003e\u003ca id=\"user-content-other-useful-wrappers\" class=\"anchor\" aria-label=\"Permalink: Other useful wrappers\" href=\"#other-useful-wrappers\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eThe above wrapper streamline the most common processing of bulk RNA\nsequencing data. Other useful wrappers are listed above.\u003c/p\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFrom BAM/SAM to tibble of gene counts\u003c/h2\u003e\u003ca id=\"user-content-from-bamsam-to-tibble-of-gene-counts\" class=\"anchor\" aria-label=\"Permalink: From BAM/SAM to tibble of gene counts\" href=\"#from-bamsam-to-tibble-of-gene-counts\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe can calculate gene counts (using FeatureCounts; Liao Y et al.,\n10.1093/nar/gkz114) from a list of BAM/SAM files and format them into a\ntidy structure (similar to counts).\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts = tidybulk_SAM_BAM(\n    file_names,\n    genome = \u0026quot;hg38\u0026quot;,\n    isPairedEnd = TRUE,\n    requireBothEndsMapped = TRUE,\n    checkFragLength = FALSE,\n    useMetaFeatures = TRUE\n)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-v\"\u003ecounts\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e tidybulk_SAM_BAM(\n    \u003cspan class=\"pl-smi\"\u003efile_names\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003egenome\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-s\"\u003e\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003ehg38\u003cspan class=\"pl-pds\"\u003e\"\u003c/span\u003e\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003eisPairedEnd\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003erequireBothEndsMapped\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003echeckFragLength\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eFALSE\u003c/span\u003e,\n    \u003cspan class=\"pl-v\"\u003euseMetaFeatures\u003c/span\u003e \u003cspan class=\"pl-k\"\u003e=\u003c/span\u003e \u003cspan class=\"pl-c1\"\u003eTRUE\u003c/span\u003e\n)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFrom ensembl IDs to gene symbol IDs\u003c/h2\u003e\u003ca id=\"user-content-from-ensembl-ids-to-gene-symbol-ids\" class=\"anchor\" aria-label=\"Permalink: From ensembl IDs to gene symbol IDs\" href=\"#from-ensembl-ids-to-gene-symbol-ids\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe can add gene symbols from ensembl identifiers. This is useful since\ndifferent resources use ensembl IDs while others use gene symbol IDs.\nThis currently works for human and mouse.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_ensembl %\u0026gt;% ensembl_to_symbol(ens)\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_ensembl\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% ensembl_to_symbol(\u003cspan class=\"pl-smi\"\u003eens\u003c/span\u003e)\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## # A tibble: 119 × 8\n##    ens             iso   `read count` sample     cases…¹ cases…² trans…³ ref_g…⁴\n##    \u0026lt;chr\u0026gt;           \u0026lt;chr\u0026gt;        \u0026lt;dbl\u0026gt; \u0026lt;chr\u0026gt;      \u0026lt;chr\u0026gt;   \u0026lt;chr\u0026gt;   \u0026lt;chr\u0026gt;   \u0026lt;chr\u0026gt;  \n##  1 ENSG00000000003 13             144 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  2 ENSG00000000003 13              72 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  3 ENSG00000000003 13               0 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  4 ENSG00000000003 13            1099 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  5 ENSG00000000003 13              11 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  6 ENSG00000000003 13               2 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  7 ENSG00000000003 13               3 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  8 ENSG00000000003 13            2678 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  9 ENSG00000000003 13             751 TARGET-20… Acute … Primar… TSPAN6  hg38   \n## 10 ENSG00000000003 13               1 TARGET-20… Acute … Primar… TSPAN6  hg38   \n## # … with 109 more rows, and abbreviated variable names\n## #   ¹​cases_0_project_disease_type, ²​cases_0_samples_0_sample_type, ³​transcript,\n## #   ⁴​ref_genome\n## # ℹ Use `print(n = ...)` to see more rows\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## # A tibble: 119 × 8\n##    ens             iso   `read count` sample     cases…¹ cases…² trans…³ ref_g…⁴\n##    \u0026lt;chr\u0026gt;           \u0026lt;chr\u0026gt;        \u0026lt;dbl\u0026gt; \u0026lt;chr\u0026gt;      \u0026lt;chr\u0026gt;   \u0026lt;chr\u0026gt;   \u0026lt;chr\u0026gt;   \u0026lt;chr\u0026gt;  \n##  1 ENSG00000000003 13             144 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  2 ENSG00000000003 13              72 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  3 ENSG00000000003 13               0 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  4 ENSG00000000003 13            1099 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  5 ENSG00000000003 13              11 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  6 ENSG00000000003 13               2 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  7 ENSG00000000003 13               3 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  8 ENSG00000000003 13            2678 TARGET-20… Acute … Primar… TSPAN6  hg38   \n##  9 ENSG00000000003 13             751 TARGET-20… Acute … Primar… TSPAN6  hg38   \n## 10 ENSG00000000003 13               1 TARGET-20… Acute … Primar… TSPAN6  hg38   \n## # … with 109 more rows, and abbreviated variable names\n## #   ¹​cases_0_project_disease_type, ²​cases_0_samples_0_sample_type, ³​transcript,\n## #   ⁴​ref_genome\n## # ℹ Use `print(n = ...)` to see more rows\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"markdown-heading\" dir=\"auto\"\u003e\u003ch2 tabindex=\"-1\" class=\"heading-element\" dir=\"auto\"\u003eFrom gene symbol to gene description (gene name in full)\u003c/h2\u003e\u003ca id=\"user-content-from-gene-symbol-to-gene-description-gene-name-in-full\" class=\"anchor\" aria-label=\"Permalink: From gene symbol to gene description (gene name in full)\" href=\"#from-gene-symbol-to-gene-description-gene-name-in-full\"\u003e\u003csvg class=\"octicon octicon-link\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" height=\"16\" aria-hidden=\"true\"\u003e\u003cpath d=\"m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z\"\u003e\u003c/path\u003e\u003c/svg\u003e\u003c/a\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eWe can add gene full name (and in future description) from symbol\nidentifiers. This currently works for human and mouse.\u003c/p\u003e\n\u003cdiv class=\"highlight highlight-source-r notranslate position-relative overflow-auto\" dir=\"auto\" data-snippet-clipboard-copy-content=\"counts_SE %\u0026gt;% \n    describe_transcript() %\u0026gt;% \n    select(feature, description, everything())\"\u003e\u003cpre\u003e\u003cspan class=\"pl-smi\"\u003ecounts_SE\u003c/span\u003e %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n    describe_transcript() %\u003cspan class=\"pl-k\"\u003e\u0026gt;\u003c/span\u003e% \n    select(\u003cspan class=\"pl-smi\"\u003efeature\u003c/span\u003e, \u003cspan class=\"pl-smi\"\u003edescription\u003c/span\u003e, everything())\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"## # A SummarizedExperiment-tibble abstraction: 408,624 × 48\n## # �[90mFeatures=8513 | Samples=48 | Assays=count�[0m\n##    feature  sample     count Cell.…¹ time  condi…² batch facto…³ descr…⁴ gene_…⁵\n##    \u0026lt;chr\u0026gt;    \u0026lt;chr\u0026gt;      \u0026lt;dbl\u0026gt; \u0026lt;fct\u0026gt;   \u0026lt;fct\u0026gt; \u0026lt;lgl\u0026gt;   \u0026lt;fct\u0026gt; \u0026lt;lgl\u0026gt;   \u0026lt;chr\u0026gt;   \u0026lt;chr\u0026gt;  \n##  1 A1BG     SRR1740034   153 b_cell  0 d   TRUE    0     TRUE    alpha-… A1BG   \n##  2 A1BG-AS1 SRR1740034    83 b_cell  0 d   TRUE    0     TRUE    A1BG a… A1BG-A…\n##  3 AAAS     SRR1740034   868 b_cell  0 d   TRUE    0     TRUE    aladin… AAAS   \n##  4 AACS     SRR1740034   222 b_cell  0 d   TRUE    0     TRUE    acetoa… AACS   \n##  5 AAGAB    SRR1740034   590 b_cell  0 d   TRUE    0     TRUE    alpha … AAGAB  \n##  6 AAMDC    SRR1740034    48 b_cell  0 d   TRUE    0     TRUE    adipog… AAMDC  \n##  7 AAMP     SRR1740034  1257 b_cell  0 d   TRUE    0     TRUE    angio … AAMP   \n##  8 AANAT    SRR1740034   284 b_cell  0 d   TRUE    0     TRUE    aralky… AANAT  \n##  9 AAR2     SRR1740034   379 b_cell  0 d   TRUE    0     TRUE    AAR2 s… AAR2   \n## 10 AARS2    SRR1740034   685 b_cell  0 d   TRUE    0     TRUE    alanyl… AARS2  \n## # … with 40 more rows, and abbreviated variable names ¹​Cell.type, ²​condition,\n## #   ³​factor_of_interest, ⁴​description, ⁵​gene_name\n## # ℹ Use `print(n = ...)` to see more rows\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e## # A SummarizedExperiment-tibble abstraction: 408,624 × 48\n## # �[90mFeatures=8513 | Samples=48 | Assays=count�[0m\n##    feature  sample     count Cell.…¹ time  condi…² batch facto…³ descr…⁴ gene_…⁵\n##    \u0026lt;chr\u0026gt;    \u0026lt;chr\u0026gt;      \u0026lt;dbl\u0026gt; \u0026lt;fct\u0026gt;   \u0026lt;fct\u0026gt; \u0026lt;lgl\u0026gt;   \u0026lt;fct\u0026gt; \u0026lt;lgl\u0026gt;   \u0026lt;chr\u0026gt;   \u0026lt;chr\u0026gt;  \n##  1 A1BG     SRR1740034   153 b_cell  0 d   TRUE    0     TRUE    alpha-… A1BG   \n##  2 A1BG-AS1 SRR1740034    83 b_cell  0 d   TRUE    0     TRUE    A1BG a… A1BG-A…\n##  3 AAAS     SRR1740034   868 b_cell  0 d   TRUE    0     TRUE    aladin… AAAS   \n##  4 AACS     SRR1740034   222 b_cell  0 d   TRUE    0     TRUE    acetoa… AACS   \n##  5 AAGAB    SRR1740034   590 b_cell  0 d   TRUE    0     TRUE    alpha … AAGAB  \n##  6 AAMDC    SRR1740034    48 b_cell  0 d   TRUE    0     TRUE    adipog… AAMDC  \n##  7 AAMP     SRR1740034  1257 b_cell  0 d   TRUE    0     TRUE    angio … AAMP   \n##  8 AANAT    SRR1740034   284 b_cell  0 d   TRUE    0     TRUE    aralky… AANAT  \n##  9 AAR2     SRR1740034   379 b_cell  0 d   TRUE    0     TRUE    AAR2 s… AAR2   \n## 10 AARS2    SRR1740034   685 b_cell  0 d   TRUE    0     TRUE    alanyl… AARS2  \n## # … with 40 more rows, and abbreviated variable names ¹​Cell.type, ²​condition,\n## #   ³​factor_of_interest, ⁴​description, ⁵​gene_name\n## # ℹ Use `print(n = ...)` to see more rows\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[{"level":1,"text":"tidybulk - part of tidyTranscriptomics","anchor":"tidybulk---part-of-tidytranscriptomics","htmlText":"tidybulk - part of tidyTranscriptomics"},{"level":2,"text":"Functions/utilities available","anchor":"functionsutilities-available","htmlText":"Functions/utilities available"},{"level":2,"text":"Installation","anchor":"installation","htmlText":"Installation"},{"level":1,"text":"Data","anchor":"data","htmlText":"Data"},{"level":2,"text":"Get the bibliography of your workflow","anchor":"get-the-bibliography-of-your-workflow","htmlText":"Get the bibliography of your workflow"},{"level":2,"text":"Aggregate duplicated transcripts","anchor":"aggregate-duplicated-transcripts","htmlText":"Aggregate duplicated transcripts"},{"level":2,"text":"Scale counts","anchor":"scale-counts","htmlText":"Scale counts"},{"level":2,"text":"Filter variable transcripts","anchor":"filter-variable-transcripts","htmlText":"Filter variable transcripts"},{"level":2,"text":"Reduce dimensions","anchor":"reduce-dimensions","htmlText":"Reduce dimensions"},{"level":2,"text":"Rotate dimensions","anchor":"rotate-dimensions","htmlText":"Rotate dimensions"},{"level":2,"text":"Test differential abundance","anchor":"test-differential-abundance","htmlText":"Test differential abundance"},{"level":2,"text":"Adjust counts","anchor":"adjust-counts","htmlText":"Adjust counts"},{"level":2,"text":"Deconvolve Cell type composition","anchor":"deconvolve-cell-type-composition","htmlText":"Deconvolve Cell type composition"},{"level":2,"text":"Test differential cell-type abundance","anchor":"test-differential-cell-type-abundance","htmlText":"Test differential cell-type abundance"},{"level":2,"text":"Cluster samples","anchor":"cluster-samples","htmlText":"Cluster samples"},{"level":2,"text":"Drop redundant transcripts","anchor":"drop-redundant-transcripts","htmlText":"Drop redundant transcripts"},{"level":2,"text":"Other useful wrappers","anchor":"other-useful-wrappers","htmlText":"Other useful wrappers"},{"level":2,"text":"From BAM/SAM to tibble of gene counts","anchor":"from-bamsam-to-tibble-of-gene-counts","htmlText":"From BAM/SAM to tibble of gene counts"},{"level":2,"text":"From ensembl IDs to gene symbol IDs","anchor":"from-ensembl-ids-to-gene-symbol-ids","htmlText":"From ensembl IDs to gene symbol IDs"},{"level":2,"text":"From gene symbol to gene description (gene name in full)","anchor":"from-gene-symbol-to-gene-description-gene-name-in-full","htmlText":"From gene symbol to gene description (gene name in full)"}],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2Fstemangiola%2Ftidybulk"}}],"overviewFilesProcessingTime":25.671716,"createFromTemplatePath":"/new?template_name=tidybulk\u0026template_owner=stemangiola"}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":"https://github.dev/","enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":true,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":true}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cLeluw:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cLeluw:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cLeluw[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cLeluw:active{-webkit-transition:none;transition:none;}/*!sc*/
.cLeluw[data-inactive]{cursor:auto;}/*!sc*/
.cLeluw:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled [data-component=ButtonCounter],.cLeluw:disabled [data-component="leadingVisual"],.cLeluw:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cLeluw:focus{outline:solid 1px transparent;}}/*!sc*/
.cLeluw [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cLeluw[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cLeluw[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cLeluw[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cLeluw[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cLeluw[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cLeluw[data-block="block"]{width:100%;}/*!sc*/
.cLeluw[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cLeluw[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cLeluw [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cLeluw [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cLeluw [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cLeluw [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cLeluw:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cLeluw[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cLeluw[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cLeluw:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cLeluw:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cLeluw[data-size="small"][data-no-visuals]{margin-right:8px;height:28px;}/*!sc*/
.jPraEl{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.jPraEl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.jPraEl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.jPraEl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.jPraEl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.jPraEl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.jPraEl:active{-webkit-transition:none;transition:none;}/*!sc*/
.jPraEl[data-inactive]{cursor:auto;}/*!sc*/
.jPraEl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled [data-component=ButtonCounter],.jPraEl:disabled [data-component="leadingVisual"],.jPraEl:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.jPraEl:focus{outline:solid 1px transparent;}}/*!sc*/
.jPraEl [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.jPraEl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.jPraEl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.jPraEl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.jPraEl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.jPraEl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.jPraEl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.jPraEl[data-block="block"]{width:100%;}/*!sc*/
.jPraEl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.jPraEl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.jPraEl [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.jPraEl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.jPraEl [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.jPraEl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.jPraEl:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.jPraEl:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.jPraEl[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.jPraEl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.jPraEl[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.jPraEl:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.jPraEl:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.jPraEl:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.jPraEl{color:var(--fgColor-muted,var(--color-fg-subtle,#6e7781));padding-left:8px;padding-right:8px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,cLeluw,jPraEl,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="master branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        master
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/stemangiola/tidybulk/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/stemangiola/tidybulk/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/stemangiola/tidybulk/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/stemangiola/tidybulk/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <button hidden="">
                  </button>
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" aria-label="Add file" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R2hil5:" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-plus" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="react-directory-remove-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 pyyxt" id=":R2lil5:" tabindex="0" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Add file
                     </span>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </button>
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/stemangiola/tidybulk/commits/master/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           1,436 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/stemangiola/tidybulk/commits/master/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".github, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/.github" title=".github">
                          .github
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".github, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/.github" title=".github">
                          .github
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-1">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="R, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/R" title="R">
                          R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="R, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/R" title="R">
                          R
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-2">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="data, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/data" title="data">
                          data
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="data, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/data" title="data">
                          data
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-3">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="dev, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/dev" title="dev">
                          dev
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="dev, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/dev" title="dev">
                          dev
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-4">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="inst, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/inst" title="inst">
                          inst
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="inst, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/inst" title="inst">
                          inst
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-5">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="man, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/man" title="man">
                          man
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="man, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/man" title="man">
                          man
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-6">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="tests, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/tests" title="tests">
                          tests
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="tests, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/tests" title="tests">
                          tests
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-7">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="vignettes, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/vignettes" title="vignettes">
                          vignettes
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="icon-directory" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1A1.75 1.75 0 0 0 0 2.75v10.5C0 14.216.784 15 1.75 15h12.5A1.75 1.75 0 0 0 16 13.25v-8.5A1.75 1.75 0 0 0 14.25 3H7.5a.25.25 0 0 1-.2-.1l-.9-1.2C6.07 1.26 5.55 1 5 1H1.75Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="vignettes, (Directory)" class="Link--primary" href="/stemangiola/tidybulk/tree/master/vignettes" title="vignettes">
                          vignettes
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-8">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".Rbuildignore, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/.Rbuildignore" title=".Rbuildignore">
                          .Rbuildignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".Rbuildignore, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/.Rbuildignore" title=".Rbuildignore">
                          .Rbuildignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-9">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".coveralls.yml, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/.coveralls.yml" title=".coveralls.yml">
                          .coveralls.yml
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".coveralls.yml, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/.coveralls.yml" title=".coveralls.yml">
                          .coveralls.yml
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-10">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".gitignore, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/.gitignore" title=".gitignore">
                          .gitignore
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-11">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".travis.yml, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/.travis.yml" title=".travis.yml">
                          .travis.yml
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label=".travis.yml, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/.travis.yml" title=".travis.yml">
                          .travis.yml
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-12">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="DESCRIPTION, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/DESCRIPTION" title="DESCRIPTION">
                          DESCRIPTION
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="DESCRIPTION, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/DESCRIPTION" title="DESCRIPTION">
                          DESCRIPTION
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-13">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="NAMESPACE, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/NAMESPACE" title="NAMESPACE">
                          NAMESPACE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="NAMESPACE, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/NAMESPACE" title="NAMESPACE">
                          NAMESPACE
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-14">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.Rmd, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/README.Rmd" title="README.Rmd">
                          README.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.Rmd, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/README.Rmd" title="README.Rmd">
                          README.Rmd
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-15">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-16">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="_pkgdown.yml, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/_pkgdown.yml" title="_pkgdown.yml">
                          _pkgdown.yml
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="_pkgdown.yml, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/_pkgdown.yml" title="_pkgdown.yml">
                          _pkgdown.yml
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row truncate-for-mobile" id="folder-row-17">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="codecov.yml, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/codecov.yml" title="codecov.yml">
                          codecov.yml
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="codecov.yml, (File)" class="Link--primary" href="/stemangiola/tidybulk/blob/master/codecov.yml" title="codecov.yml">
                          codecov.yml
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd show-for-mobile" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-label="Edit file" class="types__StyledButton-sc-ws60qy-0 cLeluw" data-component="IconButton" data-no-visuals="true" data-size="small" title="Edit file" type="button">
                    <svg aria-hidden="true" class="octicon octicon-pencil" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </button>
                   <button aria-haspopup="true" aria-label="Outline" class="types__StyledButton-sc-ws60qy-0 jPraEl" id=":Rdkl5:" style="--button-color:fg.subtle" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-list-unordered" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M5.75 2.5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5Zm0 5h8.5a.75.75 0 0 1 0 1.5h-8.5a.75.75 0 0 1 0-1.5ZM2 14a1 1 0 1 1 0-2 1 1 0 0 1 0 2Zm1-6a1 1 0 1 1-2 0 1 1 0 0 1 2 0ZM2 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      tidybulk - part of tidyTranscriptomics
                     </h1>
                     <a aria-label="Permalink: tidybulk - part of tidyTranscriptomics" class="anchor" href="#tidybulk---part-of-tidytranscriptomics" id="user-content-tidybulk---part-of-tidytranscriptomics">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     <a href="https://www.tidyverse.org/lifecycle/#maturing" rel="nofollow">
                      <img alt="Lifecycle:maturing" data-canonical-src="https://img.shields.io/badge/lifecycle-maturing-blue.svg" src="https://camo.githubusercontent.com/ee7536b3f1f530e91563303a6cdb32710af61731eb2f482dcb1af6b7d1bb42b2/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f6c6966656379636c652d6d61747572696e672d626c75652e737667" style="max-width: 100%;"/>
                     </a>
                     <a href="https://github.com/stemangiola/tidybulk/actions">
                      <img alt="R build status" src="https://github.com/stemangiola/tidybulk/workflows/R-CMD-check-bioc/badge.svg" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Brings transcriptomics to the tidyverse!
                     </strong>
                    </p>
                    <p dir="auto">
                     The code is released under the version 3 of the GNU General Public
License.
                    </p>
                    <p dir="auto">
                     <a href="/stemangiola/tidybulk/blob/master/man/figures/logo.png" rel="noopener noreferrer" target="_blank">
                      <img height="139px" src="/stemangiola/tidybulk/raw/master/man/figures/logo.png" style="max-width: 100%;" width="120px"/>
                     </a>
                    </p>
                    <p dir="auto">
                     website:
                     <a href="http://stemangiola.github.io/tidybulk/" rel="nofollow">
                      stemangiola.github.io/tidybulk/
                     </a>
                    </p>
                    <p dir="auto">
                     Please have a look also to
                    </p>
                    <ul dir="auto">
                     <li>
                      <a href="https://github.com/stemangiola/tidySummarizedExperiment">
                       tidySummarizedExperiment
                      </a>
                      for bulk data tidy representation
                     </li>
                     <li>
                      <a href="https://github.com/stemangiola/tidySingleCellExperiment">
                       tidySingleCellExperiment
                      </a>
                      for single-cell data tidy representation
                     </li>
                     <li>
                      <a href="https://github.com/stemangiola/tidyseurat">
                       tidyseurat
                      </a>
                      for
single-cell data tidy representation
                     </li>
                     <li>
                      <a href="https://github.com/stemangiola/tidyHeatmap">
                       tidyHeatmap
                      </a>
                      for
heatmaps produced with tidy principles analysis and manipulation
                     </li>
                     <li>
                      <a href="https://github.com/stemangiola/nanny">
                       nanny
                      </a>
                      for tidy high-level
data analysis and manipulation
                     </li>
                     <li>
                      <a href="https://github.com/stemangiola/tidygate">
                       tidygate
                      </a>
                      for adding
custom gate information to your tibble
                     </li>
                    </ul>
                    <p dir="auto">
                     <a href="/stemangiola/tidybulk/blob/master/man/figures/new_SE_usage-01.png" rel="noopener noreferrer" target="_blank">
                      <img src="/stemangiola/tidybulk/raw/master/man/figures/new_SE_usage-01.png" style="max-width: 100%;" width="800px"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Functions/utilities available
                     </h2>
                     <a aria-label="Permalink: Functions/utilities available" class="anchor" href="#functionsutilities-available" id="user-content-functionsutilities-available">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        Function
                       </th>
                       <th>
                        Description
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        <code>
                         aggregate_duplicates
                        </code>
                       </td>
                       <td>
                        Aggregate abundance and annotation of duplicated transcripts in a robust way
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         identify_abundant
                        </code>
                        <code>
                         keep_abundant
                        </code>
                       </td>
                       <td>
                        identify or keep the abundant genes
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         keep_variable
                        </code>
                       </td>
                       <td>
                        Filter for top variable features
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         scale_abundance
                        </code>
                       </td>
                       <td>
                        Scale (normalise) abundance for RNA sequencing depth
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         reduce_dimensions
                        </code>
                       </td>
                       <td>
                        Perform dimensionality reduction (PCA, MDS, tSNE, UMAP)
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         cluster_elements
                        </code>
                       </td>
                       <td>
                        Labels elements with cluster identity (kmeans, SNN)
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         remove_redundancy
                        </code>
                       </td>
                       <td>
                        Filter out elements with highly correlated features
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         adjust_abundance
                        </code>
                       </td>
                       <td>
                        Remove known unwanted variation (Combat)
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         test_differential_abundance
                        </code>
                       </td>
                       <td>
                        Differential transcript abundance testing (DESeq2, edgeR, voom)
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         deconvolve_cellularity
                        </code>
                       </td>
                       <td>
                        Estimated tissue composition (Cibersort, llsr, epic, xCell, mcp_counter, quantiseq
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         test_differential_cellularity
                        </code>
                       </td>
                       <td>
                        Differential cell-type abundance testing
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         test_stratification_cellularity
                        </code>
                       </td>
                       <td>
                        Estimate Kaplan-Meier survival differences
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         test_gene_enrichment
                        </code>
                       </td>
                       <td>
                        Gene enrichment analyses (EGSEA)
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         test_gene_overrepresentation
                        </code>
                       </td>
                       <td>
                        Gene enrichment on list of transcript names (no rank)
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         test_gene_rank
                        </code>
                       </td>
                       <td>
                        Gene enrichment on list of transcript (GSEA)
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         impute_missing_abundance
                        </code>
                       </td>
                       <td>
                        Impute abundance for missing data points using sample groupings
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        Utilities
                       </th>
                       <th>
                        Description
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        <code>
                         get_bibliography
                        </code>
                       </td>
                       <td>
                        Get the bibliography of your workflow
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         tidybulk
                        </code>
                       </td>
                       <td>
                        add tidybulk attributes to a tibble object
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         tidybulk_SAM_BAM
                        </code>
                       </td>
                       <td>
                        Convert SAM BAM files into tidybulk tibble
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         pivot_sample
                        </code>
                       </td>
                       <td>
                        Select sample-wise columns/information
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         pivot_transcript
                        </code>
                       </td>
                       <td>
                        Select transcript-wise columns/information
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         rotate_dimensions
                        </code>
                       </td>
                       <td>
                        Rotate two dimensions of a degree
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         ensembl_to_symbol
                        </code>
                       </td>
                       <td>
                        Add gene symbol from ensembl IDs
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         symbol_to_entrez
                        </code>
                       </td>
                       <td>
                        Add entrez ID from gene symbol
                       </td>
                      </tr>
                      <tr>
                       <td>
                        <code>
                         describe_transcript
                        </code>
                       </td>
                       <td>
                        Add gene description from gene symbol
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <p dir="auto">
                     All functions are directly compatibble with
                     <code>
                      SummarizedExperiment
                     </code>
                     object.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Installation
                     </h2>
                     <a aria-label="Permalink: Installation" class="anchor" href="#installation" id="user-content-installation">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     From Bioconductor
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='BiocManager::install("tidybulk")' dir="auto">
                     <pre><span class="pl-e">BiocManager</span><span class="pl-k">::</span>install(<span class="pl-s"><span class="pl-pds">"</span>tidybulk<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <p dir="auto">
                     From Github
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='devtools::install_github("stemangiola/tidybulk")' dir="auto">
                     <pre><span class="pl-e">devtools</span><span class="pl-k">::</span>install_github(<span class="pl-s"><span class="pl-pds">"</span>stemangiola/tidybulk<span class="pl-pds">"</span></span>)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h1 class="heading-element" dir="auto" tabindex="-1">
                      Data
                     </h1>
                     <a aria-label="Permalink: Data" class="anchor" href="#data" id="user-content-data">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We will use a
                     <code>
                      SummarizedExperiment
                     </code>
                     object
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_SE" dir="auto">
                     <pre><span class="pl-smi">counts_SE</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## # A SummarizedExperiment-tibble abstraction: 408,624 × 48
## # �[90mFeatures=8513 | Samples=48 | Assays=count�[0m
##    .feature .sample    count Cell.type time  condition batch factor_of_interest
##    &lt;chr&gt;    &lt;chr&gt;      &lt;dbl&gt; &lt;fct&gt;     &lt;fct&gt; &lt;lgl&gt;     &lt;fct&gt; &lt;lgl&gt;             
##  1 A1BG     SRR1740034   153 b_cell    0 d   TRUE      0     TRUE              
##  2 A1BG-AS1 SRR1740034    83 b_cell    0 d   TRUE      0     TRUE              
##  3 AAAS     SRR1740034   868 b_cell    0 d   TRUE      0     TRUE              
##  4 AACS     SRR1740034   222 b_cell    0 d   TRUE      0     TRUE              
##  5 AAGAB    SRR1740034   590 b_cell    0 d   TRUE      0     TRUE              
##  6 AAMDC    SRR1740034    48 b_cell    0 d   TRUE      0     TRUE              
##  7 AAMP     SRR1740034  1257 b_cell    0 d   TRUE      0     TRUE              
##  8 AANAT    SRR1740034   284 b_cell    0 d   TRUE      0     TRUE              
##  9 AAR2     SRR1740034   379 b_cell    0 d   TRUE      0     TRUE              
## 10 AARS2    SRR1740034   685 b_cell    0 d   TRUE      0     TRUE              
## # … with 40 more rows
## # ℹ Use `print(n = ...)` to see more rows">
                     <pre class="notranslate"><code>## # A SummarizedExperiment-tibble abstraction: 408,624 × 48
## # �[90mFeatures=8513 | Samples=48 | Assays=count�[0m
##    .feature .sample    count Cell.type time  condition batch factor_of_interest
##    &lt;chr&gt;    &lt;chr&gt;      &lt;dbl&gt; &lt;fct&gt;     &lt;fct&gt; &lt;lgl&gt;     &lt;fct&gt; &lt;lgl&gt;             
##  1 A1BG     SRR1740034   153 b_cell    0 d   TRUE      0     TRUE              
##  2 A1BG-AS1 SRR1740034    83 b_cell    0 d   TRUE      0     TRUE              
##  3 AAAS     SRR1740034   868 b_cell    0 d   TRUE      0     TRUE              
##  4 AACS     SRR1740034   222 b_cell    0 d   TRUE      0     TRUE              
##  5 AAGAB    SRR1740034   590 b_cell    0 d   TRUE      0     TRUE              
##  6 AAMDC    SRR1740034    48 b_cell    0 d   TRUE      0     TRUE              
##  7 AAMP     SRR1740034  1257 b_cell    0 d   TRUE      0     TRUE              
##  8 AANAT    SRR1740034   284 b_cell    0 d   TRUE      0     TRUE              
##  9 AAR2     SRR1740034   379 b_cell    0 d   TRUE      0     TRUE              
## 10 AARS2    SRR1740034   685 b_cell    0 d   TRUE      0     TRUE              
## # … with 40 more rows
## # ℹ Use `print(n = ...)` to see more rows
</code></pre>
                    </div>
                    <p dir="auto">
                     Loading
                     <code>
                      tidySummarizedExperiment
                     </code>
                     will automatically abstract this
object as
                     <code>
                      tibble
                     </code>
                     , so we can display it and manipulate it with tidy
tools. Although it looks different, and more tools (tidyverse) are
available to us, this object is in fact a
                     <code>
                      SummarizedExperiment
                     </code>
                     object.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="class(counts_SE)" dir="auto">
                     <pre>class(<span class="pl-smi">counts_SE</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='## [1] "SummarizedExperiment"
## attr(,"package")
## [1] "SummarizedExperiment"'>
                     <pre class="notranslate"><code>## [1] "SummarizedExperiment"
## attr(,"package")
## [1] "SummarizedExperiment"
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Get the bibliography of your workflow
                     </h2>
                     <a aria-label="Permalink: Get the bibliography of your workflow" class="anchor" href="#get-the-bibliography-of-your-workflow" id="user-content-get-the-bibliography-of-your-workflow">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     First of all, you can cite all articles utilised within your workflow
automatically from any tidybulk tibble
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_SE %&gt;%   get_bibliography()" dir="auto">
                     <pre><span class="pl-smi">counts_SE</span> %<span class="pl-k">&gt;</span>%   get_bibliography()</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Aggregate duplicated
                      <code>
                       transcripts
                      </code>
                     </h2>
                     <a aria-label="Permalink: Aggregate duplicated transcripts" class="anchor" href="#aggregate-duplicated-transcripts" id="user-content-aggregate-duplicated-transcripts">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     tidybulk provide the
                     <code>
                      aggregate_duplicates
                     </code>
                     function to aggregate
duplicated transcripts (e.g., isoforms, ensembl). For example, we often
have to convert ensembl symbols to gene/transcript symbol, but in doing
so we have to deal with duplicates.
                     <code>
                      aggregate_duplicates
                     </code>
                     takes a
tibble and column names (as symbols; for
                     <code>
                      sample
                     </code>
                     ,
                     <code>
                      transcript
                     </code>
                     and
                     <code>
                      count
                     </code>
                     ) as arguments and returns a tibble with transcripts with the
same name aggregated. All the rest of the columns are appended, and
factors and boolean are appended as characters.
                    </p>
                    <div dir="auto">
                     <p dir="auto">
                      TidyTranscriptomics
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="rowData(counts_SE)$gene_name = rownames(counts_SE)
counts_SE.aggr = counts_SE %&gt;% aggregate_duplicates(.transcript = gene_name)" dir="auto">
                      <pre>rowData(<span class="pl-smi">counts_SE</span>)<span class="pl-k">$</span><span class="pl-v">gene_name</span> <span class="pl-k">=</span> rownames(<span class="pl-smi">counts_SE</span>)
<span class="pl-v">counts_SE.aggr</span> <span class="pl-k">=</span> <span class="pl-smi">counts_SE</span> %<span class="pl-k">&gt;</span>% aggregate_duplicates(<span class="pl-v">.transcript</span> <span class="pl-k">=</span> <span class="pl-smi">gene_name</span>)</pre>
                     </div>
                    </div>
                    <div dir="auto">
                     <p dir="auto">
                      Standard procedure (comparative purpose)
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='temp = data.frame(
    symbol = dge_list$genes$symbol,
    dge_list$counts
)
dge_list.nr &lt;- by(temp, temp$symbol,
    function(df)
        if(length(df[1,1])&gt;0)
            matrixStats:::colSums(as.matrix(df[,-1]))
)
dge_list.nr &lt;- do.call("rbind", dge_list.nr)
colnames(dge_list.nr) &lt;- colnames(dge_list)' dir="auto">
                      <pre><span class="pl-v">temp</span> <span class="pl-k">=</span> <span class="pl-k">data.frame</span>(
    <span class="pl-v">symbol</span> <span class="pl-k">=</span> <span class="pl-smi">dge_list</span><span class="pl-k">$</span><span class="pl-smi">genes</span><span class="pl-k">$</span><span class="pl-smi">symbol</span>,
    <span class="pl-smi">dge_list</span><span class="pl-k">$</span><span class="pl-smi">counts</span>
)
<span class="pl-smi">dge_list.nr</span> <span class="pl-k">&lt;-</span> by(<span class="pl-smi">temp</span>, <span class="pl-smi">temp</span><span class="pl-k">$</span><span class="pl-smi">symbol</span>,
    <span class="pl-k">function</span>(<span class="pl-smi">df</span>)
        <span class="pl-k">if</span>(length(<span class="pl-smi">df</span>[<span class="pl-c1">1</span>,<span class="pl-c1">1</span>])<span class="pl-k">&gt;</span><span class="pl-c1">0</span>)
            <span class="pl-e">matrixStats</span><span class="pl-k">:::</span>colSums(as.matrix(<span class="pl-smi">df</span>[,<span class="pl-k">-</span><span class="pl-c1">1</span>]))
)
<span class="pl-smi">dge_list.nr</span> <span class="pl-k">&lt;-</span> do.call(<span class="pl-s"><span class="pl-pds">"</span>rbind<span class="pl-pds">"</span></span>, <span class="pl-smi">dge_list.nr</span>)
colnames(<span class="pl-smi">dge_list.nr</span>) <span class="pl-k">&lt;-</span> colnames(<span class="pl-smi">dge_list</span>)</pre>
                     </div>
                    </div>
                    <div dir="auto">
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Scale
                      <code>
                       counts
                      </code>
                     </h2>
                     <a aria-label="Permalink: Scale counts" class="anchor" href="#scale-counts" id="user-content-scale-counts">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We may want to compensate for sequencing depth, scaling the transcript
abundance (e.g., with TMM algorithm, Robinson and Oshlack
doi.org/10.1186/gb-2010-11-3-r25).
                     <code>
                      scale_abundance
                     </code>
                     takes a tibble,
column names (as symbols; for
                     <code>
                      sample
                     </code>
                     ,
                     <code>
                      transcript
                     </code>
                     and
                     <code>
                      count
                     </code>
                     ) and a
method as arguments and returns a tibble with additional columns with
scaled data as
                     <code>
                      &lt;NAME OF COUNT COLUMN&gt;_scaled
                     </code>
                     .
                    </p>
                    <div dir="auto">
                     <p dir="auto">
                      TidyTranscriptomics
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_SE.norm = counts_SE.aggr %&gt;% identify_abundant(factor_of_interest = condition) %&gt;% scale_abundance()" dir="auto">
                      <pre><span class="pl-v">counts_SE.norm</span> <span class="pl-k">=</span> <span class="pl-smi">counts_SE.aggr</span> %<span class="pl-k">&gt;</span>% identify_abundant(<span class="pl-v">factor_of_interest</span> <span class="pl-k">=</span> <span class="pl-smi">condition</span>) %<span class="pl-k">&gt;</span>% scale_abundance()</pre>
                     </div>
                    </div>
                    <div dir="auto">
                     <p dir="auto">
                      Standard procedure (comparative purpose)
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(edgeR)

dgList &lt;- DGEList(count_m=x,group=group)
keep &lt;- filterByExpr(dgList)
dgList &lt;- dgList[keep,,keep.lib.sizes=FALSE]
[...]
dgList &lt;- calcNormFactors(dgList, method="TMM")
norm_counts.table &lt;- cpm(dgList)' dir="auto">
                      <pre>library(<span class="pl-smi">edgeR</span>)

<span class="pl-smi">dgList</span> <span class="pl-k">&lt;-</span> DGEList(<span class="pl-v">count_m</span><span class="pl-k">=</span><span class="pl-smi">x</span>,<span class="pl-v">group</span><span class="pl-k">=</span><span class="pl-smi">group</span>)
<span class="pl-smi">keep</span> <span class="pl-k">&lt;-</span> filterByExpr(<span class="pl-smi">dgList</span>)
<span class="pl-smi">dgList</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">dgList</span>[<span class="pl-smi">keep</span>,,<span class="pl-v">keep.lib.sizes</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>]
[<span class="pl-k">...</span>]
<span class="pl-smi">dgList</span> <span class="pl-k">&lt;-</span> calcNormFactors(<span class="pl-smi">dgList</span>, <span class="pl-v">method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>TMM<span class="pl-pds">"</span></span>)
<span class="pl-smi">norm_counts.table</span> <span class="pl-k">&lt;-</span> cpm(<span class="pl-smi">dgList</span>)</pre>
                     </div>
                    </div>
                    <div dir="auto">
                    </div>
                    <p dir="auto">
                     We can easily plot the scaled density to check the scaling outcome. On
the x axis we have the log scaled counts, on the y axes we have the
density, data is grouped by sample and coloured by cell type.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_SE.norm %&gt;%
    ggplot(aes(count_scaled + 1, group=sample, color=`Cell.type`)) +
    geom_density() +
    scale_x_log10() +
    my_theme" dir="auto">
                     <pre><span class="pl-smi">counts_SE.norm</span> %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-smi">count_scaled</span> <span class="pl-k">+</span> <span class="pl-c1">1</span>, <span class="pl-v">group</span><span class="pl-k">=</span><span class="pl-smi">sample</span>, <span class="pl-v">color</span><span class="pl-k">=</span><span class="pl-smi">`Cell.type`</span>)) <span class="pl-k">+</span>
    geom_density() <span class="pl-k">+</span>
    scale_x_log10() <span class="pl-k">+</span>
    <span class="pl-smi">my_theme</span></pre>
                    </div>
                    <p dir="auto">
                     <a href="/stemangiola/tidybulk/blob/master/man/figures/plot_normalise-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/stemangiola/tidybulk/raw/master/man/figures/plot_normalise-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Filter
                      <code>
                       variable transcripts
                      </code>
                     </h2>
                     <a aria-label="Permalink: Filter variable transcripts" class="anchor" href="#filter-variable-transcripts" id="user-content-filter-variable-transcripts">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We may want to identify and filter variable transcripts.
                    </p>
                    <div dir="auto">
                     <p dir="auto">
                      TidyTranscriptomics
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_SE.norm.variable = counts_SE.norm %&gt;% keep_variable()" dir="auto">
                      <pre><span class="pl-v">counts_SE.norm.variable</span> <span class="pl-k">=</span> <span class="pl-smi">counts_SE.norm</span> %<span class="pl-k">&gt;</span>% keep_variable()</pre>
                     </div>
                    </div>
                    <div dir="auto">
                     <p dir="auto">
                      Standard procedure (comparative purpose)
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(edgeR)

x = norm_counts.table

s &lt;- rowMeans((x-rowMeans(x))^2)
o &lt;- order(s,decreasing=TRUE)
x &lt;- x[o[1L:top],,drop=FALSE]

norm_counts.table = norm_counts.table[rownames(x)]

norm_counts.table$cell_type = tibble_counts[
    match(
        tibble_counts$sample,
        rownames(norm_counts.table)
    ),
    "Cell.type"
]' dir="auto">
                      <pre>library(<span class="pl-smi">edgeR</span>)

<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">norm_counts.table</span>

<span class="pl-smi">s</span> <span class="pl-k">&lt;-</span> rowMeans((<span class="pl-smi">x</span><span class="pl-k">-</span>rowMeans(<span class="pl-smi">x</span>))<span class="pl-k">^</span><span class="pl-c1">2</span>)
<span class="pl-smi">o</span> <span class="pl-k">&lt;-</span> order(<span class="pl-smi">s</span>,<span class="pl-v">decreasing</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>)
<span class="pl-smi">x</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">x</span>[<span class="pl-smi">o</span>[<span class="pl-c1">1L</span><span class="pl-k">:</span><span class="pl-smi">top</span>],,<span class="pl-v">drop</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>]

<span class="pl-v">norm_counts.table</span> <span class="pl-k">=</span> <span class="pl-smi">norm_counts.table</span>[rownames(<span class="pl-smi">x</span>)]

<span class="pl-smi">norm_counts.table</span><span class="pl-k">$</span><span class="pl-v">cell_type</span> <span class="pl-k">=</span> <span class="pl-smi">tibble_counts</span>[
    match(
        <span class="pl-smi">tibble_counts</span><span class="pl-k">$</span><span class="pl-smi">sample</span>,
        rownames(<span class="pl-smi">norm_counts.table</span>)
    ),
    <span class="pl-s"><span class="pl-pds">"</span>Cell.type<span class="pl-pds">"</span></span>
]</pre>
                     </div>
                    </div>
                    <div dir="auto">
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Reduce
                      <code>
                       dimensions
                      </code>
                     </h2>
                     <a aria-label="Permalink: Reduce dimensions" class="anchor" href="#reduce-dimensions" id="user-content-reduce-dimensions">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We may want to reduce the dimensions of our data, for example using PCA
or MDS algorithms.
                     <code>
                      reduce_dimensions
                     </code>
                     takes a tibble, column names (as
symbols; for
                     <code>
                      sample
                     </code>
                     ,
                     <code>
                      transcript
                     </code>
                     and
                     <code>
                      count
                     </code>
                     ) and a method (e.g., MDS
or PCA) as arguments and returns a tibble with additional columns for
the reduced dimensions.
                    </p>
                    <p dir="auto">
                     <strong>
                      MDS
                     </strong>
                     (Robinson et al., 10.1093/bioinformatics/btp616)
                    </p>
                    <div dir="auto">
                     <p dir="auto">
                      TidyTranscriptomics
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.norm.MDS =
  counts_SE.norm %&gt;%
  reduce_dimensions(method="MDS", .dims = 6)' dir="auto">
                      <pre><span class="pl-v">counts_SE.norm.MDS</span> <span class="pl-k">=</span>
  <span class="pl-smi">counts_SE.norm</span> %<span class="pl-k">&gt;</span>%
  reduce_dimensions(<span class="pl-v">method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>MDS<span class="pl-pds">"</span></span>, <span class="pl-v">.dims</span> <span class="pl-k">=</span> <span class="pl-c1">6</span>)</pre>
                     </div>
                    </div>
                    <div dir="auto">
                     <p dir="auto">
                      Standard procedure (comparative purpose)
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(limma)

count_m_log = log(count_m + 1)
cmds = limma::plotMDS(ndim = .dims, plot = FALSE)

cmds = cmds %$% 
    cmdscale.out %&gt;%
    setNames(sprintf("Dim%s", 1:6))

cmds$cell_type = tibble_counts[
    match(tibble_counts$sample, rownames(cmds)),
    "Cell.type"
]' dir="auto">
                      <pre>library(<span class="pl-smi">limma</span>)

<span class="pl-v">count_m_log</span> <span class="pl-k">=</span> log(<span class="pl-smi">count_m</span> <span class="pl-k">+</span> <span class="pl-c1">1</span>)
<span class="pl-v">cmds</span> <span class="pl-k">=</span> <span class="pl-e">limma</span><span class="pl-k">::</span>plotMDS(<span class="pl-v">ndim</span> <span class="pl-k">=</span> <span class="pl-smi">.dims</span>, <span class="pl-v">plot</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>)

<span class="pl-v">cmds</span> <span class="pl-k">=</span> <span class="pl-smi">cmds</span> %<span class="pl-k">$</span>% 
    <span class="pl-smi">cmdscale.out</span> %<span class="pl-k">&gt;</span>%
    setNames(sprintf(<span class="pl-s"><span class="pl-pds">"</span>Dim%s<span class="pl-pds">"</span></span>, <span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">6</span>))

<span class="pl-smi">cmds</span><span class="pl-k">$</span><span class="pl-v">cell_type</span> <span class="pl-k">=</span> <span class="pl-smi">tibble_counts</span>[
    match(<span class="pl-smi">tibble_counts</span><span class="pl-k">$</span><span class="pl-smi">sample</span>, rownames(<span class="pl-smi">cmds</span>)),
    <span class="pl-s"><span class="pl-pds">"</span>Cell.type<span class="pl-pds">"</span></span>
]</pre>
                     </div>
                    </div>
                    <div dir="auto">
                    </div>
                    <p dir="auto">
                     On the x and y axes axis we have the reduced dimensions 1 to 3, data is
coloured by cell type.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.norm.MDS %&gt;% pivot_sample()  %&gt;% select(contains("Dim"), everything())' dir="auto">
                     <pre><span class="pl-smi">counts_SE.norm.MDS</span> %<span class="pl-k">&gt;</span>% pivot_sample()  %<span class="pl-k">&gt;</span>% select(contains(<span class="pl-s"><span class="pl-pds">"</span>Dim<span class="pl-pds">"</span></span>), everything())</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## # A tibble: 48 × 14
##      Dim1   Dim2   Dim3     Dim4    Dim5    Dim6 .sample   Cell.…¹ time  condi…²
##     &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;     &lt;fct&gt;   &lt;fct&gt; &lt;lgl&gt;  
##  1 -1.46   0.220 -1.68   0.0553   0.0658 -0.126  SRR17400… b_cell  0 d   TRUE   
##  2 -1.46   0.226 -1.71   0.0300   0.0454 -0.137  SRR17400… b_cell  1 d   TRUE   
##  3 -1.44   0.193 -1.60   0.0890   0.0503 -0.121  SRR17400… b_cell  3 d   TRUE   
##  4 -1.44   0.198 -1.67   0.0891   0.0543 -0.110  SRR17400… b_cell  7 d   TRUE   
##  5  0.243 -1.42   0.182  0.00642 -0.503  -0.131  SRR17400… dendri… 0 d   FALSE  
##  6  0.191 -1.42   0.195  0.0180  -0.457  -0.130  SRR17400… dendri… 1 d   FALSE  
##  7  0.257 -1.42   0.152  0.0130  -0.582  -0.0927 SRR17400… dendri… 3 d   FALSE  
##  8  0.162 -1.43   0.189  0.0232  -0.452  -0.109  SRR17400… dendri… 7 d   FALSE  
##  9  0.516 -1.47   0.240 -0.251    0.457  -0.119  SRR17400… monocy… 0 d   FALSE  
## 10  0.514 -1.41   0.231 -0.219    0.458  -0.131  SRR17400… monocy… 1 d   FALSE  
## # … with 38 more rows, 4 more variables: batch &lt;fct&gt;, factor_of_interest &lt;lgl&gt;,
## #   TMM &lt;dbl&gt;, multiplier &lt;dbl&gt;, and abbreviated variable names ¹​Cell.type,
## #   ²​condition
## # ℹ Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names">
                     <pre class="notranslate"><code>## # A tibble: 48 × 14
##      Dim1   Dim2   Dim3     Dim4    Dim5    Dim6 .sample   Cell.…¹ time  condi…²
##     &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;     &lt;fct&gt;   &lt;fct&gt; &lt;lgl&gt;  
##  1 -1.46   0.220 -1.68   0.0553   0.0658 -0.126  SRR17400… b_cell  0 d   TRUE   
##  2 -1.46   0.226 -1.71   0.0300   0.0454 -0.137  SRR17400… b_cell  1 d   TRUE   
##  3 -1.44   0.193 -1.60   0.0890   0.0503 -0.121  SRR17400… b_cell  3 d   TRUE   
##  4 -1.44   0.198 -1.67   0.0891   0.0543 -0.110  SRR17400… b_cell  7 d   TRUE   
##  5  0.243 -1.42   0.182  0.00642 -0.503  -0.131  SRR17400… dendri… 0 d   FALSE  
##  6  0.191 -1.42   0.195  0.0180  -0.457  -0.130  SRR17400… dendri… 1 d   FALSE  
##  7  0.257 -1.42   0.152  0.0130  -0.582  -0.0927 SRR17400… dendri… 3 d   FALSE  
##  8  0.162 -1.43   0.189  0.0232  -0.452  -0.109  SRR17400… dendri… 7 d   FALSE  
##  9  0.516 -1.47   0.240 -0.251    0.457  -0.119  SRR17400… monocy… 0 d   FALSE  
## 10  0.514 -1.41   0.231 -0.219    0.458  -0.131  SRR17400… monocy… 1 d   FALSE  
## # … with 38 more rows, 4 more variables: batch &lt;fct&gt;, factor_of_interest &lt;lgl&gt;,
## #   TMM &lt;dbl&gt;, multiplier &lt;dbl&gt;, and abbreviated variable names ¹​Cell.type,
## #   ²​condition
## # ℹ Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_SE.norm.MDS %&gt;%
    pivot_sample() %&gt;%
  GGally::ggpairs(columns = 6:(6+5), ggplot2::aes(colour=`Cell.type`))" dir="auto">
                     <pre><span class="pl-smi">counts_SE.norm.MDS</span> %<span class="pl-k">&gt;</span>%
    pivot_sample() %<span class="pl-k">&gt;</span>%
  <span class="pl-e">GGally</span><span class="pl-k">::</span>ggpairs(<span class="pl-v">columns</span> <span class="pl-k">=</span> <span class="pl-c1">6</span><span class="pl-k">:</span>(<span class="pl-c1">6</span><span class="pl-k">+</span><span class="pl-c1">5</span>), <span class="pl-e">ggplot2</span><span class="pl-k">::</span>aes(<span class="pl-v">colour</span><span class="pl-k">=</span><span class="pl-smi">`Cell.type`</span>))</pre>
                    </div>
                    <p dir="auto">
                     <a href="/stemangiola/tidybulk/blob/master/man/figures/plot_mds-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/stemangiola/tidybulk/raw/master/man/figures/plot_mds-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      PCA
                     </strong>
                    </p>
                    <div dir="auto">
                     <p dir="auto">
                      TidyTranscriptomics
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.norm.PCA =
  counts_SE.norm %&gt;%
  reduce_dimensions(method="PCA", .dims = 6)' dir="auto">
                      <pre><span class="pl-v">counts_SE.norm.PCA</span> <span class="pl-k">=</span>
  <span class="pl-smi">counts_SE.norm</span> %<span class="pl-k">&gt;</span>%
  reduce_dimensions(<span class="pl-v">method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>PCA<span class="pl-pds">"</span></span>, <span class="pl-v">.dims</span> <span class="pl-k">=</span> <span class="pl-c1">6</span>)</pre>
                     </div>
                    </div>
                    <div dir="auto">
                     <p dir="auto">
                      Standard procedure (comparative purpose)
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='count_m_log = log(count_m + 1)
pc = count_m_log %&gt;% prcomp(scale = TRUE)
variance = pc$sdev^2
variance = (variance / sum(variance))[1:6]
pc$cell_type = counts[
    match(counts$sample, rownames(pc)),
    "Cell.type"
]' dir="auto">
                      <pre><span class="pl-v">count_m_log</span> <span class="pl-k">=</span> log(<span class="pl-smi">count_m</span> <span class="pl-k">+</span> <span class="pl-c1">1</span>)
<span class="pl-v">pc</span> <span class="pl-k">=</span> <span class="pl-smi">count_m_log</span> %<span class="pl-k">&gt;</span>% prcomp(<span class="pl-v">scale</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
<span class="pl-v">variance</span> <span class="pl-k">=</span> <span class="pl-smi">pc</span><span class="pl-k">$</span><span class="pl-smi">sdev</span><span class="pl-k">^</span><span class="pl-c1">2</span>
<span class="pl-v">variance</span> <span class="pl-k">=</span> (<span class="pl-smi">variance</span> <span class="pl-k">/</span> sum(<span class="pl-smi">variance</span>))[<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">6</span>]
<span class="pl-smi">pc</span><span class="pl-k">$</span><span class="pl-v">cell_type</span> <span class="pl-k">=</span> <span class="pl-smi">counts</span>[
    match(<span class="pl-smi">counts</span><span class="pl-k">$</span><span class="pl-smi">sample</span>, rownames(<span class="pl-smi">pc</span>)),
    <span class="pl-s"><span class="pl-pds">"</span>Cell.type<span class="pl-pds">"</span></span>
]</pre>
                     </div>
                    </div>
                    <div dir="auto">
                    </div>
                    <p dir="auto">
                     On the x and y axes axis we have the reduced dimensions 1 to 3, data is
coloured by cell type.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.norm.PCA %&gt;% pivot_sample() %&gt;% select(contains("PC"), everything())' dir="auto">
                     <pre><span class="pl-smi">counts_SE.norm.PCA</span> %<span class="pl-k">&gt;</span>% pivot_sample() %<span class="pl-k">&gt;</span>% select(contains(<span class="pl-s"><span class="pl-pds">"</span>PC<span class="pl-pds">"</span></span>), everything())</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## # A tibble: 48 × 14
##        PC1   PC2    PC3     PC4    PC5   PC6 .sample Cell.…¹ time  condi…² batch
##      &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;   &lt;fct&gt;   &lt;fct&gt; &lt;lgl&gt;   &lt;fct&gt;
##  1 -12.6   -2.52 -14.9  -0.424  -0.592 -1.22 SRR174… b_cell  0 d   TRUE    0    
##  2 -12.6   -2.57 -15.2  -0.140  -0.388 -1.30 SRR174… b_cell  1 d   TRUE    1    
##  3 -12.6   -2.41 -14.5  -0.714  -0.344 -1.10 SRR174… b_cell  3 d   TRUE    1    
##  4 -12.5   -2.34 -14.9  -0.816  -0.427 -1.00 SRR174… b_cell  7 d   TRUE    1    
##  5   0.189 13.0    1.66 -0.0269  4.64  -1.35 SRR174… dendri… 0 d   FALSE   0    
##  6  -0.293 12.9    1.76 -0.0727  4.21  -1.28 SRR174… dendri… 1 d   FALSE   0    
##  7   0.407 13.0    1.42 -0.0529  5.37  -1.01 SRR174… dendri… 3 d   FALSE   1    
##  8  -0.620 13.0    1.73 -0.201   4.17  -1.07 SRR174… dendri… 7 d   FALSE   0    
##  9   2.56  13.5    2.32  2.03   -4.32  -1.22 SRR174… monocy… 0 d   FALSE   1    
## 10   2.65  13.1    2.21  1.80   -4.29  -1.30 SRR174… monocy… 1 d   FALSE   1    
## # … with 38 more rows, 3 more variables: factor_of_interest &lt;lgl&gt;, TMM &lt;dbl&gt;,
## #   multiplier &lt;dbl&gt;, and abbreviated variable names ¹​Cell.type, ²​condition
## # ℹ Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names">
                     <pre class="notranslate"><code>## # A tibble: 48 × 14
##        PC1   PC2    PC3     PC4    PC5   PC6 .sample Cell.…¹ time  condi…² batch
##      &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;   &lt;fct&gt;   &lt;fct&gt; &lt;lgl&gt;   &lt;fct&gt;
##  1 -12.6   -2.52 -14.9  -0.424  -0.592 -1.22 SRR174… b_cell  0 d   TRUE    0    
##  2 -12.6   -2.57 -15.2  -0.140  -0.388 -1.30 SRR174… b_cell  1 d   TRUE    1    
##  3 -12.6   -2.41 -14.5  -0.714  -0.344 -1.10 SRR174… b_cell  3 d   TRUE    1    
##  4 -12.5   -2.34 -14.9  -0.816  -0.427 -1.00 SRR174… b_cell  7 d   TRUE    1    
##  5   0.189 13.0    1.66 -0.0269  4.64  -1.35 SRR174… dendri… 0 d   FALSE   0    
##  6  -0.293 12.9    1.76 -0.0727  4.21  -1.28 SRR174… dendri… 1 d   FALSE   0    
##  7   0.407 13.0    1.42 -0.0529  5.37  -1.01 SRR174… dendri… 3 d   FALSE   1    
##  8  -0.620 13.0    1.73 -0.201   4.17  -1.07 SRR174… dendri… 7 d   FALSE   0    
##  9   2.56  13.5    2.32  2.03   -4.32  -1.22 SRR174… monocy… 0 d   FALSE   1    
## 10   2.65  13.1    2.21  1.80   -4.29  -1.30 SRR174… monocy… 1 d   FALSE   1    
## # … with 38 more rows, 3 more variables: factor_of_interest &lt;lgl&gt;, TMM &lt;dbl&gt;,
## #   multiplier &lt;dbl&gt;, and abbreviated variable names ¹​Cell.type, ²​condition
## # ℹ Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_SE.norm.PCA %&gt;%
     pivot_sample() %&gt;%
  GGally::ggpairs(columns = 11:13, ggplot2::aes(colour=`Cell.type`))" dir="auto">
                     <pre><span class="pl-smi">counts_SE.norm.PCA</span> %<span class="pl-k">&gt;</span>%
     pivot_sample() %<span class="pl-k">&gt;</span>%
  <span class="pl-e">GGally</span><span class="pl-k">::</span>ggpairs(<span class="pl-v">columns</span> <span class="pl-k">=</span> <span class="pl-c1">11</span><span class="pl-k">:</span><span class="pl-c1">13</span>, <span class="pl-e">ggplot2</span><span class="pl-k">::</span>aes(<span class="pl-v">colour</span><span class="pl-k">=</span><span class="pl-smi">`Cell.type`</span>))</pre>
                    </div>
                    <p dir="auto">
                     <a href="/stemangiola/tidybulk/blob/master/man/figures/plot_pca-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/stemangiola/tidybulk/raw/master/man/figures/plot_pca-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      tSNE
                     </strong>
                    </p>
                    <div dir="auto">
                     <p dir="auto">
                      TidyTranscriptomics
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.norm.tSNE =
    breast_tcga_mini_SE %&gt;%
    identify_abundant() %&gt;%
    reduce_dimensions(
        method = "tSNE",
        perplexity=10,
        pca_scale =TRUE
    )' dir="auto">
                      <pre><span class="pl-v">counts_SE.norm.tSNE</span> <span class="pl-k">=</span>
    <span class="pl-smi">breast_tcga_mini_SE</span> %<span class="pl-k">&gt;</span>%
    identify_abundant() %<span class="pl-k">&gt;</span>%
    reduce_dimensions(
        <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>tSNE<span class="pl-pds">"</span></span>,
        <span class="pl-v">perplexity</span><span class="pl-k">=</span><span class="pl-c1">10</span>,
        <span class="pl-v">pca_scale</span> <span class="pl-k">=</span><span class="pl-c1">TRUE</span>
    )</pre>
                     </div>
                    </div>
                    <div dir="auto">
                     <p dir="auto">
                      Standard procedure (comparative purpose)
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='count_m_log = log(count_m + 1)

tsne = Rtsne::Rtsne(
    t(count_m_log),
    perplexity=10,
        pca_scale =TRUE
)$Y
tsne$cell_type = tibble_counts[
    match(tibble_counts$sample, rownames(tsne)),
    "Cell.type"
]' dir="auto">
                      <pre><span class="pl-v">count_m_log</span> <span class="pl-k">=</span> log(<span class="pl-smi">count_m</span> <span class="pl-k">+</span> <span class="pl-c1">1</span>)

<span class="pl-v">tsne</span> <span class="pl-k">=</span> <span class="pl-e">Rtsne</span><span class="pl-k">::</span>Rtsne(
    t(<span class="pl-smi">count_m_log</span>),
    <span class="pl-v">perplexity</span><span class="pl-k">=</span><span class="pl-c1">10</span>,
        <span class="pl-v">pca_scale</span> <span class="pl-k">=</span><span class="pl-c1">TRUE</span>
)<span class="pl-k">$</span><span class="pl-smi">Y</span>
<span class="pl-smi">tsne</span><span class="pl-k">$</span><span class="pl-v">cell_type</span> <span class="pl-k">=</span> <span class="pl-smi">tibble_counts</span>[
    match(<span class="pl-smi">tibble_counts</span><span class="pl-k">$</span><span class="pl-smi">sample</span>, rownames(<span class="pl-smi">tsne</span>)),
    <span class="pl-s"><span class="pl-pds">"</span>Cell.type<span class="pl-pds">"</span></span>
]</pre>
                     </div>
                    </div>
                    <div dir="auto">
                    </div>
                    <p dir="auto">
                     Plot
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.norm.tSNE %&gt;%
    pivot_sample() %&gt;%
    select(contains("tSNE"), everything()) ' dir="auto">
                     <pre><span class="pl-smi">counts_SE.norm.tSNE</span> %<span class="pl-k">&gt;</span>%
    pivot_sample() %<span class="pl-k">&gt;</span>%
    select(contains(<span class="pl-s"><span class="pl-pds">"</span>tSNE<span class="pl-pds">"</span></span>), everything()) </pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## # A tibble: 251 × 4
##     tSNE1  tSNE2 .sample                      Call 
##     &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;                        &lt;fct&gt;
##  1  -5.25  10.2  TCGA-A1-A0SD-01A-11R-A115-07 LumA 
##  2   6.41   2.79 TCGA-A1-A0SF-01A-11R-A144-07 LumA 
##  3  -9.28   6.63 TCGA-A1-A0SG-01A-11R-A144-07 LumA 
##  4  -1.76   4.82 TCGA-A1-A0SH-01A-11R-A084-07 LumA 
##  5  -1.41  12.2  TCGA-A1-A0SI-01A-11R-A144-07 LumB 
##  6  -1.89  -3.60 TCGA-A1-A0SJ-01A-11R-A084-07 LumA 
##  7  18.5  -13.4  TCGA-A1-A0SK-01A-12R-A084-07 Basal
##  8  -4.03 -10.4  TCGA-A1-A0SM-01A-11R-A084-07 LumA 
##  9  -2.84 -10.8  TCGA-A1-A0SN-01A-11R-A144-07 LumB 
## 10 -19.3    5.03 TCGA-A1-A0SQ-01A-21R-A144-07 LumA 
## # … with 241 more rows
## # ℹ Use `print(n = ...)` to see more rows">
                     <pre class="notranslate"><code>## # A tibble: 251 × 4
##     tSNE1  tSNE2 .sample                      Call 
##     &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;                        &lt;fct&gt;
##  1  -5.25  10.2  TCGA-A1-A0SD-01A-11R-A115-07 LumA 
##  2   6.41   2.79 TCGA-A1-A0SF-01A-11R-A144-07 LumA 
##  3  -9.28   6.63 TCGA-A1-A0SG-01A-11R-A144-07 LumA 
##  4  -1.76   4.82 TCGA-A1-A0SH-01A-11R-A084-07 LumA 
##  5  -1.41  12.2  TCGA-A1-A0SI-01A-11R-A144-07 LumB 
##  6  -1.89  -3.60 TCGA-A1-A0SJ-01A-11R-A084-07 LumA 
##  7  18.5  -13.4  TCGA-A1-A0SK-01A-12R-A084-07 Basal
##  8  -4.03 -10.4  TCGA-A1-A0SM-01A-11R-A084-07 LumA 
##  9  -2.84 -10.8  TCGA-A1-A0SN-01A-11R-A144-07 LumB 
## 10 -19.3    5.03 TCGA-A1-A0SQ-01A-21R-A144-07 LumA 
## # … with 241 more rows
## # ℹ Use `print(n = ...)` to see more rows
</code></pre>
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_SE.norm.tSNE %&gt;%
    pivot_sample() %&gt;%
    ggplot(aes(x = `tSNE1`, y = `tSNE2`, color=Call)) + geom_point() + my_theme" dir="auto">
                     <pre><span class="pl-smi">counts_SE.norm.tSNE</span> %<span class="pl-k">&gt;</span>%
    pivot_sample() %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">`tSNE1`</span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">`tSNE2`</span>, <span class="pl-v">color</span><span class="pl-k">=</span><span class="pl-smi">Call</span>)) <span class="pl-k">+</span> geom_point() <span class="pl-k">+</span> <span class="pl-smi">my_theme</span></pre>
                    </div>
                    <p dir="auto">
                     <a href="/stemangiola/tidybulk/blob/master/man/figures/unnamed-chunk-14-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/stemangiola/tidybulk/raw/master/man/figures/unnamed-chunk-14-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Rotate
                      <code>
                       dimensions
                      </code>
                     </h2>
                     <a aria-label="Permalink: Rotate dimensions" class="anchor" href="#rotate-dimensions" id="user-content-rotate-dimensions">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We may want to rotate the reduced dimensions (or any two numeric columns
really) of our data, of a set angle.
                     <code>
                      rotate_dimensions
                     </code>
                     takes a tibble,
column names (as symbols; for
                     <code>
                      sample
                     </code>
                     ,
                     <code>
                      transcript
                     </code>
                     and
                     <code>
                      count
                     </code>
                     ) and an
angle as arguments and returns a tibble with additional columns for the
rotated dimensions. The rotated dimensions will be added to the original
data set as
                     <code>
                      &lt;NAME OF DIMENSION&gt; rotated &lt;ANGLE&gt;
                     </code>
                     by default, or as
specified in the input arguments.
                    </p>
                    <div dir="auto">
                     <p dir="auto">
                      TidyTranscriptomics
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.norm.MDS.rotated =
  counts_SE.norm.MDS %&gt;%
    rotate_dimensions(`Dim1`, `Dim2`, rotation_degrees = 45, action="get")' dir="auto">
                      <pre><span class="pl-v">counts_SE.norm.MDS.rotated</span> <span class="pl-k">=</span>
  <span class="pl-smi">counts_SE.norm.MDS</span> %<span class="pl-k">&gt;</span>%
    rotate_dimensions(<span class="pl-smi">`Dim1`</span>, <span class="pl-smi">`Dim2`</span>, <span class="pl-v">rotation_degrees</span> <span class="pl-k">=</span> <span class="pl-c1">45</span>, <span class="pl-v">action</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>get<span class="pl-pds">"</span></span>)</pre>
                     </div>
                    </div>
                    <div dir="auto">
                     <p dir="auto">
                      Standard procedure (comparative purpose)
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='rotation = function(m, d) {
    r = d * pi / 180
    ((bind_rows(
        c(`1` = cos(r), `2` = -sin(r)),
        c(`1` = sin(r), `2` = cos(r))
    ) %&gt;% as_matrix) %*% m)
}
mds_r = pca %&gt;% rotation(rotation_degrees)
mds_r$cell_type = counts[
    match(counts$sample, rownames(mds_r)),
    "Cell.type"
]' dir="auto">
                      <pre><span class="pl-en">rotation</span> <span class="pl-k">=</span> <span class="pl-k">function</span>(<span class="pl-smi">m</span>, <span class="pl-smi">d</span>) {
    <span class="pl-v">r</span> <span class="pl-k">=</span> <span class="pl-smi">d</span> <span class="pl-k">*</span> <span class="pl-c1">pi</span> <span class="pl-k">/</span> <span class="pl-c1">180</span>
    ((bind_rows(
        c(<span class="pl-smi">`1`</span> <span class="pl-k">=</span> cos(<span class="pl-smi">r</span>), <span class="pl-smi">`2`</span> <span class="pl-k">=</span> <span class="pl-k">-</span>sin(<span class="pl-smi">r</span>)),
        c(<span class="pl-smi">`1`</span> <span class="pl-k">=</span> sin(<span class="pl-smi">r</span>), <span class="pl-smi">`2`</span> <span class="pl-k">=</span> cos(<span class="pl-smi">r</span>))
    ) %<span class="pl-k">&gt;</span>% <span class="pl-smi">as_matrix</span>) <span class="pl-k">%*%</span> <span class="pl-smi">m</span>)
}
<span class="pl-v">mds_r</span> <span class="pl-k">=</span> <span class="pl-smi">pca</span> %<span class="pl-k">&gt;</span>% rotation(<span class="pl-smi">rotation_degrees</span>)
<span class="pl-smi">mds_r</span><span class="pl-k">$</span><span class="pl-v">cell_type</span> <span class="pl-k">=</span> <span class="pl-smi">counts</span>[
    match(<span class="pl-smi">counts</span><span class="pl-k">$</span><span class="pl-smi">sample</span>, rownames(<span class="pl-smi">mds_r</span>)),
    <span class="pl-s"><span class="pl-pds">"</span>Cell.type<span class="pl-pds">"</span></span>
]</pre>
                     </div>
                    </div>
                    <div dir="auto">
                    </div>
                    <p dir="auto">
                     <strong>
                      Original
                     </strong>
                     On the x and y axes axis we have the first two reduced
dimensions, data is coloured by cell type.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_SE.norm.MDS.rotated %&gt;%
    ggplot(aes(x=`Dim1`, y=`Dim2`, color=`Cell.type` )) +
  geom_point() +
  my_theme" dir="auto">
                     <pre><span class="pl-smi">counts_SE.norm.MDS.rotated</span> %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">`Dim1`</span>, <span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-smi">`Dim2`</span>, <span class="pl-v">color</span><span class="pl-k">=</span><span class="pl-smi">`Cell.type`</span> )) <span class="pl-k">+</span>
  geom_point() <span class="pl-k">+</span>
  <span class="pl-smi">my_theme</span></pre>
                    </div>
                    <p dir="auto">
                     <a href="/stemangiola/tidybulk/blob/master/man/figures/plot_rotate_1-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/stemangiola/tidybulk/raw/master/man/figures/plot_rotate_1-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Rotated
                     </strong>
                     On the x and y axes axis we have the first two reduced
dimensions rotated of 45 degrees, data is coloured by cell type.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_SE.norm.MDS.rotated %&gt;%
    pivot_sample() %&gt;%
    ggplot(aes(x=`Dim1_rotated_45`, y=`Dim2_rotated_45`, color=`Cell.type` )) +
  geom_point() +
  my_theme" dir="auto">
                     <pre><span class="pl-smi">counts_SE.norm.MDS.rotated</span> %<span class="pl-k">&gt;</span>%
    pivot_sample() %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">`Dim1_rotated_45`</span>, <span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-smi">`Dim2_rotated_45`</span>, <span class="pl-v">color</span><span class="pl-k">=</span><span class="pl-smi">`Cell.type`</span> )) <span class="pl-k">+</span>
  geom_point() <span class="pl-k">+</span>
  <span class="pl-smi">my_theme</span></pre>
                    </div>
                    <p dir="auto">
                     <a href="/stemangiola/tidybulk/blob/master/man/figures/plot_rotate_2-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/stemangiola/tidybulk/raw/master/man/figures/plot_rotate_2-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Test
                      <code>
                       differential abundance
                      </code>
                     </h2>
                     <a aria-label="Permalink: Test differential abundance" class="anchor" href="#test-differential-abundance" id="user-content-test-differential-abundance">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We may want to test for differential transcription between sample-wise
factors of interest (e.g., with edgeR).
                     <code>
                      test_differential_abundance
                     </code>
                     takes a tibble, column names (as symbols; for
                     <code>
                      sample
                     </code>
                     ,
                     <code>
                      transcript
                     </code>
                     and
                     <code>
                      count
                     </code>
                     ) and a formula representing the desired linear model as
arguments and returns a tibble with additional columns for the
statistics from the hypothesis test (e.g., log fold change, p-value and
false discovery rate).
                    </p>
                    <div dir="auto">
                     <p dir="auto">
                      TidyTranscriptomics
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.de =
    counts_SE %&gt;%
    test_differential_abundance( ~ condition, action="get")
counts_SE.de' dir="auto">
                      <pre><span class="pl-v">counts_SE.de</span> <span class="pl-k">=</span>
    <span class="pl-smi">counts_SE</span> %<span class="pl-k">&gt;</span>%
    test_differential_abundance( <span class="pl-k">~</span> <span class="pl-smi">condition</span>, <span class="pl-v">action</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>get<span class="pl-pds">"</span></span>)
<span class="pl-smi">counts_SE.de</span></pre>
                     </div>
                    </div>
                    <div dir="auto">
                     <p dir="auto">
                      Standard procedure (comparative purpose)
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="library(edgeR)

dgList &lt;- DGEList(counts=counts_m,group=group)
keep &lt;- filterByExpr(dgList)
dgList &lt;- dgList[keep,,keep.lib.sizes=FALSE]
dgList &lt;- calcNormFactors(dgList)
design &lt;- model.matrix(~group)
dgList &lt;- estimateDisp(dgList,design)
fit &lt;- glmQLFit(dgList,design)
qlf &lt;- glmQLFTest(fit,coef=2)
topTags(qlf, n=Inf)" dir="auto">
                      <pre>library(<span class="pl-smi">edgeR</span>)

<span class="pl-smi">dgList</span> <span class="pl-k">&lt;-</span> DGEList(<span class="pl-v">counts</span><span class="pl-k">=</span><span class="pl-smi">counts_m</span>,<span class="pl-v">group</span><span class="pl-k">=</span><span class="pl-smi">group</span>)
<span class="pl-smi">keep</span> <span class="pl-k">&lt;-</span> filterByExpr(<span class="pl-smi">dgList</span>)
<span class="pl-smi">dgList</span> <span class="pl-k">&lt;-</span> <span class="pl-smi">dgList</span>[<span class="pl-smi">keep</span>,,<span class="pl-v">keep.lib.sizes</span><span class="pl-k">=</span><span class="pl-c1">FALSE</span>]
<span class="pl-smi">dgList</span> <span class="pl-k">&lt;-</span> calcNormFactors(<span class="pl-smi">dgList</span>)
<span class="pl-smi">design</span> <span class="pl-k">&lt;-</span> model.matrix(<span class="pl-k">~</span><span class="pl-smi">group</span>)
<span class="pl-smi">dgList</span> <span class="pl-k">&lt;-</span> estimateDisp(<span class="pl-smi">dgList</span>,<span class="pl-smi">design</span>)
<span class="pl-smi">fit</span> <span class="pl-k">&lt;-</span> glmQLFit(<span class="pl-smi">dgList</span>,<span class="pl-smi">design</span>)
<span class="pl-smi">qlf</span> <span class="pl-k">&lt;-</span> glmQLFTest(<span class="pl-smi">fit</span>,<span class="pl-v">coef</span><span class="pl-k">=</span><span class="pl-c1">2</span>)
topTags(<span class="pl-smi">qlf</span>, <span class="pl-v">n</span><span class="pl-k">=</span><span class="pl-c1">Inf</span>)</pre>
                     </div>
                    </div>
                    <div dir="auto">
                    </div>
                    <p dir="auto">
                     The functon
                     <code>
                      test_differential_abundance
                     </code>
                     operated with contrasts too.
The constrasts hve the name of the design matrix (generally
&lt;NAME_COLUMN_COVARIATE&gt;&lt;VALUES_OF_COVARIATE&gt;)
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.de =
    counts_SE %&gt;%
    identify_abundant(factor_of_interest = condition) %&gt;%
    test_differential_abundance(
        ~ 0 + condition,                  
        .contrasts = c( "conditionTRUE - conditionFALSE"),
        action="get"
    )' dir="auto">
                     <pre><span class="pl-v">counts_SE.de</span> <span class="pl-k">=</span>
    <span class="pl-smi">counts_SE</span> %<span class="pl-k">&gt;</span>%
    identify_abundant(<span class="pl-v">factor_of_interest</span> <span class="pl-k">=</span> <span class="pl-smi">condition</span>) %<span class="pl-k">&gt;</span>%
    test_differential_abundance(
        <span class="pl-k">~</span> <span class="pl-c1">0</span> <span class="pl-k">+</span> <span class="pl-smi">condition</span>,                  
        <span class="pl-v">.contrasts</span> <span class="pl-k">=</span> c( <span class="pl-s"><span class="pl-pds">"</span>conditionTRUE - conditionFALSE<span class="pl-pds">"</span></span>),
        <span class="pl-v">action</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>get<span class="pl-pds">"</span></span>
    )</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Adjust
                      <code>
                       counts
                      </code>
                     </h2>
                     <a aria-label="Permalink: Adjust counts" class="anchor" href="#adjust-counts" id="user-content-adjust-counts">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We may want to adjust
                     <code>
                      counts
                     </code>
                     for (known) unwanted variation.
                     <code>
                      adjust_abundance
                     </code>
                     takes as arguments a tibble, column names (as
symbols; for
                     <code>
                      sample
                     </code>
                     ,
                     <code>
                      transcript
                     </code>
                     and
                     <code>
                      count
                     </code>
                     ) and a formula
representing the desired linear model where the first covariate is the
factor of interest and the second covariate is the unwanted variation,
and returns a tibble with additional columns for the adjusted counts as
                     <code>
                      &lt;COUNT COLUMN&gt;_adjusted
                     </code>
                     . At the moment just an unwanted covariated is
allowed at a time.
                    </p>
                    <div dir="auto">
                     <p dir="auto">
                      TidyTranscriptomics
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_SE.norm.adj =
    counts_SE.norm %&gt;% adjust_abundance(    ~ factor_of_interest + batch)" dir="auto">
                      <pre><span class="pl-v">counts_SE.norm.adj</span> <span class="pl-k">=</span>
    <span class="pl-smi">counts_SE.norm</span> %<span class="pl-k">&gt;</span>% adjust_abundance(    <span class="pl-k">~</span> <span class="pl-smi">factor_of_interest</span> <span class="pl-k">+</span> <span class="pl-smi">batch</span>)</pre>
                     </div>
                    </div>
                    <div dir="auto">
                     <p dir="auto">
                      Standard procedure (comparative purpose)
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(sva)

count_m_log = log(count_m + 1)

design =
        model.matrix(
            object = ~ factor_of_interest + batch,
            data = annotation
        )

count_m_log.sva =
    ComBat(
            batch = design[,2],
            mod = design,
            ...
        )

count_m_log.sva = ceiling(exp(count_m_log.sva) -1)
count_m_log.sva$cell_type = counts[
    match(counts$sample, rownames(count_m_log.sva)),
    "Cell.type"
]' dir="auto">
                      <pre>library(<span class="pl-smi">sva</span>)

<span class="pl-v">count_m_log</span> <span class="pl-k">=</span> log(<span class="pl-smi">count_m</span> <span class="pl-k">+</span> <span class="pl-c1">1</span>)

<span class="pl-v">design</span> <span class="pl-k">=</span>
        model.matrix(
            <span class="pl-v">object</span> <span class="pl-k">=</span> <span class="pl-k">~</span> <span class="pl-smi">factor_of_interest</span> <span class="pl-k">+</span> <span class="pl-smi">batch</span>,
            <span class="pl-v">data</span> <span class="pl-k">=</span> <span class="pl-smi">annotation</span>
        )

<span class="pl-v">count_m_log.sva</span> <span class="pl-k">=</span>
    ComBat(
            <span class="pl-v">batch</span> <span class="pl-k">=</span> <span class="pl-smi">design</span>[,<span class="pl-c1">2</span>],
            <span class="pl-v">mod</span> <span class="pl-k">=</span> <span class="pl-smi">design</span>,
            <span class="pl-k">...</span>
        )

<span class="pl-v">count_m_log.sva</span> <span class="pl-k">=</span> ceiling(exp(<span class="pl-smi">count_m_log.sva</span>) <span class="pl-k">-</span><span class="pl-c1">1</span>)
<span class="pl-smi">count_m_log.sva</span><span class="pl-k">$</span><span class="pl-v">cell_type</span> <span class="pl-k">=</span> <span class="pl-smi">counts</span>[
    match(<span class="pl-smi">counts</span><span class="pl-k">$</span><span class="pl-smi">sample</span>, rownames(<span class="pl-smi">count_m_log.sva</span>)),
    <span class="pl-s"><span class="pl-pds">"</span>Cell.type<span class="pl-pds">"</span></span>
]</pre>
                     </div>
                    </div>
                    <div dir="auto">
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Deconvolve
                      <code>
                       Cell type composition
                      </code>
                     </h2>
                     <a aria-label="Permalink: Deconvolve Cell type composition" class="anchor" href="#deconvolve-cell-type-composition" id="user-content-deconvolve-cell-type-composition">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We may want to infer the cell type composition of our samples (with the
algorithm Cibersort; Newman et al., 10.1038/nmeth.3337).
                     <code>
                      deconvolve_cellularity
                     </code>
                     takes as arguments a tibble, column names (as
symbols; for
                     <code>
                      sample
                     </code>
                     ,
                     <code>
                      transcript
                     </code>
                     and
                     <code>
                      count
                     </code>
                     ) and returns a tibble
with additional columns for the adjusted cell type proportions.
                    </p>
                    <div dir="auto">
                     <p dir="auto">
                      TidyTranscriptomics
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.cibersort =
    counts_SE %&gt;%
    deconvolve_cellularity(action="get", cores=1, prefix = "cibersort__") ' dir="auto">
                      <pre><span class="pl-v">counts_SE.cibersort</span> <span class="pl-k">=</span>
    <span class="pl-smi">counts_SE</span> %<span class="pl-k">&gt;</span>%
    deconvolve_cellularity(<span class="pl-v">action</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>get<span class="pl-pds">"</span></span>, <span class="pl-v">cores</span><span class="pl-k">=</span><span class="pl-c1">1</span>, <span class="pl-v">prefix</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>cibersort__<span class="pl-pds">"</span></span>) </pre>
                     </div>
                     <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## 
## The downloaded binary packages are in
##  /var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T//RtmpIi5KN6/downloaded_packages">
                      <pre class="notranslate"><code>## 
## The downloaded binary packages are in
##  /var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T//RtmpIi5KN6/downloaded_packages
</code></pre>
                     </div>
                    </div>
                    <div dir="auto">
                     <p dir="auto">
                      Standard procedure (comparative purpose)
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='source(‘CIBERSORT.R’)
count_m %&gt;% write.table("mixture_file.txt")
results &lt;- CIBERSORT(
    "sig_matrix_file.txt",
    "mixture_file.txt",
    perm=100, QN=TRUE
)
results$cell_type = tibble_counts[
    match(tibble_counts$sample, rownames(results)),
    "Cell.type"
]' dir="auto">
                      <pre>source(‘<span class="pl-smi">CIBERSORT.R</span>’)
<span class="pl-smi">count_m</span> %<span class="pl-k">&gt;</span>% write.table(<span class="pl-s"><span class="pl-pds">"</span>mixture_file.txt<span class="pl-pds">"</span></span>)
<span class="pl-smi">results</span> <span class="pl-k">&lt;-</span> CIBERSORT(
    <span class="pl-s"><span class="pl-pds">"</span>sig_matrix_file.txt<span class="pl-pds">"</span></span>,
    <span class="pl-s"><span class="pl-pds">"</span>mixture_file.txt<span class="pl-pds">"</span></span>,
    <span class="pl-v">perm</span><span class="pl-k">=</span><span class="pl-c1">100</span>, <span class="pl-v">QN</span><span class="pl-k">=</span><span class="pl-c1">TRUE</span>
)
<span class="pl-smi">results</span><span class="pl-k">$</span><span class="pl-v">cell_type</span> <span class="pl-k">=</span> <span class="pl-smi">tibble_counts</span>[
    match(<span class="pl-smi">tibble_counts</span><span class="pl-k">$</span><span class="pl-smi">sample</span>, rownames(<span class="pl-smi">results</span>)),
    <span class="pl-s"><span class="pl-pds">"</span>Cell.type<span class="pl-pds">"</span></span>
]</pre>
                     </div>
                    </div>
                    <div dir="auto">
                    </div>
                    <p dir="auto">
                     With the new annotated data frame, we can plot the distributions of cell
types across samples, and compare them with the nominal cell type labels
to check for the purity of isolation. On the x axis we have the cell
types inferred by Cibersort, on the y axis we have the inferred
proportions. The data is facetted and coloured by nominal cell types
(annotation given by the researcher after FACS sorting).
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.cibersort %&gt;%
    pivot_longer(
        names_to= "Cell_type_inferred", 
        values_to = "proportion", 
        names_prefix ="cibersort__", 
        cols=contains("cibersort__")
    ) %&gt;%
  ggplot(aes(x=`Cell_type_inferred`, y=proportion, fill=`Cell.type`)) +
  geom_boxplot() +
  facet_wrap(~`Cell.type`) +
  my_theme +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5), aspect.ratio=1/5)' dir="auto">
                     <pre><span class="pl-smi">counts_SE.cibersort</span> %<span class="pl-k">&gt;</span>%
    pivot_longer(
        <span class="pl-v">names_to</span><span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>Cell_type_inferred<span class="pl-pds">"</span></span>, 
        <span class="pl-v">values_to</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>proportion<span class="pl-pds">"</span></span>, 
        <span class="pl-v">names_prefix</span> <span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>cibersort__<span class="pl-pds">"</span></span>, 
        <span class="pl-v">cols</span><span class="pl-k">=</span>contains(<span class="pl-s"><span class="pl-pds">"</span>cibersort__<span class="pl-pds">"</span></span>)
    ) %<span class="pl-k">&gt;</span>%
  ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">`Cell_type_inferred`</span>, <span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-smi">proportion</span>, <span class="pl-v">fill</span><span class="pl-k">=</span><span class="pl-smi">`Cell.type`</span>)) <span class="pl-k">+</span>
  geom_boxplot() <span class="pl-k">+</span>
  facet_wrap(<span class="pl-k">~</span><span class="pl-smi">`Cell.type`</span>) <span class="pl-k">+</span>
  <span class="pl-smi">my_theme</span> <span class="pl-k">+</span>
  theme(<span class="pl-v">axis.text.x</span> <span class="pl-k">=</span> element_text(<span class="pl-v">angle</span> <span class="pl-k">=</span> <span class="pl-c1">90</span>, <span class="pl-v">hjust</span> <span class="pl-k">=</span> <span class="pl-c1">1</span>, <span class="pl-v">vjust</span> <span class="pl-k">=</span> <span class="pl-c1">0.5</span>), <span class="pl-v">aspect.ratio</span><span class="pl-k">=</span><span class="pl-c1">1</span><span class="pl-k">/</span><span class="pl-c1">5</span>)</pre>
                    </div>
                    <p dir="auto">
                     <a href="/stemangiola/tidybulk/blob/master/man/figures/plot_cibersort-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/stemangiola/tidybulk/raw/master/man/figures/plot_cibersort-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Test differential cell-type abundance
                     </h2>
                     <a aria-label="Permalink: Test differential cell-type abundance" class="anchor" href="#test-differential-cell-type-abundance" id="user-content-test-differential-cell-type-abundance">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We can also perform a statistical test on the differential cell-type
abundance across conditions
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    counts_SE %&gt;%
    test_differential_cellularity(. ~ condition )" dir="auto">
                     <pre>    <span class="pl-smi">counts_SE</span> %<span class="pl-k">&gt;</span>%
    test_differential_cellularity(<span class="pl-smi">.</span> <span class="pl-k">~</span> <span class="pl-smi">condition</span> )</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## 
## The downloaded binary packages are in
##  /var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T//RtmpIi5KN6/downloaded_packages

## # A tibble: 22 × 7
##    .cell_type                  cell_t…¹ estim…² estim…³ std.e…⁴ stati…⁵ p.valu…⁶
##    &lt;chr&gt;                       &lt;list&gt;     &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;
##  1 cibersort.B.cells.naive     &lt;tibble&gt;   -2.94   2.25    0.367   6.13  8.77e-10
##  2 cibersort.B.cells.memory    &lt;tibble&gt;   -4.86   1.48    0.436   3.40  6.77e- 4
##  3 cibersort.Plasma.cells      &lt;tibble&gt;   -5.33  -0.487   0.507  -0.960 3.37e- 1
##  4 cibersort.T.cells.CD8       &lt;tibble&gt;   -2.33   0.924   0.475   1.94  5.18e- 2
##  5 cibersort.T.cells.CD4.naive &lt;tibble&gt;   -2.83  -0.620   0.531  -1.17  2.43e- 1
##  6 cibersort.T.cells.CD4.memo… &lt;tibble&gt;   -2.46   0.190   0.500   0.380 7.04e- 1
##  7 cibersort.T.cells.CD4.memo… &lt;tibble&gt;   -3.67   2.23    0.427   5.22  1.80e- 7
##  8 cibersort.T.cells.follicul… &lt;tibble&gt;   -5.68  -0.217   0.507  -0.427 6.69e- 1
##  9 cibersort.T.cells.regulato… &lt;tibble&gt;   -5.04   1.94    0.360   5.39  6.86e- 8
## 10 cibersort.T.cells.gamma.de… &lt;tibble&gt;   -4.78  -0.250   0.514  -0.486 6.27e- 1
## # … with 12 more rows, and abbreviated variable names ¹​cell_type_proportions,
## #   ²​`estimate_(Intercept)`, ³​estimate_conditionTRUE, ⁴​std.error_conditionTRUE,
## #   ⁵​statistic_conditionTRUE, ⁶​p.value_conditionTRUE
## # ℹ Use `print(n = ...)` to see more rows">
                     <pre class="notranslate"><code>## 
## The downloaded binary packages are in
##  /var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T//RtmpIi5KN6/downloaded_packages

## # A tibble: 22 × 7
##    .cell_type                  cell_t…¹ estim…² estim…³ std.e…⁴ stati…⁵ p.valu…⁶
##    &lt;chr&gt;                       &lt;list&gt;     &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;
##  1 cibersort.B.cells.naive     &lt;tibble&gt;   -2.94   2.25    0.367   6.13  8.77e-10
##  2 cibersort.B.cells.memory    &lt;tibble&gt;   -4.86   1.48    0.436   3.40  6.77e- 4
##  3 cibersort.Plasma.cells      &lt;tibble&gt;   -5.33  -0.487   0.507  -0.960 3.37e- 1
##  4 cibersort.T.cells.CD8       &lt;tibble&gt;   -2.33   0.924   0.475   1.94  5.18e- 2
##  5 cibersort.T.cells.CD4.naive &lt;tibble&gt;   -2.83  -0.620   0.531  -1.17  2.43e- 1
##  6 cibersort.T.cells.CD4.memo… &lt;tibble&gt;   -2.46   0.190   0.500   0.380 7.04e- 1
##  7 cibersort.T.cells.CD4.memo… &lt;tibble&gt;   -3.67   2.23    0.427   5.22  1.80e- 7
##  8 cibersort.T.cells.follicul… &lt;tibble&gt;   -5.68  -0.217   0.507  -0.427 6.69e- 1
##  9 cibersort.T.cells.regulato… &lt;tibble&gt;   -5.04   1.94    0.360   5.39  6.86e- 8
## 10 cibersort.T.cells.gamma.de… &lt;tibble&gt;   -4.78  -0.250   0.514  -0.486 6.27e- 1
## # … with 12 more rows, and abbreviated variable names ¹​cell_type_proportions,
## #   ²​`estimate_(Intercept)`, ³​estimate_conditionTRUE, ⁴​std.error_conditionTRUE,
## #   ⁵​statistic_conditionTRUE, ⁶​p.value_conditionTRUE
## # ℹ Use `print(n = ...)` to see more rows
</code></pre>
                    </div>
                    <p dir="auto">
                     We can also perform regression analysis with censored data (coxph).
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="    # Add survival data

counts_SE_survival = 
    counts_SE %&gt;%
    nest(data = -sample) %&gt;%
        mutate(
            days = sample(1:1000, size = n()),
            dead = sample(c(0,1), size = n(), replace = TRUE)
        ) %&gt;%
    unnest(data) 

# Test
counts_SE_survival %&gt;%
    test_differential_cellularity(survival::Surv(days, dead) ~ .)" dir="auto">
                     <pre>    <span class="pl-c"><span class="pl-c">#</span> Add survival data</span>

<span class="pl-v">counts_SE_survival</span> <span class="pl-k">=</span> 
    <span class="pl-smi">counts_SE</span> %<span class="pl-k">&gt;</span>%
    nest(<span class="pl-v">data</span> <span class="pl-k">=</span> <span class="pl-k">-</span><span class="pl-smi">sample</span>) %<span class="pl-k">&gt;</span>%
        mutate(
            <span class="pl-v">days</span> <span class="pl-k">=</span> sample(<span class="pl-c1">1</span><span class="pl-k">:</span><span class="pl-c1">1000</span>, <span class="pl-v">size</span> <span class="pl-k">=</span> n()),
            <span class="pl-v">dead</span> <span class="pl-k">=</span> sample(c(<span class="pl-c1">0</span>,<span class="pl-c1">1</span>), <span class="pl-v">size</span> <span class="pl-k">=</span> n(), <span class="pl-v">replace</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>)
        ) %<span class="pl-k">&gt;</span>%
    unnest(<span class="pl-smi">data</span>) 

<span class="pl-c"><span class="pl-c">#</span> Test</span>
<span class="pl-smi">counts_SE_survival</span> %<span class="pl-k">&gt;</span>%
    test_differential_cellularity(<span class="pl-e">survival</span><span class="pl-k">::</span>Surv(<span class="pl-smi">days</span>, <span class="pl-smi">dead</span>) <span class="pl-k">~</span> <span class="pl-smi">.</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## # A tibble: 22 × 6
##    .cell_type                           cell_t…¹ estim…² std.e…³ stati…⁴ p.value
##    &lt;chr&gt;                                &lt;list&gt;     &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
##  1 cibersort.B.cells.naive              &lt;tibble&gt;  -0.224   0.415  -0.540  0.589 
##  2 cibersort.B.cells.memory             &lt;tibble&gt;   0.510   0.346   1.48   0.140 
##  3 cibersort.Plasma.cells               &lt;tibble&gt;   0.892   0.449   1.99   0.0467
##  4 cibersort.T.cells.CD8                &lt;tibble&gt;   0.531   0.639   0.831  0.406 
##  5 cibersort.T.cells.CD4.naive          &lt;tibble&gt;   0.112   0.386   0.290  0.772 
##  6 cibersort.T.cells.CD4.memory.resting &lt;tibble&gt;   0.498   0.540   0.921  0.357 
##  7 cibersort.T.cells.CD4.memory.activa… &lt;tibble&gt;   2.37    0.939   2.52   0.0117
##  8 cibersort.T.cells.follicular.helper  &lt;tibble&gt;  -0.544   0.421  -1.29   0.197 
##  9 cibersort.T.cells.regulatory..Tregs. &lt;tibble&gt;   1.59    0.656   2.42   0.0157
## 10 cibersort.T.cells.gamma.delta        &lt;tibble&gt;   0.510   0.688   0.741  0.459 
## # … with 12 more rows, and abbreviated variable names ¹​cell_type_proportions,
## #   ²​estimate, ³​std.error, ⁴​statistic
## # ℹ Use `print(n = ...)` to see more rows">
                     <pre class="notranslate"><code>## # A tibble: 22 × 6
##    .cell_type                           cell_t…¹ estim…² std.e…³ stati…⁴ p.value
##    &lt;chr&gt;                                &lt;list&gt;     &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
##  1 cibersort.B.cells.naive              &lt;tibble&gt;  -0.224   0.415  -0.540  0.589 
##  2 cibersort.B.cells.memory             &lt;tibble&gt;   0.510   0.346   1.48   0.140 
##  3 cibersort.Plasma.cells               &lt;tibble&gt;   0.892   0.449   1.99   0.0467
##  4 cibersort.T.cells.CD8                &lt;tibble&gt;   0.531   0.639   0.831  0.406 
##  5 cibersort.T.cells.CD4.naive          &lt;tibble&gt;   0.112   0.386   0.290  0.772 
##  6 cibersort.T.cells.CD4.memory.resting &lt;tibble&gt;   0.498   0.540   0.921  0.357 
##  7 cibersort.T.cells.CD4.memory.activa… &lt;tibble&gt;   2.37    0.939   2.52   0.0117
##  8 cibersort.T.cells.follicular.helper  &lt;tibble&gt;  -0.544   0.421  -1.29   0.197 
##  9 cibersort.T.cells.regulatory..Tregs. &lt;tibble&gt;   1.59    0.656   2.42   0.0157
## 10 cibersort.T.cells.gamma.delta        &lt;tibble&gt;   0.510   0.688   0.741  0.459 
## # … with 12 more rows, and abbreviated variable names ¹​cell_type_proportions,
## #   ²​estimate, ³​std.error, ⁴​statistic
## # ℹ Use `print(n = ...)` to see more rows
</code></pre>
                    </div>
                    <p dir="auto">
                     We can also perform test of Kaplan-Meier curves.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_stratified = 
    counts_SE_survival %&gt;%

    # Test
    test_stratification_cellularity(
        survival::Surv(days, dead) ~ .,
        sample, transcript, count
    )

counts_stratified" dir="auto">
                     <pre><span class="pl-v">counts_stratified</span> <span class="pl-k">=</span> 
    <span class="pl-smi">counts_SE_survival</span> %<span class="pl-k">&gt;</span>%

    <span class="pl-c"><span class="pl-c">#</span> Test</span>
    test_stratification_cellularity(
        <span class="pl-e">survival</span><span class="pl-k">::</span>Surv(<span class="pl-smi">days</span>, <span class="pl-smi">dead</span>) <span class="pl-k">~</span> <span class="pl-smi">.</span>,
        <span class="pl-smi">sample</span>, <span class="pl-smi">transcript</span>, <span class="pl-smi">count</span>
    )

<span class="pl-smi">counts_stratified</span></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## # A tibble: 22 × 6
##    .cell_type                         cell_t…¹ .low_…² .high…³ pvalue plot      
##    &lt;chr&gt;                              &lt;list&gt;     &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt; &lt;list&gt;    
##  1 cibersort.B.cells.naive            &lt;tibble&gt;   14.4     7.56  0.506 &lt;ggsrvplt&gt;
##  2 cibersort.B.cells.memory           &lt;tibble&gt;   17.2     4.77  0.500 &lt;ggsrvplt&gt;
##  3 cibersort.Plasma.cells             &lt;tibble&gt;   13.3     8.73  0.903 &lt;ggsrvplt&gt;
##  4 cibersort.T.cells.CD8              &lt;tibble&gt;   13.9     8.06  0.369 &lt;ggsrvplt&gt;
##  5 cibersort.T.cells.CD4.naive        &lt;tibble&gt;   12.8     9.15  0.407 &lt;ggsrvplt&gt;
##  6 cibersort.T.cells.CD4.memory.rest… &lt;tibble&gt;    7.65   14.4   0.105 &lt;ggsrvplt&gt;
##  7 cibersort.T.cells.CD4.memory.acti… &lt;tibble&gt;   15.7     6.26  0.392 &lt;ggsrvplt&gt;
##  8 cibersort.T.cells.follicular.help… &lt;tibble&gt;   17.1     4.88  0.949 &lt;ggsrvplt&gt;
##  9 cibersort.T.cells.regulatory..Tre… &lt;tibble&gt;   13.7     8.35  0.771 &lt;ggsrvplt&gt;
## 10 cibersort.T.cells.gamma.delta      &lt;tibble&gt;   16.2     5.76  0.379 &lt;ggsrvplt&gt;
## # … with 12 more rows, and abbreviated variable names ¹​cell_type_proportions,
## #   ²​.low_cellularity_expected, ³​.high_cellularity_expected
## # ℹ Use `print(n = ...)` to see more rows">
                     <pre class="notranslate"><code>## # A tibble: 22 × 6
##    .cell_type                         cell_t…¹ .low_…² .high…³ pvalue plot      
##    &lt;chr&gt;                              &lt;list&gt;     &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt; &lt;list&gt;    
##  1 cibersort.B.cells.naive            &lt;tibble&gt;   14.4     7.56  0.506 &lt;ggsrvplt&gt;
##  2 cibersort.B.cells.memory           &lt;tibble&gt;   17.2     4.77  0.500 &lt;ggsrvplt&gt;
##  3 cibersort.Plasma.cells             &lt;tibble&gt;   13.3     8.73  0.903 &lt;ggsrvplt&gt;
##  4 cibersort.T.cells.CD8              &lt;tibble&gt;   13.9     8.06  0.369 &lt;ggsrvplt&gt;
##  5 cibersort.T.cells.CD4.naive        &lt;tibble&gt;   12.8     9.15  0.407 &lt;ggsrvplt&gt;
##  6 cibersort.T.cells.CD4.memory.rest… &lt;tibble&gt;    7.65   14.4   0.105 &lt;ggsrvplt&gt;
##  7 cibersort.T.cells.CD4.memory.acti… &lt;tibble&gt;   15.7     6.26  0.392 &lt;ggsrvplt&gt;
##  8 cibersort.T.cells.follicular.help… &lt;tibble&gt;   17.1     4.88  0.949 &lt;ggsrvplt&gt;
##  9 cibersort.T.cells.regulatory..Tre… &lt;tibble&gt;   13.7     8.35  0.771 &lt;ggsrvplt&gt;
## 10 cibersort.T.cells.gamma.delta      &lt;tibble&gt;   16.2     5.76  0.379 &lt;ggsrvplt&gt;
## # … with 12 more rows, and abbreviated variable names ¹​cell_type_proportions,
## #   ²​.low_cellularity_expected, ³​.high_cellularity_expected
## # ℹ Use `print(n = ...)` to see more rows
</code></pre>
                    </div>
                    <p dir="auto">
                     Plot Kaplan-Meier curves
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_stratified$plot[[1]]" dir="auto">
                     <pre><span class="pl-smi">counts_stratified</span><span class="pl-k">$</span><span class="pl-smi">plot</span>[[<span class="pl-c1">1</span>]]</pre>
                    </div>
                    <p dir="auto">
                     <a href="/stemangiola/tidybulk/blob/master/man/figures/unnamed-chunk-19-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/stemangiola/tidybulk/raw/master/man/figures/unnamed-chunk-19-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Cluster
                      <code>
                       samples
                      </code>
                     </h2>
                     <a aria-label="Permalink: Cluster samples" class="anchor" href="#cluster-samples" id="user-content-cluster-samples">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We may want to cluster our data (e.g., using k-means sample-wise).
                     <code>
                      cluster_elements
                     </code>
                     takes as arguments a tibble, column names (as
symbols; for
                     <code>
                      sample
                     </code>
                     ,
                     <code>
                      transcript
                     </code>
                     and
                     <code>
                      count
                     </code>
                     ) and returns a tibble
with additional columns for the cluster annotation. At the moment only
k-means clustering is supported, the plan is to introduce more
clustering methods.
                    </p>
                    <p dir="auto">
                     <strong>
                      k-means
                     </strong>
                    </p>
                    <div dir="auto">
                     <p dir="auto">
                      TidyTranscriptomics
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.norm.cluster = counts_SE.norm.MDS %&gt;%
  cluster_elements(method="kmeans", centers = 2, action="get" )' dir="auto">
                      <pre><span class="pl-v">counts_SE.norm.cluster</span> <span class="pl-k">=</span> <span class="pl-smi">counts_SE.norm.MDS</span> %<span class="pl-k">&gt;</span>%
  cluster_elements(<span class="pl-v">method</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>kmeans<span class="pl-pds">"</span></span>, <span class="pl-v">centers</span> <span class="pl-k">=</span> <span class="pl-c1">2</span>, <span class="pl-v">action</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>get<span class="pl-pds">"</span></span> )</pre>
                     </div>
                    </div>
                    <div dir="auto">
                     <p dir="auto">
                      Standard procedure (comparative purpose)
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='count_m_log = log(count_m + 1)

k = kmeans(count_m_log, iter.max = 1000, ...)
cluster = k$cluster

cluster$cell_type = tibble_counts[
    match(tibble_counts$sample, rownames(cluster)),
    c("Cell.type", "Dim1", "Dim2")
]' dir="auto">
                      <pre><span class="pl-v">count_m_log</span> <span class="pl-k">=</span> log(<span class="pl-smi">count_m</span> <span class="pl-k">+</span> <span class="pl-c1">1</span>)

<span class="pl-v">k</span> <span class="pl-k">=</span> kmeans(<span class="pl-smi">count_m_log</span>, <span class="pl-v">iter.max</span> <span class="pl-k">=</span> <span class="pl-c1">1000</span>, <span class="pl-k">...</span>)
<span class="pl-v">cluster</span> <span class="pl-k">=</span> <span class="pl-smi">k</span><span class="pl-k">$</span><span class="pl-smi">cluster</span>

<span class="pl-smi">cluster</span><span class="pl-k">$</span><span class="pl-v">cell_type</span> <span class="pl-k">=</span> <span class="pl-smi">tibble_counts</span>[
    match(<span class="pl-smi">tibble_counts</span><span class="pl-k">$</span><span class="pl-smi">sample</span>, rownames(<span class="pl-smi">cluster</span>)),
    c(<span class="pl-s"><span class="pl-pds">"</span>Cell.type<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Dim1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Dim2<span class="pl-pds">"</span></span>)
]</pre>
                     </div>
                    </div>
                    <div dir="auto">
                    </div>
                    <p dir="auto">
                     We can add cluster annotation to the MDS dimension reduced data set and
plot.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content=" counts_SE.norm.cluster %&gt;%
    ggplot(aes(x=`Dim1`, y=`Dim2`, color=`cluster_kmeans`)) +
  geom_point() +
  my_theme" dir="auto">
                     <pre> <span class="pl-smi">counts_SE.norm.cluster</span> %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">`Dim1`</span>, <span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-smi">`Dim2`</span>, <span class="pl-v">color</span><span class="pl-k">=</span><span class="pl-smi">`cluster_kmeans`</span>)) <span class="pl-k">+</span>
  geom_point() <span class="pl-k">+</span>
  <span class="pl-smi">my_theme</span></pre>
                    </div>
                    <p dir="auto">
                     <a href="/stemangiola/tidybulk/blob/master/man/figures/plot_cluster-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/stemangiola/tidybulk/raw/master/man/figures/plot_cluster-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      SNN
                     </strong>
                    </p>
                    <p dir="auto">
                     Matrix package (v1.3-3) causes an error with Seurat::FindNeighbors used
in this method. We are trying to solve this issue. At the moment this
option in unaviable.
                    </p>
                    <div dir="auto">
                     <p dir="auto">
                      TidyTranscriptomics
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.norm.SNN =
    counts_SE.norm.tSNE %&gt;%
    cluster_elements(method = "SNN")' dir="auto">
                      <pre><span class="pl-v">counts_SE.norm.SNN</span> <span class="pl-k">=</span>
    <span class="pl-smi">counts_SE.norm.tSNE</span> %<span class="pl-k">&gt;</span>%
    cluster_elements(<span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>SNN<span class="pl-pds">"</span></span>)</pre>
                     </div>
                    </div>
                    <div dir="auto">
                     <p dir="auto">
                      Standard procedure (comparative purpose)
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library(Seurat)

snn = CreateSeuratObject(count_m)
snn = ScaleData(
    snn, display.progress = TRUE,
    num.cores=4, do.par = TRUE
)
snn = FindVariableFeatures(snn, selection.method = "vst")
snn = FindVariableFeatures(snn, selection.method = "vst")
snn = RunPCA(snn, npcs = 30)
snn = FindNeighbors(snn)
snn = FindClusters(snn, method = "igraph", ...)
snn = snn[["seurat_clusters"]]

snn$cell_type = tibble_counts[
    match(tibble_counts$sample, rownames(snn)),
    c("Cell.type", "Dim1", "Dim2")
]' dir="auto">
                      <pre>library(<span class="pl-smi">Seurat</span>)

<span class="pl-v">snn</span> <span class="pl-k">=</span> CreateSeuratObject(<span class="pl-smi">count_m</span>)
<span class="pl-v">snn</span> <span class="pl-k">=</span> ScaleData(
    <span class="pl-smi">snn</span>, <span class="pl-v">display.progress</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
    <span class="pl-v">num.cores</span><span class="pl-k">=</span><span class="pl-c1">4</span>, <span class="pl-v">do.par</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>
)
<span class="pl-v">snn</span> <span class="pl-k">=</span> FindVariableFeatures(<span class="pl-smi">snn</span>, <span class="pl-v">selection.method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>vst<span class="pl-pds">"</span></span>)
<span class="pl-v">snn</span> <span class="pl-k">=</span> FindVariableFeatures(<span class="pl-smi">snn</span>, <span class="pl-v">selection.method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>vst<span class="pl-pds">"</span></span>)
<span class="pl-v">snn</span> <span class="pl-k">=</span> RunPCA(<span class="pl-smi">snn</span>, <span class="pl-v">npcs</span> <span class="pl-k">=</span> <span class="pl-c1">30</span>)
<span class="pl-v">snn</span> <span class="pl-k">=</span> FindNeighbors(<span class="pl-smi">snn</span>)
<span class="pl-v">snn</span> <span class="pl-k">=</span> FindClusters(<span class="pl-smi">snn</span>, <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>igraph<span class="pl-pds">"</span></span>, <span class="pl-k">...</span>)
<span class="pl-v">snn</span> <span class="pl-k">=</span> <span class="pl-smi">snn</span>[[<span class="pl-s"><span class="pl-pds">"</span>seurat_clusters<span class="pl-pds">"</span></span>]]

<span class="pl-smi">snn</span><span class="pl-k">$</span><span class="pl-v">cell_type</span> <span class="pl-k">=</span> <span class="pl-smi">tibble_counts</span>[
    match(<span class="pl-smi">tibble_counts</span><span class="pl-k">$</span><span class="pl-smi">sample</span>, rownames(<span class="pl-smi">snn</span>)),
    c(<span class="pl-s"><span class="pl-pds">"</span>Cell.type<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Dim1<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Dim2<span class="pl-pds">"</span></span>)
]</pre>
                     </div>
                    </div>
                    <div dir="auto">
                    </div>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.norm.SNN %&gt;%
    pivot_sample() %&gt;%
    select(contains("tSNE"), everything()) 

counts_SE.norm.SNN %&gt;%
    pivot_sample() %&gt;%
    gather(source, Call, c("cluster_SNN", "Call")) %&gt;%
    distinct() %&gt;%
    ggplot(aes(x = `tSNE1`, y = `tSNE2`, color=Call)) + geom_point() + facet_grid(~source) + my_theme


# Do differential transcription between clusters
counts_SE.norm.SNN %&gt;%
    mutate(factor_of_interest = `cluster_SNN` == 3) %&gt;%
    test_differential_abundance(
    ~ factor_of_interest,
    action="get"
   )' dir="auto">
                     <pre><span class="pl-smi">counts_SE.norm.SNN</span> %<span class="pl-k">&gt;</span>%
    pivot_sample() %<span class="pl-k">&gt;</span>%
    select(contains(<span class="pl-s"><span class="pl-pds">"</span>tSNE<span class="pl-pds">"</span></span>), everything()) 

<span class="pl-smi">counts_SE.norm.SNN</span> %<span class="pl-k">&gt;</span>%
    pivot_sample() %<span class="pl-k">&gt;</span>%
    gather(<span class="pl-smi">source</span>, <span class="pl-smi">Call</span>, c(<span class="pl-s"><span class="pl-pds">"</span>cluster_SNN<span class="pl-pds">"</span></span>, <span class="pl-s"><span class="pl-pds">"</span>Call<span class="pl-pds">"</span></span>)) %<span class="pl-k">&gt;</span>%
    distinct() %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-v">x</span> <span class="pl-k">=</span> <span class="pl-smi">`tSNE1`</span>, <span class="pl-v">y</span> <span class="pl-k">=</span> <span class="pl-smi">`tSNE2`</span>, <span class="pl-v">color</span><span class="pl-k">=</span><span class="pl-smi">Call</span>)) <span class="pl-k">+</span> geom_point() <span class="pl-k">+</span> facet_grid(<span class="pl-k">~</span><span class="pl-smi">source</span>) <span class="pl-k">+</span> <span class="pl-smi">my_theme</span>


<span class="pl-c"><span class="pl-c">#</span> Do differential transcription between clusters</span>
<span class="pl-smi">counts_SE.norm.SNN</span> %<span class="pl-k">&gt;</span>%
    mutate(<span class="pl-v">factor_of_interest</span> <span class="pl-k">=</span> <span class="pl-smi">`cluster_SNN`</span> <span class="pl-k">==</span> <span class="pl-c1">3</span>) %<span class="pl-k">&gt;</span>%
    test_differential_abundance(
    <span class="pl-k">~</span> <span class="pl-smi">factor_of_interest</span>,
    <span class="pl-v">action</span><span class="pl-k">=</span><span class="pl-s"><span class="pl-pds">"</span>get<span class="pl-pds">"</span></span>
   )</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Drop
                      <code>
                       redundant
                      </code>
                      transcripts
                     </h2>
                     <a aria-label="Permalink: Drop redundant transcripts" class="anchor" href="#drop-redundant-transcripts" id="user-content-drop-redundant-transcripts">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We may want to remove redundant elements from the original data set
(e.g., samples or transcripts), for example if we want to define
cell-type specific signatures with low sample redundancy.
                     <code>
                      remove_redundancy
                     </code>
                     takes as arguments a tibble, column names (as
symbols; for
                     <code>
                      sample
                     </code>
                     ,
                     <code>
                      transcript
                     </code>
                     and
                     <code>
                      count
                     </code>
                     ) and returns a tibble
with redundant elements removed (e.g., samples). Two redundancy
estimation approaches are supported:
                    </p>
                    <ul dir="auto">
                     <li>
                      removal of highly correlated clusters of elements (keeping a
representative) with method=“correlation”
                     </li>
                     <li>
                      removal of most proximal element pairs in a reduced dimensional
space.
                     </li>
                    </ul>
                    <p dir="auto">
                     <strong>
                      Approach 1
                     </strong>
                    </p>
                    <div dir="auto">
                     <p dir="auto">
                      TidyTranscriptomics
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.norm.non_redundant =
    counts_SE.norm.MDS %&gt;%
  remove_redundancy(    method = "correlation" )' dir="auto">
                      <pre><span class="pl-v">counts_SE.norm.non_redundant</span> <span class="pl-k">=</span>
    <span class="pl-smi">counts_SE.norm.MDS</span> %<span class="pl-k">&gt;</span>%
  remove_redundancy(    <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>correlation<span class="pl-pds">"</span></span> )</pre>
                     </div>
                     <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## 
## The downloaded binary packages are in
##  /var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T//RtmpIi5KN6/downloaded_packages">
                      <pre class="notranslate"><code>## 
## The downloaded binary packages are in
##  /var/folders/zn/m_qvr9zd7tq0wdtsbq255f8xypj_zg/T//RtmpIi5KN6/downloaded_packages
</code></pre>
                     </div>
                    </div>
                    <div dir="auto">
                     <p dir="auto">
                      Standard procedure (comparative purpose)
                     </p>
                     <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="library(widyr)

.data.correlated =
    pairwise_cor(
        counts,
        sample,
        transcript,
        rc,
        sort = TRUE,
        diag = FALSE,
        upper = FALSE
    ) %&gt;%
    filter(correlation &gt; correlation_threshold) %&gt;%
    distinct(item1) %&gt;%
    rename(!!.element := item1)

# Return non redudant data frame
counts %&gt;% anti_join(.data.correlated) %&gt;%
    spread(sample, rc, - transcript) %&gt;%
    left_join(annotation)" dir="auto">
                      <pre>library(<span class="pl-smi">widyr</span>)

<span class="pl-v">.data.correlated</span> <span class="pl-k">=</span>
    pairwise_cor(
        <span class="pl-smi">counts</span>,
        <span class="pl-smi">sample</span>,
        <span class="pl-smi">transcript</span>,
        <span class="pl-smi">rc</span>,
        <span class="pl-v">sort</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
        <span class="pl-v">diag</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>,
        <span class="pl-v">upper</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>
    ) %<span class="pl-k">&gt;</span>%
    filter(<span class="pl-smi">correlation</span> <span class="pl-k">&gt;</span> <span class="pl-smi">correlation_threshold</span>) %<span class="pl-k">&gt;</span>%
    distinct(<span class="pl-smi">item1</span>) %<span class="pl-k">&gt;</span>%
    rename(<span class="pl-k">!!</span><span class="pl-smi">.element</span> <span class="pl-k">:</span><span class="pl-k">=</span> <span class="pl-smi">item1</span>)

<span class="pl-c"><span class="pl-c">#</span> Return non redudant data frame</span>
<span class="pl-smi">counts</span> %<span class="pl-k">&gt;</span>% anti_join(<span class="pl-smi">.data.correlated</span>) %<span class="pl-k">&gt;</span>%
    spread(<span class="pl-smi">sample</span>, <span class="pl-smi">rc</span>, <span class="pl-k">-</span> <span class="pl-smi">transcript</span>) %<span class="pl-k">&gt;</span>%
    left_join(<span class="pl-smi">annotation</span>)</pre>
                     </div>
                    </div>
                    <div dir="auto">
                    </div>
                    <p dir="auto">
                     We can visualise how the reduced redundancy with the reduced dimentions
look like
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_SE.norm.non_redundant %&gt;%
    pivot_sample() %&gt;%
    ggplot(aes(x=`Dim1`, y=`Dim2`, color=`Cell.type`)) +
  geom_point() +
  my_theme" dir="auto">
                     <pre><span class="pl-smi">counts_SE.norm.non_redundant</span> %<span class="pl-k">&gt;</span>%
    pivot_sample() %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">`Dim1`</span>, <span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-smi">`Dim2`</span>, <span class="pl-v">color</span><span class="pl-k">=</span><span class="pl-smi">`Cell.type`</span>)) <span class="pl-k">+</span>
  geom_point() <span class="pl-k">+</span>
  <span class="pl-smi">my_theme</span></pre>
                    </div>
                    <p dir="auto">
                     <a href="/stemangiola/tidybulk/blob/master/man/figures/plot_drop-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/stemangiola/tidybulk/raw/master/man/figures/plot_drop-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <p dir="auto">
                     <strong>
                      Approach 2
                     </strong>
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts_SE.norm.non_redundant =
    counts_SE.norm.MDS %&gt;%
  remove_redundancy(
    method = "reduced_dimensions",
    Dim_a_column = `Dim1`,
    Dim_b_column = `Dim2`
  )' dir="auto">
                     <pre><span class="pl-v">counts_SE.norm.non_redundant</span> <span class="pl-k">=</span>
    <span class="pl-smi">counts_SE.norm.MDS</span> %<span class="pl-k">&gt;</span>%
  remove_redundancy(
    <span class="pl-v">method</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>reduced_dimensions<span class="pl-pds">"</span></span>,
    <span class="pl-v">Dim_a_column</span> <span class="pl-k">=</span> <span class="pl-smi">`Dim1`</span>,
    <span class="pl-v">Dim_b_column</span> <span class="pl-k">=</span> <span class="pl-smi">`Dim2`</span>
  )</pre>
                    </div>
                    <p dir="auto">
                     We can visualise MDS reduced dimensions of the samples with the closest
pair removed.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_SE.norm.non_redundant %&gt;%
    pivot_sample() %&gt;%
    ggplot(aes(x=`Dim1`, y=`Dim2`, color=`Cell.type`)) +
  geom_point() +
  my_theme" dir="auto">
                     <pre><span class="pl-smi">counts_SE.norm.non_redundant</span> %<span class="pl-k">&gt;</span>%
    pivot_sample() %<span class="pl-k">&gt;</span>%
    ggplot(aes(<span class="pl-v">x</span><span class="pl-k">=</span><span class="pl-smi">`Dim1`</span>, <span class="pl-v">y</span><span class="pl-k">=</span><span class="pl-smi">`Dim2`</span>, <span class="pl-v">color</span><span class="pl-k">=</span><span class="pl-smi">`Cell.type`</span>)) <span class="pl-k">+</span>
  geom_point() <span class="pl-k">+</span>
  <span class="pl-smi">my_theme</span></pre>
                    </div>
                    <p dir="auto">
                     <a href="/stemangiola/tidybulk/blob/master/man/figures/plot_drop2-1.png" rel="noopener noreferrer" target="_blank">
                      <img alt="" src="/stemangiola/tidybulk/raw/master/man/figures/plot_drop2-1.png" style="max-width: 100%;"/>
                     </a>
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      Other useful wrappers
                     </h2>
                     <a aria-label="Permalink: Other useful wrappers" class="anchor" href="#other-useful-wrappers" id="user-content-other-useful-wrappers">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     The above wrapper streamline the most common processing of bulk RNA
sequencing data. Other useful wrappers are listed above.
                    </p>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      From BAM/SAM to tibble of gene counts
                     </h2>
                     <a aria-label="Permalink: From BAM/SAM to tibble of gene counts" class="anchor" href="#from-bamsam-to-tibble-of-gene-counts" id="user-content-from-bamsam-to-tibble-of-gene-counts">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We can calculate gene counts (using FeatureCounts; Liao Y et al.,
10.1093/nar/gkz114) from a list of BAM/SAM files and format them into a
tidy structure (similar to counts).
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='counts = tidybulk_SAM_BAM(
    file_names,
    genome = "hg38",
    isPairedEnd = TRUE,
    requireBothEndsMapped = TRUE,
    checkFragLength = FALSE,
    useMetaFeatures = TRUE
)' dir="auto">
                     <pre><span class="pl-v">counts</span> <span class="pl-k">=</span> tidybulk_SAM_BAM(
    <span class="pl-smi">file_names</span>,
    <span class="pl-v">genome</span> <span class="pl-k">=</span> <span class="pl-s"><span class="pl-pds">"</span>hg38<span class="pl-pds">"</span></span>,
    <span class="pl-v">isPairedEnd</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
    <span class="pl-v">requireBothEndsMapped</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>,
    <span class="pl-v">checkFragLength</span> <span class="pl-k">=</span> <span class="pl-c1">FALSE</span>,
    <span class="pl-v">useMetaFeatures</span> <span class="pl-k">=</span> <span class="pl-c1">TRUE</span>
)</pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      From ensembl IDs to gene symbol IDs
                     </h2>
                     <a aria-label="Permalink: From ensembl IDs to gene symbol IDs" class="anchor" href="#from-ensembl-ids-to-gene-symbol-ids" id="user-content-from-ensembl-ids-to-gene-symbol-ids">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We can add gene symbols from ensembl identifiers. This is useful since
different resources use ensembl IDs while others use gene symbol IDs.
This currently works for human and mouse.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_ensembl %&gt;% ensembl_to_symbol(ens)" dir="auto">
                     <pre><span class="pl-smi">counts_ensembl</span> %<span class="pl-k">&gt;</span>% ensembl_to_symbol(<span class="pl-smi">ens</span>)</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## # A tibble: 119 × 8
##    ens             iso   `read count` sample     cases…¹ cases…² trans…³ ref_g…⁴
##    &lt;chr&gt;           &lt;chr&gt;        &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  
##  1 ENSG00000000003 13             144 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  2 ENSG00000000003 13              72 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  3 ENSG00000000003 13               0 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  4 ENSG00000000003 13            1099 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  5 ENSG00000000003 13              11 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  6 ENSG00000000003 13               2 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  7 ENSG00000000003 13               3 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  8 ENSG00000000003 13            2678 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  9 ENSG00000000003 13             751 TARGET-20… Acute … Primar… TSPAN6  hg38   
## 10 ENSG00000000003 13               1 TARGET-20… Acute … Primar… TSPAN6  hg38   
## # … with 109 more rows, and abbreviated variable names
## #   ¹​cases_0_project_disease_type, ²​cases_0_samples_0_sample_type, ³​transcript,
## #   ⁴​ref_genome
## # ℹ Use `print(n = ...)` to see more rows">
                     <pre class="notranslate"><code>## # A tibble: 119 × 8
##    ens             iso   `read count` sample     cases…¹ cases…² trans…³ ref_g…⁴
##    &lt;chr&gt;           &lt;chr&gt;        &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  
##  1 ENSG00000000003 13             144 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  2 ENSG00000000003 13              72 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  3 ENSG00000000003 13               0 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  4 ENSG00000000003 13            1099 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  5 ENSG00000000003 13              11 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  6 ENSG00000000003 13               2 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  7 ENSG00000000003 13               3 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  8 ENSG00000000003 13            2678 TARGET-20… Acute … Primar… TSPAN6  hg38   
##  9 ENSG00000000003 13             751 TARGET-20… Acute … Primar… TSPAN6  hg38   
## 10 ENSG00000000003 13               1 TARGET-20… Acute … Primar… TSPAN6  hg38   
## # … with 109 more rows, and abbreviated variable names
## #   ¹​cases_0_project_disease_type, ²​cases_0_samples_0_sample_type, ³​transcript,
## #   ⁴​ref_genome
## # ℹ Use `print(n = ...)` to see more rows
</code></pre>
                    </div>
                    <div class="markdown-heading" dir="auto">
                     <h2 class="heading-element" dir="auto" tabindex="-1">
                      From gene symbol to gene description (gene name in full)
                     </h2>
                     <a aria-label="Permalink: From gene symbol to gene description (gene name in full)" class="anchor" href="#from-gene-symbol-to-gene-description-gene-name-in-full" id="user-content-from-gene-symbol-to-gene-description-gene-name-in-full">
                      <svg aria-hidden="true" class="octicon octicon-link" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z">
                       </path>
                      </svg>
                     </a>
                    </div>
                    <p dir="auto">
                     We can add gene full name (and in future description) from symbol
identifiers. This currently works for human and mouse.
                    </p>
                    <div class="highlight highlight-source-r notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="counts_SE %&gt;% 
    describe_transcript() %&gt;% 
    select(feature, description, everything())" dir="auto">
                     <pre><span class="pl-smi">counts_SE</span> %<span class="pl-k">&gt;</span>% 
    describe_transcript() %<span class="pl-k">&gt;</span>% 
    select(<span class="pl-smi">feature</span>, <span class="pl-smi">description</span>, everything())</pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="## # A SummarizedExperiment-tibble abstraction: 408,624 × 48
## # �[90mFeatures=8513 | Samples=48 | Assays=count�[0m
##    feature  sample     count Cell.…¹ time  condi…² batch facto…³ descr…⁴ gene_…⁵
##    &lt;chr&gt;    &lt;chr&gt;      &lt;dbl&gt; &lt;fct&gt;   &lt;fct&gt; &lt;lgl&gt;   &lt;fct&gt; &lt;lgl&gt;   &lt;chr&gt;   &lt;chr&gt;  
##  1 A1BG     SRR1740034   153 b_cell  0 d   TRUE    0     TRUE    alpha-… A1BG   
##  2 A1BG-AS1 SRR1740034    83 b_cell  0 d   TRUE    0     TRUE    A1BG a… A1BG-A…
##  3 AAAS     SRR1740034   868 b_cell  0 d   TRUE    0     TRUE    aladin… AAAS   
##  4 AACS     SRR1740034   222 b_cell  0 d   TRUE    0     TRUE    acetoa… AACS   
##  5 AAGAB    SRR1740034   590 b_cell  0 d   TRUE    0     TRUE    alpha … AAGAB  
##  6 AAMDC    SRR1740034    48 b_cell  0 d   TRUE    0     TRUE    adipog… AAMDC  
##  7 AAMP     SRR1740034  1257 b_cell  0 d   TRUE    0     TRUE    angio … AAMP   
##  8 AANAT    SRR1740034   284 b_cell  0 d   TRUE    0     TRUE    aralky… AANAT  
##  9 AAR2     SRR1740034   379 b_cell  0 d   TRUE    0     TRUE    AAR2 s… AAR2   
## 10 AARS2    SRR1740034   685 b_cell  0 d   TRUE    0     TRUE    alanyl… AARS2  
## # … with 40 more rows, and abbreviated variable names ¹​Cell.type, ²​condition,
## #   ³​factor_of_interest, ⁴​description, ⁵​gene_name
## # ℹ Use `print(n = ...)` to see more rows">
                     <pre class="notranslate"><code>## # A SummarizedExperiment-tibble abstraction: 408,624 × 48
## # �[90mFeatures=8513 | Samples=48 | Assays=count�[0m
##    feature  sample     count Cell.…¹ time  condi…² batch facto…³ descr…⁴ gene_…⁵
##    &lt;chr&gt;    &lt;chr&gt;      &lt;dbl&gt; &lt;fct&gt;   &lt;fct&gt; &lt;lgl&gt;   &lt;fct&gt; &lt;lgl&gt;   &lt;chr&gt;   &lt;chr&gt;  
##  1 A1BG     SRR1740034   153 b_cell  0 d   TRUE    0     TRUE    alpha-… A1BG   
##  2 A1BG-AS1 SRR1740034    83 b_cell  0 d   TRUE    0     TRUE    A1BG a… A1BG-A…
##  3 AAAS     SRR1740034   868 b_cell  0 d   TRUE    0     TRUE    aladin… AAAS   
##  4 AACS     SRR1740034   222 b_cell  0 d   TRUE    0     TRUE    acetoa… AACS   
##  5 AAGAB    SRR1740034   590 b_cell  0 d   TRUE    0     TRUE    alpha … AAGAB  
##  6 AAMDC    SRR1740034    48 b_cell  0 d   TRUE    0     TRUE    adipog… AAMDC  
##  7 AAMP     SRR1740034  1257 b_cell  0 d   TRUE    0     TRUE    angio … AAMP   
##  8 AANAT    SRR1740034   284 b_cell  0 d   TRUE    0     TRUE    aralky… AANAT  
##  9 AAR2     SRR1740034   379 b_cell  0 d   TRUE    0     TRUE    AAR2 s… AAR2   
## 10 AARS2    SRR1740034   685 b_cell  0 d   TRUE    0     TRUE    alanyl… AARS2  
## # … with 40 more rows, and abbreviated variable names ¹​Cell.type, ²​condition,
## #   ³​factor_of_interest, ⁴​description, ⁵​gene_name
## # ℹ Use `print(n = ...)` to see more rows
</code></pre>
                    </div>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" id="react-codespace-csrf" type="hidden" value="3v0P8XA-DdrNwU3YFeNNLXWlRuGM7qFeivBJaw9TYyvJJZBe7DV1us6Nt6FDt9DjAx9rAmxH2sT7I23ACvD9CA">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <p class="f4 my-3">
                 Brings bulk and pseudobulk transcriptomics to the tidyverse
                </p>
                <h3 class="sr-only">
                 Topics
                </h3>
                <div class="my-3">
                 <div class="f6">
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:pipe" data-view-component="true" href="/topics/pipe" title="Topic: pipe">
                   pipe
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:tidy-data" data-view-component="true" href="/topics/tidy-data" title="Topic: tidy-data">
                   tidy-data
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:tidyverse" data-view-component="true" href="/topics/tidyverse" title="Topic: tidyverse">
                   tidyverse
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:pca" data-view-component="true" href="/topics/pca" title="Topic: pca">
                   pca
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:bioconductor" data-view-component="true" href="/topics/bioconductor" title="Topic: bioconductor">
                   bioconductor
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:deseq2" data-view-component="true" href="/topics/deseq2" title="Topic: deseq2">
                   deseq2
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:entrez" data-view-component="true" href="/topics/entrez" title="Topic: entrez">
                   entrez
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:tidy" data-view-component="true" href="/topics/tidy" title="Topic: tidy">
                   tidy
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:transcripts" data-view-component="true" href="/topics/transcripts" title="Topic: transcripts">
                   transcripts
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:transcriptomics" data-view-component="true" href="/topics/transcriptomics" title="Topic: transcriptomics">
                   transcriptomics
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:tsne" data-view-component="true" href="/topics/tsne" title="Topic: tsne">
                   tsne
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:differential-expression" data-view-component="true" href="/topics/differential-expression" title="Topic: differential-expression">
                   differential-expression
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:edger" data-view-component="true" href="/topics/edger" title="Topic: edger">
                   edger
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:redundancy" data-view-component="true" href="/topics/redundancy" title="Topic: redundancy">
                   redundancy
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:gsea" data-view-component="true" href="/topics/gsea" title="Topic: gsea">
                   gsea
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:tibble" data-view-component="true" href="/topics/tibble" title="Topic: tibble">
                   tibble
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:gene-symbols" data-view-component="true" href="/topics/gene-symbols" title="Topic: gene-symbols">
                   gene-symbols
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:bulk-transcriptional-analyses" data-view-component="true" href="/topics/bulk-transcriptional-analyses" title="Topic: bulk-transcriptional-analyses">
                   bulk-transcriptional-analyses
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:mds-dimensions" data-view-component="true" href="/topics/mds-dimensions" title="Topic: mds-dimensions">
                   mds-dimensions
                  </a>
                  <a class="topic-tag topic-tag-link" data-ga-click="Topic, repository page" data-octo-click="topic_click" data-octo-dimensions="topic:ensembl-ids" data-view-component="true" href="/topics/ensembl-ids" title="Topic: ensembl-ids">
                   ensembl-ids
                  </a>
                 </div>
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/stemangiola/tidybulk/hovercards/citation/sidebar_partial?tree_name=master">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/stemangiola/tidybulk/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/stemangiola/tidybulk/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   158
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/stemangiola/tidybulk/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   5
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/stemangiola/tidybulk/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   23
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2Fstemangiola%2Ftidybulk&amp;report=stemangiola+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/stemangiola/tidybulk/releases">
                 Releases
                 <span class="Counter" data-view-component="true" title="15">
                  15
                 </span>
                </a>
               </h2>
               <a class="Link--primary d-flex no-underline" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/stemangiola/tidybulk/releases/tag/v1.11.7">
                <svg aria-hidden="true" class="octicon octicon-tag flex-shrink-0 mt-1 color-fg-success" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                 </path>
                </svg>
                <div class="ml-2 min-width-0">
                 <div class="d-flex">
                  <span class="css-truncate css-truncate-target text-bold mr-2" style="max-width: none;">
                   v1.11.7
                  </span>
                  <span class="Label Label--success flex-shrink-0" data-view-component="true" title="Label: Latest">
                   Latest
                  </span>
                 </div>
                 <div class="text-small color-fg-muted">
                  <relative-time class="no-wrap" datetime="2023-08-31T00:53:51Z">
                   Aug 31, 2023
                  </relative-time>
                 </div>
                </div>
               </a>
               <div class="mt-3" data-view-component="true">
                <a class="Link" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/stemangiola/tidybulk/releases" text="small">
                 + 14 releases
                </a>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/stemangiola/tidybulk/used_by_list">
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/stemangiola/tidybulk/graphs/contributors">
                 Contributors
                 <span class="Counter ml-1" data-view-component="true" title="9">
                  9
                 </span>
                </a>
               </h2>
               <include-fragment aria-busy="true" aria-label="Loading contributors" src="/stemangiola/tidybulk/contributors_list?count=9&amp;current_repository=tidybulk&amp;items_to_show=9">
                <ul class="list-style-none d-flex flex-wrap mb-n2">
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                 <li class="mb-2">
                  <div class="Skeleton avatar avatar-user mr-2" style="width:32px;height:32px;">
                  </div>
                 </li>
                </ul>
               </include-fragment>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/stemangiola/tidybulk/deployments">
                 Deployments
                 <span class="Counter ml-1" data-view-component="true" title="74">
                  74
                 </span>
                </a>
               </h2>
               <ul class="list-style-none">
                <li class="mt-2">
                 <include-fragment aria-label="Loading github-pages deployment status." src="/stemangiola/tidybulk/environment_status?environment=github-pages">
                  <div class="Skeleton Skeleton--text">
                   <span class="Label Label--warning" data-view-component="true" title="Deployment Status Label: Queued">
                    Queued
                   </span>
                   github-pages
                   <span class="color-fg-subtle" data-view-component="true">
                    queued
                   </span>
                  </div>
                 </include-fragment>
                </li>
               </ul>
               <div class="mt-3" data-view-component="true">
                <a class="Link--inTextBlock Link" data-view-component="true" href="/stemangiola/tidybulk/deployments" text="small">
                 + 73 deployments
                </a>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                Languages
               </h2>
               <div class="mb-2">
                <span class="Progress" data-view-component="true">
                 <span aria-label="R 100.0" class="Progress-item color-bg-success-emphasis" data-view-component="true" itemprop="keywords" style="background-color:#198CE7 !important;;width: 100.0%;">
                 </span>
                </span>
               </div>
               <ul class="list-style-none">
                <li class="d-inline">
                 <a class="d-inline-flex flex-items-center flex-nowrap Link--secondary no-underline text-small mr-3" data-ga-click="Repository, language stats search click, location:repo overview" href="/stemangiola/tidybulk/search?l=r">
                  <svg aria-hidden="true" class="octicon octicon-dot-fill mr-2" data-view-component="true" height="16" style="color:#198CE7;" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                   </path>
                  </svg>
                  <span class="color-fg-default text-bold mr-1">
                   R
                  </span>
                  <span>
                   100.0%
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="false" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <style>
    .user-mention[href$="/ninenia"] {
        color: var(--color-user-mention-fg);
        background-color: var(--color-user-mention-bg);
        border-radius: 2px;
        margin-left: -2px;
        margin-right: -2px;
        padding: 0 2px;
      }
   </style>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
