<!DOCTYPE html>
<html data-a11y-animated-images="system" data-a11y-link-underlines="true" data-color-mode="auto" data-dark-theme="dark" data-light-theme="light" lang="en">
 <head>
  <meta charset="utf-8"/>
  <link href="https://github.githubassets.com" rel="dns-prefetch"/>
  <link href="https://avatars.githubusercontent.com" rel="dns-prefetch"/>
  <link href="https://github-cloud.s3.amazonaws.com" rel="dns-prefetch"/>
  <link href="https://user-images.githubusercontent.com/" rel="dns-prefetch"/>
  <link crossorigin="" href="https://github.githubassets.com" rel="preconnect"/>
  <link href="https://avatars.githubusercontent.com" rel="preconnect"/>
  <link crossorigin="anonymous" href="https://github.githubassets.com/assets/light_v2-e830ce163aae.css" media="all" rel="stylesheet">
   <link crossorigin="anonymous" href="https://github.githubassets.com/assets/dark_v2-bc3c552285ea.css" media="all" rel="stylesheet">
    <link crossorigin="anonymous" data-color-theme="dark_dimmed" data-href="https://github.githubassets.com/assets/dark_dimmed_v2-191796a86a54.css" media="all" rel="stylesheet">
     <link crossorigin="anonymous" data-color-theme="dark_high_contrast" data-href="https://github.githubassets.com/assets/dark_high_contrast_v2-e0a0b6e42616.css" media="all" rel="stylesheet">
      <link crossorigin="anonymous" data-color-theme="dark_colorblind" data-href="https://github.githubassets.com/assets/dark_colorblind_v2-4fd40c9324e8.css" media="all" rel="stylesheet">
       <link crossorigin="anonymous" data-color-theme="light_colorblind" data-href="https://github.githubassets.com/assets/light_colorblind_v2-2f299c61eceb.css" media="all" rel="stylesheet">
        <link crossorigin="anonymous" data-color-theme="light_high_contrast" data-href="https://github.githubassets.com/assets/light_high_contrast_v2-05d723322773.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="light_tritanopia" data-href="https://github.githubassets.com/assets/light_tritanopia_v2-5679d6101565.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" data-color-theme="dark_tritanopia" data-href="https://github.githubassets.com/assets/dark_tritanopia_v2-0ff66d76cc55.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-primitives-366b5c973fad.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/primer-f3607eccaaae.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/global-bac48eb8df38.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/github-19c85be4af9c.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/repository-6247ca238fd4.css" media="all" rel="stylesheet"/>
        <link crossorigin="anonymous" href="https://github.githubassets.com/assets/code-111be5e4092d.css" media="all" rel="stylesheet"/>
        <script id="client-env" type="application/json">
         {"locale":"en","featureFlags":["code_vulnerability_scanning","copilot_conversational_ux_history_refs","copilot_smell_icebreaker_ux","copilot_implicit_context","failbot_handle_non_errors","geojson_azure_maps","image_metric_tracking","marketing_forms_api_integration_contact_request","marketing_pages_search_explore_provider","repository_suggester_elastic_search","turbo_experiment_risky","sample_network_conn_type","no_character_key_shortcuts_in_inputs","react_start_transition_for_navigations","custom_inp","remove_child_patch","site_features_copilot_cli_ga","copilot_code_chat_diff_header_button"]}
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/wp-runtime-358aabcf8669.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_dompurify_dist_purify_js-6890e890956f.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_stacktrace-parser_dist_stack-trace-parser_esm_js-node_modules_github_bro-a4c183-79f9611c275b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_oddbird_popover-polyfill_dist_popover_js-7bd350d761f4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_failbot_failbot_ts-5bd9ba639cc0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/environment-27057bd9ed0b.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_selector-observer_dist_index_esm_js-9f960d9b217c.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_focus-zone_js-086f7a27bac0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_relative-time-element_dist_index_js-c76945c5961a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_combobox-nav_dist_index_js-node_modules_github_markdown-toolbar-e-820fc0-bc8f02b96749.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_auto-complete-element-81d69b-d1813ba335d8.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_text-expander-element_dist_index_js-8a621df59e80.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_filter-input-element_dist_index_js-node_modules_github_remote-inp-b7d8f4-654130b7cde5.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_primer_view-co-3959a9-68b3d6c8feb2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/github-elements-369bd99876f6.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/element-registry-f00a3bf302c1.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_index_js-node_modules_github_alive-client_dist-bf5aa2-5a0e291a0298.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_lit-html_lit-html_js-5b376145beff.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_morphdom_dist_morphdom-esm_js-5bff297a06de.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_turbo_dist_turbo_es2017-esm_js-c91f4ad18b62.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_scroll-anchoring_dist_scro-52dc4b-4fecca2d00e4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_color-convert_index_js-72c9fbde5ad4.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_behaviors_dist_esm_dimensions_js-node_modules_github_jtml_lib_index_js-95b84ee6bc34.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_quote-selection_dist_index_js-node_modules_github_session-resume_-ff65ee-c202d20e2d3d.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_updatable-content_ts-ee3fc84d7fb0.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_task-list_ts-app_assets_modules_github_onfocus_ts-app_ass-421cec-9de4213015af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_sticky-scroll-into-view_ts-94209c43e6af.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_ajax-error_ts-app_assets_modules_github_behaviors_include-467754-244ee9d9ed77.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_behaviors_commenting_edit_ts-app_assets_modules_github_behaviors_ht-83c235-9285faa0e011.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/behaviors-119a649bdc24.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_delegated-events_dist_index_js-node_modules_github_catalyst_lib_index_js-d0256ebff5cd.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/notifications-global-352d84c6cc82.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_virtualized-list_es_index_js-node_modules_github_template-parts_lib_index_js-878844713bc9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_remote-form_dist_index_js-node_modules_delegated-events_dist_inde-c537341-b93616e2fc14.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_ref-selector_ts-2b432e185ab2.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/codespaces-1a8626dd714a.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_mini-throttle_dist_decorators_js-node_modules_github_remote-form_-01f9fa-075cdbe1c5e3.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_file-attachment-element_dist_index_js-node_modules_github_filter--b2311f-4c891ec4eeb9.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repositories-b481106ce166.js" type="application/javascript">
        </script>
        <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/code-menu-1ede15b5453d.js" type="application/javascript">
        </script>
        <title>
         Sumeet460/Fear-Conditioning-Genomics
        </title>
        <meta content="/:user_id/:repository" data-turbo-transient="" name="route-pattern"/>
        <meta content="files" data-turbo-transient="" name="route-controller"/>
        <meta content="disambiguate" data-turbo-transient="" name="route-action"/>
        <meta content="82c569b93da5c18ed649ebd4c2c79437db4611a6a1373e805a3cb001c64130b7" name="current-catalog-service-hash"/>
        <meta content="D1B8:323CA:79326F:854DAB:66051AD4" data-turbo-transient="true" name="request-id">
         <meta content="22f598a296c639e8b750dc2cf1a20323836185b0d8b152a2e56d5461ad76529a" data-turbo-transient="true" name="html-safe-nonce">
          <meta content="eyJyZWZlcnJlciI6bnVsbCwicmVxdWVzdF9pZCI6IkQxQjg6MzIzQ0E6NzkzMjZGOjg1NERBQjo2NjA1MUFENCIsInZpc2l0b3JfaWQiOiI1NTYzNjg1OTkxMDg0ODI5Mjg2IiwicmVnaW9uX2VkZ2UiOiJzb3V0aGVhc3Rhc2lhIiwicmVnaW9uX3JlbmRlciI6ImlhZCJ9" data-turbo-transient="true" name="visitor-payload">
           <meta content="ec337ee1668c7fc5c9a89828ced1c7943077f6f3a343ff2e2b061254d371ca22" data-turbo-transient="true" name="visitor-hmac">
            <meta content="repository:45267387" data-turbo-transient="" name="hovercard-subject-tag"/>
            <meta content="repository,copilot" data-turbo-transient="true" name="github-keyboard-shortcuts">
             <meta data-turbo-transient="" name="selected-link" value="repo_source"/>
             <link href="https://github.githubassets.com/" rel="assets"/>
             <meta content="c1kuD-K2HIVF635lypcsWPoD4kilo5-jA_wBFyT4uMY" name="google-site-verification"/>
             <meta content="KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU" name="google-site-verification"/>
             <meta content="ZzhVyEFwb7w3e0-uOTltm8Jsck2F5StVihD0exw2fsA" name="google-site-verification"/>
             <meta content="GXs5KoUUkNCoaAZn7wPN-t01Pywp9M3sEjnt_3_ZWPc" name="google-site-verification"/>
             <meta content="Apib7-x98H0j5cPqHWwSMm6dNU4GmODRoqxLiDzdx9I" name="google-site-verification"/>
             <meta content="https://collector.github.com/github/collect" name="octolytics-url">
              <meta content="126681664" name="octolytics-actor-id">
               <meta content="ninenia" name="octolytics-actor-login">
                <meta content="57d6abdd600494be8ac2f82b25b957e4c0d029a4d648c10497790171de0492de" name="octolytics-actor-hash">
                 <meta content="/&lt;user-name&gt;/&lt;repo-name&gt;" data-turbo-transient="true" name="analytics-location">
                  <meta content="ninenia" name="user-login"/>
                  <link href="/sessions/sudo_modal" rel="sudo-modal"/>
                  <meta content="width=device-width" name="viewport"/>
                  <meta content="Contribute to Sumeet460/Fear-Conditioning-Genomics development by creating an account on GitHub." name="description"/>
                  <link href="/opensearch.xml" rel="search" title="GitHub" type="application/opensearchdescription+xml"/>
                  <link href="https://github.com/fluidicon.png" rel="fluid-icon" title="GitHub"/>
                  <meta content="1401488693436528" property="fb:app_id"/>
                  <meta content="app-id=1477376905, app-argument=https://github.com/Sumeet460/Fear-Conditioning-Genomics" name="apple-itunes-app">
                   <meta content="https://opengraph.githubassets.com/1c3e88b2c21f2c320ac47a0c3252e61b02205417929acd911ae4aa7a3b9244f5/Sumeet460/Fear-Conditioning-Genomics" name="twitter:image:src">
                    <meta content="@github" name="twitter:site">
                     <meta content="summary_large_image" name="twitter:card">
                      <meta content="Sumeet460/Fear-Conditioning-Genomics" name="twitter:title">
                       <meta content="Contribute to Sumeet460/Fear-Conditioning-Genomics development by creating an account on GitHub." name="twitter:description">
                        <meta content="https://opengraph.githubassets.com/1c3e88b2c21f2c320ac47a0c3252e61b02205417929acd911ae4aa7a3b9244f5/Sumeet460/Fear-Conditioning-Genomics" property="og:image"/>
                        <meta content="Contribute to Sumeet460/Fear-Conditioning-Genomics development by creating an account on GitHub." property="og:image:alt"/>
                        <meta content="1200" property="og:image:width"/>
                        <meta content="600" property="og:image:height"/>
                        <meta content="GitHub" property="og:site_name"/>
                        <meta content="object" property="og:type"/>
                        <meta content="Sumeet460/Fear-Conditioning-Genomics" property="og:title"/>
                        <meta content="https://github.com/Sumeet460/Fear-Conditioning-Genomics" property="og:url"/>
                        <meta content="Contribute to Sumeet460/Fear-Conditioning-Genomics development by creating an account on GitHub." property="og:description"/>
                        <link data-refresh-url="/_alive" data-session-id="fae53153a7a7c8d08a43b9810520b9ce4325ce8b4e99a06c5fc3f52c85cc22fe" href="wss://alive.github.com/_sockets/u/126681664/ws?session=eyJ2IjoiVjMiLCJ1IjoxMjY2ODE2NjQsInMiOjEzMDQyODIxMTYsImMiOjIzODY2NDQ2NDgsInQiOjE3MTE2MTA1ODV9--8e00e1b01b52a582027ac943eba0566b0b182e195e604b70d9aa89b5762d3232" rel="shared-web-socket"/>
                        <link href="/assets-cdn/worker/socket-worker-9cc1149b224c.js" rel="shared-web-socket-src"/>
                        <meta content="github.com" name="hostname"/>
                        <meta content="all" name="keyboard-shortcuts-preference"/>
                        <meta content="github.com" name="expected-hostname"/>
                        <meta content="6ca657df46826de483413fac9c14cb327f0bcec7cdafdf04039354bca4236790" data-turbo-track="reload" http-equiv="x-pjax-version"/>
                        <meta content="f226bf37af9c33162063db3eb018fed7f088f86d0a20ca54c013fda96c7f2e05" data-turbo-track="reload" http-equiv="x-pjax-csp-version"/>
                        <meta content="401c342494272ee398dfb0fe4138a70b543d2bb230a3a34442ce5e819c1ec7ce" data-turbo-track="reload" http-equiv="x-pjax-css-version"/>
                        <meta content="2f265fcdc8f8b7e87f03b97de19c27c064e4e34637536b9120d7119cf377461e" data-turbo-track="reload" http-equiv="x-pjax-js-version"/>
                        <meta content="no-preview" data-turbo-transient="" name="turbo-cache-control"/>
                        <meta data-hydrostats="publish"/>
                        <meta content="github.com/Sumeet460/Fear-Conditioning-Genomics git https://github.com/Sumeet460/Fear-Conditioning-Genomics.git" name="go-import"/>
                        <meta content="8104884" name="octolytics-dimension-user_id">
                         <meta content="Sumeet460" name="octolytics-dimension-user_login">
                          <meta content="45267387" name="octolytics-dimension-repository_id">
                           <meta content="Sumeet460/Fear-Conditioning-Genomics" name="octolytics-dimension-repository_nwo">
                            <meta content="true" name="octolytics-dimension-repository_public">
                             <meta content="false" name="octolytics-dimension-repository_is_fork">
                              <meta content="45267387" name="octolytics-dimension-repository_network_root_id">
                               <meta content="Sumeet460/Fear-Conditioning-Genomics" name="octolytics-dimension-repository_network_root_nwo">
                                <link data-turbo-transient="" href="https://github.com/Sumeet460/Fear-Conditioning-Genomics" rel="canonical"/>
                                <meta content="logged-in env-production page-responsive" name="turbo-body-classes"/>
                                <meta content="https://api.github.com/_private/browser/stats" name="browser-stats-url"/>
                                <meta content="https://api.github.com/_private/browser/errors" name="browser-errors-url"/>
                                <link color="#000000" href="https://github.githubassets.com/assets/pinned-octocat-093da3e6fa40.svg" rel="mask-icon"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.png" rel="alternate icon" type="image/png"/>
                                <link class="js-site-favicon" href="https://github.githubassets.com/favicons/favicon.svg" rel="icon" type="image/svg+xml"/>
                                <meta content="#1e2327" name="theme-color"/>
                                <meta content="light dark" name="color-scheme">
                                 <meta content="/windows-tile.png" name="msapplication-TileImage"/>
                                 <meta content="#ffffff" name="msapplication-TileColor"/>
                                 <link crossorigin="use-credentials" href="/manifest.json" rel="manifest"/>
                                </meta>
                               </meta>
                              </meta>
                             </meta>
                            </meta>
                           </meta>
                          </meta>
                         </meta>
                        </meta>
                       </meta>
                      </meta>
                     </meta>
                    </meta>
                   </meta>
                  </meta>
                 </meta>
                </meta>
               </meta>
              </meta>
             </meta>
            </meta>
           </meta>
          </meta>
         </meta>
        </meta>
       </link>
      </link>
     </link>
    </link>
   </link>
  </link>
 </head>
 <body class="logged-in env-production page-responsive" style="word-wrap: break-word;">
  <div class="logged-in env-production page-responsive" data-turbo-body="" style="word-wrap: break-word;">
   <div class="position-relative js-header-wrapper">
    <a class="p-3 color-bg-accent-emphasis color-fg-on-emphasis show-on-focus js-skip-to-content" href="#start-of-content">
     Skip to content
    </a>
    <span class="progress-pjax-loader Progress position-fixed width-full" data-view-component="true">
     <span class="Progress-item progress-pjax-loader-bar left-0 top-0 color-bg-accent-emphasis" data-view-component="true" style="width: 0%;">
     </span>
    </span>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_IconButton_js-node_modules_primer_react_lib--23bcad-01764c79fa41.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/keyboard-shortcuts-dialog-ae7cd7fa1372.js" type="application/javascript">
    </script>
    <react-partial data-ssr="false" partial-name="keyboard-shortcuts-dialog">
     <script data-target="react-partial.embeddedData" type="application/json">
      {"props":{}}
     </script>
     <div data-target="react-partial.reactRoot">
     </div>
    </react-partial>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_allex_crc32_lib_crc32_esm_js-node_modules_github_mini-throttle_dist_deco-b38cad-748e74df23ce.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_clipboard-copy-element_dist_index_esm_js-node_modules_delegated-e-b37f7d-2f24d321a3fb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_github_command-palette_items_help-item_ts-app_assets_modules_github_comman-48ad9d-9041b775dcfb.js" type="application/javascript">
    </script>
    <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/command-palette-f620e569db0b.js" type="application/javascript">
    </script>
    <header class="AppHeader">
     <div class="AppHeader-globalBar pb-2 js-global-bar">
      <div class="AppHeader-globalBar-start">
       <deferred-side-panel data-url="/_side-panels/global">
        <include-fragment data-target="deferred-side-panel.fragment">
         <button aria-label="Open global navigation menu" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-bg-transparent p-0 color-fg-muted" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-3b23d566-af18-4c54-8546-cf4e0d95f844" data-view-component="true" id="dialog-show-dialog-3b23d566-af18-4c54-8546-cf4e0d95f844" type="button">
          <svg aria-hidden="true" class="octicon octicon-three-bars Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1 2.75A.75.75 0 0 1 1.75 2h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 2.75Zm0 5A.75.75 0 0 1 1.75 7h12.5a.75.75 0 0 1 0 1.5H1.75A.75.75 0 0 1 1 7.75ZM1.75 12h12.5a.75.75 0 0 1 0 1.5H1.75a.75.75 0 0 1 0-1.5Z">
           </path>
          </svg>
         </button>
         <dialog-helper>
          <dialog aria-describedby="dialog-3b23d566-af18-4c54-8546-cf4e0d95f844-description" aria-labelledby="dialog-3b23d566-af18-4c54-8546-cf4e0d95f844-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-left SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-3b23d566-af18-4c54-8546-cf4e0d95f844">
           <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title sr-only" id="dialog-3b23d566-af18-4c54-8546-cf4e0d95f844-title">
               Global navigation
              </h1>
              <div class="d-flex" data-view-component="true">
               <div class="AppHeader-logo position-relative" data-view-component="true">
                <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
                 <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
                 </path>
                </svg>
               </div>
              </div>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-3b23d566-af18-4c54-8546-cf4e0d95f844" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="dialog-3b23d566-af18-4c54-8546-cf4e0d95f844-title">
            <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
             <div class="d-flex flex-column mb-3" data-view-component="true">
              <nav aria-label="Site navigation" class="ActionList" data-view-component="true">
               <nav-list>
                <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"HOME","label":null}' data-hotkey="g d" data-view-component="true" href="/dashboard" id="item-77ea7d91-3af4-4090-b9b0-4ccddc2d3bdf">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-home" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M6.906.664a1.749 1.749 0 0 1 2.187 0l5.25 4.2c.415.332.657.835.657 1.367v7.019A1.75 1.75 0 0 1 13.25 15h-3.5a.75.75 0 0 1-.75-.75V9H7v5.25a.75.75 0 0 1-.75.75h-3.5A1.75 1.75 0 0 1 1 13.25V6.23c0-.531.242-1.034.657-1.366l5.25-4.2Zm1.25 1.171a.25.25 0 0 0-.312 0l-5.25 4.2a.25.25 0 0 0-.094.196v7.019c0 .138.112.25.25.25H5.5V8.25a.75.75 0 0 1 .75-.75h3.5a.75.75 0 0 1 .75.75v5.25h2.75a.25.25 0 0 0 .25-.25V6.23a.25.25 0 0 0-.094-.195Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Home
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"ISSUES","label":null}' data-hotkey="g i" data-view-component="true" href="/issues" id="item-28e72866-6cb5-45af-a530-0b02bd0d9302">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                     </path>
                     <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Issues
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS","label":null}' data-hotkey="g p" data-view-component="true" href="/pulls" id="item-43bfdd4c-d2f4-4692-96dc-2ee621b0c660">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Pull requests
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="projects" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROJECTS","label":null}' data-view-component="true" href="/projects" id="item-53784535-a6d2-49fd-be9c-015b84a55ade">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Projects
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"DISCUSSIONS","label":null}' data-view-component="true" href="/discussions" id="item-12f067aa-7f8c-4be5-982f-7cf125a0f5bc">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Discussions
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"CODESPACES","label":null}' data-view-component="true" href="https://github.com/codespaces" id="item-dc169de2-6506-40cf-a89a-c881bafe7fb3">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                     </path>
                     <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Codespaces
                   </span>
                  </a>
                 </li>
                 <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"EXPLORE","label":null}' data-view-component="true" href="/explore" id="item-d1849440-385f-442c-885a-14a4d4ffded8">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-telescope" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M14.184 1.143v-.001l1.422 2.464a1.75 1.75 0 0 1-.757 2.451L3.104 11.713a1.75 1.75 0 0 1-2.275-.702l-.447-.775a1.75 1.75 0 0 1 .53-2.32L11.682.573a1.748 1.748 0 0 1 2.502.57Zm-4.709 9.32h-.001l2.644 3.863a.75.75 0 1 1-1.238.848l-1.881-2.75v2.826a.75.75 0 0 1-1.5 0v-2.826l-1.881 2.75a.75.75 0 1 1-1.238-.848l2.049-2.992a.746.746 0 0 1 .293-.253l1.809-.87a.749.749 0 0 1 .944.252ZM9.436 3.92h-.001l-4.97 3.39.942 1.63 5.42-2.61Zm3.091-2.108h.001l-1.85 1.26 1.505 2.605 2.016-.97a.247.247 0 0 0 .13-.151.247.247 0 0 0-.022-.199l-1.422-2.464a.253.253 0 0 0-.161-.119.254.254 0 0 0-.197.038ZM1.756 9.157a.25.25 0 0 0-.075.33l.447.775a.25.25 0 0 0 .325.1l1.598-.769-.83-1.436-1.465 1Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Explore
                   </span>
                  </a>
                 </li>
                 <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                  <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"MARKETPLACE","label":null}' data-view-component="true" href="/marketplace" id="item-ca1d26ca-88e2-4ce1-b6b5-317b9263a1de">
                   <span class="ActionListItem-visual ActionListItem-visual--leading">
                    <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                     </path>
                    </svg>
                   </span>
                   <span class="ActionListItem-label" data-view-component="true">
                    Marketplace
                   </span>
                  </a>
                 </li>
                </ul>
               </nav-list>
              </nav>
              <div class="my-3 d-flex flex-justify-center height-full" data-view-component="true">
               <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                </circle>
                <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                </path>
               </svg>
              </div>
             </div>
             <div class="flex-1" data-view-component="true">
             </div>
             <div class="px-2" data-view-component="true">
              <p class="color-fg-subtle text-small text-light">
               © 2024 GitHub, Inc.
              </p>
              <div class="d-flex flex-wrap text-small text-light" data-view-component="true">
               <a class="Link mr-2" data-view-component="true" href="https://github.com/about" target="_blank">
                About
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.blog" target="_blank">
                Blog
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service" target="_blank">
                Terms
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement" target="_blank">
                Privacy
               </a>
               <a class="Link mr-2" data-view-component="true" href="https://github.com/security" target="_blank">
                Security
               </a>
               <a class="Link mr-3" data-view-component="true" href="https://www.githubstatus.com/" target="_blank">
                Status
               </a>
              </div>
             </div>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </include-fragment>
       </deferred-side-panel>
       <a aria-label="Homepage " class="AppHeader-logo ml-2" data-analytics-event='{"category":"Header","action":"go to dashboard","label":"icon:logo"}' data-hotkey="g d" data-turbo="false" href="https://github.com/">
        <svg aria-hidden="true" class="octicon octicon-mark-github v-align-middle color-fg-default" data-view-component="true" height="32" version="1.1" viewbox="0 0 16 16" width="32">
         <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
         </path>
        </svg>
       </a>
       <div class="AppHeader-context">
        <div class="AppHeader-context-compact">
         <button aria-expanded="false" aria-haspopup="dialog" aria-label="Page context: Sumeet460 / Fear-Conditioning-Genomics" class="AppHeader-context-compact-trigger Truncate Button--secondary Button--medium Button box-shadow-none" data-show-dialog-id="context-region-dialog" data-view-component="true" id="dialog-show-context-region-dialog" type="button">
          <span class="Button-content">
           <span class="Button-label">
            <span class="AppHeader-context-compact-lead">
             <span class="AppHeader-context-compact-parentItem">
              Sumeet460
             </span>
             <span class="no-wrap">
              /
             </span>
            </span>
            <strong class="AppHeader-context-compact-mainItem d-flex flex-items-center Truncate">
             <span class="Truncate-text">
              Fear-Conditioning-Genomics
             </span>
            </strong>
           </span>
          </span>
         </button>
         <dialog-helper>
          <dialog aria-describedby="context-region-dialog-description" aria-labelledby="context-region-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-view-component="true" id="context-region-dialog">
           <div class="Overlay-header" data-view-component="true">
            <div class="Overlay-headerContentWrap">
             <div class="Overlay-titleWrap">
              <h1 class="Overlay-title" id="context-region-dialog-title">
               Navigate back to
              </h1>
             </div>
             <div class="Overlay-actionWrap">
              <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="context-region-dialog" data-view-component="true" type="button">
               <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                </path>
               </svg>
              </button>
             </div>
            </div>
           </div>
           <scrollable-region data-labelled-by="context-region-dialog-title">
            <div class="Overlay-body" data-view-component="true">
             <ul class="list-style-none" role="list">
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"Sumeet460","screen_size":"compact"}' data-view-component="true" href="/Sumeet460">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-person mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                  </path>
                 </svg>
                 Sumeet460
                </span>
               </a>
              </li>
              <li>
               <a class="Link--primary Truncate d-flex flex-items-center py-1" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"Fear-Conditioning-Genomics","screen_size":"compact"}' data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics">
                <span class="AppHeader-context-item-label Truncate-text">
                 <svg aria-hidden="true" class="octicon octicon-repo mr-1" data-view-component="true" height="12" version="1.1" viewbox="0 0 16 16" width="12">
                  <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                  </path>
                 </svg>
                 Fear-Conditioning-Genomics
                </span>
               </a>
              </li>
             </ul>
            </div>
           </scrollable-region>
          </dialog>
         </dialog-helper>
        </div>
        <div class="AppHeader-context-full">
         <nav aria-label="Page context" role="navigation">
          <ul class="list-style-none" role="list">
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"Sumeet460","screen_size":"full"}' data-hovercard-type="user" data-hovercard-url="/users/Sumeet460/hovercard" data-octo-click="hovercard-link-click" data-octo-dimensions="link_type:self" data-view-component="true" href="/Sumeet460">
             <span class="AppHeader-context-item-label">
              Sumeet460
             </span>
            </a>
            <span class="AppHeader-context-item-separator">
             /
            </span>
           </li>
           <li>
            <a class="AppHeader-context-item" data-analytics-event='{"category":"SiteHeaderComponent","action":"context_region_crumb","label":"Fear-Conditioning-Genomics","screen_size":"full"}' data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics">
             <span class="AppHeader-context-item-label">
              Fear-Conditioning-Genomics
             </span>
            </a>
           </li>
          </ul>
         </nav>
        </div>
       </div>
      </div>
      <div class="AppHeader-globalBar-end">
       <div class="AppHeader-search">
        <qbsearch-input class="search-input" data-blackbird-indexed-repo-csrf='&lt;input type="hidden" value="iJINzTzTieKdd-NcL--_mqQGgXriOciXdA0489z4jSTYsQFFrl5WLSrtu3KKeF3BkIpBkdYSfQaqdkZNcTwwzA" data-csrf="true" /&gt;' data-blackbird-suggestions-path="/search/suggestions" data-copilot-chat-enabled="false" data-current-org="" data-current-owner="Sumeet460" data-current-repository="Sumeet460/Fear-Conditioning-Genomics" data-custom-scopes-path="/search/custom_scopes" data-delete-custom-scopes-csrf="yNtXvslKA3D3yVPzBEc0PkIiZoDePc_B3Mrj5x0ycNqAEUCU7wxKBaBlISriQsQvEIYkKfhaP66wKKJM-gxrlg" data-header-redesign-enabled="true" data-initial-value="" data-jump-to-suggestions-path="/_graphql/GetSuggestedNavigationDestinations" data-logged-in="true" data-max-custom-scopes="10" data-scope="repo:Sumeet460/Fear-Conditioning-Genomics">
         <div class="search-input-container search-with-dialog position-relative d-flex flex-row flex-items-center height-auto color-bg-transparent border-0 color-fg-subtle mx-0" data-action="click:qbsearch-input#searchInputContainerClicked">
          <button aria-expanded="false" aria-haspopup="dialog" aria-label="Search or jump to…" class="AppHeader-button AppHeader-search-whenNarrow" data-action="click:qbsearch-input#handleExpand" type="button">
           <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
            </path>
           </svg>
          </button>
          <div class="AppHeader-search-whenRegular">
           <div class="AppHeader-search-wrap AppHeader-search-wrap--hasTrailing">
            <div class="AppHeader-search-control">
             <label aria-label="Search or jump to…" class="AppHeader-search-visual--leading" for="AppHeader-searchInput">
              <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
               </path>
              </svg>
             </label>
             <button aria-describedby="search-error-message-flash" class="AppHeader-searchButton form-control input-contrast text-left color-fg-subtle no-wrap" data-action="click:qbsearch-input#handleExpand" data-analytics-event='{"category":"Global navigation","action":"SEARCH","label":null}' data-hotkey="s,/" data-target="qbsearch-input.inputButton" type="button">
              <div class="overflow-hidden">
               <span data-target="qbsearch-input.inputButtonText" id="qb-input-query">
                Type
                <kbd class="AppHeader-search-kbd">
                 /
                </kbd>
                to search
               </span>
              </div>
             </button>
            </div>
            <button class="AppHeader-search-action--trailing js-activate-command-palette" data-analytics-event='{"category":"SiteHeaderComponent","action":"command_palette","label":"open command palette"}' id="AppHeader-commandPalette-button" type="button">
             <svg aria-hidden="true" class="octicon octicon-command-palette" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m6.354 8.04-4.773 4.773a.75.75 0 1 0 1.061 1.06L7.945 8.57a.75.75 0 0 0 0-1.06L2.642 2.206a.75.75 0 0 0-1.06 1.061L6.353 8.04ZM8.75 11.5a.75.75 0 0 0 0 1.5h5.5a.75.75 0 0 0 0-1.5h-5.5Z">
              </path>
             </svg>
            </button>
            <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="AppHeader-commandPalette-button" id="tooltip-94f0cc4a-7e5e-42fc-b490-058b19368f17" popover="manual">
             Command palette
            </tool-tip>
           </div>
          </div>
          <input class="js-site-search-type-field" name="type" type="hidden"/>
          <div class="Overlay--hidden" data-modal-dialog-overlay="">
           <modal-dialog aria-labelledby="search-suggestions-dialog-header" aria-modal="true" class="Overlay Overlay--width-medium Overlay--height-auto" data-action="close:qbsearch-input#handleClose cancel:qbsearch-input#handleClose" data-target="qbsearch-input.searchSuggestionsDialog" data-view-component="true" id="search-suggestions-dialog" role="dialog">
            <h1 class="sr-only" id="search-suggestions-dialog-header">
             Search code, repositories, users, issues, pull requests...
            </h1>
            <div class="Overlay-body Overlay-body--paddingNone">
             <div data-view-component="true">
              <div class="search-suggestions position-absolute width-full color-shadow-large border color-fg-default color-bg-default overflow-hidden d-flex flex-column query-builder-container" data-target="qbsearch-input.queryBuilderContainer" hidden="" style="border-radius: 12px;">
               <!-- '"` -->
               <!-- </textarea></xmp> -->
               <form accept-charset="UTF-8" action="" id="query-builder-test-form" method="get">
                <query-builder class="QueryBuilder search-query-builder" data-filter-key=":" data-target="qbsearch-input.queryBuilder" data-view-component="true" id="query-builder-query-builder-test">
                 <div class="FormControl FormControl--fullWidth">
                  <label class="FormControl-label sr-only" for="query-builder-test" id="query-builder-test-label">
                   Search
                  </label>
                  <div class="QueryBuilder-StyledInput width-fit" data-target="query-builder.styledInput">
                   <span class="FormControl-input-leadingVisualWrap QueryBuilder-leadingVisualWrap" id="query-builder-test-leadingvisual-wrap">
                    <svg aria-hidden="true" class="octicon octicon-search FormControl-input-leadingVisual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                     </path>
                    </svg>
                   </span>
                   <div class="QueryBuilder-StyledInputContainer" data-target="query-builder.styledInputContainer">
                    <div aria-hidden="true" class="QueryBuilder-StyledInputContent" data-target="query-builder.styledInputContent">
                    </div>
                    <div class="QueryBuilder-InputWrapper">
                     <div aria-hidden="true" class="QueryBuilder-Sizer" data-target="query-builder.sizer">
                     </div>
                     <input aria-describedby="validation-4bf775ec-b4ae-4890-b8a9-dfd54c14652f" aria-expanded="false" autocomplete="off" class="FormControl-input QueryBuilder-Input FormControl-medium" data-action="
          input:query-builder#inputChange
          blur:query-builder#inputBlur
          keydown:query-builder#inputKeydown
          focus:query-builder#inputFocus
        " data-target="query-builder.input" data-view-component="true" id="query-builder-test" name="query-builder-test" role="combobox" spellcheck="false" type="text" value="">
                     </input>
                    </div>
                   </div>
                   <span class="sr-only" id="query-builder-test-clear">
                    Clear
                   </span>
                   <button aria-labelledby="query-builder-test-clear query-builder-test-label" class="Button Button--iconOnly Button--invisible Button--medium mr-1 px-2 py-0 d-flex flex-items-center rounded-1 color-fg-muted" data-action="
                click:query-builder#clear
                focus:query-builder#clearButtonFocus
                blur:query-builder#clearButtonBlur
              " data-target="query-builder.clearButton" data-view-component="true" hidden="hidden" id="query-builder-test-clear-button" role="button" type="button" variant="small">
                    <svg aria-hidden="true" class="octicon octicon-x-circle-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                     <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <template id="search-icon">
                   <svg aria-hidden="true" class="octicon octicon-search" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-icon">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="file-code-icon">
                   <svg aria-hidden="true" class="octicon octicon-file-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M4 1.75C4 .784 4.784 0 5.75 0h5.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v8.586A1.75 1.75 0 0 1 14.25 15h-9a.75.75 0 0 1 0-1.5h9a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 10 4.25V1.5H5.75a.25.25 0 0 0-.25.25v2.5a.75.75 0 0 1-1.5 0Zm1.72 4.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.47-1.47-1.47-1.47a.75.75 0 0 1 0-1.06ZM3.28 7.78 1.81 9.25l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Zm8.22-6.218V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                    </path>
                   </svg>
                  </template>
                  <template id="history-icon">
                   <svg aria-hidden="true" class="octicon octicon-history" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                    </path>
                   </svg>
                  </template>
                  <template id="repo-icon">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="bookmark-icon">
                   <svg aria-hidden="true" class="octicon octicon-bookmark" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3 2.75C3 1.784 3.784 1 4.75 1h6.5c.966 0 1.75.784 1.75 1.75v11.5a.75.75 0 0 1-1.227.579L8 11.722l-3.773 3.107A.751.751 0 0 1 3 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v9.91l3.023-2.489a.75.75 0 0 1 .954 0l3.023 2.49V2.75a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="plus-circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-plus-circle" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm7.25-3.25v2.5h2.5a.75.75 0 0 1 0 1.5h-2.5v2.5a.75.75 0 0 1-1.5 0v-2.5h-2.5a.75.75 0 0 1 0-1.5h2.5v-2.5a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="circle-icon">
                   <svg aria-hidden="true" class="octicon octicon-dot-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 4a4 4 0 1 1 0 8 4 4 0 0 1 0-8Z">
                    </path>
                   </svg>
                  </template>
                  <template id="trash-icon">
                   <svg aria-hidden="true" class="octicon octicon-trash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="team-icon">
                   <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="project-icon">
                   <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                    </path>
                   </svg>
                  </template>
                  <template id="pencil-icon">
                   <svg aria-hidden="true" class="octicon octicon-pencil" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                    </path>
                   </svg>
                  </template>
                  <template id="copilot-icon">
                   <svg aria-hidden="true" class="octicon octicon-copilot" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.998 15.035c-4.562 0-7.873-2.914-7.998-3.749V9.338c.085-.628.677-1.686 1.588-2.065.013-.07.024-.143.036-.218.029-.183.06-.384.126-.612-.201-.508-.254-1.084-.254-1.656 0-.87.128-1.769.693-2.484.579-.733 1.494-1.124 2.724-1.261 1.206-.134 2.262.034 2.944.765.05.053.096.108.139.165.044-.057.094-.112.143-.165.682-.731 1.738-.899 2.944-.765 1.23.137 2.145.528 2.724 1.261.566.715.693 1.614.693 2.484 0 .572-.053 1.148-.254 1.656.066.228.098.429.126.612.012.076.024.148.037.218.924.385 1.522 1.471 1.591 2.095v1.872c0 .766-3.351 3.795-8.002 3.795Zm0-1.485c2.28 0 4.584-1.11 5.002-1.433V7.862l-.023-.116c-.49.21-1.075.291-1.727.291-1.146 0-2.059-.327-2.71-.991A3.222 3.222 0 0 1 8 6.303a3.24 3.24 0 0 1-.544.743c-.65.664-1.563.991-2.71.991-.652 0-1.236-.081-1.727-.291l-.023.116v4.255c.419.323 2.722 1.433 5.002 1.433ZM6.762 2.83c-.193-.206-.637-.413-1.682-.297-1.019.113-1.479.404-1.713.7-.247.312-.369.789-.369 1.554 0 .793.129 1.171.308 1.371.162.181.519.379 1.442.379.853 0 1.339-.235 1.638-.54.315-.322.527-.827.617-1.553.117-.935-.037-1.395-.241-1.614Zm4.155-.297c-1.044-.116-1.488.091-1.681.297-.204.219-.359.679-.242 1.614.091.726.303 1.231.618 1.553.299.305.784.54 1.638.54.922 0 1.28-.198 1.442-.379.179-.2.308-.578.308-1.371 0-.765-.123-1.242-.37-1.554-.233-.296-.693-.587-1.713-.7Z">
                    </path>
                    <path d="M6.25 9.037a.75.75 0 0 1 .75.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 .75-.75Zm4.25.75v1.501a.75.75 0 0 1-1.5 0V9.787a.75.75 0 0 1 1.5 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="workflow-icon">
                   <svg aria-hidden="true" class="octicon octicon-workflow" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h3.5C6.216 0 7 .784 7 1.75v3.5A1.75 1.75 0 0 1 5.25 7H4v4a1 1 0 0 0 1 1h4v-1.25C9 9.784 9.784 9 10.75 9h3.5c.966 0 1.75.784 1.75 1.75v3.5A1.75 1.75 0 0 1 14.25 16h-3.5A1.75 1.75 0 0 1 9 14.25v-.75H5A2.5 2.5 0 0 1 2.5 11V7h-.75A1.75 1.75 0 0 1 0 5.25Zm1.75-.25a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Zm9 9a.25.25 0 0 0-.25.25v3.5c0 .138.112.25.25.25h3.5a.25.25 0 0 0 .25-.25v-3.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="book-icon">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-review-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-review" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 13H8.061l-2.574 2.573A1.458 1.458 0 0 1 3 14.543V13H1.75A1.75 1.75 0 0 1 0 11.25v-8.5C0 1.784.784 1 1.75 1ZM1.5 2.75v8.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h6.5a.25.25 0 0 0 .25-.25v-8.5a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm5.28 1.72a.75.75 0 0 1 0 1.06L5.31 7l1.47 1.47a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-2-2a.75.75 0 0 1 0-1.06l2-2a.75.75 0 0 1 1.06 0Zm2.44 0a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L10.69 7 9.22 5.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </template>
                  <template id="codespaces-icon">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.75C1 1.784 1.784 1 2.75 1h10.5c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 13.25 12H9.06l-2.573 2.573A1.458 1.458 0 0 1 4 13.543V12H2.75A1.75 1.75 0 0 1 1 10.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h2a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h4.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="comment-discussion-icon">
                   <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                    </path>
                   </svg>
                  </template>
                  <template id="organization-icon">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="rocket-icon">
                   <svg aria-hidden="true" class="octicon octicon-rocket" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.064 0h.186C15.216 0 16 .784 16 1.75v.186a8.752 8.752 0 0 1-2.564 6.186l-.458.459c-.314.314-.641.616-.979.904v3.207c0 .608-.315 1.172-.833 1.49l-2.774 1.707a.749.749 0 0 1-1.11-.418l-.954-3.102a1.214 1.214 0 0 1-.145-.125L3.754 9.816a1.218 1.218 0 0 1-.124-.145L.528 8.717a.749.749 0 0 1-.418-1.11l1.71-2.774A1.748 1.748 0 0 1 3.31 4h3.204c.288-.338.59-.665.904-.979l.459-.458A8.749 8.749 0 0 1 14.064 0ZM8.938 3.623h-.002l-.458.458c-.76.76-1.437 1.598-2.02 2.5l-1.5 2.317 2.143 2.143 2.317-1.5c.902-.583 1.74-1.26 2.499-2.02l.459-.458a7.25 7.25 0 0 0 2.123-5.127V1.75a.25.25 0 0 0-.25-.25h-.186a7.249 7.249 0 0 0-5.125 2.123ZM3.56 14.56c-.732.732-2.334 1.045-3.005 1.148a.234.234 0 0 1-.201-.064.234.234 0 0 1-.064-.201c.103-.671.416-2.273 1.15-3.003a1.502 1.502 0 1 1 2.12 2.12Zm6.94-3.935c-.088.06-.177.118-.266.175l-2.35 1.521.548 1.783 1.949-1.2a.25.25 0 0 0 .119-.213ZM3.678 8.116 5.2 5.766c.058-.09.117-.178.176-.266H3.309a.25.25 0 0 0-.213.119l-1.2 1.95ZM12 5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="shield-check-icon">
                   <svg aria-hidden="true" class="octicon octicon-shield-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM11.28 6.28l-3.5 3.5a.75.75 0 0 1-1.06 0l-1.5-1.5a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l.97.97 2.97-2.97a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="heart-icon">
                   <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="server-icon">
                   <svg aria-hidden="true" class="octicon octicon-server" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 1h12.5c.966 0 1.75.784 1.75 1.75v4c0 .372-.116.717-.314 1 .198.283.314.628.314 1v4a1.75 1.75 0 0 1-1.75 1.75H1.75A1.75 1.75 0 0 1 0 12.75v-4c0-.358.109-.707.314-1a1.739 1.739 0 0 1-.314-1v-4C0 1.784.784 1 1.75 1ZM1.5 2.75v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25Zm.25 5.75a.25.25 0 0 0-.25.25v4c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-4a.25.25 0 0 0-.25-.25ZM7 4.75A.75.75 0 0 1 7.75 4h4.5a.75.75 0 0 1 0 1.5h-4.5A.75.75 0 0 1 7 4.75ZM7.75 10h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1 0-1.5ZM3 4.75A.75.75 0 0 1 3.75 4h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 4.75ZM3.75 10h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </template>
                  <template id="globe-icon">
                   <svg aria-hidden="true" class="octicon octicon-globe" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
                    </path>
                   </svg>
                  </template>
                  <template id="issue-opened-icon">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-mobile-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </template>
                  <template id="package-icon">
                   <svg aria-hidden="true" class="octicon octicon-package" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m8.878.392 5.25 3.045c.54.314.872.89.872 1.514v6.098a1.75 1.75 0 0 1-.872 1.514l-5.25 3.045a1.75 1.75 0 0 1-1.756 0l-5.25-3.045A1.75 1.75 0 0 1 1 11.049V4.951c0-.624.332-1.201.872-1.514L7.122.392a1.75 1.75 0 0 1 1.756 0ZM7.875 1.69l-4.63 2.685L8 7.133l4.755-2.758-4.63-2.685a.248.248 0 0 0-.25 0ZM2.5 5.677v5.372c0 .09.047.171.125.216l4.625 2.683V8.432Zm6.25 8.271 4.625-2.683a.25.25 0 0 0 .125-.216V5.677L8.75 8.432Z">
                    </path>
                   </svg>
                  </template>
                  <template id="credit-card-icon">
                   <svg aria-hidden="true" class="octicon octicon-credit-card" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M10.75 9a.75.75 0 0 0 0 1.5h1.5a.75.75 0 0 0 0-1.5h-1.5Z">
                    </path>
                    <path d="M0 3.75C0 2.784.784 2 1.75 2h12.5c.966 0 1.75.784 1.75 1.75v8.5A1.75 1.75 0 0 1 14.25 14H1.75A1.75 1.75 0 0 1 0 12.25ZM14.5 6.5h-13v5.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Zm0-2.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25V5h13Z">
                    </path>
                   </svg>
                  </template>
                  <template id="play-icon">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </template>
                  <template id="gift-icon">
                   <svg aria-hidden="true" class="octicon octicon-gift" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.75A2.75 2.75 0 0 1 4.75 0c.983 0 1.873.42 2.57 1.232.268.318.497.668.68 1.042.183-.375.411-.725.68-1.044C9.376.42 10.266 0 11.25 0a2.75 2.75 0 0 1 2.45 4h.55c.966 0 1.75.784 1.75 1.75v2c0 .698-.409 1.301-1 1.582v4.918A1.75 1.75 0 0 1 13.25 16H2.75A1.75 1.75 0 0 1 1 14.25V9.332C.409 9.05 0 8.448 0 7.75v-2C0 4.784.784 4 1.75 4h.55c-.192-.375-.3-.8-.3-1.25ZM7.25 9.5H2.5v4.75c0 .138.112.25.25.25h4.5Zm1.5 0v5h4.5a.25.25 0 0 0 .25-.25V9.5Zm0-4V8h5.5a.25.25 0 0 0 .25-.25v-2a.25.25 0 0 0-.25-.25Zm-7 0a.25.25 0 0 0-.25.25v2c0 .138.112.25.25.25h5.5V5.5h-5.5Zm3-4a1.25 1.25 0 0 0 0 2.5h2.309c-.233-.818-.542-1.401-.878-1.793-.43-.502-.915-.707-1.431-.707ZM8.941 4h2.309a1.25 1.25 0 0 0 0-2.5c-.516 0-1 .205-1.43.707-.337.392-.646.975-.879 1.793Z">
                    </path>
                   </svg>
                  </template>
                  <template id="code-square-icon">
                   <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </template>
                  <template id="device-desktop-icon">
                   <svg aria-hidden="true" class="octicon octicon-device-desktop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M14.25 1c.966 0 1.75.784 1.75 1.75v7.5A1.75 1.75 0 0 1 14.25 12h-3.727c.099 1.041.52 1.872 1.292 2.757A.752.752 0 0 1 11.25 16h-6.5a.75.75 0 0 1-.565-1.243c.772-.885 1.192-1.716 1.292-2.757H1.75A1.75 1.75 0 0 1 0 10.25v-7.5C0 1.784.784 1 1.75 1ZM1.75 2.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25ZM9.018 12H6.982a5.72 5.72 0 0 1-.765 2.5h3.566a5.72 5.72 0 0 1-.765-2.5Z">
                    </path>
                   </svg>
                  </template>
                  <div class="position-relative">
                   <ul aria-label="Suggestions" class="ActionListWrap QueryBuilder-ListWrap" data-action="
                    combobox-commit:query-builder#comboboxCommit
                    mousedown:query-builder#resultsMousedown
                  " data-persist-list="false" data-target="query-builder.resultsList" id="query-builder-test-results" role="listbox">
                   </ul>
                  </div>
                  <div class="FormControl-inlineValidation" hidden="hidden" id="validation-4bf775ec-b4ae-4890-b8a9-dfd54c14652f">
                   <span class="FormControl-inlineValidation--visual">
                    <svg aria-hidden="true" class="octicon octicon-alert-fill" data-view-component="true" height="12" version="1.1" viewbox="0 0 12 12" width="12">
                     <path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 0 1-1.146 1.954H1.33A1.313 1.313 0 0 1 .183 9.058ZM7 7V3H5v4Zm-1 3a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z">
                     </path>
                    </svg>
                   </span>
                   <span>
                   </span>
                  </div>
                 </div>
                 <div aria-atomic="true" aria-live="polite" class="sr-only" data-target="query-builder.screenReaderFeedback">
                 </div>
                </query-builder>
               </form>
               <div class="d-flex flex-row color-fg-muted px-3 text-small color-bg-default search-feedback-prompt">
                <a class="Link color-fg-accent text-normal ml-2" data-view-component="true" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax" target="_blank">
                 Search syntax tips
                </a>
                <div class="d-flex flex-1">
                </div>
                <button class="Button--link Button--medium Button color-fg-accent text-normal ml-2" data-action="click:qbsearch-input#showFeedbackDialog" data-view-component="true" type="button">
                 <span class="Button-content">
                  <span class="Button-label">
                   Give feedback
                  </span>
                 </span>
                </button>
               </div>
              </div>
             </div>
            </div>
           </modal-dialog>
          </div>
         </div>
         <div class="dark-backdrop position-fixed" data-action="click:qbsearch-input#retract" data-target="qbsearch-input.darkBackdrop" hidden="">
         </div>
         <div class="color-fg-default">
          <dialog-helper>
           <dialog aria-describedby="feedback-dialog-description" aria-labelledby="feedback-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="qbsearch-input.feedbackDialog" data-view-component="true" id="feedback-dialog">
            <div class="Overlay-header" data-view-component="true">
             <div class="Overlay-headerContentWrap">
              <div class="Overlay-titleWrap">
               <h1 class="Overlay-title" id="feedback-dialog-title">
                Provide feedback
               </h1>
              </div>
              <div class="Overlay-actionWrap">
               <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
              </div>
             </div>
            </div>
            <scrollable-region data-labelled-by="feedback-dialog-title">
             <div class="Overlay-body" data-view-component="true">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="/search/feedback" data-turbo="false" id="code-search-feedback-form" method="post">
               <input name="authenticity_token" type="hidden" value="3n2QR_AdgrUMB0rcdsXMdlAxeKU6ynMHSelemnNfXDhMz4l_OMuxL7wlRNhI3M2HjTq1ZmDqlVJHaUsghQ5Zgw"/>
               <p>
                We read every piece of feedback, and take your input very seriously.
               </p>
               <textarea class="form-control width-full mb-2" id="feedback" name="feedback" style="height: 120px"></textarea>
               <input aria-label="Include my email address so I can be contacted" class="form-control mr-2" id="include_email" name="include_email" type="checkbox"/>
               <label for="include_email" style="font-weight: normal">
                Include my email address so I can be contacted
               </label>
              </form>
             </div>
            </scrollable-region>
            <div class="Overlay-footer Overlay-footer--alignEnd" data-view-component="true">
             <button class="btn" data-close-dialog-id="feedback-dialog" data-view-component="true" type="button">
              Cancel
             </button>
             <button class="btn-primary btn" data-action="click:qbsearch-input#submitFeedback" data-view-component="true" form="code-search-feedback-form" type="submit">
              Submit feedback
             </button>
            </div>
           </dialog>
          </dialog-helper>
          <custom-scopes data-target="qbsearch-input.customScopesManager">
           <dialog-helper>
            <dialog aria-describedby="custom-scopes-dialog-description" aria-labelledby="custom-scopes-dialog-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-medium Overlay--motion-scaleFade" data-action="close:qbsearch-input#handleDialogClose cancel:qbsearch-input#handleDialogClose" data-target="custom-scopes.customScopesModalDialog" data-view-component="true" id="custom-scopes-dialog">
             <div class="Overlay-header Overlay-header--divided" data-view-component="true">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title" id="custom-scopes-dialog-title">
                 Saved searches
                </h1>
                <h2 class="Overlay-description" id="custom-scopes-dialog-description">
                 Use saved searches to filter your results more quickly
                </h2>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="custom-scopes-dialog" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="custom-scopes-dialog-title">
              <div class="Overlay-body" data-view-component="true">
               <div data-target="custom-scopes.customScopesModalDialogFlash">
               </div>
               <div class="create-custom-scope-form" data-target="custom-scopes.createCustomScopeForm" hidden="">
                <!-- '"` -->
                <!-- </textarea></xmp> -->
                <form accept-charset="UTF-8" action="/search/custom_scopes" data-turbo="false" id="custom-scopes-dialog-form" method="post">
                 <input name="authenticity_token" type="hidden" value="0q1JZ3xtmo_gD9_FWR4-bR3MUnXSJnePlA497JqSjN2eXyvdJOzEVqtmYmgTy_NQ1Jw2YoQjNVQ3VuxUhdXViA">
                  <div data-target="custom-scopes.customScopesModalDialogFlash">
                  </div>
                  <input data-target="custom-scopes.customScopesIdField" id="custom_scope_id" name="custom_scope_id" type="hidden"/>
                  <div class="form-group">
                   <label for="custom_scope_name">
                    Name
                   </label>
                   <auto-check required="" src="/search/custom_scopes/check_name">
                    <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesNameField" id="custom_scope_name" maxlength="50" name="custom_scope_name" placeholder="github-ruby" required="" type="text"/>
                    <input data-csrf="true" type="hidden" value="WzSIaJAzXe2jSm_1_nv25lRgZR8KhP6Yfc9gaXLo58hDacXOx7H07iBeJna-KaiM4qdgmCRdHgAQb8K7Soc0AA">
                    </input>
                   </auto-check>
                  </div>
                  <div class="form-group">
                   <label for="custom_scope_query">
                    Query
                   </label>
                   <input autocomplete="off" class="form-control" data-target="custom-scopes.customScopesQueryField" id="custom_scope_query" maxlength="500" name="custom_scope_query" placeholder="(repo:mona/a OR repo:mona/b) AND lang:python" required="" type="text"/>
                  </div>
                  <p class="text-small color-fg-muted">
                   To see all available qualifiers, see our
                   <a class="Link--inTextBlock" href="https://docs.github.com/search-github/github-code-search/understanding-github-code-search-syntax">
                    documentation
                   </a>
                   .
                  </p>
                 </input>
                </form>
               </div>
               <div data-target="custom-scopes.manageCustomScopesForm">
                <div data-target="custom-scopes.list">
                </div>
               </div>
              </div>
             </scrollable-region>
             <div class="Overlay-footer Overlay-footer--alignEnd Overlay-footer--divided" data-view-component="true">
              <button class="btn" data-action="click:custom-scopes#customScopesCancel" data-view-component="true" type="button">
               Cancel
              </button>
              <button class="btn-primary btn" data-action="click:custom-scopes#customScopesSubmit" data-target="custom-scopes.customScopesSubmitButton" data-view-component="true" form="custom-scopes-dialog-form" type="submit">
               Create saved search
              </button>
             </div>
            </dialog>
           </dialog-helper>
          </custom-scopes>
         </div>
        </qbsearch-input>
        <input class="js-data-jump-to-suggestions-path-csrf" data-csrf="true" type="hidden" value="YydSschMKjRUvDDDQffaDloQlgF57ingXIBpVYL16t-kcj2T0UvsVrNnqj6ITKEdiqhOy_xI79BT4zfwGmGL1Q">
        </input>
       </div>
       <div class="AppHeader-actions position-relative">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="global-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="global-create-menu-button" popovertarget="global-create-menu-overlay" type="button">
           <span class="Button-content">
            <span class="Button-visual Button-leadingVisual">
             <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
              </path>
             </svg>
            </span>
            <span class="Button-label">
             <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
              </path>
             </svg>
            </span>
           </span>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="global-create-menu-button" id="tooltip-8d7e5563-dff0-4d6a-860f-6a547ee860d6" popover="manual">
           Create new...
          </tool-tip>
          <anchored-position align="end" anchor="global-create-menu-button" anchor-offset="normal" data-view-component="true" id="global-create-menu-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="global-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="global-create-menu-list" role="menu">
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-9d014ab0-be9b-4605-9e9a-507c657e94d9" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New repository
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-5beb2ced-1349-47d5-b3e7-86d0ef811d6f" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Import repository
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-612a133c-a2d5-41ec-89a6-efdde8a3e84f" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                    </path>
                    <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New codespace
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-850e8a66-13ce-4dd7-9b3e-f061e4c44114" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New gist
                  </span>
                 </a>
                </li>
                <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                </li>
                <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-c0997f5a-8aa1-448e-b1b7-08237f0a7a86" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   New organization
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
        <a aria-labelledby="tooltip-7f8fa49f-3d81-44f4-a908-bcb6aa934885" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"ISSUES_HEADER","label":null}' data-view-component="true" href="/issues" id="icon-button-dc7cbcea-037c-449b-bb7f-0c157f38f454">
         <svg aria-hidden="true" class="octicon octicon-issue-opened Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-dc7cbcea-037c-449b-bb7f-0c157f38f454" id="tooltip-7f8fa49f-3d81-44f4-a908-bcb6aa934885" popover="manual">
         Issues
        </tool-tip>
        <a aria-labelledby="tooltip-982ee82e-bd0c-4779-ae9c-9bfe93910ed4" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"PULL_REQUESTS_HEADER","label":null}' data-view-component="true" href="/pulls" id="icon-button-d606e618-7421-4f94-8d6c-d3b5d5fe8f2b">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-d606e618-7421-4f94-8d6c-d3b5d5fe8f2b" id="tooltip-982ee82e-bd0c-4779-ae9c-9bfe93910ed4" popover="manual">
         Pull requests
        </tool-tip>
       </div>
       <notification-indicator class="js-socket-channel" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTYxMDU4NX0=--389f99684f1fedea5edad777c83da8a1f52507aa52662e979f26d3389ee57841" data-fetch-indicator-enabled="true" data-fetch-indicator-src="/notifications/indicator" data-header-redesign-enabled="true" data-indicator-mode="none" data-tooltip-global="You have unread notifications" data-tooltip-none="You have no unread notifications" data-tooltip-unavailable="Notifications are unavailable at the moment." data-view-component="true">
        <a aria-label="Notifications" class="Button Button--iconOnly Button--secondary Button--medium AppHeader-button color-fg-muted" data-analytics-event='{"category":"Global navigation","action":"NOTIFICATIONS_HEADER","label":null}' data-hotkey="g n" data-target="notification-indicator.link" data-view-component="true" href="/notifications" id="AppHeader-notifications-button">
         <svg aria-hidden="true" class="octicon octicon-inbox Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.8 2.06A1.75 1.75 0 0 1 4.41 1h7.18c.7 0 1.333.417 1.61 1.06l2.74 6.395c.04.093.06.194.06.295v4.5A1.75 1.75 0 0 1 14.25 15H1.75A1.75 1.75 0 0 1 0 13.25v-4.5c0-.101.02-.202.06-.295Zm1.61.44a.25.25 0 0 0-.23.152L1.887 8H4.75a.75.75 0 0 1 .6.3L6.625 10h2.75l1.275-1.7a.75.75 0 0 1 .6-.3h2.863L11.82 2.652a.25.25 0 0 0-.23-.152Zm10.09 7h-2.875l-1.275 1.7a.75.75 0 0 1-.6.3h-3.5a.75.75 0 0 1-.6-.3L4.375 9.5H1.5v3.75c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </a>
        <tool-tip class="sr-only position-absolute" data-direction="s" data-target="notification-indicator.tooltip" data-type="description" data-view-component="true" for="AppHeader-notifications-button" id="tooltip-193e0639-0dc7-468f-8040-9000755a4a0e" popover="manual">
         Notifications
        </tool-tip>
       </notification-indicator>
       <div class="AppHeader-user">
        <deferred-side-panel data-url="/_side-panels/user?react_global_nav=false&amp;repository_id=45267387">
         <include-fragment data-target="deferred-side-panel.fragment">
          <user-drawer-side-panel>
           <button aria-label="Open user account menu" class="AppHeader-logo Button--invisible Button--medium Button Button--invisible-noVisuals color-bg-transparent p-0" data-action="click:deferred-side-panel#loadPanel click:deferred-side-panel#panelOpened" data-show-dialog-id="dialog-c06f3ef5-d39e-4611-a677-31e023ec4619" data-view-component="true" id="dialog-show-dialog-c06f3ef5-d39e-4611-a677-31e023ec4619" type="button">
            <span class="Button-content">
             <span class="Button-label">
              <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
             </span>
            </span>
           </button>
           <dialog-helper>
            <dialog aria-describedby="dialog-c06f3ef5-d39e-4611-a677-31e023ec4619-description" aria-labelledby="dialog-c06f3ef5-d39e-4611-a677-31e023ec4619-title" aria-modal="true" class="Overlay Overlay-whenNarrow Overlay--size-small-portrait Overlay--motion-scaleFade Overlay--placement-right SidePanel" data-target="deferred-side-panel.panel" data-view-component="true" id="dialog-c06f3ef5-d39e-4611-a677-31e023ec4619">
             <div class="Overlay-header" data-view-component="true" styles="flex-direction: row;">
              <div class="Overlay-headerContentWrap">
               <div class="Overlay-titleWrap">
                <h1 class="Overlay-title sr-only" id="dialog-c06f3ef5-d39e-4611-a677-31e023ec4619-title">
                 Account menu
                </h1>
                <div class="d-flex" data-view-component="true">
                 <div class="AppHeader-logo position-relative" data-view-component="true">
                  <img alt="" class="avatar circle" data-view-component="true" height="32" size="32" src="https://avatars.githubusercontent.com/u/126681664?v=4" width="32"/>
                 </div>
                 <div class="overflow-hidden d-flex width-full" data-view-component="true">
                  <div class="lh-condensed overflow-hidden d-flex flex-column flex-justify-center ml-2 f5 mr-auto width-full" data-view-component="true">
                   <span class="Truncate text-bold" data-view-component="true">
                    <span class="Truncate-text" data-view-component="true">
                     ninenia
                    </span>
                   </span>
                  </div>
                  <action-menu class="d-sm-none d-md-none d-lg-none" data-select-variant="none" data-view-component="true">
                   <focus-group direction="vertical" mnemonics="" retain="">
                    <button aria-controls="user-create-menu-list" aria-haspopup="true" aria-label="Create something new" class="AppHeader-button Button--secondary Button--medium Button width-auto color-fg-muted" data-view-component="true" id="user-create-menu-button" popovertarget="user-create-menu-overlay" type="button">
                     <span class="Button-content">
                      <span class="Button-visual Button-leadingVisual">
                       <svg aria-hidden="true" class="octicon octicon-plus" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                        </path>
                       </svg>
                      </span>
                      <span class="Button-label">
                       <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                        <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                        </path>
                       </svg>
                      </span>
                     </span>
                    </button>
                    <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="user-create-menu-button" id="tooltip-703defc9-6cc3-42bb-bc4e-57c28ec9788a" popover="manual">
                     Create new...
                    </tool-tip>
                    <anchored-position align="end" anchor="user-create-menu-button" anchor-offset="normal" data-view-component="true" id="user-create-menu-overlay" popover="auto" side="outside-bottom">
                     <div class="Overlay Overlay--size-auto" data-view-component="true">
                      <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
                       <action-list>
                        <div data-view-component="true">
                         <ul aria-labelledby="user-create-menu-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="user-create-menu-list" role="menu">
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new" id="item-c7c6fa35-d33e-4112-a4f8-d31322bca3e0" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New repository
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"import repository"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/new/import" id="item-2a399dd9-02e8-4b5f-9793-54c6dc2c5811" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-repo-push" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1 2.5A2.5 2.5 0 0 1 3.5 0h8.75a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0V1.5h-8a1 1 0 0 0-1 1v6.708A2.493 2.493 0 0 1 3.5 9h3.25a.75.75 0 0 1 0 1.5H3.5a1 1 0 0 0 0 2h5.75a.75.75 0 0 1 0 1.5H3.5A2.5 2.5 0 0 1 1 11.5Zm13.23 7.79h-.001l-1.224-1.224v6.184a.75.75 0 0 1-1.5 0V9.066L10.28 10.29a.75.75 0 0 1-1.06-1.061l2.505-2.504a.75.75 0 0 1 1.06 0L15.29 9.23a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             Import repository
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new codespace"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/codespaces/new" id="item-4e135a9d-fd63-403f-9295-6aede44310e2" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-codespaces" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
                              </path>
                              <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New codespace
                            </span>
                           </a>
                          </li>
                          <li class="ActionListItem" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new gist"}' data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="https://gist.github.com/" id="item-f6b4fedc-17cb-4e73-b46a-973243f257ce" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New gist
                            </span>
                           </a>
                          </li>
                          <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                          </li>
                          <li class="ActionListItem" data-targets="action-list.items" data-view-component="true" role="none">
                           <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"SiteHeaderComponent","action":"add_dropdown","label":"new organization"}' data-dont-follow-via-test="true" data-view-component="true" href="/account/organizations/new" id="item-78c025b4-b308-46c8-91f3-c37db45ad32d" role="menuitem" tabindex="-1">
                            <span class="ActionListItem-visual ActionListItem-visual--leading">
                             <svg aria-hidden="true" class="octicon octicon-organization" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                              <path d="M1.75 16A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0h8.5C11.216 0 12 .784 12 1.75v12.5c0 .085-.006.168-.018.25h2.268a.25.25 0 0 0 .25-.25V8.285a.25.25 0 0 0-.111-.208l-1.055-.703a.749.749 0 1 1 .832-1.248l1.055.703c.487.325.779.871.779 1.456v5.965A1.75 1.75 0 0 1 14.25 16h-3.5a.766.766 0 0 1-.197-.026c-.099.017-.2.026-.303.026h-3a.75.75 0 0 1-.75-.75V14h-1v1.25a.75.75 0 0 1-.75.75Zm-.25-1.75c0 .138.112.25.25.25H4v-1.25a.75.75 0 0 1 .75-.75h2.5a.75.75 0 0 1 .75.75v1.25h2.25a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM3.75 6h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 3.75A.75.75 0 0 1 3.75 3h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 3.75Zm4 3A.75.75 0 0 1 7.75 6h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 7 6.75ZM7.75 3h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5ZM3 9.75A.75.75 0 0 1 3.75 9h.5a.75.75 0 0 1 0 1.5h-.5A.75.75 0 0 1 3 9.75ZM7.75 9h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1 0-1.5Z">
                              </path>
                             </svg>
                            </span>
                            <span class="ActionListItem-label" data-view-component="true">
                             New organization
                            </span>
                           </a>
                          </li>
                         </ul>
                        </div>
                       </action-list>
                      </div>
                     </div>
                    </anchored-position>
                   </focus-group>
                  </action-menu>
                 </div>
                </div>
               </div>
               <div class="Overlay-actionWrap">
                <button aria-label="Close" class="close-button Overlay-closeButton" data-close-dialog-id="dialog-c06f3ef5-d39e-4611-a677-31e023ec4619" data-view-component="true" type="button">
                 <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                  </path>
                 </svg>
                </button>
               </div>
              </div>
             </div>
             <scrollable-region data-labelled-by="dialog-c06f3ef5-d39e-4611-a677-31e023ec4619-title">
              <div class="Overlay-body d-flex flex-column px-2" data-view-component="true">
               <div class="d-flex flex-column mb-3" data-view-component="true">
                <nav aria-label="User navigation" class="ActionList" data-view-component="true">
                 <nav-list>
                  <ul class="ActionListWrap" data-target="nav-list.topLevelList" data-view-component="true">
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-ade98f75-3517-43f6-b17b-ff879f9c0960" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"PROFILE","label":null}' data-view-component="true" href="https://github.com/ninenia" id="item-3dfbea26-157d-441b-85af-7ebc747e1d4c">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-person" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your profile
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-05599603-950c-4cef-a727-cfcb796d9947" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_REPOSITORIES","label":null}' data-view-component="true" href="/ninenia?tab=repositories" id="item-d5b917b1-bf4c-4ce3-9cae-11589fe8f320">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-repo" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 2.5A2.5 2.5 0 0 1 4.5 0h8.75a.75.75 0 0 1 .75.75v12.5a.75.75 0 0 1-.75.75h-2.5a.75.75 0 0 1 0-1.5h1.75v-2h-8a1 1 0 0 0-.714 1.7.75.75 0 1 1-1.072 1.05A2.495 2.495 0 0 1 2 11.5Zm10.5-1h-8a1 1 0 0 0-1 1v6.708A2.486 2.486 0 0 1 4.5 9h8ZM5 12.25a.25.25 0 0 1 .25-.25h3.5a.25.25 0 0 1 .25.25v3.25a.25.25 0 0 1-.4.2l-1.45-1.087a.249.249 0 0 0-.3 0L5.4 15.7a.25.25 0 0 1-.4-.2Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your repositories
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_PROJECTS","label":null}' data-view-component="true" href="/ninenia?tab=projects" id="item-769850dd-7e92-46a3-a83b-8499e02657fe">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-project" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your projects
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-587a2a5e-b8ab-4b53-91c4-6ee1f94917f4" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-ec5848d7-5082-46f5-9f2b-a35fe3ab4a11" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_STARS","label":null}' data-view-component="true" href="/ninenia?tab=stars" id="item-1a8900a9-fbbc-488d-a402-30e36428701f">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-star" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your stars
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SPONSORS","label":null}' data-view-component="true" href="/sponsors/accounts" id="item-a99786f5-a01b-4858-ac7f-b74c10981197">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-heart" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="m8 14.25.345.666a.75.75 0 0 1-.69 0l-.008-.004-.018-.01a7.152 7.152 0 0 1-.31-.17 22.055 22.055 0 0 1-3.434-2.414C2.045 10.731 0 8.35 0 5.5 0 2.836 2.086 1 4.25 1 5.797 1 7.153 1.802 8 3.02 8.847 1.802 10.203 1 11.75 1 13.914 1 16 2.836 16 5.5c0 2.85-2.045 5.231-3.885 6.818a22.066 22.066 0 0 1-3.744 2.584l-.018.01-.006.003h-.002ZM4.25 2.5c-1.336 0-2.75 1.164-2.75 3 0 2.15 1.58 4.144 3.365 5.682A20.58 20.58 0 0 0 8 13.393a20.58 20.58 0 0 0 3.135-2.211C12.92 9.644 14.5 7.65 14.5 5.5c0-1.836-1.414-3-2.75-3-1.373 0-2.609.986-3.029 2.456a.749.749 0 0 1-1.442 0C6.859 3.486 5.623 2.5 4.25 2.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your sponsors
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"YOUR_GISTS","label":null}' data-view-component="true" href="https://gist.github.com/mine" id="item-0f34a731-c3aa-494f-b3d2-6e756da41fbe">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-code-square" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25Zm7.47 3.97a.75.75 0 0 1 1.06 0l2 2a.75.75 0 0 1 0 1.06l-2 2a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L10.69 8 9.22 6.53a.75.75 0 0 1 0-1.06ZM6.78 6.53 5.31 8l1.47 1.47a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-2-2a.75.75 0 0 1 0-1.06l2-2a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Your gists
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-5ad09072-5326-4f21-828d-42f9ca074413" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <button class="ActionListContent ActionListContent--visual16" data-view-component="true" id="item-9c1d3679-f9fb-4e1b-a9f5-a2159e6fab4e" type="button">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <span class="d-flex flex-items-center" data-view-component="true">
                       <svg class="anim-rotate" data-view-component="true" fill="none" height="16" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="16">
                        <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                        </circle>
                        <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                        </path>
                       </svg>
                      </span>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      <span class="color-fg-muted">
                       Loading...
                      </span>
                     </span>
                    </button>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SETTINGS","label":null}' data-view-component="true" href="/settings/profile" id="item-7941fe5d-0918-4d04-a839-088c1d7dabb4">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-gear" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      Settings
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"SUPPORT","label":null}' data-view-component="true" href="https://support.github.com" id="item-da776b01-c8c3-449b-93da-04bebd77007a">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-people" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M2 5.5a3.5 3.5 0 1 1 5.898 2.549 5.508 5.508 0 0 1 3.034 4.084.75.75 0 1 1-1.482.235 4 4 0 0 0-7.9 0 .75.75 0 0 1-1.482-.236A5.507 5.507 0 0 1 3.102 8.05 3.493 3.493 0 0 1 2 5.5ZM11 4a3.001 3.001 0 0 1 2.22 5.018 5.01 5.01 0 0 1 2.56 3.012.749.749 0 0 1-.885.954.752.752 0 0 1-.549-.514 3.507 3.507 0 0 0-2.522-2.372.75.75 0 0 1-.574-.73v-.352a.75.75 0 0 1 .416-.672A1.5 1.5 0 0 0 11 5.5.75.75 0 0 1 11 4Zm-5.5-.5a2 2 0 1 0-.001 3.999A2 2 0 0 0 5.5 3.5Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Support
                     </span>
                    </a>
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent ActionListContent--visual16" data-analytics-event='{"category":"Global navigation","action":"COMMUNITY","label":null}' data-view-component="true" href="https://community.github.com" id="item-666d2213-46b3-424a-bef1-87039520e637">
                     <span class="ActionListItem-visual ActionListItem-visual--leading">
                      <svg aria-hidden="true" class="octicon octicon-comment-discussion" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                       <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
                       </path>
                      </svg>
                     </span>
                     <span class="ActionListItem-label" data-view-component="true">
                      GitHub Community
                     </span>
                    </a>
                   </li>
                   <li aria-hidden="true" class="ActionList-sectionDivider" data-view-component="true" role="presentation">
                   </li>
                   <li class="ActionListItem" data-item-id="" data-targets="nav-list.items" data-view-component="true">
                    <a class="ActionListContent" data-analytics-event='{"category":"Global navigation","action":"LOGOUT","label":null}' data-view-component="true" href="/logout" id="item-ffac3f88-2240-4f19-9ba4-ab6e36a43432">
                     <span class="ActionListItem-label" data-view-component="true">
                      Sign out
                     </span>
                    </a>
                   </li>
                  </ul>
                 </nav-list>
                </nav>
               </div>
              </div>
             </scrollable-region>
            </dialog>
           </dialog-helper>
          </user-drawer-side-panel>
         </include-fragment>
        </deferred-side-panel>
       </div>
       <div class="position-absolute mt-2">
        <site-header-logged-in-user-menu>
        </site-header-logged-in-user-menu>
       </div>
      </div>
     </div>
     <div class="AppHeader-localBar">
      <nav aria-label="Repository" class="js-repo-nav js-sidenav-container-pjax js-responsive-underlinenav overflow-hidden UnderlineNav" data-pjax="#js-repo-pjax-container" data-view-component="true">
       <ul class="UnderlineNav-body list-style-none" data-view-component="true">
        <li class="d-inline-flex" data-view-component="true">
         <a aria-current="page" class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item selected" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Code","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g c" data-pjax="#repo-content-pjax-container" data-selected-links="repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches repo_packages repo_deployments repo_attestations /Sumeet460/Fear-Conditioning-Genomics" data-tab-item="i0code-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics" id="code-tab">
          <svg aria-hidden="true" class="octicon octicon-code UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
           </path>
          </svg>
          <span data-content="Code">
           Code
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="code-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Issues","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g i" data-pjax="#repo-content-pjax-container" data-selected-links="repo_issues repo_labels repo_milestones /Sumeet460/Fear-Conditioning-Genomics/issues" data-tab-item="i1issues-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/issues" id="issues-tab">
          <svg aria-hidden="true" class="octicon octicon-issue-opened UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
           </path>
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
           </path>
          </svg>
          <span data-content="Issues">
           Issues
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="issues-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Pull requests","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g p" data-pjax="#repo-content-pjax-container" data-selected-links="repo_pulls checks /Sumeet460/Fear-Conditioning-Genomics/pulls" data-tab-item="i2pull-requests-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/pulls" id="pull-requests-tab">
          <svg aria-hidden="true" class="octicon octicon-git-pull-request UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
           </path>
          </svg>
          <span data-content="Pull requests">
           Pull requests
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="pull-requests-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Actions","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g a" data-pjax="#repo-content-pjax-container" data-selected-links="repo_actions /Sumeet460/Fear-Conditioning-Genomics/actions" data-tab-item="i3actions-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/actions" id="actions-tab">
          <svg aria-hidden="true" class="octicon octicon-play UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
           </path>
          </svg>
          <span data-content="Actions">
           Actions
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="actions-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Projects","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g b" data-pjax="#repo-content-pjax-container" data-selected-links="repo_projects new_repo_project repo_project /Sumeet460/Fear-Conditioning-Genomics/projects" data-tab-item="i4projects-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/projects" id="projects-tab">
          <svg aria-hidden="true" class="octicon octicon-table UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
           </path>
          </svg>
          <span data-content="Projects">
           Projects
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" hidden="hidden" id="projects-repo-tab-count" title="0">
           0
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Wiki","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g w" data-pjax="#repo-content-pjax-container" data-selected-links="repo_wiki /Sumeet460/Fear-Conditioning-Genomics/wiki" data-tab-item="i5wiki-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/wiki" id="wiki-tab">
          <svg aria-hidden="true" class="octicon octicon-book UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
           </path>
          </svg>
          <span data-content="Wiki">
           Wiki
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="wiki-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Security","target":"UNDERLINE_NAV.TAB"}' data-hotkey="g s" data-pjax="#repo-content-pjax-container" data-selected-links="security overview alerts policy token_scanning code_scanning /Sumeet460/Fear-Conditioning-Genomics/security" data-tab-item="i6security-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/security" id="security-tab">
          <svg aria-hidden="true" class="octicon octicon-shield UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
           </path>
          </svg>
          <span data-content="Security">
           Security
          </span>
          <include-fragment accept="text/fragment+html" src="/Sumeet460/Fear-Conditioning-Genomics/security/overall-count">
          </include-fragment>
         </a>
        </li>
        <li class="d-inline-flex" data-view-component="true">
         <a class="UnderlineNav-item no-wrap js-responsive-underlinenav-item js-selected-navigation-item" data-analytics-event='{"category":"Underline navbar","action":"Click tab","label":"Insights","target":"UNDERLINE_NAV.TAB"}' data-pjax="#repo-content-pjax-container" data-selected-links="repo_graphs repo_contributors dependency_graph dependabot_updates pulse people community /Sumeet460/Fear-Conditioning-Genomics/pulse" data-tab-item="i7insights-tab" data-turbo-frame="repo-content-turbo-frame" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/pulse" id="insights-tab">
          <svg aria-hidden="true" class="octicon octicon-graph UnderlineNav-octicon d-none d-sm-inline" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
           <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
           </path>
          </svg>
          <span data-content="Insights">
           Insights
          </span>
          <span class="Counter" data-pjax-replace="" data-turbo-replace="" data-view-component="true" id="insights-repo-tab-count" title="Not available">
          </span>
         </a>
        </li>
       </ul>
       <div class="UnderlineNav-actions js-responsive-underlinenav-overflow position-absolute pr-3 pr-md-4 pr-lg-5 right-0" data-view-component="true" style="visibility:hidden;">
        <action-menu data-select-variant="none" data-view-component="true">
         <focus-group direction="vertical" mnemonics="" retain="">
          <button aria-controls="action-menu-990c7eb4-746f-4875-ac33-6781b81b2f35-list" aria-haspopup="true" aria-labelledby="tooltip-52ec8b44-9d75-4dd8-9685-8b2f4d67564e" class="Button Button--iconOnly Button--secondary Button--medium UnderlineNav-item" data-view-component="true" id="action-menu-990c7eb4-746f-4875-ac33-6781b81b2f35-button" popovertarget="action-menu-990c7eb4-746f-4875-ac33-6781b81b2f35-overlay" type="button">
           <svg aria-hidden="true" class="octicon octicon-kebab-horizontal Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
            <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
            </path>
           </svg>
          </button>
          <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="action-menu-990c7eb4-746f-4875-ac33-6781b81b2f35-button" id="tooltip-52ec8b44-9d75-4dd8-9685-8b2f4d67564e" popover="manual">
           Additional navigation options
          </tool-tip>
          <anchored-position align="start" anchor="action-menu-990c7eb4-746f-4875-ac33-6781b81b2f35-button" anchor-offset="normal" data-view-component="true" id="action-menu-990c7eb4-746f-4875-ac33-6781b81b2f35-overlay" popover="auto" side="outside-bottom">
           <div class="Overlay Overlay--size-auto" data-view-component="true">
            <div class="Overlay-body Overlay-body--paddingNone" data-view-component="true">
             <action-list>
              <div data-view-component="true">
               <ul aria-labelledby="action-menu-990c7eb4-746f-4875-ac33-6781b81b2f35-button" class="ActionListWrap--inset ActionListWrap" data-view-component="true" id="action-menu-990c7eb4-746f-4875-ac33-6781b81b2f35-list" role="menu">
                <li class="ActionListItem" data-menu-item="i0code-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics" id="item-6466f897-05a5-4e28-9218-f8b0f996a2bb" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-code" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Code
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i1issues-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/issues" id="item-597ee3a8-9414-419b-88b4-cb80eef5f370" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-issue-opened" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Issues
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i2pull-requests-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/pulls" id="item-a2111e75-52dd-4d21-9324-740209d32c78" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-git-pull-request" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Pull requests
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i3actions-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/actions" id="item-ac196bcb-0380-45e0-80cc-d9d367146ff7" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-play" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Zm4.879-2.773 4.264 2.559a.25.25 0 0 1 0 .428l-4.264 2.559A.25.25 0 0 1 6 10.559V5.442a.25.25 0 0 1 .379-.215Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Actions
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i4projects-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/projects" id="item-d0fdbdb2-264f-4c08-8513-5c6c93872455" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-table" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75C0 .784.784 0 1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25ZM6.5 6.5v8h7.75a.25.25 0 0 0 .25-.25V6.5Zm8-1.5V1.75a.25.25 0 0 0-.25-.25H6.5V5Zm-13 1.5v7.75c0 .138.112.25.25.25H5v-8ZM5 5V1.5H1.75a.25.25 0 0 0-.25.25V5Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Projects
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i5wiki-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/wiki" id="item-45845a6a-1dce-4bff-b8e3-d456008334f1" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-book" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Wiki
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i6security-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/security" id="item-c2025403-8342-45af-9fff-a07269f8d35f" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-shield" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.467.133a1.748 1.748 0 0 1 1.066 0l5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667Zm.61 1.429a.25.25 0 0 0-.153 0l-5.25 1.68a.25.25 0 0 0-.174.238V7c0 1.358.275 2.666 1.057 3.86.784 1.194 2.121 2.34 4.366 3.297a.196.196 0 0 0 .154 0c2.245-.956 3.582-2.104 4.366-3.298C13.225 9.666 13.5 8.36 13.5 7V3.48a.251.251 0 0 0-.174-.237l-5.25-1.68ZM8.75 4.75v3a.75.75 0 0 1-1.5 0v-3a.75.75 0 0 1 1.5 0ZM9 10.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Security
                  </span>
                 </a>
                </li>
                <li class="ActionListItem" data-menu-item="i7insights-tab" data-targets="action-list.items" data-view-component="true" hidden="hidden" role="none">
                 <a class="ActionListContent ActionListContent--visual16" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/pulse" id="item-a9b16dda-5b71-4e71-8f9f-fd3bb4e6d7bf" role="menuitem" tabindex="-1">
                  <span class="ActionListItem-visual ActionListItem-visual--leading">
                   <svg aria-hidden="true" class="octicon octicon-graph" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z">
                    </path>
                   </svg>
                  </span>
                  <span class="ActionListItem-label" data-view-component="true">
                   Insights
                  </span>
                 </a>
                </li>
               </ul>
              </div>
             </action-list>
            </div>
           </div>
          </anchored-position>
         </focus-group>
        </action-menu>
       </div>
      </nav>
     </div>
    </header>
    <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-full mb-3" data-view-component="true" hidden="hidden">
     <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
      </path>
     </svg>
     <span class="js-stale-session-flash-signed-in" hidden="">
      You signed in with another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-signed-out" hidden="">
      You signed out in another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <span class="js-stale-session-flash-switched" hidden="">
      You switched accounts on another tab or window.
      <a class="Link--inTextBlock" href="">
       Reload
      </a>
      to refresh your session.
     </span>
     <button aria-labelledby="tooltip-81ff0db2-2d42-4eb7-a0f2-89b7bbd057fd" class="Button Button--iconOnly Button--invisible Button--medium flash-close js-flash-close" data-view-component="true" id="icon-button-fd54a810-add8-4098-9547-5f147508afe8" type="button">
      <svg aria-hidden="true" class="octicon octicon-x Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
       </path>
      </svg>
     </button>
     <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-fd54a810-add8-4098-9547-5f147508afe8" id="tooltip-81ff0db2-2d42-4eb7-a0f2-89b7bbd057fd" popover="manual">
      Dismiss alert
     </tool-tip>
    </div>
   </div>
   <div class="show-on-focus" id="start-of-content">
   </div>
   <div data-turbo-replace="" id="js-flash-container">
    <template class="js-flash-template">
     <div class="flash flash-full {{ className }}">
      <div>
       <button aria-label="Dismiss this message" autofocus="" class="flash-close js-flash-close" type="button">
        <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
         </path>
        </svg>
       </button>
       <div aria-atomic="true" class="js-flash-alert" role="alert">
        <div>
         {{ message }}
        </div>
       </div>
      </div>
     </div>
    </template>
   </div>
   <notification-shelf-watcher class="js-socket-channel" data-base-url="https://github.com/notifications/beta/shelf" data-channel="eyJjIjoibm90aWZpY2F0aW9uLWNoYW5nZWQ6MTI2NjgxNjY0IiwidCI6MTcxMTYxMDU4NX0=--389f99684f1fedea5edad777c83da8a1f52507aa52662e979f26d3389ee57841" data-view-component="true">
   </notification-shelf-watcher>
   <div data-initial="" data-target="notification-shelf-watcher.placeholder" hidden="">
   </div>
   <details class="details-reset details-overlay details-overlay-dark js-command-palette-dialog" data-turbo-replace="" id="command-palette-pjax-container">
    <summary aria-label="Command palette trigger" tabindex="-1">
    </summary>
    <details-dialog aria-label="Command palette" class="command-palette-details-dialog d-flex flex-column flex-justify-center height-fit">
     <command-palette activation-hotkey="Mod+k,Mod+Alt+k" class="command-palette color-bg-default rounded-3 border color-shadow-small" command-mode-hotkey="Mod+Shift+K" data-action="
        command-palette-input-ready:command-palette#inputReady
        command-palette-page-stack-updated:command-palette#updateInputScope
        itemsUpdated:command-palette#itemsUpdated
        keydown:command-palette#onKeydown
        loadingStateChanged:command-palette#loadingStateChanged
        selectedItemChanged:command-palette#selectedItemChanged
        pageFetchError:command-palette#pageFetchError
      " return-to="/Sumeet460/Fear-Conditioning-Genomics" user-id="126681664">
      <command-palette-mode data-char="#" data-placeholder="Search issues and pull requests" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="#" data-placeholder="Search issues, pull requests, discussions, and projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="!" data-placeholder="Search projects" data-scope-types='["owner","repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a user, organization, or repository" data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="@" data-placeholder="Search or jump to a repository" data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode data-char="/" data-placeholder="Search files" data-scope-types='["repository"]'>
      </command-palette-mode>
      <command-palette-mode data-char="?">
      </command-palette-mode>
      <command-palette-mode data-char="&gt;" data-placeholder="Run a command">
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='[""]'>
      </command-palette-mode>
      <command-palette-mode data-char="" data-placeholder="Search or jump to..." data-scope-types='["owner"]'>
      </command-palette-mode>
      <command-palette-mode class="js-command-palette-default-mode" data-char="" data-placeholder="Search or jump to...">
      </command-palette-mode>
      <command-palette-input data-action="
          command-palette-input:command-palette#onInput
          command-palette-select:command-palette#onSelect
          command-palette-descope:command-palette#onDescope
          command-palette-cleared:command-palette#onInputClear
        " placeholder="Search or jump to...">
       <div class="js-search-icon d-flex flex-items-center mr-2" style="height: 26px">
        <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
         <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
         </path>
        </svg>
       </div>
       <div class="js-spinner d-flex flex-items-center mr-2 color-fg-muted" hidden="">
        <svg aria-label="Loading" class="anim-rotate" fill="none" height="16" viewbox="0 0 16 16" width="16">
         <circle cx="8" cy="8" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
         </circle>
         <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
         </path>
        </svg>
       </div>
       <command-palette-scope>
        <div class="color-fg-subtle" data-target="command-palette-scope.placeholder" hidden="">
         /
         <span class="text-semibold color-fg-default">
          ...
         </span>
         /
        </div>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDQ6VXNlcjgxMDQ4ODQ=" data-targets="command-palette-scope.tokens" data-text="Sumeet460" data-type="owner" data-value="Sumeet460" style="white-space:nowrap;line-height:20px;">
         Sumeet460
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
        <command-palette-token class="color-fg-default text-semibold" data-id="MDEwOlJlcG9zaXRvcnk0NTI2NzM4Nw==" data-targets="command-palette-scope.tokens" data-text="Fear-Conditioning-Genomics" data-type="repository" data-value="Fear-Conditioning-Genomics" style="white-space:nowrap;line-height:20px;">
         Fear-Conditioning...
         <span class="color-fg-subtle text-normal">
          /
         </span>
        </command-palette-token>
       </command-palette-scope>
       <div class="command-palette-input-group flex-1 form-control border-0 box-shadow-none" style="z-index: 0">
        <div class="command-palette-typeahead position-absolute d-flex flex-items-center Truncate">
         <span class="typeahead-segment input-mirror" data-target="command-palette-input.mirror">
         </span>
         <span class="Truncate-text" data-target="command-palette-input.typeaheadText">
         </span>
         <span class="typeahead-segment" data-target="command-palette-input.typeaheadPlaceholder">
         </span>
        </div>
        <input aria-label="Hidden input for typeahead" class="js-overlay-input typeahead-input d-none" disabled="" tabindex="-1"/>
        <input aria-autocomplete="list" aria-controls="command-palette-page-stack" aria-expanded="false" aria-haspopup="listbox" aria-label="Command palette input" autocapitalize="off" autocomplete="off" autocorrect="off" class="js-input typeahead-input form-control border-0 box-shadow-none input-block width-full no-focus-indicator" data-action="
              input:command-palette-input#onInput
              keydown:command-palette-input#onKeydown
            " role="combobox" spellcheck="false" type="text"/>
       </div>
       <div class="position-relative d-inline-block" data-view-component="true">
        <button aria-keyshortcuts="Control+Backspace" class="btn-octicon command-palette-input-clear-button" data-action="click:command-palette-input#onClear keypress:command-palette-input#onClear" data-target="command-palette-input.clearButton" data-view-component="true" hidden="hidden" id="command-palette-clear-button" type="button">
         <svg aria-hidden="true" class="octicon octicon-x-circle-fill" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2.343 13.657A8 8 0 1 1 13.658 2.343 8 8 0 0 1 2.343 13.657ZM6.03 4.97a.751.751 0 0 0-1.042.018.751.751 0 0 0-.018 1.042L6.94 8 4.97 9.97a.749.749 0 0 0 .326 1.275.749.749 0 0 0 .734-.215L8 9.06l1.97 1.97a.749.749 0 0 0 1.275-.326.749.749 0 0 0-.215-.734L9.06 8l1.97-1.97a.749.749 0 0 0-.326-1.275.749.749 0 0 0-.734.215L8 6.94Z">
          </path>
         </svg>
        </button>
        <tool-tip class="sr-only position-absolute" data-direction="w" data-type="label" data-view-component="true" for="command-palette-clear-button" id="tooltip-bab9f3f8-41e0-45bb-a602-f86b6ff208ad" popover="manual">
         Clear Command Palette
        </tool-tip>
       </div>
      </command-palette-input>
      <command-palette-page-stack data-action="command-palette-page-octicons-cached:command-palette-page-stack#cacheOcticons" data-default-scope-id="MDEwOlJlcG9zaXRvcnk0NTI2NzM4Nw==" data-default-scope-type="Repository">
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           #
          </kbd>
          to search discussions
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           !
          </kbd>
          to search projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["owner"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search teams
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='[""]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           @
          </kbd>
          to search people and organizations
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type
          <kbd class="hx_kbd">
           &gt;
          </kbd>
          to activate command mode
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Go to your accessibility settings to change your keyboard shortcuts
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type author:@me to search your content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:pr to filter to pull requests
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:issue to filter to issues
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:project to filter to projects
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="color-fg-muted f6 px-3 py-1 my-2" data-mode="#" data-scope-types='["","owner","repository"]' data-value="">
        <div class="d-flex flex-items-start flex-justify-between">
         <div>
          <span class="text-bold">
           Tip:
          </span>
          Type is:open to filter to open content
         </div>
         <div class="ml-2 flex-shrink-0">
          Type
          <kbd class="hx_kbd">
           ?
          </kbd>
          for help and tips
         </div>
        </div>
       </command-palette-tip>
       <command-palette-tip class="mx-3 my-2 flash flash-error d-flex flex-items-center" data-on-error="" data-scope-types="*">
        <div>
         <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
          </path>
         </svg>
        </div>
        <div class="px-2">
         We’ve encountered an error and some results aren't available at this time. Type a new search or try again later.
        </div>
       </command-palette-tip>
       <command-palette-tip class="h4 color-fg-default pl-3 pb-2 pt-3" data-match-mode="[^?]|^$" data-on-empty="" data-scope-types="*">
        No results matched your search
       </command-palette-tip>
       <div hidden="">
        <div data-octicon-id="arrow-right-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="arrow-right-color-fg-default" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-arrow-right color-fg-default" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="codespaces-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-codespaces color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 11.25c0-.966.784-1.75 1.75-1.75h12.5c.966 0 1.75.784 1.75 1.75v3A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25Zm2-9.5C2 .784 2.784 0 3.75 0h8.5C13.216 0 14 .784 14 1.75v5a1.75 1.75 0 0 1-1.75 1.75h-8.5A1.75 1.75 0 0 1 2 6.75Zm1.75-.25a.25.25 0 0 0-.25.25v5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5a.25.25 0 0 0-.25-.25Zm-2 9.5a.25.25 0 0 0-.25.25v3c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25v-3a.25.25 0 0 0-.25-.25Z">
          </path>
          <path d="M7 12.75a.75.75 0 0 1 .75-.75h4.5a.75.75 0 0 1 0 1.5h-4.5a.75.75 0 0 1-.75-.75Zm-4 0a.75.75 0 0 1 .75-.75h.5a.75.75 0 0 1 0 1.5h-.5a.75.75 0 0 1-.75-.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="copy-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-copy color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
          </path>
          <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="dash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-dash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 7.75A.75.75 0 0 1 2.75 7h10a.75.75 0 0 1 0 1.5h-10A.75.75 0 0 1 2 7.75Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="file-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-file color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="gear-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-gear color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 0a8.2 8.2 0 0 1 .701.031C9.444.095 9.99.645 10.16 1.29l.288 1.107c.018.066.079.158.212.224.231.114.454.243.668.386.123.082.233.09.299.071l1.103-.303c.644-.176 1.392.021 1.82.63.27.385.506.792.704 1.218.315.675.111 1.422-.364 1.891l-.814.806c-.049.048-.098.147-.088.294.016.257.016.515 0 .772-.01.147.038.246.088.294l.814.806c.475.469.679 1.216.364 1.891a7.977 7.977 0 0 1-.704 1.217c-.428.61-1.176.807-1.82.63l-1.102-.302c-.067-.019-.177-.011-.3.071a5.909 5.909 0 0 1-.668.386c-.133.066-.194.158-.211.224l-.29 1.106c-.168.646-.715 1.196-1.458 1.26a8.006 8.006 0 0 1-1.402 0c-.743-.064-1.289-.614-1.458-1.26l-.289-1.106c-.018-.066-.079-.158-.212-.224a5.738 5.738 0 0 1-.668-.386c-.123-.082-.233-.09-.299-.071l-1.103.303c-.644.176-1.392-.021-1.82-.63a8.12 8.12 0 0 1-.704-1.218c-.315-.675-.111-1.422.363-1.891l.815-.806c.05-.048.098-.147.088-.294a6.214 6.214 0 0 1 0-.772c.01-.147-.038-.246-.088-.294l-.815-.806C.635 6.045.431 5.298.746 4.623a7.92 7.92 0 0 1 .704-1.217c.428-.61 1.176-.807 1.82-.63l1.102.302c.067.019.177.011.3-.071.214-.143.437-.272.668-.386.133-.066.194-.158.211-.224l.29-1.106C6.009.645 6.556.095 7.299.03 7.53.01 7.764 0 8 0Zm-.571 1.525c-.036.003-.108.036-.137.146l-.289 1.105c-.147.561-.549.967-.998 1.189-.173.086-.34.183-.5.29-.417.278-.97.423-1.529.27l-1.103-.303c-.109-.03-.175.016-.195.045-.22.312-.412.644-.573.99-.014.031-.021.11.059.19l.815.806c.411.406.562.957.53 1.456a4.709 4.709 0 0 0 0 .582c.032.499-.119 1.05-.53 1.456l-.815.806c-.081.08-.073.159-.059.19.162.346.353.677.573.989.02.03.085.076.195.046l1.102-.303c.56-.153 1.113-.008 1.53.27.161.107.328.204.501.29.447.222.85.629.997 1.189l.289 1.105c.029.109.101.143.137.146a6.6 6.6 0 0 0 1.142 0c.036-.003.108-.036.137-.146l.289-1.105c.147-.561.549-.967.998-1.189.173-.086.34-.183.5-.29.417-.278.97-.423 1.529-.27l1.103.303c.109.029.175-.016.195-.045.22-.313.411-.644.573-.99.014-.031.021-.11-.059-.19l-.815-.806c-.411-.406-.562-.957-.53-1.456a4.709 4.709 0 0 0 0-.582c-.032-.499.119-1.05.53-1.456l.815-.806c.081-.08.073-.159.059-.19a6.464 6.464 0 0 0-.573-.989c-.02-.03-.085-.076-.195-.046l-1.102.303c-.56.153-1.113.008-1.53-.27a4.44 4.44 0 0 0-.501-.29c-.447-.222-.85-.629-.997-1.189l-.289-1.105c-.029-.11-.101-.143-.137-.146a6.6 6.6 0 0 0-1.142 0ZM11 8a3 3 0 1 1-6 0 3 3 0 0 1 6 0ZM9.5 8a1.5 1.5 0 1 0-3.001.001A1.5 1.5 0 0 0 9.5 8Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="lock-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-lock color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M4 4a4 4 0 0 1 8 0v2h.25c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 12.25 15h-8.5A1.75 1.75 0 0 1 2 13.25v-5.5C2 6.784 2.784 6 3.75 6H4Zm8.25 3.5h-8.5a.25.25 0 0 0-.25.25v5.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25ZM10.5 6V4a2.5 2.5 0 1 0-5 0v2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="moon-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-moon color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M9.598 1.591a.749.749 0 0 1 .785-.175 7.001 7.001 0 1 1-8.967 8.967.75.75 0 0 1 .961-.96 5.5 5.5 0 0 0 7.046-7.046.75.75 0 0 1 .175-.786Zm1.616 1.945a7 7 0 0 1-7.678 7.678 5.499 5.499 0 1 0 7.678-7.678Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="person-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-person color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.561 8.073a6.005 6.005 0 0 1 3.432 5.142.75.75 0 1 1-1.498.07 4.5 4.5 0 0 0-8.99 0 .75.75 0 0 1-1.498-.07 6.004 6.004 0 0 1 3.431-5.142 3.999 3.999 0 1 1 5.123 0ZM10.5 5a2.5 2.5 0 1 0-5 0 2.5 2.5 0 0 0 5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="pencil-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-pencil color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="issue-opened-open" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-issue-opened open" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
          </path>
          <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="git-pull-request-draft-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-git-pull-request-draft color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M3.25 1A2.25 2.25 0 0 1 4 5.372v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.251 2.251 0 0 1 3.25 1Zm9.5 14a2.25 2.25 0 1 1 0-4.5 2.25 2.25 0 0 1 0 4.5ZM2.5 3.25a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0ZM3.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm9.5 0a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM14 7.5a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm0-4.25a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="search-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-search color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.68 11.74a6 6 0 0 1-7.922-8.982 6 6 0 0 1 8.982 7.922l3.04 3.04a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215ZM11.5 7a4.499 4.499 0 1 0-8.997 0A4.499 4.499 0 0 0 11.5 7Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sun-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sun color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 12a4 4 0 1 1 0-8 4 4 0 0 1 0 8Zm0-1.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Zm5.657-8.157a.75.75 0 0 1 0 1.061l-1.061 1.06a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l1.06-1.06a.75.75 0 0 1 1.06 0Zm-9.193 9.193a.75.75 0 0 1 0 1.06l-1.06 1.061a.75.75 0 1 1-1.061-1.06l1.06-1.061a.75.75 0 0 1 1.061 0ZM8 0a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0V.75A.75.75 0 0 1 8 0ZM3 8a.75.75 0 0 1-.75.75H.75a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 3 8Zm13 0a.75.75 0 0 1-.75.75h-1.5a.75.75 0 0 1 0-1.5h1.5A.75.75 0 0 1 16 8Zm-8 5a.75.75 0 0 1 .75.75v1.5a.75.75 0 0 1-1.5 0v-1.5A.75.75 0 0 1 8 13Zm3.536-1.464a.75.75 0 0 1 1.06 0l1.061 1.06a.75.75 0 0 1-1.06 1.061l-1.061-1.06a.75.75 0 0 1 0-1.061ZM2.343 2.343a.75.75 0 0 1 1.061 0l1.06 1.061a.751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018l-1.06-1.06a.75.75 0 0 1 0-1.06Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="sync-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-sync color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.705 8.005a.75.75 0 0 1 .834.656 5.5 5.5 0 0 0 9.592 2.97l-1.204-1.204a.25.25 0 0 1 .177-.427h3.646a.25.25 0 0 1 .25.25v3.646a.25.25 0 0 1-.427.177l-1.38-1.38A7.002 7.002 0 0 1 1.05 8.84a.75.75 0 0 1 .656-.834ZM8 2.5a5.487 5.487 0 0 0-4.131 1.869l1.204 1.204A.25.25 0 0 1 4.896 6H1.25A.25.25 0 0 1 1 5.75V2.104a.25.25 0 0 1 .427-.177l1.38 1.38A7.002 7.002 0 0 1 14.95 7.16a.75.75 0 0 1-1.49.178A5.5 5.5 0 0 0 8 2.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="trash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-trash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11 1.75V3h2.25a.75.75 0 0 1 0 1.5H2.75a.75.75 0 0 1 0-1.5H5V1.75C5 .784 5.784 0 6.75 0h2.5C10.216 0 11 .784 11 1.75ZM4.496 6.675l.66 6.6a.25.25 0 0 0 .249.225h5.19a.25.25 0 0 0 .249-.225l.66-6.6a.75.75 0 0 1 1.492.149l-.66 6.6A1.748 1.748 0 0 1 10.595 15h-5.19a1.75 1.75 0 0 1-1.741-1.575l-.66-6.6a.75.75 0 1 1 1.492-.15ZM6.5 1.75V3h3V1.75a.25.25 0 0 0-.25-.25h-2.5a.25.25 0 0 0-.25.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="key-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-key color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M10.5 0a5.499 5.499 0 1 1-1.288 10.848l-.932.932a.749.749 0 0 1-.53.22H7v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22H5v.75a.749.749 0 0 1-.22.53l-.5.5a.749.749 0 0 1-.53.22h-2A1.75 1.75 0 0 1 0 14.25v-2c0-.199.079-.389.22-.53l4.932-4.932A5.5 5.5 0 0 1 10.5 0Zm-4 5.5c-.001.431.069.86.205 1.269a.75.75 0 0 1-.181.768L1.5 12.56v1.69c0 .138.112.25.25.25h1.69l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l.06-.06v-1.19a.75.75 0 0 1 .75-.75h1.19l1.023-1.025a.75.75 0 0 1 .768-.18A4 4 0 1 0 6.5 5.5ZM11 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="comment-discussion-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-comment-discussion color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M8 16a2 2 0 0 0 1.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 0 0 8 16ZM3 5a5 5 0 0 1 10 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.519 1.519 0 0 1 13.482 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947Zm5-3.5A3.5 3.5 0 0 0 4.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01l.001.006c0 .002.002.004.004.006l.006.004.007.001h10.964l.007-.001.006-.004.004-.006.001-.007a.017.017 0 0 0-.003-.01l-1.703-2.554a1.745 1.745 0 0 1-.294-.97V5A3.5 3.5 0 0 0 8 1.5Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="bell-slash-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-bell-slash color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
          </path>
         </svg>
        </div>
        <div data-octicon-id="paintbrush-color-fg-muted" data-targets="command-palette-page-stack.localOcticons">
         <svg aria-hidden="true" class="octicon octicon-paintbrush color-fg-muted" height="16" version="1.1" viewbox="0 0 16 16" width="16">
          <path d="M11.134 1.535c.7-.509 1.416-.942 2.076-1.155.649-.21 1.463-.267 2.069.34.603.601.568 1.411.368 2.07-.202.668-.624 1.39-1.125 2.096-1.011 1.424-2.496 2.987-3.775 4.249-1.098 1.084-2.132 1.839-3.04 2.3a3.744 3.744 0 0 1-1.055 3.217c-.431.431-1.065.691-1.657.861-.614.177-1.294.287-1.914.357A21.151 21.151 0 0 1 .797 16H.743l.007-.75H.749L.742 16a.75.75 0 0 1-.743-.742l.743-.008-.742.007v-.054a21.25 21.25 0 0 1 .13-2.284c.067-.647.187-1.287.358-1.914.17-.591.43-1.226.86-1.657a3.746 3.746 0 0 1 3.227-1.054c.466-.893 1.225-1.907 2.314-2.982 1.271-1.255 2.833-2.75 4.245-3.777ZM1.62 13.089c-.051.464-.086.929-.104 1.395.466-.018.932-.053 1.396-.104a10.511 10.511 0 0 0 1.668-.309c.526-.151.856-.325 1.011-.48a2.25 2.25 0 1 0-3.182-3.182c-.155.155-.329.485-.48 1.01a10.515 10.515 0 0 0-.309 1.67Zm10.396-10.34c-1.224.89-2.605 2.189-3.822 3.384l1.718 1.718c1.21-1.205 2.51-2.597 3.387-3.833.47-.662.78-1.227.912-1.662.134-.444.032-.551.009-.575h-.001V1.78c-.014-.014-.113-.113-.548.027-.432.14-.995.462-1.655.942Zm-4.832 7.266-.001.001a9.859 9.859 0 0 0 1.63-1.142L7.155 7.216a9.7 9.7 0 0 0-1.161 1.607c.482.302.889.71 1.19 1.192Z">
          </path>
         </svg>
        </div>
        <command-palette-item-group data-default-priority="0" data-group-hint="" data-group-id="top" data-group-limits="{}" data-group-title="Top result">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="1" data-group-hint="Type &gt; to filter" data-group-id="commands" data-group-limits='{"static_items_page":50,"issue":50,"pull_request":50,"discussion":50}' data-group-title="Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="2" data-group-hint="Type &gt; to filter" data-group-id="global_commands" data-group-limits='{"issue":0,"pull_request":0,"discussion":0}' data-group-title="Global Commands">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="3" data-group-hint="" data-group-id="this_page" data-group-limits="{}" data-group-title="This Page">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="4" data-group-hint="" data-group-id="files" data-group-limits="{}" data-group-title="Files">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="5" data-group-hint="" data-group-id="default" data-group-limits='{"static_items_page":50}' data-group-title="Default">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="6" data-group-hint="" data-group-id="pages" data-group-limits='{"repository":10}' data-group-title="Pages">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="7" data-group-hint="" data-group-id="access_policies" data-group-limits="{}" data-group-title="Access Policies">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="8" data-group-hint="" data-group-id="organizations" data-group-limits="{}" data-group-title="Organizations">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="9" data-group-hint="" data-group-id="repositories" data-group-limits="{}" data-group-title="Repositories">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="10" data-group-hint="Type # to filter" data-group-id="references" data-group-limits="{}" data-group-title="Issues, pull requests, and discussions">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="11" data-group-hint="" data-group-id="teams" data-group-limits="{}" data-group-title="Teams">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="12" data-group-hint="" data-group-id="users" data-group-limits="{}" data-group-title="Users">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="13" data-group-hint="" data-group-id="memex_projects" data-group-limits="{}" data-group-title="Projects">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="14" data-group-hint="" data-group-id="projects" data-group-limits="{}" data-group-title="Projects (classic)">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="15" data-group-hint="" data-group-id="footer" data-group-limits="{}" data-group-title="Footer">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="16" data-group-hint="" data-group-id="modes_help" data-group-limits="{}" data-group-title="Modes">
        </command-palette-item-group>
        <command-palette-item-group data-default-priority="17" data-group-hint="" data-group-id="filters_help" data-group-limits="{}" data-group-title="Use filters in issues, pull requests, discussions, and projects">
        </command-palette-item-group>
        <command-palette-page data-page-title="Sumeet460" data-scope-id="MDQ6VXNlcjgxMDQ4ODQ=" data-scope-type="owner" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
        <command-palette-page data-page-title="Fear-Conditioning-Genomics" data-scope-id="MDEwOlJlcG9zaXRvcnk0NTI2NzM4Nw==" data-scope-type="repository" data-targets="command-palette-page-stack.defaultPages" hidden="">
        </command-palette-page>
       </div>
       <command-palette-page data-is-root="">
       </command-palette-page>
       <command-palette-page data-page-title="Sumeet460" data-scope-id="MDQ6VXNlcjgxMDQ4ODQ=" data-scope-type="owner">
       </command-palette-page>
       <command-palette-page data-page-title="Fear-Conditioning-Genomics" data-scope-id="MDEwOlJlcG9zaXRvcnk0NTI2NzM4Nw==" data-scope-type="repository">
       </command-palette-page>
      </command-palette-page-stack>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="search-links">
      </server-defined-provider>
      <server-defined-provider data-targets="command-palette.serverDefinedProviderElements" data-type="help">
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues
         </strong>
         and
         <strong>
          pull requests
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="#" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          issues, pull requests, discussions,
         </strong>
         and
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          #
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="@" data-scope-types='[""]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          organizations, repositories,
         </strong>
         and
         <strong>
          users
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          @
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="!" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          projects
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          !
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="/" data-scope-types='["repository"]'>
        <span data-target="command-palette-help.titleElement">
         Search for
         <strong>
          files
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          /
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="modes_help" data-prefix="&gt;">
        <span data-target="command-palette-help.titleElement">
         Activate
         <strong>
          command mode
         </strong>
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          &gt;
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# author:@me">
        <span data-target="command-palette-help.titleElement">
         Search your issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # author:@me
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:pr">
        <span data-target="command-palette-help.titleElement">
         Filter to pull requests
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:pr
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:issue">
        <span data-target="command-palette-help.titleElement">
         Filter to issues
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:issue
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:discussion" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:discussion
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:project" data-scope-types='["owner","repository"]'>
        <span data-target="command-palette-help.titleElement">
         Filter to projects
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:project
         </kbd>
        </span>
       </command-palette-help>
       <command-palette-help data-group="filters_help" data-prefix="# is:open">
        <span data-target="command-palette-help.titleElement">
         Filter to open issues, pull requests, and discussions
        </span>
        <span data-target="command-palette-help.hintElement">
         <kbd class="hx_kbd">
          # is:open
         </kbd>
        </span>
       </command-palette-help>
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/commands" data-supported-modes="[]" data-supports-commands="" data-targets="command-palette.serverDefinedProviderElements" data-type="commands">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_page_navigation" data-supported-modes='[""]' data-supported-scope-types='["","owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to" data-supported-modes='["@","@"]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/jump_to_members_only" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/jump_to_members_only_prefetched" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/files" data-supported-modes='["/"]' data-supported-scope-types='["repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="files">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/discussions" data-supported-modes='["#"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/projects" data-supported-modes='["#","!"]' data-supported-scope-types='["owner","repository"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="0" data-src="/command_palette/recent_issues" data-supported-modes='["#","#"]' data-supported-scope-types='["owner","repository",""]' data-targets="command-palette.serverDefinedProviderElements" data-type="prefetched">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/teams" data-supported-modes='["@",""]' data-supported-scope-types='["owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
      <server-defined-provider data-fetch-debounce="200" data-src="/command_palette/name_with_owner_repository" data-supported-modes='["@","@","",""]' data-supported-scope-types='["","owner"]' data-targets="command-palette.serverDefinedProviderElements" data-type="remote">
      </server-defined-provider>
     </command-palette>
    </details-dialog>
   </details>
   <div class="position-fixed bottom-0 left-0 ml-5 mb-5 js-command-palette-toasts" style="z-index: 1000">
    <div class="Toast Toast--loading" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="Toast--spinner" height="18" viewbox="0 0 32 32" width="18">
       <path d="M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4" fill="#959da5">
       </path>
       <path d="M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z" fill="#ffffff">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--error" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-stop" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M4.47.22A.749.749 0 0 1 5 0h6c.199 0 .389.079.53.22l4.25 4.25c.141.14.22.331.22.53v6a.749.749 0 0 1-.22.53l-4.25 4.25A.749.749 0 0 1 11 16H5a.749.749 0 0 1-.53-.22L.22 11.53A.749.749 0 0 1 0 11V5c0-.199.079-.389.22-.53Zm.84 1.28L1.5 5.31v5.38l3.81 3.81h5.38l3.81-3.81V5.31L10.69 1.5ZM8 4a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 4Zm0 8a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--warning" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast Toast--success" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
    <div class="anim-fade-in fast Toast" hidden="">
     <span class="Toast-icon">
      <svg aria-hidden="true" class="octicon octicon-info" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8Zm8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13ZM6.5 7.75A.75.75 0 0 1 7.25 7h1a.75.75 0 0 1 .75.75v2.75h.25a.75.75 0 0 1 0 1.5h-2a.75.75 0 0 1 0-1.5h.25v-2h-.25a.75.75 0 0 1-.75-.75ZM8 6a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
       </path>
      </svg>
     </span>
     <span class="Toast-content">
     </span>
    </div>
   </div>
   <div class="application-main" data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="">
    <div class="" itemscope="" itemtype="http://schema.org/SoftwareSourceCode">
     <main id="js-repo-pjax-container">
      <template class="js-user-list-create-dialog-template" data-label="Create list">
       <div class="Box-header">
        <h2 class="Box-title">
         Create list
        </h2>
       </div>
       <!-- '"` -->
       <!-- </textarea></xmp> -->
       <form accept-charset="UTF-8" action="/stars/ninenia/lists" class="Box-body d-flex flex-column p-3 js-user-list-form" data-turbo="false" method="post">
        <input autocomplete="off" name="authenticity_token" type="hidden" value="2-LcA7SAUe2KGp_Qz7FVJs6UmmufyCTGTfN0-KgulVQLjnYnIVqHSmknNbdcacyAa5dRHhvvDfrTWVFNHP-D4w">
         <p class="color-fg-subtle mb-3">
          Create a list to organize your starred repositories.
         </p>
         <input name="repository_id" type="hidden" value="{{ repositoryId }}"/>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <auto-check required="" src="/stars/ninenia/list-check?attr=name">
           <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
            <input aria-label="List name" autofocus="" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="32" maxlength="32" name="user_list[name]" placeholder="⭐️ Name this list" required="" type="text" value="{{ placeholderName }}"/>
           </text-expander>
           <input data-csrf="true" type="hidden" value="yksySMvh9RQb6srqi9XVzxbFcAB9jnE5bjvqnUaCgw-qY265BM-ncM2hVa5o1xH5Fy8ucdF8zwPCEZBUpOo3oQ">
           </input>
          </auto-check>
          <p class="note error position-relative js-user-list-error" hidden="">
           Name .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           32 remaining
          </p>
         </div>
         <div class="form-group mx-0 mt-0 mb-2 js-user-list-input-container js-characters-remaining-container position-relative">
          <text-expander data-emoji-url="/autocomplete/emoji" keys=":">
           <textarea aria-label="List description" class="form-control js-user-list-input js-characters-remaining-field" data-maxlength="160" maxlength="160" name="user_list[description]" placeholder="Write a description" style="height: 74px; min-height: 74px"></textarea>
          </text-expander>
          <p class="note error position-relative js-user-list-error" hidden="">
           Description .
          </p>
          <p class="mt-1 text-small float-right js-characters-remaining" data-suffix="remaining" hidden="">
           160 remaining
          </p>
         </div>
         <div class="js-user-list-base flash flash-error mx-0 mt-0 mb-2" data-generic-message="Unable to save your list at this time." data-view-component="true" hidden="hidden">
          .
         </div>
         <button class="Button--primary Button--medium Button Button--fullWidth mt-2" data-disable-invalid="true" data-submitting-message="Creating..." data-view-component="true" disabled="disabled" type="submit">
          <span class="Button-content">
           <span class="Button-label">
            Create
           </span>
          </span>
         </button>
         <p class="note mt-2 mb-0">
          <strong>
           Tip:
          </strong>
          type
          <code>
           :
          </code>
          to add emoji to the name or description.
         </p>
        </input>
       </form>
       <div class="Box-footer Box-row--gray text-small color-fg-muted d-flex flex-items-baseline py-2" data-view-component="true">
        <span aria-label="Feature Release Label: Beta" class="Label Label--success Label--inline px-2 mr-2" data-view-component="true" title="Feature Release Label: Beta">
         Beta
        </span>
        <span class="mr-1">
         Lists are currently in beta.
        </span>
        <a href="https://github.com//github/feedback/discussions/categories/lists">
         Share feedback and report bugs.
        </a>
       </div>
      </template>
      <div class="pt-3 hide-full-screen" data-turbo-replace="" id="repository-container-header">
       <div class="d-flex flex-wrap flex-justify-end mb-3 container-xl px-3 px-md-4 px-lg-5" style="gap: 1rem;">
        <div class="flex-auto min-width-0 width-fit mr-3">
         <div class="d-flex flex-wrap flex-items-center wb-break-word f3 text-normal" id="repo-title-component">
          <img alt="Owner avatar" class="avatar mr-2 d-none d-md-block avatar-user" height="24" src="https://avatars.githubusercontent.com/u/8104884?s=48&amp;v=4" width="24"/>
          <strong class="mr-2 flex-self-stretch d-none d-md-block" itemprop="name">
           <a class="color-fg-default" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame" href="/Sumeet460/Fear-Conditioning-Genomics">
            Fear-Conditioning-Genomics
           </a>
          </strong>
          <span>
          </span>
          <span class="Label Label--secondary v-align-middle mr-1 d-none d-md-block">
           Public
          </span>
         </div>
         <div class="d-none d-md-block">
         </div>
        </div>
        <div data-turbo-replace="" id="repository-details-container">
         <ul class="pagehead-actions flex-shrink-0 d-none d-md-inline" style="padding: 2px 0;">
          <li>
           <include-fragment loading="lazy" src="/notifications/45267387/watch_subscription?aria_id_prefix=repository-details&amp;button_block=false&amp;show_count=true">
            <div data-hide-on-error="">
             <button aria-label="Watching a repository" class="tooltipped tooltipped-n Button--secondary Button--small Button" data-view-component="true" disabled="disabled" type="button">
              <span class="Button-content">
               <span class="Button-visual Button-leadingVisual">
                <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                 </path>
                </svg>
               </span>
               <span class="Button-label">
                Watch
               </span>
              </span>
             </button>
            </div>
            <p data-show-on-error="" hidden="">
             <span class="d-flex pt-1">
              <svg aria-hidden="true" class="octicon octicon-alert mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
               </path>
              </svg>
              Couldn't load subscription status.
              <button class="Button--link Button--medium Button" data-retry-button="" data-view-component="true" type="button">
               <span class="Button-content">
                <span class="Button-label">
                 Retry
                </span>
               </span>
              </button>
             </span>
            </p>
           </include-fragment>
          </li>
          <li>
           <div class="BtnGroup d-flex" data-view-component="true">
            <a class="btn-sm btn BtnGroup-item" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":45267387,"originating_url":"https://github.com/Sumeet460/Fear-Conditioning-Genomics","user_id":126681664}}' data-hydro-click-hmac="4aacf64f3f6c58f9d9a689c34ddd50164c86bcf9dccf2be32b91ab7aa7e8ae1f" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/fork" icon="repo-forked" id="fork-button">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             Fork
             <span class="Counter" data-pjax-replace="true" data-turbo-replace="true" data-view-component="true" id="repo-network-counter" title="0">
              0
             </span>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="fork-button" id="tooltip-d636cef0-7fb5-42b1-9b55-4aa71a01a1d6" popover="manual">
              Fork your own copy of Sumeet460/Fear-Conditioning-Genomics
             </tool-tip>
            </a>
            <details class="details-reset details-overlay BtnGroup-parent d-inline-block position-relative" data-view-component="true" group_item="true" id="my-forks-menu-45267387">
             <summary aria-label="See your forks of this repository" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
              <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
               </path>
              </svg>
             </summary>
             <details-menu class="SelectMenu right-0" role="menu" src="/Sumeet460/Fear-Conditioning-Genomics/my_forks_menu_content?can_fork=true">
              <div class="SelectMenu-modal">
               <button aria-label="Close menu" class="SelectMenu-closeButton position-absolute right-0 m-2" data-toggle-for="my-forks-menu-45267387" type="button">
                <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                 </path>
                </svg>
               </button>
               <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-119266">
                <div class="SelectMenu-list">
                 <include-fragment aria-label="Loading" class="SelectMenu-loading">
                  <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                   <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                   </circle>
                   <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                   </path>
                  </svg>
                 </include-fragment>
                </div>
               </div>
              </div>
             </details-menu>
            </details>
           </div>
          </li>
          <li>
           <template class="js-unstar-confirmation-dialog-template">
            <div class="Box-header">
             <h2 class="Box-title">
              Unstar this repository?
             </h2>
            </div>
            <div class="Box-body">
             <p class="mb-3">
              This will remove {{ repoNameWithOwner }} from the {{ listsWithCount }} that it's been added to.
             </p>
             <div class="form-actions">
              <!-- '"` -->
              <!-- </textarea></xmp> -->
              <form accept-charset="UTF-8" action="{{ confirmUrl }}" class="js-social-confirmation-form" data-turbo="false" method="post">
               <input name="authenticity_token" type="hidden" value="{{ confirmCsrfToken }}"/>
               <input name="confirm" type="hidden" value="true"/>
               <button class="btn-danger btn width-full" data-close-dialog="true" data-view-component="true" type="submit">
                Unstar
               </button>
              </form>
             </div>
            </div>
           </template>
           <div class="js-toggler-container js-social-container starring-container d-flex" data-view-component="true">
            <div class="starred BtnGroup flex-1 ml-0" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/Sumeet460/Fear-Conditioning-Genomics/unstar" class="js-social-form BtnGroup-parent flex-auto js-deferred-toggler-target" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="-0Dnz0hi281bdZwWzaoJYocmmBitHcP1fyG4dzEhJ4B3f-joRvi_MoHWF4vfpD4wl_U8G-760frW9Dl5Dqan8Q">
               <input class="js-confirm-csrf-token" data-csrf="true" type="hidden" value="jF6q9RCsm8FxI5GvqQvwJCs9kTUzMAaOldc-unnQHSQAYaXSHjb_PquAGjK7Bcd2O-41NnDXFIE8Ar-0RledVQ">
                <input name="context" type="hidden" value="repository"/>
                <button aria-label="Unstar this repository (0)" class="rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":45267387,"originating_url":"https://github.com/Sumeet460/Fear-Conditioning-Genomics","user_id":126681664}}' data-hydro-click-hmac="01e6eb455c36b3503eb4ec9686916fcee6b390ce521ba08ce7ff105223b06d3c" data-view-component="true" type="submit">
                 <svg aria-hidden="true" class="octicon octicon-star-fill starred-button-icon d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                  <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                  </path>
                 </svg>
                 <span class="d-inline" data-view-component="true">
                  Starred
                 </span>
                 <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-unstar" title="0">
                  0
                 </span>
                </button>
               </input>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-45267387-starred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/Sumeet460/Fear-Conditioning-Genomics/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-45267387-starred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-54eb9f">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
            <div class="unstarred BtnGroup ml-0 flex-1" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/Sumeet460/Fear-Conditioning-Genomics/star" class="js-social-form BtnGroup-parent flex-auto" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="nOloF05k0V1Unjl0e2HL4Z-vDHRiL5xWMWwoxM8BtlwaCcrgUSJZw0ObcKQ30lm_P0N-q_ErYL3JAtHftSogEw">
               <input name="context" type="hidden" value="repository"/>
               <button aria-label="Star this repository (0)" class="js-toggler-target rounded-left-2 btn-sm btn BtnGroup-item" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":45267387,"originating_url":"https://github.com/Sumeet460/Fear-Conditioning-Genomics","user_id":126681664}}' data-hydro-click-hmac="3101c5bd66ea9f0fb177196fcc3a2d97f78e877fe2d49e03962587b3a7da0d38" data-view-component="true" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star d-inline-block mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
                <span class="d-inline" data-view-component="true">
                 Star
                </span>
                <span aria-label="0 users starred this repository" class="Counter js-social-count" data-plural-suffix="users starred this repository" data-singular-suffix="user starred this repository" data-turbo-replace="true" data-view-component="true" id="repo-stars-counter-star" title="0">
                 0
                </span>
               </button>
              </input>
             </form>
             <details class="details-reset details-overlay BtnGroup-parent js-user-list-menu d-inline-block position-relative" data-view-component="true" id="details-user-list-45267387-unstarred">
              <summary aria-label="Add this repository to a list" class="btn-sm btn BtnGroup-item px-2 float-none" data-view-component="true">
               <svg aria-hidden="true" class="octicon octicon-triangle-down" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                </path>
               </svg>
              </summary>
              <details-menu class="SelectMenu right-0" role="menu" src="/Sumeet460/Fear-Conditioning-Genomics/lists">
               <div class="SelectMenu-modal">
                <header class="SelectMenu-header">
                 <h4 class="SelectMenu-title f5" id="user-lists-menu">
                  Lists
                 </h4>
                 <button aria-label="Close menu" class="SelectMenu-closeButton" data-toggle-for="details-user-list-45267387-unstarred" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                </header>
                <div class="d-flex flex-column flex-1 overflow-hidden" id="filter-menu-eeda37">
                 <div class="SelectMenu-list">
                  <include-fragment aria-label="Loading" class="SelectMenu-loading">
                   <svg class="anim-rotate" data-view-component="true" fill="none" height="32" role="menuitem" style="box-sizing: content-box; color: var(--color-icon-primary);" viewbox="0 0 16 16" width="32">
                    <circle cx="8" cy="8" fill="none" r="7" stroke="currentColor" stroke-opacity="0.25" stroke-width="2" vector-effect="non-scaling-stroke">
                    </circle>
                    <path d="M15 8a7.002 7.002 0 00-7-7" stroke="currentColor" stroke-linecap="round" stroke-width="2" vector-effect="non-scaling-stroke">
                    </path>
                   </svg>
                  </include-fragment>
                 </div>
                </div>
               </div>
              </details-menu>
             </details>
            </div>
           </div>
          </li>
          <li>
          </li>
         </ul>
        </div>
       </div>
       <div class="container-xl">
        <div data-turbo-replace="" id="responsive-meta-container">
         <div class="d-block d-md-none mb-2 px-3 px-md-4 px-lg-5">
          <div class="d-flex gap-2 mt-n3 mb-3 flex-wrap">
           <div class="d-flex flex-row gap-2">
            <notifications-list-subscription-form data-action="notifications-dialog-label-toggled:notifications-list-subscription-form#handleDialogLabelToggle">
             <details class="details-reset details-overlay details-overlay-dark" data-action="toggle:notifications-list-subscription-form#detailsToggled" data-target="notifications-list-subscription-form.details">
              <summary aria-label="Notifications" class="Button--iconOnly Button--secondary Button--medium Button p-0" data-ga-click="Repository, click Watch settings, action:files#disambiguate" data-hydro-click='{"event_type":"repository.click","payload":{"target":"WATCH_BUTTON","repository_id":45267387,"originating_url":"https://github.com/Sumeet460/Fear-Conditioning-Genomics","user_id":126681664}}' data-hydro-click-hmac="f317812d751f099352ee9c2695a649744de9d63c38175e33d89cdc9ce7e87858" data-view-component="true" id="notifications-button">
               <span class="Button-content">
                <span class="Button-label">
                 <span data-menu-button="">
                  <span data-target="notifications-list-subscription-form.unwatchButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.stopIgnoringButtonCopy" hidden="">
                   <svg aria-hidden="true" class="octicon octicon-bell-slash" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="m4.182 4.31.016.011 10.104 7.316.013.01 1.375.996a.75.75 0 1 1-.88 1.214L13.626 13H2.518a1.516 1.516 0 0 1-1.263-2.36l1.703-2.554A.255.255 0 0 0 3 7.947V5.305L.31 3.357a.75.75 0 1 1 .88-1.214Zm7.373 7.19L4.5 6.391v1.556c0 .346-.102.683-.294.97l-1.703 2.556a.017.017 0 0 0-.003.01c0 .005.002.009.005.012l.006.004.007.001ZM8 1.5c-.997 0-1.895.416-2.534 1.086A.75.75 0 1 1 4.38 1.55 5 5 0 0 1 13 5v2.373a.75.75 0 0 1-1.5 0V5A3.5 3.5 0 0 0 8 1.5ZM8 16a2 2 0 0 1-1.985-1.75c-.017-.137.097-.25.235-.25h3.5c.138 0 .252.113.235.25A2 2 0 0 1 8 16Z">
                    </path>
                   </svg>
                  </span>
                  <span data-target="notifications-list-subscription-form.watchButtonCopy">
                   <svg aria-hidden="true" class="octicon octicon-eye" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                    </path>
                   </svg>
                  </span>
                 </span>
                </span>
               </span>
              </summary>
              <details-dialog aria-label="Notification settings" class="anim-fade-in fast Box Box--overlay">
               <div data-target="notifications-list-subscription-form.menu">
                <div class="Box-header">
                 <button aria-label="Close dialog" class="Box-btn-octicon btn-octicon float-right" data-action="click:notifications-list-subscription-form#closeMenu" data-close-dialog="" type="button">
                  <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
                   </path>
                  </svg>
                 </button>
                 <h3 class="Box-title">
                  Notifications
                 </h3>
                </div>
                <div class="SelectMenu-list mb-0" role="menu">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitForm" data-target="notifications-list-subscription-form.form" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="kmuGjAOFn6oy4kIFg2SBmqGNx0IPT8-ZEqvCBGzQBJuKq4r2HTrqq97xdQ9fhAft_qhUV9oVZE0wUDDwO2G6Ag">
                   <input name="repository_id" type="hidden" value="45267387"/>
                   <button aria-checked="true" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="included">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Participating and @mentions
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Only receive notifications from this repository when participating or @mentioned.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="subscribed">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      All Activity
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Notified of all notifications on this repository.
                     </div>
                    </div>
                   </button>
                   <button aria-checked="false" class="SelectMenu-item flex-items-start" data-targets="notifications-list-subscription-form.subscriptionButtons" name="do" role="menuitemradio" type="submit" value="ignore">
                    <span class="f5">
                     <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                      </path>
                     </svg>
                    </span>
                    <div>
                     <div class="f5 text-bold">
                      Ignore
                     </div>
                     <div class="text-small color-fg-muted text-normal pb-1">
                      Never be notified.
                     </div>
                    </div>
                   </button>
                  </input>
                 </form>
                 <button aria-checked="false" aria-haspopup="true" class="SelectMenu-item flex-items-start pr-3" data-action="click:notifications-list-subscription-form#openCustomDialog" data-target="notifications-list-subscription-form.customButton" role="menuitemradio" type="button">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-check SelectMenu-icon SelectMenu-icon--check" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
                    </path>
                   </svg>
                  </span>
                  <div>
                   <div class="d-flex flex-items-start flex-justify-between">
                    <div class="f5 text-bold">
                     Custom
                    </div>
                    <div class="f5 pr-1">
                     <svg aria-hidden="true" class="octicon octicon-arrow-right" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                      <path d="M8.22 2.97a.75.75 0 0 1 1.06 0l4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l2.97-2.97H3.75a.75.75 0 0 1 0-1.5h7.44L8.22 4.03a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </div>
                   <div class="text-small color-fg-muted text-normal pb-1">
                    Select events you want to be notified of in addition to participating and @mentions.
                   </div>
                  </div>
                 </button>
                 <div class="px-3 py-2 d-flex color-bg-subtle flex-items-center">
                  <span class="f5">
                   <svg aria-hidden="true" class="octicon octicon-device-mobile SelectMenu-icon SelectMenu-icon--device-mobile" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M3.75 0h8.5C13.216 0 14 .784 14 1.75v12.5A1.75 1.75 0 0 1 12.25 16h-8.5A1.75 1.75 0 0 1 2 14.25V1.75C2 .784 2.784 0 3.75 0ZM3.5 1.75v12.5c0 .138.112.25.25.25h8.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25ZM8 13a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z">
                    </path>
                   </svg>
                  </span>
                  <span classname="text-small color-fg-muted text-normal pb-1">
                   Get push notifications on
                   <a href="https://apps.apple.com/app/apple-store/id1477376905?ct=watch-dropdown&amp;mt=8&amp;pt=524675" rel="noopener noreferrer" target="_blank">
                    iOS
                   </a>
                   or
                   <a href="https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dwatch-dropdown%26utm_medium%3Dweb%26utm_source%3Dgithub" rel="noopener noreferrer" target="_blank">
                    Android
                   </a>
                   .
                  </span>
                 </div>
                </div>
               </div>
               <div data-target="notifications-list-subscription-form.customDialog" hidden="">
                <header class="SelectMenu-header pb-0 border-bottom-0 px-2 px-sm-3">
                 <h1 class="f3 SelectMenu-title d-inline-flex">
                  <button aria-label="Return to menu" class="color-bg-default border-0 px-2 py-0 m-0 Link--secondary f5" data-action="click:notifications-list-subscription-form#closeCustomDialog" type="button">
                   <svg aria-hidden="true" class="octicon octicon-arrow-left" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                    <path d="M7.78 12.53a.75.75 0 0 1-1.06 0L2.47 8.28a.75.75 0 0 1 0-1.06l4.25-4.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L4.81 7h7.44a.75.75 0 0 1 0 1.5H4.81l2.97 2.97a.75.75 0 0 1 0 1.06Z">
                    </path>
                   </svg>
                  </button>
                  Custom
                 </h1>
                </header>
                <div class="overflow-visible">
                 <!-- '"` -->
                 <!-- </textarea></xmp> -->
                 <form accept-charset="UTF-8" action="/notifications/subscribe" data-action="submit:notifications-list-subscription-form#submitCustomForm" data-target="notifications-list-subscription-form.customform" data-turbo="false" method="post">
                  <input autocomplete="off" name="authenticity_token" type="hidden" value="NElLVPUGkKWNI6ikyf_hKiBRI3f17hpJ21Q0boxHtBQsiUcu67nlpGEwn64VH2ddf3SwYiC0sZ35r8aa2_YKjQ">
                   <input name="repository_id" type="hidden" value="45267387"/>
                   <fieldset>
                    <legend>
                     <div class="text-small color-fg-muted pt-0 pr-3 pb-3 pl-6 pl-sm-5 border-bottom mb-3">
                      Select events you want to be notified of in addition to participating and @mentions.
                     </div>
                    </legend>
                    <div data-target="notifications-list-subscription-form.labelInputs">
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Issue"/>
                      Issues
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="PullRequest"/>
                      Pull requests
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Release"/>
                      Releases
                     </label>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input aria-describedby="Discussion-disabled" aria-disabled="true" data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="Discussion"/>
                      Discussions
                     </label>
                     <div class="color-fg-muted" id="Discussion-disabled">
                      &gt;
                          Discussions are not enabled for this repository
                     </div>
                    </div>
                    <div class="form-checkbox mr-3 ml-6 ml-sm-5 mb-2 mt-0">
                     <label class="f5 text-normal">
                      <input data-action="change:notifications-list-subscription-form#threadTypeCheckboxesUpdated" data-targets="notifications-list-subscription-form.threadTypeCheckboxes" name="thread_types[]" type="checkbox" value="SecurityAlert"/>
                      Security alerts
                     </label>
                    </div>
                   </fieldset>
                   <div class="pt-2 pb-3 px-3 d-flex flex-justify-start flex-row-reverse">
                    <button class="Button--primary Button--medium Button ml-2" data-target="notifications-list-subscription-form.customSubmit" data-view-component="true" disabled="disabled" name="do" type="submit" value="custom">
                     <span class="Button-content">
                      <span class="Button-label">
                       Apply
                      </span>
                     </span>
                    </button>
                    <button class="Button--secondary Button--medium Button" data-action="click:notifications-list-subscription-form#resetForm" data-close-dialog="" data-view-component="true" type="button">
                     <span class="Button-content">
                      <span class="Button-label">
                       Cancel
                      </span>
                     </span>
                    </button>
                   </div>
                  </input>
                 </form>
                </div>
               </div>
              </details-dialog>
             </details>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="description" data-view-component="true" for="notifications-button" id="tooltip-57f2e567-44ef-4f9d-b3d0-520852ef0761" popover="manual">
              Notification settings
             </tool-tip>
            </notifications-list-subscription-form>
            <div class="BtnGroup d-flex" data-view-component="true">
             <a aria-labelledby="tooltip-488b2c81-1a6a-4b3f-980e-161c23b56377" class="Button Button--iconOnly Button--secondary Button--medium" data-ga-click="Repository, show fork modal, action:files#disambiguate; text:Fork" data-hydro-click='{"event_type":"repository.click","payload":{"target":"FORK_BUTTON","repository_id":45267387,"originating_url":"https://github.com/Sumeet460/Fear-Conditioning-Genomics","user_id":126681664}}' data-hydro-click-hmac="4aacf64f3f6c58f9d9a689c34ddd50164c86bcf9dccf2be32b91ab7aa7e8ae1f" data-view-component="true" group_item="true" href="/Sumeet460/Fear-Conditioning-Genomics/fork" id="fork-icon-button" variant="small">
              <svg aria-hidden="true" class="octicon octicon-repo-forked Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
               <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
               </path>
              </svg>
             </a>
             <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="fork-icon-button" id="tooltip-488b2c81-1a6a-4b3f-980e-161c23b56377" popover="manual">
              Fork your own copy of Sumeet460/Fear-Conditioning-Genomics
             </tool-tip>
            </div>
            <div class="js-toggler-container starring-container" data-view-component="true">
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/Sumeet460/Fear-Conditioning-Genomics/unstar" class="starred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="ncz-4kxbOSPyHxH2UiE0uoTnpM-zapdwg7Rf4EHxyb4R8_HFQsFd3Ci8mmtALwPolDQAzPCNhX8qYd7ufnZJzw">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-2a7887bb-349c-4c72-8005-d7dccef5b736" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target starred-button-icon" data-ga-click="Repository, click unstar button, action:files#disambiguate; text:Unstar" data-hydro-click='{"event_type":"repository.click","payload":{"target":"UNSTAR_BUTTON","repository_id":45267387,"originating_url":"https://github.com/Sumeet460/Fear-Conditioning-Genomics","user_id":126681664}}' data-hydro-click-hmac="01e6eb455c36b3503eb4ec9686916fcee6b390ce521ba08ce7ff105223b06d3c" data-view-component="true" id="icon-button-b21242b9-27fe-4eed-a7fe-75e8db98c84d" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star-fill Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-b21242b9-27fe-4eed-a7fe-75e8db98c84d" id="tooltip-2a7887bb-349c-4c72-8005-d7dccef5b736" popover="manual">
                Unstar this repository
               </tool-tip>
              </input>
             </form>
             <!-- '"` -->
             <!-- </textarea></xmp> -->
             <form accept-charset="UTF-8" action="/Sumeet460/Fear-Conditioning-Genomics/star" class="unstarred js-social-form" data-turbo="false" method="post">
              <input autocomplete="off" name="authenticity_token" type="hidden" value="QwBI0dQeBbv-WBNaIdHxwAG8TTeq7NWiCKLf_eb5IBXF4Oomy1iNJeldWoptYmOeoVA_6DnoKUnwzCbmnNK2Wg">
               <input name="context" type="hidden" value="repository"/>
               <button aria-labelledby="tooltip-84b3229a-09c4-4e49-a7a6-4b5b4ea4dc9b" class="Button Button--iconOnly Button--secondary Button--medium js-toggler-target" data-ga-click="Repository, click star button, action:files#disambiguate; text:Star" data-hydro-click='{"event_type":"repository.click","payload":{"target":"STAR_BUTTON","repository_id":45267387,"originating_url":"https://github.com/Sumeet460/Fear-Conditioning-Genomics","user_id":126681664}}' data-hydro-click-hmac="3101c5bd66ea9f0fb177196fcc3a2d97f78e877fe2d49e03962587b3a7da0d38" data-view-component="true" id="icon-button-e5eef20b-7358-4b4b-aad1-bb0192141364" type="submit">
                <svg aria-hidden="true" class="octicon octicon-star Button-visual" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                 <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                 </path>
                </svg>
               </button>
               <tool-tip class="sr-only position-absolute" data-direction="s" data-type="label" data-view-component="true" for="icon-button-e5eef20b-7358-4b4b-aad1-bb0192141364" id="tooltip-84b3229a-09c4-4e49-a7a6-4b5b4ea4dc9b" popover="manual">
                Star this repository
               </tool-tip>
              </input>
             </form>
            </div>
           </div>
           <div class="d-flex flex-row gap-2">
           </div>
          </div>
          <div class="mb-3">
           <ul aria-label="Repository details" class="d-flex flex-wrap mb-2 gap-2">
            <a class="Link--secondary no-underline d-block mr-2" href="/Sumeet460/Fear-Conditioning-Genomics/stargazers" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-star mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             stars
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/Sumeet460/Fear-Conditioning-Genomics/forks" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-repo-forked mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              0
             </span>
             forks
            </a>
            <div class="color-fg-muted mr-2" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-eye mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
              </path>
             </svg>
             <span class="text-bold color-fg-default">
              1
             </span>
             watching
            </div>
            <a class="Link--secondary no-underline d-block mr-2" href="/Sumeet460/Fear-Conditioning-Genomics/branches" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-git-branch mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
              </path>
             </svg>
             <include-fragment src="/Sumeet460/Fear-Conditioning-Genomics/branch-count">
              <span>
               Branches
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/Sumeet460/Fear-Conditioning-Genomics/tags" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-tag mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
              </path>
             </svg>
             <include-fragment src="/Sumeet460/Fear-Conditioning-Genomics/tag-count">
              <span>
               Tags
              </span>
             </include-fragment>
            </a>
            <a class="Link--secondary no-underline d-block mr-2" href="/Sumeet460/Fear-Conditioning-Genomics/activity" role="listitem">
             <svg aria-hidden="true" class="octicon octicon-pulse mr-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
              </path>
             </svg>
             <span>
              Activity
             </span>
            </a>
           </ul>
           <div class="mb-2 d-flex color-fg-muted">
            <div class="d-flex flex-items-center" style="height: 21px">
             <svg aria-hidden="true" class="octicon octicon-globe flex-shrink-0 mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
              <path d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM5.78 8.75a9.64 9.64 0 0 0 1.363 4.177c.255.426.542.832.857 1.215.245-.296.551-.705.857-1.215A9.64 9.64 0 0 0 10.22 8.75Zm4.44-1.5a9.64 9.64 0 0 0-1.363-4.177c-.307-.51-.612-.919-.857-1.215a9.927 9.927 0 0 0-.857 1.215A9.64 9.64 0 0 0 5.78 7.25Zm-5.944 1.5H1.543a6.507 6.507 0 0 0 4.666 5.5c-.123-.181-.24-.365-.352-.552-.715-1.192-1.437-2.874-1.581-4.948Zm-2.733-1.5h2.733c.144-2.074.866-3.756 1.58-4.948.12-.197.237-.381.353-.552a6.507 6.507 0 0 0-4.666 5.5Zm10.181 1.5c-.144 2.074-.866 3.756-1.58 4.948-.12.197-.237.381-.353.552a6.507 6.507 0 0 0 4.666-5.5Zm2.733-1.5a6.507 6.507 0 0 0-4.666-5.5c.123.181.24.365.353.552.714 1.192 1.436 2.874 1.58 4.948Z">
              </path>
             </svg>
            </div>
            <span class="flex-auto min-width-0 width-fit">
             Public repository
            </span>
           </div>
          </div>
         </div>
        </div>
        <div class="border-bottom mx-xl-5">
        </div>
       </div>
      </div>
      <turbo-frame class="" data-turbo-action="advance" id="repo-content-turbo-frame" target="_top">
       <div class="repository-content" id="repo-content-pjax-container">
        <a class="d-none js-github-dev-shortcut" data-hotkey=".,Mod+Alt+." href="https://github.dev/">
         Open in github.dev
        </a>
        <a class="d-none js-github-dev-new-tab-shortcut" data-hotkey="Shift+.,Shift+&gt;,&gt;" href="https://github.dev/" rel="noopener noreferrer" target="_blank">
         Open in a new github.dev tab
        </a>
        <a class="d-none" data-hotkey=",,Mod+Alt+," href="/codespaces/new/Sumeet460/Fear-Conditioning-Genomics?resume=1" target="_blank">
         Open in codespace
        </a>
        <h1 class="sr-only">
         Sumeet460/Fear-Conditioning-Genomics
        </h1>
        <div class="clearfix container-xl px-md-4 px-lg-5 px-3">
         <div>
          <div aria-hidden="" class="mt-0 pb-3" hidden="" id="spoof-warning">
           <div class="flash flash-warn mt-0 clearfix" data-view-component="true">
            <svg aria-hidden="true" class="octicon octicon-alert float-left mt-1" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
             <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
             </path>
            </svg>
            <div class="overflow-hidden">
             This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.
            </div>
           </div>
          </div>
          <include-fragment data-test-selector="spoofed-commit-check" src="/Sumeet460/Fear-Conditioning-Genomics/spoofed_commit_check/0eda3eef0942edca3fd071a9f3e4568b87a72270">
          </include-fragment>
          <div class="Layout Layout--flowRow-until-md react-repos-overview-margin Layout--sidebarPosition-end Layout--sidebarPosition-flowRow-end" data-view-component="true" style="max-width: 100%">
           <div class="Layout-main" data-view-component="true">
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/react-lib-1fbfc5be2c18.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_octicons-react_dist_index_esm_js-node_modules_primer_react_lib-es-2e8e7c-a58d7c11e858.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Box_Box_js-8f8c5e2a2cbf.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Button_Button_js-d5726d25c548.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionList_index_js-1501d3ef83c2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Overlay_Overlay_js-node_modules_primer_react_lib-es-fa1130-829932cf63db.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_ActionMenu_ActionMenu_js-eaf74522e470.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_github_catalyst_lib_index_js-node_modules_github_hydro-analytics-client_-978abc0-add939c751ce.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_react-router-dom_dist_index_js-3b41341d50fe.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_Dialog_js-node_modules_primer_react_lib-esm_TabNav_-8321f5-2969c7508f3a.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_UnderlineNav_index_js-89fa5806aa3c.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/vendors-node_modules_primer_react_lib-esm_AvatarStack_AvatarStack_js-node_modules_primer_reac-558b50-c6ea7afb7179.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_react-core_create-browser-history_ts-ui_packages_react-core_AppContextProvider_ts-809ab9-5bc018b15303.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_paths_index_ts-654469d743cd.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_ref-selector_RefSelector_tsx-dbbdef4348e2.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/ui_packages_commit-attribution_index_ts-ui_packages_commit-checks-status_index_ts-ui_packages-a73d65-239b92c64d22.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/app_assets_modules_react-shared_hooks_use-canonical-object_ts-ui_packages_code-view-shared_ho-3e492a-cde4692d0c71.js" type="application/javascript">
            </script>
            <script crossorigin="anonymous" defer="defer" src="https://github.githubassets.com/assets/repos-overview-009ec6ff31b8.js" type="application/javascript">
            </script>
            <react-partial data-ssr="true" partial-name="repos-overview">
             <script data-target="react-partial.embeddedData" type="application/json">
              {"props":{"initialPayload":{"allShortcutsEnabled":true,"path":"/","repo":{"id":45267387,"defaultBranch":"master","name":"Fear-Conditioning-Genomics","ownerLogin":"Sumeet460","currentUserCanPush":false,"isFork":false,"isEmpty":false,"createdAt":"2015-10-31T01:52:20.000+08:00","ownerAvatar":"https://avatars.githubusercontent.com/u/8104884?v=4","public":true,"private":false,"isOrgOwned":false},"currentUser":{"id":126681664,"login":"ninenia","userEmail":"2945518981@qq.com"},"refInfo":{"name":"master","listCacheKey":"v0:1446227541.0","canEdit":true,"refType":"branch","currentOid":"0eda3eef0942edca3fd071a9f3e4568b87a72270"},"tree":{"items":[{"name":"README.md","path":"README.md","contentType":"file"}],"templateDirectorySuggestionUrl":null,"readme":null,"totalCount":1,"showBranchInfobar":false},"fileTree":null,"fileTreeProcessingTime":null,"foldersToFetch":[],"treeExpanded":false,"symbolsExpanded":false,"isOverview":true,"overview":{"banners":{"shouldRecommendReadme":false,"isPersonalRepo":false,"showUseActionBanner":false,"actionSlug":null,"actionId":null,"showProtectBranchBanner":false,"recentlyTouchedDataChannel":null,"publishBannersInfo":{"dismissActionNoticePath":"/settings/dismiss-notice/publish_action_from_repo","releasePath":"/Sumeet460/Fear-Conditioning-Genomics/releases/new?marketplace=true","showPublishActionBanner":false},"interactionLimitBanner":null,"showInvitationBanner":false,"inviterName":null},"codeButton":{"contactPath":"/contact","isEnterprise":false,"local":{"protocolInfo":{"httpAvailable":true,"sshAvailable":true,"httpUrl":"https://github.com/Sumeet460/Fear-Conditioning-Genomics.git","showCloneWarning":false,"sshUrl":"git@github.com:Sumeet460/Fear-Conditioning-Genomics.git","sshCertificatesRequired":false,"sshCertificatesAvailable":false,"ghCliUrl":"gh repo clone Sumeet460/Fear-Conditioning-Genomics","defaultProtocol":"http","newSshKeyUrl":"/settings/ssh/new","setProtocolPath":"/users/set_protocol"},"platformInfo":{"cloneUrl":"x-github-client://openRepo/https://github.com/Sumeet460/Fear-Conditioning-Genomics","showVisualStudioCloneButton":true,"visualStudioCloneUrl":"git-client://clone?repo=https%3A%2F%2Fgithub.com%2FSumeet460%2FFear-Conditioning-Genomics","showXcodeCloneButton":false,"xcodeCloneUrl":"https://developer.apple.com","zipballUrl":"/Sumeet460/Fear-Conditioning-Genomics/archive/refs/heads/master.zip"}},"repoPolicyInfo":{"allowed":true,"canBill":true,"changesWouldBeSafe":true,"disabledByBusiness":false,"disabledByOrganization":false,"hasIpAllowLists":false},"currentUserIsEnterpriseManaged":false,"enterpriseManagedBusinessName":null,"codespacesEnabled":true,"hasAccessToCodespaces":true},"popovers":{"rename":null,"renamedParentRepo":null},"commitCount":"39","overviewFiles":[{"displayName":"README.md","repoName":"Fear-Conditioning-Genomics","refName":"master","path":"README.md","preferredFileType":"readme","tabName":"README","richText":"\u003carticle class=\"markdown-body entry-content container-lg\" itemprop=\"text\"\u003e\u003cp dir=\"auto\"\u003eFear-Conditioning-Genomics\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTABLE OF CONTENTS\u003c/p\u003e\n\u003col dir=\"auto\"\u003e\n\u003cli\u003eALIGNMENT: STAR\u003c/li\u003e\n\u003cli\u003eRNASEQ ANALYSIS, TRAPSEQ:DIFFERENTIAL EXPRESSION:CUFFDIFF\u003c/li\u003e\n\u003cli\u003eRNASEQ ANALYSIS, TRAPSEQ:DIFFERENTIAL EXPRESSION:DESEQ2\u003c/li\u003e\n\u003cli\u003eRNASEQ ANALYSIS, TRAPSEQ:DIFFERENTIAL SPLICING:DESEQ2/DEXSEQ\u003c/li\u003e\n\u003cli\u003eRNASEQ ANALYSIS, TRAPSEQ:PATHWAT ANALYSIS:CONSENSUS PATH DATABASE\u003c/li\u003e\n\u003cli\u003e5HMC ANALYSIS:PEAK CALLING:MACS\u003c/li\u003e\n\u003cli\u003ePARCE OUT REGIONS OF 5HMC, EXPRESSION, AND ISOFORM REGULATION:BEDTOOLS\u003c/li\u003e\n\u003cli\u003eANNOTATION ANALYSIS:HOMER\u003c/li\u003e\n\u003cli\u003eMOTIF ANALYSIS:HOMER:MOTIF ID AND LOCALIZATION\u003c/li\u003e\n\u003cli\u003e5HMC ENRICHMENT ANALYSIS:HOMER:HISTOGRAMS AND SCATTERPLOTS: MOTIFS, PROMOTERS, EXONS, INTRONS, EXONS-INTRONS, CTCF, OTHER GENOMIC REGIONS (CPG ISLANDS, EXONS, LNCRNA, PSEUDOGENES, LOW COMPLEXITY REGIONS, SIMPLE REPEATS, SINES, SATELLITES\n5HMC AND ALTERNATIVE SPLICING\u003c/li\u003e\n\u003cli\u003eFAST ANALYSIS: ANNOTATING REGIONS OF THE GENOME BASED ON DISEASE RELEVANT SNP CONTENT\u003c/li\u003e\n\u003cli\u003eHomology ANALYSIS:PHASTCONS\u003c/li\u003e\n\u003cli\u003eGWAS \u0026amp; GxE:IMPUTE2\u003c/li\u003e\n\u003c/ol\u003e\n\u003cp dir=\"auto\"\u003e###1. ALIGNMENT: STAR\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eBase Files\u003c/strong\u003e\n\u003cem\u003eWholeGenomeFasta Files:\u003c/em\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eensembl\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eMus_musculus.GRCm38.dna.toplevel.fa\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eUCSC\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003emm10.fa.align\u003c/li\u003e\n\u003cli\u003emm9 -\u0026gt; genome.fa\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eGTF Files:\u003c/em\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eensembl\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003eMus_musculus.GRCm38.79.gtf (genes)\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003cli\u003eUCSC\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e/home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked.gtf (repeat masked annotations only downloaded from UCSC)\u003c/li\u003e\n\u003cli\u003e/home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/genes.gtf\u003c/li\u003e\n\u003cli\u003e/home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gtf (Repeat masked annotation concatenated with the normal annotation)\u003c/li\u003e\n\u003c/ul\u003e\n\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eIndices:\u003c/em\u003e\u003c/p\u003e\n\u003cul dir=\"auto\"\u003e\n\u003cli\u003e/home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex (this one was created concatenating the normal UCSC mm10 gtf w/ the RepeatMasked GTF)\u003c/li\u003e\n\u003c/ul\u003e\n\u003cp dir=\"auto\"\u003eGenerate STAR Genome Index using repeat masked gtf file from UCSC, cat with a normal gtf file  - in order to capture the repetive elements in RepBase in the index\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runMode genomeGenerate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --genomeFastaFiles /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Sequence/WholeGenomeFasta/genome.fa --runThreadN 25 --sjdbGTFfile /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gtf --sjdbOverhang 100 --limitSjdbInsertNsj 5000000\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runMode genomeGenerate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --genomeFastaFiles /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Sequence/WholeGenomeFasta/genome.fa --runThreadN 25 --sjdbGTFfile /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gtf --sjdbOverhang 100 --limitSjdbInsertNsj 5000000\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eAlignment Runs: UCSC\u003c/strong\u003e\nAutomated - didn't implement this, but would've save a lot of time...\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"for f in `cat files`; do STAR --genomeDir ../STAR/ENSEMBL.homo_sapiens.release-75 \\\n--readFilesIn fastq/$f\\_1.fastq fastq/$f\\_2.fastq \\\n--runThreadN 12 --outFileNamePrefix aligned/$f.; done\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efor f in `cat files`; do STAR --genomeDir ../STAR/ENSEMBL.homo_sapiens.release-75 \\\n--readFilesIn fastq/$f\\_1.fastq fastq/$f\\_2.fastq \\\n--runThreadN 12 --outFileNamePrefix aligned/$f.; done\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e19_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 19_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/19_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/19_Homecage_ACAGTG_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/19_Homecage/19_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 19_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/19_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/19_Homecage_ACAGTG_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/19_Homecage/19_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e20_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/\nmkdir 20_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/20_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/20_Homecage_GCCAAT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/20_Homecage/20_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/\nmkdir 20_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/20_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/20_Homecage_GCCAAT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/20_Homecage/20_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e21_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 21_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/21_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/21_Homecage_CAGATC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/21_Homecage/21_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 21_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/21_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/21_Homecage_CAGATC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/21_Homecage/21_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e22_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 22_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/22_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/22_Homecage_CTTGTA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/22_Homecage/22_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 22_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/22_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/22_Homecage_CTTGTA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/22_Homecage/22_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e23_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 23_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/23_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/23_Homecage_AGTCAA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/23_Homecage/23_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 23_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/23_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/23_Homecage_AGTCAA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/23_Homecage/23_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e24_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 24_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/24_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/24_Homecage_AGTTCC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/24_Homecage/24_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 24_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/24_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/24_Homecage_AGTTCC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/24_Homecage/24_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e10_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 10_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/10_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/10_FC-Alone_AGTTCC_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/10_FC-Alone/10_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 10_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/10_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/10_FC-Alone_AGTTCC_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/10_FC-Alone/10_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e11_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 11_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/11_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/11_FC-Alone_CGATGT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/11_FC-Alone/11_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 11_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/11_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/11_FC-Alone_CGATGT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/11_FC-Alone/11_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e12_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 12_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/12_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/12_FC-Alone_TGACCA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/12_FC-Alone/12_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 12_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/12_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/12_FC-Alone_TGACCA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/12_FC-Alone/12_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e9_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 9_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/9_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/9_FC-Alone_AGTCAA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/9_FC-Alone/9_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 9_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/9_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/9_FC-Alone_AGTCAA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/9_FC-Alone/9_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e8_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 8_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/8_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/8_FC-Alone_CTTGTA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/8_FC-Alone/8_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 8_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/8_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/8_FC-Alone_CTTGTA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/8_FC-Alone/8_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e1_IMMO-FC\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 1_IMMO-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/1_IMMO-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/1_IMMO-FC_CGATGT_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/1_IMMO-FC/1_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker\nmkdir 1_IMMO-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/1_IMMO-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/1_IMMO-FC_CGATGT_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/1_IMMO-FC/1_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e2_IMMO-FC\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"mkdir 2_IMMO-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/2_IMMO-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/2_IMMO-FC_TGACCA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/2_IMMO-FC/2_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emkdir 2_IMMO-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/2_IMMO-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/2_IMMO-FC_TGACCA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/2_IMMO-FC/2_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e3_IMMO-FC\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/\nmkdir 3_IMMO-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/3_IMMO-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/3_IMMO-FC_ACAGTG_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/3_IMMO-FC/3_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/\nmkdir 3_IMMO-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/3_IMMO-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/3_IMMO-FC_ACAGTG_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/3_IMMO-FC/3_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e4_IMMO-FC\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/\nmkdir 4_IMMO-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/4_IMMO-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/4_IMMO-FC_GCCAAT_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/4_IMMO-FC/4_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/\nmkdir 4_IMMO-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/4_IMMO-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/4_IMMO-FC_GCCAAT_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/4_IMMO-FC/4_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e1-935-HC (RNAseq Replication)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 1-935-HC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/1-935-HC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/1-935-HC/1-935-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/1-935-HC_GTCCGC_L008_R1_001.fastq.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 1-935-HC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/1-935-HC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/1-935-HC/1-935-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/1-935-HC_GTCCGC_L008_R1_001.fastq.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e2-935-HC (RNAseq Replication)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 2-935-HC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/2-935-HC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/2-935-HC/2-935-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/2-935-HC_GTGAAA_L008_R1_001.fastq.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 2-935-HC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/2-935-HC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/2-935-HC/2-935-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/2-935-HC_GTGAAA_L008_R1_001.fastq.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e3-940-FC (RNAseq Replication)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 3-940-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/3-940-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/3-940-FC/3-940-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/3-940-FC_ATCACG_L008_R1_001.fastq.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 3-940-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/3-940-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/3-940-FC/3-940-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/3-940-FC_ATCACG_L008_R1_001.fastq.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e4-940-FC (RNAseq Replication)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 4-940-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/4-940-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/4-940-FC/4-940-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/4-940-FC_TTAGGC_L008_R1_001.fastq.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 4-940-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/4-940-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/4-940-FC/4-940-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/4-940-FC_TTAGGC_L008_R1_001.fastq.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e5-938-FC (RNAseq Replication)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 5-938-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/5-938-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/5-938-FC/5-938-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/5-938-FC_ACTTGA_L008_R1_001.fastq.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 5-938-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/5-938-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/5-938-FC/5-938-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/5-938-FC_ACTTGA_L008_R1_001.fastq.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e6-938-FC (RNAseq Replication)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 6-938-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/6-938-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/6-938-FC/6-938-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/6-938-FC_GATCAG_L008_R1_001.fastq.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 6-938-FC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/6-938-FC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/6-938-FC/6-938-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/6-938-FC_GATCAG_L008_R1_001.fastq.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e7-947-HC (RNAseq Replication)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 7-947-HC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/7-947-HC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/7-947-HC/7-947-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/7-947-HC_TAGCTT_L008_R1_001.fastq.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 7-947-HC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/7-947-HC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/7-947-HC/7-947-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/7-947-HC_TAGCTT_L008_R1_001.fastq.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e8-947-HC (RNAseq Replication)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 8-947-HC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/8-947-HC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/8-947-HC/8-947-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/8-947-HC_GGCTAC_L008_R1_001.fastq.gz\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC\nmkdir 8-947-HC\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/8-947-HC\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/8-947-HC/8-947-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/8-947-HC_GGCTAC_L008_R1_001.fastq.gz\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSTAR Alignment: Combined HC, FC 5hmC Seq with UCSC\u003c/strong\u003e*\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC\nmkdir FC_5hmC_ALLreps\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/FC_5hmC_ALLreps\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/FC_5hmC_ALLreps/FC_5hmC_ALLreps_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/fastq_combined_files/FC_5hmC_ALLreps.fastq \u0026amp;\n\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC\nmkdir HC_5hmC_ALLreps\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/HC_5hmC_ALLreps\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/HC_5hmC_ALLreps/HC_5hmC_ALLreps_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/fastq_combined_files/HC_5hmC_ALLreps.fastq \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC\nmkdir FC_5hmC_ALLreps\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/FC_5hmC_ALLreps\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/FC_5hmC_ALLreps/FC_5hmC_ALLreps_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/fastq_combined_files/FC_5hmC_ALLreps.fastq \u0026amp;\n\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC\nmkdir HC_5hmC_ALLreps\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/HC_5hmC_ALLreps\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/HC_5hmC_ALLreps/HC_5hmC_ALLreps_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/fastq_combined_files/HC_5hmC_ALLreps.fastq \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eSTAR Alignment: ensembl\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e19_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 19_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/19_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/19_Homecage_ACAGTG_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/19_Homecage/19_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 19_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/19_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/19_Homecage_ACAGTG_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/19_Homecage/19_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e20_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/\nmkdir 20_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/20_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/20_Homecage_GCCAAT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/20_Homecage/20_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/\nmkdir 20_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/20_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/20_Homecage_GCCAAT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/20_Homecage/20_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e21_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 21_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/21_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/21_Homecage_CAGATC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/21_Homecage/21_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 21_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/21_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/21_Homecage_CAGATC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/21_Homecage/21_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e22_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 22_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/22_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/22_Homecage_CTTGTA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/22_Homecage/22_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 22_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/22_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/22_Homecage_CTTGTA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/22_Homecage/22_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e23_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 23_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/23_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/23_Homecage_AGTCAA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/23_Homecage/23_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 23_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/23_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/23_Homecage_AGTCAA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/23_Homecage/23_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e24_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 24_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/24_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/24_Homecage_AGTTCC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/24_Homecage/24_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 24_Homecage\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/24_Homecage\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/24_Homecage_AGTTCC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/24_Homecage/24_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e10_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 10_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/10_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/10_FC-Alone_AGTTCC_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/10_FC-Alone/10_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 10_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/10_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/10_FC-Alone_AGTTCC_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/10_FC-Alone/10_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e11_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 11_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/11_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/11_FC-Alone_CGATGT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/11_FC-Alone/11_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 11_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/11_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/11_FC-Alone_CGATGT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/11_FC-Alone/11_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e12_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 12_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/12_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/12_FC-Alone_TGACCA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/12_FC-Alone/12_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 12_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/12_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/12_FC-Alone_TGACCA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/12_FC-Alone/12_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e9_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 9_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/9_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/9_FC-Alone_AGTCAA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/9_FC-Alone/9_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 9_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/9_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/9_FC-Alone_AGTCAA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/9_FC-Alone/9_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e8_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 8_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/8_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/8_FC-Alone_CTTGTA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/8_FC-Alone/8_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens\nmkdir 8_FC-Alone\ncd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/8_FC-Alone\n/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/8_FC-Alone_CTTGTA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/8_FC-Alone/8_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e###2. RNASEQ ANALYSIS:DIFFERENTIAL EXPRESSION:CUFFDIFF\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCUFFDIFF\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHC vs IMMO-FC Differential expression analysis: using only repeat masked gtf\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cuffdiff -o ./AmygHC_vs_AmygIMMO-FC_RepeatMasked_SMALL -L HC,IMMO-FC -p 25 -u /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked.gtf /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/19_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/20_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/21_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/22_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/23_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/24_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/1_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/2_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/3_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/4_IMMO-FC_Aligned.sortedByCoord.out.bam \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecuffdiff -o ./AmygHC_vs_AmygIMMO-FC_RepeatMasked_SMALL -L HC,IMMO-FC -p 25 -u /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked.gtf /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/19_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/20_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/21_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/22_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/23_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/24_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/1_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/2_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/3_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/4_IMMO-FC_Aligned.sortedByCoord.out.bam \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHC vs FC Differential expression analysis: using only repeat masked gtf\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cuffdiff -o ./AmygFC_vs_AmygHC_RepeatMasked_SMALL -L HC,FC -p 2 -u /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gtf /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/19_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/20_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/21_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/22_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/23_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/24_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/10_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/11_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/12_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/8_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/9_FC-Alone_Aligned.sortedByCoord.out.bam \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecuffdiff -o ./AmygFC_vs_AmygHC_RepeatMasked_SMALL -L HC,FC -p 2 -u /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gtf /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/19_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/20_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/21_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/22_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/23_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/24_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/10_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/11_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/12_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/8_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/9_FC-Alone_Aligned.sortedByCoord.out.bam \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFC vs IMMO DE\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cuffdiff -o ./AmygFC_vs_AmygIMMO_UCSC_genes -L FC,IMMOFC -p 25 -u /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/genes.gtf /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/10_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/11_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/12_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/8_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/9_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/1_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/2_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/3_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/4_IMMO-FC_Aligned.sortedByCoord.out.bam \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecuffdiff -o ./AmygFC_vs_AmygIMMO_UCSC_genes -L FC,IMMOFC -p 25 -u /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/genes.gtf /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/10_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/11_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/12_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/8_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/9_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/1_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/2_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/3_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/4_IMMO-FC_Aligned.sortedByCoord.out.bam \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\n\nAlignment Runs: Ensembl\n\n\n\n\n###3. RNASEQ ANALYSIS:DIFFERENTIAL EXPRESSION:DESEQ2\n\nse is the RangedSummarizedExperiment object containing counts and information about the samples. In this example design we control for batch and condition, which should be columns of colData(se)\n```{r}\ndds \u0026lt;- DESeqDataSet(se, design = ~ batch + condition)\ndds \u0026lt;- DESeq(dds)\nres \u0026lt;- results(dds, contrast=c(\u0026quot;condition\u0026quot;,\u0026quot;trt\u0026quot;,\u0026quot;con\u0026quot;))\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e\n\nAlignment Runs: Ensembl\n\n\n\n\n###3. RNASEQ ANALYSIS:DIFFERENTIAL EXPRESSION:DESEQ2\n\nse is the RangedSummarizedExperiment object containing counts and information about the samples. In this example design we control for batch and condition, which should be columns of colData(se)\n```{r}\ndds \u0026lt;- DESeqDataSet(se, design = ~ batch + condition)\ndds \u0026lt;- DESeq(dds)\nres \u0026lt;- results(dds, contrast=c(\"condition\",\"trt\",\"con\"))\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCombine counts from geneCounts function of STAR to generate count matrices:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eRead the Reads/Gene table\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"FC10Alone=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/10_FC-Alone/10_FC-Alone_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nFC11Alone=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/11_FC-Alone/11_FC-Alone_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nFC12Alone=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/12_FC-Alone/12_FC-Alone_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nHomecage19=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/19_Homecage/19_Homecage_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nHomecage20=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/20_Homecage/20_Homecage_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nHomecage21=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/21_Homecage/21_Homecage_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nHomecage22=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/22_Homecage/22_Homecage_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nHomecage23=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/23_Homecage/23_Homecage_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nHomecage24=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/24_Homecage/24_Homecage_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nFC8Alone=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/8_FC-Alone/8_FC-Alone_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nFC9Alone=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/9_FC-Alone/9_FC-Alone_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003eFC10Alone=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/10_FC-Alone/10_FC-Alone_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nFC11Alone=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/11_FC-Alone/11_FC-Alone_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nFC12Alone=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/12_FC-Alone/12_FC-Alone_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nHomecage19=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/19_Homecage/19_Homecage_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nHomecage20=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/20_Homecage/20_Homecage_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nHomecage21=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/21_Homecage/21_Homecage_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nHomecage22=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/22_Homecage/22_Homecage_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nHomecage23=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/23_Homecage/23_Homecage_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nHomecage24=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/24_Homecage/24_Homecage_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nFC8Alone=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/8_FC-Alone/8_FC-Alone_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nFC9Alone=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/9_FC-Alone/9_FC-Alone_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eReplication UCSC BAMs\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"HC8r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/8-947-HC/8-947-HC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nHC7r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/7-947-HC/7-947-HC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n \nFC6r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/6-938-FC/6-938-FC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n  \nFC5r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/5-938-FC/5-938-FC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nFC4r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/4-940-FC/4-940-FC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n  \nFC3r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/3-940-FC/3-940-FC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;) \n \nHC2r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/2-935-HC/2-935-HC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)  \n\nHC1r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/1-935-HC/1-935-HC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003eHC8r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/8-947-HC/8-947-HC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nHC7r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/7-947-HC/7-947-HC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n \nFC6r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/6-938-FC/6-938-FC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n  \nFC5r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/5-938-FC/5-938-FC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nFC4r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/4-940-FC/4-940-FC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n  \nFC3r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/3-940-FC/3-940-FC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\") \n \nHC2r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/2-935-HC/2-935-HC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")  \n\nHC1r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/1-935-HC/1-935-HC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eReplication ensemble BAMs\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"HC1r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/1-935-HC/1-935-HC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nFC3r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/3-940-FC/3-940-FC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n \nFC5r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/5-938-FC/5-938-FC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n  \nHC7r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/7-947-HC/7-947-HC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nHC2r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/2-935-HC/2-935-HC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n  \nFC4r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/4-940-FC/4-940-FC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;) \n \nFC6r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/6-938-FC/6-938-FC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)  \n\nHC8r=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/8-947-HC/8-947-HC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003eHC1r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/1-935-HC/1-935-HC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nFC3r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/3-940-FC/3-940-FC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n \nFC5r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/5-938-FC/5-938-FC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n  \nHC7r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/7-947-HC/7-947-HC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nHC2r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/2-935-HC/2-935-HC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n  \nFC4r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/4-940-FC/4-940-FC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\") \n \nFC6r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/6-938-FC/6-938-FC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")  \n\nHC8r=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/8-947-HC/8-947-HC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eIMMO - UCSC BAMs\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"IMMOFC3=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/3_IMMO-FC/3_IMMO-FC_ReadsPerGene.out.tab\u0026quot;, header = F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nIMMOFC1=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/1_IMMO-FC/1_IMMO-FC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nIMMOFC2=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/2_IMMO-FC/2_IMMO-FC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\n\nIMMOFC4=read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/4_IMMO-FC/4_IMMO-FC_ReadsPerGene.out.tab\u0026quot;, header =F, sep =\u0026quot;\\t\u0026quot;, fill =T, comment.char = \u0026quot;\u0026quot;)\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003eIMMOFC3=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/3_IMMO-FC/3_IMMO-FC_ReadsPerGene.out.tab\", header = F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nIMMOFC1=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/1_IMMO-FC/1_IMMO-FC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nIMMOFC2=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/2_IMMO-FC/2_IMMO-FC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\nIMMOFC4=read.table(\"/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/4_IMMO-FC/4_IMMO-FC_ReadsPerGene.out.tab\", header =F, sep =\"\\t\", fill =T, comment.char = \"\")\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCombine column 1, with column 2 from every dataframe, add a header with the name of the sample that column came from\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"HCvsFC_gene_counts = data.frame(FC10Alone$V2, FC11Alone$V2, FC12Alone$V2, Homecage19$V2, Homecage20$V2, Homecage21$V2, Homecage22$V2, Homecage23$V2, Homecage24$V2, FC8Alone$V2, FC9Alone$V2, row.names = Homecage22$V1) \ncolnames(HCvsFC_rep_gene_counts) = c(\u0026quot;FC10Alone\u0026quot;, \u0026quot;FC11Alone\u0026quot;, \u0026quot;FC12Alone\u0026quot;, \u0026quot;Homecage19\u0026quot;, \u0026quot;Homecage20\u0026quot;, \u0026quot;Homecage21\u0026quot;, \u0026quot;Homecage22\u0026quot;, \u0026quot;Homecage23\u0026quot;,\u0026quot;Homecage24\u0026quot;, \u0026quot;FC8Alone\u0026quot;, \u0026quot;FC9Alone\u0026quot;)\n\nHCvsIMMOFC_gene_counts = data.frame(Homecage19$V2, Homecage20$V2, Homecage21$V2, Homecage22$V2, Homecage23$V2, Homecage24$V2, IMMOFC3$V2, IMMOFC1$V2, IMMOFC2$V2, IMMOFC4$V2, row.names = Homecage22$V1)\ncolnames(HCvsFC_rep_gene_counts) = c(\u0026quot;Homecage19\u0026quot;, \u0026quot;Homecage20\u0026quot;, \u0026quot;Homecage21\u0026quot;, \u0026quot;Homecage22\u0026quot;, \u0026quot;Homecage23\u0026quot;, \u0026quot;Homecage24\u0026quot;, \u0026quot;IMMOFC3\u0026quot;, \u0026quot;IMMOFC1\u0026quot;, \u0026quot;IMMOFC2\u0026quot;, \u0026quot;IMMOFC4\u0026quot;)\n\nFCvsIMMOFC_genecounts = data.frame(FC10Alone$V2, FC11Alone$V2, FC12Alone$V2, FC8Alone$V2, FC9Alone$V2, IMMOFC3$V2, IMMOFC1$V2, IMMOFC2$V2, IMMOFC4$V2, row.names = Homecage22$V1)\ncolnames(HCvsFC_rep_gene_counts) = c(\u0026quot;FC10Alone\u0026quot;, \u0026quot;FC11Alone\u0026quot;, \u0026quot;FC12Alone\u0026quot;, \u0026quot;FC8Alone\u0026quot;, \u0026quot;FC9Alone\u0026quot;, \u0026quot;IMMOFC3\u0026quot;, \u0026quot;IMMOFC1\u0026quot;, \u0026quot;IMMOFC2\u0026quot;, \u0026quot;IMMOFC4\u0026quot;)\n\nHCvsFC_ALL = data.frame(HC1r$V2, HC7r$V2, HC2r$V2, HC8r$V2, Homecage19$V2, Homecage20$V2, Homecage21$V2, Homecage22$V2, Homecage23$V2, Homecage24$V2, FC3r$V2, FC5r$V2, FC4r$V2, FC6r$V2, FC10Alone$V2, FC11Alone$V2, FC12Alone$V2, FC8Alone$V2, FC9Alone$V2, row.names = Homecage22$V1)\ncolnames(HCvsFC_ALL) = c(\u0026quot;HC1r\u0026quot;, \u0026quot;HC7r\u0026quot;, \u0026quot;HC2r\u0026quot;, \u0026quot;HC8r\u0026quot;, \u0026quot;Homecage19\u0026quot;, \u0026quot;Homecage20\u0026quot;, \u0026quot;Homecage21\u0026quot;, \u0026quot;Homecage22\u0026quot;, \u0026quot;Homecage23\u0026quot;, \u0026quot;Homecage24\u0026quot;, \u0026quot;FC3r\u0026quot;, \u0026quot;FC5r\u0026quot;, \u0026quot;FC4r\u0026quot;, \u0026quot;FC6r\u0026quot;, \u0026quot;FC10Alone\u0026quot;, \u0026quot;FC11Alone\u0026quot;, \u0026quot;FC12Alone\u0026quot;, \u0026quot;FC8Alone\u0026quot;, \u0026quot;FC9Alone\u0026quot;)\n\nHCvsFC_rep_gene_counts = data.frame(HC1r$V2, FC3r$V2, FC5r$V2, HC7r$V2, HC2r$V2, FC4r$V2, FC6r$V2, HC8r$V2, row.names = HC1r$V1) \ncolnames(HCvsFC_rep_gene_counts) = c(\u0026quot;HC1r\u0026quot;, \u0026quot;FC3r\u0026quot;, \u0026quot;FC5r\u0026quot;, \u0026quot;HC7r\u0026quot;, \u0026quot;HC2r\u0026quot;, \u0026quot;FC4r\u0026quot;, \u0026quot;FC6r\u0026quot;, \u0026quot;HC8r\u0026quot;)\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003eHCvsFC_gene_counts = data.frame(FC10Alone$V2, FC11Alone$V2, FC12Alone$V2, Homecage19$V2, Homecage20$V2, Homecage21$V2, Homecage22$V2, Homecage23$V2, Homecage24$V2, FC8Alone$V2, FC9Alone$V2, row.names = Homecage22$V1) \ncolnames(HCvsFC_rep_gene_counts) = c(\"FC10Alone\", \"FC11Alone\", \"FC12Alone\", \"Homecage19\", \"Homecage20\", \"Homecage21\", \"Homecage22\", \"Homecage23\",\"Homecage24\", \"FC8Alone\", \"FC9Alone\")\n\nHCvsIMMOFC_gene_counts = data.frame(Homecage19$V2, Homecage20$V2, Homecage21$V2, Homecage22$V2, Homecage23$V2, Homecage24$V2, IMMOFC3$V2, IMMOFC1$V2, IMMOFC2$V2, IMMOFC4$V2, row.names = Homecage22$V1)\ncolnames(HCvsFC_rep_gene_counts) = c(\"Homecage19\", \"Homecage20\", \"Homecage21\", \"Homecage22\", \"Homecage23\", \"Homecage24\", \"IMMOFC3\", \"IMMOFC1\", \"IMMOFC2\", \"IMMOFC4\")\n\nFCvsIMMOFC_genecounts = data.frame(FC10Alone$V2, FC11Alone$V2, FC12Alone$V2, FC8Alone$V2, FC9Alone$V2, IMMOFC3$V2, IMMOFC1$V2, IMMOFC2$V2, IMMOFC4$V2, row.names = Homecage22$V1)\ncolnames(HCvsFC_rep_gene_counts) = c(\"FC10Alone\", \"FC11Alone\", \"FC12Alone\", \"FC8Alone\", \"FC9Alone\", \"IMMOFC3\", \"IMMOFC1\", \"IMMOFC2\", \"IMMOFC4\")\n\nHCvsFC_ALL = data.frame(HC1r$V2, HC7r$V2, HC2r$V2, HC8r$V2, Homecage19$V2, Homecage20$V2, Homecage21$V2, Homecage22$V2, Homecage23$V2, Homecage24$V2, FC3r$V2, FC5r$V2, FC4r$V2, FC6r$V2, FC10Alone$V2, FC11Alone$V2, FC12Alone$V2, FC8Alone$V2, FC9Alone$V2, row.names = Homecage22$V1)\ncolnames(HCvsFC_ALL) = c(\"HC1r\", \"HC7r\", \"HC2r\", \"HC8r\", \"Homecage19\", \"Homecage20\", \"Homecage21\", \"Homecage22\", \"Homecage23\", \"Homecage24\", \"FC3r\", \"FC5r\", \"FC4r\", \"FC6r\", \"FC10Alone\", \"FC11Alone\", \"FC12Alone\", \"FC8Alone\", \"FC9Alone\")\n\nHCvsFC_rep_gene_counts = data.frame(HC1r$V2, FC3r$V2, FC5r$V2, HC7r$V2, HC2r$V2, FC4r$V2, FC6r$V2, HC8r$V2, row.names = HC1r$V1) \ncolnames(HCvsFC_rep_gene_counts) = c(\"HC1r\", \"FC3r\", \"FC5r\", \"HC7r\", \"HC2r\", \"FC4r\", \"FC6r\", \"HC8r\")\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRead in data about each sample\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"colData = read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DESeq2/colData_HCvsFC_ALL\u0026quot;, header =T, sep =\u0026quot;\\t\u0026quot;)\n\n\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecolData = read.table(\"/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DESeq2/colData_HCvsFC_ALL\", header =T, sep =\"\\t\")\n\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDESeq library:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"library(DESeq2)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003elibrary(DESeq2)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ese is the RangedSummarizedExperiment object containing counts and information about the samples. In this example design we control for batch and condition, which should be columns of colData(se)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"dds \u0026lt;- DESeqDataSetFromMatrix(countData = HCvsFC_ALL,\n                              colData = colData,\n                              design = ~ Condition)\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003edds \u0026lt;- DESeqDataSetFromMatrix(countData = HCvsFC_ALL,\n                              colData = colData,\n                              design = ~ Condition)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRelevel so that HomeCage is the reference\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"dds$Condition = relevel(dds$Condition, \u0026quot;HomeCage\u0026quot;)\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003edds$Condition = relevel(dds$Condition, \"HomeCage\")\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDE Analysis:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"library(BiocParallel)\nregister(MulticoreParam(20))\ndds = DESeq(dds, parallel=TRUE)\nres = results(dds)\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003elibrary(BiocParallel)\nregister(MulticoreParam(20))\ndds = DESeq(dds, parallel=TRUE)\nres = results(dds)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eManipulation of results dataframe\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"resOrdered = res[order(res$padj),]\nsummary(res)\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003eresOrdered = res[order(res$padj),]\nsummary(res)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eExport Results\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"resOrdered_subset = subset(resOrdered, padj \u0026lt;0.3)\nwrite.csv(as.data.frame(resOrdered), file=\u0026quot;HCvsFC_RNASeq_Replicate_Results.csv\u0026quot;)\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003eresOrdered_subset = subset(resOrdered, padj \u0026lt;0.3)\nwrite.csv(as.data.frame(resOrdered), file=\"HCvsFC_RNASeq_Replicate_Results.csv\")\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePCA\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"rld = rlog(dds)\npdf(\u0026quot;PCA_HCvsFC_ALL\u0026quot;)\nplotPCA(rld, intgroup=c(\u0026quot;Condition\u0026quot;))\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003erld = rlog(dds)\npdf(\"PCA_HCvsFC_ALL\")\nplotPCA(rld, intgroup=c(\"Condition\"))\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e###4. RNASEQ ANALYSIS:DIFFERENTIAL SPLICING:DEXSEQ \u0026amp; CUFFDIFF\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHTSeq: generate flattened GFF of the exon counting bins\nmm10_UCSC_RepeatMasked_Genes.gtf (comprehensive gtf file)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_prepare_annotation.py /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gtf /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gff \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_prepare_annotation.py /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gtf /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gff \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003egenes.gtf (UCSC genes only)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_prepare_annotation.py -r no genes.gtf genes.gff\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_prepare_annotation.py -r no genes.gtf genes.gff\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eTOO MANY ISSUES WITH TEH UCSC GTF FILES - USE ENSEMBL FILES FOR NOW...SEE IF THEY RELEASE A PATCH TO MAKE UCSC ANNOTATIONS FUNCTIONAL LATER ON\u003c/em\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCounting reads into exon bins:\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e1-935-HC\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/1-935-HC/1-935-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/1-935-HC_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/1-935-HC/1-935-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/1-935-HC_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e2-935-HC:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/2-935-HC/2-935-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/2-935-HC_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/2-935-HC/2-935-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/2-935-HC_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e3-940-FC:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/3-940-FC/3-940-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/3-940-FC_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/3-940-FC/3-940-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/3-940-FC_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e4-940-FC:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/4-940-FC/4-940-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/4-940-FC_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/4-940-FC/4-940-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/4-940-FC_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e5-938-FC:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/5-938-FC/5-938-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/5-938-FC_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/5-938-FC/5-938-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/5-938-FC_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e6-938-FC\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/6-938-FC/6-938-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/6-938-FC_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/6-938-FC/6-938-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/6-938-FC_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e7-947-HC:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/7-947-HC/7-947-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/7-947-HC_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/7-947-HC/7-947-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/7-947-HC_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e8-947-HC:\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/8-947-HC/8-947-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/8-947-HC_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/8-947-HC/8-947-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/8-947-HC_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e19_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/19_Homecage/19_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/19_HC_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/19_Homecage/19_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/19_HC_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e20_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/20_Homecage/20_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/20_HC_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/20_Homecage/20_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/20_HC_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e21_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/21_Homecage/21_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/21_HC_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/21_Homecage/21_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/21_HC_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e22_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/22_Homecage/22_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/22_HC_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/22_Homecage/22_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/22_HC_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e23_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/23_Homecage/23_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/23_HC_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/23_Homecage/23_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/23_HC_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e24_Homecage\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/24_Homecage/24_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/24_HC_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/24_Homecage/24_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/24_HC_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e10_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/10_FC-Alone/10_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/10_FC-Alone_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/10_FC-Alone/10_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/10_FC-Alone_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e11_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/11_FC-Alone/11_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/11_FC-Alone_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/11_FC-Alone/11_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/11_FC-Alone_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e12_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/12_FC-Alone/12_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/12_FC-Alone_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/12_FC-Alone/12_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/12_FC-Alone_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e9_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/9_FC-Alone/9_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/9_FC-Alone_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/9_FC-Alone/9_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/9_FC-Alone_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e8_FC-Alone\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/8_FC-Alone/8_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/8_FC-Alone_ExonCounts.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epython /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/8_FC-Alone/8_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/8_FC-Alone_ExonCounts.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRead data into R\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"countDir = \u0026quot;/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/ExonCounts2\u0026quot;\ncountFiles = list.files(countDir, pattern=\u0026quot;ExonCounts.txt$\u0026quot;, full.names=TRUE)\nflatDir = \u0026quot;/home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes\u0026quot;\nflattenedFile = list.files(flatDir, pattern=\u0026quot;gff$\u0026quot;, full.names=TRUE)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecountDir = \"/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/ExonCounts2\"\ncountFiles = list.files(countDir, pattern=\"ExonCounts.txt$\", full.names=TRUE)\nflatDir = \"/home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes\"\nflattenedFile = list.files(flatDir, pattern=\"gff$\", full.names=TRUE)\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePrepare Sample Table\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"sampleTable = read.table(\u0026quot;/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/HCvsFC_ALL_SampleTable.DEXSeq\u0026quot;, sep =\u0026quot;\\t\u0026quot;, header = T)\n\nsampleTable = data.frame(\n   row.names = c( \u0026quot;10_FC-Alone\u0026quot;, \u0026quot;11_FC-Alone\u0026quot;, \u0026quot;12_FC-Alone\u0026quot;, \u0026quot;1-935-HC\u0026quot;, \u0026quot;19_HC\u0026quot;, \u0026quot;20_HC\u0026quot;, \u0026quot;21_HC\u0026quot;, \u0026quot;22_HC\u0026quot;, \u0026quot;23_HC\u0026quot;, \u0026quot;24_HC\u0026quot;, \u0026quot;2-935-HC\u0026quot;, \u0026quot;3-940-FC\u0026quot;, \u0026quot;4-940-FC\u0026quot;, \u0026quot;5-938-FC\u0026quot;, \u0026quot;6-938-FC\u0026quot;, \u0026quot;7-947-HC\u0026quot;, \u0026quot;8-947-HC\u0026quot;, \u0026quot;8_FC-Alone\u0026quot;, \u0026quot;9_FC-Alone\u0026quot; ),\n   condition = c(\u0026quot;FearConditioning\u0026quot;, \u0026quot;FearConditioning\u0026quot;, \u0026quot;FearConditioning\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;FearConditioning\u0026quot;, \u0026quot;FearConditioning\u0026quot;, \u0026quot;FearConditioning\u0026quot;, \u0026quot;FearConditioning\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;FearConditioning\u0026quot;, \u0026quot;FearConditioning\u0026quot; )\n   )\n   \n   \nsampleTable = data.frame(\n   row.names = c( \u0026quot;10_FC-Alone\u0026quot;, \u0026quot;11_FC-Alone\u0026quot;, \u0026quot;12_FC-Alone\u0026quot;, \u0026quot;19_HC\u0026quot;, \u0026quot;20_HC\u0026quot;, \u0026quot;21_HC\u0026quot;, \u0026quot;22_HC\u0026quot;, \u0026quot;23_HC\u0026quot;, \u0026quot;24_HC\u0026quot;, \u0026quot;8_FC-Alone\u0026quot;, \u0026quot;9_FC-Alone\u0026quot;),\n   condition = c(\u0026quot;FearConditioning\u0026quot;, \u0026quot;FearConditioning\u0026quot;, \u0026quot;FearConditioning\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;HomeCage\u0026quot;, \u0026quot;FearConditioning\u0026quot;, \u0026quot;FearConditioning\u0026quot; )\n   )\n\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003esampleTable = read.table(\"/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/HCvsFC_ALL_SampleTable.DEXSeq\", sep =\"\\t\", header = T)\n\nsampleTable = data.frame(\n   row.names = c( \"10_FC-Alone\", \"11_FC-Alone\", \"12_FC-Alone\", \"1-935-HC\", \"19_HC\", \"20_HC\", \"21_HC\", \"22_HC\", \"23_HC\", \"24_HC\", \"2-935-HC\", \"3-940-FC\", \"4-940-FC\", \"5-938-FC\", \"6-938-FC\", \"7-947-HC\", \"8-947-HC\", \"8_FC-Alone\", \"9_FC-Alone\" ),\n   condition = c(\"FearConditioning\", \"FearConditioning\", \"FearConditioning\", \"HomeCage\", \"HomeCage\", \"HomeCage\", \"HomeCage\", \"HomeCage\", \"HomeCage\", \"HomeCage\", \"HomeCage\", \"FearConditioning\", \"FearConditioning\", \"FearConditioning\", \"FearConditioning\", \"HomeCage\", \"HomeCage\", \"FearConditioning\", \"FearConditioning\" )\n   )\n   \n   \nsampleTable = data.frame(\n   row.names = c( \"10_FC-Alone\", \"11_FC-Alone\", \"12_FC-Alone\", \"19_HC\", \"20_HC\", \"21_HC\", \"22_HC\", \"23_HC\", \"24_HC\", \"8_FC-Alone\", \"9_FC-Alone\"),\n   condition = c(\"FearConditioning\", \"FearConditioning\", \"FearConditioning\", \"HomeCage\", \"HomeCage\", \"HomeCage\", \"HomeCage\", \"HomeCage\", \"HomeCage\", \"FearConditioning\", \"FearConditioning\" )\n   )\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDEXSeq Analysis\nData object creation\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"library( \u0026quot;DEXSeq\u0026quot; )\n\nBPPARAM = MulticoreParam(workers=25)\ndxd = DEXSeqDataSetFromHTSeq(countFiles, sampleData=sampleTable, design= ~sample + exon + condition:exon, flattenedfile=flattenedFile )\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003elibrary( \"DEXSeq\" )\n\nBPPARAM = MulticoreParam(workers=25)\ndxd = DEXSeqDataSetFromHTSeq(countFiles, sampleData=sampleTable, design= ~sample + exon + condition:exon, flattenedfile=flattenedFile )\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eanalyze subset of genes\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"geneDir = \u0026quot;/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq\u0026quot;\ngenesForSubset = read.table(\n  file.path(geneDir, \u0026quot;ENS_Genes\u0026quot;),\n  stringsAsFactors=FALSE)[[1]]\n  \ndxd = dxd[geneIDs( dxd ) %in% genesForSubset,]\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003egeneDir = \"/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq\"\ngenesForSubset = read.table(\n  file.path(geneDir, \"ENS_Genes\"),\n  stringsAsFactors=FALSE)[[1]]\n  \ndxd = dxd[geneIDs( dxd ) %in% genesForSubset,]\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eNormalisation \u0026amp; Dispersion estimation \u0026amp; DEU test\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"dxd = estimateSizeFactors( dxd )\ndxd = estimateDispersions( dxd, BPPARAM=BPPARAM )\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003edxd = estimateSizeFactors( dxd )\ndxd = estimateDispersions( dxd, BPPARAM=BPPARAM )\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDispersion plots\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Dispersion_estimates.pdf\u0026quot;)\nplotDispEsts( dxd )\ndev.off()\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Dispersion_estimates.pdf\")\nplotDispEsts( dxd )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDEU testing\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"dxd = testForDEU( dxd, BPPARAM=BPPARAM )\ndxd = estimateExonFoldChanges( dxd, fitExpToVar=\u0026quot;condition\u0026quot;, BPPARAM=BPPARAM )\ndxr1 = DEXSeqResults( dxd )\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003edxd = testForDEU( dxd, BPPARAM=BPPARAM )\ndxd = estimateExonFoldChanges( dxd, fitExpToVar=\"condition\", BPPARAM=BPPARAM )\ndxr1 = DEXSeqResults( dxd )\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSignificant Results\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"table ( dxr1$padj \u0026lt; 0.1 )\n\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003etable ( dxr1$padj \u0026lt; 0.1 )\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Gphn\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Gphn_2_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000047454\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Gphn_2_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000047454\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Gabrb1\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Gabrb1_2_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000029212\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Gabrb1_2_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000029212\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Gabrb2\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Gabrb2_2_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000007653\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Gabrb2_2_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000007653\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Gabrb3\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Gabrb3_2_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000033676\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Gabrb3_2_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000033676\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Gabrg1_2\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Gabrg1_3_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000001260\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Gabrg1_3_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000001260\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Cacn1e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Cacn1e.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000004110\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Cacn1e.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000004110\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Bicd1\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Bicd1.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000003452\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Bicd1.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000003452\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Glg1\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Glg1.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000003316\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Glg1.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000003316\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Plag1\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Plag1.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000003282\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Plag1.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000003282\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Ranbp2\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Ranbp2.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000003226\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Ranbp2.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000003226\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Peg3\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Peg3.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000002265\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Peg3.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000002265\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Celf2\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Celf2.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000002107\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Celf2.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000002107\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Gabarap\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Gabarap_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000018567\u0026quot;, legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Gabarap_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000018567\", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Gabarapl1\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Gabarapl1_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000030161\u0026quot;, legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Gabarapl1_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000030161\", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Gabarapl2\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Gabarapl2_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000031950\u0026quot;, legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Gabarapl2_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000031950\", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Gad1\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Gad1_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000070880\u0026quot;, legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Gad1_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000070880\", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Gad1_2\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Gad1_2_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000070880\u0026quot;, legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Gad1_2_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000070880\", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Gad1_3\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Gad1__5_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000070880\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Gad1__5_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000070880\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Gad2\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Gad2_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000026787\u0026quot;, legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Gad2_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000026787\", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Slc6a11\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Slc6a11_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000030307\u0026quot;, legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Slc6a11_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000030307\", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Slc6a12\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Slc6a12_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000030109\u0026quot;, legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Slc6a12_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000030109\", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: gabra1\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Gabra1_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000010803\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2)\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Gabra1_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000010803\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2)\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Tac2\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"dxr2 = DEXSeqResults( dxd )\npdf(\u0026quot;Tac2_1_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr1, \u0026quot;ENSMUSG00000025400\u0026quot;, legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2)\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003edxr2 = DEXSeqResults( dxd )\npdf(\"Tac2_1_DEU.pdf\")\nplotDEXSeq( dxr1, \"ENSMUSG00000025400\", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2)\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: Nts\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;Nts_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr2, \u0026quot;ENSMUSG00000019890\u0026quot;, legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"Nts_DEU.pdf\")\nplotDEXSeq( dxr2, \"ENSMUSG00000019890\", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: POMC\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;POMC_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr2, \u0026quot;ENSMUSG00000020660\u0026quot;, legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"POMC_DEU.pdf\")\nplotDEXSeq( dxr2, \"ENSMUSG00000020660\", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eplotting: ErbB4\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"pdf(\u0026quot;ErbB4_DEU.pdf\u0026quot;)\nplotDEXSeq( dxr2, \u0026quot;ENSMUSG00000062209\u0026quot;, legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003epdf(\"ErbB4_DEU.pdf\")\nplotDEXSeq( dxr2, \"ENSMUSG00000062209\", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )\ndev.off()\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e###5. RNASEQ ANALYSIS, TRAPSEQ:PATHWAT ANALYSIS:CONSENSUS PATH DATABASE\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eURL (HOME): \u003ca href=\"http://cpdb.molgen.mpg.de/MCPDB\" rel=\"nofollow\"\u003ehttp://cpdb.molgen.mpg.de/MCPDB\u003c/a\u003e\nAnalyses:\nHOME/gene set analysis/over-representation analysis/enriched pathway-based sets/pathway-based sets\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHOME/gene set analysis/over-representation analysis/enriched pathway-based sets/Network neighborhood-based entity sets (NESTs)/1-next neighbor/AZI1 set center\u003c/p\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003euser-provided identifier\u003c/th\u003e\n\u003cth\u003eentrez-gene ID\u003c/th\u003e\n\u003cth\u003eentrez-gene name\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003eDENND4A\u003c/td\u003e\n\u003ctd\u003e10260\u003c/td\u003e\n\u003ctd\u003eDENND4A : DENN/MADD domain containing 4A\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eMAP1B\u003c/td\u003e\n\u003ctd\u003e4131\u003c/td\u003e\n\u003ctd\u003eMAP1B : microtubule-associated protein 1B\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eMIB1\u003c/td\u003e\n\u003ctd\u003e57534\u003c/td\u003e\n\u003ctd\u003eMIB1 : mindbomb E3 ubiquitin protein ligase 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCC2D2A\u003c/td\u003e\n\u003ctd\u003e57545\u003c/td\u003e\n\u003ctd\u003eCC2D2A : coiled-coil and C2 domain containing 2A\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eDENND4C\u003c/td\u003e\n\u003ctd\u003e55667\u003c/td\u003e\n\u003ctd\u003eDENND4C : DENN/MADD domain containing 4C\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCPEB4\u003c/td\u003e\n\u003ctd\u003e80315\u003c/td\u003e\n\u003ctd\u003eCPEB4 : cytoplasmic polyadenylation element binding protein 4\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCDK5RAP2\u003c/td\u003e\n\u003ctd\u003e55755\u003c/td\u003e\n\u003ctd\u003eCDK5RAP2 : CDK5 regulatory subunit associated protein 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eIRS2\u003c/td\u003e\n\u003ctd\u003e8660\u003c/td\u003e\n\u003ctd\u003eIRS2 : insulin receptor substrate 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eATM\u003c/td\u003e\n\u003ctd\u003e472\u003c/td\u003e\n\u003ctd\u003eATM : ATM serine/threonine kinase\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSNTB2\u003c/td\u003e\n\u003ctd\u003e6645\u003c/td\u003e\n\u003ctd\u003eSNTB2 : syntrophin, beta 2 (dystrophin-associated protein A1, 59kDa, basic component 2)\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eRAPGEF6\u003c/td\u003e\n\u003ctd\u003e51735\u003c/td\u003e\n\u003ctd\u003eRAPGEF6 : Rap guanine nucleotide exchange factor (GEF) 6\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eKIF1B\u003c/td\u003e\n\u003ctd\u003e23095\u003c/td\u003e\n\u003ctd\u003eKIF1B : kinesin family member 1B\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eBIRC6\u003c/td\u003e\n\u003ctd\u003e57448\u003c/td\u003e\n\u003ctd\u003eBIRC6 : baculoviral IAP repeat containing 6\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eTTBK2\u003c/td\u003e\n\u003ctd\u003e146057\u003c/td\u003e\n\u003ctd\u003eTTBK2 : tau tubulin kinase 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSTRN\u003c/td\u003e\n\u003ctd\u003e6801\u003c/td\u003e\n\u003ctd\u003eSTRN : striatin, calmodulin binding protein\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eNINL\u003c/td\u003e\n\u003ctd\u003e22981\u003c/td\u003e\n\u003ctd\u003eNINL : ninein-like\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eFRYL\u003c/td\u003e\n\u003ctd\u003e285527\u003c/td\u003e\n\u003ctd\u003eFRYL : FRY-like\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCBL\u003c/td\u003e\n\u003ctd\u003e867\u003c/td\u003e\n\u003ctd\u003eCBL : Cbl proto-oncogene, E3 ubiquitin protein ligase\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eGFOD1\u003c/td\u003e\n\u003ctd\u003e54438\u003c/td\u003e\n\u003ctd\u003eGFOD1 : glucose-fructose oxidoreductase domain containing 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eMAP3K2\u003c/td\u003e\n\u003ctd\u003e10746\u003c/td\u003e\n\u003ctd\u003eMAP3K2 : mitogen-activated protein kinase kinase kinase 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ePLK1\u003c/td\u003e\n\u003ctd\u003e5347\u003c/td\u003e\n\u003ctd\u003ePLK1 : polo-like kinase 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eUTRN\u003c/td\u003e\n\u003ctd\u003e7402\u003c/td\u003e\n\u003ctd\u003eUTRN : utrophin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eRASAL2\u003c/td\u003e\n\u003ctd\u003e9462\u003c/td\u003e\n\u003ctd\u003eRASAL2 : RAS protein activator like 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eRICTOR\u003c/td\u003e\n\u003ctd\u003e253260\u003c/td\u003e\n\u003ctd\u003eRICTOR : RPTOR independent companion of MTOR, complex 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eTANC2\u003c/td\u003e\n\u003ctd\u003e26115\u003c/td\u003e\n\u003ctd\u003eTANC2 : tetratricopeptide repeat, ankyrin repeat and coiled-coil containing 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eALMS1\u003c/td\u003e\n\u003ctd\u003e7840\u003c/td\u003e\n\u003ctd\u003eALMS1 : Alstrom syndrome protein 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eNAV1\u003c/td\u003e\n\u003ctd\u003e89796\u003c/td\u003e\n\u003ctd\u003eNAV1 : neuron navigator 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eHELZ\u003c/td\u003e\n\u003ctd\u003e9931\u003c/td\u003e\n\u003ctd\u003eHELZ : helicase with zinc finger\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eDYNC1H1\u003c/td\u003e\n\u003ctd\u003e1778\u003c/td\u003e\n\u003ctd\u003eDYNC1H1 : dynein, cytoplasmic 1, heavy chain 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSHKBP1\u003c/td\u003e\n\u003ctd\u003e92799\u003c/td\u003e\n\u003ctd\u003eSHKBP1 : SH3KBP1 binding protein 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCCDC6\u003c/td\u003e\n\u003ctd\u003e8030\u003c/td\u003e\n\u003ctd\u003eCCDC6 : coiled-coil domain containing 6\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eAKAP9\u003c/td\u003e\n\u003ctd\u003e10142\u003c/td\u003e\n\u003ctd\u003eAKAP9 : A kinase (PRKA) anchor protein 9\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eLMO7\u003c/td\u003e\n\u003ctd\u003e4008\u003c/td\u003e\n\u003ctd\u003eLMO7 : LIM domain 7\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cp dir=\"auto\"\u003eHOME/gene set analysis/over-representation analysis/enriched pathway-based sets/Network neighborhood-based entity sets (NESTs)/1-next neighbor/HDAC4 set center\u003c/p\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003euser-provided identifier\u003c/th\u003e\n\u003cth\u003eentrez-gene ID\u003c/th\u003e\n\u003cth\u003eentrez-gene name\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003eDENND4A\u003c/td\u003e\n\u003ctd\u003e10260\u003c/td\u003e\n\u003ctd\u003eDENND4A : DENN/MADD domain containing 4A\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eJDP2\u003c/td\u003e\n\u003ctd\u003e122953\u003c/td\u003e\n\u003ctd\u003eJDP2 : Jun dimerization protein 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eNAV1\u003c/td\u003e\n\u003ctd\u003e89796\u003c/td\u003e\n\u003ctd\u003eNAV1 : neuron navigator 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eAR\u003c/td\u003e\n\u003ctd\u003e367\u003c/td\u003e\n\u003ctd\u003eAR : androgen receptor\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eDENND4C\u003c/td\u003e\n\u003ctd\u003e55667\u003c/td\u003e\n\u003ctd\u003eDENND4C : DENN/MADD domain containing 4C\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eIRS2\u003c/td\u003e\n\u003ctd\u003e8660\u003c/td\u003e\n\u003ctd\u003eIRS2 : insulin receptor substrate 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eLCOR\u003c/td\u003e\n\u003ctd\u003e84458\u003c/td\u003e\n\u003ctd\u003eLCOR : ligand dependent nuclear receptor corepressor\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eMAP3K2\u003c/td\u003e\n\u003ctd\u003e10746\u003c/td\u003e\n\u003ctd\u003eMAP3K2 : mitogen-activated protein kinase kinase kinase 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eRAPGEF6\u003c/td\u003e\n\u003ctd\u003e51735\u003c/td\u003e\n\u003ctd\u003eRAPGEF6 : Rap guanine nucleotide exchange factor (GEF) 6\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eATRX\u003c/td\u003e\n\u003ctd\u003e546\u003c/td\u003e\n\u003ctd\u003eATRX : alpha thalassemia/mental retardation syndrome X-linked\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eKIF1B\u003c/td\u003e\n\u003ctd\u003e23095\u003c/td\u003e\n\u003ctd\u003eKIF1B : kinesin family member 1B\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eBCL6\u003c/td\u003e\n\u003ctd\u003e604\u003c/td\u003e\n\u003ctd\u003eBCL6 : B-cell CLL/lymphoma 6\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSHKBP1\u003c/td\u003e\n\u003ctd\u003e92799\u003c/td\u003e\n\u003ctd\u003eSHKBP1 : SH3KBP1 binding protein 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCAMK4\u003c/td\u003e\n\u003ctd\u003e814\u003c/td\u003e\n\u003ctd\u003eCAMK4 : calcium/calmodulin-dependent protein kinase IV\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCBL\u003c/td\u003e\n\u003ctd\u003e867\u003c/td\u003e\n\u003ctd\u003eCBL : Cbl proto-oncogene, E3 ubiquitin protein ligase\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCDKN1A\u003c/td\u003e\n\u003ctd\u003e1026\u003c/td\u003e\n\u003ctd\u003eCDKN1A : cyclin-dependent kinase inhibitor 1A (p21, Cip1)\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eZMYND8\u003c/td\u003e\n\u003ctd\u003e23613\u003c/td\u003e\n\u003ctd\u003eZMYND8 : zinc finger, MYND-type containing 8\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eTRIP11\u003c/td\u003e\n\u003ctd\u003e9321\u003c/td\u003e\n\u003ctd\u003eTRIP11 : thyroid hormone receptor interactor 11\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eDNAJB5\u003c/td\u003e\n\u003ctd\u003e25822\u003c/td\u003e\n\u003ctd\u003eDNAJB5 : DnaJ (Hsp40) homolog, subfamily B, member 5\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eRASAL2\u003c/td\u003e\n\u003ctd\u003e9462\u003c/td\u003e\n\u003ctd\u003eRASAL2 : RAS protein activator like 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eRICTOR\u003c/td\u003e\n\u003ctd\u003e253260\u003c/td\u003e\n\u003ctd\u003eRICTOR : RPTOR independent companion of MTOR, complex 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eNCOR2\u003c/td\u003e\n\u003ctd\u003e9612\u003c/td\u003e\n\u003ctd\u003eNCOR2 : nuclear receptor corepressor 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eTANC2\u003c/td\u003e\n\u003ctd\u003e26115\u003c/td\u003e\n\u003ctd\u003eTANC2 : tetratricopeptide repeat, ankyrin repeat and coiled-coil containing 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eZBTB16\u003c/td\u003e\n\u003ctd\u003e7704\u003c/td\u003e\n\u003ctd\u003eZBTB16 : zinc finger and BTB domain containing 16\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eRANBP2\u003c/td\u003e\n\u003ctd\u003e5903\u003c/td\u003e\n\u003ctd\u003eRANBP2 : RAN binding protein 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eREV3L\u003c/td\u003e\n\u003ctd\u003e5980\u003c/td\u003e\n\u003ctd\u003eREV3L : REV3-like, polymerase (DNA directed), zeta, catalytic subunit\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCCDC6\u003c/td\u003e\n\u003ctd\u003e8030\u003c/td\u003e\n\u003ctd\u003eCCDC6 : coiled-coil domain containing 6\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eLMO7\u003c/td\u003e\n\u003ctd\u003e4008\u003c/td\u003e\n\u003ctd\u003eLMO7 : LIM domain 7\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cp dir=\"auto\"\u003eHOME/gene set analysis/over-representation analysis/enriched pathway-based sets/Network neighborhood-based entity sets (NESTs)/2-next neighbor/CaM1,CaM3,CaM2 set center\u003c/p\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003euser-provided identifier\u003c/th\u003e\n\u003cth\u003eentrez-gene ID\u003c/th\u003e\n\u003cth\u003eentrez-gene name\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003eNRIP1\u003c/td\u003e\n\u003ctd\u003e8204\u003c/td\u003e\n\u003ctd\u003eNRIP1 : nuclear receptor interacting protein 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eUBN2\u003c/td\u003e\n\u003ctd\u003e254048\u003c/td\u003e\n\u003ctd\u003eUBN2 : ubinuclein 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eESYT2\u003c/td\u003e\n\u003ctd\u003e57488\u003c/td\u003e\n\u003ctd\u003eESYT2 : extended synaptotagmin-like protein 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eADRA1A\u003c/td\u003e\n\u003ctd\u003e148\u003c/td\u003e\n\u003ctd\u003eADRA1A : adrenoceptor alpha 1A\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eITGA10\u003c/td\u003e\n\u003ctd\u003e8515\u003c/td\u003e\n\u003ctd\u003eITGA10 : integrin, alpha 10\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eARHGAP5\u003c/td\u003e\n\u003ctd\u003e394\u003c/td\u003e\n\u003ctd\u003eARHGAP5 : Rho GTPase activating protein 5\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eIRS2\u003c/td\u003e\n\u003ctd\u003e8660\u003c/td\u003e\n\u003ctd\u003eIRS2 : insulin receptor substrate 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eAVP\u003c/td\u003e\n\u003ctd\u003e551\u003c/td\u003e\n\u003ctd\u003eAVP : arginine vasopressin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eBCL6\u003c/td\u003e\n\u003ctd\u003e604\u003c/td\u003e\n\u003ctd\u003eBCL6 : B-cell CLL/lymphoma 6\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eBDNF\u003c/td\u003e\n\u003ctd\u003e627\u003c/td\u003e\n\u003ctd\u003eBDNF : brain-derived neurotrophic factor\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eDST\u003c/td\u003e\n\u003ctd\u003e667\u003c/td\u003e\n\u003ctd\u003eDST : dystonin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCAMK4\u003c/td\u003e\n\u003ctd\u003e814\u003c/td\u003e\n\u003ctd\u003eCAMK4 : calcium/calmodulin-dependent protein kinase IV\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eDLGAP2\u003c/td\u003e\n\u003ctd\u003e9228\u003c/td\u003e\n\u003ctd\u003eDLGAP2 : discs, large (Drosophila) homolog-associated protein 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eRASAL2\u003c/td\u003e\n\u003ctd\u003e9462\u003c/td\u003e\n\u003ctd\u003eRASAL2 : RAS protein activator like 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCOL19A1\u003c/td\u003e\n\u003ctd\u003e1310\u003c/td\u003e\n\u003ctd\u003eCOL19A1 : collagen, type XIX, alpha 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCLOCK\u003c/td\u003e\n\u003ctd\u003e9575\u003c/td\u003e\n\u003ctd\u003eCLOCK : clock circadian regulator\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eARC\u003c/td\u003e\n\u003ctd\u003e23237\u003c/td\u003e\n\u003ctd\u003eARC : activity-regulated cytoskeleton-associated protein\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eDCC\u003c/td\u003e\n\u003ctd\u003e1630\u003c/td\u003e\n\u003ctd\u003eDCC : DCC netrin 1 receptor\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eDCN\u003c/td\u003e\n\u003ctd\u003e1634\u003c/td\u003e\n\u003ctd\u003eDCN : decorin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eEGR1\u003c/td\u003e\n\u003ctd\u003e1958\u003c/td\u003e\n\u003ctd\u003eEGR1 : early growth response 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eERBB4\u003c/td\u003e\n\u003ctd\u003e2066\u003c/td\u003e\n\u003ctd\u003eERBB4 : erb-b2 receptor tyrosine kinase 4\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eGABRB2\u003c/td\u003e\n\u003ctd\u003e2561\u003c/td\u003e\n\u003ctd\u003eGABRB2 : gamma-aminobutyric acid (GABA) A receptor, beta 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eRAPGEF6\u003c/td\u003e\n\u003ctd\u003e51735\u003c/td\u003e\n\u003ctd\u003eRAPGEF6 : Rap guanine nucleotide exchange factor (GEF) 6\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eRSF1\u003c/td\u003e\n\u003ctd\u003e51773\u003c/td\u003e\n\u003ctd\u003eRSF1 : remodeling and spacing factor 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSHKBP1\u003c/td\u003e\n\u003ctd\u003e92799\u003c/td\u003e\n\u003ctd\u003eSHKBP1 : SH3KBP1 binding protein 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eST6GAL2\u003c/td\u003e\n\u003ctd\u003e84620\u003c/td\u003e\n\u003ctd\u003eST6GAL2 : ST6 beta-galactosamide alpha-2,6-sialyltranferase 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eDUSP4\u003c/td\u003e\n\u003ctd\u003e1846\u003c/td\u003e\n\u003ctd\u003eDUSP4 : dual specificity phosphatase 4\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eGRIN2A\u003c/td\u003e\n\u003ctd\u003e2903\u003c/td\u003e\n\u003ctd\u003eGRIN2A : glutamate receptor, ionotropic, N-methyl D-aspartate 2A\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eGRIN2B\u003c/td\u003e\n\u003ctd\u003e2904\u003c/td\u003e\n\u003ctd\u003eGRIN2B : glutamate receptor, ionotropic, N-methyl D-aspartate 2B\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eGRM4\u003c/td\u003e\n\u003ctd\u003e2914\u003c/td\u003e\n\u003ctd\u003eGRM4 : glutamate receptor, metabotropic 4\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eGRM5\u003c/td\u003e\n\u003ctd\u003e2915\u003c/td\u003e\n\u003ctd\u003eGRM5 : glutamate receptor, metabotropic 5\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ePTPRT\u003c/td\u003e\n\u003ctd\u003e11122\u003c/td\u003e\n\u003ctd\u003ePTPRT : protein tyrosine phosphatase, receptor type, T\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eHTT\u003c/td\u003e\n\u003ctd\u003e3064\u003c/td\u003e\n\u003ctd\u003eHTT : huntingtin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSYNPO\u003c/td\u003e\n\u003ctd\u003e11346\u003c/td\u003e\n\u003ctd\u003eSYNPO : synaptopodin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eNR4A1\u003c/td\u003e\n\u003ctd\u003e3164\u003c/td\u003e\n\u003ctd\u003eNR4A1 : nuclear receptor subfamily 4, group A, member 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eGAL\u003c/td\u003e\n\u003ctd\u003e51083\u003c/td\u003e\n\u003ctd\u003eGAL : galanin/GMAP prepropeptide\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eINSR\u003c/td\u003e\n\u003ctd\u003e3643\u003c/td\u003e\n\u003ctd\u003eINSR : insulin receptor\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eITGA4\u003c/td\u003e\n\u003ctd\u003e3676\u003c/td\u003e\n\u003ctd\u003eITGA4 : integrin, alpha 4 (antigen CD49D, alpha 4 subunit of VLA-4 receptor)\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eITGB8\u003c/td\u003e\n\u003ctd\u003e3696\u003c/td\u003e\n\u003ctd\u003eITGB8 : integrin, beta 8\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eITPR1\u003c/td\u003e\n\u003ctd\u003e3708\u003c/td\u003e\n\u003ctd\u003eITPR1 : inositol 1,4,5-trisphosphate receptor, type 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eKCNA2\u003c/td\u003e\n\u003ctd\u003e3737\u003c/td\u003e\n\u003ctd\u003eKCNA2 : potassium channel, voltage gated shaker related subfamily A, member 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eKCNA3\u003c/td\u003e\n\u003ctd\u003e3738\u003c/td\u003e\n\u003ctd\u003eKCNA3 : potassium channel, voltage gated shaker related subfamily A, member 3\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eKCNMA1\u003c/td\u003e\n\u003ctd\u003e3778\u003c/td\u003e\n\u003ctd\u003eKCNMA1 : potassium channel, calcium activated large conductance subfamily M alpha, member 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eKCNN3\u003c/td\u003e\n\u003ctd\u003e3782\u003c/td\u003e\n\u003ctd\u003eKCNN3 : potassium channel, calcium activated intermediate/small conductance subfamily N alpha, member 3\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eLAMC2\u003c/td\u003e\n\u003ctd\u003e3918\u003c/td\u003e\n\u003ctd\u003eLAMC2 : laminin, gamma 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eLRP6\u003c/td\u003e\n\u003ctd\u003e4040\u003c/td\u003e\n\u003ctd\u003eLRP6 : low density lipoprotein receptor-related protein 6\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCD46\u003c/td\u003e\n\u003ctd\u003e4179\u003c/td\u003e\n\u003ctd\u003eCD46 : CD46 molecule, complement regulatory protein\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eMECP2\u003c/td\u003e\n\u003ctd\u003e4204\u003c/td\u003e\n\u003ctd\u003eMECP2 : methyl CpG binding protein 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSHC3\u003c/td\u003e\n\u003ctd\u003e53358\u003c/td\u003e\n\u003ctd\u003eSHC3 : SHC (Src homology 2 domain containing) transforming protein 3\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eDENND4A\u003c/td\u003e\n\u003ctd\u003e10260\u003c/td\u003e\n\u003ctd\u003eDENND4A : DENN/MADD domain containing 4A\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eHIPK2\u003c/td\u003e\n\u003ctd\u003e28996\u003c/td\u003e\n\u003ctd\u003eHIPK2 : homeodomain interacting protein kinase 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eNEFM\u003c/td\u003e\n\u003ctd\u003e4741\u003c/td\u003e\n\u003ctd\u003eNEFM : neurofilament, medium polypeptide\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eNRCAM\u003c/td\u003e\n\u003ctd\u003e4897\u003c/td\u003e\n\u003ctd\u003eNRCAM : neuronal cell adhesion molecule\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eNTS\u003c/td\u003e\n\u003ctd\u003e4922\u003c/td\u003e\n\u003ctd\u003eNTS : neurotensin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eOPRD1\u003c/td\u003e\n\u003ctd\u003e4985\u003c/td\u003e\n\u003ctd\u003eOPRD1 : opioid receptor, delta 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eUHMK1\u003c/td\u003e\n\u003ctd\u003e127933\u003c/td\u003e\n\u003ctd\u003eUHMK1 : U2AF homology motif (UHM) kinase 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ePDYN\u003c/td\u003e\n\u003ctd\u003e5173\u003c/td\u003e\n\u003ctd\u003ePDYN : prodynorphin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ePNOC\u003c/td\u003e\n\u003ctd\u003e5368\u003c/td\u003e\n\u003ctd\u003ePNOC : prepronociceptin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ePOMC\u003c/td\u003e\n\u003ctd\u003e5443\u003c/td\u003e\n\u003ctd\u003ePOMC : proopiomelanocortin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ePRKCD\u003c/td\u003e\n\u003ctd\u003e5580\u003c/td\u003e\n\u003ctd\u003ePRKCD : protein kinase C, delta\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eMAP2K6\u003c/td\u003e\n\u003ctd\u003e5608\u003c/td\u003e\n\u003ctd\u003eMAP2K6 : mitogen-activated protein kinase kinase 6\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ePTPRB\u003c/td\u003e\n\u003ctd\u003e5787\u003c/td\u003e\n\u003ctd\u003ePTPRB : protein tyrosine phosphatase, receptor type, B\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ePTPRC\u003c/td\u003e\n\u003ctd\u003e5788\u003c/td\u003e\n\u003ctd\u003ePTPRC : protein tyrosine phosphatase, receptor type, C\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ePTPRD\u003c/td\u003e\n\u003ctd\u003e5789\u003c/td\u003e\n\u003ctd\u003ePTPRD : protein tyrosine phosphatase, receptor type, D\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eHMBOX1\u003c/td\u003e\n\u003ctd\u003e79618\u003c/td\u003e\n\u003ctd\u003eHMBOX1 : homeobox containing 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eZMYM1\u003c/td\u003e\n\u003ctd\u003e79830\u003c/td\u003e\n\u003ctd\u003eZMYM1 : zinc finger, MYM-type 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eRPS6KA3\u003c/td\u003e\n\u003ctd\u003e6197\u003c/td\u003e\n\u003ctd\u003eRPS6KA3 : ribosomal protein S6 kinase, 90kDa, polypeptide 3\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eRXRG\u003c/td\u003e\n\u003ctd\u003e6258\u003c/td\u003e\n\u003ctd\u003eRXRG : retinoid X receptor, gamma\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eRYR2\u003c/td\u003e\n\u003ctd\u003e6262\u003c/td\u003e\n\u003ctd\u003eRYR2 : ryanodine receptor 2 (cardiac)\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eNAV1\u003c/td\u003e\n\u003ctd\u003e89796\u003c/td\u003e\n\u003ctd\u003eNAV1 : neuron navigator 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSDC4\u003c/td\u003e\n\u003ctd\u003e6385\u003c/td\u003e\n\u003ctd\u003eSDC4 : syndecan 4\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCNKSR2\u003c/td\u003e\n\u003ctd\u003e22866\u003c/td\u003e\n\u003ctd\u003eCNKSR2 : connector enhancer of kinase suppressor of Ras 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eDENND4C\u003c/td\u003e\n\u003ctd\u003e55667\u003c/td\u003e\n\u003ctd\u003eDENND4C : DENN/MADD domain containing 4C\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSNTB2\u003c/td\u003e\n\u003ctd\u003e6645\u003c/td\u003e\n\u003ctd\u003eSNTB2 : syntrophin, beta 2 (dystrophin-associated protein A1, 59kDa, basic component 2)\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ePEG10\u003c/td\u003e\n\u003ctd\u003e23089\u003c/td\u003e\n\u003ctd\u003ePEG10 : paternally expressed 10\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eKIF1B\u003c/td\u003e\n\u003ctd\u003e23095\u003c/td\u003e\n\u003ctd\u003eKIF1B : kinesin family member 1B\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSSTR2\u003c/td\u003e\n\u003ctd\u003e6752\u003c/td\u003e\n\u003ctd\u003eSSTR2 : somatostatin receptor 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSTRN\u003c/td\u003e\n\u003ctd\u003e6801\u003c/td\u003e\n\u003ctd\u003eSTRN : striatin, calmodulin binding protein\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eTACR1\u003c/td\u003e\n\u003ctd\u003e6869\u003c/td\u003e\n\u003ctd\u003eTACR1 : tachykinin receptor 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eTACR3\u003c/td\u003e\n\u003ctd\u003e6870\u003c/td\u003e\n\u003ctd\u003eTACR3 : tachykinin receptor 3\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eMGA\u003c/td\u003e\n\u003ctd\u003e23269\u003c/td\u003e\n\u003ctd\u003eMGA : MGA, MAX dimerization protein\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSATB2\u003c/td\u003e\n\u003ctd\u003e23314\u003c/td\u003e\n\u003ctd\u003eSATB2 : SATB homeobox 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eMAP3K2\u003c/td\u003e\n\u003ctd\u003e10746\u003c/td\u003e\n\u003ctd\u003eMAP3K2 : mitogen-activated protein kinase kinase kinase 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eTRAF3\u003c/td\u003e\n\u003ctd\u003e7187\u003c/td\u003e\n\u003ctd\u003eTRAF3 : TNF receptor-associated factor 3\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eTRPC5\u003c/td\u003e\n\u003ctd\u003e7224\u003c/td\u003e\n\u003ctd\u003eTRPC5 : transient receptor potential cation channel, subfamily C, member 5\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eUTRN\u003c/td\u003e\n\u003ctd\u003e7402\u003c/td\u003e\n\u003ctd\u003eUTRN : utrophin\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCYP26B1\u003c/td\u003e\n\u003ctd\u003e56603\u003c/td\u003e\n\u003ctd\u003eCYP26B1 : cytochrome P450, family 26, subfamily B, polypeptide 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eRICTOR\u003c/td\u003e\n\u003ctd\u003e253260\u003c/td\u003e\n\u003ctd\u003eRICTOR : RPTOR independent companion of MTOR, complex 2\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eZBTB16\u003c/td\u003e\n\u003ctd\u003e7704\u003c/td\u003e\n\u003ctd\u003eZBTB16 : zinc finger and BTB domain containing 16\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eSESTD1\u003c/td\u003e\n\u003ctd\u003e91404\u003c/td\u003e\n\u003ctd\u003eSESTD1 : SEC14 and spectrin domains 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCOL12A1\u003c/td\u003e\n\u003ctd\u003e1303\u003c/td\u003e\n\u003ctd\u003eCOL12A1 : collagen, type XII, alpha 1\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eNAV3\u003c/td\u003e\n\u003ctd\u003e89795\u003c/td\u003e\n\u003ctd\u003eNAV3 : neuron navigator 3\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eCCDC6\u003c/td\u003e\n\u003ctd\u003e8030\u003c/td\u003e\n\u003ctd\u003eCCDC6 : coiled-coil domain containing 6\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eGPR68\u003c/td\u003e\n\u003ctd\u003e8111\u003c/td\u003e\n\u003ctd\u003eGPR68 : G protein-coupled receptor 68\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cp dir=\"auto\"\u003eHOME/gene set analysis/over-representation analysis/enriched pathway-based sets/Network neighborhood-based entity sets (NESTs)/2-next neighbor/PSD-1 set center\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHOME/gene set analysis/over-representation analysis/enriched pathway-based sets/Network neighborhood-based entity sets (NESTs)/2-next neighbor/NCAM set center\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e###6. 5HMC ANALYSIS:PEAK CALLING:MACS\u003cbr\u003e\nApproach 1: call peaks for each sample against the input, then do complex analysis on the lotof it\nMACS + peaksplitter to call subpeaks within complex regions\nMACS/FC1\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_FC1_5hmC_3 --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC1.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_FC1_5hmC_3 --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC1.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMACS/FC2\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_FC2_5hmC_MACS --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC2.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_FC2_5hmC_MACS --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC2.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMACS/HC1\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_HC1_5hmC_MACS --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC1.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_HC1_5hmC_MACS --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC1.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMACS/HC2\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_HC2_5hmC_MACS --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC2.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_HC2_5hmC_MACS --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC2.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eApproach 2: concatenate BAM files for the HC and for the FC --\u0026gt; perform MACS as HC against FC as input (so get peaks where HC is way more enriched than FC); and as FC against HC as input (so get peaks where FC is way more enriched than HC)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ecombine BAMs MACS\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\n/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name 5hmC_DecInFC --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/HC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/FC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam \u0026amp;\n\n/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name 5hmC_IncInFC --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/FC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/HC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam \u0026amp;\n\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e\n/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name 5hmC_DecInFC --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/HC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/FC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam \u0026amp;\n\n/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name 5hmC_IncInFC --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/FC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/HC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam \u0026amp;\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e###7. PARCE OUT REGIONS OF 5HMC, EXPRESSION, AND ISOFORM REGULATION:BEDTOOLS\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCalling DhMRs common to both sequencing experiments. Save the sequence JUST where two peaks overlap (no extra peak space unique to one sample or the other). Also create a \"Peakheight\" file that contains the entries that are overlapped; allows downstream analysis of relative 5hmC changes\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eNote: needed to remove the header line -- bedtools was unable to recognize it: Chromosome      Start   End     Height  SummitPosition\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eCommon Homecage peaks\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_HC1_5hmC_MACS/fear_conditioning_HC1_5hmC_MACS_peaks.subpeaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_HC2_5hmC_MACS/fear_conditioning_HC2_5hmC_MACS_peaks.subpeaks.bed -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt \u0026gt;HC1_HC2_BTIntersect.peaks.subpeaks.bed\n\n\nCommon homecage peaks; retain all the fields, so that peak height can be kept for downstream analyses\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ebedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_HC1_5hmC_MACS/fear_conditioning_HC1_5hmC_MACS_peaks.subpeaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_HC2_5hmC_MACS/fear_conditioning_HC2_5hmC_MACS_peaks.subpeaks.bed -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt \u0026gt;HC1_HC2_BTIntersect.peaks.subpeaks.bed\n\n\nCommon homecage peaks; retain all the fields, so that peak height can be kept for downstream analyses\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ebedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_HC1_5hmC_MACS/fear_conditioning_HC1_5hmC_MACS_peaks.subpeaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_HC2_5hmC_MACS/fear_conditioning_HC2_5hmC_MACS_peaks.subpeaks.bed -bed -sorted -wa -wb -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt \u0026gt;HC1_HC2_BTIntersect.peaks.subpeaks_PeakHeight.bed\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"Common fear conditioning peaks\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eCommon fear conditioning peaks\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ebedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_FC1_5hmC_MACS/fear_conditioning_FC1_5hmC_3_peaks.subpeaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_FC2_5hmC_MACS/fear_conditioning_FC2_5hmC_MACS_peaks.subpeaks.bed -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt \u0026gt;FC1_FC2_BTIntersect.peaks.subpeaks.bed\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"Common fear conditioning peaks; retain all the fields, so that peak height can be kept for downstream analyses\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eCommon fear conditioning peaks; retain all the fields, so that peak height can be kept for downstream analyses\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ebedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_FC1_5hmC_MACS/fear_conditioning_FC1_5hmC_3_peaks.subpeaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_FC2_5hmC_MACS/fear_conditioning_FC2_5hmC_MACS_peaks.subpeaks.bed -bed -sorted -wa -wb -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt \u0026gt;FC1_FC2_BTIntersect.peaks.subpeaks_PeakHeight.bed\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"\nR Script to take the average of columns 4 and 9 from the HC1_HC2_BTIntersect.peaks.subpeaks_PeakHeight.bed file and adds it as a new column to the HC1_HC2_BTIntersect.peaks.subpeaks.bed file (ditto for FC files). End up with a BED file containing DhMRs + the average peak height from both datasets at this locus.\n```{r}\nFC_peaks = read.table(\u0026quot;FC1_FC2_BTIntersect.peaks.subpeaks.bed\u0026quot;)\nFC_heights = read.table(\u0026quot;FC1_FC2_BTIntersect.peaks.subpeaks_PeakHeight.bed\u0026quot;)\nFC_avg_heights = (FC_heights$V4+FC_heights$V9)/2\nFC_peaks$FC_avg_heights = FC_avg_heights\nwrite.table(FC_peaks, sep = \u0026quot;\\t\u0026quot;, col.names = F, row.names = F, quote = F, file = \u0026quot;/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed\u0026quot;)\n\nHC_peaks = read.table(\u0026quot;HC1_HC2_BTIntersect.peaks.subpeaks.bed\u0026quot;)\nHC_heights = read.table(\u0026quot;HC1_HC2_BTIntersect.peaks.subpeaks_PeakHeight.bed\u0026quot;)\nHC_avg_heights = (HC_heights$V4+HC_heights$V9)/2\nHC_peaks$HC_avg_heights = HC_avg_heights\nwrite.table(HC_peaks, sep = \u0026quot;\\t\u0026quot;, col.names = F, row.names = F, quote = F, file = \u0026quot;/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed\u0026quot;)\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e\nR Script to take the average of columns 4 and 9 from the HC1_HC2_BTIntersect.peaks.subpeaks_PeakHeight.bed file and adds it as a new column to the HC1_HC2_BTIntersect.peaks.subpeaks.bed file (ditto for FC files). End up with a BED file containing DhMRs + the average peak height from both datasets at this locus.\n```{r}\nFC_peaks = read.table(\"FC1_FC2_BTIntersect.peaks.subpeaks.bed\")\nFC_heights = read.table(\"FC1_FC2_BTIntersect.peaks.subpeaks_PeakHeight.bed\")\nFC_avg_heights = (FC_heights$V4+FC_heights$V9)/2\nFC_peaks$FC_avg_heights = FC_avg_heights\nwrite.table(FC_peaks, sep = \"\\t\", col.names = F, row.names = F, quote = F, file = \"/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed\")\n\nHC_peaks = read.table(\"HC1_HC2_BTIntersect.peaks.subpeaks.bed\")\nHC_heights = read.table(\"HC1_HC2_BTIntersect.peaks.subpeaks_PeakHeight.bed\")\nHC_avg_heights = (HC_heights$V4+HC_heights$V9)/2\nHC_peaks$HC_avg_heights = HC_avg_heights\nwrite.table(HC_peaks, sep = \"\\t\", col.names = F, row.names = F, quote = F, file = \"/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed\")\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFC_5hmC_Inc: Peaks that appear in fear conditioning; no peak at this loci in HC\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -v -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt \u0026gt;FC_5hmC_Inc.bed\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ebedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -v -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt \u0026gt;FC_5hmC_Inc.bed\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFC_5hmC_Dec: Peaks disappear in fear conditioning; peak at this locus in HC dissapears in FC\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed  -v -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt \u0026gt;FC_5hmC_Dec.bed\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ebedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed  -v -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt \u0026gt;FC_5hmC_Dec.bed\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRemove tmp files\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"rm FC1_FC2_BTIntersect.peaks.subpeaks.bed FC1_FC2_BTIntersect.peaks.subpeaks_PeakHeight.bed FC_5hmC_Dec_2.bed HC1_HC2_BTIntersect.peaks.subpeaks.bed FC_5hmC_Inc_2.bed HC1_HC2_BTIntersect.peaks.subpeaks_PeakHeight.bed\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003erm FC1_FC2_BTIntersect.peaks.subpeaks.bed FC1_FC2_BTIntersect.peaks.subpeaks_PeakHeight.bed FC_5hmC_Dec_2.bed HC1_HC2_BTIntersect.peaks.subpeaks.bed FC_5hmC_Inc_2.bed HC1_HC2_BTIntersect.peaks.subpeaks_PeakHeight.bed\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFC_5hmC_common_Inc: Peak is present in both HC and FC samples; but peak is smaller in HC than FC (cutoff = ??)\nbedtools/FC_HC_Common.bed\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt \u0026gt;FC_HC_Common.bed\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ebedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt \u0026gt;FC_HC_Common.bed\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ebedtools/FC_HC_Common_PeakHeights.bed\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -bed -sorted -wa -wb -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt \u0026gt;FC_HC_Common_PeakHeights.bed\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ebedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -bed -sorted -wa -wb -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt \u0026gt;FC_HC_Common_PeakHeights.bed\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eConcatenate files (to include intermediate changes in peak height)\n-5hmC increased during FC\n-5hmC decreased during FC\n-all dynamic 5hmC loci during FC\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cat FC_HC_Common_FCgreaterHC.bed FC_5hmC_Inc.bed \u0026gt; FC_5hmC_Inc_increment.bed\ncat FC_HC_Common_HCgreaterFC.bed FC_5hmC_Dec.bed \u0026gt; FC_5hmC_Dec_increment.bed\ncat HCvsFC_5hmC_Inc_increment.bed HCvsFC_5hmC_Dec_increment.bed \u0026gt; HCvsFC_AllDynamic5hmC.bed\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecat FC_HC_Common_FCgreaterHC.bed FC_5hmC_Inc.bed \u0026gt; FC_5hmC_Inc_increment.bed\ncat FC_HC_Common_HCgreaterFC.bed FC_5hmC_Dec.bed \u0026gt; FC_5hmC_Dec_increment.bed\ncat HCvsFC_5hmC_Inc_increment.bed HCvsFC_5hmC_Dec_increment.bed \u0026gt; HCvsFC_AllDynamic5hmC.bed\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFC greater than HC and FC less than HC, cutoff - peak height difference \u0026gt; or \u0026lt; 25\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"FC_HC_common_peaks = read.table(\u0026quot;FC_HC_Common.bed\u0026quot;)\nFC_HC_common_heights = read.table(\u0026quot;FC_HC_Common_PeakHeights.bed\u0026quot;)\nFC_HC_common_peaks$FCminusHC \u0026lt;- FC_HC_common_heights$V4 - FC_HC_common_heights$V10\nFC_HC_common_peaks$HCminusFC \u0026lt;- FC_HC_common_heights$V10 - FC_HC_common_heights$V4\nFC_HC_Common_FCgreaterHC = subset(FC_HC_common_peaks, FCminusHC \u0026gt;=25)\nwrite.table(FC_HC_Common_FCgreaterHC, sep = \u0026quot;\\t\u0026quot;, col.names = F, row.names = F, quote = F, file = \u0026quot;/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_HC_Common_FCgreaterHC.bed\u0026quot;)\nFC_HC_Common_HCgreaterFC = subset(FC_HC_common_peaks, HCminusFC \u0026gt;=25)\nwrite.table(FC_HC_Common_HCgreaterFC, sep = \u0026quot;\\t\u0026quot;, col.names = F, row.names = F, quote = F, file = \u0026quot;/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_HC_Common_HCgreaterFC.bed\u0026quot;)\"\u003e\u003cpre lang=\"{r}\" class=\"notranslate\"\u003e\u003ccode\u003eFC_HC_common_peaks = read.table(\"FC_HC_Common.bed\")\nFC_HC_common_heights = read.table(\"FC_HC_Common_PeakHeights.bed\")\nFC_HC_common_peaks$FCminusHC \u0026lt;- FC_HC_common_heights$V4 - FC_HC_common_heights$V10\nFC_HC_common_peaks$HCminusFC \u0026lt;- FC_HC_common_heights$V10 - FC_HC_common_heights$V4\nFC_HC_Common_FCgreaterHC = subset(FC_HC_common_peaks, FCminusHC \u0026gt;=25)\nwrite.table(FC_HC_Common_FCgreaterHC, sep = \"\\t\", col.names = F, row.names = F, quote = F, file = \"/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_HC_Common_FCgreaterHC.bed\")\nFC_HC_Common_HCgreaterFC = subset(FC_HC_common_peaks, HCminusFC \u0026gt;=25)\nwrite.table(FC_HC_Common_HCgreaterFC, sep = \"\\t\", col.names = F, row.names = F, quote = F, file = \"/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_HC_Common_HCgreaterFC.bed\")\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eSorting Bedfiles\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bedtools sort -chrThenSizeA -i HCvsFC_RNAseq_wholegene_ALL.bed \u0026gt;HCvsFC_RNAseq_wholegene_ALL_sorted.bed\n5hmC-seq/all\nbedtools sort -chrThenSizeA -i HCvsFC_AllDynamic5hmC.bed \u0026gt;HCvsFC_AllDynamic5hmC_sorted.bed\n5hmC-seq/upregulated\nbedtools sort -chrThenSizeA -i HCvsFC_5hmC_Inc_increment.bed HCvsFC_5hmC_Inc_increment_sorted.bed\n5hmC-seq/downregulated\nbedtools sort -chrThenSizeA -i HCvsFC_5hmC_Dec_increment.bed \u0026gt;HCvsFC_5hmC_Dec_increment_sorted.bed\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ebedtools sort -chrThenSizeA -i HCvsFC_RNAseq_wholegene_ALL.bed \u0026gt;HCvsFC_RNAseq_wholegene_ALL_sorted.bed\n5hmC-seq/all\nbedtools sort -chrThenSizeA -i HCvsFC_AllDynamic5hmC.bed \u0026gt;HCvsFC_AllDynamic5hmC_sorted.bed\n5hmC-seq/upregulated\nbedtools sort -chrThenSizeA -i HCvsFC_5hmC_Inc_increment.bed HCvsFC_5hmC_Inc_increment_sorted.bed\n5hmC-seq/downregulated\nbedtools sort -chrThenSizeA -i HCvsFC_5hmC_Dec_increment.bed \u0026gt;HCvsFC_5hmC_Dec_increment_sorted.bed\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMISCELLANEOUS\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ebedtools/RNAseq_wholegene/5hmC_ALL\n-wb option --\u0026gt; write out the entirety of the 5hmC peak that overlaps with a 5hmC region; find any motifs even if they aren't precisely within body of gene\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb \u0026gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed\n#bedtools/RNAseq_wholegene/5hmC_Inc\nbedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb \u0026gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed\n#bedtools/RNAseq_wholegene/5hmC_Dec\nbedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb \u0026gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed\n#bedtools/RNAseq_wholegene_upregulated/5hmC_Dec\nbedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb \u0026gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed\n#bedtools/RNAseq_wholegene_downregulated/5hmC_Dec\nbedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb \u0026gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed\n#bedtools/RNAseq_wholegene/5hmC_Dec\nbedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb \u0026gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed\n#bedtools/RNAseq_wholegene/5hmC_Dec\nbedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb \u0026gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ebedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb \u0026gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed\n#bedtools/RNAseq_wholegene/5hmC_Inc\nbedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb \u0026gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed\n#bedtools/RNAseq_wholegene/5hmC_Dec\nbedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb \u0026gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed\n#bedtools/RNAseq_wholegene_upregulated/5hmC_Dec\nbedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb \u0026gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed\n#bedtools/RNAseq_wholegene_downregulated/5hmC_Dec\nbedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb \u0026gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed\n#bedtools/RNAseq_wholegene/5hmC_Dec\nbedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb \u0026gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed\n#bedtools/RNAseq_wholegene/5hmC_Dec\nbedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb \u0026gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"MISCELLANEOUS IDEAS\n -bedtools/RNAseq_all/5hmCseq_genes_TSS_all\n -bedtools/RNAseq_all/5hmCseq_genes_TSS_all\n -bedtools/RNAseq_all/5hmCseq_genes_TSS_all\n -bedtools/RNAseq_splicing_all/5hmCseq_exons_all\n -bedtools/RNAseq_splicing_all/5hmCseq_introns_all\n -bedtools/RNAseq_splicing_all/5hmCseq_exon-intron_all\n -bedtools/RNAseq_promoters/5hmC_Inc\n -bedtools/RNAseq_promoters/5hmC_Dec\n -bedtools/RNAseq_promoters_upregulated/5hmC_Inc\n -bedtools/RNAseq_promoters_downregulated/5hmC_Dec\n -bedtools/CTCFbindingsites/5hmC_all\n -bedtools/CTCFbindingsites/5hmC_inc\n -bedtools/CTCFbindingsites/5hmC_dec\n -bedtools/CTCFbindingsites/5hmC_Inc/annotation\n -bedtools/lncRNAs_all/5hmc_all\n -bedtools/lncRNAs_/5hmc_all\n -CTCF binding sites with dynamic 5hmC in FC\n -whole genome CTCF binding sites from Corces Lab\n -Enhancers with dynamic 5hmC in FC\n -FANTOM identified enhancers\n -Neural specific enhancers\n -Other Datasets? Hi-seq? TADs or mini-TADs, or CTFCF binding sites influential at these positions..proxy for conformational change\n\nOther interesting coordinate sets to generate\n -RNAseq and lncRNA\n -RNAseq and microRNAs\n -RNAseq and TRAPSeq\n -5hmCseq and enhancers\n -5hmCseq and insulators\n\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eMISCELLANEOUS IDEAS\n -bedtools/RNAseq_all/5hmCseq_genes_TSS_all\n -bedtools/RNAseq_all/5hmCseq_genes_TSS_all\n -bedtools/RNAseq_all/5hmCseq_genes_TSS_all\n -bedtools/RNAseq_splicing_all/5hmCseq_exons_all\n -bedtools/RNAseq_splicing_all/5hmCseq_introns_all\n -bedtools/RNAseq_splicing_all/5hmCseq_exon-intron_all\n -bedtools/RNAseq_promoters/5hmC_Inc\n -bedtools/RNAseq_promoters/5hmC_Dec\n -bedtools/RNAseq_promoters_upregulated/5hmC_Inc\n -bedtools/RNAseq_promoters_downregulated/5hmC_Dec\n -bedtools/CTCFbindingsites/5hmC_all\n -bedtools/CTCFbindingsites/5hmC_inc\n -bedtools/CTCFbindingsites/5hmC_dec\n -bedtools/CTCFbindingsites/5hmC_Inc/annotation\n -bedtools/lncRNAs_all/5hmc_all\n -bedtools/lncRNAs_/5hmc_all\n -CTCF binding sites with dynamic 5hmC in FC\n -whole genome CTCF binding sites from Corces Lab\n -Enhancers with dynamic 5hmC in FC\n -FANTOM identified enhancers\n -Neural specific enhancers\n -Other Datasets? Hi-seq? TADs or mini-TADs, or CTFCF binding sites influential at these positions..proxy for conformational change\n\nOther interesting coordinate sets to generate\n -RNAseq and lncRNA\n -RNAseq and microRNAs\n -RNAseq and TRAPSeq\n -5hmCseq and enhancers\n -5hmCseq and insulators\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e###8. ANNOTATION ANALYSIS:HOMER\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e-HCvsFC_5hmCseq_Inc\n-HCvsFC_5hmCseq_Dec\n-HCvsFC_5hmCseq_ALL\n-HCvsFC RNAseq\n-HCvsFC 5hmC \u0026amp; RNAseq overlap\n-HCvsFC TRAPseq: unique and distinct from regular RNAseq\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePeak sets: HCvsFC_5hmC_Dec_increment.bed, HCvsFC_5hmC_Inc_increment.bed, HCvsFC_AllDHMR.bed\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAnnotation of HCvsFC_AllDHMR.bed (using UCSC GTF used to align the files in the first place\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_AllDHMR.bed mm10 -annStats HCvsFC_AllDHMR_HomerAnnStats.txt \u0026gt;HCvsFC_AllDHMR_HomerAnnotations_2.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_AllDHMR.bed mm10 -annStats HCvsFC_AllDHMR_HomerAnnStats.txt \u0026gt;HCvsFC_AllDHMR_HomerAnnotations_2.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAnnotation of peaks called by concatenating replicate fasta files and using either HC or FC as the input\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_Dynamic5hmC_ALL.bed mm10 -annStats HCvsFC_ALLDynamic5hmC_HomerAnnStats.txt \u0026gt;HCvsFC_ALLDynamic5hmC_HomerAnnotations.txt\n\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_Dynamic5hmC_ALL.bed mm10 -annStats HCvsFC_ALLDynamic5hmC_HomerAnnStats.txt \u0026gt;HCvsFC_ALLDynamic5hmC_HomerAnnotations.txt\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAnnotation of Smarca4 binding sites:\nNote: use homer annotation files in this folder w/ all of Bing's stuff in there\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/smarca4_forbrain_chipseq.bed mm9 -annStats smarca4_forebrain_HomerAnnStats.txt \u0026gt;smarca4_forebrain_HomerAnnotations.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/smarca4_forbrain_chipseq.bed mm9 -annStats smarca4_forebrain_HomerAnnStats.txt \u0026gt;smarca4_forebrain_HomerAnnotations.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAnnotate RNAseq Genes in Homer\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL.bed mm10  \u0026gt;HCvsFC_RNAseq_wholegene_HomerAnnotations.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL.bed mm10  \u0026gt;HCvsFC_RNAseq_wholegene_HomerAnnotations.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003epython script\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"~/bin/ExtractGeneNames_HomerOutput_Down.py HCvsFC_AllDHMR_HomerAnnotations.txt HCvsFC_AllDHMR_HomerAnnotations_Craig.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e~/bin/ExtractGeneNames_HomerOutput_Down.py HCvsFC_AllDHMR_HomerAnnotations.txt HCvsFC_AllDHMR_HomerAnnotations_Craig.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eORDER OF ANNOTATION ENRICHMENT CHANGE (HC \u0026gt; FC)\nintron\u0026gt;exon\u0026gt;intergenic\u0026gt;LTR~\u0026gt;promoter\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePython Script for Homer Annotations\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"import os\nimport fnmatch\nimport pybedtools\n \n \nInputGeneNames = open('HCvsFC_AllDHMR_HomerAnnotations.txt','r')\nGeneFileGTF = open('/home/byao/homer/data/accession/mouse2gene.tsv','r')\nGeneList = {}\nAccessionNumbers = {}\n \nfor line in GeneFileGTF:\n\tline = line.strip()\n\ttry:\n\t\tRefSeqID, Unknown, HsID, Accession, EnsemblID, Unknown2, GeneID = line.split('\\t')\n\t\tGeneList[str(RefSeqID)] = GeneID\n\texcept:\n\t\tpass\n \n \n \n \nfor line in InputGeneNames:\n    line = line.strip()\n    PeakID, Chr, Start, End, Strand, PeakScore, Focus, Ratio_Region, Remainder = line.split('\\t',8)\n    try:\n        Region, Remainder = Ratio_Region.split('(')\n        try:\n            Accession, Remainder = Remainder.split(',')\n            OutputLine = str(Chr) + '\\t' + str(Start) + '\\t' + str(End) + '\\t' + str(Ratio_Region)\n            AccessionNumbers[str(Accession)] = OutputLine\n        except:\n            Accession, Remainder = Remainder.split(')')\n            OutputLine = str(Chr) + '\\t' + str(Start) + '\\t' + str(End) + '\\t' + str(Ratio_Region)\n            AccessionNumbers[str(Accession)] = OutputLine\n    except:\n        pass\n \n \nfor AccessionNumber in AccessionNumbers.keys():\n    try:\n        print str(AccessionNumbers[str(AccessionNumber)]) + '\\t' + str(GeneList[str(AccessionNumber)])\n    except:\n        print str(AccessionNumber) + ' is missing'\n\n\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eimport os\nimport fnmatch\nimport pybedtools\n \n \nInputGeneNames = open('HCvsFC_AllDHMR_HomerAnnotations.txt','r')\nGeneFileGTF = open('/home/byao/homer/data/accession/mouse2gene.tsv','r')\nGeneList = {}\nAccessionNumbers = {}\n \nfor line in GeneFileGTF:\n\tline = line.strip()\n\ttry:\n\t\tRefSeqID, Unknown, HsID, Accession, EnsemblID, Unknown2, GeneID = line.split('\\t')\n\t\tGeneList[str(RefSeqID)] = GeneID\n\texcept:\n\t\tpass\n \n \n \n \nfor line in InputGeneNames:\n    line = line.strip()\n    PeakID, Chr, Start, End, Strand, PeakScore, Focus, Ratio_Region, Remainder = line.split('\\t',8)\n    try:\n        Region, Remainder = Ratio_Region.split('(')\n        try:\n            Accession, Remainder = Remainder.split(',')\n            OutputLine = str(Chr) + '\\t' + str(Start) + '\\t' + str(End) + '\\t' + str(Ratio_Region)\n            AccessionNumbers[str(Accession)] = OutputLine\n        except:\n            Accession, Remainder = Remainder.split(')')\n            OutputLine = str(Chr) + '\\t' + str(Start) + '\\t' + str(End) + '\\t' + str(Ratio_Region)\n            AccessionNumbers[str(Accession)] = OutputLine\n    except:\n        pass\n \n \nfor AccessionNumber in AccessionNumbers.keys():\n    try:\n        print str(AccessionNumbers[str(AccessionNumber)]) + '\\t' + str(GeneList[str(AccessionNumber)])\n    except:\n        print str(AccessionNumber) + ' is missing'\n\n\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e###9. MOTIF ANALYSIS:HOMER:MOTIF ID AND LOCALIZATION\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eBackground\u003c/em\u003e Files are critical - they are how homer decides which TF sites are overrepresented\nwholegene - mm10 all genes\nexons - mm10 all exons\nintrons - mm10 all introns\n5hmC regions - random sequences from the genome\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTesting out different -size parameters (avg size of the DNA fragments in the BED file - input into the findmotifs script)\nsize 2000\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_2k/ -size 2000 -mis 1 -S 10 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_2k/ -size 2000 -mis 1 -S 10 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003esize 100000\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_100k/ -size 100000 -mis 1 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_100k/ -size 100000 -mis 1 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003esize 10000\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_10k/ -size 10000 -mis 1 -S 10 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_10k/ -size 10000 -mis 1 -S 10 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003esize 50000\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_50k/ -size 50000 -mis 1 -S 10 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_50k/ -size 50000 -mis 1 -S 10 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eConclusion about -size parameter:\u003c/em\u003e having a size closer to the real average length of the regions in the bedfile makes for more statistically significant results, but much longer run time. 100k size failed to finish.\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eAwk script to determine average size of regions in a bed file\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cat CTCFPeaksClustered.bed | awk -F'\\t' 'BEGIN{SUM=0}{ SUM+=$3-$2 }END{print SUM}'\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecat CTCFPeaksClustered.bed | awk -F'\\t' 'BEGIN{SUM=0}{ SUM+=$3-$2 }END{print SUM}'\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"_Average size of each region in bed file:_\n -RNAseq Introns: 15000 (avg was 8800)\n -RNAseq exons: 3000 (avg 325)\n -CTCFPeaksClustered = 500 (avg 362)\nOvershooting may be better than under\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e_Average size of each region in bed file:_\n -RNAseq Introns: 15000 (avg was 8800)\n -RNAseq exons: 3000 (avg 325)\n -CTCFPeaksClustered = 500 (avg 362)\nOvershooting may be better than under\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eMotif Finding:\u003c/em\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eRNAseq, whole gene, upregulated\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_upregulated.bed mm10 HCvsFC_RNAseq_wholegene_upregulated_motifs/ -size 50000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_upregulated.bed mm10 HCvsFC_RNAseq_wholegene_upregulated_motifs/ -size 50000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eRNAseq, whole gene, downregulated\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_downregulated.bed mm10 HCvsFC_RNAseq_wholegene_downregulated_motifs/ -size 50000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_downregulated.bed mm10 HCvsFC_RNAseq_wholegene_downregulated_motifs/ -size 50000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eRNAseq, promoters, all differentially expressed (DE)\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_promoters_ALL.bed mm10 HCvsFC_RNAseq_promoters_motifs/ -size 2000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allpromoters_-2000_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_promoters_ALL.bed mm10 HCvsFC_RNAseq_promoters_motifs/ -size 2000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allpromoters_-2000_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eRNAseq, promoters, upregulated (DE)\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_promoter_upregulated.bed mm10 HCvsFC_RNAseq_promoter_upregulated_motifs/ -size 2000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allpromoters_-2000_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_promoter_upregulated.bed mm10 HCvsFC_RNAseq_promoter_upregulated_motifs/ -size 2000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allpromoters_-2000_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eRNAseq, promoters, downregulated\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_promoters_downregulated.bed mm10 HCvsFC_RNAseq_promoter_downregulated_motifs/ -size 2000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allpromoters_-2000_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_promoters_downregulated.bed mm10 HCvsFC_RNAseq_promoter_downregulated_motifs/ -size 2000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allpromoters_-2000_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eRNAseq, introns, all differentially expressed (DE)\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_introns_ALL.bed mm10 HCvsFC_RNAseq_intron_motifs/ -size 15000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allintrons_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_introns_ALL.bed mm10 HCvsFC_RNAseq_intron_motifs/ -size 15000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allintrons_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eRNAseq, introns, upregulated\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_introns_upregulated.bed mm10 HCvsFC_RNAseq_intron_upregulated_motifs/ -size 15000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allintrons_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_introns_upregulated.bed mm10 HCvsFC_RNAseq_intron_upregulated_motifs/ -size 15000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allintrons_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eRNAseq, introns, downregulated\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_introns_downregulated.bed mm10 HCvsFC_RNAseq_intron_downregulated_motifs/ -size 15000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allintrons_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_introns_downregulated.bed mm10 HCvsFC_RNAseq_intron_downregulated_motifs/ -size 15000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allintrons_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eRNAseq, exons, all DE\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_exons_ALL.bed mm10 HCvsFC_RNAseq_exon_motifs/ -size 3000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allexons_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_exons_ALL.bed mm10 HCvsFC_RNAseq_exon_motifs/ -size 3000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allexons_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eRNAseq, exons, upregulated\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_exons_upregulated.bed mm10 HCvsFC_RNAseq_exon_upregulated_motifs/ -size 3000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allexons_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_exons_upregulated.bed mm10 HCvsFC_RNAseq_exon_upregulated_motifs/ -size 3000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allexons_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eRNAseq, exons, downregulated\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_exons_downregulated.bed mm10 HCvsFC_RNAseq_exon_downregulated_motifs/ -size 3000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allexons_background.bed -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_exons_downregulated.bed mm10 HCvsFC_RNAseq_exon_downregulated_motifs/ -size 3000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allexons_background.bed -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e5hmC peaks, all DhMRs\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_AllDHMR.bed mm10 HCvsFC_5hmC_motifs/ -size 700 -mis 2 -S 25 -p 25 -h \u0026amp;\"\u003e\u003cpre lang=\"_\" class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_AllDHMR.bed mm10 HCvsFC_5hmC_motifs/ -size 700 -mis 2 -S 25 -p 25 -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e5hmC peaks, 5hmC Increased\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_5hmC_Inc_increment.bed mm10 HCvsFC_5hmC_Inc_motifs/ -size 700 -mis 2 -S 25 -p 25 -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_5hmC_Inc_increment.bed mm10 HCvsFC_5hmC_Inc_motifs/ -size 700 -mis 2 -S 25 -p 25 -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e5hmC peaks, 5hmC Decreased\u003c/em\u003e\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_5hmC_Dec_increment.bed mm10 HCvsFC_5hmC_Dec_motifs/ -size 700 -mis 2 -S 25 -p 25 -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_5hmC_Dec_increment.bed mm10 HCvsFC_5hmC_Dec_motifs/ -size 700 -mis 2 -S 25 -p 25 -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCTCFPeaksClustered: dataset from Victor Corces or all CTCF binding sites genome wide\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/CTCFPeaksClustered.bed mm10 CTCFPeaksClustered_motifs/ -size 700 -mis 2 -S 25 -p 25 -h \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003efindMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/CTCFPeaksClustered.bed mm10 CTCFPeaksClustered_motifs/ -size 700 -mis 2 -S 25 -p 25 -h \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"Other Analyses\n -FindMotifs/5hmC-Seq/FC_5hmC_Inc\n -FindMotifs/5hmC-Seq/FC_5hmC_Dec\n -FindMotifs/5hmC-Seq/FC_5hmC_Common_Inc\n -FindMotifs/5hmC-Seq/FC_5hmC_Common_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Dec\n -FindMotifs/5hmC-Seq_Enhancers/5hmC_Enhancers_overlap\n -FindMotifs/5hmC-Seq_CTCF/5hmC_CTCF_overlap\n -FindMotifs/CTCF Clustered Peaks\n\nMore Other analyses:\n -FindMotifs/5hmC-Seq/FC_5hmC_Inc\n -FindMotifs/5hmC-Seq/FC_5hmC_Dec\n -FindMotifs/5hmC-Seq/FC_5hmC_Common_Inc\n -FindMotifs/5hmC-Seq/FC_5hmC_Common_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Dec\n -FindMotifs/5hmC-Seq_Enhancers/5hmC_Enhancers_overlap\n -FindMotifs/5hmC-Seq_CTCF/5hmC_CTCF_overlap\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eOther Analyses\n -FindMotifs/5hmC-Seq/FC_5hmC_Inc\n -FindMotifs/5hmC-Seq/FC_5hmC_Dec\n -FindMotifs/5hmC-Seq/FC_5hmC_Common_Inc\n -FindMotifs/5hmC-Seq/FC_5hmC_Common_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Dec\n -FindMotifs/5hmC-Seq_Enhancers/5hmC_Enhancers_overlap\n -FindMotifs/5hmC-Seq_CTCF/5hmC_CTCF_overlap\n -FindMotifs/CTCF Clustered Peaks\n\nMore Other analyses:\n -FindMotifs/5hmC-Seq/FC_5hmC_Inc\n -FindMotifs/5hmC-Seq/FC_5hmC_Dec\n -FindMotifs/5hmC-Seq/FC_5hmC_Common_Inc\n -FindMotifs/5hmC-Seq/FC_5hmC_Common_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Dec\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Inc\n -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Dec\n -FindMotifs/5hmC-Seq_Enhancers/5hmC_Enhancers_overlap\n -FindMotifs/5hmC-Seq_CTCF/5hmC_CTCF_overlap\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e###10. 5HMC ENRICHMENT ANALYSIS:HOMER:HISTOGRAMS AND SCATTERPLOTS: MOTIFS, PROMOTERS, EXONS, INTRONS, EXONS-INTRONS, CTCF, OTHER GENOMIC REGIONS (CPG ISLANDS, EXONS, LNCRNA, PSEUDOGENES, LOW COMPLEXITY REGIONS, SIMPLE REPEATS, SINES, SATELLITES\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e\u003cstrong\u003eWorkflow\u003c/strong\u003e\u003c/em\u003e\nA) Create tag directories -\u0026gt; containers with all of the bam file reads (from all replicates) in one container - used to pileup all the reads from one condition in order to get a comprehensive picture of the density of reads in particular loci\nB) Create a .motif file using seq2profile, or download the .motif file from \u003ca href=\"http://homer.salk.edu/homer/custom.motifs\" rel=\"nofollow\"\u003ehttp://homer.salk.edu/homer/custom.motifs\u003c/a\u003e\nC) Predict sites where the TF will bind in a set of coordinates using annotatepeaks -m -mbed\nD) Find intensity of epigenetic marks at predetermined set of motif binding sites (as from #2) using annotatepeaks -hist -d to feed in tag directories. Can do this with any predetermined set of coordinates (i.e. RNAseq exon boundaries, promoters, etc)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eA) Create tag directories -\u0026gt; containers with all of the bam file reads (from all replicates) in one container - used to pileup all the reads from one condition in order to get a comprehensive picture of the density of reads in particular loci\u003c/strong\u003e\nTag Directories of 5hmC BAM files to calculate enrichment of 5hmC at loci specified in BED format\nmakeTagDirectory  [options]  \u0026lt;\"2\u0026gt; \u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHC\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"makeTagDirectory HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC1.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC2.bam\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emakeTagDirectory HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC1.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC2.bam\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFC\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"makeTagDirectory FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC1.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC2.bam \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emakeTagDirectory FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC1.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC2.bam \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eInput\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"makeTagDirectory Input_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003emakeTagDirectory Input_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eB) Create a .motif file using seq2profile, or download the .motif file from \u003ca href=\"http://homer.salk.edu/homer/custom.motifs\" rel=\"nofollow\"\u003ehttp://homer.salk.edu/homer/custom.motifs\u003c/a\u003e\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eObtaining .motif files: contain the consensus sequence and the frequency distribution of each nucleotide in the consensue\nTwo methods:\n-use seq2profile to create .motif file form within Homer\n-download pre-made consensus motif from Homer: \u003ca href=\"http://homer.salk.edu/homer/custom.motifs\" rel=\"nofollow\"\u003ehttp://homer.salk.edu/homer/custom.motifs\u003c/a\u003e\n-may be best option - seq2profile doesn't appear to produce the same profile, as is displayed on URL; also the .motif profiles form seq2profile appear much less complex\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eTFs of interest and their consensus sequences:\u003c/p\u003e\n\u003ctable\u003e\n\u003cthead\u003e\n\u003ctr\u003e\n\u003cth\u003eGene\u003c/th\u003e\n\u003cth\u003eConsensus Motif\u003c/th\u003e\n\u003c/tr\u003e\n\u003c/thead\u003e\n\u003ctbody\u003e\n\u003ctr\u003e\n\u003ctd\u003eHIF-1b\u003c/td\u003e\n\u003ctd\u003eRTACGTGC\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ec-Myc\u003c/td\u003e\n\u003ctd\u003eVVCCACGTGG\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003en-Myc\u003c/td\u003e\n\u003ctd\u003eVRCCACGTGG\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eMaz\u003c/td\u003e\n\u003ctd\u003eGGGGGGGG\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eStat3\u003c/td\u003e\n\u003ctd\u003eCTTCCGGGAA\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003eHRE (HSF)\u003c/td\u003e\n\u003ctd\u003eTTCTAGAABNTTCTA\u003c/td\u003e\n\u003c/tr\u003e\n\u003ctr\u003e\n\u003ctd\u003ePRDM1/Blimp1\u003c/td\u003e\n\u003ctd\u003eACTTTCACTTTC\u003c/td\u003e\n\u003c/tr\u003e\n\u003c/tbody\u003e\n\u003c/table\u003e\n\u003cp dir=\"auto\"\u003eseq2motif: creating .motif files - least stringent: \u003cem\u003e2\u003c/em\u003e mismatches allowed\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"seq2profile.pl RTACGTGC 2 HIF-1b \u0026gt; HIF-1b.motif\nseq2profile.pl VVCCACGTGG 2 c-Myc \u0026gt; c-Myc.motif\nseq2profile.pl VRCCACGTGG 2 n-Myc \u0026gt; n-Myc.motif\nseq2profile.pl GGGGGGGG 2 Maz \u0026gt; Maz_2.motif\nseq2profile.pl CTTCCGGGAA 2 Stat3 \u0026gt; Stat3.motif\nseq2profile.pl TTCTAGAABNTTCTA 2 HRE \u0026gt; HRE.motif\nseq2profile.pl ACTTTCACTTTC 2 PRDM1 \u0026gt; PRDM1_2.motif\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eseq2profile.pl RTACGTGC 2 HIF-1b \u0026gt; HIF-1b.motif\nseq2profile.pl VVCCACGTGG 2 c-Myc \u0026gt; c-Myc.motif\nseq2profile.pl VRCCACGTGG 2 n-Myc \u0026gt; n-Myc.motif\nseq2profile.pl GGGGGGGG 2 Maz \u0026gt; Maz_2.motif\nseq2profile.pl CTTCCGGGAA 2 Stat3 \u0026gt; Stat3.motif\nseq2profile.pl TTCTAGAABNTTCTA 2 HRE \u0026gt; HRE.motif\nseq2profile.pl ACTTTCACTTTC 2 PRDM1 \u0026gt; PRDM1_2.motif\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ecreating .motif files -  medium stringent: \u003cem\u003e1\u003c/em\u003e mismatches allowed\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"seq2profile.pl ACTTTCACTTTC 1 PRDM1 \u0026gt; PRDM1_1.motif\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eseq2profile.pl ACTTTCACTTTC 1 PRDM1 \u0026gt; PRDM1_1.motif\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ecreating .motif files - most stringent: \u003cem\u003e0\u003c/em\u003e mismatches allowed\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"seq2profile.pl ACTTTCACTTTC 0 PRDM1 \u0026gt; PRDM1_0.motif\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eseq2profile.pl ACTTTCACTTTC 0 PRDM1 \u0026gt; PRDM1_0.motif\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eC) Predict sites where the TF will bind in a set of coordinates using annotatepeaks -m -mbed\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e5hmC decreases/HC unique peaks\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHIF-1b\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m HIF-1b.motif -mbed HIF-1b_HC_unique.bed -hist 10 \u0026gt; HIF_motifs_HC_unique.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m HIF-1b.motif -mbed HIF-1b_HC_unique.bed -hist 10 \u0026gt; HIF_motifs_HC_unique.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ec-Myc\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m c-Myc.motif -mbed c-Myc_HC_unique.bed \u0026gt; c-Myc_motifs_HC_unique.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m c-Myc.motif -mbed c-Myc_HC_unique.bed \u0026gt; c-Myc_motifs_HC_unique.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003en-Myc\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m n-Myc.motif -mbed n-Myc_HC_unique.bed \u0026gt; n-Myc_motifs_HC_unique.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m n-Myc.motif -mbed n-Myc_HC_unique.bed \u0026gt; n-Myc_motifs_HC_unique.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMaz\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m Maz.motif -mbed Maz_HC_unique.bed \u0026gt; Maz_motifs_HC_unique.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m Maz.motif -mbed Maz_HC_unique.bed \u0026gt; Maz_motifs_HC_unique.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eStat3\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m Stat3.motif -mbed Stat3_HC_unique.bed \u0026gt; Stat3_motifs_HC_unique.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m Stat3.motif -mbed Stat3_HC_unique.bed \u0026gt; Stat3_motifs_HC_unique.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e5hmC increases/FC unique peaks\u003c/em\u003e\nHIF-1b\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m HIF-1b.motif -mbed HIF-1b_FC_unique.bed -hist 10 \u0026gt; HIF_motifs_FC_unique.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m HIF-1b.motif -mbed HIF-1b_FC_unique.bed -hist 10 \u0026gt; HIF_motifs_FC_unique.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ec-Myc\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m c-Myc.motif -mbed c-Myc_FC_unique.bed \u0026gt; c-Myc_motifs_FC_unique.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m c-Myc.motif -mbed c-Myc_FC_unique.bed \u0026gt; c-Myc_motifs_FC_unique.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003en-Myc\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m n-Myc.motif -mbed n-Myc_FC_unique.bed \u0026gt; n-Myc_motifs_FC_unique.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m n-Myc.motif -mbed n-Myc_FC_unique.bed \u0026gt; n-Myc_motifs_FC_unique.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMaz\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m Maz.motif -mbed Maz_FC_unique.bed \u0026gt; Maz_motifs_FC_unique.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m Maz.motif -mbed Maz_FC_unique.bed \u0026gt; Maz_motifs_FC_unique.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eStat3\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m Stat3.motif -mbed Stat3.motif_FC_unique.bed \u0026gt; Stat3_motifs_FC_unique.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m Stat3.motif -mbed Stat3.motif_FC_unique.bed \u0026gt; Stat3_motifs_FC_unique.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAll mm10 annotated promoters (UCSC)\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eStat3\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_allpromoters_-2000_background.bed mm10 -size 1000 -m Stat3.motif -mbed Stat3.motif_mm10_promoters.bed \u0026gt; Stat3_motifs_mm10_promoters.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_allpromoters_-2000_background.bed mm10 -size 1000 -m Stat3.motif -mbed Stat3.motif_mm10_promoters.bed \u0026gt; Stat3_motifs_mm10_promoters.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHRE (control)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_allpromoters_-2000_background.bed mm10 -size 1000 -m HRE.motif -mbed HRE.motif_mm10_promoters.bed \u0026gt; HRE_motifs_mm10_promoters.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_allpromoters_-2000_background.bed mm10 -size 1000 -m HRE.motif -mbed HRE.motif_mm10_promoters.bed \u0026gt; HRE_motifs_mm10_promoters.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRNAseq promoters\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eMaz - Using different .motif file\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_promoters.bed mm10 -size 1000 -m Maz.motif -mbed Maz_RNAseq_promoters.bed \u0026gt; Maz_motifs_RNAseq_promoters_2.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_promoters.bed mm10 -size 1000 -m Maz.motif -mbed Maz_RNAseq_promoters.bed \u0026gt; Maz_motifs_RNAseq_promoters_2.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePRDM1\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_promoters.bed mm10 -size 2000 -m PRDM1.motif -mbed PRDM1_RNAseq_promoters.bed \u0026gt; PRDM1_motifs_RNAseq_promoters.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_promoters.bed mm10 -size 2000 -m PRDM1.motif -mbed PRDM1_RNAseq_promoters.bed \u0026gt; PRDM1_motifs_RNAseq_promoters.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRNAseq promoters - upregulated\u003c/p\u003e\n\u003cp dir=\"auto\"\u003ePRDM1\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_promoters_upregulated.bed mm10 -size 2000 -m PRDM1.motif -mbed PRDM1_RNAseq_promoters_upreg.bed \u0026gt; PRDM1_motifs_RNAseq_promoters_upreg.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_promoters_upregulated.bed mm10 -size 2000 -m PRDM1.motif -mbed PRDM1_RNAseq_promoters_upreg.bed \u0026gt; PRDM1_motifs_RNAseq_promoters_upreg.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eConcatenate all binding sites in both HC_unique and FC_unique peaks\nAlternatively, could've searched for motifs in concatenated HC anf FC peaks first\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"cat HIF-1b_HC_unique.bed HIF-1b_FC_unique.bed \u0026gt; HIF_motifs_all.bed\ncat c-Myc_HC_unique.bed c-Myc_FC_unique.bed \u0026gt; c-Myc_motifs_all.bed\ncat n-Myc_HC_unique.bed n-Myc_FC_unique.bed \u0026gt; n-Myc_motifs_all.bed\ncat Maz_HC_unique.bed Maz_FC_unique.bed \u0026gt; Maz_motifs_all.bed\ncat Stat3_motifs_HC_unique.txt Stat3_motifs_FC_unique.txt \u0026gt; Stat3_motifs_all.bed\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003ecat HIF-1b_HC_unique.bed HIF-1b_FC_unique.bed \u0026gt; HIF_motifs_all.bed\ncat c-Myc_HC_unique.bed c-Myc_FC_unique.bed \u0026gt; c-Myc_motifs_all.bed\ncat n-Myc_HC_unique.bed n-Myc_FC_unique.bed \u0026gt; n-Myc_motifs_all.bed\ncat Maz_HC_unique.bed Maz_FC_unique.bed \u0026gt; Maz_motifs_all.bed\ncat Stat3_motifs_HC_unique.txt Stat3_motifs_FC_unique.txt \u0026gt; Stat3_motifs_all.bed\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cstrong\u003eD) Find intensity of epigenetic marks at predetermined set of motif binding sites (as from #2) using annotatepeaks -hist -d to feed in tag directories. Can do this with any predetermined set of coordinates (i.e. RNAseq exon boundaries, promoters, etc)\u003c/strong\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003e5hmC distribution comparison:\u003c/em\u003e\n\u003cem\u003eHistograms\u003c/em\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHIF1b binding sites in ALL DhMRs\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/HIF_motifs_all.bed mm10 -size 1000 -m HIF-1b.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt; HIF_motifs_hist_HC_FC.txt\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/HIF_motifs_all.bed mm10 -size 1000 -m HIF-1b.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt; HIF_motifs_hist_HC_FC.txt\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ec-Myc binding sites within ALL DhMRs\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/c-Myc_motifs_all.bed mm10 -size 1000 -m c-Myc.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt; c-Myc_motifs_hist_HC_FC.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/c-Myc_motifs_all.bed mm10 -size 1000 -m c-Myc.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt; c-Myc_motifs_hist_HC_FC.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003en-Myc binding sites within ALL DhMRs\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/n-Myc_motifs_all.bed mm10 -size 1000 -m n-Myc.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt; n-Myc_motifs_hist_HC_FC.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/n-Myc_motifs_all.bed mm10 -size 1000 -m n-Myc.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt; n-Myc_motifs_hist_HC_FC.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMaz binding sites within ALL DhMRs\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/Maz_motifs_all.bed mm10 -size 1000 -m Maz.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt; Maz_motifs_hist_HC_FC.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/Maz_motifs_all.bed mm10 -size 1000 -m Maz.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt; Maz_motifs_hist_HC_FC.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eMaz binding sites within ALL RNAseq DE promoters\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/Maz_RNAseq_promoters.bed mm10 -size 10 -m Maz.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/  \u0026gt;Maz_motifs_RNAseq_promoters_hist_HC_FC_Input_10Window.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/Maz_RNAseq_promoters.bed mm10 -size 10 -m Maz.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/  \u0026gt;Maz_motifs_RNAseq_promoters_hist_HC_FC_Input_10Window.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eControl - all mm10 genes, truncated\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/ mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;trunc_allgenes.bed_hist_HC_FC.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003e/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/ mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;trunc_allgenes.bed_hist_HC_FC.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eStat3 binding sites within ALL mm10 promoters (UCSC)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/Stat3.motif_mm10_promoters.bed mm10 -size 1000 -m Stat3.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Stat3_motifs_hist_mm10proms.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/Stat3.motif_mm10_promoters.bed mm10 -size 1000 -m Stat3.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Stat3_motifs_hist_mm10proms.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHRE (HSF) binding sites within all mm10 promoters\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/HRE.motif_mm10_promoters.bed mm10 -size 1000 -m HRE.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;HRE_motifs_hist_mm10proms.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/HRE.motif_mm10_promoters.bed mm10 -size 1000 -m HRE.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;HRE_motifs_hist_mm10proms.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePRDM1 binding sites within all mm10 promoters - 2 mismatches\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1_RNAseq_promoters_upreg.bed mm10 -size 1000 -m /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ \u0026gt;PRDM1_motifs_hist_mm10proms.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1_RNAseq_promoters_upreg.bed mm10 -size 1000 -m /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ \u0026gt;PRDM1_motifs_hist_mm10proms.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003ePRDM1 binding sites within all mm10 promoters - 0 mismatches\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1_RNAseq_promoters_upreg.bed mm10 -size 1000 -m /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1_0.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ \u0026gt;PRDM1_0_motifs_hist_mm10proms.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1_RNAseq_promoters_upreg.bed mm10 -size 1000 -m /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1_0.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ \u0026gt;PRDM1_0_motifs_hist_mm10proms.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDE gene exon-boundaries (+/- 15 bp around boundary)\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_exonboundaries.bed mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ \u0026gt;RNAseq_exon_boundaries_hist_HC_FC.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_exonboundaries.bed mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ \u0026gt;RNAseq_exon_boundaries_hist_HC_FC.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eUpregulated Promoters DE promoters\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_upregulatedgenes_promoters.bed mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ \u0026gt;RNAseq_upregulated_promoters_hist_HC_FC_Input.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_upregulatedgenes_promoters.bed mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ \u0026gt;RNAseq_upregulated_promoters_hist_HC_FC_Input.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDownregulated Promoters DE promoters\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_downregulatedgenes_promoters.bed mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ \u0026gt;RNAseq_downregulated_promoters_hist_HC_FC_Input.txt \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_downregulatedgenes_promoters.bed mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ \u0026gt;RNAseq_downregulated_promoters_hist_HC_FC_Input.txt \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eAll DE gene promoters\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_promoters.bed mm10 -size 1000 -log -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_RNAseq_promoters_ComprehensiveComparison \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_promoters.bed mm10 -size 1000 -log -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_RNAseq_promoters_ComprehensiveComparison \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDE gene exons\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_exons.bed mm10 -size 1000 -fragLength 150 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/  \u0026gt;Homer_FearConditioning_5hmC_FC_RNAseq_exons_ComprehensiveComparison_2 \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_exons.bed mm10 -size 1000 -fragLength 150 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/  \u0026gt;Homer_FearConditioning_5hmC_FC_RNAseq_exons_ComprehensiveComparison_2 \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eDE gene introns\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_introns.bed mm10 -size 1000 -p 10 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_RNAseq_introns_ComprehensiveComparison_2 \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_introns.bed mm10 -size 1000 -p 10 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_RNAseq_introns_ComprehensiveComparison_2 \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCTCF binding sites\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/CTCFPeaksClustered.bed mm10 -size 1000 -fragLength 150 -annStats Homer_CTCFPeaksClustered_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_CTCFPeaksClustered_ComprehensiveComparison_2 \u0026amp; \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/CTCFPeaksClustered.bed mm10 -size 1000 -fragLength 150 -annStats Homer_CTCFPeaksClustered_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_CTCFPeaksClustered_ComprehensiveComparison_2 \u0026amp; \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003elncRNAs\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_lncRNA.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_lncRNA_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_lncRNA_ComprehensiveComparison_2 \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_lncRNA.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_lncRNA_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_lncRNA_ComprehensiveComparison_2 \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eenhancers: cell based\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_primarycell_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_primarycell_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_enhancers_primarycell_based_ComprehensiveComparison_2 \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_primarycell_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_primarycell_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_enhancers_primarycell_based_ComprehensiveComparison_2 \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eenhancers: tissue based\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_tissue_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_tissue_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_enhancers_tissue_based_ComprehensiveComparison_2 \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_tissue_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_tissue_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_enhancers_tissue_based_ComprehensiveComparison_2 \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eexon boundaries: +/- 50 bp from exons\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_exonboundaries.bed mm10 -size 1000 -fragLength 150 -annStats Homer_mm10_exonboundaries_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_mm10_exonboundaries_ComprehensiveComparison_2 \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_exonboundaries.bed mm10 -size 1000 -fragLength 150 -annStats Homer_mm10_exonboundaries_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_mm10_exonboundaries_ComprehensiveComparison_2 \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e\u003cem\u003eScatterplots\u003c/em\u003e\u003c/p\u003e\n\u003cp dir=\"auto\"\u003eHC vs HC, FC, HC_unique, FC_unique\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC.bed mm10 -size 1000 -fragLength 150 -annStats Homer_HC_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_HC_ComprehensiveComparison \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC.bed mm10 -size 1000 -fragLength 150 -annStats Homer_HC_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_HC_ComprehensiveComparison \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFC vs HC, FC, HC_unique, FC_unique\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_ComprehensiveComparison \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_ComprehensiveComparison \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eHC_unique vs HC, FC, HC_unique, FC_unique\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -fragLength 150 -annStats Homer_HC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_HC_unique_ComprehensiveComparison \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -fragLength 150 -annStats Homer_HC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_HC_unique_ComprehensiveComparison \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eFC_unique vs HC, FC, HC_unique, FC_unique\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_unique_ComprehensiveComparison \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_unique_ComprehensiveComparison \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRNAseq promoters vs\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_promoters.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_RNAseq_promoters_ComprehensiveComparison_2 \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_promoters.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_RNAseq_promoters_ComprehensiveComparison_2 \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRNAseq exons\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_exons.bed mm10 -size 1000 -fragLength 150 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/  \u0026gt;Homer_FearConditioning_5hmC_FC_RNAseq_exons_ComprehensiveComparison_2 \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_exons.bed mm10 -size 1000 -fragLength 150 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/  \u0026gt;Homer_FearConditioning_5hmC_FC_RNAseq_exons_ComprehensiveComparison_2 \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eRNAseq introns\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_introns.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_RNAseq_introns_ComprehensiveComparison_2 \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_introns.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_RNAseq_introns_ComprehensiveComparison_2 \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eCTCF binding sites\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/CTCFPeaksClustered.bed mm10 -size 1000 -fragLength 150 -annStats Homer_CTCFPeaksClustered_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_CTCFPeaksClustered_ComprehensiveComparison_2 \u0026amp; \"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/CTCFPeaksClustered.bed mm10 -size 1000 -fragLength 150 -annStats Homer_CTCFPeaksClustered_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_CTCFPeaksClustered_ComprehensiveComparison_2 \u0026amp; \n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003elncRNAs\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_lncRNA.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_lncRNA_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_lncRNA_ComprehensiveComparison_2 \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_lncRNA.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_lncRNA_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_FC_lncRNA_ComprehensiveComparison_2 \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eenhancers: cell based\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_primarycell_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_primarycell_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_enhancers_primarycell_based_ComprehensiveComparison_2 \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_primarycell_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_primarycell_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_enhancers_primarycell_based_ComprehensiveComparison_2 \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eenhancers: tissue based\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_tissue_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_tissue_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_enhancers_tissue_based_ComprehensiveComparison_2 \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_tissue_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_tissue_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_enhancers_tissue_based_ComprehensiveComparison_2 \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003eexon boundaries: +/- 50 bp from exons\u003c/p\u003e\n\u003cdiv class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_exonboundaries.bed mm10 -size 1000 -fragLength 150 -annStats Homer_mm10_exonboundaries_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_mm10_exonboundaries_ComprehensiveComparison_2 \u0026amp;\"\u003e\u003cpre class=\"notranslate\"\u003e\u003ccode\u003eannotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_exonboundaries.bed mm10 -size 1000 -fragLength 150 -annStats Homer_mm10_exonboundaries_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ \u0026gt;Homer_FearConditioning_5hmC_mm10_exonboundaries_ComprehensiveComparison_2 \u0026amp;\n\u003c/code\u003e\u003c/pre\u003e\u003c/div\u003e\n\u003cp dir=\"auto\"\u003e###11. FAST ANALYSIS: ANNOTATING REGIONS OF THE GENOME BASED ON DISEASE RELEVANT SNP CONTENT\n###12. Homology ANALYSIS:PHASTCONS\u003c/p\u003e\n\u003c/article\u003e","loaded":true,"timedOut":false,"errorMessage":null,"headerInfo":{"toc":[],"siteNavLoginPath":"/login?return_to=https%3A%2F%2Fgithub.com%2FSumeet460%2FFear-Conditioning-Genomics"}}],"overviewFilesProcessingTime":122.64320500000001,"createFromTemplatePath":"/new?template_name=Fear-Conditioning-Genomics\u0026template_owner=Sumeet460"}},"appPayload":{"helpUrl":"https://docs.github.com","findFileWorkerPath":"/assets-cdn/worker/find-file-worker-a007d7f370d6.js","findInFileWorkerPath":"/assets-cdn/worker/find-in-file-worker-d0f0ff069004.js","githubDevUrl":"https://github.dev/","enabled_features":{"code_nav_ui_events":false,"copilot_conversational_ux":false,"react_blob_overlay":false,"copilot_conversational_ux_embedding_update":false,"copilot_popover_file_editor_header":true,"copilot_smell_icebreaker_ux":true,"copilot_workspace":false,"overview_async_data_channel":true}}}}
             </script>
             <div data-target="react-partial.reactRoot">
              <style data-styled="true" data-styled-version="5.3.6">
               .cgQnMS{font-weight:600;font-size:32px;margin:0;}/*!sc*/
data-styled.g1[id="Heading__StyledHeading-sc-1c1dgg0-0"]{content:"cgQnMS,"}/*!sc*/
.izjvBm{margin-top:16px;margin-bottom:16px;}/*!sc*/
.rPQgy{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}/*!sc*/
.eUMEDg{margin-bottom:0;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;row-gap:16px;}/*!sc*/
.eLcVee{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-bottom:16px;padding-top:8px;}/*!sc*/
.hsfLlq{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;gap:8px;}/*!sc*/
@media screen and (max-width:320px){.hsfLlq{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.gpKoUz{position:relative;}/*!sc*/
@media screen and (max-width:380px){.gpKoUz .ref-selector-button-text-container{max-width:80px;}}/*!sc*/
@media screen and (max-width:320px){.gpKoUz{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}.gpKoUz .overview-ref-selector{width:100%;}.gpKoUz .overview-ref-selector > span{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;}.gpKoUz .overview-ref-selector > span > span[data-component="text"]{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}}/*!sc*/
.kkrdEu{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}/*!sc*/
.bKgizp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;}/*!sc*/
.iPGYsi{margin-right:4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.dKmYfk{font-size:14px;min-width:0;max-width:125px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}/*!sc*/
.trpoQ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;pointer-events:none;}/*!sc*/
.laYubZ{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1079px){.laYubZ{display:none;}}/*!sc*/
.swnaL{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1080px){.swnaL{display:none;}}/*!sc*/
@media screen and (max-width:543px){.swnaL{display:none;}}/*!sc*/
.bWpuBf{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;gap:8px;}/*!sc*/
.grHjNb{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
@media screen and (max-width:543px){.grHjNb{display:none;}}/*!sc*/
.dXTsqj{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (max-width:1011px){.dXTsqj{display:none;}}/*!sc*/
.dCOrmu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:1012px){.dCOrmu{display:none;}}/*!sc*/
@media screen and (max-width:544px){.bVvbgP{display:none;}}/*!sc*/
.bNDvfp{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
@media screen and (min-width:544px){.bNDvfp{display:none;}}/*!sc*/
.yfPnm{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;gap:16px;}/*!sc*/
.cAQuiW{width:100%;border-collapse:separate;border-spacing:0;border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;table-layout:fixed;overflow:unset;}/*!sc*/
.iiUlLN{height:0px;line-height:0px;}/*!sc*/
.iiUlLN tr{height:0px;font-size:0px;}/*!sc*/
.jmggSN{padding:16px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;text-align:left;height:40px;}/*!sc*/
.jmggSN th{padding-left:16px;background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));}/*!sc*/
.kvYunM{width:100%;border-top-left-radius:6px;}/*!sc*/
@media screen and (min-width:544px){.kvYunM{display:none;}}/*!sc*/
.hrLuxA{width:40%;border-top-left-radius:6px;}/*!sc*/
@media screen and (max-width:543px){.hrLuxA{display:none;}}/*!sc*/
@media screen and (max-width:543px){.ePjhhA{display:none;}}/*!sc*/
.cuEKae{text-align:right;padding-right:16px;width:136px;border-top-right-radius:6px;}/*!sc*/
.jEbBOT{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));font-size:12px;height:40px;}/*!sc*/
.bTxCvM{background-color:var(--bgColor-muted,var(--color-canvas-subtle,#f6f8fa));padding:4px;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.eYedVD{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;min-width:273px;padding-right:8px;padding-left:16px;padding-top:8px;padding-bottom:8px;}/*!sc*/
.lhFvfi{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.jGfYmh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;gap:8px;}/*!sc*/
.bqgLjk{display:inherit;}/*!sc*/
@media screen and (min-width:544px){.bqgLjk{display:none;}}/*!sc*/
@media screen and (min-width:768px){.bqgLjk{display:none;}}/*!sc*/
.epsqEd{text-align:center;vertical-align:center;height:40px;border-top:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));}/*!sc*/
.ldpruc{border-top:1px solid var(--borderColor-default,var(--color-border-default));cursor:pointer;}/*!sc*/
.ehcSsh{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;gap:16px;}/*!sc*/
.iGmlUb{border:1px solid;border-color:var(--borderColor-default,var(--color-border-default,#d0d7de));border-radius:6px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}/*!sc*/
@media screen and (max-width:543px){.iGmlUb{margin-left:-16px;margin-right:-16px;max-width:calc(100% + 32px);}}/*!sc*/
@media screen and (min-width:544px){.iGmlUb{max-width:100%;}}/*!sc*/
.iRQGXA{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid;border-bottom-color:var(--borderColor-default,var(--color-border-default,#d0d7de));-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-right:8px;position:-webkit-sticky;position:sticky;top:0;background-color:var(--bgColor-default,var(--color-canvas-default,#ffffff));z-index:1;border-top-left-radius:6px;border-top-right-radius:6px;}/*!sc*/
.dvTdPK{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;padding-left:8px;padding-right:8px;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;border-bottom:none;border-bottom-color:var(--borderColor-muted,var(--color-border-muted,hsla(210,18%,87%,1)));align:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:48px;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;max-width:100%;}/*!sc*/
.gwuIGu{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}/*!sc*/
.kOxwQs{-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;margin-right:8px;}/*!sc*/
.kOgeFj{font-weight:600;}/*!sc*/
.bJMeLZ{padding:32px;overflow:auto;}/*!sc*/
data-styled.g2[id="Box-sc-g0xbh4-0"]{content:"izjvBm,rPQgy,eUMEDg,eLcVee,hsfLlq,gpKoUz,kkrdEu,bKgizp,iPGYsi,dKmYfk,trpoQ,laYubZ,swnaL,bWpuBf,grHjNb,dXTsqj,dCOrmu,bVvbgP,bNDvfp,yfPnm,cAQuiW,iiUlLN,jmggSN,kvYunM,hrLuxA,ePjhhA,cuEKae,jEbBOT,bTxCvM,eYedVD,lhFvfi,jGfYmh,bqgLjk,epsqEd,ldpruc,ehcSsh,iGmlUb,iRQGXA,dvTdPK,gwuIGu,kOxwQs,kOgeFj,bJMeLZ,"}/*!sc*/
.bOMzPg{min-width:0;}/*!sc*/
.eUGNHp{font-weight:600;}/*!sc*/
.dALsKK{color:var(--fgColor-default,var(--color-fg-default,#1F2328));}/*!sc*/
data-styled.g6[id="Text-sc-17v1xeu-0"]{content:"bOMzPg,eUGNHp,dALsKK,"}/*!sc*/
.dheQRw{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.dheQRw:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));-webkit-text-decoration:none;text-decoration:none;position:relative;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;color:var(--fgColor-default,var(--color-fg-default,#1F2328));text-align:center;-webkit-text-decoration:none;text-decoration:none;line-height:calc(20/14);border-radius:6px;font-size:14px;padding-left:8px;padding-right:8px;padding-top:calc((2rem - 1.25rem) / 2);padding-bottom:calc((2rem - 1.25rem) / 2);}/*!sc*/
[data-a11y-link-underlines='true'] .Link__StyledLink-sc-14289xe-0[data-inline='true']{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:hover{-webkit-text-decoration:underline;text-decoration:underline;}/*!sc*/
.vLMkZ:is(button){display:inline-block;padding:0;font-size:inherit;white-space:nowrap;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;background-color:transparent;border:0;-webkit-appearance:none;-moz-appearance:none;appearance:none;}/*!sc*/
.vLMkZ span[data-component="icon"]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
@media (hover:hover){.vLMkZ:hover{background-color:var(--bgColor-neutral-muted,var(--color-neutral-muted,rgba(175,184,193,0.2)));-webkit-transition:background .12s ease-out;transition:background .12s ease-out;-webkit-text-decoration:none;text-decoration:none;}}/*!sc*/
.vLMkZ:focus{outline:2px solid transparent;}/*!sc*/
.vLMkZ:focus{box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ:focus:not(:focus-visible){box-shadow:none;}/*!sc*/
.vLMkZ:focus-visible{outline:2px solid transparent;box-shadow:inset 0 0 0 2px var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.vLMkZ span[data-content]::before{content:attr(data-content);display:block;height:0;font-weight:600;visibility:hidden;white-space:nowrap;}/*!sc*/
.vLMkZ::after{position:absolute;right:50%;bottom:calc(50% - 25px);width:100%;height:2px;content:"";background-color:var(--underlineNav-borderColor-active,var(--color-primer-border-active,#fd8c73));border-radius:0;-webkit-transform:translate(50%,-50%);-ms-transform:translate(50%,-50%);transform:translate(50%,-50%);}/*!sc*/
@media (forced-colors:active){.vLMkZ::after{background-color:LinkText;}}/*!sc*/
data-styled.g8[id="Link__StyledLink-sc-14289xe-0"]{content:"dheQRw,vLMkZ,"}/*!sc*/
.hILMMl{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.hILMMl:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.hILMMl:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.hILMMl[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.hILMMl[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.hILMMl:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.hILMMl:active{-webkit-transition:none;transition:none;}/*!sc*/
.hILMMl[data-inactive]{cursor:auto;}/*!sc*/
.hILMMl:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.hILMMl:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.hILMMl:focus{outline:solid 1px transparent;}}/*!sc*/
.hILMMl [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.hILMMl[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.hILMMl[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.hILMMl[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.hILMMl[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.hILMMl[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.hILMMl[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.hILMMl[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.hILMMl[data-block="block"]{width:100%;}/*!sc*/
.hILMMl[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.hILMMl[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.hILMMl [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.hILMMl [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.hILMMl [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.hILMMl [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.hILMMl [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.hILMMl [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.hILMMl:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.hILMMl [data-component="leadingVisual"],.hILMMl [data-component="trailingVisual"],.hILMMl [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.hILMMl[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}/*!sc*/
.hILMMl svg{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.hILMMl > span{width:inherit;}/*!sc*/
.cuOWTR{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cuOWTR:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cuOWTR:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cuOWTR[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cuOWTR[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cuOWTR:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cuOWTR:active{-webkit-transition:none;transition:none;}/*!sc*/
.cuOWTR[data-inactive]{cursor:auto;}/*!sc*/
.cuOWTR:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled [data-component=ButtonCounter],.cuOWTR:disabled [data-component="leadingVisual"],.cuOWTR:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cuOWTR:focus{outline:solid 1px transparent;}}/*!sc*/
.cuOWTR [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cuOWTR[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cuOWTR[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cuOWTR[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cuOWTR[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cuOWTR[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cuOWTR[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cuOWTR[data-block="block"]{width:100%;}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cuOWTR[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cuOWTR [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cuOWTR [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cuOWTR [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cuOWTR [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cuOWTR:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cuOWTR:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cuOWTR[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cuOWTR[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cuOWTR[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cuOWTR:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cuOWTR:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cuOWTR{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));padding-left:4px;padding-right:4px;}/*!sc*/
.cuOWTR span[data-component="leadingVisual"]{margin-right:4px !important;}/*!sc*/
.tDSzd{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.tDSzd:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.tDSzd:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.tDSzd[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.tDSzd[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.tDSzd:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.tDSzd:active{-webkit-transition:none;transition:none;}/*!sc*/
.tDSzd[data-inactive]{cursor:auto;}/*!sc*/
.tDSzd:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled [data-component=ButtonCounter],.tDSzd:disabled [data-component="leadingVisual"],.tDSzd:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.tDSzd:focus{outline:solid 1px transparent;}}/*!sc*/
.tDSzd [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.tDSzd[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.tDSzd[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.tDSzd[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.tDSzd[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.tDSzd[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.tDSzd[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.tDSzd[data-block="block"]{width:100%;}/*!sc*/
.tDSzd[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.tDSzd[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.tDSzd [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.tDSzd [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.tDSzd [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.tDSzd [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.tDSzd:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.tDSzd:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.tDSzd[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.tDSzd[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd[data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.tDSzd:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.tDSzd:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.tDSzd:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.pyyxt{border-radius:6px;border:1px solid;border-color:var(--button-default-borderColor-rest,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:var(--button-default-bgColor-rest,var(--color-btn-bg,#f6f8fa));box-shadow:var(--button-default-shadow-resting,var(--color-btn-shadow,0 1px 0 rgba(31,35,40,0.04))),var(--button-default-shadow-inset,var(--color-btn-inset-shadow,inset 0 1px 0 rgba(255,255,255,0.25)));}/*!sc*/
.pyyxt:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.pyyxt:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.pyyxt[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.pyyxt[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.pyyxt:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.pyyxt:active{-webkit-transition:none;transition:none;}/*!sc*/
.pyyxt[data-inactive]{cursor:auto;}/*!sc*/
.pyyxt:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));border-color:var(--button-default-borderColor-disabled,var(--button-default-borderColor-rest,var(--color-btn-border,rgba(31,35,40,0.15))));background-color:var(--button-default-bgColor-disabled,var(--control-bgColor-disabled,var(--color-input-disabled-bg,rgba(175,184,193,0.2))));}/*!sc*/
.pyyxt:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.pyyxt:focus{outline:solid 1px transparent;}}/*!sc*/
.pyyxt [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-default-bgColor-rest,var(--color-btn-counter-bg,rgba(31,35,40,0.08)));}/*!sc*/
.pyyxt[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.pyyxt[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.pyyxt[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.pyyxt[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.pyyxt[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.pyyxt[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.pyyxt[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.pyyxt[data-block="block"]{width:100%;}/*!sc*/
.pyyxt[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.pyyxt[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.pyyxt [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.pyyxt [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.pyyxt [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.pyyxt [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.pyyxt [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.pyyxt [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.pyyxt:hover:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-hover,var(--color-btn-hover-bg,#f3f4f6));border-color:var(--button-default-borderColor-hover,var(--button-default-borderColor-hover,var(--color-btn-hover-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt:active:not([disabled]):not([data-inactive]){background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt[aria-expanded=true]{background-color:var(--button-default-bgColor-active,var(--color-btn-active-bg,hsla(220,14%,93%,1)));border-color:var(--button-default-borderColor-active,var(--button-default-borderColor-active,var(--color-btn-active-border,rgba(31,35,40,0.15))));}/*!sc*/
.pyyxt [data-component="leadingVisual"],.pyyxt [data-component="trailingVisual"],.pyyxt [data-component="trailingAction"]{color:var(--button-color,var(--fgColor-muted,var(--color-fg-muted,#656d76)));}/*!sc*/
.pyyxt[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.gYvpXq{border-radius:6px;border:1px solid;border-color:var(--button-primary-borderColor-rest,var(--color-btn-primary-border,rgba(31,35,40,0.15)));font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));background-color:var(--button-primary-bgColor-rest,var(--color-btn-primary-bg,#1f883d));box-shadow:var(--shadow-resting-small,var(--color-btn-primary-shadow,0 1px 0 rgba(31,35,40,0.1)));}/*!sc*/
.gYvpXq:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.gYvpXq:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.gYvpXq[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.gYvpXq[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.gYvpXq:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.gYvpXq:active{-webkit-transition:none;transition:none;}/*!sc*/
.gYvpXq[data-inactive]{cursor:auto;}/*!sc*/
.gYvpXq:disabled{cursor:not-allowed;box-shadow:none;color:var(--button-primary-fgColor-disabled,var(--color-btn-primary-disabled-text,rgba(255,255,255,0.8)));background-color:var(--button-primary-bgColor-disabled,var(--color-btn-primary-disabled-bg,#94d3a2));border-color:var(--button-primary-borderColor-disabled,var(--color-btn-primary-disabled-border,rgba(31,35,40,0.15)));}/*!sc*/
.gYvpXq:disabled [data-component=ButtonCounter]{color:inherit;}/*!sc*/
@media (forced-colors:active){.gYvpXq:focus{outline:solid 1px transparent;}}/*!sc*/
.gYvpXq [data-component=ButtonCounter]{font-size:12px;background-color:var(--buttonCounter-primary-bgColor-rest,var(--color-btn-primary-counter-bg,rgba(0,45,17,0.2)));color:var(--button-primary-fgColor-rest,var(--color-btn-primary-text,#ffffff));}/*!sc*/
.gYvpXq[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.gYvpXq[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.gYvpXq[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.gYvpXq[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.gYvpXq[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.gYvpXq[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.gYvpXq[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.gYvpXq[data-block="block"]{width:100%;}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.gYvpXq[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.gYvpXq [data-component="leadingVisual"]{grid-area:leadingVisual;}/*!sc*/
.gYvpXq [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.gYvpXq [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.gYvpXq [data-component="trailingAction"]{margin-right:-4px;}/*!sc*/
.gYvpXq [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.gYvpXq [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.gYvpXq:hover:not([disabled]):not([data-inactive]){color:btn.primary.hoverText;background-color:var(--button-primary-bgColor-hover,var(--color-btn-primary-hover-bg,#1a7f37));}/*!sc*/
.gYvpXq:focus:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:focus-visible:not([disabled]){box-shadow:inset 0 0 0 3px;}/*!sc*/
.gYvpXq:active:not([disabled]):not([data-inactive]){background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq[aria-expanded=true]{background-color:var(--button-primary-bgColor-active,var(--color-btn-primary-selected-bg,hsla(137,66%,28%,1)));box-shadow:var(--button-primary-shadow-selected,var(--color-btn-primary-selected-shadow,inset 0 1px 0 rgba(0,45,17,0.2)));}/*!sc*/
.gYvpXq svg{color:fg.primary;}/*!sc*/
.fAkXQN{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--fgColor-default,var(--color-fg-default,#1F2328));background-color:transparent;box-shadow:none;}/*!sc*/
.fAkXQN:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.fAkXQN:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.fAkXQN[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.fAkXQN[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.fAkXQN:active{-webkit-transition:none;transition:none;}/*!sc*/
.fAkXQN[data-inactive]{cursor:auto;}/*!sc*/
.fAkXQN:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled [data-component=ButtonCounter],.fAkXQN:disabled [data-component="leadingVisual"],.fAkXQN:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.fAkXQN:focus{outline:solid 1px transparent;}}/*!sc*/
.fAkXQN [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.fAkXQN[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.fAkXQN[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.fAkXQN[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.fAkXQN[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.fAkXQN[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.fAkXQN[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.fAkXQN[data-block="block"]{width:100%;}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.fAkXQN[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.fAkXQN [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.fAkXQN [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.fAkXQN [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.fAkXQN [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.fAkXQN:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.fAkXQN[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.fAkXQN[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.fAkXQN[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.fAkXQN:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.fAkXQN:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.fAkXQN:focus:not([disabled]){-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw{border-radius:6px;border:1px solid;border-color:transparent;font-family:inherit;font-weight:500;font-size:14px;cursor:pointer;-webkit-appearance:none;-moz-appearance:none;appearance:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-text-decoration:none;text-decoration:none;text-align:center;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;height:32px;padding:0 12px;gap:8px;min-width:-webkit-max-content;min-width:-moz-max-content;min-width:max-content;-webkit-transition:80ms cubic-bezier(0.65,0,0.35,1);transition:80ms cubic-bezier(0.65,0,0.35,1);-webkit-transition-property:color,fill,background-color,border-color;transition-property:color,fill,background-color,border-color;color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));background-color:transparent;box-shadow:none;}/*!sc*/
.cLeluw:focus:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw:focus:not(:disabled):not(:focus-visible){outline:solid 1px transparent;}/*!sc*/
.cLeluw:focus-visible:not(:disabled){box-shadow:none;outline:2px solid var(--fgColor-accent,var(--color-accent-fg,#0969da));outline-offset:-2px;}/*!sc*/
.cLeluw[href]{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}/*!sc*/
.cLeluw[href]:hover{-webkit-text-decoration:none;text-decoration:none;}/*!sc*/
.cLeluw:hover{-webkit-transition-duration:80ms;transition-duration:80ms;}/*!sc*/
.cLeluw:active{-webkit-transition:none;transition:none;}/*!sc*/
.cLeluw[data-inactive]{cursor:auto;}/*!sc*/
.cLeluw:disabled{cursor:not-allowed;box-shadow:none;color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled [data-component=ButtonCounter],.cLeluw:disabled [data-component="leadingVisual"],.cLeluw:disabled [data-component="trailingAction"]{color:inherit;}/*!sc*/
@media (forced-colors:active){.cLeluw:focus{outline:solid 1px transparent;}}/*!sc*/
.cLeluw [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-component=IconButton]{display:inline-grid;padding:unset;place-content:center;width:32px;min-width:unset;}/*!sc*/
.cLeluw[data-size="small"]{padding:0 8px;height:28px;gap:4px;font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="text"]{line-height:calc(20 / 12);}/*!sc*/
.cLeluw[data-size="small"] [data-component=ButtonCounter]{font-size:12px;}/*!sc*/
.cLeluw[data-size="small"] [data-component="buttonContent"] > :not(:last-child){margin-right:4px;}/*!sc*/
.cLeluw[data-size="small"][data-component=IconButton]{width:28px;padding:unset;}/*!sc*/
.cLeluw[data-size="large"]{padding:0 16px;height:40px;gap:8px;}/*!sc*/
.cLeluw[data-size="large"] [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw[data-size="large"][data-component=IconButton]{width:40px;padding:unset;}/*!sc*/
.cLeluw[data-block="block"]{width:100%;}/*!sc*/
.cLeluw[data-inactive]:not([disabled]){background-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));border-color:var(--button-inactive-bgColor,var(--button-inactive-bgColor-rest,var(--color-btn-inactive-bg,#eaeef2)));color:var(--button-inactive-fgColor,var(--button-inactive-fgColor-rest,var(--color-btn-inactive-text,#57606a)));}/*!sc*/
.cLeluw[data-inactive]:not([disabled]):focus-visible{box-shadow:none;}/*!sc*/
.cLeluw [data-component="leadingVisual"]{grid-area:leadingVisual;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="text"]{grid-area:text;line-height:calc(20/14);white-space:nowrap;}/*!sc*/
.cLeluw [data-component="trailingVisual"]{grid-area:trailingVisual;}/*!sc*/
.cLeluw [data-component="trailingAction"]{margin-right:-4px;color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw [data-component="buttonContent"]{-webkit-flex:1 0 auto;-ms-flex:1 0 auto;flex:1 0 auto;display:grid;grid-template-areas:"leadingVisual text trailingVisual";grid-template-columns:min-content minmax(0,auto) min-content;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-align-content:center;-ms-flex-line-pack:center;align-content:center;}/*!sc*/
.cLeluw [data-component="buttonContent"] > :not(:last-child){margin-right:8px;}/*!sc*/
.cLeluw:hover:not([disabled]){background-color:var(--control-transparent-bgColor-hover,var(--color-action-list-item-default-hover-bg,rgba(208,215,222,0.32)));}/*!sc*/
.cLeluw:active:not([disabled]){background-color:var(--control-transparent-bgColor-active,var(--color-action-list-item-default-active-bg,rgba(208,215,222,0.48)));}/*!sc*/
.cLeluw[aria-expanded=true]{background-color:var(--control-transparent-bgColor-selected,var(--color-action-list-item-default-selected-bg,rgba(208,215,222,0.24)));}/*!sc*/
.cLeluw[data-component="IconButton"][data-no-visuals]{color:var(--fgColor-muted,var(--color-fg-muted,#656d76));}/*!sc*/
.cLeluw[data-no-visuals]{color:var(--fgColor-accent,var(--color-accent-fg,#0969da));}/*!sc*/
.cLeluw:has([data-component="ButtonCounter"]){color:var(--button-default-fgColor-rest,var(--color-btn-text,#24292f));}/*!sc*/
.cLeluw:disabled[data-no-visuals]{color:var(--fgColor-disabled,var(--color-primer-fg-disabled,#8c959f));}/*!sc*/
.cLeluw:disabled[data-no-visuals] [data-component=ButtonCounter]{color:inherit;}/*!sc*/
.cLeluw[data-size="small"][data-no-visuals]{margin-right:8px;height:28px;}/*!sc*/
data-styled.g9[id="types__StyledButton-sc-ws60qy-0"]{content:"hILMMl,cuOWTR,tDSzd,pyyxt,gYvpXq,fAkXQN,cLeluw,"}/*!sc*/
.rTZSs{position:absolute;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;-webkit-clip:rect(0,0,0,0);clip:rect(0,0,0,0);white-space:nowrap;border-width:0;}/*!sc*/
data-styled.g10[id="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0"]{content:"rTZSs,"}/*!sc*/
.fUpWeN{display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:top;white-space:nowrap;max-width:125px;max-width:100%;}/*!sc*/
data-styled.g15[id="Truncate__StyledTruncate-sc-23o1d2-0"]{content:"fUpWeN,"}/*!sc*/
.gNgnVl{position:relative;display:inline-block;}/*!sc*/
.gNgnVl::after{position:absolute;z-index:1000000;display:none;padding:0.5em 0.75em;font:normal normal 11px/1.5 -apple-system,BlinkMacSystemFont,"Segoe UI","Noto Sans",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji";-webkit-font-smoothing:subpixel-antialiased;color:var(--fgColor-onEmphasis,var(--color-fg-on-emphasis,#ffffff));text-align:center;-webkit-text-decoration:none;text-decoration:none;text-shadow:none;text-transform:none;-webkit-letter-spacing:normal;-moz-letter-spacing:normal;-ms-letter-spacing:normal;letter-spacing:normal;word-wrap:break-word;white-space:pre;pointer-events:none;content:attr(aria-label);background:var(--bgColor-emphasis,var(--color-neutral-emphasis-plus,#24292f));border-radius:6px;opacity:0;}/*!sc*/
@-webkit-keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
@keyframes tooltip-appear{from{opacity:0;}to{opacity:1;}}/*!sc*/
.gNgnVl:hover::after,.gNgnVl:active::after,.gNgnVl:focus::after,.gNgnVl:focus-within::after{display:inline-block;-webkit-text-decoration:none;text-decoration:none;-webkit-animation-name:tooltip-appear;animation-name:tooltip-appear;-webkit-animation-duration:0.1s;animation-duration:0.1s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-in;animation-timing-function:ease-in;-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-no-delay:hover::after,.gNgnVl.tooltipped-no-delay:active::after,.gNgnVl.tooltipped-no-delay:focus::after,.gNgnVl.tooltipped-no-delay:focus-within::after{-webkit-animation-delay:0s;animation-delay:0s;}/*!sc*/
.gNgnVl.tooltipped-multiline:hover::after,.gNgnVl.tooltipped-multiline:active::after,.gNgnVl.tooltipped-multiline:focus::after,.gNgnVl.tooltipped-multiline:focus-within::after{display:table-cell;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-se::after,.gNgnVl.tooltipped-sw::after{top:100%;right:50%;margin-top:6px;}/*!sc*/
.gNgnVl.tooltipped-se::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-sw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-n::after,.gNgnVl.tooltipped-ne::after,.gNgnVl.tooltipped-nw::after{right:50%;bottom:100%;margin-bottom:6px;}/*!sc*/
.gNgnVl.tooltipped-ne::after{right:auto;left:50%;margin-left:-16px;}/*!sc*/
.gNgnVl.tooltipped-nw::after{margin-right:-16px;}/*!sc*/
.gNgnVl.tooltipped-s::after,.gNgnVl.tooltipped-n::after{-webkit-transform:translateX(50%);-ms-transform:translateX(50%);transform:translateX(50%);}/*!sc*/
.gNgnVl.tooltipped-w::after{right:100%;bottom:50%;margin-right:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-e::after{bottom:50%;left:100%;margin-left:6px;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%);}/*!sc*/
.gNgnVl.tooltipped-multiline::after{width:-webkit-max-content;width:-moz-max-content;width:max-content;max-width:250px;word-wrap:break-word;white-space:pre-line;border-collapse:separate;}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-s::after,.gNgnVl.tooltipped-multiline.tooltipped-n::after{right:auto;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}/*!sc*/
.gNgnVl.tooltipped-multiline.tooltipped-w::after,.gNgnVl.tooltipped-multiline.tooltipped-e::after{right:100%;}/*!sc*/
.gNgnVl.tooltipped-align-right-2::after{right:0;margin-right:0;}/*!sc*/
.gNgnVl.tooltipped-align-left-2::after{left:0;margin-left:0;}/*!sc*/
data-styled.g18[id="Tooltip__TooltipBase-sc-17tf59c-0"]{content:"gNgnVl,"}/*!sc*/
.bPgibo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;list-style:none;white-space:nowrap;padding-top:0;padding-bottom:0;padding-left:0;padding-right:0;margin:0;margin-bottom:-1px;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;gap:8px;position:relative;}/*!sc*/
data-styled.g103[id="UnderlineNav__NavigationList-sc-1jfr31k-0"]{content:"bPgibo,"}/*!sc*/
              </style>
              <!-- -->
              <!-- -->
              <div class="Box-sc-g0xbh4-0 izjvBm">
               <div class="Box-sc-g0xbh4-0 rPQgy">
                <div class="Box-sc-g0xbh4-0 eUMEDg">
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 eLcVee">
                <div class="Box-sc-g0xbh4-0 hsfLlq">
                 <div class="Box-sc-g0xbh4-0 gpKoUz">
                  <button aria-haspopup="true" aria-label="master branch" class="types__StyledButton-sc-ws60qy-0 hILMMl overview-ref-selector" data-testid="anchor-button" id="branch-picker-repos-header-ref-selector" tabindex="0" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span data-component="text">
                     <div class="Box-sc-g0xbh4-0 bKgizp">
                      <div class="Box-sc-g0xbh4-0 iPGYsi">
                       <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                        <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                        </path>
                       </svg>
                      </div>
                      <div class="Box-sc-g0xbh4-0 dKmYfk ref-selector-button-text-container">
                       <span class="Text-sc-17v1xeu-0 bOMzPg">
                        <!-- -->
                        master
                       </span>
                      </div>
                     </div>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingVisual">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </span>
                  </button>
                  <button data-hotkey-scope="read-only-cursor-text-area" hidden="">
                  </button>
                 </div>
                 <div class="Box-sc-g0xbh4-0 laYubZ">
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/Sumeet460/Fear-Conditioning-Genomics/branches" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Branches
                    </span>
                   </span>
                  </a>
                  <a class="types__StyledButton-sc-ws60qy-0 cuOWTR" href="/Sumeet460/Fear-Conditioning-Genomics/tags" style="--button-color:fg.muted" type="button">
                   <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                     <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                      </path>
                     </svg>
                    </span>
                    <span data-component="text">
                     Tags
                    </span>
                   </span>
                  </a>
                 </div>
                 <div class="Box-sc-g0xbh4-0 swnaL">
                  <a aria-label="Go to Branches page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/Sumeet460/Fear-Conditioning-Genomics/branches" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-git-branch" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M9.5 3.25a2.25 2.25 0 1 1 3 2.122V6A2.5 2.5 0 0 1 10 8.5H6a1 1 0 0 0-1 1v1.128a2.251 2.251 0 1 1-1.5 0V5.372a2.25 2.25 0 1 1 1.5 0v1.836A2.493 2.493 0 0 1 6 7h4a1 1 0 0 0 1-1v-.628A2.25 2.25 0 0 1 9.5 3.25Zm-6 0a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Zm8.25-.75a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5ZM4.25 12a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Z">
                    </path>
                   </svg>
                  </a>
                  <a aria-label="Go to Tags page" class="types__StyledButton-sc-ws60qy-0 tDSzd" data-no-visuals="true" href="/Sumeet460/Fear-Conditioning-Genomics/tags" style="--button-color:fg.muted" type="button">
                   <svg aria-hidden="true" class="octicon octicon-tag" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M1 7.775V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.752 1.752 0 0 1 1 7.775Zm1.5 0c0 .066.026.13.073.177l6.25 6.25a.25.25 0 0 0 .354 0l5.025-5.025a.25.25 0 0 0 0-.354l-6.25-6.25a.25.25 0 0 0-.177-.073H2.75a.25.25 0 0 0-.25.25ZM6 5a1 1 0 1 1 0 2 1 1 0 0 1 0-2Z">
                    </path>
                   </svg>
                  </a>
                 </div>
                </div>
                <div class="Box-sc-g0xbh4-0 bWpuBf">
                 <div class="Box-sc-g0xbh4-0 grHjNb">
                  <button hidden="">
                  </button>
                  <div class="Box-sc-g0xbh4-0 dXTsqj">
                   <!--$!-->
                   <template>
                   </template>
                   <!--/$-->
                  </div>
                  <div class="Box-sc-g0xbh4-0 dCOrmu">
                   <button class="types__StyledButton-sc-ws60qy-0 pyyxt" data-no-visuals="true" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Go to file
                     </span>
                    </span>
                   </button>
                  </div>
                  <div class="react-directory-add-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" aria-label="Add file" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R2hil5:" tabindex="0" type="button">
                    <svg aria-hidden="true" class="octicon octicon-plus" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M7.75 2a.75.75 0 0 1 .75.75V7h4.25a.75.75 0 0 1 0 1.5H8.5v4.25a.75.75 0 0 1-1.5 0V8.5H2.75a.75.75 0 0 1 0-1.5H7V2.75A.75.75 0 0 1 7.75 2Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="react-directory-remove-file-icon">
                   <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                    Add file
                   </h2>
                   <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 pyyxt" id=":R2lil5:" tabindex="0" type="button">
                    <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                     <span data-component="text">
                      Add file
                     </span>
                    </span>
                    <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                     <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                      </path>
                     </svg>
                    </span>
                   </button>
                  </div>
                 </div>
                 <button aria-haspopup="true" class="types__StyledButton-sc-ws60qy-0 gYvpXq" id=":R2il5:" tabindex="0" type="button">
                  <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                   <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                    <div class="Box-sc-g0xbh4-0 bVvbgP">
                     <svg aria-hidden="true" class="octicon octicon-code" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                      <path d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z">
                      </path>
                     </svg>
                    </div>
                   </span>
                   <span data-component="text">
                    Code
                   </span>
                  </span>
                  <span class="Box-sc-g0xbh4-0 trpoQ" data-component="trailingAction">
                   <svg aria-hidden="true" class="octicon octicon-triangle-down" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="m4.427 7.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.396 7H4.604a.25.25 0 0 0-.177.427Z">
                    </path>
                   </svg>
                  </span>
                 </button>
                 <div class="Box-sc-g0xbh4-0 bNDvfp">
                  <button aria-haspopup="true" aria-label="Open more actions menu" class="types__StyledButton-sc-ws60qy-0 pyyxt" data-component="IconButton" data-no-visuals="true" id=":R3il5:" tabindex="0" type="button">
                   <svg aria-hidden="true" class="octicon octicon-kebab-horizontal" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                    <path d="M8 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3ZM1.5 9a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Zm13 0a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z">
                    </path>
                   </svg>
                  </button>
                 </div>
                </div>
               </div>
               <div class="Box-sc-g0xbh4-0 yfPnm">
                <div class="Box-sc-g0xbh4-0" data-hpc="true">
                 <button data-hotkey="j" data-testid="focus-next-element-button" hidden="">
                 </button>
                 <button data-hotkey="k" data-testid="focus-previous-element-button" hidden="">
                 </button>
                 <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading" id="folders-and-files">
                  Folders and files
                 </h2>
                 <table aria-labelledby="folders-and-files" class="Box-sc-g0xbh4-0 cAQuiW">
                  <thead class="Box-sc-g0xbh4-0 iiUlLN">
                   <tr class="Box-sc-g0xbh4-0 jmggSN">
                    <th class="Box-sc-g0xbh4-0 kvYunM" colspan="2">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 hrLuxA" colspan="1">
                     <span class="Text-sc-17v1xeu-0 eUGNHp">
                      Name
                     </span>
                    </th>
                    <th class="Box-sc-g0xbh4-0 ePjhhA">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit message">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit message
                      </span>
                     </div>
                    </th>
                    <th class="Box-sc-g0xbh4-0 cuEKae" colspan="1">
                     <div class="Truncate__StyledTruncate-sc-23o1d2-0 fUpWeN" title="Last commit date">
                      <span class="Text-sc-17v1xeu-0 eUGNHp">
                       Last commit date
                      </span>
                     </div>
                    </th>
                   </tr>
                  </thead>
                  <tbody>
                   <tr class="Box-sc-g0xbh4-0 jEbBOT">
                    <td class="Box-sc-g0xbh4-0 bTxCvM" colspan="3">
                     <div class="Box-sc-g0xbh4-0 eYedVD">
                      <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                       Latest commit
                      </h2>
                      <div class="Skeleton Skeleton--text" data-testid="loading" style="width:120px">
                      </div>
                      <div class="Box-sc-g0xbh4-0 lhFvfi" data-testid="latest-commit-details">
                      </div>
                      <div class="Box-sc-g0xbh4-0 jGfYmh">
                       <h2 class="Heading__StyledHeading-sc-1c1dgg0-0 cgQnMS sr-only" data-testid="screen-reader-heading">
                        History
                       </h2>
                       <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-group" data-size="small" href="/Sumeet460/Fear-Conditioning-Genomics/commits/master/">
                        <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                         <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                          <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                           <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                           </path>
                          </svg>
                         </span>
                         <span data-component="text">
                          <span class="Text-sc-17v1xeu-0 dALsKK">
                           39 Commits
                          </span>
                         </span>
                        </span>
                       </a>
                       <div class="Box-sc-g0xbh4-0 bqgLjk">
                       </div>
                       <span aria-label="Commit history" class="Tooltip__TooltipBase-sc-17tf59c-0 gNgnVl tooltipped-n" role="tooltip">
                        <a class="types__StyledButton-sc-ws60qy-0 fAkXQN react-last-commit-history-icon" href="/Sumeet460/Fear-Conditioning-Genomics/commits/master/">
                         <span class="Box-sc-g0xbh4-0 kkrdEu" data-component="buttonContent">
                          <span class="Box-sc-g0xbh4-0 trpoQ" data-component="leadingVisual">
                           <svg aria-hidden="true" class="octicon octicon-history" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                            <path d="m.427 1.927 1.215 1.215a8.002 8.002 0 1 1-1.6 5.685.75.75 0 1 1 1.493-.154 6.5 6.5 0 1 0 1.18-4.458l1.358 1.358A.25.25 0 0 1 3.896 6H.25A.25.25 0 0 1 0 5.75V2.104a.25.25 0 0 1 .427-.177ZM7.75 4a.75.75 0 0 1 .75.75v2.992l2.028.812a.75.75 0 0 1-.557 1.392l-2.5-1A.751.751 0 0 1 7 8.25v-3.5A.75.75 0 0 1 7.75 4Z">
                            </path>
                           </svg>
                          </span>
                         </span>
                        </a>
                       </span>
                      </div>
                     </div>
                    </td>
                   </tr>
                   <tr class="react-directory-row undefined" id="folder-row-0">
                    <td class="react-directory-row-name-cell-small-screen" colspan="2">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/Sumeet460/Fear-Conditioning-Genomics/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-name-cell-large-screen" colspan="1">
                     <div class="react-directory-filename-column">
                      <svg aria-hidden="true" class="color-fg-muted" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                       <path d="M2 1.75C2 .784 2.784 0 3.75 0h6.586c.464 0 .909.184 1.237.513l2.914 2.914c.329.328.513.773.513 1.237v9.586A1.75 1.75 0 0 1 13.25 16h-9.5A1.75 1.75 0 0 1 2 14.25Zm1.75-.25a.25.25 0 0 0-.25.25v12.5c0 .138.112.25.25.25h9.5a.25.25 0 0 0 .25-.25V6h-2.75A1.75 1.75 0 0 1 9 4.25V1.5Zm6.75.062V4.25c0 .138.112.25.25.25h2.688l-.011-.013-2.914-2.914-.013-.011Z">
                       </path>
                      </svg>
                      <div class="overflow-hidden">
                       <h3>
                        <div class="react-directory-truncate">
                         <a aria-label="README.md, (File)" class="Link--primary" href="/Sumeet460/Fear-Conditioning-Genomics/blob/master/README.md" title="README.md">
                          README.md
                         </a>
                        </div>
                       </h3>
                      </div>
                     </div>
                    </td>
                    <td class="react-directory-row-commit-cell">
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                    <td>
                     <div class="Skeleton Skeleton--text">
                     </div>
                    </td>
                   </tr>
                   <tr class="Box-sc-g0xbh4-0 epsqEd d-none" data-testid="view-all-files-row">
                    <td class="Box-sc-g0xbh4-0 ldpruc" colspan="3">
                     <div>
                      <button class="Link__StyledLink-sc-14289xe-0 dheQRw">
                       View all files
                      </button>
                     </div>
                    </td>
                   </tr>
                  </tbody>
                 </table>
                </div>
                <div class="Box-sc-g0xbh4-0 ehcSsh">
                 <div class="Box-sc-g0xbh4-0 iGmlUb">
                  <div class="Box-sc-g0xbh4-0 iRQGXA">
                   <h2 class="_VisuallyHidden__VisuallyHidden-sc-11jhm7a-0 rTZSs">
                    Repository files navigation
                   </h2>
                   <nav aria-label="Repository files" class="Box-sc-g0xbh4-0 dvTdPK">
                    <ul class="UnderlineNav__NavigationList-sc-1jfr31k-0 bPgibo" role="list">
                     <li class="Box-sc-g0xbh4-0 gwuIGu">
                      <a aria-current="page" class="Link__StyledLink-sc-14289xe-0 vLMkZ" href="#">
                       <span class="Box-sc-g0xbh4-0 kOxwQs" data-component="icon">
                        <svg aria-hidden="true" class="octicon octicon-book" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                         <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                         </path>
                        </svg>
                       </span>
                       <span class="Box-sc-g0xbh4-0 kOgeFj" data-component="text" data-content="README">
                        README
                       </span>
                      </a>
                     </li>
                    </ul>
                   </nav>
                   <button aria-label="Edit file" class="types__StyledButton-sc-ws60qy-0 cLeluw" data-component="IconButton" data-no-visuals="true" data-size="small" title="Edit file" type="button">
                    <svg aria-hidden="true" class="octicon octicon-pencil" fill="currentColor" focusable="false" height="16" role="img" style="display:inline-block;user-select:none;vertical-align:text-bottom;overflow:visible" viewbox="0 0 16 16" width="16">
                     <path d="M11.013 1.427a1.75 1.75 0 0 1 2.474 0l1.086 1.086a1.75 1.75 0 0 1 0 2.474l-8.61 8.61c-.21.21-.47.364-.756.445l-3.251.93a.75.75 0 0 1-.927-.928l.929-3.25c.081-.286.235-.547.445-.758l8.61-8.61Zm.176 4.823L9.75 4.81l-6.286 6.287a.253.253 0 0 0-.064.108l-.558 1.953 1.953-.558a.253.253 0 0 0 .108-.064Zm1.238-3.763a.25.25 0 0 0-.354 0L10.811 3.75l1.439 1.44 1.263-1.263a.25.25 0 0 0 0-.354Z">
                     </path>
                    </svg>
                   </button>
                  </div>
                  <div class="Box-sc-g0xbh4-0 bJMeLZ js-snippet-clipboard-copy-unpositioned" data-hpc="true">
                   <article class="markdown-body entry-content container-lg" itemprop="text">
                    <p dir="auto">
                     Fear-Conditioning-Genomics
                    </p>
                    <p dir="auto">
                     TABLE OF CONTENTS
                    </p>
                    <ol dir="auto">
                     <li>
                      ALIGNMENT: STAR
                     </li>
                     <li>
                      RNASEQ ANALYSIS, TRAPSEQ:DIFFERENTIAL EXPRESSION:CUFFDIFF
                     </li>
                     <li>
                      RNASEQ ANALYSIS, TRAPSEQ:DIFFERENTIAL EXPRESSION:DESEQ2
                     </li>
                     <li>
                      RNASEQ ANALYSIS, TRAPSEQ:DIFFERENTIAL SPLICING:DESEQ2/DEXSEQ
                     </li>
                     <li>
                      RNASEQ ANALYSIS, TRAPSEQ:PATHWAT ANALYSIS:CONSENSUS PATH DATABASE
                     </li>
                     <li>
                      5HMC ANALYSIS:PEAK CALLING:MACS
                     </li>
                     <li>
                      PARCE OUT REGIONS OF 5HMC, EXPRESSION, AND ISOFORM REGULATION:BEDTOOLS
                     </li>
                     <li>
                      ANNOTATION ANALYSIS:HOMER
                     </li>
                     <li>
                      MOTIF ANALYSIS:HOMER:MOTIF ID AND LOCALIZATION
                     </li>
                     <li>
                      5HMC ENRICHMENT ANALYSIS:HOMER:HISTOGRAMS AND SCATTERPLOTS: MOTIFS, PROMOTERS, EXONS, INTRONS, EXONS-INTRONS, CTCF, OTHER GENOMIC REGIONS (CPG ISLANDS, EXONS, LNCRNA, PSEUDOGENES, LOW COMPLEXITY REGIONS, SIMPLE REPEATS, SINES, SATELLITES
5HMC AND ALTERNATIVE SPLICING
                     </li>
                     <li>
                      FAST ANALYSIS: ANNOTATING REGIONS OF THE GENOME BASED ON DISEASE RELEVANT SNP CONTENT
                     </li>
                     <li>
                      Homology ANALYSIS:PHASTCONS
                     </li>
                     <li>
                      GWAS &amp; GxE:IMPUTE2
                     </li>
                    </ol>
                    <p dir="auto">
                     ###1. ALIGNMENT: STAR
                    </p>
                    <p dir="auto">
                     <strong>
                      Base Files
                     </strong>
                     <em>
                      WholeGenomeFasta Files:
                     </em>
                    </p>
                    <ul dir="auto">
                     <li>
                      ensembl
                      <ul dir="auto">
                       <li>
                        Mus_musculus.GRCm38.dna.toplevel.fa
                       </li>
                      </ul>
                     </li>
                     <li>
                      UCSC
                      <ul dir="auto">
                       <li>
                        mm10.fa.align
                       </li>
                       <li>
                        mm9 -&gt; genome.fa
                       </li>
                      </ul>
                     </li>
                    </ul>
                    <p dir="auto">
                     <em>
                      GTF Files:
                     </em>
                    </p>
                    <ul dir="auto">
                     <li>
                      ensembl
                      <ul dir="auto">
                       <li>
                        Mus_musculus.GRCm38.79.gtf (genes)
                       </li>
                      </ul>
                     </li>
                     <li>
                      UCSC
                      <ul dir="auto">
                       <li>
                        /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked.gtf (repeat masked annotations only downloaded from UCSC)
                       </li>
                       <li>
                        /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/genes.gtf
                       </li>
                       <li>
                        /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gtf (Repeat masked annotation concatenated with the normal annotation)
                       </li>
                      </ul>
                     </li>
                    </ul>
                    <p dir="auto">
                     <em>
                      Indices:
                     </em>
                    </p>
                    <ul dir="auto">
                     <li>
                      /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex (this one was created concatenating the normal UCSC mm10 gtf w/ the RepeatMasked GTF)
                     </li>
                    </ul>
                    <p dir="auto">
                     Generate STAR Genome Index using repeat masked gtf file from UCSC, cat with a normal gtf file  - in order to capture the repetive elements in RepBase in the index
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runMode genomeGenerate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --genomeFastaFiles /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Sequence/WholeGenomeFasta/genome.fa --runThreadN 25 --sjdbGTFfile /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gtf --sjdbOverhang 100 --limitSjdbInsertNsj 5000000">
                     <pre class="notranslate"><code>/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runMode genomeGenerate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --genomeFastaFiles /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Sequence/WholeGenomeFasta/genome.fa --runThreadN 25 --sjdbGTFfile /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gtf --sjdbOverhang 100 --limitSjdbInsertNsj 5000000
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      Alignment Runs: UCSC
                     </strong>
                     Automated - didn't implement this, but would've save a lot of time...
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="for f in `cat files`; do STAR --genomeDir ../STAR/ENSEMBL.homo_sapiens.release-75 \
--readFilesIn fastq/$f\_1.fastq fastq/$f\_2.fastq \
--runThreadN 12 --outFileNamePrefix aligned/$f.; done">
                     <pre class="notranslate"><code>for f in `cat files`; do STAR --genomeDir ../STAR/ENSEMBL.homo_sapiens.release-75 \
--readFilesIn fastq/$f\_1.fastq fastq/$f\_2.fastq \
--runThreadN 12 --outFileNamePrefix aligned/$f.; done
</code></pre>
                    </div>
                    <p dir="auto">
                     19_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 19_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/19_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/19_Homecage_ACAGTG_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/19_Homecage/19_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts ">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 19_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/19_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/19_Homecage_ACAGTG_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/19_Homecage/19_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts 
</code></pre>
                    </div>
                    <p dir="auto">
                     20_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/
mkdir 20_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/20_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/20_Homecage_GCCAAT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/20_Homecage/20_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts ">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/
mkdir 20_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/20_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/20_Homecage_GCCAAT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/20_Homecage/20_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts 
</code></pre>
                    </div>
                    <p dir="auto">
                     21_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 21_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/21_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/21_Homecage_CAGATC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/21_Homecage/21_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts ">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 21_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/21_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/21_Homecage_CAGATC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/21_Homecage/21_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts 
</code></pre>
                    </div>
                    <p dir="auto">
                     22_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 22_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/22_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/22_Homecage_CTTGTA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/22_Homecage/22_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts ">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 22_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/22_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/22_Homecage_CTTGTA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/22_Homecage/22_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts 
</code></pre>
                    </div>
                    <p dir="auto">
                     23_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 23_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/23_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/23_Homecage_AGTCAA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/23_Homecage/23_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts ">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 23_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/23_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/23_Homecage_AGTCAA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/23_Homecage/23_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts 
</code></pre>
                    </div>
                    <p dir="auto">
                     24_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 24_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/24_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/24_Homecage_AGTTCC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/24_Homecage/24_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 24_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/24_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/24_Homecage_AGTTCC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/24_Homecage/24_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     10_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 10_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/10_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/10_FC-Alone_AGTTCC_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/10_FC-Alone/10_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 10_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/10_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/10_FC-Alone_AGTTCC_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/10_FC-Alone/10_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     11_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 11_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/11_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/11_FC-Alone_CGATGT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/11_FC-Alone/11_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 11_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/11_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/11_FC-Alone_CGATGT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/11_FC-Alone/11_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     12_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 12_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/12_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/12_FC-Alone_TGACCA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/12_FC-Alone/12_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 12_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/12_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/12_FC-Alone_TGACCA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/12_FC-Alone/12_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     9_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 9_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/9_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/9_FC-Alone_AGTCAA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/9_FC-Alone/9_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 9_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/9_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/9_FC-Alone_AGTCAA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/9_FC-Alone/9_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     8_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 8_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/8_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/8_FC-Alone_CTTGTA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/8_FC-Alone/8_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 8_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/8_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/8_FC-Alone_CTTGTA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/8_FC-Alone/8_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     1_IMMO-FC
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 1_IMMO-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/1_IMMO-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/1_IMMO-FC_CGATGT_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/1_IMMO-FC/1_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker
mkdir 1_IMMO-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/1_IMMO-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/1_IMMO-FC_CGATGT_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/1_IMMO-FC/1_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     2_IMMO-FC
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="mkdir 2_IMMO-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/2_IMMO-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/2_IMMO-FC_TGACCA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/2_IMMO-FC/2_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>mkdir 2_IMMO-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/2_IMMO-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/2_IMMO-FC_TGACCA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/2_IMMO-FC/2_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     3_IMMO-FC
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/
mkdir 3_IMMO-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/3_IMMO-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/3_IMMO-FC_ACAGTG_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/3_IMMO-FC/3_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/
mkdir 3_IMMO-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/3_IMMO-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/3_IMMO-FC_ACAGTG_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/3_IMMO-FC/3_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     4_IMMO-FC
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/
mkdir 4_IMMO-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/4_IMMO-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/4_IMMO-FC_GCCAAT_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/4_IMMO-FC/4_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/
mkdir 4_IMMO-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/4_IMMO-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/4_IMMO-FC_GCCAAT_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/4_IMMO-FC/4_IMMO-FC_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     1-935-HC (RNAseq Replication)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 1-935-HC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/1-935-HC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/1-935-HC/1-935-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/1-935-HC_GTCCGC_L008_R1_001.fastq.gz">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 1-935-HC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/1-935-HC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/1-935-HC/1-935-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/1-935-HC_GTCCGC_L008_R1_001.fastq.gz
</code></pre>
                    </div>
                    <p dir="auto">
                     2-935-HC (RNAseq Replication)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 2-935-HC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/2-935-HC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/2-935-HC/2-935-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/2-935-HC_GTGAAA_L008_R1_001.fastq.gz">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 2-935-HC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/2-935-HC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/2-935-HC/2-935-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/2-935-HC_GTGAAA_L008_R1_001.fastq.gz
</code></pre>
                    </div>
                    <p dir="auto">
                     3-940-FC (RNAseq Replication)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 3-940-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/3-940-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/3-940-FC/3-940-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/3-940-FC_ATCACG_L008_R1_001.fastq.gz">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 3-940-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/3-940-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/3-940-FC/3-940-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/3-940-FC_ATCACG_L008_R1_001.fastq.gz
</code></pre>
                    </div>
                    <p dir="auto">
                     4-940-FC (RNAseq Replication)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 4-940-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/4-940-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/4-940-FC/4-940-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/4-940-FC_TTAGGC_L008_R1_001.fastq.gz">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 4-940-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/4-940-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/4-940-FC/4-940-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/4-940-FC_TTAGGC_L008_R1_001.fastq.gz
</code></pre>
                    </div>
                    <p dir="auto">
                     5-938-FC (RNAseq Replication)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 5-938-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/5-938-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/5-938-FC/5-938-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/5-938-FC_ACTTGA_L008_R1_001.fastq.gz">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 5-938-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/5-938-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/5-938-FC/5-938-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/5-938-FC_ACTTGA_L008_R1_001.fastq.gz
</code></pre>
                    </div>
                    <p dir="auto">
                     6-938-FC (RNAseq Replication)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 6-938-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/6-938-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/6-938-FC/6-938-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/6-938-FC_GATCAG_L008_R1_001.fastq.gz">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 6-938-FC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/6-938-FC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/6-938-FC/6-938-FC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/6-938-FC_GATCAG_L008_R1_001.fastq.gz
</code></pre>
                    </div>
                    <p dir="auto">
                     7-947-HC (RNAseq Replication)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 7-947-HC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/7-947-HC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/7-947-HC/7-947-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/7-947-HC_TAGCTT_L008_R1_001.fastq.gz">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 7-947-HC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/7-947-HC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/7-947-HC/7-947-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/7-947-HC_TAGCTT_L008_R1_001.fastq.gz
</code></pre>
                    </div>
                    <p dir="auto">
                     8-947-HC (RNAseq Replication)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 8-947-HC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/8-947-HC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/8-947-HC/8-947-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/8-947-HC_GGCTAC_L008_R1_001.fastq.gz">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC
mkdir 8-947-HC
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/8-947-HC
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/8-947-HC/8-947-HC_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/fastq_files/8-947-HC_GGCTAC_L008_R1_001.fastq.gz
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      STAR Alignment: Combined HC, FC 5hmC Seq with UCSC
                     </strong>
                     *
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC
mkdir FC_5hmC_ALLreps
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/FC_5hmC_ALLreps
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/FC_5hmC_ALLreps/FC_5hmC_ALLreps_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/fastq_combined_files/FC_5hmC_ALLreps.fastq &amp;

cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC
mkdir HC_5hmC_ALLreps
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/HC_5hmC_ALLreps
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/HC_5hmC_ALLreps/HC_5hmC_ALLreps_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/fastq_combined_files/HC_5hmC_ALLreps.fastq &amp;">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC
mkdir FC_5hmC_ALLreps
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/FC_5hmC_ALLreps
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/FC_5hmC_ALLreps/FC_5hmC_ALLreps_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/fastq_combined_files/FC_5hmC_ALLreps.fastq &amp;

cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC
mkdir HC_5hmC_ALLreps
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/HC_5hmC_ALLreps
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --runThreadN 25 --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Sequence/STARIndex --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/HC_5hmC_ALLreps/HC_5hmC_ALLreps_ --outSAMstrandField intronMotif --outSAMtype BAM SortedByCoordinate --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/fastq_combined_files/HC_5hmC_ALLreps.fastq &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      STAR Alignment: ensembl
                     </strong>
                    </p>
                    <p dir="auto">
                     19_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 19_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/19_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/19_Homecage_ACAGTG_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/19_Homecage/19_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts ">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 19_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/19_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/19_Homecage_ACAGTG_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/19_Homecage/19_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts 
</code></pre>
                    </div>
                    <p dir="auto">
                     20_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/
mkdir 20_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/20_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/20_Homecage_GCCAAT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/20_Homecage/20_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts ">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/
mkdir 20_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/20_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/20_Homecage_GCCAAT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/20_Homecage/20_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts 
</code></pre>
                    </div>
                    <p dir="auto">
                     21_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 21_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/21_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/21_Homecage_CAGATC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/21_Homecage/21_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts ">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 21_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/21_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/21_Homecage_CAGATC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/21_Homecage/21_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts 
</code></pre>
                    </div>
                    <p dir="auto">
                     22_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 22_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/22_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/22_Homecage_CTTGTA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/22_Homecage/22_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts ">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 22_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/22_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/22_Homecage_CTTGTA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/22_Homecage/22_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts 
</code></pre>
                    </div>
                    <p dir="auto">
                     23_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 23_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/23_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/23_Homecage_AGTCAA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/23_Homecage/23_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts ">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 23_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/23_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/23_Homecage_AGTCAA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/23_Homecage/23_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts 
</code></pre>
                    </div>
                    <p dir="auto">
                     24_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 24_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/24_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/24_Homecage_AGTTCC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/24_Homecage/24_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 24_Homecage
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/24_Homecage
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/24_Homecage_AGTTCC_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/24_Homecage/24_Homecage_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     10_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 10_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/10_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/10_FC-Alone_AGTTCC_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/10_FC-Alone/10_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 10_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/10_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/10_FC-Alone_AGTTCC_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/10_FC-Alone/10_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     11_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 11_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/11_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/11_FC-Alone_CGATGT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/11_FC-Alone/11_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 11_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/11_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/11_FC-Alone_CGATGT_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/11_FC-Alone/11_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     12_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 12_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/12_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/12_FC-Alone_TGACCA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/12_FC-Alone/12_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 12_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/12_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/12_FC-Alone_TGACCA_L002_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/12_FC-Alone/12_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     9_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 9_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/9_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/9_FC-Alone_AGTCAA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/9_FC-Alone/9_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 9_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/9_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/9_FC-Alone_AGTCAA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/9_FC-Alone/9_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     8_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 8_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/8_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/8_FC-Alone_CTTGTA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/8_FC-Alone/8_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts">
                     <pre class="notranslate"><code>cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens
mkdir 8_FC-Alone
cd /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/8_FC-Alone
/home/ssharma/bin/STAR-STAR_2.4.2a/bin/Linux_x86_64/STAR --readFilesCommand zcat --readFilesIn /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/fastq_files/8_FC-Alone_CTTGTA_L001_R1_001.fastq.gz --outFileNamePrefix /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/8_FC-Alone/8_FC-Alone_ --outSAMtype BAM SortedByCoordinate --genomeDir /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Sequence/STARIndex --runThreadN 25 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --quantMode GeneCounts
</code></pre>
                    </div>
                    <p dir="auto">
                     ###2. RNASEQ ANALYSIS:DIFFERENTIAL EXPRESSION:CUFFDIFF
                    </p>
                    <p dir="auto">
                     CUFFDIFF
                    </p>
                    <p dir="auto">
                     HC vs IMMO-FC Differential expression analysis: using only repeat masked gtf
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cuffdiff -o ./AmygHC_vs_AmygIMMO-FC_RepeatMasked_SMALL -L HC,IMMO-FC -p 25 -u /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked.gtf /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/19_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/20_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/21_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/22_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/23_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/24_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/1_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/2_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/3_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/4_IMMO-FC_Aligned.sortedByCoord.out.bam ">
                     <pre class="notranslate"><code>cuffdiff -o ./AmygHC_vs_AmygIMMO-FC_RepeatMasked_SMALL -L HC,IMMO-FC -p 25 -u /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked.gtf /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/19_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/20_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/21_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/22_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/23_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/24_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/1_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/2_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/3_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/4_IMMO-FC_Aligned.sortedByCoord.out.bam 
</code></pre>
                    </div>
                    <p dir="auto">
                     HC vs FC Differential expression analysis: using only repeat masked gtf
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cuffdiff -o ./AmygFC_vs_AmygHC_RepeatMasked_SMALL -L HC,FC -p 2 -u /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gtf /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/19_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/20_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/21_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/22_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/23_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/24_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/10_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/11_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/12_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/8_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/9_FC-Alone_Aligned.sortedByCoord.out.bam &amp;">
                     <pre class="notranslate"><code>cuffdiff -o ./AmygFC_vs_AmygHC_RepeatMasked_SMALL -L HC,FC -p 2 -u /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gtf /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/19_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/20_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/21_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/22_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/23_Homecage_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/24_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/10_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/11_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/12_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/8_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/9_FC-Alone_Aligned.sortedByCoord.out.bam &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     FC vs IMMO DE
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cuffdiff -o ./AmygFC_vs_AmygIMMO_UCSC_genes -L FC,IMMOFC -p 25 -u /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/genes.gtf /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/10_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/11_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/12_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/8_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/9_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/1_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/2_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/3_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/4_IMMO-FC_Aligned.sortedByCoord.out.bam ">
                     <pre class="notranslate"><code>cuffdiff -o ./AmygFC_vs_AmygIMMO_UCSC_genes -L FC,IMMOFC -p 25 -u /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/genes.gtf /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/10_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/11_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/12_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/8_FC-Alone_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/9_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/1_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/2_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/3_IMMO-FC_Aligned.sortedByCoord.out.bam,/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/BAMs/4_IMMO-FC_Aligned.sortedByCoord.out.bam 
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='

Alignment Runs: Ensembl




###3. RNASEQ ANALYSIS:DIFFERENTIAL EXPRESSION:DESEQ2

se is the RangedSummarizedExperiment object containing counts and information about the samples. In this example design we control for batch and condition, which should be columns of colData(se)
```{r}
dds &lt;- DESeqDataSet(se, design = ~ batch + condition)
dds &lt;- DESeq(dds)
res &lt;- results(dds, contrast=c("condition","trt","con"))'>
                     <pre class="notranslate"><code>

Alignment Runs: Ensembl




###3. RNASEQ ANALYSIS:DIFFERENTIAL EXPRESSION:DESEQ2

se is the RangedSummarizedExperiment object containing counts and information about the samples. In this example design we control for batch and condition, which should be columns of colData(se)
```{r}
dds &lt;- DESeqDataSet(se, design = ~ batch + condition)
dds &lt;- DESeq(dds)
res &lt;- results(dds, contrast=c("condition","trt","con"))
</code></pre>
                    </div>
                    <p dir="auto">
                     Combine counts from geneCounts function of STAR to generate count matrices:
                    </p>
                    <p dir="auto">
                     Read the Reads/Gene table
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='FC10Alone=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/10_FC-Alone/10_FC-Alone_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

FC11Alone=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/11_FC-Alone/11_FC-Alone_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

FC12Alone=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/12_FC-Alone/12_FC-Alone_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

Homecage19=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/19_Homecage/19_Homecage_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

Homecage20=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/20_Homecage/20_Homecage_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

Homecage21=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/21_Homecage/21_Homecage_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

Homecage22=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/22_Homecage/22_Homecage_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

Homecage23=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/23_Homecage/23_Homecage_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

Homecage24=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/24_Homecage/24_Homecage_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

FC8Alone=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/8_FC-Alone/8_FC-Alone_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

FC9Alone=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/9_FC-Alone/9_FC-Alone_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")'>
                     <pre class="notranslate" lang="{r}"><code>FC10Alone=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/10_FC-Alone/10_FC-Alone_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

FC11Alone=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/11_FC-Alone/11_FC-Alone_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

FC12Alone=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/12_FC-Alone/12_FC-Alone_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

Homecage19=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/19_Homecage/19_Homecage_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

Homecage20=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/20_Homecage/20_Homecage_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

Homecage21=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/21_Homecage/21_Homecage_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

Homecage22=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/22_Homecage/22_Homecage_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

Homecage23=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/23_Homecage/23_Homecage_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

Homecage24=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/24_Homecage/24_Homecage_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

FC8Alone=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/8_FC-Alone/8_FC-Alone_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

FC9Alone=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/9_FC-Alone/9_FC-Alone_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
</code></pre>
                    </div>
                    <p dir="auto">
                     Replication UCSC BAMs
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='HC8r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/8-947-HC/8-947-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

HC7r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/7-947-HC/7-947-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
 
FC6r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/6-938-FC/6-938-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
  
FC5r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/5-938-FC/5-938-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

FC4r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/4-940-FC/4-940-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
  
FC3r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/3-940-FC/3-940-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "") 
 
HC2r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/2-935-HC/2-935-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")  

HC1r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/1-935-HC/1-935-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
'>
                     <pre class="notranslate" lang="{r}"><code>HC8r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/8-947-HC/8-947-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

HC7r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/7-947-HC/7-947-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
 
FC6r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/6-938-FC/6-938-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
  
FC5r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/5-938-FC/5-938-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

FC4r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/4-940-FC/4-940-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
  
FC3r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/3-940-FC/3-940-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "") 
 
HC2r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/2-935-HC/2-935-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")  

HC1r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_UCSC/1-935-HC/1-935-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

</code></pre>
                    </div>
                    <p dir="auto">
                     Replication ensemble BAMs
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='HC1r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/1-935-HC/1-935-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

FC3r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/3-940-FC/3-940-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
 
FC5r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/5-938-FC/5-938-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
  
HC7r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/7-947-HC/7-947-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

HC2r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/2-935-HC/2-935-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
  
FC4r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/4-940-FC/4-940-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "") 
 
FC6r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/6-938-FC/6-938-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")  

HC8r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/8-947-HC/8-947-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")'>
                     <pre class="notranslate" lang="{r}"><code>HC1r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/1-935-HC/1-935-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

FC3r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/3-940-FC/3-940-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
 
FC5r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/5-938-FC/5-938-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
  
HC7r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/7-947-HC/7-947-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

HC2r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/2-935-HC/2-935-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
  
FC4r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/4-940-FC/4-940-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "") 
 
FC6r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/6-938-FC/6-938-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")  

HC8r=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/8-947-HC/8-947-HC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
</code></pre>
                    </div>
                    <p dir="auto">
                     IMMO - UCSC BAMs
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='IMMOFC3=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/3_IMMO-FC/3_IMMO-FC_ReadsPerGene.out.tab", header = F, sep ="\t", fill =T, comment.char = "")

IMMOFC1=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/1_IMMO-FC/1_IMMO-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

IMMOFC2=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/2_IMMO-FC/2_IMMO-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

IMMOFC4=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/4_IMMO-FC/4_IMMO-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")'>
                     <pre class="notranslate" lang="{r}"><code>IMMOFC3=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/3_IMMO-FC/3_IMMO-FC_ReadsPerGene.out.tab", header = F, sep ="\t", fill =T, comment.char = "")

IMMOFC1=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/1_IMMO-FC/1_IMMO-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

IMMOFC2=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/2_IMMO-FC/2_IMMO-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")

IMMOFC4=read.table("/home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_UCSC_RepeatMasker/4_IMMO-FC/4_IMMO-FC_ReadsPerGene.out.tab", header =F, sep ="\t", fill =T, comment.char = "")
</code></pre>
                    </div>
                    <p dir="auto">
                     Combine column 1, with column 2 from every dataframe, add a header with the name of the sample that column came from
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='HCvsFC_gene_counts = data.frame(FC10Alone$V2, FC11Alone$V2, FC12Alone$V2, Homecage19$V2, Homecage20$V2, Homecage21$V2, Homecage22$V2, Homecage23$V2, Homecage24$V2, FC8Alone$V2, FC9Alone$V2, row.names = Homecage22$V1) 
colnames(HCvsFC_rep_gene_counts) = c("FC10Alone", "FC11Alone", "FC12Alone", "Homecage19", "Homecage20", "Homecage21", "Homecage22", "Homecage23","Homecage24", "FC8Alone", "FC9Alone")

HCvsIMMOFC_gene_counts = data.frame(Homecage19$V2, Homecage20$V2, Homecage21$V2, Homecage22$V2, Homecage23$V2, Homecage24$V2, IMMOFC3$V2, IMMOFC1$V2, IMMOFC2$V2, IMMOFC4$V2, row.names = Homecage22$V1)
colnames(HCvsFC_rep_gene_counts) = c("Homecage19", "Homecage20", "Homecage21", "Homecage22", "Homecage23", "Homecage24", "IMMOFC3", "IMMOFC1", "IMMOFC2", "IMMOFC4")

FCvsIMMOFC_genecounts = data.frame(FC10Alone$V2, FC11Alone$V2, FC12Alone$V2, FC8Alone$V2, FC9Alone$V2, IMMOFC3$V2, IMMOFC1$V2, IMMOFC2$V2, IMMOFC4$V2, row.names = Homecage22$V1)
colnames(HCvsFC_rep_gene_counts) = c("FC10Alone", "FC11Alone", "FC12Alone", "FC8Alone", "FC9Alone", "IMMOFC3", "IMMOFC1", "IMMOFC2", "IMMOFC4")

HCvsFC_ALL = data.frame(HC1r$V2, HC7r$V2, HC2r$V2, HC8r$V2, Homecage19$V2, Homecage20$V2, Homecage21$V2, Homecage22$V2, Homecage23$V2, Homecage24$V2, FC3r$V2, FC5r$V2, FC4r$V2, FC6r$V2, FC10Alone$V2, FC11Alone$V2, FC12Alone$V2, FC8Alone$V2, FC9Alone$V2, row.names = Homecage22$V1)
colnames(HCvsFC_ALL) = c("HC1r", "HC7r", "HC2r", "HC8r", "Homecage19", "Homecage20", "Homecage21", "Homecage22", "Homecage23", "Homecage24", "FC3r", "FC5r", "FC4r", "FC6r", "FC10Alone", "FC11Alone", "FC12Alone", "FC8Alone", "FC9Alone")

HCvsFC_rep_gene_counts = data.frame(HC1r$V2, FC3r$V2, FC5r$V2, HC7r$V2, HC2r$V2, FC4r$V2, FC6r$V2, HC8r$V2, row.names = HC1r$V1) 
colnames(HCvsFC_rep_gene_counts) = c("HC1r", "FC3r", "FC5r", "HC7r", "HC2r", "FC4r", "FC6r", "HC8r")'>
                     <pre class="notranslate" lang="{r}"><code>HCvsFC_gene_counts = data.frame(FC10Alone$V2, FC11Alone$V2, FC12Alone$V2, Homecage19$V2, Homecage20$V2, Homecage21$V2, Homecage22$V2, Homecage23$V2, Homecage24$V2, FC8Alone$V2, FC9Alone$V2, row.names = Homecage22$V1) 
colnames(HCvsFC_rep_gene_counts) = c("FC10Alone", "FC11Alone", "FC12Alone", "Homecage19", "Homecage20", "Homecage21", "Homecage22", "Homecage23","Homecage24", "FC8Alone", "FC9Alone")

HCvsIMMOFC_gene_counts = data.frame(Homecage19$V2, Homecage20$V2, Homecage21$V2, Homecage22$V2, Homecage23$V2, Homecage24$V2, IMMOFC3$V2, IMMOFC1$V2, IMMOFC2$V2, IMMOFC4$V2, row.names = Homecage22$V1)
colnames(HCvsFC_rep_gene_counts) = c("Homecage19", "Homecage20", "Homecage21", "Homecage22", "Homecage23", "Homecage24", "IMMOFC3", "IMMOFC1", "IMMOFC2", "IMMOFC4")

FCvsIMMOFC_genecounts = data.frame(FC10Alone$V2, FC11Alone$V2, FC12Alone$V2, FC8Alone$V2, FC9Alone$V2, IMMOFC3$V2, IMMOFC1$V2, IMMOFC2$V2, IMMOFC4$V2, row.names = Homecage22$V1)
colnames(HCvsFC_rep_gene_counts) = c("FC10Alone", "FC11Alone", "FC12Alone", "FC8Alone", "FC9Alone", "IMMOFC3", "IMMOFC1", "IMMOFC2", "IMMOFC4")

HCvsFC_ALL = data.frame(HC1r$V2, HC7r$V2, HC2r$V2, HC8r$V2, Homecage19$V2, Homecage20$V2, Homecage21$V2, Homecage22$V2, Homecage23$V2, Homecage24$V2, FC3r$V2, FC5r$V2, FC4r$V2, FC6r$V2, FC10Alone$V2, FC11Alone$V2, FC12Alone$V2, FC8Alone$V2, FC9Alone$V2, row.names = Homecage22$V1)
colnames(HCvsFC_ALL) = c("HC1r", "HC7r", "HC2r", "HC8r", "Homecage19", "Homecage20", "Homecage21", "Homecage22", "Homecage23", "Homecage24", "FC3r", "FC5r", "FC4r", "FC6r", "FC10Alone", "FC11Alone", "FC12Alone", "FC8Alone", "FC9Alone")

HCvsFC_rep_gene_counts = data.frame(HC1r$V2, FC3r$V2, FC5r$V2, HC7r$V2, HC2r$V2, FC4r$V2, FC6r$V2, HC8r$V2, row.names = HC1r$V1) 
colnames(HCvsFC_rep_gene_counts) = c("HC1r", "FC3r", "FC5r", "HC7r", "HC2r", "FC4r", "FC6r", "HC8r")
</code></pre>
                    </div>
                    <p dir="auto">
                     Read in data about each sample
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='colData = read.table("/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DESeq2/colData_HCvsFC_ALL", header =T, sep ="\t")

'>
                     <pre class="notranslate"><code>colData = read.table("/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DESeq2/colData_HCvsFC_ALL", header =T, sep ="\t")


</code></pre>
                    </div>
                    <p dir="auto">
                     DESeq library:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="library(DESeq2)">
                     <pre class="notranslate"><code>library(DESeq2)
</code></pre>
                    </div>
                    <p dir="auto">
                     se is the RangedSummarizedExperiment object containing counts and information about the samples. In this example design we control for batch and condition, which should be columns of colData(se)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="dds &lt;- DESeqDataSetFromMatrix(countData = HCvsFC_ALL,
                              colData = colData,
                              design = ~ Condition)">
                     <pre class="notranslate" lang="{r}"><code>dds &lt;- DESeqDataSetFromMatrix(countData = HCvsFC_ALL,
                              colData = colData,
                              design = ~ Condition)
</code></pre>
                    </div>
                    <p dir="auto">
                     Relevel so that HomeCage is the reference
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='dds$Condition = relevel(dds$Condition, "HomeCage")'>
                     <pre class="notranslate" lang="{r}"><code>dds$Condition = relevel(dds$Condition, "HomeCage")
</code></pre>
                    </div>
                    <p dir="auto">
                     DE Analysis:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="library(BiocParallel)
register(MulticoreParam(20))
dds = DESeq(dds, parallel=TRUE)
res = results(dds)">
                     <pre class="notranslate" lang="{r}"><code>library(BiocParallel)
register(MulticoreParam(20))
dds = DESeq(dds, parallel=TRUE)
res = results(dds)
</code></pre>
                    </div>
                    <p dir="auto">
                     Manipulation of results dataframe
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="resOrdered = res[order(res$padj),]
summary(res)">
                     <pre class="notranslate" lang="{r}"><code>resOrdered = res[order(res$padj),]
summary(res)
</code></pre>
                    </div>
                    <p dir="auto">
                     Export Results
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='resOrdered_subset = subset(resOrdered, padj &lt;0.3)
write.csv(as.data.frame(resOrdered), file="HCvsFC_RNASeq_Replicate_Results.csv")'>
                     <pre class="notranslate" lang="{r}"><code>resOrdered_subset = subset(resOrdered, padj &lt;0.3)
write.csv(as.data.frame(resOrdered), file="HCvsFC_RNASeq_Replicate_Results.csv")
</code></pre>
                    </div>
                    <p dir="auto">
                     PCA
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='rld = rlog(dds)
pdf("PCA_HCvsFC_ALL")
plotPCA(rld, intgroup=c("Condition"))
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>rld = rlog(dds)
pdf("PCA_HCvsFC_ALL")
plotPCA(rld, intgroup=c("Condition"))
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     ###4. RNASEQ ANALYSIS:DIFFERENTIAL SPLICING:DEXSEQ &amp; CUFFDIFF
                    </p>
                    <p dir="auto">
                     HTSeq: generate flattened GFF of the exon counting bins
mm10_UCSC_RepeatMasked_Genes.gtf (comprehensive gtf file)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_prepare_annotation.py /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gtf /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gff &amp;">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_prepare_annotation.py /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gtf /home/Shared/PengLab/iGenomes/Mus_musculus/RepeatMasker/mm10/Annotation/mm10_UCSC_RepeatMasked_Genes.gff &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     genes.gtf (UCSC genes only)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_prepare_annotation.py -r no genes.gtf genes.gff">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_prepare_annotation.py -r no genes.gtf genes.gff
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      TOO MANY ISSUES WITH TEH UCSC GTF FILES - USE ENSEMBL FILES FOR NOW...SEE IF THEY RELEASE A PATCH TO MAKE UCSC ANNOTATIONS FUNCTIONAL LATER ON
                     </em>
                    </p>
                    <p dir="auto">
                     Counting reads into exon bins:
                    </p>
                    <p dir="auto">
                     1-935-HC
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/1-935-HC/1-935-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/1-935-HC_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/1-935-HC/1-935-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/1-935-HC_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     2-935-HC:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/2-935-HC/2-935-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/2-935-HC_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/2-935-HC/2-935-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/2-935-HC_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     3-940-FC:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/3-940-FC/3-940-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/3-940-FC_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/3-940-FC/3-940-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/3-940-FC_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     4-940-FC:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/4-940-FC/4-940-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/4-940-FC_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/4-940-FC/4-940-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/4-940-FC_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     5-938-FC:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/5-938-FC/5-938-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/5-938-FC_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/5-938-FC/5-938-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/5-938-FC_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     6-938-FC
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/6-938-FC/6-938-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/6-938-FC_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/6-938-FC/6-938-FC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/6-938-FC_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     7-947-HC:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/7-947-HC/7-947-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/7-947-HC_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/7-947-HC/7-947-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/7-947-HC_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     8-947-HC:
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/8-947-HC/8-947-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/8-947-HC_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Amygdala_RNASeq_Replication/STARAlignedBAMs_ens/8-947-HC/8-947-HC_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/8-947-HC_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     19_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/19_Homecage/19_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/19_HC_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/19_Homecage/19_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/19_HC_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     20_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/20_Homecage/20_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/20_HC_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/20_Homecage/20_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/20_HC_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     21_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/21_Homecage/21_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/21_HC_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/21_Homecage/21_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/21_HC_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     22_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/22_Homecage/22_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/22_HC_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/22_Homecage/22_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/22_HC_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     23_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/23_Homecage/23_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/23_HC_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/23_Homecage/23_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/23_HC_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     24_Homecage
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/24_Homecage/24_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/24_HC_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/24_Homecage/24_Homecage_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/24_HC_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     10_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/10_FC-Alone/10_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/10_FC-Alone_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/10_FC-Alone/10_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/10_FC-Alone_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     11_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/11_FC-Alone/11_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/11_FC-Alone_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/11_FC-Alone/11_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/11_FC-Alone_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     12_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/12_FC-Alone/12_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/12_FC-Alone_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/12_FC-Alone/12_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/12_FC-Alone_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     9_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/9_FC-Alone/9_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/9_FC-Alone_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/9_FC-Alone/9_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/9_FC-Alone_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     8_FC-Alone
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/8_FC-Alone/8_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/8_FC-Alone_ExonCounts.txt">
                     <pre class="notranslate"><code>python /mnt/icebreaker/data/home/ssharma/R/x86_64-unknown-linux-gnu-library/3.1/DEXSeq/python_scripts/dexseq_count.py -s no -f bam /home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes/Mus_musculus.GRCm38.79.gff /home/ssharma/Ressler_RNASeq/data/FearConditioning_Immobilization_Extinction_Amygdala_RNASeq/STAR_Aligned_BAMs_ens/8_FC-Alone/8_FC-Alone_Aligned.sortedByCoord.out.bam /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/8_FC-Alone_ExonCounts.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     Read data into R
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='countDir = "/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/ExonCounts2"
countFiles = list.files(countDir, pattern="ExonCounts.txt$", full.names=TRUE)
flatDir = "/home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes"
flattenedFile = list.files(flatDir, pattern="gff$", full.names=TRUE)'>
                     <pre class="notranslate"><code>countDir = "/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/ExonCounts2"
countFiles = list.files(countDir, pattern="ExonCounts.txt$", full.names=TRUE)
flatDir = "/home/Shared/PengLab/iGenomes/Mus_musculus/ensembl/GRCm38.79/Annotation/Genes"
flattenedFile = list.files(flatDir, pattern="gff$", full.names=TRUE)
</code></pre>
                    </div>
                    <p dir="auto">
                     Prepare Sample Table
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='sampleTable = read.table("/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/HCvsFC_ALL_SampleTable.DEXSeq", sep ="\t", header = T)

sampleTable = data.frame(
   row.names = c( "10_FC-Alone", "11_FC-Alone", "12_FC-Alone", "1-935-HC", "19_HC", "20_HC", "21_HC", "22_HC", "23_HC", "24_HC", "2-935-HC", "3-940-FC", "4-940-FC", "5-938-FC", "6-938-FC", "7-947-HC", "8-947-HC", "8_FC-Alone", "9_FC-Alone" ),
   condition = c("FearConditioning", "FearConditioning", "FearConditioning", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "FearConditioning", "FearConditioning", "FearConditioning", "FearConditioning", "HomeCage", "HomeCage", "FearConditioning", "FearConditioning" )
   )
   
   
sampleTable = data.frame(
   row.names = c( "10_FC-Alone", "11_FC-Alone", "12_FC-Alone", "19_HC", "20_HC", "21_HC", "22_HC", "23_HC", "24_HC", "8_FC-Alone", "9_FC-Alone"),
   condition = c("FearConditioning", "FearConditioning", "FearConditioning", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "FearConditioning", "FearConditioning" )
   )
'>
                     <pre class="notranslate"><code>sampleTable = read.table("/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq/HCvsFC_ALL_SampleTable.DEXSeq", sep ="\t", header = T)

sampleTable = data.frame(
   row.names = c( "10_FC-Alone", "11_FC-Alone", "12_FC-Alone", "1-935-HC", "19_HC", "20_HC", "21_HC", "22_HC", "23_HC", "24_HC", "2-935-HC", "3-940-FC", "4-940-FC", "5-938-FC", "6-938-FC", "7-947-HC", "8-947-HC", "8_FC-Alone", "9_FC-Alone" ),
   condition = c("FearConditioning", "FearConditioning", "FearConditioning", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "FearConditioning", "FearConditioning", "FearConditioning", "FearConditioning", "HomeCage", "HomeCage", "FearConditioning", "FearConditioning" )
   )
   
   
sampleTable = data.frame(
   row.names = c( "10_FC-Alone", "11_FC-Alone", "12_FC-Alone", "19_HC", "20_HC", "21_HC", "22_HC", "23_HC", "24_HC", "8_FC-Alone", "9_FC-Alone"),
   condition = c("FearConditioning", "FearConditioning", "FearConditioning", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "HomeCage", "FearConditioning", "FearConditioning" )
   )

</code></pre>
                    </div>
                    <p dir="auto">
                     DEXSeq Analysis
Data object creation
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='library( "DEXSeq" )

BPPARAM = MulticoreParam(workers=25)
dxd = DEXSeqDataSetFromHTSeq(countFiles, sampleData=sampleTable, design= ~sample + exon + condition:exon, flattenedfile=flattenedFile )'>
                     <pre class="notranslate" lang="{r}"><code>library( "DEXSeq" )

BPPARAM = MulticoreParam(workers=25)
dxd = DEXSeqDataSetFromHTSeq(countFiles, sampleData=sampleTable, design= ~sample + exon + condition:exon, flattenedfile=flattenedFile )
</code></pre>
                    </div>
                    <p dir="auto">
                     analyze subset of genes
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='geneDir = "/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq"
genesForSubset = read.table(
  file.path(geneDir, "ENS_Genes"),
  stringsAsFactors=FALSE)[[1]]
  
dxd = dxd[geneIDs( dxd ) %in% genesForSubset,]'>
                     <pre class="notranslate" lang="{r}"><code>geneDir = "/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_RNASeq_DEXSeq"
genesForSubset = read.table(
  file.path(geneDir, "ENS_Genes"),
  stringsAsFactors=FALSE)[[1]]
  
dxd = dxd[geneIDs( dxd ) %in% genesForSubset,]
</code></pre>
                    </div>
                    <p dir="auto">
                     Normalisation &amp; Dispersion estimation &amp; DEU test
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="dxd = estimateSizeFactors( dxd )
dxd = estimateDispersions( dxd, BPPARAM=BPPARAM )">
                     <pre class="notranslate" lang="{r}"><code>dxd = estimateSizeFactors( dxd )
dxd = estimateDispersions( dxd, BPPARAM=BPPARAM )
</code></pre>
                    </div>
                    <p dir="auto">
                     Dispersion plots
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Dispersion_estimates.pdf")
plotDispEsts( dxd )
dev.off()'>
                     <pre class="notranslate"><code>pdf("Dispersion_estimates.pdf")
plotDispEsts( dxd )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     DEU testing
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='dxd = testForDEU( dxd, BPPARAM=BPPARAM )
dxd = estimateExonFoldChanges( dxd, fitExpToVar="condition", BPPARAM=BPPARAM )
dxr1 = DEXSeqResults( dxd )'>
                     <pre class="notranslate"><code>dxd = testForDEU( dxd, BPPARAM=BPPARAM )
dxd = estimateExonFoldChanges( dxd, fitExpToVar="condition", BPPARAM=BPPARAM )
dxr1 = DEXSeqResults( dxd )
</code></pre>
                    </div>
                    <p dir="auto">
                     Significant Results
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="table ( dxr1$padj &lt; 0.1 )
">
                     <pre class="notranslate" lang="{r}"><code>table ( dxr1$padj &lt; 0.1 )

</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Gphn
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Gphn_2_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000047454", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Gphn_2_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000047454", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Gabrb1
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Gabrb1_2_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000029212", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Gabrb1_2_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000029212", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Gabrb2
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Gabrb2_2_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000007653", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Gabrb2_2_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000007653", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Gabrb3
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Gabrb3_2_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000033676", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Gabrb3_2_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000033676", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Gabrg1_2
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Gabrg1_3_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000001260", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Gabrg1_3_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000001260", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Cacn1e
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Cacn1e.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000004110", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Cacn1e.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000004110", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Bicd1
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Bicd1.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000003452", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Bicd1.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000003452", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Glg1
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Glg1.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000003316", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Glg1.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000003316", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Plag1
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Plag1.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000003282", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Plag1.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000003282", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Ranbp2
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Ranbp2.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000003226", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Ranbp2.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000003226", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Peg3
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Peg3.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000002265", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Peg3.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000002265", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Celf2
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Celf2.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000002107", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Celf2.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000002107", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Gabarap
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Gabarap_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000018567", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Gabarap_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000018567", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Gabarapl1
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Gabarapl1_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000030161", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Gabarapl1_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000030161", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Gabarapl2
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Gabarapl2_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000031950", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Gabarapl2_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000031950", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Gad1
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Gad1_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000070880", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Gad1_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000070880", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Gad1_2
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Gad1_2_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000070880", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Gad1_2_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000070880", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Gad1_3
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Gad1__5_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000070880", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Gad1__5_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000070880", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Gad2
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Gad2_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000026787", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Gad2_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000026787", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Slc6a11
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Slc6a11_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000030307", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Slc6a11_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000030307", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Slc6a12
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Slc6a12_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000030109", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Slc6a12_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000030109", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: gabra1
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Gabra1_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000010803", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2)
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Gabra1_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000010803", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2)
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Tac2
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='dxr2 = DEXSeqResults( dxd )
pdf("Tac2_1_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000025400", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2)
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>dxr2 = DEXSeqResults( dxd )
pdf("Tac2_1_DEU.pdf")
plotDEXSeq( dxr1, "ENSMUSG00000025400", legend=TRUE, displayTranscripts=TRUE, norCounts=TRUE, expression=FALSE, splicing=TRUE, cex.axis=1.2, cex=1.3, lwd=2)
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: Nts
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("Nts_DEU.pdf")
plotDEXSeq( dxr2, "ENSMUSG00000019890", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("Nts_DEU.pdf")
plotDEXSeq( dxr2, "ENSMUSG00000019890", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: POMC
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("POMC_DEU.pdf")
plotDEXSeq( dxr2, "ENSMUSG00000020660", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("POMC_DEU.pdf")
plotDEXSeq( dxr2, "ENSMUSG00000020660", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     plotting: ErbB4
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='pdf("ErbB4_DEU.pdf")
plotDEXSeq( dxr2, "ENSMUSG00000062209", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()'>
                     <pre class="notranslate" lang="{r}"><code>pdf("ErbB4_DEU.pdf")
plotDEXSeq( dxr2, "ENSMUSG00000062209", legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )
dev.off()
</code></pre>
                    </div>
                    <p dir="auto">
                     ###5. RNASEQ ANALYSIS, TRAPSEQ:PATHWAT ANALYSIS:CONSENSUS PATH DATABASE
                    </p>
                    <p dir="auto">
                     URL (HOME):
                     <a href="http://cpdb.molgen.mpg.de/MCPDB" rel="nofollow">
                      http://cpdb.molgen.mpg.de/MCPDB
                     </a>
                     Analyses:
HOME/gene set analysis/over-representation analysis/enriched pathway-based sets/pathway-based sets
                    </p>
                    <p dir="auto">
                     HOME/gene set analysis/over-representation analysis/enriched pathway-based sets/Network neighborhood-based entity sets (NESTs)/1-next neighbor/AZI1 set center
                    </p>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        user-provided identifier
                       </th>
                       <th>
                        entrez-gene ID
                       </th>
                       <th>
                        entrez-gene name
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        DENND4A
                       </td>
                       <td>
                        10260
                       </td>
                       <td>
                        DENND4A : DENN/MADD domain containing 4A
                       </td>
                      </tr>
                      <tr>
                       <td>
                        MAP1B
                       </td>
                       <td>
                        4131
                       </td>
                       <td>
                        MAP1B : microtubule-associated protein 1B
                       </td>
                      </tr>
                      <tr>
                       <td>
                        MIB1
                       </td>
                       <td>
                        57534
                       </td>
                       <td>
                        MIB1 : mindbomb E3 ubiquitin protein ligase 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CC2D2A
                       </td>
                       <td>
                        57545
                       </td>
                       <td>
                        CC2D2A : coiled-coil and C2 domain containing 2A
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DENND4C
                       </td>
                       <td>
                        55667
                       </td>
                       <td>
                        DENND4C : DENN/MADD domain containing 4C
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CPEB4
                       </td>
                       <td>
                        80315
                       </td>
                       <td>
                        CPEB4 : cytoplasmic polyadenylation element binding protein 4
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CDK5RAP2
                       </td>
                       <td>
                        55755
                       </td>
                       <td>
                        CDK5RAP2 : CDK5 regulatory subunit associated protein 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        IRS2
                       </td>
                       <td>
                        8660
                       </td>
                       <td>
                        IRS2 : insulin receptor substrate 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ATM
                       </td>
                       <td>
                        472
                       </td>
                       <td>
                        ATM : ATM serine/threonine kinase
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SNTB2
                       </td>
                       <td>
                        6645
                       </td>
                       <td>
                        SNTB2 : syntrophin, beta 2 (dystrophin-associated protein A1, 59kDa, basic component 2)
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RAPGEF6
                       </td>
                       <td>
                        51735
                       </td>
                       <td>
                        RAPGEF6 : Rap guanine nucleotide exchange factor (GEF) 6
                       </td>
                      </tr>
                      <tr>
                       <td>
                        KIF1B
                       </td>
                       <td>
                        23095
                       </td>
                       <td>
                        KIF1B : kinesin family member 1B
                       </td>
                      </tr>
                      <tr>
                       <td>
                        BIRC6
                       </td>
                       <td>
                        57448
                       </td>
                       <td>
                        BIRC6 : baculoviral IAP repeat containing 6
                       </td>
                      </tr>
                      <tr>
                       <td>
                        TTBK2
                       </td>
                       <td>
                        146057
                       </td>
                       <td>
                        TTBK2 : tau tubulin kinase 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        STRN
                       </td>
                       <td>
                        6801
                       </td>
                       <td>
                        STRN : striatin, calmodulin binding protein
                       </td>
                      </tr>
                      <tr>
                       <td>
                        NINL
                       </td>
                       <td>
                        22981
                       </td>
                       <td>
                        NINL : ninein-like
                       </td>
                      </tr>
                      <tr>
                       <td>
                        FRYL
                       </td>
                       <td>
                        285527
                       </td>
                       <td>
                        FRYL : FRY-like
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CBL
                       </td>
                       <td>
                        867
                       </td>
                       <td>
                        CBL : Cbl proto-oncogene, E3 ubiquitin protein ligase
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GFOD1
                       </td>
                       <td>
                        54438
                       </td>
                       <td>
                        GFOD1 : glucose-fructose oxidoreductase domain containing 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        MAP3K2
                       </td>
                       <td>
                        10746
                       </td>
                       <td>
                        MAP3K2 : mitogen-activated protein kinase kinase kinase 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        PLK1
                       </td>
                       <td>
                        5347
                       </td>
                       <td>
                        PLK1 : polo-like kinase 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        UTRN
                       </td>
                       <td>
                        7402
                       </td>
                       <td>
                        UTRN : utrophin
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RASAL2
                       </td>
                       <td>
                        9462
                       </td>
                       <td>
                        RASAL2 : RAS protein activator like 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RICTOR
                       </td>
                       <td>
                        253260
                       </td>
                       <td>
                        RICTOR : RPTOR independent companion of MTOR, complex 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        TANC2
                       </td>
                       <td>
                        26115
                       </td>
                       <td>
                        TANC2 : tetratricopeptide repeat, ankyrin repeat and coiled-coil containing 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ALMS1
                       </td>
                       <td>
                        7840
                       </td>
                       <td>
                        ALMS1 : Alstrom syndrome protein 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        NAV1
                       </td>
                       <td>
                        89796
                       </td>
                       <td>
                        NAV1 : neuron navigator 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        HELZ
                       </td>
                       <td>
                        9931
                       </td>
                       <td>
                        HELZ : helicase with zinc finger
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DYNC1H1
                       </td>
                       <td>
                        1778
                       </td>
                       <td>
                        DYNC1H1 : dynein, cytoplasmic 1, heavy chain 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SHKBP1
                       </td>
                       <td>
                        92799
                       </td>
                       <td>
                        SHKBP1 : SH3KBP1 binding protein 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CCDC6
                       </td>
                       <td>
                        8030
                       </td>
                       <td>
                        CCDC6 : coiled-coil domain containing 6
                       </td>
                      </tr>
                      <tr>
                       <td>
                        AKAP9
                       </td>
                       <td>
                        10142
                       </td>
                       <td>
                        AKAP9 : A kinase (PRKA) anchor protein 9
                       </td>
                      </tr>
                      <tr>
                       <td>
                        LMO7
                       </td>
                       <td>
                        4008
                       </td>
                       <td>
                        LMO7 : LIM domain 7
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <p dir="auto">
                     HOME/gene set analysis/over-representation analysis/enriched pathway-based sets/Network neighborhood-based entity sets (NESTs)/1-next neighbor/HDAC4 set center
                    </p>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        user-provided identifier
                       </th>
                       <th>
                        entrez-gene ID
                       </th>
                       <th>
                        entrez-gene name
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        DENND4A
                       </td>
                       <td>
                        10260
                       </td>
                       <td>
                        DENND4A : DENN/MADD domain containing 4A
                       </td>
                      </tr>
                      <tr>
                       <td>
                        JDP2
                       </td>
                       <td>
                        122953
                       </td>
                       <td>
                        JDP2 : Jun dimerization protein 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        NAV1
                       </td>
                       <td>
                        89796
                       </td>
                       <td>
                        NAV1 : neuron navigator 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        AR
                       </td>
                       <td>
                        367
                       </td>
                       <td>
                        AR : androgen receptor
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DENND4C
                       </td>
                       <td>
                        55667
                       </td>
                       <td>
                        DENND4C : DENN/MADD domain containing 4C
                       </td>
                      </tr>
                      <tr>
                       <td>
                        IRS2
                       </td>
                       <td>
                        8660
                       </td>
                       <td>
                        IRS2 : insulin receptor substrate 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        LCOR
                       </td>
                       <td>
                        84458
                       </td>
                       <td>
                        LCOR : ligand dependent nuclear receptor corepressor
                       </td>
                      </tr>
                      <tr>
                       <td>
                        MAP3K2
                       </td>
                       <td>
                        10746
                       </td>
                       <td>
                        MAP3K2 : mitogen-activated protein kinase kinase kinase 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RAPGEF6
                       </td>
                       <td>
                        51735
                       </td>
                       <td>
                        RAPGEF6 : Rap guanine nucleotide exchange factor (GEF) 6
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ATRX
                       </td>
                       <td>
                        546
                       </td>
                       <td>
                        ATRX : alpha thalassemia/mental retardation syndrome X-linked
                       </td>
                      </tr>
                      <tr>
                       <td>
                        KIF1B
                       </td>
                       <td>
                        23095
                       </td>
                       <td>
                        KIF1B : kinesin family member 1B
                       </td>
                      </tr>
                      <tr>
                       <td>
                        BCL6
                       </td>
                       <td>
                        604
                       </td>
                       <td>
                        BCL6 : B-cell CLL/lymphoma 6
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SHKBP1
                       </td>
                       <td>
                        92799
                       </td>
                       <td>
                        SHKBP1 : SH3KBP1 binding protein 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CAMK4
                       </td>
                       <td>
                        814
                       </td>
                       <td>
                        CAMK4 : calcium/calmodulin-dependent protein kinase IV
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CBL
                       </td>
                       <td>
                        867
                       </td>
                       <td>
                        CBL : Cbl proto-oncogene, E3 ubiquitin protein ligase
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CDKN1A
                       </td>
                       <td>
                        1026
                       </td>
                       <td>
                        CDKN1A : cyclin-dependent kinase inhibitor 1A (p21, Cip1)
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ZMYND8
                       </td>
                       <td>
                        23613
                       </td>
                       <td>
                        ZMYND8 : zinc finger, MYND-type containing 8
                       </td>
                      </tr>
                      <tr>
                       <td>
                        TRIP11
                       </td>
                       <td>
                        9321
                       </td>
                       <td>
                        TRIP11 : thyroid hormone receptor interactor 11
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DNAJB5
                       </td>
                       <td>
                        25822
                       </td>
                       <td>
                        DNAJB5 : DnaJ (Hsp40) homolog, subfamily B, member 5
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RASAL2
                       </td>
                       <td>
                        9462
                       </td>
                       <td>
                        RASAL2 : RAS protein activator like 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RICTOR
                       </td>
                       <td>
                        253260
                       </td>
                       <td>
                        RICTOR : RPTOR independent companion of MTOR, complex 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        NCOR2
                       </td>
                       <td>
                        9612
                       </td>
                       <td>
                        NCOR2 : nuclear receptor corepressor 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        TANC2
                       </td>
                       <td>
                        26115
                       </td>
                       <td>
                        TANC2 : tetratricopeptide repeat, ankyrin repeat and coiled-coil containing 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ZBTB16
                       </td>
                       <td>
                        7704
                       </td>
                       <td>
                        ZBTB16 : zinc finger and BTB domain containing 16
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RANBP2
                       </td>
                       <td>
                        5903
                       </td>
                       <td>
                        RANBP2 : RAN binding protein 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        REV3L
                       </td>
                       <td>
                        5980
                       </td>
                       <td>
                        REV3L : REV3-like, polymerase (DNA directed), zeta, catalytic subunit
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CCDC6
                       </td>
                       <td>
                        8030
                       </td>
                       <td>
                        CCDC6 : coiled-coil domain containing 6
                       </td>
                      </tr>
                      <tr>
                       <td>
                        LMO7
                       </td>
                       <td>
                        4008
                       </td>
                       <td>
                        LMO7 : LIM domain 7
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <p dir="auto">
                     HOME/gene set analysis/over-representation analysis/enriched pathway-based sets/Network neighborhood-based entity sets (NESTs)/2-next neighbor/CaM1,CaM3,CaM2 set center
                    </p>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        user-provided identifier
                       </th>
                       <th>
                        entrez-gene ID
                       </th>
                       <th>
                        entrez-gene name
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        NRIP1
                       </td>
                       <td>
                        8204
                       </td>
                       <td>
                        NRIP1 : nuclear receptor interacting protein 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        UBN2
                       </td>
                       <td>
                        254048
                       </td>
                       <td>
                        UBN2 : ubinuclein 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ESYT2
                       </td>
                       <td>
                        57488
                       </td>
                       <td>
                        ESYT2 : extended synaptotagmin-like protein 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ADRA1A
                       </td>
                       <td>
                        148
                       </td>
                       <td>
                        ADRA1A : adrenoceptor alpha 1A
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ITGA10
                       </td>
                       <td>
                        8515
                       </td>
                       <td>
                        ITGA10 : integrin, alpha 10
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ARHGAP5
                       </td>
                       <td>
                        394
                       </td>
                       <td>
                        ARHGAP5 : Rho GTPase activating protein 5
                       </td>
                      </tr>
                      <tr>
                       <td>
                        IRS2
                       </td>
                       <td>
                        8660
                       </td>
                       <td>
                        IRS2 : insulin receptor substrate 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        AVP
                       </td>
                       <td>
                        551
                       </td>
                       <td>
                        AVP : arginine vasopressin
                       </td>
                      </tr>
                      <tr>
                       <td>
                        BCL6
                       </td>
                       <td>
                        604
                       </td>
                       <td>
                        BCL6 : B-cell CLL/lymphoma 6
                       </td>
                      </tr>
                      <tr>
                       <td>
                        BDNF
                       </td>
                       <td>
                        627
                       </td>
                       <td>
                        BDNF : brain-derived neurotrophic factor
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DST
                       </td>
                       <td>
                        667
                       </td>
                       <td>
                        DST : dystonin
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CAMK4
                       </td>
                       <td>
                        814
                       </td>
                       <td>
                        CAMK4 : calcium/calmodulin-dependent protein kinase IV
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DLGAP2
                       </td>
                       <td>
                        9228
                       </td>
                       <td>
                        DLGAP2 : discs, large (Drosophila) homolog-associated protein 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RASAL2
                       </td>
                       <td>
                        9462
                       </td>
                       <td>
                        RASAL2 : RAS protein activator like 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        COL19A1
                       </td>
                       <td>
                        1310
                       </td>
                       <td>
                        COL19A1 : collagen, type XIX, alpha 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CLOCK
                       </td>
                       <td>
                        9575
                       </td>
                       <td>
                        CLOCK : clock circadian regulator
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ARC
                       </td>
                       <td>
                        23237
                       </td>
                       <td>
                        ARC : activity-regulated cytoskeleton-associated protein
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DCC
                       </td>
                       <td>
                        1630
                       </td>
                       <td>
                        DCC : DCC netrin 1 receptor
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DCN
                       </td>
                       <td>
                        1634
                       </td>
                       <td>
                        DCN : decorin
                       </td>
                      </tr>
                      <tr>
                       <td>
                        EGR1
                       </td>
                       <td>
                        1958
                       </td>
                       <td>
                        EGR1 : early growth response 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ERBB4
                       </td>
                       <td>
                        2066
                       </td>
                       <td>
                        ERBB4 : erb-b2 receptor tyrosine kinase 4
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GABRB2
                       </td>
                       <td>
                        2561
                       </td>
                       <td>
                        GABRB2 : gamma-aminobutyric acid (GABA) A receptor, beta 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RAPGEF6
                       </td>
                       <td>
                        51735
                       </td>
                       <td>
                        RAPGEF6 : Rap guanine nucleotide exchange factor (GEF) 6
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RSF1
                       </td>
                       <td>
                        51773
                       </td>
                       <td>
                        RSF1 : remodeling and spacing factor 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SHKBP1
                       </td>
                       <td>
                        92799
                       </td>
                       <td>
                        SHKBP1 : SH3KBP1 binding protein 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ST6GAL2
                       </td>
                       <td>
                        84620
                       </td>
                       <td>
                        ST6GAL2 : ST6 beta-galactosamide alpha-2,6-sialyltranferase 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DUSP4
                       </td>
                       <td>
                        1846
                       </td>
                       <td>
                        DUSP4 : dual specificity phosphatase 4
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GRIN2A
                       </td>
                       <td>
                        2903
                       </td>
                       <td>
                        GRIN2A : glutamate receptor, ionotropic, N-methyl D-aspartate 2A
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GRIN2B
                       </td>
                       <td>
                        2904
                       </td>
                       <td>
                        GRIN2B : glutamate receptor, ionotropic, N-methyl D-aspartate 2B
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GRM4
                       </td>
                       <td>
                        2914
                       </td>
                       <td>
                        GRM4 : glutamate receptor, metabotropic 4
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GRM5
                       </td>
                       <td>
                        2915
                       </td>
                       <td>
                        GRM5 : glutamate receptor, metabotropic 5
                       </td>
                      </tr>
                      <tr>
                       <td>
                        PTPRT
                       </td>
                       <td>
                        11122
                       </td>
                       <td>
                        PTPRT : protein tyrosine phosphatase, receptor type, T
                       </td>
                      </tr>
                      <tr>
                       <td>
                        HTT
                       </td>
                       <td>
                        3064
                       </td>
                       <td>
                        HTT : huntingtin
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SYNPO
                       </td>
                       <td>
                        11346
                       </td>
                       <td>
                        SYNPO : synaptopodin
                       </td>
                      </tr>
                      <tr>
                       <td>
                        NR4A1
                       </td>
                       <td>
                        3164
                       </td>
                       <td>
                        NR4A1 : nuclear receptor subfamily 4, group A, member 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GAL
                       </td>
                       <td>
                        51083
                       </td>
                       <td>
                        GAL : galanin/GMAP prepropeptide
                       </td>
                      </tr>
                      <tr>
                       <td>
                        INSR
                       </td>
                       <td>
                        3643
                       </td>
                       <td>
                        INSR : insulin receptor
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ITGA4
                       </td>
                       <td>
                        3676
                       </td>
                       <td>
                        ITGA4 : integrin, alpha 4 (antigen CD49D, alpha 4 subunit of VLA-4 receptor)
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ITGB8
                       </td>
                       <td>
                        3696
                       </td>
                       <td>
                        ITGB8 : integrin, beta 8
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ITPR1
                       </td>
                       <td>
                        3708
                       </td>
                       <td>
                        ITPR1 : inositol 1,4,5-trisphosphate receptor, type 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        KCNA2
                       </td>
                       <td>
                        3737
                       </td>
                       <td>
                        KCNA2 : potassium channel, voltage gated shaker related subfamily A, member 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        KCNA3
                       </td>
                       <td>
                        3738
                       </td>
                       <td>
                        KCNA3 : potassium channel, voltage gated shaker related subfamily A, member 3
                       </td>
                      </tr>
                      <tr>
                       <td>
                        KCNMA1
                       </td>
                       <td>
                        3778
                       </td>
                       <td>
                        KCNMA1 : potassium channel, calcium activated large conductance subfamily M alpha, member 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        KCNN3
                       </td>
                       <td>
                        3782
                       </td>
                       <td>
                        KCNN3 : potassium channel, calcium activated intermediate/small conductance subfamily N alpha, member 3
                       </td>
                      </tr>
                      <tr>
                       <td>
                        LAMC2
                       </td>
                       <td>
                        3918
                       </td>
                       <td>
                        LAMC2 : laminin, gamma 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        LRP6
                       </td>
                       <td>
                        4040
                       </td>
                       <td>
                        LRP6 : low density lipoprotein receptor-related protein 6
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CD46
                       </td>
                       <td>
                        4179
                       </td>
                       <td>
                        CD46 : CD46 molecule, complement regulatory protein
                       </td>
                      </tr>
                      <tr>
                       <td>
                        MECP2
                       </td>
                       <td>
                        4204
                       </td>
                       <td>
                        MECP2 : methyl CpG binding protein 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SHC3
                       </td>
                       <td>
                        53358
                       </td>
                       <td>
                        SHC3 : SHC (Src homology 2 domain containing) transforming protein 3
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DENND4A
                       </td>
                       <td>
                        10260
                       </td>
                       <td>
                        DENND4A : DENN/MADD domain containing 4A
                       </td>
                      </tr>
                      <tr>
                       <td>
                        HIPK2
                       </td>
                       <td>
                        28996
                       </td>
                       <td>
                        HIPK2 : homeodomain interacting protein kinase 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        NEFM
                       </td>
                       <td>
                        4741
                       </td>
                       <td>
                        NEFM : neurofilament, medium polypeptide
                       </td>
                      </tr>
                      <tr>
                       <td>
                        NRCAM
                       </td>
                       <td>
                        4897
                       </td>
                       <td>
                        NRCAM : neuronal cell adhesion molecule
                       </td>
                      </tr>
                      <tr>
                       <td>
                        NTS
                       </td>
                       <td>
                        4922
                       </td>
                       <td>
                        NTS : neurotensin
                       </td>
                      </tr>
                      <tr>
                       <td>
                        OPRD1
                       </td>
                       <td>
                        4985
                       </td>
                       <td>
                        OPRD1 : opioid receptor, delta 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        UHMK1
                       </td>
                       <td>
                        127933
                       </td>
                       <td>
                        UHMK1 : U2AF homology motif (UHM) kinase 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        PDYN
                       </td>
                       <td>
                        5173
                       </td>
                       <td>
                        PDYN : prodynorphin
                       </td>
                      </tr>
                      <tr>
                       <td>
                        PNOC
                       </td>
                       <td>
                        5368
                       </td>
                       <td>
                        PNOC : prepronociceptin
                       </td>
                      </tr>
                      <tr>
                       <td>
                        POMC
                       </td>
                       <td>
                        5443
                       </td>
                       <td>
                        POMC : proopiomelanocortin
                       </td>
                      </tr>
                      <tr>
                       <td>
                        PRKCD
                       </td>
                       <td>
                        5580
                       </td>
                       <td>
                        PRKCD : protein kinase C, delta
                       </td>
                      </tr>
                      <tr>
                       <td>
                        MAP2K6
                       </td>
                       <td>
                        5608
                       </td>
                       <td>
                        MAP2K6 : mitogen-activated protein kinase kinase 6
                       </td>
                      </tr>
                      <tr>
                       <td>
                        PTPRB
                       </td>
                       <td>
                        5787
                       </td>
                       <td>
                        PTPRB : protein tyrosine phosphatase, receptor type, B
                       </td>
                      </tr>
                      <tr>
                       <td>
                        PTPRC
                       </td>
                       <td>
                        5788
                       </td>
                       <td>
                        PTPRC : protein tyrosine phosphatase, receptor type, C
                       </td>
                      </tr>
                      <tr>
                       <td>
                        PTPRD
                       </td>
                       <td>
                        5789
                       </td>
                       <td>
                        PTPRD : protein tyrosine phosphatase, receptor type, D
                       </td>
                      </tr>
                      <tr>
                       <td>
                        HMBOX1
                       </td>
                       <td>
                        79618
                       </td>
                       <td>
                        HMBOX1 : homeobox containing 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ZMYM1
                       </td>
                       <td>
                        79830
                       </td>
                       <td>
                        ZMYM1 : zinc finger, MYM-type 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RPS6KA3
                       </td>
                       <td>
                        6197
                       </td>
                       <td>
                        RPS6KA3 : ribosomal protein S6 kinase, 90kDa, polypeptide 3
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RXRG
                       </td>
                       <td>
                        6258
                       </td>
                       <td>
                        RXRG : retinoid X receptor, gamma
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RYR2
                       </td>
                       <td>
                        6262
                       </td>
                       <td>
                        RYR2 : ryanodine receptor 2 (cardiac)
                       </td>
                      </tr>
                      <tr>
                       <td>
                        NAV1
                       </td>
                       <td>
                        89796
                       </td>
                       <td>
                        NAV1 : neuron navigator 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SDC4
                       </td>
                       <td>
                        6385
                       </td>
                       <td>
                        SDC4 : syndecan 4
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CNKSR2
                       </td>
                       <td>
                        22866
                       </td>
                       <td>
                        CNKSR2 : connector enhancer of kinase suppressor of Ras 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        DENND4C
                       </td>
                       <td>
                        55667
                       </td>
                       <td>
                        DENND4C : DENN/MADD domain containing 4C
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SNTB2
                       </td>
                       <td>
                        6645
                       </td>
                       <td>
                        SNTB2 : syntrophin, beta 2 (dystrophin-associated protein A1, 59kDa, basic component 2)
                       </td>
                      </tr>
                      <tr>
                       <td>
                        PEG10
                       </td>
                       <td>
                        23089
                       </td>
                       <td>
                        PEG10 : paternally expressed 10
                       </td>
                      </tr>
                      <tr>
                       <td>
                        KIF1B
                       </td>
                       <td>
                        23095
                       </td>
                       <td>
                        KIF1B : kinesin family member 1B
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SSTR2
                       </td>
                       <td>
                        6752
                       </td>
                       <td>
                        SSTR2 : somatostatin receptor 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        STRN
                       </td>
                       <td>
                        6801
                       </td>
                       <td>
                        STRN : striatin, calmodulin binding protein
                       </td>
                      </tr>
                      <tr>
                       <td>
                        TACR1
                       </td>
                       <td>
                        6869
                       </td>
                       <td>
                        TACR1 : tachykinin receptor 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        TACR3
                       </td>
                       <td>
                        6870
                       </td>
                       <td>
                        TACR3 : tachykinin receptor 3
                       </td>
                      </tr>
                      <tr>
                       <td>
                        MGA
                       </td>
                       <td>
                        23269
                       </td>
                       <td>
                        MGA : MGA, MAX dimerization protein
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SATB2
                       </td>
                       <td>
                        23314
                       </td>
                       <td>
                        SATB2 : SATB homeobox 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        MAP3K2
                       </td>
                       <td>
                        10746
                       </td>
                       <td>
                        MAP3K2 : mitogen-activated protein kinase kinase kinase 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        TRAF3
                       </td>
                       <td>
                        7187
                       </td>
                       <td>
                        TRAF3 : TNF receptor-associated factor 3
                       </td>
                      </tr>
                      <tr>
                       <td>
                        TRPC5
                       </td>
                       <td>
                        7224
                       </td>
                       <td>
                        TRPC5 : transient receptor potential cation channel, subfamily C, member 5
                       </td>
                      </tr>
                      <tr>
                       <td>
                        UTRN
                       </td>
                       <td>
                        7402
                       </td>
                       <td>
                        UTRN : utrophin
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CYP26B1
                       </td>
                       <td>
                        56603
                       </td>
                       <td>
                        CYP26B1 : cytochrome P450, family 26, subfamily B, polypeptide 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        RICTOR
                       </td>
                       <td>
                        253260
                       </td>
                       <td>
                        RICTOR : RPTOR independent companion of MTOR, complex 2
                       </td>
                      </tr>
                      <tr>
                       <td>
                        ZBTB16
                       </td>
                       <td>
                        7704
                       </td>
                       <td>
                        ZBTB16 : zinc finger and BTB domain containing 16
                       </td>
                      </tr>
                      <tr>
                       <td>
                        SESTD1
                       </td>
                       <td>
                        91404
                       </td>
                       <td>
                        SESTD1 : SEC14 and spectrin domains 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        COL12A1
                       </td>
                       <td>
                        1303
                       </td>
                       <td>
                        COL12A1 : collagen, type XII, alpha 1
                       </td>
                      </tr>
                      <tr>
                       <td>
                        NAV3
                       </td>
                       <td>
                        89795
                       </td>
                       <td>
                        NAV3 : neuron navigator 3
                       </td>
                      </tr>
                      <tr>
                       <td>
                        CCDC6
                       </td>
                       <td>
                        8030
                       </td>
                       <td>
                        CCDC6 : coiled-coil domain containing 6
                       </td>
                      </tr>
                      <tr>
                       <td>
                        GPR68
                       </td>
                       <td>
                        8111
                       </td>
                       <td>
                        GPR68 : G protein-coupled receptor 68
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <p dir="auto">
                     HOME/gene set analysis/over-representation analysis/enriched pathway-based sets/Network neighborhood-based entity sets (NESTs)/2-next neighbor/PSD-1 set center
                    </p>
                    <p dir="auto">
                     HOME/gene set analysis/over-representation analysis/enriched pathway-based sets/Network neighborhood-based entity sets (NESTs)/2-next neighbor/NCAM set center
                    </p>
                    <p dir="auto">
                     ###6. 5HMC ANALYSIS:PEAK CALLING:MACS
                     <br/>
                     Approach 1: call peaks for each sample against the input, then do complex analysis on the lotof it
MACS + peaksplitter to call subpeaks within complex regions
MACS/FC1
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_FC1_5hmC_3 --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC1.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam &amp;">
                     <pre class="notranslate"><code>/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_FC1_5hmC_3 --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC1.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     MACS/FC2
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_FC2_5hmC_MACS --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC2.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam &amp;">
                     <pre class="notranslate"><code>/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_FC2_5hmC_MACS --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC2.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     MACS/HC1
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_HC1_5hmC_MACS --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC1.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam &amp;">
                     <pre class="notranslate"><code>/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_HC1_5hmC_MACS --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC1.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     MACS/HC2
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_HC2_5hmC_MACS --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC2.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam &amp;">
                     <pre class="notranslate"><code>/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name fear_conditioning_HC2_5hmC_MACS --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC2.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     Approach 2: concatenate BAM files for the HC and for the FC --&gt; perform MACS as HC against FC as input (so get peaks where HC is way more enriched than FC); and as FC against HC as input (so get peaks where FC is way more enriched than HC)
                    </p>
                    <p dir="auto">
                     combine BAMs MACS
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="
/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name 5hmC_DecInFC --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/HC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/FC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam &amp;

/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name 5hmC_IncInFC --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/FC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/HC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam &amp;
">
                     <pre class="notranslate"><code>
/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name 5hmC_DecInFC --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/HC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/FC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam &amp;

/home/ssharma/bin/MACS-1.4.2/bin/macs14 --name 5hmC_IncInFC --format BAM --gsize mm --call-subpeaks --bdg --treatment /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/FC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam --control /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/HC_5hmC_ALLreps_Aligned.sortedByCoord.out.bam &amp;

</code></pre>
                    </div>
                    <p dir="auto">
                     ###7. PARCE OUT REGIONS OF 5HMC, EXPRESSION, AND ISOFORM REGULATION:BEDTOOLS
                    </p>
                    <p dir="auto">
                     Calling DhMRs common to both sequencing experiments. Save the sequence JUST where two peaks overlap (no extra peak space unique to one sample or the other). Also create a "Peakheight" file that contains the entries that are overlapped; allows downstream analysis of relative 5hmC changes
                    </p>
                    <p dir="auto">
                     Note: needed to remove the header line -- bedtools was unable to recognize it: Chromosome      Start   End     Height  SummitPosition
                    </p>
                    <p dir="auto">
                     Common Homecage peaks
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_HC1_5hmC_MACS/fear_conditioning_HC1_5hmC_MACS_peaks.subpeaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_HC2_5hmC_MACS/fear_conditioning_HC2_5hmC_MACS_peaks.subpeaks.bed -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt &gt;HC1_HC2_BTIntersect.peaks.subpeaks.bed


Common homecage peaks; retain all the fields, so that peak height can be kept for downstream analyses">
                     <pre class="notranslate"><code>bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_HC1_5hmC_MACS/fear_conditioning_HC1_5hmC_MACS_peaks.subpeaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_HC2_5hmC_MACS/fear_conditioning_HC2_5hmC_MACS_peaks.subpeaks.bed -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt &gt;HC1_HC2_BTIntersect.peaks.subpeaks.bed


Common homecage peaks; retain all the fields, so that peak height can be kept for downstream analyses
</code></pre>
                    </div>
                    <p dir="auto">
                     bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_HC1_5hmC_MACS/fear_conditioning_HC1_5hmC_MACS_peaks.subpeaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_HC2_5hmC_MACS/fear_conditioning_HC2_5hmC_MACS_peaks.subpeaks.bed -bed -sorted -wa -wb -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt &gt;HC1_HC2_BTIntersect.peaks.subpeaks_PeakHeight.bed
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Common fear conditioning peaks">
                     <pre class="notranslate"><code>Common fear conditioning peaks
</code></pre>
                    </div>
                    <p dir="auto">
                     bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_FC1_5hmC_MACS/fear_conditioning_FC1_5hmC_3_peaks.subpeaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_FC2_5hmC_MACS/fear_conditioning_FC2_5hmC_MACS_peaks.subpeaks.bed -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt &gt;FC1_FC2_BTIntersect.peaks.subpeaks.bed
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Common fear conditioning peaks; retain all the fields, so that peak height can be kept for downstream analyses">
                     <pre class="notranslate"><code>Common fear conditioning peaks; retain all the fields, so that peak height can be kept for downstream analyses
</code></pre>
                    </div>
                    <p dir="auto">
                     bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_FC1_5hmC_MACS/fear_conditioning_FC1_5hmC_3_peaks.subpeaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/fear_conditioning_FC2_5hmC_MACS/fear_conditioning_FC2_5hmC_MACS_peaks.subpeaks.bed -bed -sorted -wa -wb -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt &gt;FC1_FC2_BTIntersect.peaks.subpeaks_PeakHeight.bed
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='
R Script to take the average of columns 4 and 9 from the HC1_HC2_BTIntersect.peaks.subpeaks_PeakHeight.bed file and adds it as a new column to the HC1_HC2_BTIntersect.peaks.subpeaks.bed file (ditto for FC files). End up with a BED file containing DhMRs + the average peak height from both datasets at this locus.
```{r}
FC_peaks = read.table("FC1_FC2_BTIntersect.peaks.subpeaks.bed")
FC_heights = read.table("FC1_FC2_BTIntersect.peaks.subpeaks_PeakHeight.bed")
FC_avg_heights = (FC_heights$V4+FC_heights$V9)/2
FC_peaks$FC_avg_heights = FC_avg_heights
write.table(FC_peaks, sep = "\t", col.names = F, row.names = F, quote = F, file = "/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed")

HC_peaks = read.table("HC1_HC2_BTIntersect.peaks.subpeaks.bed")
HC_heights = read.table("HC1_HC2_BTIntersect.peaks.subpeaks_PeakHeight.bed")
HC_avg_heights = (HC_heights$V4+HC_heights$V9)/2
HC_peaks$HC_avg_heights = HC_avg_heights
write.table(HC_peaks, sep = "\t", col.names = F, row.names = F, quote = F, file = "/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed")'>
                     <pre class="notranslate"><code>
R Script to take the average of columns 4 and 9 from the HC1_HC2_BTIntersect.peaks.subpeaks_PeakHeight.bed file and adds it as a new column to the HC1_HC2_BTIntersect.peaks.subpeaks.bed file (ditto for FC files). End up with a BED file containing DhMRs + the average peak height from both datasets at this locus.
```{r}
FC_peaks = read.table("FC1_FC2_BTIntersect.peaks.subpeaks.bed")
FC_heights = read.table("FC1_FC2_BTIntersect.peaks.subpeaks_PeakHeight.bed")
FC_avg_heights = (FC_heights$V4+FC_heights$V9)/2
FC_peaks$FC_avg_heights = FC_avg_heights
write.table(FC_peaks, sep = "\t", col.names = F, row.names = F, quote = F, file = "/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed")

HC_peaks = read.table("HC1_HC2_BTIntersect.peaks.subpeaks.bed")
HC_heights = read.table("HC1_HC2_BTIntersect.peaks.subpeaks_PeakHeight.bed")
HC_avg_heights = (HC_heights$V4+HC_heights$V9)/2
HC_peaks$HC_avg_heights = HC_avg_heights
write.table(HC_peaks, sep = "\t", col.names = F, row.names = F, quote = F, file = "/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed")
</code></pre>
                    </div>
                    <p dir="auto">
                     FC_5hmC_Inc: Peaks that appear in fear conditioning; no peak at this loci in HC
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -v -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt &gt;FC_5hmC_Inc.bed">
                     <pre class="notranslate"><code>bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -v -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt &gt;FC_5hmC_Inc.bed
</code></pre>
                    </div>
                    <p dir="auto">
                     FC_5hmC_Dec: Peaks disappear in fear conditioning; peak at this locus in HC dissapears in FC
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed  -v -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt &gt;FC_5hmC_Dec.bed">
                     <pre class="notranslate"><code>bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed  -v -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt &gt;FC_5hmC_Dec.bed
</code></pre>
                    </div>
                    <p dir="auto">
                     Remove tmp files
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="rm FC1_FC2_BTIntersect.peaks.subpeaks.bed FC1_FC2_BTIntersect.peaks.subpeaks_PeakHeight.bed FC_5hmC_Dec_2.bed HC1_HC2_BTIntersect.peaks.subpeaks.bed FC_5hmC_Inc_2.bed HC1_HC2_BTIntersect.peaks.subpeaks_PeakHeight.bed">
                     <pre class="notranslate"><code>rm FC1_FC2_BTIntersect.peaks.subpeaks.bed FC1_FC2_BTIntersect.peaks.subpeaks_PeakHeight.bed FC_5hmC_Dec_2.bed HC1_HC2_BTIntersect.peaks.subpeaks.bed FC_5hmC_Inc_2.bed HC1_HC2_BTIntersect.peaks.subpeaks_PeakHeight.bed
</code></pre>
                    </div>
                    <p dir="auto">
                     FC_5hmC_common_Inc: Peak is present in both HC and FC samples; but peak is smaller in HC than FC (cutoff = ??)
bedtools/FC_HC_Common.bed
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt &gt;FC_HC_Common.bed">
                     <pre class="notranslate"><code>bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -bed -sorted -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt &gt;FC_HC_Common.bed
</code></pre>
                    </div>
                    <p dir="auto">
                     bedtools/FC_HC_Common_PeakHeights.bed
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -bed -sorted -wa -wb -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt &gt;FC_HC_Common_PeakHeights.bed">
                     <pre class="notranslate"><code>bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_peaks.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_peaks.bed -bed -sorted -wa -wb -g /home/Shared/PengLab/iGenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/ChromInfo.txt &gt;FC_HC_Common_PeakHeights.bed
</code></pre>
                    </div>
                    <p dir="auto">
                     Concatenate files (to include intermediate changes in peak height)
-5hmC increased during FC
-5hmC decreased during FC
-all dynamic 5hmC loci during FC
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cat FC_HC_Common_FCgreaterHC.bed FC_5hmC_Inc.bed &gt; FC_5hmC_Inc_increment.bed
cat FC_HC_Common_HCgreaterFC.bed FC_5hmC_Dec.bed &gt; FC_5hmC_Dec_increment.bed
cat HCvsFC_5hmC_Inc_increment.bed HCvsFC_5hmC_Dec_increment.bed &gt; HCvsFC_AllDynamic5hmC.bed">
                     <pre class="notranslate"><code>cat FC_HC_Common_FCgreaterHC.bed FC_5hmC_Inc.bed &gt; FC_5hmC_Inc_increment.bed
cat FC_HC_Common_HCgreaterFC.bed FC_5hmC_Dec.bed &gt; FC_5hmC_Dec_increment.bed
cat HCvsFC_5hmC_Inc_increment.bed HCvsFC_5hmC_Dec_increment.bed &gt; HCvsFC_AllDynamic5hmC.bed
</code></pre>
                    </div>
                    <p dir="auto">
                     FC greater than HC and FC less than HC, cutoff - peak height difference &gt; or &lt; 25
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content='FC_HC_common_peaks = read.table("FC_HC_Common.bed")
FC_HC_common_heights = read.table("FC_HC_Common_PeakHeights.bed")
FC_HC_common_peaks$FCminusHC &lt;- FC_HC_common_heights$V4 - FC_HC_common_heights$V10
FC_HC_common_peaks$HCminusFC &lt;- FC_HC_common_heights$V10 - FC_HC_common_heights$V4
FC_HC_Common_FCgreaterHC = subset(FC_HC_common_peaks, FCminusHC &gt;=25)
write.table(FC_HC_Common_FCgreaterHC, sep = "\t", col.names = F, row.names = F, quote = F, file = "/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_HC_Common_FCgreaterHC.bed")
FC_HC_Common_HCgreaterFC = subset(FC_HC_common_peaks, HCminusFC &gt;=25)
write.table(FC_HC_Common_HCgreaterFC, sep = "\t", col.names = F, row.names = F, quote = F, file = "/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_HC_Common_HCgreaterFC.bed")'>
                     <pre class="notranslate" lang="{r}"><code>FC_HC_common_peaks = read.table("FC_HC_Common.bed")
FC_HC_common_heights = read.table("FC_HC_Common_PeakHeights.bed")
FC_HC_common_peaks$FCminusHC &lt;- FC_HC_common_heights$V4 - FC_HC_common_heights$V10
FC_HC_common_peaks$HCminusFC &lt;- FC_HC_common_heights$V10 - FC_HC_common_heights$V4
FC_HC_Common_FCgreaterHC = subset(FC_HC_common_peaks, FCminusHC &gt;=25)
write.table(FC_HC_Common_FCgreaterHC, sep = "\t", col.names = F, row.names = F, quote = F, file = "/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_HC_Common_FCgreaterHC.bed")
FC_HC_Common_HCgreaterFC = subset(FC_HC_common_peaks, HCminusFC &gt;=25)
write.table(FC_HC_Common_HCgreaterFC, sep = "\t", col.names = F, row.names = F, quote = F, file = "/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_HC_Common_HCgreaterFC.bed")
</code></pre>
                    </div>
                    <p dir="auto">
                     Sorting Bedfiles
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bedtools sort -chrThenSizeA -i HCvsFC_RNAseq_wholegene_ALL.bed &gt;HCvsFC_RNAseq_wholegene_ALL_sorted.bed
5hmC-seq/all
bedtools sort -chrThenSizeA -i HCvsFC_AllDynamic5hmC.bed &gt;HCvsFC_AllDynamic5hmC_sorted.bed
5hmC-seq/upregulated
bedtools sort -chrThenSizeA -i HCvsFC_5hmC_Inc_increment.bed HCvsFC_5hmC_Inc_increment_sorted.bed
5hmC-seq/downregulated
bedtools sort -chrThenSizeA -i HCvsFC_5hmC_Dec_increment.bed &gt;HCvsFC_5hmC_Dec_increment_sorted.bed">
                     <pre class="notranslate"><code>bedtools sort -chrThenSizeA -i HCvsFC_RNAseq_wholegene_ALL.bed &gt;HCvsFC_RNAseq_wholegene_ALL_sorted.bed
5hmC-seq/all
bedtools sort -chrThenSizeA -i HCvsFC_AllDynamic5hmC.bed &gt;HCvsFC_AllDynamic5hmC_sorted.bed
5hmC-seq/upregulated
bedtools sort -chrThenSizeA -i HCvsFC_5hmC_Inc_increment.bed HCvsFC_5hmC_Inc_increment_sorted.bed
5hmC-seq/downregulated
bedtools sort -chrThenSizeA -i HCvsFC_5hmC_Dec_increment.bed &gt;HCvsFC_5hmC_Dec_increment_sorted.bed
</code></pre>
                    </div>
                    <p dir="auto">
                     MISCELLANEOUS
                    </p>
                    <p dir="auto">
                     bedtools/RNAseq_wholegene/5hmC_ALL
-wb option --&gt; write out the entirety of the 5hmC peak that overlaps with a 5hmC region; find any motifs even if they aren't precisely within body of gene
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb &gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed
#bedtools/RNAseq_wholegene/5hmC_Inc
bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb &gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed
#bedtools/RNAseq_wholegene/5hmC_Dec
bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb &gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed
#bedtools/RNAseq_wholegene_upregulated/5hmC_Dec
bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb &gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed
#bedtools/RNAseq_wholegene_downregulated/5hmC_Dec
bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb &gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed
#bedtools/RNAseq_wholegene/5hmC_Dec
bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb &gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed
#bedtools/RNAseq_wholegene/5hmC_Dec
bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb &gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed">
                     <pre class="notranslate"><code>bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb &gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed
#bedtools/RNAseq_wholegene/5hmC_Inc
bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb &gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed
#bedtools/RNAseq_wholegene/5hmC_Dec
bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb &gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed
#bedtools/RNAseq_wholegene_upregulated/5hmC_Dec
bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb &gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed
#bedtools/RNAseq_wholegene_downregulated/5hmC_Dec
bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb &gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed
#bedtools/RNAseq_wholegene/5hmC_Dec
bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb &gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed
#bedtools/RNAseq_wholegene/5hmC_Dec
bedtools intersect -a /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL_sorted.bed -b /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_AllDynamic5hmC_sorted.bed -v -bed -wb &gt;HCvsFC_RNAseq_wholegene_5hmC_all.bed
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="MISCELLANEOUS IDEAS
 -bedtools/RNAseq_all/5hmCseq_genes_TSS_all
 -bedtools/RNAseq_all/5hmCseq_genes_TSS_all
 -bedtools/RNAseq_all/5hmCseq_genes_TSS_all
 -bedtools/RNAseq_splicing_all/5hmCseq_exons_all
 -bedtools/RNAseq_splicing_all/5hmCseq_introns_all
 -bedtools/RNAseq_splicing_all/5hmCseq_exon-intron_all
 -bedtools/RNAseq_promoters/5hmC_Inc
 -bedtools/RNAseq_promoters/5hmC_Dec
 -bedtools/RNAseq_promoters_upregulated/5hmC_Inc
 -bedtools/RNAseq_promoters_downregulated/5hmC_Dec
 -bedtools/CTCFbindingsites/5hmC_all
 -bedtools/CTCFbindingsites/5hmC_inc
 -bedtools/CTCFbindingsites/5hmC_dec
 -bedtools/CTCFbindingsites/5hmC_Inc/annotation
 -bedtools/lncRNAs_all/5hmc_all
 -bedtools/lncRNAs_/5hmc_all
 -CTCF binding sites with dynamic 5hmC in FC
 -whole genome CTCF binding sites from Corces Lab
 -Enhancers with dynamic 5hmC in FC
 -FANTOM identified enhancers
 -Neural specific enhancers
 -Other Datasets? Hi-seq? TADs or mini-TADs, or CTFCF binding sites influential at these positions..proxy for conformational change

Other interesting coordinate sets to generate
 -RNAseq and lncRNA
 -RNAseq and microRNAs
 -RNAseq and TRAPSeq
 -5hmCseq and enhancers
 -5hmCseq and insulators
">
                     <pre class="notranslate"><code>MISCELLANEOUS IDEAS
 -bedtools/RNAseq_all/5hmCseq_genes_TSS_all
 -bedtools/RNAseq_all/5hmCseq_genes_TSS_all
 -bedtools/RNAseq_all/5hmCseq_genes_TSS_all
 -bedtools/RNAseq_splicing_all/5hmCseq_exons_all
 -bedtools/RNAseq_splicing_all/5hmCseq_introns_all
 -bedtools/RNAseq_splicing_all/5hmCseq_exon-intron_all
 -bedtools/RNAseq_promoters/5hmC_Inc
 -bedtools/RNAseq_promoters/5hmC_Dec
 -bedtools/RNAseq_promoters_upregulated/5hmC_Inc
 -bedtools/RNAseq_promoters_downregulated/5hmC_Dec
 -bedtools/CTCFbindingsites/5hmC_all
 -bedtools/CTCFbindingsites/5hmC_inc
 -bedtools/CTCFbindingsites/5hmC_dec
 -bedtools/CTCFbindingsites/5hmC_Inc/annotation
 -bedtools/lncRNAs_all/5hmc_all
 -bedtools/lncRNAs_/5hmc_all
 -CTCF binding sites with dynamic 5hmC in FC
 -whole genome CTCF binding sites from Corces Lab
 -Enhancers with dynamic 5hmC in FC
 -FANTOM identified enhancers
 -Neural specific enhancers
 -Other Datasets? Hi-seq? TADs or mini-TADs, or CTFCF binding sites influential at these positions..proxy for conformational change

Other interesting coordinate sets to generate
 -RNAseq and lncRNA
 -RNAseq and microRNAs
 -RNAseq and TRAPSeq
 -5hmCseq and enhancers
 -5hmCseq and insulators

</code></pre>
                    </div>
                    <p dir="auto">
                     ###8. ANNOTATION ANALYSIS:HOMER
                    </p>
                    <p dir="auto">
                     -HCvsFC_5hmCseq_Inc
-HCvsFC_5hmCseq_Dec
-HCvsFC_5hmCseq_ALL
-HCvsFC RNAseq
-HCvsFC 5hmC &amp; RNAseq overlap
-HCvsFC TRAPseq: unique and distinct from regular RNAseq
                    </p>
                    <p dir="auto">
                     Peak sets: HCvsFC_5hmC_Dec_increment.bed, HCvsFC_5hmC_Inc_increment.bed, HCvsFC_AllDHMR.bed
                    </p>
                    <p dir="auto">
                     Annotation of HCvsFC_AllDHMR.bed (using UCSC GTF used to align the files in the first place
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_AllDHMR.bed mm10 -annStats HCvsFC_AllDHMR_HomerAnnStats.txt &gt;HCvsFC_AllDHMR_HomerAnnotations_2.txt">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_AllDHMR.bed mm10 -annStats HCvsFC_AllDHMR_HomerAnnStats.txt &gt;HCvsFC_AllDHMR_HomerAnnotations_2.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     Annotation of peaks called by concatenating replicate fasta files and using either HC or FC as the input
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_Dynamic5hmC_ALL.bed mm10 -annStats HCvsFC_ALLDynamic5hmC_HomerAnnStats.txt &gt;HCvsFC_ALLDynamic5hmC_HomerAnnotations.txt
">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_Dynamic5hmC_ALL.bed mm10 -annStats HCvsFC_ALLDynamic5hmC_HomerAnnStats.txt &gt;HCvsFC_ALLDynamic5hmC_HomerAnnotations.txt

</code></pre>
                    </div>
                    <p dir="auto">
                     Annotation of Smarca4 binding sites:
Note: use homer annotation files in this folder w/ all of Bing's stuff in there
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/smarca4_forbrain_chipseq.bed mm9 -annStats smarca4_forebrain_HomerAnnStats.txt &gt;smarca4_forebrain_HomerAnnotations.txt">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/smarca4_forbrain_chipseq.bed mm9 -annStats smarca4_forebrain_HomerAnnStats.txt &gt;smarca4_forebrain_HomerAnnotations.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     Annotate RNAseq Genes in Homer
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL.bed mm10  &gt;HCvsFC_RNAseq_wholegene_HomerAnnotations.txt">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_ALL.bed mm10  &gt;HCvsFC_RNAseq_wholegene_HomerAnnotations.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     python script
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="~/bin/ExtractGeneNames_HomerOutput_Down.py HCvsFC_AllDHMR_HomerAnnotations.txt HCvsFC_AllDHMR_HomerAnnotations_Craig.txt">
                     <pre class="notranslate"><code>~/bin/ExtractGeneNames_HomerOutput_Down.py HCvsFC_AllDHMR_HomerAnnotations.txt HCvsFC_AllDHMR_HomerAnnotations_Craig.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     ORDER OF ANNOTATION ENRICHMENT CHANGE (HC &gt; FC)
intron&gt;exon&gt;intergenic&gt;LTR~&gt;promoter
                    </p>
                    <p dir="auto">
                     Python Script for Homer Annotations
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="import os
import fnmatch
import pybedtools
 
 
InputGeneNames = open('HCvsFC_AllDHMR_HomerAnnotations.txt','r')
GeneFileGTF = open('/home/byao/homer/data/accession/mouse2gene.tsv','r')
GeneList = {}
AccessionNumbers = {}
 
for line in GeneFileGTF:
	line = line.strip()
	try:
		RefSeqID, Unknown, HsID, Accession, EnsemblID, Unknown2, GeneID = line.split('\t')
		GeneList[str(RefSeqID)] = GeneID
	except:
		pass
 
 
 
 
for line in InputGeneNames:
    line = line.strip()
    PeakID, Chr, Start, End, Strand, PeakScore, Focus, Ratio_Region, Remainder = line.split('\t',8)
    try:
        Region, Remainder = Ratio_Region.split('(')
        try:
            Accession, Remainder = Remainder.split(',')
            OutputLine = str(Chr) + '\t' + str(Start) + '\t' + str(End) + '\t' + str(Ratio_Region)
            AccessionNumbers[str(Accession)] = OutputLine
        except:
            Accession, Remainder = Remainder.split(')')
            OutputLine = str(Chr) + '\t' + str(Start) + '\t' + str(End) + '\t' + str(Ratio_Region)
            AccessionNumbers[str(Accession)] = OutputLine
    except:
        pass
 
 
for AccessionNumber in AccessionNumbers.keys():
    try:
        print str(AccessionNumbers[str(AccessionNumber)]) + '\t' + str(GeneList[str(AccessionNumber)])
    except:
        print str(AccessionNumber) + ' is missing'

">
                     <pre class="notranslate"><code>import os
import fnmatch
import pybedtools
 
 
InputGeneNames = open('HCvsFC_AllDHMR_HomerAnnotations.txt','r')
GeneFileGTF = open('/home/byao/homer/data/accession/mouse2gene.tsv','r')
GeneList = {}
AccessionNumbers = {}
 
for line in GeneFileGTF:
	line = line.strip()
	try:
		RefSeqID, Unknown, HsID, Accession, EnsemblID, Unknown2, GeneID = line.split('\t')
		GeneList[str(RefSeqID)] = GeneID
	except:
		pass
 
 
 
 
for line in InputGeneNames:
    line = line.strip()
    PeakID, Chr, Start, End, Strand, PeakScore, Focus, Ratio_Region, Remainder = line.split('\t',8)
    try:
        Region, Remainder = Ratio_Region.split('(')
        try:
            Accession, Remainder = Remainder.split(',')
            OutputLine = str(Chr) + '\t' + str(Start) + '\t' + str(End) + '\t' + str(Ratio_Region)
            AccessionNumbers[str(Accession)] = OutputLine
        except:
            Accession, Remainder = Remainder.split(')')
            OutputLine = str(Chr) + '\t' + str(Start) + '\t' + str(End) + '\t' + str(Ratio_Region)
            AccessionNumbers[str(Accession)] = OutputLine
    except:
        pass
 
 
for AccessionNumber in AccessionNumbers.keys():
    try:
        print str(AccessionNumbers[str(AccessionNumber)]) + '\t' + str(GeneList[str(AccessionNumber)])
    except:
        print str(AccessionNumber) + ' is missing'


</code></pre>
                    </div>
                    <p dir="auto">
                     ###9. MOTIF ANALYSIS:HOMER:MOTIF ID AND LOCALIZATION
                    </p>
                    <p dir="auto">
                     <em>
                      Background
                     </em>
                     Files are critical - they are how homer decides which TF sites are overrepresented
wholegene - mm10 all genes
exons - mm10 all exons
introns - mm10 all introns
5hmC regions - random sequences from the genome
                    </p>
                    <p dir="auto">
                     Testing out different -size parameters (avg size of the DNA fragments in the BED file - input into the findmotifs script)
size 2000
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_2k/ -size 2000 -mis 1 -S 10 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_2k/ -size 2000 -mis 1 -S 10 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     size 100000
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_100k/ -size 100000 -mis 1 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_100k/ -size 100000 -mis 1 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     size 10000
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_10k/ -size 10000 -mis 1 -S 10 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_10k/ -size 10000 -mis 1 -S 10 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     size 50000
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_50k/ -size 50000 -mis 1 -S 10 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_expandedcoordinates.bed mm10 HCvsFC_RNAseq_wholegene_motifs_50k/ -size 50000 -mis 1 -S 10 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      Conclusion about -size parameter:
                     </em>
                     having a size closer to the real average length of the regions in the bedfile makes for more statistically significant results, but much longer run time. 100k size failed to finish.
                    </p>
                    <p dir="auto">
                     Awk script to determine average size of regions in a bed file
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cat CTCFPeaksClustered.bed | awk -F'\t' 'BEGIN{SUM=0}{ SUM+=$3-$2 }END{print SUM}'">
                     <pre class="notranslate"><code>cat CTCFPeaksClustered.bed | awk -F'\t' 'BEGIN{SUM=0}{ SUM+=$3-$2 }END{print SUM}'
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="_Average size of each region in bed file:_
 -RNAseq Introns: 15000 (avg was 8800)
 -RNAseq exons: 3000 (avg 325)
 -CTCFPeaksClustered = 500 (avg 362)
Overshooting may be better than under">
                     <pre class="notranslate"><code>_Average size of each region in bed file:_
 -RNAseq Introns: 15000 (avg was 8800)
 -RNAseq exons: 3000 (avg 325)
 -CTCFPeaksClustered = 500 (avg 362)
Overshooting may be better than under
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      Motif Finding:
                     </em>
                    </p>
                    <p dir="auto">
                     <em>
                      RNAseq, whole gene, upregulated
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_upregulated.bed mm10 HCvsFC_RNAseq_wholegene_upregulated_motifs/ -size 50000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_upregulated.bed mm10 HCvsFC_RNAseq_wholegene_upregulated_motifs/ -size 50000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      RNAseq, whole gene, downregulated
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_downregulated.bed mm10 HCvsFC_RNAseq_wholegene_downregulated_motifs/ -size 50000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_wholegene_downregulated.bed mm10 HCvsFC_RNAseq_wholegene_downregulated_motifs/ -size 50000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allgenes_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      RNAseq, promoters, all differentially expressed (DE)
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_promoters_ALL.bed mm10 HCvsFC_RNAseq_promoters_motifs/ -size 2000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allpromoters_-2000_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_promoters_ALL.bed mm10 HCvsFC_RNAseq_promoters_motifs/ -size 2000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allpromoters_-2000_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      RNAseq, promoters, upregulated (DE)
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_promoter_upregulated.bed mm10 HCvsFC_RNAseq_promoter_upregulated_motifs/ -size 2000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allpromoters_-2000_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_promoter_upregulated.bed mm10 HCvsFC_RNAseq_promoter_upregulated_motifs/ -size 2000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allpromoters_-2000_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      RNAseq, promoters, downregulated
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_promoters_downregulated.bed mm10 HCvsFC_RNAseq_promoter_downregulated_motifs/ -size 2000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allpromoters_-2000_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_promoters_downregulated.bed mm10 HCvsFC_RNAseq_promoter_downregulated_motifs/ -size 2000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allpromoters_-2000_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      RNAseq, introns, all differentially expressed (DE)
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_introns_ALL.bed mm10 HCvsFC_RNAseq_intron_motifs/ -size 15000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allintrons_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_introns_ALL.bed mm10 HCvsFC_RNAseq_intron_motifs/ -size 15000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allintrons_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      RNAseq, introns, upregulated
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_introns_upregulated.bed mm10 HCvsFC_RNAseq_intron_upregulated_motifs/ -size 15000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allintrons_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_introns_upregulated.bed mm10 HCvsFC_RNAseq_intron_upregulated_motifs/ -size 15000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allintrons_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      RNAseq, introns, downregulated
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_introns_downregulated.bed mm10 HCvsFC_RNAseq_intron_downregulated_motifs/ -size 15000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allintrons_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_introns_downregulated.bed mm10 HCvsFC_RNAseq_intron_downregulated_motifs/ -size 15000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allintrons_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      RNAseq, exons, all DE
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_exons_ALL.bed mm10 HCvsFC_RNAseq_exon_motifs/ -size 3000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allexons_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_exons_ALL.bed mm10 HCvsFC_RNAseq_exon_motifs/ -size 3000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allexons_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      RNAseq, exons, upregulated
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_exons_upregulated.bed mm10 HCvsFC_RNAseq_exon_upregulated_motifs/ -size 3000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allexons_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_exons_upregulated.bed mm10 HCvsFC_RNAseq_exon_upregulated_motifs/ -size 3000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allexons_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      RNAseq, exons, downregulated
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_exons_downregulated.bed mm10 HCvsFC_RNAseq_exon_downregulated_motifs/ -size 3000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allexons_background.bed -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/HCvsFC_RNAseq_exons_downregulated.bed mm10 HCvsFC_RNAseq_exon_downregulated_motifs/ -size 3000 -mis 2 -S 25 -p 25 -bg /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/background_files/mm10_allexons_background.bed -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      5hmC peaks, all DhMRs
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_AllDHMR.bed mm10 HCvsFC_5hmC_motifs/ -size 700 -mis 2 -S 25 -p 25 -h &amp;">
                     <pre class="notranslate" lang="_"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_AllDHMR.bed mm10 HCvsFC_5hmC_motifs/ -size 700 -mis 2 -S 25 -p 25 -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      5hmC peaks, 5hmC Increased
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_5hmC_Inc_increment.bed mm10 HCvsFC_5hmC_Inc_motifs/ -size 700 -mis 2 -S 25 -p 25 -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_5hmC_Inc_increment.bed mm10 HCvsFC_5hmC_Inc_motifs/ -size 700 -mis 2 -S 25 -p 25 -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      5hmC peaks, 5hmC Decreased
                     </em>
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_5hmC_Dec_increment.bed mm10 HCvsFC_5hmC_Dec_motifs/ -size 700 -mis 2 -S 25 -p 25 -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HCvsFC_5hmC_Dec_increment.bed mm10 HCvsFC_5hmC_Dec_motifs/ -size 700 -mis 2 -S 25 -p 25 -h &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     CTCFPeaksClustered: dataset from Victor Corces or all CTCF binding sites genome wide
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/CTCFPeaksClustered.bed mm10 CTCFPeaksClustered_motifs/ -size 700 -mis 2 -S 25 -p 25 -h &amp;">
                     <pre class="notranslate"><code>findMotifsGenome.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/CTCFPeaksClustered.bed mm10 CTCFPeaksClustered_motifs/ -size 700 -mis 2 -S 25 -p 25 -h &amp;
</code></pre>
                    </div>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="Other Analyses
 -FindMotifs/5hmC-Seq/FC_5hmC_Inc
 -FindMotifs/5hmC-Seq/FC_5hmC_Dec
 -FindMotifs/5hmC-Seq/FC_5hmC_Common_Inc
 -FindMotifs/5hmC-Seq/FC_5hmC_Common_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Dec
 -FindMotifs/5hmC-Seq_Enhancers/5hmC_Enhancers_overlap
 -FindMotifs/5hmC-Seq_CTCF/5hmC_CTCF_overlap
 -FindMotifs/CTCF Clustered Peaks

More Other analyses:
 -FindMotifs/5hmC-Seq/FC_5hmC_Inc
 -FindMotifs/5hmC-Seq/FC_5hmC_Dec
 -FindMotifs/5hmC-Seq/FC_5hmC_Common_Inc
 -FindMotifs/5hmC-Seq/FC_5hmC_Common_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Dec
 -FindMotifs/5hmC-Seq_Enhancers/5hmC_Enhancers_overlap
 -FindMotifs/5hmC-Seq_CTCF/5hmC_CTCF_overlap">
                     <pre class="notranslate"><code>Other Analyses
 -FindMotifs/5hmC-Seq/FC_5hmC_Inc
 -FindMotifs/5hmC-Seq/FC_5hmC_Dec
 -FindMotifs/5hmC-Seq/FC_5hmC_Common_Inc
 -FindMotifs/5hmC-Seq/FC_5hmC_Common_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Dec
 -FindMotifs/5hmC-Seq_Enhancers/5hmC_Enhancers_overlap
 -FindMotifs/5hmC-Seq_CTCF/5hmC_CTCF_overlap
 -FindMotifs/CTCF Clustered Peaks

More Other analyses:
 -FindMotifs/5hmC-Seq/FC_5hmC_Inc
 -FindMotifs/5hmC-Seq/FC_5hmC_Dec
 -FindMotifs/5hmC-Seq/FC_5hmC_Common_Inc
 -FindMotifs/5hmC-Seq/FC_5hmC_Common_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_wholegene_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_wholegene_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_promoter_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_promoter_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_exons_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_exons_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Inc_RNAseq_introns_Dec
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Inc
 -FindMotifs/5hmC-Seq_RNAseq/5hmC_Dec_RNAseq_introns_Dec
 -FindMotifs/5hmC-Seq_Enhancers/5hmC_Enhancers_overlap
 -FindMotifs/5hmC-Seq_CTCF/5hmC_CTCF_overlap
</code></pre>
                    </div>
                    <p dir="auto">
                     ###10. 5HMC ENRICHMENT ANALYSIS:HOMER:HISTOGRAMS AND SCATTERPLOTS: MOTIFS, PROMOTERS, EXONS, INTRONS, EXONS-INTRONS, CTCF, OTHER GENOMIC REGIONS (CPG ISLANDS, EXONS, LNCRNA, PSEUDOGENES, LOW COMPLEXITY REGIONS, SIMPLE REPEATS, SINES, SATELLITES
                    </p>
                    <p dir="auto">
                     <em>
                      <strong>
                       Workflow
                      </strong>
                     </em>
                     A) Create tag directories -&gt; containers with all of the bam file reads (from all replicates) in one container - used to pileup all the reads from one condition in order to get a comprehensive picture of the density of reads in particular loci
B) Create a .motif file using seq2profile, or download the .motif file from
                     <a href="http://homer.salk.edu/homer/custom.motifs" rel="nofollow">
                      http://homer.salk.edu/homer/custom.motifs
                     </a>
                     C) Predict sites where the TF will bind in a set of coordinates using annotatepeaks -m -mbed
D) Find intensity of epigenetic marks at predetermined set of motif binding sites (as from #2) using annotatepeaks -hist -d to feed in tag directories. Can do this with any predetermined set of coordinates (i.e. RNAseq exon boundaries, promoters, etc)
                    </p>
                    <p dir="auto">
                     <strong>
                      A) Create tag directories -&gt; containers with all of the bam file reads (from all replicates) in one container - used to pileup all the reads from one condition in order to get a comprehensive picture of the density of reads in particular loci
                     </strong>
                     Tag Directories of 5hmC BAM files to calculate enrichment of 5hmC at loci specified in BED format
makeTagDirectory  [options]  &lt;"2&gt;
                    </p>
                    <p dir="auto">
                     HC
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="makeTagDirectory HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC1.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC2.bam">
                     <pre class="notranslate"><code>makeTagDirectory HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC1.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_HC2.bam
</code></pre>
                    </div>
                    <p dir="auto">
                     FC
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="makeTagDirectory FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC1.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC2.bam &amp;">
                     <pre class="notranslate"><code>makeTagDirectory FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC1.bam /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_FC2.bam &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     Input
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="makeTagDirectory Input_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam &amp;">
                     <pre class="notranslate"><code>makeTagDirectory Input_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/data/FearConditioning_5hmC_Seq/STAR_Aligned_BAMs_UCSC/BAMs/Sample_InputHC1.bam &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      B) Create a .motif file using seq2profile, or download the .motif file from
                      <a href="http://homer.salk.edu/homer/custom.motifs" rel="nofollow">
                       http://homer.salk.edu/homer/custom.motifs
                      </a>
                     </strong>
                    </p>
                    <p dir="auto">
                     Obtaining .motif files: contain the consensus sequence and the frequency distribution of each nucleotide in the consensue
Two methods:
-use seq2profile to create .motif file form within Homer
-download pre-made consensus motif from Homer:
                     <a href="http://homer.salk.edu/homer/custom.motifs" rel="nofollow">
                      http://homer.salk.edu/homer/custom.motifs
                     </a>
                     -may be best option - seq2profile doesn't appear to produce the same profile, as is displayed on URL; also the .motif profiles form seq2profile appear much less complex
                    </p>
                    <p dir="auto">
                     TFs of interest and their consensus sequences:
                    </p>
                    <table>
                     <thead>
                      <tr>
                       <th>
                        Gene
                       </th>
                       <th>
                        Consensus Motif
                       </th>
                      </tr>
                     </thead>
                     <tbody>
                      <tr>
                       <td>
                        HIF-1b
                       </td>
                       <td>
                        RTACGTGC
                       </td>
                      </tr>
                      <tr>
                       <td>
                        c-Myc
                       </td>
                       <td>
                        VVCCACGTGG
                       </td>
                      </tr>
                      <tr>
                       <td>
                        n-Myc
                       </td>
                       <td>
                        VRCCACGTGG
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Maz
                       </td>
                       <td>
                        GGGGGGGG
                       </td>
                      </tr>
                      <tr>
                       <td>
                        Stat3
                       </td>
                       <td>
                        CTTCCGGGAA
                       </td>
                      </tr>
                      <tr>
                       <td>
                        HRE (HSF)
                       </td>
                       <td>
                        TTCTAGAABNTTCTA
                       </td>
                      </tr>
                      <tr>
                       <td>
                        PRDM1/Blimp1
                       </td>
                       <td>
                        ACTTTCACTTTC
                       </td>
                      </tr>
                     </tbody>
                    </table>
                    <p dir="auto">
                     seq2motif: creating .motif files - least stringent:
                     <em>
                      2
                     </em>
                     mismatches allowed
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="seq2profile.pl RTACGTGC 2 HIF-1b &gt; HIF-1b.motif
seq2profile.pl VVCCACGTGG 2 c-Myc &gt; c-Myc.motif
seq2profile.pl VRCCACGTGG 2 n-Myc &gt; n-Myc.motif
seq2profile.pl GGGGGGGG 2 Maz &gt; Maz_2.motif
seq2profile.pl CTTCCGGGAA 2 Stat3 &gt; Stat3.motif
seq2profile.pl TTCTAGAABNTTCTA 2 HRE &gt; HRE.motif
seq2profile.pl ACTTTCACTTTC 2 PRDM1 &gt; PRDM1_2.motif">
                     <pre class="notranslate"><code>seq2profile.pl RTACGTGC 2 HIF-1b &gt; HIF-1b.motif
seq2profile.pl VVCCACGTGG 2 c-Myc &gt; c-Myc.motif
seq2profile.pl VRCCACGTGG 2 n-Myc &gt; n-Myc.motif
seq2profile.pl GGGGGGGG 2 Maz &gt; Maz_2.motif
seq2profile.pl CTTCCGGGAA 2 Stat3 &gt; Stat3.motif
seq2profile.pl TTCTAGAABNTTCTA 2 HRE &gt; HRE.motif
seq2profile.pl ACTTTCACTTTC 2 PRDM1 &gt; PRDM1_2.motif
</code></pre>
                    </div>
                    <p dir="auto">
                     creating .motif files -  medium stringent:
                     <em>
                      1
                     </em>
                     mismatches allowed
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="seq2profile.pl ACTTTCACTTTC 1 PRDM1 &gt; PRDM1_1.motif">
                     <pre class="notranslate"><code>seq2profile.pl ACTTTCACTTTC 1 PRDM1 &gt; PRDM1_1.motif
</code></pre>
                    </div>
                    <p dir="auto">
                     creating .motif files - most stringent:
                     <em>
                      0
                     </em>
                     mismatches allowed
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="seq2profile.pl ACTTTCACTTTC 0 PRDM1 &gt; PRDM1_0.motif">
                     <pre class="notranslate"><code>seq2profile.pl ACTTTCACTTTC 0 PRDM1 &gt; PRDM1_0.motif
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      C) Predict sites where the TF will bind in a set of coordinates using annotatepeaks -m -mbed
                     </strong>
                    </p>
                    <p dir="auto">
                     5hmC decreases/HC unique peaks
                    </p>
                    <p dir="auto">
                     HIF-1b
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m HIF-1b.motif -mbed HIF-1b_HC_unique.bed -hist 10 &gt; HIF_motifs_HC_unique.txt">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m HIF-1b.motif -mbed HIF-1b_HC_unique.bed -hist 10 &gt; HIF_motifs_HC_unique.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     c-Myc
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m c-Myc.motif -mbed c-Myc_HC_unique.bed &gt; c-Myc_motifs_HC_unique.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m c-Myc.motif -mbed c-Myc_HC_unique.bed &gt; c-Myc_motifs_HC_unique.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     n-Myc
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m n-Myc.motif -mbed n-Myc_HC_unique.bed &gt; n-Myc_motifs_HC_unique.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m n-Myc.motif -mbed n-Myc_HC_unique.bed &gt; n-Myc_motifs_HC_unique.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     Maz
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m Maz.motif -mbed Maz_HC_unique.bed &gt; Maz_motifs_HC_unique.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m Maz.motif -mbed Maz_HC_unique.bed &gt; Maz_motifs_HC_unique.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     Stat3
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m Stat3.motif -mbed Stat3_HC_unique.bed &gt; Stat3_motifs_HC_unique.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -m Stat3.motif -mbed Stat3_HC_unique.bed &gt; Stat3_motifs_HC_unique.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      5hmC increases/FC unique peaks
                     </em>
                     HIF-1b
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m HIF-1b.motif -mbed HIF-1b_FC_unique.bed -hist 10 &gt; HIF_motifs_FC_unique.txt">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m HIF-1b.motif -mbed HIF-1b_FC_unique.bed -hist 10 &gt; HIF_motifs_FC_unique.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     c-Myc
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m c-Myc.motif -mbed c-Myc_FC_unique.bed &gt; c-Myc_motifs_FC_unique.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m c-Myc.motif -mbed c-Myc_FC_unique.bed &gt; c-Myc_motifs_FC_unique.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     n-Myc
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m n-Myc.motif -mbed n-Myc_FC_unique.bed &gt; n-Myc_motifs_FC_unique.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m n-Myc.motif -mbed n-Myc_FC_unique.bed &gt; n-Myc_motifs_FC_unique.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     Maz
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m Maz.motif -mbed Maz_FC_unique.bed &gt; Maz_motifs_FC_unique.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m Maz.motif -mbed Maz_FC_unique.bed &gt; Maz_motifs_FC_unique.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     Stat3
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m Stat3.motif -mbed Stat3.motif_FC_unique.bed &gt; Stat3_motifs_FC_unique.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -m Stat3.motif -mbed Stat3.motif_FC_unique.bed &gt; Stat3_motifs_FC_unique.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     All mm10 annotated promoters (UCSC)
                    </p>
                    <p dir="auto">
                     Stat3
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_allpromoters_-2000_background.bed mm10 -size 1000 -m Stat3.motif -mbed Stat3.motif_mm10_promoters.bed &gt; Stat3_motifs_mm10_promoters.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_allpromoters_-2000_background.bed mm10 -size 1000 -m Stat3.motif -mbed Stat3.motif_mm10_promoters.bed &gt; Stat3_motifs_mm10_promoters.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     HRE (control)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_allpromoters_-2000_background.bed mm10 -size 1000 -m HRE.motif -mbed HRE.motif_mm10_promoters.bed &gt; HRE_motifs_mm10_promoters.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_allpromoters_-2000_background.bed mm10 -size 1000 -m HRE.motif -mbed HRE.motif_mm10_promoters.bed &gt; HRE_motifs_mm10_promoters.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     RNAseq promoters
                    </p>
                    <p dir="auto">
                     Maz - Using different .motif file
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_promoters.bed mm10 -size 1000 -m Maz.motif -mbed Maz_RNAseq_promoters.bed &gt; Maz_motifs_RNAseq_promoters_2.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_promoters.bed mm10 -size 1000 -m Maz.motif -mbed Maz_RNAseq_promoters.bed &gt; Maz_motifs_RNAseq_promoters_2.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     PRDM1
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_promoters.bed mm10 -size 2000 -m PRDM1.motif -mbed PRDM1_RNAseq_promoters.bed &gt; PRDM1_motifs_RNAseq_promoters.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_promoters.bed mm10 -size 2000 -m PRDM1.motif -mbed PRDM1_RNAseq_promoters.bed &gt; PRDM1_motifs_RNAseq_promoters.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     RNAseq promoters - upregulated
                    </p>
                    <p dir="auto">
                     PRDM1
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_promoters_upregulated.bed mm10 -size 2000 -m PRDM1.motif -mbed PRDM1_RNAseq_promoters_upreg.bed &gt; PRDM1_motifs_RNAseq_promoters_upreg.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_promoters_upregulated.bed mm10 -size 2000 -m PRDM1.motif -mbed PRDM1_RNAseq_promoters_upreg.bed &gt; PRDM1_motifs_RNAseq_promoters_upreg.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     Concatenate all binding sites in both HC_unique and FC_unique peaks
Alternatively, could've searched for motifs in concatenated HC anf FC peaks first
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="cat HIF-1b_HC_unique.bed HIF-1b_FC_unique.bed &gt; HIF_motifs_all.bed
cat c-Myc_HC_unique.bed c-Myc_FC_unique.bed &gt; c-Myc_motifs_all.bed
cat n-Myc_HC_unique.bed n-Myc_FC_unique.bed &gt; n-Myc_motifs_all.bed
cat Maz_HC_unique.bed Maz_FC_unique.bed &gt; Maz_motifs_all.bed
cat Stat3_motifs_HC_unique.txt Stat3_motifs_FC_unique.txt &gt; Stat3_motifs_all.bed">
                     <pre class="notranslate"><code>cat HIF-1b_HC_unique.bed HIF-1b_FC_unique.bed &gt; HIF_motifs_all.bed
cat c-Myc_HC_unique.bed c-Myc_FC_unique.bed &gt; c-Myc_motifs_all.bed
cat n-Myc_HC_unique.bed n-Myc_FC_unique.bed &gt; n-Myc_motifs_all.bed
cat Maz_HC_unique.bed Maz_FC_unique.bed &gt; Maz_motifs_all.bed
cat Stat3_motifs_HC_unique.txt Stat3_motifs_FC_unique.txt &gt; Stat3_motifs_all.bed
</code></pre>
                    </div>
                    <p dir="auto">
                     <strong>
                      D) Find intensity of epigenetic marks at predetermined set of motif binding sites (as from #2) using annotatepeaks -hist -d to feed in tag directories. Can do this with any predetermined set of coordinates (i.e. RNAseq exon boundaries, promoters, etc)
                     </strong>
                    </p>
                    <p dir="auto">
                     <em>
                      5hmC distribution comparison:
                     </em>
                     <em>
                      Histograms
                     </em>
                    </p>
                    <p dir="auto">
                     HIF1b binding sites in ALL DhMRs
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/HIF_motifs_all.bed mm10 -size 1000 -m HIF-1b.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt; HIF_motifs_hist_HC_FC.txt">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/HIF_motifs_all.bed mm10 -size 1000 -m HIF-1b.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt; HIF_motifs_hist_HC_FC.txt
</code></pre>
                    </div>
                    <p dir="auto">
                     c-Myc binding sites within ALL DhMRs
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/c-Myc_motifs_all.bed mm10 -size 1000 -m c-Myc.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt; c-Myc_motifs_hist_HC_FC.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/c-Myc_motifs_all.bed mm10 -size 1000 -m c-Myc.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt; c-Myc_motifs_hist_HC_FC.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     n-Myc binding sites within ALL DhMRs
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/n-Myc_motifs_all.bed mm10 -size 1000 -m n-Myc.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt; n-Myc_motifs_hist_HC_FC.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/n-Myc_motifs_all.bed mm10 -size 1000 -m n-Myc.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt; n-Myc_motifs_hist_HC_FC.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     Maz binding sites within ALL DhMRs
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/Maz_motifs_all.bed mm10 -size 1000 -m Maz.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt; Maz_motifs_hist_HC_FC.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/Maz_motifs_all.bed mm10 -size 1000 -m Maz.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt; Maz_motifs_hist_HC_FC.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     Maz binding sites within ALL RNAseq DE promoters
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/Maz_RNAseq_promoters.bed mm10 -size 10 -m Maz.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/  &gt;Maz_motifs_RNAseq_promoters_hist_HC_FC_Input_10Window.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/Maz_RNAseq_promoters.bed mm10 -size 10 -m Maz.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/  &gt;Maz_motifs_RNAseq_promoters_hist_HC_FC_Input_10Window.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     Control - all mm10 genes, truncated
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/ mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;trunc_allgenes.bed_hist_HC_FC.txt &amp;">
                     <pre class="notranslate"><code>/home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/ mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;trunc_allgenes.bed_hist_HC_FC.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     Stat3 binding sites within ALL mm10 promoters (UCSC)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/Stat3.motif_mm10_promoters.bed mm10 -size 1000 -m Stat3.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Stat3_motifs_hist_mm10proms.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/Stat3.motif_mm10_promoters.bed mm10 -size 1000 -m Stat3.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Stat3_motifs_hist_mm10proms.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     HRE (HSF) binding sites within all mm10 promoters
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/HRE.motif_mm10_promoters.bed mm10 -size 1000 -m HRE.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;HRE_motifs_hist_mm10proms.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/HRE.motif_mm10_promoters.bed mm10 -size 1000 -m HRE.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;HRE_motifs_hist_mm10proms.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     PRDM1 binding sites within all mm10 promoters - 2 mismatches
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1_RNAseq_promoters_upreg.bed mm10 -size 1000 -m /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ &gt;PRDM1_motifs_hist_mm10proms.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1_RNAseq_promoters_upreg.bed mm10 -size 1000 -m /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ &gt;PRDM1_motifs_hist_mm10proms.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     PRDM1 binding sites within all mm10 promoters - 0 mismatches
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1_RNAseq_promoters_upreg.bed mm10 -size 1000 -m /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1_0.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ &gt;PRDM1_0_motifs_hist_mm10proms.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1_RNAseq_promoters_upreg.bed mm10 -size 1000 -m /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/motif_analysis/PRDM1_0.motif -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ &gt;PRDM1_0_motifs_hist_mm10proms.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     DE gene exon-boundaries (+/- 15 bp around boundary)
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_exonboundaries.bed mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ &gt;RNAseq_exon_boundaries_hist_HC_FC.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_RNAseq_exonboundaries.bed mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ &gt;RNAseq_exon_boundaries_hist_HC_FC.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     Upregulated Promoters DE promoters
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_upregulatedgenes_promoters.bed mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ &gt;RNAseq_upregulated_promoters_hist_HC_FC_Input.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_upregulatedgenes_promoters.bed mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ &gt;RNAseq_upregulated_promoters_hist_HC_FC_Input.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     Downregulated Promoters DE promoters
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_downregulatedgenes_promoters.bed mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ &gt;RNAseq_downregulated_promoters_hist_HC_FC_Input.txt &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/datasets/FC_downregulatedgenes_promoters.bed mm10 -size 1000 -hist 10 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/Input_hMeDIP_BAMs_tag_directory/ &gt;RNAseq_downregulated_promoters_hist_HC_FC_Input.txt &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     All DE gene promoters
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_promoters.bed mm10 -size 1000 -log -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_RNAseq_promoters_ComprehensiveComparison ">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_promoters.bed mm10 -size 1000 -log -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_RNAseq_promoters_ComprehensiveComparison 
</code></pre>
                    </div>
                    <p dir="auto">
                     DE gene exons
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_exons.bed mm10 -size 1000 -fragLength 150 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/  &gt;Homer_FearConditioning_5hmC_FC_RNAseq_exons_ComprehensiveComparison_2 &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_exons.bed mm10 -size 1000 -fragLength 150 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/  &gt;Homer_FearConditioning_5hmC_FC_RNAseq_exons_ComprehensiveComparison_2 &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     DE gene introns
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_introns.bed mm10 -size 1000 -p 10 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_RNAseq_introns_ComprehensiveComparison_2 &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_introns.bed mm10 -size 1000 -p 10 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_RNAseq_introns_ComprehensiveComparison_2 &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     CTCF binding sites
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/CTCFPeaksClustered.bed mm10 -size 1000 -fragLength 150 -annStats Homer_CTCFPeaksClustered_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_CTCFPeaksClustered_ComprehensiveComparison_2 &amp; ">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/CTCFPeaksClustered.bed mm10 -size 1000 -fragLength 150 -annStats Homer_CTCFPeaksClustered_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_CTCFPeaksClustered_ComprehensiveComparison_2 &amp; 
</code></pre>
                    </div>
                    <p dir="auto">
                     lncRNAs
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_lncRNA.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_lncRNA_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_lncRNA_ComprehensiveComparison_2 &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_lncRNA.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_lncRNA_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_lncRNA_ComprehensiveComparison_2 &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     enhancers: cell based
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_primarycell_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_primarycell_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_enhancers_primarycell_based_ComprehensiveComparison_2 &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_primarycell_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_primarycell_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_enhancers_primarycell_based_ComprehensiveComparison_2 &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     enhancers: tissue based
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_tissue_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_tissue_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_enhancers_tissue_based_ComprehensiveComparison_2 &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_tissue_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_tissue_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_enhancers_tissue_based_ComprehensiveComparison_2 &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     exon boundaries: +/- 50 bp from exons
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_exonboundaries.bed mm10 -size 1000 -fragLength 150 -annStats Homer_mm10_exonboundaries_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_mm10_exonboundaries_ComprehensiveComparison_2 &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_exonboundaries.bed mm10 -size 1000 -fragLength 150 -annStats Homer_mm10_exonboundaries_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_mm10_exonboundaries_ComprehensiveComparison_2 &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     <em>
                      Scatterplots
                     </em>
                    </p>
                    <p dir="auto">
                     HC vs HC, FC, HC_unique, FC_unique
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC.bed mm10 -size 1000 -fragLength 150 -annStats Homer_HC_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ &gt;Homer_FearConditioning_5hmC_HC_ComprehensiveComparison &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC.bed mm10 -size 1000 -fragLength 150 -annStats Homer_HC_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ &gt;Homer_FearConditioning_5hmC_HC_ComprehensiveComparison &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     FC vs HC, FC, HC_unique, FC_unique
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_ComprehensiveComparison &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_ComprehensiveComparison &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     HC_unique vs HC, FC, HC_unique, FC_unique
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -fragLength 150 -annStats Homer_HC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ &gt;Homer_FearConditioning_5hmC_HC_unique_ComprehensiveComparison &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/HC_unique.bed mm10 -size 1000 -fragLength 150 -annStats Homer_HC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ &gt;Homer_FearConditioning_5hmC_HC_unique_ComprehensiveComparison &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     FC_unique vs HC, FC, HC_unique, FC_unique
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_unique_ComprehensiveComparison &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_unique.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_tag_directory /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_unique_hMeDIP_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_unique_hMeDIP_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_unique_ComprehensiveComparison &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     RNAseq promoters vs
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_promoters.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_RNAseq_promoters_ComprehensiveComparison_2 &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_promoters.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_RNAseq_promoters_ComprehensiveComparison_2 &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     RNAseq exons
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_exons.bed mm10 -size 1000 -fragLength 150 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/  &gt;Homer_FearConditioning_5hmC_FC_RNAseq_exons_ComprehensiveComparison_2 &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_exons.bed mm10 -size 1000 -fragLength 150 -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/  &gt;Homer_FearConditioning_5hmC_FC_RNAseq_exons_ComprehensiveComparison_2 &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     RNAseq introns
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_introns.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_RNAseq_introns_ComprehensiveComparison_2 &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_RNAseq_introns.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_unique_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_RNAseq_introns_ComprehensiveComparison_2 &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     CTCF binding sites
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/CTCFPeaksClustered.bed mm10 -size 1000 -fragLength 150 -annStats Homer_CTCFPeaksClustered_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_CTCFPeaksClustered_ComprehensiveComparison_2 &amp; ">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/CTCFPeaksClustered.bed mm10 -size 1000 -fragLength 150 -annStats Homer_CTCFPeaksClustered_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_CTCFPeaksClustered_ComprehensiveComparison_2 &amp; 
</code></pre>
                    </div>
                    <p dir="auto">
                     lncRNAs
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_lncRNA.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_lncRNA_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_lncRNA_ComprehensiveComparison_2 &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/FC_lncRNA.bed mm10 -size 1000 -fragLength 150 -annStats Homer_FC_lncRNA_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_FC_lncRNA_ComprehensiveComparison_2 &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     enhancers: cell based
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_primarycell_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_primarycell_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_enhancers_primarycell_based_ComprehensiveComparison_2 &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_primarycell_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_primarycell_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_enhancers_primarycell_based_ComprehensiveComparison_2 &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     enhancers: tissue based
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_tissue_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_tissue_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_enhancers_tissue_based_ComprehensiveComparison_2 &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/enhancers_tissue_based.bed mm10 -size 1000 -fragLength 150 -annStats Homer_enhancers_tissue_based_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_enhancers_tissue_based_ComprehensiveComparison_2 &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     exon boundaries: +/- 50 bp from exons
                    </p>
                    <div class="snippet-clipboard-content notranslate position-relative overflow-auto" data-snippet-clipboard-copy-content="annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_exonboundaries.bed mm10 -size 1000 -fragLength 150 -annStats Homer_mm10_exonboundaries_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_mm10_exonboundaries_ComprehensiveComparison_2 &amp;">
                     <pre class="notranslate"><code>annotatePeaks.pl /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/bedtools/mm10_exonboundaries.bed mm10 -size 1000 -fragLength 150 -annStats Homer_mm10_exonboundaries_annStats -d /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/HC_hMeDIP_BAMs_tag_directory/ /home/ssharma/Ressler_RNASeq/analysis/FearConditioning_5hmCSeq_MACS/homer/Homer_Tag_Directories/FC_hMeDIP_BAMs_tag_directory/ &gt;Homer_FearConditioning_5hmC_mm10_exonboundaries_ComprehensiveComparison_2 &amp;
</code></pre>
                    </div>
                    <p dir="auto">
                     ###11. FAST ANALYSIS: ANNOTATING REGIONS OF THE GENOME BASED ON DISEASE RELEVANT SNP CONTENT
###12. Homology ANALYSIS:PHASTCONS
                    </p>
                   </article>
                  </div>
                 </div>
                </div>
               </div>
              </div>
              <!-- -->
              <!-- -->
              <script id="__PRIMER_DATA__" type="application/json">
               {"resolvedServerColorMode":"day"}
              </script>
             </div>
            </react-partial>
            <input data-csrf="true" id="react-codespace-csrf" type="hidden" value="rJLfEXdX6gWFbgen5een-p2lshWKxKZFW3ni5yPGG1G7SkC-61ySZYYi_d6zszo06x-f9mpt3d8qqsZMJmWFcg">
            </input>
           </div>
           <div class="Layout-sidebar" data-view-component="true">
            <div class="BorderGrid about-margin" data-pjax="">
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <div class="hide-sm hide-md">
                <h2 class="mb-3 h4">
                 About
                </h2>
                <div class="f4 my-3 color-fg-muted text-italic">
                 No description, website, or topics provided.
                </div>
                <h3 class="sr-only">
                 Resources
                </h3>
                <div class="mt-2">
                 <a class="Link--muted" data-analytics-event='{"category":"Repository Overview","action":"click","label":"location:sidebar;file:readme"}' href="#readme-ov-file">
                  <svg aria-hidden="true" class="octicon octicon-book mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M0 1.75A.75.75 0 0 1 .75 1h4.253c1.227 0 2.317.59 3 1.501A3.743 3.743 0 0 1 11.006 1h4.245a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75h-4.507a2.25 2.25 0 0 0-1.591.659l-.622.621a.75.75 0 0 1-1.06 0l-.622-.621A2.25 2.25 0 0 0 5.258 13H.75a.75.75 0 0 1-.75-.75Zm7.251 10.324.004-5.073-.002-2.253A2.25 2.25 0 0 0 5.003 2.5H1.5v9h3.757a3.75 3.75 0 0 1 1.994.574ZM8.755 4.75l-.004 7.322a3.752 3.752 0 0 1 1.992-.572H14.5v-9h-3.495a2.25 2.25 0 0 0-2.25 2.25Z">
                   </path>
                  </svg>
                  Readme
                 </a>
                </div>
                <include-fragment src="/Sumeet460/Fear-Conditioning-Genomics/hovercards/citation/sidebar_partial?tree_name=master">
                </include-fragment>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/activity">
                  <svg aria-hidden="true" class="octicon octicon-pulse mr-2" data-view-component="true" height="16" text="gray" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M6 2c.306 0 .582.187.696.471L10 10.731l1.304-3.26A.751.751 0 0 1 12 7h3.25a.75.75 0 0 1 0 1.5h-2.742l-1.812 4.528a.751.751 0 0 1-1.392 0L6 4.77 4.696 8.03A.75.75 0 0 1 4 8.5H.75a.75.75 0 0 1 0-1.5h2.742l1.812-4.529A.751.751 0 0 1 6 2Z">
                   </path>
                  </svg>
                  <span class="color-fg-muted">
                   Activity
                  </span>
                 </a>
                </div>
                <h3 class="sr-only">
                 Stars
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/stargazers">
                  <svg aria-hidden="true" class="octicon octicon-star mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 .25a.75.75 0 0 1 .673.418l1.882 3.815 4.21.612a.75.75 0 0 1 .416 1.279l-3.046 2.97.719 4.192a.751.751 0 0 1-1.088.791L8 12.347l-3.766 1.98a.75.75 0 0 1-1.088-.79l.72-4.194L.818 6.374a.75.75 0 0 1 .416-1.28l4.21-.611L7.327.668A.75.75 0 0 1 8 .25Zm0 2.445L6.615 5.5a.75.75 0 0 1-.564.41l-3.097.45 2.24 2.184a.75.75 0 0 1 .216.664l-.528 3.084 2.769-1.456a.75.75 0 0 1 .698 0l2.77 1.456-.53-3.084a.75.75 0 0 1 .216-.664l2.24-2.183-3.096-.45a.75.75 0 0 1-.564-.41L8 2.694Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  stars
                 </a>
                </div>
                <h3 class="sr-only">
                 Watchers
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/watchers">
                  <svg aria-hidden="true" class="octicon octicon-eye mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M8 2c1.981 0 3.671.992 4.933 2.078 1.27 1.091 2.187 2.345 2.637 3.023a1.62 1.62 0 0 1 0 1.798c-.45.678-1.367 1.932-2.637 3.023C11.67 13.008 9.981 14 8 14c-1.981 0-3.671-.992-4.933-2.078C1.797 10.83.88 9.576.43 8.898a1.62 1.62 0 0 1 0-1.798c.45-.677 1.367-1.931 2.637-3.022C4.33 2.992 6.019 2 8 2ZM1.679 7.932a.12.12 0 0 0 0 .136c.411.622 1.241 1.75 2.366 2.717C5.176 11.758 6.527 12.5 8 12.5c1.473 0 2.825-.742 3.955-1.715 1.124-.967 1.954-2.096 2.366-2.717a.12.12 0 0 0 0-.136c-.412-.621-1.242-1.75-2.366-2.717C10.824 4.242 9.473 3.5 8 3.5c-1.473 0-2.825.742-3.955 1.715-1.124.967-1.954 2.096-2.366 2.717ZM8 10a2 2 0 1 1-.001-3.999A2 2 0 0 1 8 10Z">
                   </path>
                  </svg>
                  <strong>
                   1
                  </strong>
                  watching
                 </a>
                </div>
                <h3 class="sr-only">
                 Forks
                </h3>
                <div class="mt-2">
                 <a class="Link Link--muted" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/forks">
                  <svg aria-hidden="true" class="octicon octicon-repo-forked mr-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
                   <path d="M5 5.372v.878c0 .414.336.75.75.75h4.5a.75.75 0 0 0 .75-.75v-.878a2.25 2.25 0 1 1 1.5 0v.878a2.25 2.25 0 0 1-2.25 2.25h-1.5v2.128a2.251 2.251 0 1 1-1.5 0V8.5h-1.5A2.25 2.25 0 0 1 3.5 6.25v-.878a2.25 2.25 0 1 1 1.5 0ZM5 3.25a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Zm6.75.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Zm-3 8.75a.75.75 0 1 0-1.5 0 .75.75 0 0 0 1.5 0Z">
                   </path>
                  </svg>
                  <strong>
                   0
                  </strong>
                  forks
                 </a>
                </div>
                <div class="mt-2">
                 <a class="Link--muted" href="/contact/report-content?content_url=https%3A%2F%2Fgithub.com%2FSumeet460%2FFear-Conditioning-Genomics&amp;report=Sumeet460+%28user%29">
                  Report repository
                 </a>
                </div>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3" data-pjax="#repo-content-pjax-container" data-turbo-frame="repo-content-turbo-frame">
                <a class="Link--primary no-underline Link" data-view-component="true" href="/Sumeet460/Fear-Conditioning-Genomics/releases">
                 Releases
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No releases published
               </div>
              </div>
             </div>
             <div class="BorderGrid-row">
              <div class="BorderGrid-cell">
               <h2 class="h4 mb-3">
                <a class="Link--primary no-underline Link d-flex flex-items-center" data-view-component="true" href="/users/Sumeet460/packages?repo_name=Fear-Conditioning-Genomics">
                 Packages
                 <span class="Counter ml-1" data-view-component="true" hidden="hidden" title="0">
                  0
                 </span>
                </a>
               </h2>
               <div class="text-small color-fg-muted">
                No packages published
                <br/>
               </div>
              </div>
             </div>
             <div class="BorderGrid-row" hidden="">
              <div class="BorderGrid-cell">
               <include-fragment accept="text/fragment+html" src="/Sumeet460/Fear-Conditioning-Genomics/used_by_list">
               </include-fragment>
              </div>
             </div>
            </div>
           </div>
          </div>
         </div>
        </div>
       </div>
      </turbo-frame>
     </main>
    </div>
   </div>
   <footer class="footer pt-8 pb-6 f6 color-fg-muted p-responsive" role="contentinfo">
    <h2 class="sr-only">
     Footer
    </h2>
    <div class="d-flex flex-justify-center flex-items-center flex-column-reverse flex-lg-row flex-wrap flex-lg-nowrap">
     <div class="d-flex flex-items-center flex-shrink-0 mx-2">
      <a aria-label="Homepage" class="footer-octicon mr-2" href="https://github.com" title="GitHub">
       <svg aria-hidden="true" class="octicon octicon-mark-github" data-view-component="true" height="24" version="1.1" viewbox="0 0 16 16" width="24">
        <path d="M8 0c4.42 0 8 3.58 8 8a8.013 8.013 0 0 1-5.45 7.59c-.4.08-.55-.17-.55-.38 0-.27.01-1.13.01-2.2 0-.75-.25-1.23-.54-1.48 1.78-.2 3.65-.88 3.65-3.95 0-.88-.31-1.59-.82-2.15.08-.2.36-1.02-.08-2.12 0 0-.67-.22-2.2.82-.64-.18-1.32-.27-2-.27-.68 0-1.36.09-2 .27-1.53-1.03-2.2-.82-2.2-.82-.44 1.1-.16 1.92-.08 2.12-.51.56-.82 1.28-.82 2.15 0 3.06 1.86 3.75 3.64 3.95-.23.2-.44.55-.51 1.07-.46.21-1.61.55-2.33-.66-.15-.24-.6-.83-1.23-.82-.67.01-.27.38.01.53.34.19.73.9.82 1.13.16.45.68 1.31 2.69.94 0 .67.01 1.3.01 1.49 0 .21-.15.45-.55.38A7.995 7.995 0 0 1 0 8c0-4.42 3.58-8 8-8Z">
        </path>
       </svg>
      </a>
      <span>
       © 2024 GitHub, Inc.
      </span>
     </div>
     <nav aria-label="Footer">
      <h3 class="sr-only" id="sr-footer-heading">
       Footer navigation
      </h3>
      <ul aria-labelledby="sr-footer-heading" class="list-style-none d-flex flex-justify-center flex-wrap mb-2 mb-lg-0">
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to Terms","label":"text:terms"}' data-view-component="true" href="https://docs.github.com/site-policy/github-terms/github-terms-of-service">
         Terms
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to privacy","label":"text:privacy"}' data-view-component="true" href="https://docs.github.com/site-policy/privacy-policies/github-privacy-statement">
         Privacy
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to security","label":"text:security"}' data-view-component="true" href="/security">
         Security
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to status","label":"text:status"}' data-view-component="true" href="https://www.githubstatus.com/">
         Status
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to docs","label":"text:docs"}' data-view-component="true" href="https://docs.github.com/">
         Docs
        </a>
       </li>
       <li class="mx-2">
        <a class="Link--secondary Link" data-analytics-event='{"category":"Footer","action":"go to contact","label":"text:contact"}' data-view-component="true" href="https://support.github.com?tags=dotcom-footer">
         Contact
        </a>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Manage cookies
         </button>
        </cookie-consent-link>
       </li>
       <li class="mx-2">
        <cookie-consent-link>
         <button class="Link--secondary underline-on-hover border-0 p-0 color-bg-transparent" data-action="click:cookie-consent-link#showConsentManagement" type="button">
          Do not share my personal information
         </button>
        </cookie-consent-link>
       </li>
      </ul>
     </nav>
    </div>
   </footer>
   <cookie-consent class="position-fixed bottom-0 left-0" data-cookie-consent-required="true" data-initial-cookie-consent-allowed="" id="cookie-consent-banner" style="z-index: 999999">
   </cookie-consent>
   <div class="ajax-error-message flash flash-error" hidden="" id="ajax-error-message">
    <svg aria-hidden="true" class="octicon octicon-alert" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
     <path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575Zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368Zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0ZM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0Z">
     </path>
    </svg>
    <button aria-label="Dismiss error" class="flash-close js-ajax-error-dismiss" type="button">
     <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
      <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
      </path>
     </svg>
    </button>
    You can’t perform that action at this time.
   </div>
   <template id="site-details-dialog">
    <details class="details-reset details-overlay details-overlay-dark lh-default color-fg-default hx_rsm" open="">
     <summary aria-label="Close dialog" role="button">
     </summary>
     <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast hx_rsm-dialog hx_rsm-modal">
      <button aria-label="Close dialog" class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" data-close-dialog="" type="button">
       <svg aria-hidden="true" class="octicon octicon-x" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
        <path d="M3.72 3.72a.75.75 0 0 1 1.06 0L8 6.94l3.22-3.22a.749.749 0 0 1 1.275.326.749.749 0 0 1-.215.734L9.06 8l3.22 3.22a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L8 9.06l-3.22 3.22a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042L6.94 8 3.72 4.78a.75.75 0 0 1 0-1.06Z">
        </path>
       </svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner">
      </div>
     </details-dialog>
    </details>
   </template>
   <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
    <div class="Popover-message Popover-message--bottom-left Popover-message--large Box color-shadow-large" style="width:360px;">
    </div>
   </div>
   <template id="snippet-clipboard-copy-button">
    <div class="zeroclipboard-container position-absolute right-0 top-0">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0 tooltipped-no-delay" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none m-2" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <template id="snippet-clipboard-copy-button-unpositioned">
    <div class="zeroclipboard-container">
     <clipboard-copy aria-label="Copy" class="ClipboardButton btn btn-invisible js-clipboard-copy m-2 p-0 tooltipped-no-delay d-flex flex-justify-center flex-items-center" data-copy-feedback="Copied!" data-tooltip-direction="w">
      <svg aria-hidden="true" class="octicon octicon-copy js-clipboard-copy-icon" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z">
       </path>
       <path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z">
       </path>
      </svg>
      <svg aria-hidden="true" class="octicon octicon-check js-clipboard-check-icon color-fg-success d-none" data-view-component="true" height="16" version="1.1" viewbox="0 0 16 16" width="16">
       <path d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0Z">
       </path>
      </svg>
     </clipboard-copy>
    </div>
   </template>
   <style>
    .user-mention[href$="/ninenia"] {
        color: var(--color-user-mention-fg);
        background-color: var(--color-user-mention-bg);
        border-radius: 2px;
        margin-left: -2px;
        margin-right: -2px;
        padding: 0 2px;
      }
   </style>
  </div>
  <div aria-atomic="true" aria-live="polite" class="sr-only" id="js-global-screen-reader-notice">
  </div>
  <div aria-atomic="true" aria-live="assertive" class="sr-only" id="js-global-screen-reader-notice-assertive">
  </div>
 </body>
</html>
